This directory contains files to assemble the archaic genomes for data analysis and filters for high-quality sites and genotypes. The scripts are listed in the order in which they should be launched.

- combine_*_vcfs.sh uses bcftools to merge individual VCFs for InDels and individual per contig VCFs for SNVs. I kept InDels and SNVs separate at this stage because bcftools does not permit merging files with different sets of samples and InDels are not presently available for Chagyrskaya. We'll later merge data in the [first notebook](https://github.com/brandcm/Archaic_Splicing/blob/main/scripts/notebooks/1_generate_archaic_dataframe.ipynb) by setting Chagyrskaya InDel positions to missing ("./."). Archaic InDel data are available from [here](http://ftp.eva.mpg.de/neandertal/Vindija/VCF/indels/). Archaic SNV data are available from the following repositories: [Altai Neanderthal](http://ftp.eva.mpg.de/neandertal/Vindija/VCF/Altai/), [Chagyrskaya](http://ftp.eva.mpg.de/neandertal/Chagyrskaya/VCF/), [Denisova](http://ftp.eva.mpg.de/neandertal/Vindija/VCF/Denisova/), and [Vindija](http://ftp.eva.mpg.de/neandertal/Vindija/VCF/Vindija33.19/).

- filter_vcfs.sh uses bcftools to filter the data for invariant sites (i.e., the archaic genomes are indifferent to the human reference) and high-quality sites and genotypes. I define a high-quality site as those with a QUAL, or Phred-scaled probability that the site has no variant, >= 30. Further, genotypes were surveyed by quality using read depth (DP), or the number of reads available for a given position, and genotype quality (GQ), or the Phred-scaled Probability that the call is incorrect. I retained genotypes with DP >= 10 and GQ >= 30 and set genotypes that did not both of these criteria to missing. These are rather strict thresholds for ancient DNA and are the same values that I use for samples from extant individuals. However, I sought to be conservative in my approach. 
