This directory contains most of the scripts used in this analysis. Scripts for the phenotype enrichment analysis are housed in a [data subdirectory](https://github.com/brandcm/Archaic_Splicing/tree/main/data/phenotype_enrichment) for ease of executing the Snakemake pipeline there. There are several directories grouped by the stage of analysis. Numbers in the directory names reflect the order of analysis. See below for more details on reproducing this analysis from start to finish.

- [1_archaic_genome_preparation_and_filtering](https://github.com/brandcm/Archaic_Splicing/tree/main/scripts/1_archaic_genome_preparation_and_filtering) contains scripts to assemble archaic InDels and SNVs separately and filter variants to retain high-quality sites and genotypes.

- [2_archaic_data_preparation_for_SpliceAI_analysis](https://github.com/brandcm/Archaic_Splicing/tree/main/scripts/2_archaic_data_preparation_for_SpliceAI_analysis) contains scripts to 1) subset the filtered VCFs above to those sites near/in exons per GENCODE, Human Release 23 for hg19, annotations and 2) split variants into smaller VCFs containing <= 5,000 variants for faster, parallelized SpliceAI analysis.

- [3_SpliceAI_analysis](https://github.com/brandcm/Archaic_Splicing/tree/main/scripts/3_SpliceAI_analysis) contains scripts to run SpliceAI on the split VCFs and reassemble the annotations. Note that in this last step, I split variants with multiple annotations into multiple rows for easier downstream analysis.

- [4_modern_data_preparation](https://github.com/brandcm/Archaic_Splicing/tree/main/scripts/4_modern_data_preparation) contains scripts to prepare additional data that is useful for contextualizing and understanding potential archaic splice variants. Many of these scripts are called in a [Jupyter notebook](https://github.com/brandcm/Archaic_Splicing/blob/main/scripts/notebooks/2_get_ancestral_alleles_frequencies_introgressed_variants.ipynb) in the [notebooks](https://github.com/brandcm/Archaic_Splicing/tree/main/scripts/notebooks) directory.

- [notebooks](https://github.com/brandcm/Archaic_Splicing/tree/main/scripts/notebooks) contains Jupyter notebooks used in data analysis and visualization. All results and figures presented in the manuscript can be found in this directory.

- [other_scripts](https://github.com/brandcm/Archaic_Splicing/tree/main/scripts/other_scripts) contains a handful of scripts used to analyze missense spliceosome variants in archaics and moderns and prepare/run the novel isoform analysis.

Reproducing the analysis: If you wish to reproduce the entire analysis, run the scripts in the order specified in each directory for [1_archaic_genome_preparation_and_filtering](https://github.com/brandcm/Archaic_Splicing/tree/main/scripts/1_archaic_genome_preparation_and_filtering), [2_archaic_data_preparation_for_SpliceAI_analysis](https://github.com/brandcm/Archaic_Splicing/tree/main/scripts/2_archaic_data_preparation_for_SpliceAI_analysis), and [3_SpliceAI_analysis](https://github.com/brandcm/Archaic_Splicing/tree/main/scripts/3_SpliceAI_analysis). Then run the notebooks in order in the [notebooks directory](https://github.com/brandcm/Archaic_Splicing/tree/main/scripts/notebooks). Many of the [4_modern_data_preparation](https://github.com/brandcm/Archaic_Splicing/tree/main/scripts/4_modern_data_preparation) scripts are launched from within notebooks in that directory. The remaining scripts can be run as needed. Scripts in the [other_scripts](https://github.com/brandcm/Archaic_Splicing/tree/main/scripts/other_scripts) directory can be run at the very end of the [analysis Jupyter notebook](https://github.com/brandcm/Archaic_Splicing/blob/main/scripts/notebooks/5_analysis.ipynb). If you have any questions about analysis order or anything else please do not hesitate to reach out via email or submit a pull request. 
