{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "97fd51da",
   "metadata": {},
   "source": [
    "# Generate Archaic Dataframe and Splice Altering List"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8f0674a9",
   "metadata": {},
   "source": [
    "This notebook gathers the SpliceAI archaic annotations, combines the InDel and SNV data, adds additional information including the distribution of alternate alleles, gene constraint, mean gene conservation, and removes invariant sites."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a4a7e7f3",
   "metadata": {},
   "source": [
    "Load library. Set the number of columns to be displayed to a large value."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "7eb13d33",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "pd.set_option('display.max_columns', 100)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5fe7cb62",
   "metadata": {},
   "source": [
    "Load indels."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "36f14d2c",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref_allele</th>\n",
       "      <th>altai_gt</th>\n",
       "      <th>denisovan_gt</th>\n",
       "      <th>vindija_gt</th>\n",
       "      <th>alt_allele</th>\n",
       "      <th>annotation</th>\n",
       "      <th>ag_delta</th>\n",
       "      <th>al_delta</th>\n",
       "      <th>dg_delta</th>\n",
       "      <th>dl_delta</th>\n",
       "      <th>ag_pos</th>\n",
       "      <th>al_pos</th>\n",
       "      <th>dg_pos</th>\n",
       "      <th>dl_pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>739130</td>\n",
       "      <td>T</td>\n",
       "      <td>0/0</td>\n",
       "      <td>./.</td>\n",
       "      <td>0/1</td>\n",
       "      <td>TA</td>\n",
       "      <td>AL669831.1</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-7</td>\n",
       "      <td>-36</td>\n",
       "      <td>-35</td>\n",
       "      <td>-31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>739130</td>\n",
       "      <td>T</td>\n",
       "      <td>0/1</td>\n",
       "      <td>./.</td>\n",
       "      <td>0/0</td>\n",
       "      <td>TAA</td>\n",
       "      <td>AL669831.1</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-31</td>\n",
       "      <td>-36</td>\n",
       "      <td>-7</td>\n",
       "      <td>-31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>867993</td>\n",
       "      <td>GTTTC</td>\n",
       "      <td>1/1</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>G</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.06</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-48</td>\n",
       "      <td>11</td>\n",
       "      <td>29</td>\n",
       "      <td>-7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>868522</td>\n",
       "      <td>A</td>\n",
       "      <td>1/1</td>\n",
       "      <td>./.</td>\n",
       "      <td>1/1</td>\n",
       "      <td>AG</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-33</td>\n",
       "      <td>2</td>\n",
       "      <td>-10</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>869121</td>\n",
       "      <td>T</td>\n",
       "      <td>./.</td>\n",
       "      <td>1/1</td>\n",
       "      <td>./.</td>\n",
       "      <td>TG</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.00</td>\n",
       "      <td>48</td>\n",
       "      <td>-36</td>\n",
       "      <td>-36</td>\n",
       "      <td>-41</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>869244</td>\n",
       "      <td>C</td>\n",
       "      <td>./.</td>\n",
       "      <td>1/1</td>\n",
       "      <td>./.</td>\n",
       "      <td>CAG</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-3</td>\n",
       "      <td>33</td>\n",
       "      <td>23</td>\n",
       "      <td>33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr1</td>\n",
       "      <td>870339</td>\n",
       "      <td>C</td>\n",
       "      <td>0/0</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>CA</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.01</td>\n",
       "      <td>-33</td>\n",
       "      <td>-5</td>\n",
       "      <td>-6</td>\n",
       "      <td>-34</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr1</td>\n",
       "      <td>870339</td>\n",
       "      <td>C</td>\n",
       "      <td>1/1</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>CG</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-33</td>\n",
       "      <td>-5</td>\n",
       "      <td>-6</td>\n",
       "      <td>-34</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr1</td>\n",
       "      <td>872534</td>\n",
       "      <td>A</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>./.</td>\n",
       "      <td>AG</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1</td>\n",
       "      <td>-32</td>\n",
       "      <td>-32</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr1</td>\n",
       "      <td>872776</td>\n",
       "      <td>GCC</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>G</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-14</td>\n",
       "      <td>18</td>\n",
       "      <td>-19</td>\n",
       "      <td>-14</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chrom     pos ref_allele altai_gt denisovan_gt vindija_gt alt_allele  \\\n",
       "0  chr1  739130          T      0/0          ./.        0/1         TA   \n",
       "1  chr1  739130          T      0/1          ./.        0/0        TAA   \n",
       "2  chr1  867993      GTTTC      1/1          ./.        ./.          G   \n",
       "3  chr1  868522          A      1/1          ./.        1/1         AG   \n",
       "4  chr1  869121          T      ./.          1/1        ./.         TG   \n",
       "5  chr1  869244          C      ./.          1/1        ./.        CAG   \n",
       "6  chr1  870339          C      0/0          ./.        ./.         CA   \n",
       "7  chr1  870339          C      1/1          ./.        ./.         CG   \n",
       "8  chr1  872534          A      1/1          1/1        ./.         AG   \n",
       "9  chr1  872776        GCC      ./.          ./.        ./.          G   \n",
       "\n",
       "   annotation  ag_delta  al_delta  dg_delta  dl_delta  ag_pos  al_pos  dg_pos  \\\n",
       "0  AL669831.1      0.00      0.00      0.00      0.00      -7     -36     -35   \n",
       "1  AL669831.1      0.00      0.00      0.00      0.00     -31     -36      -7   \n",
       "2      SAMD11      0.00      0.06      0.00      0.00     -48      11      29   \n",
       "3      SAMD11      0.00      0.00      0.00      0.00     -33       2     -10   \n",
       "4      SAMD11      0.00      0.00      0.01      0.00      48     -36     -36   \n",
       "5      SAMD11      0.00      0.00      0.00      0.00      -3      33      23   \n",
       "6      SAMD11      0.00      0.00      0.00      0.01     -33      -5      -6   \n",
       "7      SAMD11      0.00      0.00      0.00      0.00     -33      -5      -6   \n",
       "8      SAMD11      0.00      0.00      0.00      0.00       1     -32     -32   \n",
       "9      SAMD11      0.01      0.00      0.00      0.00     -14      18     -19   \n",
       "\n",
       "   dl_pos  \n",
       "0     -31  \n",
       "1     -31  \n",
       "2      -7  \n",
       "3       2  \n",
       "4     -41  \n",
       "5      33  \n",
       "6     -34  \n",
       "7     -34  \n",
       "8      17  \n",
       "9     -14  "
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "indels_header = ['chrom','pos','ref_allele','altai_gt','denisovan_gt','vindija_gt','alt_allele','annotation','ag_delta','al_delta','dg_delta','dl_delta','ag_pos','al_pos','dg_pos','dl_pos']\n",
    "indels = pd.read_csv('../../data/spliceai_outputs/spliceai_archaic_indels.txt', sep = '\\t', names = indels_header)\n",
    "indels.head(10)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d8223bbe",
   "metadata": {},
   "source": [
    "Add a column for Chagyrskaya and the variant type."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "00ccbeb4",
   "metadata": {},
   "outputs": [],
   "source": [
    "indels['chagyrskaya_gt'] = './.'\n",
    "indels['variant_type'] = 'indel'"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ed2bdb1a",
   "metadata": {},
   "source": [
    "Reorder columns so that the dataframe matches the SNV dataframe."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "8bdbb044",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref_allele</th>\n",
       "      <th>altai_gt</th>\n",
       "      <th>chagyrskaya_gt</th>\n",
       "      <th>denisovan_gt</th>\n",
       "      <th>vindija_gt</th>\n",
       "      <th>alt_allele</th>\n",
       "      <th>annotation</th>\n",
       "      <th>ag_delta</th>\n",
       "      <th>al_delta</th>\n",
       "      <th>dg_delta</th>\n",
       "      <th>dl_delta</th>\n",
       "      <th>ag_pos</th>\n",
       "      <th>al_pos</th>\n",
       "      <th>dg_pos</th>\n",
       "      <th>dl_pos</th>\n",
       "      <th>variant_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>739130</td>\n",
       "      <td>T</td>\n",
       "      <td>0/0</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>0/1</td>\n",
       "      <td>TA</td>\n",
       "      <td>AL669831.1</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-7</td>\n",
       "      <td>-36</td>\n",
       "      <td>-35</td>\n",
       "      <td>-31</td>\n",
       "      <td>indel</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>739130</td>\n",
       "      <td>T</td>\n",
       "      <td>0/1</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>0/0</td>\n",
       "      <td>TAA</td>\n",
       "      <td>AL669831.1</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-31</td>\n",
       "      <td>-36</td>\n",
       "      <td>-7</td>\n",
       "      <td>-31</td>\n",
       "      <td>indel</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>867993</td>\n",
       "      <td>GTTTC</td>\n",
       "      <td>1/1</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>G</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.06</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-48</td>\n",
       "      <td>11</td>\n",
       "      <td>29</td>\n",
       "      <td>-7</td>\n",
       "      <td>indel</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>868522</td>\n",
       "      <td>A</td>\n",
       "      <td>1/1</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>1/1</td>\n",
       "      <td>AG</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-33</td>\n",
       "      <td>2</td>\n",
       "      <td>-10</td>\n",
       "      <td>2</td>\n",
       "      <td>indel</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>869121</td>\n",
       "      <td>T</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>1/1</td>\n",
       "      <td>./.</td>\n",
       "      <td>TG</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.00</td>\n",
       "      <td>48</td>\n",
       "      <td>-36</td>\n",
       "      <td>-36</td>\n",
       "      <td>-41</td>\n",
       "      <td>indel</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>869244</td>\n",
       "      <td>C</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>1/1</td>\n",
       "      <td>./.</td>\n",
       "      <td>CAG</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-3</td>\n",
       "      <td>33</td>\n",
       "      <td>23</td>\n",
       "      <td>33</td>\n",
       "      <td>indel</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr1</td>\n",
       "      <td>870339</td>\n",
       "      <td>C</td>\n",
       "      <td>0/0</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>CA</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.01</td>\n",
       "      <td>-33</td>\n",
       "      <td>-5</td>\n",
       "      <td>-6</td>\n",
       "      <td>-34</td>\n",
       "      <td>indel</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr1</td>\n",
       "      <td>870339</td>\n",
       "      <td>C</td>\n",
       "      <td>1/1</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>CG</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-33</td>\n",
       "      <td>-5</td>\n",
       "      <td>-6</td>\n",
       "      <td>-34</td>\n",
       "      <td>indel</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr1</td>\n",
       "      <td>872534</td>\n",
       "      <td>A</td>\n",
       "      <td>1/1</td>\n",
       "      <td>./.</td>\n",
       "      <td>1/1</td>\n",
       "      <td>./.</td>\n",
       "      <td>AG</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1</td>\n",
       "      <td>-32</td>\n",
       "      <td>-32</td>\n",
       "      <td>17</td>\n",
       "      <td>indel</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr1</td>\n",
       "      <td>872776</td>\n",
       "      <td>GCC</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>G</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-14</td>\n",
       "      <td>18</td>\n",
       "      <td>-19</td>\n",
       "      <td>-14</td>\n",
       "      <td>indel</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chrom     pos ref_allele altai_gt chagyrskaya_gt denisovan_gt vindija_gt  \\\n",
       "0  chr1  739130          T      0/0            ./.          ./.        0/1   \n",
       "1  chr1  739130          T      0/1            ./.          ./.        0/0   \n",
       "2  chr1  867993      GTTTC      1/1            ./.          ./.        ./.   \n",
       "3  chr1  868522          A      1/1            ./.          ./.        1/1   \n",
       "4  chr1  869121          T      ./.            ./.          1/1        ./.   \n",
       "5  chr1  869244          C      ./.            ./.          1/1        ./.   \n",
       "6  chr1  870339          C      0/0            ./.          ./.        ./.   \n",
       "7  chr1  870339          C      1/1            ./.          ./.        ./.   \n",
       "8  chr1  872534          A      1/1            ./.          1/1        ./.   \n",
       "9  chr1  872776        GCC      ./.            ./.          ./.        ./.   \n",
       "\n",
       "  alt_allele  annotation  ag_delta  al_delta  dg_delta  dl_delta  ag_pos  \\\n",
       "0         TA  AL669831.1      0.00      0.00      0.00      0.00      -7   \n",
       "1        TAA  AL669831.1      0.00      0.00      0.00      0.00     -31   \n",
       "2          G      SAMD11      0.00      0.06      0.00      0.00     -48   \n",
       "3         AG      SAMD11      0.00      0.00      0.00      0.00     -33   \n",
       "4         TG      SAMD11      0.00      0.00      0.01      0.00      48   \n",
       "5        CAG      SAMD11      0.00      0.00      0.00      0.00      -3   \n",
       "6         CA      SAMD11      0.00      0.00      0.00      0.01     -33   \n",
       "7         CG      SAMD11      0.00      0.00      0.00      0.00     -33   \n",
       "8         AG      SAMD11      0.00      0.00      0.00      0.00       1   \n",
       "9          G      SAMD11      0.01      0.00      0.00      0.00     -14   \n",
       "\n",
       "   al_pos  dg_pos  dl_pos variant_type  \n",
       "0     -36     -35     -31        indel  \n",
       "1     -36      -7     -31        indel  \n",
       "2      11      29      -7        indel  \n",
       "3       2     -10       2        indel  \n",
       "4     -36     -36     -41        indel  \n",
       "5      33      23      33        indel  \n",
       "6      -5      -6     -34        indel  \n",
       "7      -5      -6     -34        indel  \n",
       "8     -32     -32      17        indel  \n",
       "9      18     -19     -14        indel  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "indels = indels[['chrom','pos','ref_allele','altai_gt','chagyrskaya_gt','denisovan_gt','vindija_gt','alt_allele','annotation','ag_delta','al_delta','dg_delta','dl_delta','ag_pos','al_pos','dg_pos','dl_pos','variant_type']]\n",
    "indels.head(10)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a5ccd23b",
   "metadata": {},
   "source": [
    "Now load the SNVs."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "6f3293c1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref_allele</th>\n",
       "      <th>altai_gt</th>\n",
       "      <th>chagyrskaya_gt</th>\n",
       "      <th>denisovan_gt</th>\n",
       "      <th>vindija_gt</th>\n",
       "      <th>alt_allele</th>\n",
       "      <th>annotation</th>\n",
       "      <th>ag_delta</th>\n",
       "      <th>al_delta</th>\n",
       "      <th>dg_delta</th>\n",
       "      <th>dl_delta</th>\n",
       "      <th>ag_pos</th>\n",
       "      <th>al_pos</th>\n",
       "      <th>dg_pos</th>\n",
       "      <th>dl_pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10</td>\n",
       "      <td>180887</td>\n",
       "      <td>C</td>\n",
       "      <td>0/0</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>0/0</td>\n",
       "      <td>A</td>\n",
       "      <td>ZMYND11</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>40</td>\n",
       "      <td>42</td>\n",
       "      <td>40</td>\n",
       "      <td>-2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10</td>\n",
       "      <td>180998</td>\n",
       "      <td>G</td>\n",
       "      <td>0/0</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>0/0</td>\n",
       "      <td>A</td>\n",
       "      <td>ZMYND11</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2</td>\n",
       "      <td>12</td>\n",
       "      <td>35</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>181000</td>\n",
       "      <td>T</td>\n",
       "      <td>0/0</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>0/0</td>\n",
       "      <td>A</td>\n",
       "      <td>ZMYND11</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-7</td>\n",
       "      <td>10</td>\n",
       "      <td>33</td>\n",
       "      <td>-2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>182007</td>\n",
       "      <td>C</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>T</td>\n",
       "      <td>ZMYND11</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-6</td>\n",
       "      <td>-24</td>\n",
       "      <td>40</td>\n",
       "      <td>-24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr10</td>\n",
       "      <td>182073</td>\n",
       "      <td>C</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>T</td>\n",
       "      <td>ZMYND11</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.0</td>\n",
       "      <td>12</td>\n",
       "      <td>-46</td>\n",
       "      <td>11</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr10</td>\n",
       "      <td>183370</td>\n",
       "      <td>C</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>T</td>\n",
       "      <td>ZMYND11</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-48</td>\n",
       "      <td>-30</td>\n",
       "      <td>-12</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr10</td>\n",
       "      <td>184616</td>\n",
       "      <td>T</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>G</td>\n",
       "      <td>ZMYND11</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>9</td>\n",
       "      <td>24</td>\n",
       "      <td>-1</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr10</td>\n",
       "      <td>185171</td>\n",
       "      <td>G</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>T</td>\n",
       "      <td>ZMYND11</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>38</td>\n",
       "      <td>1</td>\n",
       "      <td>37</td>\n",
       "      <td>-47</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr10</td>\n",
       "      <td>185264</td>\n",
       "      <td>G</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>A</td>\n",
       "      <td>ZMYND11</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2</td>\n",
       "      <td>19</td>\n",
       "      <td>2</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr10</td>\n",
       "      <td>185358</td>\n",
       "      <td>C</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>T</td>\n",
       "      <td>ZMYND11</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-41</td>\n",
       "      <td>-25</td>\n",
       "      <td>-29</td>\n",
       "      <td>-11</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   chrom     pos ref_allele altai_gt chagyrskaya_gt denisovan_gt vindija_gt  \\\n",
       "0  chr10  180887          C      0/0            ./.          ./.        0/0   \n",
       "1  chr10  180998          G      0/0            ./.          ./.        0/0   \n",
       "2  chr10  181000          T      0/0            ./.          ./.        0/0   \n",
       "3  chr10  182007          C      0/0            0/0          ./.        ./.   \n",
       "4  chr10  182073          C      1/1            0/0          0/0        0/0   \n",
       "5  chr10  183370          C      0/0            0/0          1/1        0/0   \n",
       "6  chr10  184616          T      1/1            1/1          0/0        1/1   \n",
       "7  chr10  185171          G      1/1            1/1          1/1        1/1   \n",
       "8  chr10  185264          G      0/0            0/0          1/1        0/0   \n",
       "9  chr10  185358          C      1/1            1/1          1/1        1/1   \n",
       "\n",
       "  alt_allele annotation  ag_delta  al_delta  dg_delta  dl_delta  ag_pos  \\\n",
       "0          A    ZMYND11      0.00      0.00      0.00       0.0      40   \n",
       "1          A    ZMYND11      0.00      0.00      0.00       0.0       2   \n",
       "2          A    ZMYND11      0.00      0.00      0.01       0.0      -7   \n",
       "3          T    ZMYND11      0.00      0.00      0.00       0.0      -6   \n",
       "4          T    ZMYND11      0.01      0.00      0.02       0.0      12   \n",
       "5          T    ZMYND11      0.00      0.00      0.00       0.0     -48   \n",
       "6          G    ZMYND11      0.00      0.00      0.00       0.0       9   \n",
       "7          T    ZMYND11      0.01      0.00      0.00       0.0      38   \n",
       "8          A    ZMYND11      0.00      0.01      0.00       0.0       2   \n",
       "9          T    ZMYND11      0.00      0.00      0.00       0.0     -41   \n",
       "\n",
       "   al_pos  dg_pos  dl_pos  \n",
       "0      42      40      -2  \n",
       "1      12      35       0  \n",
       "2      10      33      -2  \n",
       "3     -24      40     -24  \n",
       "4     -46      11       2  \n",
       "5     -30     -12      37  \n",
       "6      24      -1       3  \n",
       "7       1      37     -47  \n",
       "8      19       2      19  \n",
       "9     -25     -29     -11  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "snvs_header = ['chrom','pos','ref_allele','altai_gt','chagyrskaya_gt','denisovan_gt','vindija_gt','alt_allele','annotation','ag_delta','al_delta','dg_delta','dl_delta','ag_pos','al_pos','dg_pos','dl_pos']\n",
    "snvs = pd.read_csv('../../data/spliceai_outputs/spliceai_archaic_snvs.txt', sep = '\\t', names = snvs_header)\n",
    "snvs.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "e966548b",
   "metadata": {},
   "outputs": [],
   "source": [
    "snvs['variant_type'] = 'snv'"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4daea448",
   "metadata": {},
   "source": [
    "Concat the dataframes. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "b337f631",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref_allele</th>\n",
       "      <th>altai_gt</th>\n",
       "      <th>chagyrskaya_gt</th>\n",
       "      <th>denisovan_gt</th>\n",
       "      <th>vindija_gt</th>\n",
       "      <th>alt_allele</th>\n",
       "      <th>annotation</th>\n",
       "      <th>ag_delta</th>\n",
       "      <th>al_delta</th>\n",
       "      <th>dg_delta</th>\n",
       "      <th>dl_delta</th>\n",
       "      <th>ag_pos</th>\n",
       "      <th>al_pos</th>\n",
       "      <th>dg_pos</th>\n",
       "      <th>dl_pos</th>\n",
       "      <th>variant_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>739130</td>\n",
       "      <td>T</td>\n",
       "      <td>0/0</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>0/1</td>\n",
       "      <td>TA</td>\n",
       "      <td>AL669831.1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-7</td>\n",
       "      <td>-36</td>\n",
       "      <td>-35</td>\n",
       "      <td>-31</td>\n",
       "      <td>indel</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>739130</td>\n",
       "      <td>T</td>\n",
       "      <td>0/1</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>0/0</td>\n",
       "      <td>TAA</td>\n",
       "      <td>AL669831.1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-31</td>\n",
       "      <td>-36</td>\n",
       "      <td>-7</td>\n",
       "      <td>-31</td>\n",
       "      <td>indel</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1169649</th>\n",
       "      <td>chr1</td>\n",
       "      <td>861808</td>\n",
       "      <td>A</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>G</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-29</td>\n",
       "      <td>-27</td>\n",
       "      <td>24</td>\n",
       "      <td>-20</td>\n",
       "      <td>snv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1169650</th>\n",
       "      <td>chr1</td>\n",
       "      <td>861808</td>\n",
       "      <td>A</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>G</td>\n",
       "      <td>AL645608.1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>48</td>\n",
       "      <td>-14</td>\n",
       "      <td>-46</td>\n",
       "      <td>-14</td>\n",
       "      <td>snv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1169651</th>\n",
       "      <td>chr1</td>\n",
       "      <td>862072</td>\n",
       "      <td>C</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>T</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3</td>\n",
       "      <td>28</td>\n",
       "      <td>-38</td>\n",
       "      <td>41</td>\n",
       "      <td>snv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1169652</th>\n",
       "      <td>chr1</td>\n",
       "      <td>862072</td>\n",
       "      <td>C</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>T</td>\n",
       "      <td>AL645608.1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-6</td>\n",
       "      <td>-50</td>\n",
       "      <td>24</td>\n",
       "      <td>-48</td>\n",
       "      <td>snv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1169653</th>\n",
       "      <td>chr1</td>\n",
       "      <td>862093</td>\n",
       "      <td>T</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>C</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>7</td>\n",
       "      <td>14</td>\n",
       "      <td>-15</td>\n",
       "      <td>34</td>\n",
       "      <td>snv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1169654</th>\n",
       "      <td>chr1</td>\n",
       "      <td>862093</td>\n",
       "      <td>T</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>C</td>\n",
       "      <td>AL645608.1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-34</td>\n",
       "      <td>48</td>\n",
       "      <td>49</td>\n",
       "      <td>3</td>\n",
       "      <td>snv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1169655</th>\n",
       "      <td>chr1</td>\n",
       "      <td>862124</td>\n",
       "      <td>A</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>G</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-24</td>\n",
       "      <td>26</td>\n",
       "      <td>-11</td>\n",
       "      <td>3</td>\n",
       "      <td>snv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1169656</th>\n",
       "      <td>chr1</td>\n",
       "      <td>862124</td>\n",
       "      <td>A</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>G</td>\n",
       "      <td>AL645608.1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>17</td>\n",
       "      <td>-14</td>\n",
       "      <td>-28</td>\n",
       "      <td>35</td>\n",
       "      <td>snv</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        chrom     pos ref_allele altai_gt chagyrskaya_gt denisovan_gt  \\\n",
       "0        chr1  739130          T      0/0            ./.          ./.   \n",
       "1        chr1  739130          T      0/1            ./.          ./.   \n",
       "1169649  chr1  861808          A      0/0            0/0          1/1   \n",
       "1169650  chr1  861808          A      0/0            0/0          1/1   \n",
       "1169651  chr1  862072          C      1/1            1/1          0/0   \n",
       "1169652  chr1  862072          C      1/1            1/1          0/0   \n",
       "1169653  chr1  862093          T      0/0            0/0          1/1   \n",
       "1169654  chr1  862093          T      0/0            0/0          1/1   \n",
       "1169655  chr1  862124          A      0/0            0/0          1/1   \n",
       "1169656  chr1  862124          A      0/0            0/0          1/1   \n",
       "\n",
       "        vindija_gt alt_allele  annotation  ag_delta  al_delta  dg_delta  \\\n",
       "0              0/1         TA  AL669831.1       0.0       0.0      0.00   \n",
       "1              0/0        TAA  AL669831.1       0.0       0.0      0.00   \n",
       "1169649        0/0          G      SAMD11       0.0       0.0      0.01   \n",
       "1169650        0/0          G  AL645608.1       0.0       0.0      0.00   \n",
       "1169651        0/1          T      SAMD11       0.0       0.0      0.00   \n",
       "1169652        0/1          T  AL645608.1       0.0       0.0      0.00   \n",
       "1169653        0/0          C      SAMD11       0.0       0.0      0.00   \n",
       "1169654        0/0          C  AL645608.1       0.0       0.0      0.00   \n",
       "1169655        0/0          G      SAMD11       0.0       0.0      0.00   \n",
       "1169656        0/0          G  AL645608.1       0.0       0.0      0.00   \n",
       "\n",
       "         dl_delta  ag_pos  al_pos  dg_pos  dl_pos variant_type  \n",
       "0             0.0      -7     -36     -35     -31        indel  \n",
       "1             0.0     -31     -36      -7     -31        indel  \n",
       "1169649       0.0     -29     -27      24     -20          snv  \n",
       "1169650       0.0      48     -14     -46     -14          snv  \n",
       "1169651       0.0       3      28     -38      41          snv  \n",
       "1169652       0.0      -6     -50      24     -48          snv  \n",
       "1169653       0.0       7      14     -15      34          snv  \n",
       "1169654       0.0     -34      48      49       3          snv  \n",
       "1169655       0.0     -24      26     -11       3          snv  \n",
       "1169656       0.0      17     -14     -28      35          snv  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data = pd.concat([indels, snvs], ignore_index = True)\n",
    "data = data.sort_values(by = ['chrom', 'pos'])\n",
    "data.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "1e6ae3be",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2269396"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(data)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9fe842dc",
   "metadata": {},
   "source": [
    "Now let's add columns to designate whether at least one alternate allele is present per individual using a boolean (>= 1 alternate allele present = TRUE). "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "51265ca3",
   "metadata": {},
   "outputs": [],
   "source": [
    "data['altai_gt_boolean'] = data['altai_gt'].replace({'./.':'FALSE', '0/0':'FALSE', '1/1':'TRUE', '1/0':'TRUE', '0/1':'TRUE'})\n",
    "data['chagyrskaya_gt_boolean'] = data['chagyrskaya_gt'].replace({'./.':'FALSE', '0/0':'FALSE', '1/1':'TRUE', '1/0':'TRUE', '0/1':'TRUE'})\n",
    "data['denisovan_gt_boolean'] = data['denisovan_gt'].replace({'./.':'FALSE', '0/0':'FALSE', '1/1':'TRUE', '1/0':'TRUE', '0/1':'TRUE'})\n",
    "data['vindija_gt_boolean'] = data['vindija_gt'].replace({'./.':'FALSE', '0/0':'FALSE', '1/1':'TRUE', '1/0':'TRUE', '0/1':'TRUE'})"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "22f5d887",
   "metadata": {},
   "source": [
    "Now that we know which archaics have at least one alternate allele, let's assign a distribution variable that indicates which archaics have one or two alternate alleles. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "b1bac967",
   "metadata": {},
   "outputs": [],
   "source": [
    "def distribution(data):\n",
    "    \n",
    "    if (data['altai_gt_boolean'] == 'TRUE') & (data['chagyrskaya_gt_boolean'] == 'FALSE') & (data['denisovan_gt_boolean'] == 'FALSE') & (data['vindija_gt_boolean'] == 'FALSE'):\n",
    "        return 'Altai'\n",
    "    elif (data['altai_gt_boolean'] == 'FALSE') & (data['chagyrskaya_gt_boolean'] == 'TRUE') & (data['denisovan_gt_boolean'] == 'FALSE') & (data['vindija_gt_boolean'] == 'FALSE'):\n",
    "        return 'Chagyrskaya'\n",
    "    elif (data['altai_gt_boolean'] == 'FALSE') & (data['chagyrskaya_gt_boolean'] == 'FALSE') & (data['denisovan_gt_boolean'] == 'TRUE') & (data['vindija_gt_boolean'] == 'FALSE'):\n",
    "        return 'Denisovan'\n",
    "    elif (data['altai_gt_boolean'] == 'FALSE') & (data['chagyrskaya_gt_boolean'] == 'FALSE') & (data['denisovan_gt_boolean'] == 'FALSE') & (data['vindija_gt_boolean'] == 'TRUE'):\n",
    "        return 'Vindija'\n",
    "    elif (data['altai_gt_boolean'] == 'TRUE') & (data['chagyrskaya_gt_boolean'] == 'TRUE') & (data['denisovan_gt_boolean'] == 'FALSE') & (data['vindija_gt_boolean'] == 'TRUE'):\n",
    "        return 'Neanderthal'\n",
    "    elif (data['altai_gt_boolean'] == 'FALSE') & (data['chagyrskaya_gt_boolean'] == 'TRUE') & (data['denisovan_gt_boolean'] == 'FALSE') & (data['vindija_gt_boolean'] == 'TRUE'):\n",
    "        return 'Late Neanderthal'\n",
    "    elif (data['altai_gt_boolean'] == 'TRUE') & (data['chagyrskaya_gt_boolean'] == 'FALSE') & (data['denisovan_gt_boolean'] == 'TRUE') & (data['vindija_gt_boolean'] == 'TRUE') & (data['variant_type'] == 'indel'):\n",
    "        return 'Shared'\n",
    "    elif (data['altai_gt_boolean'] == 'TRUE') & (data['chagyrskaya_gt_boolean'] == 'TRUE') & (data['denisovan_gt_boolean'] == 'TRUE') & (data['vindija_gt_boolean'] == 'TRUE') & (data['variant_type'] == 'snv'):\n",
    "        return 'Shared'\n",
    "    else:\n",
    "        return 'Other'\n",
    "    \n",
    "data['distribution'] = data.apply(distribution, axis = 1)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ff1fa2f0",
   "metadata": {},
   "source": [
    "Some rows may actually be invariant, likely because we set low quality genotypes to missing and/or invariant sites were included in the original data. We need to remove these rows."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "a17e8a01",
   "metadata": {},
   "outputs": [],
   "source": [
    "data_drop_indices = data[ (data['altai_gt_boolean'] == 'FALSE') & (data['chagyrskaya_gt_boolean'] == 'FALSE') & (data['denisovan_gt_boolean'] == 'FALSE') & (data['vindija_gt_boolean'] == 'FALSE')].index\n",
    "data.drop(data_drop_indices, inplace = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "c2c85124",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2107054"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(data)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4ccc3699",
   "metadata": {},
   "source": [
    "In multiple downstream analyses, we will need the maximum delta value among AG, AL, DG, and DL. Let's add a row with that information now."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "cdcc3b9b",
   "metadata": {},
   "outputs": [],
   "source": [
    "data['delta_max'] = data[['ag_delta','al_delta','dg_delta','dl_delta']].max(axis = 1)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ea1ac8dd",
   "metadata": {},
   "source": [
    "Let's reorder the columns a bit so that value occurs near the other deltas. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "3cdbe732",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref_allele</th>\n",
       "      <th>altai_gt</th>\n",
       "      <th>chagyrskaya_gt</th>\n",
       "      <th>denisovan_gt</th>\n",
       "      <th>vindija_gt</th>\n",
       "      <th>alt_allele</th>\n",
       "      <th>annotation</th>\n",
       "      <th>ag_delta</th>\n",
       "      <th>al_delta</th>\n",
       "      <th>dg_delta</th>\n",
       "      <th>dl_delta</th>\n",
       "      <th>delta_max</th>\n",
       "      <th>ag_pos</th>\n",
       "      <th>al_pos</th>\n",
       "      <th>dg_pos</th>\n",
       "      <th>dl_pos</th>\n",
       "      <th>variant_type</th>\n",
       "      <th>altai_gt_boolean</th>\n",
       "      <th>chagyrskaya_gt_boolean</th>\n",
       "      <th>denisovan_gt_boolean</th>\n",
       "      <th>vindija_gt_boolean</th>\n",
       "      <th>distribution</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>739130</td>\n",
       "      <td>T</td>\n",
       "      <td>0/0</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>0/1</td>\n",
       "      <td>TA</td>\n",
       "      <td>AL669831.1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-7</td>\n",
       "      <td>-36</td>\n",
       "      <td>-35</td>\n",
       "      <td>-31</td>\n",
       "      <td>indel</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>Vindija</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>739130</td>\n",
       "      <td>T</td>\n",
       "      <td>0/1</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>0/0</td>\n",
       "      <td>TAA</td>\n",
       "      <td>AL669831.1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-31</td>\n",
       "      <td>-36</td>\n",
       "      <td>-7</td>\n",
       "      <td>-31</td>\n",
       "      <td>indel</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>Altai</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1169649</th>\n",
       "      <td>chr1</td>\n",
       "      <td>861808</td>\n",
       "      <td>A</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>G</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.01</td>\n",
       "      <td>-29</td>\n",
       "      <td>-27</td>\n",
       "      <td>24</td>\n",
       "      <td>-20</td>\n",
       "      <td>snv</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>Denisovan</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1169650</th>\n",
       "      <td>chr1</td>\n",
       "      <td>861808</td>\n",
       "      <td>A</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>G</td>\n",
       "      <td>AL645608.1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>48</td>\n",
       "      <td>-14</td>\n",
       "      <td>-46</td>\n",
       "      <td>-14</td>\n",
       "      <td>snv</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>Denisovan</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1169651</th>\n",
       "      <td>chr1</td>\n",
       "      <td>862072</td>\n",
       "      <td>C</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>T</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>3</td>\n",
       "      <td>28</td>\n",
       "      <td>-38</td>\n",
       "      <td>41</td>\n",
       "      <td>snv</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>Neanderthal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1169652</th>\n",
       "      <td>chr1</td>\n",
       "      <td>862072</td>\n",
       "      <td>C</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>T</td>\n",
       "      <td>AL645608.1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-6</td>\n",
       "      <td>-50</td>\n",
       "      <td>24</td>\n",
       "      <td>-48</td>\n",
       "      <td>snv</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>Neanderthal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1169653</th>\n",
       "      <td>chr1</td>\n",
       "      <td>862093</td>\n",
       "      <td>T</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>C</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>7</td>\n",
       "      <td>14</td>\n",
       "      <td>-15</td>\n",
       "      <td>34</td>\n",
       "      <td>snv</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>Denisovan</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1169654</th>\n",
       "      <td>chr1</td>\n",
       "      <td>862093</td>\n",
       "      <td>T</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>C</td>\n",
       "      <td>AL645608.1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-34</td>\n",
       "      <td>48</td>\n",
       "      <td>49</td>\n",
       "      <td>3</td>\n",
       "      <td>snv</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>Denisovan</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1169655</th>\n",
       "      <td>chr1</td>\n",
       "      <td>862124</td>\n",
       "      <td>A</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>G</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-24</td>\n",
       "      <td>26</td>\n",
       "      <td>-11</td>\n",
       "      <td>3</td>\n",
       "      <td>snv</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>Denisovan</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1169656</th>\n",
       "      <td>chr1</td>\n",
       "      <td>862124</td>\n",
       "      <td>A</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>G</td>\n",
       "      <td>AL645608.1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>17</td>\n",
       "      <td>-14</td>\n",
       "      <td>-28</td>\n",
       "      <td>35</td>\n",
       "      <td>snv</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>Denisovan</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        chrom     pos ref_allele altai_gt chagyrskaya_gt denisovan_gt  \\\n",
       "0        chr1  739130          T      0/0            ./.          ./.   \n",
       "1        chr1  739130          T      0/1            ./.          ./.   \n",
       "1169649  chr1  861808          A      0/0            0/0          1/1   \n",
       "1169650  chr1  861808          A      0/0            0/0          1/1   \n",
       "1169651  chr1  862072          C      1/1            1/1          0/0   \n",
       "1169652  chr1  862072          C      1/1            1/1          0/0   \n",
       "1169653  chr1  862093          T      0/0            0/0          1/1   \n",
       "1169654  chr1  862093          T      0/0            0/0          1/1   \n",
       "1169655  chr1  862124          A      0/0            0/0          1/1   \n",
       "1169656  chr1  862124          A      0/0            0/0          1/1   \n",
       "\n",
       "        vindija_gt alt_allele  annotation  ag_delta  al_delta  dg_delta  \\\n",
       "0              0/1         TA  AL669831.1       0.0       0.0      0.00   \n",
       "1              0/0        TAA  AL669831.1       0.0       0.0      0.00   \n",
       "1169649        0/0          G      SAMD11       0.0       0.0      0.01   \n",
       "1169650        0/0          G  AL645608.1       0.0       0.0      0.00   \n",
       "1169651        0/1          T      SAMD11       0.0       0.0      0.00   \n",
       "1169652        0/1          T  AL645608.1       0.0       0.0      0.00   \n",
       "1169653        0/0          C      SAMD11       0.0       0.0      0.00   \n",
       "1169654        0/0          C  AL645608.1       0.0       0.0      0.00   \n",
       "1169655        0/0          G      SAMD11       0.0       0.0      0.00   \n",
       "1169656        0/0          G  AL645608.1       0.0       0.0      0.00   \n",
       "\n",
       "         dl_delta  delta_max  ag_pos  al_pos  dg_pos  dl_pos variant_type  \\\n",
       "0             0.0       0.00      -7     -36     -35     -31        indel   \n",
       "1             0.0       0.00     -31     -36      -7     -31        indel   \n",
       "1169649       0.0       0.01     -29     -27      24     -20          snv   \n",
       "1169650       0.0       0.00      48     -14     -46     -14          snv   \n",
       "1169651       0.0       0.00       3      28     -38      41          snv   \n",
       "1169652       0.0       0.00      -6     -50      24     -48          snv   \n",
       "1169653       0.0       0.00       7      14     -15      34          snv   \n",
       "1169654       0.0       0.00     -34      48      49       3          snv   \n",
       "1169655       0.0       0.00     -24      26     -11       3          snv   \n",
       "1169656       0.0       0.00      17     -14     -28      35          snv   \n",
       "\n",
       "        altai_gt_boolean chagyrskaya_gt_boolean denisovan_gt_boolean  \\\n",
       "0                  FALSE                  FALSE                FALSE   \n",
       "1                   TRUE                  FALSE                FALSE   \n",
       "1169649            FALSE                  FALSE                 TRUE   \n",
       "1169650            FALSE                  FALSE                 TRUE   \n",
       "1169651             TRUE                   TRUE                FALSE   \n",
       "1169652             TRUE                   TRUE                FALSE   \n",
       "1169653            FALSE                  FALSE                 TRUE   \n",
       "1169654            FALSE                  FALSE                 TRUE   \n",
       "1169655            FALSE                  FALSE                 TRUE   \n",
       "1169656            FALSE                  FALSE                 TRUE   \n",
       "\n",
       "        vindija_gt_boolean distribution  \n",
       "0                     TRUE      Vindija  \n",
       "1                    FALSE        Altai  \n",
       "1169649              FALSE    Denisovan  \n",
       "1169650              FALSE    Denisovan  \n",
       "1169651               TRUE  Neanderthal  \n",
       "1169652               TRUE  Neanderthal  \n",
       "1169653              FALSE    Denisovan  \n",
       "1169654              FALSE    Denisovan  \n",
       "1169655              FALSE    Denisovan  \n",
       "1169656              FALSE    Denisovan  "
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data = data[['chrom','pos','ref_allele','altai_gt','chagyrskaya_gt','denisovan_gt','vindija_gt','alt_allele','annotation','ag_delta','al_delta','dg_delta','dl_delta','delta_max','ag_pos','al_pos','dg_pos','dl_pos','variant_type','altai_gt_boolean','chagyrskaya_gt_boolean','denisovan_gt_boolean','vindija_gt_boolean','distribution']]\n",
    "data.head(10)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "50108cb9",
   "metadata": {},
   "source": [
    "Let's load some data on constraint at the gene-level and add four relevant ones to our dataframe: 1) missense observed/expected, 2) loss-of-function observed/expected, 3) missense z-score and 4)loss-of-function z-score."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "44853c20",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>transcript</th>\n",
       "      <th>obs_mis</th>\n",
       "      <th>exp_mis</th>\n",
       "      <th>oe_mis</th>\n",
       "      <th>mu_mis</th>\n",
       "      <th>possible_mis</th>\n",
       "      <th>obs_mis_pphen</th>\n",
       "      <th>exp_mis_pphen</th>\n",
       "      <th>oe_mis_pphen</th>\n",
       "      <th>possible_mis_pphen</th>\n",
       "      <th>obs_syn</th>\n",
       "      <th>exp_syn</th>\n",
       "      <th>oe_syn</th>\n",
       "      <th>mu_syn</th>\n",
       "      <th>possible_syn</th>\n",
       "      <th>obs_lof</th>\n",
       "      <th>mu_lof</th>\n",
       "      <th>possible_lof</th>\n",
       "      <th>exp_lof</th>\n",
       "      <th>pLI</th>\n",
       "      <th>pNull</th>\n",
       "      <th>pRec</th>\n",
       "      <th>oe_lof</th>\n",
       "      <th>oe_syn_lower</th>\n",
       "      <th>oe_syn_upper</th>\n",
       "      <th>oe_mis_lower</th>\n",
       "      <th>oe_mis_upper</th>\n",
       "      <th>oe_lof_lower</th>\n",
       "      <th>oe_lof_upper</th>\n",
       "      <th>constraint_flag</th>\n",
       "      <th>syn_z</th>\n",
       "      <th>mis_z</th>\n",
       "      <th>lof_z</th>\n",
       "      <th>oe_lof_upper_rank</th>\n",
       "      <th>oe_lof_upper_bin</th>\n",
       "      <th>oe_lof_upper_bin_6</th>\n",
       "      <th>n_sites</th>\n",
       "      <th>classic_caf</th>\n",
       "      <th>max_af</th>\n",
       "      <th>no_lofs</th>\n",
       "      <th>obs_het_lof</th>\n",
       "      <th>obs_hom_lof</th>\n",
       "      <th>defined</th>\n",
       "      <th>p</th>\n",
       "      <th>exp_hom_lof</th>\n",
       "      <th>classic_caf_afr</th>\n",
       "      <th>classic_caf_amr</th>\n",
       "      <th>classic_caf_asj</th>\n",
       "      <th>classic_caf_eas</th>\n",
       "      <th>classic_caf_fin</th>\n",
       "      <th>classic_caf_nfe</th>\n",
       "      <th>classic_caf_oth</th>\n",
       "      <th>classic_caf_sas</th>\n",
       "      <th>p_afr</th>\n",
       "      <th>p_amr</th>\n",
       "      <th>p_asj</th>\n",
       "      <th>p_eas</th>\n",
       "      <th>p_fin</th>\n",
       "      <th>p_nfe</th>\n",
       "      <th>p_oth</th>\n",
       "      <th>p_sas</th>\n",
       "      <th>transcript_type</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>transcript_level</th>\n",
       "      <th>cds_length</th>\n",
       "      <th>num_coding_exons</th>\n",
       "      <th>gene_type</th>\n",
       "      <th>gene_length</th>\n",
       "      <th>exac_pLI</th>\n",
       "      <th>exac_obs_lof</th>\n",
       "      <th>exac_exp_lof</th>\n",
       "      <th>exac_oe_lof</th>\n",
       "      <th>brain_expression</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>start_position</th>\n",
       "      <th>end_position</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MED13</td>\n",
       "      <td>ENST00000397786</td>\n",
       "      <td>871</td>\n",
       "      <td>1117.80</td>\n",
       "      <td>0.77921</td>\n",
       "      <td>0.000056</td>\n",
       "      <td>14195</td>\n",
       "      <td>314.0</td>\n",
       "      <td>529.75</td>\n",
       "      <td>0.59273</td>\n",
       "      <td>6708.0</td>\n",
       "      <td>422</td>\n",
       "      <td>387.53</td>\n",
       "      <td>1.08900</td>\n",
       "      <td>0.000019</td>\n",
       "      <td>4248</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>1257.0</td>\n",
       "      <td>98.429</td>\n",
       "      <td>1.0</td>\n",
       "      <td>8.943600e-40</td>\n",
       "      <td>1.838300e-16</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.005</td>\n",
       "      <td>1.180</td>\n",
       "      <td>0.736</td>\n",
       "      <td>0.824</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.030</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.376500</td>\n",
       "      <td>2.6232</td>\n",
       "      <td>9.1935</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.000012</td>\n",
       "      <td>0.000008</td>\n",
       "      <td>124782.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>124785.0</td>\n",
       "      <td>0.000012</td>\n",
       "      <td>0.000018</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000093</td>\n",
       "      <td>0.000009</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000093</td>\n",
       "      <td>0.000009</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>ENSG00000108510</td>\n",
       "      <td>2</td>\n",
       "      <td>6522</td>\n",
       "      <td>30</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>122678</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>64.393</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>17</td>\n",
       "      <td>60019966</td>\n",
       "      <td>60142643</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NIPBL</td>\n",
       "      <td>ENST00000282516</td>\n",
       "      <td>846</td>\n",
       "      <td>1441.50</td>\n",
       "      <td>0.58688</td>\n",
       "      <td>0.000074</td>\n",
       "      <td>18540</td>\n",
       "      <td>158.0</td>\n",
       "      <td>543.10</td>\n",
       "      <td>0.29092</td>\n",
       "      <td>7135.0</td>\n",
       "      <td>496</td>\n",
       "      <td>495.01</td>\n",
       "      <td>1.00200</td>\n",
       "      <td>0.000025</td>\n",
       "      <td>5211</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.000009</td>\n",
       "      <td>1781.0</td>\n",
       "      <td>150.320</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.977300e-59</td>\n",
       "      <td>3.572400e-24</td>\n",
       "      <td>0.006653</td>\n",
       "      <td>0.930</td>\n",
       "      <td>1.079</td>\n",
       "      <td>0.554</td>\n",
       "      <td>0.621</td>\n",
       "      <td>0.001</td>\n",
       "      <td>0.032</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.035119</td>\n",
       "      <td>5.5737</td>\n",
       "      <td>11.2860</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.000012</td>\n",
       "      <td>0.000008</td>\n",
       "      <td>125693.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>125696.0</td>\n",
       "      <td>0.000012</td>\n",
       "      <td>0.000018</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000099</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000065</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000099</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000065</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>ENSG00000164190</td>\n",
       "      <td>2</td>\n",
       "      <td>8412</td>\n",
       "      <td>46</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>189655</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>110.570</td>\n",
       "      <td>0.009044</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5</td>\n",
       "      <td>36876861</td>\n",
       "      <td>37066515</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SMC3</td>\n",
       "      <td>ENST00000361804</td>\n",
       "      <td>178</td>\n",
       "      <td>630.07</td>\n",
       "      <td>0.28251</td>\n",
       "      <td>0.000032</td>\n",
       "      <td>8109</td>\n",
       "      <td>21.0</td>\n",
       "      <td>182.52</td>\n",
       "      <td>0.11506</td>\n",
       "      <td>2197.0</td>\n",
       "      <td>215</td>\n",
       "      <td>203.25</td>\n",
       "      <td>1.05780</td>\n",
       "      <td>0.000010</td>\n",
       "      <td>2091</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>937.0</td>\n",
       "      <td>79.490</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.785300e-32</td>\n",
       "      <td>2.191400e-13</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.946</td>\n",
       "      <td>1.184</td>\n",
       "      <td>0.249</td>\n",
       "      <td>0.320</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.037</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.647760</td>\n",
       "      <td>6.3999</td>\n",
       "      <td>8.2618</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>0.000032</td>\n",
       "      <td>0.000004</td>\n",
       "      <td>125731.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>125739.0</td>\n",
       "      <td>0.000032</td>\n",
       "      <td>0.000127</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000099</td>\n",
       "      <td>0.000054</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000044</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000033</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000099</td>\n",
       "      <td>0.000054</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000044</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000033</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>ENSG00000108055</td>\n",
       "      <td>2</td>\n",
       "      <td>3651</td>\n",
       "      <td>29</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>36946</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>58.523</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10</td>\n",
       "      <td>112327449</td>\n",
       "      <td>112364394</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CNOT1</td>\n",
       "      <td>ENST00000317147</td>\n",
       "      <td>561</td>\n",
       "      <td>1295.90</td>\n",
       "      <td>0.43290</td>\n",
       "      <td>0.000069</td>\n",
       "      <td>15670</td>\n",
       "      <td>51.0</td>\n",
       "      <td>290.68</td>\n",
       "      <td>0.17545</td>\n",
       "      <td>3560.0</td>\n",
       "      <td>470</td>\n",
       "      <td>456.03</td>\n",
       "      <td>1.03060</td>\n",
       "      <td>0.000024</td>\n",
       "      <td>4564</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>1440.0</td>\n",
       "      <td>125.030</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.992400e-49</td>\n",
       "      <td>4.562900e-20</td>\n",
       "      <td>0.007998</td>\n",
       "      <td>0.955</td>\n",
       "      <td>1.112</td>\n",
       "      <td>0.403</td>\n",
       "      <td>0.464</td>\n",
       "      <td>0.002</td>\n",
       "      <td>0.038</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.514100</td>\n",
       "      <td>7.2546</td>\n",
       "      <td>10.2790</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.000020</td>\n",
       "      <td>0.000004</td>\n",
       "      <td>125740.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>125744.0</td>\n",
       "      <td>0.000016</td>\n",
       "      <td>0.000032</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000029</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000055</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000026</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000029</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000054</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000018</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>ENSG00000125107</td>\n",
       "      <td>2</td>\n",
       "      <td>7128</td>\n",
       "      <td>48</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>109936</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>90.130</td>\n",
       "      <td>0.033285</td>\n",
       "      <td>NaN</td>\n",
       "      <td>16</td>\n",
       "      <td>58553855</td>\n",
       "      <td>58663790</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>RLF</td>\n",
       "      <td>ENST00000372771</td>\n",
       "      <td>669</td>\n",
       "      <td>972.87</td>\n",
       "      <td>0.68766</td>\n",
       "      <td>0.000047</td>\n",
       "      <td>12682</td>\n",
       "      <td>107.0</td>\n",
       "      <td>321.14</td>\n",
       "      <td>0.33319</td>\n",
       "      <td>4151.0</td>\n",
       "      <td>358</td>\n",
       "      <td>352.62</td>\n",
       "      <td>1.01530</td>\n",
       "      <td>0.000017</td>\n",
       "      <td>3482</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000004</td>\n",
       "      <td>1024.0</td>\n",
       "      <td>73.222</td>\n",
       "      <td>1.0</td>\n",
       "      <td>8.405500e-30</td>\n",
       "      <td>2.284200e-12</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.930</td>\n",
       "      <td>1.108</td>\n",
       "      <td>0.645</td>\n",
       "      <td>0.733</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.040</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.225180</td>\n",
       "      <td>3.4620</td>\n",
       "      <td>7.9294</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.000004</td>\n",
       "      <td>0.000004</td>\n",
       "      <td>125122.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>125123.0</td>\n",
       "      <td>0.000004</td>\n",
       "      <td>0.000002</td>\n",
       "      <td>0.000062</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000062</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>ENSG00000117000</td>\n",
       "      <td>2</td>\n",
       "      <td>5742</td>\n",
       "      <td>8</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>79549</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>43.607</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>40627045</td>\n",
       "      <td>40706593</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>PCF11</td>\n",
       "      <td>ENST00000298281</td>\n",
       "      <td>574</td>\n",
       "      <td>783.46</td>\n",
       "      <td>0.73265</td>\n",
       "      <td>0.000039</td>\n",
       "      <td>10106</td>\n",
       "      <td>130.0</td>\n",
       "      <td>264.36</td>\n",
       "      <td>0.49176</td>\n",
       "      <td>3521.0</td>\n",
       "      <td>298</td>\n",
       "      <td>272.69</td>\n",
       "      <td>1.09280</td>\n",
       "      <td>0.000013</td>\n",
       "      <td>3026</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>862.0</td>\n",
       "      <td>74.559</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.487200e-30</td>\n",
       "      <td>1.385500e-12</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.993</td>\n",
       "      <td>1.203</td>\n",
       "      <td>0.683</td>\n",
       "      <td>0.785</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.040</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.205000</td>\n",
       "      <td>2.6592</td>\n",
       "      <td>8.0014</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.000016</td>\n",
       "      <td>0.000004</td>\n",
       "      <td>124625.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>124628.0</td>\n",
       "      <td>0.000012</td>\n",
       "      <td>0.000018</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000111</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000018</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000056</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000018</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>ENSG00000165494</td>\n",
       "      <td>2</td>\n",
       "      <td>4665</td>\n",
       "      <td>16</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>30464</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>48.160</td>\n",
       "      <td>0.020764</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11</td>\n",
       "      <td>82868030</td>\n",
       "      <td>82898493</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>FNDC3B</td>\n",
       "      <td>ENST00000336824</td>\n",
       "      <td>551</td>\n",
       "      <td>689.52</td>\n",
       "      <td>0.79911</td>\n",
       "      <td>0.000038</td>\n",
       "      <td>7833</td>\n",
       "      <td>75.0</td>\n",
       "      <td>160.48</td>\n",
       "      <td>0.46735</td>\n",
       "      <td>1794.0</td>\n",
       "      <td>274</td>\n",
       "      <td>265.93</td>\n",
       "      <td>1.03040</td>\n",
       "      <td>0.000017</td>\n",
       "      <td>2360</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000004</td>\n",
       "      <td>805.0</td>\n",
       "      <td>69.811</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.879200e-28</td>\n",
       "      <td>8.179200e-12</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.933</td>\n",
       "      <td>1.139</td>\n",
       "      <td>0.744</td>\n",
       "      <td>0.857</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.042</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.389080</td>\n",
       "      <td>1.8745</td>\n",
       "      <td>7.7425</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.000008</td>\n",
       "      <td>0.000004</td>\n",
       "      <td>125745.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>125747.0</td>\n",
       "      <td>0.000008</td>\n",
       "      <td>0.000008</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000029</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000009</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000029</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000009</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>ENSG00000075420</td>\n",
       "      <td>2</td>\n",
       "      <td>3612</td>\n",
       "      <td>25</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>362038</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>49.447</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>171757418</td>\n",
       "      <td>172119455</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>TAF1</td>\n",
       "      <td>ENST00000276072</td>\n",
       "      <td>326</td>\n",
       "      <td>748.30</td>\n",
       "      <td>0.43565</td>\n",
       "      <td>0.000060</td>\n",
       "      <td>12563</td>\n",
       "      <td>82.0</td>\n",
       "      <td>312.02</td>\n",
       "      <td>0.26280</td>\n",
       "      <td>5365.0</td>\n",
       "      <td>272</td>\n",
       "      <td>250.43</td>\n",
       "      <td>1.08610</td>\n",
       "      <td>0.000018</td>\n",
       "      <td>3522</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>1203.0</td>\n",
       "      <td>68.906</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4.288100e-28</td>\n",
       "      <td>1.147700e-11</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.983</td>\n",
       "      <td>1.201</td>\n",
       "      <td>0.397</td>\n",
       "      <td>0.477</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.043</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.071200</td>\n",
       "      <td>5.4860</td>\n",
       "      <td>7.6921</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.000093</td>\n",
       "      <td>0.000087</td>\n",
       "      <td>125612.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>125629.0</td>\n",
       "      <td>0.000068</td>\n",
       "      <td>0.000575</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000535</td>\n",
       "      <td>0.000794</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000012</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000052</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000397</td>\n",
       "      <td>0.000598</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000009</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000033</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>ENSG00000147133</td>\n",
       "      <td>2</td>\n",
       "      <td>5679</td>\n",
       "      <td>38</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>166111</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>50.800</td>\n",
       "      <td>0.039370</td>\n",
       "      <td>NaN</td>\n",
       "      <td>X</td>\n",
       "      <td>70586114</td>\n",
       "      <td>70752224</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>RSF1</td>\n",
       "      <td>ENST00000308488</td>\n",
       "      <td>620</td>\n",
       "      <td>756.71</td>\n",
       "      <td>0.81933</td>\n",
       "      <td>0.000040</td>\n",
       "      <td>9552</td>\n",
       "      <td>206.0</td>\n",
       "      <td>310.49</td>\n",
       "      <td>0.66348</td>\n",
       "      <td>3722.0</td>\n",
       "      <td>259</td>\n",
       "      <td>265.17</td>\n",
       "      <td>0.97673</td>\n",
       "      <td>0.000013</td>\n",
       "      <td>2603</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000004</td>\n",
       "      <td>846.0</td>\n",
       "      <td>67.830</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.142700e-27</td>\n",
       "      <td>1.716200e-11</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.882</td>\n",
       "      <td>1.083</td>\n",
       "      <td>0.766</td>\n",
       "      <td>0.876</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.044</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.297860</td>\n",
       "      <td>1.7661</td>\n",
       "      <td>7.6318</td>\n",
       "      <td>8.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>ENSG00000048649</td>\n",
       "      <td>2</td>\n",
       "      <td>4323</td>\n",
       "      <td>16</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>161023</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>46.996</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11</td>\n",
       "      <td>77371041</td>\n",
       "      <td>77532063</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>NCKAP1</td>\n",
       "      <td>ENST00000360982</td>\n",
       "      <td>291</td>\n",
       "      <td>580.53</td>\n",
       "      <td>0.50127</td>\n",
       "      <td>0.000029</td>\n",
       "      <td>7483</td>\n",
       "      <td>56.0</td>\n",
       "      <td>154.07</td>\n",
       "      <td>0.36346</td>\n",
       "      <td>2017.0</td>\n",
       "      <td>194</td>\n",
       "      <td>197.96</td>\n",
       "      <td>0.98000</td>\n",
       "      <td>0.000010</td>\n",
       "      <td>2051</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000004</td>\n",
       "      <td>850.0</td>\n",
       "      <td>67.231</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.972600e-27</td>\n",
       "      <td>2.147300e-11</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.871</td>\n",
       "      <td>1.104</td>\n",
       "      <td>0.455</td>\n",
       "      <td>0.552</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.044</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.221170</td>\n",
       "      <td>4.2701</td>\n",
       "      <td>7.5981</td>\n",
       "      <td>9.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>ENSG00000061676</td>\n",
       "      <td>2</td>\n",
       "      <td>3402</td>\n",
       "      <td>32</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>129744</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>50.723</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>183773843</td>\n",
       "      <td>183903586</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     gene       transcript  obs_mis  exp_mis   oe_mis    mu_mis  possible_mis  \\\n",
       "0   MED13  ENST00000397786      871  1117.80  0.77921  0.000056         14195   \n",
       "1   NIPBL  ENST00000282516      846  1441.50  0.58688  0.000074         18540   \n",
       "2    SMC3  ENST00000361804      178   630.07  0.28251  0.000032          8109   \n",
       "3   CNOT1  ENST00000317147      561  1295.90  0.43290  0.000069         15670   \n",
       "4     RLF  ENST00000372771      669   972.87  0.68766  0.000047         12682   \n",
       "5   PCF11  ENST00000298281      574   783.46  0.73265  0.000039         10106   \n",
       "6  FNDC3B  ENST00000336824      551   689.52  0.79911  0.000038          7833   \n",
       "7    TAF1  ENST00000276072      326   748.30  0.43565  0.000060         12563   \n",
       "8    RSF1  ENST00000308488      620   756.71  0.81933  0.000040          9552   \n",
       "9  NCKAP1  ENST00000360982      291   580.53  0.50127  0.000029          7483   \n",
       "\n",
       "   obs_mis_pphen  exp_mis_pphen  oe_mis_pphen  possible_mis_pphen  obs_syn  \\\n",
       "0          314.0         529.75       0.59273              6708.0      422   \n",
       "1          158.0         543.10       0.29092              7135.0      496   \n",
       "2           21.0         182.52       0.11506              2197.0      215   \n",
       "3           51.0         290.68       0.17545              3560.0      470   \n",
       "4          107.0         321.14       0.33319              4151.0      358   \n",
       "5          130.0         264.36       0.49176              3521.0      298   \n",
       "6           75.0         160.48       0.46735              1794.0      274   \n",
       "7           82.0         312.02       0.26280              5365.0      272   \n",
       "8          206.0         310.49       0.66348              3722.0      259   \n",
       "9           56.0         154.07       0.36346              2017.0      194   \n",
       "\n",
       "   exp_syn   oe_syn    mu_syn  possible_syn  obs_lof    mu_lof  possible_lof  \\\n",
       "0   387.53  1.08900  0.000019          4248      0.0  0.000005        1257.0   \n",
       "1   495.01  1.00200  0.000025          5211      1.0  0.000009        1781.0   \n",
       "2   203.25  1.05780  0.000010          2091      0.0  0.000005         937.0   \n",
       "3   456.03  1.03060  0.000024          4564      1.0  0.000007        1440.0   \n",
       "4   352.62  1.01530  0.000017          3482      0.0  0.000004        1024.0   \n",
       "5   272.69  1.09280  0.000013          3026      0.0  0.000005         862.0   \n",
       "6   265.93  1.03040  0.000017          2360      0.0  0.000004         805.0   \n",
       "7   250.43  1.08610  0.000018          3522      0.0  0.000005        1203.0   \n",
       "8   265.17  0.97673  0.000013          2603      0.0  0.000004         846.0   \n",
       "9   197.96  0.98000  0.000010          2051      0.0  0.000004         850.0   \n",
       "\n",
       "   exp_lof  pLI         pNull          pRec    oe_lof  oe_syn_lower  \\\n",
       "0   98.429  1.0  8.943600e-40  1.838300e-16  0.000000         1.005   \n",
       "1  150.320  1.0  2.977300e-59  3.572400e-24  0.006653         0.930   \n",
       "2   79.490  1.0  2.785300e-32  2.191400e-13  0.000000         0.946   \n",
       "3  125.030  1.0  2.992400e-49  4.562900e-20  0.007998         0.955   \n",
       "4   73.222  1.0  8.405500e-30  2.284200e-12  0.000000         0.930   \n",
       "5   74.559  1.0  2.487200e-30  1.385500e-12  0.000000         0.993   \n",
       "6   69.811  1.0  1.879200e-28  8.179200e-12  0.000000         0.933   \n",
       "7   68.906  1.0  4.288100e-28  1.147700e-11  0.000000         0.983   \n",
       "8   67.830  1.0  1.142700e-27  1.716200e-11  0.000000         0.882   \n",
       "9   67.231  1.0  1.972600e-27  2.147300e-11  0.000000         0.871   \n",
       "\n",
       "   oe_syn_upper  oe_mis_lower  oe_mis_upper  oe_lof_lower  oe_lof_upper  \\\n",
       "0         1.180         0.736         0.824         0.000         0.030   \n",
       "1         1.079         0.554         0.621         0.001         0.032   \n",
       "2         1.184         0.249         0.320         0.000         0.037   \n",
       "3         1.112         0.403         0.464         0.002         0.038   \n",
       "4         1.108         0.645         0.733         0.000         0.040   \n",
       "5         1.203         0.683         0.785         0.000         0.040   \n",
       "6         1.139         0.744         0.857         0.000         0.042   \n",
       "7         1.201         0.397         0.477         0.000         0.043   \n",
       "8         1.083         0.766         0.876         0.000         0.044   \n",
       "9         1.104         0.455         0.552         0.000         0.044   \n",
       "\n",
       "  constraint_flag     syn_z   mis_z    lof_z  oe_lof_upper_rank  \\\n",
       "0             NaN -1.376500  2.6232   9.1935                0.0   \n",
       "1             NaN -0.035119  5.5737  11.2860                1.0   \n",
       "2             NaN -0.647760  6.3999   8.2618                2.0   \n",
       "3             NaN -0.514100  7.2546  10.2790                3.0   \n",
       "4             NaN -0.225180  3.4620   7.9294                4.0   \n",
       "5             NaN -1.205000  2.6592   8.0014                5.0   \n",
       "6             NaN -0.389080  1.8745   7.7425                6.0   \n",
       "7             NaN -1.071200  5.4860   7.6921                7.0   \n",
       "8             NaN  0.297860  1.7661   7.6318                8.0   \n",
       "9             NaN  0.221170  4.2701   7.5981                9.0   \n",
       "\n",
       "   oe_lof_upper_bin  oe_lof_upper_bin_6  n_sites  classic_caf    max_af  \\\n",
       "0               0.0                 0.0      2.0     0.000012  0.000008   \n",
       "1               0.0                 0.0      2.0     0.000012  0.000008   \n",
       "2               0.0                 0.0      8.0     0.000032  0.000004   \n",
       "3               0.0                 0.0      5.0     0.000020  0.000004   \n",
       "4               0.0                 0.0      1.0     0.000004  0.000004   \n",
       "5               0.0                 0.0      4.0     0.000016  0.000004   \n",
       "6               0.0                 0.0      2.0     0.000008  0.000004   \n",
       "7               0.0                 0.0      2.0     0.000093  0.000087   \n",
       "8               0.0                 0.0      0.0     0.000000  0.000000   \n",
       "9               0.0                 0.0      0.0     0.000000  0.000000   \n",
       "\n",
       "    no_lofs  obs_het_lof  obs_hom_lof   defined         p  exp_hom_lof  \\\n",
       "0  124782.0          3.0          0.0  124785.0  0.000012     0.000018   \n",
       "1  125693.0          3.0          0.0  125696.0  0.000012     0.000018   \n",
       "2  125731.0          8.0          0.0  125739.0  0.000032     0.000127   \n",
       "3  125740.0          4.0          0.0  125744.0  0.000016     0.000032   \n",
       "4  125122.0          1.0          0.0  125123.0  0.000004     0.000002   \n",
       "5  124625.0          3.0          0.0  124628.0  0.000012     0.000018   \n",
       "6  125745.0          2.0          0.0  125747.0  0.000008     0.000008   \n",
       "7  125612.0         14.0          3.0  125629.0  0.000068     0.000575   \n",
       "8       0.0          0.0          0.0       0.0  0.000000     0.000000   \n",
       "9       0.0          0.0          0.0       0.0  0.000000     0.000000   \n",
       "\n",
       "   classic_caf_afr  classic_caf_amr  classic_caf_asj  classic_caf_eas  \\\n",
       "0         0.000000         0.000000         0.000000         0.000000   \n",
       "1         0.000000         0.000000         0.000099         0.000000   \n",
       "2         0.000000         0.000000         0.000099         0.000054   \n",
       "3         0.000000         0.000029         0.000000         0.000055   \n",
       "4         0.000062         0.000000         0.000000         0.000000   \n",
       "5         0.000000         0.000000         0.000000         0.000111   \n",
       "6         0.000000         0.000029         0.000000         0.000000   \n",
       "7         0.000000         0.000000         0.000535         0.000794   \n",
       "8         0.000000         0.000000         0.000000         0.000000   \n",
       "9         0.000000         0.000000         0.000000         0.000000   \n",
       "\n",
       "   classic_caf_fin  classic_caf_nfe  classic_caf_oth  classic_caf_sas  \\\n",
       "0         0.000093         0.000009              0.0         0.000000   \n",
       "1         0.000000         0.000000              0.0         0.000065   \n",
       "2         0.000000         0.000044              0.0         0.000033   \n",
       "3         0.000000         0.000026              0.0         0.000000   \n",
       "4         0.000000         0.000000              0.0         0.000000   \n",
       "5         0.000000         0.000018              0.0         0.000000   \n",
       "6         0.000000         0.000009              0.0         0.000000   \n",
       "7         0.000000         0.000012              0.0         0.000052   \n",
       "8         0.000000         0.000000              0.0         0.000000   \n",
       "9         0.000000         0.000000              0.0         0.000000   \n",
       "\n",
       "      p_afr     p_amr     p_asj     p_eas     p_fin     p_nfe  p_oth  \\\n",
       "0  0.000000  0.000000  0.000000  0.000000  0.000093  0.000009    0.0   \n",
       "1  0.000000  0.000000  0.000099  0.000000  0.000000  0.000000    0.0   \n",
       "2  0.000000  0.000000  0.000099  0.000054  0.000000  0.000044    0.0   \n",
       "3  0.000000  0.000029  0.000000  0.000054  0.000000  0.000018    0.0   \n",
       "4  0.000062  0.000000  0.000000  0.000000  0.000000  0.000000    0.0   \n",
       "5  0.000000  0.000000  0.000000  0.000056  0.000000  0.000018    0.0   \n",
       "6  0.000000  0.000029  0.000000  0.000000  0.000000  0.000009    0.0   \n",
       "7  0.000000  0.000000  0.000397  0.000598  0.000000  0.000009    0.0   \n",
       "8  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000    0.0   \n",
       "9  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000    0.0   \n",
       "\n",
       "      p_sas transcript_type          gene_id  transcript_level  cds_length  \\\n",
       "0  0.000000  protein_coding  ENSG00000108510                 2        6522   \n",
       "1  0.000065  protein_coding  ENSG00000164190                 2        8412   \n",
       "2  0.000033  protein_coding  ENSG00000108055                 2        3651   \n",
       "3  0.000000  protein_coding  ENSG00000125107                 2        7128   \n",
       "4  0.000000  protein_coding  ENSG00000117000                 2        5742   \n",
       "5  0.000000  protein_coding  ENSG00000165494                 2        4665   \n",
       "6  0.000000  protein_coding  ENSG00000075420                 2        3612   \n",
       "7  0.000033  protein_coding  ENSG00000147133                 2        5679   \n",
       "8  0.000000  protein_coding  ENSG00000048649                 2        4323   \n",
       "9  0.000000  protein_coding  ENSG00000061676                 2        3402   \n",
       "\n",
       "   num_coding_exons       gene_type  gene_length  exac_pLI  exac_obs_lof  \\\n",
       "0                30  protein_coding       122678       1.0           0.0   \n",
       "1                46  protein_coding       189655       1.0           1.0   \n",
       "2                29  protein_coding        36946       1.0           0.0   \n",
       "3                48  protein_coding       109936       1.0           3.0   \n",
       "4                 8  protein_coding        79549       1.0           0.0   \n",
       "5                16  protein_coding        30464       1.0           1.0   \n",
       "6                25  protein_coding       362038       1.0           0.0   \n",
       "7                38  protein_coding       166111       1.0           2.0   \n",
       "8                16  protein_coding       161023       1.0           0.0   \n",
       "9                32  protein_coding       129744       1.0           0.0   \n",
       "\n",
       "   exac_exp_lof  exac_oe_lof  brain_expression chromosome  start_position  \\\n",
       "0        64.393     0.000000               NaN         17        60019966   \n",
       "1       110.570     0.009044               NaN          5        36876861   \n",
       "2        58.523     0.000000               NaN         10       112327449   \n",
       "3        90.130     0.033285               NaN         16        58553855   \n",
       "4        43.607     0.000000               NaN          1        40627045   \n",
       "5        48.160     0.020764               NaN         11        82868030   \n",
       "6        49.447     0.000000               NaN          3       171757418   \n",
       "7        50.800     0.039370               NaN          X        70586114   \n",
       "8        46.996     0.000000               NaN         11        77371041   \n",
       "9        50.723     0.000000               NaN          2       183773843   \n",
       "\n",
       "   end_position  \n",
       "0      60142643  \n",
       "1      37066515  \n",
       "2     112364394  \n",
       "3      58663790  \n",
       "4      40706593  \n",
       "5      82898493  \n",
       "6     172119455  \n",
       "7      70752224  \n",
       "8      77532063  \n",
       "9     183903586  "
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "constraint = pd.read_csv('../../data/annotations/gnomad.v2.1.1.lof_metrics.by_gene.txt', header = 0, sep = '\\t')\n",
    "constraint.head(10)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "28451d46",
   "metadata": {},
   "source": [
    "Now add the missense and loss-of-function z-scores to the main dataframe."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "60a8b4bb",
   "metadata": {},
   "outputs": [],
   "source": [
    "mis_oe = pd.Series(constraint.oe_mis.values, index = constraint.gene).to_dict()\n",
    "lof_oe = pd.Series(constraint.oe_lof.values, index = constraint.gene).to_dict()\n",
    "mis_z = pd.Series(constraint.mis_z.values, index = constraint.gene).to_dict()\n",
    "lof_z = pd.Series(constraint.lof_z.values, index = constraint.gene).to_dict()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "a0428760",
   "metadata": {},
   "outputs": [],
   "source": [
    "data['mis_oe'] = data['annotation'].map(mis_oe)\n",
    "data['lof_oe'] = data['annotation'].map(lof_oe)\n",
    "data['mis_z'] = data['annotation'].map(mis_z)\n",
    "data['lof_z'] = data['annotation'].map(lof_z)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "51f46873",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref_allele</th>\n",
       "      <th>altai_gt</th>\n",
       "      <th>chagyrskaya_gt</th>\n",
       "      <th>denisovan_gt</th>\n",
       "      <th>vindija_gt</th>\n",
       "      <th>alt_allele</th>\n",
       "      <th>annotation</th>\n",
       "      <th>ag_delta</th>\n",
       "      <th>al_delta</th>\n",
       "      <th>dg_delta</th>\n",
       "      <th>dl_delta</th>\n",
       "      <th>delta_max</th>\n",
       "      <th>ag_pos</th>\n",
       "      <th>al_pos</th>\n",
       "      <th>dg_pos</th>\n",
       "      <th>dl_pos</th>\n",
       "      <th>variant_type</th>\n",
       "      <th>altai_gt_boolean</th>\n",
       "      <th>chagyrskaya_gt_boolean</th>\n",
       "      <th>denisovan_gt_boolean</th>\n",
       "      <th>vindija_gt_boolean</th>\n",
       "      <th>distribution</th>\n",
       "      <th>mis_oe</th>\n",
       "      <th>lof_oe</th>\n",
       "      <th>mis_z</th>\n",
       "      <th>lof_z</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>739130</td>\n",
       "      <td>T</td>\n",
       "      <td>0/0</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>0/1</td>\n",
       "      <td>TA</td>\n",
       "      <td>AL669831.1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-7</td>\n",
       "      <td>-36</td>\n",
       "      <td>-35</td>\n",
       "      <td>-31</td>\n",
       "      <td>indel</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>Vindija</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>739130</td>\n",
       "      <td>T</td>\n",
       "      <td>0/1</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>0/0</td>\n",
       "      <td>TAA</td>\n",
       "      <td>AL669831.1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-31</td>\n",
       "      <td>-36</td>\n",
       "      <td>-7</td>\n",
       "      <td>-31</td>\n",
       "      <td>indel</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>Altai</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1169649</th>\n",
       "      <td>chr1</td>\n",
       "      <td>861808</td>\n",
       "      <td>A</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>G</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.01</td>\n",
       "      <td>-29</td>\n",
       "      <td>-27</td>\n",
       "      <td>24</td>\n",
       "      <td>-20</td>\n",
       "      <td>snv</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>1.5082</td>\n",
       "      <td>0.89656</td>\n",
       "      <td>-3.43610</td>\n",
       "      <td>0.47484</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1169650</th>\n",
       "      <td>chr1</td>\n",
       "      <td>861808</td>\n",
       "      <td>A</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>G</td>\n",
       "      <td>AL645608.1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>48</td>\n",
       "      <td>-14</td>\n",
       "      <td>-46</td>\n",
       "      <td>-14</td>\n",
       "      <td>snv</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>1.2217</td>\n",
       "      <td>0.73515</td>\n",
       "      <td>-0.64548</td>\n",
       "      <td>0.49579</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1169651</th>\n",
       "      <td>chr1</td>\n",
       "      <td>862072</td>\n",
       "      <td>C</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>T</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>3</td>\n",
       "      <td>28</td>\n",
       "      <td>-38</td>\n",
       "      <td>41</td>\n",
       "      <td>snv</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>1.5082</td>\n",
       "      <td>0.89656</td>\n",
       "      <td>-3.43610</td>\n",
       "      <td>0.47484</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1169652</th>\n",
       "      <td>chr1</td>\n",
       "      <td>862072</td>\n",
       "      <td>C</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>T</td>\n",
       "      <td>AL645608.1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-6</td>\n",
       "      <td>-50</td>\n",
       "      <td>24</td>\n",
       "      <td>-48</td>\n",
       "      <td>snv</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>1.2217</td>\n",
       "      <td>0.73515</td>\n",
       "      <td>-0.64548</td>\n",
       "      <td>0.49579</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1169653</th>\n",
       "      <td>chr1</td>\n",
       "      <td>862093</td>\n",
       "      <td>T</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>C</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>7</td>\n",
       "      <td>14</td>\n",
       "      <td>-15</td>\n",
       "      <td>34</td>\n",
       "      <td>snv</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>1.5082</td>\n",
       "      <td>0.89656</td>\n",
       "      <td>-3.43610</td>\n",
       "      <td>0.47484</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1169654</th>\n",
       "      <td>chr1</td>\n",
       "      <td>862093</td>\n",
       "      <td>T</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>C</td>\n",
       "      <td>AL645608.1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-34</td>\n",
       "      <td>48</td>\n",
       "      <td>49</td>\n",
       "      <td>3</td>\n",
       "      <td>snv</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>1.2217</td>\n",
       "      <td>0.73515</td>\n",
       "      <td>-0.64548</td>\n",
       "      <td>0.49579</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1169655</th>\n",
       "      <td>chr1</td>\n",
       "      <td>862124</td>\n",
       "      <td>A</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>G</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-24</td>\n",
       "      <td>26</td>\n",
       "      <td>-11</td>\n",
       "      <td>3</td>\n",
       "      <td>snv</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>1.5082</td>\n",
       "      <td>0.89656</td>\n",
       "      <td>-3.43610</td>\n",
       "      <td>0.47484</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1169656</th>\n",
       "      <td>chr1</td>\n",
       "      <td>862124</td>\n",
       "      <td>A</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>G</td>\n",
       "      <td>AL645608.1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>17</td>\n",
       "      <td>-14</td>\n",
       "      <td>-28</td>\n",
       "      <td>35</td>\n",
       "      <td>snv</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>1.2217</td>\n",
       "      <td>0.73515</td>\n",
       "      <td>-0.64548</td>\n",
       "      <td>0.49579</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        chrom     pos ref_allele altai_gt chagyrskaya_gt denisovan_gt  \\\n",
       "0        chr1  739130          T      0/0            ./.          ./.   \n",
       "1        chr1  739130          T      0/1            ./.          ./.   \n",
       "1169649  chr1  861808          A      0/0            0/0          1/1   \n",
       "1169650  chr1  861808          A      0/0            0/0          1/1   \n",
       "1169651  chr1  862072          C      1/1            1/1          0/0   \n",
       "1169652  chr1  862072          C      1/1            1/1          0/0   \n",
       "1169653  chr1  862093          T      0/0            0/0          1/1   \n",
       "1169654  chr1  862093          T      0/0            0/0          1/1   \n",
       "1169655  chr1  862124          A      0/0            0/0          1/1   \n",
       "1169656  chr1  862124          A      0/0            0/0          1/1   \n",
       "\n",
       "        vindija_gt alt_allele  annotation  ag_delta  al_delta  dg_delta  \\\n",
       "0              0/1         TA  AL669831.1       0.0       0.0      0.00   \n",
       "1              0/0        TAA  AL669831.1       0.0       0.0      0.00   \n",
       "1169649        0/0          G      SAMD11       0.0       0.0      0.01   \n",
       "1169650        0/0          G  AL645608.1       0.0       0.0      0.00   \n",
       "1169651        0/1          T      SAMD11       0.0       0.0      0.00   \n",
       "1169652        0/1          T  AL645608.1       0.0       0.0      0.00   \n",
       "1169653        0/0          C      SAMD11       0.0       0.0      0.00   \n",
       "1169654        0/0          C  AL645608.1       0.0       0.0      0.00   \n",
       "1169655        0/0          G      SAMD11       0.0       0.0      0.00   \n",
       "1169656        0/0          G  AL645608.1       0.0       0.0      0.00   \n",
       "\n",
       "         dl_delta  delta_max  ag_pos  al_pos  dg_pos  dl_pos variant_type  \\\n",
       "0             0.0       0.00      -7     -36     -35     -31        indel   \n",
       "1             0.0       0.00     -31     -36      -7     -31        indel   \n",
       "1169649       0.0       0.01     -29     -27      24     -20          snv   \n",
       "1169650       0.0       0.00      48     -14     -46     -14          snv   \n",
       "1169651       0.0       0.00       3      28     -38      41          snv   \n",
       "1169652       0.0       0.00      -6     -50      24     -48          snv   \n",
       "1169653       0.0       0.00       7      14     -15      34          snv   \n",
       "1169654       0.0       0.00     -34      48      49       3          snv   \n",
       "1169655       0.0       0.00     -24      26     -11       3          snv   \n",
       "1169656       0.0       0.00      17     -14     -28      35          snv   \n",
       "\n",
       "        altai_gt_boolean chagyrskaya_gt_boolean denisovan_gt_boolean  \\\n",
       "0                  FALSE                  FALSE                FALSE   \n",
       "1                   TRUE                  FALSE                FALSE   \n",
       "1169649            FALSE                  FALSE                 TRUE   \n",
       "1169650            FALSE                  FALSE                 TRUE   \n",
       "1169651             TRUE                   TRUE                FALSE   \n",
       "1169652             TRUE                   TRUE                FALSE   \n",
       "1169653            FALSE                  FALSE                 TRUE   \n",
       "1169654            FALSE                  FALSE                 TRUE   \n",
       "1169655            FALSE                  FALSE                 TRUE   \n",
       "1169656            FALSE                  FALSE                 TRUE   \n",
       "\n",
       "        vindija_gt_boolean distribution  mis_oe   lof_oe    mis_z    lof_z  \n",
       "0                     TRUE      Vindija     NaN      NaN      NaN      NaN  \n",
       "1                    FALSE        Altai     NaN      NaN      NaN      NaN  \n",
       "1169649              FALSE    Denisovan  1.5082  0.89656 -3.43610  0.47484  \n",
       "1169650              FALSE    Denisovan  1.2217  0.73515 -0.64548  0.49579  \n",
       "1169651               TRUE  Neanderthal  1.5082  0.89656 -3.43610  0.47484  \n",
       "1169652               TRUE  Neanderthal  1.2217  0.73515 -0.64548  0.49579  \n",
       "1169653              FALSE    Denisovan  1.5082  0.89656 -3.43610  0.47484  \n",
       "1169654              FALSE    Denisovan  1.2217  0.73515 -0.64548  0.49579  \n",
       "1169655              FALSE    Denisovan  1.5082  0.89656 -3.43610  0.47484  \n",
       "1169656              FALSE    Denisovan  1.2217  0.73515 -0.64548  0.49579  "
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "b0e21495",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2107054"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(data)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ae92ba80",
   "metadata": {},
   "source": [
    "Let's add per variant phyloP data."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "3bf978c8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>pos</th>\n",
       "      <th>phyloP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>739129</td>\n",
       "      <td>739130</td>\n",
       "      <td>0.155</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>739129</td>\n",
       "      <td>739130</td>\n",
       "      <td>0.155</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>861807</td>\n",
       "      <td>861808</td>\n",
       "      <td>-0.683</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>861807</td>\n",
       "      <td>861808</td>\n",
       "      <td>-0.683</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>862071</td>\n",
       "      <td>862072</td>\n",
       "      <td>0.197</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>862071</td>\n",
       "      <td>862072</td>\n",
       "      <td>0.197</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr1</td>\n",
       "      <td>862092</td>\n",
       "      <td>862093</td>\n",
       "      <td>-1.042</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr1</td>\n",
       "      <td>862092</td>\n",
       "      <td>862093</td>\n",
       "      <td>-1.042</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr1</td>\n",
       "      <td>862123</td>\n",
       "      <td>862124</td>\n",
       "      <td>-3.750</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr1</td>\n",
       "      <td>862123</td>\n",
       "      <td>862124</td>\n",
       "      <td>-3.750</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chrom   start     pos  phyloP\n",
       "0  chr1  739129  739130   0.155\n",
       "1  chr1  739129  739130   0.155\n",
       "2  chr1  861807  861808  -0.683\n",
       "3  chr1  861807  861808  -0.683\n",
       "4  chr1  862071  862072   0.197\n",
       "5  chr1  862071  862072   0.197\n",
       "6  chr1  862092  862093  -1.042\n",
       "7  chr1  862092  862093  -1.042\n",
       "8  chr1  862123  862124  -3.750\n",
       "9  chr1  862123  862124  -3.750"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "variant_phyloP_header = ['chrom','start','pos','phyloP']\n",
    "variant_phyloP = pd.read_csv('../../data/annotations/all_variant_sites_hg19_phyloP.bed', sep = '\\t', names = variant_phyloP_header)\n",
    "variant_phyloP.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "2acdf33e",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref_allele</th>\n",
       "      <th>altai_gt</th>\n",
       "      <th>chagyrskaya_gt</th>\n",
       "      <th>denisovan_gt</th>\n",
       "      <th>vindija_gt</th>\n",
       "      <th>alt_allele</th>\n",
       "      <th>annotation</th>\n",
       "      <th>ag_delta</th>\n",
       "      <th>al_delta</th>\n",
       "      <th>dg_delta</th>\n",
       "      <th>dl_delta</th>\n",
       "      <th>delta_max</th>\n",
       "      <th>ag_pos</th>\n",
       "      <th>al_pos</th>\n",
       "      <th>dg_pos</th>\n",
       "      <th>dl_pos</th>\n",
       "      <th>variant_type</th>\n",
       "      <th>altai_gt_boolean</th>\n",
       "      <th>chagyrskaya_gt_boolean</th>\n",
       "      <th>denisovan_gt_boolean</th>\n",
       "      <th>vindija_gt_boolean</th>\n",
       "      <th>distribution</th>\n",
       "      <th>mis_oe</th>\n",
       "      <th>lof_oe</th>\n",
       "      <th>mis_z</th>\n",
       "      <th>lof_z</th>\n",
       "      <th>phyloP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>739130</td>\n",
       "      <td>T</td>\n",
       "      <td>0/0</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>0/1</td>\n",
       "      <td>TA</td>\n",
       "      <td>AL669831.1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-7</td>\n",
       "      <td>-36</td>\n",
       "      <td>-35</td>\n",
       "      <td>-31</td>\n",
       "      <td>indel</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>Vindija</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.155</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>739130</td>\n",
       "      <td>T</td>\n",
       "      <td>0/0</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>0/1</td>\n",
       "      <td>TA</td>\n",
       "      <td>AL669831.1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-7</td>\n",
       "      <td>-36</td>\n",
       "      <td>-35</td>\n",
       "      <td>-31</td>\n",
       "      <td>indel</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>Vindija</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.155</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>739130</td>\n",
       "      <td>T</td>\n",
       "      <td>0/1</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>0/0</td>\n",
       "      <td>TAA</td>\n",
       "      <td>AL669831.1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-31</td>\n",
       "      <td>-36</td>\n",
       "      <td>-7</td>\n",
       "      <td>-31</td>\n",
       "      <td>indel</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>Altai</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.155</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>739130</td>\n",
       "      <td>T</td>\n",
       "      <td>0/1</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>0/0</td>\n",
       "      <td>TAA</td>\n",
       "      <td>AL669831.1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-31</td>\n",
       "      <td>-36</td>\n",
       "      <td>-7</td>\n",
       "      <td>-31</td>\n",
       "      <td>indel</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>Altai</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.155</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>861808</td>\n",
       "      <td>A</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>G</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.01</td>\n",
       "      <td>-29</td>\n",
       "      <td>-27</td>\n",
       "      <td>24</td>\n",
       "      <td>-20</td>\n",
       "      <td>snv</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>1.5082</td>\n",
       "      <td>0.89656</td>\n",
       "      <td>-3.43610</td>\n",
       "      <td>0.47484</td>\n",
       "      <td>-0.683</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>861808</td>\n",
       "      <td>A</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>G</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.01</td>\n",
       "      <td>-29</td>\n",
       "      <td>-27</td>\n",
       "      <td>24</td>\n",
       "      <td>-20</td>\n",
       "      <td>snv</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>1.5082</td>\n",
       "      <td>0.89656</td>\n",
       "      <td>-3.43610</td>\n",
       "      <td>0.47484</td>\n",
       "      <td>-0.683</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr1</td>\n",
       "      <td>861808</td>\n",
       "      <td>A</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>G</td>\n",
       "      <td>AL645608.1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>48</td>\n",
       "      <td>-14</td>\n",
       "      <td>-46</td>\n",
       "      <td>-14</td>\n",
       "      <td>snv</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>1.2217</td>\n",
       "      <td>0.73515</td>\n",
       "      <td>-0.64548</td>\n",
       "      <td>0.49579</td>\n",
       "      <td>-0.683</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr1</td>\n",
       "      <td>861808</td>\n",
       "      <td>A</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>G</td>\n",
       "      <td>AL645608.1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>48</td>\n",
       "      <td>-14</td>\n",
       "      <td>-46</td>\n",
       "      <td>-14</td>\n",
       "      <td>snv</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>1.2217</td>\n",
       "      <td>0.73515</td>\n",
       "      <td>-0.64548</td>\n",
       "      <td>0.49579</td>\n",
       "      <td>-0.683</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr1</td>\n",
       "      <td>862072</td>\n",
       "      <td>C</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>T</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>3</td>\n",
       "      <td>28</td>\n",
       "      <td>-38</td>\n",
       "      <td>41</td>\n",
       "      <td>snv</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>1.5082</td>\n",
       "      <td>0.89656</td>\n",
       "      <td>-3.43610</td>\n",
       "      <td>0.47484</td>\n",
       "      <td>0.197</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr1</td>\n",
       "      <td>862072</td>\n",
       "      <td>C</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>T</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>3</td>\n",
       "      <td>28</td>\n",
       "      <td>-38</td>\n",
       "      <td>41</td>\n",
       "      <td>snv</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>1.5082</td>\n",
       "      <td>0.89656</td>\n",
       "      <td>-3.43610</td>\n",
       "      <td>0.47484</td>\n",
       "      <td>0.197</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chrom     pos ref_allele altai_gt chagyrskaya_gt denisovan_gt vindija_gt  \\\n",
       "0  chr1  739130          T      0/0            ./.          ./.        0/1   \n",
       "1  chr1  739130          T      0/0            ./.          ./.        0/1   \n",
       "2  chr1  739130          T      0/1            ./.          ./.        0/0   \n",
       "3  chr1  739130          T      0/1            ./.          ./.        0/0   \n",
       "4  chr1  861808          A      0/0            0/0          1/1        0/0   \n",
       "5  chr1  861808          A      0/0            0/0          1/1        0/0   \n",
       "6  chr1  861808          A      0/0            0/0          1/1        0/0   \n",
       "7  chr1  861808          A      0/0            0/0          1/1        0/0   \n",
       "8  chr1  862072          C      1/1            1/1          0/0        0/1   \n",
       "9  chr1  862072          C      1/1            1/1          0/0        0/1   \n",
       "\n",
       "  alt_allele  annotation  ag_delta  al_delta  dg_delta  dl_delta  delta_max  \\\n",
       "0         TA  AL669831.1       0.0       0.0      0.00       0.0       0.00   \n",
       "1         TA  AL669831.1       0.0       0.0      0.00       0.0       0.00   \n",
       "2        TAA  AL669831.1       0.0       0.0      0.00       0.0       0.00   \n",
       "3        TAA  AL669831.1       0.0       0.0      0.00       0.0       0.00   \n",
       "4          G      SAMD11       0.0       0.0      0.01       0.0       0.01   \n",
       "5          G      SAMD11       0.0       0.0      0.01       0.0       0.01   \n",
       "6          G  AL645608.1       0.0       0.0      0.00       0.0       0.00   \n",
       "7          G  AL645608.1       0.0       0.0      0.00       0.0       0.00   \n",
       "8          T      SAMD11       0.0       0.0      0.00       0.0       0.00   \n",
       "9          T      SAMD11       0.0       0.0      0.00       0.0       0.00   \n",
       "\n",
       "   ag_pos  al_pos  dg_pos  dl_pos variant_type altai_gt_boolean  \\\n",
       "0      -7     -36     -35     -31        indel            FALSE   \n",
       "1      -7     -36     -35     -31        indel            FALSE   \n",
       "2     -31     -36      -7     -31        indel             TRUE   \n",
       "3     -31     -36      -7     -31        indel             TRUE   \n",
       "4     -29     -27      24     -20          snv            FALSE   \n",
       "5     -29     -27      24     -20          snv            FALSE   \n",
       "6      48     -14     -46     -14          snv            FALSE   \n",
       "7      48     -14     -46     -14          snv            FALSE   \n",
       "8       3      28     -38      41          snv             TRUE   \n",
       "9       3      28     -38      41          snv             TRUE   \n",
       "\n",
       "  chagyrskaya_gt_boolean denisovan_gt_boolean vindija_gt_boolean distribution  \\\n",
       "0                  FALSE                FALSE               TRUE      Vindija   \n",
       "1                  FALSE                FALSE               TRUE      Vindija   \n",
       "2                  FALSE                FALSE              FALSE        Altai   \n",
       "3                  FALSE                FALSE              FALSE        Altai   \n",
       "4                  FALSE                 TRUE              FALSE    Denisovan   \n",
       "5                  FALSE                 TRUE              FALSE    Denisovan   \n",
       "6                  FALSE                 TRUE              FALSE    Denisovan   \n",
       "7                  FALSE                 TRUE              FALSE    Denisovan   \n",
       "8                   TRUE                FALSE               TRUE  Neanderthal   \n",
       "9                   TRUE                FALSE               TRUE  Neanderthal   \n",
       "\n",
       "   mis_oe   lof_oe    mis_z    lof_z  phyloP  \n",
       "0     NaN      NaN      NaN      NaN   0.155  \n",
       "1     NaN      NaN      NaN      NaN   0.155  \n",
       "2     NaN      NaN      NaN      NaN   0.155  \n",
       "3     NaN      NaN      NaN      NaN   0.155  \n",
       "4  1.5082  0.89656 -3.43610  0.47484  -0.683  \n",
       "5  1.5082  0.89656 -3.43610  0.47484  -0.683  \n",
       "6  1.2217  0.73515 -0.64548  0.49579  -0.683  \n",
       "7  1.2217  0.73515 -0.64548  0.49579  -0.683  \n",
       "8  1.5082  0.89656 -3.43610  0.47484   0.197  \n",
       "9  1.5082  0.89656 -3.43610  0.47484   0.197  "
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data = pd.merge(data, variant_phyloP[['chrom','pos','phyloP']], on = ['chrom','pos'], how = 'left', indicator = False)\n",
    "data.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "cda6cd6e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2358420"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "de6e957e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2107041"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data = data.drop_duplicates()\n",
    "len(data)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e2a3480c",
   "metadata": {},
   "source": [
    "Let's reorder the dataframe a bit to group similar columns."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "acf86a5f",
   "metadata": {},
   "outputs": [],
   "source": [
    "data = data[['chrom','pos','ref_allele','alt_allele','variant_type','altai_gt','chagyrskaya_gt','denisovan_gt','vindija_gt','altai_gt_boolean','chagyrskaya_gt_boolean','denisovan_gt_boolean','vindija_gt_boolean','distribution','annotation','mis_oe','mis_z','lof_oe','lof_z','phyloP','ag_delta','al_delta','dg_delta','dl_delta','delta_max','ag_pos','al_pos','dg_pos','dl_pos']]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c4daba1f",
   "metadata": {},
   "source": [
    "Now let's export this dataframe as we will use it again in downstream analyses."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "23c87420",
   "metadata": {},
   "outputs": [],
   "source": [
    "data.to_csv('../../data/dataframes/archaic_data_with_constraint.txt', sep = '\\t', header = True, index = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "52930961",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
