{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "52492749",
   "metadata": {},
   "source": [
    "# Analysis"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "59f168ff",
   "metadata": {},
   "source": [
    "This notebook contains code to analyze putative splicing altering variants in four archaic hominin genomes. The initial code \"picks up\" after the previous notebook. If you wish to run the analyses below using the full dataframe, load the libraries, navigate to your desired directory, and load the full dataframe starting [here](#startwithdataframe).\n",
    "\n",
    "I have organized the analyses that follow to largely reflect the order in which they appear in the manuscript. Feel free to navigate to a specific section using the Table of Contents below."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "18953edd",
   "metadata": {},
   "source": [
    "# Table of Contents\n",
    "- [Load Dataframe](#loaddataframe)\n",
    "- [Load sQTLs](#loadsqtls)\n",
    "- [Spliceosome Variants](#spliceosomevariants)\n",
    "- [Data Description](#datadescription)\n",
    "    - [Multiple Annotations](#multipleannotations)\n",
    "    - [Multiple Alleles](#multiplealleles)\n",
    "    - [Genotypes](#genotypes)\n",
    "    - [Annotations](#annotations)\n",
    "    - [Delta Thresholds](#deltathresholds)\n",
    "    - [Variant Distribution](#variantdistribution)\n",
    "    - [Delta Correlations](#deltacorrelations)\n",
    "    - [Multiple Deltas](#multipledeltas)\n",
    "    - [SAV Genotypes](#SAVgenotypes)\n",
    "    - [1KG vs gnomAD SAVs](#1KGgnomADSAVs)\n",
    "    - [SAV Allele Origin](#SAValleleorigin)\n",
    "    - [Introgression Set Overlap](#introgressionsetoverlap)\n",
    "    - [SAV Distribution](#SAVdistribution)\n",
    "- [SAV Genes](#SAVEgenes)\n",
    "    - [Top 20 AG, AL, DG, and DL](#top20)\n",
    "    - [N Genes](#ngenes)\n",
    "    - [N Genes Distribution](#ngenesdistribution)\n",
    "    - [N Genes by Origin](#ngenesbyorigin)\n",
    "    - [N Genes by Distribution](#ngenesbydistribution)\n",
    "    - [Archaic-Specific Genes](#archaicspecificgenes)\n",
    "    - [Gene Overlap](#geneoverlap)\n",
    "- [Comparisons](#comparions)\n",
    "    - [SAV Genes and DR genes](#drgenes)\n",
    "    - [SAV Genes and Circadian Genes](#circadiangenes)\n",
    "- [Phenotype Enrichment](#phenotypeenrichment)\n",
    "- [Novel Isoforms](#novelisoforms)\n",
    "- [Gene Characteristics](#genecharacteristics)\n",
    "    - [N Exons, CDS Length, and Gene Length](#physical)\n",
    "    - [N Isoforms](#isoforms)\n",
    "    - [Constraint and Conservation](#constraintconservation)\n",
    "- [Purifying Selection](#purifyingselection)\n",
    "    - [Varied Deltas](#varieddeltas)\n",
    "    - [Lineage-Specific](#lineagespecific)\n",
    "- [SAVs in Moderns](#SAVsinmoderns)\n",
    "    - [Introgressed SAV Distribution](#introgressedSAVdistribution)\n",
    "    - [Allele Frequency and Max Delta](#allelefrequencymaxdelta)\n",
    "    - [Introgressed Genes](#introgressedgenes)\n",
    "    - [ASE](#ASE)\n",
    "- [Gene Expression](#geneexpression)\n",
    "    - [sQTLs](#sqtls)\n",
    "        - [Allele Origin Differences](#sqtlalleleorigindifferences)\n",
    "        - [Tissue Specificity](#tissuespecificity)\n",
    "        - [European sQTLs](#eursqtls)\n",
    "    - [Gene-Level](#genelevel)\n",
    "        - [Allele Origin Differences](#genealleleorigindifferences)\n",
    "- [Genes of Evolutionary Significance](#evolutionarysignificantgenes)\n",
    "- [Average SAVs in 1KG](#avg1KGSAVs)\n",
    "- [Introgressed Reference Alleles](#introgressedrefs)\n",
    "- [Introgressed Fixed Alleles](#introgressedfixed)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1f481600",
   "metadata": {},
   "source": [
    "Load libraries and change directories."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "dae2adc8",
   "metadata": {},
   "outputs": [],
   "source": [
    "from pingouin import partial_corr\n",
    "from math import sqrt\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import re\n",
    "import sys\n",
    "from scipy.stats import chi2\n",
    "from scipy.stats import chi2_contingency\n",
    "from scipy.stats import fisher_exact\n",
    "from scipy.stats import kruskal\n",
    "from scipy.stats import mannwhitneyu\n",
    "from scipy.stats import spearmanr\n",
    "pd.set_option('display.max_columns', 200)\n",
    "pd.set_option('display.max_rows', 110)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "bb4f613d",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/wynton/group/capra/projects/archaic_splicing/data/dataframes\n"
     ]
    }
   ],
   "source": [
    "cd ../../data/dataframes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "d924b60c",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/wynton/home/capra/cbrand/miniconda3/envs/jupyter/lib/python3.10/site-packages/outdated/utils.py:14: OutdatedPackageWarning: The package pingouin is out of date. Your version is 0.5.2, the latest is 0.5.3.\n",
      "Set the environment variable OUTDATED_IGNORE=1 to disable these warnings.\n",
      "  return warn(\n",
      "/wynton/home/capra/cbrand/miniconda3/envs/jupyter/lib/python3.10/site-packages/outdated/utils.py:14: OutdatedPackageWarning: The package outdated is out of date. Your version is 0.2.1, the latest is 0.2.2.\n",
      "Set the environment variable OUTDATED_IGNORE=1 to disable these warnings.\n",
      "  return warn(\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "2.2250738585072014e-308"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sys.float_info.min"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "51abbe87",
   "metadata": {},
   "source": [
    "# Load Dataframe <a class = 'anchor' id = 'loaddataframe'></a>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "fde0b621",
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "#data = pd.read_csv('archaic_data_with_constraint_moderns_introgression.txt', sep='\\t', header=0)\n",
    "#data.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "6738e5d2",
   "metadata": {},
   "outputs": [],
   "source": [
    "#len(data)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4c3b6991",
   "metadata": {},
   "source": [
    "# Load sQTLs <a class = 'anchor' id = 'loadsqtls'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2b966fa5",
   "metadata": {},
   "source": [
    "Let's immediately add our sQTLs and make sure we're only adding sQTLs that match the ref and alt allele. Load the sQTL dataframe."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "6e23f9c5",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "#sQTLs_header = ['chrom','start','pos','sQTL_ref_allele','sQTL_alt_allele','Adipose_Subcutaneous','Adipose_Visceral_Omentum','Adrenal_Gland','Artery_Aorta','Artery_Coronary','Artery_Tibial','Brain_Amygdala','Brain_Anterior_cingulate_cortex_BA24','Brain_Caudate_basal_ganglia','Brain_Cerebellar_Hemisphere','Brain_Cerebellum','Brain_Cortex','Brain_Frontal_Cortex_BA9','Brain_Hippocampus','Brain_Hypothalamus','Brain_Nucleus_accumbens_basal_ganglia','Brain_Putamen_basal_ganglia','Brain_Spinal_cord_cervical_c-1','Brain_Substantia_nigra','Breast_Mammary_Tissue','Cells_Cultured_fibroblasts','Cells_EBV-transformed_lymphocytes','Colon_Sigmoid','Colon_Transverse','Esophagus_Gastroesophageal_Junction','Esophagus_Mucosa','Esophagus_Muscularis','Heart_Atrial_Appendage','Heart_Left_Ventricle','Kidney_Cortex','Liver','Lung','Minor_Salivary_Gland','Muscle_Skeletal','Nerve_Tibial','Ovary','Pancreas','Pituitary','Prostate','Skin_Not_Sun_Exposed_Suprapubic','Skin_Sun_Exposed_Lower_leg','Small_Intestine_Terminal_Ileum','Spleen','Stomach','Testis','Thyroid','Uterus','Vagina','Whole_Blood']\n",
    "#sQTLs = pd.read_csv('../GTEx_sQTLs/sQTLs_hg19.txt', sep = '\\t', names = sQTLs_header)\n",
    "#sQTLs.head(10)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a9e9fd2b",
   "metadata": {},
   "source": [
    "Add a column with the GTEx tissue N per variant."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "cf4d3766",
   "metadata": {},
   "outputs": [],
   "source": [
    "#sQTLs['N_GTEx_tissues'] = sQTLs[['Adipose_Subcutaneous','Adipose_Visceral_Omentum','Adrenal_Gland','Artery_Aorta','Artery_Coronary','Artery_Tibial','Brain_Amygdala','Brain_Anterior_cingulate_cortex_BA24','Brain_Caudate_basal_ganglia','Brain_Cerebellar_Hemisphere','Brain_Cerebellum','Brain_Cortex','Brain_Frontal_Cortex_BA9','Brain_Hippocampus','Brain_Hypothalamus','Brain_Nucleus_accumbens_basal_ganglia','Brain_Putamen_basal_ganglia','Brain_Spinal_cord_cervical_c-1','Brain_Substantia_nigra','Breast_Mammary_Tissue','Cells_Cultured_fibroblasts','Cells_EBV-transformed_lymphocytes','Colon_Sigmoid','Colon_Transverse','Esophagus_Gastroesophageal_Junction','Esophagus_Mucosa','Esophagus_Muscularis','Heart_Atrial_Appendage','Heart_Left_Ventricle','Kidney_Cortex','Liver','Lung','Minor_Salivary_Gland','Muscle_Skeletal','Nerve_Tibial','Ovary','Pancreas','Pituitary','Prostate','Skin_Not_Sun_Exposed_Suprapubic','Skin_Sun_Exposed_Lower_leg','Small_Intestine_Terminal_Ileum','Spleen','Stomach','Testis','Thyroid','Uterus','Vagina','Whole_Blood']].sum(axis=1)\n",
    "#sQTLs.head(10)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "af2f03b0",
   "metadata": {},
   "source": [
    "Merge the data using an indicator."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "942a66c3",
   "metadata": {},
   "outputs": [],
   "source": [
    "#data = pd.merge(data, sQTLs, on = ['chrom','pos'], how = 'left', indicator = True)\n",
    "#data = data.drop(columns = ['start'])\n",
    "#data.head(10)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c998e180",
   "metadata": {},
   "source": [
    "Classify variants as an sQTL or not depending on whether the position and ref/alt alleles match."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "b0c43e87",
   "metadata": {},
   "outputs": [],
   "source": [
    "#def sQTL(data):\n",
    "#    if (data['_merge'] == 'left_only'):\n",
    "#        return 'no'\n",
    "#    elif (data['_merge'] == 'both') & (data['ref_allele'] != data['sQTL_ref_allele']) & (data['alt_allele'] != data['sQTL_alt_allele']):\n",
    "#        return 'no'\n",
    "#    elif (data['_merge'] == 'both') & (data['ref_allele'] != data['sQTL_ref_allele']) & (data['alt_allele'] == data['sQTL_alt_allele']):\n",
    "#        return 'no'\n",
    "#    elif (data['_merge'] == 'both') & (data['ref_allele'] == data['sQTL_ref_allele']) & (data['alt_allele'] != data['sQTL_alt_allele']):\n",
    "#        return 'no'\n",
    "#    elif (data['_merge'] == 'both') & (data['ref_allele'] == data['sQTL_ref_allele']) & (data['alt_allele'] == data['sQTL_alt_allele']):\n",
    "#        return 'yes'\n",
    "\n",
    "#data['sQTL'] = data.apply(sQTL, axis = 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "80fc3857",
   "metadata": {},
   "outputs": [],
   "source": [
    "#data.head(10)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "bd3361ef",
   "metadata": {},
   "source": [
    "Let's also set the GTEx columns to NaN for variants that are not sQTLs."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "72d24a0e",
   "metadata": {},
   "outputs": [],
   "source": [
    "#data.loc[data['sQTL'] == 'no',['Adipose_Subcutaneous','Adipose_Visceral_Omentum','Adrenal_Gland','Artery_Aorta','Artery_Coronary','Artery_Tibial','Brain_Amygdala','Brain_Anterior_cingulate_cortex_BA24','Brain_Caudate_basal_ganglia','Brain_Cerebellar_Hemisphere','Brain_Cerebellum','Brain_Cortex','Brain_Frontal_Cortex_BA9','Brain_Hippocampus','Brain_Hypothalamus','Brain_Nucleus_accumbens_basal_ganglia','Brain_Putamen_basal_ganglia','Brain_Spinal_cord_cervical_c-1','Brain_Substantia_nigra','Breast_Mammary_Tissue','Cells_Cultured_fibroblasts','Cells_EBV-transformed_lymphocytes','Colon_Sigmoid','Colon_Transverse','Esophagus_Gastroesophageal_Junction','Esophagus_Mucosa','Esophagus_Muscularis','Heart_Atrial_Appendage','Heart_Left_Ventricle','Kidney_Cortex','Liver','Lung','Minor_Salivary_Gland','Muscle_Skeletal','Nerve_Tibial','Ovary','Pancreas','Pituitary','Prostate','Skin_Not_Sun_Exposed_Suprapubic','Skin_Sun_Exposed_Lower_leg','Small_Intestine_Terminal_Ileum','Spleen','Stomach','Testis','Thyroid','Uterus','Vagina','Whole_Blood','N_GTEx_tissues']] = np.nan"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "dc615ed9",
   "metadata": {},
   "source": [
    "Check that that worked."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "65439959",
   "metadata": {},
   "outputs": [],
   "source": [
    "#data.groupby(['sQTL','N_GTEx_tissues']).size().to_frame('count')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a8da44a7",
   "metadata": {},
   "source": [
    "Drop columns we no longer need."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "aeaa4466",
   "metadata": {},
   "outputs": [],
   "source": [
    "#data = data.drop(columns = ['sQTL_ref_allele','sQTL_alt_allele','_merge'])\n",
    "#data.head(10)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d3c419cc",
   "metadata": {},
   "source": [
    "Pandas panicked and created lots of duplicates. Let's get our N back to where it should be: 1,607,350."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "a67aea87",
   "metadata": {},
   "outputs": [],
   "source": [
    "#data = data.drop_duplicates(['chrom','pos','ref_allele','alt_allele','annotation'])\n",
    "#len(data)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "256f009f",
   "metadata": {},
   "source": [
    "Finally, while many sQTLs appear to also be variant in 1KG and gnomAD, there appear to be some novel variants. Let's reclassify these from 'archaic-specific' to 'low-confidence ancient' for both Vernot and Browning."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "92e7a9e0",
   "metadata": {},
   "outputs": [],
   "source": [
    "#def new_Vernot_allele_origin(data):\n",
    "#    if (data['Vernot_allele_origin'] == 'ancient'):\n",
    "#        return 'ancient'\n",
    "#    elif (data['Vernot_allele_origin'] == 'introgressed'):\n",
    "#        return 'introgressed'\n",
    "#    elif (data['Vernot_allele_origin'] == 'low-confidence ancient'):\n",
    "#        return 'low-confidence ancient'\n",
    "#    elif (data['Vernot_allele_origin'] == 'archaic-specific') & (data['sQTL'] == 'yes'):\n",
    "#        return 'low-confidence ancient'\n",
    "#    elif (data['Vernot_allele_origin'] == 'archaic-specific') & (data['sQTL'] == 'no'):\n",
    "#        return 'archaic-specific'\n",
    "\n",
    "#data['new_Vernot_allele_origin'] = data.apply(new_Vernot_allele_origin, axis = 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "3c39c108",
   "metadata": {},
   "outputs": [],
   "source": [
    "#def new_Browning_allele_origin(data):\n",
    "#    if (data['Browning_allele_origin'] == 'ancient'):\n",
    "#        return 'ancient'\n",
    "#    elif (data['Browning_allele_origin'] == 'introgressed'):\n",
    "#        return 'introgressed'\n",
    "#    elif (data['Browning_allele_origin'] == 'low-confidence ancient'):\n",
    "#        return 'low-confidence ancient'\n",
    "#    elif (data['Browning_allele_origin'] == 'archaic-specific') & (data['sQTL'] == 'yes'):\n",
    "#        return 'low-confidence ancient'\n",
    "#    elif (data['Browning_allele_origin'] == 'archaic-specific') & (data['sQTL'] == 'no'):\n",
    "#        return 'archaic-specific'\n",
    "\n",
    "#data['new_Browning_allele_origin'] = data.apply(new_Browning_allele_origin, axis = 1)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "769235ce",
   "metadata": {},
   "source": [
    "Drop the old columns."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "7bf6bd2e",
   "metadata": {},
   "outputs": [],
   "source": [
    "#data = data.drop(columns = ['Vernot_allele_origin'])\n",
    "#data.rename(columns={'new_Vernot_allele_origin': 'Vernot_allele_origin'}, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "4afa1856",
   "metadata": {},
   "outputs": [],
   "source": [
    "#data = data.drop(columns = ['Browning_allele_origin'])\n",
    "#data.rename(columns={'new_Browning_allele_origin': 'Browning_allele_origin'}, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "c67def82",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "#data.head(5)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "bd11c9ff",
   "metadata": {},
   "source": [
    "Save this dataframe."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "d5ca2700",
   "metadata": {},
   "outputs": [],
   "source": [
    "#data.to_csv('archaic_data_with_constraint_moderns_introgression_sQTLs.txt', sep='\\t', header = True, index = False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c7527cec",
   "metadata": {},
   "source": [
    "Add 'n/a' to blank cells for the published dataset."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "3cc4fd50",
   "metadata": {},
   "outputs": [],
   "source": [
    "#data_with_NAs = data.fillna('n/a')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "34a382cb",
   "metadata": {},
   "source": [
    "Save this dataframe."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "676e5af8",
   "metadata": {},
   "outputs": [],
   "source": [
    "#data_with_NAs.to_csv('archaic_data_with_constraint_moderns_introgression_sQTLs_with_NAs.txt', sep = '\\t', header = True, index = False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c1425e8e",
   "metadata": {},
   "source": [
    "**Start here with the full dataframe.** <a class = 'anchor' id = 'startwithdataframe'></a>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "4da21b19",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/scratch/cbrand/ipykernel_47833/1210275765.py:1: DtypeWarning: Columns (43) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  data = pd.read_csv('archaic_data_with_constraint_moderns_introgression_sQTLs.txt', sep='\\t', header=0)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref_allele</th>\n",
       "      <th>alt_allele</th>\n",
       "      <th>ancestral_allele</th>\n",
       "      <th>anc_dev</th>\n",
       "      <th>variant_type</th>\n",
       "      <th>altai_gt</th>\n",
       "      <th>chagyrskaya_gt</th>\n",
       "      <th>denisovan_gt</th>\n",
       "      <th>vindija_gt</th>\n",
       "      <th>altai_gt_boolean</th>\n",
       "      <th>chagyrskaya_gt_boolean</th>\n",
       "      <th>denisovan_gt_boolean</th>\n",
       "      <th>vindija_gt_boolean</th>\n",
       "      <th>distribution</th>\n",
       "      <th>present_in_1KG</th>\n",
       "      <th>1KG_allele_count</th>\n",
       "      <th>1KG_allele_number</th>\n",
       "      <th>1KG_allele_frequency</th>\n",
       "      <th>1KG_EAS_AF</th>\n",
       "      <th>1KG_EUR_AF</th>\n",
       "      <th>1KG_AFR_AF</th>\n",
       "      <th>1KG_AMR_AF</th>\n",
       "      <th>1KG_SAS_AF</th>\n",
       "      <th>1KG_non_ASW_AFR_AF</th>\n",
       "      <th>present_in_gnomAD</th>\n",
       "      <th>gnomAD_allele_count</th>\n",
       "      <th>gnomAD_allele_number</th>\n",
       "      <th>gnomAD_allele_frequency</th>\n",
       "      <th>Vernot_introgressed</th>\n",
       "      <th>Vernot_ancestral_allele</th>\n",
       "      <th>Vernot_derived_allele</th>\n",
       "      <th>Vernot_ancestral_derived_code</th>\n",
       "      <th>Vernot_AFA_AF</th>\n",
       "      <th>Vernot_AFR_AF</th>\n",
       "      <th>Vernot_AMR_AF</th>\n",
       "      <th>Vernot_EAS_AF</th>\n",
       "      <th>Vernot_EUR_AF</th>\n",
       "      <th>Vernot_PNG_AF</th>\n",
       "      <th>Vernot_SAS_AF</th>\n",
       "      <th>Vernot_Denisovan_base</th>\n",
       "      <th>Vernot_Neanderthal_base_1</th>\n",
       "      <th>Vernot_Neanderthal_base_2</th>\n",
       "      <th>Vernot_haplotype_tag</th>\n",
       "      <th>Vernot_introgressed_AF</th>\n",
       "      <th>Browning_introgressed</th>\n",
       "      <th>Browning_ref_alt</th>\n",
       "      <th>Browning_introgressed_AF</th>\n",
       "      <th>annotation</th>\n",
       "      <th>mis_oe</th>\n",
       "      <th>mis_z</th>\n",
       "      <th>lof_oe</th>\n",
       "      <th>lof_z</th>\n",
       "      <th>phyloP</th>\n",
       "      <th>ag_delta</th>\n",
       "      <th>al_delta</th>\n",
       "      <th>dg_delta</th>\n",
       "      <th>dl_delta</th>\n",
       "      <th>delta_max</th>\n",
       "      <th>ag_pos</th>\n",
       "      <th>al_pos</th>\n",
       "      <th>dg_pos</th>\n",
       "      <th>dl_pos</th>\n",
       "      <th>Adipose_Subcutaneous</th>\n",
       "      <th>Adipose_Visceral_Omentum</th>\n",
       "      <th>Adrenal_Gland</th>\n",
       "      <th>Artery_Aorta</th>\n",
       "      <th>Artery_Coronary</th>\n",
       "      <th>Artery_Tibial</th>\n",
       "      <th>Brain_Amygdala</th>\n",
       "      <th>Brain_Anterior_cingulate_cortex_BA24</th>\n",
       "      <th>Brain_Caudate_basal_ganglia</th>\n",
       "      <th>Brain_Cerebellar_Hemisphere</th>\n",
       "      <th>Brain_Cerebellum</th>\n",
       "      <th>Brain_Cortex</th>\n",
       "      <th>Brain_Frontal_Cortex_BA9</th>\n",
       "      <th>Brain_Hippocampus</th>\n",
       "      <th>Brain_Hypothalamus</th>\n",
       "      <th>Brain_Nucleus_accumbens_basal_ganglia</th>\n",
       "      <th>Brain_Putamen_basal_ganglia</th>\n",
       "      <th>Brain_Spinal_cord_cervical_c-1</th>\n",
       "      <th>Brain_Substantia_nigra</th>\n",
       "      <th>Breast_Mammary_Tissue</th>\n",
       "      <th>Cells_Cultured_fibroblasts</th>\n",
       "      <th>Cells_EBV-transformed_lymphocytes</th>\n",
       "      <th>Colon_Sigmoid</th>\n",
       "      <th>Colon_Transverse</th>\n",
       "      <th>Esophagus_Gastroesophageal_Junction</th>\n",
       "      <th>Esophagus_Mucosa</th>\n",
       "      <th>Esophagus_Muscularis</th>\n",
       "      <th>Heart_Atrial_Appendage</th>\n",
       "      <th>Heart_Left_Ventricle</th>\n",
       "      <th>Kidney_Cortex</th>\n",
       "      <th>Liver</th>\n",
       "      <th>Lung</th>\n",
       "      <th>Minor_Salivary_Gland</th>\n",
       "      <th>Muscle_Skeletal</th>\n",
       "      <th>Nerve_Tibial</th>\n",
       "      <th>Ovary</th>\n",
       "      <th>Pancreas</th>\n",
       "      <th>Pituitary</th>\n",
       "      <th>Prostate</th>\n",
       "      <th>Skin_Not_Sun_Exposed_Suprapubic</th>\n",
       "      <th>Skin_Sun_Exposed_Lower_leg</th>\n",
       "      <th>Small_Intestine_Terminal_Ileum</th>\n",
       "      <th>Spleen</th>\n",
       "      <th>Stomach</th>\n",
       "      <th>Testis</th>\n",
       "      <th>Thyroid</th>\n",
       "      <th>Uterus</th>\n",
       "      <th>Vagina</th>\n",
       "      <th>Whole_Blood</th>\n",
       "      <th>N_GTEx_tissues</th>\n",
       "      <th>sQTL</th>\n",
       "      <th>Vernot_allele_origin</th>\n",
       "      <th>Browning_allele_origin</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>861808</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>yes</td>\n",
       "      <td>3479.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.68</td>\n",
       "      <td>0.55</td>\n",
       "      <td>0.97</td>\n",
       "      <td>0.34</td>\n",
       "      <td>0.83</td>\n",
       "      <td>0.88</td>\n",
       "      <td>0.313840</td>\n",
       "      <td>yes</td>\n",
       "      <td>112289.0</td>\n",
       "      <td>143198.0</td>\n",
       "      <td>0.784152</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>1.5082</td>\n",
       "      <td>-3.43610</td>\n",
       "      <td>0.89656</td>\n",
       "      <td>0.47484</td>\n",
       "      <td>-0.683</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.01</td>\n",
       "      <td>-29</td>\n",
       "      <td>-27</td>\n",
       "      <td>24</td>\n",
       "      <td>-20</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>ancient</td>\n",
       "      <td>ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>861808</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>yes</td>\n",
       "      <td>3479.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.68</td>\n",
       "      <td>0.55</td>\n",
       "      <td>0.97</td>\n",
       "      <td>0.34</td>\n",
       "      <td>0.83</td>\n",
       "      <td>0.88</td>\n",
       "      <td>0.313840</td>\n",
       "      <td>yes</td>\n",
       "      <td>112289.0</td>\n",
       "      <td>143198.0</td>\n",
       "      <td>0.784152</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AL645608.1</td>\n",
       "      <td>1.2217</td>\n",
       "      <td>-0.64548</td>\n",
       "      <td>0.73515</td>\n",
       "      <td>0.49579</td>\n",
       "      <td>-0.683</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>48</td>\n",
       "      <td>-14</td>\n",
       "      <td>-46</td>\n",
       "      <td>-14</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>ancient</td>\n",
       "      <td>ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>862072</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>yes</td>\n",
       "      <td>9.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>yes</td>\n",
       "      <td>8.0</td>\n",
       "      <td>143246.0</td>\n",
       "      <td>0.000056</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>1.5082</td>\n",
       "      <td>-3.43610</td>\n",
       "      <td>0.89656</td>\n",
       "      <td>0.47484</td>\n",
       "      <td>0.197</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>3</td>\n",
       "      <td>28</td>\n",
       "      <td>-38</td>\n",
       "      <td>41</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>862072</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>yes</td>\n",
       "      <td>9.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>yes</td>\n",
       "      <td>8.0</td>\n",
       "      <td>143246.0</td>\n",
       "      <td>0.000056</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AL645608.1</td>\n",
       "      <td>1.2217</td>\n",
       "      <td>-0.64548</td>\n",
       "      <td>0.73515</td>\n",
       "      <td>0.49579</td>\n",
       "      <td>0.197</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-6</td>\n",
       "      <td>-50</td>\n",
       "      <td>24</td>\n",
       "      <td>-48</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>862093</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>ancestral</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>yes</td>\n",
       "      <td>3484.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.68</td>\n",
       "      <td>0.55</td>\n",
       "      <td>0.97</td>\n",
       "      <td>0.34</td>\n",
       "      <td>0.83</td>\n",
       "      <td>0.88</td>\n",
       "      <td>0.315789</td>\n",
       "      <td>yes</td>\n",
       "      <td>111981.0</td>\n",
       "      <td>143126.0</td>\n",
       "      <td>0.782395</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>1.5082</td>\n",
       "      <td>-3.43610</td>\n",
       "      <td>0.89656</td>\n",
       "      <td>0.47484</td>\n",
       "      <td>-1.042</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>7</td>\n",
       "      <td>14</td>\n",
       "      <td>-15</td>\n",
       "      <td>34</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>ancient</td>\n",
       "      <td>ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>862093</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>ancestral</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>yes</td>\n",
       "      <td>3484.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.68</td>\n",
       "      <td>0.55</td>\n",
       "      <td>0.97</td>\n",
       "      <td>0.34</td>\n",
       "      <td>0.83</td>\n",
       "      <td>0.88</td>\n",
       "      <td>0.315789</td>\n",
       "      <td>yes</td>\n",
       "      <td>111981.0</td>\n",
       "      <td>143126.0</td>\n",
       "      <td>0.782395</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AL645608.1</td>\n",
       "      <td>1.2217</td>\n",
       "      <td>-0.64548</td>\n",
       "      <td>0.73515</td>\n",
       "      <td>0.49579</td>\n",
       "      <td>-1.042</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-34</td>\n",
       "      <td>48</td>\n",
       "      <td>49</td>\n",
       "      <td>3</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>ancient</td>\n",
       "      <td>ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr1</td>\n",
       "      <td>862124</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>ancestral</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>yes</td>\n",
       "      <td>3485.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.68</td>\n",
       "      <td>0.55</td>\n",
       "      <td>0.97</td>\n",
       "      <td>0.34</td>\n",
       "      <td>0.83</td>\n",
       "      <td>0.88</td>\n",
       "      <td>0.316764</td>\n",
       "      <td>yes</td>\n",
       "      <td>112240.0</td>\n",
       "      <td>143138.0</td>\n",
       "      <td>0.784138</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>1.5082</td>\n",
       "      <td>-3.43610</td>\n",
       "      <td>0.89656</td>\n",
       "      <td>0.47484</td>\n",
       "      <td>-3.750</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-24</td>\n",
       "      <td>26</td>\n",
       "      <td>-11</td>\n",
       "      <td>3</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>ancient</td>\n",
       "      <td>ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr1</td>\n",
       "      <td>862124</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>ancestral</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>yes</td>\n",
       "      <td>3485.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.68</td>\n",
       "      <td>0.55</td>\n",
       "      <td>0.97</td>\n",
       "      <td>0.34</td>\n",
       "      <td>0.83</td>\n",
       "      <td>0.88</td>\n",
       "      <td>0.316764</td>\n",
       "      <td>yes</td>\n",
       "      <td>112240.0</td>\n",
       "      <td>143138.0</td>\n",
       "      <td>0.784138</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AL645608.1</td>\n",
       "      <td>1.2217</td>\n",
       "      <td>-0.64548</td>\n",
       "      <td>0.73515</td>\n",
       "      <td>0.49579</td>\n",
       "      <td>-3.750</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>17</td>\n",
       "      <td>-14</td>\n",
       "      <td>-28</td>\n",
       "      <td>35</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>ancient</td>\n",
       "      <td>ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr1</td>\n",
       "      <td>862383</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>yes</td>\n",
       "      <td>3476.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.68</td>\n",
       "      <td>0.55</td>\n",
       "      <td>0.97</td>\n",
       "      <td>0.34</td>\n",
       "      <td>0.83</td>\n",
       "      <td>0.88</td>\n",
       "      <td>0.310916</td>\n",
       "      <td>yes</td>\n",
       "      <td>111632.0</td>\n",
       "      <td>142788.0</td>\n",
       "      <td>0.781802</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>1.5082</td>\n",
       "      <td>-3.43610</td>\n",
       "      <td>0.89656</td>\n",
       "      <td>0.47484</td>\n",
       "      <td>-1.932</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>22</td>\n",
       "      <td>-26</td>\n",
       "      <td>-21</td>\n",
       "      <td>29</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>ancient</td>\n",
       "      <td>ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr1</td>\n",
       "      <td>862383</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>yes</td>\n",
       "      <td>3476.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.68</td>\n",
       "      <td>0.55</td>\n",
       "      <td>0.97</td>\n",
       "      <td>0.34</td>\n",
       "      <td>0.83</td>\n",
       "      <td>0.88</td>\n",
       "      <td>0.310916</td>\n",
       "      <td>yes</td>\n",
       "      <td>111632.0</td>\n",
       "      <td>142788.0</td>\n",
       "      <td>0.781802</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AL645608.1</td>\n",
       "      <td>1.2217</td>\n",
       "      <td>-0.64548</td>\n",
       "      <td>0.73515</td>\n",
       "      <td>0.49579</td>\n",
       "      <td>-1.932</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-23</td>\n",
       "      <td>7</td>\n",
       "      <td>28</td>\n",
       "      <td>46</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>ancient</td>\n",
       "      <td>ancient</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chrom     pos ref_allele alt_allele ancestral_allele    anc_dev  \\\n",
       "0  chr1  861808          A          G                A    derived   \n",
       "1  chr1  861808          A          G                A    derived   \n",
       "2  chr1  862072          C          T                C    derived   \n",
       "3  chr1  862072          C          T                C    derived   \n",
       "4  chr1  862093          T          C                C  ancestral   \n",
       "5  chr1  862093          T          C                C  ancestral   \n",
       "6  chr1  862124          A          G                G  ancestral   \n",
       "7  chr1  862124          A          G                G  ancestral   \n",
       "8  chr1  862383          C          T                C    derived   \n",
       "9  chr1  862383          C          T                C    derived   \n",
       "\n",
       "  variant_type altai_gt chagyrskaya_gt denisovan_gt vindija_gt  \\\n",
       "0          snv      0/0            0/0          1/1        0/0   \n",
       "1          snv      0/0            0/0          1/1        0/0   \n",
       "2          snv      1/1            1/1          0/0        0/1   \n",
       "3          snv      1/1            1/1          0/0        0/1   \n",
       "4          snv      0/0            0/0          1/1        0/0   \n",
       "5          snv      0/0            0/0          1/1        0/0   \n",
       "6          snv      0/0            0/0          1/1        0/0   \n",
       "7          snv      0/0            0/0          1/1        0/0   \n",
       "8          snv      0/0            0/0          1/1        0/0   \n",
       "9          snv      0/0            0/0          1/1        0/0   \n",
       "\n",
       "   altai_gt_boolean  chagyrskaya_gt_boolean  denisovan_gt_boolean  \\\n",
       "0             False                   False                  True   \n",
       "1             False                   False                  True   \n",
       "2              True                    True                 False   \n",
       "3              True                    True                 False   \n",
       "4             False                   False                  True   \n",
       "5             False                   False                  True   \n",
       "6             False                   False                  True   \n",
       "7             False                   False                  True   \n",
       "8             False                   False                  True   \n",
       "9             False                   False                  True   \n",
       "\n",
       "   vindija_gt_boolean distribution present_in_1KG  1KG_allele_count  \\\n",
       "0               False    Denisovan            yes            3479.0   \n",
       "1               False    Denisovan            yes            3479.0   \n",
       "2                True  Neanderthal            yes               9.0   \n",
       "3                True  Neanderthal            yes               9.0   \n",
       "4               False    Denisovan            yes            3484.0   \n",
       "5               False    Denisovan            yes            3484.0   \n",
       "6               False    Denisovan            yes            3485.0   \n",
       "7               False    Denisovan            yes            3485.0   \n",
       "8               False    Denisovan            yes            3476.0   \n",
       "9               False    Denisovan            yes            3476.0   \n",
       "\n",
       "   1KG_allele_number  1KG_allele_frequency  1KG_EAS_AF  1KG_EUR_AF  \\\n",
       "0             5096.0                  0.68        0.55        0.97   \n",
       "1             5096.0                  0.68        0.55        0.97   \n",
       "2             5096.0                  0.00        0.01        0.00   \n",
       "3             5096.0                  0.00        0.01        0.00   \n",
       "4             5096.0                  0.68        0.55        0.97   \n",
       "5             5096.0                  0.68        0.55        0.97   \n",
       "6             5096.0                  0.68        0.55        0.97   \n",
       "7             5096.0                  0.68        0.55        0.97   \n",
       "8             5096.0                  0.68        0.55        0.97   \n",
       "9             5096.0                  0.68        0.55        0.97   \n",
       "\n",
       "   1KG_AFR_AF  1KG_AMR_AF  1KG_SAS_AF  1KG_non_ASW_AFR_AF present_in_gnomAD  \\\n",
       "0        0.34        0.83        0.88            0.313840               yes   \n",
       "1        0.34        0.83        0.88            0.313840               yes   \n",
       "2        0.00        0.00        0.00            0.000000               yes   \n",
       "3        0.00        0.00        0.00            0.000000               yes   \n",
       "4        0.34        0.83        0.88            0.315789               yes   \n",
       "5        0.34        0.83        0.88            0.315789               yes   \n",
       "6        0.34        0.83        0.88            0.316764               yes   \n",
       "7        0.34        0.83        0.88            0.316764               yes   \n",
       "8        0.34        0.83        0.88            0.310916               yes   \n",
       "9        0.34        0.83        0.88            0.310916               yes   \n",
       "\n",
       "   gnomAD_allele_count  gnomAD_allele_number  gnomAD_allele_frequency  \\\n",
       "0             112289.0              143198.0                 0.784152   \n",
       "1             112289.0              143198.0                 0.784152   \n",
       "2                  8.0              143246.0                 0.000056   \n",
       "3                  8.0              143246.0                 0.000056   \n",
       "4             111981.0              143126.0                 0.782395   \n",
       "5             111981.0              143126.0                 0.782395   \n",
       "6             112240.0              143138.0                 0.784138   \n",
       "7             112240.0              143138.0                 0.784138   \n",
       "8             111632.0              142788.0                 0.781802   \n",
       "9             111632.0              142788.0                 0.781802   \n",
       "\n",
       "  Vernot_introgressed Vernot_ancestral_allele Vernot_derived_allele  \\\n",
       "0                  no                     NaN                   NaN   \n",
       "1                  no                     NaN                   NaN   \n",
       "2                  no                     NaN                   NaN   \n",
       "3                  no                     NaN                   NaN   \n",
       "4                  no                     NaN                   NaN   \n",
       "5                  no                     NaN                   NaN   \n",
       "6                  no                     NaN                   NaN   \n",
       "7                  no                     NaN                   NaN   \n",
       "8                  no                     NaN                   NaN   \n",
       "9                  no                     NaN                   NaN   \n",
       "\n",
       "   Vernot_ancestral_derived_code  Vernot_AFA_AF  Vernot_AFR_AF  Vernot_AMR_AF  \\\n",
       "0                            NaN            NaN            NaN            NaN   \n",
       "1                            NaN            NaN            NaN            NaN   \n",
       "2                            NaN            NaN            NaN            NaN   \n",
       "3                            NaN            NaN            NaN            NaN   \n",
       "4                            NaN            NaN            NaN            NaN   \n",
       "5                            NaN            NaN            NaN            NaN   \n",
       "6                            NaN            NaN            NaN            NaN   \n",
       "7                            NaN            NaN            NaN            NaN   \n",
       "8                            NaN            NaN            NaN            NaN   \n",
       "9                            NaN            NaN            NaN            NaN   \n",
       "\n",
       "   Vernot_EAS_AF  Vernot_EUR_AF  Vernot_PNG_AF  Vernot_SAS_AF  \\\n",
       "0            NaN            NaN            NaN            NaN   \n",
       "1            NaN            NaN            NaN            NaN   \n",
       "2            NaN            NaN            NaN            NaN   \n",
       "3            NaN            NaN            NaN            NaN   \n",
       "4            NaN            NaN            NaN            NaN   \n",
       "5            NaN            NaN            NaN            NaN   \n",
       "6            NaN            NaN            NaN            NaN   \n",
       "7            NaN            NaN            NaN            NaN   \n",
       "8            NaN            NaN            NaN            NaN   \n",
       "9            NaN            NaN            NaN            NaN   \n",
       "\n",
       "  Vernot_Denisovan_base Vernot_Neanderthal_base_1 Vernot_Neanderthal_base_2  \\\n",
       "0                   NaN                       NaN                       NaN   \n",
       "1                   NaN                       NaN                       NaN   \n",
       "2                   NaN                       NaN                       NaN   \n",
       "3                   NaN                       NaN                       NaN   \n",
       "4                   NaN                       NaN                       NaN   \n",
       "5                   NaN                       NaN                       NaN   \n",
       "6                   NaN                       NaN                       NaN   \n",
       "7                   NaN                       NaN                       NaN   \n",
       "8                   NaN                       NaN                       NaN   \n",
       "9                   NaN                       NaN                       NaN   \n",
       "\n",
       "  Vernot_haplotype_tag  Vernot_introgressed_AF Browning_introgressed  \\\n",
       "0                  NaN                     NaN                    no   \n",
       "1                  NaN                     NaN                    no   \n",
       "2                  NaN                     NaN                    no   \n",
       "3                  NaN                     NaN                    no   \n",
       "4                  NaN                     NaN                    no   \n",
       "5                  NaN                     NaN                    no   \n",
       "6                  NaN                     NaN                    no   \n",
       "7                  NaN                     NaN                    no   \n",
       "8                  NaN                     NaN                    no   \n",
       "9                  NaN                     NaN                    no   \n",
       "\n",
       "   Browning_ref_alt  Browning_introgressed_AF  annotation  mis_oe    mis_z  \\\n",
       "0               NaN                       NaN      SAMD11  1.5082 -3.43610   \n",
       "1               NaN                       NaN  AL645608.1  1.2217 -0.64548   \n",
       "2               NaN                       NaN      SAMD11  1.5082 -3.43610   \n",
       "3               NaN                       NaN  AL645608.1  1.2217 -0.64548   \n",
       "4               NaN                       NaN      SAMD11  1.5082 -3.43610   \n",
       "5               NaN                       NaN  AL645608.1  1.2217 -0.64548   \n",
       "6               NaN                       NaN      SAMD11  1.5082 -3.43610   \n",
       "7               NaN                       NaN  AL645608.1  1.2217 -0.64548   \n",
       "8               NaN                       NaN      SAMD11  1.5082 -3.43610   \n",
       "9               NaN                       NaN  AL645608.1  1.2217 -0.64548   \n",
       "\n",
       "    lof_oe    lof_z  phyloP  ag_delta  al_delta  dg_delta  dl_delta  \\\n",
       "0  0.89656  0.47484  -0.683       0.0       0.0      0.01       0.0   \n",
       "1  0.73515  0.49579  -0.683       0.0       0.0      0.00       0.0   \n",
       "2  0.89656  0.47484   0.197       0.0       0.0      0.00       0.0   \n",
       "3  0.73515  0.49579   0.197       0.0       0.0      0.00       0.0   \n",
       "4  0.89656  0.47484  -1.042       0.0       0.0      0.00       0.0   \n",
       "5  0.73515  0.49579  -1.042       0.0       0.0      0.00       0.0   \n",
       "6  0.89656  0.47484  -3.750       0.0       0.0      0.00       0.0   \n",
       "7  0.73515  0.49579  -3.750       0.0       0.0      0.00       0.0   \n",
       "8  0.89656  0.47484  -1.932       0.0       0.0      0.00       0.0   \n",
       "9  0.73515  0.49579  -1.932       0.0       0.0      0.00       0.0   \n",
       "\n",
       "   delta_max  ag_pos  al_pos  dg_pos  dl_pos  Adipose_Subcutaneous  \\\n",
       "0       0.01     -29     -27      24     -20                   0.0   \n",
       "1       0.00      48     -14     -46     -14                   0.0   \n",
       "2       0.00       3      28     -38      41                   NaN   \n",
       "3       0.00      -6     -50      24     -48                   NaN   \n",
       "4       0.00       7      14     -15      34                   0.0   \n",
       "5       0.00     -34      48      49       3                   0.0   \n",
       "6       0.00     -24      26     -11       3                   0.0   \n",
       "7       0.00      17     -14     -28      35                   0.0   \n",
       "8       0.00      22     -26     -21      29                   0.0   \n",
       "9       0.00     -23       7      28      46                   0.0   \n",
       "\n",
       "   Adipose_Visceral_Omentum  Adrenal_Gland  Artery_Aorta  Artery_Coronary  \\\n",
       "0                       0.0            0.0           0.0              0.0   \n",
       "1                       0.0            0.0           0.0              0.0   \n",
       "2                       NaN            NaN           NaN              NaN   \n",
       "3                       NaN            NaN           NaN              NaN   \n",
       "4                       0.0            0.0           0.0              0.0   \n",
       "5                       0.0            0.0           0.0              0.0   \n",
       "6                       0.0            0.0           0.0              0.0   \n",
       "7                       0.0            0.0           0.0              0.0   \n",
       "8                       0.0            0.0           0.0              0.0   \n",
       "9                       0.0            0.0           0.0              0.0   \n",
       "\n",
       "   Artery_Tibial  Brain_Amygdala  Brain_Anterior_cingulate_cortex_BA24  \\\n",
       "0            0.0             0.0                                   0.0   \n",
       "1            0.0             0.0                                   0.0   \n",
       "2            NaN             NaN                                   NaN   \n",
       "3            NaN             NaN                                   NaN   \n",
       "4            0.0             0.0                                   0.0   \n",
       "5            0.0             0.0                                   0.0   \n",
       "6            0.0             0.0                                   0.0   \n",
       "7            0.0             0.0                                   0.0   \n",
       "8            0.0             0.0                                   0.0   \n",
       "9            0.0             0.0                                   0.0   \n",
       "\n",
       "   Brain_Caudate_basal_ganglia  Brain_Cerebellar_Hemisphere  Brain_Cerebellum  \\\n",
       "0                          0.0                          0.0               0.0   \n",
       "1                          0.0                          0.0               0.0   \n",
       "2                          NaN                          NaN               NaN   \n",
       "3                          NaN                          NaN               NaN   \n",
       "4                          0.0                          0.0               0.0   \n",
       "5                          0.0                          0.0               0.0   \n",
       "6                          0.0                          0.0               0.0   \n",
       "7                          0.0                          0.0               0.0   \n",
       "8                          0.0                          0.0               0.0   \n",
       "9                          0.0                          0.0               0.0   \n",
       "\n",
       "   Brain_Cortex  Brain_Frontal_Cortex_BA9  Brain_Hippocampus  \\\n",
       "0           0.0                       0.0                0.0   \n",
       "1           0.0                       0.0                0.0   \n",
       "2           NaN                       NaN                NaN   \n",
       "3           NaN                       NaN                NaN   \n",
       "4           0.0                       0.0                0.0   \n",
       "5           0.0                       0.0                0.0   \n",
       "6           0.0                       0.0                0.0   \n",
       "7           0.0                       0.0                0.0   \n",
       "8           0.0                       0.0                0.0   \n",
       "9           0.0                       0.0                0.0   \n",
       "\n",
       "   Brain_Hypothalamus  Brain_Nucleus_accumbens_basal_ganglia  \\\n",
       "0                 0.0                                    0.0   \n",
       "1                 0.0                                    0.0   \n",
       "2                 NaN                                    NaN   \n",
       "3                 NaN                                    NaN   \n",
       "4                 0.0                                    0.0   \n",
       "5                 0.0                                    0.0   \n",
       "6                 0.0                                    0.0   \n",
       "7                 0.0                                    0.0   \n",
       "8                 0.0                                    0.0   \n",
       "9                 0.0                                    0.0   \n",
       "\n",
       "   Brain_Putamen_basal_ganglia  Brain_Spinal_cord_cervical_c-1  \\\n",
       "0                          0.0                             0.0   \n",
       "1                          0.0                             0.0   \n",
       "2                          NaN                             NaN   \n",
       "3                          NaN                             NaN   \n",
       "4                          0.0                             0.0   \n",
       "5                          0.0                             0.0   \n",
       "6                          0.0                             0.0   \n",
       "7                          0.0                             0.0   \n",
       "8                          0.0                             0.0   \n",
       "9                          0.0                             0.0   \n",
       "\n",
       "   Brain_Substantia_nigra  Breast_Mammary_Tissue  Cells_Cultured_fibroblasts  \\\n",
       "0                     0.0                    0.0                         0.0   \n",
       "1                     0.0                    0.0                         0.0   \n",
       "2                     NaN                    NaN                         NaN   \n",
       "3                     NaN                    NaN                         NaN   \n",
       "4                     0.0                    0.0                         0.0   \n",
       "5                     0.0                    0.0                         0.0   \n",
       "6                     0.0                    0.0                         0.0   \n",
       "7                     0.0                    0.0                         0.0   \n",
       "8                     0.0                    0.0                         0.0   \n",
       "9                     0.0                    0.0                         0.0   \n",
       "\n",
       "   Cells_EBV-transformed_lymphocytes  Colon_Sigmoid  Colon_Transverse  \\\n",
       "0                                0.0            0.0               0.0   \n",
       "1                                0.0            0.0               0.0   \n",
       "2                                NaN            NaN               NaN   \n",
       "3                                NaN            NaN               NaN   \n",
       "4                                0.0            0.0               0.0   \n",
       "5                                0.0            0.0               0.0   \n",
       "6                                0.0            0.0               0.0   \n",
       "7                                0.0            0.0               0.0   \n",
       "8                                0.0            0.0               0.0   \n",
       "9                                0.0            0.0               0.0   \n",
       "\n",
       "   Esophagus_Gastroesophageal_Junction  Esophagus_Mucosa  \\\n",
       "0                                  0.0               0.0   \n",
       "1                                  0.0               0.0   \n",
       "2                                  NaN               NaN   \n",
       "3                                  NaN               NaN   \n",
       "4                                  0.0               0.0   \n",
       "5                                  0.0               0.0   \n",
       "6                                  0.0               0.0   \n",
       "7                                  0.0               0.0   \n",
       "8                                  0.0               0.0   \n",
       "9                                  0.0               0.0   \n",
       "\n",
       "   Esophagus_Muscularis  Heart_Atrial_Appendage  Heart_Left_Ventricle  \\\n",
       "0                   0.0                     0.0                   0.0   \n",
       "1                   0.0                     0.0                   0.0   \n",
       "2                   NaN                     NaN                   NaN   \n",
       "3                   NaN                     NaN                   NaN   \n",
       "4                   0.0                     0.0                   0.0   \n",
       "5                   0.0                     0.0                   0.0   \n",
       "6                   0.0                     0.0                   0.0   \n",
       "7                   0.0                     0.0                   0.0   \n",
       "8                   0.0                     0.0                   0.0   \n",
       "9                   0.0                     0.0                   0.0   \n",
       "\n",
       "   Kidney_Cortex  Liver  Lung  Minor_Salivary_Gland  Muscle_Skeletal  \\\n",
       "0            0.0    0.0   0.0                   0.0              0.0   \n",
       "1            0.0    0.0   0.0                   0.0              0.0   \n",
       "2            NaN    NaN   NaN                   NaN              NaN   \n",
       "3            NaN    NaN   NaN                   NaN              NaN   \n",
       "4            0.0    0.0   0.0                   0.0              0.0   \n",
       "5            0.0    0.0   0.0                   0.0              0.0   \n",
       "6            0.0    0.0   0.0                   0.0              0.0   \n",
       "7            0.0    0.0   0.0                   0.0              0.0   \n",
       "8            0.0    0.0   0.0                   0.0              0.0   \n",
       "9            0.0    0.0   0.0                   0.0              0.0   \n",
       "\n",
       "   Nerve_Tibial  Ovary  Pancreas  Pituitary  Prostate  \\\n",
       "0           0.0    0.0       0.0        0.0       0.0   \n",
       "1           0.0    0.0       0.0        0.0       0.0   \n",
       "2           NaN    NaN       NaN        NaN       NaN   \n",
       "3           NaN    NaN       NaN        NaN       NaN   \n",
       "4           0.0    0.0       0.0        0.0       0.0   \n",
       "5           0.0    0.0       0.0        0.0       0.0   \n",
       "6           0.0    0.0       0.0        0.0       0.0   \n",
       "7           0.0    0.0       0.0        0.0       0.0   \n",
       "8           0.0    0.0       0.0        0.0       0.0   \n",
       "9           0.0    0.0       0.0        0.0       0.0   \n",
       "\n",
       "   Skin_Not_Sun_Exposed_Suprapubic  Skin_Sun_Exposed_Lower_leg  \\\n",
       "0                              0.0                         0.0   \n",
       "1                              0.0                         0.0   \n",
       "2                              NaN                         NaN   \n",
       "3                              NaN                         NaN   \n",
       "4                              0.0                         0.0   \n",
       "5                              0.0                         0.0   \n",
       "6                              0.0                         0.0   \n",
       "7                              0.0                         0.0   \n",
       "8                              0.0                         0.0   \n",
       "9                              0.0                         0.0   \n",
       "\n",
       "   Small_Intestine_Terminal_Ileum  Spleen  Stomach  Testis  Thyroid  Uterus  \\\n",
       "0                             0.0     0.0      0.0     0.0      0.0     0.0   \n",
       "1                             0.0     0.0      0.0     0.0      0.0     0.0   \n",
       "2                             NaN     NaN      NaN     NaN      NaN     NaN   \n",
       "3                             NaN     NaN      NaN     NaN      NaN     NaN   \n",
       "4                             0.0     0.0      0.0     0.0      0.0     0.0   \n",
       "5                             0.0     0.0      0.0     0.0      0.0     0.0   \n",
       "6                             0.0     0.0      0.0     0.0      0.0     0.0   \n",
       "7                             0.0     0.0      0.0     0.0      0.0     0.0   \n",
       "8                             0.0     0.0      0.0     0.0      0.0     0.0   \n",
       "9                             0.0     0.0      0.0     0.0      0.0     0.0   \n",
       "\n",
       "   Vagina  Whole_Blood  N_GTEx_tissues sQTL    Vernot_allele_origin  \\\n",
       "0     0.0          1.0             1.0  yes                 ancient   \n",
       "1     0.0          1.0             1.0  yes                 ancient   \n",
       "2     NaN          NaN             NaN   no  low-confidence ancient   \n",
       "3     NaN          NaN             NaN   no  low-confidence ancient   \n",
       "4     0.0          1.0             1.0  yes                 ancient   \n",
       "5     0.0          1.0             1.0  yes                 ancient   \n",
       "6     0.0          1.0             1.0  yes                 ancient   \n",
       "7     0.0          1.0             1.0  yes                 ancient   \n",
       "8     0.0          1.0             1.0  yes                 ancient   \n",
       "9     0.0          1.0             1.0  yes                 ancient   \n",
       "\n",
       "   Browning_allele_origin  \n",
       "0                 ancient  \n",
       "1                 ancient  \n",
       "2  low-confidence ancient  \n",
       "3  low-confidence ancient  \n",
       "4                 ancient  \n",
       "5                 ancient  \n",
       "6                 ancient  \n",
       "7                 ancient  \n",
       "8                 ancient  \n",
       "9                 ancient  "
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data = pd.read_csv('archaic_data_with_constraint_moderns_introgression_sQTLs.txt', sep='\\t', header=0)\n",
    "data.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "9ab2c104",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1607350"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(data)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7c471f40",
   "metadata": {},
   "source": [
    "# Spliceosome Variants <a class = 'anchor' id = 'spliceosomevariants'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "398498eb",
   "metadata": {},
   "source": [
    "Let's take a quick detour to examine variants in components of the major spliceosome. We'll start with genes identified in the HGNC: https://www.genenames.org/data/genegroup/#!/group/1518."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "94241a8b",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HGNC ID</th>\n",
       "      <th>Approved symbol</th>\n",
       "      <th>Approved name</th>\n",
       "      <th>Status</th>\n",
       "      <th>Locus type</th>\n",
       "      <th>Previous symbols</th>\n",
       "      <th>Alias symbols</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>NCBI Gene ID</th>\n",
       "      <th>Ensembl gene ID</th>\n",
       "      <th>Vega gene ID</th>\n",
       "      <th>Group ID</th>\n",
       "      <th>Group name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HGNC:17040</td>\n",
       "      <td>CASC3</td>\n",
       "      <td>CASC3 exon junction complex subunit</td>\n",
       "      <td>Approved</td>\n",
       "      <td>gene with protein product</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MLN51, BTZ</td>\n",
       "      <td>17q21.1</td>\n",
       "      <td>22794</td>\n",
       "      <td>ENSG00000108349</td>\n",
       "      <td>OTTHUMG00000133323</td>\n",
       "      <td>1238</td>\n",
       "      <td>Exon junction complex</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HGNC:18683</td>\n",
       "      <td>EIF4A3</td>\n",
       "      <td>eukaryotic translation initiation factor 4A3</td>\n",
       "      <td>Approved</td>\n",
       "      <td>gene with protein product</td>\n",
       "      <td>DDX48</td>\n",
       "      <td>KIAA0111, EIF4AIII, Fal1</td>\n",
       "      <td>17q25.3</td>\n",
       "      <td>9775</td>\n",
       "      <td>ENSG00000141543</td>\n",
       "      <td>OTTHUMG00000177538</td>\n",
       "      <td>1238</td>\n",
       "      <td>Exon junction complex</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HGNC:6815</td>\n",
       "      <td>MAGOH</td>\n",
       "      <td>mago homolog, exon junction complex subunit</td>\n",
       "      <td>Approved</td>\n",
       "      <td>gene with protein product</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MAGOHA, MAGOH1</td>\n",
       "      <td>1p32.3</td>\n",
       "      <td>4116</td>\n",
       "      <td>ENSG00000162385</td>\n",
       "      <td>OTTHUMG00000008932</td>\n",
       "      <td>1238</td>\n",
       "      <td>Exon junction complex</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HGNC:25504</td>\n",
       "      <td>MAGOHB</td>\n",
       "      <td>mago homolog B, exon junction complex subunit</td>\n",
       "      <td>Approved</td>\n",
       "      <td>gene with protein product</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FLJ10292, MGN2</td>\n",
       "      <td>12p13.2</td>\n",
       "      <td>55110</td>\n",
       "      <td>ENSG00000111196</td>\n",
       "      <td>OTTHUMG00000168407</td>\n",
       "      <td>1238</td>\n",
       "      <td>Exon junction complex</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HGNC:9905</td>\n",
       "      <td>RBM8A</td>\n",
       "      <td>RNA binding motif protein 8A</td>\n",
       "      <td>Approved</td>\n",
       "      <td>gene with protein product</td>\n",
       "      <td>RBM8</td>\n",
       "      <td>ZNRP, BOV-1A, BOV-1B, BOV-1C, RBM8B, Y14</td>\n",
       "      <td>1q21.1</td>\n",
       "      <td>9939</td>\n",
       "      <td>ENSG00000265241</td>\n",
       "      <td>OTTHUMG00000013736</td>\n",
       "      <td>1238</td>\n",
       "      <td>Exon junction complex</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>HGNC:20472</td>\n",
       "      <td>LSM1</td>\n",
       "      <td>LSM1 homolog, mRNA degradation associated</td>\n",
       "      <td>Approved</td>\n",
       "      <td>gene with protein product</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CASM, YJL124C</td>\n",
       "      <td>8p11.23</td>\n",
       "      <td>27257</td>\n",
       "      <td>ENSG00000175324</td>\n",
       "      <td>OTTHUMG00000164051</td>\n",
       "      <td>1505</td>\n",
       "      <td>LSm proteins</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>HGNC:17562</td>\n",
       "      <td>LSM10</td>\n",
       "      <td>LSM10, U7 small nuclear RNA associated</td>\n",
       "      <td>Approved</td>\n",
       "      <td>gene with protein product</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MGC15749</td>\n",
       "      <td>1p34.3</td>\n",
       "      <td>84967</td>\n",
       "      <td>ENSG00000181817</td>\n",
       "      <td>OTTHUMG00000008140</td>\n",
       "      <td>1505</td>\n",
       "      <td>LSm proteins</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>HGNC:30860</td>\n",
       "      <td>LSM11</td>\n",
       "      <td>LSM11, U7 small nuclear RNA associated</td>\n",
       "      <td>Approved</td>\n",
       "      <td>gene with protein product</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FLJ38273</td>\n",
       "      <td>5q33.3</td>\n",
       "      <td>134353</td>\n",
       "      <td>ENSG00000155858</td>\n",
       "      <td>OTTHUMG00000130255</td>\n",
       "      <td>1505</td>\n",
       "      <td>LSm proteins</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>HGNC:26407</td>\n",
       "      <td>LSM12</td>\n",
       "      <td>LSM12 homolog</td>\n",
       "      <td>Approved</td>\n",
       "      <td>gene with protein product</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FLJ30656</td>\n",
       "      <td>17q21.31</td>\n",
       "      <td>124801</td>\n",
       "      <td>ENSG00000161654</td>\n",
       "      <td>OTTHUMG00000181804</td>\n",
       "      <td>1505</td>\n",
       "      <td>LSm proteins</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>HGNC:24489</td>\n",
       "      <td>LSM14A</td>\n",
       "      <td>LSM14A mRNA processing body assembly factor</td>\n",
       "      <td>Approved</td>\n",
       "      <td>gene with protein product</td>\n",
       "      <td>C19orf13, FAM61A</td>\n",
       "      <td>DKFZP434D1335, RAP55A, RAP55</td>\n",
       "      <td>19q13.11</td>\n",
       "      <td>26065</td>\n",
       "      <td>ENSG00000257103</td>\n",
       "      <td>OTTHUMG00000180491</td>\n",
       "      <td>1505</td>\n",
       "      <td>LSm proteins</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      HGNC ID Approved symbol                                  Approved name  \\\n",
       "0  HGNC:17040           CASC3            CASC3 exon junction complex subunit   \n",
       "1  HGNC:18683          EIF4A3   eukaryotic translation initiation factor 4A3   \n",
       "2   HGNC:6815           MAGOH    mago homolog, exon junction complex subunit   \n",
       "3  HGNC:25504          MAGOHB  mago homolog B, exon junction complex subunit   \n",
       "4   HGNC:9905           RBM8A                   RNA binding motif protein 8A   \n",
       "5  HGNC:20472            LSM1      LSM1 homolog, mRNA degradation associated   \n",
       "6  HGNC:17562           LSM10         LSM10, U7 small nuclear RNA associated   \n",
       "7  HGNC:30860           LSM11         LSM11, U7 small nuclear RNA associated   \n",
       "8  HGNC:26407           LSM12                                  LSM12 homolog   \n",
       "9  HGNC:24489          LSM14A    LSM14A mRNA processing body assembly factor   \n",
       "\n",
       "     Status                 Locus type  Previous symbols  \\\n",
       "0  Approved  gene with protein product               NaN   \n",
       "1  Approved  gene with protein product             DDX48   \n",
       "2  Approved  gene with protein product               NaN   \n",
       "3  Approved  gene with protein product               NaN   \n",
       "4  Approved  gene with protein product              RBM8   \n",
       "5  Approved  gene with protein product               NaN   \n",
       "6  Approved  gene with protein product               NaN   \n",
       "7  Approved  gene with protein product               NaN   \n",
       "8  Approved  gene with protein product               NaN   \n",
       "9  Approved  gene with protein product  C19orf13, FAM61A   \n",
       "\n",
       "                              Alias symbols Chromosome  NCBI Gene ID  \\\n",
       "0                                MLN51, BTZ    17q21.1         22794   \n",
       "1                  KIAA0111, EIF4AIII, Fal1    17q25.3          9775   \n",
       "2                            MAGOHA, MAGOH1     1p32.3          4116   \n",
       "3                            FLJ10292, MGN2    12p13.2         55110   \n",
       "4  ZNRP, BOV-1A, BOV-1B, BOV-1C, RBM8B, Y14     1q21.1          9939   \n",
       "5                             CASM, YJL124C    8p11.23         27257   \n",
       "6                                  MGC15749     1p34.3         84967   \n",
       "7                                  FLJ38273     5q33.3        134353   \n",
       "8                                  FLJ30656   17q21.31        124801   \n",
       "9              DKFZP434D1335, RAP55A, RAP55   19q13.11         26065   \n",
       "\n",
       "   Ensembl gene ID        Vega gene ID  Group ID             Group name  \n",
       "0  ENSG00000108349  OTTHUMG00000133323      1238  Exon junction complex  \n",
       "1  ENSG00000141543  OTTHUMG00000177538      1238  Exon junction complex  \n",
       "2  ENSG00000162385  OTTHUMG00000008932      1238  Exon junction complex  \n",
       "3  ENSG00000111196  OTTHUMG00000168407      1238  Exon junction complex  \n",
       "4  ENSG00000265241  OTTHUMG00000013736      1238  Exon junction complex  \n",
       "5  ENSG00000175324  OTTHUMG00000164051      1505           LSm proteins  \n",
       "6  ENSG00000181817  OTTHUMG00000008140      1505           LSm proteins  \n",
       "7  ENSG00000155858  OTTHUMG00000130255      1505           LSm proteins  \n",
       "8  ENSG00000161654  OTTHUMG00000181804      1505           LSm proteins  \n",
       "9  ENSG00000257103  OTTHUMG00000180491      1505           LSm proteins  "
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "spliceosome = pd.read_csv('../annotations/major_spliceosome_genes.txt', sep='\\t', header=0)\n",
    "spliceosome.head(10)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "544addf8",
   "metadata": {},
   "source": [
    "Let's get just the gene names."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "9955009f",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['CASC3', 'EIF4A3', 'MAGOH', 'MAGOHB', 'RBM8A', 'LSM1', 'LSM10',\n",
       "       'LSM11', 'LSM12', 'LSM14A', 'LSM14B', 'LSM2', 'LSM3', 'LSM4',\n",
       "       'LSM5', 'LSM6', 'LSM7', 'LSM8', 'NAA38', 'BCAS2', 'CDC5L',\n",
       "       'CTNNBL1', 'CWC15', 'PLRG1', 'PRPF19', 'SF3A1', 'SF3A2', 'SF3A3',\n",
       "       'PHF5A', 'SF3B1', 'SF3B2', 'SF3B3', 'SF3B4', 'SF3B5', 'SF3B6',\n",
       "       'SNRPB', 'SNRPD1', 'SNRPD2', 'SNRPD3', 'SNRPE', 'SNRPF', 'SNRPG',\n",
       "       'SNRPN', 'CCAR1', 'CHERP', 'DDX46', 'DHX15', 'HNRNPA1', 'HNRNPAB',\n",
       "       'PRPF40A', 'PUF60', 'RBM10', 'RBM17', 'RBM25', 'RBM5', 'SF1',\n",
       "       'SMNDC1', 'SUGP1', 'THRAP3', 'U2AF1', 'U2AF2', 'U2SURP', 'AQR',\n",
       "       'BUD31', 'CRNKL1', 'IK', 'ISY1', 'MFAP1', 'PPIE', 'PPIL1', 'PQBP1',\n",
       "       'PRPF38A', 'RBM22', 'SMU1', 'SNW1', 'TFIP11', 'WBP11', 'WBP4',\n",
       "       'XAB2', 'ZMAT2', 'CCDC12', 'CDC40', 'CWC22', 'CWC25', 'CWC27',\n",
       "       'DHX16', 'EFTUD2', 'GPATCH1', 'GPKOW', 'PPIL2', 'PRCC', 'PRPF8',\n",
       "       'RNF113A', 'RNU5A-1', 'RNU6-1', 'SAP18', 'SNRNP200', 'SNRNP40',\n",
       "       'ZNF830', 'CACTIN', 'CDK10', 'STEEP1', 'DDX41', 'DHX35', 'DHX8',\n",
       "       'FAM32A', 'FAM50A', 'FRA10AC1', 'HNRNPC', 'HSPA8', 'LENG1',\n",
       "       'NOSIP', 'PPIG', 'PPIL3', 'PPWD1', 'PRPF18', 'SDE2', 'SLU7',\n",
       "       'SRRM2', 'SYF2', 'WDR83', 'DDX39B', 'BUD13', 'C9orf78', 'ESS2',\n",
       "       'SNIP1', 'RBM42', 'SART1', 'SNRNP27', 'USP39', 'RNU1-1', 'SNRNP70',\n",
       "       'SNRPA', 'SNRPC', 'RNU2-1', 'SNRPA1', 'SNRPB2', 'PPIH', 'PRPF3',\n",
       "       'PRPF31', 'PRPF4', 'RNU4-1', 'SNU13', 'CD2BP2', 'DDX23', 'PRPF6',\n",
       "       'TXNL4A'], dtype=object)"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "spliceosome_genes = spliceosome['Approved symbol'].unique()\n",
    "spliceosome_genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "764a8927",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "147"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(spliceosome_genes)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0c6c87a9",
   "metadata": {},
   "source": [
    "Now to subset all our variants for just those in spliceosome genes."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "5ad7135b",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref_allele</th>\n",
       "      <th>alt_allele</th>\n",
       "      <th>ancestral_allele</th>\n",
       "      <th>anc_dev</th>\n",
       "      <th>variant_type</th>\n",
       "      <th>altai_gt</th>\n",
       "      <th>chagyrskaya_gt</th>\n",
       "      <th>denisovan_gt</th>\n",
       "      <th>vindija_gt</th>\n",
       "      <th>altai_gt_boolean</th>\n",
       "      <th>chagyrskaya_gt_boolean</th>\n",
       "      <th>denisovan_gt_boolean</th>\n",
       "      <th>vindija_gt_boolean</th>\n",
       "      <th>distribution</th>\n",
       "      <th>present_in_1KG</th>\n",
       "      <th>1KG_allele_count</th>\n",
       "      <th>1KG_allele_number</th>\n",
       "      <th>1KG_allele_frequency</th>\n",
       "      <th>1KG_EAS_AF</th>\n",
       "      <th>1KG_EUR_AF</th>\n",
       "      <th>1KG_AFR_AF</th>\n",
       "      <th>1KG_AMR_AF</th>\n",
       "      <th>1KG_SAS_AF</th>\n",
       "      <th>1KG_non_ASW_AFR_AF</th>\n",
       "      <th>present_in_gnomAD</th>\n",
       "      <th>gnomAD_allele_count</th>\n",
       "      <th>gnomAD_allele_number</th>\n",
       "      <th>gnomAD_allele_frequency</th>\n",
       "      <th>Vernot_introgressed</th>\n",
       "      <th>Vernot_ancestral_allele</th>\n",
       "      <th>Vernot_derived_allele</th>\n",
       "      <th>Vernot_ancestral_derived_code</th>\n",
       "      <th>Vernot_AFA_AF</th>\n",
       "      <th>Vernot_AFR_AF</th>\n",
       "      <th>Vernot_AMR_AF</th>\n",
       "      <th>Vernot_EAS_AF</th>\n",
       "      <th>Vernot_EUR_AF</th>\n",
       "      <th>Vernot_PNG_AF</th>\n",
       "      <th>Vernot_SAS_AF</th>\n",
       "      <th>Vernot_Denisovan_base</th>\n",
       "      <th>Vernot_Neanderthal_base_1</th>\n",
       "      <th>Vernot_Neanderthal_base_2</th>\n",
       "      <th>Vernot_haplotype_tag</th>\n",
       "      <th>Vernot_introgressed_AF</th>\n",
       "      <th>Browning_introgressed</th>\n",
       "      <th>Browning_ref_alt</th>\n",
       "      <th>Browning_introgressed_AF</th>\n",
       "      <th>annotation</th>\n",
       "      <th>mis_oe</th>\n",
       "      <th>mis_z</th>\n",
       "      <th>lof_oe</th>\n",
       "      <th>lof_z</th>\n",
       "      <th>phyloP</th>\n",
       "      <th>ag_delta</th>\n",
       "      <th>al_delta</th>\n",
       "      <th>dg_delta</th>\n",
       "      <th>dl_delta</th>\n",
       "      <th>delta_max</th>\n",
       "      <th>ag_pos</th>\n",
       "      <th>al_pos</th>\n",
       "      <th>dg_pos</th>\n",
       "      <th>dl_pos</th>\n",
       "      <th>Adipose_Subcutaneous</th>\n",
       "      <th>Adipose_Visceral_Omentum</th>\n",
       "      <th>Adrenal_Gland</th>\n",
       "      <th>Artery_Aorta</th>\n",
       "      <th>Artery_Coronary</th>\n",
       "      <th>Artery_Tibial</th>\n",
       "      <th>Brain_Amygdala</th>\n",
       "      <th>Brain_Anterior_cingulate_cortex_BA24</th>\n",
       "      <th>Brain_Caudate_basal_ganglia</th>\n",
       "      <th>Brain_Cerebellar_Hemisphere</th>\n",
       "      <th>Brain_Cerebellum</th>\n",
       "      <th>Brain_Cortex</th>\n",
       "      <th>Brain_Frontal_Cortex_BA9</th>\n",
       "      <th>Brain_Hippocampus</th>\n",
       "      <th>Brain_Hypothalamus</th>\n",
       "      <th>Brain_Nucleus_accumbens_basal_ganglia</th>\n",
       "      <th>Brain_Putamen_basal_ganglia</th>\n",
       "      <th>Brain_Spinal_cord_cervical_c-1</th>\n",
       "      <th>Brain_Substantia_nigra</th>\n",
       "      <th>Breast_Mammary_Tissue</th>\n",
       "      <th>Cells_Cultured_fibroblasts</th>\n",
       "      <th>Cells_EBV-transformed_lymphocytes</th>\n",
       "      <th>Colon_Sigmoid</th>\n",
       "      <th>Colon_Transverse</th>\n",
       "      <th>Esophagus_Gastroesophageal_Junction</th>\n",
       "      <th>Esophagus_Mucosa</th>\n",
       "      <th>Esophagus_Muscularis</th>\n",
       "      <th>Heart_Atrial_Appendage</th>\n",
       "      <th>Heart_Left_Ventricle</th>\n",
       "      <th>Kidney_Cortex</th>\n",
       "      <th>Liver</th>\n",
       "      <th>Lung</th>\n",
       "      <th>Minor_Salivary_Gland</th>\n",
       "      <th>Muscle_Skeletal</th>\n",
       "      <th>Nerve_Tibial</th>\n",
       "      <th>Ovary</th>\n",
       "      <th>Pancreas</th>\n",
       "      <th>Pituitary</th>\n",
       "      <th>Prostate</th>\n",
       "      <th>Skin_Not_Sun_Exposed_Suprapubic</th>\n",
       "      <th>Skin_Sun_Exposed_Lower_leg</th>\n",
       "      <th>Small_Intestine_Terminal_Ileum</th>\n",
       "      <th>Spleen</th>\n",
       "      <th>Stomach</th>\n",
       "      <th>Testis</th>\n",
       "      <th>Thyroid</th>\n",
       "      <th>Uterus</th>\n",
       "      <th>Vagina</th>\n",
       "      <th>Whole_Blood</th>\n",
       "      <th>N_GTEx_tissues</th>\n",
       "      <th>sQTL</th>\n",
       "      <th>Vernot_allele_origin</th>\n",
       "      <th>Browning_allele_origin</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>21285</th>\n",
       "      <td>chr1</td>\n",
       "      <td>25549345</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>yes</td>\n",
       "      <td>37.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.03</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.029240</td>\n",
       "      <td>yes</td>\n",
       "      <td>1116.0</td>\n",
       "      <td>143230.0</td>\n",
       "      <td>0.007792</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SYF2</td>\n",
       "      <td>0.93809</td>\n",
       "      <td>0.25699</td>\n",
       "      <td>0.42189</td>\n",
       "      <td>2.0202</td>\n",
       "      <td>0.650</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>33</td>\n",
       "      <td>-6</td>\n",
       "      <td>48</td>\n",
       "      <td>-5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21286</th>\n",
       "      <td>chr1</td>\n",
       "      <td>25549970</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>g</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>yes</td>\n",
       "      <td>242.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.17</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.173489</td>\n",
       "      <td>yes</td>\n",
       "      <td>6387.0</td>\n",
       "      <td>143170.0</td>\n",
       "      <td>0.044611</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SYF2</td>\n",
       "      <td>0.93809</td>\n",
       "      <td>0.25699</td>\n",
       "      <td>0.42189</td>\n",
       "      <td>2.0202</td>\n",
       "      <td>-0.187</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-48</td>\n",
       "      <td>-30</td>\n",
       "      <td>38</td>\n",
       "      <td>44</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21287</th>\n",
       "      <td>chr1</td>\n",
       "      <td>25550063</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>yes</td>\n",
       "      <td>23.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.018519</td>\n",
       "      <td>yes</td>\n",
       "      <td>752.0</td>\n",
       "      <td>143264.0</td>\n",
       "      <td>0.005249</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SYF2</td>\n",
       "      <td>0.93809</td>\n",
       "      <td>0.25699</td>\n",
       "      <td>0.42189</td>\n",
       "      <td>2.0202</td>\n",
       "      <td>-0.797</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>28</td>\n",
       "      <td>-30</td>\n",
       "      <td>28</td>\n",
       "      <td>-41</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21288</th>\n",
       "      <td>chr1</td>\n",
       "      <td>25551415</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>yes</td>\n",
       "      <td>12.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.010721</td>\n",
       "      <td>yes</td>\n",
       "      <td>317.0</td>\n",
       "      <td>143254.0</td>\n",
       "      <td>0.002213</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SYF2</td>\n",
       "      <td>0.93809</td>\n",
       "      <td>0.25699</td>\n",
       "      <td>0.42189</td>\n",
       "      <td>2.0202</td>\n",
       "      <td>-0.467</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-9</td>\n",
       "      <td>-23</td>\n",
       "      <td>0</td>\n",
       "      <td>-31</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21289</th>\n",
       "      <td>chr1</td>\n",
       "      <td>25554032</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>yes</td>\n",
       "      <td>37.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.03</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.029240</td>\n",
       "      <td>yes</td>\n",
       "      <td>1110.0</td>\n",
       "      <td>143290.0</td>\n",
       "      <td>0.007747</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SYF2</td>\n",
       "      <td>0.93809</td>\n",
       "      <td>0.25699</td>\n",
       "      <td>0.42189</td>\n",
       "      <td>2.0202</td>\n",
       "      <td>0.455</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.01</td>\n",
       "      <td>44</td>\n",
       "      <td>-9</td>\n",
       "      <td>-9</td>\n",
       "      <td>-33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21290</th>\n",
       "      <td>chr1</td>\n",
       "      <td>25554269</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SYF2</td>\n",
       "      <td>0.93809</td>\n",
       "      <td>0.25699</td>\n",
       "      <td>0.42189</td>\n",
       "      <td>2.0202</td>\n",
       "      <td>-0.250</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-47</td>\n",
       "      <td>44</td>\n",
       "      <td>31</td>\n",
       "      <td>44</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21291</th>\n",
       "      <td>chr1</td>\n",
       "      <td>25555672</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>ancestral</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>yes</td>\n",
       "      <td>923.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.18</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.65</td>\n",
       "      <td>0.07</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.663743</td>\n",
       "      <td>yes</td>\n",
       "      <td>24307.0</td>\n",
       "      <td>143150.0</td>\n",
       "      <td>0.169801</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SYF2</td>\n",
       "      <td>0.93809</td>\n",
       "      <td>0.25699</td>\n",
       "      <td>0.42189</td>\n",
       "      <td>2.0202</td>\n",
       "      <td>-0.269</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.02</td>\n",
       "      <td>-29</td>\n",
       "      <td>-50</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>ancient</td>\n",
       "      <td>ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21292</th>\n",
       "      <td>chr1</td>\n",
       "      <td>25556806</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>36.0</td>\n",
       "      <td>142174.0</td>\n",
       "      <td>0.000253</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SYF2</td>\n",
       "      <td>0.93809</td>\n",
       "      <td>0.25699</td>\n",
       "      <td>0.42189</td>\n",
       "      <td>2.0202</td>\n",
       "      <td>0.400</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-4</td>\n",
       "      <td>23</td>\n",
       "      <td>-4</td>\n",
       "      <td>-46</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21293</th>\n",
       "      <td>chr1</td>\n",
       "      <td>25557165</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>g</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>yes</td>\n",
       "      <td>180.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.04</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.13</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.129630</td>\n",
       "      <td>yes</td>\n",
       "      <td>4375.0</td>\n",
       "      <td>143170.0</td>\n",
       "      <td>0.030558</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SYF2</td>\n",
       "      <td>0.93809</td>\n",
       "      <td>0.25699</td>\n",
       "      <td>0.42189</td>\n",
       "      <td>2.0202</td>\n",
       "      <td>-0.279</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.02</td>\n",
       "      <td>-5</td>\n",
       "      <td>-27</td>\n",
       "      <td>-27</td>\n",
       "      <td>-47</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21294</th>\n",
       "      <td>chr1</td>\n",
       "      <td>25557681</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>ancestral</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>yes</td>\n",
       "      <td>470.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.09</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.33</td>\n",
       "      <td>0.03</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.340156</td>\n",
       "      <td>yes</td>\n",
       "      <td>12267.0</td>\n",
       "      <td>143220.0</td>\n",
       "      <td>0.085651</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SYF2</td>\n",
       "      <td>0.93809</td>\n",
       "      <td>0.25699</td>\n",
       "      <td>0.42189</td>\n",
       "      <td>2.0202</td>\n",
       "      <td>-0.971</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-1</td>\n",
       "      <td>-11</td>\n",
       "      <td>-4</td>\n",
       "      <td>-15</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      chrom       pos ref_allele alt_allele ancestral_allele    anc_dev  \\\n",
       "21285  chr1  25549345          G          C                G    derived   \n",
       "21286  chr1  25549970          G          C                g    derived   \n",
       "21287  chr1  25550063          G          A                G    derived   \n",
       "21288  chr1  25551415          G          A                G    derived   \n",
       "21289  chr1  25554032          A          G                A    derived   \n",
       "21290  chr1  25554269          T          A                T    derived   \n",
       "21291  chr1  25555672          C          T                T  ancestral   \n",
       "21292  chr1  25556806          G          T                G    derived   \n",
       "21293  chr1  25557165          G          A                g    derived   \n",
       "21294  chr1  25557681          T          C                C  ancestral   \n",
       "\n",
       "      variant_type altai_gt chagyrskaya_gt denisovan_gt vindija_gt  \\\n",
       "21285          snv      1/1            1/1          0/1        1/1   \n",
       "21286          snv      0/0            0/0          0/1        0/0   \n",
       "21287          snv      1/1            1/1          0/0        1/1   \n",
       "21288          snv      0/0            0/0          0/1        0/0   \n",
       "21289          snv      1/1            1/1          0/1        1/1   \n",
       "21290          snv      0/0            0/0          0/1        0/0   \n",
       "21291          snv      1/1            1/1          1/1        1/1   \n",
       "21292          snv      1/1            1/1          0/0        1/1   \n",
       "21293          snv      0/0            0/0          0/1        0/0   \n",
       "21294          snv      1/1            1/1          1/1        1/1   \n",
       "\n",
       "       altai_gt_boolean  chagyrskaya_gt_boolean  denisovan_gt_boolean  \\\n",
       "21285              True                    True                  True   \n",
       "21286             False                   False                  True   \n",
       "21287              True                    True                 False   \n",
       "21288             False                   False                  True   \n",
       "21289              True                    True                  True   \n",
       "21290             False                   False                  True   \n",
       "21291              True                    True                  True   \n",
       "21292              True                    True                 False   \n",
       "21293             False                   False                  True   \n",
       "21294              True                    True                  True   \n",
       "\n",
       "       vindija_gt_boolean distribution present_in_1KG  1KG_allele_count  \\\n",
       "21285                True       Shared            yes              37.0   \n",
       "21286               False    Denisovan            yes             242.0   \n",
       "21287                True  Neanderthal            yes              23.0   \n",
       "21288               False    Denisovan            yes              12.0   \n",
       "21289                True       Shared            yes              37.0   \n",
       "21290               False    Denisovan             no               NaN   \n",
       "21291                True       Shared            yes             923.0   \n",
       "21292                True  Neanderthal             no               NaN   \n",
       "21293               False    Denisovan            yes             180.0   \n",
       "21294                True       Shared            yes             470.0   \n",
       "\n",
       "       1KG_allele_number  1KG_allele_frequency  1KG_EAS_AF  1KG_EUR_AF  \\\n",
       "21285             5096.0                  0.01         0.0        0.00   \n",
       "21286             5096.0                  0.05         0.0        0.00   \n",
       "21287             5096.0                  0.00         0.0        0.00   \n",
       "21288             5096.0                  0.00         0.0        0.00   \n",
       "21289             5096.0                  0.01         0.0        0.00   \n",
       "21290                NaN                   NaN         NaN         NaN   \n",
       "21291             5096.0                  0.18         0.0        0.01   \n",
       "21292                NaN                   NaN         NaN         NaN   \n",
       "21293             5096.0                  0.04         0.0        0.00   \n",
       "21294             5096.0                  0.09         0.0        0.00   \n",
       "\n",
       "       1KG_AFR_AF  1KG_AMR_AF  1KG_SAS_AF  1KG_non_ASW_AFR_AF  \\\n",
       "21285        0.03        0.00         0.0            0.029240   \n",
       "21286        0.17        0.01         0.0            0.173489   \n",
       "21287        0.02        0.00         0.0            0.018519   \n",
       "21288        0.01        0.00         0.0            0.010721   \n",
       "21289        0.03        0.00         0.0            0.029240   \n",
       "21290         NaN         NaN         NaN                 NaN   \n",
       "21291        0.65        0.07         0.0            0.663743   \n",
       "21292         NaN         NaN         NaN                 NaN   \n",
       "21293        0.13        0.01         0.0            0.129630   \n",
       "21294        0.33        0.03         0.0            0.340156   \n",
       "\n",
       "      present_in_gnomAD  gnomAD_allele_count  gnomAD_allele_number  \\\n",
       "21285               yes               1116.0              143230.0   \n",
       "21286               yes               6387.0              143170.0   \n",
       "21287               yes                752.0              143264.0   \n",
       "21288               yes                317.0              143254.0   \n",
       "21289               yes               1110.0              143290.0   \n",
       "21290                no                  NaN                   NaN   \n",
       "21291               yes              24307.0              143150.0   \n",
       "21292               yes                 36.0              142174.0   \n",
       "21293               yes               4375.0              143170.0   \n",
       "21294               yes              12267.0              143220.0   \n",
       "\n",
       "       gnomAD_allele_frequency Vernot_introgressed Vernot_ancestral_allele  \\\n",
       "21285                 0.007792                  no                     NaN   \n",
       "21286                 0.044611                  no                     NaN   \n",
       "21287                 0.005249                  no                     NaN   \n",
       "21288                 0.002213                  no                     NaN   \n",
       "21289                 0.007747                  no                     NaN   \n",
       "21290                      NaN                  no                     NaN   \n",
       "21291                 0.169801                  no                     NaN   \n",
       "21292                 0.000253                  no                     NaN   \n",
       "21293                 0.030558                  no                     NaN   \n",
       "21294                 0.085651                  no                     NaN   \n",
       "\n",
       "      Vernot_derived_allele  Vernot_ancestral_derived_code  Vernot_AFA_AF  \\\n",
       "21285                   NaN                            NaN            NaN   \n",
       "21286                   NaN                            NaN            NaN   \n",
       "21287                   NaN                            NaN            NaN   \n",
       "21288                   NaN                            NaN            NaN   \n",
       "21289                   NaN                            NaN            NaN   \n",
       "21290                   NaN                            NaN            NaN   \n",
       "21291                   NaN                            NaN            NaN   \n",
       "21292                   NaN                            NaN            NaN   \n",
       "21293                   NaN                            NaN            NaN   \n",
       "21294                   NaN                            NaN            NaN   \n",
       "\n",
       "       Vernot_AFR_AF  Vernot_AMR_AF  Vernot_EAS_AF  Vernot_EUR_AF  \\\n",
       "21285            NaN            NaN            NaN            NaN   \n",
       "21286            NaN            NaN            NaN            NaN   \n",
       "21287            NaN            NaN            NaN            NaN   \n",
       "21288            NaN            NaN            NaN            NaN   \n",
       "21289            NaN            NaN            NaN            NaN   \n",
       "21290            NaN            NaN            NaN            NaN   \n",
       "21291            NaN            NaN            NaN            NaN   \n",
       "21292            NaN            NaN            NaN            NaN   \n",
       "21293            NaN            NaN            NaN            NaN   \n",
       "21294            NaN            NaN            NaN            NaN   \n",
       "\n",
       "       Vernot_PNG_AF  Vernot_SAS_AF Vernot_Denisovan_base  \\\n",
       "21285            NaN            NaN                   NaN   \n",
       "21286            NaN            NaN                   NaN   \n",
       "21287            NaN            NaN                   NaN   \n",
       "21288            NaN            NaN                   NaN   \n",
       "21289            NaN            NaN                   NaN   \n",
       "21290            NaN            NaN                   NaN   \n",
       "21291            NaN            NaN                   NaN   \n",
       "21292            NaN            NaN                   NaN   \n",
       "21293            NaN            NaN                   NaN   \n",
       "21294            NaN            NaN                   NaN   \n",
       "\n",
       "      Vernot_Neanderthal_base_1 Vernot_Neanderthal_base_2  \\\n",
       "21285                       NaN                       NaN   \n",
       "21286                       NaN                       NaN   \n",
       "21287                       NaN                       NaN   \n",
       "21288                       NaN                       NaN   \n",
       "21289                       NaN                       NaN   \n",
       "21290                       NaN                       NaN   \n",
       "21291                       NaN                       NaN   \n",
       "21292                       NaN                       NaN   \n",
       "21293                       NaN                       NaN   \n",
       "21294                       NaN                       NaN   \n",
       "\n",
       "      Vernot_haplotype_tag  Vernot_introgressed_AF Browning_introgressed  \\\n",
       "21285                  NaN                     NaN                    no   \n",
       "21286                  NaN                     NaN                    no   \n",
       "21287                  NaN                     NaN                    no   \n",
       "21288                  NaN                     NaN                    no   \n",
       "21289                  NaN                     NaN                    no   \n",
       "21290                  NaN                     NaN                    no   \n",
       "21291                  NaN                     NaN                    no   \n",
       "21292                  NaN                     NaN                    no   \n",
       "21293                  NaN                     NaN                    no   \n",
       "21294                  NaN                     NaN                    no   \n",
       "\n",
       "       Browning_ref_alt  Browning_introgressed_AF annotation   mis_oe  \\\n",
       "21285               NaN                       NaN       SYF2  0.93809   \n",
       "21286               NaN                       NaN       SYF2  0.93809   \n",
       "21287               NaN                       NaN       SYF2  0.93809   \n",
       "21288               NaN                       NaN       SYF2  0.93809   \n",
       "21289               NaN                       NaN       SYF2  0.93809   \n",
       "21290               NaN                       NaN       SYF2  0.93809   \n",
       "21291               NaN                       NaN       SYF2  0.93809   \n",
       "21292               NaN                       NaN       SYF2  0.93809   \n",
       "21293               NaN                       NaN       SYF2  0.93809   \n",
       "21294               NaN                       NaN       SYF2  0.93809   \n",
       "\n",
       "         mis_z   lof_oe   lof_z  phyloP  ag_delta  al_delta  dg_delta  \\\n",
       "21285  0.25699  0.42189  2.0202   0.650      0.00       0.0       0.0   \n",
       "21286  0.25699  0.42189  2.0202  -0.187      0.00       0.0       0.0   \n",
       "21287  0.25699  0.42189  2.0202  -0.797      0.00       0.0       0.0   \n",
       "21288  0.25699  0.42189  2.0202  -0.467      0.00       0.0       0.0   \n",
       "21289  0.25699  0.42189  2.0202   0.455      0.01       0.0       0.0   \n",
       "21290  0.25699  0.42189  2.0202  -0.250      0.00       0.0       0.0   \n",
       "21291  0.25699  0.42189  2.0202  -0.269      0.02       0.0       0.0   \n",
       "21292  0.25699  0.42189  2.0202   0.400      0.00       0.0       0.0   \n",
       "21293  0.25699  0.42189  2.0202  -0.279      0.00       0.0       0.0   \n",
       "21294  0.25699  0.42189  2.0202  -0.971      0.00       0.0       0.0   \n",
       "\n",
       "       dl_delta  delta_max  ag_pos  al_pos  dg_pos  dl_pos  \\\n",
       "21285      0.00       0.00      33      -6      48      -5   \n",
       "21286      0.00       0.00     -48     -30      38      44   \n",
       "21287      0.00       0.00      28     -30      28     -41   \n",
       "21288      0.00       0.00      -9     -23       0     -31   \n",
       "21289      0.00       0.01      44      -9      -9     -33   \n",
       "21290      0.00       0.00     -47      44      31      44   \n",
       "21291      0.00       0.02     -29     -50       6       1   \n",
       "21292      0.00       0.00      -4      23      -4     -46   \n",
       "21293      0.02       0.02      -5     -27     -27     -47   \n",
       "21294      0.00       0.00      -1     -11      -4     -15   \n",
       "\n",
       "       Adipose_Subcutaneous  Adipose_Visceral_Omentum  Adrenal_Gland  \\\n",
       "21285                   NaN                       NaN            NaN   \n",
       "21286                   NaN                       NaN            NaN   \n",
       "21287                   NaN                       NaN            NaN   \n",
       "21288                   NaN                       NaN            NaN   \n",
       "21289                   NaN                       NaN            NaN   \n",
       "21290                   NaN                       NaN            NaN   \n",
       "21291                   NaN                       NaN            NaN   \n",
       "21292                   NaN                       NaN            NaN   \n",
       "21293                   NaN                       NaN            NaN   \n",
       "21294                   NaN                       NaN            NaN   \n",
       "\n",
       "       Artery_Aorta  Artery_Coronary  Artery_Tibial  Brain_Amygdala  \\\n",
       "21285           NaN              NaN            NaN             NaN   \n",
       "21286           NaN              NaN            NaN             NaN   \n",
       "21287           NaN              NaN            NaN             NaN   \n",
       "21288           NaN              NaN            NaN             NaN   \n",
       "21289           NaN              NaN            NaN             NaN   \n",
       "21290           NaN              NaN            NaN             NaN   \n",
       "21291           NaN              NaN            NaN             NaN   \n",
       "21292           NaN              NaN            NaN             NaN   \n",
       "21293           NaN              NaN            NaN             NaN   \n",
       "21294           NaN              NaN            NaN             NaN   \n",
       "\n",
       "       Brain_Anterior_cingulate_cortex_BA24  Brain_Caudate_basal_ganglia  \\\n",
       "21285                                   NaN                          NaN   \n",
       "21286                                   NaN                          NaN   \n",
       "21287                                   NaN                          NaN   \n",
       "21288                                   NaN                          NaN   \n",
       "21289                                   NaN                          NaN   \n",
       "21290                                   NaN                          NaN   \n",
       "21291                                   NaN                          NaN   \n",
       "21292                                   NaN                          NaN   \n",
       "21293                                   NaN                          NaN   \n",
       "21294                                   NaN                          NaN   \n",
       "\n",
       "       Brain_Cerebellar_Hemisphere  Brain_Cerebellum  Brain_Cortex  \\\n",
       "21285                          NaN               NaN           NaN   \n",
       "21286                          NaN               NaN           NaN   \n",
       "21287                          NaN               NaN           NaN   \n",
       "21288                          NaN               NaN           NaN   \n",
       "21289                          NaN               NaN           NaN   \n",
       "21290                          NaN               NaN           NaN   \n",
       "21291                          NaN               NaN           NaN   \n",
       "21292                          NaN               NaN           NaN   \n",
       "21293                          NaN               NaN           NaN   \n",
       "21294                          NaN               NaN           NaN   \n",
       "\n",
       "       Brain_Frontal_Cortex_BA9  Brain_Hippocampus  Brain_Hypothalamus  \\\n",
       "21285                       NaN                NaN                 NaN   \n",
       "21286                       NaN                NaN                 NaN   \n",
       "21287                       NaN                NaN                 NaN   \n",
       "21288                       NaN                NaN                 NaN   \n",
       "21289                       NaN                NaN                 NaN   \n",
       "21290                       NaN                NaN                 NaN   \n",
       "21291                       NaN                NaN                 NaN   \n",
       "21292                       NaN                NaN                 NaN   \n",
       "21293                       NaN                NaN                 NaN   \n",
       "21294                       NaN                NaN                 NaN   \n",
       "\n",
       "       Brain_Nucleus_accumbens_basal_ganglia  Brain_Putamen_basal_ganglia  \\\n",
       "21285                                    NaN                          NaN   \n",
       "21286                                    NaN                          NaN   \n",
       "21287                                    NaN                          NaN   \n",
       "21288                                    NaN                          NaN   \n",
       "21289                                    NaN                          NaN   \n",
       "21290                                    NaN                          NaN   \n",
       "21291                                    NaN                          NaN   \n",
       "21292                                    NaN                          NaN   \n",
       "21293                                    NaN                          NaN   \n",
       "21294                                    NaN                          NaN   \n",
       "\n",
       "       Brain_Spinal_cord_cervical_c-1  Brain_Substantia_nigra  \\\n",
       "21285                             NaN                     NaN   \n",
       "21286                             NaN                     NaN   \n",
       "21287                             NaN                     NaN   \n",
       "21288                             NaN                     NaN   \n",
       "21289                             NaN                     NaN   \n",
       "21290                             NaN                     NaN   \n",
       "21291                             NaN                     NaN   \n",
       "21292                             NaN                     NaN   \n",
       "21293                             NaN                     NaN   \n",
       "21294                             NaN                     NaN   \n",
       "\n",
       "       Breast_Mammary_Tissue  Cells_Cultured_fibroblasts  \\\n",
       "21285                    NaN                         NaN   \n",
       "21286                    NaN                         NaN   \n",
       "21287                    NaN                         NaN   \n",
       "21288                    NaN                         NaN   \n",
       "21289                    NaN                         NaN   \n",
       "21290                    NaN                         NaN   \n",
       "21291                    NaN                         NaN   \n",
       "21292                    NaN                         NaN   \n",
       "21293                    NaN                         NaN   \n",
       "21294                    NaN                         NaN   \n",
       "\n",
       "       Cells_EBV-transformed_lymphocytes  Colon_Sigmoid  Colon_Transverse  \\\n",
       "21285                                NaN            NaN               NaN   \n",
       "21286                                NaN            NaN               NaN   \n",
       "21287                                NaN            NaN               NaN   \n",
       "21288                                NaN            NaN               NaN   \n",
       "21289                                NaN            NaN               NaN   \n",
       "21290                                NaN            NaN               NaN   \n",
       "21291                                NaN            NaN               NaN   \n",
       "21292                                NaN            NaN               NaN   \n",
       "21293                                NaN            NaN               NaN   \n",
       "21294                                NaN            NaN               NaN   \n",
       "\n",
       "       Esophagus_Gastroesophageal_Junction  Esophagus_Mucosa  \\\n",
       "21285                                  NaN               NaN   \n",
       "21286                                  NaN               NaN   \n",
       "21287                                  NaN               NaN   \n",
       "21288                                  NaN               NaN   \n",
       "21289                                  NaN               NaN   \n",
       "21290                                  NaN               NaN   \n",
       "21291                                  NaN               NaN   \n",
       "21292                                  NaN               NaN   \n",
       "21293                                  NaN               NaN   \n",
       "21294                                  NaN               NaN   \n",
       "\n",
       "       Esophagus_Muscularis  Heart_Atrial_Appendage  Heart_Left_Ventricle  \\\n",
       "21285                   NaN                     NaN                   NaN   \n",
       "21286                   NaN                     NaN                   NaN   \n",
       "21287                   NaN                     NaN                   NaN   \n",
       "21288                   NaN                     NaN                   NaN   \n",
       "21289                   NaN                     NaN                   NaN   \n",
       "21290                   NaN                     NaN                   NaN   \n",
       "21291                   NaN                     NaN                   NaN   \n",
       "21292                   NaN                     NaN                   NaN   \n",
       "21293                   NaN                     NaN                   NaN   \n",
       "21294                   NaN                     NaN                   NaN   \n",
       "\n",
       "       Kidney_Cortex  Liver  Lung  Minor_Salivary_Gland  Muscle_Skeletal  \\\n",
       "21285            NaN    NaN   NaN                   NaN              NaN   \n",
       "21286            NaN    NaN   NaN                   NaN              NaN   \n",
       "21287            NaN    NaN   NaN                   NaN              NaN   \n",
       "21288            NaN    NaN   NaN                   NaN              NaN   \n",
       "21289            NaN    NaN   NaN                   NaN              NaN   \n",
       "21290            NaN    NaN   NaN                   NaN              NaN   \n",
       "21291            NaN    NaN   NaN                   NaN              NaN   \n",
       "21292            NaN    NaN   NaN                   NaN              NaN   \n",
       "21293            NaN    NaN   NaN                   NaN              NaN   \n",
       "21294            NaN    NaN   NaN                   NaN              NaN   \n",
       "\n",
       "       Nerve_Tibial  Ovary  Pancreas  Pituitary  Prostate  \\\n",
       "21285           NaN    NaN       NaN        NaN       NaN   \n",
       "21286           NaN    NaN       NaN        NaN       NaN   \n",
       "21287           NaN    NaN       NaN        NaN       NaN   \n",
       "21288           NaN    NaN       NaN        NaN       NaN   \n",
       "21289           NaN    NaN       NaN        NaN       NaN   \n",
       "21290           NaN    NaN       NaN        NaN       NaN   \n",
       "21291           NaN    NaN       NaN        NaN       NaN   \n",
       "21292           NaN    NaN       NaN        NaN       NaN   \n",
       "21293           NaN    NaN       NaN        NaN       NaN   \n",
       "21294           NaN    NaN       NaN        NaN       NaN   \n",
       "\n",
       "       Skin_Not_Sun_Exposed_Suprapubic  Skin_Sun_Exposed_Lower_leg  \\\n",
       "21285                              NaN                         NaN   \n",
       "21286                              NaN                         NaN   \n",
       "21287                              NaN                         NaN   \n",
       "21288                              NaN                         NaN   \n",
       "21289                              NaN                         NaN   \n",
       "21290                              NaN                         NaN   \n",
       "21291                              NaN                         NaN   \n",
       "21292                              NaN                         NaN   \n",
       "21293                              NaN                         NaN   \n",
       "21294                              NaN                         NaN   \n",
       "\n",
       "       Small_Intestine_Terminal_Ileum  Spleen  Stomach  Testis  Thyroid  \\\n",
       "21285                             NaN     NaN      NaN     NaN      NaN   \n",
       "21286                             NaN     NaN      NaN     NaN      NaN   \n",
       "21287                             NaN     NaN      NaN     NaN      NaN   \n",
       "21288                             NaN     NaN      NaN     NaN      NaN   \n",
       "21289                             NaN     NaN      NaN     NaN      NaN   \n",
       "21290                             NaN     NaN      NaN     NaN      NaN   \n",
       "21291                             NaN     NaN      NaN     NaN      NaN   \n",
       "21292                             NaN     NaN      NaN     NaN      NaN   \n",
       "21293                             NaN     NaN      NaN     NaN      NaN   \n",
       "21294                             NaN     NaN      NaN     NaN      NaN   \n",
       "\n",
       "       Uterus  Vagina  Whole_Blood  N_GTEx_tissues sQTL  \\\n",
       "21285     NaN     NaN          NaN             NaN   no   \n",
       "21286     NaN     NaN          NaN             NaN   no   \n",
       "21287     NaN     NaN          NaN             NaN   no   \n",
       "21288     NaN     NaN          NaN             NaN   no   \n",
       "21289     NaN     NaN          NaN             NaN   no   \n",
       "21290     NaN     NaN          NaN             NaN   no   \n",
       "21291     NaN     NaN          NaN             NaN   no   \n",
       "21292     NaN     NaN          NaN             NaN   no   \n",
       "21293     NaN     NaN          NaN             NaN   no   \n",
       "21294     NaN     NaN          NaN             NaN   no   \n",
       "\n",
       "         Vernot_allele_origin  Browning_allele_origin  \n",
       "21285  low-confidence ancient  low-confidence ancient  \n",
       "21286  low-confidence ancient  low-confidence ancient  \n",
       "21287  low-confidence ancient  low-confidence ancient  \n",
       "21288  low-confidence ancient  low-confidence ancient  \n",
       "21289  low-confidence ancient  low-confidence ancient  \n",
       "21290        archaic-specific        archaic-specific  \n",
       "21291                 ancient                 ancient  \n",
       "21292  low-confidence ancient  low-confidence ancient  \n",
       "21293  low-confidence ancient  low-confidence ancient  \n",
       "21294  low-confidence ancient  low-confidence ancient  "
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "spliceosome_variants = data[data['annotation'].isin(spliceosome_genes)]\n",
    "spliceosome_variants.head(10)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "32a34726",
   "metadata": {},
   "source": [
    "How many variants did we get?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "9f6600eb",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "4571"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(spliceosome_variants.groupby(['chrom','pos']))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0f8ae130",
   "metadata": {},
   "source": [
    "How many variants are uniquely archaic (i.e., absent from 1KG and gnomAD)?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "2e881db6",
   "metadata": {},
   "outputs": [],
   "source": [
    "archaic_specific_spliceosome_variants = spliceosome_variants[spliceosome_variants['Vernot_allele_origin'] == 'archaic-specific']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "d805c0a9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1746"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(archaic_specific_spliceosome_variants.groupby(['chrom','pos']))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cf4dde0f",
   "metadata": {},
   "source": [
    "Now let's format the data to run through Ensembl's Variant Effect Predictor. We will do this will all archaic-specific variants as well as all variants (archaic-specific or not) per archaic individual."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "da2e8eda",
   "metadata": {},
   "outputs": [],
   "source": [
    "spliceosome_variants_for_VEP = spliceosome_variants[['chrom','pos','ref_allele','alt_allele','altai_gt_boolean','chagyrskaya_gt_boolean','denisovan_gt_boolean','vindija_gt_boolean']].copy()\n",
    "spliceosome_variants_for_VEP['null'] = '.'\n",
    "spliceosome_variants_for_VEP['chrom'] = spliceosome_variants_for_VEP['chrom'].str.replace('chr', '')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "ca808f26",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2907"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "altai_spliceosome_variants_for_VEP = spliceosome_variants_for_VEP[spliceosome_variants_for_VEP['altai_gt_boolean'] == True]\n",
    "altai_spliceosome_variants_for_VEP = altai_spliceosome_variants_for_VEP[['chrom','pos','null','ref_allele','alt_allele','null','null','null']]\n",
    "len(altai_spliceosome_variants_for_VEP)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "8ba21073",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2831"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "chagyrskaya_spliceosome_variants_for_VEP = spliceosome_variants_for_VEP[spliceosome_variants_for_VEP['chagyrskaya_gt_boolean'] == True]\n",
    "chagyrskaya_spliceosome_variants_for_VEP = chagyrskaya_spliceosome_variants_for_VEP[['chrom','pos','null','ref_allele','alt_allele','null','null','null']]\n",
    "len(chagyrskaya_spliceosome_variants_for_VEP)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "a2aa7816",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2973"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "denisovan_spliceosome_variants_for_VEP = spliceosome_variants_for_VEP[spliceosome_variants_for_VEP['denisovan_gt_boolean'] == True]\n",
    "denisovan_spliceosome_variants_for_VEP = denisovan_spliceosome_variants_for_VEP[['chrom','pos','null','ref_allele','alt_allele','null','null','null']]\n",
    "len(denisovan_spliceosome_variants_for_VEP)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "0746dd23",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2846"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vindija_spliceosome_variants_for_VEP = spliceosome_variants_for_VEP[spliceosome_variants_for_VEP['vindija_gt_boolean'] == True]\n",
    "vindija_spliceosome_variants_for_VEP = vindija_spliceosome_variants_for_VEP[['chrom','pos','null','ref_allele','alt_allele','null','null','null']]\n",
    "len(vindija_spliceosome_variants_for_VEP)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "322beffe",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>pos</th>\n",
       "      <th>null</th>\n",
       "      <th>ref_allele</th>\n",
       "      <th>alt_allele</th>\n",
       "      <th>null</th>\n",
       "      <th>null</th>\n",
       "      <th>null</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>21290</th>\n",
       "      <td>1</td>\n",
       "      <td>25554269</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24432</th>\n",
       "      <td>1</td>\n",
       "      <td>31734327</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24433</th>\n",
       "      <td>1</td>\n",
       "      <td>31735461</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24440</th>\n",
       "      <td>1</td>\n",
       "      <td>31741832</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24442</th>\n",
       "      <td>1</td>\n",
       "      <td>31743354</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24443</th>\n",
       "      <td>1</td>\n",
       "      <td>31743490</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24446</th>\n",
       "      <td>1</td>\n",
       "      <td>31745979</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24451</th>\n",
       "      <td>1</td>\n",
       "      <td>31751687</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24454</th>\n",
       "      <td>1</td>\n",
       "      <td>31754447</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24458</th>\n",
       "      <td>1</td>\n",
       "      <td>31759381</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      chrom       pos null ref_allele alt_allele null null null\n",
       "21290     1  25554269    .          T          A    .    .    .\n",
       "24432     1  31734327    .          T          C    .    .    .\n",
       "24433     1  31735461    .          G          A    .    .    .\n",
       "24440     1  31741832    .          C          T    .    .    .\n",
       "24442     1  31743354    .          C          T    .    .    .\n",
       "24443     1  31743490    .          T          C    .    .    .\n",
       "24446     1  31745979    .          T          G    .    .    .\n",
       "24451     1  31751687    .          G          A    .    .    .\n",
       "24454     1  31754447    .          C          T    .    .    .\n",
       "24458     1  31759381    .          A          G    .    .    ."
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "archaic_specific_spliceosome_variants_for_VEP = archaic_specific_spliceosome_variants[['chrom','pos','ref_allele','alt_allele']].copy()\n",
    "archaic_specific_spliceosome_variants_for_VEP['null'] = '.'\n",
    "archaic_specific_spliceosome_variants_for_VEP['chrom'] = archaic_specific_spliceosome_variants_for_VEP['chrom'].str.replace('chr', '')\n",
    "archaic_specific_spliceosome_variants_for_VEP = archaic_specific_spliceosome_variants_for_VEP[['chrom','pos','null','ref_allele','alt_allele','null','null','null']]\n",
    "archaic_specific_spliceosome_variants_for_VEP.head(10)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6774a7c3",
   "metadata": {},
   "source": [
    "Save the files."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "82a05b1c",
   "metadata": {},
   "outputs": [],
   "source": [
    "altai_spliceosome_variants_for_VEP.to_csv('../spliceosome/altai_spliceosome_variants_for_VEP.txt', sep=\"\\t\", header = False, index = False)\n",
    "chagyrskaya_spliceosome_variants_for_VEP.to_csv('../spliceosome/chagyrskaya_spliceosome_variants_for_VEP.txt', sep=\"\\t\", header = False, index = False)\n",
    "denisovan_spliceosome_variants_for_VEP.to_csv('../spliceosome/denisovan_spliceosome_variants_for_VEP.txt', sep=\"\\t\", header = False, index = False)\n",
    "vindija_spliceosome_variants_for_VEP.to_csv('../spliceosome/vindija_spliceosome_variants_for_VEP.txt', sep=\"\\t\", header = False, index = False)\n",
    "archaic_specific_spliceosome_variants_for_VEP.to_csv('../spliceosome/archaic_specific_spliceosome_variants_for_VEP.txt', sep=\"\\t\", header = False, index = False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7876bbc0",
   "metadata": {},
   "source": [
    "Run this file through the Ensembl Variant Effect Predictor and upload the results to the directory. "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f57ca2f7",
   "metadata": {},
   "source": [
    "Let's examine the archaic-specific results here."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "3ac64b16",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Uploaded_variation</th>\n",
       "      <th>Location</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Feature</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>cDNA_position</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>Codons</th>\n",
       "      <th>Existing_variation</th>\n",
       "      <th>DISTANCE</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>FLAGS</th>\n",
       "      <th>SYMBOL_SOURCE</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>MANE_SELECT</th>\n",
       "      <th>MANE_PLUS_CLINICAL</th>\n",
       "      <th>TSL</th>\n",
       "      <th>APPRIS</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>SOMATIC</th>\n",
       "      <th>PHENO</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>MOTIF_NAME</th>\n",
       "      <th>MOTIF_POS</th>\n",
       "      <th>HIGH_INF_POS</th>\n",
       "      <th>MOTIF_SCORE_CHANGE</th>\n",
       "      <th>TRANSCRIPTION_FACTORS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>.</td>\n",
       "      <td>1:25554269-25554269</td>\n",
       "      <td>A</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>SYF2</td>\n",
       "      <td>ENSG00000117614</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000236273.4</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>-</td>\n",
       "      <td>4/6</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-1</td>\n",
       "      <td>-</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>19824</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>.</td>\n",
       "      <td>1:25554269-25554269</td>\n",
       "      <td>A</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>SYF2</td>\n",
       "      <td>ENSG00000117614</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000354361.3</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>-</td>\n",
       "      <td>3/5</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-1</td>\n",
       "      <td>-</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>19824</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>.</td>\n",
       "      <td>1:25554269-25554269</td>\n",
       "      <td>A</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>SYF2</td>\n",
       "      <td>ENSG00000117614</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000474160.1</td>\n",
       "      <td>processed_transcript</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>1237</td>\n",
       "      <td>-1</td>\n",
       "      <td>-</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>19824</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>.</td>\n",
       "      <td>1:25554269-25554269</td>\n",
       "      <td>A</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>SYF2</td>\n",
       "      <td>ENSG00000117614</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000476231.1</td>\n",
       "      <td>processed_transcript</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>2531</td>\n",
       "      <td>-1</td>\n",
       "      <td>-</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>19824</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>.</td>\n",
       "      <td>1:31734327-31734327</td>\n",
       "      <td>C</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>SNRNP40</td>\n",
       "      <td>ENSG00000060688</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000263694.4</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>-</td>\n",
       "      <td>9/9</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-1</td>\n",
       "      <td>-</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>30857</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>.</td>\n",
       "      <td>1:31734327-31734327</td>\n",
       "      <td>C</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>SNRNP40</td>\n",
       "      <td>ENSG00000060688</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000373720.3</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>-</td>\n",
       "      <td>3/3</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-1</td>\n",
       "      <td>-</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>30857</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>.</td>\n",
       "      <td>1:31734327-31734327</td>\n",
       "      <td>C</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>SNRNP40</td>\n",
       "      <td>ENSG00000060688</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000446633.2</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>-</td>\n",
       "      <td>9/10</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-1</td>\n",
       "      <td>-</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>30857</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>.</td>\n",
       "      <td>1:31734327-31734327</td>\n",
       "      <td>C</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>RP11-490K7.4</td>\n",
       "      <td>ENSG00000229447</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000452686.2</td>\n",
       "      <td>processed_pseudogene</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>1765</td>\n",
       "      <td>-1</td>\n",
       "      <td>-</td>\n",
       "      <td>Clone_based_vega_gene</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>.</td>\n",
       "      <td>1:31734327-31734327</td>\n",
       "      <td>C</td>\n",
       "      <td>intron_variant,non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>SNRNP40</td>\n",
       "      <td>ENSG00000060688</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000486941.1</td>\n",
       "      <td>processed_transcript</td>\n",
       "      <td>-</td>\n",
       "      <td>1/1</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-1</td>\n",
       "      <td>-</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>30857</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>.</td>\n",
       "      <td>1:31734327-31734327</td>\n",
       "      <td>C</td>\n",
       "      <td>intron_variant,non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>SNRNP40</td>\n",
       "      <td>ENSG00000060688</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000489853.1</td>\n",
       "      <td>processed_transcript</td>\n",
       "      <td>-</td>\n",
       "      <td>5/5</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-1</td>\n",
       "      <td>-</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>30857</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  #Uploaded_variation             Location Allele  \\\n",
       "0                   .  1:25554269-25554269      A   \n",
       "1                   .  1:25554269-25554269      A   \n",
       "2                   .  1:25554269-25554269      A   \n",
       "3                   .  1:25554269-25554269      A   \n",
       "4                   .  1:31734327-31734327      C   \n",
       "5                   .  1:31734327-31734327      C   \n",
       "6                   .  1:31734327-31734327      C   \n",
       "7                   .  1:31734327-31734327      C   \n",
       "8                   .  1:31734327-31734327      C   \n",
       "9                   .  1:31734327-31734327      C   \n",
       "\n",
       "                                    Consequence    IMPACT        SYMBOL  \\\n",
       "0                                intron_variant  MODIFIER          SYF2   \n",
       "1                                intron_variant  MODIFIER          SYF2   \n",
       "2                       downstream_gene_variant  MODIFIER          SYF2   \n",
       "3                       downstream_gene_variant  MODIFIER          SYF2   \n",
       "4                                intron_variant  MODIFIER       SNRNP40   \n",
       "5                                intron_variant  MODIFIER       SNRNP40   \n",
       "6                                intron_variant  MODIFIER       SNRNP40   \n",
       "7                       downstream_gene_variant  MODIFIER  RP11-490K7.4   \n",
       "8  intron_variant,non_coding_transcript_variant  MODIFIER       SNRNP40   \n",
       "9  intron_variant,non_coding_transcript_variant  MODIFIER       SNRNP40   \n",
       "\n",
       "              Gene Feature_type            Feature               BIOTYPE EXON  \\\n",
       "0  ENSG00000117614   Transcript  ENST00000236273.4        protein_coding    -   \n",
       "1  ENSG00000117614   Transcript  ENST00000354361.3        protein_coding    -   \n",
       "2  ENSG00000117614   Transcript  ENST00000474160.1  processed_transcript    -   \n",
       "3  ENSG00000117614   Transcript  ENST00000476231.1  processed_transcript    -   \n",
       "4  ENSG00000060688   Transcript  ENST00000263694.4        protein_coding    -   \n",
       "5  ENSG00000060688   Transcript  ENST00000373720.3        protein_coding    -   \n",
       "6  ENSG00000060688   Transcript  ENST00000446633.2        protein_coding    -   \n",
       "7  ENSG00000229447   Transcript  ENST00000452686.2  processed_pseudogene    -   \n",
       "8  ENSG00000060688   Transcript  ENST00000486941.1  processed_transcript    -   \n",
       "9  ENSG00000060688   Transcript  ENST00000489853.1  processed_transcript    -   \n",
       "\n",
       "  INTRON HGVSc HGVSp cDNA_position CDS_position Protein_position Amino_acids  \\\n",
       "0    4/6     -     -             -            -                -           -   \n",
       "1    3/5     -     -             -            -                -           -   \n",
       "2      -     -     -             -            -                -           -   \n",
       "3      -     -     -             -            -                -           -   \n",
       "4    9/9     -     -             -            -                -           -   \n",
       "5    3/3     -     -             -            -                -           -   \n",
       "6   9/10     -     -             -            -                -           -   \n",
       "7      -     -     -             -            -                -           -   \n",
       "8    1/1     -     -             -            -                -           -   \n",
       "9    5/5     -     -             -            -                -           -   \n",
       "\n",
       "  Codons Existing_variation DISTANCE STRAND FLAGS          SYMBOL_SOURCE  \\\n",
       "0      -                  -        -     -1     -                   HGNC   \n",
       "1      -                  -        -     -1     -                   HGNC   \n",
       "2      -                  -     1237     -1     -                   HGNC   \n",
       "3      -                  -     2531     -1     -                   HGNC   \n",
       "4      -                  -        -     -1     -                   HGNC   \n",
       "5      -                  -        -     -1     -                   HGNC   \n",
       "6      -                  -        -     -1     -                   HGNC   \n",
       "7      -                  -     1765     -1     -  Clone_based_vega_gene   \n",
       "8      -                  -        -     -1     -                   HGNC   \n",
       "9      -                  -        -     -1     -                   HGNC   \n",
       "\n",
       "  HGNC_ID MANE_SELECT MANE_PLUS_CLINICAL TSL APPRIS SIFT PolyPhen AF CLIN_SIG  \\\n",
       "0   19824           -                  -   -      -    -        -  -        -   \n",
       "1   19824           -                  -   -      -    -        -  -        -   \n",
       "2   19824           -                  -   -      -    -        -  -        -   \n",
       "3   19824           -                  -   -      -    -        -  -        -   \n",
       "4   30857           -                  -   -      -    -        -  -        -   \n",
       "5   30857           -                  -   -      -    -        -  -        -   \n",
       "6   30857           -                  -   -      -    -        -  -        -   \n",
       "7       -           -                  -   -      -    -        -  -        -   \n",
       "8   30857           -                  -   -      -    -        -  -        -   \n",
       "9   30857           -                  -   -      -    -        -  -        -   \n",
       "\n",
       "  SOMATIC PHENO PUBMED MOTIF_NAME MOTIF_POS HIGH_INF_POS MOTIF_SCORE_CHANGE  \\\n",
       "0       -     -      -          -         -            -                  -   \n",
       "1       -     -      -          -         -            -                  -   \n",
       "2       -     -      -          -         -            -                  -   \n",
       "3       -     -      -          -         -            -                  -   \n",
       "4       -     -      -          -         -            -                  -   \n",
       "5       -     -      -          -         -            -                  -   \n",
       "6       -     -      -          -         -            -                  -   \n",
       "7       -     -      -          -         -            -                  -   \n",
       "8       -     -      -          -         -            -                  -   \n",
       "9       -     -      -          -         -            -                  -   \n",
       "\n",
       "  TRANSCRIPTION_FACTORS  \n",
       "0                     -  \n",
       "1                     -  \n",
       "2                     -  \n",
       "3                     -  \n",
       "4                     -  \n",
       "5                     -  \n",
       "6                     -  \n",
       "7                     -  \n",
       "8                     -  \n",
       "9                     -  "
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "archaic_specific_spliceosome_variants = pd.read_csv(\"../spliceosome/archaic_specific_spliceosome_variants.vep\", sep='\\t', header=0)\n",
    "archaic_specific_spliceosome_variants.head(10)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "44e8ddae",
   "metadata": {},
   "source": [
    "Subset to variants with non-synonymous effects."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "e1a1f4a0",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Uploaded_variation</th>\n",
       "      <th>Location</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Feature</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>cDNA_position</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>Codons</th>\n",
       "      <th>Existing_variation</th>\n",
       "      <th>DISTANCE</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>FLAGS</th>\n",
       "      <th>SYMBOL_SOURCE</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>MANE_SELECT</th>\n",
       "      <th>MANE_PLUS_CLINICAL</th>\n",
       "      <th>TSL</th>\n",
       "      <th>APPRIS</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>AF</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>SOMATIC</th>\n",
       "      <th>PHENO</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>MOTIF_NAME</th>\n",
       "      <th>MOTIF_POS</th>\n",
       "      <th>HIGH_INF_POS</th>\n",
       "      <th>MOTIF_SCORE_CHANGE</th>\n",
       "      <th>TRANSCRIPTION_FACTORS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>559</th>\n",
       "      <td>.</td>\n",
       "      <td>1:150297443-150297443</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>PRPF3</td>\n",
       "      <td>ENSG00000117360</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000324862.6</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>2/16</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>208</td>\n",
       "      <td>43</td>\n",
       "      <td>15</td>\n",
       "      <td>I/V</td>\n",
       "      <td>Ata/Gta</td>\n",
       "      <td>rs1395340408</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>17348</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>tolerated(0.67)</td>\n",
       "      <td>benign(0.001)</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>560</th>\n",
       "      <td>.</td>\n",
       "      <td>1:150297443-150297443</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>PRPF3</td>\n",
       "      <td>ENSG00000117360</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000414970.2</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>2/15</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>131</td>\n",
       "      <td>43</td>\n",
       "      <td>15</td>\n",
       "      <td>I/V</td>\n",
       "      <td>Ata/Gta</td>\n",
       "      <td>rs1395340408</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>17348</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>tolerated(0.75)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1560</th>\n",
       "      <td>.</td>\n",
       "      <td>11:116633557-116633557</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>BUD13</td>\n",
       "      <td>ENSG00000137656</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000260210.4</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>4/10</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>772</td>\n",
       "      <td>748</td>\n",
       "      <td>250</td>\n",
       "      <td>D/H</td>\n",
       "      <td>Gac/Cac</td>\n",
       "      <td>rs1425200283</td>\n",
       "      <td>-</td>\n",
       "      <td>-1</td>\n",
       "      <td>-</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>28199</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>deleterious(0)</td>\n",
       "      <td>benign(0.341)</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1891</th>\n",
       "      <td>.</td>\n",
       "      <td>13:21714974-21714974</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>SAP18</td>\n",
       "      <td>ENSG00000150459</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000450573.1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>1/3</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>62</td>\n",
       "      <td>13</td>\n",
       "      <td>5</td>\n",
       "      <td>F/V</td>\n",
       "      <td>Ttc/Gtc</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>cds_end_NF</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>10530</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>tolerated_low_confidence(1)</td>\n",
       "      <td>unknown(0)</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3692</th>\n",
       "      <td>.</td>\n",
       "      <td>15:35219290-35219290</td>\n",
       "      <td>C</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>AQR</td>\n",
       "      <td>ENSG00000021776</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000156471.5</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>13/35</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>1290</td>\n",
       "      <td>1064</td>\n",
       "      <td>355</td>\n",
       "      <td>N/S</td>\n",
       "      <td>aAt/aGt</td>\n",
       "      <td>rs767183682</td>\n",
       "      <td>-</td>\n",
       "      <td>-1</td>\n",
       "      <td>-</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>29513</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>tolerated(0.09)</td>\n",
       "      <td>benign(0.062)</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3963</th>\n",
       "      <td>.</td>\n",
       "      <td>16:2810331-2810331</td>\n",
       "      <td>T</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>SRRM2</td>\n",
       "      <td>ENSG00000167978</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000301740.8</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>10/15</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>1412</td>\n",
       "      <td>863</td>\n",
       "      <td>288</td>\n",
       "      <td>T/I</td>\n",
       "      <td>aCa/aTa</td>\n",
       "      <td>rs747144756</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>16639</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>deleterious_low_confidence(0.04)</td>\n",
       "      <td>benign(0.006)</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3967</th>\n",
       "      <td>.</td>\n",
       "      <td>16:2810331-2810331</td>\n",
       "      <td>T</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>SRRM2</td>\n",
       "      <td>ENSG00000167978</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000571378.1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>9/10</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>760</td>\n",
       "      <td>575</td>\n",
       "      <td>192</td>\n",
       "      <td>T/I</td>\n",
       "      <td>aCa/aTa</td>\n",
       "      <td>rs747144756</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>cds_end_NF</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>16639</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>tolerated_low_confidence(0.18)</td>\n",
       "      <td>benign(0.006)</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3973</th>\n",
       "      <td>.</td>\n",
       "      <td>16:2810331-2810331</td>\n",
       "      <td>T</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>SRRM2</td>\n",
       "      <td>ENSG00000167978</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000575009.1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>9/10</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>830</td>\n",
       "      <td>575</td>\n",
       "      <td>192</td>\n",
       "      <td>T/I</td>\n",
       "      <td>aCa/aTa</td>\n",
       "      <td>rs747144756</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>cds_end_NF</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>16639</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>tolerated_low_confidence(0.22)</td>\n",
       "      <td>benign(0.006)</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3980</th>\n",
       "      <td>.</td>\n",
       "      <td>16:2810331-2810331</td>\n",
       "      <td>T</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>SRRM2</td>\n",
       "      <td>ENSG00000167978</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000576924.1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>10/11</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>1123</td>\n",
       "      <td>863</td>\n",
       "      <td>288</td>\n",
       "      <td>T/I</td>\n",
       "      <td>aCa/aTa</td>\n",
       "      <td>rs747144756</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>cds_end_NF</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>16639</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>tolerated_low_confidence(0.13)</td>\n",
       "      <td>benign(0.006)</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4017</th>\n",
       "      <td>.</td>\n",
       "      <td>16:2813610-2813610</td>\n",
       "      <td>C</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>SRRM2</td>\n",
       "      <td>ENSG00000167978</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000301740.8</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>11/15</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>3630</td>\n",
       "      <td>3081</td>\n",
       "      <td>1027</td>\n",
       "      <td>Q/H</td>\n",
       "      <td>caA/caC</td>\n",
       "      <td>rs530922815</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>16639</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>deleterious_low_confidence(0.04)</td>\n",
       "      <td>benign(0.365)</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     #Uploaded_variation                Location Allele       Consequence  \\\n",
       "559                    .   1:150297443-150297443      G  missense_variant   \n",
       "560                    .   1:150297443-150297443      G  missense_variant   \n",
       "1560                   .  11:116633557-116633557      G  missense_variant   \n",
       "1891                   .    13:21714974-21714974      G  missense_variant   \n",
       "3692                   .    15:35219290-35219290      C  missense_variant   \n",
       "3963                   .      16:2810331-2810331      T  missense_variant   \n",
       "3967                   .      16:2810331-2810331      T  missense_variant   \n",
       "3973                   .      16:2810331-2810331      T  missense_variant   \n",
       "3980                   .      16:2810331-2810331      T  missense_variant   \n",
       "4017                   .      16:2813610-2813610      C  missense_variant   \n",
       "\n",
       "        IMPACT SYMBOL             Gene Feature_type            Feature  \\\n",
       "559   MODERATE  PRPF3  ENSG00000117360   Transcript  ENST00000324862.6   \n",
       "560   MODERATE  PRPF3  ENSG00000117360   Transcript  ENST00000414970.2   \n",
       "1560  MODERATE  BUD13  ENSG00000137656   Transcript  ENST00000260210.4   \n",
       "1891  MODERATE  SAP18  ENSG00000150459   Transcript  ENST00000450573.1   \n",
       "3692  MODERATE    AQR  ENSG00000021776   Transcript  ENST00000156471.5   \n",
       "3963  MODERATE  SRRM2  ENSG00000167978   Transcript  ENST00000301740.8   \n",
       "3967  MODERATE  SRRM2  ENSG00000167978   Transcript  ENST00000571378.1   \n",
       "3973  MODERATE  SRRM2  ENSG00000167978   Transcript  ENST00000575009.1   \n",
       "3980  MODERATE  SRRM2  ENSG00000167978   Transcript  ENST00000576924.1   \n",
       "4017  MODERATE  SRRM2  ENSG00000167978   Transcript  ENST00000301740.8   \n",
       "\n",
       "             BIOTYPE   EXON INTRON HGVSc HGVSp cDNA_position CDS_position  \\\n",
       "559   protein_coding   2/16      -     -     -           208           43   \n",
       "560   protein_coding   2/15      -     -     -           131           43   \n",
       "1560  protein_coding   4/10      -     -     -           772          748   \n",
       "1891  protein_coding    1/3      -     -     -            62           13   \n",
       "3692  protein_coding  13/35      -     -     -          1290         1064   \n",
       "3963  protein_coding  10/15      -     -     -          1412          863   \n",
       "3967  protein_coding   9/10      -     -     -           760          575   \n",
       "3973  protein_coding   9/10      -     -     -           830          575   \n",
       "3980  protein_coding  10/11      -     -     -          1123          863   \n",
       "4017  protein_coding  11/15      -     -     -          3630         3081   \n",
       "\n",
       "     Protein_position Amino_acids   Codons Existing_variation DISTANCE STRAND  \\\n",
       "559                15         I/V  Ata/Gta       rs1395340408        -      1   \n",
       "560                15         I/V  Ata/Gta       rs1395340408        -      1   \n",
       "1560              250         D/H  Gac/Cac       rs1425200283        -     -1   \n",
       "1891                5         F/V  Ttc/Gtc                  -        -      1   \n",
       "3692              355         N/S  aAt/aGt        rs767183682        -     -1   \n",
       "3963              288         T/I  aCa/aTa        rs747144756        -      1   \n",
       "3967              192         T/I  aCa/aTa        rs747144756        -      1   \n",
       "3973              192         T/I  aCa/aTa        rs747144756        -      1   \n",
       "3980              288         T/I  aCa/aTa        rs747144756        -      1   \n",
       "4017             1027         Q/H  caA/caC        rs530922815        -      1   \n",
       "\n",
       "           FLAGS SYMBOL_SOURCE HGNC_ID MANE_SELECT MANE_PLUS_CLINICAL TSL  \\\n",
       "559            -          HGNC   17348           -                  -   -   \n",
       "560            -          HGNC   17348           -                  -   -   \n",
       "1560           -          HGNC   28199           -                  -   -   \n",
       "1891  cds_end_NF          HGNC   10530           -                  -   -   \n",
       "3692           -          HGNC   29513           -                  -   -   \n",
       "3963           -          HGNC   16639           -                  -   -   \n",
       "3967  cds_end_NF          HGNC   16639           -                  -   -   \n",
       "3973  cds_end_NF          HGNC   16639           -                  -   -   \n",
       "3980  cds_end_NF          HGNC   16639           -                  -   -   \n",
       "4017           -          HGNC   16639           -                  -   -   \n",
       "\n",
       "     APPRIS                              SIFT       PolyPhen AF CLIN_SIG  \\\n",
       "559       -                   tolerated(0.67)  benign(0.001)  -        -   \n",
       "560       -                   tolerated(0.75)      benign(0)  -        -   \n",
       "1560      -                    deleterious(0)  benign(0.341)  -        -   \n",
       "1891      -       tolerated_low_confidence(1)     unknown(0)  -        -   \n",
       "3692      -                   tolerated(0.09)  benign(0.062)  -        -   \n",
       "3963      -  deleterious_low_confidence(0.04)  benign(0.006)  -        -   \n",
       "3967      -    tolerated_low_confidence(0.18)  benign(0.006)  -        -   \n",
       "3973      -    tolerated_low_confidence(0.22)  benign(0.006)  -        -   \n",
       "3980      -    tolerated_low_confidence(0.13)  benign(0.006)  -        -   \n",
       "4017      -  deleterious_low_confidence(0.04)  benign(0.365)  -        -   \n",
       "\n",
       "     SOMATIC PHENO PUBMED MOTIF_NAME MOTIF_POS HIGH_INF_POS  \\\n",
       "559        -     -      -          -         -            -   \n",
       "560        -     -      -          -         -            -   \n",
       "1560       -     -      -          -         -            -   \n",
       "1891       -     -      -          -         -            -   \n",
       "3692       -     -      -          -         -            -   \n",
       "3963       -     -      -          -         -            -   \n",
       "3967       -     -      -          -         -            -   \n",
       "3973       -     -      -          -         -            -   \n",
       "3980       -     -      -          -         -            -   \n",
       "4017       -     -      -          -         -            -   \n",
       "\n",
       "     MOTIF_SCORE_CHANGE TRANSCRIPTION_FACTORS  \n",
       "559                   -                     -  \n",
       "560                   -                     -  \n",
       "1560                  -                     -  \n",
       "1891                  -                     -  \n",
       "3692                  -                     -  \n",
       "3963                  -                     -  \n",
       "3967                  -                     -  \n",
       "3973                  -                     -  \n",
       "3980                  -                     -  \n",
       "4017                  -                     -  "
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nonsynonymous_archaic_specific_spliceosome_variants = archaic_specific_spliceosome_variants[archaic_specific_spliceosome_variants['Consequence'] == 'missense_variant']\n",
    "nonsynonymous_archaic_specific_spliceosome_variants.head(10)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b5862eaf",
   "metadata": {},
   "source": [
    "How many transcripts are impacted?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "4544ef72",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "36"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(nonsynonymous_archaic_specific_spliceosome_variants)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0bc069b2",
   "metadata": {},
   "source": [
    "How many variants does this represent?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "25f37e72",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Location</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Gene</th>\n",
       "      <th>transcript_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>11:116633557-116633557</td>\n",
       "      <td>G</td>\n",
       "      <td>ENSG00000137656</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>13:21714974-21714974</td>\n",
       "      <td>G</td>\n",
       "      <td>ENSG00000150459</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>15:35219290-35219290</td>\n",
       "      <td>C</td>\n",
       "      <td>ENSG00000021776</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>16:2810331-2810331</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000167978</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>16:2813610-2813610</td>\n",
       "      <td>C</td>\n",
       "      <td>ENSG00000167978</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>16:2817499-2817499</td>\n",
       "      <td>G</td>\n",
       "      <td>ENSG00000167978</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>16:2819197-2819197</td>\n",
       "      <td>C</td>\n",
       "      <td>ENSG00000167978</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>17:1554120-1554120</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000174231</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>17:1557188-1557188</td>\n",
       "      <td>A</td>\n",
       "      <td>ENSG00000174231</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>17:36963046-36963046</td>\n",
       "      <td>C</td>\n",
       "      <td>ENSG00000108296</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>19:12786433-12786433</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000123154</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>19:36124120-36124120</td>\n",
       "      <td>A</td>\n",
       "      <td>ENSG00000126254</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>19:54663336-54663336</td>\n",
       "      <td>G</td>\n",
       "      <td>ENSG00000105617</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>1:150297443-150297443</td>\n",
       "      <td>G</td>\n",
       "      <td>ENSG00000117360</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>20:20031139-20031139</td>\n",
       "      <td>C</td>\n",
       "      <td>ENSG00000101343</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>20:37632435-37632435</td>\n",
       "      <td>A</td>\n",
       "      <td>ENSG00000101452</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>5:150080190-150080190</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000086589</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>5:176940736-176940736</td>\n",
       "      <td>G</td>\n",
       "      <td>ENSG00000183258</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>6:34725658-34725658</td>\n",
       "      <td>A</td>\n",
       "      <td>ENSG00000124562</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  Location Allele             Gene  transcript_count\n",
       "0   11:116633557-116633557      G  ENSG00000137656                 1\n",
       "1     13:21714974-21714974      G  ENSG00000150459                 1\n",
       "2     15:35219290-35219290      C  ENSG00000021776                 1\n",
       "3       16:2810331-2810331      T  ENSG00000167978                 4\n",
       "4       16:2813610-2813610      C  ENSG00000167978                 1\n",
       "5       16:2817499-2817499      G  ENSG00000167978                 1\n",
       "6       16:2819197-2819197      C  ENSG00000167978                 1\n",
       "7       17:1554120-1554120      T  ENSG00000174231                 1\n",
       "8       17:1557188-1557188      A  ENSG00000174231                 2\n",
       "9     17:36963046-36963046      C  ENSG00000108296                 2\n",
       "10    19:12786433-12786433      T  ENSG00000123154                 2\n",
       "11    19:36124120-36124120      A  ENSG00000126254                 6\n",
       "12    19:54663336-54663336      G  ENSG00000105617                 1\n",
       "13   1:150297443-150297443      G  ENSG00000117360                 2\n",
       "14    20:20031139-20031139      C  ENSG00000101343                 3\n",
       "15    20:37632435-37632435      A  ENSG00000101452                 3\n",
       "16   5:150080190-150080190      T  ENSG00000086589                 1\n",
       "17   5:176940736-176940736      G  ENSG00000183258                 2\n",
       "18     6:34725658-34725658      A  ENSG00000124562                 1"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nonsynonymous_archaic_specific_spliceosome_variants.groupby(by=['Location', 'Allele', 'Gene']).size().reset_index(name='transcript_count')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1d03adc2",
   "metadata": {},
   "source": [
    "Now let's look at the potential effects starting with SIFT designations."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "629be4bb",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Location</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>SIFT_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>11:116633557-116633557</td>\n",
       "      <td>deleterious(0)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>13:21714974-21714974</td>\n",
       "      <td>tolerated_low_confidence(1)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>15:35219290-35219290</td>\n",
       "      <td>tolerated(0.09)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>16:2810331-2810331</td>\n",
       "      <td>deleterious_low_confidence(0.04)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>16:2810331-2810331</td>\n",
       "      <td>tolerated_low_confidence(0.13)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>16:2810331-2810331</td>\n",
       "      <td>tolerated_low_confidence(0.18)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>16:2810331-2810331</td>\n",
       "      <td>tolerated_low_confidence(0.22)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>16:2813610-2813610</td>\n",
       "      <td>deleterious_low_confidence(0.04)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>16:2817499-2817499</td>\n",
       "      <td>deleterious_low_confidence(0)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>16:2819197-2819197</td>\n",
       "      <td>deleterious_low_confidence(0)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>17:1554120-1554120</td>\n",
       "      <td>deleterious_low_confidence(0)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>17:1557188-1557188</td>\n",
       "      <td>tolerated(0.07)</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>17:36963046-36963046</td>\n",
       "      <td>tolerated(0.15)</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>19:12786433-12786433</td>\n",
       "      <td>tolerated(0.06)</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>19:36124120-36124120</td>\n",
       "      <td>deleterious_low_confidence(0)</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>19:36124120-36124120</td>\n",
       "      <td>tolerated(0.1)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>19:54663336-54663336</td>\n",
       "      <td>deleterious(0.02)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>1:150297443-150297443</td>\n",
       "      <td>tolerated(0.67)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>1:150297443-150297443</td>\n",
       "      <td>tolerated(0.75)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>20:20031139-20031139</td>\n",
       "      <td>tolerated(0.48)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>20:20031139-20031139</td>\n",
       "      <td>tolerated(0.53)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>20:20031139-20031139</td>\n",
       "      <td>tolerated(0.75)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>20:37632435-37632435</td>\n",
       "      <td>tolerated(0.38)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>20:37632435-37632435</td>\n",
       "      <td>tolerated(0.42)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>20:37632435-37632435</td>\n",
       "      <td>tolerated(0.44)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>5:150080190-150080190</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>5:176940736-176940736</td>\n",
       "      <td>tolerated(0.25)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>5:176940736-176940736</td>\n",
       "      <td>tolerated(0.32)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>6:34725658-34725658</td>\n",
       "      <td>tolerated_low_confidence(0.25)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  Location                              SIFT  SIFT_count\n",
       "0   11:116633557-116633557                    deleterious(0)           1\n",
       "1     13:21714974-21714974       tolerated_low_confidence(1)           1\n",
       "2     15:35219290-35219290                   tolerated(0.09)           1\n",
       "3       16:2810331-2810331  deleterious_low_confidence(0.04)           1\n",
       "4       16:2810331-2810331    tolerated_low_confidence(0.13)           1\n",
       "5       16:2810331-2810331    tolerated_low_confidence(0.18)           1\n",
       "6       16:2810331-2810331    tolerated_low_confidence(0.22)           1\n",
       "7       16:2813610-2813610  deleterious_low_confidence(0.04)           1\n",
       "8       16:2817499-2817499     deleterious_low_confidence(0)           1\n",
       "9       16:2819197-2819197     deleterious_low_confidence(0)           1\n",
       "10      17:1554120-1554120     deleterious_low_confidence(0)           1\n",
       "11      17:1557188-1557188                   tolerated(0.07)           2\n",
       "12    17:36963046-36963046                   tolerated(0.15)           2\n",
       "13    19:12786433-12786433                   tolerated(0.06)           2\n",
       "14    19:36124120-36124120     deleterious_low_confidence(0)           5\n",
       "15    19:36124120-36124120                    tolerated(0.1)           1\n",
       "16    19:54663336-54663336                 deleterious(0.02)           1\n",
       "17   1:150297443-150297443                   tolerated(0.67)           1\n",
       "18   1:150297443-150297443                   tolerated(0.75)           1\n",
       "19    20:20031139-20031139                   tolerated(0.48)           1\n",
       "20    20:20031139-20031139                   tolerated(0.53)           1\n",
       "21    20:20031139-20031139                   tolerated(0.75)           1\n",
       "22    20:37632435-37632435                   tolerated(0.38)           1\n",
       "23    20:37632435-37632435                   tolerated(0.42)           1\n",
       "24    20:37632435-37632435                   tolerated(0.44)           1\n",
       "25   5:150080190-150080190                                 -           1\n",
       "26   5:176940736-176940736                   tolerated(0.25)           1\n",
       "27   5:176940736-176940736                   tolerated(0.32)           1\n",
       "28     6:34725658-34725658    tolerated_low_confidence(0.25)           1"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nonsynonymous_archaic_specific_spliceosome_variants_SIFT = nonsynonymous_archaic_specific_spliceosome_variants.groupby(by=['Location','SIFT']).size().reset_index(name='SIFT_count')\n",
    "nonsynonymous_archaic_specific_spliceosome_variants_SIFT"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "2c82813f",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Location</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>SIFT_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>11:116633557-116633557</td>\n",
       "      <td>deleterious(0)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>16:2810331-2810331</td>\n",
       "      <td>deleterious_low_confidence(0.04)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>16:2813610-2813610</td>\n",
       "      <td>deleterious_low_confidence(0.04)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>16:2817499-2817499</td>\n",
       "      <td>deleterious_low_confidence(0)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>16:2819197-2819197</td>\n",
       "      <td>deleterious_low_confidence(0)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>17:1554120-1554120</td>\n",
       "      <td>deleterious_low_confidence(0)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>19:36124120-36124120</td>\n",
       "      <td>deleterious_low_confidence(0)</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>19:54663336-54663336</td>\n",
       "      <td>deleterious(0.02)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  Location                              SIFT  SIFT_count\n",
       "0   11:116633557-116633557                    deleterious(0)           1\n",
       "3       16:2810331-2810331  deleterious_low_confidence(0.04)           1\n",
       "7       16:2813610-2813610  deleterious_low_confidence(0.04)           1\n",
       "8       16:2817499-2817499     deleterious_low_confidence(0)           1\n",
       "9       16:2819197-2819197     deleterious_low_confidence(0)           1\n",
       "10      17:1554120-1554120     deleterious_low_confidence(0)           1\n",
       "14    19:36124120-36124120     deleterious_low_confidence(0)           5\n",
       "16    19:54663336-54663336                 deleterious(0.02)           1"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SIFT_deleterious = nonsynonymous_archaic_specific_spliceosome_variants_SIFT[nonsynonymous_archaic_specific_spliceosome_variants_SIFT['SIFT'].str.contains('deleterious')]\n",
    "SIFT_deleterious"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b6e0a5ce",
   "metadata": {},
   "source": [
    "And PolyPhen."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "d76ddb3b",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Location</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>PolyPhen_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>11:116633557-116633557</td>\n",
       "      <td>benign(0.341)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>13:21714974-21714974</td>\n",
       "      <td>unknown(0)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>15:35219290-35219290</td>\n",
       "      <td>benign(0.062)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>16:2810331-2810331</td>\n",
       "      <td>benign(0.006)</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>16:2813610-2813610</td>\n",
       "      <td>benign(0.365)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>16:2817499-2817499</td>\n",
       "      <td>benign(0.007)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>16:2819197-2819197</td>\n",
       "      <td>benign(0.106)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>17:1554120-1554120</td>\n",
       "      <td>unknown(0)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>17:1557188-1557188</td>\n",
       "      <td>benign(0.063)</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>17:36963046-36963046</td>\n",
       "      <td>benign(0.034)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>17:36963046-36963046</td>\n",
       "      <td>benign(0.117)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>19:12786433-12786433</td>\n",
       "      <td>benign(0.319)</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>19:36124120-36124120</td>\n",
       "      <td>probably_damaging(0.993)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>19:36124120-36124120</td>\n",
       "      <td>probably_damaging(0.997)</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>19:54663336-54663336</td>\n",
       "      <td>possibly_damaging(0.592)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>1:150297443-150297443</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>1:150297443-150297443</td>\n",
       "      <td>benign(0.001)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>20:20031139-20031139</td>\n",
       "      <td>benign(0.017)</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>20:37632435-37632435</td>\n",
       "      <td>benign(0.249)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>20:37632435-37632435</td>\n",
       "      <td>possibly_damaging(0.454)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>20:37632435-37632435</td>\n",
       "      <td>possibly_damaging(0.506)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>5:150080190-150080190</td>\n",
       "      <td>possibly_damaging(0.711)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>5:176940736-176940736</td>\n",
       "      <td>benign(0.003)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>5:176940736-176940736</td>\n",
       "      <td>possibly_damaging(0.451)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>6:34725658-34725658</td>\n",
       "      <td>benign(0.011)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  Location                  PolyPhen  PolyPhen_count\n",
       "0   11:116633557-116633557             benign(0.341)               1\n",
       "1     13:21714974-21714974                unknown(0)               1\n",
       "2     15:35219290-35219290             benign(0.062)               1\n",
       "3       16:2810331-2810331             benign(0.006)               4\n",
       "4       16:2813610-2813610             benign(0.365)               1\n",
       "5       16:2817499-2817499             benign(0.007)               1\n",
       "6       16:2819197-2819197             benign(0.106)               1\n",
       "7       17:1554120-1554120                unknown(0)               1\n",
       "8       17:1557188-1557188             benign(0.063)               2\n",
       "9     17:36963046-36963046             benign(0.034)               1\n",
       "10    17:36963046-36963046             benign(0.117)               1\n",
       "11    19:12786433-12786433             benign(0.319)               2\n",
       "12    19:36124120-36124120  probably_damaging(0.993)               1\n",
       "13    19:36124120-36124120  probably_damaging(0.997)               5\n",
       "14    19:54663336-54663336  possibly_damaging(0.592)               1\n",
       "15   1:150297443-150297443                 benign(0)               1\n",
       "16   1:150297443-150297443             benign(0.001)               1\n",
       "17    20:20031139-20031139             benign(0.017)               3\n",
       "18    20:37632435-37632435             benign(0.249)               1\n",
       "19    20:37632435-37632435  possibly_damaging(0.454)               1\n",
       "20    20:37632435-37632435  possibly_damaging(0.506)               1\n",
       "21   5:150080190-150080190  possibly_damaging(0.711)               1\n",
       "22   5:176940736-176940736             benign(0.003)               1\n",
       "23   5:176940736-176940736  possibly_damaging(0.451)               1\n",
       "24     6:34725658-34725658             benign(0.011)               1"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nonsynonymous_archaic_specific_spliceosome_variants_PolyPhen = nonsynonymous_archaic_specific_spliceosome_variants.groupby(by=['Location','PolyPhen']).size().reset_index(name='PolyPhen_count')\n",
    "nonsynonymous_archaic_specific_spliceosome_variants_PolyPhen"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "b017a933",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Location</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>PolyPhen_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>19:36124120-36124120</td>\n",
       "      <td>probably_damaging(0.993)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>19:36124120-36124120</td>\n",
       "      <td>probably_damaging(0.997)</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>19:54663336-54663336</td>\n",
       "      <td>possibly_damaging(0.592)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>20:37632435-37632435</td>\n",
       "      <td>possibly_damaging(0.454)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>20:37632435-37632435</td>\n",
       "      <td>possibly_damaging(0.506)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>5:150080190-150080190</td>\n",
       "      <td>possibly_damaging(0.711)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>5:176940736-176940736</td>\n",
       "      <td>possibly_damaging(0.451)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                 Location                  PolyPhen  PolyPhen_count\n",
       "12   19:36124120-36124120  probably_damaging(0.993)               1\n",
       "13   19:36124120-36124120  probably_damaging(0.997)               5\n",
       "14   19:54663336-54663336  possibly_damaging(0.592)               1\n",
       "19   20:37632435-37632435  possibly_damaging(0.454)               1\n",
       "20   20:37632435-37632435  possibly_damaging(0.506)               1\n",
       "21  5:150080190-150080190  possibly_damaging(0.711)               1\n",
       "23  5:176940736-176940736  possibly_damaging(0.451)               1"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PolyPhen_damaging = nonsynonymous_archaic_specific_spliceosome_variants_PolyPhen[nonsynonymous_archaic_specific_spliceosome_variants_PolyPhen['PolyPhen'].str.contains('damaging')]\n",
    "PolyPhen_damaging"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "72b7c91c",
   "metadata": {},
   "source": [
    "Do SIFT and PolyPhen agree on any positions?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "a872a2f1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Location</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>SIFT_count</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>PolyPhen_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>19:36124120-36124120</td>\n",
       "      <td>deleterious_low_confidence(0)</td>\n",
       "      <td>5</td>\n",
       "      <td>probably_damaging(0.993)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>19:36124120-36124120</td>\n",
       "      <td>deleterious_low_confidence(0)</td>\n",
       "      <td>5</td>\n",
       "      <td>probably_damaging(0.997)</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>19:54663336-54663336</td>\n",
       "      <td>deleterious(0.02)</td>\n",
       "      <td>1</td>\n",
       "      <td>possibly_damaging(0.592)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               Location                           SIFT  SIFT_count  \\\n",
       "0  19:36124120-36124120  deleterious_low_confidence(0)           5   \n",
       "1  19:36124120-36124120  deleterious_low_confidence(0)           5   \n",
       "2  19:54663336-54663336              deleterious(0.02)           1   \n",
       "\n",
       "                   PolyPhen  PolyPhen_count  \n",
       "0  probably_damaging(0.993)               1  \n",
       "1  probably_damaging(0.997)               5  \n",
       "2  possibly_damaging(0.592)               1  "
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.merge(SIFT_deleterious, PolyPhen_damaging, on = ['Location'])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b29a0fb5",
   "metadata": {},
   "source": [
    "Now let's save the dataframe of non-synonymous variants for any downstream analyses."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "5b658e19",
   "metadata": {},
   "outputs": [],
   "source": [
    "nonsynonymous_archaic_specific_spliceosome_variants.to_csv('../spliceosome/nonsynonymous_archaic_specific_spliceosome_variants.txt', sep=\"\\t\", header = True, index = False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0798d0ba",
   "metadata": {},
   "source": [
    "# Data Description <a class = 'anchor' id = 'datadescription'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8ebe75e6",
   "metadata": {},
   "source": [
    "Let's get a handle on the dataset as a whole. Then we will dive into the SAVs."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9d7409d3",
   "metadata": {},
   "source": [
    "Count the number of positions."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "86ef1778",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1567894"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(data.groupby(['chrom','pos']))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c2b41a11",
   "metadata": {},
   "source": [
    "Count the number of variants."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "f9a87d1f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1607350"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(data)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "48047223",
   "metadata": {},
   "source": [
    "## Multiple Annotations <a class = 'anchor' id = 'multipleannotations'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a2200106",
   "metadata": {},
   "source": [
    "Let's assess the degree of multiple annotation in the data."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "31a72276",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>annotation_count</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1537451</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>30448</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1154</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>58</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>34</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    count\n",
       "annotation_count         \n",
       "1                 1537451\n",
       "2                   30448\n",
       "3                    1154\n",
       "4                      75\n",
       "5                      36\n",
       "6                      15\n",
       "7                      25\n",
       "8                      22\n",
       "9                      36\n",
       "10                     20\n",
       "11                     20\n",
       "12                     15\n",
       "13                     58\n",
       "14                     70\n",
       "15                     34\n",
       "16                      6\n",
       "17                      6\n",
       "18                     32\n",
       "19                      6\n",
       "20                      3\n",
       "21                     24"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "multi_annotation = data.groupby(by=['chrom','pos','ref_allele','altai_gt','chagyrskaya_gt','denisovan_gt','vindija_gt','alt_allele']).size().reset_index(name='annotation_count')\n",
    "multi_annotation.groupby('annotation_count').size().to_frame('count')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "68da21bc",
   "metadata": {},
   "source": [
    "## Multiple Alleles <a class = 'anchor' id = 'multiplealleles'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "39c645c6",
   "metadata": {},
   "source": [
    "Let's assess the degree of multi-allelism in the data (excluding positions with multiple annotations). "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "91680e3c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1569556"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "single_annotation_data = data.drop_duplicates(subset=['chrom','pos','ref_allele','altai_gt','chagyrskaya_gt','denisovan_gt','vindija_gt','alt_allele','variant_type'], keep='first').reset_index()\n",
    "len(single_annotation_data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "0c1db494",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>pos</th>\n",
       "      <th>allele_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>861808</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>862072</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>862093</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>862124</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>862383</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>862389</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr1</td>\n",
       "      <td>863124</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr1</td>\n",
       "      <td>863843</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr1</td>\n",
       "      <td>863863</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr1</td>\n",
       "      <td>863978</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chrom     pos  allele_count\n",
       "0  chr1  861808             1\n",
       "1  chr1  862072             1\n",
       "2  chr1  862093             1\n",
       "3  chr1  862124             1\n",
       "4  chr1  862383             1\n",
       "5  chr1  862389             1\n",
       "6  chr1  863124             1\n",
       "7  chr1  863843             1\n",
       "8  chr1  863863             1\n",
       "9  chr1  863978             1"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "multi_allelic = single_annotation_data.groupby(['chrom','pos',]).size().reset_index(name='allele_count')\n",
    "multi_allelic.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "a82a50a7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>allele count</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>allele_count</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1566233</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1660</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              allele count\n",
       "allele_count              \n",
       "1                  1566233\n",
       "2                     1660\n",
       "3                        1"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "multi_allelic.groupby('allele_count').size().to_frame('allele count')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a8c4680a",
   "metadata": {},
   "source": [
    "## Genotypes <a class = 'anchor' id = 'genotypes'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3e4d11b3",
   "metadata": {},
   "source": [
    "Now let's take a look at the distribution of genotypes across individuals. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "id": "d92784de",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>size</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>altai_gt</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>./.</th>\n",
       "      <td>6226</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0/0</th>\n",
       "      <td>591371</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0/1</th>\n",
       "      <td>126120</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1/0</th>\n",
       "      <td>320</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1/1</th>\n",
       "      <td>883313</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            size\n",
       "altai_gt        \n",
       "./.         6226\n",
       "0/0       591371\n",
       "0/1       126120\n",
       "1/0          320\n",
       "1/1       883313"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data.groupby(['altai_gt']).size().to_frame('size')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "id": "94c5871f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>size</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chagyrskaya_gt</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>./.</th>\n",
       "      <td>37156</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0/0</th>\n",
       "      <td>583787</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0/1</th>\n",
       "      <td>97932</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1/0</th>\n",
       "      <td>144</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1/1</th>\n",
       "      <td>888331</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  size\n",
       "chagyrskaya_gt        \n",
       "./.              37156\n",
       "0/0             583787\n",
       "0/1              97932\n",
       "1/0                144\n",
       "1/1             888331"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data.groupby(['chagyrskaya_gt']).size().to_frame('size')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "90ffeec5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>size</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>denisovan_gt</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>./.</th>\n",
       "      <td>21574</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0/0</th>\n",
       "      <td>546792</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0/1</th>\n",
       "      <td>139522</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1/0</th>\n",
       "      <td>252</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1/1</th>\n",
       "      <td>899210</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                size\n",
       "denisovan_gt        \n",
       "./.            21574\n",
       "0/0           546792\n",
       "0/1           139522\n",
       "1/0              252\n",
       "1/1           899210"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data.groupby(['denisovan_gt']).size().to_frame('size')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "c6d9457d",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>size</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>vindija_gt</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>./.</th>\n",
       "      <td>29672</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0/0</th>\n",
       "      <td>570175</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0/1</th>\n",
       "      <td>119718</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1/0</th>\n",
       "      <td>200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1/1</th>\n",
       "      <td>887585</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              size\n",
       "vindija_gt        \n",
       "./.          29672\n",
       "0/0         570175\n",
       "0/1         119718\n",
       "1/0            200\n",
       "1/1         887585"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data.groupby(['vindija_gt']).size().to_frame('size')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2e1fc0d0",
   "metadata": {},
   "source": [
    "## Annotations <a class = 'anchor' id = 'annotations'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "417aeffe",
   "metadata": {},
   "source": [
    "How many annotations are represented in the total data?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "id": "6e9cf238",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "17631"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(data.groupby(['annotation']))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "171d500d",
   "metadata": {},
   "source": [
    "What is the distribution of these annotations?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "id": "c99946de",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>n</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>count</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>577</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>546</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>473</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>475</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>427</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3868</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3956</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4747</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5413</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9504</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>921 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         n\n",
       "count     \n",
       "1      577\n",
       "2      546\n",
       "3      473\n",
       "4      475\n",
       "5      427\n",
       "...    ...\n",
       "3868     1\n",
       "3956     1\n",
       "4747     1\n",
       "5413     1\n",
       "9504     1\n",
       "\n",
       "[921 rows x 1 columns]"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "n_annotations = data.groupby(['annotation']).size().to_frame('count')\n",
    "n_annotations.groupby('count').size().to_frame('n')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2cc7e428",
   "metadata": {},
   "source": [
    "## Delta Thresholds <a class = 'anchor' id = 'deltathresholds'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ab43e558",
   "metadata": {},
   "source": [
    "Let's do a quick subset of the data for variants that include one delta >= 0.2 and another for deltas >= 0.5."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "id": "df9f3e69",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref_allele</th>\n",
       "      <th>alt_allele</th>\n",
       "      <th>ancestral_allele</th>\n",
       "      <th>anc_dev</th>\n",
       "      <th>variant_type</th>\n",
       "      <th>altai_gt</th>\n",
       "      <th>chagyrskaya_gt</th>\n",
       "      <th>denisovan_gt</th>\n",
       "      <th>vindija_gt</th>\n",
       "      <th>altai_gt_boolean</th>\n",
       "      <th>chagyrskaya_gt_boolean</th>\n",
       "      <th>denisovan_gt_boolean</th>\n",
       "      <th>vindija_gt_boolean</th>\n",
       "      <th>distribution</th>\n",
       "      <th>present_in_1KG</th>\n",
       "      <th>1KG_allele_count</th>\n",
       "      <th>1KG_allele_number</th>\n",
       "      <th>1KG_allele_frequency</th>\n",
       "      <th>1KG_EAS_AF</th>\n",
       "      <th>1KG_EUR_AF</th>\n",
       "      <th>1KG_AFR_AF</th>\n",
       "      <th>1KG_AMR_AF</th>\n",
       "      <th>1KG_SAS_AF</th>\n",
       "      <th>1KG_non_ASW_AFR_AF</th>\n",
       "      <th>present_in_gnomAD</th>\n",
       "      <th>gnomAD_allele_count</th>\n",
       "      <th>gnomAD_allele_number</th>\n",
       "      <th>gnomAD_allele_frequency</th>\n",
       "      <th>Vernot_introgressed</th>\n",
       "      <th>Vernot_ancestral_allele</th>\n",
       "      <th>Vernot_derived_allele</th>\n",
       "      <th>Vernot_ancestral_derived_code</th>\n",
       "      <th>Vernot_AFA_AF</th>\n",
       "      <th>Vernot_AFR_AF</th>\n",
       "      <th>Vernot_AMR_AF</th>\n",
       "      <th>Vernot_EAS_AF</th>\n",
       "      <th>Vernot_EUR_AF</th>\n",
       "      <th>Vernot_PNG_AF</th>\n",
       "      <th>Vernot_SAS_AF</th>\n",
       "      <th>Vernot_Denisovan_base</th>\n",
       "      <th>Vernot_Neanderthal_base_1</th>\n",
       "      <th>Vernot_Neanderthal_base_2</th>\n",
       "      <th>Vernot_haplotype_tag</th>\n",
       "      <th>Vernot_introgressed_AF</th>\n",
       "      <th>Browning_introgressed</th>\n",
       "      <th>Browning_ref_alt</th>\n",
       "      <th>Browning_introgressed_AF</th>\n",
       "      <th>annotation</th>\n",
       "      <th>mis_oe</th>\n",
       "      <th>mis_z</th>\n",
       "      <th>lof_oe</th>\n",
       "      <th>lof_z</th>\n",
       "      <th>phyloP</th>\n",
       "      <th>ag_delta</th>\n",
       "      <th>al_delta</th>\n",
       "      <th>dg_delta</th>\n",
       "      <th>dl_delta</th>\n",
       "      <th>delta_max</th>\n",
       "      <th>ag_pos</th>\n",
       "      <th>al_pos</th>\n",
       "      <th>dg_pos</th>\n",
       "      <th>dl_pos</th>\n",
       "      <th>Adipose_Subcutaneous</th>\n",
       "      <th>Adipose_Visceral_Omentum</th>\n",
       "      <th>Adrenal_Gland</th>\n",
       "      <th>Artery_Aorta</th>\n",
       "      <th>Artery_Coronary</th>\n",
       "      <th>Artery_Tibial</th>\n",
       "      <th>Brain_Amygdala</th>\n",
       "      <th>Brain_Anterior_cingulate_cortex_BA24</th>\n",
       "      <th>Brain_Caudate_basal_ganglia</th>\n",
       "      <th>Brain_Cerebellar_Hemisphere</th>\n",
       "      <th>Brain_Cerebellum</th>\n",
       "      <th>Brain_Cortex</th>\n",
       "      <th>Brain_Frontal_Cortex_BA9</th>\n",
       "      <th>Brain_Hippocampus</th>\n",
       "      <th>Brain_Hypothalamus</th>\n",
       "      <th>Brain_Nucleus_accumbens_basal_ganglia</th>\n",
       "      <th>Brain_Putamen_basal_ganglia</th>\n",
       "      <th>Brain_Spinal_cord_cervical_c-1</th>\n",
       "      <th>Brain_Substantia_nigra</th>\n",
       "      <th>Breast_Mammary_Tissue</th>\n",
       "      <th>Cells_Cultured_fibroblasts</th>\n",
       "      <th>Cells_EBV-transformed_lymphocytes</th>\n",
       "      <th>Colon_Sigmoid</th>\n",
       "      <th>Colon_Transverse</th>\n",
       "      <th>Esophagus_Gastroesophageal_Junction</th>\n",
       "      <th>Esophagus_Mucosa</th>\n",
       "      <th>Esophagus_Muscularis</th>\n",
       "      <th>Heart_Atrial_Appendage</th>\n",
       "      <th>Heart_Left_Ventricle</th>\n",
       "      <th>Kidney_Cortex</th>\n",
       "      <th>Liver</th>\n",
       "      <th>Lung</th>\n",
       "      <th>Minor_Salivary_Gland</th>\n",
       "      <th>Muscle_Skeletal</th>\n",
       "      <th>Nerve_Tibial</th>\n",
       "      <th>Ovary</th>\n",
       "      <th>Pancreas</th>\n",
       "      <th>Pituitary</th>\n",
       "      <th>Prostate</th>\n",
       "      <th>Skin_Not_Sun_Exposed_Suprapubic</th>\n",
       "      <th>Skin_Sun_Exposed_Lower_leg</th>\n",
       "      <th>Small_Intestine_Terminal_Ileum</th>\n",
       "      <th>Spleen</th>\n",
       "      <th>Stomach</th>\n",
       "      <th>Testis</th>\n",
       "      <th>Thyroid</th>\n",
       "      <th>Uterus</th>\n",
       "      <th>Vagina</th>\n",
       "      <th>Whole_Blood</th>\n",
       "      <th>N_GTEx_tissues</th>\n",
       "      <th>sQTL</th>\n",
       "      <th>Vernot_allele_origin</th>\n",
       "      <th>Browning_allele_origin</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>chr1</td>\n",
       "      <td>864726</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>yes</td>\n",
       "      <td>4859.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.95</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.83</td>\n",
       "      <td>0.99</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.812865</td>\n",
       "      <td>yes</td>\n",
       "      <td>137320.0</td>\n",
       "      <td>143300.0</td>\n",
       "      <td>0.958269</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>1.50820</td>\n",
       "      <td>-3.43610</td>\n",
       "      <td>0.89656</td>\n",
       "      <td>0.47484</td>\n",
       "      <td>0.398</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.22</td>\n",
       "      <td>0.22</td>\n",
       "      <td>17</td>\n",
       "      <td>-2</td>\n",
       "      <td>43</td>\n",
       "      <td>-2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>ancient</td>\n",
       "      <td>ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>chr1</td>\n",
       "      <td>875708</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>Altai</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>1.0</td>\n",
       "      <td>139672.0</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>1.50820</td>\n",
       "      <td>-3.43610</td>\n",
       "      <td>0.89656</td>\n",
       "      <td>0.47484</td>\n",
       "      <td>-0.847</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.29</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.29</td>\n",
       "      <td>-43</td>\n",
       "      <td>33</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>130</th>\n",
       "      <td>chr1</td>\n",
       "      <td>909768</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>yes</td>\n",
       "      <td>5094.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.998051</td>\n",
       "      <td>yes</td>\n",
       "      <td>143298.0</td>\n",
       "      <td>143312.0</td>\n",
       "      <td>0.999902</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>1.25930</td>\n",
       "      <td>-1.79880</td>\n",
       "      <td>0.80467</td>\n",
       "      <td>1.00890</td>\n",
       "      <td>-1.259</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.29</td>\n",
       "      <td>0.13</td>\n",
       "      <td>0.03</td>\n",
       "      <td>0.29</td>\n",
       "      <td>7</td>\n",
       "      <td>26</td>\n",
       "      <td>-24</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>ancient</td>\n",
       "      <td>ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>155</th>\n",
       "      <td>chr1</td>\n",
       "      <td>956622</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>yes</td>\n",
       "      <td>4.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.002924</td>\n",
       "      <td>yes</td>\n",
       "      <td>62.0</td>\n",
       "      <td>143332.0</td>\n",
       "      <td>0.000433</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>0.98257</td>\n",
       "      <td>0.22552</td>\n",
       "      <td>0.31914</td>\n",
       "      <td>6.01430</td>\n",
       "      <td>-0.542</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.24</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.24</td>\n",
       "      <td>-2</td>\n",
       "      <td>50</td>\n",
       "      <td>-2</td>\n",
       "      <td>40</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>210</th>\n",
       "      <td>chr1</td>\n",
       "      <td>975133</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>yes</td>\n",
       "      <td>4557.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.89</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.92</td>\n",
       "      <td>0.72</td>\n",
       "      <td>0.92</td>\n",
       "      <td>0.97</td>\n",
       "      <td>0.728070</td>\n",
       "      <td>yes</td>\n",
       "      <td>125462.0</td>\n",
       "      <td>143014.0</td>\n",
       "      <td>0.877271</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>0.98257</td>\n",
       "      <td>0.22552</td>\n",
       "      <td>0.31914</td>\n",
       "      <td>6.01430</td>\n",
       "      <td>0.364</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.22</td>\n",
       "      <td>0.08</td>\n",
       "      <td>0.22</td>\n",
       "      <td>-3</td>\n",
       "      <td>14</td>\n",
       "      <td>-3</td>\n",
       "      <td>14</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>ancient</td>\n",
       "      <td>ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>218</th>\n",
       "      <td>chr1</td>\n",
       "      <td>980460</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>./.</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Other</td>\n",
       "      <td>yes</td>\n",
       "      <td>4642.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.91</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.93</td>\n",
       "      <td>0.77</td>\n",
       "      <td>0.93</td>\n",
       "      <td>0.97</td>\n",
       "      <td>0.781676</td>\n",
       "      <td>yes</td>\n",
       "      <td>128061.0</td>\n",
       "      <td>143108.0</td>\n",
       "      <td>0.894856</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>0.98257</td>\n",
       "      <td>0.22552</td>\n",
       "      <td>0.31914</td>\n",
       "      <td>6.01430</td>\n",
       "      <td>0.561</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.03</td>\n",
       "      <td>0.43</td>\n",
       "      <td>0.43</td>\n",
       "      <td>30</td>\n",
       "      <td>-33</td>\n",
       "      <td>30</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>ancient</td>\n",
       "      <td>ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>224</th>\n",
       "      <td>chr1</td>\n",
       "      <td>982099</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>1.0</td>\n",
       "      <td>142512.0</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>0.98257</td>\n",
       "      <td>0.22552</td>\n",
       "      <td>0.31914</td>\n",
       "      <td>6.01430</td>\n",
       "      <td>-0.375</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.04</td>\n",
       "      <td>0.48</td>\n",
       "      <td>0.48</td>\n",
       "      <td>-48</td>\n",
       "      <td>-6</td>\n",
       "      <td>16</td>\n",
       "      <td>-6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>330</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1117486</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>a</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>yes</td>\n",
       "      <td>1493.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.29</td>\n",
       "      <td>0.15</td>\n",
       "      <td>0.13</td>\n",
       "      <td>0.54</td>\n",
       "      <td>0.21</td>\n",
       "      <td>0.32</td>\n",
       "      <td>0.547758</td>\n",
       "      <td>yes</td>\n",
       "      <td>36989.0</td>\n",
       "      <td>143186.0</td>\n",
       "      <td>0.258328</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TTLL10</td>\n",
       "      <td>0.95390</td>\n",
       "      <td>0.32998</td>\n",
       "      <td>0.77894</td>\n",
       "      <td>1.03800</td>\n",
       "      <td>-2.522</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.27</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.27</td>\n",
       "      <td>-1</td>\n",
       "      <td>33</td>\n",
       "      <td>-1</td>\n",
       "      <td>-48</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>ancient</td>\n",
       "      <td>ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>377</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1153630</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>6.0</td>\n",
       "      <td>143306.0</td>\n",
       "      <td>0.000042</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SDF4</td>\n",
       "      <td>0.90199</td>\n",
       "      <td>0.55251</td>\n",
       "      <td>0.38720</td>\n",
       "      <td>2.23540</td>\n",
       "      <td>-1.558</td>\n",
       "      <td>0.04</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.38</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.38</td>\n",
       "      <td>22</td>\n",
       "      <td>-45</td>\n",
       "      <td>2</td>\n",
       "      <td>23</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>409</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1166887</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>yes</td>\n",
       "      <td>898.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.18</td>\n",
       "      <td>0.17</td>\n",
       "      <td>0.13</td>\n",
       "      <td>0.26</td>\n",
       "      <td>0.11</td>\n",
       "      <td>0.17</td>\n",
       "      <td>0.269981</td>\n",
       "      <td>yes</td>\n",
       "      <td>23482.0</td>\n",
       "      <td>143206.0</td>\n",
       "      <td>0.163974</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SDF4</td>\n",
       "      <td>0.90199</td>\n",
       "      <td>0.55251</td>\n",
       "      <td>0.38720</td>\n",
       "      <td>2.23540</td>\n",
       "      <td>0.260</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.52</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.52</td>\n",
       "      <td>-9</td>\n",
       "      <td>-2</td>\n",
       "      <td>0</td>\n",
       "      <td>-4</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>49.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>ancient</td>\n",
       "      <td>ancient</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    chrom      pos ref_allele alt_allele ancestral_allele  anc_dev  \\\n",
       "24   chr1   864726          T          A                T  derived   \n",
       "65   chr1   875708          C          G                T  derived   \n",
       "130  chr1   909768          A          G                -        .   \n",
       "155  chr1   956622          C          T                C  derived   \n",
       "210  chr1   975133          T          A                T  derived   \n",
       "218  chr1   980460          G          A                G  derived   \n",
       "224  chr1   982099          T          C                T  derived   \n",
       "330  chr1  1117486          A          G                a  derived   \n",
       "377  chr1  1153630          G          A                G  derived   \n",
       "409  chr1  1166887          T          C                T  derived   \n",
       "\n",
       "    variant_type altai_gt chagyrskaya_gt denisovan_gt vindija_gt  \\\n",
       "24           snv      1/1            1/1          1/1        1/1   \n",
       "65           snv      1/1            ./.          ./.        ./.   \n",
       "130          snv      1/1            1/1          1/1        1/1   \n",
       "155          snv      1/1            1/1          0/0        1/1   \n",
       "210          snv      1/1            1/1          1/1        1/1   \n",
       "218          snv      1/1            ./.          1/1        1/1   \n",
       "224          snv      1/1            1/1          1/1        1/1   \n",
       "330          snv      0/0            0/0          1/1        0/0   \n",
       "377          snv      1/1            1/1          0/0        1/1   \n",
       "409          snv      0/0            0/0          1/1        0/0   \n",
       "\n",
       "     altai_gt_boolean  chagyrskaya_gt_boolean  denisovan_gt_boolean  \\\n",
       "24               True                    True                  True   \n",
       "65               True                   False                 False   \n",
       "130              True                    True                  True   \n",
       "155              True                    True                 False   \n",
       "210              True                    True                  True   \n",
       "218              True                   False                  True   \n",
       "224              True                    True                  True   \n",
       "330             False                   False                  True   \n",
       "377              True                    True                 False   \n",
       "409             False                   False                  True   \n",
       "\n",
       "     vindija_gt_boolean distribution present_in_1KG  1KG_allele_count  \\\n",
       "24                 True       Shared            yes            4859.0   \n",
       "65                False        Altai             no               NaN   \n",
       "130                True       Shared            yes            5094.0   \n",
       "155                True  Neanderthal            yes               4.0   \n",
       "210                True       Shared            yes            4557.0   \n",
       "218                True        Other            yes            4642.0   \n",
       "224                True       Shared             no               NaN   \n",
       "330               False    Denisovan            yes            1493.0   \n",
       "377                True  Neanderthal             no               NaN   \n",
       "409               False    Denisovan            yes             898.0   \n",
       "\n",
       "     1KG_allele_number  1KG_allele_frequency  1KG_EAS_AF  1KG_EUR_AF  \\\n",
       "24              5096.0                  0.95        1.00        1.00   \n",
       "65                 NaN                   NaN         NaN         NaN   \n",
       "130             5096.0                  1.00        1.00        1.00   \n",
       "155             5096.0                  0.00        0.00        0.00   \n",
       "210             5096.0                  0.89        1.00        0.92   \n",
       "218             5096.0                  0.91        1.00        0.93   \n",
       "224                NaN                   NaN         NaN         NaN   \n",
       "330             5096.0                  0.29        0.15        0.13   \n",
       "377                NaN                   NaN         NaN         NaN   \n",
       "409             5096.0                  0.18        0.17        0.13   \n",
       "\n",
       "     1KG_AFR_AF  1KG_AMR_AF  1KG_SAS_AF  1KG_non_ASW_AFR_AF present_in_gnomAD  \\\n",
       "24         0.83        0.99        1.00            0.812865               yes   \n",
       "65          NaN         NaN         NaN                 NaN               yes   \n",
       "130        1.00        1.00        1.00            0.998051               yes   \n",
       "155        0.00        0.00        0.00            0.002924               yes   \n",
       "210        0.72        0.92        0.97            0.728070               yes   \n",
       "218        0.77        0.93        0.97            0.781676               yes   \n",
       "224         NaN         NaN         NaN                 NaN               yes   \n",
       "330        0.54        0.21        0.32            0.547758               yes   \n",
       "377         NaN         NaN         NaN                 NaN               yes   \n",
       "409        0.26        0.11        0.17            0.269981               yes   \n",
       "\n",
       "     gnomAD_allele_count  gnomAD_allele_number  gnomAD_allele_frequency  \\\n",
       "24              137320.0              143300.0                 0.958269   \n",
       "65                   1.0              139672.0                 0.000007   \n",
       "130             143298.0              143312.0                 0.999902   \n",
       "155                 62.0              143332.0                 0.000433   \n",
       "210             125462.0              143014.0                 0.877271   \n",
       "218             128061.0              143108.0                 0.894856   \n",
       "224                  1.0              142512.0                 0.000007   \n",
       "330              36989.0              143186.0                 0.258328   \n",
       "377                  6.0              143306.0                 0.000042   \n",
       "409              23482.0              143206.0                 0.163974   \n",
       "\n",
       "    Vernot_introgressed Vernot_ancestral_allele Vernot_derived_allele  \\\n",
       "24                   no                     NaN                   NaN   \n",
       "65                   no                     NaN                   NaN   \n",
       "130                  no                     NaN                   NaN   \n",
       "155                  no                     NaN                   NaN   \n",
       "210                  no                     NaN                   NaN   \n",
       "218                  no                     NaN                   NaN   \n",
       "224                  no                     NaN                   NaN   \n",
       "330                  no                     NaN                   NaN   \n",
       "377                  no                     NaN                   NaN   \n",
       "409                  no                     NaN                   NaN   \n",
       "\n",
       "     Vernot_ancestral_derived_code  Vernot_AFA_AF  Vernot_AFR_AF  \\\n",
       "24                             NaN            NaN            NaN   \n",
       "65                             NaN            NaN            NaN   \n",
       "130                            NaN            NaN            NaN   \n",
       "155                            NaN            NaN            NaN   \n",
       "210                            NaN            NaN            NaN   \n",
       "218                            NaN            NaN            NaN   \n",
       "224                            NaN            NaN            NaN   \n",
       "330                            NaN            NaN            NaN   \n",
       "377                            NaN            NaN            NaN   \n",
       "409                            NaN            NaN            NaN   \n",
       "\n",
       "     Vernot_AMR_AF  Vernot_EAS_AF  Vernot_EUR_AF  Vernot_PNG_AF  \\\n",
       "24             NaN            NaN            NaN            NaN   \n",
       "65             NaN            NaN            NaN            NaN   \n",
       "130            NaN            NaN            NaN            NaN   \n",
       "155            NaN            NaN            NaN            NaN   \n",
       "210            NaN            NaN            NaN            NaN   \n",
       "218            NaN            NaN            NaN            NaN   \n",
       "224            NaN            NaN            NaN            NaN   \n",
       "330            NaN            NaN            NaN            NaN   \n",
       "377            NaN            NaN            NaN            NaN   \n",
       "409            NaN            NaN            NaN            NaN   \n",
       "\n",
       "     Vernot_SAS_AF Vernot_Denisovan_base Vernot_Neanderthal_base_1  \\\n",
       "24             NaN                   NaN                       NaN   \n",
       "65             NaN                   NaN                       NaN   \n",
       "130            NaN                   NaN                       NaN   \n",
       "155            NaN                   NaN                       NaN   \n",
       "210            NaN                   NaN                       NaN   \n",
       "218            NaN                   NaN                       NaN   \n",
       "224            NaN                   NaN                       NaN   \n",
       "330            NaN                   NaN                       NaN   \n",
       "377            NaN                   NaN                       NaN   \n",
       "409            NaN                   NaN                       NaN   \n",
       "\n",
       "    Vernot_Neanderthal_base_2 Vernot_haplotype_tag  Vernot_introgressed_AF  \\\n",
       "24                        NaN                  NaN                     NaN   \n",
       "65                        NaN                  NaN                     NaN   \n",
       "130                       NaN                  NaN                     NaN   \n",
       "155                       NaN                  NaN                     NaN   \n",
       "210                       NaN                  NaN                     NaN   \n",
       "218                       NaN                  NaN                     NaN   \n",
       "224                       NaN                  NaN                     NaN   \n",
       "330                       NaN                  NaN                     NaN   \n",
       "377                       NaN                  NaN                     NaN   \n",
       "409                       NaN                  NaN                     NaN   \n",
       "\n",
       "    Browning_introgressed  Browning_ref_alt  Browning_introgressed_AF  \\\n",
       "24                     no               NaN                       NaN   \n",
       "65                     no               NaN                       NaN   \n",
       "130                    no               NaN                       NaN   \n",
       "155                    no               NaN                       NaN   \n",
       "210                    no               NaN                       NaN   \n",
       "218                    no               NaN                       NaN   \n",
       "224                    no               NaN                       NaN   \n",
       "330                    no               NaN                       NaN   \n",
       "377                    no               NaN                       NaN   \n",
       "409                    no               NaN                       NaN   \n",
       "\n",
       "    annotation   mis_oe    mis_z   lof_oe    lof_z  phyloP  ag_delta  \\\n",
       "24      SAMD11  1.50820 -3.43610  0.89656  0.47484   0.398      0.00   \n",
       "65      SAMD11  1.50820 -3.43610  0.89656  0.47484  -0.847      0.00   \n",
       "130    PLEKHN1  1.25930 -1.79880  0.80467  1.00890  -1.259      0.00   \n",
       "155       AGRN  0.98257  0.22552  0.31914  6.01430  -0.542      0.00   \n",
       "210       AGRN  0.98257  0.22552  0.31914  6.01430   0.364      0.00   \n",
       "218       AGRN  0.98257  0.22552  0.31914  6.01430   0.561      0.00   \n",
       "224       AGRN  0.98257  0.22552  0.31914  6.01430  -0.375      0.00   \n",
       "330     TTLL10  0.95390  0.32998  0.77894  1.03800  -2.522      0.00   \n",
       "377       SDF4  0.90199  0.55251  0.38720  2.23540  -1.558      0.04   \n",
       "409       SDF4  0.90199  0.55251  0.38720  2.23540   0.260      0.00   \n",
       "\n",
       "     al_delta  dg_delta  dl_delta  delta_max  ag_pos  al_pos  dg_pos  dl_pos  \\\n",
       "24       0.00      0.02      0.22       0.22      17      -2      43      -2   \n",
       "65       0.00      0.29      0.01       0.29     -43      33       0      12   \n",
       "130      0.29      0.13      0.03       0.29       7      26     -24       1   \n",
       "155      0.00      0.24      0.02       0.24      -2      50      -2      40   \n",
       "210      0.00      0.22      0.08       0.22      -3      14      -3      14   \n",
       "218      0.00      0.03      0.43       0.43      30     -33      30       0   \n",
       "224      0.00      0.04      0.48       0.48     -48      -6      16      -6   \n",
       "330      0.00      0.27      0.00       0.27      -1      33      -1     -48   \n",
       "377      0.00      0.38      0.00       0.38      22     -45       2      23   \n",
       "409      0.00      0.52      0.00       0.52      -9      -2       0      -4   \n",
       "\n",
       "     Adipose_Subcutaneous  Adipose_Visceral_Omentum  Adrenal_Gland  \\\n",
       "24                    NaN                       NaN            NaN   \n",
       "65                    NaN                       NaN            NaN   \n",
       "130                   NaN                       NaN            NaN   \n",
       "155                   NaN                       NaN            NaN   \n",
       "210                   0.0                       1.0            0.0   \n",
       "218                   0.0                       1.0            0.0   \n",
       "224                   NaN                       NaN            NaN   \n",
       "330                   0.0                       0.0            0.0   \n",
       "377                   NaN                       NaN            NaN   \n",
       "409                   1.0                       1.0            1.0   \n",
       "\n",
       "     Artery_Aorta  Artery_Coronary  Artery_Tibial  Brain_Amygdala  \\\n",
       "24            NaN              NaN            NaN             NaN   \n",
       "65            NaN              NaN            NaN             NaN   \n",
       "130           NaN              NaN            NaN             NaN   \n",
       "155           NaN              NaN            NaN             NaN   \n",
       "210           1.0              0.0            1.0             0.0   \n",
       "218           1.0              0.0            1.0             0.0   \n",
       "224           NaN              NaN            NaN             NaN   \n",
       "330           0.0              0.0            0.0             0.0   \n",
       "377           NaN              NaN            NaN             NaN   \n",
       "409           1.0              1.0            1.0             1.0   \n",
       "\n",
       "     Brain_Anterior_cingulate_cortex_BA24  Brain_Caudate_basal_ganglia  \\\n",
       "24                                    NaN                          NaN   \n",
       "65                                    NaN                          NaN   \n",
       "130                                   NaN                          NaN   \n",
       "155                                   NaN                          NaN   \n",
       "210                                   0.0                          0.0   \n",
       "218                                   0.0                          0.0   \n",
       "224                                   NaN                          NaN   \n",
       "330                                   0.0                          0.0   \n",
       "377                                   NaN                          NaN   \n",
       "409                                   1.0                          1.0   \n",
       "\n",
       "     Brain_Cerebellar_Hemisphere  Brain_Cerebellum  Brain_Cortex  \\\n",
       "24                           NaN               NaN           NaN   \n",
       "65                           NaN               NaN           NaN   \n",
       "130                          NaN               NaN           NaN   \n",
       "155                          NaN               NaN           NaN   \n",
       "210                          0.0               0.0           0.0   \n",
       "218                          0.0               0.0           0.0   \n",
       "224                          NaN               NaN           NaN   \n",
       "330                          0.0               1.0           0.0   \n",
       "377                          NaN               NaN           NaN   \n",
       "409                          1.0               1.0           1.0   \n",
       "\n",
       "     Brain_Frontal_Cortex_BA9  Brain_Hippocampus  Brain_Hypothalamus  \\\n",
       "24                        NaN                NaN                 NaN   \n",
       "65                        NaN                NaN                 NaN   \n",
       "130                       NaN                NaN                 NaN   \n",
       "155                       NaN                NaN                 NaN   \n",
       "210                       0.0                0.0                 0.0   \n",
       "218                       0.0                0.0                 0.0   \n",
       "224                       NaN                NaN                 NaN   \n",
       "330                       0.0                0.0                 0.0   \n",
       "377                       NaN                NaN                 NaN   \n",
       "409                       1.0                1.0                 1.0   \n",
       "\n",
       "     Brain_Nucleus_accumbens_basal_ganglia  Brain_Putamen_basal_ganglia  \\\n",
       "24                                     NaN                          NaN   \n",
       "65                                     NaN                          NaN   \n",
       "130                                    NaN                          NaN   \n",
       "155                                    NaN                          NaN   \n",
       "210                                    0.0                          0.0   \n",
       "218                                    0.0                          0.0   \n",
       "224                                    NaN                          NaN   \n",
       "330                                    0.0                          0.0   \n",
       "377                                    NaN                          NaN   \n",
       "409                                    1.0                          1.0   \n",
       "\n",
       "     Brain_Spinal_cord_cervical_c-1  Brain_Substantia_nigra  \\\n",
       "24                              NaN                     NaN   \n",
       "65                              NaN                     NaN   \n",
       "130                             NaN                     NaN   \n",
       "155                             NaN                     NaN   \n",
       "210                             0.0                     0.0   \n",
       "218                             0.0                     0.0   \n",
       "224                             NaN                     NaN   \n",
       "330                             0.0                     0.0   \n",
       "377                             NaN                     NaN   \n",
       "409                             1.0                     1.0   \n",
       "\n",
       "     Breast_Mammary_Tissue  Cells_Cultured_fibroblasts  \\\n",
       "24                     NaN                         NaN   \n",
       "65                     NaN                         NaN   \n",
       "130                    NaN                         NaN   \n",
       "155                    NaN                         NaN   \n",
       "210                    1.0                         0.0   \n",
       "218                    1.0                         0.0   \n",
       "224                    NaN                         NaN   \n",
       "330                    0.0                         1.0   \n",
       "377                    NaN                         NaN   \n",
       "409                    1.0                         1.0   \n",
       "\n",
       "     Cells_EBV-transformed_lymphocytes  Colon_Sigmoid  Colon_Transverse  \\\n",
       "24                                 NaN            NaN               NaN   \n",
       "65                                 NaN            NaN               NaN   \n",
       "130                                NaN            NaN               NaN   \n",
       "155                                NaN            NaN               NaN   \n",
       "210                                0.0            1.0               0.0   \n",
       "218                                0.0            1.0               0.0   \n",
       "224                                NaN            NaN               NaN   \n",
       "330                                0.0            0.0               0.0   \n",
       "377                                NaN            NaN               NaN   \n",
       "409                                1.0            1.0               1.0   \n",
       "\n",
       "     Esophagus_Gastroesophageal_Junction  Esophagus_Mucosa  \\\n",
       "24                                   NaN               NaN   \n",
       "65                                   NaN               NaN   \n",
       "130                                  NaN               NaN   \n",
       "155                                  NaN               NaN   \n",
       "210                                  1.0               1.0   \n",
       "218                                  1.0               1.0   \n",
       "224                                  NaN               NaN   \n",
       "330                                  0.0               0.0   \n",
       "377                                  NaN               NaN   \n",
       "409                                  1.0               1.0   \n",
       "\n",
       "     Esophagus_Muscularis  Heart_Atrial_Appendage  Heart_Left_Ventricle  \\\n",
       "24                    NaN                     NaN                   NaN   \n",
       "65                    NaN                     NaN                   NaN   \n",
       "130                   NaN                     NaN                   NaN   \n",
       "155                   NaN                     NaN                   NaN   \n",
       "210                   1.0                     0.0                   0.0   \n",
       "218                   1.0                     0.0                   0.0   \n",
       "224                   NaN                     NaN                   NaN   \n",
       "330                   0.0                     0.0                   0.0   \n",
       "377                   NaN                     NaN                   NaN   \n",
       "409                   1.0                     1.0                   1.0   \n",
       "\n",
       "     Kidney_Cortex  Liver  Lung  Minor_Salivary_Gland  Muscle_Skeletal  \\\n",
       "24             NaN    NaN   NaN                   NaN              NaN   \n",
       "65             NaN    NaN   NaN                   NaN              NaN   \n",
       "130            NaN    NaN   NaN                   NaN              NaN   \n",
       "155            NaN    NaN   NaN                   NaN              NaN   \n",
       "210            0.0    0.0   1.0                   0.0              0.0   \n",
       "218            0.0    0.0   1.0                   0.0              0.0   \n",
       "224            NaN    NaN   NaN                   NaN              NaN   \n",
       "330            0.0    0.0   0.0                   0.0              0.0   \n",
       "377            NaN    NaN   NaN                   NaN              NaN   \n",
       "409            1.0    1.0   1.0                   1.0              1.0   \n",
       "\n",
       "     Nerve_Tibial  Ovary  Pancreas  Pituitary  Prostate  \\\n",
       "24            NaN    NaN       NaN        NaN       NaN   \n",
       "65            NaN    NaN       NaN        NaN       NaN   \n",
       "130           NaN    NaN       NaN        NaN       NaN   \n",
       "155           NaN    NaN       NaN        NaN       NaN   \n",
       "210           1.0    0.0       0.0        0.0       0.0   \n",
       "218           1.0    0.0       1.0        0.0       0.0   \n",
       "224           NaN    NaN       NaN        NaN       NaN   \n",
       "330           0.0    0.0       0.0        1.0       0.0   \n",
       "377           NaN    NaN       NaN        NaN       NaN   \n",
       "409           1.0    1.0       1.0        1.0       1.0   \n",
       "\n",
       "     Skin_Not_Sun_Exposed_Suprapubic  Skin_Sun_Exposed_Lower_leg  \\\n",
       "24                               NaN                         NaN   \n",
       "65                               NaN                         NaN   \n",
       "130                              NaN                         NaN   \n",
       "155                              NaN                         NaN   \n",
       "210                              1.0                         1.0   \n",
       "218                              1.0                         1.0   \n",
       "224                              NaN                         NaN   \n",
       "330                              0.0                         0.0   \n",
       "377                              NaN                         NaN   \n",
       "409                              1.0                         1.0   \n",
       "\n",
       "     Small_Intestine_Terminal_Ileum  Spleen  Stomach  Testis  Thyroid  Uterus  \\\n",
       "24                              NaN     NaN      NaN     NaN      NaN     NaN   \n",
       "65                              NaN     NaN      NaN     NaN      NaN     NaN   \n",
       "130                             NaN     NaN      NaN     NaN      NaN     NaN   \n",
       "155                             NaN     NaN      NaN     NaN      NaN     NaN   \n",
       "210                             0.0     0.0      1.0     1.0      1.0     0.0   \n",
       "218                             0.0     0.0      0.0     0.0      1.0     0.0   \n",
       "224                             NaN     NaN      NaN     NaN      NaN     NaN   \n",
       "330                             0.0     0.0      0.0     1.0      0.0     0.0   \n",
       "377                             NaN     NaN      NaN     NaN      NaN     NaN   \n",
       "409                             1.0     1.0      1.0     1.0      1.0     1.0   \n",
       "\n",
       "     Vagina  Whole_Blood  N_GTEx_tissues sQTL    Vernot_allele_origin  \\\n",
       "24      NaN          NaN             NaN   no                 ancient   \n",
       "65      NaN          NaN             NaN   no        archaic-specific   \n",
       "130     NaN          NaN             NaN   no                 ancient   \n",
       "155     NaN          NaN             NaN   no  low-confidence ancient   \n",
       "210     0.0          0.0            15.0  yes                 ancient   \n",
       "218     0.0          0.0            14.0  yes                 ancient   \n",
       "224     NaN          NaN             NaN   no        archaic-specific   \n",
       "330     0.0          1.0             5.0  yes                 ancient   \n",
       "377     NaN          NaN             NaN   no        archaic-specific   \n",
       "409     1.0          1.0            49.0  yes                 ancient   \n",
       "\n",
       "     Browning_allele_origin  \n",
       "24                  ancient  \n",
       "65         archaic-specific  \n",
       "130                 ancient  \n",
       "155  low-confidence ancient  \n",
       "210                 ancient  \n",
       "218                 ancient  \n",
       "224        archaic-specific  \n",
       "330                 ancient  \n",
       "377        archaic-specific  \n",
       "409                 ancient  "
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_2 = data[data['delta_max']>=0.2]\n",
    "data_2.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "id": "d802723a",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "5950"
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(data_2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "id": "5380c373",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "5811"
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(data_2.groupby(['chrom','pos']))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f2bc27a0",
   "metadata": {},
   "source": [
    "Now let's examine multiple annotations in these data."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "id": "59db21a5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SAV_multiple_annotation</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>5679</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>129</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                         count\n",
       "SAV_multiple_annotation       \n",
       "1                         5679\n",
       "2                          129\n",
       "3                            2\n",
       "7                            1"
      ]
     },
     "execution_count": 64,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SAV_2_multiple_annotation = data_2.groupby(['chrom','pos',]).size().reset_index(name='SAV_multiple_annotation')\n",
    "SAV_2_multiple_annotation.groupby('SAV_multiple_annotation').size().to_frame('count')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "83db4855",
   "metadata": {},
   "source": [
    "Now for delta >= 0.5."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "id": "54cda25b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref_allele</th>\n",
       "      <th>alt_allele</th>\n",
       "      <th>ancestral_allele</th>\n",
       "      <th>anc_dev</th>\n",
       "      <th>variant_type</th>\n",
       "      <th>altai_gt</th>\n",
       "      <th>chagyrskaya_gt</th>\n",
       "      <th>denisovan_gt</th>\n",
       "      <th>vindija_gt</th>\n",
       "      <th>altai_gt_boolean</th>\n",
       "      <th>chagyrskaya_gt_boolean</th>\n",
       "      <th>denisovan_gt_boolean</th>\n",
       "      <th>vindija_gt_boolean</th>\n",
       "      <th>distribution</th>\n",
       "      <th>present_in_1KG</th>\n",
       "      <th>1KG_allele_count</th>\n",
       "      <th>1KG_allele_number</th>\n",
       "      <th>1KG_allele_frequency</th>\n",
       "      <th>1KG_EAS_AF</th>\n",
       "      <th>1KG_EUR_AF</th>\n",
       "      <th>1KG_AFR_AF</th>\n",
       "      <th>1KG_AMR_AF</th>\n",
       "      <th>1KG_SAS_AF</th>\n",
       "      <th>1KG_non_ASW_AFR_AF</th>\n",
       "      <th>present_in_gnomAD</th>\n",
       "      <th>gnomAD_allele_count</th>\n",
       "      <th>gnomAD_allele_number</th>\n",
       "      <th>gnomAD_allele_frequency</th>\n",
       "      <th>Vernot_introgressed</th>\n",
       "      <th>Vernot_ancestral_allele</th>\n",
       "      <th>Vernot_derived_allele</th>\n",
       "      <th>Vernot_ancestral_derived_code</th>\n",
       "      <th>Vernot_AFA_AF</th>\n",
       "      <th>Vernot_AFR_AF</th>\n",
       "      <th>Vernot_AMR_AF</th>\n",
       "      <th>Vernot_EAS_AF</th>\n",
       "      <th>Vernot_EUR_AF</th>\n",
       "      <th>Vernot_PNG_AF</th>\n",
       "      <th>Vernot_SAS_AF</th>\n",
       "      <th>Vernot_Denisovan_base</th>\n",
       "      <th>Vernot_Neanderthal_base_1</th>\n",
       "      <th>Vernot_Neanderthal_base_2</th>\n",
       "      <th>Vernot_haplotype_tag</th>\n",
       "      <th>Vernot_introgressed_AF</th>\n",
       "      <th>Browning_introgressed</th>\n",
       "      <th>Browning_ref_alt</th>\n",
       "      <th>Browning_introgressed_AF</th>\n",
       "      <th>annotation</th>\n",
       "      <th>mis_oe</th>\n",
       "      <th>mis_z</th>\n",
       "      <th>lof_oe</th>\n",
       "      <th>lof_z</th>\n",
       "      <th>phyloP</th>\n",
       "      <th>ag_delta</th>\n",
       "      <th>al_delta</th>\n",
       "      <th>dg_delta</th>\n",
       "      <th>dl_delta</th>\n",
       "      <th>delta_max</th>\n",
       "      <th>ag_pos</th>\n",
       "      <th>al_pos</th>\n",
       "      <th>dg_pos</th>\n",
       "      <th>dl_pos</th>\n",
       "      <th>Adipose_Subcutaneous</th>\n",
       "      <th>Adipose_Visceral_Omentum</th>\n",
       "      <th>Adrenal_Gland</th>\n",
       "      <th>Artery_Aorta</th>\n",
       "      <th>Artery_Coronary</th>\n",
       "      <th>Artery_Tibial</th>\n",
       "      <th>Brain_Amygdala</th>\n",
       "      <th>Brain_Anterior_cingulate_cortex_BA24</th>\n",
       "      <th>Brain_Caudate_basal_ganglia</th>\n",
       "      <th>Brain_Cerebellar_Hemisphere</th>\n",
       "      <th>Brain_Cerebellum</th>\n",
       "      <th>Brain_Cortex</th>\n",
       "      <th>Brain_Frontal_Cortex_BA9</th>\n",
       "      <th>Brain_Hippocampus</th>\n",
       "      <th>Brain_Hypothalamus</th>\n",
       "      <th>Brain_Nucleus_accumbens_basal_ganglia</th>\n",
       "      <th>Brain_Putamen_basal_ganglia</th>\n",
       "      <th>Brain_Spinal_cord_cervical_c-1</th>\n",
       "      <th>Brain_Substantia_nigra</th>\n",
       "      <th>Breast_Mammary_Tissue</th>\n",
       "      <th>Cells_Cultured_fibroblasts</th>\n",
       "      <th>Cells_EBV-transformed_lymphocytes</th>\n",
       "      <th>Colon_Sigmoid</th>\n",
       "      <th>Colon_Transverse</th>\n",
       "      <th>Esophagus_Gastroesophageal_Junction</th>\n",
       "      <th>Esophagus_Mucosa</th>\n",
       "      <th>Esophagus_Muscularis</th>\n",
       "      <th>Heart_Atrial_Appendage</th>\n",
       "      <th>Heart_Left_Ventricle</th>\n",
       "      <th>Kidney_Cortex</th>\n",
       "      <th>Liver</th>\n",
       "      <th>Lung</th>\n",
       "      <th>Minor_Salivary_Gland</th>\n",
       "      <th>Muscle_Skeletal</th>\n",
       "      <th>Nerve_Tibial</th>\n",
       "      <th>Ovary</th>\n",
       "      <th>Pancreas</th>\n",
       "      <th>Pituitary</th>\n",
       "      <th>Prostate</th>\n",
       "      <th>Skin_Not_Sun_Exposed_Suprapubic</th>\n",
       "      <th>Skin_Sun_Exposed_Lower_leg</th>\n",
       "      <th>Small_Intestine_Terminal_Ileum</th>\n",
       "      <th>Spleen</th>\n",
       "      <th>Stomach</th>\n",
       "      <th>Testis</th>\n",
       "      <th>Thyroid</th>\n",
       "      <th>Uterus</th>\n",
       "      <th>Vagina</th>\n",
       "      <th>Whole_Blood</th>\n",
       "      <th>N_GTEx_tissues</th>\n",
       "      <th>sQTL</th>\n",
       "      <th>Vernot_allele_origin</th>\n",
       "      <th>Browning_allele_origin</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>409</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1166887</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>yes</td>\n",
       "      <td>898.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.18</td>\n",
       "      <td>0.17</td>\n",
       "      <td>0.13</td>\n",
       "      <td>0.26</td>\n",
       "      <td>0.11</td>\n",
       "      <td>0.17</td>\n",
       "      <td>0.269981</td>\n",
       "      <td>yes</td>\n",
       "      <td>23482.0</td>\n",
       "      <td>143206.0</td>\n",
       "      <td>0.163974</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SDF4</td>\n",
       "      <td>0.90199</td>\n",
       "      <td>0.55251</td>\n",
       "      <td>0.387200</td>\n",
       "      <td>2.23540</td>\n",
       "      <td>0.260</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.52</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.52</td>\n",
       "      <td>-9</td>\n",
       "      <td>-2</td>\n",
       "      <td>0</td>\n",
       "      <td>-4</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>49.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>ancient</td>\n",
       "      <td>ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1881</th>\n",
       "      <td>chr1</td>\n",
       "      <td>2524237</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>ancestral</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>yes</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>yes</td>\n",
       "      <td>2.0</td>\n",
       "      <td>142446.0</td>\n",
       "      <td>0.000014</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MMEL1</td>\n",
       "      <td>0.98394</td>\n",
       "      <td>0.12602</td>\n",
       "      <td>0.768240</td>\n",
       "      <td>1.56890</td>\n",
       "      <td>-0.920</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.56</td>\n",
       "      <td>0.56</td>\n",
       "      <td>-42</td>\n",
       "      <td>10</td>\n",
       "      <td>35</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3147</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3445088</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>Altai</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MEGF6</td>\n",
       "      <td>0.95847</td>\n",
       "      <td>0.45803</td>\n",
       "      <td>0.738560</td>\n",
       "      <td>2.27280</td>\n",
       "      <td>0.205</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.73</td>\n",
       "      <td>0.73</td>\n",
       "      <td>-4</td>\n",
       "      <td>44</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4234</th>\n",
       "      <td>chr1</td>\n",
       "      <td>5935162</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>ancestral</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NPHP4</td>\n",
       "      <td>1.02290</td>\n",
       "      <td>-0.24249</td>\n",
       "      <td>0.781550</td>\n",
       "      <td>1.66690</td>\n",
       "      <td>-0.557</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.91</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1.00</td>\n",
       "      <td>-2</td>\n",
       "      <td>-8</td>\n",
       "      <td>-43</td>\n",
       "      <td>-44</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4624</th>\n",
       "      <td>chr1</td>\n",
       "      <td>6207798</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>yes</td>\n",
       "      <td>2126.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.42</td>\n",
       "      <td>0.64</td>\n",
       "      <td>0.23</td>\n",
       "      <td>0.54</td>\n",
       "      <td>0.17</td>\n",
       "      <td>0.39</td>\n",
       "      <td>0.550682</td>\n",
       "      <td>yes</td>\n",
       "      <td>45620.0</td>\n",
       "      <td>143004.0</td>\n",
       "      <td>0.319012</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHD5</td>\n",
       "      <td>0.56491</td>\n",
       "      <td>5.31680</td>\n",
       "      <td>0.089818</td>\n",
       "      <td>8.44280</td>\n",
       "      <td>-0.331</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.06</td>\n",
       "      <td>0.61</td>\n",
       "      <td>0.61</td>\n",
       "      <td>-31</td>\n",
       "      <td>36</td>\n",
       "      <td>26</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>ancient</td>\n",
       "      <td>ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5440</th>\n",
       "      <td>chr1</td>\n",
       "      <td>6930684</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CAMTA1</td>\n",
       "      <td>0.71229</td>\n",
       "      <td>3.26190</td>\n",
       "      <td>0.091344</td>\n",
       "      <td>7.37100</td>\n",
       "      <td>0.555</td>\n",
       "      <td>0.64</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.64</td>\n",
       "      <td>2</td>\n",
       "      <td>-9</td>\n",
       "      <td>23</td>\n",
       "      <td>-9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7121</th>\n",
       "      <td>chr1</td>\n",
       "      <td>7731131</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CAMTA1</td>\n",
       "      <td>0.71229</td>\n",
       "      <td>3.26190</td>\n",
       "      <td>0.091344</td>\n",
       "      <td>7.37100</td>\n",
       "      <td>0.655</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.71</td>\n",
       "      <td>0.71</td>\n",
       "      <td>48</td>\n",
       "      <td>-5</td>\n",
       "      <td>-34</td>\n",
       "      <td>-5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7464</th>\n",
       "      <td>chr1</td>\n",
       "      <td>8395370</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SLC45A1</td>\n",
       "      <td>0.81558</td>\n",
       "      <td>1.46580</td>\n",
       "      <td>0.459530</td>\n",
       "      <td>2.55930</td>\n",
       "      <td>-0.441</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.67</td>\n",
       "      <td>0.03</td>\n",
       "      <td>0.67</td>\n",
       "      <td>-31</td>\n",
       "      <td>33</td>\n",
       "      <td>-1</td>\n",
       "      <td>-32</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9996</th>\n",
       "      <td>chr1</td>\n",
       "      <td>11009679</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>yes</td>\n",
       "      <td>352.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.07</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.04</td>\n",
       "      <td>0.20</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.207602</td>\n",
       "      <td>yes</td>\n",
       "      <td>11521.0</td>\n",
       "      <td>143214.0</td>\n",
       "      <td>0.080446</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C1orf127</td>\n",
       "      <td>0.91367</td>\n",
       "      <td>0.66009</td>\n",
       "      <td>0.723200</td>\n",
       "      <td>1.38220</td>\n",
       "      <td>-0.354</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.83</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.83</td>\n",
       "      <td>2</td>\n",
       "      <td>-44</td>\n",
       "      <td>2</td>\n",
       "      <td>-28</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14209</th>\n",
       "      <td>chr1</td>\n",
       "      <td>16373634</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>a</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CLCNKB</td>\n",
       "      <td>1.07910</td>\n",
       "      <td>-0.56265</td>\n",
       "      <td>0.934780</td>\n",
       "      <td>0.36979</td>\n",
       "      <td>0.260</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.75</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.75</td>\n",
       "      <td>-1</td>\n",
       "      <td>28</td>\n",
       "      <td>-1</td>\n",
       "      <td>-50</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      chrom       pos ref_allele alt_allele ancestral_allele    anc_dev  \\\n",
       "409    chr1   1166887          T          C                T    derived   \n",
       "1881   chr1   2524237          A          C                C  ancestral   \n",
       "3147   chr1   3445088          C          A                C    derived   \n",
       "4234   chr1   5935162          A          T                T  ancestral   \n",
       "4624   chr1   6207798          A          G                A    derived   \n",
       "5440   chr1   6930684          G          A                G    derived   \n",
       "7121   chr1   7731131          G          A                G    derived   \n",
       "7464   chr1   8395370          A          G                A    derived   \n",
       "9996   chr1  11009679          G          A                G    derived   \n",
       "14209  chr1  16373634          A          G                a    derived   \n",
       "\n",
       "      variant_type altai_gt chagyrskaya_gt denisovan_gt vindija_gt  \\\n",
       "409            snv      0/0            0/0          1/1        0/0   \n",
       "1881           snv      0/0            0/0          1/1        0/0   \n",
       "3147           snv      1/1            0/0          0/0        0/0   \n",
       "4234           snv      1/1            1/1          1/1        1/1   \n",
       "4624           snv      1/1            1/1          1/1        1/1   \n",
       "5440           snv      0/0            0/0          0/1        0/0   \n",
       "7121           snv      0/0            0/0          1/1        0/0   \n",
       "7464           snv      0/0            0/0          1/1        0/0   \n",
       "9996           snv      1/1            1/1          1/1        1/1   \n",
       "14209          snv      0/0            0/0          1/1        0/0   \n",
       "\n",
       "       altai_gt_boolean  chagyrskaya_gt_boolean  denisovan_gt_boolean  \\\n",
       "409               False                   False                  True   \n",
       "1881              False                   False                  True   \n",
       "3147               True                   False                 False   \n",
       "4234               True                    True                  True   \n",
       "4624               True                    True                  True   \n",
       "5440              False                   False                  True   \n",
       "7121              False                   False                  True   \n",
       "7464              False                   False                  True   \n",
       "9996               True                    True                  True   \n",
       "14209             False                   False                  True   \n",
       "\n",
       "       vindija_gt_boolean distribution present_in_1KG  1KG_allele_count  \\\n",
       "409                 False    Denisovan            yes             898.0   \n",
       "1881                False    Denisovan            yes               1.0   \n",
       "3147                False        Altai             no               NaN   \n",
       "4234                 True       Shared             no               NaN   \n",
       "4624                 True       Shared            yes            2126.0   \n",
       "5440                False    Denisovan             no               NaN   \n",
       "7121                False    Denisovan             no               NaN   \n",
       "7464                False    Denisovan             no               NaN   \n",
       "9996                 True       Shared            yes             352.0   \n",
       "14209               False    Denisovan             no               NaN   \n",
       "\n",
       "       1KG_allele_number  1KG_allele_frequency  1KG_EAS_AF  1KG_EUR_AF  \\\n",
       "409               5096.0                  0.18        0.17        0.13   \n",
       "1881              5096.0                  0.00        0.00        0.00   \n",
       "3147                 NaN                   NaN         NaN         NaN   \n",
       "4234                 NaN                   NaN         NaN         NaN   \n",
       "4624              5096.0                  0.42        0.64        0.23   \n",
       "5440                 NaN                   NaN         NaN         NaN   \n",
       "7121                 NaN                   NaN         NaN         NaN   \n",
       "7464                 NaN                   NaN         NaN         NaN   \n",
       "9996              5096.0                  0.07        0.00        0.04   \n",
       "14209                NaN                   NaN         NaN         NaN   \n",
       "\n",
       "       1KG_AFR_AF  1KG_AMR_AF  1KG_SAS_AF  1KG_non_ASW_AFR_AF  \\\n",
       "409          0.26        0.11        0.17            0.269981   \n",
       "1881         0.00        0.00        0.00            0.000000   \n",
       "3147          NaN         NaN         NaN                 NaN   \n",
       "4234          NaN         NaN         NaN                 NaN   \n",
       "4624         0.54        0.17        0.39            0.550682   \n",
       "5440          NaN         NaN         NaN                 NaN   \n",
       "7121          NaN         NaN         NaN                 NaN   \n",
       "7464          NaN         NaN         NaN                 NaN   \n",
       "9996         0.20        0.01        0.02            0.207602   \n",
       "14209         NaN         NaN         NaN                 NaN   \n",
       "\n",
       "      present_in_gnomAD  gnomAD_allele_count  gnomAD_allele_number  \\\n",
       "409                 yes              23482.0              143206.0   \n",
       "1881                yes                  2.0              142446.0   \n",
       "3147                 no                  NaN                   NaN   \n",
       "4234                 no                  NaN                   NaN   \n",
       "4624                yes              45620.0              143004.0   \n",
       "5440                 no                  NaN                   NaN   \n",
       "7121                 no                  NaN                   NaN   \n",
       "7464                 no                  NaN                   NaN   \n",
       "9996                yes              11521.0              143214.0   \n",
       "14209                no                  NaN                   NaN   \n",
       "\n",
       "       gnomAD_allele_frequency Vernot_introgressed Vernot_ancestral_allele  \\\n",
       "409                   0.163974                  no                     NaN   \n",
       "1881                  0.000014                  no                     NaN   \n",
       "3147                       NaN                  no                     NaN   \n",
       "4234                       NaN                  no                     NaN   \n",
       "4624                  0.319012                  no                     NaN   \n",
       "5440                       NaN                  no                     NaN   \n",
       "7121                       NaN                  no                     NaN   \n",
       "7464                       NaN                  no                     NaN   \n",
       "9996                  0.080446                  no                     NaN   \n",
       "14209                      NaN                  no                     NaN   \n",
       "\n",
       "      Vernot_derived_allele  Vernot_ancestral_derived_code  Vernot_AFA_AF  \\\n",
       "409                     NaN                            NaN            NaN   \n",
       "1881                    NaN                            NaN            NaN   \n",
       "3147                    NaN                            NaN            NaN   \n",
       "4234                    NaN                            NaN            NaN   \n",
       "4624                    NaN                            NaN            NaN   \n",
       "5440                    NaN                            NaN            NaN   \n",
       "7121                    NaN                            NaN            NaN   \n",
       "7464                    NaN                            NaN            NaN   \n",
       "9996                    NaN                            NaN            NaN   \n",
       "14209                   NaN                            NaN            NaN   \n",
       "\n",
       "       Vernot_AFR_AF  Vernot_AMR_AF  Vernot_EAS_AF  Vernot_EUR_AF  \\\n",
       "409              NaN            NaN            NaN            NaN   \n",
       "1881             NaN            NaN            NaN            NaN   \n",
       "3147             NaN            NaN            NaN            NaN   \n",
       "4234             NaN            NaN            NaN            NaN   \n",
       "4624             NaN            NaN            NaN            NaN   \n",
       "5440             NaN            NaN            NaN            NaN   \n",
       "7121             NaN            NaN            NaN            NaN   \n",
       "7464             NaN            NaN            NaN            NaN   \n",
       "9996             NaN            NaN            NaN            NaN   \n",
       "14209            NaN            NaN            NaN            NaN   \n",
       "\n",
       "       Vernot_PNG_AF  Vernot_SAS_AF Vernot_Denisovan_base  \\\n",
       "409              NaN            NaN                   NaN   \n",
       "1881             NaN            NaN                   NaN   \n",
       "3147             NaN            NaN                   NaN   \n",
       "4234             NaN            NaN                   NaN   \n",
       "4624             NaN            NaN                   NaN   \n",
       "5440             NaN            NaN                   NaN   \n",
       "7121             NaN            NaN                   NaN   \n",
       "7464             NaN            NaN                   NaN   \n",
       "9996             NaN            NaN                   NaN   \n",
       "14209            NaN            NaN                   NaN   \n",
       "\n",
       "      Vernot_Neanderthal_base_1 Vernot_Neanderthal_base_2  \\\n",
       "409                         NaN                       NaN   \n",
       "1881                        NaN                       NaN   \n",
       "3147                        NaN                       NaN   \n",
       "4234                        NaN                       NaN   \n",
       "4624                        NaN                       NaN   \n",
       "5440                        NaN                       NaN   \n",
       "7121                        NaN                       NaN   \n",
       "7464                        NaN                       NaN   \n",
       "9996                        NaN                       NaN   \n",
       "14209                       NaN                       NaN   \n",
       "\n",
       "      Vernot_haplotype_tag  Vernot_introgressed_AF Browning_introgressed  \\\n",
       "409                    NaN                     NaN                    no   \n",
       "1881                   NaN                     NaN                    no   \n",
       "3147                   NaN                     NaN                    no   \n",
       "4234                   NaN                     NaN                    no   \n",
       "4624                   NaN                     NaN                    no   \n",
       "5440                   NaN                     NaN                    no   \n",
       "7121                   NaN                     NaN                    no   \n",
       "7464                   NaN                     NaN                    no   \n",
       "9996                   NaN                     NaN                    no   \n",
       "14209                  NaN                     NaN                    no   \n",
       "\n",
       "       Browning_ref_alt  Browning_introgressed_AF annotation   mis_oe  \\\n",
       "409                 NaN                       NaN       SDF4  0.90199   \n",
       "1881                NaN                       NaN      MMEL1  0.98394   \n",
       "3147                NaN                       NaN      MEGF6  0.95847   \n",
       "4234                NaN                       NaN      NPHP4  1.02290   \n",
       "4624                NaN                       NaN       CHD5  0.56491   \n",
       "5440                NaN                       NaN     CAMTA1  0.71229   \n",
       "7121                NaN                       NaN     CAMTA1  0.71229   \n",
       "7464                NaN                       NaN    SLC45A1  0.81558   \n",
       "9996                NaN                       NaN   C1orf127  0.91367   \n",
       "14209               NaN                       NaN     CLCNKB  1.07910   \n",
       "\n",
       "         mis_z    lof_oe    lof_z  phyloP  ag_delta  al_delta  dg_delta  \\\n",
       "409    0.55251  0.387200  2.23540   0.260      0.00      0.00      0.52   \n",
       "1881   0.12602  0.768240  1.56890  -0.920      0.01      0.00      0.00   \n",
       "3147   0.45803  0.738560  2.27280   0.205      0.00      0.00      0.00   \n",
       "4234  -0.24249  0.781550  1.66690  -0.557      1.00      0.91      0.00   \n",
       "4624   5.31680  0.089818  8.44280  -0.331      0.00      0.00      0.06   \n",
       "5440   3.26190  0.091344  7.37100   0.555      0.64      0.00      0.01   \n",
       "7121   3.26190  0.091344  7.37100   0.655      0.00      0.00      0.02   \n",
       "7464   1.46580  0.459530  2.55930  -0.441      0.01      0.00      0.67   \n",
       "9996   0.66009  0.723200  1.38220  -0.354      0.00      0.00      0.83   \n",
       "14209 -0.56265  0.934780  0.36979   0.260      0.00      0.00      0.75   \n",
       "\n",
       "       dl_delta  delta_max  ag_pos  al_pos  dg_pos  dl_pos  \\\n",
       "409        0.00       0.52      -9      -2       0      -4   \n",
       "1881       0.56       0.56     -42      10      35       2   \n",
       "3147       0.73       0.73      -4      44       2       1   \n",
       "4234       0.00       1.00      -2      -8     -43     -44   \n",
       "4624       0.61       0.61     -31      36      26       2   \n",
       "5440       0.00       0.64       2      -9      23      -9   \n",
       "7121       0.71       0.71      48      -5     -34      -5   \n",
       "7464       0.03       0.67     -31      33      -1     -32   \n",
       "9996       0.00       0.83       2     -44       2     -28   \n",
       "14209      0.00       0.75      -1      28      -1     -50   \n",
       "\n",
       "       Adipose_Subcutaneous  Adipose_Visceral_Omentum  Adrenal_Gland  \\\n",
       "409                     1.0                       1.0            1.0   \n",
       "1881                    NaN                       NaN            NaN   \n",
       "3147                    NaN                       NaN            NaN   \n",
       "4234                    NaN                       NaN            NaN   \n",
       "4624                    NaN                       NaN            NaN   \n",
       "5440                    NaN                       NaN            NaN   \n",
       "7121                    NaN                       NaN            NaN   \n",
       "7464                    NaN                       NaN            NaN   \n",
       "9996                    0.0                       0.0            0.0   \n",
       "14209                   NaN                       NaN            NaN   \n",
       "\n",
       "       Artery_Aorta  Artery_Coronary  Artery_Tibial  Brain_Amygdala  \\\n",
       "409             1.0              1.0            1.0             1.0   \n",
       "1881            NaN              NaN            NaN             NaN   \n",
       "3147            NaN              NaN            NaN             NaN   \n",
       "4234            NaN              NaN            NaN             NaN   \n",
       "4624            NaN              NaN            NaN             NaN   \n",
       "5440            NaN              NaN            NaN             NaN   \n",
       "7121            NaN              NaN            NaN             NaN   \n",
       "7464            NaN              NaN            NaN             NaN   \n",
       "9996            0.0              0.0            0.0             0.0   \n",
       "14209           NaN              NaN            NaN             NaN   \n",
       "\n",
       "       Brain_Anterior_cingulate_cortex_BA24  Brain_Caudate_basal_ganglia  \\\n",
       "409                                     1.0                          1.0   \n",
       "1881                                    NaN                          NaN   \n",
       "3147                                    NaN                          NaN   \n",
       "4234                                    NaN                          NaN   \n",
       "4624                                    NaN                          NaN   \n",
       "5440                                    NaN                          NaN   \n",
       "7121                                    NaN                          NaN   \n",
       "7464                                    NaN                          NaN   \n",
       "9996                                    0.0                          0.0   \n",
       "14209                                   NaN                          NaN   \n",
       "\n",
       "       Brain_Cerebellar_Hemisphere  Brain_Cerebellum  Brain_Cortex  \\\n",
       "409                            1.0               1.0           1.0   \n",
       "1881                           NaN               NaN           NaN   \n",
       "3147                           NaN               NaN           NaN   \n",
       "4234                           NaN               NaN           NaN   \n",
       "4624                           NaN               NaN           NaN   \n",
       "5440                           NaN               NaN           NaN   \n",
       "7121                           NaN               NaN           NaN   \n",
       "7464                           NaN               NaN           NaN   \n",
       "9996                           0.0               0.0           0.0   \n",
       "14209                          NaN               NaN           NaN   \n",
       "\n",
       "       Brain_Frontal_Cortex_BA9  Brain_Hippocampus  Brain_Hypothalamus  \\\n",
       "409                         1.0                1.0                 1.0   \n",
       "1881                        NaN                NaN                 NaN   \n",
       "3147                        NaN                NaN                 NaN   \n",
       "4234                        NaN                NaN                 NaN   \n",
       "4624                        NaN                NaN                 NaN   \n",
       "5440                        NaN                NaN                 NaN   \n",
       "7121                        NaN                NaN                 NaN   \n",
       "7464                        NaN                NaN                 NaN   \n",
       "9996                        0.0                0.0                 0.0   \n",
       "14209                       NaN                NaN                 NaN   \n",
       "\n",
       "       Brain_Nucleus_accumbens_basal_ganglia  Brain_Putamen_basal_ganglia  \\\n",
       "409                                      1.0                          1.0   \n",
       "1881                                     NaN                          NaN   \n",
       "3147                                     NaN                          NaN   \n",
       "4234                                     NaN                          NaN   \n",
       "4624                                     NaN                          NaN   \n",
       "5440                                     NaN                          NaN   \n",
       "7121                                     NaN                          NaN   \n",
       "7464                                     NaN                          NaN   \n",
       "9996                                     0.0                          0.0   \n",
       "14209                                    NaN                          NaN   \n",
       "\n",
       "       Brain_Spinal_cord_cervical_c-1  Brain_Substantia_nigra  \\\n",
       "409                               1.0                     1.0   \n",
       "1881                              NaN                     NaN   \n",
       "3147                              NaN                     NaN   \n",
       "4234                              NaN                     NaN   \n",
       "4624                              NaN                     NaN   \n",
       "5440                              NaN                     NaN   \n",
       "7121                              NaN                     NaN   \n",
       "7464                              NaN                     NaN   \n",
       "9996                              0.0                     0.0   \n",
       "14209                             NaN                     NaN   \n",
       "\n",
       "       Breast_Mammary_Tissue  Cells_Cultured_fibroblasts  \\\n",
       "409                      1.0                         1.0   \n",
       "1881                     NaN                         NaN   \n",
       "3147                     NaN                         NaN   \n",
       "4234                     NaN                         NaN   \n",
       "4624                     NaN                         NaN   \n",
       "5440                     NaN                         NaN   \n",
       "7121                     NaN                         NaN   \n",
       "7464                     NaN                         NaN   \n",
       "9996                     0.0                         0.0   \n",
       "14209                    NaN                         NaN   \n",
       "\n",
       "       Cells_EBV-transformed_lymphocytes  Colon_Sigmoid  Colon_Transverse  \\\n",
       "409                                  1.0            1.0               1.0   \n",
       "1881                                 NaN            NaN               NaN   \n",
       "3147                                 NaN            NaN               NaN   \n",
       "4234                                 NaN            NaN               NaN   \n",
       "4624                                 NaN            NaN               NaN   \n",
       "5440                                 NaN            NaN               NaN   \n",
       "7121                                 NaN            NaN               NaN   \n",
       "7464                                 NaN            NaN               NaN   \n",
       "9996                                 0.0            0.0               0.0   \n",
       "14209                                NaN            NaN               NaN   \n",
       "\n",
       "       Esophagus_Gastroesophageal_Junction  Esophagus_Mucosa  \\\n",
       "409                                    1.0               1.0   \n",
       "1881                                   NaN               NaN   \n",
       "3147                                   NaN               NaN   \n",
       "4234                                   NaN               NaN   \n",
       "4624                                   NaN               NaN   \n",
       "5440                                   NaN               NaN   \n",
       "7121                                   NaN               NaN   \n",
       "7464                                   NaN               NaN   \n",
       "9996                                   0.0               0.0   \n",
       "14209                                  NaN               NaN   \n",
       "\n",
       "       Esophagus_Muscularis  Heart_Atrial_Appendage  Heart_Left_Ventricle  \\\n",
       "409                     1.0                     1.0                   1.0   \n",
       "1881                    NaN                     NaN                   NaN   \n",
       "3147                    NaN                     NaN                   NaN   \n",
       "4234                    NaN                     NaN                   NaN   \n",
       "4624                    NaN                     NaN                   NaN   \n",
       "5440                    NaN                     NaN                   NaN   \n",
       "7121                    NaN                     NaN                   NaN   \n",
       "7464                    NaN                     NaN                   NaN   \n",
       "9996                    0.0                     0.0                   0.0   \n",
       "14209                   NaN                     NaN                   NaN   \n",
       "\n",
       "       Kidney_Cortex  Liver  Lung  Minor_Salivary_Gland  Muscle_Skeletal  \\\n",
       "409              1.0    1.0   1.0                   1.0              1.0   \n",
       "1881             NaN    NaN   NaN                   NaN              NaN   \n",
       "3147             NaN    NaN   NaN                   NaN              NaN   \n",
       "4234             NaN    NaN   NaN                   NaN              NaN   \n",
       "4624             NaN    NaN   NaN                   NaN              NaN   \n",
       "5440             NaN    NaN   NaN                   NaN              NaN   \n",
       "7121             NaN    NaN   NaN                   NaN              NaN   \n",
       "7464             NaN    NaN   NaN                   NaN              NaN   \n",
       "9996             0.0    0.0   0.0                   0.0              1.0   \n",
       "14209            NaN    NaN   NaN                   NaN              NaN   \n",
       "\n",
       "       Nerve_Tibial  Ovary  Pancreas  Pituitary  Prostate  \\\n",
       "409             1.0    1.0       1.0        1.0       1.0   \n",
       "1881            NaN    NaN       NaN        NaN       NaN   \n",
       "3147            NaN    NaN       NaN        NaN       NaN   \n",
       "4234            NaN    NaN       NaN        NaN       NaN   \n",
       "4624            NaN    NaN       NaN        NaN       NaN   \n",
       "5440            NaN    NaN       NaN        NaN       NaN   \n",
       "7121            NaN    NaN       NaN        NaN       NaN   \n",
       "7464            NaN    NaN       NaN        NaN       NaN   \n",
       "9996            0.0    0.0       0.0        0.0       0.0   \n",
       "14209           NaN    NaN       NaN        NaN       NaN   \n",
       "\n",
       "       Skin_Not_Sun_Exposed_Suprapubic  Skin_Sun_Exposed_Lower_leg  \\\n",
       "409                                1.0                         1.0   \n",
       "1881                               NaN                         NaN   \n",
       "3147                               NaN                         NaN   \n",
       "4234                               NaN                         NaN   \n",
       "4624                               NaN                         NaN   \n",
       "5440                               NaN                         NaN   \n",
       "7121                               NaN                         NaN   \n",
       "7464                               NaN                         NaN   \n",
       "9996                               0.0                         0.0   \n",
       "14209                              NaN                         NaN   \n",
       "\n",
       "       Small_Intestine_Terminal_Ileum  Spleen  Stomach  Testis  Thyroid  \\\n",
       "409                               1.0     1.0      1.0     1.0      1.0   \n",
       "1881                              NaN     NaN      NaN     NaN      NaN   \n",
       "3147                              NaN     NaN      NaN     NaN      NaN   \n",
       "4234                              NaN     NaN      NaN     NaN      NaN   \n",
       "4624                              NaN     NaN      NaN     NaN      NaN   \n",
       "5440                              NaN     NaN      NaN     NaN      NaN   \n",
       "7121                              NaN     NaN      NaN     NaN      NaN   \n",
       "7464                              NaN     NaN      NaN     NaN      NaN   \n",
       "9996                              0.0     0.0      0.0     0.0      0.0   \n",
       "14209                             NaN     NaN      NaN     NaN      NaN   \n",
       "\n",
       "       Uterus  Vagina  Whole_Blood  N_GTEx_tissues sQTL  \\\n",
       "409       1.0     1.0          1.0            49.0  yes   \n",
       "1881      NaN     NaN          NaN             NaN   no   \n",
       "3147      NaN     NaN          NaN             NaN   no   \n",
       "4234      NaN     NaN          NaN             NaN   no   \n",
       "4624      NaN     NaN          NaN             NaN   no   \n",
       "5440      NaN     NaN          NaN             NaN   no   \n",
       "7121      NaN     NaN          NaN             NaN   no   \n",
       "7464      NaN     NaN          NaN             NaN   no   \n",
       "9996      0.0     0.0          0.0             1.0  yes   \n",
       "14209     NaN     NaN          NaN             NaN   no   \n",
       "\n",
       "         Vernot_allele_origin  Browning_allele_origin  \n",
       "409                   ancient                 ancient  \n",
       "1881   low-confidence ancient  low-confidence ancient  \n",
       "3147         archaic-specific        archaic-specific  \n",
       "4234         archaic-specific        archaic-specific  \n",
       "4624                  ancient                 ancient  \n",
       "5440         archaic-specific        archaic-specific  \n",
       "7121         archaic-specific        archaic-specific  \n",
       "7464         archaic-specific        archaic-specific  \n",
       "9996   low-confidence ancient  low-confidence ancient  \n",
       "14209        archaic-specific        archaic-specific  "
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_5 = data[data['delta_max']>=0.5]\n",
    "data_5.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "id": "1b31fefd",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1049"
      ]
     },
     "execution_count": 66,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(data_5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "id": "a7a3fd56",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1017"
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(data_5.groupby(['chrom','pos']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "id": "14cc6efa",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SAV_multiple_annotation</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>986</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                         count\n",
       "SAV_multiple_annotation       \n",
       "1                          986\n",
       "2                           30\n",
       "3                            1"
      ]
     },
     "execution_count": 68,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SAV_5_multiple_annotation = data_5.groupby(['chrom','pos',]).size().reset_index(name='SAV_multiple_annotation')\n",
    "SAV_5_multiple_annotation.groupby('SAV_multiple_annotation').size().to_frame('count')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cd1dcacb",
   "metadata": {},
   "source": [
    "Now let's count the number of each class."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "id": "88064271",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1511"
      ]
     },
     "execution_count": 69,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(data_2[(data_2['ag_delta'] >= 0.2)])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "id": "c5a6dd47",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1377"
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(data_2[(data_2['al_delta'] >= 0.2)])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "id": "785af1ef",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1894"
      ]
     },
     "execution_count": 71,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(data_2[(data_2['dg_delta'] >= 0.2)])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "id": "6f6f0b81",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1647"
      ]
     },
     "execution_count": 72,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(data_2[(data_2['dl_delta'] >= 0.2)])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "id": "31e135a3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "180"
      ]
     },
     "execution_count": 73,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(data_5[(data_5['ag_delta'] >= 0.5)])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "id": "df794585",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "191"
      ]
     },
     "execution_count": 74,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(data_5[(data_5['al_delta'] >= 0.5)])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "id": "27275785",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "390"
      ]
     },
     "execution_count": 75,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(data_5[(data_5['dg_delta'] >= 0.5)])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "id": "ce5bee97",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "349"
      ]
     },
     "execution_count": 76,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(data_5[(data_5['dl_delta'] >= 0.5)])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e27c3a19",
   "metadata": {},
   "source": [
    "Let's get a count of each class where the effect is unique to that class per threshold."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "id": "0e74c132",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "616"
      ]
     },
     "execution_count": 77,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(data_2[(data_2['ag_delta'] >= 0.2) & (data_2['al_delta'] == 0) & (data_2['dg_delta'] == 0) & (data_2['dl_delta'] == 0)])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "id": "d0f5140b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "569"
      ]
     },
     "execution_count": 78,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(data_2[(data_2['al_delta'] >= 0.2) & (data_2['ag_delta'] == 0) & (data_2['dg_delta'] == 0) & (data_2['dl_delta'] == 0)])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "id": "310737a9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "899"
      ]
     },
     "execution_count": 79,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(data_2[(data_2['dg_delta'] >= 0.2) & (data_2['ag_delta'] == 0) & (data_2['al_delta'] == 0) & (data_2['dl_delta'] == 0) ])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "id": "33a5881e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "761"
      ]
     },
     "execution_count": 80,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(data_2[(data_2['dl_delta'] >= 0.2) & (data_2['ag_delta'] == 0) & (data_2['al_delta'] == 0) & (data_2['dg_delta'] == 0)])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "id": "9b61cd67",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "39"
      ]
     },
     "execution_count": 81,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(data_5[(data_5['ag_delta'] >= 0.5) & (data_5['al_delta'] == 0) & (data_5['dg_delta'] == 0) & (data_5['dl_delta'] == 0)])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "id": "d699f5de",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "37"
      ]
     },
     "execution_count": 82,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(data_5[(data_5['al_delta'] >= 0.5) & (data_5['ag_delta'] == 0) & (data_5['dg_delta'] == 0) & (data_5['dl_delta'] == 0)])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "id": "f42dd065",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "146"
      ]
     },
     "execution_count": 83,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(data_5[(data_5['dg_delta'] >= 0.5) & (data_5['ag_delta'] == 0) & (data_5['al_delta'] == 0) & (data_5['dl_delta'] == 0)])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "id": "361739e7",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "133"
      ]
     },
     "execution_count": 84,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(data_5[(data_5['dl_delta'] >= 0.5) & (data_5['ag_delta'] == 0) & (data_5['al_delta'] == 0) & (data_5['dg_delta'] == 0)])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c31e5219",
   "metadata": {},
   "source": [
    "## Variant Distribution <a class = 'anchor' id = 'variantdistribution'></a>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "id": "c1668eca",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>distribution</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>distribution</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Altai</th>\n",
       "      <td>81916</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Chagyrskaya</th>\n",
       "      <td>53675</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Denisovan</th>\n",
       "      <td>411492</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Late Neanderthal</th>\n",
       "      <td>44651</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Neanderthal</th>\n",
       "      <td>266965</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Other</th>\n",
       "      <td>104455</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Shared</th>\n",
       "      <td>573197</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Vindija</th>\n",
       "      <td>70999</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  distribution\n",
       "distribution                  \n",
       "Altai                    81916\n",
       "Chagyrskaya              53675\n",
       "Denisovan               411492\n",
       "Late Neanderthal         44651\n",
       "Neanderthal             266965\n",
       "Other                   104455\n",
       "Shared                  573197\n",
       "Vindija                  70999"
      ]
     },
     "execution_count": 85,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data.groupby(['distribution']).size().to_frame('distribution')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "id": "08d22aec",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>distribution</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>distribution</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Altai</th>\n",
       "      <td>399</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Chagyrskaya</th>\n",
       "      <td>218</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Denisovan</th>\n",
       "      <td>1492</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Late Neanderthal</th>\n",
       "      <td>172</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Neanderthal</th>\n",
       "      <td>973</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Other</th>\n",
       "      <td>456</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Shared</th>\n",
       "      <td>1933</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Vindija</th>\n",
       "      <td>307</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  distribution\n",
       "distribution                  \n",
       "Altai                      399\n",
       "Chagyrskaya                218\n",
       "Denisovan                 1492\n",
       "Late Neanderthal           172\n",
       "Neanderthal                973\n",
       "Other                      456\n",
       "Shared                    1933\n",
       "Vindija                    307"
      ]
     },
     "execution_count": 86,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_2.groupby(['distribution']).size().to_frame('distribution')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "id": "81786359",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>distribution</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>distribution</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Altai</th>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Chagyrskaya</th>\n",
       "      <td>35</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Denisovan</th>\n",
       "      <td>254</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Late Neanderthal</th>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Neanderthal</th>\n",
       "      <td>178</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Other</th>\n",
       "      <td>84</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Shared</th>\n",
       "      <td>328</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Vindija</th>\n",
       "      <td>73</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  distribution\n",
       "distribution                  \n",
       "Altai                       75\n",
       "Chagyrskaya                 35\n",
       "Denisovan                  254\n",
       "Late Neanderthal            22\n",
       "Neanderthal                178\n",
       "Other                       84\n",
       "Shared                     328\n",
       "Vindija                     73"
      ]
     },
     "execution_count": 87,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_5.groupby(['distribution']).size().to_frame('distribution')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5413c4f0",
   "metadata": {},
   "source": [
    "Let's get the total count per individual for delta >= 0.2."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "id": "a9706a02",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "3705"
      ]
     },
     "execution_count": 88,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(data_2[data_2['altai_gt_boolean'] == True])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "id": "1d700c7a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "3482"
      ]
     },
     "execution_count": 89,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(data_2[data_2['chagyrskaya_gt_boolean'] == True])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "id": "ff3b6d9c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "3671"
      ]
     },
     "execution_count": 90,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(data_2[data_2['denisovan_gt_boolean'] == True])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "id": "4a2ac4b0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "3627"
      ]
     },
     "execution_count": 91,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(data_2[data_2['vindija_gt_boolean'] == True])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8bc2cae2",
   "metadata": {},
   "source": [
    "## Delta Correlations <a class = 'anchor' id = 'deltacorrelations'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "124e782d",
   "metadata": {},
   "source": [
    "Let's examine the correlations between deltas. There are a lot of zeroes so we'll first exclude rows with all zeroes as this will bias our analysis."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "id": "e0e38a92",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref_allele</th>\n",
       "      <th>alt_allele</th>\n",
       "      <th>ancestral_allele</th>\n",
       "      <th>anc_dev</th>\n",
       "      <th>variant_type</th>\n",
       "      <th>altai_gt</th>\n",
       "      <th>chagyrskaya_gt</th>\n",
       "      <th>denisovan_gt</th>\n",
       "      <th>vindija_gt</th>\n",
       "      <th>altai_gt_boolean</th>\n",
       "      <th>chagyrskaya_gt_boolean</th>\n",
       "      <th>denisovan_gt_boolean</th>\n",
       "      <th>vindija_gt_boolean</th>\n",
       "      <th>distribution</th>\n",
       "      <th>present_in_1KG</th>\n",
       "      <th>1KG_allele_count</th>\n",
       "      <th>1KG_allele_number</th>\n",
       "      <th>1KG_allele_frequency</th>\n",
       "      <th>1KG_EAS_AF</th>\n",
       "      <th>1KG_EUR_AF</th>\n",
       "      <th>1KG_AFR_AF</th>\n",
       "      <th>1KG_AMR_AF</th>\n",
       "      <th>1KG_SAS_AF</th>\n",
       "      <th>1KG_non_ASW_AFR_AF</th>\n",
       "      <th>present_in_gnomAD</th>\n",
       "      <th>gnomAD_allele_count</th>\n",
       "      <th>gnomAD_allele_number</th>\n",
       "      <th>gnomAD_allele_frequency</th>\n",
       "      <th>Vernot_introgressed</th>\n",
       "      <th>Vernot_ancestral_allele</th>\n",
       "      <th>Vernot_derived_allele</th>\n",
       "      <th>Vernot_ancestral_derived_code</th>\n",
       "      <th>Vernot_AFA_AF</th>\n",
       "      <th>Vernot_AFR_AF</th>\n",
       "      <th>Vernot_AMR_AF</th>\n",
       "      <th>Vernot_EAS_AF</th>\n",
       "      <th>Vernot_EUR_AF</th>\n",
       "      <th>Vernot_PNG_AF</th>\n",
       "      <th>Vernot_SAS_AF</th>\n",
       "      <th>Vernot_Denisovan_base</th>\n",
       "      <th>Vernot_Neanderthal_base_1</th>\n",
       "      <th>Vernot_Neanderthal_base_2</th>\n",
       "      <th>Vernot_haplotype_tag</th>\n",
       "      <th>Vernot_introgressed_AF</th>\n",
       "      <th>Browning_introgressed</th>\n",
       "      <th>Browning_ref_alt</th>\n",
       "      <th>Browning_introgressed_AF</th>\n",
       "      <th>annotation</th>\n",
       "      <th>mis_oe</th>\n",
       "      <th>mis_z</th>\n",
       "      <th>lof_oe</th>\n",
       "      <th>lof_z</th>\n",
       "      <th>phyloP</th>\n",
       "      <th>ag_delta</th>\n",
       "      <th>al_delta</th>\n",
       "      <th>dg_delta</th>\n",
       "      <th>dl_delta</th>\n",
       "      <th>delta_max</th>\n",
       "      <th>ag_pos</th>\n",
       "      <th>al_pos</th>\n",
       "      <th>dg_pos</th>\n",
       "      <th>dl_pos</th>\n",
       "      <th>Adipose_Subcutaneous</th>\n",
       "      <th>Adipose_Visceral_Omentum</th>\n",
       "      <th>Adrenal_Gland</th>\n",
       "      <th>Artery_Aorta</th>\n",
       "      <th>Artery_Coronary</th>\n",
       "      <th>Artery_Tibial</th>\n",
       "      <th>Brain_Amygdala</th>\n",
       "      <th>Brain_Anterior_cingulate_cortex_BA24</th>\n",
       "      <th>Brain_Caudate_basal_ganglia</th>\n",
       "      <th>Brain_Cerebellar_Hemisphere</th>\n",
       "      <th>Brain_Cerebellum</th>\n",
       "      <th>Brain_Cortex</th>\n",
       "      <th>Brain_Frontal_Cortex_BA9</th>\n",
       "      <th>Brain_Hippocampus</th>\n",
       "      <th>Brain_Hypothalamus</th>\n",
       "      <th>Brain_Nucleus_accumbens_basal_ganglia</th>\n",
       "      <th>Brain_Putamen_basal_ganglia</th>\n",
       "      <th>Brain_Spinal_cord_cervical_c-1</th>\n",
       "      <th>Brain_Substantia_nigra</th>\n",
       "      <th>Breast_Mammary_Tissue</th>\n",
       "      <th>Cells_Cultured_fibroblasts</th>\n",
       "      <th>Cells_EBV-transformed_lymphocytes</th>\n",
       "      <th>Colon_Sigmoid</th>\n",
       "      <th>Colon_Transverse</th>\n",
       "      <th>Esophagus_Gastroesophageal_Junction</th>\n",
       "      <th>Esophagus_Mucosa</th>\n",
       "      <th>Esophagus_Muscularis</th>\n",
       "      <th>Heart_Atrial_Appendage</th>\n",
       "      <th>Heart_Left_Ventricle</th>\n",
       "      <th>Kidney_Cortex</th>\n",
       "      <th>Liver</th>\n",
       "      <th>Lung</th>\n",
       "      <th>Minor_Salivary_Gland</th>\n",
       "      <th>Muscle_Skeletal</th>\n",
       "      <th>Nerve_Tibial</th>\n",
       "      <th>Ovary</th>\n",
       "      <th>Pancreas</th>\n",
       "      <th>Pituitary</th>\n",
       "      <th>Prostate</th>\n",
       "      <th>Skin_Not_Sun_Exposed_Suprapubic</th>\n",
       "      <th>Skin_Sun_Exposed_Lower_leg</th>\n",
       "      <th>Small_Intestine_Terminal_Ileum</th>\n",
       "      <th>Spleen</th>\n",
       "      <th>Stomach</th>\n",
       "      <th>Testis</th>\n",
       "      <th>Thyroid</th>\n",
       "      <th>Uterus</th>\n",
       "      <th>Vagina</th>\n",
       "      <th>Whole_Blood</th>\n",
       "      <th>N_GTEx_tissues</th>\n",
       "      <th>sQTL</th>\n",
       "      <th>Vernot_allele_origin</th>\n",
       "      <th>Browning_allele_origin</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>861808</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>yes</td>\n",
       "      <td>3479.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.68</td>\n",
       "      <td>0.55</td>\n",
       "      <td>0.97</td>\n",
       "      <td>0.34</td>\n",
       "      <td>0.83</td>\n",
       "      <td>0.88</td>\n",
       "      <td>0.313840</td>\n",
       "      <td>yes</td>\n",
       "      <td>112289.0</td>\n",
       "      <td>143198.0</td>\n",
       "      <td>0.784152</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>1.5082</td>\n",
       "      <td>-3.43610</td>\n",
       "      <td>0.89656</td>\n",
       "      <td>0.47484</td>\n",
       "      <td>-0.683</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.01</td>\n",
       "      <td>-29</td>\n",
       "      <td>-27</td>\n",
       "      <td>24</td>\n",
       "      <td>-20</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>ancient</td>\n",
       "      <td>ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>chr1</td>\n",
       "      <td>863978</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>g</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>yes</td>\n",
       "      <td>497.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.15</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.18</td>\n",
       "      <td>0.11</td>\n",
       "      <td>0.03</td>\n",
       "      <td>0.180312</td>\n",
       "      <td>yes</td>\n",
       "      <td>9475.0</td>\n",
       "      <td>143334.0</td>\n",
       "      <td>0.066104</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AL645608.1</td>\n",
       "      <td>1.2217</td>\n",
       "      <td>-0.64548</td>\n",
       "      <td>0.73515</td>\n",
       "      <td>0.49579</td>\n",
       "      <td>0.197</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.01</td>\n",
       "      <td>-15</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>-14</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>ancient</td>\n",
       "      <td>ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>chr1</td>\n",
       "      <td>864678</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>1.5082</td>\n",
       "      <td>-3.43610</td>\n",
       "      <td>0.89656</td>\n",
       "      <td>0.47484</td>\n",
       "      <td>0.486</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.03</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.03</td>\n",
       "      <td>-48</td>\n",
       "      <td>26</td>\n",
       "      <td>46</td>\n",
       "      <td>-14</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>chr1</td>\n",
       "      <td>864678</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AL645608.1</td>\n",
       "      <td>1.2217</td>\n",
       "      <td>-0.64548</td>\n",
       "      <td>0.73515</td>\n",
       "      <td>0.49579</td>\n",
       "      <td>0.486</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.02</td>\n",
       "      <td>-2</td>\n",
       "      <td>44</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>chr1</td>\n",
       "      <td>864701</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>yes</td>\n",
       "      <td>10.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>yes</td>\n",
       "      <td>14.0</td>\n",
       "      <td>143284.0</td>\n",
       "      <td>0.000098</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>1.5082</td>\n",
       "      <td>-3.43610</td>\n",
       "      <td>0.89656</td>\n",
       "      <td>0.47484</td>\n",
       "      <td>-0.423</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.08</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.08</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>23</td>\n",
       "      <td>-10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "      <td>introgressed</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>chr1</td>\n",
       "      <td>864701</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>yes</td>\n",
       "      <td>10.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>yes</td>\n",
       "      <td>14.0</td>\n",
       "      <td>143284.0</td>\n",
       "      <td>0.000098</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>AL645608.1</td>\n",
       "      <td>1.2217</td>\n",
       "      <td>-0.64548</td>\n",
       "      <td>0.73515</td>\n",
       "      <td>0.49579</td>\n",
       "      <td>-0.423</td>\n",
       "      <td>0.04</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.04</td>\n",
       "      <td>21</td>\n",
       "      <td>-18</td>\n",
       "      <td>-28</td>\n",
       "      <td>8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "      <td>introgressed</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>chr1</td>\n",
       "      <td>864726</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>yes</td>\n",
       "      <td>4859.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.95</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.83</td>\n",
       "      <td>0.99</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.812865</td>\n",
       "      <td>yes</td>\n",
       "      <td>137320.0</td>\n",
       "      <td>143300.0</td>\n",
       "      <td>0.958269</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>1.5082</td>\n",
       "      <td>-3.43610</td>\n",
       "      <td>0.89656</td>\n",
       "      <td>0.47484</td>\n",
       "      <td>0.398</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.22</td>\n",
       "      <td>0.22</td>\n",
       "      <td>17</td>\n",
       "      <td>-2</td>\n",
       "      <td>43</td>\n",
       "      <td>-2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>ancient</td>\n",
       "      <td>ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>chr1</td>\n",
       "      <td>864726</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>yes</td>\n",
       "      <td>4859.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.95</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.83</td>\n",
       "      <td>0.99</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.812865</td>\n",
       "      <td>yes</td>\n",
       "      <td>137320.0</td>\n",
       "      <td>143300.0</td>\n",
       "      <td>0.958269</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AL645608.1</td>\n",
       "      <td>1.2217</td>\n",
       "      <td>-0.64548</td>\n",
       "      <td>0.73515</td>\n",
       "      <td>0.49579</td>\n",
       "      <td>0.398</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.03</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.03</td>\n",
       "      <td>1</td>\n",
       "      <td>-4</td>\n",
       "      <td>36</td>\n",
       "      <td>-43</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>ancient</td>\n",
       "      <td>ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>chr1</td>\n",
       "      <td>864898</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>yes</td>\n",
       "      <td>10.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>yes</td>\n",
       "      <td>8.0</td>\n",
       "      <td>143316.0</td>\n",
       "      <td>0.000056</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>1.5082</td>\n",
       "      <td>-3.43610</td>\n",
       "      <td>0.89656</td>\n",
       "      <td>0.47484</td>\n",
       "      <td>0.454</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.01</td>\n",
       "      <td>-1</td>\n",
       "      <td>48</td>\n",
       "      <td>-1</td>\n",
       "      <td>48</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "      <td>introgressed</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>chr1</td>\n",
       "      <td>864898</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>yes</td>\n",
       "      <td>10.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>yes</td>\n",
       "      <td>8.0</td>\n",
       "      <td>143316.0</td>\n",
       "      <td>0.000056</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>AL645608.1</td>\n",
       "      <td>1.2217</td>\n",
       "      <td>-0.64548</td>\n",
       "      <td>0.73515</td>\n",
       "      <td>0.49579</td>\n",
       "      <td>0.454</td>\n",
       "      <td>0.03</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.03</td>\n",
       "      <td>-4</td>\n",
       "      <td>1</td>\n",
       "      <td>-4</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "      <td>introgressed</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   chrom     pos ref_allele alt_allele ancestral_allele  anc_dev variant_type  \\\n",
       "0   chr1  861808          A          G                A  derived          snv   \n",
       "19  chr1  863978          G          A                g  derived          snv   \n",
       "20  chr1  864678          C          T                C  derived          snv   \n",
       "21  chr1  864678          C          T                C  derived          snv   \n",
       "22  chr1  864701          A          G                A  derived          snv   \n",
       "23  chr1  864701          A          G                A  derived          snv   \n",
       "24  chr1  864726          T          A                T  derived          snv   \n",
       "25  chr1  864726          T          A                T  derived          snv   \n",
       "26  chr1  864898          A          G                A  derived          snv   \n",
       "27  chr1  864898          A          G                A  derived          snv   \n",
       "\n",
       "   altai_gt chagyrskaya_gt denisovan_gt vindija_gt  altai_gt_boolean  \\\n",
       "0       0/0            0/0          1/1        0/0             False   \n",
       "19      1/1            1/1          0/0        1/1              True   \n",
       "20      0/0            0/0          1/1        0/0             False   \n",
       "21      0/0            0/0          1/1        0/0             False   \n",
       "22      1/1            1/1          0/0        1/1              True   \n",
       "23      1/1            1/1          0/0        1/1              True   \n",
       "24      1/1            1/1          1/1        1/1              True   \n",
       "25      1/1            1/1          1/1        1/1              True   \n",
       "26      1/1            1/1          0/0        0/1              True   \n",
       "27      1/1            1/1          0/0        0/1              True   \n",
       "\n",
       "    chagyrskaya_gt_boolean  denisovan_gt_boolean  vindija_gt_boolean  \\\n",
       "0                    False                  True               False   \n",
       "19                    True                 False                True   \n",
       "20                   False                  True               False   \n",
       "21                   False                  True               False   \n",
       "22                    True                 False                True   \n",
       "23                    True                 False                True   \n",
       "24                    True                  True                True   \n",
       "25                    True                  True                True   \n",
       "26                    True                 False                True   \n",
       "27                    True                 False                True   \n",
       "\n",
       "   distribution present_in_1KG  1KG_allele_count  1KG_allele_number  \\\n",
       "0     Denisovan            yes            3479.0             5096.0   \n",
       "19  Neanderthal            yes             497.0             5096.0   \n",
       "20    Denisovan             no               NaN                NaN   \n",
       "21    Denisovan             no               NaN                NaN   \n",
       "22  Neanderthal            yes              10.0             5096.0   \n",
       "23  Neanderthal            yes              10.0             5096.0   \n",
       "24       Shared            yes            4859.0             5096.0   \n",
       "25       Shared            yes            4859.0             5096.0   \n",
       "26  Neanderthal            yes              10.0             5096.0   \n",
       "27  Neanderthal            yes              10.0             5096.0   \n",
       "\n",
       "    1KG_allele_frequency  1KG_EAS_AF  1KG_EUR_AF  1KG_AFR_AF  1KG_AMR_AF  \\\n",
       "0                   0.68        0.55        0.97        0.34        0.83   \n",
       "19                  0.10        0.15        0.01        0.18        0.11   \n",
       "20                   NaN         NaN         NaN         NaN         NaN   \n",
       "21                   NaN         NaN         NaN         NaN         NaN   \n",
       "22                  0.00        0.01        0.00        0.00        0.00   \n",
       "23                  0.00        0.01        0.00        0.00        0.00   \n",
       "24                  0.95        1.00        1.00        0.83        0.99   \n",
       "25                  0.95        1.00        1.00        0.83        0.99   \n",
       "26                  0.00        0.01        0.00        0.00        0.00   \n",
       "27                  0.00        0.01        0.00        0.00        0.00   \n",
       "\n",
       "    1KG_SAS_AF  1KG_non_ASW_AFR_AF present_in_gnomAD  gnomAD_allele_count  \\\n",
       "0         0.88            0.313840               yes             112289.0   \n",
       "19        0.03            0.180312               yes               9475.0   \n",
       "20         NaN                 NaN                no                  NaN   \n",
       "21         NaN                 NaN                no                  NaN   \n",
       "22        0.00            0.000000               yes                 14.0   \n",
       "23        0.00            0.000000               yes                 14.0   \n",
       "24        1.00            0.812865               yes             137320.0   \n",
       "25        1.00            0.812865               yes             137320.0   \n",
       "26        0.00            0.000000               yes                  8.0   \n",
       "27        0.00            0.000000               yes                  8.0   \n",
       "\n",
       "    gnomAD_allele_number  gnomAD_allele_frequency Vernot_introgressed  \\\n",
       "0               143198.0                 0.784152                  no   \n",
       "19              143334.0                 0.066104                  no   \n",
       "20                   NaN                      NaN                  no   \n",
       "21                   NaN                      NaN                  no   \n",
       "22              143284.0                 0.000098                  no   \n",
       "23              143284.0                 0.000098                  no   \n",
       "24              143300.0                 0.958269                  no   \n",
       "25              143300.0                 0.958269                  no   \n",
       "26              143316.0                 0.000056                  no   \n",
       "27              143316.0                 0.000056                  no   \n",
       "\n",
       "   Vernot_ancestral_allele Vernot_derived_allele  \\\n",
       "0                      NaN                   NaN   \n",
       "19                     NaN                   NaN   \n",
       "20                     NaN                   NaN   \n",
       "21                     NaN                   NaN   \n",
       "22                     NaN                   NaN   \n",
       "23                     NaN                   NaN   \n",
       "24                     NaN                   NaN   \n",
       "25                     NaN                   NaN   \n",
       "26                     NaN                   NaN   \n",
       "27                     NaN                   NaN   \n",
       "\n",
       "    Vernot_ancestral_derived_code  Vernot_AFA_AF  Vernot_AFR_AF  \\\n",
       "0                             NaN            NaN            NaN   \n",
       "19                            NaN            NaN            NaN   \n",
       "20                            NaN            NaN            NaN   \n",
       "21                            NaN            NaN            NaN   \n",
       "22                            NaN            NaN            NaN   \n",
       "23                            NaN            NaN            NaN   \n",
       "24                            NaN            NaN            NaN   \n",
       "25                            NaN            NaN            NaN   \n",
       "26                            NaN            NaN            NaN   \n",
       "27                            NaN            NaN            NaN   \n",
       "\n",
       "    Vernot_AMR_AF  Vernot_EAS_AF  Vernot_EUR_AF  Vernot_PNG_AF  Vernot_SAS_AF  \\\n",
       "0             NaN            NaN            NaN            NaN            NaN   \n",
       "19            NaN            NaN            NaN            NaN            NaN   \n",
       "20            NaN            NaN            NaN            NaN            NaN   \n",
       "21            NaN            NaN            NaN            NaN            NaN   \n",
       "22            NaN            NaN            NaN            NaN            NaN   \n",
       "23            NaN            NaN            NaN            NaN            NaN   \n",
       "24            NaN            NaN            NaN            NaN            NaN   \n",
       "25            NaN            NaN            NaN            NaN            NaN   \n",
       "26            NaN            NaN            NaN            NaN            NaN   \n",
       "27            NaN            NaN            NaN            NaN            NaN   \n",
       "\n",
       "   Vernot_Denisovan_base Vernot_Neanderthal_base_1 Vernot_Neanderthal_base_2  \\\n",
       "0                    NaN                       NaN                       NaN   \n",
       "19                   NaN                       NaN                       NaN   \n",
       "20                   NaN                       NaN                       NaN   \n",
       "21                   NaN                       NaN                       NaN   \n",
       "22                   NaN                       NaN                       NaN   \n",
       "23                   NaN                       NaN                       NaN   \n",
       "24                   NaN                       NaN                       NaN   \n",
       "25                   NaN                       NaN                       NaN   \n",
       "26                   NaN                       NaN                       NaN   \n",
       "27                   NaN                       NaN                       NaN   \n",
       "\n",
       "   Vernot_haplotype_tag  Vernot_introgressed_AF Browning_introgressed  \\\n",
       "0                   NaN                     NaN                    no   \n",
       "19                  NaN                     NaN                    no   \n",
       "20                  NaN                     NaN                    no   \n",
       "21                  NaN                     NaN                    no   \n",
       "22                  NaN                     NaN                   yes   \n",
       "23                  NaN                     NaN                   yes   \n",
       "24                  NaN                     NaN                    no   \n",
       "25                  NaN                     NaN                    no   \n",
       "26                  NaN                     NaN                   yes   \n",
       "27                  NaN                     NaN                   yes   \n",
       "\n",
       "    Browning_ref_alt  Browning_introgressed_AF  annotation  mis_oe    mis_z  \\\n",
       "0                NaN                       NaN      SAMD11  1.5082 -3.43610   \n",
       "19               NaN                       NaN  AL645608.1  1.2217 -0.64548   \n",
       "20               NaN                       NaN      SAMD11  1.5082 -3.43610   \n",
       "21               NaN                       NaN  AL645608.1  1.2217 -0.64548   \n",
       "22               1.0                       0.0      SAMD11  1.5082 -3.43610   \n",
       "23               1.0                       0.0  AL645608.1  1.2217 -0.64548   \n",
       "24               NaN                       NaN      SAMD11  1.5082 -3.43610   \n",
       "25               NaN                       NaN  AL645608.1  1.2217 -0.64548   \n",
       "26               1.0                       0.0      SAMD11  1.5082 -3.43610   \n",
       "27               1.0                       0.0  AL645608.1  1.2217 -0.64548   \n",
       "\n",
       "     lof_oe    lof_z  phyloP  ag_delta  al_delta  dg_delta  dl_delta  \\\n",
       "0   0.89656  0.47484  -0.683      0.00      0.00      0.01      0.00   \n",
       "19  0.73515  0.49579   0.197      0.00      0.00      0.00      0.01   \n",
       "20  0.89656  0.47484   0.486      0.00      0.00      0.03      0.00   \n",
       "21  0.73515  0.49579   0.486      0.01      0.02      0.00      0.00   \n",
       "22  0.89656  0.47484  -0.423      0.00      0.00      0.08      0.02   \n",
       "23  0.73515  0.49579  -0.423      0.04      0.00      0.00      0.00   \n",
       "24  0.89656  0.47484   0.398      0.00      0.00      0.02      0.22   \n",
       "25  0.73515  0.49579   0.398      0.00      0.03      0.00      0.00   \n",
       "26  0.89656  0.47484   0.454      0.00      0.00      0.01      0.00   \n",
       "27  0.73515  0.49579   0.454      0.03      0.00      0.00      0.00   \n",
       "\n",
       "    delta_max  ag_pos  al_pos  dg_pos  dl_pos  Adipose_Subcutaneous  \\\n",
       "0        0.01     -29     -27      24     -20                   0.0   \n",
       "19       0.01     -15       4       2     -14                   NaN   \n",
       "20       0.03     -48      26      46     -14                   NaN   \n",
       "21       0.02      -2      44       3       5                   NaN   \n",
       "22       0.08       1       3      23     -10                   NaN   \n",
       "23       0.04      21     -18     -28       8                   NaN   \n",
       "24       0.22      17      -2      43      -2                   NaN   \n",
       "25       0.03       1      -4      36     -43                   NaN   \n",
       "26       0.01      -1      48      -1      48                   NaN   \n",
       "27       0.03      -4       1      -4       2                   NaN   \n",
       "\n",
       "    Adipose_Visceral_Omentum  Adrenal_Gland  Artery_Aorta  Artery_Coronary  \\\n",
       "0                        0.0            0.0           0.0              0.0   \n",
       "19                       NaN            NaN           NaN              NaN   \n",
       "20                       NaN            NaN           NaN              NaN   \n",
       "21                       NaN            NaN           NaN              NaN   \n",
       "22                       NaN            NaN           NaN              NaN   \n",
       "23                       NaN            NaN           NaN              NaN   \n",
       "24                       NaN            NaN           NaN              NaN   \n",
       "25                       NaN            NaN           NaN              NaN   \n",
       "26                       NaN            NaN           NaN              NaN   \n",
       "27                       NaN            NaN           NaN              NaN   \n",
       "\n",
       "    Artery_Tibial  Brain_Amygdala  Brain_Anterior_cingulate_cortex_BA24  \\\n",
       "0             0.0             0.0                                   0.0   \n",
       "19            NaN             NaN                                   NaN   \n",
       "20            NaN             NaN                                   NaN   \n",
       "21            NaN             NaN                                   NaN   \n",
       "22            NaN             NaN                                   NaN   \n",
       "23            NaN             NaN                                   NaN   \n",
       "24            NaN             NaN                                   NaN   \n",
       "25            NaN             NaN                                   NaN   \n",
       "26            NaN             NaN                                   NaN   \n",
       "27            NaN             NaN                                   NaN   \n",
       "\n",
       "    Brain_Caudate_basal_ganglia  Brain_Cerebellar_Hemisphere  \\\n",
       "0                           0.0                          0.0   \n",
       "19                          NaN                          NaN   \n",
       "20                          NaN                          NaN   \n",
       "21                          NaN                          NaN   \n",
       "22                          NaN                          NaN   \n",
       "23                          NaN                          NaN   \n",
       "24                          NaN                          NaN   \n",
       "25                          NaN                          NaN   \n",
       "26                          NaN                          NaN   \n",
       "27                          NaN                          NaN   \n",
       "\n",
       "    Brain_Cerebellum  Brain_Cortex  Brain_Frontal_Cortex_BA9  \\\n",
       "0                0.0           0.0                       0.0   \n",
       "19               NaN           NaN                       NaN   \n",
       "20               NaN           NaN                       NaN   \n",
       "21               NaN           NaN                       NaN   \n",
       "22               NaN           NaN                       NaN   \n",
       "23               NaN           NaN                       NaN   \n",
       "24               NaN           NaN                       NaN   \n",
       "25               NaN           NaN                       NaN   \n",
       "26               NaN           NaN                       NaN   \n",
       "27               NaN           NaN                       NaN   \n",
       "\n",
       "    Brain_Hippocampus  Brain_Hypothalamus  \\\n",
       "0                 0.0                 0.0   \n",
       "19                NaN                 NaN   \n",
       "20                NaN                 NaN   \n",
       "21                NaN                 NaN   \n",
       "22                NaN                 NaN   \n",
       "23                NaN                 NaN   \n",
       "24                NaN                 NaN   \n",
       "25                NaN                 NaN   \n",
       "26                NaN                 NaN   \n",
       "27                NaN                 NaN   \n",
       "\n",
       "    Brain_Nucleus_accumbens_basal_ganglia  Brain_Putamen_basal_ganglia  \\\n",
       "0                                     0.0                          0.0   \n",
       "19                                    NaN                          NaN   \n",
       "20                                    NaN                          NaN   \n",
       "21                                    NaN                          NaN   \n",
       "22                                    NaN                          NaN   \n",
       "23                                    NaN                          NaN   \n",
       "24                                    NaN                          NaN   \n",
       "25                                    NaN                          NaN   \n",
       "26                                    NaN                          NaN   \n",
       "27                                    NaN                          NaN   \n",
       "\n",
       "    Brain_Spinal_cord_cervical_c-1  Brain_Substantia_nigra  \\\n",
       "0                              0.0                     0.0   \n",
       "19                             NaN                     NaN   \n",
       "20                             NaN                     NaN   \n",
       "21                             NaN                     NaN   \n",
       "22                             NaN                     NaN   \n",
       "23                             NaN                     NaN   \n",
       "24                             NaN                     NaN   \n",
       "25                             NaN                     NaN   \n",
       "26                             NaN                     NaN   \n",
       "27                             NaN                     NaN   \n",
       "\n",
       "    Breast_Mammary_Tissue  Cells_Cultured_fibroblasts  \\\n",
       "0                     0.0                         0.0   \n",
       "19                    NaN                         NaN   \n",
       "20                    NaN                         NaN   \n",
       "21                    NaN                         NaN   \n",
       "22                    NaN                         NaN   \n",
       "23                    NaN                         NaN   \n",
       "24                    NaN                         NaN   \n",
       "25                    NaN                         NaN   \n",
       "26                    NaN                         NaN   \n",
       "27                    NaN                         NaN   \n",
       "\n",
       "    Cells_EBV-transformed_lymphocytes  Colon_Sigmoid  Colon_Transverse  \\\n",
       "0                                 0.0            0.0               0.0   \n",
       "19                                NaN            NaN               NaN   \n",
       "20                                NaN            NaN               NaN   \n",
       "21                                NaN            NaN               NaN   \n",
       "22                                NaN            NaN               NaN   \n",
       "23                                NaN            NaN               NaN   \n",
       "24                                NaN            NaN               NaN   \n",
       "25                                NaN            NaN               NaN   \n",
       "26                                NaN            NaN               NaN   \n",
       "27                                NaN            NaN               NaN   \n",
       "\n",
       "    Esophagus_Gastroesophageal_Junction  Esophagus_Mucosa  \\\n",
       "0                                   0.0               0.0   \n",
       "19                                  NaN               NaN   \n",
       "20                                  NaN               NaN   \n",
       "21                                  NaN               NaN   \n",
       "22                                  NaN               NaN   \n",
       "23                                  NaN               NaN   \n",
       "24                                  NaN               NaN   \n",
       "25                                  NaN               NaN   \n",
       "26                                  NaN               NaN   \n",
       "27                                  NaN               NaN   \n",
       "\n",
       "    Esophagus_Muscularis  Heart_Atrial_Appendage  Heart_Left_Ventricle  \\\n",
       "0                    0.0                     0.0                   0.0   \n",
       "19                   NaN                     NaN                   NaN   \n",
       "20                   NaN                     NaN                   NaN   \n",
       "21                   NaN                     NaN                   NaN   \n",
       "22                   NaN                     NaN                   NaN   \n",
       "23                   NaN                     NaN                   NaN   \n",
       "24                   NaN                     NaN                   NaN   \n",
       "25                   NaN                     NaN                   NaN   \n",
       "26                   NaN                     NaN                   NaN   \n",
       "27                   NaN                     NaN                   NaN   \n",
       "\n",
       "    Kidney_Cortex  Liver  Lung  Minor_Salivary_Gland  Muscle_Skeletal  \\\n",
       "0             0.0    0.0   0.0                   0.0              0.0   \n",
       "19            NaN    NaN   NaN                   NaN              NaN   \n",
       "20            NaN    NaN   NaN                   NaN              NaN   \n",
       "21            NaN    NaN   NaN                   NaN              NaN   \n",
       "22            NaN    NaN   NaN                   NaN              NaN   \n",
       "23            NaN    NaN   NaN                   NaN              NaN   \n",
       "24            NaN    NaN   NaN                   NaN              NaN   \n",
       "25            NaN    NaN   NaN                   NaN              NaN   \n",
       "26            NaN    NaN   NaN                   NaN              NaN   \n",
       "27            NaN    NaN   NaN                   NaN              NaN   \n",
       "\n",
       "    Nerve_Tibial  Ovary  Pancreas  Pituitary  Prostate  \\\n",
       "0            0.0    0.0       0.0        0.0       0.0   \n",
       "19           NaN    NaN       NaN        NaN       NaN   \n",
       "20           NaN    NaN       NaN        NaN       NaN   \n",
       "21           NaN    NaN       NaN        NaN       NaN   \n",
       "22           NaN    NaN       NaN        NaN       NaN   \n",
       "23           NaN    NaN       NaN        NaN       NaN   \n",
       "24           NaN    NaN       NaN        NaN       NaN   \n",
       "25           NaN    NaN       NaN        NaN       NaN   \n",
       "26           NaN    NaN       NaN        NaN       NaN   \n",
       "27           NaN    NaN       NaN        NaN       NaN   \n",
       "\n",
       "    Skin_Not_Sun_Exposed_Suprapubic  Skin_Sun_Exposed_Lower_leg  \\\n",
       "0                               0.0                         0.0   \n",
       "19                              NaN                         NaN   \n",
       "20                              NaN                         NaN   \n",
       "21                              NaN                         NaN   \n",
       "22                              NaN                         NaN   \n",
       "23                              NaN                         NaN   \n",
       "24                              NaN                         NaN   \n",
       "25                              NaN                         NaN   \n",
       "26                              NaN                         NaN   \n",
       "27                              NaN                         NaN   \n",
       "\n",
       "    Small_Intestine_Terminal_Ileum  Spleen  Stomach  Testis  Thyroid  Uterus  \\\n",
       "0                              0.0     0.0      0.0     0.0      0.0     0.0   \n",
       "19                             NaN     NaN      NaN     NaN      NaN     NaN   \n",
       "20                             NaN     NaN      NaN     NaN      NaN     NaN   \n",
       "21                             NaN     NaN      NaN     NaN      NaN     NaN   \n",
       "22                             NaN     NaN      NaN     NaN      NaN     NaN   \n",
       "23                             NaN     NaN      NaN     NaN      NaN     NaN   \n",
       "24                             NaN     NaN      NaN     NaN      NaN     NaN   \n",
       "25                             NaN     NaN      NaN     NaN      NaN     NaN   \n",
       "26                             NaN     NaN      NaN     NaN      NaN     NaN   \n",
       "27                             NaN     NaN      NaN     NaN      NaN     NaN   \n",
       "\n",
       "    Vagina  Whole_Blood  N_GTEx_tissues sQTL    Vernot_allele_origin  \\\n",
       "0      0.0          1.0             1.0  yes                 ancient   \n",
       "19     NaN          NaN             NaN   no                 ancient   \n",
       "20     NaN          NaN             NaN   no        archaic-specific   \n",
       "21     NaN          NaN             NaN   no        archaic-specific   \n",
       "22     NaN          NaN             NaN   no  low-confidence ancient   \n",
       "23     NaN          NaN             NaN   no  low-confidence ancient   \n",
       "24     NaN          NaN             NaN   no                 ancient   \n",
       "25     NaN          NaN             NaN   no                 ancient   \n",
       "26     NaN          NaN             NaN   no  low-confidence ancient   \n",
       "27     NaN          NaN             NaN   no  low-confidence ancient   \n",
       "\n",
       "   Browning_allele_origin  \n",
       "0                 ancient  \n",
       "19                ancient  \n",
       "20       archaic-specific  \n",
       "21       archaic-specific  \n",
       "22           introgressed  \n",
       "23           introgressed  \n",
       "24                ancient  \n",
       "25                ancient  \n",
       "26           introgressed  \n",
       "27           introgressed  "
      ]
     },
     "execution_count": 92,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_nonzero = data[(data['ag_delta']>0) | (data['al_delta']>0) | (data['dg_delta']>0) | (data['dl_delta']>0)]\n",
    "data_nonzero.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "id": "569d520a",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "188302"
      ]
     },
     "execution_count": 93,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(data_nonzero)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "edff43fd",
   "metadata": {},
   "source": [
    "Run correlations."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "id": "f7c268d8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-0.14267746909602813 0.0\n"
     ]
    }
   ],
   "source": [
    "rho, p = spearmanr(data_nonzero['ag_delta'], data_nonzero['al_delta'])\n",
    "print(rho,p)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "id": "8675ef06",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-0.2239766968064798 0.0\n"
     ]
    }
   ],
   "source": [
    "rho, p = spearmanr(data_nonzero['ag_delta'], data_nonzero['dg_delta'])\n",
    "print(rho,p)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "id": "463d1861",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-0.34023522469107176 0.0\n"
     ]
    }
   ],
   "source": [
    "rho, p = spearmanr(data_nonzero['ag_delta'], data_nonzero['dl_delta'])\n",
    "print(rho,p)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "id": "7a816ddc",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-0.33985645661346087 0.0\n"
     ]
    }
   ],
   "source": [
    "rho, p = spearmanr(data_nonzero['al_delta'], data_nonzero['dg_delta'])\n",
    "print(rho,p)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "id": "32f74ccd",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-0.2146624536999448 0.0\n"
     ]
    }
   ],
   "source": [
    "rho, p = spearmanr(data_nonzero['al_delta'], data_nonzero['dl_delta'])\n",
    "print(rho,p)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "id": "e789875c",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-0.2036868319219267 0.0\n"
     ]
    }
   ],
   "source": [
    "rho, p = spearmanr(data_nonzero['dg_delta'], data_nonzero['dl_delta'])\n",
    "print(rho,p)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "bcd791c5",
   "metadata": {},
   "source": [
    "Repeat for splice altering variants."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "id": "afc64ba0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.1500250528757491 2.7088750503541857e-31\n"
     ]
    }
   ],
   "source": [
    "rho, p = spearmanr(data_2['ag_delta'], data_2['al_delta'])\n",
    "print(rho,p)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "id": "a123d68c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-0.32319408269234423 1.0314796151065124e-144\n"
     ]
    }
   ],
   "source": [
    "rho, p = spearmanr(data_2['ag_delta'], data_2['dg_delta'])\n",
    "print(rho,p)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "id": "857507b1",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-0.46608147540520983 0.0\n"
     ]
    }
   ],
   "source": [
    "rho, p = spearmanr(data_2['ag_delta'], data_2['dl_delta'])\n",
    "print(rho,p)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "id": "92e50027",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-0.48795009291846625 0.0\n"
     ]
    }
   ],
   "source": [
    "rho, p = spearmanr(data_2['al_delta'], data_2['dg_delta'])\n",
    "print(rho,p)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "id": "b6978e77",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-0.27635622346475713 9.11316178310797e-105\n"
     ]
    }
   ],
   "source": [
    "rho, p = spearmanr(data_2['al_delta'], data_2['dl_delta'])\n",
    "print(rho,p)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "id": "fb06b89d",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-0.02349927032949021 0.06990668106164923\n"
     ]
    }
   ],
   "source": [
    "rho, p = spearmanr(data_2['dg_delta'], data_2['dl_delta'])\n",
    "print(rho,p)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0a293b08",
   "metadata": {},
   "source": [
    "The p-values above are listed as 0, which means those values are less than the floating point minimum. Let's get that value for the manuscript."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "id": "1d1899ae",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2.2250738585072014e-308"
      ]
     },
     "execution_count": 106,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sys.float_info.min"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "38d16619",
   "metadata": {},
   "source": [
    "## Multiple Deltas <a class = 'anchor' id = 'multipledeltas'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2a385fab",
   "metadata": {},
   "source": [
    "How many variants have a single variant effect? This result should match the sum of the four individual counts above."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "id": "fb21aac0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2845"
      ]
     },
     "execution_count": 107,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "delta_counts = (data_2[['ag_delta','al_delta','dg_delta','dl_delta']] > 0).sum(1).to_frame('count')\n",
    "len(delta_counts[delta_counts['count'] == 1])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "65b572f9",
   "metadata": {},
   "source": [
    "And it does. Bravo."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c1b43b1a",
   "metadata": {},
   "source": [
    "Let's assess the number of variants that result in both a gain and a loss for the donor and acceptor. Presumably, these are strong candidates for alternative splicing."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "id": "0459ac9a",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "119"
      ]
     },
     "execution_count": 108,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(data_nonzero[(data_nonzero['ag_delta'] >= 0.2) & (data_nonzero['al_delta'] >= 0.2)])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "id": "fadde7cb",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "108"
      ]
     },
     "execution_count": 109,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(data_nonzero[(data_nonzero['dg_delta'] >= 0.2) & (data_nonzero['dl_delta'] >= 0.2)])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1f008987",
   "metadata": {},
   "source": [
    "Now let's examine when both the donor and acceptor are affected. These events likely represent exon skipping (Jaganathan et al. 2019)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "id": "37514d77",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "119"
      ]
     },
     "execution_count": 110,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(data_nonzero[(data_nonzero['ag_delta'] >= 0.2) & (data_nonzero['dg_delta'] >= 0.2)])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "id": "2b8cb04d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "6"
      ]
     },
     "execution_count": 111,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(data_nonzero[(data_nonzero['ag_delta'] >= 0.2) & (data_nonzero['dl_delta'] >= 0.2)])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "id": "c338e9f0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "6"
      ]
     },
     "execution_count": 112,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(data_nonzero[(data_nonzero['al_delta'] >= 0.2) & (data_nonzero['dg_delta'] >= 0.2)])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "id": "7a28ad2e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "126"
      ]
     },
     "execution_count": 113,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(data_nonzero[(data_nonzero['al_delta'] >= 0.2) & (data_nonzero['dl_delta'] >= 0.2)])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a0cff6b7",
   "metadata": {},
   "source": [
    "Let's save these cases of exon skipping to a dataframe."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "id": "f4b174ad",
   "metadata": {},
   "outputs": [],
   "source": [
    "exon_skipping_2 = data_nonzero[((data_nonzero['ag_delta'] >= 0.2) & (data_nonzero['dg_delta'] >= 0.2)) | ((data_nonzero['ag_delta'] >= 0.2) & (data_nonzero['dl_delta'] >= 0.2)) | ((data_nonzero['al_delta'] >= 0.2) & (data_nonzero['dg_delta'] >= 0.2)) | ((data_nonzero['al_delta'] >= 0.2) & (data_nonzero['dl_delta'] >= 0.2))] "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "id": "3c7c46a4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "252"
      ]
     },
     "execution_count": 115,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(exon_skipping_2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "id": "1b004658",
   "metadata": {},
   "outputs": [],
   "source": [
    "exon_skipping_5 = data_nonzero[((data_nonzero['ag_delta'] >= 0.5) & (data_nonzero['dg_delta'] >= 0.5)) | ((data_nonzero['ag_delta'] >= 0.5) & (data_nonzero['dl_delta'] >= 0.5)) | ((data_nonzero['al_delta'] >= 0.5) & (data_nonzero['dg_delta'] >= 0.5)) | ((data_nonzero['al_delta'] >= 0.5) & (data_nonzero['dl_delta'] >= 0.5))] "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "id": "32ad523c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "24"
      ]
     },
     "execution_count": 117,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(exon_skipping_5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "id": "7e486c1a",
   "metadata": {},
   "outputs": [],
   "source": [
    "exon_skipping_2.to_csv('../dataframes/exon_skipping_2.txt', sep=\"\\t\", header = True, index = False)\n",
    "exon_skipping_5.to_csv('../dataframes/exon_skipping_5.txt', sep=\"\\t\", header = True, index = False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "eb59de53",
   "metadata": {},
   "source": [
    "Let's check the distribution and allele origin of these exon skipping variants out of curiosity."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "id": "50a9ef27",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>distribution</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>distribution</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Altai</th>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Chagyrskaya</th>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Denisovan</th>\n",
       "      <td>63</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Late Neanderthal</th>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Neanderthal</th>\n",
       "      <td>49</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Other</th>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Shared</th>\n",
       "      <td>88</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Vindija</th>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  distribution\n",
       "distribution                  \n",
       "Altai                       11\n",
       "Chagyrskaya                  8\n",
       "Denisovan                   63\n",
       "Late Neanderthal             9\n",
       "Neanderthal                 49\n",
       "Other                       15\n",
       "Shared                      88\n",
       "Vindija                      9"
      ]
     },
     "execution_count": 119,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "exon_skipping_2.groupby(['distribution']).size().to_frame('distribution')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "id": "7aa98260",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>origin</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Vernot_allele_origin</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ancient</th>\n",
       "      <td>92</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>archaic-specific</th>\n",
       "      <td>88</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>introgressed</th>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>low-confidence ancient</th>\n",
       "      <td>57</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                        origin\n",
       "Vernot_allele_origin          \n",
       "ancient                     92\n",
       "archaic-specific            88\n",
       "introgressed                15\n",
       "low-confidence ancient      57"
      ]
     },
     "execution_count": 120,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "exon_skipping_2.groupby(['Vernot_allele_origin']).size().to_frame('origin')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "id": "6cec14f3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>origin</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Browning_allele_origin</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ancient</th>\n",
       "      <td>88</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>archaic-specific</th>\n",
       "      <td>88</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>introgressed</th>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>low-confidence ancient</th>\n",
       "      <td>52</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                        origin\n",
       "Browning_allele_origin        \n",
       "ancient                     88\n",
       "archaic-specific            88\n",
       "introgressed                24\n",
       "low-confidence ancient      52"
      ]
     },
     "execution_count": 121,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "exon_skipping_2.groupby(['Browning_allele_origin']).size().to_frame('origin')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ffa9fb1f",
   "metadata": {},
   "source": [
    "## SAV Genotypes <a class = 'anchor' id = 'SAVgenotypes'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a7296c3a",
   "metadata": {},
   "source": [
    "We just examined the distribution of genotypes but let's do this again for SAVs."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "id": "7e4c63d1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>size</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>altai_gt</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>./.</th>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0/0</th>\n",
       "      <td>2227</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0/1</th>\n",
       "      <td>537</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1/0</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1/1</th>\n",
       "      <td>3167</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          size\n",
       "altai_gt      \n",
       "./.         18\n",
       "0/0       2227\n",
       "0/1        537\n",
       "1/0          1\n",
       "1/1       3167"
      ]
     },
     "execution_count": 122,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_2.groupby(['altai_gt']).size().to_frame('size')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "id": "7d184aa5",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>size</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>altai_gt</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>./.</th>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0/0</th>\n",
       "      <td>386</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0/1</th>\n",
       "      <td>101</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1/1</th>\n",
       "      <td>558</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          size\n",
       "altai_gt      \n",
       "./.          4\n",
       "0/0        386\n",
       "0/1        101\n",
       "1/1        558"
      ]
     },
     "execution_count": 123,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_5.groupby(['altai_gt']).size().to_frame('size')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "id": "477800e4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>size</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chagyrskaya_gt</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>./.</th>\n",
       "      <td>266</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0/0</th>\n",
       "      <td>2202</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0/1</th>\n",
       "      <td>395</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1/0</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1/1</th>\n",
       "      <td>3085</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                size\n",
       "chagyrskaya_gt      \n",
       "./.              266\n",
       "0/0             2202\n",
       "0/1              395\n",
       "1/0                2\n",
       "1/1             3085"
      ]
     },
     "execution_count": 124,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_2.groupby(['chagyrskaya_gt']).size().to_frame('size')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "id": "640a286d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>size</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chagyrskaya_gt</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>./.</th>\n",
       "      <td>69</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0/0</th>\n",
       "      <td>394</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0/1</th>\n",
       "      <td>60</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1/0</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1/1</th>\n",
       "      <td>525</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                size\n",
       "chagyrskaya_gt      \n",
       "./.               69\n",
       "0/0              394\n",
       "0/1               60\n",
       "1/0                1\n",
       "1/1              525"
      ]
     },
     "execution_count": 125,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_5.groupby(['chagyrskaya_gt']).size().to_frame('size')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "id": "9d4d23fd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>size</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>denisovan_gt</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>./.</th>\n",
       "      <td>157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0/0</th>\n",
       "      <td>2122</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0/1</th>\n",
       "      <td>584</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1/0</th>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1/1</th>\n",
       "      <td>3084</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              size\n",
       "denisovan_gt      \n",
       "./.            157\n",
       "0/0           2122\n",
       "0/1            584\n",
       "1/0              3\n",
       "1/1           3084"
      ]
     },
     "execution_count": 126,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_2.groupby(['denisovan_gt']).size().to_frame('size')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "id": "cea8c55f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>size</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>denisovan_gt</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>./.</th>\n",
       "      <td>45</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0/0</th>\n",
       "      <td>376</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0/1</th>\n",
       "      <td>106</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1/0</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1/1</th>\n",
       "      <td>521</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              size\n",
       "denisovan_gt      \n",
       "./.             45\n",
       "0/0            376\n",
       "0/1            106\n",
       "1/0              1\n",
       "1/1            521"
      ]
     },
     "execution_count": 127,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_5.groupby(['denisovan_gt']).size().to_frame('size')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 128,
   "id": "04e4bcf7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>size</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>vindija_gt</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>./.</th>\n",
       "      <td>163</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0/0</th>\n",
       "      <td>2160</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0/1</th>\n",
       "      <td>496</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1/1</th>\n",
       "      <td>3131</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            size\n",
       "vindija_gt      \n",
       "./.          163\n",
       "0/0         2160\n",
       "0/1          496\n",
       "1/1         3131"
      ]
     },
     "execution_count": 128,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_2.groupby(['vindija_gt']).size().to_frame('size')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "id": "eb4d72df",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>size</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>vindija_gt</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>./.</th>\n",
       "      <td>35</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0/0</th>\n",
       "      <td>361</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0/1</th>\n",
       "      <td>108</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1/1</th>\n",
       "      <td>545</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            size\n",
       "vindija_gt      \n",
       "./.           35\n",
       "0/0          361\n",
       "0/1          108\n",
       "1/1          545"
      ]
     },
     "execution_count": 129,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_5.groupby(['vindija_gt']).size().to_frame('size')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "819c67e7",
   "metadata": {},
   "source": [
    "## 1KG vs gnomAD SAVs <a class = 'anchor' id = '1KGgnomADSAVs'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "15c27a4b",
   "metadata": {},
   "source": [
    "Let's briefly compare 1KG and gnomAD SAVs. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 130,
   "id": "923f3543",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>N</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>present_in_gnomAD</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>no</th>\n",
       "      <td>1705</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>yes</th>\n",
       "      <td>4245</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                      N\n",
       "present_in_gnomAD      \n",
       "no                 1705\n",
       "yes                4245"
      ]
     },
     "execution_count": 130,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_2.groupby(['present_in_gnomAD']).size().to_frame('N')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "id": "12be64cd",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>origin</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>present_in_gnomAD</th>\n",
       "      <th>present_in_1KG</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">no</th>\n",
       "      <th>no</th>\n",
       "      <td>1674</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>yes</th>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">yes</th>\n",
       "      <th>no</th>\n",
       "      <td>751</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>yes</th>\n",
       "      <td>3494</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                  origin\n",
       "present_in_gnomAD present_in_1KG        \n",
       "no                no                1674\n",
       "                  yes                 31\n",
       "yes               no                 751\n",
       "                  yes               3494"
      ]
     },
     "execution_count": 131,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_2.groupby(['present_in_gnomAD','present_in_1KG']).size().to_frame('origin')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1b90ae30",
   "metadata": {},
   "source": [
    "How many rare variants (allele frequency < 0.0001) occur only in gnomAD? These are likely recurrent mutations."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 132,
   "id": "0ec92245",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "512"
      ]
     },
     "execution_count": 132,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(data_2[(data_2['present_in_gnomAD'] == 'yes') & (data_2['present_in_1KG'] == 'no') & (data_2['gnomAD_allele_frequency'] < 0.0001)])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "91e9a4f8",
   "metadata": {},
   "source": [
    "Let's test for CpG enrichment among those sites. Enrichment for CpGs supports the hypothesis that they are recurrent mutations."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "id": "38225e0d",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>ref_allele</th>\n",
       "      <th>alt_allele</th>\n",
       "      <th>allele_count</th>\n",
       "      <th>allele_number</th>\n",
       "      <th>allele_frequency</th>\n",
       "      <th>C_dinucleotide</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10</td>\n",
       "      <td>183370</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>1</td>\n",
       "      <td>143172</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>CT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10</td>\n",
       "      <td>184616</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>437</td>\n",
       "      <td>143284</td>\n",
       "      <td>0.003050</td>\n",
       "      <td>n\u0007</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>185171</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>132752</td>\n",
       "      <td>143248</td>\n",
       "      <td>0.926728</td>\n",
       "      <td>n\u0007</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>185358</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>132720</td>\n",
       "      <td>143208</td>\n",
       "      <td>0.926764</td>\n",
       "      <td>CA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr10</td>\n",
       "      <td>185611</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>1</td>\n",
       "      <td>141008</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>CG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr10</td>\n",
       "      <td>185611</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "      <td>141008</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>CG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr10</td>\n",
       "      <td>185612</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>432</td>\n",
       "      <td>138944</td>\n",
       "      <td>0.003109</td>\n",
       "      <td>n\u0007</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr10</td>\n",
       "      <td>188101</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>7</td>\n",
       "      <td>143212</td>\n",
       "      <td>0.000049</td>\n",
       "      <td>CG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr10</td>\n",
       "      <td>188859</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>437</td>\n",
       "      <td>143332</td>\n",
       "      <td>0.003049</td>\n",
       "      <td>n\u0007</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr10</td>\n",
       "      <td>190851</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>129638</td>\n",
       "      <td>140998</td>\n",
       "      <td>0.919431</td>\n",
       "      <td>n\u0007</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   chrom   start ref_allele alt_allele  allele_count  allele_number  \\\n",
       "0  chr10  183370          C          G             1         143172   \n",
       "1  chr10  184616          T          G           437         143284   \n",
       "2  chr10  185171          G          T        132752         143248   \n",
       "3  chr10  185358          C          T        132720         143208   \n",
       "4  chr10  185611          C          G             1         141008   \n",
       "5  chr10  185611          C          T             1         141008   \n",
       "6  chr10  185612          G          A           432         138944   \n",
       "7  chr10  188101          C          T             7         143212   \n",
       "8  chr10  188859          A          G           437         143332   \n",
       "9  chr10  190851          T          C        129638         140998   \n",
       "\n",
       "   allele_frequency C_dinucleotide  \n",
       "0          0.000007             CT  \n",
       "1          0.003050             n\u0007  \n",
       "2          0.926728             n\u0007  \n",
       "3          0.926764             CA  \n",
       "4          0.000007             CG  \n",
       "5          0.000007             CG  \n",
       "6          0.003109             n\u0007  \n",
       "7          0.000049             CG  \n",
       "8          0.003049             n\u0007  \n",
       "9          0.919431             n\u0007  "
      ]
     },
     "execution_count": 133,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gnomAD_CpGs_header = ['chrom','start','ref_allele','alt_allele','allele_count','allele_number','allele_frequency','C_dinucleotide']\n",
    "gnomAD_CpGs = pd.read_csv('../archaic_variants_in_humans/archaic_SNVs_in_gnomAD_allele_frequencies_hg19_with_CpG_annotation.txt', sep = '\\t', names = gnomAD_CpGs_header)\n",
    "gnomAD_CpGs.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "id": "875ade0b",
   "metadata": {},
   "outputs": [],
   "source": [
    "def CpG_enrichment_rare_common(rare_allele_frequency, common_allele_frequency):\n",
    "    rare_CpGs = len(gnomAD_CpGs[(gnomAD_CpGs['allele_frequency'] < rare_allele_frequency) & (gnomAD_CpGs['C_dinucleotide'] == 'CG')])\n",
    "    rare_non_CpGs = len(gnomAD_CpGs[(gnomAD_CpGs['allele_frequency'] < rare_allele_frequency) & (gnomAD_CpGs['C_dinucleotide'] != 'CG')])\n",
    "    common_CpGs = len(gnomAD_CpGs[(gnomAD_CpGs['allele_frequency'] >= common_allele_frequency) & (gnomAD_CpGs['C_dinucleotide'] == 'CG')])\n",
    "    common_non_CpGs = len(gnomAD_CpGs[(gnomAD_CpGs['allele_frequency'] >= common_allele_frequency) & (gnomAD_CpGs['C_dinucleotide'] != 'CG')])\n",
    "    \n",
    "    oddsratio, pvalue = fisher_exact([[rare_CpGs, rare_non_CpGs], [common_CpGs, common_non_CpGs]])\n",
    "    return rare_CpGs, rare_non_CpGs, common_CpGs, common_non_CpGs, oddsratio, pvalue"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 135,
   "id": "f2e3c76c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(62328, 402159, 65449, 794925, 1.8823853718697245, 0.0)"
      ]
     },
     "execution_count": 135,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "CpG_enrichment_rare_common(0.0001, 0.01)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3b2563ee",
   "metadata": {},
   "source": [
    "## SAV Allele Origin <a class = 'anchor' id = 'SAValleleorigin'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5b7f2122",
   "metadata": {},
   "source": [
    "Let's consider the allele origin for our data. Let's start with Vernot."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "id": "2b75f7c1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>origin</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Vernot_allele_origin</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ancient</th>\n",
       "      <td>681416</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>archaic-specific</th>\n",
       "      <td>523407</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>introgressed</th>\n",
       "      <td>68046</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>low-confidence ancient</th>\n",
       "      <td>334481</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                        origin\n",
       "Vernot_allele_origin          \n",
       "ancient                 681416\n",
       "archaic-specific        523407\n",
       "introgressed             68046\n",
       "low-confidence ancient  334481"
      ]
     },
     "execution_count": 136,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data.groupby(['Vernot_allele_origin']).size().to_frame('origin')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 137,
   "id": "a1913e1b",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>origin</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Vernot_allele_origin</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ancient</th>\n",
       "      <td>2252</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>archaic-specific</th>\n",
       "      <td>2186</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>introgressed</th>\n",
       "      <td>244</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>low-confidence ancient</th>\n",
       "      <td>1268</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                        origin\n",
       "Vernot_allele_origin          \n",
       "ancient                   2252\n",
       "archaic-specific          2186\n",
       "introgressed               244\n",
       "low-confidence ancient    1268"
      ]
     },
     "execution_count": 137,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_2.groupby(['Vernot_allele_origin']).size().to_frame('origin')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 138,
   "id": "17785dc6",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>origin</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Vernot_allele_origin</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ancient</th>\n",
       "      <td>383</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>archaic-specific</th>\n",
       "      <td>404</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>introgressed</th>\n",
       "      <td>59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>low-confidence ancient</th>\n",
       "      <td>203</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                        origin\n",
       "Vernot_allele_origin          \n",
       "ancient                    383\n",
       "archaic-specific           404\n",
       "introgressed                59\n",
       "low-confidence ancient     203"
      ]
     },
     "execution_count": 138,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_5.groupby(['Vernot_allele_origin']).size().to_frame('origin')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "aa7cac12",
   "metadata": {},
   "source": [
    "Now for Browning."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 139,
   "id": "1f7303db",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>origin</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Browning_allele_origin</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ancient</th>\n",
       "      <td>670611</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>archaic-specific</th>\n",
       "      <td>523407</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>introgressed</th>\n",
       "      <td>98189</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>low-confidence ancient</th>\n",
       "      <td>315143</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                        origin\n",
       "Browning_allele_origin        \n",
       "ancient                 670611\n",
       "archaic-specific        523407\n",
       "introgressed             98189\n",
       "low-confidence ancient  315143"
      ]
     },
     "execution_count": 139,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data.groupby(['Browning_allele_origin']).size().to_frame('origin')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 140,
   "id": "e9595fae",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>origin</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Browning_allele_origin</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ancient</th>\n",
       "      <td>2195</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>archaic-specific</th>\n",
       "      <td>2186</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>introgressed</th>\n",
       "      <td>386</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>low-confidence ancient</th>\n",
       "      <td>1183</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                        origin\n",
       "Browning_allele_origin        \n",
       "ancient                   2195\n",
       "archaic-specific          2186\n",
       "introgressed               386\n",
       "low-confidence ancient    1183"
      ]
     },
     "execution_count": 140,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_2.groupby(['Browning_allele_origin']).size().to_frame('origin')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 141,
   "id": "5ee59abc",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>origin</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Browning_allele_origin</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ancient</th>\n",
       "      <td>376</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>archaic-specific</th>\n",
       "      <td>404</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>introgressed</th>\n",
       "      <td>71</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>low-confidence ancient</th>\n",
       "      <td>198</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                        origin\n",
       "Browning_allele_origin        \n",
       "ancient                    376\n",
       "archaic-specific           404\n",
       "introgressed                71\n",
       "low-confidence ancient     198"
      ]
     },
     "execution_count": 141,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_5.groupby(['Browning_allele_origin']).size().to_frame('origin')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c4cca01a",
   "metadata": {
    "scrolled": false
   },
   "source": [
    "Let's take a quick peak at the ancient category and add some additional parameters to test the robustness of that category. Presumably, these variants should be shared in the archaics and occur at high frequency in Africans."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 142,
   "id": "b1c6e0f9",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref_allele</th>\n",
       "      <th>alt_allele</th>\n",
       "      <th>ancestral_allele</th>\n",
       "      <th>anc_dev</th>\n",
       "      <th>variant_type</th>\n",
       "      <th>altai_gt</th>\n",
       "      <th>chagyrskaya_gt</th>\n",
       "      <th>denisovan_gt</th>\n",
       "      <th>vindija_gt</th>\n",
       "      <th>altai_gt_boolean</th>\n",
       "      <th>chagyrskaya_gt_boolean</th>\n",
       "      <th>denisovan_gt_boolean</th>\n",
       "      <th>vindija_gt_boolean</th>\n",
       "      <th>distribution</th>\n",
       "      <th>present_in_1KG</th>\n",
       "      <th>1KG_allele_count</th>\n",
       "      <th>1KG_allele_number</th>\n",
       "      <th>1KG_allele_frequency</th>\n",
       "      <th>1KG_EAS_AF</th>\n",
       "      <th>1KG_EUR_AF</th>\n",
       "      <th>1KG_AFR_AF</th>\n",
       "      <th>1KG_AMR_AF</th>\n",
       "      <th>1KG_SAS_AF</th>\n",
       "      <th>1KG_non_ASW_AFR_AF</th>\n",
       "      <th>present_in_gnomAD</th>\n",
       "      <th>gnomAD_allele_count</th>\n",
       "      <th>gnomAD_allele_number</th>\n",
       "      <th>gnomAD_allele_frequency</th>\n",
       "      <th>Vernot_introgressed</th>\n",
       "      <th>Vernot_ancestral_allele</th>\n",
       "      <th>Vernot_derived_allele</th>\n",
       "      <th>Vernot_ancestral_derived_code</th>\n",
       "      <th>Vernot_AFA_AF</th>\n",
       "      <th>Vernot_AFR_AF</th>\n",
       "      <th>Vernot_AMR_AF</th>\n",
       "      <th>Vernot_EAS_AF</th>\n",
       "      <th>Vernot_EUR_AF</th>\n",
       "      <th>Vernot_PNG_AF</th>\n",
       "      <th>Vernot_SAS_AF</th>\n",
       "      <th>Vernot_Denisovan_base</th>\n",
       "      <th>Vernot_Neanderthal_base_1</th>\n",
       "      <th>Vernot_Neanderthal_base_2</th>\n",
       "      <th>Vernot_haplotype_tag</th>\n",
       "      <th>Vernot_introgressed_AF</th>\n",
       "      <th>Browning_introgressed</th>\n",
       "      <th>Browning_ref_alt</th>\n",
       "      <th>Browning_introgressed_AF</th>\n",
       "      <th>annotation</th>\n",
       "      <th>mis_oe</th>\n",
       "      <th>mis_z</th>\n",
       "      <th>lof_oe</th>\n",
       "      <th>lof_z</th>\n",
       "      <th>phyloP</th>\n",
       "      <th>ag_delta</th>\n",
       "      <th>al_delta</th>\n",
       "      <th>dg_delta</th>\n",
       "      <th>dl_delta</th>\n",
       "      <th>delta_max</th>\n",
       "      <th>ag_pos</th>\n",
       "      <th>al_pos</th>\n",
       "      <th>dg_pos</th>\n",
       "      <th>dl_pos</th>\n",
       "      <th>Adipose_Subcutaneous</th>\n",
       "      <th>Adipose_Visceral_Omentum</th>\n",
       "      <th>Adrenal_Gland</th>\n",
       "      <th>Artery_Aorta</th>\n",
       "      <th>Artery_Coronary</th>\n",
       "      <th>Artery_Tibial</th>\n",
       "      <th>Brain_Amygdala</th>\n",
       "      <th>Brain_Anterior_cingulate_cortex_BA24</th>\n",
       "      <th>Brain_Caudate_basal_ganglia</th>\n",
       "      <th>Brain_Cerebellar_Hemisphere</th>\n",
       "      <th>Brain_Cerebellum</th>\n",
       "      <th>Brain_Cortex</th>\n",
       "      <th>Brain_Frontal_Cortex_BA9</th>\n",
       "      <th>Brain_Hippocampus</th>\n",
       "      <th>Brain_Hypothalamus</th>\n",
       "      <th>Brain_Nucleus_accumbens_basal_ganglia</th>\n",
       "      <th>Brain_Putamen_basal_ganglia</th>\n",
       "      <th>Brain_Spinal_cord_cervical_c-1</th>\n",
       "      <th>Brain_Substantia_nigra</th>\n",
       "      <th>Breast_Mammary_Tissue</th>\n",
       "      <th>Cells_Cultured_fibroblasts</th>\n",
       "      <th>Cells_EBV-transformed_lymphocytes</th>\n",
       "      <th>Colon_Sigmoid</th>\n",
       "      <th>Colon_Transverse</th>\n",
       "      <th>Esophagus_Gastroesophageal_Junction</th>\n",
       "      <th>Esophagus_Mucosa</th>\n",
       "      <th>Esophagus_Muscularis</th>\n",
       "      <th>Heart_Atrial_Appendage</th>\n",
       "      <th>Heart_Left_Ventricle</th>\n",
       "      <th>Kidney_Cortex</th>\n",
       "      <th>Liver</th>\n",
       "      <th>Lung</th>\n",
       "      <th>Minor_Salivary_Gland</th>\n",
       "      <th>Muscle_Skeletal</th>\n",
       "      <th>Nerve_Tibial</th>\n",
       "      <th>Ovary</th>\n",
       "      <th>Pancreas</th>\n",
       "      <th>Pituitary</th>\n",
       "      <th>Prostate</th>\n",
       "      <th>Skin_Not_Sun_Exposed_Suprapubic</th>\n",
       "      <th>Skin_Sun_Exposed_Lower_leg</th>\n",
       "      <th>Small_Intestine_Terminal_Ileum</th>\n",
       "      <th>Spleen</th>\n",
       "      <th>Stomach</th>\n",
       "      <th>Testis</th>\n",
       "      <th>Thyroid</th>\n",
       "      <th>Uterus</th>\n",
       "      <th>Vagina</th>\n",
       "      <th>Whole_Blood</th>\n",
       "      <th>N_GTEx_tissues</th>\n",
       "      <th>sQTL</th>\n",
       "      <th>Vernot_allele_origin</th>\n",
       "      <th>Browning_allele_origin</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>chr1</td>\n",
       "      <td>864726</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>yes</td>\n",
       "      <td>4859.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.95</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.83</td>\n",
       "      <td>0.99</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.812865</td>\n",
       "      <td>yes</td>\n",
       "      <td>137320.0</td>\n",
       "      <td>143300.0</td>\n",
       "      <td>0.958269</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>1.50820</td>\n",
       "      <td>-3.43610</td>\n",
       "      <td>0.89656</td>\n",
       "      <td>0.47484</td>\n",
       "      <td>0.398</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.22</td>\n",
       "      <td>0.22</td>\n",
       "      <td>17</td>\n",
       "      <td>-2</td>\n",
       "      <td>43</td>\n",
       "      <td>-2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>ancient</td>\n",
       "      <td>ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>chr1</td>\n",
       "      <td>875708</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>Altai</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>1.0</td>\n",
       "      <td>139672.0</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>1.50820</td>\n",
       "      <td>-3.43610</td>\n",
       "      <td>0.89656</td>\n",
       "      <td>0.47484</td>\n",
       "      <td>-0.847</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.29</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.29</td>\n",
       "      <td>-43</td>\n",
       "      <td>33</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>130</th>\n",
       "      <td>chr1</td>\n",
       "      <td>909768</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>yes</td>\n",
       "      <td>5094.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.998051</td>\n",
       "      <td>yes</td>\n",
       "      <td>143298.0</td>\n",
       "      <td>143312.0</td>\n",
       "      <td>0.999902</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>1.25930</td>\n",
       "      <td>-1.79880</td>\n",
       "      <td>0.80467</td>\n",
       "      <td>1.00890</td>\n",
       "      <td>-1.259</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.29</td>\n",
       "      <td>0.13</td>\n",
       "      <td>0.03</td>\n",
       "      <td>0.29</td>\n",
       "      <td>7</td>\n",
       "      <td>26</td>\n",
       "      <td>-24</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>ancient</td>\n",
       "      <td>ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>155</th>\n",
       "      <td>chr1</td>\n",
       "      <td>956622</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>yes</td>\n",
       "      <td>4.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.002924</td>\n",
       "      <td>yes</td>\n",
       "      <td>62.0</td>\n",
       "      <td>143332.0</td>\n",
       "      <td>0.000433</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>0.98257</td>\n",
       "      <td>0.22552</td>\n",
       "      <td>0.31914</td>\n",
       "      <td>6.01430</td>\n",
       "      <td>-0.542</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.24</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.24</td>\n",
       "      <td>-2</td>\n",
       "      <td>50</td>\n",
       "      <td>-2</td>\n",
       "      <td>40</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>210</th>\n",
       "      <td>chr1</td>\n",
       "      <td>975133</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>yes</td>\n",
       "      <td>4557.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.89</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.92</td>\n",
       "      <td>0.72</td>\n",
       "      <td>0.92</td>\n",
       "      <td>0.97</td>\n",
       "      <td>0.728070</td>\n",
       "      <td>yes</td>\n",
       "      <td>125462.0</td>\n",
       "      <td>143014.0</td>\n",
       "      <td>0.877271</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>0.98257</td>\n",
       "      <td>0.22552</td>\n",
       "      <td>0.31914</td>\n",
       "      <td>6.01430</td>\n",
       "      <td>0.364</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.22</td>\n",
       "      <td>0.08</td>\n",
       "      <td>0.22</td>\n",
       "      <td>-3</td>\n",
       "      <td>14</td>\n",
       "      <td>-3</td>\n",
       "      <td>14</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>ancient</td>\n",
       "      <td>ancient</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    chrom     pos ref_allele alt_allele ancestral_allele  anc_dev  \\\n",
       "24   chr1  864726          T          A                T  derived   \n",
       "65   chr1  875708          C          G                T  derived   \n",
       "130  chr1  909768          A          G                -        .   \n",
       "155  chr1  956622          C          T                C  derived   \n",
       "210  chr1  975133          T          A                T  derived   \n",
       "\n",
       "    variant_type altai_gt chagyrskaya_gt denisovan_gt vindija_gt  \\\n",
       "24           snv      1/1            1/1          1/1        1/1   \n",
       "65           snv      1/1            ./.          ./.        ./.   \n",
       "130          snv      1/1            1/1          1/1        1/1   \n",
       "155          snv      1/1            1/1          0/0        1/1   \n",
       "210          snv      1/1            1/1          1/1        1/1   \n",
       "\n",
       "     altai_gt_boolean  chagyrskaya_gt_boolean  denisovan_gt_boolean  \\\n",
       "24               True                    True                  True   \n",
       "65               True                   False                 False   \n",
       "130              True                    True                  True   \n",
       "155              True                    True                 False   \n",
       "210              True                    True                  True   \n",
       "\n",
       "     vindija_gt_boolean distribution present_in_1KG  1KG_allele_count  \\\n",
       "24                 True       Shared            yes            4859.0   \n",
       "65                False        Altai             no               NaN   \n",
       "130                True       Shared            yes            5094.0   \n",
       "155                True  Neanderthal            yes               4.0   \n",
       "210                True       Shared            yes            4557.0   \n",
       "\n",
       "     1KG_allele_number  1KG_allele_frequency  1KG_EAS_AF  1KG_EUR_AF  \\\n",
       "24              5096.0                  0.95         1.0        1.00   \n",
       "65                 NaN                   NaN         NaN         NaN   \n",
       "130             5096.0                  1.00         1.0        1.00   \n",
       "155             5096.0                  0.00         0.0        0.00   \n",
       "210             5096.0                  0.89         1.0        0.92   \n",
       "\n",
       "     1KG_AFR_AF  1KG_AMR_AF  1KG_SAS_AF  1KG_non_ASW_AFR_AF present_in_gnomAD  \\\n",
       "24         0.83        0.99        1.00            0.812865               yes   \n",
       "65          NaN         NaN         NaN                 NaN               yes   \n",
       "130        1.00        1.00        1.00            0.998051               yes   \n",
       "155        0.00        0.00        0.00            0.002924               yes   \n",
       "210        0.72        0.92        0.97            0.728070               yes   \n",
       "\n",
       "     gnomAD_allele_count  gnomAD_allele_number  gnomAD_allele_frequency  \\\n",
       "24              137320.0              143300.0                 0.958269   \n",
       "65                   1.0              139672.0                 0.000007   \n",
       "130             143298.0              143312.0                 0.999902   \n",
       "155                 62.0              143332.0                 0.000433   \n",
       "210             125462.0              143014.0                 0.877271   \n",
       "\n",
       "    Vernot_introgressed Vernot_ancestral_allele Vernot_derived_allele  \\\n",
       "24                   no                     NaN                   NaN   \n",
       "65                   no                     NaN                   NaN   \n",
       "130                  no                     NaN                   NaN   \n",
       "155                  no                     NaN                   NaN   \n",
       "210                  no                     NaN                   NaN   \n",
       "\n",
       "     Vernot_ancestral_derived_code  Vernot_AFA_AF  Vernot_AFR_AF  \\\n",
       "24                             NaN            NaN            NaN   \n",
       "65                             NaN            NaN            NaN   \n",
       "130                            NaN            NaN            NaN   \n",
       "155                            NaN            NaN            NaN   \n",
       "210                            NaN            NaN            NaN   \n",
       "\n",
       "     Vernot_AMR_AF  Vernot_EAS_AF  Vernot_EUR_AF  Vernot_PNG_AF  \\\n",
       "24             NaN            NaN            NaN            NaN   \n",
       "65             NaN            NaN            NaN            NaN   \n",
       "130            NaN            NaN            NaN            NaN   \n",
       "155            NaN            NaN            NaN            NaN   \n",
       "210            NaN            NaN            NaN            NaN   \n",
       "\n",
       "     Vernot_SAS_AF Vernot_Denisovan_base Vernot_Neanderthal_base_1  \\\n",
       "24             NaN                   NaN                       NaN   \n",
       "65             NaN                   NaN                       NaN   \n",
       "130            NaN                   NaN                       NaN   \n",
       "155            NaN                   NaN                       NaN   \n",
       "210            NaN                   NaN                       NaN   \n",
       "\n",
       "    Vernot_Neanderthal_base_2 Vernot_haplotype_tag  Vernot_introgressed_AF  \\\n",
       "24                        NaN                  NaN                     NaN   \n",
       "65                        NaN                  NaN                     NaN   \n",
       "130                       NaN                  NaN                     NaN   \n",
       "155                       NaN                  NaN                     NaN   \n",
       "210                       NaN                  NaN                     NaN   \n",
       "\n",
       "    Browning_introgressed  Browning_ref_alt  Browning_introgressed_AF  \\\n",
       "24                     no               NaN                       NaN   \n",
       "65                     no               NaN                       NaN   \n",
       "130                    no               NaN                       NaN   \n",
       "155                    no               NaN                       NaN   \n",
       "210                    no               NaN                       NaN   \n",
       "\n",
       "    annotation   mis_oe    mis_z   lof_oe    lof_z  phyloP  ag_delta  \\\n",
       "24      SAMD11  1.50820 -3.43610  0.89656  0.47484   0.398       0.0   \n",
       "65      SAMD11  1.50820 -3.43610  0.89656  0.47484  -0.847       0.0   \n",
       "130    PLEKHN1  1.25930 -1.79880  0.80467  1.00890  -1.259       0.0   \n",
       "155       AGRN  0.98257  0.22552  0.31914  6.01430  -0.542       0.0   \n",
       "210       AGRN  0.98257  0.22552  0.31914  6.01430   0.364       0.0   \n",
       "\n",
       "     al_delta  dg_delta  dl_delta  delta_max  ag_pos  al_pos  dg_pos  dl_pos  \\\n",
       "24       0.00      0.02      0.22       0.22      17      -2      43      -2   \n",
       "65       0.00      0.29      0.01       0.29     -43      33       0      12   \n",
       "130      0.29      0.13      0.03       0.29       7      26     -24       1   \n",
       "155      0.00      0.24      0.02       0.24      -2      50      -2      40   \n",
       "210      0.00      0.22      0.08       0.22      -3      14      -3      14   \n",
       "\n",
       "     Adipose_Subcutaneous  Adipose_Visceral_Omentum  Adrenal_Gland  \\\n",
       "24                    NaN                       NaN            NaN   \n",
       "65                    NaN                       NaN            NaN   \n",
       "130                   NaN                       NaN            NaN   \n",
       "155                   NaN                       NaN            NaN   \n",
       "210                   0.0                       1.0            0.0   \n",
       "\n",
       "     Artery_Aorta  Artery_Coronary  Artery_Tibial  Brain_Amygdala  \\\n",
       "24            NaN              NaN            NaN             NaN   \n",
       "65            NaN              NaN            NaN             NaN   \n",
       "130           NaN              NaN            NaN             NaN   \n",
       "155           NaN              NaN            NaN             NaN   \n",
       "210           1.0              0.0            1.0             0.0   \n",
       "\n",
       "     Brain_Anterior_cingulate_cortex_BA24  Brain_Caudate_basal_ganglia  \\\n",
       "24                                    NaN                          NaN   \n",
       "65                                    NaN                          NaN   \n",
       "130                                   NaN                          NaN   \n",
       "155                                   NaN                          NaN   \n",
       "210                                   0.0                          0.0   \n",
       "\n",
       "     Brain_Cerebellar_Hemisphere  Brain_Cerebellum  Brain_Cortex  \\\n",
       "24                           NaN               NaN           NaN   \n",
       "65                           NaN               NaN           NaN   \n",
       "130                          NaN               NaN           NaN   \n",
       "155                          NaN               NaN           NaN   \n",
       "210                          0.0               0.0           0.0   \n",
       "\n",
       "     Brain_Frontal_Cortex_BA9  Brain_Hippocampus  Brain_Hypothalamus  \\\n",
       "24                        NaN                NaN                 NaN   \n",
       "65                        NaN                NaN                 NaN   \n",
       "130                       NaN                NaN                 NaN   \n",
       "155                       NaN                NaN                 NaN   \n",
       "210                       0.0                0.0                 0.0   \n",
       "\n",
       "     Brain_Nucleus_accumbens_basal_ganglia  Brain_Putamen_basal_ganglia  \\\n",
       "24                                     NaN                          NaN   \n",
       "65                                     NaN                          NaN   \n",
       "130                                    NaN                          NaN   \n",
       "155                                    NaN                          NaN   \n",
       "210                                    0.0                          0.0   \n",
       "\n",
       "     Brain_Spinal_cord_cervical_c-1  Brain_Substantia_nigra  \\\n",
       "24                              NaN                     NaN   \n",
       "65                              NaN                     NaN   \n",
       "130                             NaN                     NaN   \n",
       "155                             NaN                     NaN   \n",
       "210                             0.0                     0.0   \n",
       "\n",
       "     Breast_Mammary_Tissue  Cells_Cultured_fibroblasts  \\\n",
       "24                     NaN                         NaN   \n",
       "65                     NaN                         NaN   \n",
       "130                    NaN                         NaN   \n",
       "155                    NaN                         NaN   \n",
       "210                    1.0                         0.0   \n",
       "\n",
       "     Cells_EBV-transformed_lymphocytes  Colon_Sigmoid  Colon_Transverse  \\\n",
       "24                                 NaN            NaN               NaN   \n",
       "65                                 NaN            NaN               NaN   \n",
       "130                                NaN            NaN               NaN   \n",
       "155                                NaN            NaN               NaN   \n",
       "210                                0.0            1.0               0.0   \n",
       "\n",
       "     Esophagus_Gastroesophageal_Junction  Esophagus_Mucosa  \\\n",
       "24                                   NaN               NaN   \n",
       "65                                   NaN               NaN   \n",
       "130                                  NaN               NaN   \n",
       "155                                  NaN               NaN   \n",
       "210                                  1.0               1.0   \n",
       "\n",
       "     Esophagus_Muscularis  Heart_Atrial_Appendage  Heart_Left_Ventricle  \\\n",
       "24                    NaN                     NaN                   NaN   \n",
       "65                    NaN                     NaN                   NaN   \n",
       "130                   NaN                     NaN                   NaN   \n",
       "155                   NaN                     NaN                   NaN   \n",
       "210                   1.0                     0.0                   0.0   \n",
       "\n",
       "     Kidney_Cortex  Liver  Lung  Minor_Salivary_Gland  Muscle_Skeletal  \\\n",
       "24             NaN    NaN   NaN                   NaN              NaN   \n",
       "65             NaN    NaN   NaN                   NaN              NaN   \n",
       "130            NaN    NaN   NaN                   NaN              NaN   \n",
       "155            NaN    NaN   NaN                   NaN              NaN   \n",
       "210            0.0    0.0   1.0                   0.0              0.0   \n",
       "\n",
       "     Nerve_Tibial  Ovary  Pancreas  Pituitary  Prostate  \\\n",
       "24            NaN    NaN       NaN        NaN       NaN   \n",
       "65            NaN    NaN       NaN        NaN       NaN   \n",
       "130           NaN    NaN       NaN        NaN       NaN   \n",
       "155           NaN    NaN       NaN        NaN       NaN   \n",
       "210           1.0    0.0       0.0        0.0       0.0   \n",
       "\n",
       "     Skin_Not_Sun_Exposed_Suprapubic  Skin_Sun_Exposed_Lower_leg  \\\n",
       "24                               NaN                         NaN   \n",
       "65                               NaN                         NaN   \n",
       "130                              NaN                         NaN   \n",
       "155                              NaN                         NaN   \n",
       "210                              1.0                         1.0   \n",
       "\n",
       "     Small_Intestine_Terminal_Ileum  Spleen  Stomach  Testis  Thyroid  Uterus  \\\n",
       "24                              NaN     NaN      NaN     NaN      NaN     NaN   \n",
       "65                              NaN     NaN      NaN     NaN      NaN     NaN   \n",
       "130                             NaN     NaN      NaN     NaN      NaN     NaN   \n",
       "155                             NaN     NaN      NaN     NaN      NaN     NaN   \n",
       "210                             0.0     0.0      1.0     1.0      1.0     0.0   \n",
       "\n",
       "     Vagina  Whole_Blood  N_GTEx_tissues sQTL    Vernot_allele_origin  \\\n",
       "24      NaN          NaN             NaN   no                 ancient   \n",
       "65      NaN          NaN             NaN   no        archaic-specific   \n",
       "130     NaN          NaN             NaN   no                 ancient   \n",
       "155     NaN          NaN             NaN   no  low-confidence ancient   \n",
       "210     0.0          0.0            15.0  yes                 ancient   \n",
       "\n",
       "     Browning_allele_origin  \n",
       "24                  ancient  \n",
       "65         archaic-specific  \n",
       "130                 ancient  \n",
       "155  low-confidence ancient  \n",
       "210                 ancient  "
      ]
     },
     "execution_count": 142,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_2.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 143,
   "id": "47aa5e4b",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1380"
      ]
     },
     "execution_count": 143,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(data_2[(data_2['Vernot_allele_origin'] == 'ancient') & (data_2['distribution'] == 'Shared') & (data_2['1KG_non_ASW_AFR_AF'] >= 0.05)])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 144,
   "id": "af8807c7",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.6127886323268206"
      ]
     },
     "execution_count": 144,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "1380/2252"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 145,
   "id": "b08a0d53",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1380"
      ]
     },
     "execution_count": 145,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(data_2[(data_2['Browning_allele_origin'] == 'ancient') & (data_2['distribution'] == 'Shared') & (data_2['1KG_non_ASW_AFR_AF'] >= 0.05)])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 146,
   "id": "f21b4247",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.6287015945330297"
      ]
     },
     "execution_count": 146,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "1380/2195"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f4723b30",
   "metadata": {},
   "source": [
    "What about dropping the shared archaic part?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 147,
   "id": "61e3bb32",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2098"
      ]
     },
     "execution_count": 147,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(data_2[(data_2['Vernot_allele_origin'] == 'ancient') & (data_2['1KG_non_ASW_AFR_AF'] >= 0.05)])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 148,
   "id": "16720255",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.9316163410301954"
      ]
     },
     "execution_count": 148,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "2098/2252"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 149,
   "id": "c8545c46",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2097"
      ]
     },
     "execution_count": 149,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(data_2[(data_2['Browning_allele_origin'] == 'ancient') & (data_2['1KG_non_ASW_AFR_AF'] >= 0.05)])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 150,
   "id": "ff00f700",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.9553530751708428"
      ]
     },
     "execution_count": 150,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "2097/2195"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "954fbac2",
   "metadata": {},
   "source": [
    "How many introgressed variants may have evolved prior to the archaic-modern split?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 151,
   "id": "2328ff43",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "68"
      ]
     },
     "execution_count": 151,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(data_2[(data_2['Vernot_allele_origin'] == 'introgressed') & (data_2['distribution'] == 'Shared')])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 152,
   "id": "24ff7db9",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.2786885245901639"
      ]
     },
     "execution_count": 152,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "68/244"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 153,
   "id": "f75e4876",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "122"
      ]
     },
     "execution_count": 153,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(data_2[(data_2['Browning_allele_origin'] == 'introgressed') & (data_2['distribution'] == 'Shared')])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 154,
   "id": "60dc09b7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.3160621761658031"
      ]
     },
     "execution_count": 154,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "122/386"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 155,
   "id": "d5329092",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1"
      ]
     },
     "execution_count": 155,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(data_2[(data_2['Vernot_allele_origin'] == 'introgressed') & (data_2['1KG_non_ASW_AFR_AF'] >= 0.05)])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 156,
   "id": "0d97264d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2"
      ]
     },
     "execution_count": 156,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(data_2[(data_2['Browning_allele_origin'] == 'introgressed') & (data_2['1KG_non_ASW_AFR_AF'] >= 0.05)])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 157,
   "id": "2676d54c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2"
      ]
     },
     "execution_count": 157,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(data_2[(data_2['Vernot_allele_origin'] == 'introgressed') & (data_2['1KG_non_ASW_AFR_AF'] >= 0.01)])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 158,
   "id": "0fb544c6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "16"
      ]
     },
     "execution_count": 158,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(data_2[(data_2['Browning_allele_origin'] == 'introgressed') & (data_2['1KG_non_ASW_AFR_AF'] >= 0.01)])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "65b732d8",
   "metadata": {},
   "source": [
    "## Introgression Set Overlap <a class = 'anchor' id = 'introgressionsetoverlap'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0b91ba6b",
   "metadata": {},
   "source": [
    "How well do Vernot et al. 2016 and Browning et al. 2018 overlap for this set of variants? Let's look at all variants and then SAVs at delta >= 0.2."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 159,
   "id": "155c5ac9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "116144"
      ]
     },
     "execution_count": 159,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "introgressed_vars = data[(data['Vernot_allele_origin'] == 'introgressed') | (data['Browning_allele_origin'] == 'introgressed')]\n",
    "len(introgressed_vars)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 160,
   "id": "c60d36a2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "68046"
      ]
     },
     "execution_count": 160,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(introgressed_vars[introgressed_vars['Vernot_allele_origin'] == 'introgressed'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 161,
   "id": "240ae841",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "98189"
      ]
     },
     "execution_count": 161,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(introgressed_vars[introgressed_vars['Browning_allele_origin'] == 'introgressed'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 162,
   "id": "41bc454e",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "50091"
      ]
     },
     "execution_count": 162,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(introgressed_vars[(introgressed_vars['Vernot_allele_origin'] == 'introgressed') & (introgressed_vars['Browning_allele_origin'] == 'introgressed')])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 163,
   "id": "58d41010",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "460"
      ]
     },
     "execution_count": 163,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "introgressed_SAVs = data_2[(data_2['Vernot_allele_origin'] == 'introgressed') | (data_2['Browning_allele_origin'] == 'introgressed')]\n",
    "len(introgressed_SAVs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 164,
   "id": "dc78af02",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "244"
      ]
     },
     "execution_count": 164,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(introgressed_SAVs[introgressed_SAVs['Vernot_allele_origin'] == 'introgressed'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 165,
   "id": "42312a22",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "386"
      ]
     },
     "execution_count": 165,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(introgressed_SAVs[introgressed_SAVs['Browning_allele_origin'] == 'introgressed'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 166,
   "id": "39d2471d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "170"
      ]
     },
     "execution_count": 166,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(introgressed_SAVs[(introgressed_SAVs['Vernot_allele_origin'] == 'introgressed') & (introgressed_SAVs['Browning_allele_origin'] == 'introgressed')])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9a770a33",
   "metadata": {},
   "source": [
    "## SAV Distribution <a class = 'anchor' id = 'SAVdistribution'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2ccf0169",
   "metadata": {},
   "source": [
    "Get distribution of archaic-specific variants for Figure 1. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 167,
   "id": "c6ab6252",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>distribution</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>distribution</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Altai</th>\n",
       "      <td>300</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Chagyrskaya</th>\n",
       "      <td>169</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Denisovan</th>\n",
       "      <td>916</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Late Neanderthal</th>\n",
       "      <td>94</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Neanderthal</th>\n",
       "      <td>236</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Other</th>\n",
       "      <td>128</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Shared</th>\n",
       "      <td>112</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Vindija</th>\n",
       "      <td>231</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  distribution\n",
       "distribution                  \n",
       "Altai                      300\n",
       "Chagyrskaya                169\n",
       "Denisovan                  916\n",
       "Late Neanderthal            94\n",
       "Neanderthal                236\n",
       "Other                      128\n",
       "Shared                     112\n",
       "Vindija                    231"
      ]
     },
     "execution_count": 167,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "archaic_specific = data_2[(data_2['Vernot_allele_origin'] == 'archaic-specific')]\n",
    "archaic_specific.groupby(['distribution']).size().to_frame('distribution')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 168,
   "id": "dd191be6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>distribution</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>distribution</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Altai</th>\n",
       "      <td>300</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Chagyrskaya</th>\n",
       "      <td>169</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Denisovan</th>\n",
       "      <td>916</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Late Neanderthal</th>\n",
       "      <td>94</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Neanderthal</th>\n",
       "      <td>236</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Other</th>\n",
       "      <td>128</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Shared</th>\n",
       "      <td>112</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Vindija</th>\n",
       "      <td>231</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  distribution\n",
       "distribution                  \n",
       "Altai                      300\n",
       "Chagyrskaya                169\n",
       "Denisovan                  916\n",
       "Late Neanderthal            94\n",
       "Neanderthal                236\n",
       "Other                      128\n",
       "Shared                     112\n",
       "Vindija                    231"
      ]
     },
     "execution_count": 168,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "archaic_specific = data_2[(data_2['Browning_allele_origin'] == 'archaic-specific')]\n",
    "archaic_specific.groupby(['distribution']).size().to_frame('distribution')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2d850ecd",
   "metadata": {},
   "source": [
    "Let's use parsimony to categorize the 'Other' distribution category. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 169,
   "id": "02bee11c",
   "metadata": {},
   "outputs": [],
   "source": [
    "archaic_specific_other = archaic_specific[(archaic_specific['distribution'] == 'Other')]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1ce15951",
   "metadata": {},
   "source": [
    "DAC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 170,
   "id": "b312cce8",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "5"
      ]
     },
     "execution_count": 170,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(archaic_specific_other[(archaic_specific_other['altai_gt_boolean'] == True) & (archaic_specific_other['chagyrskaya_gt_boolean'] == True) & (archaic_specific_other['denisovan_gt_boolean'] == True) & (archaic_specific_other['vindija_gt_boolean'] == False)])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6da1c1fe",
   "metadata": {},
   "source": [
    "DAV"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 171,
   "id": "b0e8f079",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "7"
      ]
     },
     "execution_count": 171,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(archaic_specific_other[(archaic_specific_other['altai_gt_boolean'] == True) & (archaic_specific_other['chagyrskaya_gt_boolean'] == False) & (archaic_specific_other['denisovan_gt_boolean'] == True) & (archaic_specific_other['vindija_gt_boolean'] == True)])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "65ed2f80",
   "metadata": {},
   "source": [
    "DCV"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 172,
   "id": "79159b53",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2"
      ]
     },
     "execution_count": 172,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(archaic_specific_other[(archaic_specific_other['altai_gt_boolean'] == False) & (archaic_specific_other['chagyrskaya_gt_boolean'] == True) & (archaic_specific_other['denisovan_gt_boolean'] == True) & (archaic_specific_other['vindija_gt_boolean'] == True)])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f3e3512f",
   "metadata": {},
   "source": [
    "DA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 173,
   "id": "d47a61ba",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "20"
      ]
     },
     "execution_count": 173,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(archaic_specific_other[(archaic_specific_other['altai_gt_boolean'] == True) & (archaic_specific_other['chagyrskaya_gt_boolean'] == False) & (archaic_specific_other['denisovan_gt_boolean'] == True) & (archaic_specific_other['vindija_gt_boolean'] == False)])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "12252853",
   "metadata": {},
   "source": [
    "DC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 174,
   "id": "2c9e8d37",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2"
      ]
     },
     "execution_count": 174,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(archaic_specific_other[(archaic_specific_other['altai_gt_boolean'] == False) & (archaic_specific_other['chagyrskaya_gt_boolean'] == True) & (archaic_specific_other['denisovan_gt_boolean'] == True) & (archaic_specific_other['vindija_gt_boolean'] == False)])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ec3b5bc6",
   "metadata": {},
   "source": [
    "DV"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 175,
   "id": "dd397a1a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2"
      ]
     },
     "execution_count": 175,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(archaic_specific_other[(archaic_specific_other['altai_gt_boolean'] == False) & (archaic_specific_other['chagyrskaya_gt_boolean'] == False) & (archaic_specific_other['denisovan_gt_boolean'] == True) & (archaic_specific_other['vindija_gt_boolean'] == True)])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d1cd7dc8",
   "metadata": {},
   "source": [
    "AC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 176,
   "id": "f5b956bb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "38"
      ]
     },
     "execution_count": 176,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(archaic_specific_other[(archaic_specific_other['altai_gt_boolean'] == True) & (archaic_specific_other['chagyrskaya_gt_boolean'] == True) & (archaic_specific_other['denisovan_gt_boolean'] == False) & (archaic_specific_other['vindija_gt_boolean'] == False)])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3ef61b7d",
   "metadata": {},
   "source": [
    "AV"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 177,
   "id": "f00303f8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "52"
      ]
     },
     "execution_count": 177,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(archaic_specific_other[(archaic_specific_other['altai_gt_boolean'] == True) & (archaic_specific_other['chagyrskaya_gt_boolean'] == False) & (archaic_specific_other['denisovan_gt_boolean'] == False) & (archaic_specific_other['vindija_gt_boolean'] == True)])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "83565f29",
   "metadata": {},
   "source": [
    "Let's add 38 (5+7+2+20+2+2) to the archaic common ancestor and 90 (38+52) to the Neanderthal common ancestor in Figure 1."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2252c11e",
   "metadata": {},
   "source": [
    "# SAV Genes <a class = 'anchor' id = 'SAVgenes'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5bedbe48",
   "metadata": {},
   "source": [
    "## Top 20 AG, AL, DG, and DL <a class = 'anchor' id = 'top20'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "09450e65",
   "metadata": {},
   "source": [
    "Now let's take a look at the top 20 variants with the highest deltas for each of the four classes."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 178,
   "id": "b01cdd87",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref_allele</th>\n",
       "      <th>alt_allele</th>\n",
       "      <th>ancestral_allele</th>\n",
       "      <th>anc_dev</th>\n",
       "      <th>variant_type</th>\n",
       "      <th>altai_gt</th>\n",
       "      <th>chagyrskaya_gt</th>\n",
       "      <th>denisovan_gt</th>\n",
       "      <th>vindija_gt</th>\n",
       "      <th>altai_gt_boolean</th>\n",
       "      <th>chagyrskaya_gt_boolean</th>\n",
       "      <th>denisovan_gt_boolean</th>\n",
       "      <th>vindija_gt_boolean</th>\n",
       "      <th>distribution</th>\n",
       "      <th>present_in_1KG</th>\n",
       "      <th>1KG_allele_count</th>\n",
       "      <th>1KG_allele_number</th>\n",
       "      <th>1KG_allele_frequency</th>\n",
       "      <th>1KG_EAS_AF</th>\n",
       "      <th>1KG_EUR_AF</th>\n",
       "      <th>1KG_AFR_AF</th>\n",
       "      <th>1KG_AMR_AF</th>\n",
       "      <th>1KG_SAS_AF</th>\n",
       "      <th>1KG_non_ASW_AFR_AF</th>\n",
       "      <th>present_in_gnomAD</th>\n",
       "      <th>gnomAD_allele_count</th>\n",
       "      <th>gnomAD_allele_number</th>\n",
       "      <th>gnomAD_allele_frequency</th>\n",
       "      <th>Vernot_introgressed</th>\n",
       "      <th>Vernot_ancestral_allele</th>\n",
       "      <th>Vernot_derived_allele</th>\n",
       "      <th>Vernot_ancestral_derived_code</th>\n",
       "      <th>Vernot_AFA_AF</th>\n",
       "      <th>Vernot_AFR_AF</th>\n",
       "      <th>Vernot_AMR_AF</th>\n",
       "      <th>Vernot_EAS_AF</th>\n",
       "      <th>Vernot_EUR_AF</th>\n",
       "      <th>Vernot_PNG_AF</th>\n",
       "      <th>Vernot_SAS_AF</th>\n",
       "      <th>Vernot_Denisovan_base</th>\n",
       "      <th>Vernot_Neanderthal_base_1</th>\n",
       "      <th>Vernot_Neanderthal_base_2</th>\n",
       "      <th>Vernot_haplotype_tag</th>\n",
       "      <th>Vernot_introgressed_AF</th>\n",
       "      <th>Browning_introgressed</th>\n",
       "      <th>Browning_ref_alt</th>\n",
       "      <th>Browning_introgressed_AF</th>\n",
       "      <th>annotation</th>\n",
       "      <th>mis_oe</th>\n",
       "      <th>mis_z</th>\n",
       "      <th>lof_oe</th>\n",
       "      <th>lof_z</th>\n",
       "      <th>phyloP</th>\n",
       "      <th>ag_delta</th>\n",
       "      <th>al_delta</th>\n",
       "      <th>dg_delta</th>\n",
       "      <th>dl_delta</th>\n",
       "      <th>delta_max</th>\n",
       "      <th>ag_pos</th>\n",
       "      <th>al_pos</th>\n",
       "      <th>dg_pos</th>\n",
       "      <th>dl_pos</th>\n",
       "      <th>Adipose_Subcutaneous</th>\n",
       "      <th>Adipose_Visceral_Omentum</th>\n",
       "      <th>Adrenal_Gland</th>\n",
       "      <th>Artery_Aorta</th>\n",
       "      <th>Artery_Coronary</th>\n",
       "      <th>Artery_Tibial</th>\n",
       "      <th>Brain_Amygdala</th>\n",
       "      <th>Brain_Anterior_cingulate_cortex_BA24</th>\n",
       "      <th>Brain_Caudate_basal_ganglia</th>\n",
       "      <th>Brain_Cerebellar_Hemisphere</th>\n",
       "      <th>Brain_Cerebellum</th>\n",
       "      <th>Brain_Cortex</th>\n",
       "      <th>Brain_Frontal_Cortex_BA9</th>\n",
       "      <th>Brain_Hippocampus</th>\n",
       "      <th>Brain_Hypothalamus</th>\n",
       "      <th>Brain_Nucleus_accumbens_basal_ganglia</th>\n",
       "      <th>Brain_Putamen_basal_ganglia</th>\n",
       "      <th>Brain_Spinal_cord_cervical_c-1</th>\n",
       "      <th>Brain_Substantia_nigra</th>\n",
       "      <th>Breast_Mammary_Tissue</th>\n",
       "      <th>Cells_Cultured_fibroblasts</th>\n",
       "      <th>Cells_EBV-transformed_lymphocytes</th>\n",
       "      <th>Colon_Sigmoid</th>\n",
       "      <th>Colon_Transverse</th>\n",
       "      <th>Esophagus_Gastroesophageal_Junction</th>\n",
       "      <th>Esophagus_Mucosa</th>\n",
       "      <th>Esophagus_Muscularis</th>\n",
       "      <th>Heart_Atrial_Appendage</th>\n",
       "      <th>Heart_Left_Ventricle</th>\n",
       "      <th>Kidney_Cortex</th>\n",
       "      <th>Liver</th>\n",
       "      <th>Lung</th>\n",
       "      <th>Minor_Salivary_Gland</th>\n",
       "      <th>Muscle_Skeletal</th>\n",
       "      <th>Nerve_Tibial</th>\n",
       "      <th>Ovary</th>\n",
       "      <th>Pancreas</th>\n",
       "      <th>Pituitary</th>\n",
       "      <th>Prostate</th>\n",
       "      <th>Skin_Not_Sun_Exposed_Suprapubic</th>\n",
       "      <th>Skin_Sun_Exposed_Lower_leg</th>\n",
       "      <th>Small_Intestine_Terminal_Ileum</th>\n",
       "      <th>Spleen</th>\n",
       "      <th>Stomach</th>\n",
       "      <th>Testis</th>\n",
       "      <th>Thyroid</th>\n",
       "      <th>Uterus</th>\n",
       "      <th>Vagina</th>\n",
       "      <th>Whole_Blood</th>\n",
       "      <th>N_GTEx_tissues</th>\n",
       "      <th>sQTL</th>\n",
       "      <th>Vernot_allele_origin</th>\n",
       "      <th>Browning_allele_origin</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>78185</th>\n",
       "      <td>chr1</td>\n",
       "      <td>144923837</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>./.</td>\n",
       "      <td>0/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PDE4DIP</td>\n",
       "      <td>1.12420</td>\n",
       "      <td>-1.40300</td>\n",
       "      <td>0.68045</td>\n",
       "      <td>3.12940</td>\n",
       "      <td>-0.264</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.97</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.00</td>\n",
       "      <td>-2</td>\n",
       "      <td>-16</td>\n",
       "      <td>-16</td>\n",
       "      <td>-29</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>302444</th>\n",
       "      <td>chr11</td>\n",
       "      <td>108810969</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Other</td>\n",
       "      <td>yes</td>\n",
       "      <td>2.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>yes</td>\n",
       "      <td>8.0</td>\n",
       "      <td>143086.0</td>\n",
       "      <td>0.000056</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DDX10</td>\n",
       "      <td>0.96014</td>\n",
       "      <td>0.30359</td>\n",
       "      <td>0.70688</td>\n",
       "      <td>1.82750</td>\n",
       "      <td>-0.175</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4234</th>\n",
       "      <td>chr1</td>\n",
       "      <td>5935162</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>ancestral</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NPHP4</td>\n",
       "      <td>1.02290</td>\n",
       "      <td>-0.24249</td>\n",
       "      <td>0.78155</td>\n",
       "      <td>1.66690</td>\n",
       "      <td>-0.557</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.91</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.00</td>\n",
       "      <td>-2</td>\n",
       "      <td>-8</td>\n",
       "      <td>-43</td>\n",
       "      <td>-44</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121279</th>\n",
       "      <td>chr1</td>\n",
       "      <td>218475625</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>ancestral</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>yes</td>\n",
       "      <td>436.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.09</td>\n",
       "      <td>0.06</td>\n",
       "      <td>0.04</td>\n",
       "      <td>0.23</td>\n",
       "      <td>0.04</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.230019</td>\n",
       "      <td>yes</td>\n",
       "      <td>11299.0</td>\n",
       "      <td>143136.0</td>\n",
       "      <td>0.078939</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>RRP15</td>\n",
       "      <td>1.10360</td>\n",
       "      <td>-0.45285</td>\n",
       "      <td>0.62321</td>\n",
       "      <td>1.17020</td>\n",
       "      <td>-0.372</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.70</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.00</td>\n",
       "      <td>2</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>15</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>ancient</td>\n",
       "      <td>ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1455913</th>\n",
       "      <td>chr8</td>\n",
       "      <td>8999187</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>ancestral</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>yes</td>\n",
       "      <td>2597.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.51</td>\n",
       "      <td>0.82</td>\n",
       "      <td>0.31</td>\n",
       "      <td>0.37</td>\n",
       "      <td>0.41</td>\n",
       "      <td>0.66</td>\n",
       "      <td>0.367446</td>\n",
       "      <td>yes</td>\n",
       "      <td>52466.0</td>\n",
       "      <td>142958.0</td>\n",
       "      <td>0.367003</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PPP1R3B</td>\n",
       "      <td>1.24760</td>\n",
       "      <td>-1.10860</td>\n",
       "      <td>0.38350</td>\n",
       "      <td>1.59780</td>\n",
       "      <td>-0.216</td>\n",
       "      <td>0.99</td>\n",
       "      <td>0.95</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.99</td>\n",
       "      <td>-1</td>\n",
       "      <td>-9</td>\n",
       "      <td>-45</td>\n",
       "      <td>-22</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>ancient</td>\n",
       "      <td>ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>737199</th>\n",
       "      <td>chr19</td>\n",
       "      <td>35506729</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>yes</td>\n",
       "      <td>1292.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.25</td>\n",
       "      <td>0.22</td>\n",
       "      <td>0.34</td>\n",
       "      <td>0.27</td>\n",
       "      <td>0.30</td>\n",
       "      <td>0.15</td>\n",
       "      <td>0.276803</td>\n",
       "      <td>yes</td>\n",
       "      <td>43574.0</td>\n",
       "      <td>143070.0</td>\n",
       "      <td>0.304564</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GRAMD1A</td>\n",
       "      <td>0.83225</td>\n",
       "      <td>1.29050</td>\n",
       "      <td>0.31602</td>\n",
       "      <td>4.06520</td>\n",
       "      <td>-1.373</td>\n",
       "      <td>0.99</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.99</td>\n",
       "      <td>2</td>\n",
       "      <td>-1</td>\n",
       "      <td>2</td>\n",
       "      <td>-1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>ancient</td>\n",
       "      <td>ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1531582</th>\n",
       "      <td>chr8</td>\n",
       "      <td>142505038</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>ancestral</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MROH5</td>\n",
       "      <td>1.06700</td>\n",
       "      <td>-0.19014</td>\n",
       "      <td>0.78720</td>\n",
       "      <td>0.38495</td>\n",
       "      <td>-0.215</td>\n",
       "      <td>0.99</td>\n",
       "      <td>0.82</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.99</td>\n",
       "      <td>-1</td>\n",
       "      <td>-9</td>\n",
       "      <td>-11</td>\n",
       "      <td>-12</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1342297</th>\n",
       "      <td>chr7</td>\n",
       "      <td>4215403</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>yes</td>\n",
       "      <td>174.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.03</td>\n",
       "      <td>0.03</td>\n",
       "      <td>0.03</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>yes</td>\n",
       "      <td>2616.0</td>\n",
       "      <td>143312.0</td>\n",
       "      <td>0.018254</td>\n",
       "      <td>yes</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.00637</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00865</td>\n",
       "      <td>0.02976</td>\n",
       "      <td>0.0338</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.10327</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr7_4179789_4232380</td>\n",
       "      <td>0.035096</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SDK1</td>\n",
       "      <td>1.16140</td>\n",
       "      <td>-2.10040</td>\n",
       "      <td>0.29493</td>\n",
       "      <td>6.69840</td>\n",
       "      <td>0.533</td>\n",
       "      <td>0.99</td>\n",
       "      <td>0.03</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.99</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>23</td>\n",
       "      <td>14</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1268358</th>\n",
       "      <td>chr6</td>\n",
       "      <td>53529252</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>yes</td>\n",
       "      <td>265.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.09</td>\n",
       "      <td>0.08</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.06</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>yes</td>\n",
       "      <td>7496.0</td>\n",
       "      <td>143150.0</td>\n",
       "      <td>0.052365</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.05</td>\n",
       "      <td>KLHL31</td>\n",
       "      <td>0.88648</td>\n",
       "      <td>0.76527</td>\n",
       "      <td>0.56321</td>\n",
       "      <td>1.86830</td>\n",
       "      <td>0.563</td>\n",
       "      <td>0.99</td>\n",
       "      <td>0.04</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.99</td>\n",
       "      <td>-2</td>\n",
       "      <td>35</td>\n",
       "      <td>-1</td>\n",
       "      <td>35</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>ancient</td>\n",
       "      <td>introgressed</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>705859</th>\n",
       "      <td>chr18</td>\n",
       "      <td>59805532</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>ancestral</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PIGN</td>\n",
       "      <td>0.94399</td>\n",
       "      <td>0.41579</td>\n",
       "      <td>0.78236</td>\n",
       "      <td>1.40550</td>\n",
       "      <td>-0.186</td>\n",
       "      <td>0.99</td>\n",
       "      <td>0.17</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.99</td>\n",
       "      <td>-1</td>\n",
       "      <td>-2</td>\n",
       "      <td>0</td>\n",
       "      <td>-2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>685458</th>\n",
       "      <td>chr18</td>\n",
       "      <td>28611139</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>ancestral</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DSC3</td>\n",
       "      <td>1.08270</td>\n",
       "      <td>-0.64799</td>\n",
       "      <td>0.89818</td>\n",
       "      <td>0.58048</td>\n",
       "      <td>-0.843</td>\n",
       "      <td>0.98</td>\n",
       "      <td>0.57</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.98</td>\n",
       "      <td>-1</td>\n",
       "      <td>-13</td>\n",
       "      <td>-22</td>\n",
       "      <td>-13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>171355</th>\n",
       "      <td>chr10</td>\n",
       "      <td>43869097</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>ancestral</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>46002.0</td>\n",
       "      <td>143016.0</td>\n",
       "      <td>0.321656</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FXYD4</td>\n",
       "      <td>1.12530</td>\n",
       "      <td>-0.30260</td>\n",
       "      <td>0.76663</td>\n",
       "      <td>0.60500</td>\n",
       "      <td>-1.852</td>\n",
       "      <td>0.98</td>\n",
       "      <td>0.45</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.98</td>\n",
       "      <td>2</td>\n",
       "      <td>15</td>\n",
       "      <td>2</td>\n",
       "      <td>16</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17156</th>\n",
       "      <td>chr1</td>\n",
       "      <td>19633923</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>4.0</td>\n",
       "      <td>143222.0</td>\n",
       "      <td>0.000028</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AKR7A2</td>\n",
       "      <td>1.08410</td>\n",
       "      <td>-0.41709</td>\n",
       "      <td>0.99174</td>\n",
       "      <td>0.02978</td>\n",
       "      <td>-1.031</td>\n",
       "      <td>0.98</td>\n",
       "      <td>0.35</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.98</td>\n",
       "      <td>-2</td>\n",
       "      <td>-19</td>\n",
       "      <td>-19</td>\n",
       "      <td>-45</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>555035</th>\n",
       "      <td>chr16</td>\n",
       "      <td>920071</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Vindija</td>\n",
       "      <td>yes</td>\n",
       "      <td>57.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.04</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>yes</td>\n",
       "      <td>257.0</td>\n",
       "      <td>143258.0</td>\n",
       "      <td>0.001794</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LMF1</td>\n",
       "      <td>1.26920</td>\n",
       "      <td>-1.79340</td>\n",
       "      <td>0.90896</td>\n",
       "      <td>0.44243</td>\n",
       "      <td>-1.083</td>\n",
       "      <td>0.97</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.97</td>\n",
       "      <td>-2</td>\n",
       "      <td>-5</td>\n",
       "      <td>-5</td>\n",
       "      <td>-6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>680581</th>\n",
       "      <td>chr18</td>\n",
       "      <td>13095591</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>ancestral</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>19100.0</td>\n",
       "      <td>143184.0</td>\n",
       "      <td>0.133395</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CEP192</td>\n",
       "      <td>0.92191</td>\n",
       "      <td>0.99822</td>\n",
       "      <td>0.48300</td>\n",
       "      <td>5.20440</td>\n",
       "      <td>-0.254</td>\n",
       "      <td>0.97</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.97</td>\n",
       "      <td>2</td>\n",
       "      <td>-33</td>\n",
       "      <td>2</td>\n",
       "      <td>-28</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>126381</th>\n",
       "      <td>chr1</td>\n",
       "      <td>228566387</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>yes</td>\n",
       "      <td>17.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>yes</td>\n",
       "      <td>115.0</td>\n",
       "      <td>143086.0</td>\n",
       "      <td>0.000804</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>OBSCN</td>\n",
       "      <td>1.04430</td>\n",
       "      <td>-1.14490</td>\n",
       "      <td>0.79291</td>\n",
       "      <td>3.71400</td>\n",
       "      <td>0.177</td>\n",
       "      <td>0.97</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.97</td>\n",
       "      <td>2</td>\n",
       "      <td>-47</td>\n",
       "      <td>2</td>\n",
       "      <td>-7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>737762</th>\n",
       "      <td>chr19</td>\n",
       "      <td>36243813</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>ancestral</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AC002398.9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.688</td>\n",
       "      <td>0.96</td>\n",
       "      <td>0.45</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.96</td>\n",
       "      <td>2</td>\n",
       "      <td>27</td>\n",
       "      <td>49</td>\n",
       "      <td>27</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>737763</th>\n",
       "      <td>chr19</td>\n",
       "      <td>36243813</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>ancestral</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LIN37</td>\n",
       "      <td>0.77853</td>\n",
       "      <td>0.98522</td>\n",
       "      <td>0.41053</td>\n",
       "      <td>2.08820</td>\n",
       "      <td>-0.688</td>\n",
       "      <td>0.96</td>\n",
       "      <td>0.45</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.96</td>\n",
       "      <td>2</td>\n",
       "      <td>27</td>\n",
       "      <td>49</td>\n",
       "      <td>27</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>741704</th>\n",
       "      <td>chr19</td>\n",
       "      <td>42092815</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>a</td>\n",
       "      <td>ancestral</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CEACAM21</td>\n",
       "      <td>1.06900</td>\n",
       "      <td>-0.31618</td>\n",
       "      <td>1.14870</td>\n",
       "      <td>-0.53013</td>\n",
       "      <td>-2.214</td>\n",
       "      <td>0.95</td>\n",
       "      <td>0.02</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.95</td>\n",
       "      <td>2</td>\n",
       "      <td>26</td>\n",
       "      <td>-49</td>\n",
       "      <td>19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>773401</th>\n",
       "      <td>chr2</td>\n",
       "      <td>31133923</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Late Neanderthal</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>4.0</td>\n",
       "      <td>143204.0</td>\n",
       "      <td>0.000028</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GALNT14</td>\n",
       "      <td>0.93309</td>\n",
       "      <td>0.43475</td>\n",
       "      <td>0.80467</td>\n",
       "      <td>1.08660</td>\n",
       "      <td>0.533</td>\n",
       "      <td>0.95</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.95</td>\n",
       "      <td>-1</td>\n",
       "      <td>31</td>\n",
       "      <td>-1</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         chrom        pos ref_allele alt_allele ancestral_allele    anc_dev  \\\n",
       "78185     chr1  144923837          A          T              NaN    derived   \n",
       "302444   chr11  108810969          C          G                C    derived   \n",
       "4234      chr1    5935162          A          T                T  ancestral   \n",
       "121279    chr1  218475625          T          A                A  ancestral   \n",
       "1455913   chr8    8999187          G          C                C  ancestral   \n",
       "737199   chr19   35506729          G          A                G    derived   \n",
       "1531582   chr8  142505038          T          C                C  ancestral   \n",
       "1342297   chr7    4215403          A          G                A    derived   \n",
       "1268358   chr6   53529252          C          T                C    derived   \n",
       "705859   chr18   59805532          T          C                C  ancestral   \n",
       "685458   chr18   28611139          A          C                C  ancestral   \n",
       "171355   chr10   43869097          C          A                A  ancestral   \n",
       "17156     chr1   19633923          C          T                C    derived   \n",
       "555035   chr16     920071          C          T                C    derived   \n",
       "680581   chr18   13095591          G          A                A  ancestral   \n",
       "126381    chr1  228566387          G          A                G    derived   \n",
       "737762   chr19   36243813          G          A                A  ancestral   \n",
       "737763   chr19   36243813          G          A                A  ancestral   \n",
       "741704   chr19   42092815          T          A                a  ancestral   \n",
       "773401    chr2   31133923          T          C                T    derived   \n",
       "\n",
       "        variant_type altai_gt chagyrskaya_gt denisovan_gt vindija_gt  \\\n",
       "78185            snv      0/1            0/1          ./.        0/1   \n",
       "302444           snv      0/1            0/0          0/0        1/1   \n",
       "4234             snv      1/1            1/1          1/1        1/1   \n",
       "121279           snv      1/1            1/1          1/1        1/1   \n",
       "1455913          snv      1/1            1/1          1/1        1/1   \n",
       "737199           snv      0/0            0/0          1/1        0/0   \n",
       "1531582          snv      1/1            1/1          0/0        1/1   \n",
       "1342297          snv      1/1            1/1          0/0        1/1   \n",
       "1268358          snv      0/1            1/1          0/0        1/1   \n",
       "705859           snv      1/1            1/1          1/1        1/1   \n",
       "685458           snv      1/1            1/1          1/1        1/1   \n",
       "171355           snv      1/1            1/1          1/1        1/1   \n",
       "17156            snv      0/0            0/0          0/1        0/0   \n",
       "555035           snv      0/0            0/0          0/0        0/1   \n",
       "680581           snv      1/1            1/1          0/0        1/1   \n",
       "126381           snv      0/0            0/0          1/1        0/0   \n",
       "737762           snv      1/1            1/1          1/1        1/1   \n",
       "737763           snv      1/1            1/1          1/1        1/1   \n",
       "741704           snv      1/1            1/1          1/1        1/1   \n",
       "773401           snv      0/0            0/1          0/0        0/1   \n",
       "\n",
       "         altai_gt_boolean  chagyrskaya_gt_boolean  denisovan_gt_boolean  \\\n",
       "78185                True                    True                 False   \n",
       "302444               True                   False                 False   \n",
       "4234                 True                    True                  True   \n",
       "121279               True                    True                  True   \n",
       "1455913              True                    True                  True   \n",
       "737199              False                   False                  True   \n",
       "1531582              True                    True                 False   \n",
       "1342297              True                    True                 False   \n",
       "1268358              True                    True                 False   \n",
       "705859               True                    True                  True   \n",
       "685458               True                    True                  True   \n",
       "171355               True                    True                  True   \n",
       "17156               False                   False                  True   \n",
       "555035              False                   False                 False   \n",
       "680581               True                    True                 False   \n",
       "126381              False                   False                  True   \n",
       "737762               True                    True                  True   \n",
       "737763               True                    True                  True   \n",
       "741704               True                    True                  True   \n",
       "773401              False                    True                 False   \n",
       "\n",
       "         vindija_gt_boolean      distribution present_in_1KG  \\\n",
       "78185                  True       Neanderthal             no   \n",
       "302444                 True             Other            yes   \n",
       "4234                   True            Shared             no   \n",
       "121279                 True            Shared            yes   \n",
       "1455913                True            Shared            yes   \n",
       "737199                False         Denisovan            yes   \n",
       "1531582                True       Neanderthal             no   \n",
       "1342297                True       Neanderthal            yes   \n",
       "1268358                True       Neanderthal            yes   \n",
       "705859                 True            Shared             no   \n",
       "685458                 True            Shared             no   \n",
       "171355                 True            Shared             no   \n",
       "17156                 False         Denisovan             no   \n",
       "555035                 True           Vindija            yes   \n",
       "680581                 True       Neanderthal             no   \n",
       "126381                False         Denisovan            yes   \n",
       "737762                 True            Shared             no   \n",
       "737763                 True            Shared             no   \n",
       "741704                 True            Shared             no   \n",
       "773401                 True  Late Neanderthal             no   \n",
       "\n",
       "         1KG_allele_count  1KG_allele_number  1KG_allele_frequency  \\\n",
       "78185                 NaN                NaN                   NaN   \n",
       "302444                2.0             5096.0                  0.00   \n",
       "4234                  NaN                NaN                   NaN   \n",
       "121279              436.0             5096.0                  0.09   \n",
       "1455913            2597.0             5096.0                  0.51   \n",
       "737199             1292.0             5096.0                  0.25   \n",
       "1531582               NaN                NaN                   NaN   \n",
       "1342297             174.0             5096.0                  0.03   \n",
       "1268358             265.0             5096.0                  0.05   \n",
       "705859                NaN                NaN                   NaN   \n",
       "685458                NaN                NaN                   NaN   \n",
       "171355                NaN                NaN                   NaN   \n",
       "17156                 NaN                NaN                   NaN   \n",
       "555035               57.0             5096.0                  0.01   \n",
       "680581                NaN                NaN                   NaN   \n",
       "126381               17.0             5096.0                  0.00   \n",
       "737762                NaN                NaN                   NaN   \n",
       "737763                NaN                NaN                   NaN   \n",
       "741704                NaN                NaN                   NaN   \n",
       "773401                NaN                NaN                   NaN   \n",
       "\n",
       "         1KG_EAS_AF  1KG_EUR_AF  1KG_AFR_AF  1KG_AMR_AF  1KG_SAS_AF  \\\n",
       "78185           NaN         NaN         NaN         NaN         NaN   \n",
       "302444         0.00        0.00        0.00        0.00        0.00   \n",
       "4234            NaN         NaN         NaN         NaN         NaN   \n",
       "121279         0.06        0.04        0.23        0.04        0.01   \n",
       "1455913        0.82        0.31        0.37        0.41        0.66   \n",
       "737199         0.22        0.34        0.27        0.30        0.15   \n",
       "1531582         NaN         NaN         NaN         NaN         NaN   \n",
       "1342297        0.03        0.03        0.00        0.01        0.10   \n",
       "1268358        0.09        0.08        0.00        0.06        0.05   \n",
       "705859          NaN         NaN         NaN         NaN         NaN   \n",
       "685458          NaN         NaN         NaN         NaN         NaN   \n",
       "171355          NaN         NaN         NaN         NaN         NaN   \n",
       "17156           NaN         NaN         NaN         NaN         NaN   \n",
       "555035         0.01        0.00        0.00        0.01        0.04   \n",
       "680581          NaN         NaN         NaN         NaN         NaN   \n",
       "126381         0.01        0.00        0.00        0.00        0.00   \n",
       "737762          NaN         NaN         NaN         NaN         NaN   \n",
       "737763          NaN         NaN         NaN         NaN         NaN   \n",
       "741704          NaN         NaN         NaN         NaN         NaN   \n",
       "773401          NaN         NaN         NaN         NaN         NaN   \n",
       "\n",
       "         1KG_non_ASW_AFR_AF present_in_gnomAD  gnomAD_allele_count  \\\n",
       "78185                   NaN                no                  NaN   \n",
       "302444             0.000000               yes                  8.0   \n",
       "4234                    NaN                no                  NaN   \n",
       "121279             0.230019               yes              11299.0   \n",
       "1455913            0.367446               yes              52466.0   \n",
       "737199             0.276803               yes              43574.0   \n",
       "1531582                 NaN                no                  NaN   \n",
       "1342297            0.000000               yes               2616.0   \n",
       "1268358            0.000000               yes               7496.0   \n",
       "705859                  NaN                no                  NaN   \n",
       "685458                  NaN                no                  NaN   \n",
       "171355                  NaN               yes              46002.0   \n",
       "17156                   NaN               yes                  4.0   \n",
       "555035             0.000000               yes                257.0   \n",
       "680581                  NaN               yes              19100.0   \n",
       "126381             0.000000               yes                115.0   \n",
       "737762                  NaN                no                  NaN   \n",
       "737763                  NaN                no                  NaN   \n",
       "741704                  NaN                no                  NaN   \n",
       "773401                  NaN               yes                  4.0   \n",
       "\n",
       "         gnomAD_allele_number  gnomAD_allele_frequency Vernot_introgressed  \\\n",
       "78185                     NaN                      NaN                  no   \n",
       "302444               143086.0                 0.000056                  no   \n",
       "4234                      NaN                      NaN                  no   \n",
       "121279               143136.0                 0.078939                  no   \n",
       "1455913              142958.0                 0.367003                  no   \n",
       "737199               143070.0                 0.304564                  no   \n",
       "1531582                   NaN                      NaN                  no   \n",
       "1342297              143312.0                 0.018254                 yes   \n",
       "1268358              143150.0                 0.052365                  no   \n",
       "705859                    NaN                      NaN                  no   \n",
       "685458                    NaN                      NaN                  no   \n",
       "171355               143016.0                 0.321656                  no   \n",
       "17156                143222.0                 0.000028                  no   \n",
       "555035               143258.0                 0.001794                  no   \n",
       "680581               143184.0                 0.133395                  no   \n",
       "126381               143086.0                 0.000804                  no   \n",
       "737762                    NaN                      NaN                  no   \n",
       "737763                    NaN                      NaN                  no   \n",
       "741704                    NaN                      NaN                  no   \n",
       "773401               143204.0                 0.000028                  no   \n",
       "\n",
       "        Vernot_ancestral_allele Vernot_derived_allele  \\\n",
       "78185                       NaN                   NaN   \n",
       "302444                      NaN                   NaN   \n",
       "4234                        NaN                   NaN   \n",
       "121279                      NaN                   NaN   \n",
       "1455913                     NaN                   NaN   \n",
       "737199                      NaN                   NaN   \n",
       "1531582                     NaN                   NaN   \n",
       "1342297                       A                     G   \n",
       "1268358                     NaN                   NaN   \n",
       "705859                      NaN                   NaN   \n",
       "685458                      NaN                   NaN   \n",
       "171355                      NaN                   NaN   \n",
       "17156                       NaN                   NaN   \n",
       "555035                      NaN                   NaN   \n",
       "680581                      NaN                   NaN   \n",
       "126381                      NaN                   NaN   \n",
       "737762                      NaN                   NaN   \n",
       "737763                      NaN                   NaN   \n",
       "741704                      NaN                   NaN   \n",
       "773401                      NaN                   NaN   \n",
       "\n",
       "         Vernot_ancestral_derived_code  Vernot_AFA_AF  Vernot_AFR_AF  \\\n",
       "78185                              NaN            NaN            NaN   \n",
       "302444                             NaN            NaN            NaN   \n",
       "4234                               NaN            NaN            NaN   \n",
       "121279                             NaN            NaN            NaN   \n",
       "1455913                            NaN            NaN            NaN   \n",
       "737199                             NaN            NaN            NaN   \n",
       "1531582                            NaN            NaN            NaN   \n",
       "1342297                            1.0        0.00637            0.0   \n",
       "1268358                            NaN            NaN            NaN   \n",
       "705859                             NaN            NaN            NaN   \n",
       "685458                             NaN            NaN            NaN   \n",
       "171355                             NaN            NaN            NaN   \n",
       "17156                              NaN            NaN            NaN   \n",
       "555035                             NaN            NaN            NaN   \n",
       "680581                             NaN            NaN            NaN   \n",
       "126381                             NaN            NaN            NaN   \n",
       "737762                             NaN            NaN            NaN   \n",
       "737763                             NaN            NaN            NaN   \n",
       "741704                             NaN            NaN            NaN   \n",
       "773401                             NaN            NaN            NaN   \n",
       "\n",
       "         Vernot_AMR_AF  Vernot_EAS_AF  Vernot_EUR_AF  Vernot_PNG_AF  \\\n",
       "78185              NaN            NaN            NaN            NaN   \n",
       "302444             NaN            NaN            NaN            NaN   \n",
       "4234               NaN            NaN            NaN            NaN   \n",
       "121279             NaN            NaN            NaN            NaN   \n",
       "1455913            NaN            NaN            NaN            NaN   \n",
       "737199             NaN            NaN            NaN            NaN   \n",
       "1531582            NaN            NaN            NaN            NaN   \n",
       "1342297        0.00865        0.02976         0.0338            0.0   \n",
       "1268358            NaN            NaN            NaN            NaN   \n",
       "705859             NaN            NaN            NaN            NaN   \n",
       "685458             NaN            NaN            NaN            NaN   \n",
       "171355             NaN            NaN            NaN            NaN   \n",
       "17156              NaN            NaN            NaN            NaN   \n",
       "555035             NaN            NaN            NaN            NaN   \n",
       "680581             NaN            NaN            NaN            NaN   \n",
       "126381             NaN            NaN            NaN            NaN   \n",
       "737762             NaN            NaN            NaN            NaN   \n",
       "737763             NaN            NaN            NaN            NaN   \n",
       "741704             NaN            NaN            NaN            NaN   \n",
       "773401             NaN            NaN            NaN            NaN   \n",
       "\n",
       "         Vernot_SAS_AF Vernot_Denisovan_base Vernot_Neanderthal_base_1  \\\n",
       "78185              NaN                   NaN                       NaN   \n",
       "302444             NaN                   NaN                       NaN   \n",
       "4234               NaN                   NaN                       NaN   \n",
       "121279             NaN                   NaN                       NaN   \n",
       "1455913            NaN                   NaN                       NaN   \n",
       "737199             NaN                   NaN                       NaN   \n",
       "1531582            NaN                   NaN                       NaN   \n",
       "1342297        0.10327                     A                         G   \n",
       "1268358            NaN                   NaN                       NaN   \n",
       "705859             NaN                   NaN                       NaN   \n",
       "685458             NaN                   NaN                       NaN   \n",
       "171355             NaN                   NaN                       NaN   \n",
       "17156              NaN                   NaN                       NaN   \n",
       "555035             NaN                   NaN                       NaN   \n",
       "680581             NaN                   NaN                       NaN   \n",
       "126381             NaN                   NaN                       NaN   \n",
       "737762             NaN                   NaN                       NaN   \n",
       "737763             NaN                   NaN                       NaN   \n",
       "741704             NaN                   NaN                       NaN   \n",
       "773401             NaN                   NaN                       NaN   \n",
       "\n",
       "        Vernot_Neanderthal_base_2  Vernot_haplotype_tag  \\\n",
       "78185                         NaN                   NaN   \n",
       "302444                        NaN                   NaN   \n",
       "4234                          NaN                   NaN   \n",
       "121279                        NaN                   NaN   \n",
       "1455913                       NaN                   NaN   \n",
       "737199                        NaN                   NaN   \n",
       "1531582                       NaN                   NaN   \n",
       "1342297                       NaN  chr7_4179789_4232380   \n",
       "1268358                       NaN                   NaN   \n",
       "705859                        NaN                   NaN   \n",
       "685458                        NaN                   NaN   \n",
       "171355                        NaN                   NaN   \n",
       "17156                         NaN                   NaN   \n",
       "555035                        NaN                   NaN   \n",
       "680581                        NaN                   NaN   \n",
       "126381                        NaN                   NaN   \n",
       "737762                        NaN                   NaN   \n",
       "737763                        NaN                   NaN   \n",
       "741704                        NaN                   NaN   \n",
       "773401                        NaN                   NaN   \n",
       "\n",
       "         Vernot_introgressed_AF Browning_introgressed  Browning_ref_alt  \\\n",
       "78185                       NaN                    no               NaN   \n",
       "302444                      NaN                    no               NaN   \n",
       "4234                        NaN                    no               NaN   \n",
       "121279                      NaN                    no               NaN   \n",
       "1455913                     NaN                    no               NaN   \n",
       "737199                      NaN                    no               NaN   \n",
       "1531582                     NaN                    no               NaN   \n",
       "1342297                0.035096                    no               NaN   \n",
       "1268358                     NaN                   yes               1.0   \n",
       "705859                      NaN                    no               NaN   \n",
       "685458                      NaN                    no               NaN   \n",
       "171355                      NaN                    no               NaN   \n",
       "17156                       NaN                    no               NaN   \n",
       "555035                      NaN                    no               NaN   \n",
       "680581                      NaN                    no               NaN   \n",
       "126381                      NaN                    no               NaN   \n",
       "737762                      NaN                    no               NaN   \n",
       "737763                      NaN                    no               NaN   \n",
       "741704                      NaN                    no               NaN   \n",
       "773401                      NaN                    no               NaN   \n",
       "\n",
       "         Browning_introgressed_AF  annotation   mis_oe    mis_z   lof_oe  \\\n",
       "78185                         NaN     PDE4DIP  1.12420 -1.40300  0.68045   \n",
       "302444                        NaN       DDX10  0.96014  0.30359  0.70688   \n",
       "4234                          NaN       NPHP4  1.02290 -0.24249  0.78155   \n",
       "121279                        NaN       RRP15  1.10360 -0.45285  0.62321   \n",
       "1455913                       NaN     PPP1R3B  1.24760 -1.10860  0.38350   \n",
       "737199                        NaN     GRAMD1A  0.83225  1.29050  0.31602   \n",
       "1531582                       NaN       MROH5  1.06700 -0.19014  0.78720   \n",
       "1342297                       NaN        SDK1  1.16140 -2.10040  0.29493   \n",
       "1268358                      0.05      KLHL31  0.88648  0.76527  0.56321   \n",
       "705859                        NaN        PIGN  0.94399  0.41579  0.78236   \n",
       "685458                        NaN        DSC3  1.08270 -0.64799  0.89818   \n",
       "171355                        NaN       FXYD4  1.12530 -0.30260  0.76663   \n",
       "17156                         NaN      AKR7A2  1.08410 -0.41709  0.99174   \n",
       "555035                        NaN        LMF1  1.26920 -1.79340  0.90896   \n",
       "680581                        NaN      CEP192  0.92191  0.99822  0.48300   \n",
       "126381                        NaN       OBSCN  1.04430 -1.14490  0.79291   \n",
       "737762                        NaN  AC002398.9      NaN      NaN      NaN   \n",
       "737763                        NaN       LIN37  0.77853  0.98522  0.41053   \n",
       "741704                        NaN    CEACAM21  1.06900 -0.31618  1.14870   \n",
       "773401                        NaN     GALNT14  0.93309  0.43475  0.80467   \n",
       "\n",
       "           lof_z  phyloP  ag_delta  al_delta  dg_delta  dl_delta  delta_max  \\\n",
       "78185    3.12940  -0.264      1.00      0.97       0.0       0.0       1.00   \n",
       "302444   1.82750  -0.175      1.00      0.01       0.0       0.0       1.00   \n",
       "4234     1.66690  -0.557      1.00      0.91       0.0       0.0       1.00   \n",
       "121279   1.17020  -0.372      1.00      0.70       0.0       0.0       1.00   \n",
       "1455913  1.59780  -0.216      0.99      0.95       0.0       0.0       0.99   \n",
       "737199   4.06520  -1.373      0.99      0.01       0.0       0.0       0.99   \n",
       "1531582  0.38495  -0.215      0.99      0.82       0.0       0.0       0.99   \n",
       "1342297  6.69840   0.533      0.99      0.03       0.0       0.0       0.99   \n",
       "1268358  1.86830   0.563      0.99      0.04       0.0       0.0       0.99   \n",
       "705859   1.40550  -0.186      0.99      0.17       0.0       0.0       0.99   \n",
       "685458   0.58048  -0.843      0.98      0.57       0.0       0.0       0.98   \n",
       "171355   0.60500  -1.852      0.98      0.45       0.0       0.0       0.98   \n",
       "17156    0.02978  -1.031      0.98      0.35       0.0       0.0       0.98   \n",
       "555035   0.44243  -1.083      0.97      0.02       0.0       0.0       0.97   \n",
       "680581   5.20440  -0.254      0.97      0.05       0.0       0.0       0.97   \n",
       "126381   3.71400   0.177      0.97      0.10       0.0       0.0       0.97   \n",
       "737762       NaN  -0.688      0.96      0.45       0.0       0.0       0.96   \n",
       "737763   2.08820  -0.688      0.96      0.45       0.0       0.0       0.96   \n",
       "741704  -0.53013  -2.214      0.95      0.02      -0.0       0.0       0.95   \n",
       "773401   1.08660   0.533      0.95      0.00       0.0       0.0       0.95   \n",
       "\n",
       "         ag_pos  al_pos  dg_pos  dl_pos  Adipose_Subcutaneous  \\\n",
       "78185        -2     -16     -16     -29                   NaN   \n",
       "302444        1       4       4       5                   NaN   \n",
       "4234         -2      -8     -43     -44                   NaN   \n",
       "121279        2      11      11      15                   0.0   \n",
       "1455913      -1      -9     -45     -22                   0.0   \n",
       "737199        2      -1       2      -1                   1.0   \n",
       "1531582      -1      -9     -11     -12                   NaN   \n",
       "1342297       1       4      23      14                   0.0   \n",
       "1268358      -2      35      -1      35                   NaN   \n",
       "705859       -1      -2       0      -2                   NaN   \n",
       "685458       -1     -13     -22     -13                   NaN   \n",
       "171355        2      15       2      16                   0.0   \n",
       "17156        -2     -19     -19     -45                   NaN   \n",
       "555035       -2      -5      -5      -6                   NaN   \n",
       "680581        2     -33       2     -28                   1.0   \n",
       "126381        2     -47       2      -7                   NaN   \n",
       "737762        2      27      49      27                   NaN   \n",
       "737763        2      27      49      27                   NaN   \n",
       "741704        2      26     -49      19                   NaN   \n",
       "773401       -1      31      -1       2                   NaN   \n",
       "\n",
       "         Adipose_Visceral_Omentum  Adrenal_Gland  Artery_Aorta  \\\n",
       "78185                         NaN            NaN           NaN   \n",
       "302444                        NaN            NaN           NaN   \n",
       "4234                          NaN            NaN           NaN   \n",
       "121279                        0.0            0.0           0.0   \n",
       "1455913                       0.0            0.0           0.0   \n",
       "737199                        1.0            0.0           1.0   \n",
       "1531582                       NaN            NaN           NaN   \n",
       "1342297                       0.0            0.0           0.0   \n",
       "1268358                       NaN            NaN           NaN   \n",
       "705859                        NaN            NaN           NaN   \n",
       "685458                        NaN            NaN           NaN   \n",
       "171355                        1.0            0.0           0.0   \n",
       "17156                         NaN            NaN           NaN   \n",
       "555035                        NaN            NaN           NaN   \n",
       "680581                        0.0            0.0           1.0   \n",
       "126381                        NaN            NaN           NaN   \n",
       "737762                        NaN            NaN           NaN   \n",
       "737763                        NaN            NaN           NaN   \n",
       "741704                        NaN            NaN           NaN   \n",
       "773401                        NaN            NaN           NaN   \n",
       "\n",
       "         Artery_Coronary  Artery_Tibial  Brain_Amygdala  \\\n",
       "78185                NaN            NaN             NaN   \n",
       "302444               NaN            NaN             NaN   \n",
       "4234                 NaN            NaN             NaN   \n",
       "121279               0.0            0.0             0.0   \n",
       "1455913              0.0            1.0             0.0   \n",
       "737199               0.0            1.0             0.0   \n",
       "1531582              NaN            NaN             NaN   \n",
       "1342297              0.0            0.0             0.0   \n",
       "1268358              NaN            NaN             NaN   \n",
       "705859               NaN            NaN             NaN   \n",
       "685458               NaN            NaN             NaN   \n",
       "171355               0.0            0.0             0.0   \n",
       "17156                NaN            NaN             NaN   \n",
       "555035               NaN            NaN             NaN   \n",
       "680581               0.0            0.0             0.0   \n",
       "126381               NaN            NaN             NaN   \n",
       "737762               NaN            NaN             NaN   \n",
       "737763               NaN            NaN             NaN   \n",
       "741704               NaN            NaN             NaN   \n",
       "773401               NaN            NaN             NaN   \n",
       "\n",
       "         Brain_Anterior_cingulate_cortex_BA24  Brain_Caudate_basal_ganglia  \\\n",
       "78185                                     NaN                          NaN   \n",
       "302444                                    NaN                          NaN   \n",
       "4234                                      NaN                          NaN   \n",
       "121279                                    0.0                          0.0   \n",
       "1455913                                   0.0                          0.0   \n",
       "737199                                    1.0                          0.0   \n",
       "1531582                                   NaN                          NaN   \n",
       "1342297                                   0.0                          0.0   \n",
       "1268358                                   NaN                          NaN   \n",
       "705859                                    NaN                          NaN   \n",
       "685458                                    NaN                          NaN   \n",
       "171355                                    0.0                          0.0   \n",
       "17156                                     NaN                          NaN   \n",
       "555035                                    NaN                          NaN   \n",
       "680581                                    0.0                          0.0   \n",
       "126381                                    NaN                          NaN   \n",
       "737762                                    NaN                          NaN   \n",
       "737763                                    NaN                          NaN   \n",
       "741704                                    NaN                          NaN   \n",
       "773401                                    NaN                          NaN   \n",
       "\n",
       "         Brain_Cerebellar_Hemisphere  Brain_Cerebellum  Brain_Cortex  \\\n",
       "78185                            NaN               NaN           NaN   \n",
       "302444                           NaN               NaN           NaN   \n",
       "4234                             NaN               NaN           NaN   \n",
       "121279                           0.0               0.0           0.0   \n",
       "1455913                          0.0               0.0           0.0   \n",
       "737199                           0.0               0.0           1.0   \n",
       "1531582                          NaN               NaN           NaN   \n",
       "1342297                          0.0               0.0           0.0   \n",
       "1268358                          NaN               NaN           NaN   \n",
       "705859                           NaN               NaN           NaN   \n",
       "685458                           NaN               NaN           NaN   \n",
       "171355                           0.0               0.0           1.0   \n",
       "17156                            NaN               NaN           NaN   \n",
       "555035                           NaN               NaN           NaN   \n",
       "680581                           1.0               0.0           0.0   \n",
       "126381                           NaN               NaN           NaN   \n",
       "737762                           NaN               NaN           NaN   \n",
       "737763                           NaN               NaN           NaN   \n",
       "741704                           NaN               NaN           NaN   \n",
       "773401                           NaN               NaN           NaN   \n",
       "\n",
       "         Brain_Frontal_Cortex_BA9  Brain_Hippocampus  Brain_Hypothalamus  \\\n",
       "78185                         NaN                NaN                 NaN   \n",
       "302444                        NaN                NaN                 NaN   \n",
       "4234                          NaN                NaN                 NaN   \n",
       "121279                        0.0                0.0                 0.0   \n",
       "1455913                       0.0                0.0                 0.0   \n",
       "737199                        0.0                0.0                 0.0   \n",
       "1531582                       NaN                NaN                 NaN   \n",
       "1342297                       0.0                0.0                 0.0   \n",
       "1268358                       NaN                NaN                 NaN   \n",
       "705859                        NaN                NaN                 NaN   \n",
       "685458                        NaN                NaN                 NaN   \n",
       "171355                        0.0                1.0                 0.0   \n",
       "17156                         NaN                NaN                 NaN   \n",
       "555035                        NaN                NaN                 NaN   \n",
       "680581                        0.0                0.0                 0.0   \n",
       "126381                        NaN                NaN                 NaN   \n",
       "737762                        NaN                NaN                 NaN   \n",
       "737763                        NaN                NaN                 NaN   \n",
       "741704                        NaN                NaN                 NaN   \n",
       "773401                        NaN                NaN                 NaN   \n",
       "\n",
       "         Brain_Nucleus_accumbens_basal_ganglia  Brain_Putamen_basal_ganglia  \\\n",
       "78185                                      NaN                          NaN   \n",
       "302444                                     NaN                          NaN   \n",
       "4234                                       NaN                          NaN   \n",
       "121279                                     0.0                          0.0   \n",
       "1455913                                    0.0                          0.0   \n",
       "737199                                     0.0                          0.0   \n",
       "1531582                                    NaN                          NaN   \n",
       "1342297                                    0.0                          0.0   \n",
       "1268358                                    NaN                          NaN   \n",
       "705859                                     NaN                          NaN   \n",
       "685458                                     NaN                          NaN   \n",
       "171355                                     0.0                          1.0   \n",
       "17156                                      NaN                          NaN   \n",
       "555035                                     NaN                          NaN   \n",
       "680581                                     0.0                          0.0   \n",
       "126381                                     NaN                          NaN   \n",
       "737762                                     NaN                          NaN   \n",
       "737763                                     NaN                          NaN   \n",
       "741704                                     NaN                          NaN   \n",
       "773401                                     NaN                          NaN   \n",
       "\n",
       "         Brain_Spinal_cord_cervical_c-1  Brain_Substantia_nigra  \\\n",
       "78185                               NaN                     NaN   \n",
       "302444                              NaN                     NaN   \n",
       "4234                                NaN                     NaN   \n",
       "121279                              0.0                     0.0   \n",
       "1455913                             0.0                     0.0   \n",
       "737199                              0.0                     0.0   \n",
       "1531582                             NaN                     NaN   \n",
       "1342297                             0.0                     0.0   \n",
       "1268358                             NaN                     NaN   \n",
       "705859                              NaN                     NaN   \n",
       "685458                              NaN                     NaN   \n",
       "171355                              0.0                     0.0   \n",
       "17156                               NaN                     NaN   \n",
       "555035                              NaN                     NaN   \n",
       "680581                              0.0                     0.0   \n",
       "126381                              NaN                     NaN   \n",
       "737762                              NaN                     NaN   \n",
       "737763                              NaN                     NaN   \n",
       "741704                              NaN                     NaN   \n",
       "773401                              NaN                     NaN   \n",
       "\n",
       "         Breast_Mammary_Tissue  Cells_Cultured_fibroblasts  \\\n",
       "78185                      NaN                         NaN   \n",
       "302444                     NaN                         NaN   \n",
       "4234                       NaN                         NaN   \n",
       "121279                     0.0                         0.0   \n",
       "1455913                    1.0                         1.0   \n",
       "737199                     1.0                         0.0   \n",
       "1531582                    NaN                         NaN   \n",
       "1342297                    0.0                         0.0   \n",
       "1268358                    NaN                         NaN   \n",
       "705859                     NaN                         NaN   \n",
       "685458                     NaN                         NaN   \n",
       "171355                     0.0                         1.0   \n",
       "17156                      NaN                         NaN   \n",
       "555035                     NaN                         NaN   \n",
       "680581                     0.0                         0.0   \n",
       "126381                     NaN                         NaN   \n",
       "737762                     NaN                         NaN   \n",
       "737763                     NaN                         NaN   \n",
       "741704                     NaN                         NaN   \n",
       "773401                     NaN                         NaN   \n",
       "\n",
       "         Cells_EBV-transformed_lymphocytes  Colon_Sigmoid  Colon_Transverse  \\\n",
       "78185                                  NaN            NaN               NaN   \n",
       "302444                                 NaN            NaN               NaN   \n",
       "4234                                   NaN            NaN               NaN   \n",
       "121279                                 0.0            0.0               0.0   \n",
       "1455913                                0.0            1.0               0.0   \n",
       "737199                                 0.0            0.0               0.0   \n",
       "1531582                                NaN            NaN               NaN   \n",
       "1342297                                0.0            0.0               0.0   \n",
       "1268358                                NaN            NaN               NaN   \n",
       "705859                                 NaN            NaN               NaN   \n",
       "685458                                 NaN            NaN               NaN   \n",
       "171355                                 0.0            0.0               1.0   \n",
       "17156                                  NaN            NaN               NaN   \n",
       "555035                                 NaN            NaN               NaN   \n",
       "680581                                 0.0            0.0               0.0   \n",
       "126381                                 NaN            NaN               NaN   \n",
       "737762                                 NaN            NaN               NaN   \n",
       "737763                                 NaN            NaN               NaN   \n",
       "741704                                 NaN            NaN               NaN   \n",
       "773401                                 NaN            NaN               NaN   \n",
       "\n",
       "         Esophagus_Gastroesophageal_Junction  Esophagus_Mucosa  \\\n",
       "78185                                    NaN               NaN   \n",
       "302444                                   NaN               NaN   \n",
       "4234                                     NaN               NaN   \n",
       "121279                                   0.0               0.0   \n",
       "1455913                                  0.0               0.0   \n",
       "737199                                   0.0               1.0   \n",
       "1531582                                  NaN               NaN   \n",
       "1342297                                  0.0               0.0   \n",
       "1268358                                  NaN               NaN   \n",
       "705859                                   NaN               NaN   \n",
       "685458                                   NaN               NaN   \n",
       "171355                                   0.0               0.0   \n",
       "17156                                    NaN               NaN   \n",
       "555035                                   NaN               NaN   \n",
       "680581                                   1.0               1.0   \n",
       "126381                                   NaN               NaN   \n",
       "737762                                   NaN               NaN   \n",
       "737763                                   NaN               NaN   \n",
       "741704                                   NaN               NaN   \n",
       "773401                                   NaN               NaN   \n",
       "\n",
       "         Esophagus_Muscularis  Heart_Atrial_Appendage  Heart_Left_Ventricle  \\\n",
       "78185                     NaN                     NaN                   NaN   \n",
       "302444                    NaN                     NaN                   NaN   \n",
       "4234                      NaN                     NaN                   NaN   \n",
       "121279                    0.0                     0.0                   0.0   \n",
       "1455913                   0.0                     0.0                   0.0   \n",
       "737199                    1.0                     1.0                   1.0   \n",
       "1531582                   NaN                     NaN                   NaN   \n",
       "1342297                   0.0                     0.0                   0.0   \n",
       "1268358                   NaN                     NaN                   NaN   \n",
       "705859                    NaN                     NaN                   NaN   \n",
       "685458                    NaN                     NaN                   NaN   \n",
       "171355                    0.0                     0.0                   0.0   \n",
       "17156                     NaN                     NaN                   NaN   \n",
       "555035                    NaN                     NaN                   NaN   \n",
       "680581                    0.0                     0.0                   0.0   \n",
       "126381                    NaN                     NaN                   NaN   \n",
       "737762                    NaN                     NaN                   NaN   \n",
       "737763                    NaN                     NaN                   NaN   \n",
       "741704                    NaN                     NaN                   NaN   \n",
       "773401                    NaN                     NaN                   NaN   \n",
       "\n",
       "         Kidney_Cortex  Liver  Lung  Minor_Salivary_Gland  Muscle_Skeletal  \\\n",
       "78185              NaN    NaN   NaN                   NaN              NaN   \n",
       "302444             NaN    NaN   NaN                   NaN              NaN   \n",
       "4234               NaN    NaN   NaN                   NaN              NaN   \n",
       "121279             0.0    0.0   1.0                   0.0              0.0   \n",
       "1455913            0.0    0.0   0.0                   0.0              0.0   \n",
       "737199             0.0    0.0   1.0                   0.0              0.0   \n",
       "1531582            NaN    NaN   NaN                   NaN              NaN   \n",
       "1342297            0.0    0.0   0.0                   0.0              0.0   \n",
       "1268358            NaN    NaN   NaN                   NaN              NaN   \n",
       "705859             NaN    NaN   NaN                   NaN              NaN   \n",
       "685458             NaN    NaN   NaN                   NaN              NaN   \n",
       "171355             1.0    0.0   1.0                   0.0              0.0   \n",
       "17156              NaN    NaN   NaN                   NaN              NaN   \n",
       "555035             NaN    NaN   NaN                   NaN              NaN   \n",
       "680581             0.0    0.0   0.0                   0.0              0.0   \n",
       "126381             NaN    NaN   NaN                   NaN              NaN   \n",
       "737762             NaN    NaN   NaN                   NaN              NaN   \n",
       "737763             NaN    NaN   NaN                   NaN              NaN   \n",
       "741704             NaN    NaN   NaN                   NaN              NaN   \n",
       "773401             NaN    NaN   NaN                   NaN              NaN   \n",
       "\n",
       "         Nerve_Tibial  Ovary  Pancreas  Pituitary  Prostate  \\\n",
       "78185             NaN    NaN       NaN        NaN       NaN   \n",
       "302444            NaN    NaN       NaN        NaN       NaN   \n",
       "4234              NaN    NaN       NaN        NaN       NaN   \n",
       "121279            0.0    0.0       0.0        0.0       0.0   \n",
       "1455913           1.0    0.0       0.0        0.0       0.0   \n",
       "737199            1.0    0.0       0.0        1.0       0.0   \n",
       "1531582           NaN    NaN       NaN        NaN       NaN   \n",
       "1342297           1.0    0.0       0.0        0.0       0.0   \n",
       "1268358           NaN    NaN       NaN        NaN       NaN   \n",
       "705859            NaN    NaN       NaN        NaN       NaN   \n",
       "685458            NaN    NaN       NaN        NaN       NaN   \n",
       "171355            0.0    0.0       0.0        0.0       0.0   \n",
       "17156             NaN    NaN       NaN        NaN       NaN   \n",
       "555035            NaN    NaN       NaN        NaN       NaN   \n",
       "680581            0.0    0.0       0.0        1.0       0.0   \n",
       "126381            NaN    NaN       NaN        NaN       NaN   \n",
       "737762            NaN    NaN       NaN        NaN       NaN   \n",
       "737763            NaN    NaN       NaN        NaN       NaN   \n",
       "741704            NaN    NaN       NaN        NaN       NaN   \n",
       "773401            NaN    NaN       NaN        NaN       NaN   \n",
       "\n",
       "         Skin_Not_Sun_Exposed_Suprapubic  Skin_Sun_Exposed_Lower_leg  \\\n",
       "78185                                NaN                         NaN   \n",
       "302444                               NaN                         NaN   \n",
       "4234                                 NaN                         NaN   \n",
       "121279                               0.0                         0.0   \n",
       "1455913                              0.0                         0.0   \n",
       "737199                               1.0                         1.0   \n",
       "1531582                              NaN                         NaN   \n",
       "1342297                              0.0                         0.0   \n",
       "1268358                              NaN                         NaN   \n",
       "705859                               NaN                         NaN   \n",
       "685458                               NaN                         NaN   \n",
       "171355                               0.0                         0.0   \n",
       "17156                                NaN                         NaN   \n",
       "555035                               NaN                         NaN   \n",
       "680581                               1.0                         1.0   \n",
       "126381                               NaN                         NaN   \n",
       "737762                               NaN                         NaN   \n",
       "737763                               NaN                         NaN   \n",
       "741704                               NaN                         NaN   \n",
       "773401                               NaN                         NaN   \n",
       "\n",
       "         Small_Intestine_Terminal_Ileum  Spleen  Stomach  Testis  Thyroid  \\\n",
       "78185                               NaN     NaN      NaN     NaN      NaN   \n",
       "302444                              NaN     NaN      NaN     NaN      NaN   \n",
       "4234                                NaN     NaN      NaN     NaN      NaN   \n",
       "121279                              0.0     0.0      0.0     0.0      0.0   \n",
       "1455913                             0.0     0.0      0.0     0.0      1.0   \n",
       "737199                              0.0     0.0      1.0     1.0      1.0   \n",
       "1531582                             NaN     NaN      NaN     NaN      NaN   \n",
       "1342297                             0.0     0.0      0.0     1.0      0.0   \n",
       "1268358                             NaN     NaN      NaN     NaN      NaN   \n",
       "705859                              NaN     NaN      NaN     NaN      NaN   \n",
       "685458                              NaN     NaN      NaN     NaN      NaN   \n",
       "171355                              0.0     0.0      1.0     1.0      0.0   \n",
       "17156                               NaN     NaN      NaN     NaN      NaN   \n",
       "555035                              NaN     NaN      NaN     NaN      NaN   \n",
       "680581                              0.0     0.0      0.0     0.0      0.0   \n",
       "126381                              NaN     NaN      NaN     NaN      NaN   \n",
       "737762                              NaN     NaN      NaN     NaN      NaN   \n",
       "737763                              NaN     NaN      NaN     NaN      NaN   \n",
       "741704                              NaN     NaN      NaN     NaN      NaN   \n",
       "773401                              NaN     NaN      NaN     NaN      NaN   \n",
       "\n",
       "         Uterus  Vagina  Whole_Blood  N_GTEx_tissues sQTL  \\\n",
       "78185       NaN     NaN          NaN             NaN   no   \n",
       "302444      NaN     NaN          NaN             NaN   no   \n",
       "4234        NaN     NaN          NaN             NaN   no   \n",
       "121279      0.0     0.0          0.0             1.0  yes   \n",
       "1455913     0.0     0.0          1.0             7.0  yes   \n",
       "737199      0.0     0.0          1.0            20.0  yes   \n",
       "1531582     NaN     NaN          NaN             NaN   no   \n",
       "1342297     0.0     0.0          0.0             2.0  yes   \n",
       "1268358     NaN     NaN          NaN             NaN   no   \n",
       "705859      NaN     NaN          NaN             NaN   no   \n",
       "685458      NaN     NaN          NaN             NaN   no   \n",
       "171355      0.0     0.0          0.0            10.0  yes   \n",
       "17156       NaN     NaN          NaN             NaN   no   \n",
       "555035      NaN     NaN          NaN             NaN   no   \n",
       "680581      0.0     0.0          0.0             8.0  yes   \n",
       "126381      NaN     NaN          NaN             NaN   no   \n",
       "737762      NaN     NaN          NaN             NaN   no   \n",
       "737763      NaN     NaN          NaN             NaN   no   \n",
       "741704      NaN     NaN          NaN             NaN   no   \n",
       "773401      NaN     NaN          NaN             NaN   no   \n",
       "\n",
       "           Vernot_allele_origin  Browning_allele_origin  \n",
       "78185          archaic-specific        archaic-specific  \n",
       "302444   low-confidence ancient  low-confidence ancient  \n",
       "4234           archaic-specific        archaic-specific  \n",
       "121279                  ancient                 ancient  \n",
       "1455913                 ancient                 ancient  \n",
       "737199                  ancient                 ancient  \n",
       "1531582        archaic-specific        archaic-specific  \n",
       "1342297            introgressed  low-confidence ancient  \n",
       "1268358                 ancient            introgressed  \n",
       "705859         archaic-specific        archaic-specific  \n",
       "685458         archaic-specific        archaic-specific  \n",
       "171355   low-confidence ancient  low-confidence ancient  \n",
       "17156          archaic-specific        archaic-specific  \n",
       "555035   low-confidence ancient  low-confidence ancient  \n",
       "680581   low-confidence ancient  low-confidence ancient  \n",
       "126381   low-confidence ancient  low-confidence ancient  \n",
       "737762         archaic-specific        archaic-specific  \n",
       "737763         archaic-specific        archaic-specific  \n",
       "741704         archaic-specific        archaic-specific  \n",
       "773401         archaic-specific        archaic-specific  "
      ]
     },
     "execution_count": 178,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_2_ag = data_2.sort_values(['ag_delta'], ascending = [False])\n",
    "data_2_ag.head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 179,
   "id": "17e2dd31",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref_allele</th>\n",
       "      <th>alt_allele</th>\n",
       "      <th>ancestral_allele</th>\n",
       "      <th>anc_dev</th>\n",
       "      <th>variant_type</th>\n",
       "      <th>altai_gt</th>\n",
       "      <th>chagyrskaya_gt</th>\n",
       "      <th>denisovan_gt</th>\n",
       "      <th>vindija_gt</th>\n",
       "      <th>altai_gt_boolean</th>\n",
       "      <th>chagyrskaya_gt_boolean</th>\n",
       "      <th>denisovan_gt_boolean</th>\n",
       "      <th>vindija_gt_boolean</th>\n",
       "      <th>distribution</th>\n",
       "      <th>present_in_1KG</th>\n",
       "      <th>1KG_allele_count</th>\n",
       "      <th>1KG_allele_number</th>\n",
       "      <th>1KG_allele_frequency</th>\n",
       "      <th>1KG_EAS_AF</th>\n",
       "      <th>1KG_EUR_AF</th>\n",
       "      <th>1KG_AFR_AF</th>\n",
       "      <th>1KG_AMR_AF</th>\n",
       "      <th>1KG_SAS_AF</th>\n",
       "      <th>1KG_non_ASW_AFR_AF</th>\n",
       "      <th>present_in_gnomAD</th>\n",
       "      <th>gnomAD_allele_count</th>\n",
       "      <th>gnomAD_allele_number</th>\n",
       "      <th>gnomAD_allele_frequency</th>\n",
       "      <th>Vernot_introgressed</th>\n",
       "      <th>Vernot_ancestral_allele</th>\n",
       "      <th>Vernot_derived_allele</th>\n",
       "      <th>Vernot_ancestral_derived_code</th>\n",
       "      <th>Vernot_AFA_AF</th>\n",
       "      <th>Vernot_AFR_AF</th>\n",
       "      <th>Vernot_AMR_AF</th>\n",
       "      <th>Vernot_EAS_AF</th>\n",
       "      <th>Vernot_EUR_AF</th>\n",
       "      <th>Vernot_PNG_AF</th>\n",
       "      <th>Vernot_SAS_AF</th>\n",
       "      <th>Vernot_Denisovan_base</th>\n",
       "      <th>Vernot_Neanderthal_base_1</th>\n",
       "      <th>Vernot_Neanderthal_base_2</th>\n",
       "      <th>Vernot_haplotype_tag</th>\n",
       "      <th>Vernot_introgressed_AF</th>\n",
       "      <th>Browning_introgressed</th>\n",
       "      <th>Browning_ref_alt</th>\n",
       "      <th>Browning_introgressed_AF</th>\n",
       "      <th>annotation</th>\n",
       "      <th>mis_oe</th>\n",
       "      <th>mis_z</th>\n",
       "      <th>lof_oe</th>\n",
       "      <th>lof_z</th>\n",
       "      <th>phyloP</th>\n",
       "      <th>ag_delta</th>\n",
       "      <th>al_delta</th>\n",
       "      <th>dg_delta</th>\n",
       "      <th>dl_delta</th>\n",
       "      <th>delta_max</th>\n",
       "      <th>ag_pos</th>\n",
       "      <th>al_pos</th>\n",
       "      <th>dg_pos</th>\n",
       "      <th>dl_pos</th>\n",
       "      <th>Adipose_Subcutaneous</th>\n",
       "      <th>Adipose_Visceral_Omentum</th>\n",
       "      <th>Adrenal_Gland</th>\n",
       "      <th>Artery_Aorta</th>\n",
       "      <th>Artery_Coronary</th>\n",
       "      <th>Artery_Tibial</th>\n",
       "      <th>Brain_Amygdala</th>\n",
       "      <th>Brain_Anterior_cingulate_cortex_BA24</th>\n",
       "      <th>Brain_Caudate_basal_ganglia</th>\n",
       "      <th>Brain_Cerebellar_Hemisphere</th>\n",
       "      <th>Brain_Cerebellum</th>\n",
       "      <th>Brain_Cortex</th>\n",
       "      <th>Brain_Frontal_Cortex_BA9</th>\n",
       "      <th>Brain_Hippocampus</th>\n",
       "      <th>Brain_Hypothalamus</th>\n",
       "      <th>Brain_Nucleus_accumbens_basal_ganglia</th>\n",
       "      <th>Brain_Putamen_basal_ganglia</th>\n",
       "      <th>Brain_Spinal_cord_cervical_c-1</th>\n",
       "      <th>Brain_Substantia_nigra</th>\n",
       "      <th>Breast_Mammary_Tissue</th>\n",
       "      <th>Cells_Cultured_fibroblasts</th>\n",
       "      <th>Cells_EBV-transformed_lymphocytes</th>\n",
       "      <th>Colon_Sigmoid</th>\n",
       "      <th>Colon_Transverse</th>\n",
       "      <th>Esophagus_Gastroesophageal_Junction</th>\n",
       "      <th>Esophagus_Mucosa</th>\n",
       "      <th>Esophagus_Muscularis</th>\n",
       "      <th>Heart_Atrial_Appendage</th>\n",
       "      <th>Heart_Left_Ventricle</th>\n",
       "      <th>Kidney_Cortex</th>\n",
       "      <th>Liver</th>\n",
       "      <th>Lung</th>\n",
       "      <th>Minor_Salivary_Gland</th>\n",
       "      <th>Muscle_Skeletal</th>\n",
       "      <th>Nerve_Tibial</th>\n",
       "      <th>Ovary</th>\n",
       "      <th>Pancreas</th>\n",
       "      <th>Pituitary</th>\n",
       "      <th>Prostate</th>\n",
       "      <th>Skin_Not_Sun_Exposed_Suprapubic</th>\n",
       "      <th>Skin_Sun_Exposed_Lower_leg</th>\n",
       "      <th>Small_Intestine_Terminal_Ileum</th>\n",
       "      <th>Spleen</th>\n",
       "      <th>Stomach</th>\n",
       "      <th>Testis</th>\n",
       "      <th>Thyroid</th>\n",
       "      <th>Uterus</th>\n",
       "      <th>Vagina</th>\n",
       "      <th>Whole_Blood</th>\n",
       "      <th>N_GTEx_tissues</th>\n",
       "      <th>sQTL</th>\n",
       "      <th>Vernot_allele_origin</th>\n",
       "      <th>Browning_allele_origin</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>607092</th>\n",
       "      <td>chr16</td>\n",
       "      <td>85105387</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>0/1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Vindija</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>KIAA0513</td>\n",
       "      <td>1.05040</td>\n",
       "      <td>-0.28542</td>\n",
       "      <td>0.241480</td>\n",
       "      <td>3.503700</td>\n",
       "      <td>0.533</td>\n",
       "      <td>0.86</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.00</td>\n",
       "      <td>6</td>\n",
       "      <td>2</td>\n",
       "      <td>-40</td>\n",
       "      <td>31</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1326583</th>\n",
       "      <td>chr6</td>\n",
       "      <td>160482533</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Other</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>IGF2R</td>\n",
       "      <td>0.84588</td>\n",
       "      <td>2.06550</td>\n",
       "      <td>0.141620</td>\n",
       "      <td>8.714900</td>\n",
       "      <td>0.533</td>\n",
       "      <td>0.02</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.00</td>\n",
       "      <td>19</td>\n",
       "      <td>2</td>\n",
       "      <td>44</td>\n",
       "      <td>17</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1073592</th>\n",
       "      <td>chr4</td>\n",
       "      <td>71339723</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>yes</td>\n",
       "      <td>1060.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.21</td>\n",
       "      <td>0.20</td>\n",
       "      <td>0.13</td>\n",
       "      <td>0.28</td>\n",
       "      <td>0.14</td>\n",
       "      <td>0.25</td>\n",
       "      <td>0.290448</td>\n",
       "      <td>yes</td>\n",
       "      <td>26632.0</td>\n",
       "      <td>142744.0</td>\n",
       "      <td>0.186572</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MUC7</td>\n",
       "      <td>1.06550</td>\n",
       "      <td>-0.32621</td>\n",
       "      <td>1.789600</td>\n",
       "      <td>-0.773480</td>\n",
       "      <td>0.563</td>\n",
       "      <td>0.86</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.00</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>5</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>ancient</td>\n",
       "      <td>ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1002905</th>\n",
       "      <td>chr3</td>\n",
       "      <td>14965507</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>ancestral</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>yes</td>\n",
       "      <td>4927.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.97</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.92</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.96</td>\n",
       "      <td>0.94</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>yes</td>\n",
       "      <td>134716.0</td>\n",
       "      <td>143286.0</td>\n",
       "      <td>0.940190</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FGD5</td>\n",
       "      <td>0.98381</td>\n",
       "      <td>0.16707</td>\n",
       "      <td>0.070985</td>\n",
       "      <td>6.462300</td>\n",
       "      <td>-0.684</td>\n",
       "      <td>0.49</td>\n",
       "      <td>0.99</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.99</td>\n",
       "      <td>8</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>27.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>ancient</td>\n",
       "      <td>ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>341118</th>\n",
       "      <td>chr12</td>\n",
       "      <td>27543028</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>yes</td>\n",
       "      <td>31.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.023392</td>\n",
       "      <td>yes</td>\n",
       "      <td>1197.0</td>\n",
       "      <td>143232.0</td>\n",
       "      <td>0.008357</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ARNTL2</td>\n",
       "      <td>0.93937</td>\n",
       "      <td>0.38887</td>\n",
       "      <td>0.604430</td>\n",
       "      <td>2.108500</td>\n",
       "      <td>0.655</td>\n",
       "      <td>0.79</td>\n",
       "      <td>0.99</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.99</td>\n",
       "      <td>13</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>42</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>791229</th>\n",
       "      <td>chr2</td>\n",
       "      <td>55200335</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Other</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>RTN4</td>\n",
       "      <td>1.13290</td>\n",
       "      <td>-1.11760</td>\n",
       "      <td>0.220240</td>\n",
       "      <td>4.354800</td>\n",
       "      <td>0.585</td>\n",
       "      <td>0.03</td>\n",
       "      <td>0.99</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.99</td>\n",
       "      <td>-9</td>\n",
       "      <td>-1</td>\n",
       "      <td>-9</td>\n",
       "      <td>-27</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1176567</th>\n",
       "      <td>chr5</td>\n",
       "      <td>82648943</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>ancestral</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>yes</td>\n",
       "      <td>1907.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.37</td>\n",
       "      <td>0.70</td>\n",
       "      <td>0.14</td>\n",
       "      <td>0.48</td>\n",
       "      <td>0.33</td>\n",
       "      <td>0.17</td>\n",
       "      <td>0.488304</td>\n",
       "      <td>yes</td>\n",
       "      <td>35999.0</td>\n",
       "      <td>142846.0</td>\n",
       "      <td>0.252013</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>XRCC4</td>\n",
       "      <td>0.93418</td>\n",
       "      <td>0.29835</td>\n",
       "      <td>0.675330</td>\n",
       "      <td>1.268200</td>\n",
       "      <td>-2.023</td>\n",
       "      <td>0.77</td>\n",
       "      <td>0.99</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.99</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>32</td>\n",
       "      <td>7</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>49.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>ancient</td>\n",
       "      <td>ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>670290</th>\n",
       "      <td>chr18</td>\n",
       "      <td>633297</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>Altai</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CLUL1</td>\n",
       "      <td>0.88393</td>\n",
       "      <td>0.63877</td>\n",
       "      <td>0.883220</td>\n",
       "      <td>0.488520</td>\n",
       "      <td>0.655</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.99</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.99</td>\n",
       "      <td>16</td>\n",
       "      <td>1</td>\n",
       "      <td>-47</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1074868</th>\n",
       "      <td>chr4</td>\n",
       "      <td>72994301</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Vindija</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NPFFR2</td>\n",
       "      <td>1.19560</td>\n",
       "      <td>-1.16550</td>\n",
       "      <td>0.710990</td>\n",
       "      <td>1.145200</td>\n",
       "      <td>0.557</td>\n",
       "      <td>0.32</td>\n",
       "      <td>0.99</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.99</td>\n",
       "      <td>17</td>\n",
       "      <td>1</td>\n",
       "      <td>17</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>272304</th>\n",
       "      <td>chr11</td>\n",
       "      <td>61725616</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>0/0</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>Altai</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BEST1</td>\n",
       "      <td>0.90296</td>\n",
       "      <td>0.61695</td>\n",
       "      <td>1.005900</td>\n",
       "      <td>-0.023276</td>\n",
       "      <td>0.164</td>\n",
       "      <td>0.30</td>\n",
       "      <td>0.99</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.99</td>\n",
       "      <td>45</td>\n",
       "      <td>2</td>\n",
       "      <td>-44</td>\n",
       "      <td>16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>272305</th>\n",
       "      <td>chr11</td>\n",
       "      <td>61725617</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>0/0</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>Altai</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BEST1</td>\n",
       "      <td>0.90296</td>\n",
       "      <td>0.61695</td>\n",
       "      <td>1.005900</td>\n",
       "      <td>-0.023276</td>\n",
       "      <td>0.195</td>\n",
       "      <td>0.29</td>\n",
       "      <td>0.99</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.99</td>\n",
       "      <td>24</td>\n",
       "      <td>1</td>\n",
       "      <td>-45</td>\n",
       "      <td>15</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>171531</th>\n",
       "      <td>chr10</td>\n",
       "      <td>45799950</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>ancestral</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Other</td>\n",
       "      <td>yes</td>\n",
       "      <td>835.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.16</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.13</td>\n",
       "      <td>0.45</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.03</td>\n",
       "      <td>0.468811</td>\n",
       "      <td>yes</td>\n",
       "      <td>27693.0</td>\n",
       "      <td>142972.0</td>\n",
       "      <td>0.193695</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>OR13A1</td>\n",
       "      <td>1.21950</td>\n",
       "      <td>-1.10120</td>\n",
       "      <td>0.687900</td>\n",
       "      <td>0.779730</td>\n",
       "      <td>-2.035</td>\n",
       "      <td>0.13</td>\n",
       "      <td>0.98</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.98</td>\n",
       "      <td>-11</td>\n",
       "      <td>-2</td>\n",
       "      <td>48</td>\n",
       "      <td>-10</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>ancient</td>\n",
       "      <td>ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>583261</th>\n",
       "      <td>chr16</td>\n",
       "      <td>56701294</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>ancestral</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>yes</td>\n",
       "      <td>1217.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.24</td>\n",
       "      <td>0.41</td>\n",
       "      <td>0.12</td>\n",
       "      <td>0.15</td>\n",
       "      <td>0.33</td>\n",
       "      <td>0.24</td>\n",
       "      <td>0.150097</td>\n",
       "      <td>yes</td>\n",
       "      <td>21394.0</td>\n",
       "      <td>143136.0</td>\n",
       "      <td>0.149466</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MT1G</td>\n",
       "      <td>1.38180</td>\n",
       "      <td>-0.79967</td>\n",
       "      <td>0.818810</td>\n",
       "      <td>0.262410</td>\n",
       "      <td>-1.760</td>\n",
       "      <td>0.08</td>\n",
       "      <td>0.98</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.98</td>\n",
       "      <td>-5</td>\n",
       "      <td>-2</td>\n",
       "      <td>-35</td>\n",
       "      <td>-5</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>46.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>ancient</td>\n",
       "      <td>ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1241798</th>\n",
       "      <td>chr6</td>\n",
       "      <td>14135338</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>Chagyrskaya</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CD83</td>\n",
       "      <td>0.91542</td>\n",
       "      <td>0.31570</td>\n",
       "      <td>0.508910</td>\n",
       "      <td>1.426400</td>\n",
       "      <td>0.655</td>\n",
       "      <td>0.22</td>\n",
       "      <td>0.98</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.98</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>19</td>\n",
       "      <td>12</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>790320</th>\n",
       "      <td>chr2</td>\n",
       "      <td>54586095</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C2orf73</td>\n",
       "      <td>1.14080</td>\n",
       "      <td>-0.57371</td>\n",
       "      <td>0.973580</td>\n",
       "      <td>0.078459</td>\n",
       "      <td>0.655</td>\n",
       "      <td>0.31</td>\n",
       "      <td>0.97</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.97</td>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>28</td>\n",
       "      <td>18</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78185</th>\n",
       "      <td>chr1</td>\n",
       "      <td>144923837</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>./.</td>\n",
       "      <td>0/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PDE4DIP</td>\n",
       "      <td>1.12420</td>\n",
       "      <td>-1.40300</td>\n",
       "      <td>0.680450</td>\n",
       "      <td>3.129400</td>\n",
       "      <td>-0.264</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.97</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.00</td>\n",
       "      <td>-2</td>\n",
       "      <td>-16</td>\n",
       "      <td>-16</td>\n",
       "      <td>-29</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>98475</th>\n",
       "      <td>chr1</td>\n",
       "      <td>179858444</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>yes</td>\n",
       "      <td>3238.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.64</td>\n",
       "      <td>0.78</td>\n",
       "      <td>0.61</td>\n",
       "      <td>0.46</td>\n",
       "      <td>0.69</td>\n",
       "      <td>0.71</td>\n",
       "      <td>0.458090</td>\n",
       "      <td>yes</td>\n",
       "      <td>84628.0</td>\n",
       "      <td>142954.0</td>\n",
       "      <td>0.591995</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TOR1AIP1</td>\n",
       "      <td>1.14220</td>\n",
       "      <td>-0.88308</td>\n",
       "      <td>0.618180</td>\n",
       "      <td>1.961600</td>\n",
       "      <td>0.557</td>\n",
       "      <td>0.15</td>\n",
       "      <td>0.97</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.97</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>-50</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>49.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>ancient</td>\n",
       "      <td>ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>724074</th>\n",
       "      <td>chr19</td>\n",
       "      <td>8191238</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>ancestral</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>yes</td>\n",
       "      <td>5052.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.99</td>\n",
       "      <td>0.98</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.98</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>yes</td>\n",
       "      <td>143051.0</td>\n",
       "      <td>143322.0</td>\n",
       "      <td>0.998109</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ELAVL1</td>\n",
       "      <td>0.28002</td>\n",
       "      <td>3.80700</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>3.399500</td>\n",
       "      <td>-0.524</td>\n",
       "      <td>0.44</td>\n",
       "      <td>0.97</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.97</td>\n",
       "      <td>-7</td>\n",
       "      <td>-1</td>\n",
       "      <td>-39</td>\n",
       "      <td>-7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>ancient</td>\n",
       "      <td>ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>724075</th>\n",
       "      <td>chr19</td>\n",
       "      <td>8191238</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>ancestral</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>yes</td>\n",
       "      <td>5052.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.99</td>\n",
       "      <td>0.98</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.98</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>yes</td>\n",
       "      <td>143051.0</td>\n",
       "      <td>143322.0</td>\n",
       "      <td>0.998109</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FBN3</td>\n",
       "      <td>0.93583</td>\n",
       "      <td>0.95840</td>\n",
       "      <td>0.623530</td>\n",
       "      <td>4.191300</td>\n",
       "      <td>-0.524</td>\n",
       "      <td>0.44</td>\n",
       "      <td>0.97</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.97</td>\n",
       "      <td>-7</td>\n",
       "      <td>-1</td>\n",
       "      <td>-39</td>\n",
       "      <td>-7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>ancient</td>\n",
       "      <td>ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1455913</th>\n",
       "      <td>chr8</td>\n",
       "      <td>8999187</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>ancestral</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>yes</td>\n",
       "      <td>2597.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.51</td>\n",
       "      <td>0.82</td>\n",
       "      <td>0.31</td>\n",
       "      <td>0.37</td>\n",
       "      <td>0.41</td>\n",
       "      <td>0.66</td>\n",
       "      <td>0.367446</td>\n",
       "      <td>yes</td>\n",
       "      <td>52466.0</td>\n",
       "      <td>142958.0</td>\n",
       "      <td>0.367003</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PPP1R3B</td>\n",
       "      <td>1.24760</td>\n",
       "      <td>-1.10860</td>\n",
       "      <td>0.383500</td>\n",
       "      <td>1.597800</td>\n",
       "      <td>-0.216</td>\n",
       "      <td>0.99</td>\n",
       "      <td>0.95</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.99</td>\n",
       "      <td>-1</td>\n",
       "      <td>-9</td>\n",
       "      <td>-45</td>\n",
       "      <td>-22</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>ancient</td>\n",
       "      <td>ancient</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         chrom        pos ref_allele alt_allele ancestral_allele    anc_dev  \\\n",
       "607092   chr16   85105387          A          T                A    derived   \n",
       "1326583   chr6  160482533          A          C                A    derived   \n",
       "1073592   chr4   71339723          G          A                G    derived   \n",
       "1002905   chr3   14965507          A          G                G  ancestral   \n",
       "341118   chr12   27543028          G          A                G    derived   \n",
       "791229    chr2   55200335          C          T                C    derived   \n",
       "1176567   chr5   82648943          G          A                A  ancestral   \n",
       "670290   chr18     633297          G          A                G    derived   \n",
       "1074868   chr4   72994301          G          T                G    derived   \n",
       "272304   chr11   61725616          A          C                A    derived   \n",
       "272305   chr11   61725617          G          C                G    derived   \n",
       "171531   chr10   45799950          T          G                G  ancestral   \n",
       "583261   chr16   56701294          T          A                A  ancestral   \n",
       "1241798   chr6   14135338          G          C                G    derived   \n",
       "790320    chr2   54586095          G          A                G    derived   \n",
       "78185     chr1  144923837          A          T              NaN    derived   \n",
       "98475     chr1  179858444          G          A                G    derived   \n",
       "724074   chr19    8191238          C          T                T  ancestral   \n",
       "724075   chr19    8191238          C          T                T  ancestral   \n",
       "1455913   chr8    8999187          G          C                C  ancestral   \n",
       "\n",
       "        variant_type altai_gt chagyrskaya_gt denisovan_gt vindija_gt  \\\n",
       "607092           snv      0/0            ./.          ./.        0/1   \n",
       "1326583          snv      0/1            0/0          0/1        0/0   \n",
       "1073592          snv      0/0            0/0          1/1        0/0   \n",
       "1002905          snv      1/1            1/1          1/1        1/1   \n",
       "341118           snv      0/0            0/0          1/1        0/0   \n",
       "791229           snv      0/1            0/0          0/1        0/0   \n",
       "1176567          snv      1/1            1/1          1/1        1/1   \n",
       "670290           snv      0/1            0/0          0/0        0/0   \n",
       "1074868          snv      0/0            0/0          0/0        0/1   \n",
       "272304           snv      0/1            ./.          ./.        0/0   \n",
       "272305           snv      0/1            ./.          ./.        0/0   \n",
       "171531           snv      0/1            0/0          1/1        0/1   \n",
       "583261           snv      1/1            1/1          1/1        1/1   \n",
       "1241798          snv      0/0            0/1          0/0        0/0   \n",
       "790320           snv      0/0            0/0          1/1        0/0   \n",
       "78185            snv      0/1            0/1          ./.        0/1   \n",
       "98475            snv      1/1            1/1          0/0        1/1   \n",
       "724074           snv      1/1            1/1          1/1        1/1   \n",
       "724075           snv      1/1            1/1          1/1        1/1   \n",
       "1455913          snv      1/1            1/1          1/1        1/1   \n",
       "\n",
       "         altai_gt_boolean  chagyrskaya_gt_boolean  denisovan_gt_boolean  \\\n",
       "607092              False                   False                 False   \n",
       "1326583              True                   False                  True   \n",
       "1073592             False                   False                  True   \n",
       "1002905              True                    True                  True   \n",
       "341118              False                   False                  True   \n",
       "791229               True                   False                  True   \n",
       "1176567              True                    True                  True   \n",
       "670290               True                   False                 False   \n",
       "1074868             False                   False                 False   \n",
       "272304               True                   False                 False   \n",
       "272305               True                   False                 False   \n",
       "171531               True                   False                  True   \n",
       "583261               True                    True                  True   \n",
       "1241798             False                    True                 False   \n",
       "790320              False                   False                  True   \n",
       "78185                True                    True                 False   \n",
       "98475                True                    True                 False   \n",
       "724074               True                    True                  True   \n",
       "724075               True                    True                  True   \n",
       "1455913              True                    True                  True   \n",
       "\n",
       "         vindija_gt_boolean distribution present_in_1KG  1KG_allele_count  \\\n",
       "607092                 True      Vindija             no               NaN   \n",
       "1326583               False        Other             no               NaN   \n",
       "1073592               False    Denisovan            yes            1060.0   \n",
       "1002905                True       Shared            yes            4927.0   \n",
       "341118                False    Denisovan            yes              31.0   \n",
       "791229                False        Other             no               NaN   \n",
       "1176567                True       Shared            yes            1907.0   \n",
       "670290                False        Altai             no               NaN   \n",
       "1074868                True      Vindija             no               NaN   \n",
       "272304                False        Altai             no               NaN   \n",
       "272305                False        Altai             no               NaN   \n",
       "171531                 True        Other            yes             835.0   \n",
       "583261                 True       Shared            yes            1217.0   \n",
       "1241798               False  Chagyrskaya             no               NaN   \n",
       "790320                False    Denisovan             no               NaN   \n",
       "78185                  True  Neanderthal             no               NaN   \n",
       "98475                  True  Neanderthal            yes            3238.0   \n",
       "724074                 True       Shared            yes            5052.0   \n",
       "724075                 True       Shared            yes            5052.0   \n",
       "1455913                True       Shared            yes            2597.0   \n",
       "\n",
       "         1KG_allele_number  1KG_allele_frequency  1KG_EAS_AF  1KG_EUR_AF  \\\n",
       "607092                 NaN                   NaN         NaN         NaN   \n",
       "1326583                NaN                   NaN         NaN         NaN   \n",
       "1073592             5096.0                  0.21        0.20        0.13   \n",
       "1002905             5096.0                  0.97        1.00        0.92   \n",
       "341118              5096.0                  0.01        0.00        0.00   \n",
       "791229                 NaN                   NaN         NaN         NaN   \n",
       "1176567             5096.0                  0.37        0.70        0.14   \n",
       "670290                 NaN                   NaN         NaN         NaN   \n",
       "1074868                NaN                   NaN         NaN         NaN   \n",
       "272304                 NaN                   NaN         NaN         NaN   \n",
       "272305                 NaN                   NaN         NaN         NaN   \n",
       "171531              5096.0                  0.16        0.00        0.13   \n",
       "583261              5096.0                  0.24        0.41        0.12   \n",
       "1241798                NaN                   NaN         NaN         NaN   \n",
       "790320                 NaN                   NaN         NaN         NaN   \n",
       "78185                  NaN                   NaN         NaN         NaN   \n",
       "98475               5096.0                  0.64        0.78        0.61   \n",
       "724074              5096.0                  0.99        0.98        1.00   \n",
       "724075              5096.0                  0.99        0.98        1.00   \n",
       "1455913             5096.0                  0.51        0.82        0.31   \n",
       "\n",
       "         1KG_AFR_AF  1KG_AMR_AF  1KG_SAS_AF  1KG_non_ASW_AFR_AF  \\\n",
       "607092          NaN         NaN         NaN                 NaN   \n",
       "1326583         NaN         NaN         NaN                 NaN   \n",
       "1073592        0.28        0.14        0.25            0.290448   \n",
       "1002905        1.00        0.96        0.94            1.000000   \n",
       "341118         0.02        0.00        0.00            0.023392   \n",
       "791229          NaN         NaN         NaN                 NaN   \n",
       "1176567        0.48        0.33        0.17            0.488304   \n",
       "670290          NaN         NaN         NaN                 NaN   \n",
       "1074868         NaN         NaN         NaN                 NaN   \n",
       "272304          NaN         NaN         NaN                 NaN   \n",
       "272305          NaN         NaN         NaN                 NaN   \n",
       "171531         0.45        0.10        0.03            0.468811   \n",
       "583261         0.15        0.33        0.24            0.150097   \n",
       "1241798         NaN         NaN         NaN                 NaN   \n",
       "790320          NaN         NaN         NaN                 NaN   \n",
       "78185           NaN         NaN         NaN                 NaN   \n",
       "98475          0.46        0.69        0.71            0.458090   \n",
       "724074         1.00        1.00        0.98            1.000000   \n",
       "724075         1.00        1.00        0.98            1.000000   \n",
       "1455913        0.37        0.41        0.66            0.367446   \n",
       "\n",
       "        present_in_gnomAD  gnomAD_allele_count  gnomAD_allele_number  \\\n",
       "607092                 no                  NaN                   NaN   \n",
       "1326583                no                  NaN                   NaN   \n",
       "1073592               yes              26632.0              142744.0   \n",
       "1002905               yes             134716.0              143286.0   \n",
       "341118                yes               1197.0              143232.0   \n",
       "791229                 no                  NaN                   NaN   \n",
       "1176567               yes              35999.0              142846.0   \n",
       "670290                 no                  NaN                   NaN   \n",
       "1074868                no                  NaN                   NaN   \n",
       "272304                 no                  NaN                   NaN   \n",
       "272305                 no                  NaN                   NaN   \n",
       "171531                yes              27693.0              142972.0   \n",
       "583261                yes              21394.0              143136.0   \n",
       "1241798                no                  NaN                   NaN   \n",
       "790320                 no                  NaN                   NaN   \n",
       "78185                  no                  NaN                   NaN   \n",
       "98475                 yes              84628.0              142954.0   \n",
       "724074                yes             143051.0              143322.0   \n",
       "724075                yes             143051.0              143322.0   \n",
       "1455913               yes              52466.0              142958.0   \n",
       "\n",
       "         gnomAD_allele_frequency Vernot_introgressed Vernot_ancestral_allele  \\\n",
       "607092                       NaN                  no                     NaN   \n",
       "1326583                      NaN                  no                     NaN   \n",
       "1073592                 0.186572                  no                     NaN   \n",
       "1002905                 0.940190                  no                     NaN   \n",
       "341118                  0.008357                  no                     NaN   \n",
       "791229                       NaN                  no                     NaN   \n",
       "1176567                 0.252013                  no                     NaN   \n",
       "670290                       NaN                  no                     NaN   \n",
       "1074868                      NaN                  no                     NaN   \n",
       "272304                       NaN                  no                     NaN   \n",
       "272305                       NaN                  no                     NaN   \n",
       "171531                  0.193695                  no                     NaN   \n",
       "583261                  0.149466                  no                     NaN   \n",
       "1241798                      NaN                  no                     NaN   \n",
       "790320                       NaN                  no                     NaN   \n",
       "78185                        NaN                  no                     NaN   \n",
       "98475                   0.591995                  no                     NaN   \n",
       "724074                  0.998109                  no                     NaN   \n",
       "724075                  0.998109                  no                     NaN   \n",
       "1455913                 0.367003                  no                     NaN   \n",
       "\n",
       "        Vernot_derived_allele  Vernot_ancestral_derived_code  Vernot_AFA_AF  \\\n",
       "607092                    NaN                            NaN            NaN   \n",
       "1326583                   NaN                            NaN            NaN   \n",
       "1073592                   NaN                            NaN            NaN   \n",
       "1002905                   NaN                            NaN            NaN   \n",
       "341118                    NaN                            NaN            NaN   \n",
       "791229                    NaN                            NaN            NaN   \n",
       "1176567                   NaN                            NaN            NaN   \n",
       "670290                    NaN                            NaN            NaN   \n",
       "1074868                   NaN                            NaN            NaN   \n",
       "272304                    NaN                            NaN            NaN   \n",
       "272305                    NaN                            NaN            NaN   \n",
       "171531                    NaN                            NaN            NaN   \n",
       "583261                    NaN                            NaN            NaN   \n",
       "1241798                   NaN                            NaN            NaN   \n",
       "790320                    NaN                            NaN            NaN   \n",
       "78185                     NaN                            NaN            NaN   \n",
       "98475                     NaN                            NaN            NaN   \n",
       "724074                    NaN                            NaN            NaN   \n",
       "724075                    NaN                            NaN            NaN   \n",
       "1455913                   NaN                            NaN            NaN   \n",
       "\n",
       "         Vernot_AFR_AF  Vernot_AMR_AF  Vernot_EAS_AF  Vernot_EUR_AF  \\\n",
       "607092             NaN            NaN            NaN            NaN   \n",
       "1326583            NaN            NaN            NaN            NaN   \n",
       "1073592            NaN            NaN            NaN            NaN   \n",
       "1002905            NaN            NaN            NaN            NaN   \n",
       "341118             NaN            NaN            NaN            NaN   \n",
       "791229             NaN            NaN            NaN            NaN   \n",
       "1176567            NaN            NaN            NaN            NaN   \n",
       "670290             NaN            NaN            NaN            NaN   \n",
       "1074868            NaN            NaN            NaN            NaN   \n",
       "272304             NaN            NaN            NaN            NaN   \n",
       "272305             NaN            NaN            NaN            NaN   \n",
       "171531             NaN            NaN            NaN            NaN   \n",
       "583261             NaN            NaN            NaN            NaN   \n",
       "1241798            NaN            NaN            NaN            NaN   \n",
       "790320             NaN            NaN            NaN            NaN   \n",
       "78185              NaN            NaN            NaN            NaN   \n",
       "98475              NaN            NaN            NaN            NaN   \n",
       "724074             NaN            NaN            NaN            NaN   \n",
       "724075             NaN            NaN            NaN            NaN   \n",
       "1455913            NaN            NaN            NaN            NaN   \n",
       "\n",
       "         Vernot_PNG_AF  Vernot_SAS_AF Vernot_Denisovan_base  \\\n",
       "607092             NaN            NaN                   NaN   \n",
       "1326583            NaN            NaN                   NaN   \n",
       "1073592            NaN            NaN                   NaN   \n",
       "1002905            NaN            NaN                   NaN   \n",
       "341118             NaN            NaN                   NaN   \n",
       "791229             NaN            NaN                   NaN   \n",
       "1176567            NaN            NaN                   NaN   \n",
       "670290             NaN            NaN                   NaN   \n",
       "1074868            NaN            NaN                   NaN   \n",
       "272304             NaN            NaN                   NaN   \n",
       "272305             NaN            NaN                   NaN   \n",
       "171531             NaN            NaN                   NaN   \n",
       "583261             NaN            NaN                   NaN   \n",
       "1241798            NaN            NaN                   NaN   \n",
       "790320             NaN            NaN                   NaN   \n",
       "78185              NaN            NaN                   NaN   \n",
       "98475              NaN            NaN                   NaN   \n",
       "724074             NaN            NaN                   NaN   \n",
       "724075             NaN            NaN                   NaN   \n",
       "1455913            NaN            NaN                   NaN   \n",
       "\n",
       "        Vernot_Neanderthal_base_1 Vernot_Neanderthal_base_2  \\\n",
       "607092                        NaN                       NaN   \n",
       "1326583                       NaN                       NaN   \n",
       "1073592                       NaN                       NaN   \n",
       "1002905                       NaN                       NaN   \n",
       "341118                        NaN                       NaN   \n",
       "791229                        NaN                       NaN   \n",
       "1176567                       NaN                       NaN   \n",
       "670290                        NaN                       NaN   \n",
       "1074868                       NaN                       NaN   \n",
       "272304                        NaN                       NaN   \n",
       "272305                        NaN                       NaN   \n",
       "171531                        NaN                       NaN   \n",
       "583261                        NaN                       NaN   \n",
       "1241798                       NaN                       NaN   \n",
       "790320                        NaN                       NaN   \n",
       "78185                         NaN                       NaN   \n",
       "98475                         NaN                       NaN   \n",
       "724074                        NaN                       NaN   \n",
       "724075                        NaN                       NaN   \n",
       "1455913                       NaN                       NaN   \n",
       "\n",
       "        Vernot_haplotype_tag  Vernot_introgressed_AF Browning_introgressed  \\\n",
       "607092                   NaN                     NaN                    no   \n",
       "1326583                  NaN                     NaN                    no   \n",
       "1073592                  NaN                     NaN                    no   \n",
       "1002905                  NaN                     NaN                    no   \n",
       "341118                   NaN                     NaN                    no   \n",
       "791229                   NaN                     NaN                    no   \n",
       "1176567                  NaN                     NaN                    no   \n",
       "670290                   NaN                     NaN                    no   \n",
       "1074868                  NaN                     NaN                    no   \n",
       "272304                   NaN                     NaN                    no   \n",
       "272305                   NaN                     NaN                    no   \n",
       "171531                   NaN                     NaN                    no   \n",
       "583261                   NaN                     NaN                    no   \n",
       "1241798                  NaN                     NaN                    no   \n",
       "790320                   NaN                     NaN                    no   \n",
       "78185                    NaN                     NaN                    no   \n",
       "98475                    NaN                     NaN                    no   \n",
       "724074                   NaN                     NaN                    no   \n",
       "724075                   NaN                     NaN                    no   \n",
       "1455913                  NaN                     NaN                    no   \n",
       "\n",
       "         Browning_ref_alt  Browning_introgressed_AF annotation   mis_oe  \\\n",
       "607092                NaN                       NaN   KIAA0513  1.05040   \n",
       "1326583               NaN                       NaN      IGF2R  0.84588   \n",
       "1073592               NaN                       NaN       MUC7  1.06550   \n",
       "1002905               NaN                       NaN       FGD5  0.98381   \n",
       "341118                NaN                       NaN     ARNTL2  0.93937   \n",
       "791229                NaN                       NaN       RTN4  1.13290   \n",
       "1176567               NaN                       NaN      XRCC4  0.93418   \n",
       "670290                NaN                       NaN      CLUL1  0.88393   \n",
       "1074868               NaN                       NaN     NPFFR2  1.19560   \n",
       "272304                NaN                       NaN      BEST1  0.90296   \n",
       "272305                NaN                       NaN      BEST1  0.90296   \n",
       "171531                NaN                       NaN     OR13A1  1.21950   \n",
       "583261                NaN                       NaN       MT1G  1.38180   \n",
       "1241798               NaN                       NaN       CD83  0.91542   \n",
       "790320                NaN                       NaN    C2orf73  1.14080   \n",
       "78185                 NaN                       NaN    PDE4DIP  1.12420   \n",
       "98475                 NaN                       NaN   TOR1AIP1  1.14220   \n",
       "724074                NaN                       NaN     ELAVL1  0.28002   \n",
       "724075                NaN                       NaN       FBN3  0.93583   \n",
       "1455913               NaN                       NaN    PPP1R3B  1.24760   \n",
       "\n",
       "           mis_z    lof_oe     lof_z  phyloP  ag_delta  al_delta  dg_delta  \\\n",
       "607092  -0.28542  0.241480  3.503700   0.533      0.86      1.00       0.0   \n",
       "1326583  2.06550  0.141620  8.714900   0.533      0.02      1.00       0.0   \n",
       "1073592 -0.32621  1.789600 -0.773480   0.563      0.86      1.00       0.0   \n",
       "1002905  0.16707  0.070985  6.462300  -0.684      0.49      0.99       0.0   \n",
       "341118   0.38887  0.604430  2.108500   0.655      0.79      0.99       0.0   \n",
       "791229  -1.11760  0.220240  4.354800   0.585      0.03      0.99       0.0   \n",
       "1176567  0.29835  0.675330  1.268200  -2.023      0.77      0.99       0.0   \n",
       "670290   0.63877  0.883220  0.488520   0.655      0.05      0.99       0.0   \n",
       "1074868 -1.16550  0.710990  1.145200   0.557      0.32      0.99       0.0   \n",
       "272304   0.61695  1.005900 -0.023276   0.164      0.30      0.99       0.0   \n",
       "272305   0.61695  1.005900 -0.023276   0.195      0.29      0.99       0.0   \n",
       "171531  -1.10120  0.687900  0.779730  -2.035      0.13      0.98       0.0   \n",
       "583261  -0.79967  0.818810  0.262410  -1.760      0.08      0.98       0.0   \n",
       "1241798  0.31570  0.508910  1.426400   0.655      0.22      0.98       0.0   \n",
       "790320  -0.57371  0.973580  0.078459   0.655      0.31      0.97       0.0   \n",
       "78185   -1.40300  0.680450  3.129400  -0.264      1.00      0.97       0.0   \n",
       "98475   -0.88308  0.618180  1.961600   0.557      0.15      0.97       0.0   \n",
       "724074   3.80700  0.000000  3.399500  -0.524      0.44      0.97       0.0   \n",
       "724075   0.95840  0.623530  4.191300  -0.524      0.44      0.97       0.0   \n",
       "1455913 -1.10860  0.383500  1.597800  -0.216      0.99      0.95       0.0   \n",
       "\n",
       "         dl_delta  delta_max  ag_pos  al_pos  dg_pos  dl_pos  \\\n",
       "607092        0.0       1.00       6       2     -40      31   \n",
       "1326583       0.0       1.00      19       2      44      17   \n",
       "1073592       0.0       1.00       2       1       4       5   \n",
       "1002905       0.0       0.99       8       2       0      15   \n",
       "341118        0.0       0.99      13       1       3      42   \n",
       "791229        0.0       0.99      -9      -1      -9     -27   \n",
       "1176567       0.0       0.99       7       1      32       7   \n",
       "670290        0.0       0.99      16       1     -47       1   \n",
       "1074868       0.0       0.99      17       1      17       0   \n",
       "272304        0.0       0.99      45       2     -44      16   \n",
       "272305        0.0       0.99      24       1     -45      15   \n",
       "171531        0.0       0.98     -11      -2      48     -10   \n",
       "583261        0.0       0.98      -5      -2     -35      -5   \n",
       "1241798       0.0       0.98       4       1      19      12   \n",
       "790320        0.0       0.97      31       1      28      18   \n",
       "78185         0.0       1.00      -2     -16     -16     -29   \n",
       "98475         0.0       0.97       4       1       2     -50   \n",
       "724074        0.0       0.97      -7      -1     -39      -7   \n",
       "724075        0.0       0.97      -7      -1     -39      -7   \n",
       "1455913       0.0       0.99      -1      -9     -45     -22   \n",
       "\n",
       "         Adipose_Subcutaneous  Adipose_Visceral_Omentum  Adrenal_Gland  \\\n",
       "607092                    NaN                       NaN            NaN   \n",
       "1326583                   NaN                       NaN            NaN   \n",
       "1073592                   0.0                       0.0            0.0   \n",
       "1002905                   1.0                       1.0            1.0   \n",
       "341118                    NaN                       NaN            NaN   \n",
       "791229                    NaN                       NaN            NaN   \n",
       "1176567                   1.0                       1.0            1.0   \n",
       "670290                    NaN                       NaN            NaN   \n",
       "1074868                   NaN                       NaN            NaN   \n",
       "272304                    NaN                       NaN            NaN   \n",
       "272305                    NaN                       NaN            NaN   \n",
       "171531                    0.0                       0.0            0.0   \n",
       "583261                    1.0                       1.0            1.0   \n",
       "1241798                   NaN                       NaN            NaN   \n",
       "790320                    NaN                       NaN            NaN   \n",
       "78185                     NaN                       NaN            NaN   \n",
       "98475                     1.0                       1.0            1.0   \n",
       "724074                    NaN                       NaN            NaN   \n",
       "724075                    NaN                       NaN            NaN   \n",
       "1455913                   0.0                       0.0            0.0   \n",
       "\n",
       "         Artery_Aorta  Artery_Coronary  Artery_Tibial  Brain_Amygdala  \\\n",
       "607092            NaN              NaN            NaN             NaN   \n",
       "1326583           NaN              NaN            NaN             NaN   \n",
       "1073592           0.0              0.0            0.0             0.0   \n",
       "1002905           1.0              1.0            0.0             0.0   \n",
       "341118            NaN              NaN            NaN             NaN   \n",
       "791229            NaN              NaN            NaN             NaN   \n",
       "1176567           1.0              1.0            1.0             1.0   \n",
       "670290            NaN              NaN            NaN             NaN   \n",
       "1074868           NaN              NaN            NaN             NaN   \n",
       "272304            NaN              NaN            NaN             NaN   \n",
       "272305            NaN              NaN            NaN             NaN   \n",
       "171531            0.0              0.0            0.0             0.0   \n",
       "583261            1.0              1.0            1.0             1.0   \n",
       "1241798           NaN              NaN            NaN             NaN   \n",
       "790320            NaN              NaN            NaN             NaN   \n",
       "78185             NaN              NaN            NaN             NaN   \n",
       "98475             1.0              1.0            1.0             1.0   \n",
       "724074            NaN              NaN            NaN             NaN   \n",
       "724075            NaN              NaN            NaN             NaN   \n",
       "1455913           0.0              0.0            1.0             0.0   \n",
       "\n",
       "         Brain_Anterior_cingulate_cortex_BA24  Brain_Caudate_basal_ganglia  \\\n",
       "607092                                    NaN                          NaN   \n",
       "1326583                                   NaN                          NaN   \n",
       "1073592                                   0.0                          0.0   \n",
       "1002905                                   0.0                          1.0   \n",
       "341118                                    NaN                          NaN   \n",
       "791229                                    NaN                          NaN   \n",
       "1176567                                   1.0                          1.0   \n",
       "670290                                    NaN                          NaN   \n",
       "1074868                                   NaN                          NaN   \n",
       "272304                                    NaN                          NaN   \n",
       "272305                                    NaN                          NaN   \n",
       "171531                                    0.0                          0.0   \n",
       "583261                                    1.0                          1.0   \n",
       "1241798                                   NaN                          NaN   \n",
       "790320                                    NaN                          NaN   \n",
       "78185                                     NaN                          NaN   \n",
       "98475                                     1.0                          1.0   \n",
       "724074                                    NaN                          NaN   \n",
       "724075                                    NaN                          NaN   \n",
       "1455913                                   0.0                          0.0   \n",
       "\n",
       "         Brain_Cerebellar_Hemisphere  Brain_Cerebellum  Brain_Cortex  \\\n",
       "607092                           NaN               NaN           NaN   \n",
       "1326583                          NaN               NaN           NaN   \n",
       "1073592                          0.0               0.0           0.0   \n",
       "1002905                          0.0               0.0           0.0   \n",
       "341118                           NaN               NaN           NaN   \n",
       "791229                           NaN               NaN           NaN   \n",
       "1176567                          1.0               1.0           1.0   \n",
       "670290                           NaN               NaN           NaN   \n",
       "1074868                          NaN               NaN           NaN   \n",
       "272304                           NaN               NaN           NaN   \n",
       "272305                           NaN               NaN           NaN   \n",
       "171531                           0.0               0.0           0.0   \n",
       "583261                           1.0               1.0           1.0   \n",
       "1241798                          NaN               NaN           NaN   \n",
       "790320                           NaN               NaN           NaN   \n",
       "78185                            NaN               NaN           NaN   \n",
       "98475                            1.0               1.0           1.0   \n",
       "724074                           NaN               NaN           NaN   \n",
       "724075                           NaN               NaN           NaN   \n",
       "1455913                          0.0               0.0           0.0   \n",
       "\n",
       "         Brain_Frontal_Cortex_BA9  Brain_Hippocampus  Brain_Hypothalamus  \\\n",
       "607092                        NaN                NaN                 NaN   \n",
       "1326583                       NaN                NaN                 NaN   \n",
       "1073592                       0.0                0.0                 0.0   \n",
       "1002905                       0.0                0.0                 0.0   \n",
       "341118                        NaN                NaN                 NaN   \n",
       "791229                        NaN                NaN                 NaN   \n",
       "1176567                       1.0                1.0                 1.0   \n",
       "670290                        NaN                NaN                 NaN   \n",
       "1074868                       NaN                NaN                 NaN   \n",
       "272304                        NaN                NaN                 NaN   \n",
       "272305                        NaN                NaN                 NaN   \n",
       "171531                        0.0                0.0                 0.0   \n",
       "583261                        1.0                1.0                 1.0   \n",
       "1241798                       NaN                NaN                 NaN   \n",
       "790320                        NaN                NaN                 NaN   \n",
       "78185                         NaN                NaN                 NaN   \n",
       "98475                         1.0                1.0                 1.0   \n",
       "724074                        NaN                NaN                 NaN   \n",
       "724075                        NaN                NaN                 NaN   \n",
       "1455913                       0.0                0.0                 0.0   \n",
       "\n",
       "         Brain_Nucleus_accumbens_basal_ganglia  Brain_Putamen_basal_ganglia  \\\n",
       "607092                                     NaN                          NaN   \n",
       "1326583                                    NaN                          NaN   \n",
       "1073592                                    0.0                          0.0   \n",
       "1002905                                    0.0                          0.0   \n",
       "341118                                     NaN                          NaN   \n",
       "791229                                     NaN                          NaN   \n",
       "1176567                                    1.0                          1.0   \n",
       "670290                                     NaN                          NaN   \n",
       "1074868                                    NaN                          NaN   \n",
       "272304                                     NaN                          NaN   \n",
       "272305                                     NaN                          NaN   \n",
       "171531                                     0.0                          0.0   \n",
       "583261                                     1.0                          1.0   \n",
       "1241798                                    NaN                          NaN   \n",
       "790320                                     NaN                          NaN   \n",
       "78185                                      NaN                          NaN   \n",
       "98475                                      1.0                          1.0   \n",
       "724074                                     NaN                          NaN   \n",
       "724075                                     NaN                          NaN   \n",
       "1455913                                    0.0                          0.0   \n",
       "\n",
       "         Brain_Spinal_cord_cervical_c-1  Brain_Substantia_nigra  \\\n",
       "607092                              NaN                     NaN   \n",
       "1326583                             NaN                     NaN   \n",
       "1073592                             0.0                     0.0   \n",
       "1002905                             0.0                     0.0   \n",
       "341118                              NaN                     NaN   \n",
       "791229                              NaN                     NaN   \n",
       "1176567                             1.0                     1.0   \n",
       "670290                              NaN                     NaN   \n",
       "1074868                             NaN                     NaN   \n",
       "272304                              NaN                     NaN   \n",
       "272305                              NaN                     NaN   \n",
       "171531                              0.0                     0.0   \n",
       "583261                              1.0                     1.0   \n",
       "1241798                             NaN                     NaN   \n",
       "790320                              NaN                     NaN   \n",
       "78185                               NaN                     NaN   \n",
       "98475                               1.0                     1.0   \n",
       "724074                              NaN                     NaN   \n",
       "724075                              NaN                     NaN   \n",
       "1455913                             0.0                     0.0   \n",
       "\n",
       "         Breast_Mammary_Tissue  Cells_Cultured_fibroblasts  \\\n",
       "607092                     NaN                         NaN   \n",
       "1326583                    NaN                         NaN   \n",
       "1073592                    0.0                         0.0   \n",
       "1002905                    1.0                         0.0   \n",
       "341118                     NaN                         NaN   \n",
       "791229                     NaN                         NaN   \n",
       "1176567                    1.0                         1.0   \n",
       "670290                     NaN                         NaN   \n",
       "1074868                    NaN                         NaN   \n",
       "272304                     NaN                         NaN   \n",
       "272305                     NaN                         NaN   \n",
       "171531                     0.0                         0.0   \n",
       "583261                     1.0                         1.0   \n",
       "1241798                    NaN                         NaN   \n",
       "790320                     NaN                         NaN   \n",
       "78185                      NaN                         NaN   \n",
       "98475                      1.0                         1.0   \n",
       "724074                     NaN                         NaN   \n",
       "724075                     NaN                         NaN   \n",
       "1455913                    1.0                         1.0   \n",
       "\n",
       "         Cells_EBV-transformed_lymphocytes  Colon_Sigmoid  Colon_Transverse  \\\n",
       "607092                                 NaN            NaN               NaN   \n",
       "1326583                                NaN            NaN               NaN   \n",
       "1073592                                0.0            0.0               0.0   \n",
       "1002905                                0.0            1.0               1.0   \n",
       "341118                                 NaN            NaN               NaN   \n",
       "791229                                 NaN            NaN               NaN   \n",
       "1176567                                1.0            1.0               1.0   \n",
       "670290                                 NaN            NaN               NaN   \n",
       "1074868                                NaN            NaN               NaN   \n",
       "272304                                 NaN            NaN               NaN   \n",
       "272305                                 NaN            NaN               NaN   \n",
       "171531                                 0.0            0.0               0.0   \n",
       "583261                                 0.0            1.0               1.0   \n",
       "1241798                                NaN            NaN               NaN   \n",
       "790320                                 NaN            NaN               NaN   \n",
       "78185                                  NaN            NaN               NaN   \n",
       "98475                                  1.0            1.0               1.0   \n",
       "724074                                 NaN            NaN               NaN   \n",
       "724075                                 NaN            NaN               NaN   \n",
       "1455913                                0.0            1.0               0.0   \n",
       "\n",
       "         Esophagus_Gastroesophageal_Junction  Esophagus_Mucosa  \\\n",
       "607092                                   NaN               NaN   \n",
       "1326583                                  NaN               NaN   \n",
       "1073592                                  0.0               0.0   \n",
       "1002905                                  1.0               1.0   \n",
       "341118                                   NaN               NaN   \n",
       "791229                                   NaN               NaN   \n",
       "1176567                                  1.0               1.0   \n",
       "670290                                   NaN               NaN   \n",
       "1074868                                  NaN               NaN   \n",
       "272304                                   NaN               NaN   \n",
       "272305                                   NaN               NaN   \n",
       "171531                                   0.0               0.0   \n",
       "583261                                   1.0               1.0   \n",
       "1241798                                  NaN               NaN   \n",
       "790320                                   NaN               NaN   \n",
       "78185                                    NaN               NaN   \n",
       "98475                                    1.0               1.0   \n",
       "724074                                   NaN               NaN   \n",
       "724075                                   NaN               NaN   \n",
       "1455913                                  0.0               0.0   \n",
       "\n",
       "         Esophagus_Muscularis  Heart_Atrial_Appendage  Heart_Left_Ventricle  \\\n",
       "607092                    NaN                     NaN                   NaN   \n",
       "1326583                   NaN                     NaN                   NaN   \n",
       "1073592                   0.0                     0.0                   0.0   \n",
       "1002905                   1.0                     1.0                   1.0   \n",
       "341118                    NaN                     NaN                   NaN   \n",
       "791229                    NaN                     NaN                   NaN   \n",
       "1176567                   1.0                     1.0                   1.0   \n",
       "670290                    NaN                     NaN                   NaN   \n",
       "1074868                   NaN                     NaN                   NaN   \n",
       "272304                    NaN                     NaN                   NaN   \n",
       "272305                    NaN                     NaN                   NaN   \n",
       "171531                    0.0                     0.0                   0.0   \n",
       "583261                    1.0                     1.0                   1.0   \n",
       "1241798                   NaN                     NaN                   NaN   \n",
       "790320                    NaN                     NaN                   NaN   \n",
       "78185                     NaN                     NaN                   NaN   \n",
       "98475                     1.0                     1.0                   1.0   \n",
       "724074                    NaN                     NaN                   NaN   \n",
       "724075                    NaN                     NaN                   NaN   \n",
       "1455913                   0.0                     0.0                   0.0   \n",
       "\n",
       "         Kidney_Cortex  Liver  Lung  Minor_Salivary_Gland  Muscle_Skeletal  \\\n",
       "607092             NaN    NaN   NaN                   NaN              NaN   \n",
       "1326583            NaN    NaN   NaN                   NaN              NaN   \n",
       "1073592            0.0    0.0   0.0                   1.0              0.0   \n",
       "1002905            1.0    0.0   1.0                   1.0              1.0   \n",
       "341118             NaN    NaN   NaN                   NaN              NaN   \n",
       "791229             NaN    NaN   NaN                   NaN              NaN   \n",
       "1176567            1.0    1.0   1.0                   1.0              1.0   \n",
       "670290             NaN    NaN   NaN                   NaN              NaN   \n",
       "1074868            NaN    NaN   NaN                   NaN              NaN   \n",
       "272304             NaN    NaN   NaN                   NaN              NaN   \n",
       "272305             NaN    NaN   NaN                   NaN              NaN   \n",
       "171531             0.0    0.0   0.0                   0.0              0.0   \n",
       "583261             1.0    1.0   1.0                   1.0              1.0   \n",
       "1241798            NaN    NaN   NaN                   NaN              NaN   \n",
       "790320             NaN    NaN   NaN                   NaN              NaN   \n",
       "78185              NaN    NaN   NaN                   NaN              NaN   \n",
       "98475              1.0    1.0   1.0                   1.0              1.0   \n",
       "724074             NaN    NaN   NaN                   NaN              NaN   \n",
       "724075             NaN    NaN   NaN                   NaN              NaN   \n",
       "1455913            0.0    0.0   0.0                   0.0              0.0   \n",
       "\n",
       "         Nerve_Tibial  Ovary  Pancreas  Pituitary  Prostate  \\\n",
       "607092            NaN    NaN       NaN        NaN       NaN   \n",
       "1326583           NaN    NaN       NaN        NaN       NaN   \n",
       "1073592           0.0    0.0       0.0        0.0       0.0   \n",
       "1002905           0.0    1.0       0.0        1.0       1.0   \n",
       "341118            NaN    NaN       NaN        NaN       NaN   \n",
       "791229            NaN    NaN       NaN        NaN       NaN   \n",
       "1176567           1.0    1.0       1.0        1.0       1.0   \n",
       "670290            NaN    NaN       NaN        NaN       NaN   \n",
       "1074868           NaN    NaN       NaN        NaN       NaN   \n",
       "272304            NaN    NaN       NaN        NaN       NaN   \n",
       "272305            NaN    NaN       NaN        NaN       NaN   \n",
       "171531            0.0    0.0       0.0        0.0       0.0   \n",
       "583261            1.0    0.0       1.0        1.0       1.0   \n",
       "1241798           NaN    NaN       NaN        NaN       NaN   \n",
       "790320            NaN    NaN       NaN        NaN       NaN   \n",
       "78185             NaN    NaN       NaN        NaN       NaN   \n",
       "98475             1.0    1.0       1.0        1.0       1.0   \n",
       "724074            NaN    NaN       NaN        NaN       NaN   \n",
       "724075            NaN    NaN       NaN        NaN       NaN   \n",
       "1455913           1.0    0.0       0.0        0.0       0.0   \n",
       "\n",
       "         Skin_Not_Sun_Exposed_Suprapubic  Skin_Sun_Exposed_Lower_leg  \\\n",
       "607092                               NaN                         NaN   \n",
       "1326583                              NaN                         NaN   \n",
       "1073592                              0.0                         0.0   \n",
       "1002905                              0.0                         0.0   \n",
       "341118                               NaN                         NaN   \n",
       "791229                               NaN                         NaN   \n",
       "1176567                              1.0                         1.0   \n",
       "670290                               NaN                         NaN   \n",
       "1074868                              NaN                         NaN   \n",
       "272304                               NaN                         NaN   \n",
       "272305                               NaN                         NaN   \n",
       "171531                               0.0                         0.0   \n",
       "583261                               1.0                         1.0   \n",
       "1241798                              NaN                         NaN   \n",
       "790320                               NaN                         NaN   \n",
       "78185                                NaN                         NaN   \n",
       "98475                                1.0                         1.0   \n",
       "724074                               NaN                         NaN   \n",
       "724075                               NaN                         NaN   \n",
       "1455913                              0.0                         0.0   \n",
       "\n",
       "         Small_Intestine_Terminal_Ileum  Spleen  Stomach  Testis  Thyroid  \\\n",
       "607092                              NaN     NaN      NaN     NaN      NaN   \n",
       "1326583                             NaN     NaN      NaN     NaN      NaN   \n",
       "1073592                             0.0     0.0      0.0     1.0      0.0   \n",
       "1002905                             1.0     0.0      1.0     1.0      1.0   \n",
       "341118                              NaN     NaN      NaN     NaN      NaN   \n",
       "791229                              NaN     NaN      NaN     NaN      NaN   \n",
       "1176567                             1.0     1.0      1.0     1.0      1.0   \n",
       "670290                              NaN     NaN      NaN     NaN      NaN   \n",
       "1074868                             NaN     NaN      NaN     NaN      NaN   \n",
       "272304                              NaN     NaN      NaN     NaN      NaN   \n",
       "272305                              NaN     NaN      NaN     NaN      NaN   \n",
       "171531                              0.0     0.0      0.0     1.0      0.0   \n",
       "583261                              1.0     1.0      1.0     1.0      1.0   \n",
       "1241798                             NaN     NaN      NaN     NaN      NaN   \n",
       "790320                              NaN     NaN      NaN     NaN      NaN   \n",
       "78185                               NaN     NaN      NaN     NaN      NaN   \n",
       "98475                               1.0     1.0      1.0     1.0      1.0   \n",
       "724074                              NaN     NaN      NaN     NaN      NaN   \n",
       "724075                              NaN     NaN      NaN     NaN      NaN   \n",
       "1455913                             0.0     0.0      0.0     0.0      1.0   \n",
       "\n",
       "         Uterus  Vagina  Whole_Blood  N_GTEx_tissues sQTL  \\\n",
       "607092      NaN     NaN          NaN             NaN   no   \n",
       "1326583     NaN     NaN          NaN             NaN   no   \n",
       "1073592     0.0     0.0          0.0             2.0  yes   \n",
       "1002905     1.0     1.0          0.0            27.0  yes   \n",
       "341118      NaN     NaN          NaN             NaN   no   \n",
       "791229      NaN     NaN          NaN             NaN   no   \n",
       "1176567     1.0     1.0          1.0            49.0  yes   \n",
       "670290      NaN     NaN          NaN             NaN   no   \n",
       "1074868     NaN     NaN          NaN             NaN   no   \n",
       "272304      NaN     NaN          NaN             NaN   no   \n",
       "272305      NaN     NaN          NaN             NaN   no   \n",
       "171531      0.0     0.0          0.0             1.0  yes   \n",
       "583261      0.0     1.0          1.0            46.0  yes   \n",
       "1241798     NaN     NaN          NaN             NaN   no   \n",
       "790320      NaN     NaN          NaN             NaN   no   \n",
       "78185       NaN     NaN          NaN             NaN   no   \n",
       "98475       1.0     1.0          1.0            49.0  yes   \n",
       "724074      NaN     NaN          NaN             NaN   no   \n",
       "724075      NaN     NaN          NaN             NaN   no   \n",
       "1455913     0.0     0.0          1.0             7.0  yes   \n",
       "\n",
       "           Vernot_allele_origin  Browning_allele_origin  \n",
       "607092         archaic-specific        archaic-specific  \n",
       "1326583        archaic-specific        archaic-specific  \n",
       "1073592                 ancient                 ancient  \n",
       "1002905                 ancient                 ancient  \n",
       "341118   low-confidence ancient  low-confidence ancient  \n",
       "791229         archaic-specific        archaic-specific  \n",
       "1176567                 ancient                 ancient  \n",
       "670290         archaic-specific        archaic-specific  \n",
       "1074868        archaic-specific        archaic-specific  \n",
       "272304         archaic-specific        archaic-specific  \n",
       "272305         archaic-specific        archaic-specific  \n",
       "171531                  ancient                 ancient  \n",
       "583261                  ancient                 ancient  \n",
       "1241798        archaic-specific        archaic-specific  \n",
       "790320         archaic-specific        archaic-specific  \n",
       "78185          archaic-specific        archaic-specific  \n",
       "98475                   ancient                 ancient  \n",
       "724074                  ancient                 ancient  \n",
       "724075                  ancient                 ancient  \n",
       "1455913                 ancient                 ancient  "
      ]
     },
     "execution_count": 179,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_2_al = data_2.sort_values(['al_delta'], ascending = [False])\n",
    "data_2_al.head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 180,
   "id": "22ad71c8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref_allele</th>\n",
       "      <th>alt_allele</th>\n",
       "      <th>ancestral_allele</th>\n",
       "      <th>anc_dev</th>\n",
       "      <th>variant_type</th>\n",
       "      <th>altai_gt</th>\n",
       "      <th>chagyrskaya_gt</th>\n",
       "      <th>denisovan_gt</th>\n",
       "      <th>vindija_gt</th>\n",
       "      <th>altai_gt_boolean</th>\n",
       "      <th>chagyrskaya_gt_boolean</th>\n",
       "      <th>denisovan_gt_boolean</th>\n",
       "      <th>vindija_gt_boolean</th>\n",
       "      <th>distribution</th>\n",
       "      <th>present_in_1KG</th>\n",
       "      <th>1KG_allele_count</th>\n",
       "      <th>1KG_allele_number</th>\n",
       "      <th>1KG_allele_frequency</th>\n",
       "      <th>1KG_EAS_AF</th>\n",
       "      <th>1KG_EUR_AF</th>\n",
       "      <th>1KG_AFR_AF</th>\n",
       "      <th>1KG_AMR_AF</th>\n",
       "      <th>1KG_SAS_AF</th>\n",
       "      <th>1KG_non_ASW_AFR_AF</th>\n",
       "      <th>present_in_gnomAD</th>\n",
       "      <th>gnomAD_allele_count</th>\n",
       "      <th>gnomAD_allele_number</th>\n",
       "      <th>gnomAD_allele_frequency</th>\n",
       "      <th>Vernot_introgressed</th>\n",
       "      <th>Vernot_ancestral_allele</th>\n",
       "      <th>Vernot_derived_allele</th>\n",
       "      <th>Vernot_ancestral_derived_code</th>\n",
       "      <th>Vernot_AFA_AF</th>\n",
       "      <th>Vernot_AFR_AF</th>\n",
       "      <th>Vernot_AMR_AF</th>\n",
       "      <th>Vernot_EAS_AF</th>\n",
       "      <th>Vernot_EUR_AF</th>\n",
       "      <th>Vernot_PNG_AF</th>\n",
       "      <th>Vernot_SAS_AF</th>\n",
       "      <th>Vernot_Denisovan_base</th>\n",
       "      <th>Vernot_Neanderthal_base_1</th>\n",
       "      <th>Vernot_Neanderthal_base_2</th>\n",
       "      <th>Vernot_haplotype_tag</th>\n",
       "      <th>Vernot_introgressed_AF</th>\n",
       "      <th>Browning_introgressed</th>\n",
       "      <th>Browning_ref_alt</th>\n",
       "      <th>Browning_introgressed_AF</th>\n",
       "      <th>annotation</th>\n",
       "      <th>mis_oe</th>\n",
       "      <th>mis_z</th>\n",
       "      <th>lof_oe</th>\n",
       "      <th>lof_z</th>\n",
       "      <th>phyloP</th>\n",
       "      <th>ag_delta</th>\n",
       "      <th>al_delta</th>\n",
       "      <th>dg_delta</th>\n",
       "      <th>dl_delta</th>\n",
       "      <th>delta_max</th>\n",
       "      <th>ag_pos</th>\n",
       "      <th>al_pos</th>\n",
       "      <th>dg_pos</th>\n",
       "      <th>dl_pos</th>\n",
       "      <th>Adipose_Subcutaneous</th>\n",
       "      <th>Adipose_Visceral_Omentum</th>\n",
       "      <th>Adrenal_Gland</th>\n",
       "      <th>Artery_Aorta</th>\n",
       "      <th>Artery_Coronary</th>\n",
       "      <th>Artery_Tibial</th>\n",
       "      <th>Brain_Amygdala</th>\n",
       "      <th>Brain_Anterior_cingulate_cortex_BA24</th>\n",
       "      <th>Brain_Caudate_basal_ganglia</th>\n",
       "      <th>Brain_Cerebellar_Hemisphere</th>\n",
       "      <th>Brain_Cerebellum</th>\n",
       "      <th>Brain_Cortex</th>\n",
       "      <th>Brain_Frontal_Cortex_BA9</th>\n",
       "      <th>Brain_Hippocampus</th>\n",
       "      <th>Brain_Hypothalamus</th>\n",
       "      <th>Brain_Nucleus_accumbens_basal_ganglia</th>\n",
       "      <th>Brain_Putamen_basal_ganglia</th>\n",
       "      <th>Brain_Spinal_cord_cervical_c-1</th>\n",
       "      <th>Brain_Substantia_nigra</th>\n",
       "      <th>Breast_Mammary_Tissue</th>\n",
       "      <th>Cells_Cultured_fibroblasts</th>\n",
       "      <th>Cells_EBV-transformed_lymphocytes</th>\n",
       "      <th>Colon_Sigmoid</th>\n",
       "      <th>Colon_Transverse</th>\n",
       "      <th>Esophagus_Gastroesophageal_Junction</th>\n",
       "      <th>Esophagus_Mucosa</th>\n",
       "      <th>Esophagus_Muscularis</th>\n",
       "      <th>Heart_Atrial_Appendage</th>\n",
       "      <th>Heart_Left_Ventricle</th>\n",
       "      <th>Kidney_Cortex</th>\n",
       "      <th>Liver</th>\n",
       "      <th>Lung</th>\n",
       "      <th>Minor_Salivary_Gland</th>\n",
       "      <th>Muscle_Skeletal</th>\n",
       "      <th>Nerve_Tibial</th>\n",
       "      <th>Ovary</th>\n",
       "      <th>Pancreas</th>\n",
       "      <th>Pituitary</th>\n",
       "      <th>Prostate</th>\n",
       "      <th>Skin_Not_Sun_Exposed_Suprapubic</th>\n",
       "      <th>Skin_Sun_Exposed_Lower_leg</th>\n",
       "      <th>Small_Intestine_Terminal_Ileum</th>\n",
       "      <th>Spleen</th>\n",
       "      <th>Stomach</th>\n",
       "      <th>Testis</th>\n",
       "      <th>Thyroid</th>\n",
       "      <th>Uterus</th>\n",
       "      <th>Vagina</th>\n",
       "      <th>Whole_Blood</th>\n",
       "      <th>N_GTEx_tissues</th>\n",
       "      <th>sQTL</th>\n",
       "      <th>Vernot_allele_origin</th>\n",
       "      <th>Browning_allele_origin</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>299707</th>\n",
       "      <td>chr11</td>\n",
       "      <td>104761100</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>ancestral</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CASP12</td>\n",
       "      <td>0.72169</td>\n",
       "      <td>1.259300</td>\n",
       "      <td>0.603180</td>\n",
       "      <td>1.33920</td>\n",
       "      <td>-0.282</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1.00</td>\n",
       "      <td>46</td>\n",
       "      <td>-42</td>\n",
       "      <td>1</td>\n",
       "      <td>-3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>642626</th>\n",
       "      <td>chr17</td>\n",
       "      <td>45485046</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>yes</td>\n",
       "      <td>760.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.15</td>\n",
       "      <td>0.33</td>\n",
       "      <td>0.08</td>\n",
       "      <td>0.07</td>\n",
       "      <td>0.11</td>\n",
       "      <td>0.17</td>\n",
       "      <td>0.058480</td>\n",
       "      <td>yes</td>\n",
       "      <td>11933.0</td>\n",
       "      <td>143208.0</td>\n",
       "      <td>0.083326</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>EFCAB13</td>\n",
       "      <td>0.87046</td>\n",
       "      <td>1.001500</td>\n",
       "      <td>1.068500</td>\n",
       "      <td>-0.41674</td>\n",
       "      <td>-0.364</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.99</td>\n",
       "      <td>0.16</td>\n",
       "      <td>0.99</td>\n",
       "      <td>-27</td>\n",
       "      <td>28</td>\n",
       "      <td>-1</td>\n",
       "      <td>30</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>ancient</td>\n",
       "      <td>ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>939053</th>\n",
       "      <td>chr20</td>\n",
       "      <td>61467445</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>yes</td>\n",
       "      <td>541.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.11</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.32</td>\n",
       "      <td>0.03</td>\n",
       "      <td>0.03</td>\n",
       "      <td>0.328460</td>\n",
       "      <td>yes</td>\n",
       "      <td>18606.0</td>\n",
       "      <td>143286.0</td>\n",
       "      <td>0.129852</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>COL9A3</td>\n",
       "      <td>1.05080</td>\n",
       "      <td>-0.382240</td>\n",
       "      <td>0.541080</td>\n",
       "      <td>3.00410</td>\n",
       "      <td>0.313</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.99</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.99</td>\n",
       "      <td>-39</td>\n",
       "      <td>-6</td>\n",
       "      <td>-2</td>\n",
       "      <td>-31</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>ancient</td>\n",
       "      <td>ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1376781</th>\n",
       "      <td>chr7</td>\n",
       "      <td>56088811</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PSPH</td>\n",
       "      <td>0.80477</td>\n",
       "      <td>0.777190</td>\n",
       "      <td>0.602230</td>\n",
       "      <td>1.16340</td>\n",
       "      <td>-1.431</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.99</td>\n",
       "      <td>0.62</td>\n",
       "      <td>0.99</td>\n",
       "      <td>1</td>\n",
       "      <td>26</td>\n",
       "      <td>1</td>\n",
       "      <td>-45</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19011</th>\n",
       "      <td>chr1</td>\n",
       "      <td>22174518</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>yes</td>\n",
       "      <td>176.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.03</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.11</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>yes</td>\n",
       "      <td>5691.0</td>\n",
       "      <td>143278.0</td>\n",
       "      <td>0.039720</td>\n",
       "      <td>yes</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.00637</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0245</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.04871</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.10838</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr1_22145179_22179036</td>\n",
       "      <td>0.036318</td>\n",
       "      <td>yes</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.03</td>\n",
       "      <td>HSPG2</td>\n",
       "      <td>0.93887</td>\n",
       "      <td>1.136700</td>\n",
       "      <td>0.341210</td>\n",
       "      <td>9.23020</td>\n",
       "      <td>0.655</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.98</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.98</td>\n",
       "      <td>-14</td>\n",
       "      <td>38</td>\n",
       "      <td>3</td>\n",
       "      <td>-49</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>introgressed</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1023894</th>\n",
       "      <td>chr3</td>\n",
       "      <td>46785355</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>ancestral</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>yes</td>\n",
       "      <td>1519.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.30</td>\n",
       "      <td>0.12</td>\n",
       "      <td>0.15</td>\n",
       "      <td>0.65</td>\n",
       "      <td>0.20</td>\n",
       "      <td>0.24</td>\n",
       "      <td>0.648148</td>\n",
       "      <td>yes</td>\n",
       "      <td>39029.0</td>\n",
       "      <td>142438.0</td>\n",
       "      <td>0.274007</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PRSS45</td>\n",
       "      <td>1.05520</td>\n",
       "      <td>-0.222030</td>\n",
       "      <td>0.846290</td>\n",
       "      <td>0.37926</td>\n",
       "      <td>-1.142</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.98</td>\n",
       "      <td>0.18</td>\n",
       "      <td>0.98</td>\n",
       "      <td>43</td>\n",
       "      <td>20</td>\n",
       "      <td>1</td>\n",
       "      <td>20</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>ancient</td>\n",
       "      <td>ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>615932</th>\n",
       "      <td>chr17</td>\n",
       "      <td>3588807</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>ancestral</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>yes</td>\n",
       "      <td>929.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.18</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.67</td>\n",
       "      <td>0.04</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.685185</td>\n",
       "      <td>yes</td>\n",
       "      <td>23096.0</td>\n",
       "      <td>143126.0</td>\n",
       "      <td>0.161368</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P2RX5</td>\n",
       "      <td>1.13210</td>\n",
       "      <td>-0.742140</td>\n",
       "      <td>0.879570</td>\n",
       "      <td>0.54528</td>\n",
       "      <td>-0.539</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.98</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.98</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>32</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>615931</th>\n",
       "      <td>chr17</td>\n",
       "      <td>3588807</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>ancestral</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>yes</td>\n",
       "      <td>929.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.18</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.67</td>\n",
       "      <td>0.04</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.685185</td>\n",
       "      <td>yes</td>\n",
       "      <td>23096.0</td>\n",
       "      <td>143126.0</td>\n",
       "      <td>0.161368</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P2RX5-TAX1BP3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.539</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.98</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.98</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>32</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>114787</th>\n",
       "      <td>chr1</td>\n",
       "      <td>210025062</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DIEXF</td>\n",
       "      <td>0.91990</td>\n",
       "      <td>0.575430</td>\n",
       "      <td>0.447770</td>\n",
       "      <td>3.05890</td>\n",
       "      <td>0.533</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.97</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.97</td>\n",
       "      <td>-5</td>\n",
       "      <td>7</td>\n",
       "      <td>-5</td>\n",
       "      <td>7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>635204</th>\n",
       "      <td>chr17</td>\n",
       "      <td>32690090</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>Altai</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CCL1</td>\n",
       "      <td>0.90113</td>\n",
       "      <td>0.251030</td>\n",
       "      <td>0.895940</td>\n",
       "      <td>0.17645</td>\n",
       "      <td>-0.902</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.97</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.97</td>\n",
       "      <td>50</td>\n",
       "      <td>-25</td>\n",
       "      <td>2</td>\n",
       "      <td>15</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>574423</th>\n",
       "      <td>chr16</td>\n",
       "      <td>23724753</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>ancestral</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>yes</td>\n",
       "      <td>3182.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.62</td>\n",
       "      <td>0.34</td>\n",
       "      <td>0.66</td>\n",
       "      <td>0.94</td>\n",
       "      <td>0.53</td>\n",
       "      <td>0.51</td>\n",
       "      <td>0.949318</td>\n",
       "      <td>yes</td>\n",
       "      <td>101345.0</td>\n",
       "      <td>142998.0</td>\n",
       "      <td>0.708716</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ERN2</td>\n",
       "      <td>1.03480</td>\n",
       "      <td>-0.294750</td>\n",
       "      <td>0.795300</td>\n",
       "      <td>1.37840</td>\n",
       "      <td>-1.390</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.97</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.97</td>\n",
       "      <td>-15</td>\n",
       "      <td>49</td>\n",
       "      <td>2</td>\n",
       "      <td>9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>ancient</td>\n",
       "      <td>ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>656165</th>\n",
       "      <td>chr17</td>\n",
       "      <td>67323155</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Other</td>\n",
       "      <td>yes</td>\n",
       "      <td>340.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.07</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.07</td>\n",
       "      <td>0.18</td>\n",
       "      <td>0.04</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.183236</td>\n",
       "      <td>yes</td>\n",
       "      <td>13231.0</td>\n",
       "      <td>143098.0</td>\n",
       "      <td>0.092461</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ABCA5</td>\n",
       "      <td>0.96225</td>\n",
       "      <td>0.374040</td>\n",
       "      <td>0.669050</td>\n",
       "      <td>2.78060</td>\n",
       "      <td>0.462</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.96</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.96</td>\n",
       "      <td>2</td>\n",
       "      <td>-29</td>\n",
       "      <td>2</td>\n",
       "      <td>38</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>ancient</td>\n",
       "      <td>ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1472211</th>\n",
       "      <td>chr8</td>\n",
       "      <td>25316661</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>./.</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Other</td>\n",
       "      <td>yes</td>\n",
       "      <td>2072.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.41</td>\n",
       "      <td>0.34</td>\n",
       "      <td>0.36</td>\n",
       "      <td>0.53</td>\n",
       "      <td>0.36</td>\n",
       "      <td>0.39</td>\n",
       "      <td>0.532164</td>\n",
       "      <td>yes</td>\n",
       "      <td>57377.0</td>\n",
       "      <td>142508.0</td>\n",
       "      <td>0.402623</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CDCA2</td>\n",
       "      <td>1.05260</td>\n",
       "      <td>-0.433550</td>\n",
       "      <td>0.722500</td>\n",
       "      <td>1.38640</td>\n",
       "      <td>0.484</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.96</td>\n",
       "      <td>0.19</td>\n",
       "      <td>0.96</td>\n",
       "      <td>-2</td>\n",
       "      <td>23</td>\n",
       "      <td>-2</td>\n",
       "      <td>32</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>ancient</td>\n",
       "      <td>ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>403424</th>\n",
       "      <td>chr13</td>\n",
       "      <td>25453420</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>yes</td>\n",
       "      <td>166.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.03</td>\n",
       "      <td>0.08</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.03</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.048733</td>\n",
       "      <td>yes</td>\n",
       "      <td>2312.0</td>\n",
       "      <td>143292.0</td>\n",
       "      <td>0.016135</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>RNF17</td>\n",
       "      <td>0.82604</td>\n",
       "      <td>1.772300</td>\n",
       "      <td>0.034290</td>\n",
       "      <td>8.37040</td>\n",
       "      <td>0.533</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.96</td>\n",
       "      <td>0.33</td>\n",
       "      <td>0.96</td>\n",
       "      <td>-1</td>\n",
       "      <td>13</td>\n",
       "      <td>-1</td>\n",
       "      <td>13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>554672</th>\n",
       "      <td>chr16</td>\n",
       "      <td>711712</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>yes</td>\n",
       "      <td>3549.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.70</td>\n",
       "      <td>0.98</td>\n",
       "      <td>0.37</td>\n",
       "      <td>0.88</td>\n",
       "      <td>0.60</td>\n",
       "      <td>0.56</td>\n",
       "      <td>0.902534</td>\n",
       "      <td>yes</td>\n",
       "      <td>77384.0</td>\n",
       "      <td>143170.0</td>\n",
       "      <td>0.540504</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>WDR90</td>\n",
       "      <td>1.07780</td>\n",
       "      <td>-0.925830</td>\n",
       "      <td>1.091700</td>\n",
       "      <td>-0.78467</td>\n",
       "      <td>0.561</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.96</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.96</td>\n",
       "      <td>-2</td>\n",
       "      <td>-38</td>\n",
       "      <td>-2</td>\n",
       "      <td>-29</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>43.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>ancient</td>\n",
       "      <td>ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47872</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65613729</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>ancestral</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>yes</td>\n",
       "      <td>2010.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.39</td>\n",
       "      <td>0.37</td>\n",
       "      <td>0.09</td>\n",
       "      <td>0.82</td>\n",
       "      <td>0.39</td>\n",
       "      <td>0.16</td>\n",
       "      <td>0.854776</td>\n",
       "      <td>yes</td>\n",
       "      <td>45021.0</td>\n",
       "      <td>141604.0</td>\n",
       "      <td>0.317936</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AK4</td>\n",
       "      <td>0.69409</td>\n",
       "      <td>1.230900</td>\n",
       "      <td>0.304940</td>\n",
       "      <td>2.02020</td>\n",
       "      <td>-0.677</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.95</td>\n",
       "      <td>0.79</td>\n",
       "      <td>0.95</td>\n",
       "      <td>-1</td>\n",
       "      <td>11</td>\n",
       "      <td>-1</td>\n",
       "      <td>11</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>ancient</td>\n",
       "      <td>ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>727597</th>\n",
       "      <td>chr19</td>\n",
       "      <td>12802164</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FBXW9</td>\n",
       "      <td>0.89706</td>\n",
       "      <td>0.621590</td>\n",
       "      <td>0.667920</td>\n",
       "      <td>1.55250</td>\n",
       "      <td>0.462</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.95</td>\n",
       "      <td>0.68</td>\n",
       "      <td>0.95</td>\n",
       "      <td>19</td>\n",
       "      <td>-29</td>\n",
       "      <td>2</td>\n",
       "      <td>-8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>610909</th>\n",
       "      <td>chr16</td>\n",
       "      <td>89520101</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ANKRD11</td>\n",
       "      <td>1.03900</td>\n",
       "      <td>-0.553630</td>\n",
       "      <td>0.046779</td>\n",
       "      <td>8.16800</td>\n",
       "      <td>0.313</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.95</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.95</td>\n",
       "      <td>14</td>\n",
       "      <td>-16</td>\n",
       "      <td>1</td>\n",
       "      <td>-16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1594464</th>\n",
       "      <td>chr9</td>\n",
       "      <td>127965372</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>Altai</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>RABEPK</td>\n",
       "      <td>0.99070</td>\n",
       "      <td>0.047745</td>\n",
       "      <td>0.928380</td>\n",
       "      <td>0.28401</td>\n",
       "      <td>0.477</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.95</td>\n",
       "      <td>0.06</td>\n",
       "      <td>0.95</td>\n",
       "      <td>-27</td>\n",
       "      <td>30</td>\n",
       "      <td>-2</td>\n",
       "      <td>3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1291956</th>\n",
       "      <td>chr6</td>\n",
       "      <td>105626514</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>1.0</td>\n",
       "      <td>142928.0</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>POPDC3</td>\n",
       "      <td>0.85007</td>\n",
       "      <td>0.666430</td>\n",
       "      <td>0.429120</td>\n",
       "      <td>1.80570</td>\n",
       "      <td>-0.360</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.94</td>\n",
       "      <td>0.07</td>\n",
       "      <td>0.94</td>\n",
       "      <td>47</td>\n",
       "      <td>-10</td>\n",
       "      <td>1</td>\n",
       "      <td>-10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         chrom        pos ref_allele alt_allele ancestral_allele    anc_dev  \\\n",
       "299707   chr11  104761100          T          C                C  ancestral   \n",
       "642626   chr17   45485046          A          G                -          .   \n",
       "939053   chr20   61467445          C          T                C    derived   \n",
       "1376781   chr7   56088811          T          C                T    derived   \n",
       "19011     chr1   22174518          G          T                G    derived   \n",
       "1023894   chr3   46785355          T          C                C  ancestral   \n",
       "615932   chr17    3588807          C          A                A  ancestral   \n",
       "615931   chr17    3588807          C          A                A  ancestral   \n",
       "114787    chr1  210025062          A          G                A    derived   \n",
       "635204   chr17   32690090          G          A                G    derived   \n",
       "574423   chr16   23724753          G          A                A  ancestral   \n",
       "656165   chr17   67323155          G          A                G    derived   \n",
       "1472211   chr8   25316661          G          T                G    derived   \n",
       "403424   chr13   25453420          A          G                A    derived   \n",
       "554672   chr16     711712          C          T                C    derived   \n",
       "47872     chr1   65613729          T          G                G  ancestral   \n",
       "727597   chr19   12802164          G          A                G    derived   \n",
       "610909   chr16   89520101          G          C                G    derived   \n",
       "1594464   chr9  127965372          A          T                A    derived   \n",
       "1291956   chr6  105626514          T          C                T    derived   \n",
       "\n",
       "        variant_type altai_gt chagyrskaya_gt denisovan_gt vindija_gt  \\\n",
       "299707           snv      1/1            1/1          1/1        1/1   \n",
       "642626           snv      1/1            1/1          1/1        1/1   \n",
       "939053           snv      1/1            1/1          0/0        1/1   \n",
       "1376781          snv      0/1            0/1          0/1        0/1   \n",
       "19011            snv      1/1            1/1          0/0        1/1   \n",
       "1023894          snv      1/1            1/1          1/1        1/1   \n",
       "615932           snv      1/1            1/1          1/1        1/1   \n",
       "615931           snv      1/1            1/1          1/1        1/1   \n",
       "114787           snv      0/0            0/0          1/1        0/0   \n",
       "635204           snv      1/1            0/0          0/0        0/0   \n",
       "574423           snv      1/1            1/1          1/1        1/1   \n",
       "656165           snv      1/1            ./.          ./.        1/1   \n",
       "1472211          snv      1/1            ./.          1/1        1/1   \n",
       "403424           snv      1/1            1/1          0/0        0/1   \n",
       "554672           snv      1/1            1/1          1/1        1/1   \n",
       "47872            snv      1/1            1/1          1/1        1/1   \n",
       "727597           snv      0/0            0/0          0/1        0/0   \n",
       "610909           snv      1/1            1/1          1/1        1/1   \n",
       "1594464          snv      1/1            0/0          0/0        0/0   \n",
       "1291956          snv      0/0            0/0          1/1        0/0   \n",
       "\n",
       "         altai_gt_boolean  chagyrskaya_gt_boolean  denisovan_gt_boolean  \\\n",
       "299707               True                    True                  True   \n",
       "642626               True                    True                  True   \n",
       "939053               True                    True                 False   \n",
       "1376781              True                    True                  True   \n",
       "19011                True                    True                 False   \n",
       "1023894              True                    True                  True   \n",
       "615932               True                    True                  True   \n",
       "615931               True                    True                  True   \n",
       "114787              False                   False                  True   \n",
       "635204               True                   False                 False   \n",
       "574423               True                    True                  True   \n",
       "656165               True                   False                 False   \n",
       "1472211              True                   False                  True   \n",
       "403424               True                    True                 False   \n",
       "554672               True                    True                  True   \n",
       "47872                True                    True                  True   \n",
       "727597              False                   False                  True   \n",
       "610909               True                    True                  True   \n",
       "1594464              True                   False                 False   \n",
       "1291956             False                   False                  True   \n",
       "\n",
       "         vindija_gt_boolean distribution present_in_1KG  1KG_allele_count  \\\n",
       "299707                 True       Shared             no               NaN   \n",
       "642626                 True       Shared            yes             760.0   \n",
       "939053                 True  Neanderthal            yes             541.0   \n",
       "1376781                True       Shared             no               NaN   \n",
       "19011                  True  Neanderthal            yes             176.0   \n",
       "1023894                True       Shared            yes            1519.0   \n",
       "615932                 True       Shared            yes             929.0   \n",
       "615931                 True       Shared            yes             929.0   \n",
       "114787                False    Denisovan             no               NaN   \n",
       "635204                False        Altai             no               NaN   \n",
       "574423                 True       Shared            yes            3182.0   \n",
       "656165                 True        Other            yes             340.0   \n",
       "1472211                True        Other            yes            2072.0   \n",
       "403424                 True  Neanderthal            yes             166.0   \n",
       "554672                 True       Shared            yes            3549.0   \n",
       "47872                  True       Shared            yes            2010.0   \n",
       "727597                False    Denisovan             no               NaN   \n",
       "610909                 True       Shared             no               NaN   \n",
       "1594464               False        Altai             no               NaN   \n",
       "1291956               False    Denisovan             no               NaN   \n",
       "\n",
       "         1KG_allele_number  1KG_allele_frequency  1KG_EAS_AF  1KG_EUR_AF  \\\n",
       "299707                 NaN                   NaN         NaN         NaN   \n",
       "642626              5096.0                  0.15        0.33        0.08   \n",
       "939053              5096.0                  0.11        0.01        0.05   \n",
       "1376781                NaN                   NaN         NaN         NaN   \n",
       "19011               5096.0                  0.03        0.00        0.05   \n",
       "1023894             5096.0                  0.30        0.12        0.15   \n",
       "615932              5096.0                  0.18        0.00        0.00   \n",
       "615931              5096.0                  0.18        0.00        0.00   \n",
       "114787                 NaN                   NaN         NaN         NaN   \n",
       "635204                 NaN                   NaN         NaN         NaN   \n",
       "574423              5096.0                  0.62        0.34        0.66   \n",
       "656165              5096.0                  0.07        0.00        0.07   \n",
       "1472211             5096.0                  0.41        0.34        0.36   \n",
       "403424              5096.0                  0.03        0.08        0.00   \n",
       "554672              5096.0                  0.70        0.98        0.37   \n",
       "47872               5096.0                  0.39        0.37        0.09   \n",
       "727597                 NaN                   NaN         NaN         NaN   \n",
       "610909                 NaN                   NaN         NaN         NaN   \n",
       "1594464                NaN                   NaN         NaN         NaN   \n",
       "1291956                NaN                   NaN         NaN         NaN   \n",
       "\n",
       "         1KG_AFR_AF  1KG_AMR_AF  1KG_SAS_AF  1KG_non_ASW_AFR_AF  \\\n",
       "299707          NaN         NaN         NaN                 NaN   \n",
       "642626         0.07        0.11        0.17            0.058480   \n",
       "939053         0.32        0.03        0.03            0.328460   \n",
       "1376781         NaN         NaN         NaN                 NaN   \n",
       "19011          0.00        0.02        0.11            0.000000   \n",
       "1023894        0.65        0.20        0.24            0.648148   \n",
       "615932         0.67        0.04        0.00            0.685185   \n",
       "615931         0.67        0.04        0.00            0.685185   \n",
       "114787          NaN         NaN         NaN                 NaN   \n",
       "635204          NaN         NaN         NaN                 NaN   \n",
       "574423         0.94        0.53        0.51            0.949318   \n",
       "656165         0.18        0.04        0.01            0.183236   \n",
       "1472211        0.53        0.36        0.39            0.532164   \n",
       "403424         0.05        0.03        0.00            0.048733   \n",
       "554672         0.88        0.60        0.56            0.902534   \n",
       "47872          0.82        0.39        0.16            0.854776   \n",
       "727597          NaN         NaN         NaN                 NaN   \n",
       "610909          NaN         NaN         NaN                 NaN   \n",
       "1594464         NaN         NaN         NaN                 NaN   \n",
       "1291956         NaN         NaN         NaN                 NaN   \n",
       "\n",
       "        present_in_gnomAD  gnomAD_allele_count  gnomAD_allele_number  \\\n",
       "299707                 no                  NaN                   NaN   \n",
       "642626                yes              11933.0              143208.0   \n",
       "939053                yes              18606.0              143286.0   \n",
       "1376781                no                  NaN                   NaN   \n",
       "19011                 yes               5691.0              143278.0   \n",
       "1023894               yes              39029.0              142438.0   \n",
       "615932                yes              23096.0              143126.0   \n",
       "615931                yes              23096.0              143126.0   \n",
       "114787                 no                  NaN                   NaN   \n",
       "635204                 no                  NaN                   NaN   \n",
       "574423                yes             101345.0              142998.0   \n",
       "656165                yes              13231.0              143098.0   \n",
       "1472211               yes              57377.0              142508.0   \n",
       "403424                yes               2312.0              143292.0   \n",
       "554672                yes              77384.0              143170.0   \n",
       "47872                 yes              45021.0              141604.0   \n",
       "727597                 no                  NaN                   NaN   \n",
       "610909                 no                  NaN                   NaN   \n",
       "1594464                no                  NaN                   NaN   \n",
       "1291956               yes                  1.0              142928.0   \n",
       "\n",
       "         gnomAD_allele_frequency Vernot_introgressed Vernot_ancestral_allele  \\\n",
       "299707                       NaN                  no                     NaN   \n",
       "642626                  0.083326                  no                     NaN   \n",
       "939053                  0.129852                  no                     NaN   \n",
       "1376781                      NaN                  no                     NaN   \n",
       "19011                   0.039720                 yes                       G   \n",
       "1023894                 0.274007                  no                     NaN   \n",
       "615932                  0.161368                  no                     NaN   \n",
       "615931                  0.161368                  no                     NaN   \n",
       "114787                       NaN                  no                     NaN   \n",
       "635204                       NaN                  no                     NaN   \n",
       "574423                  0.708716                  no                     NaN   \n",
       "656165                  0.092461                  no                     NaN   \n",
       "1472211                 0.402623                  no                     NaN   \n",
       "403424                  0.016135                  no                     NaN   \n",
       "554672                  0.540504                  no                     NaN   \n",
       "47872                   0.317936                  no                     NaN   \n",
       "727597                       NaN                  no                     NaN   \n",
       "610909                       NaN                  no                     NaN   \n",
       "1594464                      NaN                  no                     NaN   \n",
       "1291956                 0.000007                  no                     NaN   \n",
       "\n",
       "        Vernot_derived_allele  Vernot_ancestral_derived_code  Vernot_AFA_AF  \\\n",
       "299707                    NaN                            NaN            NaN   \n",
       "642626                    NaN                            NaN            NaN   \n",
       "939053                    NaN                            NaN            NaN   \n",
       "1376781                   NaN                            NaN            NaN   \n",
       "19011                       T                            1.0        0.00637   \n",
       "1023894                   NaN                            NaN            NaN   \n",
       "615932                    NaN                            NaN            NaN   \n",
       "615931                    NaN                            NaN            NaN   \n",
       "114787                    NaN                            NaN            NaN   \n",
       "635204                    NaN                            NaN            NaN   \n",
       "574423                    NaN                            NaN            NaN   \n",
       "656165                    NaN                            NaN            NaN   \n",
       "1472211                   NaN                            NaN            NaN   \n",
       "403424                    NaN                            NaN            NaN   \n",
       "554672                    NaN                            NaN            NaN   \n",
       "47872                     NaN                            NaN            NaN   \n",
       "727597                    NaN                            NaN            NaN   \n",
       "610909                    NaN                            NaN            NaN   \n",
       "1594464                   NaN                            NaN            NaN   \n",
       "1291956                   NaN                            NaN            NaN   \n",
       "\n",
       "         Vernot_AFR_AF  Vernot_AMR_AF  Vernot_EAS_AF  Vernot_EUR_AF  \\\n",
       "299707             NaN            NaN            NaN            NaN   \n",
       "642626             NaN            NaN            NaN            NaN   \n",
       "939053             NaN            NaN            NaN            NaN   \n",
       "1376781            NaN            NaN            NaN            NaN   \n",
       "19011              0.0         0.0245            0.0        0.04871   \n",
       "1023894            NaN            NaN            NaN            NaN   \n",
       "615932             NaN            NaN            NaN            NaN   \n",
       "615931             NaN            NaN            NaN            NaN   \n",
       "114787             NaN            NaN            NaN            NaN   \n",
       "635204             NaN            NaN            NaN            NaN   \n",
       "574423             NaN            NaN            NaN            NaN   \n",
       "656165             NaN            NaN            NaN            NaN   \n",
       "1472211            NaN            NaN            NaN            NaN   \n",
       "403424             NaN            NaN            NaN            NaN   \n",
       "554672             NaN            NaN            NaN            NaN   \n",
       "47872              NaN            NaN            NaN            NaN   \n",
       "727597             NaN            NaN            NaN            NaN   \n",
       "610909             NaN            NaN            NaN            NaN   \n",
       "1594464            NaN            NaN            NaN            NaN   \n",
       "1291956            NaN            NaN            NaN            NaN   \n",
       "\n",
       "         Vernot_PNG_AF  Vernot_SAS_AF Vernot_Denisovan_base  \\\n",
       "299707             NaN            NaN                   NaN   \n",
       "642626             NaN            NaN                   NaN   \n",
       "939053             NaN            NaN                   NaN   \n",
       "1376781            NaN            NaN                   NaN   \n",
       "19011              0.0        0.10838                     G   \n",
       "1023894            NaN            NaN                   NaN   \n",
       "615932             NaN            NaN                   NaN   \n",
       "615931             NaN            NaN                   NaN   \n",
       "114787             NaN            NaN                   NaN   \n",
       "635204             NaN            NaN                   NaN   \n",
       "574423             NaN            NaN                   NaN   \n",
       "656165             NaN            NaN                   NaN   \n",
       "1472211            NaN            NaN                   NaN   \n",
       "403424             NaN            NaN                   NaN   \n",
       "554672             NaN            NaN                   NaN   \n",
       "47872              NaN            NaN                   NaN   \n",
       "727597             NaN            NaN                   NaN   \n",
       "610909             NaN            NaN                   NaN   \n",
       "1594464            NaN            NaN                   NaN   \n",
       "1291956            NaN            NaN                   NaN   \n",
       "\n",
       "        Vernot_Neanderthal_base_1 Vernot_Neanderthal_base_2  \\\n",
       "299707                        NaN                       NaN   \n",
       "642626                        NaN                       NaN   \n",
       "939053                        NaN                       NaN   \n",
       "1376781                       NaN                       NaN   \n",
       "19011                           T                       NaN   \n",
       "1023894                       NaN                       NaN   \n",
       "615932                        NaN                       NaN   \n",
       "615931                        NaN                       NaN   \n",
       "114787                        NaN                       NaN   \n",
       "635204                        NaN                       NaN   \n",
       "574423                        NaN                       NaN   \n",
       "656165                        NaN                       NaN   \n",
       "1472211                       NaN                       NaN   \n",
       "403424                        NaN                       NaN   \n",
       "554672                        NaN                       NaN   \n",
       "47872                         NaN                       NaN   \n",
       "727597                        NaN                       NaN   \n",
       "610909                        NaN                       NaN   \n",
       "1594464                       NaN                       NaN   \n",
       "1291956                       NaN                       NaN   \n",
       "\n",
       "           Vernot_haplotype_tag  Vernot_introgressed_AF Browning_introgressed  \\\n",
       "299707                      NaN                     NaN                    no   \n",
       "642626                      NaN                     NaN                    no   \n",
       "939053                      NaN                     NaN                    no   \n",
       "1376781                     NaN                     NaN                    no   \n",
       "19011    chr1_22145179_22179036                0.036318                   yes   \n",
       "1023894                     NaN                     NaN                    no   \n",
       "615932                      NaN                     NaN                    no   \n",
       "615931                      NaN                     NaN                    no   \n",
       "114787                      NaN                     NaN                    no   \n",
       "635204                      NaN                     NaN                    no   \n",
       "574423                      NaN                     NaN                    no   \n",
       "656165                      NaN                     NaN                    no   \n",
       "1472211                     NaN                     NaN                    no   \n",
       "403424                      NaN                     NaN                    no   \n",
       "554672                      NaN                     NaN                    no   \n",
       "47872                       NaN                     NaN                    no   \n",
       "727597                      NaN                     NaN                    no   \n",
       "610909                      NaN                     NaN                    no   \n",
       "1594464                     NaN                     NaN                    no   \n",
       "1291956                     NaN                     NaN                    no   \n",
       "\n",
       "         Browning_ref_alt  Browning_introgressed_AF     annotation   mis_oe  \\\n",
       "299707                NaN                       NaN         CASP12  0.72169   \n",
       "642626                NaN                       NaN        EFCAB13  0.87046   \n",
       "939053                NaN                       NaN         COL9A3  1.05080   \n",
       "1376781               NaN                       NaN           PSPH  0.80477   \n",
       "19011                 1.0                      0.03          HSPG2  0.93887   \n",
       "1023894               NaN                       NaN         PRSS45  1.05520   \n",
       "615932                NaN                       NaN          P2RX5  1.13210   \n",
       "615931                NaN                       NaN  P2RX5-TAX1BP3      NaN   \n",
       "114787                NaN                       NaN          DIEXF  0.91990   \n",
       "635204                NaN                       NaN           CCL1  0.90113   \n",
       "574423                NaN                       NaN           ERN2  1.03480   \n",
       "656165                NaN                       NaN          ABCA5  0.96225   \n",
       "1472211               NaN                       NaN          CDCA2  1.05260   \n",
       "403424                NaN                       NaN          RNF17  0.82604   \n",
       "554672                NaN                       NaN          WDR90  1.07780   \n",
       "47872                 NaN                       NaN            AK4  0.69409   \n",
       "727597                NaN                       NaN          FBXW9  0.89706   \n",
       "610909                NaN                       NaN        ANKRD11  1.03900   \n",
       "1594464               NaN                       NaN         RABEPK  0.99070   \n",
       "1291956               NaN                       NaN         POPDC3  0.85007   \n",
       "\n",
       "            mis_z    lof_oe    lof_z  phyloP  ag_delta  al_delta  dg_delta  \\\n",
       "299707   1.259300  0.603180  1.33920  -0.282       0.0      0.00      1.00   \n",
       "642626   1.001500  1.068500 -0.41674  -0.364       0.0      0.00      0.99   \n",
       "939053  -0.382240  0.541080  3.00410   0.313       0.0      0.00      0.99   \n",
       "1376781  0.777190  0.602230  1.16340  -1.431       0.0      0.00      0.99   \n",
       "19011    1.136700  0.341210  9.23020   0.655       0.0      0.01      0.98   \n",
       "1023894 -0.222030  0.846290  0.37926  -1.142       0.0      0.00      0.98   \n",
       "615932  -0.742140  0.879570  0.54528  -0.539       0.0      0.00      0.98   \n",
       "615931        NaN       NaN      NaN  -0.539       0.0      0.00      0.98   \n",
       "114787   0.575430  0.447770  3.05890   0.533       0.0      0.00      0.97   \n",
       "635204   0.251030  0.895940  0.17645  -0.902       0.0      0.00      0.97   \n",
       "574423  -0.294750  0.795300  1.37840  -1.390       0.0      0.00      0.97   \n",
       "656165   0.374040  0.669050  2.78060   0.462       0.0      0.00      0.96   \n",
       "1472211 -0.433550  0.722500  1.38640   0.484       0.0      0.00      0.96   \n",
       "403424   1.772300  0.034290  8.37040   0.533       0.0      0.00      0.96   \n",
       "554672  -0.925830  1.091700 -0.78467   0.561       0.0      0.00      0.96   \n",
       "47872    1.230900  0.304940  2.02020  -0.677       0.0      0.00      0.95   \n",
       "727597   0.621590  0.667920  1.55250   0.462       0.0      0.00      0.95   \n",
       "610909  -0.553630  0.046779  8.16800   0.313       0.0      0.00      0.95   \n",
       "1594464  0.047745  0.928380  0.28401   0.477       0.0      0.00      0.95   \n",
       "1291956  0.666430  0.429120  1.80570  -0.360       0.0      0.00      0.94   \n",
       "\n",
       "         dl_delta  delta_max  ag_pos  al_pos  dg_pos  dl_pos  \\\n",
       "299707       0.00       1.00      46     -42       1      -3   \n",
       "642626       0.16       0.99     -27      28      -1      30   \n",
       "939053       0.10       0.99     -39      -6      -2     -31   \n",
       "1376781      0.62       0.99       1      26       1     -45   \n",
       "19011        0.00       0.98     -14      38       3     -49   \n",
       "1023894      0.18       0.98      43      20       1      20   \n",
       "615932       0.01       0.98       2       1       2      32   \n",
       "615931       0.01       0.98       2       1       2      32   \n",
       "114787       0.00       0.97      -5       7      -5       7   \n",
       "635204       0.05       0.97      50     -25       2      15   \n",
       "574423       0.01       0.97     -15      49       2       9   \n",
       "656165       0.02       0.96       2     -29       2      38   \n",
       "1472211      0.19       0.96      -2      23      -2      32   \n",
       "403424       0.33       0.96      -1      13      -1      13   \n",
       "554672       0.00       0.96      -2     -38      -2     -29   \n",
       "47872        0.79       0.95      -1      11      -1      11   \n",
       "727597       0.68       0.95      19     -29       2      -8   \n",
       "610909       0.00       0.95      14     -16       1     -16   \n",
       "1594464      0.06       0.95     -27      30      -2       3   \n",
       "1291956      0.07       0.94      47     -10       1     -10   \n",
       "\n",
       "         Adipose_Subcutaneous  Adipose_Visceral_Omentum  Adrenal_Gland  \\\n",
       "299707                    NaN                       NaN            NaN   \n",
       "642626                    0.0                       0.0            0.0   \n",
       "939053                    NaN                       NaN            NaN   \n",
       "1376781                   NaN                       NaN            NaN   \n",
       "19011                     1.0                       0.0            0.0   \n",
       "1023894                   0.0                       0.0            0.0   \n",
       "615932                    0.0                       0.0            0.0   \n",
       "615931                    0.0                       0.0            0.0   \n",
       "114787                    NaN                       NaN            NaN   \n",
       "635204                    NaN                       NaN            NaN   \n",
       "574423                    NaN                       NaN            NaN   \n",
       "656165                    NaN                       NaN            NaN   \n",
       "1472211                   1.0                       0.0            0.0   \n",
       "403424                    NaN                       NaN            NaN   \n",
       "554672                    1.0                       1.0            1.0   \n",
       "47872                     0.0                       0.0            0.0   \n",
       "727597                    NaN                       NaN            NaN   \n",
       "610909                    NaN                       NaN            NaN   \n",
       "1594464                   NaN                       NaN            NaN   \n",
       "1291956                   NaN                       NaN            NaN   \n",
       "\n",
       "         Artery_Aorta  Artery_Coronary  Artery_Tibial  Brain_Amygdala  \\\n",
       "299707            NaN              NaN            NaN             NaN   \n",
       "642626            0.0              1.0            1.0             0.0   \n",
       "939053            NaN              NaN            NaN             NaN   \n",
       "1376781           NaN              NaN            NaN             NaN   \n",
       "19011             0.0              0.0            1.0             0.0   \n",
       "1023894           0.0              0.0            1.0             0.0   \n",
       "615932            0.0              0.0            0.0             0.0   \n",
       "615931            0.0              0.0            0.0             0.0   \n",
       "114787            NaN              NaN            NaN             NaN   \n",
       "635204            NaN              NaN            NaN             NaN   \n",
       "574423            NaN              NaN            NaN             NaN   \n",
       "656165            NaN              NaN            NaN             NaN   \n",
       "1472211           0.0              0.0            0.0             0.0   \n",
       "403424            NaN              NaN            NaN             NaN   \n",
       "554672            1.0              1.0            1.0             1.0   \n",
       "47872             0.0              0.0            0.0             0.0   \n",
       "727597            NaN              NaN            NaN             NaN   \n",
       "610909            NaN              NaN            NaN             NaN   \n",
       "1594464           NaN              NaN            NaN             NaN   \n",
       "1291956           NaN              NaN            NaN             NaN   \n",
       "\n",
       "         Brain_Anterior_cingulate_cortex_BA24  Brain_Caudate_basal_ganglia  \\\n",
       "299707                                    NaN                          NaN   \n",
       "642626                                    0.0                          0.0   \n",
       "939053                                    NaN                          NaN   \n",
       "1376781                                   NaN                          NaN   \n",
       "19011                                     0.0                          0.0   \n",
       "1023894                                   0.0                          0.0   \n",
       "615932                                    0.0                          1.0   \n",
       "615931                                    0.0                          1.0   \n",
       "114787                                    NaN                          NaN   \n",
       "635204                                    NaN                          NaN   \n",
       "574423                                    NaN                          NaN   \n",
       "656165                                    NaN                          NaN   \n",
       "1472211                                   0.0                          0.0   \n",
       "403424                                    NaN                          NaN   \n",
       "554672                                    1.0                          1.0   \n",
       "47872                                     0.0                          0.0   \n",
       "727597                                    NaN                          NaN   \n",
       "610909                                    NaN                          NaN   \n",
       "1594464                                   NaN                          NaN   \n",
       "1291956                                   NaN                          NaN   \n",
       "\n",
       "         Brain_Cerebellar_Hemisphere  Brain_Cerebellum  Brain_Cortex  \\\n",
       "299707                           NaN               NaN           NaN   \n",
       "642626                           0.0               0.0           0.0   \n",
       "939053                           NaN               NaN           NaN   \n",
       "1376781                          NaN               NaN           NaN   \n",
       "19011                            0.0               0.0           0.0   \n",
       "1023894                          0.0               0.0           0.0   \n",
       "615932                           1.0               1.0           1.0   \n",
       "615931                           1.0               1.0           1.0   \n",
       "114787                           NaN               NaN           NaN   \n",
       "635204                           NaN               NaN           NaN   \n",
       "574423                           NaN               NaN           NaN   \n",
       "656165                           NaN               NaN           NaN   \n",
       "1472211                          0.0               0.0           0.0   \n",
       "403424                           NaN               NaN           NaN   \n",
       "554672                           1.0               1.0           1.0   \n",
       "47872                            0.0               0.0           0.0   \n",
       "727597                           NaN               NaN           NaN   \n",
       "610909                           NaN               NaN           NaN   \n",
       "1594464                          NaN               NaN           NaN   \n",
       "1291956                          NaN               NaN           NaN   \n",
       "\n",
       "         Brain_Frontal_Cortex_BA9  Brain_Hippocampus  Brain_Hypothalamus  \\\n",
       "299707                        NaN                NaN                 NaN   \n",
       "642626                        0.0                0.0                 0.0   \n",
       "939053                        NaN                NaN                 NaN   \n",
       "1376781                       NaN                NaN                 NaN   \n",
       "19011                         0.0                0.0                 0.0   \n",
       "1023894                       0.0                0.0                 0.0   \n",
       "615932                        1.0                0.0                 1.0   \n",
       "615931                        1.0                0.0                 1.0   \n",
       "114787                        NaN                NaN                 NaN   \n",
       "635204                        NaN                NaN                 NaN   \n",
       "574423                        NaN                NaN                 NaN   \n",
       "656165                        NaN                NaN                 NaN   \n",
       "1472211                       0.0                0.0                 0.0   \n",
       "403424                        NaN                NaN                 NaN   \n",
       "554672                        1.0                1.0                 1.0   \n",
       "47872                         0.0                0.0                 0.0   \n",
       "727597                        NaN                NaN                 NaN   \n",
       "610909                        NaN                NaN                 NaN   \n",
       "1594464                       NaN                NaN                 NaN   \n",
       "1291956                       NaN                NaN                 NaN   \n",
       "\n",
       "         Brain_Nucleus_accumbens_basal_ganglia  Brain_Putamen_basal_ganglia  \\\n",
       "299707                                     NaN                          NaN   \n",
       "642626                                     0.0                          0.0   \n",
       "939053                                     NaN                          NaN   \n",
       "1376781                                    NaN                          NaN   \n",
       "19011                                      0.0                          0.0   \n",
       "1023894                                    0.0                          0.0   \n",
       "615932                                     1.0                          1.0   \n",
       "615931                                     1.0                          1.0   \n",
       "114787                                     NaN                          NaN   \n",
       "635204                                     NaN                          NaN   \n",
       "574423                                     NaN                          NaN   \n",
       "656165                                     NaN                          NaN   \n",
       "1472211                                    0.0                          0.0   \n",
       "403424                                     NaN                          NaN   \n",
       "554672                                     1.0                          1.0   \n",
       "47872                                      0.0                          0.0   \n",
       "727597                                     NaN                          NaN   \n",
       "610909                                     NaN                          NaN   \n",
       "1594464                                    NaN                          NaN   \n",
       "1291956                                    NaN                          NaN   \n",
       "\n",
       "         Brain_Spinal_cord_cervical_c-1  Brain_Substantia_nigra  \\\n",
       "299707                              NaN                     NaN   \n",
       "642626                              0.0                     0.0   \n",
       "939053                              NaN                     NaN   \n",
       "1376781                             NaN                     NaN   \n",
       "19011                               0.0                     0.0   \n",
       "1023894                             0.0                     0.0   \n",
       "615932                              0.0                     1.0   \n",
       "615931                              0.0                     1.0   \n",
       "114787                              NaN                     NaN   \n",
       "635204                              NaN                     NaN   \n",
       "574423                              NaN                     NaN   \n",
       "656165                              NaN                     NaN   \n",
       "1472211                             0.0                     0.0   \n",
       "403424                              NaN                     NaN   \n",
       "554672                              0.0                     0.0   \n",
       "47872                               0.0                     0.0   \n",
       "727597                              NaN                     NaN   \n",
       "610909                              NaN                     NaN   \n",
       "1594464                             NaN                     NaN   \n",
       "1291956                             NaN                     NaN   \n",
       "\n",
       "         Breast_Mammary_Tissue  Cells_Cultured_fibroblasts  \\\n",
       "299707                     NaN                         NaN   \n",
       "642626                     0.0                         0.0   \n",
       "939053                     NaN                         NaN   \n",
       "1376781                    NaN                         NaN   \n",
       "19011                      0.0                         0.0   \n",
       "1023894                    1.0                         0.0   \n",
       "615932                     0.0                         0.0   \n",
       "615931                     0.0                         0.0   \n",
       "114787                     NaN                         NaN   \n",
       "635204                     NaN                         NaN   \n",
       "574423                     NaN                         NaN   \n",
       "656165                     NaN                         NaN   \n",
       "1472211                    0.0                         1.0   \n",
       "403424                     NaN                         NaN   \n",
       "554672                     1.0                         1.0   \n",
       "47872                      0.0                         0.0   \n",
       "727597                     NaN                         NaN   \n",
       "610909                     NaN                         NaN   \n",
       "1594464                    NaN                         NaN   \n",
       "1291956                    NaN                         NaN   \n",
       "\n",
       "         Cells_EBV-transformed_lymphocytes  Colon_Sigmoid  Colon_Transverse  \\\n",
       "299707                                 NaN            NaN               NaN   \n",
       "642626                                 0.0            0.0               0.0   \n",
       "939053                                 NaN            NaN               NaN   \n",
       "1376781                                NaN            NaN               NaN   \n",
       "19011                                  0.0            0.0               0.0   \n",
       "1023894                                0.0            1.0               0.0   \n",
       "615932                                 1.0            0.0               1.0   \n",
       "615931                                 1.0            0.0               1.0   \n",
       "114787                                 NaN            NaN               NaN   \n",
       "635204                                 NaN            NaN               NaN   \n",
       "574423                                 NaN            NaN               NaN   \n",
       "656165                                 NaN            NaN               NaN   \n",
       "1472211                                0.0            0.0               0.0   \n",
       "403424                                 NaN            NaN               NaN   \n",
       "554672                                 1.0            1.0               1.0   \n",
       "47872                                  0.0            0.0               0.0   \n",
       "727597                                 NaN            NaN               NaN   \n",
       "610909                                 NaN            NaN               NaN   \n",
       "1594464                                NaN            NaN               NaN   \n",
       "1291956                                NaN            NaN               NaN   \n",
       "\n",
       "         Esophagus_Gastroesophageal_Junction  Esophagus_Mucosa  \\\n",
       "299707                                   NaN               NaN   \n",
       "642626                                   0.0               0.0   \n",
       "939053                                   NaN               NaN   \n",
       "1376781                                  NaN               NaN   \n",
       "19011                                    0.0               0.0   \n",
       "1023894                                  1.0               0.0   \n",
       "615932                                   0.0               1.0   \n",
       "615931                                   0.0               1.0   \n",
       "114787                                   NaN               NaN   \n",
       "635204                                   NaN               NaN   \n",
       "574423                                   NaN               NaN   \n",
       "656165                                   NaN               NaN   \n",
       "1472211                                  0.0               0.0   \n",
       "403424                                   NaN               NaN   \n",
       "554672                                   1.0               1.0   \n",
       "47872                                    0.0               0.0   \n",
       "727597                                   NaN               NaN   \n",
       "610909                                   NaN               NaN   \n",
       "1594464                                  NaN               NaN   \n",
       "1291956                                  NaN               NaN   \n",
       "\n",
       "         Esophagus_Muscularis  Heart_Atrial_Appendage  Heart_Left_Ventricle  \\\n",
       "299707                    NaN                     NaN                   NaN   \n",
       "642626                    0.0                     0.0                   0.0   \n",
       "939053                    NaN                     NaN                   NaN   \n",
       "1376781                   NaN                     NaN                   NaN   \n",
       "19011                     0.0                     0.0                   0.0   \n",
       "1023894                   1.0                     0.0                   0.0   \n",
       "615932                    0.0                     1.0                   1.0   \n",
       "615931                    0.0                     1.0                   1.0   \n",
       "114787                    NaN                     NaN                   NaN   \n",
       "635204                    NaN                     NaN                   NaN   \n",
       "574423                    NaN                     NaN                   NaN   \n",
       "656165                    NaN                     NaN                   NaN   \n",
       "1472211                   0.0                     0.0                   0.0   \n",
       "403424                    NaN                     NaN                   NaN   \n",
       "554672                    1.0                     1.0                   1.0   \n",
       "47872                     0.0                     0.0                   0.0   \n",
       "727597                    NaN                     NaN                   NaN   \n",
       "610909                    NaN                     NaN                   NaN   \n",
       "1594464                   NaN                     NaN                   NaN   \n",
       "1291956                   NaN                     NaN                   NaN   \n",
       "\n",
       "         Kidney_Cortex  Liver  Lung  Minor_Salivary_Gland  Muscle_Skeletal  \\\n",
       "299707             NaN    NaN   NaN                   NaN              NaN   \n",
       "642626             1.0    0.0   1.0                   0.0              0.0   \n",
       "939053             NaN    NaN   NaN                   NaN              NaN   \n",
       "1376781            NaN    NaN   NaN                   NaN              NaN   \n",
       "19011              0.0    0.0   0.0                   0.0              0.0   \n",
       "1023894            0.0    0.0   1.0                   0.0              0.0   \n",
       "615932             0.0    0.0   1.0                   1.0              1.0   \n",
       "615931             0.0    0.0   1.0                   1.0              1.0   \n",
       "114787             NaN    NaN   NaN                   NaN              NaN   \n",
       "635204             NaN    NaN   NaN                   NaN              NaN   \n",
       "574423             NaN    NaN   NaN                   NaN              NaN   \n",
       "656165             NaN    NaN   NaN                   NaN              NaN   \n",
       "1472211            0.0    0.0   0.0                   0.0              0.0   \n",
       "403424             NaN    NaN   NaN                   NaN              NaN   \n",
       "554672             0.0    1.0   1.0                   0.0              1.0   \n",
       "47872              0.0    0.0   0.0                   0.0              0.0   \n",
       "727597             NaN    NaN   NaN                   NaN              NaN   \n",
       "610909             NaN    NaN   NaN                   NaN              NaN   \n",
       "1594464            NaN    NaN   NaN                   NaN              NaN   \n",
       "1291956            NaN    NaN   NaN                   NaN              NaN   \n",
       "\n",
       "         Nerve_Tibial  Ovary  Pancreas  Pituitary  Prostate  \\\n",
       "299707            NaN    NaN       NaN        NaN       NaN   \n",
       "642626            0.0    0.0       0.0        0.0       0.0   \n",
       "939053            NaN    NaN       NaN        NaN       NaN   \n",
       "1376781           NaN    NaN       NaN        NaN       NaN   \n",
       "19011             1.0    0.0       0.0        0.0       0.0   \n",
       "1023894           0.0    0.0       1.0        1.0       1.0   \n",
       "615932            0.0    0.0       0.0        0.0       0.0   \n",
       "615931            0.0    0.0       0.0        0.0       0.0   \n",
       "114787            NaN    NaN       NaN        NaN       NaN   \n",
       "635204            NaN    NaN       NaN        NaN       NaN   \n",
       "574423            NaN    NaN       NaN        NaN       NaN   \n",
       "656165            NaN    NaN       NaN        NaN       NaN   \n",
       "1472211           1.0    0.0       0.0        0.0       0.0   \n",
       "403424            NaN    NaN       NaN        NaN       NaN   \n",
       "554672            1.0    0.0       1.0        1.0       1.0   \n",
       "47872             0.0    0.0       0.0        0.0       0.0   \n",
       "727597            NaN    NaN       NaN        NaN       NaN   \n",
       "610909            NaN    NaN       NaN        NaN       NaN   \n",
       "1594464           NaN    NaN       NaN        NaN       NaN   \n",
       "1291956           NaN    NaN       NaN        NaN       NaN   \n",
       "\n",
       "         Skin_Not_Sun_Exposed_Suprapubic  Skin_Sun_Exposed_Lower_leg  \\\n",
       "299707                               NaN                         NaN   \n",
       "642626                               0.0                         1.0   \n",
       "939053                               NaN                         NaN   \n",
       "1376781                              NaN                         NaN   \n",
       "19011                                0.0                         0.0   \n",
       "1023894                              0.0                         0.0   \n",
       "615932                               0.0                         0.0   \n",
       "615931                               0.0                         0.0   \n",
       "114787                               NaN                         NaN   \n",
       "635204                               NaN                         NaN   \n",
       "574423                               NaN                         NaN   \n",
       "656165                               NaN                         NaN   \n",
       "1472211                              0.0                         1.0   \n",
       "403424                               NaN                         NaN   \n",
       "554672                               1.0                         1.0   \n",
       "47872                                0.0                         1.0   \n",
       "727597                               NaN                         NaN   \n",
       "610909                               NaN                         NaN   \n",
       "1594464                              NaN                         NaN   \n",
       "1291956                              NaN                         NaN   \n",
       "\n",
       "         Small_Intestine_Terminal_Ileum  Spleen  Stomach  Testis  Thyroid  \\\n",
       "299707                              NaN     NaN      NaN     NaN      NaN   \n",
       "642626                              0.0     0.0      0.0     0.0      1.0   \n",
       "939053                              NaN     NaN      NaN     NaN      NaN   \n",
       "1376781                             NaN     NaN      NaN     NaN      NaN   \n",
       "19011                               0.0     0.0      0.0     0.0      1.0   \n",
       "1023894                             0.0     0.0      0.0     0.0      1.0   \n",
       "615932                              0.0     1.0      1.0     0.0      0.0   \n",
       "615931                              0.0     1.0      1.0     0.0      0.0   \n",
       "114787                              NaN     NaN      NaN     NaN      NaN   \n",
       "635204                              NaN     NaN      NaN     NaN      NaN   \n",
       "574423                              NaN     NaN      NaN     NaN      NaN   \n",
       "656165                              NaN     NaN      NaN     NaN      NaN   \n",
       "1472211                             0.0     0.0      0.0     1.0      1.0   \n",
       "403424                              NaN     NaN      NaN     NaN      NaN   \n",
       "554672                              1.0     1.0      1.0     1.0      1.0   \n",
       "47872                               0.0     0.0      0.0     0.0      0.0   \n",
       "727597                              NaN     NaN      NaN     NaN      NaN   \n",
       "610909                              NaN     NaN      NaN     NaN      NaN   \n",
       "1594464                             NaN     NaN      NaN     NaN      NaN   \n",
       "1291956                             NaN     NaN      NaN     NaN      NaN   \n",
       "\n",
       "         Uterus  Vagina  Whole_Blood  N_GTEx_tissues sQTL  \\\n",
       "299707      NaN     NaN          NaN             NaN   no   \n",
       "642626      1.0     0.0          0.0             7.0  yes   \n",
       "939053      NaN     NaN          NaN             NaN   no   \n",
       "1376781     NaN     NaN          NaN             NaN   no   \n",
       "19011       0.0     0.0          0.0             4.0  yes   \n",
       "1023894     1.0     0.0          0.0            11.0  yes   \n",
       "615932      0.0     0.0          1.0            20.0  yes   \n",
       "615931      0.0     0.0          1.0            20.0  yes   \n",
       "114787      NaN     NaN          NaN             NaN   no   \n",
       "635204      NaN     NaN          NaN             NaN   no   \n",
       "574423      NaN     NaN          NaN             NaN   no   \n",
       "656165      NaN     NaN          NaN             NaN   no   \n",
       "1472211     0.0     0.0          0.0             6.0  yes   \n",
       "403424      NaN     NaN          NaN             NaN   no   \n",
       "554672      1.0     0.0          1.0            43.0  yes   \n",
       "47872       0.0     0.0          0.0             1.0  yes   \n",
       "727597      NaN     NaN          NaN             NaN   no   \n",
       "610909      NaN     NaN          NaN             NaN   no   \n",
       "1594464     NaN     NaN          NaN             NaN   no   \n",
       "1291956     NaN     NaN          NaN             NaN   no   \n",
       "\n",
       "           Vernot_allele_origin  Browning_allele_origin  \n",
       "299707         archaic-specific        archaic-specific  \n",
       "642626                  ancient                 ancient  \n",
       "939053                  ancient                 ancient  \n",
       "1376781        archaic-specific        archaic-specific  \n",
       "19011              introgressed            introgressed  \n",
       "1023894                 ancient                 ancient  \n",
       "615932   low-confidence ancient  low-confidence ancient  \n",
       "615931   low-confidence ancient  low-confidence ancient  \n",
       "114787         archaic-specific        archaic-specific  \n",
       "635204         archaic-specific        archaic-specific  \n",
       "574423                  ancient                 ancient  \n",
       "656165                  ancient                 ancient  \n",
       "1472211                 ancient                 ancient  \n",
       "403424   low-confidence ancient  low-confidence ancient  \n",
       "554672                  ancient                 ancient  \n",
       "47872                   ancient                 ancient  \n",
       "727597         archaic-specific        archaic-specific  \n",
       "610909         archaic-specific        archaic-specific  \n",
       "1594464        archaic-specific        archaic-specific  \n",
       "1291956        archaic-specific        archaic-specific  "
      ]
     },
     "execution_count": 180,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_2_dg = data_2.sort_values(['dg_delta'], ascending = [False])\n",
    "data_2_dg.head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 181,
   "id": "fc8759b0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref_allele</th>\n",
       "      <th>alt_allele</th>\n",
       "      <th>ancestral_allele</th>\n",
       "      <th>anc_dev</th>\n",
       "      <th>variant_type</th>\n",
       "      <th>altai_gt</th>\n",
       "      <th>chagyrskaya_gt</th>\n",
       "      <th>denisovan_gt</th>\n",
       "      <th>vindija_gt</th>\n",
       "      <th>altai_gt_boolean</th>\n",
       "      <th>chagyrskaya_gt_boolean</th>\n",
       "      <th>denisovan_gt_boolean</th>\n",
       "      <th>vindija_gt_boolean</th>\n",
       "      <th>distribution</th>\n",
       "      <th>present_in_1KG</th>\n",
       "      <th>1KG_allele_count</th>\n",
       "      <th>1KG_allele_number</th>\n",
       "      <th>1KG_allele_frequency</th>\n",
       "      <th>1KG_EAS_AF</th>\n",
       "      <th>1KG_EUR_AF</th>\n",
       "      <th>1KG_AFR_AF</th>\n",
       "      <th>1KG_AMR_AF</th>\n",
       "      <th>1KG_SAS_AF</th>\n",
       "      <th>1KG_non_ASW_AFR_AF</th>\n",
       "      <th>present_in_gnomAD</th>\n",
       "      <th>gnomAD_allele_count</th>\n",
       "      <th>gnomAD_allele_number</th>\n",
       "      <th>gnomAD_allele_frequency</th>\n",
       "      <th>Vernot_introgressed</th>\n",
       "      <th>Vernot_ancestral_allele</th>\n",
       "      <th>Vernot_derived_allele</th>\n",
       "      <th>Vernot_ancestral_derived_code</th>\n",
       "      <th>Vernot_AFA_AF</th>\n",
       "      <th>Vernot_AFR_AF</th>\n",
       "      <th>Vernot_AMR_AF</th>\n",
       "      <th>Vernot_EAS_AF</th>\n",
       "      <th>Vernot_EUR_AF</th>\n",
       "      <th>Vernot_PNG_AF</th>\n",
       "      <th>Vernot_SAS_AF</th>\n",
       "      <th>Vernot_Denisovan_base</th>\n",
       "      <th>Vernot_Neanderthal_base_1</th>\n",
       "      <th>Vernot_Neanderthal_base_2</th>\n",
       "      <th>Vernot_haplotype_tag</th>\n",
       "      <th>Vernot_introgressed_AF</th>\n",
       "      <th>Browning_introgressed</th>\n",
       "      <th>Browning_ref_alt</th>\n",
       "      <th>Browning_introgressed_AF</th>\n",
       "      <th>annotation</th>\n",
       "      <th>mis_oe</th>\n",
       "      <th>mis_z</th>\n",
       "      <th>lof_oe</th>\n",
       "      <th>lof_z</th>\n",
       "      <th>phyloP</th>\n",
       "      <th>ag_delta</th>\n",
       "      <th>al_delta</th>\n",
       "      <th>dg_delta</th>\n",
       "      <th>dl_delta</th>\n",
       "      <th>delta_max</th>\n",
       "      <th>ag_pos</th>\n",
       "      <th>al_pos</th>\n",
       "      <th>dg_pos</th>\n",
       "      <th>dl_pos</th>\n",
       "      <th>Adipose_Subcutaneous</th>\n",
       "      <th>Adipose_Visceral_Omentum</th>\n",
       "      <th>Adrenal_Gland</th>\n",
       "      <th>Artery_Aorta</th>\n",
       "      <th>Artery_Coronary</th>\n",
       "      <th>Artery_Tibial</th>\n",
       "      <th>Brain_Amygdala</th>\n",
       "      <th>Brain_Anterior_cingulate_cortex_BA24</th>\n",
       "      <th>Brain_Caudate_basal_ganglia</th>\n",
       "      <th>Brain_Cerebellar_Hemisphere</th>\n",
       "      <th>Brain_Cerebellum</th>\n",
       "      <th>Brain_Cortex</th>\n",
       "      <th>Brain_Frontal_Cortex_BA9</th>\n",
       "      <th>Brain_Hippocampus</th>\n",
       "      <th>Brain_Hypothalamus</th>\n",
       "      <th>Brain_Nucleus_accumbens_basal_ganglia</th>\n",
       "      <th>Brain_Putamen_basal_ganglia</th>\n",
       "      <th>Brain_Spinal_cord_cervical_c-1</th>\n",
       "      <th>Brain_Substantia_nigra</th>\n",
       "      <th>Breast_Mammary_Tissue</th>\n",
       "      <th>Cells_Cultured_fibroblasts</th>\n",
       "      <th>Cells_EBV-transformed_lymphocytes</th>\n",
       "      <th>Colon_Sigmoid</th>\n",
       "      <th>Colon_Transverse</th>\n",
       "      <th>Esophagus_Gastroesophageal_Junction</th>\n",
       "      <th>Esophagus_Mucosa</th>\n",
       "      <th>Esophagus_Muscularis</th>\n",
       "      <th>Heart_Atrial_Appendage</th>\n",
       "      <th>Heart_Left_Ventricle</th>\n",
       "      <th>Kidney_Cortex</th>\n",
       "      <th>Liver</th>\n",
       "      <th>Lung</th>\n",
       "      <th>Minor_Salivary_Gland</th>\n",
       "      <th>Muscle_Skeletal</th>\n",
       "      <th>Nerve_Tibial</th>\n",
       "      <th>Ovary</th>\n",
       "      <th>Pancreas</th>\n",
       "      <th>Pituitary</th>\n",
       "      <th>Prostate</th>\n",
       "      <th>Skin_Not_Sun_Exposed_Suprapubic</th>\n",
       "      <th>Skin_Sun_Exposed_Lower_leg</th>\n",
       "      <th>Small_Intestine_Terminal_Ileum</th>\n",
       "      <th>Spleen</th>\n",
       "      <th>Stomach</th>\n",
       "      <th>Testis</th>\n",
       "      <th>Thyroid</th>\n",
       "      <th>Uterus</th>\n",
       "      <th>Vagina</th>\n",
       "      <th>Whole_Blood</th>\n",
       "      <th>N_GTEx_tissues</th>\n",
       "      <th>sQTL</th>\n",
       "      <th>Vernot_allele_origin</th>\n",
       "      <th>Browning_allele_origin</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>549948</th>\n",
       "      <td>chr15</td>\n",
       "      <td>99514264</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Vindija</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PGPEP1L</td>\n",
       "      <td>1.28090</td>\n",
       "      <td>-1.094500</td>\n",
       "      <td>1.68290</td>\n",
       "      <td>-1.54260</td>\n",
       "      <td>0.650</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.01</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1.00</td>\n",
       "      <td>-3</td>\n",
       "      <td>49</td>\n",
       "      <td>-3</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>613603</th>\n",
       "      <td>chr17</td>\n",
       "      <td>1386153</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>Other</td>\n",
       "      <td>yes</td>\n",
       "      <td>886.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.17</td>\n",
       "      <td>0.13</td>\n",
       "      <td>0.24</td>\n",
       "      <td>0.17</td>\n",
       "      <td>0.17</td>\n",
       "      <td>0.16</td>\n",
       "      <td>0.154971</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MYO1C</td>\n",
       "      <td>1.06990</td>\n",
       "      <td>-0.636630</td>\n",
       "      <td>0.51054</td>\n",
       "      <td>3.70140</td>\n",
       "      <td>0.334</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.92</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1.00</td>\n",
       "      <td>11</td>\n",
       "      <td>19</td>\n",
       "      <td>46</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>ancient</td>\n",
       "      <td>ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>549796</th>\n",
       "      <td>chr15</td>\n",
       "      <td>99442852</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>./.</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>IGF1R</td>\n",
       "      <td>0.73351</td>\n",
       "      <td>2.728600</td>\n",
       "      <td>0.18874</td>\n",
       "      <td>5.99420</td>\n",
       "      <td>0.460</td>\n",
       "      <td>0.04</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1.00</td>\n",
       "      <td>-20</td>\n",
       "      <td>-31</td>\n",
       "      <td>-20</td>\n",
       "      <td>-2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>933419</th>\n",
       "      <td>chr20</td>\n",
       "      <td>54963211</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>./.</td>\n",
       "      <td>0/1</td>\n",
       "      <td>./.</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AURKA</td>\n",
       "      <td>0.68585</td>\n",
       "      <td>1.595000</td>\n",
       "      <td>0.15147</td>\n",
       "      <td>3.49930</td>\n",
       "      <td>0.585</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.26</td>\n",
       "      <td>0.41</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1.00</td>\n",
       "      <td>-49</td>\n",
       "      <td>47</td>\n",
       "      <td>-49</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1476084</th>\n",
       "      <td>chr8</td>\n",
       "      <td>33361265</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Vindija</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TTI2</td>\n",
       "      <td>1.04940</td>\n",
       "      <td>-0.288910</td>\n",
       "      <td>0.65730</td>\n",
       "      <td>1.41230</td>\n",
       "      <td>0.650</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.02</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1.00</td>\n",
       "      <td>-39</td>\n",
       "      <td>36</td>\n",
       "      <td>-4</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>572769</th>\n",
       "      <td>chr16</td>\n",
       "      <td>20638576</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>ancestral</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>yes</td>\n",
       "      <td>3083.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.60</td>\n",
       "      <td>0.85</td>\n",
       "      <td>0.32</td>\n",
       "      <td>0.78</td>\n",
       "      <td>0.34</td>\n",
       "      <td>0.59</td>\n",
       "      <td>0.812865</td>\n",
       "      <td>yes</td>\n",
       "      <td>67770.0</td>\n",
       "      <td>142920.0</td>\n",
       "      <td>0.474181</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ACSM1</td>\n",
       "      <td>0.96843</td>\n",
       "      <td>0.203920</td>\n",
       "      <td>0.77860</td>\n",
       "      <td>1.20820</td>\n",
       "      <td>-0.257</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.99</td>\n",
       "      <td>0.99</td>\n",
       "      <td>47</td>\n",
       "      <td>2</td>\n",
       "      <td>-25</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>ancient</td>\n",
       "      <td>ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>448731</th>\n",
       "      <td>chr14</td>\n",
       "      <td>24568446</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Vindija</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PCK2</td>\n",
       "      <td>1.02830</td>\n",
       "      <td>-0.202710</td>\n",
       "      <td>1.04620</td>\n",
       "      <td>-0.24419</td>\n",
       "      <td>0.563</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.07</td>\n",
       "      <td>0.99</td>\n",
       "      <td>0.99</td>\n",
       "      <td>10</td>\n",
       "      <td>-1</td>\n",
       "      <td>10</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1347914</th>\n",
       "      <td>chr7</td>\n",
       "      <td>12393974</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>yes</td>\n",
       "      <td>4333.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.85</td>\n",
       "      <td>0.81</td>\n",
       "      <td>0.90</td>\n",
       "      <td>0.81</td>\n",
       "      <td>0.91</td>\n",
       "      <td>0.86</td>\n",
       "      <td>0.803119</td>\n",
       "      <td>yes</td>\n",
       "      <td>124274.0</td>\n",
       "      <td>142956.0</td>\n",
       "      <td>0.869316</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>VWDE</td>\n",
       "      <td>1.01570</td>\n",
       "      <td>-0.155870</td>\n",
       "      <td>0.90845</td>\n",
       "      <td>0.71763</td>\n",
       "      <td>0.533</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.99</td>\n",
       "      <td>0.99</td>\n",
       "      <td>-26</td>\n",
       "      <td>34</td>\n",
       "      <td>47</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>ancient</td>\n",
       "      <td>ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1169648</th>\n",
       "      <td>chr5</td>\n",
       "      <td>74062762</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>./.</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>yes</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GFM2</td>\n",
       "      <td>0.88906</td>\n",
       "      <td>0.796580</td>\n",
       "      <td>0.77687</td>\n",
       "      <td>1.34760</td>\n",
       "      <td>0.462</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.14</td>\n",
       "      <td>0.99</td>\n",
       "      <td>0.99</td>\n",
       "      <td>45</td>\n",
       "      <td>0</td>\n",
       "      <td>37</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>749275</th>\n",
       "      <td>chr19</td>\n",
       "      <td>51535130</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>yes</td>\n",
       "      <td>2725.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.53</td>\n",
       "      <td>0.64</td>\n",
       "      <td>0.56</td>\n",
       "      <td>0.36</td>\n",
       "      <td>0.56</td>\n",
       "      <td>0.62</td>\n",
       "      <td>0.359649</td>\n",
       "      <td>yes</td>\n",
       "      <td>74328.0</td>\n",
       "      <td>142512.0</td>\n",
       "      <td>0.521556</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>KLK12</td>\n",
       "      <td>1.07420</td>\n",
       "      <td>-0.329600</td>\n",
       "      <td>0.75715</td>\n",
       "      <td>0.73148</td>\n",
       "      <td>0.528</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.18</td>\n",
       "      <td>0.99</td>\n",
       "      <td>0.99</td>\n",
       "      <td>-19</td>\n",
       "      <td>28</td>\n",
       "      <td>-26</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>ancient</td>\n",
       "      <td>ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1483683</th>\n",
       "      <td>chr8</td>\n",
       "      <td>52469399</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>Chagyrskaya</td>\n",
       "      <td>yes</td>\n",
       "      <td>24.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>yes</td>\n",
       "      <td>57.0</td>\n",
       "      <td>143244.0</td>\n",
       "      <td>0.000398</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PXDNL</td>\n",
       "      <td>1.01930</td>\n",
       "      <td>-0.196550</td>\n",
       "      <td>0.87676</td>\n",
       "      <td>0.92884</td>\n",
       "      <td>0.650</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.99</td>\n",
       "      <td>0.99</td>\n",
       "      <td>-3</td>\n",
       "      <td>22</td>\n",
       "      <td>-3</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>701098</th>\n",
       "      <td>chr18</td>\n",
       "      <td>51731367</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Other</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MBD2</td>\n",
       "      <td>0.53061</td>\n",
       "      <td>2.111100</td>\n",
       "      <td>0.22818</td>\n",
       "      <td>2.99450</td>\n",
       "      <td>0.643</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.99</td>\n",
       "      <td>0.99</td>\n",
       "      <td>-42</td>\n",
       "      <td>45</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>575663</th>\n",
       "      <td>chr16</td>\n",
       "      <td>25255105</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ZKSCAN2</td>\n",
       "      <td>0.96998</td>\n",
       "      <td>0.248880</td>\n",
       "      <td>0.36273</td>\n",
       "      <td>3.92200</td>\n",
       "      <td>0.563</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.57</td>\n",
       "      <td>0.99</td>\n",
       "      <td>0.99</td>\n",
       "      <td>-4</td>\n",
       "      <td>1</td>\n",
       "      <td>-4</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>573334</th>\n",
       "      <td>chr16</td>\n",
       "      <td>22019730</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>t</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C16orf52</td>\n",
       "      <td>0.54356</td>\n",
       "      <td>1.425800</td>\n",
       "      <td>0.26140</td>\n",
       "      <td>1.89320</td>\n",
       "      <td>0.409</td>\n",
       "      <td>0.01</td>\n",
       "      <td>-0.00</td>\n",
       "      <td>0.03</td>\n",
       "      <td>0.99</td>\n",
       "      <td>0.99</td>\n",
       "      <td>-42</td>\n",
       "      <td>-2</td>\n",
       "      <td>-10</td>\n",
       "      <td>-2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1055487</th>\n",
       "      <td>chr4</td>\n",
       "      <td>24981689</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>ancestral</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>yes</td>\n",
       "      <td>2347.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.46</td>\n",
       "      <td>0.51</td>\n",
       "      <td>0.24</td>\n",
       "      <td>0.62</td>\n",
       "      <td>0.37</td>\n",
       "      <td>0.48</td>\n",
       "      <td>0.642300</td>\n",
       "      <td>yes</td>\n",
       "      <td>52419.0</td>\n",
       "      <td>143106.0</td>\n",
       "      <td>0.366295</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CCDC149</td>\n",
       "      <td>0.79749</td>\n",
       "      <td>1.178900</td>\n",
       "      <td>0.47219</td>\n",
       "      <td>2.46560</td>\n",
       "      <td>-0.257</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.13</td>\n",
       "      <td>0.99</td>\n",
       "      <td>0.99</td>\n",
       "      <td>48</td>\n",
       "      <td>2</td>\n",
       "      <td>7</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>ancient</td>\n",
       "      <td>ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>587259</th>\n",
       "      <td>chr16</td>\n",
       "      <td>67995477</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>Chagyrskaya</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SLC12A4</td>\n",
       "      <td>0.85544</td>\n",
       "      <td>1.353700</td>\n",
       "      <td>0.49267</td>\n",
       "      <td>3.48020</td>\n",
       "      <td>0.462</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.98</td>\n",
       "      <td>0.98</td>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "      <td>-7</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>808374</th>\n",
       "      <td>chr2</td>\n",
       "      <td>88826016</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TEX37</td>\n",
       "      <td>1.18320</td>\n",
       "      <td>-0.658200</td>\n",
       "      <td>1.32610</td>\n",
       "      <td>-0.64277</td>\n",
       "      <td>0.462</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.08</td>\n",
       "      <td>0.98</td>\n",
       "      <td>0.98</td>\n",
       "      <td>42</td>\n",
       "      <td>-42</td>\n",
       "      <td>7</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1476102</th>\n",
       "      <td>chr8</td>\n",
       "      <td>33370599</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>./.</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>yes</td>\n",
       "      <td>73.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.056530</td>\n",
       "      <td>yes</td>\n",
       "      <td>2365.0</td>\n",
       "      <td>143230.0</td>\n",
       "      <td>0.016512</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TTI2</td>\n",
       "      <td>1.04940</td>\n",
       "      <td>-0.288910</td>\n",
       "      <td>0.65730</td>\n",
       "      <td>1.41230</td>\n",
       "      <td>0.655</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.45</td>\n",
       "      <td>0.98</td>\n",
       "      <td>0.98</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>43</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>852344</th>\n",
       "      <td>chr2</td>\n",
       "      <td>172544138</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DYNC1I2</td>\n",
       "      <td>0.61314</td>\n",
       "      <td>2.451600</td>\n",
       "      <td>0.52411</td>\n",
       "      <td>2.65520</td>\n",
       "      <td>0.467</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.98</td>\n",
       "      <td>0.98</td>\n",
       "      <td>0</td>\n",
       "      <td>32</td>\n",
       "      <td>-2</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>656470</th>\n",
       "      <td>chr17</td>\n",
       "      <td>68127117</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>ancestral</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>yes</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>yes</td>\n",
       "      <td>2.0</td>\n",
       "      <td>143212.0</td>\n",
       "      <td>0.000014</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>KCNJ16</td>\n",
       "      <td>1.00040</td>\n",
       "      <td>-0.002204</td>\n",
       "      <td>1.11810</td>\n",
       "      <td>-0.41397</td>\n",
       "      <td>-4.309</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.98</td>\n",
       "      <td>0.98</td>\n",
       "      <td>35</td>\n",
       "      <td>-43</td>\n",
       "      <td>-49</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         chrom        pos ref_allele alt_allele ancestral_allele    anc_dev  \\\n",
       "549948   chr15   99514264          C          A                C    derived   \n",
       "613603   chr17    1386153          A          C                A    derived   \n",
       "549796   chr15   99442852          T          A                T    derived   \n",
       "933419   chr20   54963211          C          G                C    derived   \n",
       "1476084   chr8   33361265          C          T                C    derived   \n",
       "572769   chr16   20638576          A          T                T  ancestral   \n",
       "448731   chr14   24568446          G          A                G    derived   \n",
       "1347914   chr7   12393974          A          C                A    derived   \n",
       "1169648   chr5   74062762          C          G                C    derived   \n",
       "749275   chr19   51535130          A          G                A    derived   \n",
       "1483683   chr8   52469399          C          T                C    derived   \n",
       "701098   chr18   51731367          C          G                C    derived   \n",
       "575663   chr16   25255105          C          A                C    derived   \n",
       "573334   chr16   22019730          T          A                t    derived   \n",
       "1055487   chr4   24981689          A          T                T  ancestral   \n",
       "587259   chr16   67995477          C          T                C    derived   \n",
       "808374    chr2   88826016          G          T                G    derived   \n",
       "1476102   chr8   33370599          C          T                C    derived   \n",
       "852344    chr2  172544138          G          A                G    derived   \n",
       "656470   chr17   68127117          G          A                A  ancestral   \n",
       "\n",
       "        variant_type altai_gt chagyrskaya_gt denisovan_gt vindija_gt  \\\n",
       "549948           snv      0/0            0/0          0/0        0/1   \n",
       "613603           snv      0/1            0/1          ./.        ./.   \n",
       "549796           snv      ./.            0/0          0/1        0/0   \n",
       "933419           snv      0/0            ./.          0/1        ./.   \n",
       "1476084          snv      0/0            0/0          0/0        0/1   \n",
       "572769           snv      1/1            1/1          1/1        1/1   \n",
       "448731           snv      0/0            0/0          0/0        0/1   \n",
       "1347914          snv      0/0            0/0          1/1        0/0   \n",
       "1169648          snv      0/0            0/0          1/1        ./.   \n",
       "749275           snv      1/1            1/1          1/1        1/1   \n",
       "1483683          snv      0/0            0/1          0/0        0/0   \n",
       "701098           snv      0/1            0/0          0/0        0/1   \n",
       "575663           snv      0/0            0/0          0/1        0/0   \n",
       "573334           snv      0/0            0/0          0/1        0/0   \n",
       "1055487          snv      0/0            0/0          1/1        0/0   \n",
       "587259           snv      0/0            0/1          0/0        0/0   \n",
       "808374           snv      0/0            0/0          1/1        0/0   \n",
       "1476102          snv      0/0            0/0          1/1        ./.   \n",
       "852344           snv      0/0            0/0          0/1        0/0   \n",
       "656470           snv      1/1            1/1          1/1        1/1   \n",
       "\n",
       "         altai_gt_boolean  chagyrskaya_gt_boolean  denisovan_gt_boolean  \\\n",
       "549948              False                   False                 False   \n",
       "613603               True                    True                 False   \n",
       "549796              False                   False                  True   \n",
       "933419              False                   False                  True   \n",
       "1476084             False                   False                 False   \n",
       "572769               True                    True                  True   \n",
       "448731              False                   False                 False   \n",
       "1347914             False                   False                  True   \n",
       "1169648             False                   False                  True   \n",
       "749275               True                    True                  True   \n",
       "1483683             False                    True                 False   \n",
       "701098               True                   False                 False   \n",
       "575663              False                   False                  True   \n",
       "573334              False                   False                  True   \n",
       "1055487             False                   False                  True   \n",
       "587259              False                    True                 False   \n",
       "808374              False                   False                  True   \n",
       "1476102             False                   False                  True   \n",
       "852344              False                   False                  True   \n",
       "656470               True                    True                  True   \n",
       "\n",
       "         vindija_gt_boolean distribution present_in_1KG  1KG_allele_count  \\\n",
       "549948                 True      Vindija             no               NaN   \n",
       "613603                False        Other            yes             886.0   \n",
       "549796                False    Denisovan             no               NaN   \n",
       "933419                False    Denisovan             no               NaN   \n",
       "1476084                True      Vindija             no               NaN   \n",
       "572769                 True       Shared            yes            3083.0   \n",
       "448731                 True      Vindija             no               NaN   \n",
       "1347914               False    Denisovan            yes            4333.0   \n",
       "1169648               False    Denisovan            yes               1.0   \n",
       "749275                 True       Shared            yes            2725.0   \n",
       "1483683               False  Chagyrskaya            yes              24.0   \n",
       "701098                 True        Other             no               NaN   \n",
       "575663                False    Denisovan             no               NaN   \n",
       "573334                False    Denisovan             no               NaN   \n",
       "1055487               False    Denisovan            yes            2347.0   \n",
       "587259                False  Chagyrskaya             no               NaN   \n",
       "808374                False    Denisovan             no               NaN   \n",
       "1476102               False    Denisovan            yes              73.0   \n",
       "852344                False    Denisovan             no               NaN   \n",
       "656470                 True       Shared            yes               1.0   \n",
       "\n",
       "         1KG_allele_number  1KG_allele_frequency  1KG_EAS_AF  1KG_EUR_AF  \\\n",
       "549948                 NaN                   NaN         NaN         NaN   \n",
       "613603              5096.0                  0.17        0.13        0.24   \n",
       "549796                 NaN                   NaN         NaN         NaN   \n",
       "933419                 NaN                   NaN         NaN         NaN   \n",
       "1476084                NaN                   NaN         NaN         NaN   \n",
       "572769              5096.0                  0.60        0.85        0.32   \n",
       "448731                 NaN                   NaN         NaN         NaN   \n",
       "1347914             5096.0                  0.85        0.81        0.90   \n",
       "1169648             5096.0                  0.00        0.00        0.00   \n",
       "749275              5096.0                  0.53        0.64        0.56   \n",
       "1483683             5096.0                  0.00        0.00        0.00   \n",
       "701098                 NaN                   NaN         NaN         NaN   \n",
       "575663                 NaN                   NaN         NaN         NaN   \n",
       "573334                 NaN                   NaN         NaN         NaN   \n",
       "1055487             5096.0                  0.46        0.51        0.24   \n",
       "587259                 NaN                   NaN         NaN         NaN   \n",
       "808374                 NaN                   NaN         NaN         NaN   \n",
       "1476102             5096.0                  0.01        0.00        0.00   \n",
       "852344                 NaN                   NaN         NaN         NaN   \n",
       "656470              5096.0                  0.00        0.00        0.00   \n",
       "\n",
       "         1KG_AFR_AF  1KG_AMR_AF  1KG_SAS_AF  1KG_non_ASW_AFR_AF  \\\n",
       "549948          NaN         NaN         NaN                 NaN   \n",
       "613603         0.17        0.17        0.16            0.154971   \n",
       "549796          NaN         NaN         NaN                 NaN   \n",
       "933419          NaN         NaN         NaN                 NaN   \n",
       "1476084         NaN         NaN         NaN                 NaN   \n",
       "572769         0.78        0.34        0.59            0.812865   \n",
       "448731          NaN         NaN         NaN                 NaN   \n",
       "1347914        0.81        0.91        0.86            0.803119   \n",
       "1169648        0.00        0.00        0.00            0.000000   \n",
       "749275         0.36        0.56        0.62            0.359649   \n",
       "1483683        0.00        0.00        0.02            0.000000   \n",
       "701098          NaN         NaN         NaN                 NaN   \n",
       "575663          NaN         NaN         NaN                 NaN   \n",
       "573334          NaN         NaN         NaN                 NaN   \n",
       "1055487        0.62        0.37        0.48            0.642300   \n",
       "587259          NaN         NaN         NaN                 NaN   \n",
       "808374          NaN         NaN         NaN                 NaN   \n",
       "1476102        0.05        0.00        0.00            0.056530   \n",
       "852344          NaN         NaN         NaN                 NaN   \n",
       "656470         0.00        0.00        0.00            0.000000   \n",
       "\n",
       "        present_in_gnomAD  gnomAD_allele_count  gnomAD_allele_number  \\\n",
       "549948                 no                  NaN                   NaN   \n",
       "613603                 no                  NaN                   NaN   \n",
       "549796                 no                  NaN                   NaN   \n",
       "933419                 no                  NaN                   NaN   \n",
       "1476084                no                  NaN                   NaN   \n",
       "572769                yes              67770.0              142920.0   \n",
       "448731                 no                  NaN                   NaN   \n",
       "1347914               yes             124274.0              142956.0   \n",
       "1169648                no                  NaN                   NaN   \n",
       "749275                yes              74328.0              142512.0   \n",
       "1483683               yes                 57.0              143244.0   \n",
       "701098                 no                  NaN                   NaN   \n",
       "575663                 no                  NaN                   NaN   \n",
       "573334                 no                  NaN                   NaN   \n",
       "1055487               yes              52419.0              143106.0   \n",
       "587259                 no                  NaN                   NaN   \n",
       "808374                 no                  NaN                   NaN   \n",
       "1476102               yes               2365.0              143230.0   \n",
       "852344                 no                  NaN                   NaN   \n",
       "656470                yes                  2.0              143212.0   \n",
       "\n",
       "         gnomAD_allele_frequency Vernot_introgressed Vernot_ancestral_allele  \\\n",
       "549948                       NaN                  no                     NaN   \n",
       "613603                       NaN                  no                     NaN   \n",
       "549796                       NaN                  no                     NaN   \n",
       "933419                       NaN                  no                     NaN   \n",
       "1476084                      NaN                  no                     NaN   \n",
       "572769                  0.474181                  no                     NaN   \n",
       "448731                       NaN                  no                     NaN   \n",
       "1347914                 0.869316                  no                     NaN   \n",
       "1169648                      NaN                  no                     NaN   \n",
       "749275                  0.521556                  no                     NaN   \n",
       "1483683                 0.000398                  no                     NaN   \n",
       "701098                       NaN                  no                     NaN   \n",
       "575663                       NaN                  no                     NaN   \n",
       "573334                       NaN                  no                     NaN   \n",
       "1055487                 0.366295                  no                     NaN   \n",
       "587259                       NaN                  no                     NaN   \n",
       "808374                       NaN                  no                     NaN   \n",
       "1476102                 0.016512                  no                     NaN   \n",
       "852344                       NaN                  no                     NaN   \n",
       "656470                  0.000014                  no                     NaN   \n",
       "\n",
       "        Vernot_derived_allele  Vernot_ancestral_derived_code  Vernot_AFA_AF  \\\n",
       "549948                    NaN                            NaN            NaN   \n",
       "613603                    NaN                            NaN            NaN   \n",
       "549796                    NaN                            NaN            NaN   \n",
       "933419                    NaN                            NaN            NaN   \n",
       "1476084                   NaN                            NaN            NaN   \n",
       "572769                    NaN                            NaN            NaN   \n",
       "448731                    NaN                            NaN            NaN   \n",
       "1347914                   NaN                            NaN            NaN   \n",
       "1169648                   NaN                            NaN            NaN   \n",
       "749275                    NaN                            NaN            NaN   \n",
       "1483683                   NaN                            NaN            NaN   \n",
       "701098                    NaN                            NaN            NaN   \n",
       "575663                    NaN                            NaN            NaN   \n",
       "573334                    NaN                            NaN            NaN   \n",
       "1055487                   NaN                            NaN            NaN   \n",
       "587259                    NaN                            NaN            NaN   \n",
       "808374                    NaN                            NaN            NaN   \n",
       "1476102                   NaN                            NaN            NaN   \n",
       "852344                    NaN                            NaN            NaN   \n",
       "656470                    NaN                            NaN            NaN   \n",
       "\n",
       "         Vernot_AFR_AF  Vernot_AMR_AF  Vernot_EAS_AF  Vernot_EUR_AF  \\\n",
       "549948             NaN            NaN            NaN            NaN   \n",
       "613603             NaN            NaN            NaN            NaN   \n",
       "549796             NaN            NaN            NaN            NaN   \n",
       "933419             NaN            NaN            NaN            NaN   \n",
       "1476084            NaN            NaN            NaN            NaN   \n",
       "572769             NaN            NaN            NaN            NaN   \n",
       "448731             NaN            NaN            NaN            NaN   \n",
       "1347914            NaN            NaN            NaN            NaN   \n",
       "1169648            NaN            NaN            NaN            NaN   \n",
       "749275             NaN            NaN            NaN            NaN   \n",
       "1483683            NaN            NaN            NaN            NaN   \n",
       "701098             NaN            NaN            NaN            NaN   \n",
       "575663             NaN            NaN            NaN            NaN   \n",
       "573334             NaN            NaN            NaN            NaN   \n",
       "1055487            NaN            NaN            NaN            NaN   \n",
       "587259             NaN            NaN            NaN            NaN   \n",
       "808374             NaN            NaN            NaN            NaN   \n",
       "1476102            NaN            NaN            NaN            NaN   \n",
       "852344             NaN            NaN            NaN            NaN   \n",
       "656470             NaN            NaN            NaN            NaN   \n",
       "\n",
       "         Vernot_PNG_AF  Vernot_SAS_AF Vernot_Denisovan_base  \\\n",
       "549948             NaN            NaN                   NaN   \n",
       "613603             NaN            NaN                   NaN   \n",
       "549796             NaN            NaN                   NaN   \n",
       "933419             NaN            NaN                   NaN   \n",
       "1476084            NaN            NaN                   NaN   \n",
       "572769             NaN            NaN                   NaN   \n",
       "448731             NaN            NaN                   NaN   \n",
       "1347914            NaN            NaN                   NaN   \n",
       "1169648            NaN            NaN                   NaN   \n",
       "749275             NaN            NaN                   NaN   \n",
       "1483683            NaN            NaN                   NaN   \n",
       "701098             NaN            NaN                   NaN   \n",
       "575663             NaN            NaN                   NaN   \n",
       "573334             NaN            NaN                   NaN   \n",
       "1055487            NaN            NaN                   NaN   \n",
       "587259             NaN            NaN                   NaN   \n",
       "808374             NaN            NaN                   NaN   \n",
       "1476102            NaN            NaN                   NaN   \n",
       "852344             NaN            NaN                   NaN   \n",
       "656470             NaN            NaN                   NaN   \n",
       "\n",
       "        Vernot_Neanderthal_base_1 Vernot_Neanderthal_base_2  \\\n",
       "549948                        NaN                       NaN   \n",
       "613603                        NaN                       NaN   \n",
       "549796                        NaN                       NaN   \n",
       "933419                        NaN                       NaN   \n",
       "1476084                       NaN                       NaN   \n",
       "572769                        NaN                       NaN   \n",
       "448731                        NaN                       NaN   \n",
       "1347914                       NaN                       NaN   \n",
       "1169648                       NaN                       NaN   \n",
       "749275                        NaN                       NaN   \n",
       "1483683                       NaN                       NaN   \n",
       "701098                        NaN                       NaN   \n",
       "575663                        NaN                       NaN   \n",
       "573334                        NaN                       NaN   \n",
       "1055487                       NaN                       NaN   \n",
       "587259                        NaN                       NaN   \n",
       "808374                        NaN                       NaN   \n",
       "1476102                       NaN                       NaN   \n",
       "852344                        NaN                       NaN   \n",
       "656470                        NaN                       NaN   \n",
       "\n",
       "        Vernot_haplotype_tag  Vernot_introgressed_AF Browning_introgressed  \\\n",
       "549948                   NaN                     NaN                    no   \n",
       "613603                   NaN                     NaN                    no   \n",
       "549796                   NaN                     NaN                    no   \n",
       "933419                   NaN                     NaN                    no   \n",
       "1476084                  NaN                     NaN                    no   \n",
       "572769                   NaN                     NaN                    no   \n",
       "448731                   NaN                     NaN                    no   \n",
       "1347914                  NaN                     NaN                    no   \n",
       "1169648                  NaN                     NaN                    no   \n",
       "749275                   NaN                     NaN                    no   \n",
       "1483683                  NaN                     NaN                    no   \n",
       "701098                   NaN                     NaN                    no   \n",
       "575663                   NaN                     NaN                    no   \n",
       "573334                   NaN                     NaN                    no   \n",
       "1055487                  NaN                     NaN                    no   \n",
       "587259                   NaN                     NaN                    no   \n",
       "808374                   NaN                     NaN                    no   \n",
       "1476102                  NaN                     NaN                    no   \n",
       "852344                   NaN                     NaN                    no   \n",
       "656470                   NaN                     NaN                    no   \n",
       "\n",
       "         Browning_ref_alt  Browning_introgressed_AF annotation   mis_oe  \\\n",
       "549948                NaN                       NaN    PGPEP1L  1.28090   \n",
       "613603                NaN                       NaN      MYO1C  1.06990   \n",
       "549796                NaN                       NaN      IGF1R  0.73351   \n",
       "933419                NaN                       NaN      AURKA  0.68585   \n",
       "1476084               NaN                       NaN       TTI2  1.04940   \n",
       "572769                NaN                       NaN      ACSM1  0.96843   \n",
       "448731                NaN                       NaN       PCK2  1.02830   \n",
       "1347914               NaN                       NaN       VWDE  1.01570   \n",
       "1169648               NaN                       NaN       GFM2  0.88906   \n",
       "749275                NaN                       NaN      KLK12  1.07420   \n",
       "1483683               NaN                       NaN      PXDNL  1.01930   \n",
       "701098                NaN                       NaN       MBD2  0.53061   \n",
       "575663                NaN                       NaN    ZKSCAN2  0.96998   \n",
       "573334                NaN                       NaN   C16orf52  0.54356   \n",
       "1055487               NaN                       NaN    CCDC149  0.79749   \n",
       "587259                NaN                       NaN    SLC12A4  0.85544   \n",
       "808374                NaN                       NaN      TEX37  1.18320   \n",
       "1476102               NaN                       NaN       TTI2  1.04940   \n",
       "852344                NaN                       NaN    DYNC1I2  0.61314   \n",
       "656470                NaN                       NaN     KCNJ16  1.00040   \n",
       "\n",
       "            mis_z   lof_oe    lof_z  phyloP  ag_delta  al_delta  dg_delta  \\\n",
       "549948  -1.094500  1.68290 -1.54260   0.650      0.00      0.00      0.01   \n",
       "613603  -0.636630  0.51054  3.70140   0.334      0.00      0.00      0.92   \n",
       "549796   2.728600  0.18874  5.99420   0.460      0.04      0.00      0.00   \n",
       "933419   1.595000  0.15147  3.49930   0.585      0.00      0.26      0.41   \n",
       "1476084 -0.288910  0.65730  1.41230   0.650      0.00      0.01      0.02   \n",
       "572769   0.203920  0.77860  1.20820  -0.257      0.00      0.00      0.00   \n",
       "448731  -0.202710  1.04620 -0.24419   0.563      0.00      0.00      0.07   \n",
       "1347914 -0.155870  0.90845  0.71763   0.533      0.00      0.00      0.01   \n",
       "1169648  0.796580  0.77687  1.34760   0.462      0.00      0.00      0.14   \n",
       "749275  -0.329600  0.75715  0.73148   0.528      0.00      0.00      0.18   \n",
       "1483683 -0.196550  0.87676  0.92884   0.650      0.00      0.00      0.00   \n",
       "701098   2.111100  0.22818  2.99450   0.643      0.00      0.00      0.02   \n",
       "575663   0.248880  0.36273  3.92200   0.563      0.00      0.00      0.57   \n",
       "573334   1.425800  0.26140  1.89320   0.409      0.01     -0.00      0.03   \n",
       "1055487  1.178900  0.47219  2.46560  -0.257      0.00      0.00      0.13   \n",
       "587259   1.353700  0.49267  3.48020   0.462      0.00      0.00      0.00   \n",
       "808374  -0.658200  1.32610 -0.64277   0.462      0.00      0.00      0.08   \n",
       "1476102 -0.288910  0.65730  1.41230   0.655      0.00      0.00      0.45   \n",
       "852344   2.451600  0.52411  2.65520   0.467      0.01      0.00      0.00   \n",
       "656470  -0.002204  1.11810 -0.41397  -4.309      0.00      0.00      0.00   \n",
       "\n",
       "         dl_delta  delta_max  ag_pos  al_pos  dg_pos  dl_pos  \\\n",
       "549948       1.00       1.00      -3      49      -3       1   \n",
       "613603       1.00       1.00      11      19      46       2   \n",
       "549796       1.00       1.00     -20     -31     -20      -2   \n",
       "933419       1.00       1.00     -49      47     -49       1   \n",
       "1476084      1.00       1.00     -39      36      -4       1   \n",
       "572769       0.99       0.99      47       2     -25       2   \n",
       "448731       0.99       0.99      10      -1      10      -1   \n",
       "1347914      0.99       0.99     -26      34      47       2   \n",
       "1169648      0.99       0.99      45       0      37       1   \n",
       "749275       0.99       0.99     -19      28     -26       2   \n",
       "1483683      0.99       0.99      -3      22      -3       1   \n",
       "701098       0.99       0.99     -42      45       6       1   \n",
       "575663       0.99       0.99      -4       1      -4       1   \n",
       "573334       0.99       0.99     -42      -2     -10      -2   \n",
       "1055487      0.99       0.99      48       2       7       2   \n",
       "587259       0.98       0.98       5       1      -7       1   \n",
       "808374       0.98       0.98      42     -42       7      -1   \n",
       "1476102      0.98       0.98       0       9      43       1   \n",
       "852344       0.98       0.98       0      32      -2      -1   \n",
       "656470       0.98       0.98      35     -43     -49      -1   \n",
       "\n",
       "         Adipose_Subcutaneous  Adipose_Visceral_Omentum  Adrenal_Gland  \\\n",
       "549948                    NaN                       NaN            NaN   \n",
       "613603                    NaN                       NaN            NaN   \n",
       "549796                    NaN                       NaN            NaN   \n",
       "933419                    NaN                       NaN            NaN   \n",
       "1476084                   NaN                       NaN            NaN   \n",
       "572769                    0.0                       0.0            0.0   \n",
       "448731                    NaN                       NaN            NaN   \n",
       "1347914                   0.0                       0.0            0.0   \n",
       "1169648                   NaN                       NaN            NaN   \n",
       "749275                    0.0                       0.0            0.0   \n",
       "1483683                   NaN                       NaN            NaN   \n",
       "701098                    NaN                       NaN            NaN   \n",
       "575663                    NaN                       NaN            NaN   \n",
       "573334                    NaN                       NaN            NaN   \n",
       "1055487                   NaN                       NaN            NaN   \n",
       "587259                    NaN                       NaN            NaN   \n",
       "808374                    NaN                       NaN            NaN   \n",
       "1476102                   NaN                       NaN            NaN   \n",
       "852344                    NaN                       NaN            NaN   \n",
       "656470                    NaN                       NaN            NaN   \n",
       "\n",
       "         Artery_Aorta  Artery_Coronary  Artery_Tibial  Brain_Amygdala  \\\n",
       "549948            NaN              NaN            NaN             NaN   \n",
       "613603            NaN              NaN            NaN             NaN   \n",
       "549796            NaN              NaN            NaN             NaN   \n",
       "933419            NaN              NaN            NaN             NaN   \n",
       "1476084           NaN              NaN            NaN             NaN   \n",
       "572769            1.0              1.0            1.0             0.0   \n",
       "448731            NaN              NaN            NaN             NaN   \n",
       "1347914           0.0              0.0            0.0             0.0   \n",
       "1169648           NaN              NaN            NaN             NaN   \n",
       "749275            0.0              0.0            0.0             0.0   \n",
       "1483683           NaN              NaN            NaN             NaN   \n",
       "701098            NaN              NaN            NaN             NaN   \n",
       "575663            NaN              NaN            NaN             NaN   \n",
       "573334            NaN              NaN            NaN             NaN   \n",
       "1055487           NaN              NaN            NaN             NaN   \n",
       "587259            NaN              NaN            NaN             NaN   \n",
       "808374            NaN              NaN            NaN             NaN   \n",
       "1476102           NaN              NaN            NaN             NaN   \n",
       "852344            NaN              NaN            NaN             NaN   \n",
       "656470            NaN              NaN            NaN             NaN   \n",
       "\n",
       "         Brain_Anterior_cingulate_cortex_BA24  Brain_Caudate_basal_ganglia  \\\n",
       "549948                                    NaN                          NaN   \n",
       "613603                                    NaN                          NaN   \n",
       "549796                                    NaN                          NaN   \n",
       "933419                                    NaN                          NaN   \n",
       "1476084                                   NaN                          NaN   \n",
       "572769                                    0.0                          0.0   \n",
       "448731                                    NaN                          NaN   \n",
       "1347914                                   0.0                          0.0   \n",
       "1169648                                   NaN                          NaN   \n",
       "749275                                    0.0                          0.0   \n",
       "1483683                                   NaN                          NaN   \n",
       "701098                                    NaN                          NaN   \n",
       "575663                                    NaN                          NaN   \n",
       "573334                                    NaN                          NaN   \n",
       "1055487                                   NaN                          NaN   \n",
       "587259                                    NaN                          NaN   \n",
       "808374                                    NaN                          NaN   \n",
       "1476102                                   NaN                          NaN   \n",
       "852344                                    NaN                          NaN   \n",
       "656470                                    NaN                          NaN   \n",
       "\n",
       "         Brain_Cerebellar_Hemisphere  Brain_Cerebellum  Brain_Cortex  \\\n",
       "549948                           NaN               NaN           NaN   \n",
       "613603                           NaN               NaN           NaN   \n",
       "549796                           NaN               NaN           NaN   \n",
       "933419                           NaN               NaN           NaN   \n",
       "1476084                          NaN               NaN           NaN   \n",
       "572769                           0.0               0.0           0.0   \n",
       "448731                           NaN               NaN           NaN   \n",
       "1347914                          0.0               0.0           0.0   \n",
       "1169648                          NaN               NaN           NaN   \n",
       "749275                           0.0               0.0           0.0   \n",
       "1483683                          NaN               NaN           NaN   \n",
       "701098                           NaN               NaN           NaN   \n",
       "575663                           NaN               NaN           NaN   \n",
       "573334                           NaN               NaN           NaN   \n",
       "1055487                          NaN               NaN           NaN   \n",
       "587259                           NaN               NaN           NaN   \n",
       "808374                           NaN               NaN           NaN   \n",
       "1476102                          NaN               NaN           NaN   \n",
       "852344                           NaN               NaN           NaN   \n",
       "656470                           NaN               NaN           NaN   \n",
       "\n",
       "         Brain_Frontal_Cortex_BA9  Brain_Hippocampus  Brain_Hypothalamus  \\\n",
       "549948                        NaN                NaN                 NaN   \n",
       "613603                        NaN                NaN                 NaN   \n",
       "549796                        NaN                NaN                 NaN   \n",
       "933419                        NaN                NaN                 NaN   \n",
       "1476084                       NaN                NaN                 NaN   \n",
       "572769                        0.0                0.0                 0.0   \n",
       "448731                        NaN                NaN                 NaN   \n",
       "1347914                       0.0                0.0                 0.0   \n",
       "1169648                       NaN                NaN                 NaN   \n",
       "749275                        0.0                0.0                 0.0   \n",
       "1483683                       NaN                NaN                 NaN   \n",
       "701098                        NaN                NaN                 NaN   \n",
       "575663                        NaN                NaN                 NaN   \n",
       "573334                        NaN                NaN                 NaN   \n",
       "1055487                       NaN                NaN                 NaN   \n",
       "587259                        NaN                NaN                 NaN   \n",
       "808374                        NaN                NaN                 NaN   \n",
       "1476102                       NaN                NaN                 NaN   \n",
       "852344                        NaN                NaN                 NaN   \n",
       "656470                        NaN                NaN                 NaN   \n",
       "\n",
       "         Brain_Nucleus_accumbens_basal_ganglia  Brain_Putamen_basal_ganglia  \\\n",
       "549948                                     NaN                          NaN   \n",
       "613603                                     NaN                          NaN   \n",
       "549796                                     NaN                          NaN   \n",
       "933419                                     NaN                          NaN   \n",
       "1476084                                    NaN                          NaN   \n",
       "572769                                     0.0                          0.0   \n",
       "448731                                     NaN                          NaN   \n",
       "1347914                                    0.0                          0.0   \n",
       "1169648                                    NaN                          NaN   \n",
       "749275                                     0.0                          0.0   \n",
       "1483683                                    NaN                          NaN   \n",
       "701098                                     NaN                          NaN   \n",
       "575663                                     NaN                          NaN   \n",
       "573334                                     NaN                          NaN   \n",
       "1055487                                    NaN                          NaN   \n",
       "587259                                     NaN                          NaN   \n",
       "808374                                     NaN                          NaN   \n",
       "1476102                                    NaN                          NaN   \n",
       "852344                                     NaN                          NaN   \n",
       "656470                                     NaN                          NaN   \n",
       "\n",
       "         Brain_Spinal_cord_cervical_c-1  Brain_Substantia_nigra  \\\n",
       "549948                              NaN                     NaN   \n",
       "613603                              NaN                     NaN   \n",
       "549796                              NaN                     NaN   \n",
       "933419                              NaN                     NaN   \n",
       "1476084                             NaN                     NaN   \n",
       "572769                              0.0                     0.0   \n",
       "448731                              NaN                     NaN   \n",
       "1347914                             0.0                     0.0   \n",
       "1169648                             NaN                     NaN   \n",
       "749275                              0.0                     0.0   \n",
       "1483683                             NaN                     NaN   \n",
       "701098                              NaN                     NaN   \n",
       "575663                              NaN                     NaN   \n",
       "573334                              NaN                     NaN   \n",
       "1055487                             NaN                     NaN   \n",
       "587259                              NaN                     NaN   \n",
       "808374                              NaN                     NaN   \n",
       "1476102                             NaN                     NaN   \n",
       "852344                              NaN                     NaN   \n",
       "656470                              NaN                     NaN   \n",
       "\n",
       "         Breast_Mammary_Tissue  Cells_Cultured_fibroblasts  \\\n",
       "549948                     NaN                         NaN   \n",
       "613603                     NaN                         NaN   \n",
       "549796                     NaN                         NaN   \n",
       "933419                     NaN                         NaN   \n",
       "1476084                    NaN                         NaN   \n",
       "572769                     0.0                         0.0   \n",
       "448731                     NaN                         NaN   \n",
       "1347914                    0.0                         0.0   \n",
       "1169648                    NaN                         NaN   \n",
       "749275                     0.0                         0.0   \n",
       "1483683                    NaN                         NaN   \n",
       "701098                     NaN                         NaN   \n",
       "575663                     NaN                         NaN   \n",
       "573334                     NaN                         NaN   \n",
       "1055487                    NaN                         NaN   \n",
       "587259                     NaN                         NaN   \n",
       "808374                     NaN                         NaN   \n",
       "1476102                    NaN                         NaN   \n",
       "852344                     NaN                         NaN   \n",
       "656470                     NaN                         NaN   \n",
       "\n",
       "         Cells_EBV-transformed_lymphocytes  Colon_Sigmoid  Colon_Transverse  \\\n",
       "549948                                 NaN            NaN               NaN   \n",
       "613603                                 NaN            NaN               NaN   \n",
       "549796                                 NaN            NaN               NaN   \n",
       "933419                                 NaN            NaN               NaN   \n",
       "1476084                                NaN            NaN               NaN   \n",
       "572769                                 0.0            0.0               1.0   \n",
       "448731                                 NaN            NaN               NaN   \n",
       "1347914                                0.0            0.0               0.0   \n",
       "1169648                                NaN            NaN               NaN   \n",
       "749275                                 0.0            0.0               0.0   \n",
       "1483683                                NaN            NaN               NaN   \n",
       "701098                                 NaN            NaN               NaN   \n",
       "575663                                 NaN            NaN               NaN   \n",
       "573334                                 NaN            NaN               NaN   \n",
       "1055487                                NaN            NaN               NaN   \n",
       "587259                                 NaN            NaN               NaN   \n",
       "808374                                 NaN            NaN               NaN   \n",
       "1476102                                NaN            NaN               NaN   \n",
       "852344                                 NaN            NaN               NaN   \n",
       "656470                                 NaN            NaN               NaN   \n",
       "\n",
       "         Esophagus_Gastroesophageal_Junction  Esophagus_Mucosa  \\\n",
       "549948                                   NaN               NaN   \n",
       "613603                                   NaN               NaN   \n",
       "549796                                   NaN               NaN   \n",
       "933419                                   NaN               NaN   \n",
       "1476084                                  NaN               NaN   \n",
       "572769                                   0.0               0.0   \n",
       "448731                                   NaN               NaN   \n",
       "1347914                                  0.0               0.0   \n",
       "1169648                                  NaN               NaN   \n",
       "749275                                   0.0               1.0   \n",
       "1483683                                  NaN               NaN   \n",
       "701098                                   NaN               NaN   \n",
       "575663                                   NaN               NaN   \n",
       "573334                                   NaN               NaN   \n",
       "1055487                                  NaN               NaN   \n",
       "587259                                   NaN               NaN   \n",
       "808374                                   NaN               NaN   \n",
       "1476102                                  NaN               NaN   \n",
       "852344                                   NaN               NaN   \n",
       "656470                                   NaN               NaN   \n",
       "\n",
       "         Esophagus_Muscularis  Heart_Atrial_Appendage  Heart_Left_Ventricle  \\\n",
       "549948                    NaN                     NaN                   NaN   \n",
       "613603                    NaN                     NaN                   NaN   \n",
       "549796                    NaN                     NaN                   NaN   \n",
       "933419                    NaN                     NaN                   NaN   \n",
       "1476084                   NaN                     NaN                   NaN   \n",
       "572769                    0.0                     0.0                   0.0   \n",
       "448731                    NaN                     NaN                   NaN   \n",
       "1347914                   0.0                     1.0                   0.0   \n",
       "1169648                   NaN                     NaN                   NaN   \n",
       "749275                    0.0                     0.0                   0.0   \n",
       "1483683                   NaN                     NaN                   NaN   \n",
       "701098                    NaN                     NaN                   NaN   \n",
       "575663                    NaN                     NaN                   NaN   \n",
       "573334                    NaN                     NaN                   NaN   \n",
       "1055487                   NaN                     NaN                   NaN   \n",
       "587259                    NaN                     NaN                   NaN   \n",
       "808374                    NaN                     NaN                   NaN   \n",
       "1476102                   NaN                     NaN                   NaN   \n",
       "852344                    NaN                     NaN                   NaN   \n",
       "656470                    NaN                     NaN                   NaN   \n",
       "\n",
       "         Kidney_Cortex  Liver  Lung  Minor_Salivary_Gland  Muscle_Skeletal  \\\n",
       "549948             NaN    NaN   NaN                   NaN              NaN   \n",
       "613603             NaN    NaN   NaN                   NaN              NaN   \n",
       "549796             NaN    NaN   NaN                   NaN              NaN   \n",
       "933419             NaN    NaN   NaN                   NaN              NaN   \n",
       "1476084            NaN    NaN   NaN                   NaN              NaN   \n",
       "572769             0.0    1.0   0.0                   1.0              0.0   \n",
       "448731             NaN    NaN   NaN                   NaN              NaN   \n",
       "1347914            0.0    0.0   0.0                   0.0              0.0   \n",
       "1169648            NaN    NaN   NaN                   NaN              NaN   \n",
       "749275             0.0    0.0   0.0                   1.0              0.0   \n",
       "1483683            NaN    NaN   NaN                   NaN              NaN   \n",
       "701098             NaN    NaN   NaN                   NaN              NaN   \n",
       "575663             NaN    NaN   NaN                   NaN              NaN   \n",
       "573334             NaN    NaN   NaN                   NaN              NaN   \n",
       "1055487            NaN    NaN   NaN                   NaN              NaN   \n",
       "587259             NaN    NaN   NaN                   NaN              NaN   \n",
       "808374             NaN    NaN   NaN                   NaN              NaN   \n",
       "1476102            NaN    NaN   NaN                   NaN              NaN   \n",
       "852344             NaN    NaN   NaN                   NaN              NaN   \n",
       "656470             NaN    NaN   NaN                   NaN              NaN   \n",
       "\n",
       "         Nerve_Tibial  Ovary  Pancreas  Pituitary  Prostate  \\\n",
       "549948            NaN    NaN       NaN        NaN       NaN   \n",
       "613603            NaN    NaN       NaN        NaN       NaN   \n",
       "549796            NaN    NaN       NaN        NaN       NaN   \n",
       "933419            NaN    NaN       NaN        NaN       NaN   \n",
       "1476084           NaN    NaN       NaN        NaN       NaN   \n",
       "572769            0.0    1.0       1.0        0.0       1.0   \n",
       "448731            NaN    NaN       NaN        NaN       NaN   \n",
       "1347914           0.0    0.0       0.0        1.0       1.0   \n",
       "1169648           NaN    NaN       NaN        NaN       NaN   \n",
       "749275            0.0    0.0       0.0        1.0       0.0   \n",
       "1483683           NaN    NaN       NaN        NaN       NaN   \n",
       "701098            NaN    NaN       NaN        NaN       NaN   \n",
       "575663            NaN    NaN       NaN        NaN       NaN   \n",
       "573334            NaN    NaN       NaN        NaN       NaN   \n",
       "1055487           NaN    NaN       NaN        NaN       NaN   \n",
       "587259            NaN    NaN       NaN        NaN       NaN   \n",
       "808374            NaN    NaN       NaN        NaN       NaN   \n",
       "1476102           NaN    NaN       NaN        NaN       NaN   \n",
       "852344            NaN    NaN       NaN        NaN       NaN   \n",
       "656470            NaN    NaN       NaN        NaN       NaN   \n",
       "\n",
       "         Skin_Not_Sun_Exposed_Suprapubic  Skin_Sun_Exposed_Lower_leg  \\\n",
       "549948                               NaN                         NaN   \n",
       "613603                               NaN                         NaN   \n",
       "549796                               NaN                         NaN   \n",
       "933419                               NaN                         NaN   \n",
       "1476084                              NaN                         NaN   \n",
       "572769                               1.0                         1.0   \n",
       "448731                               NaN                         NaN   \n",
       "1347914                              0.0                         0.0   \n",
       "1169648                              NaN                         NaN   \n",
       "749275                               0.0                         1.0   \n",
       "1483683                              NaN                         NaN   \n",
       "701098                               NaN                         NaN   \n",
       "575663                               NaN                         NaN   \n",
       "573334                               NaN                         NaN   \n",
       "1055487                              NaN                         NaN   \n",
       "587259                               NaN                         NaN   \n",
       "808374                               NaN                         NaN   \n",
       "1476102                              NaN                         NaN   \n",
       "852344                               NaN                         NaN   \n",
       "656470                               NaN                         NaN   \n",
       "\n",
       "         Small_Intestine_Terminal_Ileum  Spleen  Stomach  Testis  Thyroid  \\\n",
       "549948                              NaN     NaN      NaN     NaN      NaN   \n",
       "613603                              NaN     NaN      NaN     NaN      NaN   \n",
       "549796                              NaN     NaN      NaN     NaN      NaN   \n",
       "933419                              NaN     NaN      NaN     NaN      NaN   \n",
       "1476084                             NaN     NaN      NaN     NaN      NaN   \n",
       "572769                              1.0     0.0      1.0     1.0      0.0   \n",
       "448731                              NaN     NaN      NaN     NaN      NaN   \n",
       "1347914                             0.0     0.0      0.0     0.0      1.0   \n",
       "1169648                             NaN     NaN      NaN     NaN      NaN   \n",
       "749275                              1.0     0.0      0.0     1.0      0.0   \n",
       "1483683                             NaN     NaN      NaN     NaN      NaN   \n",
       "701098                              NaN     NaN      NaN     NaN      NaN   \n",
       "575663                              NaN     NaN      NaN     NaN      NaN   \n",
       "573334                              NaN     NaN      NaN     NaN      NaN   \n",
       "1055487                             NaN     NaN      NaN     NaN      NaN   \n",
       "587259                              NaN     NaN      NaN     NaN      NaN   \n",
       "808374                              NaN     NaN      NaN     NaN      NaN   \n",
       "1476102                             NaN     NaN      NaN     NaN      NaN   \n",
       "852344                              NaN     NaN      NaN     NaN      NaN   \n",
       "656470                              NaN     NaN      NaN     NaN      NaN   \n",
       "\n",
       "         Uterus  Vagina  Whole_Blood  N_GTEx_tissues sQTL  \\\n",
       "549948      NaN     NaN          NaN             NaN   no   \n",
       "613603      NaN     NaN          NaN             NaN   no   \n",
       "549796      NaN     NaN          NaN             NaN   no   \n",
       "933419      NaN     NaN          NaN             NaN   no   \n",
       "1476084     NaN     NaN          NaN             NaN   no   \n",
       "572769      1.0     1.0          0.0            16.0  yes   \n",
       "448731      NaN     NaN          NaN             NaN   no   \n",
       "1347914     0.0     0.0          0.0             4.0  yes   \n",
       "1169648     NaN     NaN          NaN             NaN   no   \n",
       "749275      0.0     1.0          0.0             7.0  yes   \n",
       "1483683     NaN     NaN          NaN             NaN   no   \n",
       "701098      NaN     NaN          NaN             NaN   no   \n",
       "575663      NaN     NaN          NaN             NaN   no   \n",
       "573334      NaN     NaN          NaN             NaN   no   \n",
       "1055487     NaN     NaN          NaN             NaN   no   \n",
       "587259      NaN     NaN          NaN             NaN   no   \n",
       "808374      NaN     NaN          NaN             NaN   no   \n",
       "1476102     NaN     NaN          NaN             NaN   no   \n",
       "852344      NaN     NaN          NaN             NaN   no   \n",
       "656470      NaN     NaN          NaN             NaN   no   \n",
       "\n",
       "           Vernot_allele_origin  Browning_allele_origin  \n",
       "549948         archaic-specific        archaic-specific  \n",
       "613603                  ancient                 ancient  \n",
       "549796         archaic-specific        archaic-specific  \n",
       "933419         archaic-specific        archaic-specific  \n",
       "1476084        archaic-specific        archaic-specific  \n",
       "572769                  ancient                 ancient  \n",
       "448731         archaic-specific        archaic-specific  \n",
       "1347914                 ancient                 ancient  \n",
       "1169648  low-confidence ancient  low-confidence ancient  \n",
       "749275                  ancient                 ancient  \n",
       "1483683  low-confidence ancient  low-confidence ancient  \n",
       "701098         archaic-specific        archaic-specific  \n",
       "575663         archaic-specific        archaic-specific  \n",
       "573334         archaic-specific        archaic-specific  \n",
       "1055487                 ancient                 ancient  \n",
       "587259         archaic-specific        archaic-specific  \n",
       "808374         archaic-specific        archaic-specific  \n",
       "1476102  low-confidence ancient  low-confidence ancient  \n",
       "852344         archaic-specific        archaic-specific  \n",
       "656470   low-confidence ancient  low-confidence ancient  "
      ]
     },
     "execution_count": 181,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_2_dl = data_2.sort_values(['dl_delta'], ascending = [False])\n",
    "data_2_dl.head(20)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "41fcaf05",
   "metadata": {},
   "source": [
    "## N Genes <a class = 'anchor' id = 'ngenes'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "65ef12c4",
   "metadata": {},
   "source": [
    "Now let's get a list of genes for all variants with delta >= 0.2."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 182,
   "id": "69bed403",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>755832</th>\n",
       "      <td>A1BG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11555</th>\n",
       "      <td>AADACL4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1127491</th>\n",
       "      <td>AADAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>879182</th>\n",
       "      <td>AAMP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>879179</th>\n",
       "      <td>AAMP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>920210</th>\n",
       "      <td>AAR2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1263240</th>\n",
       "      <td>AARS2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>636461</th>\n",
       "      <td>AATF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>666602</th>\n",
       "      <td>AATK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>562265</th>\n",
       "      <td>ABAT</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            gene\n",
       "755832      A1BG\n",
       "11555    AADACL4\n",
       "1127491    AADAT\n",
       "879182      AAMP\n",
       "879179      AAMP\n",
       "920210      AAR2\n",
       "1263240    AARS2\n",
       "636461      AATF\n",
       "666602      AATK\n",
       "562265      ABAT"
      ]
     },
     "execution_count": 182,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_genes_2 = data_2['annotation'].to_frame('gene')\n",
    "all_genes_2 = all_genes_2.sort_values(['gene'], ascending = [True])\n",
    "all_genes_2.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 183,
   "id": "4d91e103",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "5950"
      ]
     },
     "execution_count": 183,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(all_genes_2)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "af93d893",
   "metadata": {},
   "source": [
    "How many unique genes are contained in these vectors?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 184,
   "id": "2295d567",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "4242"
      ]
     },
     "execution_count": 184,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(all_genes_2.drop_duplicates())"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d9fd4b7e",
   "metadata": {},
   "source": [
    "Save this gene list."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 185,
   "id": "2b189af4",
   "metadata": {},
   "outputs": [],
   "source": [
    "all_genes_2.to_csv('../genes/all_genes_2.txt', sep=\"\\t\", header=False, index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "aa588a59",
   "metadata": {},
   "source": [
    "Repeat for delta >= 0.5."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 186,
   "id": "95505c70",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>879182</th>\n",
       "      <td>AAMP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>920210</th>\n",
       "      <td>AAR2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>636461</th>\n",
       "      <td>AATF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>656165</th>\n",
       "      <td>ABCA5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>715609</th>\n",
       "      <td>ABCA7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1433756</th>\n",
       "      <td>ABCB8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>570288</th>\n",
       "      <td>ABCC1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>570362</th>\n",
       "      <td>ABCC1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>579405</th>\n",
       "      <td>ABCC12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>570587</th>\n",
       "      <td>ABCC6</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           gene\n",
       "879182     AAMP\n",
       "920210     AAR2\n",
       "636461     AATF\n",
       "656165    ABCA5\n",
       "715609    ABCA7\n",
       "1433756   ABCB8\n",
       "570288    ABCC1\n",
       "570362    ABCC1\n",
       "579405   ABCC12\n",
       "570587    ABCC6"
      ]
     },
     "execution_count": 186,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_genes_5 = data_5['annotation'].to_frame('gene')\n",
    "all_genes_5 = all_genes_5.sort_values(['gene'], ascending = [True])\n",
    "all_genes_5.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 187,
   "id": "fd1af765",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1049"
      ]
     },
     "execution_count": 187,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(all_genes_5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 188,
   "id": "44ed636b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "962"
      ]
     },
     "execution_count": 188,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(all_genes_5.drop_duplicates())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 189,
   "id": "289335d0",
   "metadata": {},
   "outputs": [],
   "source": [
    "all_genes_5.to_csv('../genes/all_genes_5.txt', sep=\"\\t\", header=False, index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5c9c34e0",
   "metadata": {},
   "source": [
    "## Distribution of N Genes <a class = 'anchor' id = 'ngenedistribution'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f3ed5648",
   "metadata": {},
   "source": [
    "Let's get the distribution of N SAVs per gene."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 190,
   "id": "e486d678",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>N</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>N_SAVs</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3111</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>769</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>246</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>71</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           N\n",
       "N_SAVs      \n",
       "1       3111\n",
       "2        769\n",
       "3        246\n",
       "4         71\n",
       "5         19\n",
       "6         14\n",
       "7          6\n",
       "8          1\n",
       "9          2\n",
       "10         1\n",
       "11         2"
      ]
     },
     "execution_count": 190,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SAVs_per_gene = data_2.groupby(by=['annotation']).size().reset_index(name='N_SAVs')\n",
    "SAVs_per_gene.groupby('N_SAVs').size().to_frame('N')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 191,
   "id": "890e90cc",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>annotation</th>\n",
       "      <th>N_SAVs</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>682</th>\n",
       "      <td>CDH13</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>686</th>\n",
       "      <td>CDH23</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>907</th>\n",
       "      <td>CSMD1</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1059</th>\n",
       "      <td>DNAH17</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1662</th>\n",
       "      <td>HLA-DPA1</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1942</th>\n",
       "      <td>LAMA5</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     annotation  N_SAVs\n",
       "682       CDH13       7\n",
       "686       CDH23       7\n",
       "907       CSMD1       7\n",
       "1059     DNAH17       7\n",
       "1662   HLA-DPA1       7\n",
       "1942      LAMA5       7"
      ]
     },
     "execution_count": 191,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SAVs_per_gene[SAVs_per_gene['N_SAVs']==7]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 192,
   "id": "5bc096aa",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>annotation</th>\n",
       "      <th>N_SAVs</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>108</th>\n",
       "      <td>ADARB2</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    annotation  N_SAVs\n",
       "108     ADARB2       8"
      ]
     },
     "execution_count": 192,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SAVs_per_gene[SAVs_per_gene['N_SAVs']==8]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 193,
   "id": "30fdc695",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>annotation</th>\n",
       "      <th>N_SAVs</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3198</th>\n",
       "      <td>SDK1</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4093</th>\n",
       "      <td>WWOX</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     annotation  N_SAVs\n",
       "3198       SDK1       9\n",
       "4093       WWOX       9"
      ]
     },
     "execution_count": 193,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SAVs_per_gene[SAVs_per_gene['N_SAVs']==9]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 194,
   "id": "406bea5f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>annotation</th>\n",
       "      <th>N_SAVs</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1663</th>\n",
       "      <td>HLA-DPB1</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     annotation  N_SAVs\n",
       "1663   HLA-DPB1      10"
      ]
     },
     "execution_count": 194,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SAVs_per_gene[SAVs_per_gene['N_SAVs']==10]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 195,
   "id": "c324b267",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>annotation</th>\n",
       "      <th>N_SAVs</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>829</th>\n",
       "      <td>CNTNAP2</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3508</th>\n",
       "      <td>SSPO</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     annotation  N_SAVs\n",
       "829     CNTNAP2      11\n",
       "3508       SSPO      11"
      ]
     },
     "execution_count": 195,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SAVs_per_gene[SAVs_per_gene['N_SAVs']==11]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e6640500",
   "metadata": {},
   "source": [
    "Let's take a closer look at the genes with multiple SAVs. "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "40a2fe9c",
   "metadata": {},
   "source": [
    "How often do genes with >= 2 SAVs match in distribution or allele origin?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 196,
   "id": "80726960",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref_allele</th>\n",
       "      <th>alt_allele</th>\n",
       "      <th>ancestral_allele</th>\n",
       "      <th>anc_dev</th>\n",
       "      <th>variant_type</th>\n",
       "      <th>altai_gt</th>\n",
       "      <th>chagyrskaya_gt</th>\n",
       "      <th>denisovan_gt</th>\n",
       "      <th>vindija_gt</th>\n",
       "      <th>altai_gt_boolean</th>\n",
       "      <th>chagyrskaya_gt_boolean</th>\n",
       "      <th>denisovan_gt_boolean</th>\n",
       "      <th>vindija_gt_boolean</th>\n",
       "      <th>distribution</th>\n",
       "      <th>present_in_1KG</th>\n",
       "      <th>1KG_allele_count</th>\n",
       "      <th>1KG_allele_number</th>\n",
       "      <th>1KG_allele_frequency</th>\n",
       "      <th>1KG_EAS_AF</th>\n",
       "      <th>1KG_EUR_AF</th>\n",
       "      <th>1KG_AFR_AF</th>\n",
       "      <th>1KG_AMR_AF</th>\n",
       "      <th>1KG_SAS_AF</th>\n",
       "      <th>1KG_non_ASW_AFR_AF</th>\n",
       "      <th>present_in_gnomAD</th>\n",
       "      <th>gnomAD_allele_count</th>\n",
       "      <th>gnomAD_allele_number</th>\n",
       "      <th>gnomAD_allele_frequency</th>\n",
       "      <th>Vernot_introgressed</th>\n",
       "      <th>Vernot_ancestral_allele</th>\n",
       "      <th>Vernot_derived_allele</th>\n",
       "      <th>Vernot_ancestral_derived_code</th>\n",
       "      <th>Vernot_AFA_AF</th>\n",
       "      <th>Vernot_AFR_AF</th>\n",
       "      <th>Vernot_AMR_AF</th>\n",
       "      <th>Vernot_EAS_AF</th>\n",
       "      <th>Vernot_EUR_AF</th>\n",
       "      <th>Vernot_PNG_AF</th>\n",
       "      <th>Vernot_SAS_AF</th>\n",
       "      <th>Vernot_Denisovan_base</th>\n",
       "      <th>Vernot_Neanderthal_base_1</th>\n",
       "      <th>Vernot_Neanderthal_base_2</th>\n",
       "      <th>Vernot_haplotype_tag</th>\n",
       "      <th>Vernot_introgressed_AF</th>\n",
       "      <th>Browning_introgressed</th>\n",
       "      <th>Browning_ref_alt</th>\n",
       "      <th>Browning_introgressed_AF</th>\n",
       "      <th>annotation</th>\n",
       "      <th>mis_oe</th>\n",
       "      <th>mis_z</th>\n",
       "      <th>lof_oe</th>\n",
       "      <th>lof_z</th>\n",
       "      <th>phyloP</th>\n",
       "      <th>ag_delta</th>\n",
       "      <th>al_delta</th>\n",
       "      <th>dg_delta</th>\n",
       "      <th>dl_delta</th>\n",
       "      <th>delta_max</th>\n",
       "      <th>ag_pos</th>\n",
       "      <th>al_pos</th>\n",
       "      <th>dg_pos</th>\n",
       "      <th>dl_pos</th>\n",
       "      <th>Adipose_Subcutaneous</th>\n",
       "      <th>Adipose_Visceral_Omentum</th>\n",
       "      <th>Adrenal_Gland</th>\n",
       "      <th>Artery_Aorta</th>\n",
       "      <th>Artery_Coronary</th>\n",
       "      <th>Artery_Tibial</th>\n",
       "      <th>Brain_Amygdala</th>\n",
       "      <th>Brain_Anterior_cingulate_cortex_BA24</th>\n",
       "      <th>Brain_Caudate_basal_ganglia</th>\n",
       "      <th>Brain_Cerebellar_Hemisphere</th>\n",
       "      <th>Brain_Cerebellum</th>\n",
       "      <th>Brain_Cortex</th>\n",
       "      <th>Brain_Frontal_Cortex_BA9</th>\n",
       "      <th>Brain_Hippocampus</th>\n",
       "      <th>Brain_Hypothalamus</th>\n",
       "      <th>Brain_Nucleus_accumbens_basal_ganglia</th>\n",
       "      <th>Brain_Putamen_basal_ganglia</th>\n",
       "      <th>Brain_Spinal_cord_cervical_c-1</th>\n",
       "      <th>Brain_Substantia_nigra</th>\n",
       "      <th>Breast_Mammary_Tissue</th>\n",
       "      <th>Cells_Cultured_fibroblasts</th>\n",
       "      <th>Cells_EBV-transformed_lymphocytes</th>\n",
       "      <th>Colon_Sigmoid</th>\n",
       "      <th>Colon_Transverse</th>\n",
       "      <th>Esophagus_Gastroesophageal_Junction</th>\n",
       "      <th>Esophagus_Mucosa</th>\n",
       "      <th>Esophagus_Muscularis</th>\n",
       "      <th>Heart_Atrial_Appendage</th>\n",
       "      <th>Heart_Left_Ventricle</th>\n",
       "      <th>Kidney_Cortex</th>\n",
       "      <th>Liver</th>\n",
       "      <th>Lung</th>\n",
       "      <th>Minor_Salivary_Gland</th>\n",
       "      <th>Muscle_Skeletal</th>\n",
       "      <th>Nerve_Tibial</th>\n",
       "      <th>Ovary</th>\n",
       "      <th>Pancreas</th>\n",
       "      <th>Pituitary</th>\n",
       "      <th>Prostate</th>\n",
       "      <th>Skin_Not_Sun_Exposed_Suprapubic</th>\n",
       "      <th>Skin_Sun_Exposed_Lower_leg</th>\n",
       "      <th>Small_Intestine_Terminal_Ileum</th>\n",
       "      <th>Spleen</th>\n",
       "      <th>Stomach</th>\n",
       "      <th>Testis</th>\n",
       "      <th>Thyroid</th>\n",
       "      <th>Uterus</th>\n",
       "      <th>Vagina</th>\n",
       "      <th>Whole_Blood</th>\n",
       "      <th>N_GTEx_tissues</th>\n",
       "      <th>sQTL</th>\n",
       "      <th>Vernot_allele_origin</th>\n",
       "      <th>Browning_allele_origin</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>224</th>\n",
       "      <td>chr1</td>\n",
       "      <td>982099</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>1.0</td>\n",
       "      <td>142512.0</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>0.98257</td>\n",
       "      <td>0.225520</td>\n",
       "      <td>0.319140</td>\n",
       "      <td>6.0143</td>\n",
       "      <td>-0.375</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.04</td>\n",
       "      <td>0.48</td>\n",
       "      <td>0.48</td>\n",
       "      <td>-48</td>\n",
       "      <td>-6</td>\n",
       "      <td>16</td>\n",
       "      <td>-6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1176</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1872053</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>7.0</td>\n",
       "      <td>143266.0</td>\n",
       "      <td>0.000049</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.000049</td>\n",
       "      <td>CFAP74</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.669</td>\n",
       "      <td>0.22</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.22</td>\n",
       "      <td>49</td>\n",
       "      <td>-2</td>\n",
       "      <td>49</td>\n",
       "      <td>-2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1889</th>\n",
       "      <td>chr1</td>\n",
       "      <td>2526280</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>2.0</td>\n",
       "      <td>143228.0</td>\n",
       "      <td>0.000014</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MMEL1</td>\n",
       "      <td>0.98394</td>\n",
       "      <td>0.126020</td>\n",
       "      <td>0.768240</td>\n",
       "      <td>1.5689</td>\n",
       "      <td>-0.242</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.03</td>\n",
       "      <td>0.25</td>\n",
       "      <td>0.25</td>\n",
       "      <td>-37</td>\n",
       "      <td>39</td>\n",
       "      <td>-37</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3354</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3520127</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>ancestral</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>yes</td>\n",
       "      <td>4811.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.94</td>\n",
       "      <td>0.98</td>\n",
       "      <td>0.97</td>\n",
       "      <td>0.86</td>\n",
       "      <td>0.99</td>\n",
       "      <td>0.95</td>\n",
       "      <td>0.855750</td>\n",
       "      <td>yes</td>\n",
       "      <td>134537.0</td>\n",
       "      <td>143196.0</td>\n",
       "      <td>0.939530</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MEGF6</td>\n",
       "      <td>0.95847</td>\n",
       "      <td>0.458030</td>\n",
       "      <td>0.738560</td>\n",
       "      <td>2.2728</td>\n",
       "      <td>-1.183</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.24</td>\n",
       "      <td>0.24</td>\n",
       "      <td>-28</td>\n",
       "      <td>47</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>ancient</td>\n",
       "      <td>ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7121</th>\n",
       "      <td>chr1</td>\n",
       "      <td>7731131</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CAMTA1</td>\n",
       "      <td>0.71229</td>\n",
       "      <td>3.261900</td>\n",
       "      <td>0.091344</td>\n",
       "      <td>7.3710</td>\n",
       "      <td>0.655</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.71</td>\n",
       "      <td>0.71</td>\n",
       "      <td>48</td>\n",
       "      <td>-5</td>\n",
       "      <td>-34</td>\n",
       "      <td>-5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15321</th>\n",
       "      <td>chr1</td>\n",
       "      <td>17715723</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>695.0</td>\n",
       "      <td>143346.0</td>\n",
       "      <td>0.004848</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PADI6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.561</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.30</td>\n",
       "      <td>0.30</td>\n",
       "      <td>3</td>\n",
       "      <td>-5</td>\n",
       "      <td>16</td>\n",
       "      <td>-5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21905</th>\n",
       "      <td>chr1</td>\n",
       "      <td>26359246</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>ancestral</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>yes</td>\n",
       "      <td>4456.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.87</td>\n",
       "      <td>0.93</td>\n",
       "      <td>0.89</td>\n",
       "      <td>0.81</td>\n",
       "      <td>0.93</td>\n",
       "      <td>0.83</td>\n",
       "      <td>0.810916</td>\n",
       "      <td>yes</td>\n",
       "      <td>125748.0</td>\n",
       "      <td>143146.0</td>\n",
       "      <td>0.878460</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>EXTL1</td>\n",
       "      <td>0.87054</td>\n",
       "      <td>0.903800</td>\n",
       "      <td>0.751620</td>\n",
       "      <td>1.2166</td>\n",
       "      <td>-0.527</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.04</td>\n",
       "      <td>0.61</td>\n",
       "      <td>0.61</td>\n",
       "      <td>-50</td>\n",
       "      <td>-2</td>\n",
       "      <td>10</td>\n",
       "      <td>-2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>ancient</td>\n",
       "      <td>ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22075</th>\n",
       "      <td>chr1</td>\n",
       "      <td>26584637</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>yes</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>yes</td>\n",
       "      <td>30.0</td>\n",
       "      <td>143294.0</td>\n",
       "      <td>0.000209</td>\n",
       "      <td>yes</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00099</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr1_26583921_26646198</td>\n",
       "      <td>0.000198</td>\n",
       "      <td>yes</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>CEP85</td>\n",
       "      <td>0.96911</td>\n",
       "      <td>0.224110</td>\n",
       "      <td>0.338850</td>\n",
       "      <td>3.9380</td>\n",
       "      <td>0.563</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.26</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.26</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>-4</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>introgressed</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34032</th>\n",
       "      <td>chr1</td>\n",
       "      <td>46522577</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>ancestral</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>yes</td>\n",
       "      <td>3241.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.64</td>\n",
       "      <td>0.68</td>\n",
       "      <td>0.73</td>\n",
       "      <td>0.59</td>\n",
       "      <td>0.62</td>\n",
       "      <td>0.56</td>\n",
       "      <td>0.583821</td>\n",
       "      <td>yes</td>\n",
       "      <td>96369.0</td>\n",
       "      <td>143076.0</td>\n",
       "      <td>0.673551</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PIK3R3</td>\n",
       "      <td>0.99209</td>\n",
       "      <td>0.044445</td>\n",
       "      <td>0.684270</td>\n",
       "      <td>1.5417</td>\n",
       "      <td>-2.533</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.43</td>\n",
       "      <td>0.04</td>\n",
       "      <td>0.43</td>\n",
       "      <td>22</td>\n",
       "      <td>-4</td>\n",
       "      <td>0</td>\n",
       "      <td>-4</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>19.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>ancient</td>\n",
       "      <td>ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34364</th>\n",
       "      <td>chr1</td>\n",
       "      <td>46877284</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>ancestral</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>yes</td>\n",
       "      <td>42.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.004873</td>\n",
       "      <td>yes</td>\n",
       "      <td>215.0</td>\n",
       "      <td>143268.0</td>\n",
       "      <td>0.001501</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.010000</td>\n",
       "      <td>FAAH</td>\n",
       "      <td>0.89767</td>\n",
       "      <td>0.637630</td>\n",
       "      <td>0.725440</td>\n",
       "      <td>1.3359</td>\n",
       "      <td>-0.121</td>\n",
       "      <td>0.28</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.03</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.28</td>\n",
       "      <td>0</td>\n",
       "      <td>-49</td>\n",
       "      <td>39</td>\n",
       "      <td>-24</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "      <td>introgressed</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      chrom       pos ref_allele alt_allele ancestral_allele    anc_dev  \\\n",
       "224    chr1    982099          T          C                T    derived   \n",
       "1176   chr1   1872053          T          C                T    derived   \n",
       "1889   chr1   2526280          A          G                A    derived   \n",
       "3354   chr1   3520127          T          C                C  ancestral   \n",
       "7121   chr1   7731131          G          A                G    derived   \n",
       "15321  chr1  17715723          G          C                G    derived   \n",
       "21905  chr1  26359246          T          A                A  ancestral   \n",
       "22075  chr1  26584637          G          A                G    derived   \n",
       "34032  chr1  46522577          A          C                C  ancestral   \n",
       "34364  chr1  46877284          T          G                G  ancestral   \n",
       "\n",
       "      variant_type altai_gt chagyrskaya_gt denisovan_gt vindija_gt  \\\n",
       "224            snv      1/1            1/1          1/1        1/1   \n",
       "1176           snv      1/1            1/1          1/1        1/1   \n",
       "1889           snv      0/0            0/0          1/1        0/0   \n",
       "3354           snv      1/1            1/1          1/1        1/1   \n",
       "7121           snv      0/0            0/0          1/1        0/0   \n",
       "15321          snv      0/0            0/0          1/1        0/0   \n",
       "21905          snv      1/1            1/1          1/1        1/1   \n",
       "22075          snv      1/1            1/1          1/1        1/1   \n",
       "34032          snv      1/1            1/1          1/1        1/1   \n",
       "34364          snv      1/1            1/1          1/1        1/1   \n",
       "\n",
       "       altai_gt_boolean  chagyrskaya_gt_boolean  denisovan_gt_boolean  \\\n",
       "224                True                    True                  True   \n",
       "1176               True                    True                  True   \n",
       "1889              False                   False                  True   \n",
       "3354               True                    True                  True   \n",
       "7121              False                   False                  True   \n",
       "15321             False                   False                  True   \n",
       "21905              True                    True                  True   \n",
       "22075              True                    True                  True   \n",
       "34032              True                    True                  True   \n",
       "34364              True                    True                  True   \n",
       "\n",
       "       vindija_gt_boolean distribution present_in_1KG  1KG_allele_count  \\\n",
       "224                  True       Shared             no               NaN   \n",
       "1176                 True       Shared             no               NaN   \n",
       "1889                False    Denisovan             no               NaN   \n",
       "3354                 True       Shared            yes            4811.0   \n",
       "7121                False    Denisovan             no               NaN   \n",
       "15321               False    Denisovan             no               NaN   \n",
       "21905                True       Shared            yes            4456.0   \n",
       "22075                True       Shared            yes               1.0   \n",
       "34032                True       Shared            yes            3241.0   \n",
       "34364                True       Shared            yes              42.0   \n",
       "\n",
       "       1KG_allele_number  1KG_allele_frequency  1KG_EAS_AF  1KG_EUR_AF  \\\n",
       "224                  NaN                   NaN         NaN         NaN   \n",
       "1176                 NaN                   NaN         NaN         NaN   \n",
       "1889                 NaN                   NaN         NaN         NaN   \n",
       "3354              5096.0                  0.94        0.98        0.97   \n",
       "7121                 NaN                   NaN         NaN         NaN   \n",
       "15321                NaN                   NaN         NaN         NaN   \n",
       "21905             5096.0                  0.87        0.93        0.89   \n",
       "22075             5096.0                  0.00        0.00        0.00   \n",
       "34032             5096.0                  0.64        0.68        0.73   \n",
       "34364             5096.0                  0.01        0.02        0.00   \n",
       "\n",
       "       1KG_AFR_AF  1KG_AMR_AF  1KG_SAS_AF  1KG_non_ASW_AFR_AF  \\\n",
       "224           NaN         NaN         NaN                 NaN   \n",
       "1176          NaN         NaN         NaN                 NaN   \n",
       "1889          NaN         NaN         NaN                 NaN   \n",
       "3354         0.86        0.99        0.95            0.855750   \n",
       "7121          NaN         NaN         NaN                 NaN   \n",
       "15321         NaN         NaN         NaN                 NaN   \n",
       "21905        0.81        0.93        0.83            0.810916   \n",
       "22075        0.00        0.00        0.00            0.000000   \n",
       "34032        0.59        0.62        0.56            0.583821   \n",
       "34364        0.01        0.00        0.02            0.004873   \n",
       "\n",
       "      present_in_gnomAD  gnomAD_allele_count  gnomAD_allele_number  \\\n",
       "224                 yes                  1.0              142512.0   \n",
       "1176                yes                  7.0              143266.0   \n",
       "1889                yes                  2.0              143228.0   \n",
       "3354                yes             134537.0              143196.0   \n",
       "7121                 no                  NaN                   NaN   \n",
       "15321               yes                695.0              143346.0   \n",
       "21905               yes             125748.0              143146.0   \n",
       "22075               yes                 30.0              143294.0   \n",
       "34032               yes              96369.0              143076.0   \n",
       "34364               yes                215.0              143268.0   \n",
       "\n",
       "       gnomAD_allele_frequency Vernot_introgressed Vernot_ancestral_allele  \\\n",
       "224                   0.000007                  no                     NaN   \n",
       "1176                  0.000049                  no                     NaN   \n",
       "1889                  0.000014                  no                     NaN   \n",
       "3354                  0.939530                  no                     NaN   \n",
       "7121                       NaN                  no                     NaN   \n",
       "15321                 0.004848                  no                     NaN   \n",
       "21905                 0.878460                  no                     NaN   \n",
       "22075                 0.000209                 yes                       G   \n",
       "34032                 0.673551                  no                     NaN   \n",
       "34364                 0.001501                  no                     NaN   \n",
       "\n",
       "      Vernot_derived_allele  Vernot_ancestral_derived_code  Vernot_AFA_AF  \\\n",
       "224                     NaN                            NaN            NaN   \n",
       "1176                    NaN                            NaN            NaN   \n",
       "1889                    NaN                            NaN            NaN   \n",
       "3354                    NaN                            NaN            NaN   \n",
       "7121                    NaN                            NaN            NaN   \n",
       "15321                   NaN                            NaN            NaN   \n",
       "21905                   NaN                            NaN            NaN   \n",
       "22075                     A                            1.0            0.0   \n",
       "34032                   NaN                            NaN            NaN   \n",
       "34364                   NaN                            NaN            NaN   \n",
       "\n",
       "       Vernot_AFR_AF  Vernot_AMR_AF  Vernot_EAS_AF  Vernot_EUR_AF  \\\n",
       "224              NaN            NaN            NaN            NaN   \n",
       "1176             NaN            NaN            NaN            NaN   \n",
       "1889             NaN            NaN            NaN            NaN   \n",
       "3354             NaN            NaN            NaN            NaN   \n",
       "7121             NaN            NaN            NaN            NaN   \n",
       "15321            NaN            NaN            NaN            NaN   \n",
       "21905            NaN            NaN            NaN            NaN   \n",
       "22075            0.0            0.0            0.0        0.00099   \n",
       "34032            NaN            NaN            NaN            NaN   \n",
       "34364            NaN            NaN            NaN            NaN   \n",
       "\n",
       "       Vernot_PNG_AF  Vernot_SAS_AF Vernot_Denisovan_base  \\\n",
       "224              NaN            NaN                   NaN   \n",
       "1176             NaN            NaN                   NaN   \n",
       "1889             NaN            NaN                   NaN   \n",
       "3354             NaN            NaN                   NaN   \n",
       "7121             NaN            NaN                   NaN   \n",
       "15321            NaN            NaN                   NaN   \n",
       "21905            NaN            NaN                   NaN   \n",
       "22075            0.0            0.0                     A   \n",
       "34032            NaN            NaN                   NaN   \n",
       "34364            NaN            NaN                   NaN   \n",
       "\n",
       "      Vernot_Neanderthal_base_1 Vernot_Neanderthal_base_2  \\\n",
       "224                         NaN                       NaN   \n",
       "1176                        NaN                       NaN   \n",
       "1889                        NaN                       NaN   \n",
       "3354                        NaN                       NaN   \n",
       "7121                        NaN                       NaN   \n",
       "15321                       NaN                       NaN   \n",
       "21905                       NaN                       NaN   \n",
       "22075                         A                       NaN   \n",
       "34032                       NaN                       NaN   \n",
       "34364                       NaN                       NaN   \n",
       "\n",
       "         Vernot_haplotype_tag  Vernot_introgressed_AF Browning_introgressed  \\\n",
       "224                       NaN                     NaN                    no   \n",
       "1176                      NaN                     NaN                   yes   \n",
       "1889                      NaN                     NaN                    no   \n",
       "3354                      NaN                     NaN                    no   \n",
       "7121                      NaN                     NaN                    no   \n",
       "15321                     NaN                     NaN                    no   \n",
       "21905                     NaN                     NaN                    no   \n",
       "22075  chr1_26583921_26646198                0.000198                   yes   \n",
       "34032                     NaN                     NaN                    no   \n",
       "34364                     NaN                     NaN                   yes   \n",
       "\n",
       "       Browning_ref_alt  Browning_introgressed_AF annotation   mis_oe  \\\n",
       "224                 NaN                       NaN       AGRN  0.98257   \n",
       "1176                1.0                  0.000049     CFAP74      NaN   \n",
       "1889                NaN                       NaN      MMEL1  0.98394   \n",
       "3354                NaN                       NaN      MEGF6  0.95847   \n",
       "7121                NaN                       NaN     CAMTA1  0.71229   \n",
       "15321               NaN                       NaN      PADI6      NaN   \n",
       "21905               NaN                       NaN      EXTL1  0.87054   \n",
       "22075               1.0                  0.000000      CEP85  0.96911   \n",
       "34032               NaN                       NaN     PIK3R3  0.99209   \n",
       "34364               1.0                  0.010000       FAAH  0.89767   \n",
       "\n",
       "          mis_z    lof_oe   lof_z  phyloP  ag_delta  al_delta  dg_delta  \\\n",
       "224    0.225520  0.319140  6.0143  -0.375      0.00      0.00      0.04   \n",
       "1176        NaN       NaN     NaN  -0.669      0.22      0.00      0.00   \n",
       "1889   0.126020  0.768240  1.5689  -0.242      0.00      0.01      0.03   \n",
       "3354   0.458030  0.738560  2.2728  -1.183      0.00      0.00      0.00   \n",
       "7121   3.261900  0.091344  7.3710   0.655      0.00      0.00      0.02   \n",
       "15321       NaN       NaN     NaN   0.561      0.00      0.00      0.00   \n",
       "21905  0.903800  0.751620  1.2166  -0.527      0.00      0.00      0.04   \n",
       "22075  0.224110  0.338850  3.9380   0.563      0.00      0.26      0.00   \n",
       "34032  0.044445  0.684270  1.5417  -2.533      0.00      0.00      0.43   \n",
       "34364  0.637630  0.725440  1.3359  -0.121      0.28      0.00      0.03   \n",
       "\n",
       "       dl_delta  delta_max  ag_pos  al_pos  dg_pos  dl_pos  \\\n",
       "224        0.48       0.48     -48      -6      16      -6   \n",
       "1176       0.00       0.22      49      -2      49      -2   \n",
       "1889       0.25       0.25     -37      39     -37       2   \n",
       "3354       0.24       0.24     -28      47       1       3   \n",
       "7121       0.71       0.71      48      -5     -34      -5   \n",
       "15321      0.30       0.30       3      -5      16      -5   \n",
       "21905      0.61       0.61     -50      -2      10      -2   \n",
       "22075      0.00       0.26       3       1      -4       1   \n",
       "34032      0.04       0.43      22      -4       0      -4   \n",
       "34364      0.00       0.28       0     -49      39     -24   \n",
       "\n",
       "       Adipose_Subcutaneous  Adipose_Visceral_Omentum  Adrenal_Gland  \\\n",
       "224                     NaN                       NaN            NaN   \n",
       "1176                    NaN                       NaN            NaN   \n",
       "1889                    NaN                       NaN            NaN   \n",
       "3354                    NaN                       NaN            NaN   \n",
       "7121                    NaN                       NaN            NaN   \n",
       "15321                   NaN                       NaN            NaN   \n",
       "21905                   0.0                       0.0            0.0   \n",
       "22075                   NaN                       NaN            NaN   \n",
       "34032                   1.0                       1.0            0.0   \n",
       "34364                   NaN                       NaN            NaN   \n",
       "\n",
       "       Artery_Aorta  Artery_Coronary  Artery_Tibial  Brain_Amygdala  \\\n",
       "224             NaN              NaN            NaN             NaN   \n",
       "1176            NaN              NaN            NaN             NaN   \n",
       "1889            NaN              NaN            NaN             NaN   \n",
       "3354            NaN              NaN            NaN             NaN   \n",
       "7121            NaN              NaN            NaN             NaN   \n",
       "15321           NaN              NaN            NaN             NaN   \n",
       "21905           0.0              0.0            0.0             0.0   \n",
       "22075           NaN              NaN            NaN             NaN   \n",
       "34032           0.0              0.0            1.0             0.0   \n",
       "34364           NaN              NaN            NaN             NaN   \n",
       "\n",
       "       Brain_Anterior_cingulate_cortex_BA24  Brain_Caudate_basal_ganglia  \\\n",
       "224                                     NaN                          NaN   \n",
       "1176                                    NaN                          NaN   \n",
       "1889                                    NaN                          NaN   \n",
       "3354                                    NaN                          NaN   \n",
       "7121                                    NaN                          NaN   \n",
       "15321                                   NaN                          NaN   \n",
       "21905                                   0.0                          0.0   \n",
       "22075                                   NaN                          NaN   \n",
       "34032                                   0.0                          0.0   \n",
       "34364                                   NaN                          NaN   \n",
       "\n",
       "       Brain_Cerebellar_Hemisphere  Brain_Cerebellum  Brain_Cortex  \\\n",
       "224                            NaN               NaN           NaN   \n",
       "1176                           NaN               NaN           NaN   \n",
       "1889                           NaN               NaN           NaN   \n",
       "3354                           NaN               NaN           NaN   \n",
       "7121                           NaN               NaN           NaN   \n",
       "15321                          NaN               NaN           NaN   \n",
       "21905                          0.0               0.0           0.0   \n",
       "22075                          NaN               NaN           NaN   \n",
       "34032                          0.0               0.0           0.0   \n",
       "34364                          NaN               NaN           NaN   \n",
       "\n",
       "       Brain_Frontal_Cortex_BA9  Brain_Hippocampus  Brain_Hypothalamus  \\\n",
       "224                         NaN                NaN                 NaN   \n",
       "1176                        NaN                NaN                 NaN   \n",
       "1889                        NaN                NaN                 NaN   \n",
       "3354                        NaN                NaN                 NaN   \n",
       "7121                        NaN                NaN                 NaN   \n",
       "15321                       NaN                NaN                 NaN   \n",
       "21905                       0.0                0.0                 0.0   \n",
       "22075                       NaN                NaN                 NaN   \n",
       "34032                       0.0                0.0                 0.0   \n",
       "34364                       NaN                NaN                 NaN   \n",
       "\n",
       "       Brain_Nucleus_accumbens_basal_ganglia  Brain_Putamen_basal_ganglia  \\\n",
       "224                                      NaN                          NaN   \n",
       "1176                                     NaN                          NaN   \n",
       "1889                                     NaN                          NaN   \n",
       "3354                                     NaN                          NaN   \n",
       "7121                                     NaN                          NaN   \n",
       "15321                                    NaN                          NaN   \n",
       "21905                                    0.0                          0.0   \n",
       "22075                                    NaN                          NaN   \n",
       "34032                                    0.0                          0.0   \n",
       "34364                                    NaN                          NaN   \n",
       "\n",
       "       Brain_Spinal_cord_cervical_c-1  Brain_Substantia_nigra  \\\n",
       "224                               NaN                     NaN   \n",
       "1176                              NaN                     NaN   \n",
       "1889                              NaN                     NaN   \n",
       "3354                              NaN                     NaN   \n",
       "7121                              NaN                     NaN   \n",
       "15321                             NaN                     NaN   \n",
       "21905                             0.0                     0.0   \n",
       "22075                             NaN                     NaN   \n",
       "34032                             0.0                     0.0   \n",
       "34364                             NaN                     NaN   \n",
       "\n",
       "       Breast_Mammary_Tissue  Cells_Cultured_fibroblasts  \\\n",
       "224                      NaN                         NaN   \n",
       "1176                     NaN                         NaN   \n",
       "1889                     NaN                         NaN   \n",
       "3354                     NaN                         NaN   \n",
       "7121                     NaN                         NaN   \n",
       "15321                    NaN                         NaN   \n",
       "21905                    0.0                         0.0   \n",
       "22075                    NaN                         NaN   \n",
       "34032                    0.0                         1.0   \n",
       "34364                    NaN                         NaN   \n",
       "\n",
       "       Cells_EBV-transformed_lymphocytes  Colon_Sigmoid  Colon_Transverse  \\\n",
       "224                                  NaN            NaN               NaN   \n",
       "1176                                 NaN            NaN               NaN   \n",
       "1889                                 NaN            NaN               NaN   \n",
       "3354                                 NaN            NaN               NaN   \n",
       "7121                                 NaN            NaN               NaN   \n",
       "15321                                NaN            NaN               NaN   \n",
       "21905                                0.0            0.0               0.0   \n",
       "22075                                NaN            NaN               NaN   \n",
       "34032                                0.0            1.0               0.0   \n",
       "34364                                NaN            NaN               NaN   \n",
       "\n",
       "       Esophagus_Gastroesophageal_Junction  Esophagus_Mucosa  \\\n",
       "224                                    NaN               NaN   \n",
       "1176                                   NaN               NaN   \n",
       "1889                                   NaN               NaN   \n",
       "3354                                   NaN               NaN   \n",
       "7121                                   NaN               NaN   \n",
       "15321                                  NaN               NaN   \n",
       "21905                                  0.0               0.0   \n",
       "22075                                  NaN               NaN   \n",
       "34032                                  1.0               1.0   \n",
       "34364                                  NaN               NaN   \n",
       "\n",
       "       Esophagus_Muscularis  Heart_Atrial_Appendage  Heart_Left_Ventricle  \\\n",
       "224                     NaN                     NaN                   NaN   \n",
       "1176                    NaN                     NaN                   NaN   \n",
       "1889                    NaN                     NaN                   NaN   \n",
       "3354                    NaN                     NaN                   NaN   \n",
       "7121                    NaN                     NaN                   NaN   \n",
       "15321                   NaN                     NaN                   NaN   \n",
       "21905                   0.0                     0.0                   0.0   \n",
       "22075                   NaN                     NaN                   NaN   \n",
       "34032                   1.0                     0.0                   0.0   \n",
       "34364                   NaN                     NaN                   NaN   \n",
       "\n",
       "       Kidney_Cortex  Liver  Lung  Minor_Salivary_Gland  Muscle_Skeletal  \\\n",
       "224              NaN    NaN   NaN                   NaN              NaN   \n",
       "1176             NaN    NaN   NaN                   NaN              NaN   \n",
       "1889             NaN    NaN   NaN                   NaN              NaN   \n",
       "3354             NaN    NaN   NaN                   NaN              NaN   \n",
       "7121             NaN    NaN   NaN                   NaN              NaN   \n",
       "15321            NaN    NaN   NaN                   NaN              NaN   \n",
       "21905            0.0    0.0   0.0                   0.0              0.0   \n",
       "22075            NaN    NaN   NaN                   NaN              NaN   \n",
       "34032            0.0    0.0   1.0                   0.0              1.0   \n",
       "34364            NaN    NaN   NaN                   NaN              NaN   \n",
       "\n",
       "       Nerve_Tibial  Ovary  Pancreas  Pituitary  Prostate  \\\n",
       "224             NaN    NaN       NaN        NaN       NaN   \n",
       "1176            NaN    NaN       NaN        NaN       NaN   \n",
       "1889            NaN    NaN       NaN        NaN       NaN   \n",
       "3354            NaN    NaN       NaN        NaN       NaN   \n",
       "7121            NaN    NaN       NaN        NaN       NaN   \n",
       "15321           NaN    NaN       NaN        NaN       NaN   \n",
       "21905           1.0    0.0       0.0        0.0       0.0   \n",
       "22075           NaN    NaN       NaN        NaN       NaN   \n",
       "34032           1.0    0.0       1.0        0.0       1.0   \n",
       "34364           NaN    NaN       NaN        NaN       NaN   \n",
       "\n",
       "       Skin_Not_Sun_Exposed_Suprapubic  Skin_Sun_Exposed_Lower_leg  \\\n",
       "224                                NaN                         NaN   \n",
       "1176                               NaN                         NaN   \n",
       "1889                               NaN                         NaN   \n",
       "3354                               NaN                         NaN   \n",
       "7121                               NaN                         NaN   \n",
       "15321                              NaN                         NaN   \n",
       "21905                              0.0                         1.0   \n",
       "22075                              NaN                         NaN   \n",
       "34032                              1.0                         1.0   \n",
       "34364                              NaN                         NaN   \n",
       "\n",
       "       Small_Intestine_Terminal_Ileum  Spleen  Stomach  Testis  Thyroid  \\\n",
       "224                               NaN     NaN      NaN     NaN      NaN   \n",
       "1176                              NaN     NaN      NaN     NaN      NaN   \n",
       "1889                              NaN     NaN      NaN     NaN      NaN   \n",
       "3354                              NaN     NaN      NaN     NaN      NaN   \n",
       "7121                              NaN     NaN      NaN     NaN      NaN   \n",
       "15321                             NaN     NaN      NaN     NaN      NaN   \n",
       "21905                             0.0     1.0      0.0     0.0      0.0   \n",
       "22075                             NaN     NaN      NaN     NaN      NaN   \n",
       "34032                             0.0     0.0      1.0     1.0      1.0   \n",
       "34364                             NaN     NaN      NaN     NaN      NaN   \n",
       "\n",
       "       Uterus  Vagina  Whole_Blood  N_GTEx_tissues sQTL  \\\n",
       "224       NaN     NaN          NaN             NaN   no   \n",
       "1176      NaN     NaN          NaN             NaN   no   \n",
       "1889      NaN     NaN          NaN             NaN   no   \n",
       "3354      NaN     NaN          NaN             NaN   no   \n",
       "7121      NaN     NaN          NaN             NaN   no   \n",
       "15321     NaN     NaN          NaN             NaN   no   \n",
       "21905     0.0     0.0          0.0             3.0  yes   \n",
       "22075     NaN     NaN          NaN             NaN   no   \n",
       "34032     0.0     0.0          1.0            19.0  yes   \n",
       "34364     NaN     NaN          NaN             NaN   no   \n",
       "\n",
       "         Vernot_allele_origin  Browning_allele_origin  \n",
       "224          archaic-specific        archaic-specific  \n",
       "1176         archaic-specific        archaic-specific  \n",
       "1889         archaic-specific        archaic-specific  \n",
       "3354                  ancient                 ancient  \n",
       "7121         archaic-specific        archaic-specific  \n",
       "15321  low-confidence ancient  low-confidence ancient  \n",
       "21905                 ancient                 ancient  \n",
       "22075            introgressed            introgressed  \n",
       "34032                 ancient                 ancient  \n",
       "34364  low-confidence ancient            introgressed  "
      ]
     },
     "execution_count": 196,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "multiple_SAVs_per_gene = data_2[data_2.duplicated(subset=['annotation','distribution'])]\n",
    "multiple_SAVs_per_gene.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 197,
   "id": "0532b38a",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "591"
      ]
     },
     "execution_count": 197,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(multiple_SAVs_per_gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 198,
   "id": "b456951b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.20817189151109547"
      ]
     },
     "execution_count": 198,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "591/2839"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 199,
   "id": "aa7c9aa2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref_allele</th>\n",
       "      <th>alt_allele</th>\n",
       "      <th>ancestral_allele</th>\n",
       "      <th>anc_dev</th>\n",
       "      <th>variant_type</th>\n",
       "      <th>altai_gt</th>\n",
       "      <th>chagyrskaya_gt</th>\n",
       "      <th>denisovan_gt</th>\n",
       "      <th>vindija_gt</th>\n",
       "      <th>altai_gt_boolean</th>\n",
       "      <th>chagyrskaya_gt_boolean</th>\n",
       "      <th>denisovan_gt_boolean</th>\n",
       "      <th>vindija_gt_boolean</th>\n",
       "      <th>distribution</th>\n",
       "      <th>present_in_1KG</th>\n",
       "      <th>1KG_allele_count</th>\n",
       "      <th>1KG_allele_number</th>\n",
       "      <th>1KG_allele_frequency</th>\n",
       "      <th>1KG_EAS_AF</th>\n",
       "      <th>1KG_EUR_AF</th>\n",
       "      <th>1KG_AFR_AF</th>\n",
       "      <th>1KG_AMR_AF</th>\n",
       "      <th>1KG_SAS_AF</th>\n",
       "      <th>1KG_non_ASW_AFR_AF</th>\n",
       "      <th>present_in_gnomAD</th>\n",
       "      <th>gnomAD_allele_count</th>\n",
       "      <th>gnomAD_allele_number</th>\n",
       "      <th>gnomAD_allele_frequency</th>\n",
       "      <th>Vernot_introgressed</th>\n",
       "      <th>Vernot_ancestral_allele</th>\n",
       "      <th>Vernot_derived_allele</th>\n",
       "      <th>Vernot_ancestral_derived_code</th>\n",
       "      <th>Vernot_AFA_AF</th>\n",
       "      <th>Vernot_AFR_AF</th>\n",
       "      <th>Vernot_AMR_AF</th>\n",
       "      <th>Vernot_EAS_AF</th>\n",
       "      <th>Vernot_EUR_AF</th>\n",
       "      <th>Vernot_PNG_AF</th>\n",
       "      <th>Vernot_SAS_AF</th>\n",
       "      <th>Vernot_Denisovan_base</th>\n",
       "      <th>Vernot_Neanderthal_base_1</th>\n",
       "      <th>Vernot_Neanderthal_base_2</th>\n",
       "      <th>Vernot_haplotype_tag</th>\n",
       "      <th>Vernot_introgressed_AF</th>\n",
       "      <th>Browning_introgressed</th>\n",
       "      <th>Browning_ref_alt</th>\n",
       "      <th>Browning_introgressed_AF</th>\n",
       "      <th>annotation</th>\n",
       "      <th>mis_oe</th>\n",
       "      <th>mis_z</th>\n",
       "      <th>lof_oe</th>\n",
       "      <th>lof_z</th>\n",
       "      <th>phyloP</th>\n",
       "      <th>ag_delta</th>\n",
       "      <th>al_delta</th>\n",
       "      <th>dg_delta</th>\n",
       "      <th>dl_delta</th>\n",
       "      <th>delta_max</th>\n",
       "      <th>ag_pos</th>\n",
       "      <th>al_pos</th>\n",
       "      <th>dg_pos</th>\n",
       "      <th>dl_pos</th>\n",
       "      <th>Adipose_Subcutaneous</th>\n",
       "      <th>Adipose_Visceral_Omentum</th>\n",
       "      <th>Adrenal_Gland</th>\n",
       "      <th>Artery_Aorta</th>\n",
       "      <th>Artery_Coronary</th>\n",
       "      <th>Artery_Tibial</th>\n",
       "      <th>Brain_Amygdala</th>\n",
       "      <th>Brain_Anterior_cingulate_cortex_BA24</th>\n",
       "      <th>Brain_Caudate_basal_ganglia</th>\n",
       "      <th>Brain_Cerebellar_Hemisphere</th>\n",
       "      <th>Brain_Cerebellum</th>\n",
       "      <th>Brain_Cortex</th>\n",
       "      <th>Brain_Frontal_Cortex_BA9</th>\n",
       "      <th>Brain_Hippocampus</th>\n",
       "      <th>Brain_Hypothalamus</th>\n",
       "      <th>Brain_Nucleus_accumbens_basal_ganglia</th>\n",
       "      <th>Brain_Putamen_basal_ganglia</th>\n",
       "      <th>Brain_Spinal_cord_cervical_c-1</th>\n",
       "      <th>Brain_Substantia_nigra</th>\n",
       "      <th>Breast_Mammary_Tissue</th>\n",
       "      <th>Cells_Cultured_fibroblasts</th>\n",
       "      <th>Cells_EBV-transformed_lymphocytes</th>\n",
       "      <th>Colon_Sigmoid</th>\n",
       "      <th>Colon_Transverse</th>\n",
       "      <th>Esophagus_Gastroesophageal_Junction</th>\n",
       "      <th>Esophagus_Mucosa</th>\n",
       "      <th>Esophagus_Muscularis</th>\n",
       "      <th>Heart_Atrial_Appendage</th>\n",
       "      <th>Heart_Left_Ventricle</th>\n",
       "      <th>Kidney_Cortex</th>\n",
       "      <th>Liver</th>\n",
       "      <th>Lung</th>\n",
       "      <th>Minor_Salivary_Gland</th>\n",
       "      <th>Muscle_Skeletal</th>\n",
       "      <th>Nerve_Tibial</th>\n",
       "      <th>Ovary</th>\n",
       "      <th>Pancreas</th>\n",
       "      <th>Pituitary</th>\n",
       "      <th>Prostate</th>\n",
       "      <th>Skin_Not_Sun_Exposed_Suprapubic</th>\n",
       "      <th>Skin_Sun_Exposed_Lower_leg</th>\n",
       "      <th>Small_Intestine_Terminal_Ileum</th>\n",
       "      <th>Spleen</th>\n",
       "      <th>Stomach</th>\n",
       "      <th>Testis</th>\n",
       "      <th>Thyroid</th>\n",
       "      <th>Uterus</th>\n",
       "      <th>Vagina</th>\n",
       "      <th>Whole_Blood</th>\n",
       "      <th>N_GTEx_tissues</th>\n",
       "      <th>sQTL</th>\n",
       "      <th>Vernot_allele_origin</th>\n",
       "      <th>Browning_allele_origin</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>218</th>\n",
       "      <td>chr1</td>\n",
       "      <td>980460</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>./.</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Other</td>\n",
       "      <td>yes</td>\n",
       "      <td>4642.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.91</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.93</td>\n",
       "      <td>0.77</td>\n",
       "      <td>0.93</td>\n",
       "      <td>0.97</td>\n",
       "      <td>0.781676</td>\n",
       "      <td>yes</td>\n",
       "      <td>128061.0</td>\n",
       "      <td>143108.0</td>\n",
       "      <td>0.894856</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>0.98257</td>\n",
       "      <td>0.22552</td>\n",
       "      <td>0.319140</td>\n",
       "      <td>6.0143</td>\n",
       "      <td>0.561</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.03</td>\n",
       "      <td>0.43</td>\n",
       "      <td>0.43</td>\n",
       "      <td>30</td>\n",
       "      <td>-33</td>\n",
       "      <td>30</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>ancient</td>\n",
       "      <td>ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>479</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1226233</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>g</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>yes</td>\n",
       "      <td>211.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.04</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.15</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.160819</td>\n",
       "      <td>yes</td>\n",
       "      <td>5454.0</td>\n",
       "      <td>143298.0</td>\n",
       "      <td>0.038060</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SCNN1D</td>\n",
       "      <td>1.10050</td>\n",
       "      <td>-0.78105</td>\n",
       "      <td>1.237300</td>\n",
       "      <td>-1.3261</td>\n",
       "      <td>0.299</td>\n",
       "      <td>0.40</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.40</td>\n",
       "      <td>-10</td>\n",
       "      <td>42</td>\n",
       "      <td>41</td>\n",
       "      <td>-44</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1176</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1872053</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>7.0</td>\n",
       "      <td>143266.0</td>\n",
       "      <td>0.000049</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.000049</td>\n",
       "      <td>CFAP74</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.669</td>\n",
       "      <td>0.22</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.22</td>\n",
       "      <td>49</td>\n",
       "      <td>-2</td>\n",
       "      <td>49</td>\n",
       "      <td>-2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3281</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3498077</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>Chagyrskaya</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>2.0</td>\n",
       "      <td>143324.0</td>\n",
       "      <td>0.000014</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MEGF6</td>\n",
       "      <td>0.95847</td>\n",
       "      <td>0.45803</td>\n",
       "      <td>0.738560</td>\n",
       "      <td>2.2728</td>\n",
       "      <td>-1.579</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.30</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.30</td>\n",
       "      <td>-2</td>\n",
       "      <td>-44</td>\n",
       "      <td>-1</td>\n",
       "      <td>-44</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3354</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3520127</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>ancestral</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>yes</td>\n",
       "      <td>4811.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.94</td>\n",
       "      <td>0.98</td>\n",
       "      <td>0.97</td>\n",
       "      <td>0.86</td>\n",
       "      <td>0.99</td>\n",
       "      <td>0.95</td>\n",
       "      <td>0.855750</td>\n",
       "      <td>yes</td>\n",
       "      <td>134537.0</td>\n",
       "      <td>143196.0</td>\n",
       "      <td>0.939530</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MEGF6</td>\n",
       "      <td>0.95847</td>\n",
       "      <td>0.45803</td>\n",
       "      <td>0.738560</td>\n",
       "      <td>2.2728</td>\n",
       "      <td>-1.183</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.24</td>\n",
       "      <td>0.24</td>\n",
       "      <td>-28</td>\n",
       "      <td>47</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>ancient</td>\n",
       "      <td>ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4234</th>\n",
       "      <td>chr1</td>\n",
       "      <td>5935162</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>ancestral</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NPHP4</td>\n",
       "      <td>1.02290</td>\n",
       "      <td>-0.24249</td>\n",
       "      <td>0.781550</td>\n",
       "      <td>1.6669</td>\n",
       "      <td>-0.557</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.91</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1.00</td>\n",
       "      <td>-2</td>\n",
       "      <td>-8</td>\n",
       "      <td>-43</td>\n",
       "      <td>-44</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4452</th>\n",
       "      <td>chr1</td>\n",
       "      <td>6104558</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>KCNAB2</td>\n",
       "      <td>0.55002</td>\n",
       "      <td>2.59630</td>\n",
       "      <td>0.183150</td>\n",
       "      <td>3.9550</td>\n",
       "      <td>0.377</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.25</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.25</td>\n",
       "      <td>-4</td>\n",
       "      <td>32</td>\n",
       "      <td>-4</td>\n",
       "      <td>-36</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6747</th>\n",
       "      <td>chr1</td>\n",
       "      <td>7568548</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>ancestral</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>1.0</td>\n",
       "      <td>143320.0</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CAMTA1</td>\n",
       "      <td>0.71229</td>\n",
       "      <td>3.26190</td>\n",
       "      <td>0.091344</td>\n",
       "      <td>7.3710</td>\n",
       "      <td>-1.546</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.32</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.21</td>\n",
       "      <td>0.32</td>\n",
       "      <td>-2</td>\n",
       "      <td>-42</td>\n",
       "      <td>-6</td>\n",
       "      <td>36</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7121</th>\n",
       "      <td>chr1</td>\n",
       "      <td>7731131</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CAMTA1</td>\n",
       "      <td>0.71229</td>\n",
       "      <td>3.26190</td>\n",
       "      <td>0.091344</td>\n",
       "      <td>7.3710</td>\n",
       "      <td>0.655</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.71</td>\n",
       "      <td>0.71</td>\n",
       "      <td>48</td>\n",
       "      <td>-5</td>\n",
       "      <td>-34</td>\n",
       "      <td>-5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9122</th>\n",
       "      <td>chr1</td>\n",
       "      <td>10231046</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>UBE4B</td>\n",
       "      <td>0.62481</td>\n",
       "      <td>3.72600</td>\n",
       "      <td>0.074661</td>\n",
       "      <td>7.0171</td>\n",
       "      <td>0.591</td>\n",
       "      <td>0.35</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.35</td>\n",
       "      <td>12</td>\n",
       "      <td>36</td>\n",
       "      <td>30</td>\n",
       "      <td>-4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     chrom       pos ref_allele alt_allele ancestral_allele    anc_dev  \\\n",
       "218   chr1    980460          G          A                G    derived   \n",
       "479   chr1   1226233          G          A                g    derived   \n",
       "1176  chr1   1872053          T          C                T    derived   \n",
       "3281  chr1   3498077          C          T                C    derived   \n",
       "3354  chr1   3520127          T          C                C  ancestral   \n",
       "4234  chr1   5935162          A          T                T  ancestral   \n",
       "4452  chr1   6104558          T          A                T    derived   \n",
       "6747  chr1   7568548          C          T                T  ancestral   \n",
       "7121  chr1   7731131          G          A                G    derived   \n",
       "9122  chr1  10231046          G          T                G    derived   \n",
       "\n",
       "     variant_type altai_gt chagyrskaya_gt denisovan_gt vindija_gt  \\\n",
       "218           snv      1/1            ./.          1/1        1/1   \n",
       "479           snv      0/0            0/0          1/1        0/0   \n",
       "1176          snv      1/1            1/1          1/1        1/1   \n",
       "3281          snv      0/0            1/1          0/0        0/0   \n",
       "3354          snv      1/1            1/1          1/1        1/1   \n",
       "4234          snv      1/1            1/1          1/1        1/1   \n",
       "4452          snv      0/0            0/0          0/1        0/0   \n",
       "6747          snv      1/1            1/1          1/1        1/1   \n",
       "7121          snv      0/0            0/0          1/1        0/0   \n",
       "9122          snv      0/0            0/0          1/1        0/0   \n",
       "\n",
       "      altai_gt_boolean  chagyrskaya_gt_boolean  denisovan_gt_boolean  \\\n",
       "218               True                   False                  True   \n",
       "479              False                   False                  True   \n",
       "1176              True                    True                  True   \n",
       "3281             False                    True                 False   \n",
       "3354              True                    True                  True   \n",
       "4234              True                    True                  True   \n",
       "4452             False                   False                  True   \n",
       "6747              True                    True                  True   \n",
       "7121             False                   False                  True   \n",
       "9122             False                   False                  True   \n",
       "\n",
       "      vindija_gt_boolean distribution present_in_1KG  1KG_allele_count  \\\n",
       "218                 True        Other            yes            4642.0   \n",
       "479                False    Denisovan            yes             211.0   \n",
       "1176                True       Shared             no               NaN   \n",
       "3281               False  Chagyrskaya             no               NaN   \n",
       "3354                True       Shared            yes            4811.0   \n",
       "4234                True       Shared             no               NaN   \n",
       "4452               False    Denisovan             no               NaN   \n",
       "6747                True       Shared             no               NaN   \n",
       "7121               False    Denisovan             no               NaN   \n",
       "9122               False    Denisovan             no               NaN   \n",
       "\n",
       "      1KG_allele_number  1KG_allele_frequency  1KG_EAS_AF  1KG_EUR_AF  \\\n",
       "218              5096.0                  0.91        1.00        0.93   \n",
       "479              5096.0                  0.04        0.00        0.00   \n",
       "1176                NaN                   NaN         NaN         NaN   \n",
       "3281                NaN                   NaN         NaN         NaN   \n",
       "3354             5096.0                  0.94        0.98        0.97   \n",
       "4234                NaN                   NaN         NaN         NaN   \n",
       "4452                NaN                   NaN         NaN         NaN   \n",
       "6747                NaN                   NaN         NaN         NaN   \n",
       "7121                NaN                   NaN         NaN         NaN   \n",
       "9122                NaN                   NaN         NaN         NaN   \n",
       "\n",
       "      1KG_AFR_AF  1KG_AMR_AF  1KG_SAS_AF  1KG_non_ASW_AFR_AF  \\\n",
       "218         0.77        0.93        0.97            0.781676   \n",
       "479         0.15        0.01        0.00            0.160819   \n",
       "1176         NaN         NaN         NaN                 NaN   \n",
       "3281         NaN         NaN         NaN                 NaN   \n",
       "3354        0.86        0.99        0.95            0.855750   \n",
       "4234         NaN         NaN         NaN                 NaN   \n",
       "4452         NaN         NaN         NaN                 NaN   \n",
       "6747         NaN         NaN         NaN                 NaN   \n",
       "7121         NaN         NaN         NaN                 NaN   \n",
       "9122         NaN         NaN         NaN                 NaN   \n",
       "\n",
       "     present_in_gnomAD  gnomAD_allele_count  gnomAD_allele_number  \\\n",
       "218                yes             128061.0              143108.0   \n",
       "479                yes               5454.0              143298.0   \n",
       "1176               yes                  7.0              143266.0   \n",
       "3281               yes                  2.0              143324.0   \n",
       "3354               yes             134537.0              143196.0   \n",
       "4234                no                  NaN                   NaN   \n",
       "4452                no                  NaN                   NaN   \n",
       "6747               yes                  1.0              143320.0   \n",
       "7121                no                  NaN                   NaN   \n",
       "9122                no                  NaN                   NaN   \n",
       "\n",
       "      gnomAD_allele_frequency Vernot_introgressed Vernot_ancestral_allele  \\\n",
       "218                  0.894856                  no                     NaN   \n",
       "479                  0.038060                  no                     NaN   \n",
       "1176                 0.000049                  no                     NaN   \n",
       "3281                 0.000014                  no                     NaN   \n",
       "3354                 0.939530                  no                     NaN   \n",
       "4234                      NaN                  no                     NaN   \n",
       "4452                      NaN                  no                     NaN   \n",
       "6747                 0.000007                  no                     NaN   \n",
       "7121                      NaN                  no                     NaN   \n",
       "9122                      NaN                  no                     NaN   \n",
       "\n",
       "     Vernot_derived_allele  Vernot_ancestral_derived_code  Vernot_AFA_AF  \\\n",
       "218                    NaN                            NaN            NaN   \n",
       "479                    NaN                            NaN            NaN   \n",
       "1176                   NaN                            NaN            NaN   \n",
       "3281                   NaN                            NaN            NaN   \n",
       "3354                   NaN                            NaN            NaN   \n",
       "4234                   NaN                            NaN            NaN   \n",
       "4452                   NaN                            NaN            NaN   \n",
       "6747                   NaN                            NaN            NaN   \n",
       "7121                   NaN                            NaN            NaN   \n",
       "9122                   NaN                            NaN            NaN   \n",
       "\n",
       "      Vernot_AFR_AF  Vernot_AMR_AF  Vernot_EAS_AF  Vernot_EUR_AF  \\\n",
       "218             NaN            NaN            NaN            NaN   \n",
       "479             NaN            NaN            NaN            NaN   \n",
       "1176            NaN            NaN            NaN            NaN   \n",
       "3281            NaN            NaN            NaN            NaN   \n",
       "3354            NaN            NaN            NaN            NaN   \n",
       "4234            NaN            NaN            NaN            NaN   \n",
       "4452            NaN            NaN            NaN            NaN   \n",
       "6747            NaN            NaN            NaN            NaN   \n",
       "7121            NaN            NaN            NaN            NaN   \n",
       "9122            NaN            NaN            NaN            NaN   \n",
       "\n",
       "      Vernot_PNG_AF  Vernot_SAS_AF Vernot_Denisovan_base  \\\n",
       "218             NaN            NaN                   NaN   \n",
       "479             NaN            NaN                   NaN   \n",
       "1176            NaN            NaN                   NaN   \n",
       "3281            NaN            NaN                   NaN   \n",
       "3354            NaN            NaN                   NaN   \n",
       "4234            NaN            NaN                   NaN   \n",
       "4452            NaN            NaN                   NaN   \n",
       "6747            NaN            NaN                   NaN   \n",
       "7121            NaN            NaN                   NaN   \n",
       "9122            NaN            NaN                   NaN   \n",
       "\n",
       "     Vernot_Neanderthal_base_1 Vernot_Neanderthal_base_2 Vernot_haplotype_tag  \\\n",
       "218                        NaN                       NaN                  NaN   \n",
       "479                        NaN                       NaN                  NaN   \n",
       "1176                       NaN                       NaN                  NaN   \n",
       "3281                       NaN                       NaN                  NaN   \n",
       "3354                       NaN                       NaN                  NaN   \n",
       "4234                       NaN                       NaN                  NaN   \n",
       "4452                       NaN                       NaN                  NaN   \n",
       "6747                       NaN                       NaN                  NaN   \n",
       "7121                       NaN                       NaN                  NaN   \n",
       "9122                       NaN                       NaN                  NaN   \n",
       "\n",
       "      Vernot_introgressed_AF Browning_introgressed  Browning_ref_alt  \\\n",
       "218                      NaN                    no               NaN   \n",
       "479                      NaN                    no               NaN   \n",
       "1176                     NaN                   yes               1.0   \n",
       "3281                     NaN                    no               NaN   \n",
       "3354                     NaN                    no               NaN   \n",
       "4234                     NaN                    no               NaN   \n",
       "4452                     NaN                    no               NaN   \n",
       "6747                     NaN                    no               NaN   \n",
       "7121                     NaN                    no               NaN   \n",
       "9122                     NaN                    no               NaN   \n",
       "\n",
       "      Browning_introgressed_AF annotation   mis_oe    mis_z    lof_oe   lof_z  \\\n",
       "218                        NaN       AGRN  0.98257  0.22552  0.319140  6.0143   \n",
       "479                        NaN     SCNN1D  1.10050 -0.78105  1.237300 -1.3261   \n",
       "1176                  0.000049     CFAP74      NaN      NaN       NaN     NaN   \n",
       "3281                       NaN      MEGF6  0.95847  0.45803  0.738560  2.2728   \n",
       "3354                       NaN      MEGF6  0.95847  0.45803  0.738560  2.2728   \n",
       "4234                       NaN      NPHP4  1.02290 -0.24249  0.781550  1.6669   \n",
       "4452                       NaN     KCNAB2  0.55002  2.59630  0.183150  3.9550   \n",
       "6747                       NaN     CAMTA1  0.71229  3.26190  0.091344  7.3710   \n",
       "7121                       NaN     CAMTA1  0.71229  3.26190  0.091344  7.3710   \n",
       "9122                       NaN      UBE4B  0.62481  3.72600  0.074661  7.0171   \n",
       "\n",
       "      phyloP  ag_delta  al_delta  dg_delta  dl_delta  delta_max  ag_pos  \\\n",
       "218    0.561      0.00      0.00      0.03      0.43       0.43      30   \n",
       "479    0.299      0.40      0.01      0.00      0.00       0.40     -10   \n",
       "1176  -0.669      0.22      0.00      0.00      0.00       0.22      49   \n",
       "3281  -1.579      0.00      0.00      0.30      0.00       0.30      -2   \n",
       "3354  -1.183      0.00      0.00      0.00      0.24       0.24     -28   \n",
       "4234  -0.557      1.00      0.91      0.00      0.00       1.00      -2   \n",
       "4452   0.377      0.00      0.00      0.25      0.00       0.25      -4   \n",
       "6747  -1.546      0.00      0.32      0.00      0.21       0.32      -2   \n",
       "7121   0.655      0.00      0.00      0.02      0.71       0.71      48   \n",
       "9122   0.591      0.35      0.00      0.01      0.00       0.35      12   \n",
       "\n",
       "      al_pos  dg_pos  dl_pos  Adipose_Subcutaneous  Adipose_Visceral_Omentum  \\\n",
       "218      -33      30       0                   0.0                       1.0   \n",
       "479       42      41     -44                   1.0                       0.0   \n",
       "1176      -2      49      -2                   NaN                       NaN   \n",
       "3281     -44      -1     -44                   NaN                       NaN   \n",
       "3354      47       1       3                   NaN                       NaN   \n",
       "4234      -8     -43     -44                   NaN                       NaN   \n",
       "4452      32      -4     -36                   NaN                       NaN   \n",
       "6747     -42      -6      36                   NaN                       NaN   \n",
       "7121      -5     -34      -5                   NaN                       NaN   \n",
       "9122      36      30      -4                   NaN                       NaN   \n",
       "\n",
       "      Adrenal_Gland  Artery_Aorta  Artery_Coronary  Artery_Tibial  \\\n",
       "218             0.0           1.0              0.0            1.0   \n",
       "479             0.0           0.0              0.0            1.0   \n",
       "1176            NaN           NaN              NaN            NaN   \n",
       "3281            NaN           NaN              NaN            NaN   \n",
       "3354            NaN           NaN              NaN            NaN   \n",
       "4234            NaN           NaN              NaN            NaN   \n",
       "4452            NaN           NaN              NaN            NaN   \n",
       "6747            NaN           NaN              NaN            NaN   \n",
       "7121            NaN           NaN              NaN            NaN   \n",
       "9122            NaN           NaN              NaN            NaN   \n",
       "\n",
       "      Brain_Amygdala  Brain_Anterior_cingulate_cortex_BA24  \\\n",
       "218              0.0                                   0.0   \n",
       "479              0.0                                   0.0   \n",
       "1176             NaN                                   NaN   \n",
       "3281             NaN                                   NaN   \n",
       "3354             NaN                                   NaN   \n",
       "4234             NaN                                   NaN   \n",
       "4452             NaN                                   NaN   \n",
       "6747             NaN                                   NaN   \n",
       "7121             NaN                                   NaN   \n",
       "9122             NaN                                   NaN   \n",
       "\n",
       "      Brain_Caudate_basal_ganglia  Brain_Cerebellar_Hemisphere  \\\n",
       "218                           0.0                          0.0   \n",
       "479                           0.0                          0.0   \n",
       "1176                          NaN                          NaN   \n",
       "3281                          NaN                          NaN   \n",
       "3354                          NaN                          NaN   \n",
       "4234                          NaN                          NaN   \n",
       "4452                          NaN                          NaN   \n",
       "6747                          NaN                          NaN   \n",
       "7121                          NaN                          NaN   \n",
       "9122                          NaN                          NaN   \n",
       "\n",
       "      Brain_Cerebellum  Brain_Cortex  Brain_Frontal_Cortex_BA9  \\\n",
       "218                0.0           0.0                       0.0   \n",
       "479                0.0           0.0                       0.0   \n",
       "1176               NaN           NaN                       NaN   \n",
       "3281               NaN           NaN                       NaN   \n",
       "3354               NaN           NaN                       NaN   \n",
       "4234               NaN           NaN                       NaN   \n",
       "4452               NaN           NaN                       NaN   \n",
       "6747               NaN           NaN                       NaN   \n",
       "7121               NaN           NaN                       NaN   \n",
       "9122               NaN           NaN                       NaN   \n",
       "\n",
       "      Brain_Hippocampus  Brain_Hypothalamus  \\\n",
       "218                 0.0                 0.0   \n",
       "479                 0.0                 0.0   \n",
       "1176                NaN                 NaN   \n",
       "3281                NaN                 NaN   \n",
       "3354                NaN                 NaN   \n",
       "4234                NaN                 NaN   \n",
       "4452                NaN                 NaN   \n",
       "6747                NaN                 NaN   \n",
       "7121                NaN                 NaN   \n",
       "9122                NaN                 NaN   \n",
       "\n",
       "      Brain_Nucleus_accumbens_basal_ganglia  Brain_Putamen_basal_ganglia  \\\n",
       "218                                     0.0                          0.0   \n",
       "479                                     0.0                          0.0   \n",
       "1176                                    NaN                          NaN   \n",
       "3281                                    NaN                          NaN   \n",
       "3354                                    NaN                          NaN   \n",
       "4234                                    NaN                          NaN   \n",
       "4452                                    NaN                          NaN   \n",
       "6747                                    NaN                          NaN   \n",
       "7121                                    NaN                          NaN   \n",
       "9122                                    NaN                          NaN   \n",
       "\n",
       "      Brain_Spinal_cord_cervical_c-1  Brain_Substantia_nigra  \\\n",
       "218                              0.0                     0.0   \n",
       "479                              0.0                     0.0   \n",
       "1176                             NaN                     NaN   \n",
       "3281                             NaN                     NaN   \n",
       "3354                             NaN                     NaN   \n",
       "4234                             NaN                     NaN   \n",
       "4452                             NaN                     NaN   \n",
       "6747                             NaN                     NaN   \n",
       "7121                             NaN                     NaN   \n",
       "9122                             NaN                     NaN   \n",
       "\n",
       "      Breast_Mammary_Tissue  Cells_Cultured_fibroblasts  \\\n",
       "218                     1.0                         0.0   \n",
       "479                     1.0                         0.0   \n",
       "1176                    NaN                         NaN   \n",
       "3281                    NaN                         NaN   \n",
       "3354                    NaN                         NaN   \n",
       "4234                    NaN                         NaN   \n",
       "4452                    NaN                         NaN   \n",
       "6747                    NaN                         NaN   \n",
       "7121                    NaN                         NaN   \n",
       "9122                    NaN                         NaN   \n",
       "\n",
       "      Cells_EBV-transformed_lymphocytes  Colon_Sigmoid  Colon_Transverse  \\\n",
       "218                                 0.0            1.0               0.0   \n",
       "479                                 0.0            0.0               0.0   \n",
       "1176                                NaN            NaN               NaN   \n",
       "3281                                NaN            NaN               NaN   \n",
       "3354                                NaN            NaN               NaN   \n",
       "4234                                NaN            NaN               NaN   \n",
       "4452                                NaN            NaN               NaN   \n",
       "6747                                NaN            NaN               NaN   \n",
       "7121                                NaN            NaN               NaN   \n",
       "9122                                NaN            NaN               NaN   \n",
       "\n",
       "      Esophagus_Gastroesophageal_Junction  Esophagus_Mucosa  \\\n",
       "218                                   1.0               1.0   \n",
       "479                                   0.0               0.0   \n",
       "1176                                  NaN               NaN   \n",
       "3281                                  NaN               NaN   \n",
       "3354                                  NaN               NaN   \n",
       "4234                                  NaN               NaN   \n",
       "4452                                  NaN               NaN   \n",
       "6747                                  NaN               NaN   \n",
       "7121                                  NaN               NaN   \n",
       "9122                                  NaN               NaN   \n",
       "\n",
       "      Esophagus_Muscularis  Heart_Atrial_Appendage  Heart_Left_Ventricle  \\\n",
       "218                    1.0                     0.0                   0.0   \n",
       "479                    0.0                     0.0                   0.0   \n",
       "1176                   NaN                     NaN                   NaN   \n",
       "3281                   NaN                     NaN                   NaN   \n",
       "3354                   NaN                     NaN                   NaN   \n",
       "4234                   NaN                     NaN                   NaN   \n",
       "4452                   NaN                     NaN                   NaN   \n",
       "6747                   NaN                     NaN                   NaN   \n",
       "7121                   NaN                     NaN                   NaN   \n",
       "9122                   NaN                     NaN                   NaN   \n",
       "\n",
       "      Kidney_Cortex  Liver  Lung  Minor_Salivary_Gland  Muscle_Skeletal  \\\n",
       "218             0.0    0.0   1.0                   0.0              0.0   \n",
       "479             0.0    0.0   0.0                   0.0              0.0   \n",
       "1176            NaN    NaN   NaN                   NaN              NaN   \n",
       "3281            NaN    NaN   NaN                   NaN              NaN   \n",
       "3354            NaN    NaN   NaN                   NaN              NaN   \n",
       "4234            NaN    NaN   NaN                   NaN              NaN   \n",
       "4452            NaN    NaN   NaN                   NaN              NaN   \n",
       "6747            NaN    NaN   NaN                   NaN              NaN   \n",
       "7121            NaN    NaN   NaN                   NaN              NaN   \n",
       "9122            NaN    NaN   NaN                   NaN              NaN   \n",
       "\n",
       "      Nerve_Tibial  Ovary  Pancreas  Pituitary  Prostate  \\\n",
       "218            1.0    0.0       1.0        0.0       0.0   \n",
       "479            0.0    0.0       0.0        0.0       0.0   \n",
       "1176           NaN    NaN       NaN        NaN       NaN   \n",
       "3281           NaN    NaN       NaN        NaN       NaN   \n",
       "3354           NaN    NaN       NaN        NaN       NaN   \n",
       "4234           NaN    NaN       NaN        NaN       NaN   \n",
       "4452           NaN    NaN       NaN        NaN       NaN   \n",
       "6747           NaN    NaN       NaN        NaN       NaN   \n",
       "7121           NaN    NaN       NaN        NaN       NaN   \n",
       "9122           NaN    NaN       NaN        NaN       NaN   \n",
       "\n",
       "      Skin_Not_Sun_Exposed_Suprapubic  Skin_Sun_Exposed_Lower_leg  \\\n",
       "218                               1.0                         1.0   \n",
       "479                               0.0                         0.0   \n",
       "1176                              NaN                         NaN   \n",
       "3281                              NaN                         NaN   \n",
       "3354                              NaN                         NaN   \n",
       "4234                              NaN                         NaN   \n",
       "4452                              NaN                         NaN   \n",
       "6747                              NaN                         NaN   \n",
       "7121                              NaN                         NaN   \n",
       "9122                              NaN                         NaN   \n",
       "\n",
       "      Small_Intestine_Terminal_Ileum  Spleen  Stomach  Testis  Thyroid  \\\n",
       "218                              0.0     0.0      0.0     0.0      1.0   \n",
       "479                              0.0     0.0      0.0     0.0      0.0   \n",
       "1176                             NaN     NaN      NaN     NaN      NaN   \n",
       "3281                             NaN     NaN      NaN     NaN      NaN   \n",
       "3354                             NaN     NaN      NaN     NaN      NaN   \n",
       "4234                             NaN     NaN      NaN     NaN      NaN   \n",
       "4452                             NaN     NaN      NaN     NaN      NaN   \n",
       "6747                             NaN     NaN      NaN     NaN      NaN   \n",
       "7121                             NaN     NaN      NaN     NaN      NaN   \n",
       "9122                             NaN     NaN      NaN     NaN      NaN   \n",
       "\n",
       "      Uterus  Vagina  Whole_Blood  N_GTEx_tissues sQTL  \\\n",
       "218      0.0     0.0          0.0            14.0  yes   \n",
       "479      0.0     0.0          0.0             3.0  yes   \n",
       "1176     NaN     NaN          NaN             NaN   no   \n",
       "3281     NaN     NaN          NaN             NaN   no   \n",
       "3354     NaN     NaN          NaN             NaN   no   \n",
       "4234     NaN     NaN          NaN             NaN   no   \n",
       "4452     NaN     NaN          NaN             NaN   no   \n",
       "6747     NaN     NaN          NaN             NaN   no   \n",
       "7121     NaN     NaN          NaN             NaN   no   \n",
       "9122     NaN     NaN          NaN             NaN   no   \n",
       "\n",
       "        Vernot_allele_origin  Browning_allele_origin  \n",
       "218                  ancient                 ancient  \n",
       "479   low-confidence ancient  low-confidence ancient  \n",
       "1176        archaic-specific        archaic-specific  \n",
       "3281        archaic-specific        archaic-specific  \n",
       "3354                 ancient                 ancient  \n",
       "4234        archaic-specific        archaic-specific  \n",
       "4452        archaic-specific        archaic-specific  \n",
       "6747        archaic-specific        archaic-specific  \n",
       "7121        archaic-specific        archaic-specific  \n",
       "9122        archaic-specific        archaic-specific  "
      ]
     },
     "execution_count": 199,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "multiple_SAVs_per_gene = data_2[data_2.duplicated(subset=['annotation','Vernot_allele_origin'])]\n",
    "multiple_SAVs_per_gene.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 200,
   "id": "3bc785ca",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "810"
      ]
     },
     "execution_count": 200,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(multiple_SAVs_per_gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 201,
   "id": "88293ce9",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.28531172948221206"
      ]
     },
     "execution_count": 201,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "810/2839"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 202,
   "id": "a15af182",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Vernot_introgressed</th>\n",
       "      <th>Vernot_allele_origin</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>no</td>\n",
       "      <td>ancient</td>\n",
       "      <td>2252</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>2169</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>no</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "      <td>1268</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>yes</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>yes</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>244</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Vernot_introgressed    Vernot_allele_origin  count\n",
       "0                  no                 ancient   2252\n",
       "1                  no        archaic-specific   2169\n",
       "2                  no  low-confidence ancient   1268\n",
       "3                 yes        archaic-specific     17\n",
       "4                 yes            introgressed    244"
      ]
     },
     "execution_count": 202,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_2.groupby(by=['Vernot_introgressed','Vernot_allele_origin']).size().reset_index(name='count')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 203,
   "id": "b136994e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>present_in_1KG</th>\n",
       "      <th>Vernot_allele_origin</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>2186</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>no</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>no</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "      <td>232</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>yes</td>\n",
       "      <td>ancient</td>\n",
       "      <td>2252</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>yes</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>237</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>yes</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "      <td>1036</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  present_in_1KG    Vernot_allele_origin  count\n",
       "0             no        archaic-specific   2186\n",
       "1             no            introgressed      7\n",
       "2             no  low-confidence ancient    232\n",
       "3            yes                 ancient   2252\n",
       "4            yes            introgressed    237\n",
       "5            yes  low-confidence ancient   1036"
      ]
     },
     "execution_count": 203,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_2.groupby(by=['present_in_1KG','Vernot_allele_origin']).size().reset_index(name='count')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 204,
   "id": "329a5450",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Browning_introgressed</th>\n",
       "      <th>Browning_allele_origin</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>no</td>\n",
       "      <td>ancient</td>\n",
       "      <td>2195</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>2180</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>no</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "      <td>1183</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>yes</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>yes</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>386</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Browning_introgressed  Browning_allele_origin  count\n",
       "0                    no                 ancient   2195\n",
       "1                    no        archaic-specific   2180\n",
       "2                    no  low-confidence ancient   1183\n",
       "3                   yes        archaic-specific      6\n",
       "4                   yes            introgressed    386"
      ]
     },
     "execution_count": 204,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_2.groupby(by=['Browning_introgressed','Browning_allele_origin']).size().reset_index(name='count')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 205,
   "id": "31483699",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>present_in_1KG</th>\n",
       "      <th>Browning_allele_origin</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>2186</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>no</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>no</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "      <td>230</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>yes</td>\n",
       "      <td>ancient</td>\n",
       "      <td>2195</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>yes</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>377</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>yes</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "      <td>953</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  present_in_1KG  Browning_allele_origin  count\n",
       "0             no        archaic-specific   2186\n",
       "1             no            introgressed      9\n",
       "2             no  low-confidence ancient    230\n",
       "3            yes                 ancient   2195\n",
       "4            yes            introgressed    377\n",
       "5            yes  low-confidence ancient    953"
      ]
     },
     "execution_count": 205,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_2.groupby(by=['present_in_1KG','Browning_allele_origin']).size().reset_index(name='count')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3b7a8edf",
   "metadata": {},
   "source": [
    "## N Genes by Origin <a class = 'anchor' id = 'ngenesbyorigin'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "06bca2c5",
   "metadata": {},
   "source": [
    "Let's get gene lists for archaic-specific variants. Note that our current filtering has resulted in complete agreement between Vernot and Browning for archaic-specific variants so we don't need to specify the introgression callset."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 206,
   "id": "d3a03007",
   "metadata": {},
   "outputs": [],
   "source": [
    "archaic_specific_genes_2 = data_2.loc[(data_2['Vernot_allele_origin']=='archaic-specific'), 'annotation'].to_frame('gene')\n",
    "archaic_specific_genes_2 = archaic_specific_genes_2.sort_values(['gene'], ascending = [True])\n",
    "archaic_specific_genes_2.to_csv('../genes/archaic_specific_genes_2.txt', sep=\"\\t\", header=False, index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 207,
   "id": "e0e0e743",
   "metadata": {},
   "outputs": [],
   "source": [
    "archaic_specific_genes_5 = data_5.loc[(data_5['Vernot_allele_origin']=='archaic-specific'), 'annotation'].to_frame('gene')\n",
    "archaic_specific_genes_5 = archaic_specific_genes_5.sort_values(['gene'], ascending = [True])\n",
    "archaic_specific_genes_5.to_csv('../genes/archaic_specific_genes_5.txt', sep=\"\\t\", header=False, index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a44b3d08",
   "metadata": {},
   "source": [
    "How many unique genes are contained in archaic-specific vectors?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 208,
   "id": "032db1ac",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1907"
      ]
     },
     "execution_count": 208,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(archaic_specific_genes_2.drop_duplicates())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 209,
   "id": "0ac54978",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "392"
      ]
     },
     "execution_count": 209,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(archaic_specific_genes_5.drop_duplicates())"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c7915160",
   "metadata": {},
   "source": [
    "And for ancient and introgressed alleles."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 210,
   "id": "928caa94",
   "metadata": {},
   "outputs": [],
   "source": [
    "Vernot_ancient_genes_2 = data_2.loc[(data_2['Vernot_allele_origin']=='ancient'), 'annotation'].to_frame('gene')\n",
    "Vernot_ancient_genes_2 = Vernot_ancient_genes_2.sort_values(['gene'], ascending = [True])\n",
    "Vernot_ancient_genes_2.to_csv('../genes/Vernot_ancient_genes_2.txt', sep=\"\\t\", header=False, index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 211,
   "id": "4e15d8cf",
   "metadata": {},
   "outputs": [],
   "source": [
    "Vernot_ancient_genes_5 = data_5.loc[(data_5['Vernot_allele_origin']=='ancient'), 'annotation'].to_frame('gene')\n",
    "Vernot_ancient_genes_5 = Vernot_ancient_genes_5.sort_values(['gene'], ascending = [True])\n",
    "Vernot_ancient_genes_5.to_csv('../genes/Vernot_ancient_genes_5.txt', sep=\"\\t\", header=False, index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 212,
   "id": "f211a3d1",
   "metadata": {},
   "outputs": [],
   "source": [
    "Browning_ancient_genes_2 = data_2.loc[(data_2['Browning_allele_origin']=='ancient'), 'annotation'].to_frame('gene')\n",
    "Browning_ancient_genes_2 = Browning_ancient_genes_2.sort_values(['gene'], ascending = [True])\n",
    "Browning_ancient_genes_2.to_csv('../genes/Browning_ancient_genes_2.txt', sep=\"\\t\", header=False, index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 213,
   "id": "77d78010",
   "metadata": {},
   "outputs": [],
   "source": [
    "Browning_ancient_genes_5 = data_5.loc[(data_5['Browning_allele_origin']=='ancient'), 'annotation'].to_frame('gene')\n",
    "Browning_ancient_genes_5 = Browning_ancient_genes_5.sort_values(['gene'], ascending = [True])\n",
    "Browning_ancient_genes_5.to_csv('../genes/Browning_ancient_genes_5.txt', sep=\"\\t\", header=False, index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "97518559",
   "metadata": {},
   "source": [
    "Check the numebr of ancient genes."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 214,
   "id": "8dc4fc07",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1896"
      ]
     },
     "execution_count": 214,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(Vernot_ancient_genes_2.drop_duplicates())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 215,
   "id": "e85ba7e2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "370"
      ]
     },
     "execution_count": 215,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(Vernot_ancient_genes_5.drop_duplicates())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 216,
   "id": "a23a6c8e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1856"
      ]
     },
     "execution_count": 216,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(Browning_ancient_genes_2.drop_duplicates())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 217,
   "id": "b05f0676",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "363"
      ]
     },
     "execution_count": 217,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(Browning_ancient_genes_5.drop_duplicates())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 218,
   "id": "a76d649d",
   "metadata": {},
   "outputs": [],
   "source": [
    "Vernot_introgressed_genes_2 = data_2.loc[(data_2['Vernot_allele_origin']=='introgressed'), 'annotation'].to_frame('gene')\n",
    "Vernot_introgressed_genes_2 = Vernot_introgressed_genes_2.sort_values(['gene'], ascending = [True])\n",
    "Vernot_introgressed_genes_2.to_csv('../genes/Vernot_introgressed_genes_2.txt', sep=\"\\t\", header=False, index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 219,
   "id": "8a86cf7c",
   "metadata": {},
   "outputs": [],
   "source": [
    "Vernot_introgressed_genes_5 = data_5.loc[(data_5['Vernot_allele_origin']=='introgressed'), 'annotation'].to_frame('gene')\n",
    "Vernot_introgressed_genes_5 = Vernot_introgressed_genes_5.sort_values(['gene'], ascending = [True])\n",
    "Vernot_introgressed_genes_5.to_csv('../genes/Vernot_introgressed_genes_5.txt', sep=\"\\t\", header=False, index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 220,
   "id": "ca4b3d28",
   "metadata": {},
   "outputs": [],
   "source": [
    "Browning_introgressed_genes_2 = data_2.loc[(data_2['Browning_allele_origin']=='introgressed'), 'annotation'].to_frame('gene')\n",
    "Browning_introgressed_genes_2 = Browning_introgressed_genes_2.sort_values(['gene'], ascending = [True])\n",
    "Browning_introgressed_genes_2.to_csv('../genes/Browning_introgressed_genes_2.txt', sep=\"\\t\", header=False, index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 221,
   "id": "6bc279ce",
   "metadata": {},
   "outputs": [],
   "source": [
    "Browning_introgressed_genes_5 = data_5.loc[(data_5['Browning_allele_origin']=='introgressed'), 'annotation'].to_frame('gene')\n",
    "Browning_introgressed_genes_5 = Browning_introgressed_genes_5.sort_values(['gene'], ascending = [True])\n",
    "Browning_introgressed_genes_5.to_csv('../genes/Browning_introgressed_genes_5.txt', sep=\"\\t\", header=False, index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d5606b63",
   "metadata": {},
   "source": [
    "Check the numebr of introgressed genes."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 222,
   "id": "50fc6e60",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "239"
      ]
     },
     "execution_count": 222,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(Vernot_introgressed_genes_2.drop_duplicates())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 223,
   "id": "aad62d13",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "59"
      ]
     },
     "execution_count": 223,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(Vernot_introgressed_genes_5.drop_duplicates())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 224,
   "id": "9cf96512",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "361"
      ]
     },
     "execution_count": 224,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(Browning_introgressed_genes_2.drop_duplicates())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 225,
   "id": "0275dfa4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "67"
      ]
     },
     "execution_count": 225,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(Browning_introgressed_genes_5.drop_duplicates())"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ebd79fd6",
   "metadata": {},
   "source": [
    "## N Genes by Distribution <a class = 'anchor' id = 'ngenesbydistribution'></a>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 226,
   "id": "810565e1",
   "metadata": {},
   "outputs": [],
   "source": [
    "altai_genes = data_2.loc[(data_2['altai_gt_boolean']==True), 'annotation'].to_frame('gene')\n",
    "chagyrskaya_genes = data_2.loc[(data_2['chagyrskaya_gt_boolean']==True), 'annotation'].to_frame('gene')\n",
    "denisovan_genes = data_2.loc[(data_2['denisovan_gt_boolean']==True), 'annotation'].to_frame('gene')\n",
    "vindija_genes = data_2.loc[(data_2['vindija_gt_boolean']==True), 'annotation'].to_frame('gene')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 227,
   "id": "83ba01a1",
   "metadata": {},
   "outputs": [],
   "source": [
    "altai_genes = altai_genes.drop_duplicates()\n",
    "chagyrskaya_genes = chagyrskaya_genes.drop_duplicates()\n",
    "denisovan_genes = denisovan_genes.drop_duplicates()\n",
    "vindija_genes = vindija_genes.drop_duplicates()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 228,
   "id": "474900e6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2914"
      ]
     },
     "execution_count": 228,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(altai_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 229,
   "id": "3096152a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2755"
      ]
     },
     "execution_count": 229,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(chagyrskaya_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 230,
   "id": "98d14ba0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2910"
      ]
     },
     "execution_count": 230,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(denisovan_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 231,
   "id": "4c27dbc4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2889"
      ]
     },
     "execution_count": 231,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(vindija_genes)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3b11be71",
   "metadata": {},
   "source": [
    "## Archaic-Specific Genes <a class = 'anchor' id = 'archaicspecificgenes'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b894fd00",
   "metadata": {},
   "source": [
    "Now let's dig into the various archaic-specific distributions."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 232,
   "id": "913aeea8",
   "metadata": {},
   "outputs": [],
   "source": [
    "archaic_specific_altai_genes_2 = data_2.loc[(data_2['distribution']=='Altai') & (data_2['Vernot_allele_origin']=='archaic-specific'), 'annotation'].to_frame('gene')\n",
    "archaic_specific_altai_genes_2 = archaic_specific_altai_genes_2.sort_values(['gene'], ascending = [True])\n",
    "\n",
    "archaic_specific_chagyrskaya_genes_2 = data_2.loc[(data_2['distribution']=='Chagyrskaya') & (data_2['Vernot_allele_origin']=='archaic-specific'), 'annotation'].to_frame('gene')\n",
    "archaic_specific_chagyrskaya_genes_2 = archaic_specific_chagyrskaya_genes_2.sort_values(['gene'], ascending = [True])\n",
    "\n",
    "archaic_specific_denisovan_genes_2 = data_2.loc[(data_2['distribution']=='Denisovan') & (data_2['Vernot_allele_origin']=='archaic-specific'), 'annotation'].to_frame('gene')\n",
    "archaic_specific_denisovan_genes_2 = archaic_specific_denisovan_genes_2.sort_values(['gene'], ascending = [True])\n",
    "\n",
    "archaic_specific_vindija_genes_2 = data_2.loc[(data_2['distribution']=='Vindija') & (data_2['Vernot_allele_origin']=='archaic-specific'), 'annotation'].to_frame('gene')\n",
    "archaic_specific_vindija_genes_2 = archaic_specific_vindija_genes_2.sort_values(['gene'], ascending = [True])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 233,
   "id": "4f0fcc86",
   "metadata": {},
   "outputs": [],
   "source": [
    "archaic_specific_late_neanderthal_genes_2 = data_2.loc[(data_2['distribution']=='Late Neanderthal') & (data_2['Vernot_allele_origin']=='archaic-specific'), 'annotation'].to_frame('gene')\n",
    "archaic_specific_late_neanderthal_genes_2 = archaic_specific_late_neanderthal_genes_2.sort_values(['gene'], ascending = [True])\n",
    "\n",
    "archaic_specific_neanderthal_genes_2 = data_2.loc[(data_2['distribution']=='Neanderthal') & (data_2['Vernot_allele_origin']=='archaic-specific'), 'annotation'].to_frame('gene')\n",
    "archaic_specific_neanderthal_genes_2 = archaic_specific_neanderthal_genes_2.sort_values(['gene'], ascending = [True])\n",
    "\n",
    "archaic_specific_shared_genes_2 = data_2.loc[(data_2['distribution']=='Shared') & (data_2['Vernot_allele_origin']=='archaic-specific'), 'annotation'].to_frame('gene')\n",
    "archaic_specific_shared_genes_2 = archaic_specific_shared_genes_2.sort_values(['gene'], ascending = [True])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 234,
   "id": "7d756afa",
   "metadata": {},
   "outputs": [],
   "source": [
    "archaic_specific_altai_genes_2.to_csv('../genes/archaic_specific_altai_genes_2.txt', sep=\"\\t\", header=False, index=False)\n",
    "archaic_specific_chagyrskaya_genes_2.to_csv('../genes/archaic_specific_chagyrskaya_genes_2.txt', sep=\"\\t\", header=False, index=False)\n",
    "archaic_specific_denisovan_genes_2.to_csv('../genes/archaic_specific_denisovan_genes_2.txt', sep=\"\\t\", header=False, index=False)\n",
    "archaic_specific_vindija_genes_2.to_csv('../genes/archaic_specific_vindija_genes_2.txt', sep=\"\\t\", header=False, index=False)\n",
    "archaic_specific_late_neanderthal_genes_2.to_csv('../genes/archaic_specific_late_neanderthal_genes_2.txt', sep=\"\\t\", header=False, index=False)\n",
    "archaic_specific_neanderthal_genes_2.to_csv('../genes/archaic_specific_neanderthal_genes_2.txt', sep=\"\\t\", header=False, index=False)\n",
    "archaic_specific_shared_genes_2.to_csv('../genes/archaic_specific_shared_genes_2.txt', sep=\"\\t\", header=False, index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "90e73251",
   "metadata": {},
   "source": [
    "How many unique genes are contained in lineage-specific vectors?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 235,
   "id": "bb0691af",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "283"
      ]
     },
     "execution_count": 235,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "archaic_specific_altai_genes_2_no_dups = archaic_specific_altai_genes_2.drop_duplicates()\n",
    "len(archaic_specific_altai_genes_2_no_dups)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 236,
   "id": "4c816187",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "165"
      ]
     },
     "execution_count": 236,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "archaic_specific_chagyrskaya_genes_2_no_dups = archaic_specific_chagyrskaya_genes_2.drop_duplicates()\n",
    "len(archaic_specific_chagyrskaya_genes_2_no_dups)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 237,
   "id": "b2c0bc07",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "859"
      ]
     },
     "execution_count": 237,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "archaic_specific_denisovan_genes_2_no_dups = archaic_specific_denisovan_genes_2.drop_duplicates()\n",
    "len(archaic_specific_denisovan_genes_2_no_dups)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 238,
   "id": "7cec75d5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "228"
      ]
     },
     "execution_count": 238,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "archaic_specific_vindija_genes_2_no_dups = archaic_specific_vindija_genes_2.drop_duplicates()\n",
    "len(archaic_specific_vindija_genes_2_no_dups)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 239,
   "id": "06d4f631",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "93"
      ]
     },
     "execution_count": 239,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "archaic_specific_late_neanderthal_genes_2_no_dups = archaic_specific_late_neanderthal_genes_2.drop_duplicates()\n",
    "len(archaic_specific_late_neanderthal_genes_2_no_dups)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 240,
   "id": "f23645f6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "227"
      ]
     },
     "execution_count": 240,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "archaic_specific_neanderthal_genes_2_no_dups = archaic_specific_neanderthal_genes_2.drop_duplicates()\n",
    "len(archaic_specific_neanderthal_genes_2_no_dups)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 241,
   "id": "4bc16407",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "106"
      ]
     },
     "execution_count": 241,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "archaic_specific_shared_genes_2_no_dups = archaic_specific_shared_genes_2.drop_duplicates()\n",
    "len(archaic_specific_shared_genes_2_no_dups)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e47da429",
   "metadata": {},
   "source": [
    "Repeat for delta >= 0.5."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 242,
   "id": "a4cdcc35",
   "metadata": {},
   "outputs": [],
   "source": [
    "archaic_specific_altai_genes_5 = data_5.loc[(data_5['distribution']=='Altai') & (data_5['Vernot_allele_origin']=='archaic-specific'), 'annotation'].to_frame('gene')\n",
    "archaic_specific_altai_genes_5 = archaic_specific_altai_genes_5.sort_values(['gene'], ascending = [True])\n",
    "\n",
    "archaic_specific_chagyrskaya_genes_5 = data_5.loc[(data_5['distribution']=='Chagyrskaya') & (data_5['Vernot_allele_origin']=='archaic-specific'), 'annotation'].to_frame('gene')\n",
    "archaic_specific_chagyrskaya_genes_5 = archaic_specific_chagyrskaya_genes_5.sort_values(['gene'], ascending = [True])\n",
    "\n",
    "archaic_specific_denisovan_genes_5 = data_5.loc[(data_5['distribution']=='Denisovan') & (data_5['Vernot_allele_origin']=='archaic-specific'), 'annotation'].to_frame('gene')\n",
    "archaic_specific_denisovan_genes_5 = archaic_specific_denisovan_genes_5.sort_values(['gene'], ascending = [True])\n",
    "\n",
    "archaic_specific_vindija_genes_5 = data_5.loc[(data_5['distribution']=='Vindija') & (data_5['Vernot_allele_origin']=='archaic-specific'), 'annotation'].to_frame('gene')\n",
    "archaic_specific_vindija_genes_5 = archaic_specific_vindija_genes_5.sort_values(['gene'], ascending = [True])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 243,
   "id": "78ab9732",
   "metadata": {},
   "outputs": [],
   "source": [
    "archaic_specific_late_neanderthal_genes_5 = data_5.loc[(data_5['distribution']=='Late Neanderthal') & (data_5['Vernot_allele_origin']=='archaic-specific'), 'annotation'].to_frame('gene')\n",
    "archaic_specific_late_neanderthal_genes_5 = archaic_specific_late_neanderthal_genes_5.sort_values(['gene'], ascending = [True])\n",
    "\n",
    "archaic_specific_neanderthal_genes_5 = data_5.loc[(data_5['distribution']=='Neanderthal') & (data_5['Vernot_allele_origin']=='archaic-specific'), 'annotation'].to_frame('gene')\n",
    "archaic_specific_neanderthal_genes_5 = archaic_specific_neanderthal_genes_5.sort_values(['gene'], ascending = [True])\n",
    "\n",
    "archaic_specific_shared_genes_5 = data_5.loc[(data_5['distribution']=='Shared') & (data_5['Vernot_allele_origin']=='archaic-specific'), 'annotation'].to_frame('gene')\n",
    "archaic_specific_shared_genes_5 = archaic_specific_shared_genes_5.sort_values(['gene'], ascending = [True])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 244,
   "id": "b000b8da",
   "metadata": {},
   "outputs": [],
   "source": [
    "archaic_specific_altai_genes_5.to_csv('../genes/archaic_specific_altai_genes_5.txt', sep='\\t', header=False, index=False)\n",
    "archaic_specific_chagyrskaya_genes_5.to_csv('../genes/archaic_specific_chagyrskaya_genes_5.txt', sep='\\t', header=False, index=False)\n",
    "archaic_specific_denisovan_genes_5.to_csv('../genes/archaic_specific_denisovan_genes_5.txt', sep='\\t', header=False, index=False)\n",
    "archaic_specific_vindija_genes_5.to_csv('../genes/archaic_specific_vindija_genes_5.txt', sep='\\t', header=False, index=False)\n",
    "archaic_specific_late_neanderthal_genes_5.to_csv('../genes/archaic_specific_late_neanderthal_genes_5.txt', sep='\\t', header=False, index=False)\n",
    "archaic_specific_neanderthal_genes_5.to_csv('../genes/archaic_specific_neanderthal_genes_5.txt', sep='\\t', header=False, index=False)\n",
    "archaic_specific_shared_genes_5.to_csv('../genes/archaic_specific_shared_genes_5.txt', sep='\\t', header=False, index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 245,
   "id": "220430a7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "52"
      ]
     },
     "execution_count": 245,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(archaic_specific_altai_genes_5.drop_duplicates())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 246,
   "id": "fe61b6cb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "27"
      ]
     },
     "execution_count": 246,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(archaic_specific_chagyrskaya_genes_5.drop_duplicates())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 247,
   "id": "f6b85907",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "161"
      ]
     },
     "execution_count": 247,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(archaic_specific_denisovan_genes_5.drop_duplicates())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 248,
   "id": "f0b1cd48",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "55"
      ]
     },
     "execution_count": 248,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(archaic_specific_vindija_genes_5.drop_duplicates())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 249,
   "id": "88c3d018",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "10"
      ]
     },
     "execution_count": 249,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(archaic_specific_late_neanderthal_genes_5.drop_duplicates())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 250,
   "id": "d2e88f92",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "36"
      ]
     },
     "execution_count": 250,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(archaic_specific_neanderthal_genes_5.drop_duplicates())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 251,
   "id": "4253f004",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "26"
      ]
     },
     "execution_count": 251,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(archaic_specific_shared_genes_5.drop_duplicates())"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "18fde8c1",
   "metadata": {},
   "source": [
    "## Gene Overlap <a class = 'anchor' id = 'geneoverlap'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ff1f4c34",
   "metadata": {},
   "source": [
    "Do genes overlap in the archaic-specific sets?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 252,
   "id": "e42bce4f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "5"
      ]
     },
     "execution_count": 252,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "neanderthal_altai_genes = pd.merge(archaic_specific_neanderthal_genes_2_no_dups, archaic_specific_altai_genes_2_no_dups)\n",
    "len(neanderthal_altai_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 253,
   "id": "f0700d55",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1"
      ]
     },
     "execution_count": 253,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "neanderthal_chagyrskaya_genes = pd.merge(archaic_specific_neanderthal_genes_2_no_dups, archaic_specific_chagyrskaya_genes_2_no_dups)\n",
    "len(neanderthal_chagyrskaya_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 254,
   "id": "62ff9de0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "20"
      ]
     },
     "execution_count": 254,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "neanderthal_denisovan_genes = pd.merge(archaic_specific_neanderthal_genes_2_no_dups, archaic_specific_denisovan_genes_2_no_dups)\n",
    "len(neanderthal_denisovan_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 255,
   "id": "1a8a8fb2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "3"
      ]
     },
     "execution_count": 255,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "neanderthal_vindija_genes = pd.merge(archaic_specific_neanderthal_genes_2_no_dups, archaic_specific_vindija_genes_2_no_dups)\n",
    "len(neanderthal_vindija_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 256,
   "id": "5c836e49",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "9"
      ]
     },
     "execution_count": 256,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "altai_chagyrskaya_genes = pd.merge(archaic_specific_altai_genes_2_no_dups, archaic_specific_chagyrskaya_genes_2_no_dups)\n",
    "len(altai_chagyrskaya_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 257,
   "id": "cfa29ef7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "34"
      ]
     },
     "execution_count": 257,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "altai_denisovan_genes = pd.merge(archaic_specific_altai_genes_2_no_dups, archaic_specific_denisovan_genes_2_no_dups)\n",
    "len(altai_denisovan_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 258,
   "id": "3d147805",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "3"
      ]
     },
     "execution_count": 258,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "altai_vindija_genes = pd.merge(archaic_specific_altai_genes_2_no_dups, archaic_specific_vindija_genes_2_no_dups)\n",
    "len(altai_vindija_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 259,
   "id": "89d730ea",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "20"
      ]
     },
     "execution_count": 259,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "chagyrskaya_denisovan_genes = pd.merge(archaic_specific_chagyrskaya_genes_2_no_dups, archaic_specific_denisovan_genes_2_no_dups)\n",
    "len(chagyrskaya_denisovan_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 260,
   "id": "992cb2c4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "6"
      ]
     },
     "execution_count": 260,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "chagyrskaya_vindija_genes = pd.merge(archaic_specific_chagyrskaya_genes_2_no_dups, archaic_specific_vindija_genes_2_no_dups)\n",
    "len(chagyrskaya_vindija_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 261,
   "id": "50269a16",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "19"
      ]
     },
     "execution_count": 261,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "denisovan_vindija_genes = pd.merge(archaic_specific_denisovan_genes_2_no_dups, archaic_specific_vindija_genes_2_no_dups)\n",
    "len(denisovan_vindija_genes)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b5e73eaa",
   "metadata": {},
   "source": [
    "# Comparisons <a class = 'anchor' id = 'comparisons'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "120c63dc",
   "metadata": {},
   "source": [
    "## SAV Genes and DR Genes <a class = 'anchor' id = 'drgenes'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "38662c20",
   "metadata": {},
   "source": [
    "Let's take a second to compare genes with SAVs with divergently regulated genes from Colbran et al. 2019."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 262,
   "id": "d8965485",
   "metadata": {},
   "outputs": [],
   "source": [
    "altai_genes = data_2.loc[(data_2['altai_gt_boolean']==True), 'annotation'].to_frame('gene')\n",
    "denisovan_genes = data_2.loc[(data_2['denisovan_gt_boolean']==True), 'annotation'].to_frame('gene')\n",
    "vindija_genes = data_2.loc[(data_2['vindija_gt_boolean']==True), 'annotation'].to_frame('gene')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 263,
   "id": "94ecf15d",
   "metadata": {},
   "outputs": [],
   "source": [
    "altai_genes = altai_genes.drop_duplicates()\n",
    "denisovan_genes = denisovan_genes.drop_duplicates()\n",
    "vindija_genes = vindija_genes.drop_duplicates()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 264,
   "id": "cb234290",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2914"
      ]
     },
     "execution_count": 264,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(altai_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 265,
   "id": "f2c49e4c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2910"
      ]
     },
     "execution_count": 265,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(denisovan_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 266,
   "id": "32383760",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2889"
      ]
     },
     "execution_count": 266,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(vindija_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 267,
   "id": "8afa1213",
   "metadata": {},
   "outputs": [],
   "source": [
    "altai_DR_genes = pd.read_csv('../DR_genes/altai_DR_genes.txt', sep = '\\t', names = ['gene'])\n",
    "denisovan_DR_genes = pd.read_csv('../DR_genes/denisovan_DR_genes.txt', sep = '\\t', names = ['gene'])\n",
    "vindija_DR_genes = pd.read_csv('../DR_genes/vindija_DR_genes.txt', sep = '\\t', names = ['gene'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 268,
   "id": "370943d8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1419"
      ]
     },
     "execution_count": 268,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(altai_DR_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 269,
   "id": "d2129af7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1171"
      ]
     },
     "execution_count": 269,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(denisovan_DR_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 270,
   "id": "7e44a9da",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1484"
      ]
     },
     "execution_count": 270,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(vindija_DR_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 271,
   "id": "05ecf1f4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "213"
      ]
     },
     "execution_count": 271,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "altai_DR_overlap = pd.merge(altai_genes, altai_DR_genes)\n",
    "len(altai_DR_overlap)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 272,
   "id": "fff54f31",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "154"
      ]
     },
     "execution_count": 272,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "denisovan_DR_overlap = pd.merge(denisovan_genes, denisovan_DR_genes)\n",
    "len(denisovan_DR_overlap)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 273,
   "id": "8bb4248d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "222"
      ]
     },
     "execution_count": 273,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vindija_DR_overlap = pd.merge(vindija_genes, vindija_DR_genes)\n",
    "len(vindija_DR_overlap)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 274,
   "id": "4e16a4bc",
   "metadata": {},
   "outputs": [],
   "source": [
    "altai_DR_overlap.to_csv('../DR_genes/altai_DR_overlap.txt', sep='\\t', header=False, index=False)\n",
    "denisovan_DR_overlap.to_csv('../DR_genes/denisovan_DR_overlap.txt', sep='\\t', header=False, index=False)\n",
    "vindija_DR_overlap.to_csv('../DR_genes/vindija_DR_overlap.txt', sep='\\t', header=False, index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "64aef042",
   "metadata": {},
   "source": [
    "## SAV Genes and Circadian Genes <a class = 'anchor' id = 'circadiangenes'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "223ff657",
   "metadata": {},
   "source": [
    "Let's also compare the overlap between genes involved in circadian biology (Velázquez-Arcelay et al., in prep) and our genes with SAVs."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 275,
   "id": "a174ef20",
   "metadata": {},
   "outputs": [],
   "source": [
    "archaic_specific_genes_2_no_dups = archaic_specific_genes_2.drop_duplicates()\n",
    "Vernot_introgressed_genes_2_no_dups = Vernot_introgressed_genes_2.drop_duplicates()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 276,
   "id": "387ed1e4",
   "metadata": {},
   "outputs": [],
   "source": [
    "circadian_genes = pd.read_csv('../circadian_genes/circadian_genes.txt', sep = '\\t', header = 0)\n",
    "circadian_genes = circadian_genes[['GeneName']]\n",
    "circadian_genes.rename(columns={'GeneName': 'gene'}, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 277,
   "id": "3198710e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "27"
      ]
     },
     "execution_count": 277,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "circadian_archaic_specific_overlap = pd.merge(archaic_specific_genes_2_no_dups, circadian_genes)\n",
    "len(circadian_archaic_specific_overlap)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 278,
   "id": "84c52d8a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "4"
      ]
     },
     "execution_count": 278,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "circadian_introgressed_overlap = pd.merge(Vernot_introgressed_genes_2_no_dups, circadian_genes)\n",
    "len(circadian_introgressed_overlap)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 279,
   "id": "9e95caaa",
   "metadata": {},
   "outputs": [],
   "source": [
    "circadian_archaic_specific_overlap.to_csv('../circadian_genes/circadian_genes_archaic_specific_overlap.txt', sep='\\t', header=False, index=False)\n",
    "circadian_introgressed_overlap.to_csv('../circadian_genes/circadian_genes_introgressed_overlap.txt', sep='\\t', header=False, index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3e674a31",
   "metadata": {},
   "source": [
    "Let's also subset the main dataframe for any variants that occur in circadian genes."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 280,
   "id": "c294e238",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PER3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>RERE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>DNAJC16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ECE1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HTR1D</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      gene\n",
       "0     PER3\n",
       "1     RERE\n",
       "2  DNAJC16\n",
       "3     ECE1\n",
       "4    HTR1D"
      ]
     },
     "execution_count": 280,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "circadian_genes.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 281,
   "id": "838e92db",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref_allele</th>\n",
       "      <th>alt_allele</th>\n",
       "      <th>ancestral_allele</th>\n",
       "      <th>anc_dev</th>\n",
       "      <th>variant_type</th>\n",
       "      <th>altai_gt</th>\n",
       "      <th>chagyrskaya_gt</th>\n",
       "      <th>denisovan_gt</th>\n",
       "      <th>vindija_gt</th>\n",
       "      <th>altai_gt_boolean</th>\n",
       "      <th>chagyrskaya_gt_boolean</th>\n",
       "      <th>denisovan_gt_boolean</th>\n",
       "      <th>vindija_gt_boolean</th>\n",
       "      <th>distribution</th>\n",
       "      <th>present_in_1KG</th>\n",
       "      <th>1KG_allele_count</th>\n",
       "      <th>1KG_allele_number</th>\n",
       "      <th>1KG_allele_frequency</th>\n",
       "      <th>1KG_EAS_AF</th>\n",
       "      <th>1KG_EUR_AF</th>\n",
       "      <th>1KG_AFR_AF</th>\n",
       "      <th>1KG_AMR_AF</th>\n",
       "      <th>1KG_SAS_AF</th>\n",
       "      <th>1KG_non_ASW_AFR_AF</th>\n",
       "      <th>present_in_gnomAD</th>\n",
       "      <th>gnomAD_allele_count</th>\n",
       "      <th>gnomAD_allele_number</th>\n",
       "      <th>gnomAD_allele_frequency</th>\n",
       "      <th>Vernot_introgressed</th>\n",
       "      <th>Vernot_ancestral_allele</th>\n",
       "      <th>Vernot_derived_allele</th>\n",
       "      <th>Vernot_ancestral_derived_code</th>\n",
       "      <th>Vernot_AFA_AF</th>\n",
       "      <th>Vernot_AFR_AF</th>\n",
       "      <th>Vernot_AMR_AF</th>\n",
       "      <th>Vernot_EAS_AF</th>\n",
       "      <th>Vernot_EUR_AF</th>\n",
       "      <th>Vernot_PNG_AF</th>\n",
       "      <th>Vernot_SAS_AF</th>\n",
       "      <th>Vernot_Denisovan_base</th>\n",
       "      <th>Vernot_Neanderthal_base_1</th>\n",
       "      <th>Vernot_Neanderthal_base_2</th>\n",
       "      <th>Vernot_haplotype_tag</th>\n",
       "      <th>Vernot_introgressed_AF</th>\n",
       "      <th>Browning_introgressed</th>\n",
       "      <th>Browning_ref_alt</th>\n",
       "      <th>Browning_introgressed_AF</th>\n",
       "      <th>annotation</th>\n",
       "      <th>mis_oe</th>\n",
       "      <th>mis_z</th>\n",
       "      <th>lof_oe</th>\n",
       "      <th>lof_z</th>\n",
       "      <th>phyloP</th>\n",
       "      <th>ag_delta</th>\n",
       "      <th>al_delta</th>\n",
       "      <th>dg_delta</th>\n",
       "      <th>dl_delta</th>\n",
       "      <th>delta_max</th>\n",
       "      <th>ag_pos</th>\n",
       "      <th>al_pos</th>\n",
       "      <th>dg_pos</th>\n",
       "      <th>dl_pos</th>\n",
       "      <th>Adipose_Subcutaneous</th>\n",
       "      <th>Adipose_Visceral_Omentum</th>\n",
       "      <th>Adrenal_Gland</th>\n",
       "      <th>Artery_Aorta</th>\n",
       "      <th>Artery_Coronary</th>\n",
       "      <th>Artery_Tibial</th>\n",
       "      <th>Brain_Amygdala</th>\n",
       "      <th>Brain_Anterior_cingulate_cortex_BA24</th>\n",
       "      <th>Brain_Caudate_basal_ganglia</th>\n",
       "      <th>Brain_Cerebellar_Hemisphere</th>\n",
       "      <th>Brain_Cerebellum</th>\n",
       "      <th>Brain_Cortex</th>\n",
       "      <th>Brain_Frontal_Cortex_BA9</th>\n",
       "      <th>Brain_Hippocampus</th>\n",
       "      <th>Brain_Hypothalamus</th>\n",
       "      <th>Brain_Nucleus_accumbens_basal_ganglia</th>\n",
       "      <th>Brain_Putamen_basal_ganglia</th>\n",
       "      <th>Brain_Spinal_cord_cervical_c-1</th>\n",
       "      <th>Brain_Substantia_nigra</th>\n",
       "      <th>Breast_Mammary_Tissue</th>\n",
       "      <th>Cells_Cultured_fibroblasts</th>\n",
       "      <th>Cells_EBV-transformed_lymphocytes</th>\n",
       "      <th>Colon_Sigmoid</th>\n",
       "      <th>Colon_Transverse</th>\n",
       "      <th>Esophagus_Gastroesophageal_Junction</th>\n",
       "      <th>Esophagus_Mucosa</th>\n",
       "      <th>Esophagus_Muscularis</th>\n",
       "      <th>Heart_Atrial_Appendage</th>\n",
       "      <th>Heart_Left_Ventricle</th>\n",
       "      <th>Kidney_Cortex</th>\n",
       "      <th>Liver</th>\n",
       "      <th>Lung</th>\n",
       "      <th>Minor_Salivary_Gland</th>\n",
       "      <th>Muscle_Skeletal</th>\n",
       "      <th>Nerve_Tibial</th>\n",
       "      <th>Ovary</th>\n",
       "      <th>Pancreas</th>\n",
       "      <th>Pituitary</th>\n",
       "      <th>Prostate</th>\n",
       "      <th>Skin_Not_Sun_Exposed_Suprapubic</th>\n",
       "      <th>Skin_Sun_Exposed_Lower_leg</th>\n",
       "      <th>Small_Intestine_Terminal_Ileum</th>\n",
       "      <th>Spleen</th>\n",
       "      <th>Stomach</th>\n",
       "      <th>Testis</th>\n",
       "      <th>Thyroid</th>\n",
       "      <th>Uterus</th>\n",
       "      <th>Vagina</th>\n",
       "      <th>Whole_Blood</th>\n",
       "      <th>N_GTEx_tissues</th>\n",
       "      <th>sQTL</th>\n",
       "      <th>Vernot_allele_origin</th>\n",
       "      <th>Browning_allele_origin</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>7274</th>\n",
       "      <td>chr1</td>\n",
       "      <td>7845695</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>ancestral</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>yes</td>\n",
       "      <td>3690.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.72</td>\n",
       "      <td>0.76</td>\n",
       "      <td>0.66</td>\n",
       "      <td>0.78</td>\n",
       "      <td>0.76</td>\n",
       "      <td>0.65</td>\n",
       "      <td>0.785575</td>\n",
       "      <td>yes</td>\n",
       "      <td>98348.0</td>\n",
       "      <td>143106.0</td>\n",
       "      <td>0.687239</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PER3</td>\n",
       "      <td>1.0116</td>\n",
       "      <td>-0.10855</td>\n",
       "      <td>0.63671</td>\n",
       "      <td>2.5313</td>\n",
       "      <td>-0.215</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.02</td>\n",
       "      <td>36</td>\n",
       "      <td>19</td>\n",
       "      <td>-49</td>\n",
       "      <td>19</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>ancient</td>\n",
       "      <td>ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7275</th>\n",
       "      <td>chr1</td>\n",
       "      <td>7846527</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>yes</td>\n",
       "      <td>2249.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.44</td>\n",
       "      <td>0.45</td>\n",
       "      <td>0.37</td>\n",
       "      <td>0.62</td>\n",
       "      <td>0.26</td>\n",
       "      <td>0.38</td>\n",
       "      <td>0.639376</td>\n",
       "      <td>yes</td>\n",
       "      <td>59442.0</td>\n",
       "      <td>142192.0</td>\n",
       "      <td>0.418040</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PER3</td>\n",
       "      <td>1.0116</td>\n",
       "      <td>-0.10855</td>\n",
       "      <td>0.63671</td>\n",
       "      <td>2.5313</td>\n",
       "      <td>-0.379</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>13</td>\n",
       "      <td>-7</td>\n",
       "      <td>-12</td>\n",
       "      <td>-7</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>ancient</td>\n",
       "      <td>ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7276</th>\n",
       "      <td>chr1</td>\n",
       "      <td>7849200</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>./.</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Other</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>2.0</td>\n",
       "      <td>125184.0</td>\n",
       "      <td>0.000016</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PER3</td>\n",
       "      <td>1.0116</td>\n",
       "      <td>-0.10855</td>\n",
       "      <td>0.63671</td>\n",
       "      <td>2.5313</td>\n",
       "      <td>0.172</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>22</td>\n",
       "      <td>-20</td>\n",
       "      <td>-2</td>\n",
       "      <td>-27</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7277</th>\n",
       "      <td>chr1</td>\n",
       "      <td>7853294</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>6.0</td>\n",
       "      <td>143348.0</td>\n",
       "      <td>0.000042</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PER3</td>\n",
       "      <td>1.0116</td>\n",
       "      <td>-0.10855</td>\n",
       "      <td>0.63671</td>\n",
       "      <td>2.5313</td>\n",
       "      <td>-0.371</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-47</td>\n",
       "      <td>17</td>\n",
       "      <td>-47</td>\n",
       "      <td>3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7278</th>\n",
       "      <td>chr1</td>\n",
       "      <td>7853884</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>ancestral</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PER3</td>\n",
       "      <td>1.0116</td>\n",
       "      <td>-0.10855</td>\n",
       "      <td>0.63671</td>\n",
       "      <td>2.5313</td>\n",
       "      <td>-2.062</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-30</td>\n",
       "      <td>47</td>\n",
       "      <td>45</td>\n",
       "      <td>-48</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7279</th>\n",
       "      <td>chr1</td>\n",
       "      <td>7855814</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>yes</td>\n",
       "      <td>2757.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.54</td>\n",
       "      <td>0.80</td>\n",
       "      <td>0.37</td>\n",
       "      <td>0.62</td>\n",
       "      <td>0.46</td>\n",
       "      <td>0.39</td>\n",
       "      <td>0.638402</td>\n",
       "      <td>yes</td>\n",
       "      <td>63786.0</td>\n",
       "      <td>143078.0</td>\n",
       "      <td>0.445813</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PER3</td>\n",
       "      <td>1.0116</td>\n",
       "      <td>-0.10855</td>\n",
       "      <td>0.63671</td>\n",
       "      <td>2.5313</td>\n",
       "      <td>0.482</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.02</td>\n",
       "      <td>1</td>\n",
       "      <td>40</td>\n",
       "      <td>1</td>\n",
       "      <td>-36</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>ancient</td>\n",
       "      <td>ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7280</th>\n",
       "      <td>chr1</td>\n",
       "      <td>7856346</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>yes</td>\n",
       "      <td>1584.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.31</td>\n",
       "      <td>0.19</td>\n",
       "      <td>0.38</td>\n",
       "      <td>0.27</td>\n",
       "      <td>0.29</td>\n",
       "      <td>0.43</td>\n",
       "      <td>0.264133</td>\n",
       "      <td>yes</td>\n",
       "      <td>49122.0</td>\n",
       "      <td>140236.0</td>\n",
       "      <td>0.350281</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PER3</td>\n",
       "      <td>1.0116</td>\n",
       "      <td>-0.10855</td>\n",
       "      <td>0.63671</td>\n",
       "      <td>2.5313</td>\n",
       "      <td>0.358</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-28</td>\n",
       "      <td>29</td>\n",
       "      <td>8</td>\n",
       "      <td>-42</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>ancient</td>\n",
       "      <td>ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7281</th>\n",
       "      <td>chr1</td>\n",
       "      <td>7857156</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PER3</td>\n",
       "      <td>1.0116</td>\n",
       "      <td>-0.10855</td>\n",
       "      <td>0.63671</td>\n",
       "      <td>2.5313</td>\n",
       "      <td>-0.372</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-12</td>\n",
       "      <td>15</td>\n",
       "      <td>17</td>\n",
       "      <td>-6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7282</th>\n",
       "      <td>chr1</td>\n",
       "      <td>7858042</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>2.0</td>\n",
       "      <td>143206.0</td>\n",
       "      <td>0.000014</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PER3</td>\n",
       "      <td>1.0116</td>\n",
       "      <td>-0.10855</td>\n",
       "      <td>0.63671</td>\n",
       "      <td>2.5313</td>\n",
       "      <td>-0.140</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>2</td>\n",
       "      <td>33</td>\n",
       "      <td>2</td>\n",
       "      <td>-39</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7283</th>\n",
       "      <td>chr1</td>\n",
       "      <td>7858925</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>ancestral</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>yes</td>\n",
       "      <td>4973.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.98</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.95</td>\n",
       "      <td>0.98</td>\n",
       "      <td>0.97</td>\n",
       "      <td>0.98</td>\n",
       "      <td>0.978558</td>\n",
       "      <td>yes</td>\n",
       "      <td>138460.0</td>\n",
       "      <td>143308.0</td>\n",
       "      <td>0.966171</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PER3</td>\n",
       "      <td>1.0116</td>\n",
       "      <td>-0.10855</td>\n",
       "      <td>0.63671</td>\n",
       "      <td>2.5313</td>\n",
       "      <td>-0.726</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-18</td>\n",
       "      <td>13</td>\n",
       "      <td>46</td>\n",
       "      <td>-2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>ancient</td>\n",
       "      <td>ancient</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     chrom      pos ref_allele alt_allele ancestral_allele    anc_dev  \\\n",
       "7274  chr1  7845695          T          C                C  ancestral   \n",
       "7275  chr1  7846527          A          C                A    derived   \n",
       "7276  chr1  7849200          A          T                A    derived   \n",
       "7277  chr1  7853294          A          G                A    derived   \n",
       "7278  chr1  7853884          G          A                A  ancestral   \n",
       "7279  chr1  7855814          A          G                A    derived   \n",
       "7280  chr1  7856346          T          C                T    derived   \n",
       "7281  chr1  7857156          T          A                T    derived   \n",
       "7282  chr1  7858042          G          A                G    derived   \n",
       "7283  chr1  7858925          T          C                C  ancestral   \n",
       "\n",
       "     variant_type altai_gt chagyrskaya_gt denisovan_gt vindija_gt  \\\n",
       "7274          snv      1/1            1/1          1/1        1/1   \n",
       "7275          snv      1/1            1/1          1/1        1/1   \n",
       "7276          snv      1/1            ./.          1/1        1/1   \n",
       "7277          snv      1/1            1/1          1/1        1/1   \n",
       "7278          snv      1/1            1/1          1/1        1/1   \n",
       "7279          snv      1/1            1/1          1/1        1/1   \n",
       "7280          snv      1/1            1/1          1/1        1/1   \n",
       "7281          snv      1/1            1/1          0/0        1/1   \n",
       "7282          snv      1/1            1/1          1/1        1/1   \n",
       "7283          snv      1/1            1/1          1/1        1/1   \n",
       "\n",
       "      altai_gt_boolean  chagyrskaya_gt_boolean  denisovan_gt_boolean  \\\n",
       "7274              True                    True                  True   \n",
       "7275              True                    True                  True   \n",
       "7276              True                   False                  True   \n",
       "7277              True                    True                  True   \n",
       "7278              True                    True                  True   \n",
       "7279              True                    True                  True   \n",
       "7280              True                    True                  True   \n",
       "7281              True                    True                 False   \n",
       "7282              True                    True                  True   \n",
       "7283              True                    True                  True   \n",
       "\n",
       "      vindija_gt_boolean distribution present_in_1KG  1KG_allele_count  \\\n",
       "7274                True       Shared            yes            3690.0   \n",
       "7275                True       Shared            yes            2249.0   \n",
       "7276                True        Other             no               NaN   \n",
       "7277                True       Shared             no               NaN   \n",
       "7278                True       Shared             no               NaN   \n",
       "7279                True       Shared            yes            2757.0   \n",
       "7280                True       Shared            yes            1584.0   \n",
       "7281                True  Neanderthal             no               NaN   \n",
       "7282                True       Shared             no               NaN   \n",
       "7283                True       Shared            yes            4973.0   \n",
       "\n",
       "      1KG_allele_number  1KG_allele_frequency  1KG_EAS_AF  1KG_EUR_AF  \\\n",
       "7274             5096.0                  0.72        0.76        0.66   \n",
       "7275             5096.0                  0.44        0.45        0.37   \n",
       "7276                NaN                   NaN         NaN         NaN   \n",
       "7277                NaN                   NaN         NaN         NaN   \n",
       "7278                NaN                   NaN         NaN         NaN   \n",
       "7279             5096.0                  0.54        0.80        0.37   \n",
       "7280             5096.0                  0.31        0.19        0.38   \n",
       "7281                NaN                   NaN         NaN         NaN   \n",
       "7282                NaN                   NaN         NaN         NaN   \n",
       "7283             5096.0                  0.98        1.00        0.95   \n",
       "\n",
       "      1KG_AFR_AF  1KG_AMR_AF  1KG_SAS_AF  1KG_non_ASW_AFR_AF  \\\n",
       "7274        0.78        0.76        0.65            0.785575   \n",
       "7275        0.62        0.26        0.38            0.639376   \n",
       "7276         NaN         NaN         NaN                 NaN   \n",
       "7277         NaN         NaN         NaN                 NaN   \n",
       "7278         NaN         NaN         NaN                 NaN   \n",
       "7279        0.62        0.46        0.39            0.638402   \n",
       "7280        0.27        0.29        0.43            0.264133   \n",
       "7281         NaN         NaN         NaN                 NaN   \n",
       "7282         NaN         NaN         NaN                 NaN   \n",
       "7283        0.98        0.97        0.98            0.978558   \n",
       "\n",
       "     present_in_gnomAD  gnomAD_allele_count  gnomAD_allele_number  \\\n",
       "7274               yes              98348.0              143106.0   \n",
       "7275               yes              59442.0              142192.0   \n",
       "7276               yes                  2.0              125184.0   \n",
       "7277               yes                  6.0              143348.0   \n",
       "7278                no                  NaN                   NaN   \n",
       "7279               yes              63786.0              143078.0   \n",
       "7280               yes              49122.0              140236.0   \n",
       "7281                no                  NaN                   NaN   \n",
       "7282               yes                  2.0              143206.0   \n",
       "7283               yes             138460.0              143308.0   \n",
       "\n",
       "      gnomAD_allele_frequency Vernot_introgressed Vernot_ancestral_allele  \\\n",
       "7274                 0.687239                  no                     NaN   \n",
       "7275                 0.418040                  no                     NaN   \n",
       "7276                 0.000016                  no                     NaN   \n",
       "7277                 0.000042                  no                     NaN   \n",
       "7278                      NaN                  no                     NaN   \n",
       "7279                 0.445813                  no                     NaN   \n",
       "7280                 0.350281                  no                     NaN   \n",
       "7281                      NaN                  no                     NaN   \n",
       "7282                 0.000014                  no                     NaN   \n",
       "7283                 0.966171                  no                     NaN   \n",
       "\n",
       "     Vernot_derived_allele  Vernot_ancestral_derived_code  Vernot_AFA_AF  \\\n",
       "7274                   NaN                            NaN            NaN   \n",
       "7275                   NaN                            NaN            NaN   \n",
       "7276                   NaN                            NaN            NaN   \n",
       "7277                   NaN                            NaN            NaN   \n",
       "7278                   NaN                            NaN            NaN   \n",
       "7279                   NaN                            NaN            NaN   \n",
       "7280                   NaN                            NaN            NaN   \n",
       "7281                   NaN                            NaN            NaN   \n",
       "7282                   NaN                            NaN            NaN   \n",
       "7283                   NaN                            NaN            NaN   \n",
       "\n",
       "      Vernot_AFR_AF  Vernot_AMR_AF  Vernot_EAS_AF  Vernot_EUR_AF  \\\n",
       "7274            NaN            NaN            NaN            NaN   \n",
       "7275            NaN            NaN            NaN            NaN   \n",
       "7276            NaN            NaN            NaN            NaN   \n",
       "7277            NaN            NaN            NaN            NaN   \n",
       "7278            NaN            NaN            NaN            NaN   \n",
       "7279            NaN            NaN            NaN            NaN   \n",
       "7280            NaN            NaN            NaN            NaN   \n",
       "7281            NaN            NaN            NaN            NaN   \n",
       "7282            NaN            NaN            NaN            NaN   \n",
       "7283            NaN            NaN            NaN            NaN   \n",
       "\n",
       "      Vernot_PNG_AF  Vernot_SAS_AF Vernot_Denisovan_base  \\\n",
       "7274            NaN            NaN                   NaN   \n",
       "7275            NaN            NaN                   NaN   \n",
       "7276            NaN            NaN                   NaN   \n",
       "7277            NaN            NaN                   NaN   \n",
       "7278            NaN            NaN                   NaN   \n",
       "7279            NaN            NaN                   NaN   \n",
       "7280            NaN            NaN                   NaN   \n",
       "7281            NaN            NaN                   NaN   \n",
       "7282            NaN            NaN                   NaN   \n",
       "7283            NaN            NaN                   NaN   \n",
       "\n",
       "     Vernot_Neanderthal_base_1 Vernot_Neanderthal_base_2 Vernot_haplotype_tag  \\\n",
       "7274                       NaN                       NaN                  NaN   \n",
       "7275                       NaN                       NaN                  NaN   \n",
       "7276                       NaN                       NaN                  NaN   \n",
       "7277                       NaN                       NaN                  NaN   \n",
       "7278                       NaN                       NaN                  NaN   \n",
       "7279                       NaN                       NaN                  NaN   \n",
       "7280                       NaN                       NaN                  NaN   \n",
       "7281                       NaN                       NaN                  NaN   \n",
       "7282                       NaN                       NaN                  NaN   \n",
       "7283                       NaN                       NaN                  NaN   \n",
       "\n",
       "      Vernot_introgressed_AF Browning_introgressed  Browning_ref_alt  \\\n",
       "7274                     NaN                    no               NaN   \n",
       "7275                     NaN                    no               NaN   \n",
       "7276                     NaN                    no               NaN   \n",
       "7277                     NaN                    no               NaN   \n",
       "7278                     NaN                    no               NaN   \n",
       "7279                     NaN                    no               NaN   \n",
       "7280                     NaN                    no               NaN   \n",
       "7281                     NaN                    no               NaN   \n",
       "7282                     NaN                    no               NaN   \n",
       "7283                     NaN                    no               NaN   \n",
       "\n",
       "      Browning_introgressed_AF annotation  mis_oe    mis_z   lof_oe   lof_z  \\\n",
       "7274                       NaN       PER3  1.0116 -0.10855  0.63671  2.5313   \n",
       "7275                       NaN       PER3  1.0116 -0.10855  0.63671  2.5313   \n",
       "7276                       NaN       PER3  1.0116 -0.10855  0.63671  2.5313   \n",
       "7277                       NaN       PER3  1.0116 -0.10855  0.63671  2.5313   \n",
       "7278                       NaN       PER3  1.0116 -0.10855  0.63671  2.5313   \n",
       "7279                       NaN       PER3  1.0116 -0.10855  0.63671  2.5313   \n",
       "7280                       NaN       PER3  1.0116 -0.10855  0.63671  2.5313   \n",
       "7281                       NaN       PER3  1.0116 -0.10855  0.63671  2.5313   \n",
       "7282                       NaN       PER3  1.0116 -0.10855  0.63671  2.5313   \n",
       "7283                       NaN       PER3  1.0116 -0.10855  0.63671  2.5313   \n",
       "\n",
       "      phyloP  ag_delta  al_delta  dg_delta  dl_delta  delta_max  ag_pos  \\\n",
       "7274  -0.215      0.00       0.0       0.0      0.02       0.02      36   \n",
       "7275  -0.379      0.00       0.0       0.0      0.00       0.00      13   \n",
       "7276   0.172      0.00       0.0       0.0      0.00       0.00      22   \n",
       "7277  -0.371      0.00       0.0       0.0      0.00       0.00     -47   \n",
       "7278  -2.062      0.00       0.0       0.0      0.00       0.00     -30   \n",
       "7279   0.482      0.02       0.0       0.0      0.00       0.02       1   \n",
       "7280   0.358      0.00       0.0       0.0      0.00       0.00     -28   \n",
       "7281  -0.372      0.00       0.0       0.0      0.00       0.00     -12   \n",
       "7282  -0.140      0.00       0.0       0.0      0.00       0.00       2   \n",
       "7283  -0.726      0.00       0.0       0.0      0.00       0.00     -18   \n",
       "\n",
       "      al_pos  dg_pos  dl_pos  Adipose_Subcutaneous  Adipose_Visceral_Omentum  \\\n",
       "7274      19     -49      19                   1.0                       1.0   \n",
       "7275      -7     -12      -7                   1.0                       1.0   \n",
       "7276     -20      -2     -27                   NaN                       NaN   \n",
       "7277      17     -47       3                   NaN                       NaN   \n",
       "7278      47      45     -48                   NaN                       NaN   \n",
       "7279      40       1     -36                   1.0                       0.0   \n",
       "7280      29       8     -42                   1.0                       0.0   \n",
       "7281      15      17      -6                   NaN                       NaN   \n",
       "7282      33       2     -39                   NaN                       NaN   \n",
       "7283      13      46      -2                   NaN                       NaN   \n",
       "\n",
       "      Adrenal_Gland  Artery_Aorta  Artery_Coronary  Artery_Tibial  \\\n",
       "7274            1.0           1.0              0.0            1.0   \n",
       "7275            0.0           1.0              0.0            1.0   \n",
       "7276            NaN           NaN              NaN            NaN   \n",
       "7277            NaN           NaN              NaN            NaN   \n",
       "7278            NaN           NaN              NaN            NaN   \n",
       "7279            0.0           1.0              0.0            1.0   \n",
       "7280            1.0           0.0              0.0            1.0   \n",
       "7281            NaN           NaN              NaN            NaN   \n",
       "7282            NaN           NaN              NaN            NaN   \n",
       "7283            NaN           NaN              NaN            NaN   \n",
       "\n",
       "      Brain_Amygdala  Brain_Anterior_cingulate_cortex_BA24  \\\n",
       "7274             0.0                                   0.0   \n",
       "7275             0.0                                   0.0   \n",
       "7276             NaN                                   NaN   \n",
       "7277             NaN                                   NaN   \n",
       "7278             NaN                                   NaN   \n",
       "7279             0.0                                   0.0   \n",
       "7280             0.0                                   0.0   \n",
       "7281             NaN                                   NaN   \n",
       "7282             NaN                                   NaN   \n",
       "7283             NaN                                   NaN   \n",
       "\n",
       "      Brain_Caudate_basal_ganglia  Brain_Cerebellar_Hemisphere  \\\n",
       "7274                          0.0                          0.0   \n",
       "7275                          1.0                          0.0   \n",
       "7276                          NaN                          NaN   \n",
       "7277                          NaN                          NaN   \n",
       "7278                          NaN                          NaN   \n",
       "7279                          1.0                          0.0   \n",
       "7280                          0.0                          0.0   \n",
       "7281                          NaN                          NaN   \n",
       "7282                          NaN                          NaN   \n",
       "7283                          NaN                          NaN   \n",
       "\n",
       "      Brain_Cerebellum  Brain_Cortex  Brain_Frontal_Cortex_BA9  \\\n",
       "7274               0.0           0.0                       0.0   \n",
       "7275               0.0           0.0                       1.0   \n",
       "7276               NaN           NaN                       NaN   \n",
       "7277               NaN           NaN                       NaN   \n",
       "7278               NaN           NaN                       NaN   \n",
       "7279               0.0           0.0                       1.0   \n",
       "7280               1.0           0.0                       0.0   \n",
       "7281               NaN           NaN                       NaN   \n",
       "7282               NaN           NaN                       NaN   \n",
       "7283               NaN           NaN                       NaN   \n",
       "\n",
       "      Brain_Hippocampus  Brain_Hypothalamus  \\\n",
       "7274                0.0                 0.0   \n",
       "7275                0.0                 0.0   \n",
       "7276                NaN                 NaN   \n",
       "7277                NaN                 NaN   \n",
       "7278                NaN                 NaN   \n",
       "7279                0.0                 0.0   \n",
       "7280                0.0                 0.0   \n",
       "7281                NaN                 NaN   \n",
       "7282                NaN                 NaN   \n",
       "7283                NaN                 NaN   \n",
       "\n",
       "      Brain_Nucleus_accumbens_basal_ganglia  Brain_Putamen_basal_ganglia  \\\n",
       "7274                                    0.0                          0.0   \n",
       "7275                                    0.0                          0.0   \n",
       "7276                                    NaN                          NaN   \n",
       "7277                                    NaN                          NaN   \n",
       "7278                                    NaN                          NaN   \n",
       "7279                                    0.0                          0.0   \n",
       "7280                                    0.0                          1.0   \n",
       "7281                                    NaN                          NaN   \n",
       "7282                                    NaN                          NaN   \n",
       "7283                                    NaN                          NaN   \n",
       "\n",
       "      Brain_Spinal_cord_cervical_c-1  Brain_Substantia_nigra  \\\n",
       "7274                             0.0                     0.0   \n",
       "7275                             0.0                     0.0   \n",
       "7276                             NaN                     NaN   \n",
       "7277                             NaN                     NaN   \n",
       "7278                             NaN                     NaN   \n",
       "7279                             0.0                     0.0   \n",
       "7280                             0.0                     0.0   \n",
       "7281                             NaN                     NaN   \n",
       "7282                             NaN                     NaN   \n",
       "7283                             NaN                     NaN   \n",
       "\n",
       "      Breast_Mammary_Tissue  Cells_Cultured_fibroblasts  \\\n",
       "7274                    1.0                         0.0   \n",
       "7275                    1.0                         1.0   \n",
       "7276                    NaN                         NaN   \n",
       "7277                    NaN                         NaN   \n",
       "7278                    NaN                         NaN   \n",
       "7279                    1.0                         1.0   \n",
       "7280                    0.0                         0.0   \n",
       "7281                    NaN                         NaN   \n",
       "7282                    NaN                         NaN   \n",
       "7283                    NaN                         NaN   \n",
       "\n",
       "      Cells_EBV-transformed_lymphocytes  Colon_Sigmoid  Colon_Transverse  \\\n",
       "7274                                0.0            0.0               0.0   \n",
       "7275                                0.0            0.0               1.0   \n",
       "7276                                NaN            NaN               NaN   \n",
       "7277                                NaN            NaN               NaN   \n",
       "7278                                NaN            NaN               NaN   \n",
       "7279                                0.0            0.0               0.0   \n",
       "7280                                0.0            0.0               0.0   \n",
       "7281                                NaN            NaN               NaN   \n",
       "7282                                NaN            NaN               NaN   \n",
       "7283                                NaN            NaN               NaN   \n",
       "\n",
       "      Esophagus_Gastroesophageal_Junction  Esophagus_Mucosa  \\\n",
       "7274                                  1.0               1.0   \n",
       "7275                                  1.0               1.0   \n",
       "7276                                  NaN               NaN   \n",
       "7277                                  NaN               NaN   \n",
       "7278                                  NaN               NaN   \n",
       "7279                                  1.0               1.0   \n",
       "7280                                  1.0               1.0   \n",
       "7281                                  NaN               NaN   \n",
       "7282                                  NaN               NaN   \n",
       "7283                                  NaN               NaN   \n",
       "\n",
       "      Esophagus_Muscularis  Heart_Atrial_Appendage  Heart_Left_Ventricle  \\\n",
       "7274                   0.0                     1.0                   0.0   \n",
       "7275                   1.0                     1.0                   0.0   \n",
       "7276                   NaN                     NaN                   NaN   \n",
       "7277                   NaN                     NaN                   NaN   \n",
       "7278                   NaN                     NaN                   NaN   \n",
       "7279                   1.0                     1.0                   0.0   \n",
       "7280                   0.0                     1.0                   0.0   \n",
       "7281                   NaN                     NaN                   NaN   \n",
       "7282                   NaN                     NaN                   NaN   \n",
       "7283                   NaN                     NaN                   NaN   \n",
       "\n",
       "      Kidney_Cortex  Liver  Lung  Minor_Salivary_Gland  Muscle_Skeletal  \\\n",
       "7274            0.0    0.0   1.0                   0.0              1.0   \n",
       "7275            0.0    0.0   1.0                   0.0              1.0   \n",
       "7276            NaN    NaN   NaN                   NaN              NaN   \n",
       "7277            NaN    NaN   NaN                   NaN              NaN   \n",
       "7278            NaN    NaN   NaN                   NaN              NaN   \n",
       "7279            0.0    0.0   0.0                   0.0              1.0   \n",
       "7280            0.0    0.0   0.0                   0.0              1.0   \n",
       "7281            NaN    NaN   NaN                   NaN              NaN   \n",
       "7282            NaN    NaN   NaN                   NaN              NaN   \n",
       "7283            NaN    NaN   NaN                   NaN              NaN   \n",
       "\n",
       "      Nerve_Tibial  Ovary  Pancreas  Pituitary  Prostate  \\\n",
       "7274           1.0    0.0       0.0        0.0       0.0   \n",
       "7275           1.0    0.0       0.0        0.0       0.0   \n",
       "7276           NaN    NaN       NaN        NaN       NaN   \n",
       "7277           NaN    NaN       NaN        NaN       NaN   \n",
       "7278           NaN    NaN       NaN        NaN       NaN   \n",
       "7279           1.0    0.0       0.0        0.0       0.0   \n",
       "7280           1.0    0.0       0.0        0.0       0.0   \n",
       "7281           NaN    NaN       NaN        NaN       NaN   \n",
       "7282           NaN    NaN       NaN        NaN       NaN   \n",
       "7283           NaN    NaN       NaN        NaN       NaN   \n",
       "\n",
       "      Skin_Not_Sun_Exposed_Suprapubic  Skin_Sun_Exposed_Lower_leg  \\\n",
       "7274                              1.0                         1.0   \n",
       "7275                              1.0                         1.0   \n",
       "7276                              NaN                         NaN   \n",
       "7277                              NaN                         NaN   \n",
       "7278                              NaN                         NaN   \n",
       "7279                              1.0                         1.0   \n",
       "7280                              1.0                         1.0   \n",
       "7281                              NaN                         NaN   \n",
       "7282                              NaN                         NaN   \n",
       "7283                              NaN                         NaN   \n",
       "\n",
       "      Small_Intestine_Terminal_Ileum  Spleen  Stomach  Testis  Thyroid  \\\n",
       "7274                             0.0     0.0      0.0     0.0      1.0   \n",
       "7275                             0.0     0.0      0.0     0.0      1.0   \n",
       "7276                             NaN     NaN      NaN     NaN      NaN   \n",
       "7277                             NaN     NaN      NaN     NaN      NaN   \n",
       "7278                             NaN     NaN      NaN     NaN      NaN   \n",
       "7279                             0.0     0.0      1.0     0.0      1.0   \n",
       "7280                             0.0     0.0      0.0     0.0      0.0   \n",
       "7281                             NaN     NaN      NaN     NaN      NaN   \n",
       "7282                             NaN     NaN      NaN     NaN      NaN   \n",
       "7283                             NaN     NaN      NaN     NaN      NaN   \n",
       "\n",
       "      Uterus  Vagina  Whole_Blood  N_GTEx_tissues sQTL Vernot_allele_origin  \\\n",
       "7274     0.0     0.0          1.0            16.0  yes              ancient   \n",
       "7275     0.0     0.0          1.0            20.0  yes              ancient   \n",
       "7276     NaN     NaN          NaN             NaN   no     archaic-specific   \n",
       "7277     NaN     NaN          NaN             NaN   no     archaic-specific   \n",
       "7278     NaN     NaN          NaN             NaN   no     archaic-specific   \n",
       "7279     0.0     0.0          1.0            18.0  yes              ancient   \n",
       "7280     0.0     0.0          1.0            13.0  yes              ancient   \n",
       "7281     NaN     NaN          NaN             NaN   no     archaic-specific   \n",
       "7282     NaN     NaN          NaN             NaN   no     archaic-specific   \n",
       "7283     NaN     NaN          NaN             NaN   no              ancient   \n",
       "\n",
       "     Browning_allele_origin  \n",
       "7274                ancient  \n",
       "7275                ancient  \n",
       "7276       archaic-specific  \n",
       "7277       archaic-specific  \n",
       "7278       archaic-specific  \n",
       "7279                ancient  \n",
       "7280                ancient  \n",
       "7281       archaic-specific  \n",
       "7282       archaic-specific  \n",
       "7283                ancient  "
      ]
     },
     "execution_count": 281,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "circadian_mask = data['annotation'].isin(circadian_genes['gene'])\n",
    "circadian_data = data[circadian_mask]\n",
    "circadian_data.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 282,
   "id": "4ba4bb81",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "31775"
      ]
     },
     "execution_count": 282,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(circadian_data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 283,
   "id": "4f2899d8",
   "metadata": {},
   "outputs": [],
   "source": [
    "circadian_data.to_csv('../circadian_genes/all_circadian_variants.txt', sep='\\t', header=True, index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "decc8b18",
   "metadata": {},
   "source": [
    "Now just the SAVs."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 284,
   "id": "10c59801",
   "metadata": {},
   "outputs": [],
   "source": [
    "circadian_SAVs = circadian_data[circadian_data['delta_max'] >= 0.2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 285,
   "id": "0ed668a0",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "91"
      ]
     },
     "execution_count": 285,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(circadian_SAVs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 286,
   "id": "56a0961c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref_allele</th>\n",
       "      <th>alt_allele</th>\n",
       "      <th>ancestral_allele</th>\n",
       "      <th>anc_dev</th>\n",
       "      <th>variant_type</th>\n",
       "      <th>altai_gt</th>\n",
       "      <th>chagyrskaya_gt</th>\n",
       "      <th>denisovan_gt</th>\n",
       "      <th>vindija_gt</th>\n",
       "      <th>altai_gt_boolean</th>\n",
       "      <th>chagyrskaya_gt_boolean</th>\n",
       "      <th>denisovan_gt_boolean</th>\n",
       "      <th>vindija_gt_boolean</th>\n",
       "      <th>distribution</th>\n",
       "      <th>present_in_1KG</th>\n",
       "      <th>1KG_allele_count</th>\n",
       "      <th>1KG_allele_number</th>\n",
       "      <th>1KG_allele_frequency</th>\n",
       "      <th>1KG_EAS_AF</th>\n",
       "      <th>1KG_EUR_AF</th>\n",
       "      <th>1KG_AFR_AF</th>\n",
       "      <th>1KG_AMR_AF</th>\n",
       "      <th>1KG_SAS_AF</th>\n",
       "      <th>1KG_non_ASW_AFR_AF</th>\n",
       "      <th>present_in_gnomAD</th>\n",
       "      <th>gnomAD_allele_count</th>\n",
       "      <th>gnomAD_allele_number</th>\n",
       "      <th>gnomAD_allele_frequency</th>\n",
       "      <th>Vernot_introgressed</th>\n",
       "      <th>Vernot_ancestral_allele</th>\n",
       "      <th>Vernot_derived_allele</th>\n",
       "      <th>Vernot_ancestral_derived_code</th>\n",
       "      <th>Vernot_AFA_AF</th>\n",
       "      <th>Vernot_AFR_AF</th>\n",
       "      <th>Vernot_AMR_AF</th>\n",
       "      <th>Vernot_EAS_AF</th>\n",
       "      <th>Vernot_EUR_AF</th>\n",
       "      <th>Vernot_PNG_AF</th>\n",
       "      <th>Vernot_SAS_AF</th>\n",
       "      <th>Vernot_Denisovan_base</th>\n",
       "      <th>Vernot_Neanderthal_base_1</th>\n",
       "      <th>Vernot_Neanderthal_base_2</th>\n",
       "      <th>Vernot_haplotype_tag</th>\n",
       "      <th>Vernot_introgressed_AF</th>\n",
       "      <th>Browning_introgressed</th>\n",
       "      <th>Browning_ref_alt</th>\n",
       "      <th>Browning_introgressed_AF</th>\n",
       "      <th>annotation</th>\n",
       "      <th>mis_oe</th>\n",
       "      <th>mis_z</th>\n",
       "      <th>lof_oe</th>\n",
       "      <th>lof_z</th>\n",
       "      <th>phyloP</th>\n",
       "      <th>ag_delta</th>\n",
       "      <th>al_delta</th>\n",
       "      <th>dg_delta</th>\n",
       "      <th>dl_delta</th>\n",
       "      <th>delta_max</th>\n",
       "      <th>ag_pos</th>\n",
       "      <th>al_pos</th>\n",
       "      <th>dg_pos</th>\n",
       "      <th>dl_pos</th>\n",
       "      <th>Adipose_Subcutaneous</th>\n",
       "      <th>Adipose_Visceral_Omentum</th>\n",
       "      <th>Adrenal_Gland</th>\n",
       "      <th>Artery_Aorta</th>\n",
       "      <th>Artery_Coronary</th>\n",
       "      <th>Artery_Tibial</th>\n",
       "      <th>Brain_Amygdala</th>\n",
       "      <th>Brain_Anterior_cingulate_cortex_BA24</th>\n",
       "      <th>Brain_Caudate_basal_ganglia</th>\n",
       "      <th>Brain_Cerebellar_Hemisphere</th>\n",
       "      <th>Brain_Cerebellum</th>\n",
       "      <th>Brain_Cortex</th>\n",
       "      <th>Brain_Frontal_Cortex_BA9</th>\n",
       "      <th>Brain_Hippocampus</th>\n",
       "      <th>Brain_Hypothalamus</th>\n",
       "      <th>Brain_Nucleus_accumbens_basal_ganglia</th>\n",
       "      <th>Brain_Putamen_basal_ganglia</th>\n",
       "      <th>Brain_Spinal_cord_cervical_c-1</th>\n",
       "      <th>Brain_Substantia_nigra</th>\n",
       "      <th>Breast_Mammary_Tissue</th>\n",
       "      <th>Cells_Cultured_fibroblasts</th>\n",
       "      <th>Cells_EBV-transformed_lymphocytes</th>\n",
       "      <th>Colon_Sigmoid</th>\n",
       "      <th>Colon_Transverse</th>\n",
       "      <th>Esophagus_Gastroesophageal_Junction</th>\n",
       "      <th>Esophagus_Mucosa</th>\n",
       "      <th>Esophagus_Muscularis</th>\n",
       "      <th>Heart_Atrial_Appendage</th>\n",
       "      <th>Heart_Left_Ventricle</th>\n",
       "      <th>Kidney_Cortex</th>\n",
       "      <th>Liver</th>\n",
       "      <th>Lung</th>\n",
       "      <th>Minor_Salivary_Gland</th>\n",
       "      <th>Muscle_Skeletal</th>\n",
       "      <th>Nerve_Tibial</th>\n",
       "      <th>Ovary</th>\n",
       "      <th>Pancreas</th>\n",
       "      <th>Pituitary</th>\n",
       "      <th>Prostate</th>\n",
       "      <th>Skin_Not_Sun_Exposed_Suprapubic</th>\n",
       "      <th>Skin_Sun_Exposed_Lower_leg</th>\n",
       "      <th>Small_Intestine_Terminal_Ileum</th>\n",
       "      <th>Spleen</th>\n",
       "      <th>Stomach</th>\n",
       "      <th>Testis</th>\n",
       "      <th>Thyroid</th>\n",
       "      <th>Uterus</th>\n",
       "      <th>Vagina</th>\n",
       "      <th>Whole_Blood</th>\n",
       "      <th>N_GTEx_tissues</th>\n",
       "      <th>sQTL</th>\n",
       "      <th>Vernot_allele_origin</th>\n",
       "      <th>Browning_allele_origin</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>18537</th>\n",
       "      <td>chr1</td>\n",
       "      <td>21588603</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>ancestral</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>yes</td>\n",
       "      <td>3942.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.77</td>\n",
       "      <td>0.89</td>\n",
       "      <td>0.61</td>\n",
       "      <td>0.91</td>\n",
       "      <td>0.69</td>\n",
       "      <td>0.69</td>\n",
       "      <td>0.932749</td>\n",
       "      <td>yes</td>\n",
       "      <td>99055.0</td>\n",
       "      <td>143024.0</td>\n",
       "      <td>0.692576</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ECE1</td>\n",
       "      <td>0.69508</td>\n",
       "      <td>2.3066</td>\n",
       "      <td>0.171130</td>\n",
       "      <td>4.9124</td>\n",
       "      <td>-0.748</td>\n",
       "      <td>0.22</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.22</td>\n",
       "      <td>47</td>\n",
       "      <td>0</td>\n",
       "      <td>-44</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>ancient</td>\n",
       "      <td>ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25058</th>\n",
       "      <td>chr1</td>\n",
       "      <td>32797241</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>yes</td>\n",
       "      <td>90.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.06</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.067251</td>\n",
       "      <td>yes</td>\n",
       "      <td>2708.0</td>\n",
       "      <td>143162.0</td>\n",
       "      <td>0.018916</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HDAC1</td>\n",
       "      <td>0.50102</td>\n",
       "      <td>3.0373</td>\n",
       "      <td>0.205120</td>\n",
       "      <td>3.9837</td>\n",
       "      <td>-1.113</td>\n",
       "      <td>0.21</td>\n",
       "      <td>0.13</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.21</td>\n",
       "      <td>6</td>\n",
       "      <td>40</td>\n",
       "      <td>-11</td>\n",
       "      <td>40</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79980</th>\n",
       "      <td>chr1</td>\n",
       "      <td>151796494</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>6.0</td>\n",
       "      <td>143212.0</td>\n",
       "      <td>0.000042</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>RORC</td>\n",
       "      <td>0.63127</td>\n",
       "      <td>2.2852</td>\n",
       "      <td>0.033914</td>\n",
       "      <td>4.8612</td>\n",
       "      <td>0.561</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.20</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.20</td>\n",
       "      <td>2</td>\n",
       "      <td>-43</td>\n",
       "      <td>2</td>\n",
       "      <td>-43</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80982</th>\n",
       "      <td>chr1</td>\n",
       "      <td>154542319</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>yes</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.000975</td>\n",
       "      <td>yes</td>\n",
       "      <td>5.0</td>\n",
       "      <td>143174.0</td>\n",
       "      <td>0.000035</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHRNB2</td>\n",
       "      <td>0.67329</td>\n",
       "      <td>2.1034</td>\n",
       "      <td>0.455610</td>\n",
       "      <td>2.1139</td>\n",
       "      <td>-0.752</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.46</td>\n",
       "      <td>0.46</td>\n",
       "      <td>-36</td>\n",
       "      <td>-1</td>\n",
       "      <td>-13</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>81092</th>\n",
       "      <td>chr1</td>\n",
       "      <td>154706620</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Other</td>\n",
       "      <td>yes</td>\n",
       "      <td>4096.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.80</td>\n",
       "      <td>0.72</td>\n",
       "      <td>0.94</td>\n",
       "      <td>0.64</td>\n",
       "      <td>0.89</td>\n",
       "      <td>0.91</td>\n",
       "      <td>0.622807</td>\n",
       "      <td>yes</td>\n",
       "      <td>121787.0</td>\n",
       "      <td>143186.0</td>\n",
       "      <td>0.850551</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>KCNN3</td>\n",
       "      <td>0.55469</td>\n",
       "      <td>3.3258</td>\n",
       "      <td>0.139380</td>\n",
       "      <td>4.2723</td>\n",
       "      <td>-0.313</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.22</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.22</td>\n",
       "      <td>1</td>\n",
       "      <td>-36</td>\n",
       "      <td>1</td>\n",
       "      <td>-22</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>ancient</td>\n",
       "      <td>ancient</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      chrom        pos ref_allele alt_allele ancestral_allele    anc_dev  \\\n",
       "18537  chr1   21588603          C          G                G  ancestral   \n",
       "25058  chr1   32797241          G          C                G    derived   \n",
       "79980  chr1  151796494          G          A                G    derived   \n",
       "80982  chr1  154542319          G          A                G    derived   \n",
       "81092  chr1  154706620          T          C                T    derived   \n",
       "\n",
       "      variant_type altai_gt chagyrskaya_gt denisovan_gt vindija_gt  \\\n",
       "18537          snv      1/1            1/1          1/1        1/1   \n",
       "25058          snv      1/1            1/1          1/1        1/1   \n",
       "79980          snv      1/1            1/1          0/0        1/1   \n",
       "80982          snv      0/0            0/0          0/1        0/0   \n",
       "81092          snv      1/1            0/0          0/0        0/1   \n",
       "\n",
       "       altai_gt_boolean  chagyrskaya_gt_boolean  denisovan_gt_boolean  \\\n",
       "18537              True                    True                  True   \n",
       "25058              True                    True                  True   \n",
       "79980              True                    True                 False   \n",
       "80982             False                   False                  True   \n",
       "81092              True                   False                 False   \n",
       "\n",
       "       vindija_gt_boolean distribution present_in_1KG  1KG_allele_count  \\\n",
       "18537                True       Shared            yes            3942.0   \n",
       "25058                True       Shared            yes              90.0   \n",
       "79980                True  Neanderthal             no               NaN   \n",
       "80982               False    Denisovan            yes               1.0   \n",
       "81092                True        Other            yes            4096.0   \n",
       "\n",
       "       1KG_allele_number  1KG_allele_frequency  1KG_EAS_AF  1KG_EUR_AF  \\\n",
       "18537             5096.0                  0.77        0.89        0.61   \n",
       "25058             5096.0                  0.02        0.00        0.00   \n",
       "79980                NaN                   NaN         NaN         NaN   \n",
       "80982             5096.0                  0.00        0.00        0.00   \n",
       "81092             5096.0                  0.80        0.72        0.94   \n",
       "\n",
       "       1KG_AFR_AF  1KG_AMR_AF  1KG_SAS_AF  1KG_non_ASW_AFR_AF  \\\n",
       "18537        0.91        0.69        0.69            0.932749   \n",
       "25058        0.06        0.01        0.00            0.067251   \n",
       "79980         NaN         NaN         NaN                 NaN   \n",
       "80982        0.00        0.00        0.00            0.000975   \n",
       "81092        0.64        0.89        0.91            0.622807   \n",
       "\n",
       "      present_in_gnomAD  gnomAD_allele_count  gnomAD_allele_number  \\\n",
       "18537               yes              99055.0              143024.0   \n",
       "25058               yes               2708.0              143162.0   \n",
       "79980               yes                  6.0              143212.0   \n",
       "80982               yes                  5.0              143174.0   \n",
       "81092               yes             121787.0              143186.0   \n",
       "\n",
       "       gnomAD_allele_frequency Vernot_introgressed Vernot_ancestral_allele  \\\n",
       "18537                 0.692576                  no                     NaN   \n",
       "25058                 0.018916                  no                     NaN   \n",
       "79980                 0.000042                  no                     NaN   \n",
       "80982                 0.000035                  no                     NaN   \n",
       "81092                 0.850551                  no                     NaN   \n",
       "\n",
       "      Vernot_derived_allele  Vernot_ancestral_derived_code  Vernot_AFA_AF  \\\n",
       "18537                   NaN                            NaN            NaN   \n",
       "25058                   NaN                            NaN            NaN   \n",
       "79980                   NaN                            NaN            NaN   \n",
       "80982                   NaN                            NaN            NaN   \n",
       "81092                   NaN                            NaN            NaN   \n",
       "\n",
       "       Vernot_AFR_AF  Vernot_AMR_AF  Vernot_EAS_AF  Vernot_EUR_AF  \\\n",
       "18537            NaN            NaN            NaN            NaN   \n",
       "25058            NaN            NaN            NaN            NaN   \n",
       "79980            NaN            NaN            NaN            NaN   \n",
       "80982            NaN            NaN            NaN            NaN   \n",
       "81092            NaN            NaN            NaN            NaN   \n",
       "\n",
       "       Vernot_PNG_AF  Vernot_SAS_AF Vernot_Denisovan_base  \\\n",
       "18537            NaN            NaN                   NaN   \n",
       "25058            NaN            NaN                   NaN   \n",
       "79980            NaN            NaN                   NaN   \n",
       "80982            NaN            NaN                   NaN   \n",
       "81092            NaN            NaN                   NaN   \n",
       "\n",
       "      Vernot_Neanderthal_base_1 Vernot_Neanderthal_base_2  \\\n",
       "18537                       NaN                       NaN   \n",
       "25058                       NaN                       NaN   \n",
       "79980                       NaN                       NaN   \n",
       "80982                       NaN                       NaN   \n",
       "81092                       NaN                       NaN   \n",
       "\n",
       "      Vernot_haplotype_tag  Vernot_introgressed_AF Browning_introgressed  \\\n",
       "18537                  NaN                     NaN                    no   \n",
       "25058                  NaN                     NaN                    no   \n",
       "79980                  NaN                     NaN                    no   \n",
       "80982                  NaN                     NaN                    no   \n",
       "81092                  NaN                     NaN                    no   \n",
       "\n",
       "       Browning_ref_alt  Browning_introgressed_AF annotation   mis_oe   mis_z  \\\n",
       "18537               NaN                       NaN       ECE1  0.69508  2.3066   \n",
       "25058               NaN                       NaN      HDAC1  0.50102  3.0373   \n",
       "79980               NaN                       NaN       RORC  0.63127  2.2852   \n",
       "80982               NaN                       NaN     CHRNB2  0.67329  2.1034   \n",
       "81092               NaN                       NaN      KCNN3  0.55469  3.3258   \n",
       "\n",
       "         lof_oe   lof_z  phyloP  ag_delta  al_delta  dg_delta  dl_delta  \\\n",
       "18537  0.171130  4.9124  -0.748      0.22      0.00      0.00      0.00   \n",
       "25058  0.205120  3.9837  -1.113      0.21      0.13      0.00      0.00   \n",
       "79980  0.033914  4.8612   0.561      0.00      0.00      0.20      0.00   \n",
       "80982  0.455610  2.1139  -0.752      0.00      0.00      0.00      0.46   \n",
       "81092  0.139380  4.2723  -0.313      0.00      0.00      0.22      0.00   \n",
       "\n",
       "       delta_max  ag_pos  al_pos  dg_pos  dl_pos  Adipose_Subcutaneous  \\\n",
       "18537       0.22      47       0     -44       0                   NaN   \n",
       "25058       0.21       6      40     -11      40                   NaN   \n",
       "79980       0.20       2     -43       2     -43                   NaN   \n",
       "80982       0.46     -36      -1     -13      -1                   NaN   \n",
       "81092       0.22       1     -36       1     -22                   NaN   \n",
       "\n",
       "       Adipose_Visceral_Omentum  Adrenal_Gland  Artery_Aorta  Artery_Coronary  \\\n",
       "18537                       NaN            NaN           NaN              NaN   \n",
       "25058                       NaN            NaN           NaN              NaN   \n",
       "79980                       NaN            NaN           NaN              NaN   \n",
       "80982                       NaN            NaN           NaN              NaN   \n",
       "81092                       NaN            NaN           NaN              NaN   \n",
       "\n",
       "       Artery_Tibial  Brain_Amygdala  Brain_Anterior_cingulate_cortex_BA24  \\\n",
       "18537            NaN             NaN                                   NaN   \n",
       "25058            NaN             NaN                                   NaN   \n",
       "79980            NaN             NaN                                   NaN   \n",
       "80982            NaN             NaN                                   NaN   \n",
       "81092            NaN             NaN                                   NaN   \n",
       "\n",
       "       Brain_Caudate_basal_ganglia  Brain_Cerebellar_Hemisphere  \\\n",
       "18537                          NaN                          NaN   \n",
       "25058                          NaN                          NaN   \n",
       "79980                          NaN                          NaN   \n",
       "80982                          NaN                          NaN   \n",
       "81092                          NaN                          NaN   \n",
       "\n",
       "       Brain_Cerebellum  Brain_Cortex  Brain_Frontal_Cortex_BA9  \\\n",
       "18537               NaN           NaN                       NaN   \n",
       "25058               NaN           NaN                       NaN   \n",
       "79980               NaN           NaN                       NaN   \n",
       "80982               NaN           NaN                       NaN   \n",
       "81092               NaN           NaN                       NaN   \n",
       "\n",
       "       Brain_Hippocampus  Brain_Hypothalamus  \\\n",
       "18537                NaN                 NaN   \n",
       "25058                NaN                 NaN   \n",
       "79980                NaN                 NaN   \n",
       "80982                NaN                 NaN   \n",
       "81092                NaN                 NaN   \n",
       "\n",
       "       Brain_Nucleus_accumbens_basal_ganglia  Brain_Putamen_basal_ganglia  \\\n",
       "18537                                    NaN                          NaN   \n",
       "25058                                    NaN                          NaN   \n",
       "79980                                    NaN                          NaN   \n",
       "80982                                    NaN                          NaN   \n",
       "81092                                    NaN                          NaN   \n",
       "\n",
       "       Brain_Spinal_cord_cervical_c-1  Brain_Substantia_nigra  \\\n",
       "18537                             NaN                     NaN   \n",
       "25058                             NaN                     NaN   \n",
       "79980                             NaN                     NaN   \n",
       "80982                             NaN                     NaN   \n",
       "81092                             NaN                     NaN   \n",
       "\n",
       "       Breast_Mammary_Tissue  Cells_Cultured_fibroblasts  \\\n",
       "18537                    NaN                         NaN   \n",
       "25058                    NaN                         NaN   \n",
       "79980                    NaN                         NaN   \n",
       "80982                    NaN                         NaN   \n",
       "81092                    NaN                         NaN   \n",
       "\n",
       "       Cells_EBV-transformed_lymphocytes  Colon_Sigmoid  Colon_Transverse  \\\n",
       "18537                                NaN            NaN               NaN   \n",
       "25058                                NaN            NaN               NaN   \n",
       "79980                                NaN            NaN               NaN   \n",
       "80982                                NaN            NaN               NaN   \n",
       "81092                                NaN            NaN               NaN   \n",
       "\n",
       "       Esophagus_Gastroesophageal_Junction  Esophagus_Mucosa  \\\n",
       "18537                                  NaN               NaN   \n",
       "25058                                  NaN               NaN   \n",
       "79980                                  NaN               NaN   \n",
       "80982                                  NaN               NaN   \n",
       "81092                                  NaN               NaN   \n",
       "\n",
       "       Esophagus_Muscularis  Heart_Atrial_Appendage  Heart_Left_Ventricle  \\\n",
       "18537                   NaN                     NaN                   NaN   \n",
       "25058                   NaN                     NaN                   NaN   \n",
       "79980                   NaN                     NaN                   NaN   \n",
       "80982                   NaN                     NaN                   NaN   \n",
       "81092                   NaN                     NaN                   NaN   \n",
       "\n",
       "       Kidney_Cortex  Liver  Lung  Minor_Salivary_Gland  Muscle_Skeletal  \\\n",
       "18537            NaN    NaN   NaN                   NaN              NaN   \n",
       "25058            NaN    NaN   NaN                   NaN              NaN   \n",
       "79980            NaN    NaN   NaN                   NaN              NaN   \n",
       "80982            NaN    NaN   NaN                   NaN              NaN   \n",
       "81092            NaN    NaN   NaN                   NaN              NaN   \n",
       "\n",
       "       Nerve_Tibial  Ovary  Pancreas  Pituitary  Prostate  \\\n",
       "18537           NaN    NaN       NaN        NaN       NaN   \n",
       "25058           NaN    NaN       NaN        NaN       NaN   \n",
       "79980           NaN    NaN       NaN        NaN       NaN   \n",
       "80982           NaN    NaN       NaN        NaN       NaN   \n",
       "81092           NaN    NaN       NaN        NaN       NaN   \n",
       "\n",
       "       Skin_Not_Sun_Exposed_Suprapubic  Skin_Sun_Exposed_Lower_leg  \\\n",
       "18537                              NaN                         NaN   \n",
       "25058                              NaN                         NaN   \n",
       "79980                              NaN                         NaN   \n",
       "80982                              NaN                         NaN   \n",
       "81092                              NaN                         NaN   \n",
       "\n",
       "       Small_Intestine_Terminal_Ileum  Spleen  Stomach  Testis  Thyroid  \\\n",
       "18537                             NaN     NaN      NaN     NaN      NaN   \n",
       "25058                             NaN     NaN      NaN     NaN      NaN   \n",
       "79980                             NaN     NaN      NaN     NaN      NaN   \n",
       "80982                             NaN     NaN      NaN     NaN      NaN   \n",
       "81092                             NaN     NaN      NaN     NaN      NaN   \n",
       "\n",
       "       Uterus  Vagina  Whole_Blood  N_GTEx_tissues sQTL  \\\n",
       "18537     NaN     NaN          NaN             NaN   no   \n",
       "25058     NaN     NaN          NaN             NaN   no   \n",
       "79980     NaN     NaN          NaN             NaN   no   \n",
       "80982     NaN     NaN          NaN             NaN   no   \n",
       "81092     NaN     NaN          NaN             NaN   no   \n",
       "\n",
       "         Vernot_allele_origin  Browning_allele_origin  \n",
       "18537                 ancient                 ancient  \n",
       "25058  low-confidence ancient  low-confidence ancient  \n",
       "79980        archaic-specific        archaic-specific  \n",
       "80982  low-confidence ancient  low-confidence ancient  \n",
       "81092                 ancient                 ancient  "
      ]
     },
     "execution_count": 286,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "circadian_SAVs.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 287,
   "id": "9991261f",
   "metadata": {},
   "outputs": [],
   "source": [
    "circadian_SAVs.to_csv('../circadian_genes/all_circadian_SAVs.txt', sep='\\t', header=True, index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "927f1d9b",
   "metadata": {},
   "source": [
    "# Phenotype Enrichment <a class = 'anchor' id = 'phenotypeenrichment'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6d504ae4",
   "metadata": {},
   "source": [
    "Let's subset data for our phenotype enrichment analyses. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 288,
   "id": "0f8f0a27",
   "metadata": {},
   "outputs": [],
   "source": [
    "enrichment_data = data[['delta_max','annotation','distribution','Vernot_allele_origin','Browning_allele_origin']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 289,
   "id": "529fcacd",
   "metadata": {},
   "outputs": [],
   "source": [
    "enrichment_data.to_csv('archaic_data_for_enrichment.txt', sep='\\t', header=True, index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b230bf22",
   "metadata": {},
   "source": [
    "We have plenty of GWAS and HPO terms. Let's make life a bit easier by categorizing them into systems starting with GWAS. Write a function to generate a new enrichment summary file with an added column that lists the system per trait."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 290,
   "id": "9426da2a",
   "metadata": {},
   "outputs": [],
   "source": [
    "def map_GWAS_systems():\n",
    "    GWAS_mapping = pd.read_csv(f'../phenotype_enrichment/ontologies/GWAS_terms_and_systems.txt', sep = '\\t', header = 0)\n",
    "    GWAS_system = pd.Series(GWAS_mapping['system'].values, index = GWAS_mapping['GWAS_term']).to_dict()\n",
    "    setnames = ['altai','archaic_specific','Browning_introgressed','chagyrskaya','denisovan','neanderthal','shared','Vernot_introgressed','vindija']\n",
    "    \n",
    "    for set in setnames:\n",
    "        dataframe = pd.read_csv(f'../phenotype_enrichment/enrichment/{set}_GWAS_enrichment.txt', sep = '\\t', header = 0)\n",
    "        dataframe['system'] = dataframe['label'].map(GWAS_system)\n",
    "        dataframe.to_csv(f'../phenotype_enrichment/enrichment/{set}_GWAS_enrichment_with_system.txt', sep='\\t', header=True, index=False)   "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "087d1f98",
   "metadata": {},
   "source": [
    "Run the function."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 291,
   "id": "354d0774",
   "metadata": {},
   "outputs": [],
   "source": [
    "map_GWAS_systems()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8683a29d",
   "metadata": {},
   "source": [
    "Repeat with HPO."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 292,
   "id": "b1626e89",
   "metadata": {},
   "outputs": [],
   "source": [
    "def map_HPO_systems():\n",
    "    HPO_mapping = pd.read_csv(f'../phenotype_enrichment/ontologies/HPO_terms_and_systems.txt', sep = '\\t', header = 0)\n",
    "    HPO_system = pd.Series(HPO_mapping['system'].values, index = HPO_mapping['HPO_term']).to_dict()\n",
    "    setnames = ['altai','archaic_specific','Browning_introgressed','chagyrskaya','denisovan','neanderthal','shared','Vernot_introgressed','vindija']\n",
    "    \n",
    "    for set in setnames:\n",
    "        dataframe = pd.read_csv(f'../phenotype_enrichment/enrichment/{set}_HPO_enrichment.txt', sep = '\\t', header = 0)\n",
    "        dataframe['system'] = dataframe['label'].map(HPO_system)\n",
    "        dataframe.to_csv(f'../phenotype_enrichment/enrichment/{set}_HPO_enrichment_with_system.txt', sep='\\t', header=True, index=False)   "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 293,
   "id": "c8dca0ca",
   "metadata": {},
   "outputs": [],
   "source": [
    "map_HPO_systems()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "20d0aafb",
   "metadata": {},
   "source": [
    "Now it's time to get FDR corrected p-values for our gene enrichment analyses. Let's write a function to start."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 294,
   "id": "8c784b51",
   "metadata": {},
   "outputs": [],
   "source": [
    "fdr_table = []"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 295,
   "id": "8a1bcc02",
   "metadata": {},
   "outputs": [],
   "source": [
    "def reportFDRcorrectedPthreshold(set_name, ontology, q_value_threshold, resolution=0.0001, minStart=0):\n",
    "    fdr_empiric = pd.read_csv(f'../phenotype_enrichment/empiric_FDR/{set_name}_{ontology}_empiric_FDR.txt', sep = '\\t', header = None, index_col = 0)\n",
    "    obs = pd.read_csv(f'../phenotype_enrichment/enrichment/{set_name}_{ontology}_enrichment.txt', sep = '\\t')\n",
    "\n",
    "    fdr_threshold = []\n",
    "    for i in np.arange(minStart,0.05,resolution):\n",
    "        \n",
    "        observed_positive = sum(obs['p_value'] <= i)\n",
    "        average_false_positive = (fdr_empiric <= i).sum().mean()\n",
    "        q = average_false_positive/observed_positive\n",
    "        fdr_threshold.append([set_name, ontology, q_value_threshold, i, observed_positive, average_false_positive, q])\n",
    "        \n",
    "        if (q != np.inf) & (q > q_value_threshold):\n",
    "            break\n",
    "    \n",
    "    threshold = fdr_threshold[-2]\n",
    "    fdr_table.append(threshold)\n",
    "    #fdr_threshold = pd.DataFrame(fdr_threshold, columns = ['pval_threshold','obsPos','avgFalsePos','q'])\n",
    "    #return fdr_threshold.tail(2).head(1)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "941557f7",
   "metadata": {},
   "source": [
    "Now to generate all the combinations for which to get a corrected p-value."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 296,
   "id": "fd1ad2cd",
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "combinations = [(set_name,ontology,q_value_threshold) for set_name in ['altai','archaic_specific','Browning_introgressed','chagyrskaya','denisovan','neanderthal','shared','Vernot_introgressed','vindija'] for ontology in ['GWAS','HPO'] for q_value_threshold in [0.05,0.1]]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9cddc781",
   "metadata": {},
   "source": [
    "And run!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 297,
   "id": "9ffb3035",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/scratch/cbrand/ipykernel_47833/47096580.py:10: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  q = average_false_positive/observed_positive\n",
      "/scratch/cbrand/ipykernel_47833/47096580.py:10: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  q = average_false_positive/observed_positive\n",
      "/scratch/cbrand/ipykernel_47833/47096580.py:10: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  q = average_false_positive/observed_positive\n",
      "/scratch/cbrand/ipykernel_47833/47096580.py:10: RuntimeWarning: divide by zero encountered in double_scalars\n",
      "  q = average_false_positive/observed_positive\n",
      "/scratch/cbrand/ipykernel_47833/47096580.py:10: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  q = average_false_positive/observed_positive\n",
      "/scratch/cbrand/ipykernel_47833/47096580.py:10: RuntimeWarning: divide by zero encountered in double_scalars\n",
      "  q = average_false_positive/observed_positive\n",
      "/scratch/cbrand/ipykernel_47833/47096580.py:10: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  q = average_false_positive/observed_positive\n",
      "/scratch/cbrand/ipykernel_47833/47096580.py:10: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  q = average_false_positive/observed_positive\n",
      "/scratch/cbrand/ipykernel_47833/47096580.py:10: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  q = average_false_positive/observed_positive\n",
      "/scratch/cbrand/ipykernel_47833/47096580.py:10: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  q = average_false_positive/observed_positive\n",
      "/scratch/cbrand/ipykernel_47833/47096580.py:10: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  q = average_false_positive/observed_positive\n",
      "/scratch/cbrand/ipykernel_47833/47096580.py:10: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  q = average_false_positive/observed_positive\n",
      "/scratch/cbrand/ipykernel_47833/47096580.py:10: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  q = average_false_positive/observed_positive\n",
      "/scratch/cbrand/ipykernel_47833/47096580.py:10: RuntimeWarning: divide by zero encountered in double_scalars\n",
      "  q = average_false_positive/observed_positive\n",
      "/scratch/cbrand/ipykernel_47833/47096580.py:10: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  q = average_false_positive/observed_positive\n",
      "/scratch/cbrand/ipykernel_47833/47096580.py:10: RuntimeWarning: divide by zero encountered in double_scalars\n",
      "  q = average_false_positive/observed_positive\n",
      "/scratch/cbrand/ipykernel_47833/47096580.py:10: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  q = average_false_positive/observed_positive\n",
      "/scratch/cbrand/ipykernel_47833/47096580.py:10: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  q = average_false_positive/observed_positive\n",
      "/scratch/cbrand/ipykernel_47833/47096580.py:10: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  q = average_false_positive/observed_positive\n",
      "/scratch/cbrand/ipykernel_47833/47096580.py:10: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  q = average_false_positive/observed_positive\n",
      "/scratch/cbrand/ipykernel_47833/47096580.py:10: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  q = average_false_positive/observed_positive\n",
      "/scratch/cbrand/ipykernel_47833/47096580.py:10: RuntimeWarning: divide by zero encountered in double_scalars\n",
      "  q = average_false_positive/observed_positive\n",
      "/scratch/cbrand/ipykernel_47833/47096580.py:10: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  q = average_false_positive/observed_positive\n",
      "/scratch/cbrand/ipykernel_47833/47096580.py:10: RuntimeWarning: divide by zero encountered in double_scalars\n",
      "  q = average_false_positive/observed_positive\n",
      "/scratch/cbrand/ipykernel_47833/47096580.py:10: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  q = average_false_positive/observed_positive\n",
      "/scratch/cbrand/ipykernel_47833/47096580.py:10: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  q = average_false_positive/observed_positive\n",
      "/scratch/cbrand/ipykernel_47833/47096580.py:10: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  q = average_false_positive/observed_positive\n",
      "/scratch/cbrand/ipykernel_47833/47096580.py:10: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  q = average_false_positive/observed_positive\n",
      "/scratch/cbrand/ipykernel_47833/47096580.py:10: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  q = average_false_positive/observed_positive\n",
      "/scratch/cbrand/ipykernel_47833/47096580.py:10: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  q = average_false_positive/observed_positive\n",
      "/scratch/cbrand/ipykernel_47833/47096580.py:10: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  q = average_false_positive/observed_positive\n",
      "/scratch/cbrand/ipykernel_47833/47096580.py:10: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  q = average_false_positive/observed_positive\n",
      "/scratch/cbrand/ipykernel_47833/47096580.py:10: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  q = average_false_positive/observed_positive\n",
      "/scratch/cbrand/ipykernel_47833/47096580.py:10: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  q = average_false_positive/observed_positive\n",
      "/scratch/cbrand/ipykernel_47833/47096580.py:10: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  q = average_false_positive/observed_positive\n",
      "/scratch/cbrand/ipykernel_47833/47096580.py:10: RuntimeWarning: divide by zero encountered in double_scalars\n",
      "  q = average_false_positive/observed_positive\n",
      "/scratch/cbrand/ipykernel_47833/47096580.py:10: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  q = average_false_positive/observed_positive\n",
      "/scratch/cbrand/ipykernel_47833/47096580.py:10: RuntimeWarning: divide by zero encountered in double_scalars\n",
      "  q = average_false_positive/observed_positive\n",
      "/scratch/cbrand/ipykernel_47833/47096580.py:10: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  q = average_false_positive/observed_positive\n",
      "/scratch/cbrand/ipykernel_47833/47096580.py:10: RuntimeWarning: divide by zero encountered in double_scalars\n",
      "  q = average_false_positive/observed_positive\n",
      "/scratch/cbrand/ipykernel_47833/47096580.py:10: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  q = average_false_positive/observed_positive\n",
      "/scratch/cbrand/ipykernel_47833/47096580.py:10: RuntimeWarning: divide by zero encountered in double_scalars\n",
      "  q = average_false_positive/observed_positive\n",
      "/scratch/cbrand/ipykernel_47833/47096580.py:10: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  q = average_false_positive/observed_positive\n",
      "/scratch/cbrand/ipykernel_47833/47096580.py:10: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  q = average_false_positive/observed_positive\n",
      "/scratch/cbrand/ipykernel_47833/47096580.py:10: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  q = average_false_positive/observed_positive\n",
      "/scratch/cbrand/ipykernel_47833/47096580.py:10: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  q = average_false_positive/observed_positive\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "[None,\n",
       " None,\n",
       " None,\n",
       " None,\n",
       " None,\n",
       " None,\n",
       " None,\n",
       " None,\n",
       " None,\n",
       " None,\n",
       " None,\n",
       " None,\n",
       " None,\n",
       " None,\n",
       " None,\n",
       " None,\n",
       " None,\n",
       " None,\n",
       " None,\n",
       " None,\n",
       " None,\n",
       " None,\n",
       " None,\n",
       " None,\n",
       " None,\n",
       " None,\n",
       " None,\n",
       " None,\n",
       " None,\n",
       " None,\n",
       " None,\n",
       " None,\n",
       " None,\n",
       " None,\n",
       " None,\n",
       " None]"
      ]
     },
     "execution_count": 297,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "[reportFDRcorrectedPthreshold(set_name, ontology, q_value_threshold) for set_name, ontology, q_value_threshold in combinations]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 298,
   "id": "a6faef84",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>set</th>\n",
       "      <th>ontology</th>\n",
       "      <th>q_value_threshold</th>\n",
       "      <th>p_value_threshold</th>\n",
       "      <th>observed_positive</th>\n",
       "      <th>average_false_positive</th>\n",
       "      <th>q</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>altai</td>\n",
       "      <td>GWAS</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.0009</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>altai</td>\n",
       "      <td>GWAS</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.0009</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>altai</td>\n",
       "      <td>HPO</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.0010</td>\n",
       "      <td>0</td>\n",
       "      <td>0.221</td>\n",
       "      <td>inf</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>altai</td>\n",
       "      <td>HPO</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.0010</td>\n",
       "      <td>0</td>\n",
       "      <td>0.221</td>\n",
       "      <td>inf</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>archaic_specific</td>\n",
       "      <td>GWAS</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.0009</td>\n",
       "      <td>2</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>archaic_specific</td>\n",
       "      <td>GWAS</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.0009</td>\n",
       "      <td>2</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>archaic_specific</td>\n",
       "      <td>HPO</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.0049</td>\n",
       "      <td>48</td>\n",
       "      <td>1.958</td>\n",
       "      <td>0.040792</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>archaic_specific</td>\n",
       "      <td>HPO</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.0129</td>\n",
       "      <td>74</td>\n",
       "      <td>7.026</td>\n",
       "      <td>0.094946</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Browning_introgressed</td>\n",
       "      <td>GWAS</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.0019</td>\n",
       "      <td>7</td>\n",
       "      <td>0.223</td>\n",
       "      <td>0.031857</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Browning_introgressed</td>\n",
       "      <td>GWAS</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.0069</td>\n",
       "      <td>19</td>\n",
       "      <td>1.720</td>\n",
       "      <td>0.090526</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Browning_introgressed</td>\n",
       "      <td>HPO</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.0020</td>\n",
       "      <td>0</td>\n",
       "      <td>0.558</td>\n",
       "      <td>inf</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Browning_introgressed</td>\n",
       "      <td>HPO</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.0020</td>\n",
       "      <td>0</td>\n",
       "      <td>0.558</td>\n",
       "      <td>inf</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chagyrskaya</td>\n",
       "      <td>GWAS</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.0009</td>\n",
       "      <td>2</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chagyrskaya</td>\n",
       "      <td>GWAS</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.0019</td>\n",
       "      <td>2</td>\n",
       "      <td>0.143</td>\n",
       "      <td>0.071500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>chagyrskaya</td>\n",
       "      <td>HPO</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.0019</td>\n",
       "      <td>4</td>\n",
       "      <td>0.142</td>\n",
       "      <td>0.035500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>chagyrskaya</td>\n",
       "      <td>HPO</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.0029</td>\n",
       "      <td>4</td>\n",
       "      <td>0.271</td>\n",
       "      <td>0.067750</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>denisovan</td>\n",
       "      <td>GWAS</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.0022</td>\n",
       "      <td>0</td>\n",
       "      <td>0.524</td>\n",
       "      <td>inf</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>denisovan</td>\n",
       "      <td>GWAS</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.0022</td>\n",
       "      <td>0</td>\n",
       "      <td>0.524</td>\n",
       "      <td>inf</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>denisovan</td>\n",
       "      <td>HPO</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.0019</td>\n",
       "      <td>15</td>\n",
       "      <td>0.414</td>\n",
       "      <td>0.027600</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>denisovan</td>\n",
       "      <td>HPO</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.0039</td>\n",
       "      <td>17</td>\n",
       "      <td>1.237</td>\n",
       "      <td>0.072765</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>neanderthal</td>\n",
       "      <td>GWAS</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.0009</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>neanderthal</td>\n",
       "      <td>GWAS</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.0009</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>neanderthal</td>\n",
       "      <td>HPO</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.0029</td>\n",
       "      <td>10</td>\n",
       "      <td>0.395</td>\n",
       "      <td>0.039500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>neanderthal</td>\n",
       "      <td>HPO</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.0099</td>\n",
       "      <td>21</td>\n",
       "      <td>1.889</td>\n",
       "      <td>0.089952</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>shared</td>\n",
       "      <td>GWAS</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.0009</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>shared</td>\n",
       "      <td>GWAS</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.0009</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>shared</td>\n",
       "      <td>HPO</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.0009</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>shared</td>\n",
       "      <td>HPO</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.0009</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>Vernot_introgressed</td>\n",
       "      <td>GWAS</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.0032</td>\n",
       "      <td>0</td>\n",
       "      <td>0.748</td>\n",
       "      <td>inf</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>Vernot_introgressed</td>\n",
       "      <td>GWAS</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.0032</td>\n",
       "      <td>0</td>\n",
       "      <td>0.748</td>\n",
       "      <td>inf</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>Vernot_introgressed</td>\n",
       "      <td>HPO</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.0011</td>\n",
       "      <td>0</td>\n",
       "      <td>0.233</td>\n",
       "      <td>inf</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>Vernot_introgressed</td>\n",
       "      <td>HPO</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.0011</td>\n",
       "      <td>0</td>\n",
       "      <td>0.233</td>\n",
       "      <td>inf</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>vindija</td>\n",
       "      <td>GWAS</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.0009</td>\n",
       "      <td>2</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>vindija</td>\n",
       "      <td>GWAS</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.0019</td>\n",
       "      <td>2</td>\n",
       "      <td>0.173</td>\n",
       "      <td>0.086500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>vindija</td>\n",
       "      <td>HPO</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.0009</td>\n",
       "      <td>2</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>vindija</td>\n",
       "      <td>HPO</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.0019</td>\n",
       "      <td>3</td>\n",
       "      <td>0.182</td>\n",
       "      <td>0.060667</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                      set ontology  q_value_threshold  p_value_threshold  \\\n",
       "0                   altai     GWAS               0.05             0.0009   \n",
       "1                   altai     GWAS               0.10             0.0009   \n",
       "2                   altai      HPO               0.05             0.0010   \n",
       "3                   altai      HPO               0.10             0.0010   \n",
       "4        archaic_specific     GWAS               0.05             0.0009   \n",
       "5        archaic_specific     GWAS               0.10             0.0009   \n",
       "6        archaic_specific      HPO               0.05             0.0049   \n",
       "7        archaic_specific      HPO               0.10             0.0129   \n",
       "8   Browning_introgressed     GWAS               0.05             0.0019   \n",
       "9   Browning_introgressed     GWAS               0.10             0.0069   \n",
       "10  Browning_introgressed      HPO               0.05             0.0020   \n",
       "11  Browning_introgressed      HPO               0.10             0.0020   \n",
       "12            chagyrskaya     GWAS               0.05             0.0009   \n",
       "13            chagyrskaya     GWAS               0.10             0.0019   \n",
       "14            chagyrskaya      HPO               0.05             0.0019   \n",
       "15            chagyrskaya      HPO               0.10             0.0029   \n",
       "16              denisovan     GWAS               0.05             0.0022   \n",
       "17              denisovan     GWAS               0.10             0.0022   \n",
       "18              denisovan      HPO               0.05             0.0019   \n",
       "19              denisovan      HPO               0.10             0.0039   \n",
       "20            neanderthal     GWAS               0.05             0.0009   \n",
       "21            neanderthal     GWAS               0.10             0.0009   \n",
       "22            neanderthal      HPO               0.05             0.0029   \n",
       "23            neanderthal      HPO               0.10             0.0099   \n",
       "24                 shared     GWAS               0.05             0.0009   \n",
       "25                 shared     GWAS               0.10             0.0009   \n",
       "26                 shared      HPO               0.05             0.0009   \n",
       "27                 shared      HPO               0.10             0.0009   \n",
       "28    Vernot_introgressed     GWAS               0.05             0.0032   \n",
       "29    Vernot_introgressed     GWAS               0.10             0.0032   \n",
       "30    Vernot_introgressed      HPO               0.05             0.0011   \n",
       "31    Vernot_introgressed      HPO               0.10             0.0011   \n",
       "32                vindija     GWAS               0.05             0.0009   \n",
       "33                vindija     GWAS               0.10             0.0019   \n",
       "34                vindija      HPO               0.05             0.0009   \n",
       "35                vindija      HPO               0.10             0.0019   \n",
       "\n",
       "    observed_positive  average_false_positive         q  \n",
       "0                   1                   0.000  0.000000  \n",
       "1                   1                   0.000  0.000000  \n",
       "2                   0                   0.221       inf  \n",
       "3                   0                   0.221       inf  \n",
       "4                   2                   0.000  0.000000  \n",
       "5                   2                   0.000  0.000000  \n",
       "6                  48                   1.958  0.040792  \n",
       "7                  74                   7.026  0.094946  \n",
       "8                   7                   0.223  0.031857  \n",
       "9                  19                   1.720  0.090526  \n",
       "10                  0                   0.558       inf  \n",
       "11                  0                   0.558       inf  \n",
       "12                  2                   0.000  0.000000  \n",
       "13                  2                   0.143  0.071500  \n",
       "14                  4                   0.142  0.035500  \n",
       "15                  4                   0.271  0.067750  \n",
       "16                  0                   0.524       inf  \n",
       "17                  0                   0.524       inf  \n",
       "18                 15                   0.414  0.027600  \n",
       "19                 17                   1.237  0.072765  \n",
       "20                  1                   0.000  0.000000  \n",
       "21                  1                   0.000  0.000000  \n",
       "22                 10                   0.395  0.039500  \n",
       "23                 21                   1.889  0.089952  \n",
       "24                  1                   0.000  0.000000  \n",
       "25                  1                   0.000  0.000000  \n",
       "26                  1                   0.000  0.000000  \n",
       "27                  1                   0.000  0.000000  \n",
       "28                  0                   0.748       inf  \n",
       "29                  0                   0.748       inf  \n",
       "30                  0                   0.233       inf  \n",
       "31                  0                   0.233       inf  \n",
       "32                  2                   0.000  0.000000  \n",
       "33                  2                   0.173  0.086500  \n",
       "34                  2                   0.000  0.000000  \n",
       "35                  3                   0.182  0.060667  "
      ]
     },
     "execution_count": 298,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fdr_table = pd.DataFrame(fdr_table, columns = ['set', 'ontology', 'q_value_threshold', 'p_value_threshold','observed_positive','average_false_positive','q'])\n",
    "fdr_table"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "daab6a20",
   "metadata": {},
   "source": [
    "# Novel Isoforms <a class = 'anchor' id = 'novelisoforms'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3cbaec61",
   "metadata": {},
   "source": [
    "Let's subset the main dataframe to a simpler file to estimate the effects of a given SAV on the canonical transcript/protein."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 299,
   "id": "b21906e8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref_allele</th>\n",
       "      <th>alt_allele</th>\n",
       "      <th>annotation</th>\n",
       "      <th>ag_delta</th>\n",
       "      <th>al_delta</th>\n",
       "      <th>dg_delta</th>\n",
       "      <th>dl_delta</th>\n",
       "      <th>ag_pos</th>\n",
       "      <th>al_pos</th>\n",
       "      <th>dg_pos</th>\n",
       "      <th>dl_pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>chr1</td>\n",
       "      <td>864726</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.22</td>\n",
       "      <td>17</td>\n",
       "      <td>-2</td>\n",
       "      <td>43</td>\n",
       "      <td>-2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>chr1</td>\n",
       "      <td>875708</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.29</td>\n",
       "      <td>0.01</td>\n",
       "      <td>-43</td>\n",
       "      <td>33</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>130</th>\n",
       "      <td>chr1</td>\n",
       "      <td>909768</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.29</td>\n",
       "      <td>0.13</td>\n",
       "      <td>0.03</td>\n",
       "      <td>7</td>\n",
       "      <td>26</td>\n",
       "      <td>-24</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>155</th>\n",
       "      <td>chr1</td>\n",
       "      <td>956622</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.24</td>\n",
       "      <td>0.02</td>\n",
       "      <td>-2</td>\n",
       "      <td>50</td>\n",
       "      <td>-2</td>\n",
       "      <td>40</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>210</th>\n",
       "      <td>chr1</td>\n",
       "      <td>975133</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.22</td>\n",
       "      <td>0.08</td>\n",
       "      <td>-3</td>\n",
       "      <td>14</td>\n",
       "      <td>-3</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    chrom     pos ref_allele alt_allele annotation  ag_delta  al_delta  \\\n",
       "24   chr1  864726          T          A     SAMD11       0.0      0.00   \n",
       "65   chr1  875708          C          G     SAMD11       0.0      0.00   \n",
       "130  chr1  909768          A          G    PLEKHN1       0.0      0.29   \n",
       "155  chr1  956622          C          T       AGRN       0.0      0.00   \n",
       "210  chr1  975133          T          A       AGRN       0.0      0.00   \n",
       "\n",
       "     dg_delta  dl_delta  ag_pos  al_pos  dg_pos  dl_pos  \n",
       "24       0.02      0.22      17      -2      43      -2  \n",
       "65       0.29      0.01     -43      33       0      12  \n",
       "130      0.13      0.03       7      26     -24       1  \n",
       "155      0.24      0.02      -2      50      -2      40  \n",
       "210      0.22      0.08      -3      14      -3      14  "
      ]
     },
     "execution_count": 299,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "isoform_builder_input = data_2[['chrom','pos','ref_allele','alt_allele','annotation','ag_delta','al_delta','dg_delta','dl_delta','ag_pos','al_pos','dg_pos','dl_pos']]\n",
    "isoform_builder_input.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 300,
   "id": "62334ab4",
   "metadata": {},
   "outputs": [],
   "source": [
    "isoform_builder_input.to_csv('../novel_isoforms/isoform_builder_input.txt', sep = '\\t', header = False, index = False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6011fd8e",
   "metadata": {},
   "source": [
    "Let's analyze the data."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 301,
   "id": "2bc0a3fd",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref_allele</th>\n",
       "      <th>alt_allele</th>\n",
       "      <th>gene</th>\n",
       "      <th>strand</th>\n",
       "      <th>consequence</th>\n",
       "      <th>delta_pos</th>\n",
       "      <th>exon_number</th>\n",
       "      <th>exonic_intronic</th>\n",
       "      <th>sequence_effect</th>\n",
       "      <th>canonical_transcript_len</th>\n",
       "      <th>novel_transcript_len</th>\n",
       "      <th>canonical_len-novel_len</th>\n",
       "      <th>n_amino_acids</th>\n",
       "      <th>canonical_start_codon</th>\n",
       "      <th>novel_start_codon</th>\n",
       "      <th>n_premature_stops</th>\n",
       "      <th>result</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>864726</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>donor loss</td>\n",
       "      <td>-2</td>\n",
       "      <td>2</td>\n",
       "      <td>intronic</td>\n",
       "      <td>intron not retained, variant excluded</td>\n",
       "      <td>2551</td>\n",
       "      <td>2551</td>\n",
       "      <td>0</td>\n",
       "      <td>681</td>\n",
       "      <td>ATG</td>\n",
       "      <td>ATG</td>\n",
       "      <td>.</td>\n",
       "      <td>No effect</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>875708</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>donor gain</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>intronic</td>\n",
       "      <td>upstream effect, variant excluded, sequence ad...</td>\n",
       "      <td>2551</td>\n",
       "      <td>3419</td>\n",
       "      <td>868</td>\n",
       "      <td>681</td>\n",
       "      <td>ATG</td>\n",
       "      <td>ATG</td>\n",
       "      <td>8</td>\n",
       "      <td>Longer protein w/ PTCs</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>909768</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>acceptor loss</td>\n",
       "      <td>26</td>\n",
       "      <td>16</td>\n",
       "      <td>intronic</td>\n",
       "      <td>exon not skipped, variant excluded</td>\n",
       "      <td>2404</td>\n",
       "      <td>2404</td>\n",
       "      <td>0</td>\n",
       "      <td>611</td>\n",
       "      <td>ATG</td>\n",
       "      <td>ATG</td>\n",
       "      <td>.</td>\n",
       "      <td>No effect</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>956622</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>donor gain</td>\n",
       "      <td>-2</td>\n",
       "      <td>1</td>\n",
       "      <td>intronic</td>\n",
       "      <td>upstream effect, variant excluded, sequence ad...</td>\n",
       "      <td>7323</td>\n",
       "      <td>8190</td>\n",
       "      <td>867</td>\n",
       "      <td>1907</td>\n",
       "      <td>ATG</td>\n",
       "      <td>CCG</td>\n",
       "      <td>.</td>\n",
       "      <td>Longer UTR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>975133</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>donor gain</td>\n",
       "      <td>-3</td>\n",
       "      <td>3</td>\n",
       "      <td>intronic</td>\n",
       "      <td>upstream effect, variant excluded, sequence ad...</td>\n",
       "      <td>7323</td>\n",
       "      <td>11749</td>\n",
       "      <td>4426</td>\n",
       "      <td>1907</td>\n",
       "      <td>ATG</td>\n",
       "      <td>ATG</td>\n",
       "      <td>51</td>\n",
       "      <td>Longer protein w/ PTCs</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>980460</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>donor loss</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>intronic</td>\n",
       "      <td>intron not retained, variant excluded</td>\n",
       "      <td>7323</td>\n",
       "      <td>7323</td>\n",
       "      <td>0</td>\n",
       "      <td>1907</td>\n",
       "      <td>ATG</td>\n",
       "      <td>ATG</td>\n",
       "      <td>.</td>\n",
       "      <td>No effect</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr1</td>\n",
       "      <td>982099</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>donor loss</td>\n",
       "      <td>-6</td>\n",
       "      <td>18</td>\n",
       "      <td>exonic</td>\n",
       "      <td>intron not retained, variant included</td>\n",
       "      <td>7323</td>\n",
       "      <td>7323</td>\n",
       "      <td>0</td>\n",
       "      <td>1907</td>\n",
       "      <td>ATG</td>\n",
       "      <td>ATG</td>\n",
       "      <td>.</td>\n",
       "      <td>Single missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1117486</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>TTLL10</td>\n",
       "      <td>+</td>\n",
       "      <td>donor gain</td>\n",
       "      <td>-1</td>\n",
       "      <td>9</td>\n",
       "      <td>intronic</td>\n",
       "      <td>upstream effect, variant excluded, sequence ad...</td>\n",
       "      <td>2283</td>\n",
       "      <td>2573</td>\n",
       "      <td>290</td>\n",
       "      <td>673</td>\n",
       "      <td>ATG</td>\n",
       "      <td>ATG</td>\n",
       "      <td>3</td>\n",
       "      <td>Longer protein w/ PTCs</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1153630</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SDF4</td>\n",
       "      <td>-</td>\n",
       "      <td>donor gain</td>\n",
       "      <td>-2</td>\n",
       "      <td>6</td>\n",
       "      <td>intronic</td>\n",
       "      <td>upstream effect, variant excluded, sequence ad...</td>\n",
       "      <td>2079</td>\n",
       "      <td>2285</td>\n",
       "      <td>206</td>\n",
       "      <td>348</td>\n",
       "      <td>ATG</td>\n",
       "      <td>ATG</td>\n",
       "      <td>2</td>\n",
       "      <td>Longer protein w/ PTCs</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1166887</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SDF4</td>\n",
       "      <td>-</td>\n",
       "      <td>donor gain</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>intronic</td>\n",
       "      <td>upstream effect, variant excluded, sequence ad...</td>\n",
       "      <td>2079</td>\n",
       "      <td>2464</td>\n",
       "      <td>385</td>\n",
       "      <td>348</td>\n",
       "      <td>ATG</td>\n",
       "      <td>CCA</td>\n",
       "      <td>.</td>\n",
       "      <td>Longer UTR</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chrom      pos ref_allele alt_allele     gene strand    consequence  \\\n",
       "0  chr1   864726          T          A   SAMD11      +     donor loss   \n",
       "1  chr1   875708          C          G   SAMD11      +     donor gain   \n",
       "2  chr1   909768          A          G  PLEKHN1      +  acceptor loss   \n",
       "3  chr1   956622          C          T     AGRN      +     donor gain   \n",
       "4  chr1   975133          T          A     AGRN      +     donor gain   \n",
       "5  chr1   980460          G          A     AGRN      +     donor loss   \n",
       "6  chr1   982099          T          C     AGRN      +     donor loss   \n",
       "7  chr1  1117486          A          G   TTLL10      +     donor gain   \n",
       "8  chr1  1153630          G          A     SDF4      -     donor gain   \n",
       "9  chr1  1166887          T          C     SDF4      -     donor gain   \n",
       "\n",
       "   delta_pos  exon_number exonic_intronic  \\\n",
       "0         -2            2        intronic   \n",
       "1          0            7        intronic   \n",
       "2         26           16        intronic   \n",
       "3         -2            1        intronic   \n",
       "4         -3            3        intronic   \n",
       "5          0           12        intronic   \n",
       "6         -6           18          exonic   \n",
       "7         -1            9        intronic   \n",
       "8         -2            6        intronic   \n",
       "9          0            1        intronic   \n",
       "\n",
       "                                     sequence_effect  \\\n",
       "0              intron not retained, variant excluded   \n",
       "1  upstream effect, variant excluded, sequence ad...   \n",
       "2                 exon not skipped, variant excluded   \n",
       "3  upstream effect, variant excluded, sequence ad...   \n",
       "4  upstream effect, variant excluded, sequence ad...   \n",
       "5              intron not retained, variant excluded   \n",
       "6              intron not retained, variant included   \n",
       "7  upstream effect, variant excluded, sequence ad...   \n",
       "8  upstream effect, variant excluded, sequence ad...   \n",
       "9  upstream effect, variant excluded, sequence ad...   \n",
       "\n",
       "   canonical_transcript_len  novel_transcript_len  canonical_len-novel_len  \\\n",
       "0                      2551                  2551                        0   \n",
       "1                      2551                  3419                      868   \n",
       "2                      2404                  2404                        0   \n",
       "3                      7323                  8190                      867   \n",
       "4                      7323                 11749                     4426   \n",
       "5                      7323                  7323                        0   \n",
       "6                      7323                  7323                        0   \n",
       "7                      2283                  2573                      290   \n",
       "8                      2079                  2285                      206   \n",
       "9                      2079                  2464                      385   \n",
       "\n",
       "  n_amino_acids canonical_start_codon novel_start_codon n_premature_stops  \\\n",
       "0           681                   ATG               ATG                 .   \n",
       "1           681                   ATG               ATG                 8   \n",
       "2           611                   ATG               ATG                 .   \n",
       "3          1907                   ATG               CCG                 .   \n",
       "4          1907                   ATG               ATG                51   \n",
       "5          1907                   ATG               ATG                 .   \n",
       "6          1907                   ATG               ATG                 .   \n",
       "7           673                   ATG               ATG                 3   \n",
       "8           348                   ATG               ATG                 2   \n",
       "9           348                   ATG               CCA                 .   \n",
       "\n",
       "                   result  \n",
       "0               No effect  \n",
       "1  Longer protein w/ PTCs  \n",
       "2               No effect  \n",
       "3              Longer UTR  \n",
       "4  Longer protein w/ PTCs  \n",
       "5               No effect  \n",
       "6         Single missense  \n",
       "7  Longer protein w/ PTCs  \n",
       "8  Longer protein w/ PTCs  \n",
       "9              Longer UTR  "
      ]
     },
     "execution_count": 301,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "novel_isoforms = pd.read_csv('../novel_isoforms/novel_isoforms.txt', sep = '\\t', header = 0)\n",
    "novel_isoforms['pos'] = novel_isoforms['pos']\n",
    "novel_isoforms.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 302,
   "id": "22514550",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "5950"
      ]
     },
     "execution_count": 302,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(novel_isoforms)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 303,
   "id": "fed167bc",
   "metadata": {},
   "outputs": [],
   "source": [
    "Vernot_allele_origin_series = data_2['Vernot_allele_origin'].reset_index(drop = True)\n",
    "Browning_allele_origin_series = data_2['Browning_allele_origin'].reset_index(drop = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 304,
   "id": "c23b9ca4",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref_allele</th>\n",
       "      <th>alt_allele</th>\n",
       "      <th>gene</th>\n",
       "      <th>strand</th>\n",
       "      <th>consequence</th>\n",
       "      <th>delta_pos</th>\n",
       "      <th>exon_number</th>\n",
       "      <th>exonic_intronic</th>\n",
       "      <th>sequence_effect</th>\n",
       "      <th>canonical_transcript_len</th>\n",
       "      <th>novel_transcript_len</th>\n",
       "      <th>canonical_len-novel_len</th>\n",
       "      <th>n_amino_acids</th>\n",
       "      <th>canonical_start_codon</th>\n",
       "      <th>novel_start_codon</th>\n",
       "      <th>n_premature_stops</th>\n",
       "      <th>result</th>\n",
       "      <th>Vernot_allele_origin</th>\n",
       "      <th>Browning_allele_origin</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>864726</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>donor loss</td>\n",
       "      <td>-2</td>\n",
       "      <td>2</td>\n",
       "      <td>intronic</td>\n",
       "      <td>intron not retained, variant excluded</td>\n",
       "      <td>2551</td>\n",
       "      <td>2551</td>\n",
       "      <td>0</td>\n",
       "      <td>681</td>\n",
       "      <td>ATG</td>\n",
       "      <td>ATG</td>\n",
       "      <td>.</td>\n",
       "      <td>No effect</td>\n",
       "      <td>ancient</td>\n",
       "      <td>ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>875708</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>donor gain</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>intronic</td>\n",
       "      <td>upstream effect, variant excluded, sequence ad...</td>\n",
       "      <td>2551</td>\n",
       "      <td>3419</td>\n",
       "      <td>868</td>\n",
       "      <td>681</td>\n",
       "      <td>ATG</td>\n",
       "      <td>ATG</td>\n",
       "      <td>8</td>\n",
       "      <td>Longer protein w/ PTCs</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>909768</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>acceptor loss</td>\n",
       "      <td>26</td>\n",
       "      <td>16</td>\n",
       "      <td>intronic</td>\n",
       "      <td>exon not skipped, variant excluded</td>\n",
       "      <td>2404</td>\n",
       "      <td>2404</td>\n",
       "      <td>0</td>\n",
       "      <td>611</td>\n",
       "      <td>ATG</td>\n",
       "      <td>ATG</td>\n",
       "      <td>.</td>\n",
       "      <td>No effect</td>\n",
       "      <td>ancient</td>\n",
       "      <td>ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>956622</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>donor gain</td>\n",
       "      <td>-2</td>\n",
       "      <td>1</td>\n",
       "      <td>intronic</td>\n",
       "      <td>upstream effect, variant excluded, sequence ad...</td>\n",
       "      <td>7323</td>\n",
       "      <td>8190</td>\n",
       "      <td>867</td>\n",
       "      <td>1907</td>\n",
       "      <td>ATG</td>\n",
       "      <td>CCG</td>\n",
       "      <td>.</td>\n",
       "      <td>Longer UTR</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>975133</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>donor gain</td>\n",
       "      <td>-3</td>\n",
       "      <td>3</td>\n",
       "      <td>intronic</td>\n",
       "      <td>upstream effect, variant excluded, sequence ad...</td>\n",
       "      <td>7323</td>\n",
       "      <td>11749</td>\n",
       "      <td>4426</td>\n",
       "      <td>1907</td>\n",
       "      <td>ATG</td>\n",
       "      <td>ATG</td>\n",
       "      <td>51</td>\n",
       "      <td>Longer protein w/ PTCs</td>\n",
       "      <td>ancient</td>\n",
       "      <td>ancient</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chrom     pos ref_allele alt_allele     gene strand    consequence  \\\n",
       "0  chr1  864726          T          A   SAMD11      +     donor loss   \n",
       "1  chr1  875708          C          G   SAMD11      +     donor gain   \n",
       "2  chr1  909768          A          G  PLEKHN1      +  acceptor loss   \n",
       "3  chr1  956622          C          T     AGRN      +     donor gain   \n",
       "4  chr1  975133          T          A     AGRN      +     donor gain   \n",
       "\n",
       "   delta_pos  exon_number exonic_intronic  \\\n",
       "0         -2            2        intronic   \n",
       "1          0            7        intronic   \n",
       "2         26           16        intronic   \n",
       "3         -2            1        intronic   \n",
       "4         -3            3        intronic   \n",
       "\n",
       "                                     sequence_effect  \\\n",
       "0              intron not retained, variant excluded   \n",
       "1  upstream effect, variant excluded, sequence ad...   \n",
       "2                 exon not skipped, variant excluded   \n",
       "3  upstream effect, variant excluded, sequence ad...   \n",
       "4  upstream effect, variant excluded, sequence ad...   \n",
       "\n",
       "   canonical_transcript_len  novel_transcript_len  canonical_len-novel_len  \\\n",
       "0                      2551                  2551                        0   \n",
       "1                      2551                  3419                      868   \n",
       "2                      2404                  2404                        0   \n",
       "3                      7323                  8190                      867   \n",
       "4                      7323                 11749                     4426   \n",
       "\n",
       "  n_amino_acids canonical_start_codon novel_start_codon n_premature_stops  \\\n",
       "0           681                   ATG               ATG                 .   \n",
       "1           681                   ATG               ATG                 8   \n",
       "2           611                   ATG               ATG                 .   \n",
       "3          1907                   ATG               CCG                 .   \n",
       "4          1907                   ATG               ATG                51   \n",
       "\n",
       "                   result    Vernot_allele_origin  Browning_allele_origin  \n",
       "0               No effect                 ancient                 ancient  \n",
       "1  Longer protein w/ PTCs        archaic-specific        archaic-specific  \n",
       "2               No effect                 ancient                 ancient  \n",
       "3              Longer UTR  low-confidence ancient  low-confidence ancient  \n",
       "4  Longer protein w/ PTCs                 ancient                 ancient  "
      ]
     },
     "execution_count": 304,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "novel_isoforms_with_allele_origin = novel_isoforms.copy()\n",
    "novel_isoforms_with_allele_origin['Vernot_allele_origin'] = Vernot_allele_origin_series\n",
    "novel_isoforms_with_allele_origin['Browning_allele_origin'] = Browning_allele_origin_series\n",
    "novel_isoforms_with_allele_origin.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 305,
   "id": "91853cb1",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "5950"
      ]
     },
     "execution_count": 305,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(novel_isoforms_with_allele_origin)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 306,
   "id": "66d0e216",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>N</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>result</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Longer UTR</th>\n",
       "      <td>479</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Longer protein w/ PTCs</th>\n",
       "      <td>1897</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Longer protein w/o PTCs</th>\n",
       "      <td>333</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>No effect</th>\n",
       "      <td>2391</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Single missense</th>\n",
       "      <td>396</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Translated product unknown</th>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Truncated UTR</th>\n",
       "      <td>116</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Truncated protein</th>\n",
       "      <td>332</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                               N\n",
       "result                          \n",
       "Longer UTR                   479\n",
       "Longer protein w/ PTCs      1897\n",
       "Longer protein w/o PTCs      333\n",
       "No effect                   2391\n",
       "Single missense              396\n",
       "Translated product unknown     6\n",
       "Truncated UTR                116\n",
       "Truncated protein            332"
      ]
     },
     "execution_count": 306,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "novel_isoforms_with_allele_origin.groupby('result').size().to_frame('N')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 307,
   "id": "fc6a353a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>N</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Vernot_allele_origin</th>\n",
       "      <th>result</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"8\" valign=\"top\">ancient</th>\n",
       "      <th>Longer UTR</th>\n",
       "      <td>151</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Longer protein w/ PTCs</th>\n",
       "      <td>634</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Longer protein w/o PTCs</th>\n",
       "      <td>95</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>No effect</th>\n",
       "      <td>1092</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Single missense</th>\n",
       "      <td>142</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Translated product unknown</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Truncated UTR</th>\n",
       "      <td>44</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Truncated protein</th>\n",
       "      <td>93</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"8\" valign=\"top\">archaic-specific</th>\n",
       "      <th>Longer UTR</th>\n",
       "      <td>206</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Longer protein w/ PTCs</th>\n",
       "      <td>753</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Longer protein w/o PTCs</th>\n",
       "      <td>153</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>No effect</th>\n",
       "      <td>729</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Single missense</th>\n",
       "      <td>137</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Translated product unknown</th>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Truncated UTR</th>\n",
       "      <td>48</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Truncated protein</th>\n",
       "      <td>155</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"7\" valign=\"top\">introgressed</th>\n",
       "      <th>Longer UTR</th>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Longer protein w/ PTCs</th>\n",
       "      <td>91</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Longer protein w/o PTCs</th>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>No effect</th>\n",
       "      <td>69</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Single missense</th>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Truncated UTR</th>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Truncated protein</th>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"7\" valign=\"top\">low-confidence ancient</th>\n",
       "      <th>Longer UTR</th>\n",
       "      <td>96</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Longer protein w/ PTCs</th>\n",
       "      <td>419</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Longer protein w/o PTCs</th>\n",
       "      <td>71</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>No effect</th>\n",
       "      <td>501</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Single missense</th>\n",
       "      <td>96</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Truncated UTR</th>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Truncated protein</th>\n",
       "      <td>66</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                      N\n",
       "Vernot_allele_origin   result                          \n",
       "ancient                Longer UTR                   151\n",
       "                       Longer protein w/ PTCs       634\n",
       "                       Longer protein w/o PTCs       95\n",
       "                       No effect                   1092\n",
       "                       Single missense              142\n",
       "                       Translated product unknown     1\n",
       "                       Truncated UTR                 44\n",
       "                       Truncated protein             93\n",
       "archaic-specific       Longer UTR                   206\n",
       "                       Longer protein w/ PTCs       753\n",
       "                       Longer protein w/o PTCs      153\n",
       "                       No effect                    729\n",
       "                       Single missense              137\n",
       "                       Translated product unknown     5\n",
       "                       Truncated UTR                 48\n",
       "                       Truncated protein            155\n",
       "introgressed           Longer UTR                    26\n",
       "                       Longer protein w/ PTCs        91\n",
       "                       Longer protein w/o PTCs       14\n",
       "                       No effect                     69\n",
       "                       Single missense               21\n",
       "                       Truncated UTR                  5\n",
       "                       Truncated protein             18\n",
       "low-confidence ancient Longer UTR                    96\n",
       "                       Longer protein w/ PTCs       419\n",
       "                       Longer protein w/o PTCs       71\n",
       "                       No effect                    501\n",
       "                       Single missense               96\n",
       "                       Truncated UTR                 19\n",
       "                       Truncated protein             66"
      ]
     },
     "execution_count": 307,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "novel_isoforms_with_allele_origin.groupby(['Vernot_allele_origin','result']).size().to_frame('N')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 308,
   "id": "758ab2e6",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>N</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Browning_allele_origin</th>\n",
       "      <th>result</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"8\" valign=\"top\">ancient</th>\n",
       "      <th>Longer UTR</th>\n",
       "      <td>149</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Longer protein w/ PTCs</th>\n",
       "      <td>620</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Longer protein w/o PTCs</th>\n",
       "      <td>90</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>No effect</th>\n",
       "      <td>1061</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Single missense</th>\n",
       "      <td>139</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Translated product unknown</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Truncated UTR</th>\n",
       "      <td>42</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Truncated protein</th>\n",
       "      <td>93</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"8\" valign=\"top\">archaic-specific</th>\n",
       "      <th>Longer UTR</th>\n",
       "      <td>206</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Longer protein w/ PTCs</th>\n",
       "      <td>753</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Longer protein w/o PTCs</th>\n",
       "      <td>153</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>No effect</th>\n",
       "      <td>729</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Single missense</th>\n",
       "      <td>137</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Translated product unknown</th>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Truncated UTR</th>\n",
       "      <td>48</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Truncated protein</th>\n",
       "      <td>155</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"7\" valign=\"top\">introgressed</th>\n",
       "      <th>Longer UTR</th>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Longer protein w/ PTCs</th>\n",
       "      <td>133</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Longer protein w/o PTCs</th>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>No effect</th>\n",
       "      <td>148</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Single missense</th>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Truncated UTR</th>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Truncated protein</th>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"7\" valign=\"top\">low-confidence ancient</th>\n",
       "      <th>Longer UTR</th>\n",
       "      <td>99</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Longer protein w/ PTCs</th>\n",
       "      <td>391</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Longer protein w/o PTCs</th>\n",
       "      <td>65</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>No effect</th>\n",
       "      <td>453</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Single missense</th>\n",
       "      <td>92</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Truncated UTR</th>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Truncated protein</th>\n",
       "      <td>66</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                      N\n",
       "Browning_allele_origin result                          \n",
       "ancient                Longer UTR                   149\n",
       "                       Longer protein w/ PTCs       620\n",
       "                       Longer protein w/o PTCs       90\n",
       "                       No effect                   1061\n",
       "                       Single missense              139\n",
       "                       Translated product unknown     1\n",
       "                       Truncated UTR                 42\n",
       "                       Truncated protein             93\n",
       "archaic-specific       Longer UTR                   206\n",
       "                       Longer protein w/ PTCs       753\n",
       "                       Longer protein w/o PTCs      153\n",
       "                       No effect                    729\n",
       "                       Single missense              137\n",
       "                       Translated product unknown     5\n",
       "                       Truncated UTR                 48\n",
       "                       Truncated protein            155\n",
       "introgressed           Longer UTR                    25\n",
       "                       Longer protein w/ PTCs       133\n",
       "                       Longer protein w/o PTCs       25\n",
       "                       No effect                    148\n",
       "                       Single missense               28\n",
       "                       Truncated UTR                  9\n",
       "                       Truncated protein             18\n",
       "low-confidence ancient Longer UTR                    99\n",
       "                       Longer protein w/ PTCs       391\n",
       "                       Longer protein w/o PTCs       65\n",
       "                       No effect                    453\n",
       "                       Single missense               92\n",
       "                       Truncated UTR                 17\n",
       "                       Truncated protein             66"
      ]
     },
     "execution_count": 308,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "novel_isoforms_with_allele_origin.groupby(['Browning_allele_origin','result']).size().to_frame('N')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "06149265",
   "metadata": {},
   "source": [
    "Let's test whether these distributions are statistically different to one another."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 309,
   "id": "a4093cef",
   "metadata": {},
   "outputs": [],
   "source": [
    "vernot_effect_counts = np.array([[151,634,95,1092,142,44,93],[206,753,153,729,137,48,155],[26,91,14,69,21,5,18]])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 310,
   "id": "b81aafa1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(137.72250812383166, 1.725254484851994e-23)"
      ]
     },
     "execution_count": 310,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "g, p, dof, expctd = chi2_contingency(vernot_effect_counts, lambda_ = 'log-likelihood')\n",
    "g, p"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 311,
   "id": "e0429fbc",
   "metadata": {},
   "outputs": [],
   "source": [
    "browning_effect_counts = np.array([[149,620,90,1061,139,42,93],[206,753,153,729,137,48,155],[25,133,25,148,28,9,18]])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 312,
   "id": "e0ad7ce7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(121.2718284505004, 3.446077675710777e-20)"
      ]
     },
     "execution_count": 312,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "g, p, dof, expctd = chi2_contingency(browning_effect_counts, lambda_ = 'log-likelihood')\n",
    "g, p"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2df5a536",
   "metadata": {},
   "source": [
    "# Gene Characteristics <a class = 'anchor' id = 'genecharacteristics'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d6b42bcd",
   "metadata": {},
   "source": [
    "## N Exons, CDS Length, and Gene Length <a class = 'anchor' id = 'physical'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "46c9f9c2",
   "metadata": {},
   "source": [
    "Do genes with more exons or longer genes have more SAVs? Let's load the annotation data first."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 313,
   "id": "9521d3a3",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>chrom</th>\n",
       "      <th>strand</th>\n",
       "      <th>tx_start</th>\n",
       "      <th>tx_end</th>\n",
       "      <th>exon_start</th>\n",
       "      <th>exon_end</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>chr1</td>\n",
       "      <td>+</td>\n",
       "      <td>69090</td>\n",
       "      <td>70008</td>\n",
       "      <td>69090,</td>\n",
       "      <td>70008,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AL627309.1</td>\n",
       "      <td>chr1</td>\n",
       "      <td>-</td>\n",
       "      <td>134900</td>\n",
       "      <td>139379</td>\n",
       "      <td>134900,137620,</td>\n",
       "      <td>135802,139379,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>OR4F29</td>\n",
       "      <td>chr1</td>\n",
       "      <td>+</td>\n",
       "      <td>367639</td>\n",
       "      <td>368634</td>\n",
       "      <td>367639,</td>\n",
       "      <td>368634,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>OR4F16</td>\n",
       "      <td>chr1</td>\n",
       "      <td>-</td>\n",
       "      <td>621095</td>\n",
       "      <td>622034</td>\n",
       "      <td>621095,</td>\n",
       "      <td>622034,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AL669831.1</td>\n",
       "      <td>chr1</td>\n",
       "      <td>-</td>\n",
       "      <td>738531</td>\n",
       "      <td>739137</td>\n",
       "      <td>738531,738787,739120,</td>\n",
       "      <td>738618,738812,739137,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>AL645608.2</td>\n",
       "      <td>chr1</td>\n",
       "      <td>+</td>\n",
       "      <td>818042</td>\n",
       "      <td>819983</td>\n",
       "      <td>818042,819495,819960,</td>\n",
       "      <td>818058,819513,819983,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>chr1</td>\n",
       "      <td>+</td>\n",
       "      <td>861117</td>\n",
       "      <td>879955</td>\n",
       "      <td>861117,861301,865534,866418,871151,874419,8746...</td>\n",
       "      <td>861180,861393,865716,866469,871276,874509,8748...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>AL645608.1</td>\n",
       "      <td>chr1</td>\n",
       "      <td>-</td>\n",
       "      <td>861263</td>\n",
       "      <td>866445</td>\n",
       "      <td>861263,863254,865555,865665,865989,866425,</td>\n",
       "      <td>861406,863261,865660,865719,865996,866445,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>NOC2L</td>\n",
       "      <td>chr1</td>\n",
       "      <td>-</td>\n",
       "      <td>879583</td>\n",
       "      <td>894670</td>\n",
       "      <td>879583,880436,880897,881552,881781,883510,8838...</td>\n",
       "      <td>880180,880526,881033,881666,881925,883612,8839...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>KLHL17</td>\n",
       "      <td>chr1</td>\n",
       "      <td>+</td>\n",
       "      <td>895966</td>\n",
       "      <td>901095</td>\n",
       "      <td>895966,896672,897008,897205,897734,898083,8984...</td>\n",
       "      <td>896180,896932,897130,897427,897851,898297,8986...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         name chrom strand  tx_start  tx_end  \\\n",
       "0       OR4F5  chr1      +     69090   70008   \n",
       "1  AL627309.1  chr1      -    134900  139379   \n",
       "2      OR4F29  chr1      +    367639  368634   \n",
       "3      OR4F16  chr1      -    621095  622034   \n",
       "4  AL669831.1  chr1      -    738531  739137   \n",
       "5  AL645608.2  chr1      +    818042  819983   \n",
       "6      SAMD11  chr1      +    861117  879955   \n",
       "7  AL645608.1  chr1      -    861263  866445   \n",
       "8       NOC2L  chr1      -    879583  894670   \n",
       "9      KLHL17  chr1      +    895966  901095   \n",
       "\n",
       "                                          exon_start  \\\n",
       "0                                             69090,   \n",
       "1                                     134900,137620,   \n",
       "2                                            367639,   \n",
       "3                                            621095,   \n",
       "4                              738531,738787,739120,   \n",
       "5                              818042,819495,819960,   \n",
       "6  861117,861301,865534,866418,871151,874419,8746...   \n",
       "7         861263,863254,865555,865665,865989,866425,   \n",
       "8  879583,880436,880897,881552,881781,883510,8838...   \n",
       "9  895966,896672,897008,897205,897734,898083,8984...   \n",
       "\n",
       "                                            exon_end  \n",
       "0                                             70008,  \n",
       "1                                     135802,139379,  \n",
       "2                                            368634,  \n",
       "3                                            622034,  \n",
       "4                              738618,738812,739137,  \n",
       "5                              818058,819513,819983,  \n",
       "6  861180,861393,865716,866469,871276,874509,8748...  \n",
       "7         861406,863261,865660,865719,865996,866445,  \n",
       "8  880180,880526,881033,881666,881925,883612,8839...  \n",
       "9  896180,896932,897130,897427,897851,898297,8986...  "
      ]
     },
     "execution_count": 313,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "annotation_header = ['name','chrom','strand','tx_start','tx_end','exon_start','exon_end']\n",
    "annotations = pd.read_csv('../annotations/grch37_exon_annotations.txt', sep='\\t', skiprows=1, names=annotation_header)\n",
    "annotations.head(10)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "05d5a1c6",
   "metadata": {},
   "source": [
    "Get the length."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 314,
   "id": "af3e3f16",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "20274"
      ]
     },
     "execution_count": 314,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(annotations)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "925fc373",
   "metadata": {},
   "source": [
    "Now let's count the number of exons using the delimiter in either of the last columns."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 315,
   "id": "1ba2ea58",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>chrom</th>\n",
       "      <th>strand</th>\n",
       "      <th>tx_start</th>\n",
       "      <th>tx_end</th>\n",
       "      <th>exon_start</th>\n",
       "      <th>exon_end</th>\n",
       "      <th>n_exons</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>chr1</td>\n",
       "      <td>+</td>\n",
       "      <td>69090</td>\n",
       "      <td>70008</td>\n",
       "      <td>69090,</td>\n",
       "      <td>70008,</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AL627309.1</td>\n",
       "      <td>chr1</td>\n",
       "      <td>-</td>\n",
       "      <td>134900</td>\n",
       "      <td>139379</td>\n",
       "      <td>134900,137620,</td>\n",
       "      <td>135802,139379,</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>OR4F29</td>\n",
       "      <td>chr1</td>\n",
       "      <td>+</td>\n",
       "      <td>367639</td>\n",
       "      <td>368634</td>\n",
       "      <td>367639,</td>\n",
       "      <td>368634,</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>OR4F16</td>\n",
       "      <td>chr1</td>\n",
       "      <td>-</td>\n",
       "      <td>621095</td>\n",
       "      <td>622034</td>\n",
       "      <td>621095,</td>\n",
       "      <td>622034,</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AL669831.1</td>\n",
       "      <td>chr1</td>\n",
       "      <td>-</td>\n",
       "      <td>738531</td>\n",
       "      <td>739137</td>\n",
       "      <td>738531,738787,739120,</td>\n",
       "      <td>738618,738812,739137,</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>AL645608.2</td>\n",
       "      <td>chr1</td>\n",
       "      <td>+</td>\n",
       "      <td>818042</td>\n",
       "      <td>819983</td>\n",
       "      <td>818042,819495,819960,</td>\n",
       "      <td>818058,819513,819983,</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>chr1</td>\n",
       "      <td>+</td>\n",
       "      <td>861117</td>\n",
       "      <td>879955</td>\n",
       "      <td>861117,861301,865534,866418,871151,874419,8746...</td>\n",
       "      <td>861180,861393,865716,866469,871276,874509,8748...</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>AL645608.1</td>\n",
       "      <td>chr1</td>\n",
       "      <td>-</td>\n",
       "      <td>861263</td>\n",
       "      <td>866445</td>\n",
       "      <td>861263,863254,865555,865665,865989,866425,</td>\n",
       "      <td>861406,863261,865660,865719,865996,866445,</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>NOC2L</td>\n",
       "      <td>chr1</td>\n",
       "      <td>-</td>\n",
       "      <td>879583</td>\n",
       "      <td>894670</td>\n",
       "      <td>879583,880436,880897,881552,881781,883510,8838...</td>\n",
       "      <td>880180,880526,881033,881666,881925,883612,8839...</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>KLHL17</td>\n",
       "      <td>chr1</td>\n",
       "      <td>+</td>\n",
       "      <td>895966</td>\n",
       "      <td>901095</td>\n",
       "      <td>895966,896672,897008,897205,897734,898083,8984...</td>\n",
       "      <td>896180,896932,897130,897427,897851,898297,8986...</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         name chrom strand  tx_start  tx_end  \\\n",
       "0       OR4F5  chr1      +     69090   70008   \n",
       "1  AL627309.1  chr1      -    134900  139379   \n",
       "2      OR4F29  chr1      +    367639  368634   \n",
       "3      OR4F16  chr1      -    621095  622034   \n",
       "4  AL669831.1  chr1      -    738531  739137   \n",
       "5  AL645608.2  chr1      +    818042  819983   \n",
       "6      SAMD11  chr1      +    861117  879955   \n",
       "7  AL645608.1  chr1      -    861263  866445   \n",
       "8       NOC2L  chr1      -    879583  894670   \n",
       "9      KLHL17  chr1      +    895966  901095   \n",
       "\n",
       "                                          exon_start  \\\n",
       "0                                             69090,   \n",
       "1                                     134900,137620,   \n",
       "2                                            367639,   \n",
       "3                                            621095,   \n",
       "4                              738531,738787,739120,   \n",
       "5                              818042,819495,819960,   \n",
       "6  861117,861301,865534,866418,871151,874419,8746...   \n",
       "7         861263,863254,865555,865665,865989,866425,   \n",
       "8  879583,880436,880897,881552,881781,883510,8838...   \n",
       "9  895966,896672,897008,897205,897734,898083,8984...   \n",
       "\n",
       "                                            exon_end  n_exons  \n",
       "0                                             70008,        1  \n",
       "1                                     135802,139379,        2  \n",
       "2                                            368634,        1  \n",
       "3                                            622034,        1  \n",
       "4                              738618,738812,739137,        3  \n",
       "5                              818058,819513,819983,        3  \n",
       "6  861180,861393,865716,866469,871276,874509,8748...       14  \n",
       "7         861406,863261,865660,865719,865996,866445,        6  \n",
       "8  880180,880526,881033,881666,881925,883612,8839...       19  \n",
       "9  896180,896932,897130,897427,897851,898297,8986...       12  "
      ]
     },
     "execution_count": 315,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "annotations['n_exons'] = annotations['exon_start'].str.count(',')\n",
    "annotations.head(10)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fa904129",
   "metadata": {},
   "source": [
    "Now the total length of the gene from the start of the first exon to the end of the last exon."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 316,
   "id": "0e74246b",
   "metadata": {},
   "outputs": [],
   "source": [
    "annotations['gene_length'] = annotations['tx_end'] - annotations['tx_start']"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "414836e9",
   "metadata": {},
   "source": [
    "Now let's get the total exon length. Let's extract information as lists to make this easier and then bring everything back together. First, get a list of gene names copied by the number of exons present in that gene."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 317,
   "id": "0bdca705",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_names = annotations['name'].loc[annotations.index.repeat(annotations['n_exons'])].tolist()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c714c1f3",
   "metadata": {},
   "source": [
    "Let's also get a list of chromosome names."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 318,
   "id": "39b36fc4",
   "metadata": {},
   "outputs": [],
   "source": [
    "chroms = annotations['chrom'].loc[annotations.index.repeat(annotations['n_exons'])].tolist()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1a8395ab",
   "metadata": {},
   "source": [
    "Then grab a list of exon starts and stops."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 319,
   "id": "1f792581",
   "metadata": {},
   "outputs": [],
   "source": [
    "exon_starts = annotations['exon_start'].str.split(',').sum()\n",
    "exon_stops = annotations['exon_end'].str.split(',').sum()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cc538c0c",
   "metadata": {},
   "source": [
    "The trailing commas are creating blanks in the list so let's get rid of those."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 320,
   "id": "77270545",
   "metadata": {},
   "outputs": [],
   "source": [
    "exon_starts = list(filter(None, exon_starts))\n",
    "exon_stops = list(filter(None, exon_stops))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3c2943e6",
   "metadata": {},
   "source": [
    "Create a new dataframe and bring everything together."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 321,
   "id": "a2baf953",
   "metadata": {},
   "outputs": [],
   "source": [
    "exon_length = pd.DataFrame()\n",
    "exon_length['name'] = gene_names\n",
    "exon_length['exon_start'] = exon_starts\n",
    "exon_length['exon_stop'] = exon_stops"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "db7a3cda",
   "metadata": {},
   "source": [
    "Make sure it looks the way it should!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 322,
   "id": "044d42ae",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>exon_start</th>\n",
       "      <th>exon_stop</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>69090</td>\n",
       "      <td>70008</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AL627309.1</td>\n",
       "      <td>134900</td>\n",
       "      <td>135802</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AL627309.1</td>\n",
       "      <td>137620</td>\n",
       "      <td>139379</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>OR4F29</td>\n",
       "      <td>367639</td>\n",
       "      <td>368634</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>OR4F16</td>\n",
       "      <td>621095</td>\n",
       "      <td>622034</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>AL669831.1</td>\n",
       "      <td>738531</td>\n",
       "      <td>738618</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>AL669831.1</td>\n",
       "      <td>738787</td>\n",
       "      <td>738812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>AL669831.1</td>\n",
       "      <td>739120</td>\n",
       "      <td>739137</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>AL645608.2</td>\n",
       "      <td>818042</td>\n",
       "      <td>818058</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>AL645608.2</td>\n",
       "      <td>819495</td>\n",
       "      <td>819513</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         name exon_start exon_stop\n",
       "0       OR4F5      69090     70008\n",
       "1  AL627309.1     134900    135802\n",
       "2  AL627309.1     137620    139379\n",
       "3      OR4F29     367639    368634\n",
       "4      OR4F16     621095    622034\n",
       "5  AL669831.1     738531    738618\n",
       "6  AL669831.1     738787    738812\n",
       "7  AL669831.1     739120    739137\n",
       "8  AL645608.2     818042    818058\n",
       "9  AL645608.2     819495    819513"
      ]
     },
     "execution_count": 322,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "exon_length.head(10)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c8336b83",
   "metadata": {},
   "source": [
    "Quick sidebar! This information would be very useful as annotation data. Let's also include the strand."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 323,
   "id": "4f0b24ef",
   "metadata": {},
   "outputs": [],
   "source": [
    "strands = annotations['strand'].loc[annotations.index.repeat(annotations['n_exons'])].tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 324,
   "id": "d2189a2c",
   "metadata": {},
   "outputs": [],
   "source": [
    "hg19_exons = pd.DataFrame()\n",
    "hg19_exons['chrom'] = chroms\n",
    "hg19_exons['exon_start'] = exon_length['exon_start'].astype(int)\n",
    "hg19_exons['exon_stop'] = exon_stops\n",
    "hg19_exons['strand'] = strands\n",
    "hg19_exons['gene'] = gene_names"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 325,
   "id": "5746220b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>exon_start</th>\n",
       "      <th>exon_stop</th>\n",
       "      <th>strand</th>\n",
       "      <th>gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69090</td>\n",
       "      <td>70008</td>\n",
       "      <td>+</td>\n",
       "      <td>OR4F5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>134900</td>\n",
       "      <td>135802</td>\n",
       "      <td>-</td>\n",
       "      <td>AL627309.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>137620</td>\n",
       "      <td>139379</td>\n",
       "      <td>-</td>\n",
       "      <td>AL627309.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>367639</td>\n",
       "      <td>368634</td>\n",
       "      <td>+</td>\n",
       "      <td>OR4F29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>621095</td>\n",
       "      <td>622034</td>\n",
       "      <td>-</td>\n",
       "      <td>OR4F16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>738531</td>\n",
       "      <td>738618</td>\n",
       "      <td>-</td>\n",
       "      <td>AL669831.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr1</td>\n",
       "      <td>738787</td>\n",
       "      <td>738812</td>\n",
       "      <td>-</td>\n",
       "      <td>AL669831.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr1</td>\n",
       "      <td>739120</td>\n",
       "      <td>739137</td>\n",
       "      <td>-</td>\n",
       "      <td>AL669831.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr1</td>\n",
       "      <td>818042</td>\n",
       "      <td>818058</td>\n",
       "      <td>+</td>\n",
       "      <td>AL645608.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr1</td>\n",
       "      <td>819495</td>\n",
       "      <td>819513</td>\n",
       "      <td>+</td>\n",
       "      <td>AL645608.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr1</td>\n",
       "      <td>819960</td>\n",
       "      <td>819983</td>\n",
       "      <td>+</td>\n",
       "      <td>AL645608.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr1</td>\n",
       "      <td>861117</td>\n",
       "      <td>861180</td>\n",
       "      <td>+</td>\n",
       "      <td>SAMD11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr1</td>\n",
       "      <td>861301</td>\n",
       "      <td>861393</td>\n",
       "      <td>+</td>\n",
       "      <td>SAMD11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr1</td>\n",
       "      <td>865534</td>\n",
       "      <td>865716</td>\n",
       "      <td>+</td>\n",
       "      <td>SAMD11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>chr1</td>\n",
       "      <td>866418</td>\n",
       "      <td>866469</td>\n",
       "      <td>+</td>\n",
       "      <td>SAMD11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>chr1</td>\n",
       "      <td>871151</td>\n",
       "      <td>871276</td>\n",
       "      <td>+</td>\n",
       "      <td>SAMD11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>chr1</td>\n",
       "      <td>874419</td>\n",
       "      <td>874509</td>\n",
       "      <td>+</td>\n",
       "      <td>SAMD11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>chr1</td>\n",
       "      <td>874654</td>\n",
       "      <td>874840</td>\n",
       "      <td>+</td>\n",
       "      <td>SAMD11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>chr1</td>\n",
       "      <td>876523</td>\n",
       "      <td>876686</td>\n",
       "      <td>+</td>\n",
       "      <td>SAMD11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>chr1</td>\n",
       "      <td>877515</td>\n",
       "      <td>877631</td>\n",
       "      <td>+</td>\n",
       "      <td>SAMD11</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   chrom  exon_start exon_stop strand        gene\n",
       "0   chr1       69090     70008      +       OR4F5\n",
       "1   chr1      134900    135802      -  AL627309.1\n",
       "2   chr1      137620    139379      -  AL627309.1\n",
       "3   chr1      367639    368634      +      OR4F29\n",
       "4   chr1      621095    622034      -      OR4F16\n",
       "5   chr1      738531    738618      -  AL669831.1\n",
       "6   chr1      738787    738812      -  AL669831.1\n",
       "7   chr1      739120    739137      -  AL669831.1\n",
       "8   chr1      818042    818058      +  AL645608.2\n",
       "9   chr1      819495    819513      +  AL645608.2\n",
       "10  chr1      819960    819983      +  AL645608.2\n",
       "11  chr1      861117    861180      +      SAMD11\n",
       "12  chr1      861301    861393      +      SAMD11\n",
       "13  chr1      865534    865716      +      SAMD11\n",
       "14  chr1      866418    866469      +      SAMD11\n",
       "15  chr1      871151    871276      +      SAMD11\n",
       "16  chr1      874419    874509      +      SAMD11\n",
       "17  chr1      874654    874840      +      SAMD11\n",
       "18  chr1      876523    876686      +      SAMD11\n",
       "19  chr1      877515    877631      +      SAMD11"
      ]
     },
     "execution_count": 325,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hg19_exons.head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 326,
   "id": "6e81a309",
   "metadata": {},
   "outputs": [],
   "source": [
    "hg19_exons.to_csv('../annotations/hg19_exons.bed', sep=\"\\t\", header = False, index = False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5b1736f0",
   "metadata": {},
   "source": [
    "Now change the data type for the starts and stops and create a length column."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 327,
   "id": "b795d776",
   "metadata": {},
   "outputs": [],
   "source": [
    "exon_length['exon_stop'] = exon_length['exon_stop'].astype(int)\n",
    "exon_length['exon_start'] = exon_length['exon_start'].astype(int)\n",
    "exon_length['exon_length'] = exon_length['exon_stop'] - exon_length['exon_start']"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "97921cf1",
   "metadata": {},
   "source": [
    "Check to see that it worked."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 328,
   "id": "a938b6c2",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>exon_start</th>\n",
       "      <th>exon_stop</th>\n",
       "      <th>exon_length</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>69090</td>\n",
       "      <td>70008</td>\n",
       "      <td>918</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AL627309.1</td>\n",
       "      <td>134900</td>\n",
       "      <td>135802</td>\n",
       "      <td>902</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AL627309.1</td>\n",
       "      <td>137620</td>\n",
       "      <td>139379</td>\n",
       "      <td>1759</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>OR4F29</td>\n",
       "      <td>367639</td>\n",
       "      <td>368634</td>\n",
       "      <td>995</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>OR4F16</td>\n",
       "      <td>621095</td>\n",
       "      <td>622034</td>\n",
       "      <td>939</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>AL669831.1</td>\n",
       "      <td>738531</td>\n",
       "      <td>738618</td>\n",
       "      <td>87</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>AL669831.1</td>\n",
       "      <td>738787</td>\n",
       "      <td>738812</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>AL669831.1</td>\n",
       "      <td>739120</td>\n",
       "      <td>739137</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>AL645608.2</td>\n",
       "      <td>818042</td>\n",
       "      <td>818058</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>AL645608.2</td>\n",
       "      <td>819495</td>\n",
       "      <td>819513</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         name  exon_start  exon_stop  exon_length\n",
       "0       OR4F5       69090      70008          918\n",
       "1  AL627309.1      134900     135802          902\n",
       "2  AL627309.1      137620     139379         1759\n",
       "3      OR4F29      367639     368634          995\n",
       "4      OR4F16      621095     622034          939\n",
       "5  AL669831.1      738531     738618           87\n",
       "6  AL669831.1      738787     738812           25\n",
       "7  AL669831.1      739120     739137           17\n",
       "8  AL645608.2      818042     818058           16\n",
       "9  AL645608.2      819495     819513           18"
      ]
     },
     "execution_count": 328,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "exon_length.head(10)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ff463442",
   "metadata": {},
   "source": [
    "Now let's group by the gene to get the total exon length."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 329,
   "id": "e7630c82",
   "metadata": {},
   "outputs": [],
   "source": [
    "exon_length = exon_length.groupby('name')['exon_length'].sum().to_frame('exon_length')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6169a2dc",
   "metadata": {},
   "source": [
    "Create a new dataframe."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 330,
   "id": "3d872619",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>n_exons</th>\n",
       "      <th>gene_length</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>1</td>\n",
       "      <td>918</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AL627309.1</td>\n",
       "      <td>2</td>\n",
       "      <td>4479</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>OR4F29</td>\n",
       "      <td>1</td>\n",
       "      <td>995</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>OR4F16</td>\n",
       "      <td>1</td>\n",
       "      <td>939</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AL669831.1</td>\n",
       "      <td>3</td>\n",
       "      <td>606</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>AL645608.2</td>\n",
       "      <td>3</td>\n",
       "      <td>1941</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>14</td>\n",
       "      <td>18838</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>AL645608.1</td>\n",
       "      <td>6</td>\n",
       "      <td>5182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>NOC2L</td>\n",
       "      <td>19</td>\n",
       "      <td>15087</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>KLHL17</td>\n",
       "      <td>12</td>\n",
       "      <td>5129</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         name  n_exons  gene_length\n",
       "0       OR4F5        1          918\n",
       "1  AL627309.1        2         4479\n",
       "2      OR4F29        1          995\n",
       "3      OR4F16        1          939\n",
       "4  AL669831.1        3          606\n",
       "5  AL645608.2        3         1941\n",
       "6      SAMD11       14        18838\n",
       "7  AL645608.1        6         5182\n",
       "8       NOC2L       19        15087\n",
       "9      KLHL17       12         5129"
      ]
     },
     "execution_count": 330,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "n_exons_length = annotations[['name','n_exons','gene_length']]\n",
    "n_exons_length.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 331,
   "id": "e461430e",
   "metadata": {},
   "outputs": [],
   "source": [
    "n_exons_length = pd.merge(n_exons_length, exon_length['exon_length'], on='name')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c8feae2a",
   "metadata": {},
   "source": [
    "Let's take a peek at the distributions of these variables."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 332,
   "id": "18190913",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>n_exons</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1597</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1525</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1482</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1604</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>1517</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>1389</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>1152</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>1088</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>1024</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>858</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>829</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>723</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>602</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>536</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>490</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>423</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>348</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>327</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>277</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>275</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>238</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>217</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>216</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>137</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>115</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>107</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>110</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>81</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>86</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>66</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>52</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>34</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>35</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58</th>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66</th>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67</th>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>68</th>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69</th>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71</th>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74</th>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>77</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78</th>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79</th>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>81</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83</th>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84</th>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86</th>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>90</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>93</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>98</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>104</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>105</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>106</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>107</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>108</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>111</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>116</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>118</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>183</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>363</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         count\n",
       "n_exons       \n",
       "1         1597\n",
       "2         1525\n",
       "3         1482\n",
       "4         1604\n",
       "5         1517\n",
       "6         1389\n",
       "7         1152\n",
       "8         1088\n",
       "9         1024\n",
       "10         858\n",
       "11         829\n",
       "12         723\n",
       "13         602\n",
       "14         536\n",
       "15         490\n",
       "16         423\n",
       "17         348\n",
       "18         327\n",
       "19         277\n",
       "20         275\n",
       "21         238\n",
       "22         217\n",
       "23         216\n",
       "24         157\n",
       "25         137\n",
       "26         115\n",
       "27         107\n",
       "28         110\n",
       "29          81\n",
       "30          86\n",
       "31          66\n",
       "32          52\n",
       "33          53\n",
       "34          37\n",
       "35          37\n",
       "36          38\n",
       "37          34\n",
       "38          35\n",
       "39          36\n",
       "40          32\n",
       "41          31\n",
       "42          17\n",
       "43          21\n",
       "44          15\n",
       "45          10\n",
       "46          10\n",
       "47          17\n",
       "48          19\n",
       "49          13\n",
       "50           5\n",
       "51           8\n",
       "52          11\n",
       "53           9\n",
       "54           6\n",
       "55           5\n",
       "56           6\n",
       "57           7\n",
       "58           8\n",
       "59           4\n",
       "60           1\n",
       "61           4\n",
       "62           4\n",
       "63           4\n",
       "64           4\n",
       "65           4\n",
       "66           4\n",
       "67           6\n",
       "68           5\n",
       "69           3\n",
       "70           2\n",
       "71           8\n",
       "72           2\n",
       "74           3\n",
       "75           2\n",
       "76           1\n",
       "77           1\n",
       "78           5\n",
       "79           3\n",
       "80           2\n",
       "81           1\n",
       "82           2\n",
       "83           3\n",
       "84           3\n",
       "86           3\n",
       "89           1\n",
       "90           2\n",
       "91           1\n",
       "93           2\n",
       "97           2\n",
       "98           1\n",
       "102          1\n",
       "104          1\n",
       "105          1\n",
       "106          2\n",
       "107          1\n",
       "108          1\n",
       "111          1\n",
       "116          2\n",
       "118          1\n",
       "183          1\n",
       "363          1"
      ]
     },
     "execution_count": 332,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "n_exons_hist = n_exons_length.groupby('n_exons').size().to_frame('count')\n",
    "n_exons_hist"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 333,
   "id": "75f9eb3a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene_length</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1987243</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2056874</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2057828</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2092292</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2304640</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>17237 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             count\n",
       "gene_length       \n",
       "9                2\n",
       "11               1\n",
       "13               1\n",
       "16               3\n",
       "17               6\n",
       "...            ...\n",
       "1987243          1\n",
       "2056874          1\n",
       "2057828          1\n",
       "2092292          1\n",
       "2304640          1\n",
       "\n",
       "[17237 rows x 1 columns]"
      ]
     },
     "execution_count": 333,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_length_hist = n_exons_length.groupby('gene_length').size().to_frame('count')\n",
    "gene_length_hist"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 334,
   "id": "febd23fb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>exon_length</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30355</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33679</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34526</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43816</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>109224</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>7186 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             count\n",
       "exon_length       \n",
       "9                2\n",
       "11               1\n",
       "13               1\n",
       "16               3\n",
       "17               6\n",
       "...            ...\n",
       "30355            1\n",
       "33679            1\n",
       "34526            1\n",
       "43816            1\n",
       "109224           1\n",
       "\n",
       "[7186 rows x 1 columns]"
      ]
     },
     "execution_count": 334,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "exon_length_hist = n_exons_length.groupby('exon_length').size().to_frame('count')\n",
    "exon_length_hist"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a170cb4f",
   "metadata": {},
   "source": [
    "Now to actually test their relationship to variant count at our two thresholds."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 335,
   "id": "42ae4ff5",
   "metadata": {},
   "outputs": [],
   "source": [
    "n_variants = data.rename(columns={'annotation':'name'})\n",
    "n_variants = n_variants.groupby('name').size().to_frame('n_variants')\n",
    "variant_characterstics = pd.merge(n_exons_length, n_variants, on='name', how='left')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 336,
   "id": "946c393a",
   "metadata": {},
   "outputs": [],
   "source": [
    "n_splice_variants_2 = data_2.rename(columns={'annotation':'name'})\n",
    "n_splice_variants_2 = n_splice_variants_2.groupby('name').size().to_frame('variant_count_2')\n",
    "variant_characterstics = pd.merge(variant_characterstics, n_splice_variants_2, on='name', how='left')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 337,
   "id": "664e98a2",
   "metadata": {},
   "outputs": [],
   "source": [
    "n_splice_variants_5 = data_5.rename(columns={'annotation':'name'})\n",
    "n_splice_variants_5 = n_splice_variants_5.groupby('name').size().to_frame('variant_count_5')\n",
    "variant_characterstics = pd.merge(variant_characterstics, n_splice_variants_5, on='name', how='left')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 338,
   "id": "f8742c11",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>n_exons</th>\n",
       "      <th>gene_length</th>\n",
       "      <th>exon_length</th>\n",
       "      <th>n_variants</th>\n",
       "      <th>variant_count_2</th>\n",
       "      <th>variant_count_5</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>1</td>\n",
       "      <td>918</td>\n",
       "      <td>918</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AL627309.1</td>\n",
       "      <td>2</td>\n",
       "      <td>4479</td>\n",
       "      <td>2661</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>OR4F29</td>\n",
       "      <td>1</td>\n",
       "      <td>995</td>\n",
       "      <td>995</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>OR4F16</td>\n",
       "      <td>1</td>\n",
       "      <td>939</td>\n",
       "      <td>939</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AL669831.1</td>\n",
       "      <td>3</td>\n",
       "      <td>606</td>\n",
       "      <td>129</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>AL645608.2</td>\n",
       "      <td>3</td>\n",
       "      <td>1941</td>\n",
       "      <td>57</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>14</td>\n",
       "      <td>18838</td>\n",
       "      <td>2551</td>\n",
       "      <td>53.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>AL645608.1</td>\n",
       "      <td>6</td>\n",
       "      <td>5182</td>\n",
       "      <td>336</td>\n",
       "      <td>17.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>NOC2L</td>\n",
       "      <td>19</td>\n",
       "      <td>15087</td>\n",
       "      <td>2790</td>\n",
       "      <td>37.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>KLHL17</td>\n",
       "      <td>12</td>\n",
       "      <td>5129</td>\n",
       "      <td>2560</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>16</td>\n",
       "      <td>8612</td>\n",
       "      <td>2404</td>\n",
       "      <td>21.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>PERM1</td>\n",
       "      <td>3</td>\n",
       "      <td>5825</td>\n",
       "      <td>3340</td>\n",
       "      <td>10.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>HES4</td>\n",
       "      <td>4</td>\n",
       "      <td>1148</td>\n",
       "      <td>899</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>ISG15</td>\n",
       "      <td>2</td>\n",
       "      <td>1118</td>\n",
       "      <td>711</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>AGRN</td>\n",
       "      <td>36</td>\n",
       "      <td>35994</td>\n",
       "      <td>7323</td>\n",
       "      <td>85.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>RNF223</td>\n",
       "      <td>2</td>\n",
       "      <td>3342</td>\n",
       "      <td>1902</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>C1orf159</td>\n",
       "      <td>10</td>\n",
       "      <td>34276</td>\n",
       "      <td>1841</td>\n",
       "      <td>52.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>TTLL10</td>\n",
       "      <td>16</td>\n",
       "      <td>24033</td>\n",
       "      <td>2283</td>\n",
       "      <td>66.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>TNFRSF18</td>\n",
       "      <td>4</td>\n",
       "      <td>3064</td>\n",
       "      <td>851</td>\n",
       "      <td>9.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>TNFRSF4</td>\n",
       "      <td>7</td>\n",
       "      <td>2807</td>\n",
       "      <td>1068</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          name  n_exons  gene_length  exon_length  n_variants  \\\n",
       "0        OR4F5        1          918          918         0.0   \n",
       "1   AL627309.1        2         4479         2661         0.0   \n",
       "2       OR4F29        1          995          995         0.0   \n",
       "3       OR4F16        1          939          939         0.0   \n",
       "4   AL669831.1        3          606          129         0.0   \n",
       "5   AL645608.2        3         1941           57         0.0   \n",
       "6       SAMD11       14        18838         2551        53.0   \n",
       "7   AL645608.1        6         5182          336        17.0   \n",
       "8        NOC2L       19        15087         2790        37.0   \n",
       "9       KLHL17       12         5129         2560         7.0   \n",
       "10     PLEKHN1       16         8612         2404        21.0   \n",
       "11       PERM1        3         5825         3340        10.0   \n",
       "12        HES4        4         1148          899         2.0   \n",
       "13       ISG15        2         1118          711         7.0   \n",
       "14        AGRN       36        35994         7323        85.0   \n",
       "15      RNF223        2         3342         1902         3.0   \n",
       "16    C1orf159       10        34276         1841        52.0   \n",
       "17      TTLL10       16        24033         2283        66.0   \n",
       "18    TNFRSF18        4         3064          851         9.0   \n",
       "19     TNFRSF4        7         2807         1068         3.0   \n",
       "\n",
       "    variant_count_2  variant_count_5  \n",
       "0               0.0              0.0  \n",
       "1               0.0              0.0  \n",
       "2               0.0              0.0  \n",
       "3               0.0              0.0  \n",
       "4               0.0              0.0  \n",
       "5               0.0              0.0  \n",
       "6               2.0              0.0  \n",
       "7               0.0              0.0  \n",
       "8               0.0              0.0  \n",
       "9               0.0              0.0  \n",
       "10              1.0              0.0  \n",
       "11              0.0              0.0  \n",
       "12              0.0              0.0  \n",
       "13              0.0              0.0  \n",
       "14              4.0              0.0  \n",
       "15              0.0              0.0  \n",
       "16              0.0              0.0  \n",
       "17              1.0              0.0  \n",
       "18              0.0              0.0  \n",
       "19              0.0              0.0  "
      ]
     },
     "execution_count": 338,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "variant_characterstics = variant_characterstics.fillna(0)\n",
    "variant_characterstics.head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 339,
   "id": "1895dd5e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "20274"
      ]
     },
     "execution_count": 339,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(variant_characterstics)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5ffb3a1d",
   "metadata": {},
   "source": [
    "Save this dataframe for plotting."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 340,
   "id": "58d61c17",
   "metadata": {},
   "outputs": [],
   "source": [
    "variant_characterstics.to_csv('../dataframes/variant_characteristics.txt', sep=\"\\t\", header = True, index = False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "40082eb2",
   "metadata": {},
   "source": [
    "Subset to genes with at least one variant."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 341,
   "id": "d716b803",
   "metadata": {},
   "outputs": [],
   "source": [
    "variant_characterstics = variant_characterstics[variant_characterstics['n_variants'] > 0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 342,
   "id": "193858c2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "17631"
      ]
     },
     "execution_count": 342,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(variant_characterstics)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0706369c",
   "metadata": {},
   "source": [
    "Let's run some correlations on the characteristics."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 343,
   "id": "3dbd70f1",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.3160848388955107 0.0\n"
     ]
    }
   ],
   "source": [
    "rho, p = spearmanr(variant_characterstics['n_exons'], variant_characterstics['variant_count_2'])\n",
    "print(rho,p)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 344,
   "id": "a8792110",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.1851170800689867 1.0667367197057646e-135\n"
     ]
    }
   ],
   "source": [
    "rho, p = spearmanr(variant_characterstics['exon_length'], variant_characterstics['variant_count_2'])\n",
    "print(rho,p)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 345,
   "id": "bb7bf1b2",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.2919566369123417 0.0\n"
     ]
    }
   ],
   "source": [
    "rho, p = spearmanr(variant_characterstics['gene_length'], variant_characterstics['variant_count_2'])\n",
    "print(rho,p)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 346,
   "id": "ba3b8e80",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.13299027146082146 2.2009095588908162e-70\n"
     ]
    }
   ],
   "source": [
    "rho, p = spearmanr(variant_characterstics['n_exons'], variant_characterstics['variant_count_5'])\n",
    "print(rho,p)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 347,
   "id": "55f36acc",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.0624799065089255 1.009030502488749e-16\n"
     ]
    }
   ],
   "source": [
    "rho, p = spearmanr(variant_characterstics['exon_length'], variant_characterstics['variant_count_5'])\n",
    "print(rho,p)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 348,
   "id": "e14e1776",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.1017201812950686 8.985235346295254e-42\n"
     ]
    }
   ],
   "source": [
    "rho, p = spearmanr(variant_characterstics['gene_length'], variant_characterstics['variant_count_5'])\n",
    "print(rho,p)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a55bda51",
   "metadata": {},
   "source": [
    "## N Isoforms <a class = 'anchor' id = 'isoforms'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c8cb71a9",
   "metadata": {},
   "source": [
    "Does the number of the known isoforms associate with N SAVs?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 349,
   "id": "594de846",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>isoform_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MIR1302-2HG</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>FAM138A</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000238009</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000239945</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>ENSG00000239906</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>ENSG00000241860</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>ENSG00000241599</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>ENSG00000286448</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>ENSG00000236601</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              name  isoform_count\n",
       "0      MIR1302-2HG              2\n",
       "1          FAM138A              2\n",
       "2            OR4F5              1\n",
       "3  ENSG00000238009              5\n",
       "4  ENSG00000239945              1\n",
       "5  ENSG00000239906              1\n",
       "6  ENSG00000241860              6\n",
       "7  ENSG00000241599              1\n",
       "8  ENSG00000286448              1\n",
       "9  ENSG00000236601              3"
      ]
     },
     "execution_count": 349,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "isoforms_header = ['name','isoform_count']\n",
    "isoforms = pd.read_csv('../annotations/GENCODE_Release_40_hg38_N_isoforms.txt', sep = '\\t', skiprows = 1, names = isoforms_header)\n",
    "isoforms.head(10)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "82d53ac1",
   "metadata": {},
   "source": [
    "Merge with above dataframe. Let's use a new dataframe."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 350,
   "id": "79f28155",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>n_exons</th>\n",
       "      <th>gene_length</th>\n",
       "      <th>exon_length</th>\n",
       "      <th>n_variants</th>\n",
       "      <th>variant_count_2</th>\n",
       "      <th>variant_count_5</th>\n",
       "      <th>isoform_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>14</td>\n",
       "      <td>18838</td>\n",
       "      <td>2551</td>\n",
       "      <td>53.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NOC2L</td>\n",
       "      <td>19</td>\n",
       "      <td>15087</td>\n",
       "      <td>2790</td>\n",
       "      <td>37.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>KLHL17</td>\n",
       "      <td>12</td>\n",
       "      <td>5129</td>\n",
       "      <td>2560</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>16</td>\n",
       "      <td>8612</td>\n",
       "      <td>2404</td>\n",
       "      <td>21.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PERM1</td>\n",
       "      <td>3</td>\n",
       "      <td>5825</td>\n",
       "      <td>3340</td>\n",
       "      <td>10.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>HES4</td>\n",
       "      <td>4</td>\n",
       "      <td>1148</td>\n",
       "      <td>899</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>ISG15</td>\n",
       "      <td>2</td>\n",
       "      <td>1118</td>\n",
       "      <td>711</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>AGRN</td>\n",
       "      <td>36</td>\n",
       "      <td>35994</td>\n",
       "      <td>7323</td>\n",
       "      <td>85.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>RNF223</td>\n",
       "      <td>2</td>\n",
       "      <td>3342</td>\n",
       "      <td>1902</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>C1orf159</td>\n",
       "      <td>10</td>\n",
       "      <td>34276</td>\n",
       "      <td>1841</td>\n",
       "      <td>52.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       name  n_exons  gene_length  exon_length  n_variants  variant_count_2  \\\n",
       "0    SAMD11       14        18838         2551        53.0              2.0   \n",
       "1     NOC2L       19        15087         2790        37.0              0.0   \n",
       "2    KLHL17       12         5129         2560         7.0              0.0   \n",
       "3   PLEKHN1       16         8612         2404        21.0              1.0   \n",
       "4     PERM1        3         5825         3340        10.0              0.0   \n",
       "5      HES4        4         1148          899         2.0              0.0   \n",
       "6     ISG15        2         1118          711         7.0              0.0   \n",
       "7      AGRN       36        35994         7323        85.0              4.0   \n",
       "8    RNF223        2         3342         1902         3.0              0.0   \n",
       "9  C1orf159       10        34276         1841        52.0              0.0   \n",
       "\n",
       "   variant_count_5  isoform_count  \n",
       "0              0.0             15  \n",
       "1              0.0              6  \n",
       "2              0.0              4  \n",
       "3              0.0              5  \n",
       "4              0.0              4  \n",
       "5              0.0              4  \n",
       "6              0.0              3  \n",
       "7              0.0             10  \n",
       "8              0.0              1  \n",
       "9              0.0             19  "
      ]
     },
     "execution_count": 350,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "n_isoforms = pd.merge(variant_characterstics, isoforms, on = ['name'])\n",
    "n_isoforms.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 351,
   "id": "e3340024",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "16151"
      ]
     },
     "execution_count": 351,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(n_isoforms)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 352,
   "id": "e4226207",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.1767337559003496 1.8670449850386394e-113\n"
     ]
    }
   ],
   "source": [
    "rho, p = spearmanr(n_isoforms['isoform_count'], n_isoforms['variant_count_2'])\n",
    "print(rho,p)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 353,
   "id": "9063c8a8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.07692978918607588 1.2368500958019458e-22\n"
     ]
    }
   ],
   "source": [
    "rho, p = spearmanr(n_isoforms['isoform_count'], n_isoforms['variant_count_5'])\n",
    "print(rho,p)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 354,
   "id": "a9bfc1b9",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.5639685614316065 0.0\n"
     ]
    }
   ],
   "source": [
    "rho, p = spearmanr(n_isoforms['n_exons'], n_isoforms['isoform_count'])\n",
    "print(rho,p)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 355,
   "id": "78d64488",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.09741069465365773 2.3604050238515602e-35\n"
     ]
    }
   ],
   "source": [
    "rho, p = spearmanr(n_isoforms['gene_length'], n_isoforms['variant_count_5'])\n",
    "print(rho,p)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ff269fdb",
   "metadata": {},
   "source": [
    "Run a partial correlation to control for the number of exons, which seems to be the strongest covariate."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 356,
   "id": "cf1ef8f0",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>n</th>\n",
       "      <th>r</th>\n",
       "      <th>CI95%</th>\n",
       "      <th>p-val</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>spearman</th>\n",
       "      <td>16151</td>\n",
       "      <td>-0.003008</td>\n",
       "      <td>[-0.02, 0.01]</td>\n",
       "      <td>0.702242</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              n         r          CI95%     p-val\n",
       "spearman  16151 -0.003008  [-0.02, 0.01]  0.702242"
      ]
     },
     "execution_count": 356,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "partial_corr(data = n_isoforms, x = 'isoform_count', y = 'variant_count_2', covar=['n_exons'], method = 'spearman')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 357,
   "id": "bcb0ef3a",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>n</th>\n",
       "      <th>r</th>\n",
       "      <th>CI95%</th>\n",
       "      <th>p-val</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>spearman</th>\n",
       "      <td>16151</td>\n",
       "      <td>0.000817</td>\n",
       "      <td>[-0.01, 0.02]</td>\n",
       "      <td>0.917351</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              n         r          CI95%     p-val\n",
       "spearman  16151  0.000817  [-0.01, 0.02]  0.917351"
      ]
     },
     "execution_count": 357,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "partial_corr(data = n_isoforms, x = 'isoform_count', y = 'variant_count_5', covar=['n_exons'], method = 'spearman')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "884991d8",
   "metadata": {},
   "source": [
    "## Constraint and Conservation <a class = 'anchor' id = 'constraintconservation'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9d7f3376",
   "metadata": {},
   "source": [
    "Let's check for differences in constraint and conservation starting with missense observed over expected. Start with Vernot."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 358,
   "id": "597d8f26",
   "metadata": {},
   "outputs": [],
   "source": [
    "Vernot_ancient_mis_oe = data_2.loc[data_2['Vernot_allele_origin'] == 'ancient', ['annotation', 'mis_oe']].drop_duplicates().dropna()\n",
    "Vernot_archaic_specific_mis_oe = data_2.loc[data_2['Vernot_allele_origin'] == 'archaic-specific', ['annotation', 'mis_oe']].drop_duplicates().dropna()\n",
    "Vernot_introgressed_mis_oe = data_2.loc[data_2['Vernot_allele_origin'] == 'introgressed', ['annotation', 'mis_oe']].drop_duplicates().dropna()\n",
    "non_SA_mis_oe = data.loc[~data['annotation'].isin(data_2['annotation']), ['annotation', 'mis_oe']].drop_duplicates().dropna()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 359,
   "id": "27b35560",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1817"
      ]
     },
     "execution_count": 359,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(Vernot_ancient_mis_oe)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 360,
   "id": "a324cbf3",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1832"
      ]
     },
     "execution_count": 360,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(Vernot_archaic_specific_mis_oe)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 361,
   "id": "d8d66d70",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "234"
      ]
     },
     "execution_count": 361,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(Vernot_introgressed_mis_oe)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 362,
   "id": "67765332",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "12708"
      ]
     },
     "execution_count": 362,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(non_SA_mis_oe)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 363,
   "id": "808c6c19",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "KruskalResult(statistic=18.078829007426336, pvalue=0.0004236866085398531)"
      ]
     },
     "execution_count": 363,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "kruskal(Vernot_ancient_mis_oe['mis_oe'], Vernot_archaic_specific_mis_oe['mis_oe'], Vernot_introgressed_mis_oe['mis_oe'], non_SA_mis_oe['mis_oe'])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "53fad998",
   "metadata": {},
   "source": [
    "Now for missense z-score."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 364,
   "id": "cc362016",
   "metadata": {},
   "outputs": [],
   "source": [
    "Vernot_ancient_mis_z = data_2.loc[data_2['Vernot_allele_origin'] == 'ancient', ['annotation', 'mis_z']].drop_duplicates().dropna()\n",
    "Vernot_archaic_specific_mis_z = data_2.loc[data_2['Vernot_allele_origin'] == 'archaic-specific', ['annotation', 'mis_z']].drop_duplicates().dropna()\n",
    "Vernot_introgressed_mis_z = data_2.loc[data_2['Vernot_allele_origin'] == 'introgressed', ['annotation', 'mis_z']].drop_duplicates().dropna()\n",
    "non_SA_mis_z = data.loc[~data['annotation'].isin(data_2['annotation']), ['annotation', 'mis_z']].drop_duplicates().dropna()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 365,
   "id": "4d906787",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1817"
      ]
     },
     "execution_count": 365,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(Vernot_ancient_mis_z)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 366,
   "id": "b1280050",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1832"
      ]
     },
     "execution_count": 366,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(Vernot_archaic_specific_mis_z)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 367,
   "id": "3d4ac120",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "234"
      ]
     },
     "execution_count": 367,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(Vernot_introgressed_mis_z)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 368,
   "id": "1637b1d4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "12708"
      ]
     },
     "execution_count": 368,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(non_SA_mis_z)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 369,
   "id": "3c40c973",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "KruskalResult(statistic=8.620709037824106, pvalue=0.0347828750312021)"
      ]
     },
     "execution_count": 369,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "kruskal(Vernot_ancient_mis_z['mis_z'], Vernot_archaic_specific_mis_z['mis_z'], Vernot_introgressed_mis_z['mis_z'], non_SA_mis_z['mis_z'])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "61910986",
   "metadata": {},
   "source": [
    "Now for LoF observed/expected."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 370,
   "id": "47a62dad",
   "metadata": {},
   "outputs": [],
   "source": [
    "Vernot_ancient_lof_oe = data_2.loc[data_2['Vernot_allele_origin'] == 'ancient', ['annotation', 'lof_oe']].drop_duplicates().dropna()\n",
    "Vernot_archaic_specific_lof_oe = data_2.loc[data_2['Vernot_allele_origin'] == 'archaic-specific', ['annotation', 'lof_oe']].drop_duplicates().dropna()\n",
    "Vernot_introgressed_lof_oe = data_2.loc[data_2['Vernot_allele_origin'] == 'introgressed', ['annotation', 'lof_oe']].drop_duplicates().dropna()\n",
    "non_SA_lof_oe = data.loc[~data['annotation'].isin(data_2['annotation']), ['annotation', 'lof_oe']].drop_duplicates().dropna()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 371,
   "id": "17a9d168",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1811"
      ]
     },
     "execution_count": 371,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(Vernot_ancient_lof_oe)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 372,
   "id": "ff115bc4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1824"
      ]
     },
     "execution_count": 372,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(Vernot_archaic_specific_lof_oe)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 373,
   "id": "96b5c764",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "234"
      ]
     },
     "execution_count": 373,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(Vernot_introgressed_lof_oe)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 374,
   "id": "1f9b3d1f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "12484"
      ]
     },
     "execution_count": 374,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(non_SA_lof_oe)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 375,
   "id": "0ba32930",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "KruskalResult(statistic=1.5332226862446645, pvalue=0.6746240497778113)"
      ]
     },
     "execution_count": 375,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "kruskal(Vernot_ancient_lof_oe['lof_oe'], Vernot_archaic_specific_lof_oe['lof_oe'], Vernot_introgressed_lof_oe['lof_oe'], non_SA_lof_oe['lof_oe'])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e5b35452",
   "metadata": {},
   "source": [
    "Now for LoF z-scores."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 376,
   "id": "b2245b1d",
   "metadata": {},
   "outputs": [],
   "source": [
    "Vernot_ancient_lof_z = data_2.loc[data_2['Vernot_allele_origin'] == 'ancient', ['annotation', 'lof_z']].drop_duplicates().dropna()\n",
    "Vernot_archaic_specific_lof_z = data_2.loc[data_2['Vernot_allele_origin'] == 'archaic-specific', ['annotation', 'lof_z']].drop_duplicates().dropna()\n",
    "Vernot_introgressed_lof_z = data_2.loc[data_2['Vernot_allele_origin'] == 'introgressed', ['annotation', 'lof_z']].drop_duplicates().dropna()\n",
    "non_SA_lof_z = data.loc[~data['annotation'].isin(data_2['annotation']), ['annotation', 'lof_z']].drop_duplicates().dropna()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 377,
   "id": "5519851d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1811"
      ]
     },
     "execution_count": 377,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(Vernot_ancient_lof_z)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 378,
   "id": "b345ff78",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1824"
      ]
     },
     "execution_count": 378,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(Vernot_archaic_specific_lof_z)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 379,
   "id": "52cb3a3a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "234"
      ]
     },
     "execution_count": 379,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(Vernot_introgressed_lof_z)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 380,
   "id": "870d7d3e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "12484"
      ]
     },
     "execution_count": 380,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(non_SA_lof_z)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 381,
   "id": "7eae7842",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "KruskalResult(statistic=323.3105002953148, pvalue=8.955791061419506e-70)"
      ]
     },
     "execution_count": 381,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "kruskal(Vernot_ancient_lof_z['lof_z'], Vernot_archaic_specific_lof_z['lof_z'], Vernot_introgressed_lof_z['lof_z'], non_SA_lof_z['lof_z'])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e2bf0f06",
   "metadata": {},
   "source": [
    "And phyloP."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 382,
   "id": "39414c34",
   "metadata": {},
   "outputs": [],
   "source": [
    "Vernot_ancient_phyloP = data_2.loc[data_2['Vernot_allele_origin'] == 'ancient', 'phyloP'].dropna()\n",
    "Vernot_archaic_specific_phyloP = data_2.loc[data_2['Vernot_allele_origin'] == 'archaic-specific', 'phyloP'].dropna()\n",
    "Vernot_introgressed_phyloP = data_2.loc[data_2['Vernot_allele_origin'] == 'introgressed', 'phyloP'].dropna()\n",
    "non_SA_phyloP = data.loc[data['delta_max'] < 0.2, 'phyloP'].dropna()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 383,
   "id": "7042cc26",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2251"
      ]
     },
     "execution_count": 383,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(Vernot_ancient_phyloP)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 384,
   "id": "66431a3f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2186"
      ]
     },
     "execution_count": 384,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(Vernot_archaic_specific_phyloP)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 385,
   "id": "5e8d1ea0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "244"
      ]
     },
     "execution_count": 385,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(Vernot_introgressed_phyloP)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 386,
   "id": "c0bae5e6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1600192"
      ]
     },
     "execution_count": 386,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(non_SA_phyloP)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 387,
   "id": "9f35e9be",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "KruskalResult(statistic=877.4289463274906, pvalue=6.962502007197678e-190)"
      ]
     },
     "execution_count": 387,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "kruskal(Vernot_ancient_phyloP, Vernot_archaic_specific_phyloP, Vernot_introgressed_phyloP, non_SA_phyloP)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fec2f23f",
   "metadata": {},
   "source": [
    "How many of these are not \"neutrally evolving\"?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 388,
   "id": "985482a9",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.15650646499754187"
      ]
     },
     "execution_count": 388,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "((Vernot_ancient_phyloP < -1.3).sum() + (Vernot_ancient_phyloP > 1.3).sum() + (Vernot_archaic_specific_phyloP < -1.3).sum() + (Vernot_archaic_specific_phyloP > 1.3).sum() + (Vernot_introgressed_phyloP < -1.3).sum() + (Vernot_introgressed_phyloP > 1.3).sum() + (non_SA_phyloP < -1.3).sum() + (non_SA_phyloP > 1.3).sum()) / (len(Vernot_ancient_phyloP) + len(Vernot_archaic_specific_phyloP) + len(Vernot_introgressed_phyloP) + len(non_SA_phyloP))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "46dfa837",
   "metadata": {},
   "source": [
    "And \"neutrally evolving\"?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 389,
   "id": "5e83db08",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.8434935350024582"
      ]
     },
     "execution_count": 389,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "1-0.15650646499754187"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f10dfc8b",
   "metadata": {},
   "source": [
    "Repeat for Browning."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 390,
   "id": "ac2d45c6",
   "metadata": {},
   "outputs": [],
   "source": [
    "Browning_ancient_mis_oe = data_2.loc[data_2['Browning_allele_origin'] == 'ancient', ['annotation', 'mis_oe']].drop_duplicates().dropna()\n",
    "Browning_archaic_specific_mis_oe = data_2.loc[data_2['Browning_allele_origin'] == 'archaic-specific', ['annotation', 'mis_oe']].drop_duplicates().dropna()\n",
    "Browning_introgressed_mis_oe = data_2.loc[data_2['Browning_allele_origin'] == 'introgressed', ['annotation', 'mis_oe']].drop_duplicates().dropna()\n",
    "non_SA_mis_oe = data.loc[~data['annotation'].isin(data_2['annotation']), ['annotation', 'mis_oe']].drop_duplicates().dropna()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 391,
   "id": "8011d70b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1780"
      ]
     },
     "execution_count": 391,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(Browning_ancient_mis_oe)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 392,
   "id": "e8048d11",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1832"
      ]
     },
     "execution_count": 392,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(Browning_archaic_specific_mis_oe)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 393,
   "id": "7895393a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "349"
      ]
     },
     "execution_count": 393,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(Browning_introgressed_mis_oe)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 394,
   "id": "a25a44f4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "12708"
      ]
     },
     "execution_count": 394,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(non_SA_mis_oe)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 395,
   "id": "3eacfc74",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "KruskalResult(statistic=20.889083756698465, pvalue=0.00011100996646209538)"
      ]
     },
     "execution_count": 395,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "kruskal(Browning_ancient_mis_oe['mis_oe'], Browning_archaic_specific_mis_oe['mis_oe'], Browning_introgressed_mis_oe['mis_oe'], non_SA_mis_oe['mis_oe'])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "24d95952",
   "metadata": {},
   "source": [
    "Now for missense z-score."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 396,
   "id": "119b8948",
   "metadata": {},
   "outputs": [],
   "source": [
    "Browning_ancient_mis_z = data_2.loc[data_2['Browning_allele_origin'] == 'ancient', ['annotation', 'mis_z']].drop_duplicates().dropna()\n",
    "Browning_archaic_specific_mis_z = data_2.loc[data_2['Browning_allele_origin'] == 'archaic-specific', ['annotation', 'mis_z']].drop_duplicates().dropna()\n",
    "Browning_introgressed_mis_z = data_2.loc[data_2['Browning_allele_origin'] == 'introgressed', ['annotation', 'mis_z']].drop_duplicates().dropna()\n",
    "non_SA_mis_z = data.loc[~data['annotation'].isin(data_2['annotation']), ['annotation', 'mis_z']].drop_duplicates().dropna()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 397,
   "id": "e779dad9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1780"
      ]
     },
     "execution_count": 397,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(Browning_ancient_mis_z)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 398,
   "id": "359aad05",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1832"
      ]
     },
     "execution_count": 398,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(Browning_archaic_specific_mis_z)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 399,
   "id": "0b8eba75",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "349"
      ]
     },
     "execution_count": 399,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(Browning_introgressed_mis_z)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 400,
   "id": "2535f1b2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "12708"
      ]
     },
     "execution_count": 400,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(non_SA_mis_z)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 401,
   "id": "311eab90",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "KruskalResult(statistic=7.474011387744941, pvalue=0.05822998409610527)"
      ]
     },
     "execution_count": 401,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "kruskal(Browning_ancient_mis_z['mis_z'], Browning_archaic_specific_mis_z['mis_z'], Browning_introgressed_mis_z['mis_z'], non_SA_mis_z['mis_z'])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5223f7ba",
   "metadata": {},
   "source": [
    "Now for LoF observed/expected."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 402,
   "id": "8da85cce",
   "metadata": {},
   "outputs": [],
   "source": [
    "Browning_ancient_lof_oe = data_2.loc[data_2['Browning_allele_origin'] == 'ancient', ['annotation', 'lof_oe']].drop_duplicates().dropna()\n",
    "Browning_archaic_specific_lof_oe = data_2.loc[data_2['Browning_allele_origin'] == 'archaic-specific', ['annotation', 'lof_oe']].drop_duplicates().dropna()\n",
    "Browning_introgressed_lof_oe = data_2.loc[data_2['Browning_allele_origin'] == 'introgressed', ['annotation', 'lof_oe']].drop_duplicates().dropna()\n",
    "non_SA_lof_oe = data.loc[~data['annotation'].isin(data_2['annotation']), ['annotation', 'lof_oe']].drop_duplicates().dropna()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 403,
   "id": "0afe96a3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1774"
      ]
     },
     "execution_count": 403,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(Browning_ancient_lof_oe)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 404,
   "id": "6725020b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1824"
      ]
     },
     "execution_count": 404,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(Browning_archaic_specific_lof_oe)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 405,
   "id": "d36e68d0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "349"
      ]
     },
     "execution_count": 405,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(Browning_introgressed_lof_oe)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 406,
   "id": "8f1ef733",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "12484"
      ]
     },
     "execution_count": 406,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(non_SA_lof_oe)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 407,
   "id": "e55b579b",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "KruskalResult(statistic=1.4415470445962708, pvalue=0.6958254176524867)"
      ]
     },
     "execution_count": 407,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "kruskal(Browning_ancient_lof_oe['lof_oe'], Browning_archaic_specific_lof_oe['lof_oe'], Browning_introgressed_lof_oe['lof_oe'], non_SA_lof_oe['lof_oe'])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f4545c93",
   "metadata": {},
   "source": [
    "Now for LoF z-scores."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 408,
   "id": "7a398574",
   "metadata": {},
   "outputs": [],
   "source": [
    "Browning_ancient_lof_z = data_2.loc[data_2['Browning_allele_origin'] == 'ancient', ['annotation', 'lof_z']].drop_duplicates().dropna()\n",
    "Browning_archaic_specific_lof_z = data_2.loc[data_2['Browning_allele_origin'] == 'archaic-specific', ['annotation', 'lof_z']].drop_duplicates().dropna()\n",
    "Browning_introgressed_lof_z = data_2.loc[data_2['Browning_allele_origin'] == 'introgressed', ['annotation', 'lof_z']].drop_duplicates().dropna()\n",
    "non_SA_lof_z = data.loc[~data['annotation'].isin(data_2['annotation']), ['annotation', 'lof_z']].drop_duplicates().dropna()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 409,
   "id": "027c4eb0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1774"
      ]
     },
     "execution_count": 409,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(Browning_ancient_lof_z)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 410,
   "id": "ec9832be",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1824"
      ]
     },
     "execution_count": 410,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(Browning_archaic_specific_lof_z)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 411,
   "id": "c7d9345b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "349"
      ]
     },
     "execution_count": 411,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(Browning_introgressed_lof_z)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 412,
   "id": "b207e5f4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "12484"
      ]
     },
     "execution_count": 412,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(non_SA_lof_z)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 413,
   "id": "14a81517",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "KruskalResult(statistic=315.77589291885266, pvalue=3.8294331448242675e-68)"
      ]
     },
     "execution_count": 413,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "kruskal(Browning_ancient_lof_z['lof_z'], Browning_archaic_specific_lof_z['lof_z'], Browning_introgressed_lof_z['lof_z'], non_SA_lof_z['lof_z'])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5f8ca625",
   "metadata": {},
   "source": [
    "And phyloP."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 414,
   "id": "d7c719ef",
   "metadata": {},
   "outputs": [],
   "source": [
    "Browning_ancient_phyloP = data_2.loc[data_2['Browning_allele_origin'] == 'ancient', 'phyloP'].dropna()\n",
    "Browning_archaic_specific_phyloP = data_2.loc[data_2['Browning_allele_origin'] == 'archaic-specific', 'phyloP'].dropna()\n",
    "Browning_introgressed_phyloP = data_2.loc[data_2['Browning_allele_origin'] == 'introgressed', 'phyloP'].dropna()\n",
    "non_SA_phyloP = data.loc[data['delta_max'] < 0.2, 'phyloP'].dropna()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 415,
   "id": "37b42f51",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2194"
      ]
     },
     "execution_count": 415,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(Browning_ancient_phyloP)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 416,
   "id": "034aeea3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2186"
      ]
     },
     "execution_count": 416,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(Browning_archaic_specific_phyloP)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 417,
   "id": "12f7e432",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "386"
      ]
     },
     "execution_count": 417,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(Browning_introgressed_phyloP)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 418,
   "id": "6e2379e4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1600192"
      ]
     },
     "execution_count": 418,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(non_SA_phyloP)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 419,
   "id": "259d1c76",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "KruskalResult(statistic=808.2443759754741, pvalue=7.050363136869668e-175)"
      ]
     },
     "execution_count": 419,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "kruskal(Browning_ancient_phyloP, Browning_archaic_specific_phyloP, Browning_introgressed_phyloP, non_SA_phyloP)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ec7f09e1",
   "metadata": {},
   "source": [
    "How many of these are not \"neutrally evolving\"?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 420,
   "id": "b0c3ca97",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.15650876845375394"
      ]
     },
     "execution_count": 420,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "((Browning_ancient_phyloP < -1.3).sum() + (Browning_ancient_phyloP > 1.3).sum() + (Browning_archaic_specific_phyloP < -1.3).sum() + (Browning_archaic_specific_phyloP > 1.3).sum() + (Browning_introgressed_phyloP < -1.3).sum() + (Browning_introgressed_phyloP > 1.3).sum() + (non_SA_phyloP < -1.3).sum() + (non_SA_phyloP > 1.3).sum()) / (len(Browning_ancient_phyloP) + len(Browning_archaic_specific_phyloP) + len(Browning_introgressed_phyloP) + len(non_SA_phyloP))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2d49c95c",
   "metadata": {},
   "source": [
    "And \"neutrally evolving\"?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 421,
   "id": "0eeba110",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.8434912315462461"
      ]
     },
     "execution_count": 421,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "1-0.15650876845375394"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cd8d4e0f",
   "metadata": {},
   "source": [
    "# Purifying Selection <a class = 'anchor' id = 'purifyingselection'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b83f1b74",
   "metadata": {},
   "source": [
    "Now let's consider the role of purifying selection on SAVs. Start by writing a function that will use data from a dataframe to add columns with enrichment test information: the OR, p-value, and 95% CI."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 422,
   "id": "03719dc2",
   "metadata": {},
   "outputs": [],
   "source": [
    "def purifying_enrichment(df,A,B,C,D):\n",
    "    odds_ratio = []\n",
    "    p_value = []\n",
    "    lower_CI = []\n",
    "    upper_CI = []\n",
    "\n",
    "    for i in range(len(df)):\n",
    "        table = np.array([[df.loc[i, A],\n",
    "                           df.loc[i, B]],\n",
    "                          [df.loc[i, C],\n",
    "                           df.loc[i, D]]])\n",
    "        OR, p = fisher_exact(table)\n",
    "    \n",
    "        lCI = np.exp((np.log(OR)) - (1.96 * (sqrt((1/table[0,0]) + (1/table[0,1]) + (1/table[1,0]) + (1/table[1,1])))))\n",
    "        uCI = np.exp((np.log(OR)) + (1.96 * (sqrt((1/table[0,0]) + (1/table[0,1]) + (1/table[1,0]) + (1/table[1,1])))))\n",
    "    \n",
    "        odds_ratio.append(OR)\n",
    "        p_value.append(p)\n",
    "        lower_CI.append(lCI)\n",
    "        upper_CI.append(uCI)\n",
    "    \n",
    "    df['odds_ratio'] = odds_ratio\n",
    "    df['p_value'] = p_value\n",
    "    df['lower_CI'] = lower_CI\n",
    "    df['upper_CI'] = upper_CI"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6ac7f9e0",
   "metadata": {},
   "source": [
    "## Varied Deltas <a class = 'anchor' id = 'varieddeltas'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "27ab8f62",
   "metadata": {},
   "source": [
    "Now assemble the data starting with the sum of all unique counts."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 423,
   "id": "40cc1c90",
   "metadata": {},
   "outputs": [],
   "source": [
    "all_unique_total = data[(data['distribution']=='Altai') | (data['distribution']=='Chagyrskaya') | (data['distribution']=='Denisovan') | (data['distribution']=='Vindija')]\n",
    "all_unique_total_SA_2 = all_unique_total[(all_unique_total['delta_max']>=0.2)]\n",
    "all_unique_total_SA_2_index = all_unique_total_SA_2.index\n",
    "all_unique_total_non_SA_2 = all_unique_total.drop(all_unique_total_SA_2_index)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 424,
   "id": "48606aee",
   "metadata": {},
   "outputs": [],
   "source": [
    "all_unique_total = data[(data['distribution']=='Altai') | (data['distribution']=='Chagyrskaya') | (data['distribution']=='Denisovan') | (data['distribution']=='Vindija')]\n",
    "all_unique_total_SA_3 = all_unique_total[(all_unique_total['delta_max']>=0.3)]\n",
    "all_unique_total_SA_3_index = all_unique_total_SA_3.index\n",
    "all_unique_total_non_SA_3 = all_unique_total.drop(all_unique_total_SA_3_index)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 425,
   "id": "b655a88d",
   "metadata": {},
   "outputs": [],
   "source": [
    "all_unique_total = data[(data['distribution']=='Altai') | (data['distribution']=='Chagyrskaya') | (data['distribution']=='Denisovan') | (data['distribution']=='Vindija')]\n",
    "all_unique_total_SA_4 = all_unique_total[(all_unique_total['delta_max']>=0.4)]\n",
    "all_unique_total_SA_4_index = all_unique_total_SA_4.index\n",
    "all_unique_total_non_SA_4 = all_unique_total.drop(all_unique_total_SA_4_index)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 426,
   "id": "b145f583",
   "metadata": {},
   "outputs": [],
   "source": [
    "all_unique_total = data[(data['distribution']=='Altai') | (data['distribution']=='Chagyrskaya') | (data['distribution']=='Denisovan') | (data['distribution']=='Vindija')]\n",
    "all_unique_total_SA_5 = all_unique_total[(all_unique_total['delta_max']>=0.5)]\n",
    "all_unique_total_SA_5_index = all_unique_total_SA_5.index\n",
    "all_unique_total_non_SA_5 = all_unique_total.drop(all_unique_total_SA_5_index)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "bb9d0129",
   "metadata": {},
   "source": [
    "Now for the shared SAVs. We'll use some of these again below."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 427,
   "id": "46b0631c",
   "metadata": {},
   "outputs": [],
   "source": [
    "all_shared = data[(data['distribution']=='Shared')]\n",
    "all_shared_SAVs_2 = all_shared[(all_shared['delta_max']>=0.2)]\n",
    "all_shared_SAVs_2_index = all_shared_SAVs_2.index\n",
    "all_shared_non_SAVs_2 = all_shared.drop(all_shared_SAVs_2_index)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 428,
   "id": "a818bad5",
   "metadata": {},
   "outputs": [],
   "source": [
    "all_shared = data[(data['distribution']=='Shared')]\n",
    "all_shared_SAVs_3 = all_shared[(all_shared['delta_max']>=0.3)]\n",
    "all_shared_SAVs_3_index = all_shared_SAVs_3.index\n",
    "all_shared_non_SAVs_3 = all_shared.drop(all_shared_SAVs_3_index)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 429,
   "id": "e92cb2eb",
   "metadata": {},
   "outputs": [],
   "source": [
    "all_shared = data[(data['distribution']=='Shared')]\n",
    "all_shared_SAVs_4 = all_shared[(all_shared['delta_max']>=0.4)]\n",
    "all_shared_SAVs_4_index = all_shared_SAVs_4.index\n",
    "all_shared_non_SAVs_4 = all_shared.drop(all_shared_SAVs_4_index)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 430,
   "id": "15cb541d",
   "metadata": {},
   "outputs": [],
   "source": [
    "all_shared = data[(data['distribution']=='Shared')]\n",
    "all_shared_SAVs_5 = all_shared[(all_shared['delta_max']>=0.5)]\n",
    "all_shared_SAVs_5_index = all_shared_SAVs_5.index\n",
    "all_shared_non_SAVs_5 = all_shared.drop(all_shared_SAVs_5_index)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "18137bc6",
   "metadata": {},
   "source": [
    "Build a dataframe."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 431,
   "id": "d19de204",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>delta_threshold</th>\n",
       "      <th>N_total_unique_SAVs</th>\n",
       "      <th>N_total_unique_non_SAVs</th>\n",
       "      <th>N_shared_SAVs</th>\n",
       "      <th>N_shared_non_SAVs</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0.2</td>\n",
       "      <td>2416</td>\n",
       "      <td>615666</td>\n",
       "      <td>1933</td>\n",
       "      <td>571264</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.3</td>\n",
       "      <td>1175</td>\n",
       "      <td>616907</td>\n",
       "      <td>945</td>\n",
       "      <td>572252</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.4</td>\n",
       "      <td>685</td>\n",
       "      <td>617397</td>\n",
       "      <td>533</td>\n",
       "      <td>572664</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.5</td>\n",
       "      <td>437</td>\n",
       "      <td>617645</td>\n",
       "      <td>328</td>\n",
       "      <td>572869</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  delta_threshold  N_total_unique_SAVs  N_total_unique_non_SAVs  \\\n",
       "0             0.2                 2416                   615666   \n",
       "1             0.3                 1175                   616907   \n",
       "2             0.4                  685                   617397   \n",
       "3             0.5                  437                   617645   \n",
       "\n",
       "   N_shared_SAVs  N_shared_non_SAVs  \n",
       "0           1933             571264  \n",
       "1            945             572252  \n",
       "2            533             572664  \n",
       "3            328             572869  "
      ]
     },
     "execution_count": 431,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "delta_threshold = ['0.2','0.3','0.4','0.5']\n",
    "total_unique_SAVs = [len(all_unique_total_SA_2), len(all_unique_total_SA_3), len(all_unique_total_SA_4), len(all_unique_total_SA_5)]\n",
    "total_unique_non_SAVs = [len(all_unique_total_non_SA_2), len(all_unique_total_non_SA_3), len(all_unique_total_non_SA_4), len(all_unique_total_non_SA_5)]\n",
    "shared_SAVs = [len(all_shared_SAVs_2), len(all_shared_SAVs_3), len(all_shared_SAVs_4), len(all_shared_SAVs_5)]\n",
    "shared_non_SAVs = [len(all_shared_non_SAVs_2), len(all_shared_non_SAVs_3), len(all_shared_non_SAVs_4), len(all_shared_non_SAVs_5)]\n",
    "\n",
    "total_unique_enrichment = pd.DataFrame(list(zip(delta_threshold, total_unique_SAVs, total_unique_non_SAVs, shared_SAVs, shared_non_SAVs)),\n",
    "                                          columns =['delta_threshold', 'N_total_unique_SAVs', 'N_total_unique_non_SAVs', 'N_shared_SAVs', 'N_shared_non_SAVs'])\n",
    "total_unique_enrichment"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f8546d6a",
   "metadata": {},
   "source": [
    "Run the enrichment tests."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 432,
   "id": "d3737ee3",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>delta_threshold</th>\n",
       "      <th>N_total_unique_SAVs</th>\n",
       "      <th>N_total_unique_non_SAVs</th>\n",
       "      <th>N_shared_SAVs</th>\n",
       "      <th>N_shared_non_SAVs</th>\n",
       "      <th>odds_ratio</th>\n",
       "      <th>p_value</th>\n",
       "      <th>lower_CI</th>\n",
       "      <th>upper_CI</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0.2</td>\n",
       "      <td>2416</td>\n",
       "      <td>615666</td>\n",
       "      <td>1933</td>\n",
       "      <td>571264</td>\n",
       "      <td>1.159730</td>\n",
       "      <td>0.000001</td>\n",
       "      <td>1.092280</td>\n",
       "      <td>1.231345</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.3</td>\n",
       "      <td>1175</td>\n",
       "      <td>616907</td>\n",
       "      <td>945</td>\n",
       "      <td>572252</td>\n",
       "      <td>1.153383</td>\n",
       "      <td>0.001099</td>\n",
       "      <td>1.058636</td>\n",
       "      <td>1.256610</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.4</td>\n",
       "      <td>685</td>\n",
       "      <td>617397</td>\n",
       "      <td>533</td>\n",
       "      <td>572664</td>\n",
       "      <td>1.192062</td>\n",
       "      <td>0.002351</td>\n",
       "      <td>1.064410</td>\n",
       "      <td>1.335022</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.5</td>\n",
       "      <td>437</td>\n",
       "      <td>617645</td>\n",
       "      <td>328</td>\n",
       "      <td>572869</td>\n",
       "      <td>1.235731</td>\n",
       "      <td>0.003772</td>\n",
       "      <td>1.070824</td>\n",
       "      <td>1.426033</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  delta_threshold  N_total_unique_SAVs  N_total_unique_non_SAVs  \\\n",
       "0             0.2                 2416                   615666   \n",
       "1             0.3                 1175                   616907   \n",
       "2             0.4                  685                   617397   \n",
       "3             0.5                  437                   617645   \n",
       "\n",
       "   N_shared_SAVs  N_shared_non_SAVs  odds_ratio   p_value  lower_CI  upper_CI  \n",
       "0           1933             571264    1.159730  0.000001  1.092280  1.231345  \n",
       "1            945             572252    1.153383  0.001099  1.058636  1.256610  \n",
       "2            533             572664    1.192062  0.002351  1.064410  1.335022  \n",
       "3            328             572869    1.235731  0.003772  1.070824  1.426033  "
      ]
     },
     "execution_count": 432,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "purifying_enrichment(total_unique_enrichment,'N_total_unique_SAVs','N_shared_SAVs','N_total_unique_non_SAVs','N_shared_non_SAVs')\n",
    "total_unique_enrichment"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f1ac069d",
   "metadata": {},
   "source": [
    "Save this dataframe to make plotting easier."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 433,
   "id": "e181c641",
   "metadata": {},
   "outputs": [],
   "source": [
    "total_unique_enrichment.to_csv('total_unique_enrichment_tests.txt', sep = '\\t', header = True, index = False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "68c9e48a",
   "metadata": {},
   "source": [
    "## Lineage-Specific <a class = 'anchor' id = 'lineagespecific'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f1d26496",
   "metadata": {},
   "source": [
    "What about lineage-specific effects? "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 434,
   "id": "a8cee2e3",
   "metadata": {},
   "outputs": [],
   "source": [
    "altai = data[(data['distribution'] == 'Altai')]\n",
    "altai_SAVs_2 = altai[(altai['delta_max'] >= 0.2)]\n",
    "altai_SAVs_2_index = altai_SAVs_2.index\n",
    "altai_non_SAVs_2 = altai.drop(altai_SAVs_2_index)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 435,
   "id": "aaafe3fa",
   "metadata": {},
   "outputs": [],
   "source": [
    "altai = data[(data['distribution'] == 'Altai')]\n",
    "altai_SAVs_5 = altai[(altai['delta_max'] >= 0.5)]\n",
    "altai_SAVs_5_index = altai_SAVs_5.index\n",
    "altai_non_SAVs_5 = altai.drop(altai_SAVs_5_index)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 436,
   "id": "f25656ab",
   "metadata": {},
   "outputs": [],
   "source": [
    "chagyrskaya = data[(data['distribution'] == 'Chagyrskaya')]\n",
    "chagyrskaya_SAVs_2 = chagyrskaya[(chagyrskaya['delta_max'] >= 0.2)]\n",
    "chagyrskaya_SAVs_2_index = chagyrskaya_SAVs_2.index\n",
    "chagyrskaya_non_SAVs_2 = chagyrskaya.drop(chagyrskaya_SAVs_2_index)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 437,
   "id": "98e543b2",
   "metadata": {},
   "outputs": [],
   "source": [
    "chagyrskaya = data[(data['distribution'] == 'Chagyrskaya')]\n",
    "chagyrskaya_SAVs_5 = chagyrskaya[(chagyrskaya['delta_max'] >= 0.5)]\n",
    "chagyrskaya_SAVs_5_index = chagyrskaya_SAVs_5.index\n",
    "chagyrskaya_non_SAVs_5 = chagyrskaya.drop(chagyrskaya_SAVs_5_index)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 438,
   "id": "ea38d0e7",
   "metadata": {},
   "outputs": [],
   "source": [
    "denisovan = data[(data['distribution'] == 'Denisovan')]\n",
    "denisovan_SAVs_2 = denisovan[(denisovan['delta_max'] >= 0.2)]\n",
    "denisovan_SAVs_2_index = denisovan_SAVs_2.index\n",
    "denisovan_non_SAVs_2 = denisovan.drop(denisovan_SAVs_2_index)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 439,
   "id": "da43ad48",
   "metadata": {},
   "outputs": [],
   "source": [
    "denisovan = data[(data['distribution'] == 'Denisovan')]\n",
    "denisovan_SAVs_5 = denisovan[(denisovan['delta_max'] >= 0.5)]\n",
    "denisovan_SAVs_5_index = denisovan_SAVs_5.index\n",
    "denisovan_non_SAVs_5 = denisovan.drop(denisovan_SAVs_5_index)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 440,
   "id": "0b3b165a",
   "metadata": {},
   "outputs": [],
   "source": [
    "vindija = data[(data['distribution'] == 'Vindija')]\n",
    "vindija_SAVs_2 = vindija[(vindija['delta_max'] >= 0.2)]\n",
    "vindija_SAVs_2_index = vindija_SAVs_2.index\n",
    "vindija_non_SAVs_2 = vindija.drop(vindija_SAVs_2_index)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 441,
   "id": "b4b8da38",
   "metadata": {},
   "outputs": [],
   "source": [
    "vindija = data[(data['distribution'] == 'Vindija')]\n",
    "vindija_SAVs_5 = vindija[(vindija['delta_max'] >= 0.5)]\n",
    "vindija_SAVs_5_index = vindija_SAVs_5.index\n",
    "vindija_non_SAVs_5 = vindija.drop(vindija_SAVs_5_index)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1be695fc",
   "metadata": {},
   "source": [
    "Build the dataframe."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 442,
   "id": "8cada4a9",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>delta_threshold</th>\n",
       "      <th>lineage</th>\n",
       "      <th>N_unique_SAVs</th>\n",
       "      <th>N_unique_non_SAVs</th>\n",
       "      <th>N_shared_SAVs</th>\n",
       "      <th>N_shared_non_SAVs</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0.2</td>\n",
       "      <td>altai</td>\n",
       "      <td>399</td>\n",
       "      <td>81517</td>\n",
       "      <td>1933</td>\n",
       "      <td>571264</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.2</td>\n",
       "      <td>chagyrskaya</td>\n",
       "      <td>218</td>\n",
       "      <td>53457</td>\n",
       "      <td>1933</td>\n",
       "      <td>571264</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.2</td>\n",
       "      <td>denisovan</td>\n",
       "      <td>1492</td>\n",
       "      <td>410000</td>\n",
       "      <td>1933</td>\n",
       "      <td>571264</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.2</td>\n",
       "      <td>vindija</td>\n",
       "      <td>307</td>\n",
       "      <td>70692</td>\n",
       "      <td>1933</td>\n",
       "      <td>571264</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0.5</td>\n",
       "      <td>altai</td>\n",
       "      <td>75</td>\n",
       "      <td>81841</td>\n",
       "      <td>328</td>\n",
       "      <td>572869</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>0.5</td>\n",
       "      <td>chagyrskaya</td>\n",
       "      <td>35</td>\n",
       "      <td>53640</td>\n",
       "      <td>328</td>\n",
       "      <td>572869</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>0.5</td>\n",
       "      <td>denisovan</td>\n",
       "      <td>254</td>\n",
       "      <td>411238</td>\n",
       "      <td>328</td>\n",
       "      <td>572869</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>0.5</td>\n",
       "      <td>vindija</td>\n",
       "      <td>73</td>\n",
       "      <td>70926</td>\n",
       "      <td>328</td>\n",
       "      <td>572869</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  delta_threshold      lineage  N_unique_SAVs  N_unique_non_SAVs  \\\n",
       "0             0.2        altai            399              81517   \n",
       "1             0.2  chagyrskaya            218              53457   \n",
       "2             0.2    denisovan           1492             410000   \n",
       "3             0.2      vindija            307              70692   \n",
       "4             0.5        altai             75              81841   \n",
       "5             0.5  chagyrskaya             35              53640   \n",
       "6             0.5    denisovan            254             411238   \n",
       "7             0.5      vindija             73              70926   \n",
       "\n",
       "   N_shared_SAVs  N_shared_non_SAVs  \n",
       "0           1933             571264  \n",
       "1           1933             571264  \n",
       "2           1933             571264  \n",
       "3           1933             571264  \n",
       "4            328             572869  \n",
       "5            328             572869  \n",
       "6            328             572869  \n",
       "7            328             572869  "
      ]
     },
     "execution_count": 442,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "delta_threshold = ['0.2','0.2','0.2','0.2','0.5','0.5','0.5','0.5']\n",
    "lineage = ['altai','chagyrskaya','denisovan','vindija','altai','chagyrskaya','denisovan','vindija']\n",
    "unique_SAVs = [len(altai_SAVs_2), len(chagyrskaya_SAVs_2), len(denisovan_SAVs_2), len(vindija_SAVs_2), len(altai_SAVs_5), len(chagyrskaya_SAVs_5), len(denisovan_SAVs_5), len(vindija_SAVs_5)]\n",
    "unique_non_SAVs = [len(altai_non_SAVs_2), len(chagyrskaya_non_SAVs_2), len(denisovan_non_SAVs_2), len(vindija_non_SAVs_2), len(altai_non_SAVs_5), len(chagyrskaya_non_SAVs_5), len(denisovan_non_SAVs_5), len(vindija_non_SAVs_5)]\n",
    "shared_SAVs = [len(all_shared_SAVs_2), len(all_shared_SAVs_2), len(all_shared_SAVs_2), len(all_shared_SAVs_2), len(all_shared_SAVs_5), len(all_shared_SAVs_5), len(all_shared_SAVs_5), len(all_shared_SAVs_5)]\n",
    "shared_non_SAVs = [len(all_shared_non_SAVs_2), len(all_shared_non_SAVs_2), len(all_shared_non_SAVs_2), len(all_shared_non_SAVs_2), len(all_shared_non_SAVs_5), len(all_shared_non_SAVs_5), len(all_shared_non_SAVs_5), len(all_shared_non_SAVs_5)]\n",
    "\n",
    "lineage_specific_enrichment = pd.DataFrame(list(zip(delta_threshold, lineage, unique_SAVs, unique_non_SAVs, shared_SAVs, shared_non_SAVs)),\n",
    "                                          columns =['delta_threshold', 'lineage', 'N_unique_SAVs', 'N_unique_non_SAVs', 'N_shared_SAVs', 'N_shared_non_SAVs'])\n",
    "lineage_specific_enrichment"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 443,
   "id": "4b4952d6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>delta_threshold</th>\n",
       "      <th>lineage</th>\n",
       "      <th>N_unique_SAVs</th>\n",
       "      <th>N_unique_non_SAVs</th>\n",
       "      <th>N_shared_SAVs</th>\n",
       "      <th>N_shared_non_SAVs</th>\n",
       "      <th>odds_ratio</th>\n",
       "      <th>p_value</th>\n",
       "      <th>lower_CI</th>\n",
       "      <th>upper_CI</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0.2</td>\n",
       "      <td>altai</td>\n",
       "      <td>399</td>\n",
       "      <td>81517</td>\n",
       "      <td>1933</td>\n",
       "      <td>571264</td>\n",
       "      <td>1.446538</td>\n",
       "      <td>1.152673e-10</td>\n",
       "      <td>1.298420</td>\n",
       "      <td>1.611551</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.2</td>\n",
       "      <td>chagyrskaya</td>\n",
       "      <td>218</td>\n",
       "      <td>53457</td>\n",
       "      <td>1933</td>\n",
       "      <td>571264</td>\n",
       "      <td>1.205194</td>\n",
       "      <td>1.078870e-02</td>\n",
       "      <td>1.047416</td>\n",
       "      <td>1.386738</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.2</td>\n",
       "      <td>denisovan</td>\n",
       "      <td>1492</td>\n",
       "      <td>410000</td>\n",
       "      <td>1933</td>\n",
       "      <td>571264</td>\n",
       "      <td>1.075449</td>\n",
       "      <td>3.575391e-02</td>\n",
       "      <td>1.005089</td>\n",
       "      <td>1.150736</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.2</td>\n",
       "      <td>vindija</td>\n",
       "      <td>307</td>\n",
       "      <td>70692</td>\n",
       "      <td>1933</td>\n",
       "      <td>571264</td>\n",
       "      <td>1.283433</td>\n",
       "      <td>8.655485e-05</td>\n",
       "      <td>1.137538</td>\n",
       "      <td>1.448040</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0.5</td>\n",
       "      <td>altai</td>\n",
       "      <td>75</td>\n",
       "      <td>81841</td>\n",
       "      <td>328</td>\n",
       "      <td>572869</td>\n",
       "      <td>1.600559</td>\n",
       "      <td>4.931121e-04</td>\n",
       "      <td>1.245305</td>\n",
       "      <td>2.057159</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>0.5</td>\n",
       "      <td>chagyrskaya</td>\n",
       "      <td>35</td>\n",
       "      <td>53640</td>\n",
       "      <td>328</td>\n",
       "      <td>572869</td>\n",
       "      <td>1.139622</td>\n",
       "      <td>4.524517e-01</td>\n",
       "      <td>0.804170</td>\n",
       "      <td>1.615004</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>0.5</td>\n",
       "      <td>denisovan</td>\n",
       "      <td>254</td>\n",
       "      <td>411238</td>\n",
       "      <td>328</td>\n",
       "      <td>572869</td>\n",
       "      <td>1.078753</td>\n",
       "      <td>3.774332e-01</td>\n",
       "      <td>0.915704</td>\n",
       "      <td>1.270834</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>0.5</td>\n",
       "      <td>vindija</td>\n",
       "      <td>73</td>\n",
       "      <td>70926</td>\n",
       "      <td>328</td>\n",
       "      <td>572869</td>\n",
       "      <td>1.797624</td>\n",
       "      <td>2.009186e-05</td>\n",
       "      <td>1.394727</td>\n",
       "      <td>2.316906</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  delta_threshold      lineage  N_unique_SAVs  N_unique_non_SAVs  \\\n",
       "0             0.2        altai            399              81517   \n",
       "1             0.2  chagyrskaya            218              53457   \n",
       "2             0.2    denisovan           1492             410000   \n",
       "3             0.2      vindija            307              70692   \n",
       "4             0.5        altai             75              81841   \n",
       "5             0.5  chagyrskaya             35              53640   \n",
       "6             0.5    denisovan            254             411238   \n",
       "7             0.5      vindija             73              70926   \n",
       "\n",
       "   N_shared_SAVs  N_shared_non_SAVs  odds_ratio       p_value  lower_CI  \\\n",
       "0           1933             571264    1.446538  1.152673e-10  1.298420   \n",
       "1           1933             571264    1.205194  1.078870e-02  1.047416   \n",
       "2           1933             571264    1.075449  3.575391e-02  1.005089   \n",
       "3           1933             571264    1.283433  8.655485e-05  1.137538   \n",
       "4            328             572869    1.600559  4.931121e-04  1.245305   \n",
       "5            328             572869    1.139622  4.524517e-01  0.804170   \n",
       "6            328             572869    1.078753  3.774332e-01  0.915704   \n",
       "7            328             572869    1.797624  2.009186e-05  1.394727   \n",
       "\n",
       "   upper_CI  \n",
       "0  1.611551  \n",
       "1  1.386738  \n",
       "2  1.150736  \n",
       "3  1.448040  \n",
       "4  2.057159  \n",
       "5  1.615004  \n",
       "6  1.270834  \n",
       "7  2.316906  "
      ]
     },
     "execution_count": 443,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "purifying_enrichment(lineage_specific_enrichment,'N_unique_SAVs','N_shared_SAVs','N_unique_non_SAVs','N_shared_non_SAVs')\n",
    "lineage_specific_enrichment"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 444,
   "id": "f59cf815",
   "metadata": {},
   "outputs": [],
   "source": [
    "lineage_specific_enrichment.to_csv('lineage_specific_enrichment_tests.txt', sep = '\\t', header = True, index = False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1489cdb6",
   "metadata": {},
   "source": [
    "# SAVs in Moderns <a class = 'anchor' id = 'SAVsinmoderns'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "09e2d67f",
   "metadata": {},
   "source": [
    "## Introgressed SAV Distribution <a class = 'anchor' id = 'introgressedSAVdistribution'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "589c2ce1",
   "metadata": {},
   "source": [
    "We know how splice altering variants appear to be distributed among archaics. But how are introgressed variants distributed among the archaic variants? Let's use the archaic-specific variants as a background so we will assess those first."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 445,
   "id": "7b2880bc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>distribution</th>\n",
       "      <th>archaic_distribution</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Altai</td>\n",
       "      <td>300</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Chagyrskaya</td>\n",
       "      <td>169</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Denisovan</td>\n",
       "      <td>916</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Late Neanderthal</td>\n",
       "      <td>94</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>236</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Other</td>\n",
       "      <td>128</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Shared</td>\n",
       "      <td>112</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Vindija</td>\n",
       "      <td>231</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       distribution  archaic_distribution\n",
       "0             Altai                   300\n",
       "1       Chagyrskaya                   169\n",
       "2         Denisovan                   916\n",
       "3  Late Neanderthal                    94\n",
       "4       Neanderthal                   236\n",
       "5             Other                   128\n",
       "6            Shared                   112\n",
       "7           Vindija                   231"
      ]
     },
     "execution_count": 445,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "archaic_specific_2 = data_2[data_2['Vernot_allele_origin']=='archaic-specific']\n",
    "archaic_dist_2 = archaic_specific_2.groupby(['distribution']).size().to_frame('archaic_distribution').reset_index()\n",
    "archaic_dist_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 446,
   "id": "1d6aabd2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>distribution</th>\n",
       "      <th>archaic_distribution</th>\n",
       "      <th>prop</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Altai</td>\n",
       "      <td>300</td>\n",
       "      <td>0.291262</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Chagyrskaya</td>\n",
       "      <td>169</td>\n",
       "      <td>0.164078</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Late Neanderthal</td>\n",
       "      <td>94</td>\n",
       "      <td>0.091262</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>236</td>\n",
       "      <td>0.229126</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Vindija</td>\n",
       "      <td>231</td>\n",
       "      <td>0.224272</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       distribution  archaic_distribution      prop\n",
       "0             Altai                   300  0.291262\n",
       "1       Chagyrskaya                   169  0.164078\n",
       "3  Late Neanderthal                    94  0.091262\n",
       "4       Neanderthal                   236  0.229126\n",
       "7           Vindija                   231  0.224272"
      ]
     },
     "execution_count": 446,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "archaic_dist_2 = archaic_dist_2.drop([2, 5, 6])\n",
    "archaic_dist_2['prop'] = archaic_dist_2['archaic_distribution']/archaic_dist_2['archaic_distribution'].sum()\n",
    "archaic_dist_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 447,
   "id": "1d88faaa",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>distribution</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Altai</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Denisovan</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>143</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Other</td>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Shared</td>\n",
       "      <td>68</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  distribution  count\n",
       "0        Altai      7\n",
       "1    Denisovan      5\n",
       "2  Neanderthal    143\n",
       "3        Other     21\n",
       "4       Shared     68"
      ]
     },
     "execution_count": 447,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Vernot_introgressed_dist_2 = data_2[data_2['Vernot_allele_origin']=='introgressed'].groupby(['distribution']).size().to_frame('count').reset_index()\n",
    "Vernot_introgressed_dist_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 448,
   "id": "9723142d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>distribution</th>\n",
       "      <th>count</th>\n",
       "      <th>prop</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Altai</td>\n",
       "      <td>7</td>\n",
       "      <td>0.046667</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>143</td>\n",
       "      <td>0.953333</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  distribution  count      prop\n",
       "0        Altai      7  0.046667\n",
       "2  Neanderthal    143  0.953333"
      ]
     },
     "execution_count": 448,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Vernot_introgressed_dist_2 = Vernot_introgressed_dist_2.drop([1, 3, 4])\n",
    "Vernot_introgressed_dist_2['prop'] = Vernot_introgressed_dist_2['count'] / Vernot_introgressed_dist_2['count'].sum()\n",
    "Vernot_introgressed_dist_2"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "71b811b2",
   "metadata": {},
   "source": [
    "Repeat for delta >= 0.5."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 449,
   "id": "27937859",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>distribution</th>\n",
       "      <th>archaic_distribution</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Altai</td>\n",
       "      <td>57</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Chagyrskaya</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Denisovan</td>\n",
       "      <td>165</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Late Neanderthal</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Other</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Shared</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Vindija</td>\n",
       "      <td>55</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       distribution  archaic_distribution\n",
       "0             Altai                    57\n",
       "1       Chagyrskaya                    27\n",
       "2         Denisovan                   165\n",
       "3  Late Neanderthal                    10\n",
       "4       Neanderthal                    36\n",
       "5             Other                    27\n",
       "6            Shared                    27\n",
       "7           Vindija                    55"
      ]
     },
     "execution_count": 449,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "archaic_specific_5 = data_5[data_5['Vernot_allele_origin']=='archaic-specific']\n",
    "archaic_dist_5 = archaic_specific_5.groupby(['distribution']).size().to_frame('archaic_distribution').reset_index()\n",
    "archaic_dist_5"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 450,
   "id": "3ef9ff6f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>distribution</th>\n",
       "      <th>archaic_distribution</th>\n",
       "      <th>prop</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Altai</td>\n",
       "      <td>57</td>\n",
       "      <td>0.308108</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Chagyrskaya</td>\n",
       "      <td>27</td>\n",
       "      <td>0.145946</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Late Neanderthal</td>\n",
       "      <td>10</td>\n",
       "      <td>0.054054</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>36</td>\n",
       "      <td>0.194595</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Vindija</td>\n",
       "      <td>55</td>\n",
       "      <td>0.297297</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       distribution  archaic_distribution      prop\n",
       "0             Altai                    57  0.308108\n",
       "1       Chagyrskaya                    27  0.145946\n",
       "3  Late Neanderthal                    10  0.054054\n",
       "4       Neanderthal                    36  0.194595\n",
       "7           Vindija                    55  0.297297"
      ]
     },
     "execution_count": 450,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "archaic_dist_5 = archaic_dist_5.drop([2, 5, 6])\n",
    "archaic_dist_5['prop'] = archaic_dist_5['archaic_distribution']/archaic_dist_5['archaic_distribution'].sum()\n",
    "archaic_dist_5"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 451,
   "id": "cbe98e3d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>distribution</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Altai</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Denisovan</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Other</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Shared</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  distribution  count\n",
       "0        Altai      1\n",
       "1    Denisovan      1\n",
       "2  Neanderthal     38\n",
       "3        Other      5\n",
       "4       Shared     14"
      ]
     },
     "execution_count": 451,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Vernot_introgressed_dist_5 = data_5[data_5['Vernot_allele_origin']=='introgressed'].groupby(['distribution']).size().to_frame('count').reset_index()\n",
    "Vernot_introgressed_dist_5"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 452,
   "id": "25055f03",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>distribution</th>\n",
       "      <th>count</th>\n",
       "      <th>prop</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Altai</td>\n",
       "      <td>1</td>\n",
       "      <td>0.025641</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>38</td>\n",
       "      <td>0.974359</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  distribution  count      prop\n",
       "0        Altai      1  0.025641\n",
       "2  Neanderthal     38  0.974359"
      ]
     },
     "execution_count": 452,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Vernot_introgressed_dist_5 = Vernot_introgressed_dist_5.drop([1, 3, 4])\n",
    "Vernot_introgressed_dist_5['prop'] = Vernot_introgressed_dist_5['count'] / Vernot_introgressed_dist_5['count'].sum()\n",
    "Vernot_introgressed_dist_5"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "366d9157",
   "metadata": {},
   "source": [
    "Now for Browning."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 453,
   "id": "487139b1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>distribution</th>\n",
       "      <th>archaic_distribution</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Altai</td>\n",
       "      <td>300</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Chagyrskaya</td>\n",
       "      <td>169</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Denisovan</td>\n",
       "      <td>916</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Late Neanderthal</td>\n",
       "      <td>94</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>236</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Other</td>\n",
       "      <td>128</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Shared</td>\n",
       "      <td>112</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Vindija</td>\n",
       "      <td>231</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       distribution  archaic_distribution\n",
       "0             Altai                   300\n",
       "1       Chagyrskaya                   169\n",
       "2         Denisovan                   916\n",
       "3  Late Neanderthal                    94\n",
       "4       Neanderthal                   236\n",
       "5             Other                   128\n",
       "6            Shared                   112\n",
       "7           Vindija                   231"
      ]
     },
     "execution_count": 453,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "archaic_specific_2 = data_2[data_2['Vernot_allele_origin']=='archaic-specific']\n",
    "archaic_dist_2 = archaic_specific_2.groupby(['distribution']).size().to_frame('archaic_distribution').reset_index()\n",
    "archaic_dist_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 454,
   "id": "cdf6f969",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>distribution</th>\n",
       "      <th>archaic_distribution</th>\n",
       "      <th>prop</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Altai</td>\n",
       "      <td>300</td>\n",
       "      <td>0.291262</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Chagyrskaya</td>\n",
       "      <td>169</td>\n",
       "      <td>0.164078</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Late Neanderthal</td>\n",
       "      <td>94</td>\n",
       "      <td>0.091262</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>236</td>\n",
       "      <td>0.229126</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Vindija</td>\n",
       "      <td>231</td>\n",
       "      <td>0.224272</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       distribution  archaic_distribution      prop\n",
       "0             Altai                   300  0.291262\n",
       "1       Chagyrskaya                   169  0.164078\n",
       "3  Late Neanderthal                    94  0.091262\n",
       "4       Neanderthal                   236  0.229126\n",
       "7           Vindija                   231  0.224272"
      ]
     },
     "execution_count": 454,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "archaic_dist_2 = archaic_dist_2.drop([2, 5, 6])\n",
    "archaic_dist_2['prop'] = archaic_dist_2['archaic_distribution']/archaic_dist_2['archaic_distribution'].sum()\n",
    "archaic_dist_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 455,
   "id": "93c4e7c9",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>distribution</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Altai</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Denisovan</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>207</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Other</td>\n",
       "      <td>38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Shared</td>\n",
       "      <td>122</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  distribution  count\n",
       "0        Altai     17\n",
       "1    Denisovan      2\n",
       "2  Neanderthal    207\n",
       "3        Other     38\n",
       "4       Shared    122"
      ]
     },
     "execution_count": 455,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Browning_introgressed_dist_2 = data_2[data_2['Browning_allele_origin']=='introgressed'].groupby(['distribution']).size().to_frame('count').reset_index()\n",
    "Browning_introgressed_dist_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 456,
   "id": "3122e1d9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>distribution</th>\n",
       "      <th>count</th>\n",
       "      <th>prop</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Altai</td>\n",
       "      <td>17</td>\n",
       "      <td>0.075893</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>207</td>\n",
       "      <td>0.924107</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  distribution  count      prop\n",
       "0        Altai     17  0.075893\n",
       "2  Neanderthal    207  0.924107"
      ]
     },
     "execution_count": 456,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Browning_introgressed_dist_2 = Browning_introgressed_dist_2.drop([1, 3, 4])\n",
    "Browning_introgressed_dist_2['prop'] = Browning_introgressed_dist_2['count'] / Browning_introgressed_dist_2['count'].sum()\n",
    "Browning_introgressed_dist_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 457,
   "id": "3761ea09",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>distribution</th>\n",
       "      <th>archaic_distribution</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Altai</td>\n",
       "      <td>57</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Chagyrskaya</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Denisovan</td>\n",
       "      <td>165</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Late Neanderthal</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Other</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Shared</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Vindija</td>\n",
       "      <td>55</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       distribution  archaic_distribution\n",
       "0             Altai                    57\n",
       "1       Chagyrskaya                    27\n",
       "2         Denisovan                   165\n",
       "3  Late Neanderthal                    10\n",
       "4       Neanderthal                    36\n",
       "5             Other                    27\n",
       "6            Shared                    27\n",
       "7           Vindija                    55"
      ]
     },
     "execution_count": 457,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "archaic_specific_5 = data_5[data_5['Vernot_allele_origin']=='archaic-specific']\n",
    "archaic_dist_5 = archaic_specific_5.groupby(['distribution']).size().to_frame('archaic_distribution').reset_index()\n",
    "archaic_dist_5"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 458,
   "id": "62bd5cb9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>distribution</th>\n",
       "      <th>archaic_distribution</th>\n",
       "      <th>prop</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Altai</td>\n",
       "      <td>57</td>\n",
       "      <td>0.308108</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Chagyrskaya</td>\n",
       "      <td>27</td>\n",
       "      <td>0.145946</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Late Neanderthal</td>\n",
       "      <td>10</td>\n",
       "      <td>0.054054</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>36</td>\n",
       "      <td>0.194595</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Vindija</td>\n",
       "      <td>55</td>\n",
       "      <td>0.297297</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       distribution  archaic_distribution      prop\n",
       "0             Altai                    57  0.308108\n",
       "1       Chagyrskaya                    27  0.145946\n",
       "3  Late Neanderthal                    10  0.054054\n",
       "4       Neanderthal                    36  0.194595\n",
       "7           Vindija                    55  0.297297"
      ]
     },
     "execution_count": 458,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "archaic_dist_5 = archaic_dist_5.drop([2, 5, 6])\n",
    "archaic_dist_5['prop'] = archaic_dist_5['archaic_distribution']/archaic_dist_5['archaic_distribution'].sum()\n",
    "archaic_dist_5"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 459,
   "id": "bfca857f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>distribution</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Altai</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Denisovan</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Other</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Shared</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  distribution  count\n",
       "0        Altai      4\n",
       "1    Denisovan      2\n",
       "2  Neanderthal     38\n",
       "3        Other      4\n",
       "4       Shared     23"
      ]
     },
     "execution_count": 459,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Browning_introgressed_dist_5 = data_5[data_5['Browning_allele_origin']=='introgressed'].groupby(['distribution']).size().to_frame('count').reset_index()\n",
    "Browning_introgressed_dist_5"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 460,
   "id": "0f9c8e06",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>distribution</th>\n",
       "      <th>count</th>\n",
       "      <th>prop</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Altai</td>\n",
       "      <td>4</td>\n",
       "      <td>0.095238</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>38</td>\n",
       "      <td>0.904762</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  distribution  count      prop\n",
       "0        Altai      4  0.095238\n",
       "2  Neanderthal     38  0.904762"
      ]
     },
     "execution_count": 460,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Browning_introgressed_dist_5 = Browning_introgressed_dist_5.drop([1, 3, 4])\n",
    "Browning_introgressed_dist_5['prop'] = Browning_introgressed_dist_5['count'] / Browning_introgressed_dist_5['count'].sum()\n",
    "Browning_introgressed_dist_5"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2ab15e09",
   "metadata": {},
   "source": [
    "## Allele Frequency and Max Delta <a class = 'anchor' id = 'allelefrequencymaxdelta'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cbfe4be3",
   "metadata": {},
   "source": [
    "If deltas represent the probability of a variant being splice altering, this metric may be related to allele frequency in modern human populations such that variants with small deltas may be high frequency because they are not actually splice altering."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 461,
   "id": "cb29c832",
   "metadata": {},
   "outputs": [],
   "source": [
    "present_in_1KG = data[data['present_in_1KG'] == 'yes']\n",
    "present_in_1KG_SA_2 = data_2[data_2['present_in_1KG'] == 'yes']\n",
    "present_in_1KG_SA_5 = data_5[data_5['present_in_1KG'] == 'yes']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 462,
   "id": "6ba1742e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-0.007352829220282604 8.637985526942276e-14\n"
     ]
    }
   ],
   "source": [
    "rho, p = spearmanr(present_in_1KG['1KG_allele_frequency'], present_in_1KG['delta_max'])\n",
    "print(rho,p)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 463,
   "id": "b01b94a4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-0.00015112162504674333 0.9928437359082407\n"
     ]
    }
   ],
   "source": [
    "rho, p = spearmanr(present_in_1KG_SA_2['1KG_allele_frequency'], present_in_1KG_SA_2['delta_max'])\n",
    "print(rho,p)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 464,
   "id": "71c2c34f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-0.1122804631529113 0.005817787511787715\n"
     ]
    }
   ],
   "source": [
    "rho, p = spearmanr(present_in_1KG_SA_5['1KG_allele_frequency'], present_in_1KG_SA_5['delta_max'])\n",
    "print(rho,p)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "77a215a3",
   "metadata": {},
   "source": [
    "Let's also consider ancient vs introgressed variants. First Vernot."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 465,
   "id": "947c0b9a",
   "metadata": {},
   "outputs": [],
   "source": [
    "Vernot_ancient_variants = data[data['Vernot_allele_origin'] == 'ancient']\n",
    "Vernot_introgressed_variants = data[data['Vernot_allele_origin'] == 'introgressed']\n",
    "\n",
    "Vernot_ancient_variants_SA_2 = data_2[data_2['Vernot_allele_origin'] == 'ancient']\n",
    "Vernot_introgressed_variants_SA_2 = data_2[data_2['Vernot_allele_origin'] == 'introgressed']\n",
    "\n",
    "Vernot_ancient_variants_SA_5 = data_5[data_5['Vernot_allele_origin'] == 'ancient']\n",
    "Vernot_introgressed_variants_SA_5 = data_5[data_5['Vernot_allele_origin'] == 'introgressed']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 466,
   "id": "e423f3b1",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-0.003314513124591188 0.006217887126235556\n"
     ]
    }
   ],
   "source": [
    "rho, p = spearmanr(Vernot_ancient_variants['1KG_allele_frequency'], Vernot_ancient_variants['delta_max'])\n",
    "print(rho,p)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 467,
   "id": "3605b62d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.003996022439579673 0.2972389787199452\n"
     ]
    }
   ],
   "source": [
    "rho, p = spearmanr(Vernot_introgressed_variants['Vernot_introgressed_AF'], Vernot_introgressed_variants['delta_max'])\n",
    "print(rho,p)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 468,
   "id": "9b8e5784",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.023032083134465274 0.27459969993034433\n"
     ]
    }
   ],
   "source": [
    "rho, p = spearmanr(Vernot_ancient_variants_SA_2['1KG_allele_frequency'], Vernot_ancient_variants_SA_2['delta_max'])\n",
    "print(rho,p)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 469,
   "id": "72a222a6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-0.23778363811750808 0.0001774412637169479\n"
     ]
    }
   ],
   "source": [
    "rho, p = spearmanr(Vernot_introgressed_variants_SA_2['Vernot_introgressed_AF'], Vernot_introgressed_variants_SA_2['delta_max'])\n",
    "print(rho,p)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 470,
   "id": "c5b7f2a2",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-0.12045161956844723 0.018365211572499436\n"
     ]
    }
   ],
   "source": [
    "rho, p = spearmanr(Vernot_ancient_variants_SA_5['1KG_allele_frequency'], Vernot_ancient_variants_SA_5['delta_max'])\n",
    "print(rho,p)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 471,
   "id": "61ad8d66",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-0.010180203372018522 0.9390018302513599\n"
     ]
    }
   ],
   "source": [
    "rho, p = spearmanr(Vernot_introgressed_variants_SA_5['Vernot_introgressed_AF'], Vernot_introgressed_variants_SA_5['delta_max'])\n",
    "print(rho,p)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4cdb9bb5",
   "metadata": {},
   "source": [
    "Now Browning."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 472,
   "id": "21892a9c",
   "metadata": {},
   "outputs": [],
   "source": [
    "Browning_ancient_variants = data[data['Browning_allele_origin'] == 'ancient']\n",
    "Browning_introgressed_variants = data[data['Browning_allele_origin'] == 'introgressed']\n",
    "\n",
    "Browning_ancient_variants_SA_2 = data_2[data_2['Browning_allele_origin'] == 'ancient']\n",
    "Browning_introgressed_variants_SA_2 = data_2[data_2['Browning_allele_origin'] == 'introgressed']\n",
    "\n",
    "Browning_ancient_variants_SA_5 = data_5[data_5['Browning_allele_origin'] == 'ancient']\n",
    "Browning_introgressed_variants_SA_5 = data_5[data_5['Browning_allele_origin'] == 'introgressed']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 473,
   "id": "00fe7675",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-0.0014042305212856882 0.2501708002524762\n"
     ]
    }
   ],
   "source": [
    "rho, p = spearmanr(Browning_ancient_variants['1KG_allele_frequency'], Browning_ancient_variants['delta_max'])\n",
    "print(rho,p)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 474,
   "id": "f0502481",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.004092787486773656 0.1996781446277722\n"
     ]
    }
   ],
   "source": [
    "rho, p = spearmanr(Browning_introgressed_variants['Browning_introgressed_AF'], Browning_introgressed_variants['delta_max'])\n",
    "print(rho,p)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 475,
   "id": "8a82c87c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.018438954258859618 0.3878837383270327\n"
     ]
    }
   ],
   "source": [
    "rho, p = spearmanr(Browning_ancient_variants_SA_2['1KG_allele_frequency'], Browning_ancient_variants_SA_2['delta_max'])\n",
    "print(rho,p)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 476,
   "id": "690f559d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-0.09956359455569352 0.05062683014640129\n"
     ]
    }
   ],
   "source": [
    "rho, p = spearmanr(Browning_introgressed_variants_SA_2['Browning_introgressed_AF'], Browning_introgressed_variants_SA_2['delta_max'])\n",
    "print(rho,p)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 477,
   "id": "1c41a960",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-0.11314820835053853 0.02825198996162118\n"
     ]
    }
   ],
   "source": [
    "rho, p = spearmanr(Browning_ancient_variants_SA_5['1KG_allele_frequency'], Browning_ancient_variants_SA_5['delta_max'])\n",
    "print(rho,p)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 478,
   "id": "a2bd4f13",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-0.13688008107823207 0.25500901988007507\n"
     ]
    }
   ],
   "source": [
    "rho, p = spearmanr(Browning_introgressed_variants_SA_5['Browning_introgressed_AF'], Browning_introgressed_variants_SA_5['delta_max'])\n",
    "print(rho,p)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d8f79220",
   "metadata": {},
   "source": [
    "Let's examine allele frequency between superpopulations. "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2be5306c",
   "metadata": {},
   "source": [
    "First Vernot."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 479,
   "id": "f87ce527",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.5226508643821909"
      ]
     },
     "execution_count": 479,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Vernot_ancient_SA_2_AFR = Vernot_ancient_variants_SA_2['1KG_non_ASW_AFR_AF']\n",
    "Vernot_ancient_SA_2_AFR.mean()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 480,
   "id": "63b172a4",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.4761589698046181"
      ]
     },
     "execution_count": 480,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Vernot_ancient_SA_2_non_AFR = pd.concat([Vernot_ancient_variants_SA_2['1KG_AMR_AF'], Vernot_ancient_variants_SA_2['1KG_EAS_AF'], Vernot_ancient_variants_SA_2['1KG_EUR_AF'], Vernot_ancient_variants_SA_2['1KG_SAS_AF']], axis=0)\n",
    "Vernot_ancient_SA_2_non_AFR.mean()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 481,
   "id": "bdf27e18",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "MannwhitneyuResult(statistic=10963956.0, pvalue=2.6636825400212013e-09)"
      ]
     },
     "execution_count": 481,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mannwhitneyu(Vernot_ancient_SA_2_AFR, Vernot_ancient_SA_2_non_AFR)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fc2b5961",
   "metadata": {},
   "source": [
    "Let's use the Vernot AFs here that are for the archaic allele. Important to note that AFR may include ASW samples."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 482,
   "id": "62c2ff7b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.0005038524590163934"
      ]
     },
     "execution_count": 482,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Vernot_introgressed_SA_2_AFR = Vernot_introgressed_variants_SA_2['Vernot_AFR_AF']\n",
    "Vernot_introgressed_SA_2_AFR.mean()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 483,
   "id": "a6108d56",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.027982571721311474"
      ]
     },
     "execution_count": 483,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Vernot_introgressed_SA_2_non_AFR = pd.concat([Vernot_introgressed_variants_SA_2['Vernot_AMR_AF'], Vernot_introgressed_variants_SA_2['Vernot_EAS_AF'], Vernot_introgressed_variants_SA_2['Vernot_EUR_AF'], Vernot_introgressed_variants_SA_2['Vernot_SAS_AF']], axis=0)\n",
    "Vernot_introgressed_SA_2_non_AFR.mean()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 484,
   "id": "bb6a67af",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "MannwhitneyuResult(statistic=45360.5, pvalue=5.0924222907304063e-54)"
      ]
     },
     "execution_count": 484,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mannwhitneyu(Vernot_introgressed_SA_2_AFR, Vernot_introgressed_SA_2_non_AFR)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "dee21b73",
   "metadata": {},
   "source": [
    "Now Browning."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 485,
   "id": "78d35e31",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.5356676302246378"
      ]
     },
     "execution_count": 485,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Browning_ancient_SA_2_AFR = Browning_ancient_variants_SA_2['1KG_non_ASW_AFR_AF']\n",
    "Browning_ancient_SA_2_AFR.mean()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 486,
   "id": "f568397f",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.4851947608200456"
      ]
     },
     "execution_count": 486,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Browning_ancient_SA_2_non_AFR = pd.concat([Browning_ancient_variants_SA_2['1KG_AMR_AF'], Browning_ancient_variants_SA_2['1KG_EAS_AF'], Browning_ancient_variants_SA_2['1KG_EUR_AF'], Browning_ancient_variants_SA_2['1KG_SAS_AF']], axis=0)\n",
    "Browning_ancient_SA_2_non_AFR.mean()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 487,
   "id": "7bd68ef4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "MannwhitneyuResult(statistic=10506748.0, pvalue=5.408902214406806e-11)"
      ]
     },
     "execution_count": 487,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mannwhitneyu(Browning_ancient_SA_2_AFR, Browning_ancient_SA_2_non_AFR)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "199b8158",
   "metadata": {},
   "source": [
    "We need to do a little extra work for Browning to handle the introgressed reference variants. Let's start with AFR."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 488,
   "id": "72dd0b22",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "235107    0.001949\n",
       "735814    0.000975\n",
       "Name: 1KG_non_ASW_AFR_AF, dtype: float64"
      ]
     },
     "execution_count": 488,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Browning_introgressed_SA_2_AFR_refs = Browning_introgressed_variants_SA_2.loc[(Browning_introgressed_variants_SA_2['Browning_ref_alt'] == 0), '1KG_non_ASW_AFR_AF']\n",
    "Browning_introgressed_SA_2_AFR_refs = 1-Browning_introgressed_SA_2_AFR_refs\n",
    "Browning_introgressed_SA_2_AFR_refs.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 489,
   "id": "c0533576",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "10928    0.003899\n",
       "14504    0.007797\n",
       "17107    0.000000\n",
       "17139    0.000000\n",
       "19011    0.000000\n",
       "Name: 1KG_non_ASW_AFR_AF, dtype: float64"
      ]
     },
     "execution_count": 489,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Browning_introgressed_SA_2_AFR_alts = Browning_introgressed_variants_SA_2.loc[(Browning_introgressed_variants_SA_2['Browning_ref_alt'] == 1), '1KG_non_ASW_AFR_AF']\n",
    "Browning_introgressed_SA_2_AFR_alts.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 490,
   "id": "35e60fd2",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0    0.001949\n",
       "1    0.000975\n",
       "2    0.003899\n",
       "3    0.007797\n",
       "4    0.000000\n",
       "Name: 1KG_non_ASW_AFR_AF, dtype: float64"
      ]
     },
     "execution_count": 490,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Browning_introgressed_SA_2_AFR = pd.concat([Browning_introgressed_SA_2_AFR_refs, Browning_introgressed_SA_2_AFR_alts], ignore_index=True)\n",
    "Browning_introgressed_SA_2_AFR.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 491,
   "id": "49ea6dc7",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.0014736221632773263"
      ]
     },
     "execution_count": 491,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Browning_introgressed_SA_2_AFR.mean()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c1a1ab5d",
   "metadata": {},
   "source": [
    "Now do AMR, EAS, EUR, and SAS."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 492,
   "id": "b742ebaf",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0    0.25\n",
       "1    0.07\n",
       "2    0.00\n",
       "3    0.10\n",
       "4    0.17\n",
       "5    0.14\n",
       "6    0.03\n",
       "7    0.04\n",
       "dtype: float64"
      ]
     },
     "execution_count": 492,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Browning_introgressed_SA_2_AMR_refs = Browning_introgressed_variants_SA_2.loc[(Browning_introgressed_variants_SA_2['Browning_ref_alt'] == 0), '1KG_AMR_AF']\n",
    "Browning_introgressed_SA_2_EAS_refs = Browning_introgressed_variants_SA_2.loc[(Browning_introgressed_variants_SA_2['Browning_ref_alt'] == 0), '1KG_EAS_AF']\n",
    "Browning_introgressed_SA_2_EUR_refs = Browning_introgressed_variants_SA_2.loc[(Browning_introgressed_variants_SA_2['Browning_ref_alt'] == 0), '1KG_EUR_AF']\n",
    "Browning_introgressed_SA_2_SAS_refs = Browning_introgressed_variants_SA_2.loc[(Browning_introgressed_variants_SA_2['Browning_ref_alt'] == 0), '1KG_SAS_AF']\n",
    "Browning_introgressed_SA_2_non_AFR_refs = pd.concat([Browning_introgressed_SA_2_AMR_refs, Browning_introgressed_SA_2_EAS_refs, Browning_introgressed_SA_2_EUR_refs, Browning_introgressed_SA_2_SAS_refs], ignore_index=True)\n",
    "Browning_introgressed_SA_2_non_AFR_refs = 1 - Browning_introgressed_SA_2_non_AFR_refs\n",
    "Browning_introgressed_SA_2_non_AFR_refs.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 493,
   "id": "84f57a7c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0    0.00\n",
       "1    0.06\n",
       "2    0.01\n",
       "3    0.02\n",
       "4    0.02\n",
       "5    0.00\n",
       "6    0.00\n",
       "7    0.00\n",
       "8    0.01\n",
       "9    0.00\n",
       "dtype: float64"
      ]
     },
     "execution_count": 493,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Browning_introgressed_SA_2_AMR_alts = Browning_introgressed_variants_SA_2.loc[(Browning_introgressed_variants_SA_2['Browning_ref_alt'] == 1), '1KG_AMR_AF']\n",
    "Browning_introgressed_SA_2_EAS_alts = Browning_introgressed_variants_SA_2.loc[(Browning_introgressed_variants_SA_2['Browning_ref_alt'] == 1), '1KG_EAS_AF']\n",
    "Browning_introgressed_SA_2_EUR_alts = Browning_introgressed_variants_SA_2.loc[(Browning_introgressed_variants_SA_2['Browning_ref_alt'] == 1), '1KG_EUR_AF']\n",
    "Browning_introgressed_SA_2_SAS_alts = Browning_introgressed_variants_SA_2.loc[(Browning_introgressed_variants_SA_2['Browning_ref_alt'] == 1), '1KG_SAS_AF']\n",
    "Browning_introgressed_SA_2_non_AFR_alts = pd.concat([Browning_introgressed_SA_2_AMR_alts, Browning_introgressed_SA_2_EAS_alts, Browning_introgressed_SA_2_EUR_alts, Browning_introgressed_SA_2_SAS_alts], ignore_index=True)\n",
    "Browning_introgressed_SA_2_non_AFR_alts.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 494,
   "id": "12805b3d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.03881299734748011"
      ]
     },
     "execution_count": 494,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Browning_introgressed_SA_2_non_AFR = pd.concat([Browning_introgressed_SA_2_non_AFR_refs, Browning_introgressed_SA_2_non_AFR_alts], ignore_index=True)\n",
    "Browning_introgressed_SA_2_non_AFR.mean()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 495,
   "id": "2c14719e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "MannwhitneyuResult(statistic=nan, pvalue=nan)"
      ]
     },
     "execution_count": 495,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mannwhitneyu(Browning_introgressed_SA_2_AFR, Browning_introgressed_SA_2_non_AFR)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "251a2c6a",
   "metadata": {},
   "source": [
    "## Introgressed Genes <a class = 'anchor' id = 'introgressedgenes'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "91df522b",
   "metadata": {},
   "source": [
    "Let's take a quick peek at the highest frequency introgressed genes."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 496,
   "id": "a1988539",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref_allele</th>\n",
       "      <th>alt_allele</th>\n",
       "      <th>ancestral_allele</th>\n",
       "      <th>anc_dev</th>\n",
       "      <th>variant_type</th>\n",
       "      <th>altai_gt</th>\n",
       "      <th>chagyrskaya_gt</th>\n",
       "      <th>denisovan_gt</th>\n",
       "      <th>vindija_gt</th>\n",
       "      <th>altai_gt_boolean</th>\n",
       "      <th>chagyrskaya_gt_boolean</th>\n",
       "      <th>denisovan_gt_boolean</th>\n",
       "      <th>vindija_gt_boolean</th>\n",
       "      <th>distribution</th>\n",
       "      <th>present_in_1KG</th>\n",
       "      <th>1KG_allele_count</th>\n",
       "      <th>1KG_allele_number</th>\n",
       "      <th>1KG_allele_frequency</th>\n",
       "      <th>1KG_EAS_AF</th>\n",
       "      <th>1KG_EUR_AF</th>\n",
       "      <th>1KG_AFR_AF</th>\n",
       "      <th>1KG_AMR_AF</th>\n",
       "      <th>1KG_SAS_AF</th>\n",
       "      <th>1KG_non_ASW_AFR_AF</th>\n",
       "      <th>present_in_gnomAD</th>\n",
       "      <th>gnomAD_allele_count</th>\n",
       "      <th>gnomAD_allele_number</th>\n",
       "      <th>gnomAD_allele_frequency</th>\n",
       "      <th>Vernot_introgressed</th>\n",
       "      <th>Vernot_ancestral_allele</th>\n",
       "      <th>Vernot_derived_allele</th>\n",
       "      <th>Vernot_ancestral_derived_code</th>\n",
       "      <th>Vernot_AFA_AF</th>\n",
       "      <th>Vernot_AFR_AF</th>\n",
       "      <th>Vernot_AMR_AF</th>\n",
       "      <th>Vernot_EAS_AF</th>\n",
       "      <th>Vernot_EUR_AF</th>\n",
       "      <th>Vernot_PNG_AF</th>\n",
       "      <th>Vernot_SAS_AF</th>\n",
       "      <th>Vernot_Denisovan_base</th>\n",
       "      <th>Vernot_Neanderthal_base_1</th>\n",
       "      <th>Vernot_Neanderthal_base_2</th>\n",
       "      <th>Vernot_haplotype_tag</th>\n",
       "      <th>Vernot_introgressed_AF</th>\n",
       "      <th>Browning_introgressed</th>\n",
       "      <th>Browning_ref_alt</th>\n",
       "      <th>Browning_introgressed_AF</th>\n",
       "      <th>annotation</th>\n",
       "      <th>mis_oe</th>\n",
       "      <th>mis_z</th>\n",
       "      <th>lof_oe</th>\n",
       "      <th>lof_z</th>\n",
       "      <th>phyloP</th>\n",
       "      <th>ag_delta</th>\n",
       "      <th>al_delta</th>\n",
       "      <th>dg_delta</th>\n",
       "      <th>dl_delta</th>\n",
       "      <th>delta_max</th>\n",
       "      <th>ag_pos</th>\n",
       "      <th>al_pos</th>\n",
       "      <th>dg_pos</th>\n",
       "      <th>dl_pos</th>\n",
       "      <th>Adipose_Subcutaneous</th>\n",
       "      <th>Adipose_Visceral_Omentum</th>\n",
       "      <th>Adrenal_Gland</th>\n",
       "      <th>Artery_Aorta</th>\n",
       "      <th>Artery_Coronary</th>\n",
       "      <th>Artery_Tibial</th>\n",
       "      <th>Brain_Amygdala</th>\n",
       "      <th>Brain_Anterior_cingulate_cortex_BA24</th>\n",
       "      <th>Brain_Caudate_basal_ganglia</th>\n",
       "      <th>Brain_Cerebellar_Hemisphere</th>\n",
       "      <th>Brain_Cerebellum</th>\n",
       "      <th>Brain_Cortex</th>\n",
       "      <th>Brain_Frontal_Cortex_BA9</th>\n",
       "      <th>Brain_Hippocampus</th>\n",
       "      <th>Brain_Hypothalamus</th>\n",
       "      <th>Brain_Nucleus_accumbens_basal_ganglia</th>\n",
       "      <th>Brain_Putamen_basal_ganglia</th>\n",
       "      <th>Brain_Spinal_cord_cervical_c-1</th>\n",
       "      <th>Brain_Substantia_nigra</th>\n",
       "      <th>Breast_Mammary_Tissue</th>\n",
       "      <th>Cells_Cultured_fibroblasts</th>\n",
       "      <th>Cells_EBV-transformed_lymphocytes</th>\n",
       "      <th>Colon_Sigmoid</th>\n",
       "      <th>Colon_Transverse</th>\n",
       "      <th>Esophagus_Gastroesophageal_Junction</th>\n",
       "      <th>Esophagus_Mucosa</th>\n",
       "      <th>Esophagus_Muscularis</th>\n",
       "      <th>Heart_Atrial_Appendage</th>\n",
       "      <th>Heart_Left_Ventricle</th>\n",
       "      <th>Kidney_Cortex</th>\n",
       "      <th>Liver</th>\n",
       "      <th>Lung</th>\n",
       "      <th>Minor_Salivary_Gland</th>\n",
       "      <th>Muscle_Skeletal</th>\n",
       "      <th>Nerve_Tibial</th>\n",
       "      <th>Ovary</th>\n",
       "      <th>Pancreas</th>\n",
       "      <th>Pituitary</th>\n",
       "      <th>Prostate</th>\n",
       "      <th>Skin_Not_Sun_Exposed_Suprapubic</th>\n",
       "      <th>Skin_Sun_Exposed_Lower_leg</th>\n",
       "      <th>Small_Intestine_Terminal_Ileum</th>\n",
       "      <th>Spleen</th>\n",
       "      <th>Stomach</th>\n",
       "      <th>Testis</th>\n",
       "      <th>Thyroid</th>\n",
       "      <th>Uterus</th>\n",
       "      <th>Vagina</th>\n",
       "      <th>Whole_Blood</th>\n",
       "      <th>N_GTEx_tissues</th>\n",
       "      <th>sQTL</th>\n",
       "      <th>Vernot_allele_origin</th>\n",
       "      <th>Browning_allele_origin</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>939756</th>\n",
       "      <td>chr20</td>\n",
       "      <td>62224595</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>yes</td>\n",
       "      <td>1138.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.22</td>\n",
       "      <td>0.51</td>\n",
       "      <td>0.21</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.11</td>\n",
       "      <td>0.31</td>\n",
       "      <td>0.000975</td>\n",
       "      <td>yes</td>\n",
       "      <td>23692.0</td>\n",
       "      <td>143104.0</td>\n",
       "      <td>0.165558</td>\n",
       "      <td>yes</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.03503</td>\n",
       "      <td>0.00099</td>\n",
       "      <td>0.10663</td>\n",
       "      <td>0.51488</td>\n",
       "      <td>0.21173</td>\n",
       "      <td>0.03704</td>\n",
       "      <td>0.30982</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>chr20_62195671_62229244</td>\n",
       "      <td>0.228810</td>\n",
       "      <td>yes</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.22</td>\n",
       "      <td>GMEB2</td>\n",
       "      <td>0.66048</td>\n",
       "      <td>2.202000</td>\n",
       "      <td>0.049787</td>\n",
       "      <td>3.94620</td>\n",
       "      <td>-0.837</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.41</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.41</td>\n",
       "      <td>15</td>\n",
       "      <td>-44</td>\n",
       "      <td>2</td>\n",
       "      <td>-2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>21.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>introgressed</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>246812</th>\n",
       "      <td>chr11</td>\n",
       "      <td>11507780</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>yes</td>\n",
       "      <td>917.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.18</td>\n",
       "      <td>0.14</td>\n",
       "      <td>0.31</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.21</td>\n",
       "      <td>0.28</td>\n",
       "      <td>0.003899</td>\n",
       "      <td>yes</td>\n",
       "      <td>33636.0</td>\n",
       "      <td>143182.0</td>\n",
       "      <td>0.234918</td>\n",
       "      <td>yes</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.05732</td>\n",
       "      <td>0.00397</td>\n",
       "      <td>0.21182</td>\n",
       "      <td>0.13988</td>\n",
       "      <td>0.31014</td>\n",
       "      <td>0.09259</td>\n",
       "      <td>0.28323</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr11_11494384_11530287</td>\n",
       "      <td>0.189808</td>\n",
       "      <td>yes</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.18</td>\n",
       "      <td>GALNT18</td>\n",
       "      <td>0.87026</td>\n",
       "      <td>0.879900</td>\n",
       "      <td>0.387360</td>\n",
       "      <td>3.02520</td>\n",
       "      <td>-0.302</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.21</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.21</td>\n",
       "      <td>37</td>\n",
       "      <td>-14</td>\n",
       "      <td>2</td>\n",
       "      <td>-14</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>introgressed</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1058119</th>\n",
       "      <td>chr4</td>\n",
       "      <td>38805942</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>yes</td>\n",
       "      <td>911.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.18</td>\n",
       "      <td>0.39</td>\n",
       "      <td>0.22</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.15</td>\n",
       "      <td>0.16</td>\n",
       "      <td>0.001949</td>\n",
       "      <td>yes</td>\n",
       "      <td>19931.0</td>\n",
       "      <td>143224.0</td>\n",
       "      <td>0.139160</td>\n",
       "      <td>yes</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.04459</td>\n",
       "      <td>0.00198</td>\n",
       "      <td>0.14986</td>\n",
       "      <td>0.39484</td>\n",
       "      <td>0.22068</td>\n",
       "      <td>0.05556</td>\n",
       "      <td>0.16360</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>chr4_38294200_38857896</td>\n",
       "      <td>0.186192</td>\n",
       "      <td>yes</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.18</td>\n",
       "      <td>TLR1</td>\n",
       "      <td>1.09920</td>\n",
       "      <td>-0.701570</td>\n",
       "      <td>1.411200</td>\n",
       "      <td>-1.72720</td>\n",
       "      <td>-0.219</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.38</td>\n",
       "      <td>0.38</td>\n",
       "      <td>27</td>\n",
       "      <td>-15</td>\n",
       "      <td>1</td>\n",
       "      <td>-15</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>introgressed</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1267615</th>\n",
       "      <td>chr6</td>\n",
       "      <td>52138226</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>Altai</td>\n",
       "      <td>yes</td>\n",
       "      <td>782.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.15</td>\n",
       "      <td>0.24</td>\n",
       "      <td>0.17</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.09</td>\n",
       "      <td>0.29</td>\n",
       "      <td>0.001949</td>\n",
       "      <td>yes</td>\n",
       "      <td>17455.0</td>\n",
       "      <td>143064.0</td>\n",
       "      <td>0.122008</td>\n",
       "      <td>yes</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.01274</td>\n",
       "      <td>0.00198</td>\n",
       "      <td>0.09366</td>\n",
       "      <td>0.24206</td>\n",
       "      <td>0.16600</td>\n",
       "      <td>0.09259</td>\n",
       "      <td>0.29448</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>chr6_52138225_52183126</td>\n",
       "      <td>0.159636</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MCM3</td>\n",
       "      <td>0.89914</td>\n",
       "      <td>0.804510</td>\n",
       "      <td>0.535230</td>\n",
       "      <td>2.88400</td>\n",
       "      <td>0.563</td>\n",
       "      <td>0.21</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.21</td>\n",
       "      <td>-7</td>\n",
       "      <td>-40</td>\n",
       "      <td>-7</td>\n",
       "      <td>3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>222914</th>\n",
       "      <td>chr10</td>\n",
       "      <td>121209372</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>yes</td>\n",
       "      <td>690.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.14</td>\n",
       "      <td>0.31</td>\n",
       "      <td>0.09</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.11</td>\n",
       "      <td>0.20</td>\n",
       "      <td>0.001949</td>\n",
       "      <td>yes</td>\n",
       "      <td>12162.0</td>\n",
       "      <td>143260.0</td>\n",
       "      <td>0.084895</td>\n",
       "      <td>yes</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.00318</td>\n",
       "      <td>0.00198</td>\n",
       "      <td>0.11095</td>\n",
       "      <td>0.31845</td>\n",
       "      <td>0.08946</td>\n",
       "      <td>0.11111</td>\n",
       "      <td>0.19632</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr10_121002922_121217061</td>\n",
       "      <td>0.143432</td>\n",
       "      <td>yes</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.14</td>\n",
       "      <td>GRK5</td>\n",
       "      <td>0.68002</td>\n",
       "      <td>2.158300</td>\n",
       "      <td>0.203620</td>\n",
       "      <td>4.32700</td>\n",
       "      <td>-0.355</td>\n",
       "      <td>0.23</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.23</td>\n",
       "      <td>-23</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>-14</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>introgressed</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>541237</th>\n",
       "      <td>chr15</td>\n",
       "      <td>85403496</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>yes</td>\n",
       "      <td>682.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.13</td>\n",
       "      <td>0.14</td>\n",
       "      <td>0.24</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.20</td>\n",
       "      <td>0.13</td>\n",
       "      <td>0.003899</td>\n",
       "      <td>yes</td>\n",
       "      <td>24329.0</td>\n",
       "      <td>143106.0</td>\n",
       "      <td>0.170007</td>\n",
       "      <td>yes</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.04140</td>\n",
       "      <td>0.00397</td>\n",
       "      <td>0.20029</td>\n",
       "      <td>0.13889</td>\n",
       "      <td>0.23559</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.12883</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr15_84263508_85447079</td>\n",
       "      <td>0.141514</td>\n",
       "      <td>yes</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.13</td>\n",
       "      <td>ALPK3</td>\n",
       "      <td>0.98871</td>\n",
       "      <td>0.131400</td>\n",
       "      <td>0.716100</td>\n",
       "      <td>2.15380</td>\n",
       "      <td>0.462</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.33</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.33</td>\n",
       "      <td>15</td>\n",
       "      <td>50</td>\n",
       "      <td>15</td>\n",
       "      <td>-11</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>introgressed</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1507143</th>\n",
       "      <td>chr8</td>\n",
       "      <td>99006787</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>yes</td>\n",
       "      <td>569.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.11</td>\n",
       "      <td>0.13</td>\n",
       "      <td>0.09</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.24</td>\n",
       "      <td>0.17</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>yes</td>\n",
       "      <td>12906.0</td>\n",
       "      <td>143208.0</td>\n",
       "      <td>0.090121</td>\n",
       "      <td>yes</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.00955</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.23919</td>\n",
       "      <td>0.12897</td>\n",
       "      <td>0.09344</td>\n",
       "      <td>0.05556</td>\n",
       "      <td>0.17280</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr8_98910893_99084133</td>\n",
       "      <td>0.126880</td>\n",
       "      <td>yes</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.11</td>\n",
       "      <td>MATN2</td>\n",
       "      <td>0.79916</td>\n",
       "      <td>1.668900</td>\n",
       "      <td>0.638200</td>\n",
       "      <td>2.22070</td>\n",
       "      <td>-0.373</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.24</td>\n",
       "      <td>0.24</td>\n",
       "      <td>43</td>\n",
       "      <td>8</td>\n",
       "      <td>-2</td>\n",
       "      <td>43</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>introgressed</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>247163</th>\n",
       "      <td>chr11</td>\n",
       "      <td>11944265</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>yes</td>\n",
       "      <td>591.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.12</td>\n",
       "      <td>0.19</td>\n",
       "      <td>0.19</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.09</td>\n",
       "      <td>0.11</td>\n",
       "      <td>0.009747</td>\n",
       "      <td>yes</td>\n",
       "      <td>17909.0</td>\n",
       "      <td>142902.0</td>\n",
       "      <td>0.125324</td>\n",
       "      <td>yes</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.02866</td>\n",
       "      <td>0.00893</td>\n",
       "      <td>0.09222</td>\n",
       "      <td>0.18948</td>\n",
       "      <td>0.19682</td>\n",
       "      <td>0.18519</td>\n",
       "      <td>0.11043</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr11_11691675_12000532</td>\n",
       "      <td>0.119576</td>\n",
       "      <td>yes</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.12</td>\n",
       "      <td>USP47</td>\n",
       "      <td>0.73421</td>\n",
       "      <td>2.437500</td>\n",
       "      <td>0.056706</td>\n",
       "      <td>7.34150</td>\n",
       "      <td>0.383</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.35</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.35</td>\n",
       "      <td>-13</td>\n",
       "      <td>18</td>\n",
       "      <td>-1</td>\n",
       "      <td>18</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>introgressed</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>593546</th>\n",
       "      <td>chr16</td>\n",
       "      <td>77412781</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>yes</td>\n",
       "      <td>528.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.24</td>\n",
       "      <td>0.12</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.04</td>\n",
       "      <td>0.12</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>yes</td>\n",
       "      <td>11923.0</td>\n",
       "      <td>143216.0</td>\n",
       "      <td>0.083252</td>\n",
       "      <td>yes</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.01274</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.04611</td>\n",
       "      <td>0.23909</td>\n",
       "      <td>0.12127</td>\n",
       "      <td>0.16667</td>\n",
       "      <td>0.12474</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr16_77304807_77417155</td>\n",
       "      <td>0.106242</td>\n",
       "      <td>yes</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.10</td>\n",
       "      <td>ADAMTS18</td>\n",
       "      <td>1.38300</td>\n",
       "      <td>-3.508200</td>\n",
       "      <td>0.742900</td>\n",
       "      <td>2.06850</td>\n",
       "      <td>0.455</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.41</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.41</td>\n",
       "      <td>23</td>\n",
       "      <td>-32</td>\n",
       "      <td>2</td>\n",
       "      <td>-32</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>introgressed</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1123317</th>\n",
       "      <td>chr4</td>\n",
       "      <td>163061051</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>yes</td>\n",
       "      <td>510.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.08</td>\n",
       "      <td>0.23</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.09</td>\n",
       "      <td>0.12</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>yes</td>\n",
       "      <td>19710.0</td>\n",
       "      <td>143060.0</td>\n",
       "      <td>0.137774</td>\n",
       "      <td>yes</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.01592</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.08790</td>\n",
       "      <td>0.07837</td>\n",
       "      <td>0.22962</td>\n",
       "      <td>0.14815</td>\n",
       "      <td>0.12372</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr4_162928454_163168783</td>\n",
       "      <td>0.103922</td>\n",
       "      <td>yes</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.10</td>\n",
       "      <td>FSTL5</td>\n",
       "      <td>0.99361</td>\n",
       "      <td>0.047046</td>\n",
       "      <td>0.419990</td>\n",
       "      <td>3.41950</td>\n",
       "      <td>-0.415</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.20</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.20</td>\n",
       "      <td>1</td>\n",
       "      <td>-48</td>\n",
       "      <td>1</td>\n",
       "      <td>-48</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>introgressed</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>897185</th>\n",
       "      <td>chr2</td>\n",
       "      <td>242423548</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>yes</td>\n",
       "      <td>436.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.09</td>\n",
       "      <td>0.14</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.27</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>yes</td>\n",
       "      <td>4268.0</td>\n",
       "      <td>143346.0</td>\n",
       "      <td>0.029774</td>\n",
       "      <td>yes</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.02229</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.26513</td>\n",
       "      <td>0.13492</td>\n",
       "      <td>0.00398</td>\n",
       "      <td>0.64815</td>\n",
       "      <td>0.09816</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr2_242085733_242430170</td>\n",
       "      <td>0.100438</td>\n",
       "      <td>yes</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.09</td>\n",
       "      <td>FARP2</td>\n",
       "      <td>1.03750</td>\n",
       "      <td>-0.330770</td>\n",
       "      <td>0.947030</td>\n",
       "      <td>0.37745</td>\n",
       "      <td>0.477</td>\n",
       "      <td>0.32</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.32</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>44</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>introgressed</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>706199</th>\n",
       "      <td>chr18</td>\n",
       "      <td>60003587</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>yes</td>\n",
       "      <td>444.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.09</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.25</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.12</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.003899</td>\n",
       "      <td>yes</td>\n",
       "      <td>21727.0</td>\n",
       "      <td>143202.0</td>\n",
       "      <td>0.151723</td>\n",
       "      <td>yes</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.01592</td>\n",
       "      <td>0.00397</td>\n",
       "      <td>0.11671</td>\n",
       "      <td>0.00198</td>\n",
       "      <td>0.24950</td>\n",
       "      <td>0.09259</td>\n",
       "      <td>0.09714</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>chr18_59569521_60089689</td>\n",
       "      <td>0.093860</td>\n",
       "      <td>yes</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.09</td>\n",
       "      <td>TNFRSF11A</td>\n",
       "      <td>0.80508</td>\n",
       "      <td>1.289400</td>\n",
       "      <td>0.330780</td>\n",
       "      <td>3.04970</td>\n",
       "      <td>0.650</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.31</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.28</td>\n",
       "      <td>0.31</td>\n",
       "      <td>3</td>\n",
       "      <td>-40</td>\n",
       "      <td>3</td>\n",
       "      <td>33</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>introgressed</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1126124</th>\n",
       "      <td>chr4</td>\n",
       "      <td>169348476</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>yes</td>\n",
       "      <td>439.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.09</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.20</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.07</td>\n",
       "      <td>0.15</td>\n",
       "      <td>0.001949</td>\n",
       "      <td>yes</td>\n",
       "      <td>19225.0</td>\n",
       "      <td>143000.0</td>\n",
       "      <td>0.134441</td>\n",
       "      <td>yes</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.03503</td>\n",
       "      <td>0.00198</td>\n",
       "      <td>0.07205</td>\n",
       "      <td>0.01091</td>\n",
       "      <td>0.19881</td>\n",
       "      <td>0.16667</td>\n",
       "      <td>0.15235</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr4_169073736_169412749</td>\n",
       "      <td>0.087220</td>\n",
       "      <td>yes</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.09</td>\n",
       "      <td>DDX60L</td>\n",
       "      <td>0.98853</td>\n",
       "      <td>0.114220</td>\n",
       "      <td>0.820340</td>\n",
       "      <td>1.50460</td>\n",
       "      <td>0.467</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.31</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.31</td>\n",
       "      <td>-35</td>\n",
       "      <td>-3</td>\n",
       "      <td>2</td>\n",
       "      <td>-33</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>introgressed</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1582579</th>\n",
       "      <td>chr9</td>\n",
       "      <td>112209039</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>yes</td>\n",
       "      <td>385.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.08</td>\n",
       "      <td>0.27</td>\n",
       "      <td>0.03</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.07</td>\n",
       "      <td>0.03</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>yes</td>\n",
       "      <td>5035.0</td>\n",
       "      <td>143260.0</td>\n",
       "      <td>0.035146</td>\n",
       "      <td>yes</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.00637</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.07061</td>\n",
       "      <td>0.26587</td>\n",
       "      <td>0.03181</td>\n",
       "      <td>0.05556</td>\n",
       "      <td>0.02863</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr9_112193017_112358561</td>\n",
       "      <td>0.079384</td>\n",
       "      <td>yes</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.08</td>\n",
       "      <td>PTPN3</td>\n",
       "      <td>0.91465</td>\n",
       "      <td>0.705600</td>\n",
       "      <td>0.406750</td>\n",
       "      <td>4.30980</td>\n",
       "      <td>0.455</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.30</td>\n",
       "      <td>0.30</td>\n",
       "      <td>-35</td>\n",
       "      <td>-47</td>\n",
       "      <td>-47</td>\n",
       "      <td>6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>introgressed</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1221855</th>\n",
       "      <td>chr5</td>\n",
       "      <td>167919825</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>yes</td>\n",
       "      <td>379.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.07</td>\n",
       "      <td>0.06</td>\n",
       "      <td>0.18</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.09</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.000975</td>\n",
       "      <td>yes</td>\n",
       "      <td>18806.0</td>\n",
       "      <td>143144.0</td>\n",
       "      <td>0.131378</td>\n",
       "      <td>yes</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.03503</td>\n",
       "      <td>0.00099</td>\n",
       "      <td>0.09078</td>\n",
       "      <td>0.06448</td>\n",
       "      <td>0.17992</td>\n",
       "      <td>0.16667</td>\n",
       "      <td>0.05215</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr5_167873398_168164228</td>\n",
       "      <td>0.077664</td>\n",
       "      <td>yes</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.07</td>\n",
       "      <td>RARS</td>\n",
       "      <td>0.91791</td>\n",
       "      <td>0.543820</td>\n",
       "      <td>0.516620</td>\n",
       "      <td>2.64400</td>\n",
       "      <td>-0.471</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.38</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.38</td>\n",
       "      <td>-37</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>27</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>introgressed</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>611320</th>\n",
       "      <td>chr16</td>\n",
       "      <td>89806063</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>yes</td>\n",
       "      <td>371.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.07</td>\n",
       "      <td>0.26</td>\n",
       "      <td>0.07</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>yes</td>\n",
       "      <td>8286.0</td>\n",
       "      <td>143188.0</td>\n",
       "      <td>0.057868</td>\n",
       "      <td>yes</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.01592</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.02017</td>\n",
       "      <td>0.25893</td>\n",
       "      <td>0.06461</td>\n",
       "      <td>0.20370</td>\n",
       "      <td>0.01636</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr16_89785602_90065941</td>\n",
       "      <td>0.072014</td>\n",
       "      <td>yes</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.07</td>\n",
       "      <td>FANCA</td>\n",
       "      <td>1.54640</td>\n",
       "      <td>-5.413800</td>\n",
       "      <td>1.152500</td>\n",
       "      <td>-1.28990</td>\n",
       "      <td>0.491</td>\n",
       "      <td>0.21</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.21</td>\n",
       "      <td>0</td>\n",
       "      <td>-34</td>\n",
       "      <td>1</td>\n",
       "      <td>43</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>39.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>introgressed</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>735814</th>\n",
       "      <td>chr19</td>\n",
       "      <td>33410289</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>ancestral</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>yes</td>\n",
       "      <td>4744.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.93</td>\n",
       "      <td>0.90</td>\n",
       "      <td>0.86</td>\n",
       "      <td>0.99</td>\n",
       "      <td>0.93</td>\n",
       "      <td>0.96</td>\n",
       "      <td>0.999025</td>\n",
       "      <td>yes</td>\n",
       "      <td>129412.0</td>\n",
       "      <td>143248.0</td>\n",
       "      <td>0.903412</td>\n",
       "      <td>yes</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.03503</td>\n",
       "      <td>0.00099</td>\n",
       "      <td>0.06916</td>\n",
       "      <td>0.09821</td>\n",
       "      <td>0.14612</td>\n",
       "      <td>0.07407</td>\n",
       "      <td>0.04090</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr19_33040186_33456652</td>\n",
       "      <td>0.071076</td>\n",
       "      <td>yes</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.07</td>\n",
       "      <td>CEP89</td>\n",
       "      <td>0.93498</td>\n",
       "      <td>0.474930</td>\n",
       "      <td>0.939730</td>\n",
       "      <td>0.36440</td>\n",
       "      <td>-0.642</td>\n",
       "      <td>0.47</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.71</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.71</td>\n",
       "      <td>41</td>\n",
       "      <td>-29</td>\n",
       "      <td>5</td>\n",
       "      <td>-29</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>introgressed</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1154162</th>\n",
       "      <td>chr5</td>\n",
       "      <td>36114442</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>yes</td>\n",
       "      <td>326.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.06</td>\n",
       "      <td>0.14</td>\n",
       "      <td>0.07</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.11</td>\n",
       "      <td>0.03</td>\n",
       "      <td>0.000975</td>\n",
       "      <td>yes</td>\n",
       "      <td>7014.0</td>\n",
       "      <td>143212.0</td>\n",
       "      <td>0.048976</td>\n",
       "      <td>yes</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.02229</td>\n",
       "      <td>0.00099</td>\n",
       "      <td>0.11239</td>\n",
       "      <td>0.13889</td>\n",
       "      <td>0.06660</td>\n",
       "      <td>0.01852</td>\n",
       "      <td>0.02863</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr5_35828521_36296191</td>\n",
       "      <td>0.069500</td>\n",
       "      <td>yes</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.06</td>\n",
       "      <td>LMBRD2</td>\n",
       "      <td>0.66428</td>\n",
       "      <td>2.272300</td>\n",
       "      <td>0.350190</td>\n",
       "      <td>3.80740</td>\n",
       "      <td>-0.291</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.22</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.22</td>\n",
       "      <td>21</td>\n",
       "      <td>-25</td>\n",
       "      <td>41</td>\n",
       "      <td>-25</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>introgressed</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1580224</th>\n",
       "      <td>chr9</td>\n",
       "      <td>104333384</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>yes</td>\n",
       "      <td>324.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.06</td>\n",
       "      <td>0.07</td>\n",
       "      <td>0.08</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>yes</td>\n",
       "      <td>8613.0</td>\n",
       "      <td>143190.0</td>\n",
       "      <td>0.060151</td>\n",
       "      <td>yes</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.01274</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.09654</td>\n",
       "      <td>0.07143</td>\n",
       "      <td>0.07555</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.10327</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr9_104256268_104545680</td>\n",
       "      <td>0.069358</td>\n",
       "      <td>yes</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.06</td>\n",
       "      <td>GRIN3A</td>\n",
       "      <td>0.91452</td>\n",
       "      <td>0.742910</td>\n",
       "      <td>0.220950</td>\n",
       "      <td>4.60750</td>\n",
       "      <td>0.459</td>\n",
       "      <td>0.42</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.42</td>\n",
       "      <td>9</td>\n",
       "      <td>-37</td>\n",
       "      <td>-2</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>introgressed</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>235154</th>\n",
       "      <td>chr11</td>\n",
       "      <td>460765</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>yes</td>\n",
       "      <td>282.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.06</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.21</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.000975</td>\n",
       "      <td>yes</td>\n",
       "      <td>14233.0</td>\n",
       "      <td>143242.0</td>\n",
       "      <td>0.099363</td>\n",
       "      <td>yes</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.02229</td>\n",
       "      <td>0.00099</td>\n",
       "      <td>0.21326</td>\n",
       "      <td>0.00198</td>\n",
       "      <td>0.09742</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.02352</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr11_437558_508720</td>\n",
       "      <td>0.067434</td>\n",
       "      <td>yes</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.06</td>\n",
       "      <td>PTDSS2</td>\n",
       "      <td>0.73242</td>\n",
       "      <td>1.571300</td>\n",
       "      <td>0.448660</td>\n",
       "      <td>2.75010</td>\n",
       "      <td>0.334</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.07</td>\n",
       "      <td>0.30</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.30</td>\n",
       "      <td>-5</td>\n",
       "      <td>-4</td>\n",
       "      <td>-5</td>\n",
       "      <td>-39</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>19.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>introgressed</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         chrom        pos ref_allele alt_allele ancestral_allele    anc_dev  \\\n",
       "939756   chr20   62224595          G          A                G    derived   \n",
       "246812   chr11   11507780          G          A                G    derived   \n",
       "1058119   chr4   38805942          G          C                G    derived   \n",
       "1267615   chr6   52138226          C          G                C    derived   \n",
       "222914   chr10  121209372          G          A                G    derived   \n",
       "541237   chr15   85403496          G          A                G    derived   \n",
       "1507143   chr8   99006787          C          T                C    derived   \n",
       "247163   chr11   11944265          A          G                A    derived   \n",
       "593546   chr16   77412781          T          A                T    derived   \n",
       "1123317   chr4  163061051          T          C                T    derived   \n",
       "897185    chr2  242423548          A          G                A    derived   \n",
       "706199   chr18   60003587          G          A                G    derived   \n",
       "1126124   chr4  169348476          G          A                G    derived   \n",
       "1582579   chr9  112209039          A          G                A    derived   \n",
       "1221855   chr5  167919825          G          A                G    derived   \n",
       "611320   chr16   89806063          G          C                G    derived   \n",
       "735814   chr19   33410289          T          C                C  ancestral   \n",
       "1154162   chr5   36114442          A          C                A    derived   \n",
       "1580224   chr9  104333384          A          T                A    derived   \n",
       "235154   chr11     460765          A          G                A    derived   \n",
       "\n",
       "        variant_type altai_gt chagyrskaya_gt denisovan_gt vindija_gt  \\\n",
       "939756           snv      0/1            1/1          0/0        0/1   \n",
       "246812           snv      1/1            1/1          0/0        1/1   \n",
       "1058119          snv      0/1            1/1          0/0        1/1   \n",
       "1267615          snv      0/1            0/0          0/0        0/0   \n",
       "222914           snv      1/1            1/1          0/0        1/1   \n",
       "541237           snv      1/1            1/1          0/0        1/1   \n",
       "1507143          snv      1/1            1/1          0/0        1/1   \n",
       "247163           snv      1/1            1/1          0/0        1/1   \n",
       "593546           snv      1/1            0/1          0/0        1/1   \n",
       "1123317          snv      1/1            1/1          0/0        1/1   \n",
       "897185           snv      1/1            1/1          1/1        1/1   \n",
       "706199           snv      0/1            1/1          1/1        1/1   \n",
       "1126124          snv      1/1            1/1          0/0        1/1   \n",
       "1582579          snv      1/1            1/1          1/1        1/1   \n",
       "1221855          snv      1/1            1/1          1/1        1/1   \n",
       "611320           snv      1/1            1/1          1/1        1/1   \n",
       "735814           snv      0/0            0/0          1/1        0/0   \n",
       "1154162          snv      1/1            1/1          0/0        1/1   \n",
       "1580224          snv      1/1            1/1          0/0        1/1   \n",
       "235154           snv      1/1            1/1          0/0        1/1   \n",
       "\n",
       "         altai_gt_boolean  chagyrskaya_gt_boolean  denisovan_gt_boolean  \\\n",
       "939756               True                    True                 False   \n",
       "246812               True                    True                 False   \n",
       "1058119              True                    True                 False   \n",
       "1267615              True                   False                 False   \n",
       "222914               True                    True                 False   \n",
       "541237               True                    True                 False   \n",
       "1507143              True                    True                 False   \n",
       "247163               True                    True                 False   \n",
       "593546               True                    True                 False   \n",
       "1123317              True                    True                 False   \n",
       "897185               True                    True                  True   \n",
       "706199               True                    True                  True   \n",
       "1126124              True                    True                 False   \n",
       "1582579              True                    True                  True   \n",
       "1221855              True                    True                  True   \n",
       "611320               True                    True                  True   \n",
       "735814              False                   False                  True   \n",
       "1154162              True                    True                 False   \n",
       "1580224              True                    True                 False   \n",
       "235154               True                    True                 False   \n",
       "\n",
       "         vindija_gt_boolean distribution present_in_1KG  1KG_allele_count  \\\n",
       "939756                 True  Neanderthal            yes            1138.0   \n",
       "246812                 True  Neanderthal            yes             917.0   \n",
       "1058119                True  Neanderthal            yes             911.0   \n",
       "1267615               False        Altai            yes             782.0   \n",
       "222914                 True  Neanderthal            yes             690.0   \n",
       "541237                 True  Neanderthal            yes             682.0   \n",
       "1507143                True  Neanderthal            yes             569.0   \n",
       "247163                 True  Neanderthal            yes             591.0   \n",
       "593546                 True  Neanderthal            yes             528.0   \n",
       "1123317                True  Neanderthal            yes             510.0   \n",
       "897185                 True       Shared            yes             436.0   \n",
       "706199                 True       Shared            yes             444.0   \n",
       "1126124                True  Neanderthal            yes             439.0   \n",
       "1582579                True       Shared            yes             385.0   \n",
       "1221855                True       Shared            yes             379.0   \n",
       "611320                 True       Shared            yes             371.0   \n",
       "735814                False    Denisovan            yes            4744.0   \n",
       "1154162                True  Neanderthal            yes             326.0   \n",
       "1580224                True  Neanderthal            yes             324.0   \n",
       "235154                 True  Neanderthal            yes             282.0   \n",
       "\n",
       "         1KG_allele_number  1KG_allele_frequency  1KG_EAS_AF  1KG_EUR_AF  \\\n",
       "939756              5096.0                  0.22        0.51        0.21   \n",
       "246812              5096.0                  0.18        0.14        0.31   \n",
       "1058119             5096.0                  0.18        0.39        0.22   \n",
       "1267615             5096.0                  0.15        0.24        0.17   \n",
       "222914              5096.0                  0.14        0.31        0.09   \n",
       "541237              5096.0                  0.13        0.14        0.24   \n",
       "1507143             5096.0                  0.11        0.13        0.09   \n",
       "247163              5096.0                  0.12        0.19        0.19   \n",
       "593546              5096.0                  0.10        0.24        0.12   \n",
       "1123317             5096.0                  0.10        0.08        0.23   \n",
       "897185              5096.0                  0.09        0.14        0.00   \n",
       "706199              5096.0                  0.09        0.00        0.25   \n",
       "1126124             5096.0                  0.09        0.01        0.20   \n",
       "1582579             5096.0                  0.08        0.27        0.03   \n",
       "1221855             5096.0                  0.07        0.06        0.18   \n",
       "611320              5096.0                  0.07        0.26        0.07   \n",
       "735814              5096.0                  0.93        0.90        0.86   \n",
       "1154162             5096.0                  0.06        0.14        0.07   \n",
       "1580224             5096.0                  0.06        0.07        0.08   \n",
       "235154              5096.0                  0.06        0.00        0.10   \n",
       "\n",
       "         1KG_AFR_AF  1KG_AMR_AF  1KG_SAS_AF  1KG_non_ASW_AFR_AF  \\\n",
       "939756         0.01        0.11        0.31            0.000975   \n",
       "246812         0.02        0.21        0.28            0.003899   \n",
       "1058119        0.01        0.15        0.16            0.001949   \n",
       "1267615        0.00        0.09        0.29            0.001949   \n",
       "222914         0.00        0.11        0.20            0.001949   \n",
       "541237         0.01        0.20        0.13            0.003899   \n",
       "1507143        0.00        0.24        0.17            0.000000   \n",
       "247163         0.01        0.09        0.11            0.009747   \n",
       "593546         0.00        0.04        0.12            0.000000   \n",
       "1123317        0.00        0.09        0.12            0.000000   \n",
       "897185         0.01        0.27        0.10            0.000000   \n",
       "706199         0.01        0.12        0.10            0.003899   \n",
       "1126124        0.01        0.07        0.15            0.001949   \n",
       "1582579        0.00        0.07        0.03            0.000000   \n",
       "1221855        0.01        0.09        0.05            0.000975   \n",
       "611320         0.00        0.02        0.02            0.000000   \n",
       "735814         0.99        0.93        0.96            0.999025   \n",
       "1154162        0.01        0.11        0.03            0.000975   \n",
       "1580224        0.00        0.10        0.10            0.000000   \n",
       "235154         0.01        0.21        0.02            0.000975   \n",
       "\n",
       "        present_in_gnomAD  gnomAD_allele_count  gnomAD_allele_number  \\\n",
       "939756                yes              23692.0              143104.0   \n",
       "246812                yes              33636.0              143182.0   \n",
       "1058119               yes              19931.0              143224.0   \n",
       "1267615               yes              17455.0              143064.0   \n",
       "222914                yes              12162.0              143260.0   \n",
       "541237                yes              24329.0              143106.0   \n",
       "1507143               yes              12906.0              143208.0   \n",
       "247163                yes              17909.0              142902.0   \n",
       "593546                yes              11923.0              143216.0   \n",
       "1123317               yes              19710.0              143060.0   \n",
       "897185                yes               4268.0              143346.0   \n",
       "706199                yes              21727.0              143202.0   \n",
       "1126124               yes              19225.0              143000.0   \n",
       "1582579               yes               5035.0              143260.0   \n",
       "1221855               yes              18806.0              143144.0   \n",
       "611320                yes               8286.0              143188.0   \n",
       "735814                yes             129412.0              143248.0   \n",
       "1154162               yes               7014.0              143212.0   \n",
       "1580224               yes               8613.0              143190.0   \n",
       "235154                yes              14233.0              143242.0   \n",
       "\n",
       "         gnomAD_allele_frequency Vernot_introgressed Vernot_ancestral_allele  \\\n",
       "939756                  0.165558                 yes                       G   \n",
       "246812                  0.234918                 yes                       G   \n",
       "1058119                 0.139160                 yes                       G   \n",
       "1267615                 0.122008                 yes                       C   \n",
       "222914                  0.084895                 yes                       G   \n",
       "541237                  0.170007                 yes                       G   \n",
       "1507143                 0.090121                 yes                       C   \n",
       "247163                  0.125324                 yes                       A   \n",
       "593546                  0.083252                 yes                       T   \n",
       "1123317                 0.137774                 yes                       T   \n",
       "897185                  0.029774                 yes                       A   \n",
       "706199                  0.151723                 yes                       G   \n",
       "1126124                 0.134441                 yes                       G   \n",
       "1582579                 0.035146                 yes                       A   \n",
       "1221855                 0.131378                 yes                       G   \n",
       "611320                  0.057868                 yes                       G   \n",
       "735814                  0.903412                 yes                       C   \n",
       "1154162                 0.048976                 yes                       A   \n",
       "1580224                 0.060151                 yes                       A   \n",
       "235154                  0.099363                 yes                       A   \n",
       "\n",
       "        Vernot_derived_allele  Vernot_ancestral_derived_code  Vernot_AFA_AF  \\\n",
       "939756                      A                            1.0        0.03503   \n",
       "246812                      A                            1.0        0.05732   \n",
       "1058119                     C                            1.0        0.04459   \n",
       "1267615                     G                            1.0        0.01274   \n",
       "222914                      A                            1.0        0.00318   \n",
       "541237                      A                            1.0        0.04140   \n",
       "1507143                     T                            1.0        0.00955   \n",
       "247163                      G                            1.0        0.02866   \n",
       "593546                      A                            1.0        0.01274   \n",
       "1123317                     C                            1.0        0.01592   \n",
       "897185                      G                            1.0        0.02229   \n",
       "706199                      A                            1.0        0.01592   \n",
       "1126124                     A                            1.0        0.03503   \n",
       "1582579                     G                            1.0        0.00637   \n",
       "1221855                     A                            1.0        0.03503   \n",
       "611320                      C                            1.0        0.01592   \n",
       "735814                      T                            1.0        0.03503   \n",
       "1154162                     C                            1.0        0.02229   \n",
       "1580224                     T                            1.0        0.01274   \n",
       "235154                      G                            1.0        0.02229   \n",
       "\n",
       "         Vernot_AFR_AF  Vernot_AMR_AF  Vernot_EAS_AF  Vernot_EUR_AF  \\\n",
       "939756         0.00099        0.10663        0.51488        0.21173   \n",
       "246812         0.00397        0.21182        0.13988        0.31014   \n",
       "1058119        0.00198        0.14986        0.39484        0.22068   \n",
       "1267615        0.00198        0.09366        0.24206        0.16600   \n",
       "222914         0.00198        0.11095        0.31845        0.08946   \n",
       "541237         0.00397        0.20029        0.13889        0.23559   \n",
       "1507143        0.00000        0.23919        0.12897        0.09344   \n",
       "247163         0.00893        0.09222        0.18948        0.19682   \n",
       "593546         0.00000        0.04611        0.23909        0.12127   \n",
       "1123317        0.00000        0.08790        0.07837        0.22962   \n",
       "897185         0.00000        0.26513        0.13492        0.00398   \n",
       "706199         0.00397        0.11671        0.00198        0.24950   \n",
       "1126124        0.00198        0.07205        0.01091        0.19881   \n",
       "1582579        0.00000        0.07061        0.26587        0.03181   \n",
       "1221855        0.00099        0.09078        0.06448        0.17992   \n",
       "611320         0.00000        0.02017        0.25893        0.06461   \n",
       "735814         0.00099        0.06916        0.09821        0.14612   \n",
       "1154162        0.00099        0.11239        0.13889        0.06660   \n",
       "1580224        0.00000        0.09654        0.07143        0.07555   \n",
       "235154         0.00099        0.21326        0.00198        0.09742   \n",
       "\n",
       "         Vernot_PNG_AF  Vernot_SAS_AF Vernot_Denisovan_base  \\\n",
       "939756         0.03704        0.30982                     G   \n",
       "246812         0.09259        0.28323                     G   \n",
       "1058119        0.05556        0.16360                     G   \n",
       "1267615        0.09259        0.29448                     C   \n",
       "222914         0.11111        0.19632                     G   \n",
       "541237         0.00000        0.12883                     G   \n",
       "1507143        0.05556        0.17280                     C   \n",
       "247163         0.18519        0.11043                     A   \n",
       "593546         0.16667        0.12474                     T   \n",
       "1123317        0.14815        0.12372                     T   \n",
       "897185         0.64815        0.09816                     G   \n",
       "706199         0.09259        0.09714                     A   \n",
       "1126124        0.16667        0.15235                     G   \n",
       "1582579        0.05556        0.02863                     G   \n",
       "1221855        0.16667        0.05215                     A   \n",
       "611320         0.20370        0.01636                     C   \n",
       "735814         0.07407        0.04090                     C   \n",
       "1154162        0.01852        0.02863                     A   \n",
       "1580224        0.00000        0.10327                     A   \n",
       "235154         0.00000        0.02352                     A   \n",
       "\n",
       "        Vernot_Neanderthal_base_1 Vernot_Neanderthal_base_2  \\\n",
       "939756                          G                         A   \n",
       "246812                          A                       NaN   \n",
       "1058119                         G                         C   \n",
       "1267615                         C                         G   \n",
       "222914                          A                       NaN   \n",
       "541237                          A                       NaN   \n",
       "1507143                         T                       NaN   \n",
       "247163                          G                       NaN   \n",
       "593546                          A                       NaN   \n",
       "1123317                         C                       NaN   \n",
       "897185                          G                       NaN   \n",
       "706199                          G                         A   \n",
       "1126124                         A                       NaN   \n",
       "1582579                         G                       NaN   \n",
       "1221855                         A                       NaN   \n",
       "611320                          C                       NaN   \n",
       "735814                          T                       NaN   \n",
       "1154162                         C                       NaN   \n",
       "1580224                         T                       NaN   \n",
       "235154                          G                       NaN   \n",
       "\n",
       "              Vernot_haplotype_tag  Vernot_introgressed_AF  \\\n",
       "939756     chr20_62195671_62229244                0.228810   \n",
       "246812     chr11_11494384_11530287                0.189808   \n",
       "1058119     chr4_38294200_38857896                0.186192   \n",
       "1267615     chr6_52138225_52183126                0.159636   \n",
       "222914   chr10_121002922_121217061                0.143432   \n",
       "541237     chr15_84263508_85447079                0.141514   \n",
       "1507143     chr8_98910893_99084133                0.126880   \n",
       "247163     chr11_11691675_12000532                0.119576   \n",
       "593546     chr16_77304807_77417155                0.106242   \n",
       "1123317   chr4_162928454_163168783                0.103922   \n",
       "897185    chr2_242085733_242430170                0.100438   \n",
       "706199     chr18_59569521_60089689                0.093860   \n",
       "1126124   chr4_169073736_169412749                0.087220   \n",
       "1582579   chr9_112193017_112358561                0.079384   \n",
       "1221855   chr5_167873398_168164228                0.077664   \n",
       "611320     chr16_89785602_90065941                0.072014   \n",
       "735814     chr19_33040186_33456652                0.071076   \n",
       "1154162     chr5_35828521_36296191                0.069500   \n",
       "1580224   chr9_104256268_104545680                0.069358   \n",
       "235154         chr11_437558_508720                0.067434   \n",
       "\n",
       "        Browning_introgressed  Browning_ref_alt  Browning_introgressed_AF  \\\n",
       "939756                    yes               1.0                      0.22   \n",
       "246812                    yes               1.0                      0.18   \n",
       "1058119                   yes               1.0                      0.18   \n",
       "1267615                    no               NaN                       NaN   \n",
       "222914                    yes               1.0                      0.14   \n",
       "541237                    yes               1.0                      0.13   \n",
       "1507143                   yes               1.0                      0.11   \n",
       "247163                    yes               1.0                      0.12   \n",
       "593546                    yes               1.0                      0.10   \n",
       "1123317                   yes               1.0                      0.10   \n",
       "897185                    yes               1.0                      0.09   \n",
       "706199                    yes               1.0                      0.09   \n",
       "1126124                   yes               1.0                      0.09   \n",
       "1582579                   yes               1.0                      0.08   \n",
       "1221855                   yes               1.0                      0.07   \n",
       "611320                    yes               1.0                      0.07   \n",
       "735814                    yes               0.0                      0.07   \n",
       "1154162                   yes               1.0                      0.06   \n",
       "1580224                   yes               1.0                      0.06   \n",
       "235154                    yes               1.0                      0.06   \n",
       "\n",
       "        annotation   mis_oe     mis_z    lof_oe    lof_z  phyloP  ag_delta  \\\n",
       "939756       GMEB2  0.66048  2.202000  0.049787  3.94620  -0.837      0.01   \n",
       "246812     GALNT18  0.87026  0.879900  0.387360  3.02520  -0.302      0.00   \n",
       "1058119       TLR1  1.09920 -0.701570  1.411200 -1.72720  -0.219      0.00   \n",
       "1267615       MCM3  0.89914  0.804510  0.535230  2.88400   0.563      0.21   \n",
       "222914        GRK5  0.68002  2.158300  0.203620  4.32700  -0.355      0.23   \n",
       "541237       ALPK3  0.98871  0.131400  0.716100  2.15380   0.462      0.00   \n",
       "1507143      MATN2  0.79916  1.668900  0.638200  2.22070  -0.373      0.00   \n",
       "247163       USP47  0.73421  2.437500  0.056706  7.34150   0.383      0.00   \n",
       "593546    ADAMTS18  1.38300 -3.508200  0.742900  2.06850   0.455      0.00   \n",
       "1123317      FSTL5  0.99361  0.047046  0.419990  3.41950  -0.415      0.00   \n",
       "897185       FARP2  1.03750 -0.330770  0.947030  0.37745   0.477      0.32   \n",
       "706199   TNFRSF11A  0.80508  1.289400  0.330780  3.04970   0.650      0.00   \n",
       "1126124     DDX60L  0.98853  0.114220  0.820340  1.50460   0.467      0.00   \n",
       "1582579      PTPN3  0.91465  0.705600  0.406750  4.30980   0.455      0.00   \n",
       "1221855       RARS  0.91791  0.543820  0.516620  2.64400  -0.471      0.00   \n",
       "611320       FANCA  1.54640 -5.413800  1.152500 -1.28990   0.491      0.21   \n",
       "735814       CEP89  0.93498  0.474930  0.939730  0.36440  -0.642      0.47   \n",
       "1154162     LMBRD2  0.66428  2.272300  0.350190  3.80740  -0.291      0.00   \n",
       "1580224     GRIN3A  0.91452  0.742910  0.220950  4.60750   0.459      0.42   \n",
       "235154      PTDSS2  0.73242  1.571300  0.448660  2.75010   0.334      0.00   \n",
       "\n",
       "         al_delta  dg_delta  dl_delta  delta_max  ag_pos  al_pos  dg_pos  \\\n",
       "939756       0.00      0.41      0.02       0.41      15     -44       2   \n",
       "246812       0.00      0.21      0.00       0.21      37     -14       2   \n",
       "1058119      0.00      0.00      0.38       0.38      27     -15       1   \n",
       "1267615      0.00      0.00      0.00       0.21      -7     -40      -7   \n",
       "222914       0.00      0.00      0.00       0.23     -23       1       4   \n",
       "541237       0.00      0.33      0.00       0.33      15      50      15   \n",
       "1507143      0.00      0.00      0.24       0.24      43       8      -2   \n",
       "247163       0.35      0.00      0.00       0.35     -13      18      -1   \n",
       "593546       0.00      0.41      0.00       0.41      23     -32       2   \n",
       "1123317      0.00      0.20      0.00       0.20       1     -48       1   \n",
       "897185       0.00      0.00      0.00       0.32       1       2       1   \n",
       "706199       0.31      0.00      0.28       0.31       3     -40       3   \n",
       "1126124      0.31      0.00      0.00       0.31     -35      -3       2   \n",
       "1582579      0.01      0.00      0.30       0.30     -35     -47     -47   \n",
       "1221855      0.00      0.38      0.01       0.38     -37       1       1   \n",
       "611320       0.00      0.00      0.00       0.21       0     -34       1   \n",
       "735814       0.00      0.71      0.00       0.71      41     -29       5   \n",
       "1154162      0.22      0.00      0.00       0.22      21     -25      41   \n",
       "1580224      0.00      0.00      0.00       0.42       9     -37      -2   \n",
       "235154       0.07      0.30      0.01       0.30      -5      -4      -5   \n",
       "\n",
       "         dl_pos  Adipose_Subcutaneous  Adipose_Visceral_Omentum  \\\n",
       "939756       -2                   0.0                       0.0   \n",
       "246812      -14                   NaN                       NaN   \n",
       "1058119     -15                   1.0                       1.0   \n",
       "1267615       3                   NaN                       NaN   \n",
       "222914      -14                   0.0                       0.0   \n",
       "541237      -11                   1.0                       0.0   \n",
       "1507143      43                   NaN                       NaN   \n",
       "247163       18                   NaN                       NaN   \n",
       "593546      -32                   NaN                       NaN   \n",
       "1123317     -48                   0.0                       0.0   \n",
       "897185       44                   0.0                       0.0   \n",
       "706199       33                   1.0                       1.0   \n",
       "1126124     -33                   0.0                       0.0   \n",
       "1582579       6                   NaN                       NaN   \n",
       "1221855      27                   0.0                       0.0   \n",
       "611320       43                   1.0                       1.0   \n",
       "735814      -29                   NaN                       NaN   \n",
       "1154162     -25                   NaN                       NaN   \n",
       "1580224       2                   NaN                       NaN   \n",
       "235154      -39                   1.0                       1.0   \n",
       "\n",
       "         Adrenal_Gland  Artery_Aorta  Artery_Coronary  Artery_Tibial  \\\n",
       "939756             1.0           0.0              1.0            1.0   \n",
       "246812             NaN           NaN              NaN            NaN   \n",
       "1058119            0.0           0.0              0.0            0.0   \n",
       "1267615            NaN           NaN              NaN            NaN   \n",
       "222914             0.0           0.0              0.0            1.0   \n",
       "541237             1.0           1.0              1.0            1.0   \n",
       "1507143            NaN           NaN              NaN            NaN   \n",
       "247163             NaN           NaN              NaN            NaN   \n",
       "593546             NaN           NaN              NaN            NaN   \n",
       "1123317            0.0           0.0              0.0            0.0   \n",
       "897185             0.0           0.0              0.0            0.0   \n",
       "706199             0.0           1.0              0.0            1.0   \n",
       "1126124            0.0           0.0              0.0            0.0   \n",
       "1582579            NaN           NaN              NaN            NaN   \n",
       "1221855            0.0           0.0              0.0            0.0   \n",
       "611320             1.0           1.0              1.0            1.0   \n",
       "735814             NaN           NaN              NaN            NaN   \n",
       "1154162            NaN           NaN              NaN            NaN   \n",
       "1580224            NaN           NaN              NaN            NaN   \n",
       "235154             0.0           1.0              0.0            1.0   \n",
       "\n",
       "         Brain_Amygdala  Brain_Anterior_cingulate_cortex_BA24  \\\n",
       "939756              0.0                                   0.0   \n",
       "246812              NaN                                   NaN   \n",
       "1058119             0.0                                   0.0   \n",
       "1267615             NaN                                   NaN   \n",
       "222914              0.0                                   0.0   \n",
       "541237              0.0                                   0.0   \n",
       "1507143             NaN                                   NaN   \n",
       "247163              NaN                                   NaN   \n",
       "593546              NaN                                   NaN   \n",
       "1123317             0.0                                   0.0   \n",
       "897185              0.0                                   0.0   \n",
       "706199              0.0                                   0.0   \n",
       "1126124             0.0                                   0.0   \n",
       "1582579             NaN                                   NaN   \n",
       "1221855             0.0                                   0.0   \n",
       "611320              0.0                                   0.0   \n",
       "735814              NaN                                   NaN   \n",
       "1154162             NaN                                   NaN   \n",
       "1580224             NaN                                   NaN   \n",
       "235154              0.0                                   0.0   \n",
       "\n",
       "         Brain_Caudate_basal_ganglia  Brain_Cerebellar_Hemisphere  \\\n",
       "939756                           1.0                          1.0   \n",
       "246812                           NaN                          NaN   \n",
       "1058119                          0.0                          0.0   \n",
       "1267615                          NaN                          NaN   \n",
       "222914                           0.0                          0.0   \n",
       "541237                           0.0                          0.0   \n",
       "1507143                          NaN                          NaN   \n",
       "247163                           NaN                          NaN   \n",
       "593546                           NaN                          NaN   \n",
       "1123317                          0.0                          1.0   \n",
       "897185                           0.0                          0.0   \n",
       "706199                           0.0                          0.0   \n",
       "1126124                          1.0                          0.0   \n",
       "1582579                          NaN                          NaN   \n",
       "1221855                          0.0                          1.0   \n",
       "611320                           0.0                          1.0   \n",
       "735814                           NaN                          NaN   \n",
       "1154162                          NaN                          NaN   \n",
       "1580224                          NaN                          NaN   \n",
       "235154                           0.0                          0.0   \n",
       "\n",
       "         Brain_Cerebellum  Brain_Cortex  Brain_Frontal_Cortex_BA9  \\\n",
       "939756                1.0           1.0                       1.0   \n",
       "246812                NaN           NaN                       NaN   \n",
       "1058119               0.0           0.0                       0.0   \n",
       "1267615               NaN           NaN                       NaN   \n",
       "222914                0.0           0.0                       0.0   \n",
       "541237                0.0           0.0                       0.0   \n",
       "1507143               NaN           NaN                       NaN   \n",
       "247163                NaN           NaN                       NaN   \n",
       "593546                NaN           NaN                       NaN   \n",
       "1123317               1.0           0.0                       0.0   \n",
       "897185                0.0           0.0                       0.0   \n",
       "706199                0.0           1.0                       0.0   \n",
       "1126124               0.0           0.0                       0.0   \n",
       "1582579               NaN           NaN                       NaN   \n",
       "1221855               0.0           0.0                       0.0   \n",
       "611320                1.0           1.0                       1.0   \n",
       "735814                NaN           NaN                       NaN   \n",
       "1154162               NaN           NaN                       NaN   \n",
       "1580224               NaN           NaN                       NaN   \n",
       "235154                0.0           0.0                       0.0   \n",
       "\n",
       "         Brain_Hippocampus  Brain_Hypothalamus  \\\n",
       "939756                 0.0                 1.0   \n",
       "246812                 NaN                 NaN   \n",
       "1058119                0.0                 0.0   \n",
       "1267615                NaN                 NaN   \n",
       "222914                 0.0                 0.0   \n",
       "541237                 0.0                 0.0   \n",
       "1507143                NaN                 NaN   \n",
       "247163                 NaN                 NaN   \n",
       "593546                 NaN                 NaN   \n",
       "1123317                0.0                 0.0   \n",
       "897185                 0.0                 0.0   \n",
       "706199                 0.0                 0.0   \n",
       "1126124                0.0                 0.0   \n",
       "1582579                NaN                 NaN   \n",
       "1221855                0.0                 0.0   \n",
       "611320                 0.0                 1.0   \n",
       "735814                 NaN                 NaN   \n",
       "1154162                NaN                 NaN   \n",
       "1580224                NaN                 NaN   \n",
       "235154                 0.0                 0.0   \n",
       "\n",
       "         Brain_Nucleus_accumbens_basal_ganglia  Brain_Putamen_basal_ganglia  \\\n",
       "939756                                     1.0                          0.0   \n",
       "246812                                     NaN                          NaN   \n",
       "1058119                                    0.0                          0.0   \n",
       "1267615                                    NaN                          NaN   \n",
       "222914                                     0.0                          0.0   \n",
       "541237                                     0.0                          0.0   \n",
       "1507143                                    NaN                          NaN   \n",
       "247163                                     NaN                          NaN   \n",
       "593546                                     NaN                          NaN   \n",
       "1123317                                    0.0                          0.0   \n",
       "897185                                     0.0                          0.0   \n",
       "706199                                     0.0                          0.0   \n",
       "1126124                                    1.0                          0.0   \n",
       "1582579                                    NaN                          NaN   \n",
       "1221855                                    0.0                          0.0   \n",
       "611320                                     0.0                          1.0   \n",
       "735814                                     NaN                          NaN   \n",
       "1154162                                    NaN                          NaN   \n",
       "1580224                                    NaN                          NaN   \n",
       "235154                                     0.0                          0.0   \n",
       "\n",
       "         Brain_Spinal_cord_cervical_c-1  Brain_Substantia_nigra  \\\n",
       "939756                              0.0                     0.0   \n",
       "246812                              NaN                     NaN   \n",
       "1058119                             0.0                     0.0   \n",
       "1267615                             NaN                     NaN   \n",
       "222914                              0.0                     0.0   \n",
       "541237                              0.0                     0.0   \n",
       "1507143                             NaN                     NaN   \n",
       "247163                              NaN                     NaN   \n",
       "593546                              NaN                     NaN   \n",
       "1123317                             0.0                     0.0   \n",
       "897185                              0.0                     0.0   \n",
       "706199                              0.0                     0.0   \n",
       "1126124                             0.0                     0.0   \n",
       "1582579                             NaN                     NaN   \n",
       "1221855                             0.0                     0.0   \n",
       "611320                              0.0                     0.0   \n",
       "735814                              NaN                     NaN   \n",
       "1154162                             NaN                     NaN   \n",
       "1580224                             NaN                     NaN   \n",
       "235154                              0.0                     0.0   \n",
       "\n",
       "         Breast_Mammary_Tissue  Cells_Cultured_fibroblasts  \\\n",
       "939756                     0.0                         0.0   \n",
       "246812                     NaN                         NaN   \n",
       "1058119                    0.0                         0.0   \n",
       "1267615                    NaN                         NaN   \n",
       "222914                     0.0                         0.0   \n",
       "541237                     1.0                         0.0   \n",
       "1507143                    NaN                         NaN   \n",
       "247163                     NaN                         NaN   \n",
       "593546                     NaN                         NaN   \n",
       "1123317                    0.0                         0.0   \n",
       "897185                     0.0                         0.0   \n",
       "706199                     1.0                         1.0   \n",
       "1126124                    0.0                         0.0   \n",
       "1582579                    NaN                         NaN   \n",
       "1221855                    0.0                         0.0   \n",
       "611320                     1.0                         1.0   \n",
       "735814                     NaN                         NaN   \n",
       "1154162                    NaN                         NaN   \n",
       "1580224                    NaN                         NaN   \n",
       "235154                     1.0                         0.0   \n",
       "\n",
       "         Cells_EBV-transformed_lymphocytes  Colon_Sigmoid  Colon_Transverse  \\\n",
       "939756                                 0.0            1.0               1.0   \n",
       "246812                                 NaN            NaN               NaN   \n",
       "1058119                                1.0            0.0               0.0   \n",
       "1267615                                NaN            NaN               NaN   \n",
       "222914                                 0.0            0.0               0.0   \n",
       "541237                                 0.0            1.0               1.0   \n",
       "1507143                                NaN            NaN               NaN   \n",
       "247163                                 NaN            NaN               NaN   \n",
       "593546                                 NaN            NaN               NaN   \n",
       "1123317                                0.0            0.0               0.0   \n",
       "897185                                 0.0            0.0               0.0   \n",
       "706199                                 0.0            1.0               1.0   \n",
       "1126124                                0.0            0.0               0.0   \n",
       "1582579                                NaN            NaN               NaN   \n",
       "1221855                                0.0            0.0               0.0   \n",
       "611320                                 1.0            1.0               1.0   \n",
       "735814                                 NaN            NaN               NaN   \n",
       "1154162                                NaN            NaN               NaN   \n",
       "1580224                                NaN            NaN               NaN   \n",
       "235154                                 1.0            0.0               0.0   \n",
       "\n",
       "         Esophagus_Gastroesophageal_Junction  Esophagus_Mucosa  \\\n",
       "939756                                   1.0               0.0   \n",
       "246812                                   NaN               NaN   \n",
       "1058119                                  0.0               0.0   \n",
       "1267615                                  NaN               NaN   \n",
       "222914                                   0.0               0.0   \n",
       "541237                                   1.0               0.0   \n",
       "1507143                                  NaN               NaN   \n",
       "247163                                   NaN               NaN   \n",
       "593546                                   NaN               NaN   \n",
       "1123317                                  0.0               0.0   \n",
       "897185                                   0.0               0.0   \n",
       "706199                                   1.0               1.0   \n",
       "1126124                                  0.0               0.0   \n",
       "1582579                                  NaN               NaN   \n",
       "1221855                                  0.0               1.0   \n",
       "611320                                   1.0               1.0   \n",
       "735814                                   NaN               NaN   \n",
       "1154162                                  NaN               NaN   \n",
       "1580224                                  NaN               NaN   \n",
       "235154                                   0.0               1.0   \n",
       "\n",
       "         Esophagus_Muscularis  Heart_Atrial_Appendage  Heart_Left_Ventricle  \\\n",
       "939756                    1.0                     0.0                   0.0   \n",
       "246812                    NaN                     NaN                   NaN   \n",
       "1058119                   0.0                     0.0                   0.0   \n",
       "1267615                   NaN                     NaN                   NaN   \n",
       "222914                    0.0                     0.0                   0.0   \n",
       "541237                    1.0                     1.0                   0.0   \n",
       "1507143                   NaN                     NaN                   NaN   \n",
       "247163                    NaN                     NaN                   NaN   \n",
       "593546                    NaN                     NaN                   NaN   \n",
       "1123317                   0.0                     0.0                   0.0   \n",
       "897185                    0.0                     0.0                   0.0   \n",
       "706199                    1.0                     0.0                   0.0   \n",
       "1126124                   0.0                     0.0                   0.0   \n",
       "1582579                   NaN                     NaN                   NaN   \n",
       "1221855                   0.0                     0.0                   0.0   \n",
       "611320                    1.0                     1.0                   1.0   \n",
       "735814                    NaN                     NaN                   NaN   \n",
       "1154162                   NaN                     NaN                   NaN   \n",
       "1580224                   NaN                     NaN                   NaN   \n",
       "235154                    0.0                     0.0                   1.0   \n",
       "\n",
       "         Kidney_Cortex  Liver  Lung  Minor_Salivary_Gland  Muscle_Skeletal  \\\n",
       "939756             0.0    0.0   1.0                   0.0              0.0   \n",
       "246812             NaN    NaN   NaN                   NaN              NaN   \n",
       "1058119            0.0    0.0   1.0                   0.0              0.0   \n",
       "1267615            NaN    NaN   NaN                   NaN              NaN   \n",
       "222914             0.0    0.0   0.0                   0.0              0.0   \n",
       "541237             0.0    0.0   0.0                   0.0              0.0   \n",
       "1507143            NaN    NaN   NaN                   NaN              NaN   \n",
       "247163             NaN    NaN   NaN                   NaN              NaN   \n",
       "593546             NaN    NaN   NaN                   NaN              NaN   \n",
       "1123317            0.0    0.0   0.0                   0.0              0.0   \n",
       "897185             0.0    0.0   0.0                   0.0              0.0   \n",
       "706199             0.0    0.0   1.0                   0.0              0.0   \n",
       "1126124            0.0    0.0   0.0                   0.0              0.0   \n",
       "1582579            NaN    NaN   NaN                   NaN              NaN   \n",
       "1221855            0.0    0.0   0.0                   0.0              0.0   \n",
       "611320             0.0    1.0   1.0                   1.0              1.0   \n",
       "735814             NaN    NaN   NaN                   NaN              NaN   \n",
       "1154162            NaN    NaN   NaN                   NaN              NaN   \n",
       "1580224            NaN    NaN   NaN                   NaN              NaN   \n",
       "235154             0.0    0.0   1.0                   1.0              1.0   \n",
       "\n",
       "         Nerve_Tibial  Ovary  Pancreas  Pituitary  Prostate  \\\n",
       "939756            1.0    1.0       0.0        0.0       1.0   \n",
       "246812            NaN    NaN       NaN        NaN       NaN   \n",
       "1058119           1.0    0.0       0.0        0.0       0.0   \n",
       "1267615           NaN    NaN       NaN        NaN       NaN   \n",
       "222914            0.0    0.0       0.0        0.0       0.0   \n",
       "541237            1.0    1.0       0.0        1.0       1.0   \n",
       "1507143           NaN    NaN       NaN        NaN       NaN   \n",
       "247163            NaN    NaN       NaN        NaN       NaN   \n",
       "593546            NaN    NaN       NaN        NaN       NaN   \n",
       "1123317           0.0    0.0       0.0        0.0       0.0   \n",
       "897185            0.0    0.0       0.0        0.0       0.0   \n",
       "706199            1.0    0.0       0.0        1.0       0.0   \n",
       "1126124           0.0    0.0       0.0        0.0       0.0   \n",
       "1582579           NaN    NaN       NaN        NaN       NaN   \n",
       "1221855           1.0    0.0       0.0        1.0       0.0   \n",
       "611320            1.0    1.0       1.0        1.0       1.0   \n",
       "735814            NaN    NaN       NaN        NaN       NaN   \n",
       "1154162           NaN    NaN       NaN        NaN       NaN   \n",
       "1580224           NaN    NaN       NaN        NaN       NaN   \n",
       "235154            1.0    0.0       1.0        0.0       0.0   \n",
       "\n",
       "         Skin_Not_Sun_Exposed_Suprapubic  Skin_Sun_Exposed_Lower_leg  \\\n",
       "939756                               0.0                         0.0   \n",
       "246812                               NaN                         NaN   \n",
       "1058119                              0.0                         0.0   \n",
       "1267615                              NaN                         NaN   \n",
       "222914                               0.0                         0.0   \n",
       "541237                               0.0                         0.0   \n",
       "1507143                              NaN                         NaN   \n",
       "247163                               NaN                         NaN   \n",
       "593546                               NaN                         NaN   \n",
       "1123317                              0.0                         0.0   \n",
       "897185                               0.0                         0.0   \n",
       "706199                               1.0                         1.0   \n",
       "1126124                              0.0                         0.0   \n",
       "1582579                              NaN                         NaN   \n",
       "1221855                              0.0                         0.0   \n",
       "611320                               1.0                         1.0   \n",
       "735814                               NaN                         NaN   \n",
       "1154162                              NaN                         NaN   \n",
       "1580224                              NaN                         NaN   \n",
       "235154                               1.0                         1.0   \n",
       "\n",
       "         Small_Intestine_Terminal_Ileum  Spleen  Stomach  Testis  Thyroid  \\\n",
       "939756                              1.0     0.0      0.0     1.0      1.0   \n",
       "246812                              NaN     NaN      NaN     NaN      NaN   \n",
       "1058119                             0.0     1.0      0.0     0.0      0.0   \n",
       "1267615                             NaN     NaN      NaN     NaN      NaN   \n",
       "222914                              0.0     0.0      0.0     0.0      0.0   \n",
       "541237                              0.0     0.0      1.0     1.0      0.0   \n",
       "1507143                             NaN     NaN      NaN     NaN      NaN   \n",
       "247163                              NaN     NaN      NaN     NaN      NaN   \n",
       "593546                              NaN     NaN      NaN     NaN      NaN   \n",
       "1123317                             0.0     0.0      0.0     0.0      0.0   \n",
       "897185                              1.0     0.0      0.0     0.0      0.0   \n",
       "706199                              0.0     0.0      1.0     1.0      1.0   \n",
       "1126124                             0.0     0.0      0.0     0.0      0.0   \n",
       "1582579                             NaN     NaN      NaN     NaN      NaN   \n",
       "1221855                             0.0     0.0      0.0     1.0      0.0   \n",
       "611320                              1.0     1.0      1.0     1.0      1.0   \n",
       "735814                              NaN     NaN      NaN     NaN      NaN   \n",
       "1154162                             NaN     NaN      NaN     NaN      NaN   \n",
       "1580224                             NaN     NaN      NaN     NaN      NaN   \n",
       "235154                              0.0     1.0      0.0     1.0      1.0   \n",
       "\n",
       "         Uterus  Vagina  Whole_Blood  N_GTEx_tissues sQTL  \\\n",
       "939756      0.0     0.0          0.0            21.0  yes   \n",
       "246812      NaN     NaN          NaN             NaN   no   \n",
       "1058119     0.0     0.0          1.0             7.0  yes   \n",
       "1267615     NaN     NaN          NaN             NaN   no   \n",
       "222914      0.0     0.0          0.0             1.0  yes   \n",
       "541237      0.0     0.0          0.0            17.0  yes   \n",
       "1507143     NaN     NaN          NaN             NaN   no   \n",
       "247163      NaN     NaN          NaN             NaN   no   \n",
       "593546      NaN     NaN          NaN             NaN   no   \n",
       "1123317     0.0     0.0          0.0             2.0  yes   \n",
       "897185      0.0     0.0          0.0             1.0  yes   \n",
       "706199      0.0     0.0          0.0            20.0  yes   \n",
       "1126124     0.0     0.0          0.0             2.0  yes   \n",
       "1582579     NaN     NaN          NaN             NaN   no   \n",
       "1221855     0.0     0.0          0.0             5.0  yes   \n",
       "611320      0.0     0.0          1.0            39.0  yes   \n",
       "735814      NaN     NaN          NaN             NaN   no   \n",
       "1154162     NaN     NaN          NaN             NaN   no   \n",
       "1580224     NaN     NaN          NaN             NaN   no   \n",
       "235154      0.0     0.0          1.0            19.0  yes   \n",
       "\n",
       "        Vernot_allele_origin Browning_allele_origin  \n",
       "939756          introgressed           introgressed  \n",
       "246812          introgressed           introgressed  \n",
       "1058119         introgressed           introgressed  \n",
       "1267615         introgressed                ancient  \n",
       "222914          introgressed           introgressed  \n",
       "541237          introgressed           introgressed  \n",
       "1507143         introgressed           introgressed  \n",
       "247163          introgressed           introgressed  \n",
       "593546          introgressed           introgressed  \n",
       "1123317         introgressed           introgressed  \n",
       "897185          introgressed           introgressed  \n",
       "706199          introgressed           introgressed  \n",
       "1126124         introgressed           introgressed  \n",
       "1582579         introgressed           introgressed  \n",
       "1221855         introgressed           introgressed  \n",
       "611320          introgressed           introgressed  \n",
       "735814          introgressed           introgressed  \n",
       "1154162         introgressed           introgressed  \n",
       "1580224         introgressed           introgressed  \n",
       "235154          introgressed           introgressed  "
      ]
     },
     "execution_count": 496,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SA_2_introgressed = data_2[data_2['Vernot_allele_origin']=='introgressed']\n",
    "SA_2_introgressed_AF = SA_2_introgressed.sort_values(['Vernot_introgressed_AF'], ascending = [False])\n",
    "SA_2_introgressed_AF.head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 497,
   "id": "3d275b1e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref_allele</th>\n",
       "      <th>alt_allele</th>\n",
       "      <th>ancestral_allele</th>\n",
       "      <th>anc_dev</th>\n",
       "      <th>variant_type</th>\n",
       "      <th>altai_gt</th>\n",
       "      <th>chagyrskaya_gt</th>\n",
       "      <th>denisovan_gt</th>\n",
       "      <th>vindija_gt</th>\n",
       "      <th>altai_gt_boolean</th>\n",
       "      <th>chagyrskaya_gt_boolean</th>\n",
       "      <th>denisovan_gt_boolean</th>\n",
       "      <th>vindija_gt_boolean</th>\n",
       "      <th>distribution</th>\n",
       "      <th>present_in_1KG</th>\n",
       "      <th>1KG_allele_count</th>\n",
       "      <th>1KG_allele_number</th>\n",
       "      <th>1KG_allele_frequency</th>\n",
       "      <th>1KG_EAS_AF</th>\n",
       "      <th>1KG_EUR_AF</th>\n",
       "      <th>1KG_AFR_AF</th>\n",
       "      <th>1KG_AMR_AF</th>\n",
       "      <th>1KG_SAS_AF</th>\n",
       "      <th>1KG_non_ASW_AFR_AF</th>\n",
       "      <th>present_in_gnomAD</th>\n",
       "      <th>gnomAD_allele_count</th>\n",
       "      <th>gnomAD_allele_number</th>\n",
       "      <th>gnomAD_allele_frequency</th>\n",
       "      <th>Vernot_introgressed</th>\n",
       "      <th>Vernot_ancestral_allele</th>\n",
       "      <th>Vernot_derived_allele</th>\n",
       "      <th>Vernot_ancestral_derived_code</th>\n",
       "      <th>Vernot_AFA_AF</th>\n",
       "      <th>Vernot_AFR_AF</th>\n",
       "      <th>Vernot_AMR_AF</th>\n",
       "      <th>Vernot_EAS_AF</th>\n",
       "      <th>Vernot_EUR_AF</th>\n",
       "      <th>Vernot_PNG_AF</th>\n",
       "      <th>Vernot_SAS_AF</th>\n",
       "      <th>Vernot_Denisovan_base</th>\n",
       "      <th>Vernot_Neanderthal_base_1</th>\n",
       "      <th>Vernot_Neanderthal_base_2</th>\n",
       "      <th>Vernot_haplotype_tag</th>\n",
       "      <th>Vernot_introgressed_AF</th>\n",
       "      <th>Browning_introgressed</th>\n",
       "      <th>Browning_ref_alt</th>\n",
       "      <th>Browning_introgressed_AF</th>\n",
       "      <th>annotation</th>\n",
       "      <th>mis_oe</th>\n",
       "      <th>mis_z</th>\n",
       "      <th>lof_oe</th>\n",
       "      <th>lof_z</th>\n",
       "      <th>phyloP</th>\n",
       "      <th>ag_delta</th>\n",
       "      <th>al_delta</th>\n",
       "      <th>dg_delta</th>\n",
       "      <th>dl_delta</th>\n",
       "      <th>delta_max</th>\n",
       "      <th>ag_pos</th>\n",
       "      <th>al_pos</th>\n",
       "      <th>dg_pos</th>\n",
       "      <th>dl_pos</th>\n",
       "      <th>Adipose_Subcutaneous</th>\n",
       "      <th>Adipose_Visceral_Omentum</th>\n",
       "      <th>Adrenal_Gland</th>\n",
       "      <th>Artery_Aorta</th>\n",
       "      <th>Artery_Coronary</th>\n",
       "      <th>Artery_Tibial</th>\n",
       "      <th>Brain_Amygdala</th>\n",
       "      <th>Brain_Anterior_cingulate_cortex_BA24</th>\n",
       "      <th>Brain_Caudate_basal_ganglia</th>\n",
       "      <th>Brain_Cerebellar_Hemisphere</th>\n",
       "      <th>Brain_Cerebellum</th>\n",
       "      <th>Brain_Cortex</th>\n",
       "      <th>Brain_Frontal_Cortex_BA9</th>\n",
       "      <th>Brain_Hippocampus</th>\n",
       "      <th>Brain_Hypothalamus</th>\n",
       "      <th>Brain_Nucleus_accumbens_basal_ganglia</th>\n",
       "      <th>Brain_Putamen_basal_ganglia</th>\n",
       "      <th>Brain_Spinal_cord_cervical_c-1</th>\n",
       "      <th>Brain_Substantia_nigra</th>\n",
       "      <th>Breast_Mammary_Tissue</th>\n",
       "      <th>Cells_Cultured_fibroblasts</th>\n",
       "      <th>Cells_EBV-transformed_lymphocytes</th>\n",
       "      <th>Colon_Sigmoid</th>\n",
       "      <th>Colon_Transverse</th>\n",
       "      <th>Esophagus_Gastroesophageal_Junction</th>\n",
       "      <th>Esophagus_Mucosa</th>\n",
       "      <th>Esophagus_Muscularis</th>\n",
       "      <th>Heart_Atrial_Appendage</th>\n",
       "      <th>Heart_Left_Ventricle</th>\n",
       "      <th>Kidney_Cortex</th>\n",
       "      <th>Liver</th>\n",
       "      <th>Lung</th>\n",
       "      <th>Minor_Salivary_Gland</th>\n",
       "      <th>Muscle_Skeletal</th>\n",
       "      <th>Nerve_Tibial</th>\n",
       "      <th>Ovary</th>\n",
       "      <th>Pancreas</th>\n",
       "      <th>Pituitary</th>\n",
       "      <th>Prostate</th>\n",
       "      <th>Skin_Not_Sun_Exposed_Suprapubic</th>\n",
       "      <th>Skin_Sun_Exposed_Lower_leg</th>\n",
       "      <th>Small_Intestine_Terminal_Ileum</th>\n",
       "      <th>Spleen</th>\n",
       "      <th>Stomach</th>\n",
       "      <th>Testis</th>\n",
       "      <th>Thyroid</th>\n",
       "      <th>Uterus</th>\n",
       "      <th>Vagina</th>\n",
       "      <th>Whole_Blood</th>\n",
       "      <th>N_GTEx_tissues</th>\n",
       "      <th>sQTL</th>\n",
       "      <th>Vernot_allele_origin</th>\n",
       "      <th>Browning_allele_origin</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>894154</th>\n",
       "      <td>chr2</td>\n",
       "      <td>238970511</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>yes</td>\n",
       "      <td>1274.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.25</td>\n",
       "      <td>0.23</td>\n",
       "      <td>0.45</td>\n",
       "      <td>0.04</td>\n",
       "      <td>0.43</td>\n",
       "      <td>0.23</td>\n",
       "      <td>0.018519</td>\n",
       "      <td>yes</td>\n",
       "      <td>44715.0</td>\n",
       "      <td>143004.0</td>\n",
       "      <td>0.312684</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.25</td>\n",
       "      <td>SCLY</td>\n",
       "      <td>1.01520</td>\n",
       "      <td>-0.091207</td>\n",
       "      <td>0.759470</td>\n",
       "      <td>1.02300</td>\n",
       "      <td>0.561</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.09</td>\n",
       "      <td>0.46</td>\n",
       "      <td>0.46</td>\n",
       "      <td>31</td>\n",
       "      <td>-5</td>\n",
       "      <td>-34</td>\n",
       "      <td>-5</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>ancient</td>\n",
       "      <td>introgressed</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>894150</th>\n",
       "      <td>chr2</td>\n",
       "      <td>238970308</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>yes</td>\n",
       "      <td>1274.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.25</td>\n",
       "      <td>0.23</td>\n",
       "      <td>0.45</td>\n",
       "      <td>0.04</td>\n",
       "      <td>0.43</td>\n",
       "      <td>0.23</td>\n",
       "      <td>0.018519</td>\n",
       "      <td>yes</td>\n",
       "      <td>44716.0</td>\n",
       "      <td>143046.0</td>\n",
       "      <td>0.312599</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.25</td>\n",
       "      <td>SCLY</td>\n",
       "      <td>1.01520</td>\n",
       "      <td>-0.091207</td>\n",
       "      <td>0.759470</td>\n",
       "      <td>1.02300</td>\n",
       "      <td>0.402</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.25</td>\n",
       "      <td>0.03</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.25</td>\n",
       "      <td>48</td>\n",
       "      <td>2</td>\n",
       "      <td>39</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>ancient</td>\n",
       "      <td>introgressed</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>894153</th>\n",
       "      <td>chr2</td>\n",
       "      <td>238970511</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>yes</td>\n",
       "      <td>1274.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.25</td>\n",
       "      <td>0.23</td>\n",
       "      <td>0.45</td>\n",
       "      <td>0.04</td>\n",
       "      <td>0.43</td>\n",
       "      <td>0.23</td>\n",
       "      <td>0.018519</td>\n",
       "      <td>yes</td>\n",
       "      <td>44715.0</td>\n",
       "      <td>143004.0</td>\n",
       "      <td>0.312684</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.25</td>\n",
       "      <td>UBE2F-SCLY</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.561</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.04</td>\n",
       "      <td>0.31</td>\n",
       "      <td>0.31</td>\n",
       "      <td>31</td>\n",
       "      <td>-5</td>\n",
       "      <td>-34</td>\n",
       "      <td>-5</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>ancient</td>\n",
       "      <td>introgressed</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>939756</th>\n",
       "      <td>chr20</td>\n",
       "      <td>62224595</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>yes</td>\n",
       "      <td>1138.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.22</td>\n",
       "      <td>0.51</td>\n",
       "      <td>0.21</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.11</td>\n",
       "      <td>0.31</td>\n",
       "      <td>0.000975</td>\n",
       "      <td>yes</td>\n",
       "      <td>23692.0</td>\n",
       "      <td>143104.0</td>\n",
       "      <td>0.165558</td>\n",
       "      <td>yes</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.03503</td>\n",
       "      <td>0.00099</td>\n",
       "      <td>0.10663</td>\n",
       "      <td>0.51488</td>\n",
       "      <td>0.21173</td>\n",
       "      <td>0.03704</td>\n",
       "      <td>0.30982</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>chr20_62195671_62229244</td>\n",
       "      <td>0.228810</td>\n",
       "      <td>yes</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.22</td>\n",
       "      <td>GMEB2</td>\n",
       "      <td>0.66048</td>\n",
       "      <td>2.202000</td>\n",
       "      <td>0.049787</td>\n",
       "      <td>3.94620</td>\n",
       "      <td>-0.837</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.41</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.41</td>\n",
       "      <td>15</td>\n",
       "      <td>-44</td>\n",
       "      <td>2</td>\n",
       "      <td>-2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>21.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>introgressed</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1058121</th>\n",
       "      <td>chr4</td>\n",
       "      <td>38806019</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>ancestral</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>yes</td>\n",
       "      <td>911.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.18</td>\n",
       "      <td>0.39</td>\n",
       "      <td>0.22</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.15</td>\n",
       "      <td>0.16</td>\n",
       "      <td>0.001949</td>\n",
       "      <td>yes</td>\n",
       "      <td>19926.0</td>\n",
       "      <td>143260.0</td>\n",
       "      <td>0.139090</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.18</td>\n",
       "      <td>TLR1</td>\n",
       "      <td>1.09920</td>\n",
       "      <td>-0.701570</td>\n",
       "      <td>1.411200</td>\n",
       "      <td>-1.72720</td>\n",
       "      <td>-1.664</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.21</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.21</td>\n",
       "      <td>50</td>\n",
       "      <td>-16</td>\n",
       "      <td>-16</td>\n",
       "      <td>-28</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>ancient</td>\n",
       "      <td>introgressed</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>246812</th>\n",
       "      <td>chr11</td>\n",
       "      <td>11507780</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>yes</td>\n",
       "      <td>917.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.18</td>\n",
       "      <td>0.14</td>\n",
       "      <td>0.31</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.21</td>\n",
       "      <td>0.28</td>\n",
       "      <td>0.003899</td>\n",
       "      <td>yes</td>\n",
       "      <td>33636.0</td>\n",
       "      <td>143182.0</td>\n",
       "      <td>0.234918</td>\n",
       "      <td>yes</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.05732</td>\n",
       "      <td>0.00397</td>\n",
       "      <td>0.21182</td>\n",
       "      <td>0.13988</td>\n",
       "      <td>0.31014</td>\n",
       "      <td>0.09259</td>\n",
       "      <td>0.28323</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr11_11494384_11530287</td>\n",
       "      <td>0.189808</td>\n",
       "      <td>yes</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.18</td>\n",
       "      <td>GALNT18</td>\n",
       "      <td>0.87026</td>\n",
       "      <td>0.879900</td>\n",
       "      <td>0.387360</td>\n",
       "      <td>3.02520</td>\n",
       "      <td>-0.302</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.21</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.21</td>\n",
       "      <td>37</td>\n",
       "      <td>-14</td>\n",
       "      <td>2</td>\n",
       "      <td>-14</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>introgressed</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1058119</th>\n",
       "      <td>chr4</td>\n",
       "      <td>38805942</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>yes</td>\n",
       "      <td>911.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.18</td>\n",
       "      <td>0.39</td>\n",
       "      <td>0.22</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.15</td>\n",
       "      <td>0.16</td>\n",
       "      <td>0.001949</td>\n",
       "      <td>yes</td>\n",
       "      <td>19931.0</td>\n",
       "      <td>143224.0</td>\n",
       "      <td>0.139160</td>\n",
       "      <td>yes</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.04459</td>\n",
       "      <td>0.00198</td>\n",
       "      <td>0.14986</td>\n",
       "      <td>0.39484</td>\n",
       "      <td>0.22068</td>\n",
       "      <td>0.05556</td>\n",
       "      <td>0.16360</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>chr4_38294200_38857896</td>\n",
       "      <td>0.186192</td>\n",
       "      <td>yes</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.18</td>\n",
       "      <td>TLR1</td>\n",
       "      <td>1.09920</td>\n",
       "      <td>-0.701570</td>\n",
       "      <td>1.411200</td>\n",
       "      <td>-1.72720</td>\n",
       "      <td>-0.219</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.38</td>\n",
       "      <td>0.38</td>\n",
       "      <td>27</td>\n",
       "      <td>-15</td>\n",
       "      <td>1</td>\n",
       "      <td>-15</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>introgressed</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1420111</th>\n",
       "      <td>chr7</td>\n",
       "      <td>136608159</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>yes</td>\n",
       "      <td>710.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.14</td>\n",
       "      <td>0.04</td>\n",
       "      <td>0.36</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.21</td>\n",
       "      <td>0.12</td>\n",
       "      <td>0.008772</td>\n",
       "      <td>yes</td>\n",
       "      <td>34245.0</td>\n",
       "      <td>142902.0</td>\n",
       "      <td>0.239640</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.14</td>\n",
       "      <td>CHRM2</td>\n",
       "      <td>0.65445</td>\n",
       "      <td>1.920000</td>\n",
       "      <td>0.277890</td>\n",
       "      <td>2.19860</td>\n",
       "      <td>-0.264</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.27</td>\n",
       "      <td>0.27</td>\n",
       "      <td>2</td>\n",
       "      <td>7</td>\n",
       "      <td>2</td>\n",
       "      <td>15</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>ancient</td>\n",
       "      <td>introgressed</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1058111</th>\n",
       "      <td>chr4</td>\n",
       "      <td>38802528</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Other</td>\n",
       "      <td>yes</td>\n",
       "      <td>705.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.14</td>\n",
       "      <td>0.27</td>\n",
       "      <td>0.17</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.12</td>\n",
       "      <td>0.15</td>\n",
       "      <td>0.001949</td>\n",
       "      <td>yes</td>\n",
       "      <td>17091.0</td>\n",
       "      <td>143174.0</td>\n",
       "      <td>0.119372</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.14</td>\n",
       "      <td>TLR1</td>\n",
       "      <td>1.09920</td>\n",
       "      <td>-0.701570</td>\n",
       "      <td>1.411200</td>\n",
       "      <td>-1.72720</td>\n",
       "      <td>-0.247</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.33</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.19</td>\n",
       "      <td>0.33</td>\n",
       "      <td>-50</td>\n",
       "      <td>41</td>\n",
       "      <td>0</td>\n",
       "      <td>-50</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>ancient</td>\n",
       "      <td>introgressed</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>341553</th>\n",
       "      <td>chr12</td>\n",
       "      <td>27922074</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>yes</td>\n",
       "      <td>715.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.14</td>\n",
       "      <td>0.20</td>\n",
       "      <td>0.24</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.11</td>\n",
       "      <td>0.16</td>\n",
       "      <td>0.011696</td>\n",
       "      <td>yes</td>\n",
       "      <td>23359.0</td>\n",
       "      <td>143168.0</td>\n",
       "      <td>0.163158</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.14</td>\n",
       "      <td>MANSC4</td>\n",
       "      <td>0.65585</td>\n",
       "      <td>1.619400</td>\n",
       "      <td>1.210900</td>\n",
       "      <td>-0.43512</td>\n",
       "      <td>0.454</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.22</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.22</td>\n",
       "      <td>-1</td>\n",
       "      <td>-37</td>\n",
       "      <td>26</td>\n",
       "      <td>-37</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>ancient</td>\n",
       "      <td>introgressed</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         chrom        pos ref_allele alt_allele ancestral_allele    anc_dev  \\\n",
       "894154    chr2  238970511          G          A                G    derived   \n",
       "894150    chr2  238970308          A          G                A    derived   \n",
       "894153    chr2  238970511          G          A                G    derived   \n",
       "939756   chr20   62224595          G          A                G    derived   \n",
       "1058121   chr4   38806019          A          G                G  ancestral   \n",
       "246812   chr11   11507780          G          A                G    derived   \n",
       "1058119   chr4   38805942          G          C                G    derived   \n",
       "1420111   chr7  136608159          G          A                G    derived   \n",
       "1058111   chr4   38802528          G          A                G    derived   \n",
       "341553   chr12   27922074          T          C                T    derived   \n",
       "\n",
       "        variant_type altai_gt chagyrskaya_gt denisovan_gt vindija_gt  \\\n",
       "894154           snv      1/1            1/1          0/0        1/1   \n",
       "894150           snv      1/1            1/1          0/0        1/1   \n",
       "894153           snv      1/1            1/1          0/0        1/1   \n",
       "939756           snv      0/1            1/1          0/0        0/1   \n",
       "1058121          snv      1/1            1/1          1/1        1/1   \n",
       "246812           snv      1/1            1/1          0/0        1/1   \n",
       "1058119          snv      0/1            1/1          0/0        1/1   \n",
       "1420111          snv      1/1            1/1          0/0        1/1   \n",
       "1058111          snv      0/1            0/0          0/0        1/1   \n",
       "341553           snv      1/1            1/1          1/1        1/1   \n",
       "\n",
       "         altai_gt_boolean  chagyrskaya_gt_boolean  denisovan_gt_boolean  \\\n",
       "894154               True                    True                 False   \n",
       "894150               True                    True                 False   \n",
       "894153               True                    True                 False   \n",
       "939756               True                    True                 False   \n",
       "1058121              True                    True                  True   \n",
       "246812               True                    True                 False   \n",
       "1058119              True                    True                 False   \n",
       "1420111              True                    True                 False   \n",
       "1058111              True                   False                 False   \n",
       "341553               True                    True                  True   \n",
       "\n",
       "         vindija_gt_boolean distribution present_in_1KG  1KG_allele_count  \\\n",
       "894154                 True  Neanderthal            yes            1274.0   \n",
       "894150                 True  Neanderthal            yes            1274.0   \n",
       "894153                 True  Neanderthal            yes            1274.0   \n",
       "939756                 True  Neanderthal            yes            1138.0   \n",
       "1058121                True       Shared            yes             911.0   \n",
       "246812                 True  Neanderthal            yes             917.0   \n",
       "1058119                True  Neanderthal            yes             911.0   \n",
       "1420111                True  Neanderthal            yes             710.0   \n",
       "1058111                True        Other            yes             705.0   \n",
       "341553                 True       Shared            yes             715.0   \n",
       "\n",
       "         1KG_allele_number  1KG_allele_frequency  1KG_EAS_AF  1KG_EUR_AF  \\\n",
       "894154              5096.0                  0.25        0.23        0.45   \n",
       "894150              5096.0                  0.25        0.23        0.45   \n",
       "894153              5096.0                  0.25        0.23        0.45   \n",
       "939756              5096.0                  0.22        0.51        0.21   \n",
       "1058121             5096.0                  0.18        0.39        0.22   \n",
       "246812              5096.0                  0.18        0.14        0.31   \n",
       "1058119             5096.0                  0.18        0.39        0.22   \n",
       "1420111             5096.0                  0.14        0.04        0.36   \n",
       "1058111             5096.0                  0.14        0.27        0.17   \n",
       "341553              5096.0                  0.14        0.20        0.24   \n",
       "\n",
       "         1KG_AFR_AF  1KG_AMR_AF  1KG_SAS_AF  1KG_non_ASW_AFR_AF  \\\n",
       "894154         0.04        0.43        0.23            0.018519   \n",
       "894150         0.04        0.43        0.23            0.018519   \n",
       "894153         0.04        0.43        0.23            0.018519   \n",
       "939756         0.01        0.11        0.31            0.000975   \n",
       "1058121        0.01        0.15        0.16            0.001949   \n",
       "246812         0.02        0.21        0.28            0.003899   \n",
       "1058119        0.01        0.15        0.16            0.001949   \n",
       "1420111        0.02        0.21        0.12            0.008772   \n",
       "1058111        0.01        0.12        0.15            0.001949   \n",
       "341553         0.02        0.11        0.16            0.011696   \n",
       "\n",
       "        present_in_gnomAD  gnomAD_allele_count  gnomAD_allele_number  \\\n",
       "894154                yes              44715.0              143004.0   \n",
       "894150                yes              44716.0              143046.0   \n",
       "894153                yes              44715.0              143004.0   \n",
       "939756                yes              23692.0              143104.0   \n",
       "1058121               yes              19926.0              143260.0   \n",
       "246812                yes              33636.0              143182.0   \n",
       "1058119               yes              19931.0              143224.0   \n",
       "1420111               yes              34245.0              142902.0   \n",
       "1058111               yes              17091.0              143174.0   \n",
       "341553                yes              23359.0              143168.0   \n",
       "\n",
       "         gnomAD_allele_frequency Vernot_introgressed Vernot_ancestral_allele  \\\n",
       "894154                  0.312684                  no                     NaN   \n",
       "894150                  0.312599                  no                     NaN   \n",
       "894153                  0.312684                  no                     NaN   \n",
       "939756                  0.165558                 yes                       G   \n",
       "1058121                 0.139090                  no                     NaN   \n",
       "246812                  0.234918                 yes                       G   \n",
       "1058119                 0.139160                 yes                       G   \n",
       "1420111                 0.239640                  no                     NaN   \n",
       "1058111                 0.119372                  no                     NaN   \n",
       "341553                  0.163158                  no                     NaN   \n",
       "\n",
       "        Vernot_derived_allele  Vernot_ancestral_derived_code  Vernot_AFA_AF  \\\n",
       "894154                    NaN                            NaN            NaN   \n",
       "894150                    NaN                            NaN            NaN   \n",
       "894153                    NaN                            NaN            NaN   \n",
       "939756                      A                            1.0        0.03503   \n",
       "1058121                   NaN                            NaN            NaN   \n",
       "246812                      A                            1.0        0.05732   \n",
       "1058119                     C                            1.0        0.04459   \n",
       "1420111                   NaN                            NaN            NaN   \n",
       "1058111                   NaN                            NaN            NaN   \n",
       "341553                    NaN                            NaN            NaN   \n",
       "\n",
       "         Vernot_AFR_AF  Vernot_AMR_AF  Vernot_EAS_AF  Vernot_EUR_AF  \\\n",
       "894154             NaN            NaN            NaN            NaN   \n",
       "894150             NaN            NaN            NaN            NaN   \n",
       "894153             NaN            NaN            NaN            NaN   \n",
       "939756         0.00099        0.10663        0.51488        0.21173   \n",
       "1058121            NaN            NaN            NaN            NaN   \n",
       "246812         0.00397        0.21182        0.13988        0.31014   \n",
       "1058119        0.00198        0.14986        0.39484        0.22068   \n",
       "1420111            NaN            NaN            NaN            NaN   \n",
       "1058111            NaN            NaN            NaN            NaN   \n",
       "341553             NaN            NaN            NaN            NaN   \n",
       "\n",
       "         Vernot_PNG_AF  Vernot_SAS_AF Vernot_Denisovan_base  \\\n",
       "894154             NaN            NaN                   NaN   \n",
       "894150             NaN            NaN                   NaN   \n",
       "894153             NaN            NaN                   NaN   \n",
       "939756         0.03704        0.30982                     G   \n",
       "1058121            NaN            NaN                   NaN   \n",
       "246812         0.09259        0.28323                     G   \n",
       "1058119        0.05556        0.16360                     G   \n",
       "1420111            NaN            NaN                   NaN   \n",
       "1058111            NaN            NaN                   NaN   \n",
       "341553             NaN            NaN                   NaN   \n",
       "\n",
       "        Vernot_Neanderthal_base_1 Vernot_Neanderthal_base_2  \\\n",
       "894154                        NaN                       NaN   \n",
       "894150                        NaN                       NaN   \n",
       "894153                        NaN                       NaN   \n",
       "939756                          G                         A   \n",
       "1058121                       NaN                       NaN   \n",
       "246812                          A                       NaN   \n",
       "1058119                         G                         C   \n",
       "1420111                       NaN                       NaN   \n",
       "1058111                       NaN                       NaN   \n",
       "341553                        NaN                       NaN   \n",
       "\n",
       "            Vernot_haplotype_tag  Vernot_introgressed_AF  \\\n",
       "894154                       NaN                     NaN   \n",
       "894150                       NaN                     NaN   \n",
       "894153                       NaN                     NaN   \n",
       "939756   chr20_62195671_62229244                0.228810   \n",
       "1058121                      NaN                     NaN   \n",
       "246812   chr11_11494384_11530287                0.189808   \n",
       "1058119   chr4_38294200_38857896                0.186192   \n",
       "1420111                      NaN                     NaN   \n",
       "1058111                      NaN                     NaN   \n",
       "341553                       NaN                     NaN   \n",
       "\n",
       "        Browning_introgressed  Browning_ref_alt  Browning_introgressed_AF  \\\n",
       "894154                    yes               1.0                      0.25   \n",
       "894150                    yes               1.0                      0.25   \n",
       "894153                    yes               1.0                      0.25   \n",
       "939756                    yes               1.0                      0.22   \n",
       "1058121                   yes               1.0                      0.18   \n",
       "246812                    yes               1.0                      0.18   \n",
       "1058119                   yes               1.0                      0.18   \n",
       "1420111                   yes               1.0                      0.14   \n",
       "1058111                   yes               1.0                      0.14   \n",
       "341553                    yes               1.0                      0.14   \n",
       "\n",
       "         annotation   mis_oe     mis_z    lof_oe    lof_z  phyloP  ag_delta  \\\n",
       "894154         SCLY  1.01520 -0.091207  0.759470  1.02300   0.561      0.00   \n",
       "894150         SCLY  1.01520 -0.091207  0.759470  1.02300   0.402      0.00   \n",
       "894153   UBE2F-SCLY      NaN       NaN       NaN      NaN   0.561      0.00   \n",
       "939756        GMEB2  0.66048  2.202000  0.049787  3.94620  -0.837      0.01   \n",
       "1058121        TLR1  1.09920 -0.701570  1.411200 -1.72720  -1.664      0.01   \n",
       "246812      GALNT18  0.87026  0.879900  0.387360  3.02520  -0.302      0.00   \n",
       "1058119        TLR1  1.09920 -0.701570  1.411200 -1.72720  -0.219      0.00   \n",
       "1420111       CHRM2  0.65445  1.920000  0.277890  2.19860  -0.264      0.00   \n",
       "1058111        TLR1  1.09920 -0.701570  1.411200 -1.72720  -0.247      0.00   \n",
       "341553       MANSC4  0.65585  1.619400  1.210900 -0.43512   0.454      0.00   \n",
       "\n",
       "         al_delta  dg_delta  dl_delta  delta_max  ag_pos  al_pos  dg_pos  \\\n",
       "894154       0.00      0.09      0.46       0.46      31      -5     -34   \n",
       "894150       0.25      0.03      0.00       0.25      48       2      39   \n",
       "894153       0.00      0.04      0.31       0.31      31      -5     -34   \n",
       "939756       0.00      0.41      0.02       0.41      15     -44       2   \n",
       "1058121      0.21      0.00      0.00       0.21      50     -16     -16   \n",
       "246812       0.00      0.21      0.00       0.21      37     -14       2   \n",
       "1058119      0.00      0.00      0.38       0.38      27     -15       1   \n",
       "1420111      0.00      0.00      0.27       0.27       2       7       2   \n",
       "1058111      0.33      0.00      0.19       0.33     -50      41       0   \n",
       "341553       0.22      0.00      0.00       0.22      -1     -37      26   \n",
       "\n",
       "         dl_pos  Adipose_Subcutaneous  Adipose_Visceral_Omentum  \\\n",
       "894154       -5                   0.0                       0.0   \n",
       "894150        2                   0.0                       0.0   \n",
       "894153       -5                   0.0                       0.0   \n",
       "939756       -2                   0.0                       0.0   \n",
       "1058121     -28                   1.0                       1.0   \n",
       "246812      -14                   NaN                       NaN   \n",
       "1058119     -15                   1.0                       1.0   \n",
       "1420111      15                   NaN                       NaN   \n",
       "1058111     -50                   1.0                       1.0   \n",
       "341553      -37                   0.0                       0.0   \n",
       "\n",
       "         Adrenal_Gland  Artery_Aorta  Artery_Coronary  Artery_Tibial  \\\n",
       "894154             0.0           0.0              0.0            0.0   \n",
       "894150             0.0           0.0              0.0            0.0   \n",
       "894153             0.0           0.0              0.0            0.0   \n",
       "939756             1.0           0.0              1.0            1.0   \n",
       "1058121            0.0           0.0              0.0            0.0   \n",
       "246812             NaN           NaN              NaN            NaN   \n",
       "1058119            0.0           0.0              0.0            0.0   \n",
       "1420111            NaN           NaN              NaN            NaN   \n",
       "1058111            0.0           0.0              0.0            0.0   \n",
       "341553             0.0           0.0              0.0            0.0   \n",
       "\n",
       "         Brain_Amygdala  Brain_Anterior_cingulate_cortex_BA24  \\\n",
       "894154              0.0                                   0.0   \n",
       "894150              0.0                                   0.0   \n",
       "894153              0.0                                   0.0   \n",
       "939756              0.0                                   0.0   \n",
       "1058121             0.0                                   0.0   \n",
       "246812              NaN                                   NaN   \n",
       "1058119             0.0                                   0.0   \n",
       "1420111             NaN                                   NaN   \n",
       "1058111             0.0                                   0.0   \n",
       "341553              0.0                                   0.0   \n",
       "\n",
       "         Brain_Caudate_basal_ganglia  Brain_Cerebellar_Hemisphere  \\\n",
       "894154                           0.0                          1.0   \n",
       "894150                           0.0                          1.0   \n",
       "894153                           0.0                          1.0   \n",
       "939756                           1.0                          1.0   \n",
       "1058121                          0.0                          0.0   \n",
       "246812                           NaN                          NaN   \n",
       "1058119                          0.0                          0.0   \n",
       "1420111                          NaN                          NaN   \n",
       "1058111                          0.0                          0.0   \n",
       "341553                           0.0                          0.0   \n",
       "\n",
       "         Brain_Cerebellum  Brain_Cortex  Brain_Frontal_Cortex_BA9  \\\n",
       "894154                0.0           0.0                       0.0   \n",
       "894150                0.0           0.0                       0.0   \n",
       "894153                0.0           0.0                       0.0   \n",
       "939756                1.0           1.0                       1.0   \n",
       "1058121               0.0           0.0                       0.0   \n",
       "246812                NaN           NaN                       NaN   \n",
       "1058119               0.0           0.0                       0.0   \n",
       "1420111               NaN           NaN                       NaN   \n",
       "1058111               0.0           0.0                       0.0   \n",
       "341553                0.0           0.0                       0.0   \n",
       "\n",
       "         Brain_Hippocampus  Brain_Hypothalamus  \\\n",
       "894154                 0.0                 0.0   \n",
       "894150                 0.0                 0.0   \n",
       "894153                 0.0                 0.0   \n",
       "939756                 0.0                 1.0   \n",
       "1058121                0.0                 0.0   \n",
       "246812                 NaN                 NaN   \n",
       "1058119                0.0                 0.0   \n",
       "1420111                NaN                 NaN   \n",
       "1058111                0.0                 0.0   \n",
       "341553                 0.0                 0.0   \n",
       "\n",
       "         Brain_Nucleus_accumbens_basal_ganglia  Brain_Putamen_basal_ganglia  \\\n",
       "894154                                     0.0                          0.0   \n",
       "894150                                     0.0                          0.0   \n",
       "894153                                     0.0                          0.0   \n",
       "939756                                     1.0                          0.0   \n",
       "1058121                                    0.0                          0.0   \n",
       "246812                                     NaN                          NaN   \n",
       "1058119                                    0.0                          0.0   \n",
       "1420111                                    NaN                          NaN   \n",
       "1058111                                    0.0                          0.0   \n",
       "341553                                     0.0                          0.0   \n",
       "\n",
       "         Brain_Spinal_cord_cervical_c-1  Brain_Substantia_nigra  \\\n",
       "894154                              0.0                     0.0   \n",
       "894150                              0.0                     0.0   \n",
       "894153                              0.0                     0.0   \n",
       "939756                              0.0                     0.0   \n",
       "1058121                             0.0                     0.0   \n",
       "246812                              NaN                     NaN   \n",
       "1058119                             0.0                     0.0   \n",
       "1420111                             NaN                     NaN   \n",
       "1058111                             0.0                     0.0   \n",
       "341553                              0.0                     0.0   \n",
       "\n",
       "         Breast_Mammary_Tissue  Cells_Cultured_fibroblasts  \\\n",
       "894154                     0.0                         0.0   \n",
       "894150                     0.0                         0.0   \n",
       "894153                     0.0                         0.0   \n",
       "939756                     0.0                         0.0   \n",
       "1058121                    0.0                         0.0   \n",
       "246812                     NaN                         NaN   \n",
       "1058119                    0.0                         0.0   \n",
       "1420111                    NaN                         NaN   \n",
       "1058111                    0.0                         0.0   \n",
       "341553                     0.0                         1.0   \n",
       "\n",
       "         Cells_EBV-transformed_lymphocytes  Colon_Sigmoid  Colon_Transverse  \\\n",
       "894154                                 0.0            0.0               0.0   \n",
       "894150                                 0.0            0.0               0.0   \n",
       "894153                                 0.0            0.0               0.0   \n",
       "939756                                 0.0            1.0               1.0   \n",
       "1058121                                1.0            0.0               0.0   \n",
       "246812                                 NaN            NaN               NaN   \n",
       "1058119                                1.0            0.0               0.0   \n",
       "1420111                                NaN            NaN               NaN   \n",
       "1058111                                1.0            0.0               0.0   \n",
       "341553                                 0.0            0.0               0.0   \n",
       "\n",
       "         Esophagus_Gastroesophageal_Junction  Esophagus_Mucosa  \\\n",
       "894154                                   0.0               0.0   \n",
       "894150                                   0.0               0.0   \n",
       "894153                                   0.0               0.0   \n",
       "939756                                   1.0               0.0   \n",
       "1058121                                  0.0               0.0   \n",
       "246812                                   NaN               NaN   \n",
       "1058119                                  0.0               0.0   \n",
       "1420111                                  NaN               NaN   \n",
       "1058111                                  0.0               0.0   \n",
       "341553                                   0.0               0.0   \n",
       "\n",
       "         Esophagus_Muscularis  Heart_Atrial_Appendage  Heart_Left_Ventricle  \\\n",
       "894154                    0.0                     1.0                   0.0   \n",
       "894150                    0.0                     1.0                   0.0   \n",
       "894153                    0.0                     1.0                   0.0   \n",
       "939756                    1.0                     0.0                   0.0   \n",
       "1058121                   0.0                     0.0                   0.0   \n",
       "246812                    NaN                     NaN                   NaN   \n",
       "1058119                   0.0                     0.0                   0.0   \n",
       "1420111                   NaN                     NaN                   NaN   \n",
       "1058111                   0.0                     0.0                   0.0   \n",
       "341553                    0.0                     0.0                   0.0   \n",
       "\n",
       "         Kidney_Cortex  Liver  Lung  Minor_Salivary_Gland  Muscle_Skeletal  \\\n",
       "894154             0.0    0.0   0.0                   0.0              1.0   \n",
       "894150             0.0    0.0   0.0                   0.0              1.0   \n",
       "894153             0.0    0.0   0.0                   0.0              1.0   \n",
       "939756             0.0    0.0   1.0                   0.0              0.0   \n",
       "1058121            0.0    0.0   1.0                   0.0              0.0   \n",
       "246812             NaN    NaN   NaN                   NaN              NaN   \n",
       "1058119            0.0    0.0   1.0                   0.0              0.0   \n",
       "1420111            NaN    NaN   NaN                   NaN              NaN   \n",
       "1058111            0.0    0.0   1.0                   0.0              0.0   \n",
       "341553             0.0    0.0   0.0                   0.0              1.0   \n",
       "\n",
       "         Nerve_Tibial  Ovary  Pancreas  Pituitary  Prostate  \\\n",
       "894154            0.0    0.0       0.0        0.0       0.0   \n",
       "894150            0.0    0.0       0.0        0.0       0.0   \n",
       "894153            0.0    0.0       0.0        0.0       0.0   \n",
       "939756            1.0    1.0       0.0        0.0       1.0   \n",
       "1058121           1.0    0.0       0.0        0.0       0.0   \n",
       "246812            NaN    NaN       NaN        NaN       NaN   \n",
       "1058119           1.0    0.0       0.0        0.0       0.0   \n",
       "1420111           NaN    NaN       NaN        NaN       NaN   \n",
       "1058111           1.0    0.0       0.0        0.0       0.0   \n",
       "341553            0.0    0.0       0.0        0.0       0.0   \n",
       "\n",
       "         Skin_Not_Sun_Exposed_Suprapubic  Skin_Sun_Exposed_Lower_leg  \\\n",
       "894154                               0.0                         0.0   \n",
       "894150                               0.0                         0.0   \n",
       "894153                               0.0                         0.0   \n",
       "939756                               0.0                         0.0   \n",
       "1058121                              0.0                         0.0   \n",
       "246812                               NaN                         NaN   \n",
       "1058119                              0.0                         0.0   \n",
       "1420111                              NaN                         NaN   \n",
       "1058111                              0.0                         1.0   \n",
       "341553                               0.0                         0.0   \n",
       "\n",
       "         Small_Intestine_Terminal_Ileum  Spleen  Stomach  Testis  Thyroid  \\\n",
       "894154                              0.0     0.0      0.0     1.0      0.0   \n",
       "894150                              0.0     0.0      0.0     1.0      0.0   \n",
       "894153                              0.0     0.0      0.0     1.0      0.0   \n",
       "939756                              1.0     0.0      0.0     1.0      1.0   \n",
       "1058121                             0.0     1.0      0.0     0.0      0.0   \n",
       "246812                              NaN     NaN      NaN     NaN      NaN   \n",
       "1058119                             0.0     1.0      0.0     0.0      0.0   \n",
       "1420111                             NaN     NaN      NaN     NaN      NaN   \n",
       "1058111                             0.0     1.0      0.0     0.0      0.0   \n",
       "341553                              0.0     0.0      0.0     0.0      0.0   \n",
       "\n",
       "         Uterus  Vagina  Whole_Blood  N_GTEx_tissues sQTL  \\\n",
       "894154      0.0     0.0          0.0             4.0  yes   \n",
       "894150      0.0     0.0          0.0             4.0  yes   \n",
       "894153      0.0     0.0          0.0             4.0  yes   \n",
       "939756      0.0     0.0          0.0            21.0  yes   \n",
       "1058121     0.0     0.0          1.0             7.0  yes   \n",
       "246812      NaN     NaN          NaN             NaN   no   \n",
       "1058119     0.0     0.0          1.0             7.0  yes   \n",
       "1420111     NaN     NaN          NaN             NaN   no   \n",
       "1058111     0.0     0.0          1.0             8.0  yes   \n",
       "341553      0.0     0.0          0.0             2.0  yes   \n",
       "\n",
       "        Vernot_allele_origin Browning_allele_origin  \n",
       "894154               ancient           introgressed  \n",
       "894150               ancient           introgressed  \n",
       "894153               ancient           introgressed  \n",
       "939756          introgressed           introgressed  \n",
       "1058121              ancient           introgressed  \n",
       "246812          introgressed           introgressed  \n",
       "1058119         introgressed           introgressed  \n",
       "1420111              ancient           introgressed  \n",
       "1058111              ancient           introgressed  \n",
       "341553               ancient           introgressed  "
      ]
     },
     "execution_count": 497,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SA_2_introgressed = data_2[data_2['Browning_allele_origin']=='introgressed']\n",
    "SA_2_introgressed_AF = SA_2_introgressed.sort_values(['Browning_introgressed_AF'], ascending = [False])\n",
    "SA_2_introgressed_AF.head(10)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "481569eb",
   "metadata": {},
   "source": [
    "## ASE <a class = 'anchor' id = 'ASE'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7bcc31a5",
   "metadata": {},
   "source": [
    "Is there overlap between introgressed variants implicated in allele-specific expression from McCoy et al. 2017 and our data?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 498,
   "id": "33bffada",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref_allele</th>\n",
       "      <th>alt_allele</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr9</td>\n",
       "      <td>34569494</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10</td>\n",
       "      <td>126696061</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr11</td>\n",
       "      <td>8454637</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>222987816</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr2</td>\n",
       "      <td>107872056</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr13</td>\n",
       "      <td>86949835</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr17</td>\n",
       "      <td>8026078</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr4</td>\n",
       "      <td>149556752</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr16</td>\n",
       "      <td>86754480</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr8</td>\n",
       "      <td>130718756</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   chrom        pos ref_allele alt_allele\n",
       "0   chr9   34569494          G          T\n",
       "1  chr10  126696061          T          G\n",
       "2  chr11    8454637          A          C\n",
       "3   chr1  222987816          A          T\n",
       "4   chr2  107872056          T          C\n",
       "5  chr13   86949835          C          G\n",
       "6  chr17    8026078          C          G\n",
       "7   chr4  149556752          G          A\n",
       "8  chr16   86754480          C          T\n",
       "9   chr8  130718756          C          G"
      ]
     },
     "execution_count": 498,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mccoy_header = ['chrom','pos','ref_allele','alt_allele']\n",
    "mccoy = pd.read_csv('../ASE/McCoy_et_al_2017_ASE_variants.txt', sep = '\\t', names = mccoy_header)\n",
    "mccoy.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 499,
   "id": "ceffe117",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1236"
      ]
     },
     "execution_count": 499,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(mccoy)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 500,
   "id": "e77b33a5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref_allele</th>\n",
       "      <th>alt_allele</th>\n",
       "      <th>delta_max</th>\n",
       "      <th>Vernot_allele_origin</th>\n",
       "      <th>Browning_allele_origin</th>\n",
       "      <th>annotation</th>\n",
       "      <th>_merge</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3383286</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.00</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>ARHGEF16</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3411396</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.01</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>MEGF6</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3411561</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.01</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>MEGF6</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3411652</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.04</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>MEGF6</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3551792</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.01</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>WRAP73</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>10368282</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0.00</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>KIF1B</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr1</td>\n",
       "      <td>11905995</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>0.00</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>NPPA</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr1</td>\n",
       "      <td>12247128</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.00</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>TNFRSF1B</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr1</td>\n",
       "      <td>12260399</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.01</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>TNFRSF1B</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr1</td>\n",
       "      <td>12261613</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.00</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>TNFRSF1B</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chrom       pos ref_allele alt_allele  delta_max Vernot_allele_origin  \\\n",
       "0  chr1   3383286          G          A       0.00         introgressed   \n",
       "1  chr1   3411396          T          C       0.01         introgressed   \n",
       "2  chr1   3411561          G          A       0.01         introgressed   \n",
       "3  chr1   3411652          G          A       0.04         introgressed   \n",
       "4  chr1   3551792          G          A       0.01         introgressed   \n",
       "5  chr1  10368282          G          C       0.00         introgressed   \n",
       "6  chr1  11905995          C          A       0.00         introgressed   \n",
       "7  chr1  12247128          C          T       0.00         introgressed   \n",
       "8  chr1  12260399          C          T       0.01         introgressed   \n",
       "9  chr1  12261613          T          C       0.00         introgressed   \n",
       "\n",
       "  Browning_allele_origin annotation _merge  \n",
       "0           introgressed   ARHGEF16   both  \n",
       "1           introgressed      MEGF6   both  \n",
       "2           introgressed      MEGF6   both  \n",
       "3           introgressed      MEGF6   both  \n",
       "4           introgressed     WRAP73   both  \n",
       "5           introgressed      KIF1B   both  \n",
       "6           introgressed       NPPA   both  \n",
       "7           introgressed   TNFRSF1B   both  \n",
       "8           introgressed   TNFRSF1B   both  \n",
       "9           introgressed   TNFRSF1B   both  "
      ]
     },
     "execution_count": 500,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mccoy_merge = pd.merge(data[['chrom','pos','ref_allele','alt_allele','delta_max','Vernot_allele_origin','Browning_allele_origin','annotation']], mccoy[['chrom','pos','ref_allele','alt_allele']], on=['chrom','pos','ref_allele','alt_allele'], indicator=True)\n",
    "mccoy_merge.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 501,
   "id": "09213303",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "862"
      ]
     },
     "execution_count": 501,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(mccoy_merge)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 502,
   "id": "cca78e1f",
   "metadata": {},
   "outputs": [],
   "source": [
    "mccoy_SAV = mccoy_merge[mccoy_merge['delta_max'] >= 0.2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 503,
   "id": "7816324d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "16"
      ]
     },
     "execution_count": 503,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(mccoy_SAV)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 504,
   "id": "7f3b1357",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref_allele</th>\n",
       "      <th>alt_allele</th>\n",
       "      <th>delta_max</th>\n",
       "      <th>Vernot_allele_origin</th>\n",
       "      <th>Browning_allele_origin</th>\n",
       "      <th>annotation</th>\n",
       "      <th>_merge</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>chr1</td>\n",
       "      <td>22174518</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>0.98</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>HSPG2</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>chr1</td>\n",
       "      <td>55537474</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>0.33</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "      <td>USP24</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>chr1</td>\n",
       "      <td>161681848</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.20</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>FCRLA</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>90</th>\n",
       "      <td>chr1</td>\n",
       "      <td>212985592</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.52</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>TATDN3</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170</th>\n",
       "      <td>chr11</td>\n",
       "      <td>86159859</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.26</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>ancient</td>\n",
       "      <td>ME3</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>269</th>\n",
       "      <td>chr12</td>\n",
       "      <td>133272470</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>0.26</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>PXMP2</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>270</th>\n",
       "      <td>chr12</td>\n",
       "      <td>133272470</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>0.26</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>RP13-672B3.2</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>365</th>\n",
       "      <td>chr15</td>\n",
       "      <td>85403496</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.33</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>ALPK3</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>403</th>\n",
       "      <td>chr16</td>\n",
       "      <td>88924425</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>0.41</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>TRAPPC2L</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>449</th>\n",
       "      <td>chr19</td>\n",
       "      <td>40913595</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.23</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>PRX</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>581</th>\n",
       "      <td>chr22</td>\n",
       "      <td>50684852</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0.25</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>HDAC10</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>628</th>\n",
       "      <td>chr4</td>\n",
       "      <td>170990750</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.22</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>AADAT</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>640</th>\n",
       "      <td>chr5</td>\n",
       "      <td>6753013</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.24</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>PAPD7</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>683</th>\n",
       "      <td>chr5</td>\n",
       "      <td>167919825</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.38</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>RARS</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>697</th>\n",
       "      <td>chr6</td>\n",
       "      <td>33264115</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>0.34</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "      <td>RGL2</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>803</th>\n",
       "      <td>chr8</td>\n",
       "      <td>130763783</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.87</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>GSDMC</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     chrom        pos ref_allele alt_allele  delta_max  \\\n",
       "22    chr1   22174518          G          T       0.98   \n",
       "36    chr1   55537474          C          G       0.33   \n",
       "46    chr1  161681848          C          T       0.20   \n",
       "90    chr1  212985592          G          A       0.52   \n",
       "170  chr11   86159859          G          A       0.26   \n",
       "269  chr12  133272470          G          T       0.26   \n",
       "270  chr12  133272470          G          T       0.26   \n",
       "365  chr15   85403496          G          A       0.33   \n",
       "403  chr16   88924425          C          G       0.41   \n",
       "449  chr19   40913595          G          A       0.23   \n",
       "581  chr22   50684852          G          C       0.25   \n",
       "628   chr4  170990750          G          A       0.22   \n",
       "640   chr5    6753013          C          T       0.24   \n",
       "683   chr5  167919825          G          A       0.38   \n",
       "697   chr6   33264115          A          C       0.34   \n",
       "803   chr8  130763783          C          T       0.87   \n",
       "\n",
       "       Vernot_allele_origin  Browning_allele_origin    annotation _merge  \n",
       "22             introgressed            introgressed         HSPG2   both  \n",
       "36             introgressed  low-confidence ancient         USP24   both  \n",
       "46             introgressed            introgressed         FCRLA   both  \n",
       "90             introgressed            introgressed        TATDN3   both  \n",
       "170            introgressed                 ancient           ME3   both  \n",
       "269            introgressed            introgressed         PXMP2   both  \n",
       "270            introgressed            introgressed  RP13-672B3.2   both  \n",
       "365            introgressed            introgressed         ALPK3   both  \n",
       "403            introgressed            introgressed      TRAPPC2L   both  \n",
       "449  low-confidence ancient            introgressed           PRX   both  \n",
       "581            introgressed            introgressed        HDAC10   both  \n",
       "628            introgressed            introgressed         AADAT   both  \n",
       "640            introgressed            introgressed         PAPD7   both  \n",
       "683            introgressed            introgressed          RARS   both  \n",
       "697            introgressed  low-confidence ancient          RGL2   both  \n",
       "803            introgressed            introgressed         GSDMC   both  "
      ]
     },
     "execution_count": 504,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mccoy_SAV.head(16)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fb809a71",
   "metadata": {},
   "source": [
    "# Gene Expression <a class = 'anchor' id = 'geneexpression'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cef0ebe6",
   "metadata": {},
   "source": [
    "## sQTLs <a class = 'anchor' id = 'sqtls'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "145a1dc1",
   "metadata": {},
   "source": [
    "Now let's examine overlap between our variants and sQTLs from GTEx."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 505,
   "id": "f125e285",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>N</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sQTL</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>no</th>\n",
       "      <td>1359027</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>yes</th>\n",
       "      <td>248323</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            N\n",
       "sQTL         \n",
       "no    1359027\n",
       "yes    248323"
      ]
     },
     "execution_count": 505,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data.groupby('sQTL').size().to_frame('N')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 506,
   "id": "ed142853",
   "metadata": {},
   "outputs": [],
   "source": [
    "non_sQTLs = data[data['N_GTEx_tissues'].isnull()]\n",
    "sQTLs = data[~data['N_GTEx_tissues'].isnull()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 507,
   "id": "1a2a4eb5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1359027"
      ]
     },
     "execution_count": 507,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(non_sQTLs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 508,
   "id": "7f82f1f0",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "248323"
      ]
     },
     "execution_count": 508,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(sQTLs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 509,
   "id": "52797176",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.0"
      ]
     },
     "execution_count": 509,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sQTLs['delta_max'].min()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 510,
   "id": "17364e47",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1.0"
      ]
     },
     "execution_count": 510,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sQTLs['delta_max'].max()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 511,
   "id": "9add9c71",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.0025026787676935133"
      ]
     },
     "execution_count": 511,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sQTLs['delta_max'].mean() - non_sQTLs['delta_max'].mean()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 512,
   "id": "1451e6b7",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "MannwhitneyuResult(statistic=160822988988.5, pvalue=0.0)"
      ]
     },
     "execution_count": 512,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mannwhitneyu(non_sQTLs['delta_max'], sQTLs['delta_max'])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "15820ca5",
   "metadata": {},
   "source": [
    "Let's save the delta max for sQTLs just in case."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 513,
   "id": "f3836aa1",
   "metadata": {},
   "outputs": [],
   "source": [
    "sQTLs.to_csv('sQTLs.txt', sep=\"\\t\", header=True, index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c3f3ce95",
   "metadata": {},
   "source": [
    "Now let's look at SAV sQTLs."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 514,
   "id": "88973317",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>N</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sQTL</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>no</th>\n",
       "      <td>4569</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>yes</th>\n",
       "      <td>1381</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         N\n",
       "sQTL      \n",
       "no    4569\n",
       "yes   1381"
      ]
     },
     "execution_count": 514,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_2.groupby('sQTL').size().to_frame('N')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c966a540",
   "metadata": {},
   "source": [
    "How many possible SAVs are sQTLs?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 515,
   "id": "bc035b3d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.3668969181721573"
      ]
     },
     "execution_count": 515,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "1381/(5950-2186)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 516,
   "id": "4c62ba38",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>N_GTEx_tissues</th>\n",
       "      <th>N</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1.0</td>\n",
       "      <td>297</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2.0</td>\n",
       "      <td>130</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3.0</td>\n",
       "      <td>93</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4.0</td>\n",
       "      <td>76</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5.0</td>\n",
       "      <td>57</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>6.0</td>\n",
       "      <td>39</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>7.0</td>\n",
       "      <td>36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>8.0</td>\n",
       "      <td>45</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>9.0</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>10.0</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>11.0</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>12.0</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>13.0</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>14.0</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>15.0</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>16.0</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>17.0</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>18.0</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>19.0</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>20.0</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>21.0</td>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>22.0</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>23.0</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>24.0</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>25.0</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>26.0</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>27.0</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>28.0</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>29.0</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>30.0</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>31.0</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>32.0</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>33.0</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>34.0</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>35.0</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>36.0</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>37.0</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>38.0</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>39.0</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>40.0</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>41.0</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>42.0</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>43.0</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>44.0</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>45.0</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>46.0</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>47.0</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>48.0</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>49.0</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    N_GTEx_tissues    N\n",
       "0              1.0  297\n",
       "1              2.0  130\n",
       "2              3.0   93\n",
       "3              4.0   76\n",
       "4              5.0   57\n",
       "5              6.0   39\n",
       "6              7.0   36\n",
       "7              8.0   45\n",
       "8              9.0   32\n",
       "9             10.0   25\n",
       "10            11.0   29\n",
       "11            12.0   16\n",
       "12            13.0   14\n",
       "13            14.0   16\n",
       "14            15.0   17\n",
       "15            16.0   26\n",
       "16            17.0   20\n",
       "17            18.0   14\n",
       "18            19.0   15\n",
       "19            20.0   29\n",
       "20            21.0   21\n",
       "21            22.0   10\n",
       "22            23.0   14\n",
       "23            24.0   18\n",
       "24            25.0    8\n",
       "25            26.0   13\n",
       "26            27.0   13\n",
       "27            28.0   10\n",
       "28            29.0   17\n",
       "29            30.0    8\n",
       "30            31.0   16\n",
       "31            32.0   13\n",
       "32            33.0   16\n",
       "33            34.0    8\n",
       "34            35.0    8\n",
       "35            36.0   13\n",
       "36            37.0   13\n",
       "37            38.0   14\n",
       "38            39.0    4\n",
       "39            40.0    6\n",
       "40            41.0    9\n",
       "41            42.0    8\n",
       "42            43.0   13\n",
       "43            44.0    9\n",
       "44            45.0   12\n",
       "45            46.0   13\n",
       "46            47.0   11\n",
       "47            48.0   14\n",
       "48            49.0   23"
      ]
     },
     "execution_count": 516,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_2.groupby('N_GTEx_tissues').size().reset_index(name='N')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 517,
   "id": "c6b283f7",
   "metadata": {},
   "outputs": [],
   "source": [
    "non_SAV_sQTLs = data_2[data_2['N_GTEx_tissues'].isnull()]\n",
    "SAV_sQTLs = data_2[~data_2['N_GTEx_tissues'].isnull()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 518,
   "id": "c4986d03",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "4569"
      ]
     },
     "execution_count": 518,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(non_SAV_sQTLs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 519,
   "id": "3b2a564d",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1381"
      ]
     },
     "execution_count": 519,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(SAV_sQTLs)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2662eaf3",
   "metadata": {},
   "source": [
    "What does their distribution look like?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 520,
   "id": "9dac8161",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>N</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Vernot_allele_origin</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ancient</th>\n",
       "      <td>1145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>introgressed</th>\n",
       "      <td>52</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>low-confidence ancient</th>\n",
       "      <td>184</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                           N\n",
       "Vernot_allele_origin        \n",
       "ancient                 1145\n",
       "introgressed              52\n",
       "low-confidence ancient   184"
      ]
     },
     "execution_count": 520,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SAV_sQTLs.groupby(['Vernot_allele_origin']).size().to_frame('N')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 521,
   "id": "a4f3e3cc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>N</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Browning_allele_origin</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ancient</th>\n",
       "      <td>1113</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>introgressed</th>\n",
       "      <td>95</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>low-confidence ancient</th>\n",
       "      <td>173</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                           N\n",
       "Browning_allele_origin      \n",
       "ancient                 1113\n",
       "introgressed              95\n",
       "low-confidence ancient   173"
      ]
     },
     "execution_count": 521,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SAV_sQTLs.groupby(['Browning_allele_origin']).size().to_frame('N')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "22cd42aa",
   "metadata": {},
   "source": [
    "### sQTL Allele Origin Differences <a class = 'anchor' id = 'sqtlalleleorigindifferences'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f09b1bcf",
   "metadata": {},
   "source": [
    "Check the tissue distribution for tissue specific variants. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 522,
   "id": "058649b9",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Adipose_Subcutaneous</th>\n",
       "      <th>Adipose_Visceral_Omentum</th>\n",
       "      <th>Adrenal_Gland</th>\n",
       "      <th>Artery_Aorta</th>\n",
       "      <th>Artery_Coronary</th>\n",
       "      <th>Artery_Tibial</th>\n",
       "      <th>Brain_Amygdala</th>\n",
       "      <th>Brain_Anterior_cingulate_cortex_BA24</th>\n",
       "      <th>Brain_Caudate_basal_ganglia</th>\n",
       "      <th>Brain_Cerebellar_Hemisphere</th>\n",
       "      <th>Brain_Cerebellum</th>\n",
       "      <th>Brain_Cortex</th>\n",
       "      <th>Brain_Frontal_Cortex_BA9</th>\n",
       "      <th>Brain_Hippocampus</th>\n",
       "      <th>Brain_Hypothalamus</th>\n",
       "      <th>Brain_Nucleus_accumbens_basal_ganglia</th>\n",
       "      <th>Brain_Putamen_basal_ganglia</th>\n",
       "      <th>Brain_Spinal_cord_cervical_c-1</th>\n",
       "      <th>Brain_Substantia_nigra</th>\n",
       "      <th>Breast_Mammary_Tissue</th>\n",
       "      <th>Cells_Cultured_fibroblasts</th>\n",
       "      <th>Cells_EBV-transformed_lymphocytes</th>\n",
       "      <th>Colon_Sigmoid</th>\n",
       "      <th>Colon_Transverse</th>\n",
       "      <th>Esophagus_Gastroesophageal_Junction</th>\n",
       "      <th>Esophagus_Mucosa</th>\n",
       "      <th>Esophagus_Muscularis</th>\n",
       "      <th>Heart_Atrial_Appendage</th>\n",
       "      <th>Heart_Left_Ventricle</th>\n",
       "      <th>Kidney_Cortex</th>\n",
       "      <th>Liver</th>\n",
       "      <th>Lung</th>\n",
       "      <th>Minor_Salivary_Gland</th>\n",
       "      <th>Muscle_Skeletal</th>\n",
       "      <th>Nerve_Tibial</th>\n",
       "      <th>Ovary</th>\n",
       "      <th>Pancreas</th>\n",
       "      <th>Pituitary</th>\n",
       "      <th>Prostate</th>\n",
       "      <th>Skin_Not_Sun_Exposed_Suprapubic</th>\n",
       "      <th>Skin_Sun_Exposed_Lower_leg</th>\n",
       "      <th>Small_Intestine_Terminal_Ileum</th>\n",
       "      <th>Spleen</th>\n",
       "      <th>Stomach</th>\n",
       "      <th>Testis</th>\n",
       "      <th>Thyroid</th>\n",
       "      <th>Uterus</th>\n",
       "      <th>Vagina</th>\n",
       "      <th>Whole_Blood</th>\n",
       "      <th>N_GTEx_tissues</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>130</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>155</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>210</th>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>15.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>218</th>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>14.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>224</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>330</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>377</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>409</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>49.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     Adipose_Subcutaneous  Adipose_Visceral_Omentum  Adrenal_Gland  \\\n",
       "24                    NaN                       NaN            NaN   \n",
       "65                    NaN                       NaN            NaN   \n",
       "130                   NaN                       NaN            NaN   \n",
       "155                   NaN                       NaN            NaN   \n",
       "210                   0.0                       1.0            0.0   \n",
       "218                   0.0                       1.0            0.0   \n",
       "224                   NaN                       NaN            NaN   \n",
       "330                   0.0                       0.0            0.0   \n",
       "377                   NaN                       NaN            NaN   \n",
       "409                   1.0                       1.0            1.0   \n",
       "\n",
       "     Artery_Aorta  Artery_Coronary  Artery_Tibial  Brain_Amygdala  \\\n",
       "24            NaN              NaN            NaN             NaN   \n",
       "65            NaN              NaN            NaN             NaN   \n",
       "130           NaN              NaN            NaN             NaN   \n",
       "155           NaN              NaN            NaN             NaN   \n",
       "210           1.0              0.0            1.0             0.0   \n",
       "218           1.0              0.0            1.0             0.0   \n",
       "224           NaN              NaN            NaN             NaN   \n",
       "330           0.0              0.0            0.0             0.0   \n",
       "377           NaN              NaN            NaN             NaN   \n",
       "409           1.0              1.0            1.0             1.0   \n",
       "\n",
       "     Brain_Anterior_cingulate_cortex_BA24  Brain_Caudate_basal_ganglia  \\\n",
       "24                                    NaN                          NaN   \n",
       "65                                    NaN                          NaN   \n",
       "130                                   NaN                          NaN   \n",
       "155                                   NaN                          NaN   \n",
       "210                                   0.0                          0.0   \n",
       "218                                   0.0                          0.0   \n",
       "224                                   NaN                          NaN   \n",
       "330                                   0.0                          0.0   \n",
       "377                                   NaN                          NaN   \n",
       "409                                   1.0                          1.0   \n",
       "\n",
       "     Brain_Cerebellar_Hemisphere  Brain_Cerebellum  Brain_Cortex  \\\n",
       "24                           NaN               NaN           NaN   \n",
       "65                           NaN               NaN           NaN   \n",
       "130                          NaN               NaN           NaN   \n",
       "155                          NaN               NaN           NaN   \n",
       "210                          0.0               0.0           0.0   \n",
       "218                          0.0               0.0           0.0   \n",
       "224                          NaN               NaN           NaN   \n",
       "330                          0.0               1.0           0.0   \n",
       "377                          NaN               NaN           NaN   \n",
       "409                          1.0               1.0           1.0   \n",
       "\n",
       "     Brain_Frontal_Cortex_BA9  Brain_Hippocampus  Brain_Hypothalamus  \\\n",
       "24                        NaN                NaN                 NaN   \n",
       "65                        NaN                NaN                 NaN   \n",
       "130                       NaN                NaN                 NaN   \n",
       "155                       NaN                NaN                 NaN   \n",
       "210                       0.0                0.0                 0.0   \n",
       "218                       0.0                0.0                 0.0   \n",
       "224                       NaN                NaN                 NaN   \n",
       "330                       0.0                0.0                 0.0   \n",
       "377                       NaN                NaN                 NaN   \n",
       "409                       1.0                1.0                 1.0   \n",
       "\n",
       "     Brain_Nucleus_accumbens_basal_ganglia  Brain_Putamen_basal_ganglia  \\\n",
       "24                                     NaN                          NaN   \n",
       "65                                     NaN                          NaN   \n",
       "130                                    NaN                          NaN   \n",
       "155                                    NaN                          NaN   \n",
       "210                                    0.0                          0.0   \n",
       "218                                    0.0                          0.0   \n",
       "224                                    NaN                          NaN   \n",
       "330                                    0.0                          0.0   \n",
       "377                                    NaN                          NaN   \n",
       "409                                    1.0                          1.0   \n",
       "\n",
       "     Brain_Spinal_cord_cervical_c-1  Brain_Substantia_nigra  \\\n",
       "24                              NaN                     NaN   \n",
       "65                              NaN                     NaN   \n",
       "130                             NaN                     NaN   \n",
       "155                             NaN                     NaN   \n",
       "210                             0.0                     0.0   \n",
       "218                             0.0                     0.0   \n",
       "224                             NaN                     NaN   \n",
       "330                             0.0                     0.0   \n",
       "377                             NaN                     NaN   \n",
       "409                             1.0                     1.0   \n",
       "\n",
       "     Breast_Mammary_Tissue  Cells_Cultured_fibroblasts  \\\n",
       "24                     NaN                         NaN   \n",
       "65                     NaN                         NaN   \n",
       "130                    NaN                         NaN   \n",
       "155                    NaN                         NaN   \n",
       "210                    1.0                         0.0   \n",
       "218                    1.0                         0.0   \n",
       "224                    NaN                         NaN   \n",
       "330                    0.0                         1.0   \n",
       "377                    NaN                         NaN   \n",
       "409                    1.0                         1.0   \n",
       "\n",
       "     Cells_EBV-transformed_lymphocytes  Colon_Sigmoid  Colon_Transverse  \\\n",
       "24                                 NaN            NaN               NaN   \n",
       "65                                 NaN            NaN               NaN   \n",
       "130                                NaN            NaN               NaN   \n",
       "155                                NaN            NaN               NaN   \n",
       "210                                0.0            1.0               0.0   \n",
       "218                                0.0            1.0               0.0   \n",
       "224                                NaN            NaN               NaN   \n",
       "330                                0.0            0.0               0.0   \n",
       "377                                NaN            NaN               NaN   \n",
       "409                                1.0            1.0               1.0   \n",
       "\n",
       "     Esophagus_Gastroesophageal_Junction  Esophagus_Mucosa  \\\n",
       "24                                   NaN               NaN   \n",
       "65                                   NaN               NaN   \n",
       "130                                  NaN               NaN   \n",
       "155                                  NaN               NaN   \n",
       "210                                  1.0               1.0   \n",
       "218                                  1.0               1.0   \n",
       "224                                  NaN               NaN   \n",
       "330                                  0.0               0.0   \n",
       "377                                  NaN               NaN   \n",
       "409                                  1.0               1.0   \n",
       "\n",
       "     Esophagus_Muscularis  Heart_Atrial_Appendage  Heart_Left_Ventricle  \\\n",
       "24                    NaN                     NaN                   NaN   \n",
       "65                    NaN                     NaN                   NaN   \n",
       "130                   NaN                     NaN                   NaN   \n",
       "155                   NaN                     NaN                   NaN   \n",
       "210                   1.0                     0.0                   0.0   \n",
       "218                   1.0                     0.0                   0.0   \n",
       "224                   NaN                     NaN                   NaN   \n",
       "330                   0.0                     0.0                   0.0   \n",
       "377                   NaN                     NaN                   NaN   \n",
       "409                   1.0                     1.0                   1.0   \n",
       "\n",
       "     Kidney_Cortex  Liver  Lung  Minor_Salivary_Gland  Muscle_Skeletal  \\\n",
       "24             NaN    NaN   NaN                   NaN              NaN   \n",
       "65             NaN    NaN   NaN                   NaN              NaN   \n",
       "130            NaN    NaN   NaN                   NaN              NaN   \n",
       "155            NaN    NaN   NaN                   NaN              NaN   \n",
       "210            0.0    0.0   1.0                   0.0              0.0   \n",
       "218            0.0    0.0   1.0                   0.0              0.0   \n",
       "224            NaN    NaN   NaN                   NaN              NaN   \n",
       "330            0.0    0.0   0.0                   0.0              0.0   \n",
       "377            NaN    NaN   NaN                   NaN              NaN   \n",
       "409            1.0    1.0   1.0                   1.0              1.0   \n",
       "\n",
       "     Nerve_Tibial  Ovary  Pancreas  Pituitary  Prostate  \\\n",
       "24            NaN    NaN       NaN        NaN       NaN   \n",
       "65            NaN    NaN       NaN        NaN       NaN   \n",
       "130           NaN    NaN       NaN        NaN       NaN   \n",
       "155           NaN    NaN       NaN        NaN       NaN   \n",
       "210           1.0    0.0       0.0        0.0       0.0   \n",
       "218           1.0    0.0       1.0        0.0       0.0   \n",
       "224           NaN    NaN       NaN        NaN       NaN   \n",
       "330           0.0    0.0       0.0        1.0       0.0   \n",
       "377           NaN    NaN       NaN        NaN       NaN   \n",
       "409           1.0    1.0       1.0        1.0       1.0   \n",
       "\n",
       "     Skin_Not_Sun_Exposed_Suprapubic  Skin_Sun_Exposed_Lower_leg  \\\n",
       "24                               NaN                         NaN   \n",
       "65                               NaN                         NaN   \n",
       "130                              NaN                         NaN   \n",
       "155                              NaN                         NaN   \n",
       "210                              1.0                         1.0   \n",
       "218                              1.0                         1.0   \n",
       "224                              NaN                         NaN   \n",
       "330                              0.0                         0.0   \n",
       "377                              NaN                         NaN   \n",
       "409                              1.0                         1.0   \n",
       "\n",
       "     Small_Intestine_Terminal_Ileum  Spleen  Stomach  Testis  Thyroid  Uterus  \\\n",
       "24                              NaN     NaN      NaN     NaN      NaN     NaN   \n",
       "65                              NaN     NaN      NaN     NaN      NaN     NaN   \n",
       "130                             NaN     NaN      NaN     NaN      NaN     NaN   \n",
       "155                             NaN     NaN      NaN     NaN      NaN     NaN   \n",
       "210                             0.0     0.0      1.0     1.0      1.0     0.0   \n",
       "218                             0.0     0.0      0.0     0.0      1.0     0.0   \n",
       "224                             NaN     NaN      NaN     NaN      NaN     NaN   \n",
       "330                             0.0     0.0      0.0     1.0      0.0     0.0   \n",
       "377                             NaN     NaN      NaN     NaN      NaN     NaN   \n",
       "409                             1.0     1.0      1.0     1.0      1.0     1.0   \n",
       "\n",
       "     Vagina  Whole_Blood  N_GTEx_tissues  \n",
       "24      NaN          NaN             NaN  \n",
       "65      NaN          NaN             NaN  \n",
       "130     NaN          NaN             NaN  \n",
       "155     NaN          NaN             NaN  \n",
       "210     0.0          0.0            15.0  \n",
       "218     0.0          0.0            14.0  \n",
       "224     NaN          NaN             NaN  \n",
       "330     0.0          1.0             5.0  \n",
       "377     NaN          NaN             NaN  \n",
       "409     1.0          1.0            49.0  "
      ]
     },
     "execution_count": 522,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tissues = data_2[['Adipose_Subcutaneous','Adipose_Visceral_Omentum','Adrenal_Gland','Artery_Aorta','Artery_Coronary','Artery_Tibial','Brain_Amygdala','Brain_Anterior_cingulate_cortex_BA24','Brain_Caudate_basal_ganglia','Brain_Cerebellar_Hemisphere','Brain_Cerebellum','Brain_Cortex','Brain_Frontal_Cortex_BA9','Brain_Hippocampus','Brain_Hypothalamus','Brain_Nucleus_accumbens_basal_ganglia','Brain_Putamen_basal_ganglia','Brain_Spinal_cord_cervical_c-1','Brain_Substantia_nigra','Breast_Mammary_Tissue','Cells_Cultured_fibroblasts','Cells_EBV-transformed_lymphocytes','Colon_Sigmoid','Colon_Transverse','Esophagus_Gastroesophageal_Junction','Esophagus_Mucosa','Esophagus_Muscularis','Heart_Atrial_Appendage','Heart_Left_Ventricle','Kidney_Cortex','Liver','Lung','Minor_Salivary_Gland','Muscle_Skeletal','Nerve_Tibial','Ovary','Pancreas','Pituitary','Prostate','Skin_Not_Sun_Exposed_Suprapubic','Skin_Sun_Exposed_Lower_leg','Small_Intestine_Terminal_Ileum','Spleen','Stomach','Testis','Thyroid','Uterus','Vagina','Whole_Blood','N_GTEx_tissues']]\n",
    "tissues.head(10)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "27508bba",
   "metadata": {},
   "source": [
    "Let's get totals for the 49 tissues."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 523,
   "id": "979da744",
   "metadata": {},
   "outputs": [],
   "source": [
    "tissue_totals = tissues.sum()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b519443a",
   "metadata": {},
   "source": [
    "What if we compare ancient vs. introgressed?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 524,
   "id": "af07e76e",
   "metadata": {},
   "outputs": [],
   "source": [
    "Vernot_ancient_tissues = data_2[data_2['Vernot_allele_origin'] == 'ancient']\n",
    "Vernot_ancient_tissues = Vernot_ancient_tissues[['Adipose_Subcutaneous','Adipose_Visceral_Omentum','Adrenal_Gland','Artery_Aorta','Artery_Coronary','Artery_Tibial','Brain_Amygdala','Brain_Anterior_cingulate_cortex_BA24','Brain_Caudate_basal_ganglia','Brain_Cerebellar_Hemisphere','Brain_Cerebellum','Brain_Cortex','Brain_Frontal_Cortex_BA9','Brain_Hippocampus','Brain_Hypothalamus','Brain_Nucleus_accumbens_basal_ganglia','Brain_Putamen_basal_ganglia','Brain_Spinal_cord_cervical_c-1','Brain_Substantia_nigra','Breast_Mammary_Tissue','Cells_Cultured_fibroblasts','Cells_EBV-transformed_lymphocytes','Colon_Sigmoid','Colon_Transverse','Esophagus_Gastroesophageal_Junction','Esophagus_Mucosa','Esophagus_Muscularis','Heart_Atrial_Appendage','Heart_Left_Ventricle','Kidney_Cortex','Liver','Lung','Minor_Salivary_Gland','Muscle_Skeletal','Nerve_Tibial','Ovary','Pancreas','Pituitary','Prostate','Skin_Not_Sun_Exposed_Suprapubic','Skin_Sun_Exposed_Lower_leg','Small_Intestine_Terminal_Ileum','Spleen','Stomach','Testis','Thyroid','Uterus','Vagina','Whole_Blood']]\n",
    "Vernot_ancient_tissue_totals = Vernot_ancient_tissues.sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 525,
   "id": "611cd414",
   "metadata": {},
   "outputs": [],
   "source": [
    "Vernot_introgressed_tissues = data_2[data_2['Vernot_allele_origin'] == 'introgressed']\n",
    "Vernot_introgressed_tissues = Vernot_introgressed_tissues[['Adipose_Subcutaneous','Adipose_Visceral_Omentum','Adrenal_Gland','Artery_Aorta','Artery_Coronary','Artery_Tibial','Brain_Amygdala','Brain_Anterior_cingulate_cortex_BA24','Brain_Caudate_basal_ganglia','Brain_Cerebellar_Hemisphere','Brain_Cerebellum','Brain_Cortex','Brain_Frontal_Cortex_BA9','Brain_Hippocampus','Brain_Hypothalamus','Brain_Nucleus_accumbens_basal_ganglia','Brain_Putamen_basal_ganglia','Brain_Spinal_cord_cervical_c-1','Brain_Substantia_nigra','Breast_Mammary_Tissue','Cells_Cultured_fibroblasts','Cells_EBV-transformed_lymphocytes','Colon_Sigmoid','Colon_Transverse','Esophagus_Gastroesophageal_Junction','Esophagus_Mucosa','Esophagus_Muscularis','Heart_Atrial_Appendage','Heart_Left_Ventricle','Kidney_Cortex','Liver','Lung','Minor_Salivary_Gland','Muscle_Skeletal','Nerve_Tibial','Ovary','Pancreas','Pituitary','Prostate','Skin_Not_Sun_Exposed_Suprapubic','Skin_Sun_Exposed_Lower_leg','Small_Intestine_Terminal_Ileum','Spleen','Stomach','Testis','Thyroid','Uterus','Vagina','Whole_Blood']]\n",
    "Vernot_introgressed_tissue_totals = Vernot_introgressed_tissues.sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 526,
   "id": "92008e55",
   "metadata": {},
   "outputs": [],
   "source": [
    "Browning_ancient_tissues = data_2[data_2['Browning_allele_origin'] == 'ancient']\n",
    "Browning_ancient_tissues = Browning_ancient_tissues[['Adipose_Subcutaneous','Adipose_Visceral_Omentum','Adrenal_Gland','Artery_Aorta','Artery_Coronary','Artery_Tibial','Brain_Amygdala','Brain_Anterior_cingulate_cortex_BA24','Brain_Caudate_basal_ganglia','Brain_Cerebellar_Hemisphere','Brain_Cerebellum','Brain_Cortex','Brain_Frontal_Cortex_BA9','Brain_Hippocampus','Brain_Hypothalamus','Brain_Nucleus_accumbens_basal_ganglia','Brain_Putamen_basal_ganglia','Brain_Spinal_cord_cervical_c-1','Brain_Substantia_nigra','Breast_Mammary_Tissue','Cells_Cultured_fibroblasts','Cells_EBV-transformed_lymphocytes','Colon_Sigmoid','Colon_Transverse','Esophagus_Gastroesophageal_Junction','Esophagus_Mucosa','Esophagus_Muscularis','Heart_Atrial_Appendage','Heart_Left_Ventricle','Kidney_Cortex','Liver','Lung','Minor_Salivary_Gland','Muscle_Skeletal','Nerve_Tibial','Ovary','Pancreas','Pituitary','Prostate','Skin_Not_Sun_Exposed_Suprapubic','Skin_Sun_Exposed_Lower_leg','Small_Intestine_Terminal_Ileum','Spleen','Stomach','Testis','Thyroid','Uterus','Vagina','Whole_Blood']]\n",
    "Browning_ancient_tissue_totals = Browning_ancient_tissues.sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 527,
   "id": "2b201f3d",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "Browning_introgressed_tissues = data_2[data_2['Browning_allele_origin'] == 'introgressed']\n",
    "Browning_introgressed_tissues = Browning_introgressed_tissues[['Adipose_Subcutaneous','Adipose_Visceral_Omentum','Adrenal_Gland','Artery_Aorta','Artery_Coronary','Artery_Tibial','Brain_Amygdala','Brain_Anterior_cingulate_cortex_BA24','Brain_Caudate_basal_ganglia','Brain_Cerebellar_Hemisphere','Brain_Cerebellum','Brain_Cortex','Brain_Frontal_Cortex_BA9','Brain_Hippocampus','Brain_Hypothalamus','Brain_Nucleus_accumbens_basal_ganglia','Brain_Putamen_basal_ganglia','Brain_Spinal_cord_cervical_c-1','Brain_Substantia_nigra','Breast_Mammary_Tissue','Cells_Cultured_fibroblasts','Cells_EBV-transformed_lymphocytes','Colon_Sigmoid','Colon_Transverse','Esophagus_Gastroesophageal_Junction','Esophagus_Mucosa','Esophagus_Muscularis','Heart_Atrial_Appendage','Heart_Left_Ventricle','Kidney_Cortex','Liver','Lung','Minor_Salivary_Gland','Muscle_Skeletal','Nerve_Tibial','Ovary','Pancreas','Pituitary','Prostate','Skin_Not_Sun_Exposed_Suprapubic','Skin_Sun_Exposed_Lower_leg','Small_Intestine_Terminal_Ileum','Spleen','Stomach','Testis','Thyroid','Uterus','Vagina','Whole_Blood']]\n",
    "Browning_introgressed_tissue_totals = Browning_introgressed_tissues.sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 528,
   "id": "5dcf0325",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tissue</th>\n",
       "      <th>Vernot_ancient</th>\n",
       "      <th>Vernot_introgressed</th>\n",
       "      <th>Browning_ancient</th>\n",
       "      <th>Browning_introgressed</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Adipose_Subcutaneous</td>\n",
       "      <td>530.0</td>\n",
       "      <td>25.0</td>\n",
       "      <td>511.0</td>\n",
       "      <td>45.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Adipose_Visceral_Omentum</td>\n",
       "      <td>451.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>436.0</td>\n",
       "      <td>27.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Adrenal_Gland</td>\n",
       "      <td>279.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>275.0</td>\n",
       "      <td>11.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Artery_Aorta</td>\n",
       "      <td>433.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>423.0</td>\n",
       "      <td>27.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Artery_Coronary</td>\n",
       "      <td>259.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>258.0</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Artery_Tibial</td>\n",
       "      <td>525.0</td>\n",
       "      <td>19.0</td>\n",
       "      <td>508.0</td>\n",
       "      <td>40.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Brain_Amygdala</td>\n",
       "      <td>120.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>119.0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Brain_Anterior_cingulate_cortex_BA24</td>\n",
       "      <td>158.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>158.0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Brain_Caudate_basal_ganglia</td>\n",
       "      <td>226.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>225.0</td>\n",
       "      <td>8.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Brain_Cerebellar_Hemisphere</td>\n",
       "      <td>283.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>275.0</td>\n",
       "      <td>19.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                 tissue  Vernot_ancient  Vernot_introgressed  \\\n",
       "0                  Adipose_Subcutaneous           530.0                 25.0   \n",
       "1              Adipose_Visceral_Omentum           451.0                 12.0   \n",
       "2                         Adrenal_Gland           279.0                  6.0   \n",
       "3                          Artery_Aorta           433.0                 16.0   \n",
       "4                       Artery_Coronary           259.0                  5.0   \n",
       "5                         Artery_Tibial           525.0                 19.0   \n",
       "6                        Brain_Amygdala           120.0                  2.0   \n",
       "7  Brain_Anterior_cingulate_cortex_BA24           158.0                  3.0   \n",
       "8           Brain_Caudate_basal_ganglia           226.0                  7.0   \n",
       "9           Brain_Cerebellar_Hemisphere           283.0                 10.0   \n",
       "\n",
       "   Browning_ancient  Browning_introgressed  \n",
       "0             511.0                   45.0  \n",
       "1             436.0                   27.0  \n",
       "2             275.0                   11.0  \n",
       "3             423.0                   27.0  \n",
       "4             258.0                    6.0  \n",
       "5             508.0                   40.0  \n",
       "6             119.0                    3.0  \n",
       "7             158.0                    3.0  \n",
       "8             225.0                    8.0  \n",
       "9             275.0                   19.0  "
      ]
     },
     "execution_count": 528,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "allele_origin_tissue_counts = pd.concat({'Vernot_ancient': Vernot_ancient_tissue_totals, 'Vernot_introgressed': Vernot_introgressed_tissue_totals, 'Browning_ancient': Browning_ancient_tissue_totals, 'Browning_introgressed': Browning_introgressed_tissue_totals}, axis = 1)\n",
    "allele_origin_tissue_counts['tissue'] = allele_origin_tissue_counts.index\n",
    "allele_origin_tissue_counts = allele_origin_tissue_counts.reset_index()\n",
    "allele_origin_tissue_counts = allele_origin_tissue_counts[['tissue','Vernot_ancient','Vernot_introgressed','Browning_ancient','Browning_introgressed']]\n",
    "allele_origin_tissue_counts.head(10)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "661d68ee",
   "metadata": {},
   "source": [
    "Let's also add the GTEx sample size."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 529,
   "id": "d83ef9d9",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tissue</th>\n",
       "      <th>Vernot_ancient</th>\n",
       "      <th>Vernot_introgressed</th>\n",
       "      <th>Browning_ancient</th>\n",
       "      <th>Browning_introgressed</th>\n",
       "      <th>sample_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Adipose_Subcutaneous</td>\n",
       "      <td>530.0</td>\n",
       "      <td>25.0</td>\n",
       "      <td>511.0</td>\n",
       "      <td>45.0</td>\n",
       "      <td>663</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Adipose_Visceral_Omentum</td>\n",
       "      <td>451.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>436.0</td>\n",
       "      <td>27.0</td>\n",
       "      <td>541</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Adrenal_Gland</td>\n",
       "      <td>279.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>275.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>258</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Artery_Aorta</td>\n",
       "      <td>433.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>423.0</td>\n",
       "      <td>27.0</td>\n",
       "      <td>432</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Artery_Coronary</td>\n",
       "      <td>259.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>258.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>240</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                     tissue  Vernot_ancient  Vernot_introgressed  \\\n",
       "0      Adipose_Subcutaneous           530.0                 25.0   \n",
       "1  Adipose_Visceral_Omentum           451.0                 12.0   \n",
       "2             Adrenal_Gland           279.0                  6.0   \n",
       "3              Artery_Aorta           433.0                 16.0   \n",
       "4           Artery_Coronary           259.0                  5.0   \n",
       "\n",
       "   Browning_ancient  Browning_introgressed  sample_size  \n",
       "0             511.0                   45.0          663  \n",
       "1             436.0                   27.0          541  \n",
       "2             275.0                   11.0          258  \n",
       "3             423.0                   27.0          432  \n",
       "4             258.0                    6.0          240  "
      ]
     },
     "execution_count": 529,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "allele_origin_tissue_counts['sample_size'] = [663,541,258,432,240,663,152,176,246,215,241,255,209,197,202,246,205,159,139,459,504,174,373,406,375,555,515,429,432,85,226,578,162,803,619,180,328,283,245,604,701,187,241,359,361,653,142,156,755]\n",
    "allele_origin_tissue_counts.head(5)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e277bc13",
   "metadata": {},
   "source": [
    "Check the correlations between each allele origin and sample size."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 530,
   "id": "6f8dc34d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.9133729685808691 5.732392167246112e-20\n"
     ]
    }
   ],
   "source": [
    "rho, p = spearmanr(allele_origin_tissue_counts['Browning_ancient'], allele_origin_tissue_counts['sample_size'])\n",
    "print(rho,p)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 531,
   "id": "c20c1180",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.8864031196766793 2.4665240198700857e-17\n"
     ]
    }
   ],
   "source": [
    "rho, p = spearmanr(allele_origin_tissue_counts['Browning_introgressed'], allele_origin_tissue_counts['sample_size'])\n",
    "print(rho,p)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 532,
   "id": "c2edf9b6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.9118742676565625 8.434829773157069e-20\n"
     ]
    }
   ],
   "source": [
    "rho, p = spearmanr(allele_origin_tissue_counts['Vernot_ancient'], allele_origin_tissue_counts['sample_size'])\n",
    "print(rho,p)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 533,
   "id": "d216fb01",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.848729659663223 1.3402094505815186e-14\n"
     ]
    }
   ],
   "source": [
    "rho, p = spearmanr(allele_origin_tissue_counts['Vernot_introgressed'], allele_origin_tissue_counts['sample_size'])\n",
    "print(rho,p)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d998bad7",
   "metadata": {},
   "source": [
    "Now for correlations among the Vernot and Browning allele origins."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 534,
   "id": "a74050e8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.9432898208149728 3.797623224449068e-24\n"
     ]
    }
   ],
   "source": [
    "rho, p = spearmanr(allele_origin_tissue_counts['Browning_ancient'], allele_origin_tissue_counts['Browning_introgressed'])\n",
    "print(rho,p)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 535,
   "id": "a3719f2f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.9195145487047715 1.090872224680547e-20\n"
     ]
    }
   ],
   "source": [
    "rho, p = spearmanr(allele_origin_tissue_counts['Vernot_ancient'], allele_origin_tissue_counts['Vernot_introgressed'])\n",
    "print(rho,p)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5f93b99d",
   "metadata": {},
   "source": [
    "Save this dataframe."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 536,
   "id": "55cbd0af",
   "metadata": {},
   "outputs": [],
   "source": [
    "allele_origin_tissue_counts.to_csv('../GTEx_sQTLs/sQTLs_allele_origin_tissue_counts.txt', sep = '\\t', header = True, index = False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "885dcee0",
   "metadata": {},
   "source": [
    "### Tissue Specifcity <a class = 'anchor' id = 'tissuespecificity'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5d112288",
   "metadata": {},
   "source": [
    "Now let's look at tissue specificity."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 537,
   "id": "6ea7db22",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>N_sQTLs</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Adipose_Subcutaneous</th>\n",
       "      <td>14.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Adipose_Visceral_Omentum</th>\n",
       "      <td>12.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Adrenal_Gland</th>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Artery_Aorta</th>\n",
       "      <td>11.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Artery_Coronary</th>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Artery_Tibial</th>\n",
       "      <td>29.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Brain_Amygdala</th>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Brain_Anterior_cingulate_cortex_BA24</th>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Brain_Caudate_basal_ganglia</th>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Brain_Cerebellar_Hemisphere</th>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Brain_Cerebellum</th>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Brain_Cortex</th>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Brain_Frontal_Cortex_BA9</th>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Brain_Hippocampus</th>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Brain_Hypothalamus</th>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Brain_Nucleus_accumbens_basal_ganglia</th>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Brain_Putamen_basal_ganglia</th>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Brain_Spinal_cord_cervical_c-1</th>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Brain_Substantia_nigra</th>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Breast_Mammary_Tissue</th>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Cells_Cultured_fibroblasts</th>\n",
       "      <td>29.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Cells_EBV-transformed_lymphocytes</th>\n",
       "      <td>8.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Colon_Sigmoid</th>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Colon_Transverse</th>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Esophagus_Gastroesophageal_Junction</th>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Esophagus_Mucosa</th>\n",
       "      <td>13.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Esophagus_Muscularis</th>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Heart_Atrial_Appendage</th>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Heart_Left_Ventricle</th>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Kidney_Cortex</th>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Liver</th>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Lung</th>\n",
       "      <td>17.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Minor_Salivary_Gland</th>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Muscle_Skeletal</th>\n",
       "      <td>54.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Nerve_Tibial</th>\n",
       "      <td>22.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Ovary</th>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Pancreas</th>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Pituitary</th>\n",
       "      <td>14.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Prostate</th>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Skin_Not_Sun_Exposed_Suprapubic</th>\n",
       "      <td>20.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Skin_Sun_Exposed_Lower_leg</th>\n",
       "      <td>25.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Small_Intestine_Terminal_Ileum</th>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Spleen</th>\n",
       "      <td>11.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Stomach</th>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Testis</th>\n",
       "      <td>144.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Thyroid</th>\n",
       "      <td>35.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Uterus</th>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Vagina</th>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Whole_Blood</th>\n",
       "      <td>29.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>N_GTEx_tissues</th>\n",
       "      <td>557.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                       N_sQTLs\n",
       "Adipose_Subcutaneous                      14.0\n",
       "Adipose_Visceral_Omentum                  12.0\n",
       "Adrenal_Gland                              1.0\n",
       "Artery_Aorta                              11.0\n",
       "Artery_Coronary                            1.0\n",
       "Artery_Tibial                             29.0\n",
       "Brain_Amygdala                             1.0\n",
       "Brain_Anterior_cingulate_cortex_BA24       2.0\n",
       "Brain_Caudate_basal_ganglia                2.0\n",
       "Brain_Cerebellar_Hemisphere                5.0\n",
       "Brain_Cerebellum                           6.0\n",
       "Brain_Cortex                               3.0\n",
       "Brain_Frontal_Cortex_BA9                   0.0\n",
       "Brain_Hippocampus                          0.0\n",
       "Brain_Hypothalamus                         0.0\n",
       "Brain_Nucleus_accumbens_basal_ganglia      4.0\n",
       "Brain_Putamen_basal_ganglia                0.0\n",
       "Brain_Spinal_cord_cervical_c-1             0.0\n",
       "Brain_Substantia_nigra                     0.0\n",
       "Breast_Mammary_Tissue                      3.0\n",
       "Cells_Cultured_fibroblasts                29.0\n",
       "Cells_EBV-transformed_lymphocytes          8.0\n",
       "Colon_Sigmoid                              6.0\n",
       "Colon_Transverse                           6.0\n",
       "Esophagus_Gastroesophageal_Junction        3.0\n",
       "Esophagus_Mucosa                          13.0\n",
       "Esophagus_Muscularis                       6.0\n",
       "Heart_Atrial_Appendage                     2.0\n",
       "Heart_Left_Ventricle                       2.0\n",
       "Kidney_Cortex                              0.0\n",
       "Liver                                      5.0\n",
       "Lung                                      17.0\n",
       "Minor_Salivary_Gland                       1.0\n",
       "Muscle_Skeletal                           54.0\n",
       "Nerve_Tibial                              22.0\n",
       "Ovary                                      0.0\n",
       "Pancreas                                   2.0\n",
       "Pituitary                                 14.0\n",
       "Prostate                                   2.0\n",
       "Skin_Not_Sun_Exposed_Suprapubic           20.0\n",
       "Skin_Sun_Exposed_Lower_leg                25.0\n",
       "Small_Intestine_Terminal_Ileum             3.0\n",
       "Spleen                                    11.0\n",
       "Stomach                                    3.0\n",
       "Testis                                   144.0\n",
       "Thyroid                                   35.0\n",
       "Uterus                                     1.0\n",
       "Vagina                                     0.0\n",
       "Whole_Blood                               29.0\n",
       "N_GTEx_tissues                           557.0"
      ]
     },
     "execution_count": 537,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tissues[tissues['N_GTEx_tissues'] < 3].sum().to_frame('N_sQTLs')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 538,
   "id": "9e5335bd",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>N_sQTLs</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Vernot_allele_origin</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ancient</th>\n",
       "      <td>107</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>low-confidence ancient</th>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                        N_sQTLs\n",
       "Vernot_allele_origin           \n",
       "ancient                     107\n",
       "low-confidence ancient        5"
      ]
     },
     "execution_count": 538,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_2[data_2['N_GTEx_tissues'] > 40].groupby('Vernot_allele_origin').size().to_frame('N_sQTLs')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 539,
   "id": "6d029e00",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>N_sQTLs</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Browning_allele_origin</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ancient</th>\n",
       "      <td>107</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>low-confidence ancient</th>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                        N_sQTLs\n",
       "Browning_allele_origin         \n",
       "ancient                     107\n",
       "low-confidence ancient        5"
      ]
     },
     "execution_count": 539,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_2[data_2['N_GTEx_tissues'] > 40].groupby('Browning_allele_origin').size().to_frame('N_sQTLs')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 540,
   "id": "41b1b437",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>N</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>N_GTEx_tissues</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1.0</th>\n",
       "      <td>225</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2.0</th>\n",
       "      <td>98</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3.0</th>\n",
       "      <td>76</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4.0</th>\n",
       "      <td>60</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5.0</th>\n",
       "      <td>48</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6.0</th>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7.0</th>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8.0</th>\n",
       "      <td>39</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9.0</th>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10.0</th>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11.0</th>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12.0</th>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13.0</th>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14.0</th>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15.0</th>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16.0</th>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17.0</th>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18.0</th>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19.0</th>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20.0</th>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21.0</th>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22.0</th>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23.0</th>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24.0</th>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25.0</th>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26.0</th>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27.0</th>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28.0</th>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29.0</th>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30.0</th>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31.0</th>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32.0</th>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33.0</th>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34.0</th>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35.0</th>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36.0</th>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37.0</th>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38.0</th>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39.0</th>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40.0</th>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41.0</th>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42.0</th>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43.0</th>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44.0</th>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45.0</th>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46.0</th>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47.0</th>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48.0</th>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49.0</th>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  N\n",
       "N_GTEx_tissues     \n",
       "1.0             225\n",
       "2.0              98\n",
       "3.0              76\n",
       "4.0              60\n",
       "5.0              48\n",
       "6.0              30\n",
       "7.0              32\n",
       "8.0              39\n",
       "9.0              25\n",
       "10.0             21\n",
       "11.0             27\n",
       "12.0             13\n",
       "13.0             12\n",
       "14.0             15\n",
       "15.0             14\n",
       "16.0             23\n",
       "17.0             18\n",
       "18.0             11\n",
       "19.0             14\n",
       "20.0             24\n",
       "21.0             15\n",
       "22.0              9\n",
       "23.0             14\n",
       "24.0             17\n",
       "25.0              7\n",
       "26.0             10\n",
       "27.0             13\n",
       "28.0              8\n",
       "29.0             16\n",
       "30.0              6\n",
       "31.0             15\n",
       "32.0             10\n",
       "33.0             13\n",
       "34.0              7\n",
       "35.0              5\n",
       "36.0             13\n",
       "37.0             12\n",
       "38.0             14\n",
       "39.0              3\n",
       "40.0              6\n",
       "41.0              9\n",
       "42.0              8\n",
       "43.0             13\n",
       "44.0              6\n",
       "45.0             12\n",
       "46.0             12\n",
       "47.0             11\n",
       "48.0             14\n",
       "49.0             22"
      ]
     },
     "execution_count": 540,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_2[data_2['Vernot_allele_origin'] == 'ancient'].groupby('N_GTEx_tissues').size().to_frame('N')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 541,
   "id": "9efce3f9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>N</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>N_GTEx_tissues</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1.0</th>\n",
       "      <td>220</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2.0</th>\n",
       "      <td>98</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3.0</th>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4.0</th>\n",
       "      <td>56</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5.0</th>\n",
       "      <td>47</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6.0</th>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7.0</th>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8.0</th>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9.0</th>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10.0</th>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11.0</th>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12.0</th>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13.0</th>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14.0</th>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15.0</th>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16.0</th>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17.0</th>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18.0</th>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19.0</th>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20.0</th>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21.0</th>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22.0</th>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23.0</th>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24.0</th>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25.0</th>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26.0</th>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27.0</th>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28.0</th>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29.0</th>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30.0</th>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31.0</th>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32.0</th>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33.0</th>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34.0</th>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35.0</th>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36.0</th>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37.0</th>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38.0</th>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39.0</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40.0</th>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41.0</th>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42.0</th>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43.0</th>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44.0</th>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45.0</th>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46.0</th>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47.0</th>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48.0</th>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49.0</th>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  N\n",
       "N_GTEx_tissues     \n",
       "1.0             220\n",
       "2.0              98\n",
       "3.0              75\n",
       "4.0              56\n",
       "5.0              47\n",
       "6.0              30\n",
       "7.0              31\n",
       "8.0              37\n",
       "9.0              25\n",
       "10.0             21\n",
       "11.0             23\n",
       "12.0             12\n",
       "13.0             10\n",
       "14.0             14\n",
       "15.0             13\n",
       "16.0             20\n",
       "17.0             18\n",
       "18.0             10\n",
       "19.0             13\n",
       "20.0             24\n",
       "21.0             15\n",
       "22.0              8\n",
       "23.0             13\n",
       "24.0             17\n",
       "25.0              6\n",
       "26.0             10\n",
       "27.0             13\n",
       "28.0              8\n",
       "29.0             16\n",
       "30.0              6\n",
       "31.0             15\n",
       "32.0             10\n",
       "33.0             13\n",
       "34.0              7\n",
       "35.0              5\n",
       "36.0             13\n",
       "37.0             12\n",
       "38.0             14\n",
       "39.0              2\n",
       "40.0              6\n",
       "41.0              9\n",
       "42.0              8\n",
       "43.0             13\n",
       "44.0              6\n",
       "45.0             12\n",
       "46.0             12\n",
       "47.0             11\n",
       "48.0             14\n",
       "49.0             22"
      ]
     },
     "execution_count": 541,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_2[data_2['Browning_allele_origin'] == 'ancient'].groupby('N_GTEx_tissues').size().to_frame('N')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 542,
   "id": "cf04d8f9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>N_sQTLs</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>distribution</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Altai</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Chagyrskaya</th>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Denisovan</th>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Late Neanderthal</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Neanderthal</th>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Other</th>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Shared</th>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  N_sQTLs\n",
       "distribution             \n",
       "Altai                   2\n",
       "Chagyrskaya             3\n",
       "Denisovan               8\n",
       "Late Neanderthal        2\n",
       "Neanderthal             9\n",
       "Other                  14\n",
       "Shared                 74"
      ]
     },
     "execution_count": 542,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_2[data_2['N_GTEx_tissues'] > 40].groupby('distribution').size().to_frame('N_sQTLs')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 543,
   "id": "cc6d897f",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>N</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>N_GTEx_tissues</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1.0</th>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2.0</th>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3.0</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4.0</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5.0</th>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6.0</th>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7.0</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9.0</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15.0</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16.0</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17.0</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18.0</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19.0</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20.0</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21.0</th>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28.0</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31.0</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33.0</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39.0</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                 N\n",
       "N_GTEx_tissues    \n",
       "1.0             12\n",
       "2.0             11\n",
       "3.0              1\n",
       "4.0              2\n",
       "5.0              4\n",
       "6.0              3\n",
       "7.0              1\n",
       "9.0              2\n",
       "15.0             1\n",
       "16.0             2\n",
       "17.0             1\n",
       "18.0             1\n",
       "19.0             1\n",
       "20.0             1\n",
       "21.0             4\n",
       "28.0             1\n",
       "31.0             1\n",
       "33.0             2\n",
       "39.0             1"
      ]
     },
     "execution_count": 543,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_2[data_2['Vernot_allele_origin'] == 'introgressed'].groupby('N_GTEx_tissues').size().to_frame('N')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 544,
   "id": "e3516c95",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>N</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>N_GTEx_tissues</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1.0</th>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2.0</th>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3.0</th>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4.0</th>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5.0</th>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6.0</th>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7.0</th>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8.0</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9.0</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11.0</th>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12.0</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13.0</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14.0</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15.0</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16.0</th>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17.0</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18.0</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19.0</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20.0</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21.0</th>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22.0</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23.0</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25.0</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28.0</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31.0</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33.0</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39.0</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                 N\n",
       "N_GTEx_tissues    \n",
       "1.0             20\n",
       "2.0             13\n",
       "3.0              3\n",
       "4.0              7\n",
       "5.0              6\n",
       "6.0              5\n",
       "7.0              3\n",
       "8.0              2\n",
       "9.0              2\n",
       "11.0             4\n",
       "12.0             1\n",
       "13.0             2\n",
       "14.0             1\n",
       "15.0             2\n",
       "16.0             5\n",
       "17.0             1\n",
       "18.0             2\n",
       "19.0             2\n",
       "20.0             1\n",
       "21.0             4\n",
       "22.0             1\n",
       "23.0             1\n",
       "25.0             1\n",
       "28.0             1\n",
       "31.0             1\n",
       "33.0             2\n",
       "39.0             2"
      ]
     },
     "execution_count": 544,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_2[data_2['Browning_allele_origin'] == 'introgressed'].groupby('N_GTEx_tissues').size().to_frame('N')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "46f7438e",
   "metadata": {},
   "source": [
    "Do ancient and introgressed sQTLs differ in the average number of tissues? First let's look at Vernot."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 545,
   "id": "ba42a0d2",
   "metadata": {},
   "outputs": [],
   "source": [
    "Vernot_ancient_SAV_sQTLs = SAV_sQTLs[SAV_sQTLs['Vernot_allele_origin'] == 'ancient']\n",
    "Vernot_introgressed_SAV_sQTLs = SAV_sQTLs[SAV_sQTLs['Vernot_allele_origin'] == 'introgressed']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 546,
   "id": "5c92568a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1145"
      ]
     },
     "execution_count": 546,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(Vernot_ancient_SAV_sQTLs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 547,
   "id": "b8927dc5",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "52"
      ]
     },
     "execution_count": 547,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(Vernot_introgressed_SAV_sQTLs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 548,
   "id": "86e6b785",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "4.922119583473295"
      ]
     },
     "execution_count": 548,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Vernot_ancient_SAV_sQTLs['N_GTEx_tissues'].mean() - Vernot_introgressed_SAV_sQTLs['N_GTEx_tissues'].mean()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 549,
   "id": "30db6bec",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "MannwhitneyuResult(statistic=35123.5, pvalue=0.02740137285965415)"
      ]
     },
     "execution_count": 549,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mannwhitneyu(Vernot_ancient_SAV_sQTLs['N_GTEx_tissues'], Vernot_introgressed_SAV_sQTLs['N_GTEx_tissues'])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3e4c8818",
   "metadata": {},
   "source": [
    "Now Browning."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 550,
   "id": "c50b856a",
   "metadata": {},
   "outputs": [],
   "source": [
    "Browning_ancient_SAV_sQTLs = SAV_sQTLs[SAV_sQTLs['Browning_allele_origin'] == 'ancient']\n",
    "Browning_introgressed_SAV_sQTLs = SAV_sQTLs[SAV_sQTLs['Browning_allele_origin'] == 'introgressed']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 551,
   "id": "625c50ef",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1113"
      ]
     },
     "execution_count": 551,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(Browning_ancient_SAV_sQTLs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 552,
   "id": "8637ec09",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "95"
      ]
     },
     "execution_count": 552,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(Browning_introgressed_SAV_sQTLs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 553,
   "id": "a09f0349",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "5.0007471508961086"
      ]
     },
     "execution_count": 553,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Browning_ancient_SAV_sQTLs['N_GTEx_tissues'].mean() - Browning_introgressed_SAV_sQTLs['N_GTEx_tissues'].mean()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 554,
   "id": "6be736f7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "MannwhitneyuResult(statistic=60337.0, pvalue=0.02149969439234634)"
      ]
     },
     "execution_count": 554,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mannwhitneyu(Browning_ancient_SAV_sQTLs['N_GTEx_tissues'], Browning_introgressed_SAV_sQTLs['N_GTEx_tissues'])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cc8c43a1",
   "metadata": {},
   "source": [
    "### European sQTLs <a class = 'anchor' id = 'eursqtls'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9824d1cd",
   "metadata": {},
   "source": [
    "Let's take a quick peek at European allele frequencies and N sQTL SAVs given that a lot of GTEx samples come from individuals with European ancestry."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 555,
   "id": "4c973e52",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>1KG_EUR_AF</th>\n",
       "      <th>N_SAVs</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>100</th>\n",
       "      <td>1.00</td>\n",
       "      <td>171</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>99</th>\n",
       "      <td>0.99</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>98</th>\n",
       "      <td>0.98</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>0.97</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96</th>\n",
       "      <td>0.96</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95</th>\n",
       "      <td>0.95</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>94</th>\n",
       "      <td>0.94</td>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>93</th>\n",
       "      <td>0.93</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92</th>\n",
       "      <td>0.92</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91</th>\n",
       "      <td>0.91</td>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>90</th>\n",
       "      <td>0.90</td>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89</th>\n",
       "      <td>0.89</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88</th>\n",
       "      <td>0.88</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87</th>\n",
       "      <td>0.87</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86</th>\n",
       "      <td>0.86</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>85</th>\n",
       "      <td>0.85</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84</th>\n",
       "      <td>0.84</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83</th>\n",
       "      <td>0.83</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82</th>\n",
       "      <td>0.82</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>81</th>\n",
       "      <td>0.81</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80</th>\n",
       "      <td>0.80</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79</th>\n",
       "      <td>0.79</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78</th>\n",
       "      <td>0.78</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>77</th>\n",
       "      <td>0.77</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76</th>\n",
       "      <td>0.76</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75</th>\n",
       "      <td>0.75</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74</th>\n",
       "      <td>0.74</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73</th>\n",
       "      <td>0.73</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72</th>\n",
       "      <td>0.72</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71</th>\n",
       "      <td>0.71</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70</th>\n",
       "      <td>0.70</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69</th>\n",
       "      <td>0.69</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>68</th>\n",
       "      <td>0.68</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67</th>\n",
       "      <td>0.67</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66</th>\n",
       "      <td>0.66</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>0.65</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>0.64</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>0.63</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>0.62</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>0.61</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>0.60</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>0.59</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58</th>\n",
       "      <td>0.58</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>0.57</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>0.56</td>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>0.55</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>0.54</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>0.53</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>0.52</td>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>0.51</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>0.50</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>0.49</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>0.48</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>0.47</td>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>0.46</td>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>0.45</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>0.44</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>0.43</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>0.42</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>0.41</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>0.40</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>0.39</td>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>0.38</td>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>0.37</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>0.36</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>0.35</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>0.34</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>0.33</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>0.32</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>0.31</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>0.30</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>0.29</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>0.28</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>0.27</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>0.26</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>0.25</td>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>0.24</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>0.23</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>0.22</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>0.21</td>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>0.20</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>0.19</td>\n",
       "      <td>33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>0.18</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>0.17</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>0.16</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>0.15</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>0.14</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>0.13</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>0.12</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>0.11</td>\n",
       "      <td>35</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>0.10</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>0.09</td>\n",
       "      <td>55</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>0.08</td>\n",
       "      <td>40</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>0.07</td>\n",
       "      <td>40</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>0.06</td>\n",
       "      <td>43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>0.05</td>\n",
       "      <td>47</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0.04</td>\n",
       "      <td>81</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.03</td>\n",
       "      <td>86</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.02</td>\n",
       "      <td>93</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.01</td>\n",
       "      <td>140</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0.00</td>\n",
       "      <td>985</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     1KG_EUR_AF  N_SAVs\n",
       "100        1.00     171\n",
       "99         0.99      25\n",
       "98         0.98      19\n",
       "97         0.97      11\n",
       "96         0.96      15\n",
       "95         0.95      16\n",
       "94         0.94      21\n",
       "93         0.93      11\n",
       "92         0.92      14\n",
       "91         0.91      21\n",
       "90         0.90      21\n",
       "89         0.89      13\n",
       "88         0.88      12\n",
       "87         0.87      16\n",
       "86         0.86      12\n",
       "85         0.85      14\n",
       "84         0.84      17\n",
       "83         0.83      17\n",
       "82         0.82      17\n",
       "81         0.81      12\n",
       "80         0.80      26\n",
       "79         0.79      14\n",
       "78         0.78      17\n",
       "77         0.77      11\n",
       "76         0.76      11\n",
       "75         0.75      24\n",
       "74         0.74      17\n",
       "73         0.73      25\n",
       "72         0.72      19\n",
       "71         0.71      16\n",
       "70         0.70      13\n",
       "69         0.69      12\n",
       "68         0.68      17\n",
       "67         0.67      17\n",
       "66         0.66      17\n",
       "65         0.65      15\n",
       "64         0.64      18\n",
       "63         0.63      25\n",
       "62         0.62      14\n",
       "61         0.61      19\n",
       "60         0.60      14\n",
       "59         0.59      11\n",
       "58         0.58      17\n",
       "57         0.57      20\n",
       "56         0.56      22\n",
       "55         0.55      18\n",
       "54         0.54      12\n",
       "53         0.53      14\n",
       "52         0.52      21\n",
       "51         0.51      12\n",
       "50         0.50      26\n",
       "49         0.49      18\n",
       "48         0.48      13\n",
       "47         0.47      21\n",
       "46         0.46      21\n",
       "45         0.45      30\n",
       "44         0.44      16\n",
       "43         0.43      20\n",
       "42         0.42      18\n",
       "41         0.41      13\n",
       "40         0.40      24\n",
       "39         0.39      22\n",
       "38         0.38      21\n",
       "37         0.37      18\n",
       "36         0.36      23\n",
       "35         0.35       8\n",
       "34         0.34      19\n",
       "33         0.33      18\n",
       "32         0.32      20\n",
       "31         0.31      24\n",
       "30         0.30      29\n",
       "29         0.29      23\n",
       "28         0.28      15\n",
       "27         0.27      27\n",
       "26         0.26      20\n",
       "25         0.25      22\n",
       "24         0.24      13\n",
       "23         0.23      27\n",
       "22         0.22      26\n",
       "21         0.21      22\n",
       "20         0.20      27\n",
       "19         0.19      33\n",
       "18         0.18      27\n",
       "17         0.17      24\n",
       "16         0.16      26\n",
       "15         0.15      29\n",
       "14         0.14      23\n",
       "13         0.13      32\n",
       "12         0.12      29\n",
       "11         0.11      35\n",
       "10         0.10      30\n",
       "9          0.09      55\n",
       "8          0.08      40\n",
       "7          0.07      40\n",
       "6          0.06      43\n",
       "5          0.05      47\n",
       "4          0.04      81\n",
       "3          0.03      86\n",
       "2          0.02      93\n",
       "1          0.01     140\n",
       "0          0.00     985"
      ]
     },
     "execution_count": 555,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "EUR_SAVs = data_2.groupby('1KG_EUR_AF').size().to_frame('N_SAVs')\n",
    "EUR_SAVs = EUR_SAVs.reset_index()\n",
    "EUR_SAVs = EUR_SAVs.sort_values('1KG_EUR_AF', ascending=False)\n",
    "EUR_SAVs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 556,
   "id": "7742ad55",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>1KG_EUR_AF</th>\n",
       "      <th>N_SAV_sQTLs</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0.00</td>\n",
       "      <td>49</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.01</td>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.02</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.03</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0.04</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>0.05</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>0.06</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>0.07</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>0.08</td>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>0.09</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>0.10</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>0.11</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>0.12</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>0.13</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>0.14</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>0.15</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>0.16</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>0.17</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>0.18</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>0.19</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>0.20</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>0.21</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>0.22</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>0.23</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>0.24</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>0.25</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>0.26</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>0.27</td>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>0.28</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>0.29</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>0.30</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>0.31</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>0.32</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>0.33</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>0.34</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>0.35</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>0.36</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>0.37</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>0.38</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>0.39</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>0.40</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>0.41</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>0.42</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>0.43</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>0.44</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>0.45</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>0.46</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>0.47</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>0.48</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>0.49</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>0.50</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>0.51</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>0.52</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>0.53</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>0.54</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>0.55</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>0.56</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>0.57</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58</th>\n",
       "      <td>0.58</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>0.59</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>0.60</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>0.61</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>0.62</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>0.63</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>0.64</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>0.65</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66</th>\n",
       "      <td>0.66</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67</th>\n",
       "      <td>0.67</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>68</th>\n",
       "      <td>0.68</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69</th>\n",
       "      <td>0.69</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70</th>\n",
       "      <td>0.70</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71</th>\n",
       "      <td>0.71</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72</th>\n",
       "      <td>0.72</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73</th>\n",
       "      <td>0.73</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74</th>\n",
       "      <td>0.74</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75</th>\n",
       "      <td>0.75</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76</th>\n",
       "      <td>0.76</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>77</th>\n",
       "      <td>0.77</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78</th>\n",
       "      <td>0.78</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79</th>\n",
       "      <td>0.79</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80</th>\n",
       "      <td>0.80</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>81</th>\n",
       "      <td>0.81</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82</th>\n",
       "      <td>0.82</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83</th>\n",
       "      <td>0.83</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84</th>\n",
       "      <td>0.84</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>85</th>\n",
       "      <td>0.85</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86</th>\n",
       "      <td>0.86</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87</th>\n",
       "      <td>0.87</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88</th>\n",
       "      <td>0.88</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89</th>\n",
       "      <td>0.89</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>90</th>\n",
       "      <td>0.90</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91</th>\n",
       "      <td>0.91</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92</th>\n",
       "      <td>0.92</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>93</th>\n",
       "      <td>0.93</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>94</th>\n",
       "      <td>0.94</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95</th>\n",
       "      <td>0.95</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96</th>\n",
       "      <td>0.96</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>0.98</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>98</th>\n",
       "      <td>0.99</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>99</th>\n",
       "      <td>1.00</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    1KG_EUR_AF  N_SAV_sQTLs\n",
       "0         0.00           49\n",
       "1         0.01           22\n",
       "2         0.02           29\n",
       "3         0.03           28\n",
       "4         0.04           29\n",
       "5         0.05           26\n",
       "6         0.06           23\n",
       "7         0.07           19\n",
       "8         0.08           22\n",
       "9         0.09           25\n",
       "10        0.10           19\n",
       "11        0.11           17\n",
       "12        0.12           15\n",
       "13        0.13           20\n",
       "14        0.14           14\n",
       "15        0.15           18\n",
       "16        0.16           16\n",
       "17        0.17           15\n",
       "18        0.18           20\n",
       "19        0.19           19\n",
       "20        0.20           15\n",
       "21        0.21           16\n",
       "22        0.22           18\n",
       "23        0.23           16\n",
       "24        0.24            7\n",
       "25        0.25           11\n",
       "26        0.26           14\n",
       "27        0.27           21\n",
       "28        0.28            8\n",
       "29        0.29           15\n",
       "30        0.30           17\n",
       "31        0.31           16\n",
       "32        0.32           12\n",
       "33        0.33           13\n",
       "34        0.34           16\n",
       "35        0.35            4\n",
       "36        0.36           11\n",
       "37        0.37            9\n",
       "38        0.38           12\n",
       "39        0.39           16\n",
       "40        0.40           15\n",
       "41        0.41            6\n",
       "42        0.42           14\n",
       "43        0.43           12\n",
       "44        0.44           12\n",
       "45        0.45           18\n",
       "46        0.46           14\n",
       "47        0.47           15\n",
       "48        0.48           10\n",
       "49        0.49           10\n",
       "50        0.50           17\n",
       "51        0.51            9\n",
       "52        0.52           13\n",
       "53        0.53            9\n",
       "54        0.54           10\n",
       "55        0.55           10\n",
       "56        0.56           14\n",
       "57        0.57           13\n",
       "58        0.58           12\n",
       "59        0.59            8\n",
       "60        0.60            9\n",
       "61        0.61           13\n",
       "62        0.62           12\n",
       "63        0.63           14\n",
       "64        0.64           10\n",
       "65        0.65           12\n",
       "66        0.66           13\n",
       "67        0.67           10\n",
       "68        0.68            9\n",
       "69        0.69            7\n",
       "70        0.70            8\n",
       "71        0.71            9\n",
       "72        0.72           12\n",
       "73        0.73           13\n",
       "74        0.74            9\n",
       "75        0.75           14\n",
       "76        0.76            5\n",
       "77        0.77            9\n",
       "78        0.78            9\n",
       "79        0.79           12\n",
       "80        0.80           14\n",
       "81        0.81            6\n",
       "82        0.82            7\n",
       "83        0.83            7\n",
       "84        0.84            8\n",
       "85        0.85            7\n",
       "86        0.86            5\n",
       "87        0.87            8\n",
       "88        0.88            5\n",
       "89        0.89            8\n",
       "90        0.90            8\n",
       "91        0.91           10\n",
       "92        0.92            8\n",
       "93        0.93            5\n",
       "94        0.94            6\n",
       "95        0.95            4\n",
       "96        0.96            6\n",
       "97        0.98            3\n",
       "98        0.99            2\n",
       "99        1.00           10"
      ]
     },
     "execution_count": 556,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "EUR_SAV_sQTLs = SAV_sQTLs.groupby('1KG_EUR_AF').size().to_frame('N_SAV_sQTLs')\n",
    "EUR_SAV_sQTLs = EUR_SAV_sQTLs.reset_index()\n",
    "EUR_SAV_sQTLs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 557,
   "id": "519cd39c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>binned_AF</th>\n",
       "      <th>N_SAVs</th>\n",
       "      <th>N_SAV_sQTLs</th>\n",
       "      <th>proportion_sQTLs</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>(0.0, 0.05]</td>\n",
       "      <td>447</td>\n",
       "      <td>134</td>\n",
       "      <td>0.299776</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>(0.05, 0.1]</td>\n",
       "      <td>208</td>\n",
       "      <td>108</td>\n",
       "      <td>0.519231</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>(0.1, 0.15]</td>\n",
       "      <td>148</td>\n",
       "      <td>84</td>\n",
       "      <td>0.567568</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>(0.15, 0.2]</td>\n",
       "      <td>137</td>\n",
       "      <td>85</td>\n",
       "      <td>0.620438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>(0.2, 0.25]</td>\n",
       "      <td>110</td>\n",
       "      <td>68</td>\n",
       "      <td>0.618182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>(0.25, 0.3]</td>\n",
       "      <td>114</td>\n",
       "      <td>75</td>\n",
       "      <td>0.657895</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>(0.3, 0.35]</td>\n",
       "      <td>89</td>\n",
       "      <td>61</td>\n",
       "      <td>0.685393</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>(0.35, 0.4]</td>\n",
       "      <td>108</td>\n",
       "      <td>63</td>\n",
       "      <td>0.583333</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>(0.4, 0.45]</td>\n",
       "      <td>97</td>\n",
       "      <td>62</td>\n",
       "      <td>0.639175</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>(0.45, 0.5]</td>\n",
       "      <td>99</td>\n",
       "      <td>66</td>\n",
       "      <td>0.666667</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>(0.5, 0.55]</td>\n",
       "      <td>77</td>\n",
       "      <td>51</td>\n",
       "      <td>0.662338</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>(0.55, 0.6]</td>\n",
       "      <td>84</td>\n",
       "      <td>56</td>\n",
       "      <td>0.666667</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>(0.6, 0.65]</td>\n",
       "      <td>91</td>\n",
       "      <td>61</td>\n",
       "      <td>0.670330</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>(0.65, 0.7]</td>\n",
       "      <td>76</td>\n",
       "      <td>47</td>\n",
       "      <td>0.618421</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>(0.7, 0.75]</td>\n",
       "      <td>101</td>\n",
       "      <td>57</td>\n",
       "      <td>0.564356</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>(0.75, 0.8]</td>\n",
       "      <td>79</td>\n",
       "      <td>49</td>\n",
       "      <td>0.620253</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>(0.8, 0.85]</td>\n",
       "      <td>77</td>\n",
       "      <td>35</td>\n",
       "      <td>0.454545</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>(0.85, 0.9]</td>\n",
       "      <td>74</td>\n",
       "      <td>34</td>\n",
       "      <td>0.459459</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>(0.9, 0.95]</td>\n",
       "      <td>83</td>\n",
       "      <td>33</td>\n",
       "      <td>0.397590</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>(0.95, 1.0]</td>\n",
       "      <td>230</td>\n",
       "      <td>21</td>\n",
       "      <td>0.091304</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      binned_AF  N_SAVs  N_SAV_sQTLs  proportion_sQTLs\n",
       "0   (0.0, 0.05]     447          134          0.299776\n",
       "1   (0.05, 0.1]     208          108          0.519231\n",
       "2   (0.1, 0.15]     148           84          0.567568\n",
       "3   (0.15, 0.2]     137           85          0.620438\n",
       "4   (0.2, 0.25]     110           68          0.618182\n",
       "5   (0.25, 0.3]     114           75          0.657895\n",
       "6   (0.3, 0.35]      89           61          0.685393\n",
       "7   (0.35, 0.4]     108           63          0.583333\n",
       "8   (0.4, 0.45]      97           62          0.639175\n",
       "9   (0.45, 0.5]      99           66          0.666667\n",
       "10  (0.5, 0.55]      77           51          0.662338\n",
       "11  (0.55, 0.6]      84           56          0.666667\n",
       "12  (0.6, 0.65]      91           61          0.670330\n",
       "13  (0.65, 0.7]      76           47          0.618421\n",
       "14  (0.7, 0.75]     101           57          0.564356\n",
       "15  (0.75, 0.8]      79           49          0.620253\n",
       "16  (0.8, 0.85]      77           35          0.454545\n",
       "17  (0.85, 0.9]      74           34          0.459459\n",
       "18  (0.9, 0.95]      83           33          0.397590\n",
       "19  (0.95, 1.0]     230           21          0.091304"
      ]
     },
     "execution_count": 557,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "EUR_sQTL_proportion = pd.merge(EUR_SAVs, EUR_SAV_sQTLs, on='1KG_EUR_AF')\n",
    "bins = [0,0.05,0.1,0.15,0.2,0.25,0.3,0.35,0.4,0.45,0.5,0.55,0.6,0.65,0.7,0.75,0.8,0.85,0.9,0.95,1]\n",
    "EUR_sQTL_proportion['binned_AF'] = pd.cut(EUR_sQTL_proportion['1KG_EUR_AF'], bins)\n",
    "EUR_sQTL_proportion = EUR_sQTL_proportion.groupby('binned_AF').sum().reset_index()\n",
    "EUR_sQTL_proportion['proportion_sQTLs'] = (EUR_sQTL_proportion['N_SAV_sQTLs'])/(EUR_sQTL_proportion['N_SAVs'])\n",
    "EUR_sQTL_proportion['binned_AF'] = EUR_sQTL_proportion['binned_AF'].astype(str)\n",
    "EUR_sQTL_proportion = EUR_sQTL_proportion[['binned_AF','N_SAVs','N_SAV_sQTLs','proportion_sQTLs']]\n",
    "EUR_sQTL_proportion"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 558,
   "id": "18e43f93",
   "metadata": {},
   "outputs": [],
   "source": [
    "EUR_sQTL_proportion.to_csv('../GTEx_sQTLs/EUR_sQTLs.txt', sep=\"\\t\", header=True, index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f7a3a9bf",
   "metadata": {},
   "source": [
    "## Gene-Level <a class = 'anchor' id = 'genelevel'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8ea15b0b",
   "metadata": {},
   "source": [
    "In which tissues are archaic splice variants that occur in modern humans expressed? We have data kindly provided by Mary Lauren Benton on expression specifity metrics. That dataframe contains Ensembl IDs so first we need to build a dictionary to get the GENCODE annotation."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 559,
   "id": "6aa8b18d",
   "metadata": {},
   "outputs": [],
   "source": [
    "genes = []\n",
    "ids = []\n",
    "\n",
    "with open(\"../annotations/gencode.v24lift37.basic.annotation.gff3\") as f: \n",
    "    lines = f.readlines()\n",
    "for line in lines:\n",
    "\tinfo=line.split(';')\n",
    "\tfor x in info:\n",
    "\t\tif x.startswith(\"gene_id=\"):\n",
    "\t\t\tb=re.split('=|\\.',x) # use re here because there are multiple delimiters (';' and '.')\n",
    "\t\t\tids.append(b[1])\n",
    "\t\telif x.startswith(\"gene_name=\"):\n",
    "\t\t\tc=x.split(\"=\")\n",
    "\t\t\tgenes.append(c[1])\n",
    "            \n",
    "ensembl_id_to_gene_id = dict(zip(ids, genes))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3ff2731c",
   "metadata": {},
   "source": [
    "Load in Mary Lauren's data."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 560,
   "id": "8a647b26",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>adipose tissue</th>\n",
       "      <th>adrenal gland</th>\n",
       "      <th>amygdala</th>\n",
       "      <th>basal ganglia</th>\n",
       "      <th>breast</th>\n",
       "      <th>cerebellum</th>\n",
       "      <th>cerebral cortex</th>\n",
       "      <th>cervix, uterine</th>\n",
       "      <th>colon</th>\n",
       "      <th>endometrium</th>\n",
       "      <th>esophagus</th>\n",
       "      <th>fallopian tube</th>\n",
       "      <th>heart muscle</th>\n",
       "      <th>hippocampal formation</th>\n",
       "      <th>hypothalamus</th>\n",
       "      <th>kidney</th>\n",
       "      <th>liver</th>\n",
       "      <th>lung</th>\n",
       "      <th>midbrain</th>\n",
       "      <th>ovary</th>\n",
       "      <th>pancreas</th>\n",
       "      <th>pituitary gland</th>\n",
       "      <th>prostate</th>\n",
       "      <th>salivary gland</th>\n",
       "      <th>skeletal muscle</th>\n",
       "      <th>skin</th>\n",
       "      <th>small intestine</th>\n",
       "      <th>spinal cord</th>\n",
       "      <th>spleen</th>\n",
       "      <th>stomach</th>\n",
       "      <th>testis</th>\n",
       "      <th>thyroid gland</th>\n",
       "      <th>urinary bladder</th>\n",
       "      <th>vagina</th>\n",
       "      <th>entropy</th>\n",
       "      <th>rel_entropy</th>\n",
       "      <th>tau</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000000003</td>\n",
       "      <td>27.4</td>\n",
       "      <td>15.5</td>\n",
       "      <td>7.3</td>\n",
       "      <td>7.7</td>\n",
       "      <td>32.0</td>\n",
       "      <td>2.5</td>\n",
       "      <td>5.6</td>\n",
       "      <td>30.6</td>\n",
       "      <td>32.6</td>\n",
       "      <td>22.8</td>\n",
       "      <td>35.5</td>\n",
       "      <td>27.7</td>\n",
       "      <td>4.5</td>\n",
       "      <td>7.0</td>\n",
       "      <td>9.9</td>\n",
       "      <td>16.2</td>\n",
       "      <td>23.8</td>\n",
       "      <td>12.0</td>\n",
       "      <td>6.5</td>\n",
       "      <td>50.3</td>\n",
       "      <td>7.6</td>\n",
       "      <td>36.5</td>\n",
       "      <td>17.5</td>\n",
       "      <td>30.7</td>\n",
       "      <td>2.0</td>\n",
       "      <td>8.8</td>\n",
       "      <td>16.2</td>\n",
       "      <td>6.4</td>\n",
       "      <td>7.8</td>\n",
       "      <td>8.8</td>\n",
       "      <td>49.5</td>\n",
       "      <td>18.7</td>\n",
       "      <td>37.6</td>\n",
       "      <td>23.8</td>\n",
       "      <td>0.068330</td>\n",
       "      <td>0.037445</td>\n",
       "      <td>0.639135</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000000005</td>\n",
       "      <td>20.7</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>11.9</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.1</td>\n",
       "      <td>0.2</td>\n",
       "      <td>0.6</td>\n",
       "      <td>0.1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.3</td>\n",
       "      <td>0.1</td>\n",
       "      <td>0.1</td>\n",
       "      <td>0.1</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.4</td>\n",
       "      <td>1.3</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.4</td>\n",
       "      <td>0.4</td>\n",
       "      <td>1.3</td>\n",
       "      <td>3.1</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.1</td>\n",
       "      <td>0.2</td>\n",
       "      <td>0.1</td>\n",
       "      <td>0.2</td>\n",
       "      <td>0.534839</td>\n",
       "      <td>0.404403</td>\n",
       "      <td>0.974235</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000000419</td>\n",
       "      <td>33.6</td>\n",
       "      <td>37.7</td>\n",
       "      <td>12.4</td>\n",
       "      <td>15.6</td>\n",
       "      <td>34.0</td>\n",
       "      <td>24.9</td>\n",
       "      <td>22.4</td>\n",
       "      <td>35.8</td>\n",
       "      <td>31.5</td>\n",
       "      <td>39.2</td>\n",
       "      <td>33.8</td>\n",
       "      <td>36.5</td>\n",
       "      <td>21.6</td>\n",
       "      <td>14.6</td>\n",
       "      <td>17.4</td>\n",
       "      <td>20.5</td>\n",
       "      <td>17.3</td>\n",
       "      <td>31.5</td>\n",
       "      <td>14.1</td>\n",
       "      <td>33.2</td>\n",
       "      <td>18.2</td>\n",
       "      <td>28.4</td>\n",
       "      <td>29.9</td>\n",
       "      <td>28.0</td>\n",
       "      <td>26.2</td>\n",
       "      <td>30.5</td>\n",
       "      <td>28.6</td>\n",
       "      <td>16.7</td>\n",
       "      <td>34.0</td>\n",
       "      <td>24.7</td>\n",
       "      <td>42.7</td>\n",
       "      <td>35.2</td>\n",
       "      <td>32.7</td>\n",
       "      <td>34.7</td>\n",
       "      <td>0.013113</td>\n",
       "      <td>0.006362</td>\n",
       "      <td>0.364559</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000000457</td>\n",
       "      <td>5.4</td>\n",
       "      <td>4.5</td>\n",
       "      <td>1.5</td>\n",
       "      <td>1.7</td>\n",
       "      <td>6.7</td>\n",
       "      <td>5.2</td>\n",
       "      <td>2.3</td>\n",
       "      <td>7.9</td>\n",
       "      <td>5.4</td>\n",
       "      <td>6.5</td>\n",
       "      <td>5.7</td>\n",
       "      <td>6.3</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.7</td>\n",
       "      <td>2.1</td>\n",
       "      <td>3.4</td>\n",
       "      <td>3.8</td>\n",
       "      <td>5.1</td>\n",
       "      <td>1.7</td>\n",
       "      <td>5.9</td>\n",
       "      <td>2.5</td>\n",
       "      <td>4.5</td>\n",
       "      <td>6.8</td>\n",
       "      <td>6.7</td>\n",
       "      <td>3.6</td>\n",
       "      <td>6.7</td>\n",
       "      <td>5.4</td>\n",
       "      <td>2.5</td>\n",
       "      <td>5.7</td>\n",
       "      <td>3.8</td>\n",
       "      <td>6.3</td>\n",
       "      <td>7.4</td>\n",
       "      <td>9.1</td>\n",
       "      <td>6.6</td>\n",
       "      <td>0.028105</td>\n",
       "      <td>0.010124</td>\n",
       "      <td>0.489510</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000000460</td>\n",
       "      <td>0.7</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0.2</td>\n",
       "      <td>0.2</td>\n",
       "      <td>0.7</td>\n",
       "      <td>1.5</td>\n",
       "      <td>0.2</td>\n",
       "      <td>0.9</td>\n",
       "      <td>0.6</td>\n",
       "      <td>0.7</td>\n",
       "      <td>0.7</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0.2</td>\n",
       "      <td>0.2</td>\n",
       "      <td>0.3</td>\n",
       "      <td>0.3</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0.3</td>\n",
       "      <td>0.6</td>\n",
       "      <td>0.2</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0.7</td>\n",
       "      <td>0.6</td>\n",
       "      <td>0.1</td>\n",
       "      <td>0.6</td>\n",
       "      <td>0.6</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0.9</td>\n",
       "      <td>0.4</td>\n",
       "      <td>4.9</td>\n",
       "      <td>0.8</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.7</td>\n",
       "      <td>0.104682</td>\n",
       "      <td>0.051871</td>\n",
       "      <td>0.984539</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>ENSG00000000938</td>\n",
       "      <td>19.9</td>\n",
       "      <td>4.8</td>\n",
       "      <td>2.7</td>\n",
       "      <td>3.2</td>\n",
       "      <td>13.7</td>\n",
       "      <td>2.4</td>\n",
       "      <td>3.8</td>\n",
       "      <td>9.1</td>\n",
       "      <td>5.0</td>\n",
       "      <td>8.6</td>\n",
       "      <td>3.8</td>\n",
       "      <td>18.7</td>\n",
       "      <td>3.7</td>\n",
       "      <td>3.2</td>\n",
       "      <td>3.7</td>\n",
       "      <td>4.4</td>\n",
       "      <td>4.2</td>\n",
       "      <td>86.1</td>\n",
       "      <td>4.9</td>\n",
       "      <td>2.4</td>\n",
       "      <td>1.7</td>\n",
       "      <td>2.7</td>\n",
       "      <td>6.2</td>\n",
       "      <td>5.8</td>\n",
       "      <td>1.3</td>\n",
       "      <td>3.7</td>\n",
       "      <td>8.2</td>\n",
       "      <td>6.5</td>\n",
       "      <td>124.3</td>\n",
       "      <td>4.1</td>\n",
       "      <td>3.2</td>\n",
       "      <td>7.6</td>\n",
       "      <td>7.1</td>\n",
       "      <td>6.7</td>\n",
       "      <td>0.268393</td>\n",
       "      <td>0.194874</td>\n",
       "      <td>0.933421</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>ENSG00000000971</td>\n",
       "      <td>122.4</td>\n",
       "      <td>27.3</td>\n",
       "      <td>5.0</td>\n",
       "      <td>7.5</td>\n",
       "      <td>54.3</td>\n",
       "      <td>3.5</td>\n",
       "      <td>5.5</td>\n",
       "      <td>30.9</td>\n",
       "      <td>26.3</td>\n",
       "      <td>12.5</td>\n",
       "      <td>30.7</td>\n",
       "      <td>27.5</td>\n",
       "      <td>163.3</td>\n",
       "      <td>5.8</td>\n",
       "      <td>6.0</td>\n",
       "      <td>19.1</td>\n",
       "      <td>286.4</td>\n",
       "      <td>73.2</td>\n",
       "      <td>10.1</td>\n",
       "      <td>166.9</td>\n",
       "      <td>4.3</td>\n",
       "      <td>17.5</td>\n",
       "      <td>24.2</td>\n",
       "      <td>107.1</td>\n",
       "      <td>9.9</td>\n",
       "      <td>71.9</td>\n",
       "      <td>41.9</td>\n",
       "      <td>11.3</td>\n",
       "      <td>9.0</td>\n",
       "      <td>22.1</td>\n",
       "      <td>11.5</td>\n",
       "      <td>117.8</td>\n",
       "      <td>97.4</td>\n",
       "      <td>26.6</td>\n",
       "      <td>0.170275</td>\n",
       "      <td>0.163749</td>\n",
       "      <td>0.855013</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>ENSG00000001036</td>\n",
       "      <td>27.4</td>\n",
       "      <td>35.1</td>\n",
       "      <td>7.1</td>\n",
       "      <td>7.8</td>\n",
       "      <td>24.8</td>\n",
       "      <td>5.9</td>\n",
       "      <td>6.3</td>\n",
       "      <td>32.5</td>\n",
       "      <td>23.3</td>\n",
       "      <td>26.6</td>\n",
       "      <td>11.1</td>\n",
       "      <td>27.7</td>\n",
       "      <td>14.0</td>\n",
       "      <td>7.9</td>\n",
       "      <td>10.7</td>\n",
       "      <td>20.7</td>\n",
       "      <td>15.2</td>\n",
       "      <td>27.3</td>\n",
       "      <td>10.5</td>\n",
       "      <td>26.8</td>\n",
       "      <td>12.7</td>\n",
       "      <td>16.8</td>\n",
       "      <td>16.3</td>\n",
       "      <td>21.7</td>\n",
       "      <td>3.0</td>\n",
       "      <td>11.5</td>\n",
       "      <td>23.9</td>\n",
       "      <td>17.6</td>\n",
       "      <td>22.4</td>\n",
       "      <td>18.8</td>\n",
       "      <td>11.1</td>\n",
       "      <td>26.6</td>\n",
       "      <td>24.9</td>\n",
       "      <td>20.7</td>\n",
       "      <td>0.031989</td>\n",
       "      <td>0.019567</td>\n",
       "      <td>0.497885</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>ENSG00000001084</td>\n",
       "      <td>11.0</td>\n",
       "      <td>9.9</td>\n",
       "      <td>7.6</td>\n",
       "      <td>9.8</td>\n",
       "      <td>9.1</td>\n",
       "      <td>5.7</td>\n",
       "      <td>8.6</td>\n",
       "      <td>9.6</td>\n",
       "      <td>13.9</td>\n",
       "      <td>8.4</td>\n",
       "      <td>18.0</td>\n",
       "      <td>10.5</td>\n",
       "      <td>3.1</td>\n",
       "      <td>8.2</td>\n",
       "      <td>8.1</td>\n",
       "      <td>6.1</td>\n",
       "      <td>12.6</td>\n",
       "      <td>7.0</td>\n",
       "      <td>9.6</td>\n",
       "      <td>6.0</td>\n",
       "      <td>3.7</td>\n",
       "      <td>5.5</td>\n",
       "      <td>12.1</td>\n",
       "      <td>10.7</td>\n",
       "      <td>3.7</td>\n",
       "      <td>8.6</td>\n",
       "      <td>15.6</td>\n",
       "      <td>10.9</td>\n",
       "      <td>15.8</td>\n",
       "      <td>11.0</td>\n",
       "      <td>5.2</td>\n",
       "      <td>12.4</td>\n",
       "      <td>44.5</td>\n",
       "      <td>14.3</td>\n",
       "      <td>0.043669</td>\n",
       "      <td>0.033442</td>\n",
       "      <td>0.787334</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>ENSG00000001167</td>\n",
       "      <td>9.9</td>\n",
       "      <td>7.4</td>\n",
       "      <td>6.0</td>\n",
       "      <td>5.9</td>\n",
       "      <td>10.9</td>\n",
       "      <td>11.2</td>\n",
       "      <td>6.3</td>\n",
       "      <td>12.4</td>\n",
       "      <td>7.2</td>\n",
       "      <td>10.9</td>\n",
       "      <td>8.7</td>\n",
       "      <td>11.3</td>\n",
       "      <td>4.7</td>\n",
       "      <td>6.4</td>\n",
       "      <td>6.0</td>\n",
       "      <td>5.7</td>\n",
       "      <td>4.5</td>\n",
       "      <td>12.7</td>\n",
       "      <td>6.8</td>\n",
       "      <td>14.1</td>\n",
       "      <td>4.3</td>\n",
       "      <td>9.9</td>\n",
       "      <td>9.6</td>\n",
       "      <td>8.3</td>\n",
       "      <td>4.5</td>\n",
       "      <td>8.6</td>\n",
       "      <td>8.9</td>\n",
       "      <td>10.6</td>\n",
       "      <td>10.4</td>\n",
       "      <td>6.1</td>\n",
       "      <td>42.3</td>\n",
       "      <td>17.6</td>\n",
       "      <td>11.2</td>\n",
       "      <td>9.9</td>\n",
       "      <td>0.041457</td>\n",
       "      <td>0.015504</td>\n",
       "      <td>0.793037</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              Gene  adipose tissue  adrenal gland  amygdala  basal ganglia  \\\n",
       "0  ENSG00000000003            27.4           15.5       7.3            7.7   \n",
       "1  ENSG00000000005            20.7            0.0       0.1            0.0   \n",
       "2  ENSG00000000419            33.6           37.7      12.4           15.6   \n",
       "3  ENSG00000000457             5.4            4.5       1.5            1.7   \n",
       "4  ENSG00000000460             0.7            0.5       0.2            0.2   \n",
       "5  ENSG00000000938            19.9            4.8       2.7            3.2   \n",
       "6  ENSG00000000971           122.4           27.3       5.0            7.5   \n",
       "7  ENSG00000001036            27.4           35.1       7.1            7.8   \n",
       "8  ENSG00000001084            11.0            9.9       7.6            9.8   \n",
       "9  ENSG00000001167             9.9            7.4       6.0            5.9   \n",
       "\n",
       "   breast  cerebellum  cerebral cortex  cervix, uterine  colon  endometrium  \\\n",
       "0    32.0         2.5              5.6             30.6   32.6         22.8   \n",
       "1    11.9         0.0              0.1              0.2    0.6          0.1   \n",
       "2    34.0        24.9             22.4             35.8   31.5         39.2   \n",
       "3     6.7         5.2              2.3              7.9    5.4          6.5   \n",
       "4     0.7         1.5              0.2              0.9    0.6          0.7   \n",
       "5    13.7         2.4              3.8              9.1    5.0          8.6   \n",
       "6    54.3         3.5              5.5             30.9   26.3         12.5   \n",
       "7    24.8         5.9              6.3             32.5   23.3         26.6   \n",
       "8     9.1         5.7              8.6              9.6   13.9          8.4   \n",
       "9    10.9        11.2              6.3             12.4    7.2         10.9   \n",
       "\n",
       "   esophagus  fallopian tube  heart muscle  hippocampal formation  \\\n",
       "0       35.5            27.7           4.5                    7.0   \n",
       "1        0.0             0.3           0.1                    0.1   \n",
       "2       33.8            36.5          21.6                   14.6   \n",
       "3        5.7             6.3           2.0                    1.7   \n",
       "4        0.7             0.5           0.2                    0.2   \n",
       "5        3.8            18.7           3.7                    3.2   \n",
       "6       30.7            27.5         163.3                    5.8   \n",
       "7       11.1            27.7          14.0                    7.9   \n",
       "8       18.0            10.5           3.1                    8.2   \n",
       "9        8.7            11.3           4.7                    6.4   \n",
       "\n",
       "   hypothalamus  kidney  liver  lung  midbrain  ovary  pancreas  \\\n",
       "0           9.9    16.2   23.8  12.0       6.5   50.3       7.6   \n",
       "1           0.1     0.5    0.0   0.0       0.4    1.3       0.0   \n",
       "2          17.4    20.5   17.3  31.5      14.1   33.2      18.2   \n",
       "3           2.1     3.4    3.8   5.1       1.7    5.9       2.5   \n",
       "4           0.3     0.3    0.5   0.5       0.3    0.6       0.2   \n",
       "5           3.7     4.4    4.2  86.1       4.9    2.4       1.7   \n",
       "6           6.0    19.1  286.4  73.2      10.1  166.9       4.3   \n",
       "7          10.7    20.7   15.2  27.3      10.5   26.8      12.7   \n",
       "8           8.1     6.1   12.6   7.0       9.6    6.0       3.7   \n",
       "9           6.0     5.7    4.5  12.7       6.8   14.1       4.3   \n",
       "\n",
       "   pituitary gland  prostate  salivary gland  skeletal muscle  skin  \\\n",
       "0             36.5      17.5            30.7              2.0   8.8   \n",
       "1              0.0       0.4             0.4              1.3   3.1   \n",
       "2             28.4      29.9            28.0             26.2  30.5   \n",
       "3              4.5       6.8             6.7              3.6   6.7   \n",
       "4              0.5       0.7             0.6              0.1   0.6   \n",
       "5              2.7       6.2             5.8              1.3   3.7   \n",
       "6             17.5      24.2           107.1              9.9  71.9   \n",
       "7             16.8      16.3            21.7              3.0  11.5   \n",
       "8              5.5      12.1            10.7              3.7   8.6   \n",
       "9              9.9       9.6             8.3              4.5   8.6   \n",
       "\n",
       "   small intestine  spinal cord  spleen  stomach  testis  thyroid gland  \\\n",
       "0             16.2          6.4     7.8      8.8    49.5           18.7   \n",
       "1              0.5          0.0     0.1      0.0     0.1            0.2   \n",
       "2             28.6         16.7    34.0     24.7    42.7           35.2   \n",
       "3              5.4          2.5     5.7      3.8     6.3            7.4   \n",
       "4              0.6          0.5     0.9      0.4     4.9            0.8   \n",
       "5              8.2          6.5   124.3      4.1     3.2            7.6   \n",
       "6             41.9         11.3     9.0     22.1    11.5          117.8   \n",
       "7             23.9         17.6    22.4     18.8    11.1           26.6   \n",
       "8             15.6         10.9    15.8     11.0     5.2           12.4   \n",
       "9              8.9         10.6    10.4      6.1    42.3           17.6   \n",
       "\n",
       "   urinary bladder  vagina   entropy  rel_entropy       tau  \n",
       "0             37.6    23.8  0.068330     0.037445  0.639135  \n",
       "1              0.1     0.2  0.534839     0.404403  0.974235  \n",
       "2             32.7    34.7  0.013113     0.006362  0.364559  \n",
       "3              9.1     6.6  0.028105     0.010124  0.489510  \n",
       "4              1.0     0.7  0.104682     0.051871  0.984539  \n",
       "5              7.1     6.7  0.268393     0.194874  0.933421  \n",
       "6             97.4    26.6  0.170275     0.163749  0.855013  \n",
       "7             24.9    20.7  0.031989     0.019567  0.497885  \n",
       "8             44.5    14.3  0.043669     0.033442  0.787334  \n",
       "9             11.2     9.9  0.041457     0.015504  0.793037  "
      ]
     },
     "execution_count": 560,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "expression = pd.read_csv(\"../gene_expression/gene-specificity_GTEx_TPM_alltissues.tsv\", sep='\\t', header=0)\n",
    "expression.head(10)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "21f84337",
   "metadata": {},
   "source": [
    "Now map our dictionary."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 561,
   "id": "45ff4803",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>adipose tissue</th>\n",
       "      <th>adrenal gland</th>\n",
       "      <th>amygdala</th>\n",
       "      <th>basal ganglia</th>\n",
       "      <th>breast</th>\n",
       "      <th>cerebellum</th>\n",
       "      <th>cerebral cortex</th>\n",
       "      <th>cervix, uterine</th>\n",
       "      <th>colon</th>\n",
       "      <th>endometrium</th>\n",
       "      <th>esophagus</th>\n",
       "      <th>fallopian tube</th>\n",
       "      <th>heart muscle</th>\n",
       "      <th>hippocampal formation</th>\n",
       "      <th>hypothalamus</th>\n",
       "      <th>kidney</th>\n",
       "      <th>liver</th>\n",
       "      <th>lung</th>\n",
       "      <th>midbrain</th>\n",
       "      <th>ovary</th>\n",
       "      <th>pancreas</th>\n",
       "      <th>pituitary gland</th>\n",
       "      <th>prostate</th>\n",
       "      <th>salivary gland</th>\n",
       "      <th>skeletal muscle</th>\n",
       "      <th>skin</th>\n",
       "      <th>small intestine</th>\n",
       "      <th>spinal cord</th>\n",
       "      <th>spleen</th>\n",
       "      <th>stomach</th>\n",
       "      <th>testis</th>\n",
       "      <th>thyroid gland</th>\n",
       "      <th>urinary bladder</th>\n",
       "      <th>vagina</th>\n",
       "      <th>entropy</th>\n",
       "      <th>rel_entropy</th>\n",
       "      <th>tau</th>\n",
       "      <th>annotation</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000000003</td>\n",
       "      <td>27.4</td>\n",
       "      <td>15.5</td>\n",
       "      <td>7.3</td>\n",
       "      <td>7.7</td>\n",
       "      <td>32.0</td>\n",
       "      <td>2.5</td>\n",
       "      <td>5.6</td>\n",
       "      <td>30.6</td>\n",
       "      <td>32.6</td>\n",
       "      <td>22.8</td>\n",
       "      <td>35.5</td>\n",
       "      <td>27.7</td>\n",
       "      <td>4.5</td>\n",
       "      <td>7.0</td>\n",
       "      <td>9.9</td>\n",
       "      <td>16.2</td>\n",
       "      <td>23.8</td>\n",
       "      <td>12.0</td>\n",
       "      <td>6.5</td>\n",
       "      <td>50.3</td>\n",
       "      <td>7.6</td>\n",
       "      <td>36.5</td>\n",
       "      <td>17.5</td>\n",
       "      <td>30.7</td>\n",
       "      <td>2.0</td>\n",
       "      <td>8.8</td>\n",
       "      <td>16.2</td>\n",
       "      <td>6.4</td>\n",
       "      <td>7.8</td>\n",
       "      <td>8.8</td>\n",
       "      <td>49.5</td>\n",
       "      <td>18.7</td>\n",
       "      <td>37.6</td>\n",
       "      <td>23.8</td>\n",
       "      <td>0.068330</td>\n",
       "      <td>0.037445</td>\n",
       "      <td>0.639135</td>\n",
       "      <td>TSPAN6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000000005</td>\n",
       "      <td>20.7</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>11.9</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.1</td>\n",
       "      <td>0.2</td>\n",
       "      <td>0.6</td>\n",
       "      <td>0.1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.3</td>\n",
       "      <td>0.1</td>\n",
       "      <td>0.1</td>\n",
       "      <td>0.1</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.4</td>\n",
       "      <td>1.3</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.4</td>\n",
       "      <td>0.4</td>\n",
       "      <td>1.3</td>\n",
       "      <td>3.1</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.1</td>\n",
       "      <td>0.2</td>\n",
       "      <td>0.1</td>\n",
       "      <td>0.2</td>\n",
       "      <td>0.534839</td>\n",
       "      <td>0.404403</td>\n",
       "      <td>0.974235</td>\n",
       "      <td>TNMD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000000419</td>\n",
       "      <td>33.6</td>\n",
       "      <td>37.7</td>\n",
       "      <td>12.4</td>\n",
       "      <td>15.6</td>\n",
       "      <td>34.0</td>\n",
       "      <td>24.9</td>\n",
       "      <td>22.4</td>\n",
       "      <td>35.8</td>\n",
       "      <td>31.5</td>\n",
       "      <td>39.2</td>\n",
       "      <td>33.8</td>\n",
       "      <td>36.5</td>\n",
       "      <td>21.6</td>\n",
       "      <td>14.6</td>\n",
       "      <td>17.4</td>\n",
       "      <td>20.5</td>\n",
       "      <td>17.3</td>\n",
       "      <td>31.5</td>\n",
       "      <td>14.1</td>\n",
       "      <td>33.2</td>\n",
       "      <td>18.2</td>\n",
       "      <td>28.4</td>\n",
       "      <td>29.9</td>\n",
       "      <td>28.0</td>\n",
       "      <td>26.2</td>\n",
       "      <td>30.5</td>\n",
       "      <td>28.6</td>\n",
       "      <td>16.7</td>\n",
       "      <td>34.0</td>\n",
       "      <td>24.7</td>\n",
       "      <td>42.7</td>\n",
       "      <td>35.2</td>\n",
       "      <td>32.7</td>\n",
       "      <td>34.7</td>\n",
       "      <td>0.013113</td>\n",
       "      <td>0.006362</td>\n",
       "      <td>0.364559</td>\n",
       "      <td>DPM1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000000457</td>\n",
       "      <td>5.4</td>\n",
       "      <td>4.5</td>\n",
       "      <td>1.5</td>\n",
       "      <td>1.7</td>\n",
       "      <td>6.7</td>\n",
       "      <td>5.2</td>\n",
       "      <td>2.3</td>\n",
       "      <td>7.9</td>\n",
       "      <td>5.4</td>\n",
       "      <td>6.5</td>\n",
       "      <td>5.7</td>\n",
       "      <td>6.3</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.7</td>\n",
       "      <td>2.1</td>\n",
       "      <td>3.4</td>\n",
       "      <td>3.8</td>\n",
       "      <td>5.1</td>\n",
       "      <td>1.7</td>\n",
       "      <td>5.9</td>\n",
       "      <td>2.5</td>\n",
       "      <td>4.5</td>\n",
       "      <td>6.8</td>\n",
       "      <td>6.7</td>\n",
       "      <td>3.6</td>\n",
       "      <td>6.7</td>\n",
       "      <td>5.4</td>\n",
       "      <td>2.5</td>\n",
       "      <td>5.7</td>\n",
       "      <td>3.8</td>\n",
       "      <td>6.3</td>\n",
       "      <td>7.4</td>\n",
       "      <td>9.1</td>\n",
       "      <td>6.6</td>\n",
       "      <td>0.028105</td>\n",
       "      <td>0.010124</td>\n",
       "      <td>0.489510</td>\n",
       "      <td>SCYL3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000000460</td>\n",
       "      <td>0.7</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0.2</td>\n",
       "      <td>0.2</td>\n",
       "      <td>0.7</td>\n",
       "      <td>1.5</td>\n",
       "      <td>0.2</td>\n",
       "      <td>0.9</td>\n",
       "      <td>0.6</td>\n",
       "      <td>0.7</td>\n",
       "      <td>0.7</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0.2</td>\n",
       "      <td>0.2</td>\n",
       "      <td>0.3</td>\n",
       "      <td>0.3</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0.3</td>\n",
       "      <td>0.6</td>\n",
       "      <td>0.2</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0.7</td>\n",
       "      <td>0.6</td>\n",
       "      <td>0.1</td>\n",
       "      <td>0.6</td>\n",
       "      <td>0.6</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0.9</td>\n",
       "      <td>0.4</td>\n",
       "      <td>4.9</td>\n",
       "      <td>0.8</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.7</td>\n",
       "      <td>0.104682</td>\n",
       "      <td>0.051871</td>\n",
       "      <td>0.984539</td>\n",
       "      <td>C1orf112</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>ENSG00000000938</td>\n",
       "      <td>19.9</td>\n",
       "      <td>4.8</td>\n",
       "      <td>2.7</td>\n",
       "      <td>3.2</td>\n",
       "      <td>13.7</td>\n",
       "      <td>2.4</td>\n",
       "      <td>3.8</td>\n",
       "      <td>9.1</td>\n",
       "      <td>5.0</td>\n",
       "      <td>8.6</td>\n",
       "      <td>3.8</td>\n",
       "      <td>18.7</td>\n",
       "      <td>3.7</td>\n",
       "      <td>3.2</td>\n",
       "      <td>3.7</td>\n",
       "      <td>4.4</td>\n",
       "      <td>4.2</td>\n",
       "      <td>86.1</td>\n",
       "      <td>4.9</td>\n",
       "      <td>2.4</td>\n",
       "      <td>1.7</td>\n",
       "      <td>2.7</td>\n",
       "      <td>6.2</td>\n",
       "      <td>5.8</td>\n",
       "      <td>1.3</td>\n",
       "      <td>3.7</td>\n",
       "      <td>8.2</td>\n",
       "      <td>6.5</td>\n",
       "      <td>124.3</td>\n",
       "      <td>4.1</td>\n",
       "      <td>3.2</td>\n",
       "      <td>7.6</td>\n",
       "      <td>7.1</td>\n",
       "      <td>6.7</td>\n",
       "      <td>0.268393</td>\n",
       "      <td>0.194874</td>\n",
       "      <td>0.933421</td>\n",
       "      <td>FGR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>ENSG00000000971</td>\n",
       "      <td>122.4</td>\n",
       "      <td>27.3</td>\n",
       "      <td>5.0</td>\n",
       "      <td>7.5</td>\n",
       "      <td>54.3</td>\n",
       "      <td>3.5</td>\n",
       "      <td>5.5</td>\n",
       "      <td>30.9</td>\n",
       "      <td>26.3</td>\n",
       "      <td>12.5</td>\n",
       "      <td>30.7</td>\n",
       "      <td>27.5</td>\n",
       "      <td>163.3</td>\n",
       "      <td>5.8</td>\n",
       "      <td>6.0</td>\n",
       "      <td>19.1</td>\n",
       "      <td>286.4</td>\n",
       "      <td>73.2</td>\n",
       "      <td>10.1</td>\n",
       "      <td>166.9</td>\n",
       "      <td>4.3</td>\n",
       "      <td>17.5</td>\n",
       "      <td>24.2</td>\n",
       "      <td>107.1</td>\n",
       "      <td>9.9</td>\n",
       "      <td>71.9</td>\n",
       "      <td>41.9</td>\n",
       "      <td>11.3</td>\n",
       "      <td>9.0</td>\n",
       "      <td>22.1</td>\n",
       "      <td>11.5</td>\n",
       "      <td>117.8</td>\n",
       "      <td>97.4</td>\n",
       "      <td>26.6</td>\n",
       "      <td>0.170275</td>\n",
       "      <td>0.163749</td>\n",
       "      <td>0.855013</td>\n",
       "      <td>CFH</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>ENSG00000001036</td>\n",
       "      <td>27.4</td>\n",
       "      <td>35.1</td>\n",
       "      <td>7.1</td>\n",
       "      <td>7.8</td>\n",
       "      <td>24.8</td>\n",
       "      <td>5.9</td>\n",
       "      <td>6.3</td>\n",
       "      <td>32.5</td>\n",
       "      <td>23.3</td>\n",
       "      <td>26.6</td>\n",
       "      <td>11.1</td>\n",
       "      <td>27.7</td>\n",
       "      <td>14.0</td>\n",
       "      <td>7.9</td>\n",
       "      <td>10.7</td>\n",
       "      <td>20.7</td>\n",
       "      <td>15.2</td>\n",
       "      <td>27.3</td>\n",
       "      <td>10.5</td>\n",
       "      <td>26.8</td>\n",
       "      <td>12.7</td>\n",
       "      <td>16.8</td>\n",
       "      <td>16.3</td>\n",
       "      <td>21.7</td>\n",
       "      <td>3.0</td>\n",
       "      <td>11.5</td>\n",
       "      <td>23.9</td>\n",
       "      <td>17.6</td>\n",
       "      <td>22.4</td>\n",
       "      <td>18.8</td>\n",
       "      <td>11.1</td>\n",
       "      <td>26.6</td>\n",
       "      <td>24.9</td>\n",
       "      <td>20.7</td>\n",
       "      <td>0.031989</td>\n",
       "      <td>0.019567</td>\n",
       "      <td>0.497885</td>\n",
       "      <td>FUCA2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>ENSG00000001084</td>\n",
       "      <td>11.0</td>\n",
       "      <td>9.9</td>\n",
       "      <td>7.6</td>\n",
       "      <td>9.8</td>\n",
       "      <td>9.1</td>\n",
       "      <td>5.7</td>\n",
       "      <td>8.6</td>\n",
       "      <td>9.6</td>\n",
       "      <td>13.9</td>\n",
       "      <td>8.4</td>\n",
       "      <td>18.0</td>\n",
       "      <td>10.5</td>\n",
       "      <td>3.1</td>\n",
       "      <td>8.2</td>\n",
       "      <td>8.1</td>\n",
       "      <td>6.1</td>\n",
       "      <td>12.6</td>\n",
       "      <td>7.0</td>\n",
       "      <td>9.6</td>\n",
       "      <td>6.0</td>\n",
       "      <td>3.7</td>\n",
       "      <td>5.5</td>\n",
       "      <td>12.1</td>\n",
       "      <td>10.7</td>\n",
       "      <td>3.7</td>\n",
       "      <td>8.6</td>\n",
       "      <td>15.6</td>\n",
       "      <td>10.9</td>\n",
       "      <td>15.8</td>\n",
       "      <td>11.0</td>\n",
       "      <td>5.2</td>\n",
       "      <td>12.4</td>\n",
       "      <td>44.5</td>\n",
       "      <td>14.3</td>\n",
       "      <td>0.043669</td>\n",
       "      <td>0.033442</td>\n",
       "      <td>0.787334</td>\n",
       "      <td>GCLC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>ENSG00000001167</td>\n",
       "      <td>9.9</td>\n",
       "      <td>7.4</td>\n",
       "      <td>6.0</td>\n",
       "      <td>5.9</td>\n",
       "      <td>10.9</td>\n",
       "      <td>11.2</td>\n",
       "      <td>6.3</td>\n",
       "      <td>12.4</td>\n",
       "      <td>7.2</td>\n",
       "      <td>10.9</td>\n",
       "      <td>8.7</td>\n",
       "      <td>11.3</td>\n",
       "      <td>4.7</td>\n",
       "      <td>6.4</td>\n",
       "      <td>6.0</td>\n",
       "      <td>5.7</td>\n",
       "      <td>4.5</td>\n",
       "      <td>12.7</td>\n",
       "      <td>6.8</td>\n",
       "      <td>14.1</td>\n",
       "      <td>4.3</td>\n",
       "      <td>9.9</td>\n",
       "      <td>9.6</td>\n",
       "      <td>8.3</td>\n",
       "      <td>4.5</td>\n",
       "      <td>8.6</td>\n",
       "      <td>8.9</td>\n",
       "      <td>10.6</td>\n",
       "      <td>10.4</td>\n",
       "      <td>6.1</td>\n",
       "      <td>42.3</td>\n",
       "      <td>17.6</td>\n",
       "      <td>11.2</td>\n",
       "      <td>9.9</td>\n",
       "      <td>0.041457</td>\n",
       "      <td>0.015504</td>\n",
       "      <td>0.793037</td>\n",
       "      <td>NFYA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              Gene  adipose tissue  adrenal gland  amygdala  basal ganglia  \\\n",
       "0  ENSG00000000003            27.4           15.5       7.3            7.7   \n",
       "1  ENSG00000000005            20.7            0.0       0.1            0.0   \n",
       "2  ENSG00000000419            33.6           37.7      12.4           15.6   \n",
       "3  ENSG00000000457             5.4            4.5       1.5            1.7   \n",
       "4  ENSG00000000460             0.7            0.5       0.2            0.2   \n",
       "5  ENSG00000000938            19.9            4.8       2.7            3.2   \n",
       "6  ENSG00000000971           122.4           27.3       5.0            7.5   \n",
       "7  ENSG00000001036            27.4           35.1       7.1            7.8   \n",
       "8  ENSG00000001084            11.0            9.9       7.6            9.8   \n",
       "9  ENSG00000001167             9.9            7.4       6.0            5.9   \n",
       "\n",
       "   breast  cerebellum  cerebral cortex  cervix, uterine  colon  endometrium  \\\n",
       "0    32.0         2.5              5.6             30.6   32.6         22.8   \n",
       "1    11.9         0.0              0.1              0.2    0.6          0.1   \n",
       "2    34.0        24.9             22.4             35.8   31.5         39.2   \n",
       "3     6.7         5.2              2.3              7.9    5.4          6.5   \n",
       "4     0.7         1.5              0.2              0.9    0.6          0.7   \n",
       "5    13.7         2.4              3.8              9.1    5.0          8.6   \n",
       "6    54.3         3.5              5.5             30.9   26.3         12.5   \n",
       "7    24.8         5.9              6.3             32.5   23.3         26.6   \n",
       "8     9.1         5.7              8.6              9.6   13.9          8.4   \n",
       "9    10.9        11.2              6.3             12.4    7.2         10.9   \n",
       "\n",
       "   esophagus  fallopian tube  heart muscle  hippocampal formation  \\\n",
       "0       35.5            27.7           4.5                    7.0   \n",
       "1        0.0             0.3           0.1                    0.1   \n",
       "2       33.8            36.5          21.6                   14.6   \n",
       "3        5.7             6.3           2.0                    1.7   \n",
       "4        0.7             0.5           0.2                    0.2   \n",
       "5        3.8            18.7           3.7                    3.2   \n",
       "6       30.7            27.5         163.3                    5.8   \n",
       "7       11.1            27.7          14.0                    7.9   \n",
       "8       18.0            10.5           3.1                    8.2   \n",
       "9        8.7            11.3           4.7                    6.4   \n",
       "\n",
       "   hypothalamus  kidney  liver  lung  midbrain  ovary  pancreas  \\\n",
       "0           9.9    16.2   23.8  12.0       6.5   50.3       7.6   \n",
       "1           0.1     0.5    0.0   0.0       0.4    1.3       0.0   \n",
       "2          17.4    20.5   17.3  31.5      14.1   33.2      18.2   \n",
       "3           2.1     3.4    3.8   5.1       1.7    5.9       2.5   \n",
       "4           0.3     0.3    0.5   0.5       0.3    0.6       0.2   \n",
       "5           3.7     4.4    4.2  86.1       4.9    2.4       1.7   \n",
       "6           6.0    19.1  286.4  73.2      10.1  166.9       4.3   \n",
       "7          10.7    20.7   15.2  27.3      10.5   26.8      12.7   \n",
       "8           8.1     6.1   12.6   7.0       9.6    6.0       3.7   \n",
       "9           6.0     5.7    4.5  12.7       6.8   14.1       4.3   \n",
       "\n",
       "   pituitary gland  prostate  salivary gland  skeletal muscle  skin  \\\n",
       "0             36.5      17.5            30.7              2.0   8.8   \n",
       "1              0.0       0.4             0.4              1.3   3.1   \n",
       "2             28.4      29.9            28.0             26.2  30.5   \n",
       "3              4.5       6.8             6.7              3.6   6.7   \n",
       "4              0.5       0.7             0.6              0.1   0.6   \n",
       "5              2.7       6.2             5.8              1.3   3.7   \n",
       "6             17.5      24.2           107.1              9.9  71.9   \n",
       "7             16.8      16.3            21.7              3.0  11.5   \n",
       "8              5.5      12.1            10.7              3.7   8.6   \n",
       "9              9.9       9.6             8.3              4.5   8.6   \n",
       "\n",
       "   small intestine  spinal cord  spleen  stomach  testis  thyroid gland  \\\n",
       "0             16.2          6.4     7.8      8.8    49.5           18.7   \n",
       "1              0.5          0.0     0.1      0.0     0.1            0.2   \n",
       "2             28.6         16.7    34.0     24.7    42.7           35.2   \n",
       "3              5.4          2.5     5.7      3.8     6.3            7.4   \n",
       "4              0.6          0.5     0.9      0.4     4.9            0.8   \n",
       "5              8.2          6.5   124.3      4.1     3.2            7.6   \n",
       "6             41.9         11.3     9.0     22.1    11.5          117.8   \n",
       "7             23.9         17.6    22.4     18.8    11.1           26.6   \n",
       "8             15.6         10.9    15.8     11.0     5.2           12.4   \n",
       "9              8.9         10.6    10.4      6.1    42.3           17.6   \n",
       "\n",
       "   urinary bladder  vagina   entropy  rel_entropy       tau annotation  \n",
       "0             37.6    23.8  0.068330     0.037445  0.639135     TSPAN6  \n",
       "1              0.1     0.2  0.534839     0.404403  0.974235       TNMD  \n",
       "2             32.7    34.7  0.013113     0.006362  0.364559       DPM1  \n",
       "3              9.1     6.6  0.028105     0.010124  0.489510      SCYL3  \n",
       "4              1.0     0.7  0.104682     0.051871  0.984539   C1orf112  \n",
       "5              7.1     6.7  0.268393     0.194874  0.933421        FGR  \n",
       "6             97.4    26.6  0.170275     0.163749  0.855013        CFH  \n",
       "7             24.9    20.7  0.031989     0.019567  0.497885      FUCA2  \n",
       "8             44.5    14.3  0.043669     0.033442  0.787334       GCLC  \n",
       "9             11.2     9.9  0.041457     0.015504  0.793037       NFYA  "
      ]
     },
     "execution_count": 561,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "expression['annotation'] = expression['Gene'].map(ensembl_id_to_gene_id)\n",
    "expression.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 562,
   "id": "655cc59c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "18392"
      ]
     },
     "execution_count": 562,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(expression)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7af109e6",
   "metadata": {},
   "source": [
    "Let's reorder the columns."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 563,
   "id": "6abe5870",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>annotation</th>\n",
       "      <th>entropy</th>\n",
       "      <th>rel_entropy</th>\n",
       "      <th>tau</th>\n",
       "      <th>adipose tissue</th>\n",
       "      <th>adrenal gland</th>\n",
       "      <th>amygdala</th>\n",
       "      <th>basal ganglia</th>\n",
       "      <th>breast</th>\n",
       "      <th>cerebellum</th>\n",
       "      <th>cerebral cortex</th>\n",
       "      <th>cervix, uterine</th>\n",
       "      <th>colon</th>\n",
       "      <th>endometrium</th>\n",
       "      <th>esophagus</th>\n",
       "      <th>fallopian tube</th>\n",
       "      <th>heart muscle</th>\n",
       "      <th>hippocampal formation</th>\n",
       "      <th>hypothalamus</th>\n",
       "      <th>kidney</th>\n",
       "      <th>liver</th>\n",
       "      <th>lung</th>\n",
       "      <th>midbrain</th>\n",
       "      <th>ovary</th>\n",
       "      <th>pancreas</th>\n",
       "      <th>pituitary gland</th>\n",
       "      <th>prostate</th>\n",
       "      <th>salivary gland</th>\n",
       "      <th>skeletal muscle</th>\n",
       "      <th>skin</th>\n",
       "      <th>small intestine</th>\n",
       "      <th>spinal cord</th>\n",
       "      <th>spleen</th>\n",
       "      <th>stomach</th>\n",
       "      <th>testis</th>\n",
       "      <th>thyroid gland</th>\n",
       "      <th>urinary bladder</th>\n",
       "      <th>vagina</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000000003</td>\n",
       "      <td>TSPAN6</td>\n",
       "      <td>0.068330</td>\n",
       "      <td>0.037445</td>\n",
       "      <td>0.639135</td>\n",
       "      <td>27.4</td>\n",
       "      <td>15.5</td>\n",
       "      <td>7.3</td>\n",
       "      <td>7.7</td>\n",
       "      <td>32.0</td>\n",
       "      <td>2.5</td>\n",
       "      <td>5.6</td>\n",
       "      <td>30.6</td>\n",
       "      <td>32.6</td>\n",
       "      <td>22.8</td>\n",
       "      <td>35.5</td>\n",
       "      <td>27.7</td>\n",
       "      <td>4.5</td>\n",
       "      <td>7.0</td>\n",
       "      <td>9.9</td>\n",
       "      <td>16.2</td>\n",
       "      <td>23.8</td>\n",
       "      <td>12.0</td>\n",
       "      <td>6.5</td>\n",
       "      <td>50.3</td>\n",
       "      <td>7.6</td>\n",
       "      <td>36.5</td>\n",
       "      <td>17.5</td>\n",
       "      <td>30.7</td>\n",
       "      <td>2.0</td>\n",
       "      <td>8.8</td>\n",
       "      <td>16.2</td>\n",
       "      <td>6.4</td>\n",
       "      <td>7.8</td>\n",
       "      <td>8.8</td>\n",
       "      <td>49.5</td>\n",
       "      <td>18.7</td>\n",
       "      <td>37.6</td>\n",
       "      <td>23.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000000005</td>\n",
       "      <td>TNMD</td>\n",
       "      <td>0.534839</td>\n",
       "      <td>0.404403</td>\n",
       "      <td>0.974235</td>\n",
       "      <td>20.7</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>11.9</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.1</td>\n",
       "      <td>0.2</td>\n",
       "      <td>0.6</td>\n",
       "      <td>0.1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.3</td>\n",
       "      <td>0.1</td>\n",
       "      <td>0.1</td>\n",
       "      <td>0.1</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.4</td>\n",
       "      <td>1.3</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.4</td>\n",
       "      <td>0.4</td>\n",
       "      <td>1.3</td>\n",
       "      <td>3.1</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.1</td>\n",
       "      <td>0.2</td>\n",
       "      <td>0.1</td>\n",
       "      <td>0.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000000419</td>\n",
       "      <td>DPM1</td>\n",
       "      <td>0.013113</td>\n",
       "      <td>0.006362</td>\n",
       "      <td>0.364559</td>\n",
       "      <td>33.6</td>\n",
       "      <td>37.7</td>\n",
       "      <td>12.4</td>\n",
       "      <td>15.6</td>\n",
       "      <td>34.0</td>\n",
       "      <td>24.9</td>\n",
       "      <td>22.4</td>\n",
       "      <td>35.8</td>\n",
       "      <td>31.5</td>\n",
       "      <td>39.2</td>\n",
       "      <td>33.8</td>\n",
       "      <td>36.5</td>\n",
       "      <td>21.6</td>\n",
       "      <td>14.6</td>\n",
       "      <td>17.4</td>\n",
       "      <td>20.5</td>\n",
       "      <td>17.3</td>\n",
       "      <td>31.5</td>\n",
       "      <td>14.1</td>\n",
       "      <td>33.2</td>\n",
       "      <td>18.2</td>\n",
       "      <td>28.4</td>\n",
       "      <td>29.9</td>\n",
       "      <td>28.0</td>\n",
       "      <td>26.2</td>\n",
       "      <td>30.5</td>\n",
       "      <td>28.6</td>\n",
       "      <td>16.7</td>\n",
       "      <td>34.0</td>\n",
       "      <td>24.7</td>\n",
       "      <td>42.7</td>\n",
       "      <td>35.2</td>\n",
       "      <td>32.7</td>\n",
       "      <td>34.7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000000457</td>\n",
       "      <td>SCYL3</td>\n",
       "      <td>0.028105</td>\n",
       "      <td>0.010124</td>\n",
       "      <td>0.489510</td>\n",
       "      <td>5.4</td>\n",
       "      <td>4.5</td>\n",
       "      <td>1.5</td>\n",
       "      <td>1.7</td>\n",
       "      <td>6.7</td>\n",
       "      <td>5.2</td>\n",
       "      <td>2.3</td>\n",
       "      <td>7.9</td>\n",
       "      <td>5.4</td>\n",
       "      <td>6.5</td>\n",
       "      <td>5.7</td>\n",
       "      <td>6.3</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.7</td>\n",
       "      <td>2.1</td>\n",
       "      <td>3.4</td>\n",
       "      <td>3.8</td>\n",
       "      <td>5.1</td>\n",
       "      <td>1.7</td>\n",
       "      <td>5.9</td>\n",
       "      <td>2.5</td>\n",
       "      <td>4.5</td>\n",
       "      <td>6.8</td>\n",
       "      <td>6.7</td>\n",
       "      <td>3.6</td>\n",
       "      <td>6.7</td>\n",
       "      <td>5.4</td>\n",
       "      <td>2.5</td>\n",
       "      <td>5.7</td>\n",
       "      <td>3.8</td>\n",
       "      <td>6.3</td>\n",
       "      <td>7.4</td>\n",
       "      <td>9.1</td>\n",
       "      <td>6.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000000460</td>\n",
       "      <td>C1orf112</td>\n",
       "      <td>0.104682</td>\n",
       "      <td>0.051871</td>\n",
       "      <td>0.984539</td>\n",
       "      <td>0.7</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0.2</td>\n",
       "      <td>0.2</td>\n",
       "      <td>0.7</td>\n",
       "      <td>1.5</td>\n",
       "      <td>0.2</td>\n",
       "      <td>0.9</td>\n",
       "      <td>0.6</td>\n",
       "      <td>0.7</td>\n",
       "      <td>0.7</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0.2</td>\n",
       "      <td>0.2</td>\n",
       "      <td>0.3</td>\n",
       "      <td>0.3</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0.3</td>\n",
       "      <td>0.6</td>\n",
       "      <td>0.2</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0.7</td>\n",
       "      <td>0.6</td>\n",
       "      <td>0.1</td>\n",
       "      <td>0.6</td>\n",
       "      <td>0.6</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0.9</td>\n",
       "      <td>0.4</td>\n",
       "      <td>4.9</td>\n",
       "      <td>0.8</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>ENSG00000000938</td>\n",
       "      <td>FGR</td>\n",
       "      <td>0.268393</td>\n",
       "      <td>0.194874</td>\n",
       "      <td>0.933421</td>\n",
       "      <td>19.9</td>\n",
       "      <td>4.8</td>\n",
       "      <td>2.7</td>\n",
       "      <td>3.2</td>\n",
       "      <td>13.7</td>\n",
       "      <td>2.4</td>\n",
       "      <td>3.8</td>\n",
       "      <td>9.1</td>\n",
       "      <td>5.0</td>\n",
       "      <td>8.6</td>\n",
       "      <td>3.8</td>\n",
       "      <td>18.7</td>\n",
       "      <td>3.7</td>\n",
       "      <td>3.2</td>\n",
       "      <td>3.7</td>\n",
       "      <td>4.4</td>\n",
       "      <td>4.2</td>\n",
       "      <td>86.1</td>\n",
       "      <td>4.9</td>\n",
       "      <td>2.4</td>\n",
       "      <td>1.7</td>\n",
       "      <td>2.7</td>\n",
       "      <td>6.2</td>\n",
       "      <td>5.8</td>\n",
       "      <td>1.3</td>\n",
       "      <td>3.7</td>\n",
       "      <td>8.2</td>\n",
       "      <td>6.5</td>\n",
       "      <td>124.3</td>\n",
       "      <td>4.1</td>\n",
       "      <td>3.2</td>\n",
       "      <td>7.6</td>\n",
       "      <td>7.1</td>\n",
       "      <td>6.7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>ENSG00000000971</td>\n",
       "      <td>CFH</td>\n",
       "      <td>0.170275</td>\n",
       "      <td>0.163749</td>\n",
       "      <td>0.855013</td>\n",
       "      <td>122.4</td>\n",
       "      <td>27.3</td>\n",
       "      <td>5.0</td>\n",
       "      <td>7.5</td>\n",
       "      <td>54.3</td>\n",
       "      <td>3.5</td>\n",
       "      <td>5.5</td>\n",
       "      <td>30.9</td>\n",
       "      <td>26.3</td>\n",
       "      <td>12.5</td>\n",
       "      <td>30.7</td>\n",
       "      <td>27.5</td>\n",
       "      <td>163.3</td>\n",
       "      <td>5.8</td>\n",
       "      <td>6.0</td>\n",
       "      <td>19.1</td>\n",
       "      <td>286.4</td>\n",
       "      <td>73.2</td>\n",
       "      <td>10.1</td>\n",
       "      <td>166.9</td>\n",
       "      <td>4.3</td>\n",
       "      <td>17.5</td>\n",
       "      <td>24.2</td>\n",
       "      <td>107.1</td>\n",
       "      <td>9.9</td>\n",
       "      <td>71.9</td>\n",
       "      <td>41.9</td>\n",
       "      <td>11.3</td>\n",
       "      <td>9.0</td>\n",
       "      <td>22.1</td>\n",
       "      <td>11.5</td>\n",
       "      <td>117.8</td>\n",
       "      <td>97.4</td>\n",
       "      <td>26.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>ENSG00000001036</td>\n",
       "      <td>FUCA2</td>\n",
       "      <td>0.031989</td>\n",
       "      <td>0.019567</td>\n",
       "      <td>0.497885</td>\n",
       "      <td>27.4</td>\n",
       "      <td>35.1</td>\n",
       "      <td>7.1</td>\n",
       "      <td>7.8</td>\n",
       "      <td>24.8</td>\n",
       "      <td>5.9</td>\n",
       "      <td>6.3</td>\n",
       "      <td>32.5</td>\n",
       "      <td>23.3</td>\n",
       "      <td>26.6</td>\n",
       "      <td>11.1</td>\n",
       "      <td>27.7</td>\n",
       "      <td>14.0</td>\n",
       "      <td>7.9</td>\n",
       "      <td>10.7</td>\n",
       "      <td>20.7</td>\n",
       "      <td>15.2</td>\n",
       "      <td>27.3</td>\n",
       "      <td>10.5</td>\n",
       "      <td>26.8</td>\n",
       "      <td>12.7</td>\n",
       "      <td>16.8</td>\n",
       "      <td>16.3</td>\n",
       "      <td>21.7</td>\n",
       "      <td>3.0</td>\n",
       "      <td>11.5</td>\n",
       "      <td>23.9</td>\n",
       "      <td>17.6</td>\n",
       "      <td>22.4</td>\n",
       "      <td>18.8</td>\n",
       "      <td>11.1</td>\n",
       "      <td>26.6</td>\n",
       "      <td>24.9</td>\n",
       "      <td>20.7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>ENSG00000001084</td>\n",
       "      <td>GCLC</td>\n",
       "      <td>0.043669</td>\n",
       "      <td>0.033442</td>\n",
       "      <td>0.787334</td>\n",
       "      <td>11.0</td>\n",
       "      <td>9.9</td>\n",
       "      <td>7.6</td>\n",
       "      <td>9.8</td>\n",
       "      <td>9.1</td>\n",
       "      <td>5.7</td>\n",
       "      <td>8.6</td>\n",
       "      <td>9.6</td>\n",
       "      <td>13.9</td>\n",
       "      <td>8.4</td>\n",
       "      <td>18.0</td>\n",
       "      <td>10.5</td>\n",
       "      <td>3.1</td>\n",
       "      <td>8.2</td>\n",
       "      <td>8.1</td>\n",
       "      <td>6.1</td>\n",
       "      <td>12.6</td>\n",
       "      <td>7.0</td>\n",
       "      <td>9.6</td>\n",
       "      <td>6.0</td>\n",
       "      <td>3.7</td>\n",
       "      <td>5.5</td>\n",
       "      <td>12.1</td>\n",
       "      <td>10.7</td>\n",
       "      <td>3.7</td>\n",
       "      <td>8.6</td>\n",
       "      <td>15.6</td>\n",
       "      <td>10.9</td>\n",
       "      <td>15.8</td>\n",
       "      <td>11.0</td>\n",
       "      <td>5.2</td>\n",
       "      <td>12.4</td>\n",
       "      <td>44.5</td>\n",
       "      <td>14.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>ENSG00000001167</td>\n",
       "      <td>NFYA</td>\n",
       "      <td>0.041457</td>\n",
       "      <td>0.015504</td>\n",
       "      <td>0.793037</td>\n",
       "      <td>9.9</td>\n",
       "      <td>7.4</td>\n",
       "      <td>6.0</td>\n",
       "      <td>5.9</td>\n",
       "      <td>10.9</td>\n",
       "      <td>11.2</td>\n",
       "      <td>6.3</td>\n",
       "      <td>12.4</td>\n",
       "      <td>7.2</td>\n",
       "      <td>10.9</td>\n",
       "      <td>8.7</td>\n",
       "      <td>11.3</td>\n",
       "      <td>4.7</td>\n",
       "      <td>6.4</td>\n",
       "      <td>6.0</td>\n",
       "      <td>5.7</td>\n",
       "      <td>4.5</td>\n",
       "      <td>12.7</td>\n",
       "      <td>6.8</td>\n",
       "      <td>14.1</td>\n",
       "      <td>4.3</td>\n",
       "      <td>9.9</td>\n",
       "      <td>9.6</td>\n",
       "      <td>8.3</td>\n",
       "      <td>4.5</td>\n",
       "      <td>8.6</td>\n",
       "      <td>8.9</td>\n",
       "      <td>10.6</td>\n",
       "      <td>10.4</td>\n",
       "      <td>6.1</td>\n",
       "      <td>42.3</td>\n",
       "      <td>17.6</td>\n",
       "      <td>11.2</td>\n",
       "      <td>9.9</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              Gene annotation   entropy  rel_entropy       tau  \\\n",
       "0  ENSG00000000003     TSPAN6  0.068330     0.037445  0.639135   \n",
       "1  ENSG00000000005       TNMD  0.534839     0.404403  0.974235   \n",
       "2  ENSG00000000419       DPM1  0.013113     0.006362  0.364559   \n",
       "3  ENSG00000000457      SCYL3  0.028105     0.010124  0.489510   \n",
       "4  ENSG00000000460   C1orf112  0.104682     0.051871  0.984539   \n",
       "5  ENSG00000000938        FGR  0.268393     0.194874  0.933421   \n",
       "6  ENSG00000000971        CFH  0.170275     0.163749  0.855013   \n",
       "7  ENSG00000001036      FUCA2  0.031989     0.019567  0.497885   \n",
       "8  ENSG00000001084       GCLC  0.043669     0.033442  0.787334   \n",
       "9  ENSG00000001167       NFYA  0.041457     0.015504  0.793037   \n",
       "\n",
       "   adipose tissue  adrenal gland  amygdala  basal ganglia  breast  cerebellum  \\\n",
       "0            27.4           15.5       7.3            7.7    32.0         2.5   \n",
       "1            20.7            0.0       0.1            0.0    11.9         0.0   \n",
       "2            33.6           37.7      12.4           15.6    34.0        24.9   \n",
       "3             5.4            4.5       1.5            1.7     6.7         5.2   \n",
       "4             0.7            0.5       0.2            0.2     0.7         1.5   \n",
       "5            19.9            4.8       2.7            3.2    13.7         2.4   \n",
       "6           122.4           27.3       5.0            7.5    54.3         3.5   \n",
       "7            27.4           35.1       7.1            7.8    24.8         5.9   \n",
       "8            11.0            9.9       7.6            9.8     9.1         5.7   \n",
       "9             9.9            7.4       6.0            5.9    10.9        11.2   \n",
       "\n",
       "   cerebral cortex  cervix, uterine  colon  endometrium  esophagus  \\\n",
       "0              5.6             30.6   32.6         22.8       35.5   \n",
       "1              0.1              0.2    0.6          0.1        0.0   \n",
       "2             22.4             35.8   31.5         39.2       33.8   \n",
       "3              2.3              7.9    5.4          6.5        5.7   \n",
       "4              0.2              0.9    0.6          0.7        0.7   \n",
       "5              3.8              9.1    5.0          8.6        3.8   \n",
       "6              5.5             30.9   26.3         12.5       30.7   \n",
       "7              6.3             32.5   23.3         26.6       11.1   \n",
       "8              8.6              9.6   13.9          8.4       18.0   \n",
       "9              6.3             12.4    7.2         10.9        8.7   \n",
       "\n",
       "   fallopian tube  heart muscle  hippocampal formation  hypothalamus  kidney  \\\n",
       "0            27.7           4.5                    7.0           9.9    16.2   \n",
       "1             0.3           0.1                    0.1           0.1     0.5   \n",
       "2            36.5          21.6                   14.6          17.4    20.5   \n",
       "3             6.3           2.0                    1.7           2.1     3.4   \n",
       "4             0.5           0.2                    0.2           0.3     0.3   \n",
       "5            18.7           3.7                    3.2           3.7     4.4   \n",
       "6            27.5         163.3                    5.8           6.0    19.1   \n",
       "7            27.7          14.0                    7.9          10.7    20.7   \n",
       "8            10.5           3.1                    8.2           8.1     6.1   \n",
       "9            11.3           4.7                    6.4           6.0     5.7   \n",
       "\n",
       "   liver  lung  midbrain  ovary  pancreas  pituitary gland  prostate  \\\n",
       "0   23.8  12.0       6.5   50.3       7.6             36.5      17.5   \n",
       "1    0.0   0.0       0.4    1.3       0.0              0.0       0.4   \n",
       "2   17.3  31.5      14.1   33.2      18.2             28.4      29.9   \n",
       "3    3.8   5.1       1.7    5.9       2.5              4.5       6.8   \n",
       "4    0.5   0.5       0.3    0.6       0.2              0.5       0.7   \n",
       "5    4.2  86.1       4.9    2.4       1.7              2.7       6.2   \n",
       "6  286.4  73.2      10.1  166.9       4.3             17.5      24.2   \n",
       "7   15.2  27.3      10.5   26.8      12.7             16.8      16.3   \n",
       "8   12.6   7.0       9.6    6.0       3.7              5.5      12.1   \n",
       "9    4.5  12.7       6.8   14.1       4.3              9.9       9.6   \n",
       "\n",
       "   salivary gland  skeletal muscle  skin  small intestine  spinal cord  \\\n",
       "0            30.7              2.0   8.8             16.2          6.4   \n",
       "1             0.4              1.3   3.1              0.5          0.0   \n",
       "2            28.0             26.2  30.5             28.6         16.7   \n",
       "3             6.7              3.6   6.7              5.4          2.5   \n",
       "4             0.6              0.1   0.6              0.6          0.5   \n",
       "5             5.8              1.3   3.7              8.2          6.5   \n",
       "6           107.1              9.9  71.9             41.9         11.3   \n",
       "7            21.7              3.0  11.5             23.9         17.6   \n",
       "8            10.7              3.7   8.6             15.6         10.9   \n",
       "9             8.3              4.5   8.6              8.9         10.6   \n",
       "\n",
       "   spleen  stomach  testis  thyroid gland  urinary bladder  vagina  \n",
       "0     7.8      8.8    49.5           18.7             37.6    23.8  \n",
       "1     0.1      0.0     0.1            0.2              0.1     0.2  \n",
       "2    34.0     24.7    42.7           35.2             32.7    34.7  \n",
       "3     5.7      3.8     6.3            7.4              9.1     6.6  \n",
       "4     0.9      0.4     4.9            0.8              1.0     0.7  \n",
       "5   124.3      4.1     3.2            7.6              7.1     6.7  \n",
       "6     9.0     22.1    11.5          117.8             97.4    26.6  \n",
       "7    22.4     18.8    11.1           26.6             24.9    20.7  \n",
       "8    15.8     11.0     5.2           12.4             44.5    14.3  \n",
       "9    10.4      6.1    42.3           17.6             11.2     9.9  "
      ]
     },
     "execution_count": 563,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "expression = expression[['Gene','annotation','entropy','rel_entropy','tau','adipose tissue','adrenal gland','amygdala','basal ganglia','breast','cerebellum','cerebral cortex','cervix, uterine','colon','endometrium','esophagus','fallopian tube','heart muscle','hippocampal formation','hypothalamus','kidney','liver','lung','midbrain','ovary','pancreas','pituitary gland','prostate','salivary gland','skeletal muscle','skin','small intestine','spinal cord','spleen','stomach','testis','thyroid gland','urinary bladder','vagina']]\n",
    "expression.head(10)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d7a2400e",
   "metadata": {},
   "source": [
    "And save the dataframe while we're at it."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 564,
   "id": "2596854b",
   "metadata": {},
   "outputs": [],
   "source": [
    "expression.to_csv('../gene_expression/GTEX_gene_expression_with_gene_ids.txt', sep=\"\\t\", header=True, index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "50a0feb3",
   "metadata": {},
   "source": [
    "Why don't we examine how gene expression is related to the number of SAVs. Remember our N splice variants dataframe from earlier?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 565,
   "id": "f701bcc0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>variant_count_2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AADACL4</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AADAT</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AAMP</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AAR2</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>AARS2</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>AATF</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>AATK</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>ABAT</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>ABC7-42404400C24.1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                 name  variant_count_2\n",
       "0                A1BG                1\n",
       "1             AADACL4                1\n",
       "2               AADAT                1\n",
       "3                AAMP                2\n",
       "4                AAR2                1\n",
       "5               AARS2                1\n",
       "6                AATF                1\n",
       "7                AATK                1\n",
       "8                ABAT                1\n",
       "9  ABC7-42404400C24.1                1"
      ]
     },
     "execution_count": 565,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "n_splice_variants_2.reset_index(inplace=True)\n",
    "n_splice_variants_2.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 566,
   "id": "00b284c3",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>annotation</th>\n",
       "      <th>rel_entropy</th>\n",
       "      <th>variant_count_2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>C1orf112</td>\n",
       "      <td>0.051871</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>GCLC</td>\n",
       "      <td>0.033442</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>CFTR</td>\n",
       "      <td>0.462603</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>ANKIB1</td>\n",
       "      <td>0.010808</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>LAP3</td>\n",
       "      <td>0.020556</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>AOC1</td>\n",
       "      <td>0.483678</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>HECW1</td>\n",
       "      <td>0.300240</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>MAD1L1</td>\n",
       "      <td>0.010812</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>LASP1</td>\n",
       "      <td>0.013782</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>TMEM176A</td>\n",
       "      <td>0.180257</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   annotation  rel_entropy  variant_count_2\n",
       "4    C1orf112     0.051871              1.0\n",
       "8        GCLC     0.033442              1.0\n",
       "15       CFTR     0.462603              1.0\n",
       "16     ANKIB1     0.010808              1.0\n",
       "21       LAP3     0.020556              1.0\n",
       "24       AOC1     0.483678              1.0\n",
       "26      HECW1     0.300240              1.0\n",
       "27     MAD1L1     0.010812              1.0\n",
       "28      LASP1     0.013782              1.0\n",
       "30   TMEM176A     0.180257              1.0"
      ]
     },
     "execution_count": 566,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "entropy_N_SAVs = n_splice_variants_2.rename(columns={'name':'annotation'})\n",
    "entropy_N_SAVs = pd.merge(expression[['annotation','rel_entropy']], entropy_N_SAVs, on='annotation', how='left').dropna()\n",
    "entropy_N_SAVs.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 567,
   "id": "1ec823cb",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>annotation</th>\n",
       "      <th>rel_entropy</th>\n",
       "      <th>variant_count_2</th>\n",
       "      <th>rel_entropy_bin</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>C1orf112</td>\n",
       "      <td>0.051871</td>\n",
       "      <td>1.0</td>\n",
       "      <td>low</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>GCLC</td>\n",
       "      <td>0.033442</td>\n",
       "      <td>1.0</td>\n",
       "      <td>low</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>CFTR</td>\n",
       "      <td>0.462603</td>\n",
       "      <td>1.0</td>\n",
       "      <td>medium</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>ANKIB1</td>\n",
       "      <td>0.010808</td>\n",
       "      <td>1.0</td>\n",
       "      <td>low</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>LAP3</td>\n",
       "      <td>0.020556</td>\n",
       "      <td>1.0</td>\n",
       "      <td>low</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>AOC1</td>\n",
       "      <td>0.483678</td>\n",
       "      <td>1.0</td>\n",
       "      <td>medium</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>HECW1</td>\n",
       "      <td>0.300240</td>\n",
       "      <td>1.0</td>\n",
       "      <td>medium</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>MAD1L1</td>\n",
       "      <td>0.010812</td>\n",
       "      <td>1.0</td>\n",
       "      <td>low</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>LASP1</td>\n",
       "      <td>0.013782</td>\n",
       "      <td>1.0</td>\n",
       "      <td>low</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>TMEM176A</td>\n",
       "      <td>0.180257</td>\n",
       "      <td>1.0</td>\n",
       "      <td>medium</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   annotation  rel_entropy  variant_count_2 rel_entropy_bin\n",
       "4    C1orf112     0.051871              1.0             low\n",
       "8        GCLC     0.033442              1.0             low\n",
       "15       CFTR     0.462603              1.0          medium\n",
       "16     ANKIB1     0.010808              1.0             low\n",
       "21       LAP3     0.020556              1.0             low\n",
       "24       AOC1     0.483678              1.0          medium\n",
       "26      HECW1     0.300240              1.0          medium\n",
       "27     MAD1L1     0.010812              1.0             low\n",
       "28      LASP1     0.013782              1.0             low\n",
       "30   TMEM176A     0.180257              1.0          medium"
      ]
     },
     "execution_count": 567,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "entropy_bins = [-np.inf, 0.1, 0.5, np.inf]\n",
    "entropy_N_SAVs['rel_entropy_bin'] = pd.cut(entropy_N_SAVs['rel_entropy'], entropy_bins, labels=['low', 'medium', 'high'])\n",
    "entropy_N_SAVs.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 568,
   "id": "8ec32139",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "4061"
      ]
     },
     "execution_count": 568,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(entropy_N_SAVs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 569,
   "id": "d575b7e0",
   "metadata": {},
   "outputs": [],
   "source": [
    "low_entropy = entropy_N_SAVs.loc[entropy_N_SAVs['rel_entropy_bin'] == 'low', 'variant_count_2']\n",
    "med_entropy = entropy_N_SAVs.loc[entropy_N_SAVs['rel_entropy_bin'] == 'medium', 'variant_count_2']\n",
    "high_entropy = entropy_N_SAVs.loc[entropy_N_SAVs['rel_entropy_bin'] == 'high', 'variant_count_2']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 570,
   "id": "2b15535b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2434"
      ]
     },
     "execution_count": 570,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(low_entropy)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 571,
   "id": "f72c0092",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1268"
      ]
     },
     "execution_count": 571,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(med_entropy)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 572,
   "id": "e30b7f46",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "359"
      ]
     },
     "execution_count": 572,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(high_entropy)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 573,
   "id": "0b5645bf",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "KruskalResult(statistic=1.8914270893492804, pvalue=0.3884023295875893)"
      ]
     },
     "execution_count": 573,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "kruskal(low_entropy, med_entropy, high_entropy)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c7a0319c",
   "metadata": {},
   "source": [
    "Now let's look at the relationship between gene expression and delta max."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 574,
   "id": "21ab83a3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>annotation</th>\n",
       "      <th>delta_max</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>0.22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>0.29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>130</th>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>0.29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>155</th>\n",
       "      <td>AGRN</td>\n",
       "      <td>0.24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>210</th>\n",
       "      <td>AGRN</td>\n",
       "      <td>0.22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1606940</th>\n",
       "      <td>EHMT1</td>\n",
       "      <td>0.33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1606952</th>\n",
       "      <td>EHMT1</td>\n",
       "      <td>0.31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1606989</th>\n",
       "      <td>CACNA1B</td>\n",
       "      <td>0.28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1607338</th>\n",
       "      <td>CACNA1B</td>\n",
       "      <td>0.25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1607339</th>\n",
       "      <td>CACNA1B</td>\n",
       "      <td>0.25</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5950 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        annotation  delta_max\n",
       "24          SAMD11       0.22\n",
       "65          SAMD11       0.29\n",
       "130        PLEKHN1       0.29\n",
       "155           AGRN       0.24\n",
       "210           AGRN       0.22\n",
       "...            ...        ...\n",
       "1606940      EHMT1       0.33\n",
       "1606952      EHMT1       0.31\n",
       "1606989    CACNA1B       0.28\n",
       "1607338    CACNA1B       0.25\n",
       "1607339    CACNA1B       0.25\n",
       "\n",
       "[5950 rows x 2 columns]"
      ]
     },
     "execution_count": 574,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "entropy_delta = data_2[['annotation','delta_max']]\n",
    "entropy_delta"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e65137a8",
   "metadata": {},
   "source": [
    "Map on the relative entropy values."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 575,
   "id": "cd2eabad",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>annotation</th>\n",
       "      <th>delta_max</th>\n",
       "      <th>rel_entropy</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>0.22</td>\n",
       "      <td>0.089470</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>0.29</td>\n",
       "      <td>0.089470</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>0.29</td>\n",
       "      <td>0.295620</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AGRN</td>\n",
       "      <td>0.24</td>\n",
       "      <td>0.036305</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AGRN</td>\n",
       "      <td>0.22</td>\n",
       "      <td>0.036305</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>AGRN</td>\n",
       "      <td>0.43</td>\n",
       "      <td>0.036305</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>AGRN</td>\n",
       "      <td>0.48</td>\n",
       "      <td>0.036305</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>TTLL10</td>\n",
       "      <td>0.27</td>\n",
       "      <td>0.411293</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>SDF4</td>\n",
       "      <td>0.38</td>\n",
       "      <td>0.008625</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>SDF4</td>\n",
       "      <td>0.52</td>\n",
       "      <td>0.008625</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  annotation  delta_max  rel_entropy\n",
       "0     SAMD11       0.22     0.089470\n",
       "1     SAMD11       0.29     0.089470\n",
       "2    PLEKHN1       0.29     0.295620\n",
       "3       AGRN       0.24     0.036305\n",
       "4       AGRN       0.22     0.036305\n",
       "5       AGRN       0.43     0.036305\n",
       "6       AGRN       0.48     0.036305\n",
       "7     TTLL10       0.27     0.411293\n",
       "8       SDF4       0.38     0.008625\n",
       "9       SDF4       0.52     0.008625"
      ]
     },
     "execution_count": 575,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "entropy_delta = pd.merge(entropy_delta, expression[['annotation','rel_entropy']], on='annotation', how='left')\n",
    "entropy_delta = entropy_delta.dropna(axis = 0)\n",
    "entropy_delta.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 576,
   "id": "9d8e63dd",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "5700"
      ]
     },
     "execution_count": 576,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(entropy_delta)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 577,
   "id": "bae91772",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>annotation</th>\n",
       "      <th>delta_max</th>\n",
       "      <th>rel_entropy</th>\n",
       "      <th>rel_entropy_bin</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>0.22</td>\n",
       "      <td>0.089470</td>\n",
       "      <td>low</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>0.29</td>\n",
       "      <td>0.089470</td>\n",
       "      <td>low</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>0.29</td>\n",
       "      <td>0.295620</td>\n",
       "      <td>medium</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AGRN</td>\n",
       "      <td>0.24</td>\n",
       "      <td>0.036305</td>\n",
       "      <td>low</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AGRN</td>\n",
       "      <td>0.22</td>\n",
       "      <td>0.036305</td>\n",
       "      <td>low</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>AGRN</td>\n",
       "      <td>0.43</td>\n",
       "      <td>0.036305</td>\n",
       "      <td>low</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>AGRN</td>\n",
       "      <td>0.48</td>\n",
       "      <td>0.036305</td>\n",
       "      <td>low</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>TTLL10</td>\n",
       "      <td>0.27</td>\n",
       "      <td>0.411293</td>\n",
       "      <td>medium</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>SDF4</td>\n",
       "      <td>0.38</td>\n",
       "      <td>0.008625</td>\n",
       "      <td>low</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>SDF4</td>\n",
       "      <td>0.52</td>\n",
       "      <td>0.008625</td>\n",
       "      <td>low</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  annotation  delta_max  rel_entropy rel_entropy_bin\n",
       "0     SAMD11       0.22     0.089470             low\n",
       "1     SAMD11       0.29     0.089470             low\n",
       "2    PLEKHN1       0.29     0.295620          medium\n",
       "3       AGRN       0.24     0.036305             low\n",
       "4       AGRN       0.22     0.036305             low\n",
       "5       AGRN       0.43     0.036305             low\n",
       "6       AGRN       0.48     0.036305             low\n",
       "7     TTLL10       0.27     0.411293          medium\n",
       "8       SDF4       0.38     0.008625             low\n",
       "9       SDF4       0.52     0.008625             low"
      ]
     },
     "execution_count": 577,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "entropy_delta['rel_entropy_bin'] = pd.cut(entropy_delta['rel_entropy'], entropy_bins, labels=['low', 'medium', 'high'])\n",
    "entropy_delta.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 578,
   "id": "fbdbe983",
   "metadata": {},
   "outputs": [],
   "source": [
    "low_entropy = entropy_delta.loc[entropy_delta['rel_entropy_bin'] == 'low', 'delta_max']\n",
    "med_entropy = entropy_delta.loc[entropy_delta['rel_entropy_bin'] == 'medium', 'delta_max']\n",
    "high_entropy = entropy_delta.loc[entropy_delta['rel_entropy_bin'] == 'high', 'delta_max']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 579,
   "id": "a8c7b338",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "3384"
      ]
     },
     "execution_count": 579,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(low_entropy)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 580,
   "id": "9e5e1f09",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1826"
      ]
     },
     "execution_count": 580,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(med_entropy)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 581,
   "id": "003aa8ef",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "490"
      ]
     },
     "execution_count": 581,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(high_entropy)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 582,
   "id": "ee245efc",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "KruskalResult(statistic=6.599476430123113, pvalue=0.03689282412288473)"
      ]
     },
     "execution_count": 582,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "kruskal(low_entropy, med_entropy, high_entropy)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e965ebd5",
   "metadata": {},
   "source": [
    "### Gene Allele Origin Differences <a class = 'anchor' id = 'genealleleorigindifferences'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "eddd4554",
   "metadata": {},
   "source": [
    "Now let's examine the correlation between the number of genes expressed among different allele origins. We'll use a TPM >= 1 to define a gene as \"expressed\"."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5922d758",
   "metadata": {},
   "source": [
    "First Vernot."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 583,
   "id": "dab2d111",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "Vernot_ancient_expression = expression[expression['annotation'].isin(Vernot_ancient_genes_2['gene'])]\n",
    "Vernot_ancient_expression = Vernot_ancient_expression.loc[:, ~Vernot_ancient_expression.columns.isin(['Gene', 'annotation','entropy','rel_entropy','tau'])]\n",
    "Vernot_ancient_expression[Vernot_ancient_expression < 1] = 0\n",
    "Vernot_ancient_expression[Vernot_ancient_expression >= 1] = 1\n",
    "Vernot_ancient_expression_sums = Vernot_ancient_expression.sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 584,
   "id": "e86d5182",
   "metadata": {},
   "outputs": [],
   "source": [
    "Vernot_archaic_specific_expression = expression[expression['annotation'].isin(archaic_specific_genes_2['gene'])]\n",
    "Vernot_archaic_specific_expression = Vernot_archaic_specific_expression.loc[:, ~Vernot_archaic_specific_expression.columns.isin(['Gene', 'annotation','entropy','rel_entropy','tau'])]\n",
    "Vernot_archaic_specific_expression[Vernot_archaic_specific_expression < 1] = 0\n",
    "Vernot_archaic_specific_expression[Vernot_archaic_specific_expression >= 1] = 1\n",
    "Vernot_archaic_specific_expression_sums = Vernot_archaic_specific_expression.sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 585,
   "id": "72fb244c",
   "metadata": {},
   "outputs": [],
   "source": [
    "Vernot_introgressed_expression = expression[expression['annotation'].isin(Vernot_introgressed_genes_2['gene'])]\n",
    "Vernot_introgressed_expression = Vernot_introgressed_expression.loc[:, ~Vernot_introgressed_expression.columns.isin(['Gene', 'annotation','entropy','rel_entropy','tau'])]\n",
    "Vernot_introgressed_expression[Vernot_introgressed_expression < 1] = 0\n",
    "Vernot_introgressed_expression[Vernot_introgressed_expression >= 1] = 1\n",
    "Vernot_introgressed_expression_sums = Vernot_introgressed_expression.sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 586,
   "id": "07f03627",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ancient</th>\n",
       "      <th>archaic-specific</th>\n",
       "      <th>introgressed</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>adipose tissue</th>\n",
       "      <td>1298.0</td>\n",
       "      <td>1352.0</td>\n",
       "      <td>166.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>adrenal gland</th>\n",
       "      <td>1248.0</td>\n",
       "      <td>1264.0</td>\n",
       "      <td>167.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>amygdala</th>\n",
       "      <td>1285.0</td>\n",
       "      <td>1298.0</td>\n",
       "      <td>150.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>basal ganglia</th>\n",
       "      <td>1342.0</td>\n",
       "      <td>1377.0</td>\n",
       "      <td>158.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>breast</th>\n",
       "      <td>1309.0</td>\n",
       "      <td>1367.0</td>\n",
       "      <td>169.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cerebellum</th>\n",
       "      <td>1321.0</td>\n",
       "      <td>1346.0</td>\n",
       "      <td>164.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cerebral cortex</th>\n",
       "      <td>1375.0</td>\n",
       "      <td>1379.0</td>\n",
       "      <td>169.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cervix, uterine</th>\n",
       "      <td>1420.0</td>\n",
       "      <td>1430.0</td>\n",
       "      <td>174.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>colon</th>\n",
       "      <td>1362.0</td>\n",
       "      <td>1398.0</td>\n",
       "      <td>173.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>endometrium</th>\n",
       "      <td>1295.0</td>\n",
       "      <td>1342.0</td>\n",
       "      <td>164.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>esophagus</th>\n",
       "      <td>1252.0</td>\n",
       "      <td>1274.0</td>\n",
       "      <td>160.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>fallopian tube</th>\n",
       "      <td>1372.0</td>\n",
       "      <td>1417.0</td>\n",
       "      <td>170.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>heart muscle</th>\n",
       "      <td>1225.0</td>\n",
       "      <td>1262.0</td>\n",
       "      <td>157.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>hippocampal formation</th>\n",
       "      <td>1319.0</td>\n",
       "      <td>1326.0</td>\n",
       "      <td>161.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>hypothalamus</th>\n",
       "      <td>1379.0</td>\n",
       "      <td>1399.0</td>\n",
       "      <td>174.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>kidney</th>\n",
       "      <td>1290.0</td>\n",
       "      <td>1322.0</td>\n",
       "      <td>165.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>liver</th>\n",
       "      <td>1063.0</td>\n",
       "      <td>1129.0</td>\n",
       "      <td>140.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>lung</th>\n",
       "      <td>1376.0</td>\n",
       "      <td>1400.0</td>\n",
       "      <td>175.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>midbrain</th>\n",
       "      <td>1312.0</td>\n",
       "      <td>1332.0</td>\n",
       "      <td>161.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ovary</th>\n",
       "      <td>1247.0</td>\n",
       "      <td>1309.0</td>\n",
       "      <td>159.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pancreas</th>\n",
       "      <td>1109.0</td>\n",
       "      <td>1138.0</td>\n",
       "      <td>140.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pituitary gland</th>\n",
       "      <td>1436.0</td>\n",
       "      <td>1426.0</td>\n",
       "      <td>185.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>prostate</th>\n",
       "      <td>1353.0</td>\n",
       "      <td>1392.0</td>\n",
       "      <td>170.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>salivary gland</th>\n",
       "      <td>1308.0</td>\n",
       "      <td>1354.0</td>\n",
       "      <td>164.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>skeletal muscle</th>\n",
       "      <td>1078.0</td>\n",
       "      <td>1132.0</td>\n",
       "      <td>142.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>skin</th>\n",
       "      <td>1309.0</td>\n",
       "      <td>1343.0</td>\n",
       "      <td>168.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>small intestine</th>\n",
       "      <td>1325.0</td>\n",
       "      <td>1383.0</td>\n",
       "      <td>171.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>spinal cord</th>\n",
       "      <td>1321.0</td>\n",
       "      <td>1355.0</td>\n",
       "      <td>172.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>spleen</th>\n",
       "      <td>1253.0</td>\n",
       "      <td>1295.0</td>\n",
       "      <td>156.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>stomach</th>\n",
       "      <td>1260.0</td>\n",
       "      <td>1299.0</td>\n",
       "      <td>163.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>testis</th>\n",
       "      <td>1508.0</td>\n",
       "      <td>1539.0</td>\n",
       "      <td>194.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>thyroid gland</th>\n",
       "      <td>1360.0</td>\n",
       "      <td>1382.0</td>\n",
       "      <td>171.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>urinary bladder</th>\n",
       "      <td>1330.0</td>\n",
       "      <td>1366.0</td>\n",
       "      <td>166.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>vagina</th>\n",
       "      <td>1346.0</td>\n",
       "      <td>1372.0</td>\n",
       "      <td>172.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                       ancient  archaic-specific  introgressed\n",
       "adipose tissue          1298.0            1352.0         166.0\n",
       "adrenal gland           1248.0            1264.0         167.0\n",
       "amygdala                1285.0            1298.0         150.0\n",
       "basal ganglia           1342.0            1377.0         158.0\n",
       "breast                  1309.0            1367.0         169.0\n",
       "cerebellum              1321.0            1346.0         164.0\n",
       "cerebral cortex         1375.0            1379.0         169.0\n",
       "cervix, uterine         1420.0            1430.0         174.0\n",
       "colon                   1362.0            1398.0         173.0\n",
       "endometrium             1295.0            1342.0         164.0\n",
       "esophagus               1252.0            1274.0         160.0\n",
       "fallopian tube          1372.0            1417.0         170.0\n",
       "heart muscle            1225.0            1262.0         157.0\n",
       "hippocampal formation   1319.0            1326.0         161.0\n",
       "hypothalamus            1379.0            1399.0         174.0\n",
       "kidney                  1290.0            1322.0         165.0\n",
       "liver                   1063.0            1129.0         140.0\n",
       "lung                    1376.0            1400.0         175.0\n",
       "midbrain                1312.0            1332.0         161.0\n",
       "ovary                   1247.0            1309.0         159.0\n",
       "pancreas                1109.0            1138.0         140.0\n",
       "pituitary gland         1436.0            1426.0         185.0\n",
       "prostate                1353.0            1392.0         170.0\n",
       "salivary gland          1308.0            1354.0         164.0\n",
       "skeletal muscle         1078.0            1132.0         142.0\n",
       "skin                    1309.0            1343.0         168.0\n",
       "small intestine         1325.0            1383.0         171.0\n",
       "spinal cord             1321.0            1355.0         172.0\n",
       "spleen                  1253.0            1295.0         156.0\n",
       "stomach                 1260.0            1299.0         163.0\n",
       "testis                  1508.0            1539.0         194.0\n",
       "thyroid gland           1360.0            1382.0         171.0\n",
       "urinary bladder         1330.0            1366.0         166.0\n",
       "vagina                  1346.0            1372.0         172.0"
      ]
     },
     "execution_count": 586,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Vernot_expression = pd.DataFrame({ 'ancient': Vernot_ancient_expression_sums, 'archaic-specific': Vernot_archaic_specific_expression_sums, 'introgressed': Vernot_introgressed_expression_sums })\n",
    "Vernot_expression"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 587,
   "id": "fbc1c5d6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.963019571146141 8.623592464716148e-20\n"
     ]
    }
   ],
   "source": [
    "rho, p = spearmanr(Vernot_expression['ancient'], Vernot_expression['archaic-specific'])\n",
    "print(rho,p)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 588,
   "id": "6a00f3e0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.8562253956893983 1.0711542961831134e-10\n"
     ]
    }
   ],
   "source": [
    "rho, p = spearmanr(Vernot_expression['ancient'], Vernot_expression['introgressed'])\n",
    "print(rho,p)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 589,
   "id": "238144f7",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.8808690051661383 6.3617463212295455e-12\n"
     ]
    }
   ],
   "source": [
    "rho, p = spearmanr(Vernot_expression['archaic-specific'], Vernot_expression['introgressed'])\n",
    "print(rho,p)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f8ce7ec2",
   "metadata": {},
   "source": [
    "Now Browning."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 590,
   "id": "b666b506",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "Browning_ancient_expression = expression[expression['annotation'].isin(Browning_ancient_genes_2['gene'])]\n",
    "Browning_ancient_expression = Browning_ancient_expression.loc[:, ~Browning_ancient_expression.columns.isin(['Gene', 'annotation','entropy','rel_entropy','tau'])]\n",
    "Browning_ancient_expression[Browning_ancient_expression < 1] = 0\n",
    "Browning_ancient_expression[Browning_ancient_expression >= 1] = 1\n",
    "Browning_ancient_expression_sums = Browning_ancient_expression.sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 591,
   "id": "d690a9e7",
   "metadata": {},
   "outputs": [],
   "source": [
    "Browning_archaic_specific_expression = expression[expression['annotation'].isin(archaic_specific_genes_2['gene'])]\n",
    "Browning_archaic_specific_expression = Browning_archaic_specific_expression.loc[:, ~Browning_archaic_specific_expression.columns.isin(['Gene', 'annotation','entropy','rel_entropy','tau'])]\n",
    "Browning_archaic_specific_expression[Browning_archaic_specific_expression < 1] = 0\n",
    "Browning_archaic_specific_expression[Browning_archaic_specific_expression >= 1] = 1\n",
    "Browning_archaic_specific_expression_sums = Browning_archaic_specific_expression.sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 592,
   "id": "11937f6c",
   "metadata": {},
   "outputs": [],
   "source": [
    "Browning_introgressed_expression = expression[expression['annotation'].isin(Browning_introgressed_genes_2['gene'])]\n",
    "Browning_introgressed_expression = Browning_introgressed_expression.loc[:, ~Browning_introgressed_expression.columns.isin(['Gene', 'annotation','entropy','rel_entropy','tau'])]\n",
    "Browning_introgressed_expression[Browning_introgressed_expression < 1] = 0\n",
    "Browning_introgressed_expression[Browning_introgressed_expression >= 1] = 1\n",
    "Browning_introgressed_expression_sums = Browning_introgressed_expression.sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 593,
   "id": "ac29e35e",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ancient</th>\n",
       "      <th>archaic-specific</th>\n",
       "      <th>introgressed</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>adipose tissue</th>\n",
       "      <td>1274.0</td>\n",
       "      <td>1352.0</td>\n",
       "      <td>252.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>adrenal gland</th>\n",
       "      <td>1226.0</td>\n",
       "      <td>1264.0</td>\n",
       "      <td>250.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>amygdala</th>\n",
       "      <td>1262.0</td>\n",
       "      <td>1298.0</td>\n",
       "      <td>240.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>basal ganglia</th>\n",
       "      <td>1316.0</td>\n",
       "      <td>1377.0</td>\n",
       "      <td>252.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>breast</th>\n",
       "      <td>1284.0</td>\n",
       "      <td>1367.0</td>\n",
       "      <td>253.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cerebellum</th>\n",
       "      <td>1296.0</td>\n",
       "      <td>1346.0</td>\n",
       "      <td>254.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cerebral cortex</th>\n",
       "      <td>1350.0</td>\n",
       "      <td>1379.0</td>\n",
       "      <td>266.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cervix, uterine</th>\n",
       "      <td>1396.0</td>\n",
       "      <td>1430.0</td>\n",
       "      <td>269.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>colon</th>\n",
       "      <td>1339.0</td>\n",
       "      <td>1398.0</td>\n",
       "      <td>258.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>endometrium</th>\n",
       "      <td>1271.0</td>\n",
       "      <td>1342.0</td>\n",
       "      <td>250.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>esophagus</th>\n",
       "      <td>1231.0</td>\n",
       "      <td>1274.0</td>\n",
       "      <td>241.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>fallopian tube</th>\n",
       "      <td>1346.0</td>\n",
       "      <td>1417.0</td>\n",
       "      <td>262.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>heart muscle</th>\n",
       "      <td>1203.0</td>\n",
       "      <td>1262.0</td>\n",
       "      <td>239.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>hippocampal formation</th>\n",
       "      <td>1295.0</td>\n",
       "      <td>1326.0</td>\n",
       "      <td>250.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>hypothalamus</th>\n",
       "      <td>1350.0</td>\n",
       "      <td>1399.0</td>\n",
       "      <td>266.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>kidney</th>\n",
       "      <td>1265.0</td>\n",
       "      <td>1322.0</td>\n",
       "      <td>250.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>liver</th>\n",
       "      <td>1044.0</td>\n",
       "      <td>1129.0</td>\n",
       "      <td>210.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>lung</th>\n",
       "      <td>1351.0</td>\n",
       "      <td>1400.0</td>\n",
       "      <td>271.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>midbrain</th>\n",
       "      <td>1285.0</td>\n",
       "      <td>1332.0</td>\n",
       "      <td>248.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ovary</th>\n",
       "      <td>1225.0</td>\n",
       "      <td>1309.0</td>\n",
       "      <td>239.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pancreas</th>\n",
       "      <td>1087.0</td>\n",
       "      <td>1138.0</td>\n",
       "      <td>211.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pituitary gland</th>\n",
       "      <td>1410.0</td>\n",
       "      <td>1426.0</td>\n",
       "      <td>278.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>prostate</th>\n",
       "      <td>1329.0</td>\n",
       "      <td>1392.0</td>\n",
       "      <td>253.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>salivary gland</th>\n",
       "      <td>1284.0</td>\n",
       "      <td>1354.0</td>\n",
       "      <td>249.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>skeletal muscle</th>\n",
       "      <td>1058.0</td>\n",
       "      <td>1132.0</td>\n",
       "      <td>204.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>skin</th>\n",
       "      <td>1286.0</td>\n",
       "      <td>1343.0</td>\n",
       "      <td>251.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>small intestine</th>\n",
       "      <td>1304.0</td>\n",
       "      <td>1383.0</td>\n",
       "      <td>260.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>spinal cord</th>\n",
       "      <td>1294.0</td>\n",
       "      <td>1355.0</td>\n",
       "      <td>265.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>spleen</th>\n",
       "      <td>1229.0</td>\n",
       "      <td>1295.0</td>\n",
       "      <td>245.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>stomach</th>\n",
       "      <td>1239.0</td>\n",
       "      <td>1299.0</td>\n",
       "      <td>242.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>testis</th>\n",
       "      <td>1478.0</td>\n",
       "      <td>1539.0</td>\n",
       "      <td>287.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>thyroid gland</th>\n",
       "      <td>1336.0</td>\n",
       "      <td>1382.0</td>\n",
       "      <td>266.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>urinary bladder</th>\n",
       "      <td>1306.0</td>\n",
       "      <td>1366.0</td>\n",
       "      <td>252.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>vagina</th>\n",
       "      <td>1323.0</td>\n",
       "      <td>1372.0</td>\n",
       "      <td>258.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                       ancient  archaic-specific  introgressed\n",
       "adipose tissue          1274.0            1352.0         252.0\n",
       "adrenal gland           1226.0            1264.0         250.0\n",
       "amygdala                1262.0            1298.0         240.0\n",
       "basal ganglia           1316.0            1377.0         252.0\n",
       "breast                  1284.0            1367.0         253.0\n",
       "cerebellum              1296.0            1346.0         254.0\n",
       "cerebral cortex         1350.0            1379.0         266.0\n",
       "cervix, uterine         1396.0            1430.0         269.0\n",
       "colon                   1339.0            1398.0         258.0\n",
       "endometrium             1271.0            1342.0         250.0\n",
       "esophagus               1231.0            1274.0         241.0\n",
       "fallopian tube          1346.0            1417.0         262.0\n",
       "heart muscle            1203.0            1262.0         239.0\n",
       "hippocampal formation   1295.0            1326.0         250.0\n",
       "hypothalamus            1350.0            1399.0         266.0\n",
       "kidney                  1265.0            1322.0         250.0\n",
       "liver                   1044.0            1129.0         210.0\n",
       "lung                    1351.0            1400.0         271.0\n",
       "midbrain                1285.0            1332.0         248.0\n",
       "ovary                   1225.0            1309.0         239.0\n",
       "pancreas                1087.0            1138.0         211.0\n",
       "pituitary gland         1410.0            1426.0         278.0\n",
       "prostate                1329.0            1392.0         253.0\n",
       "salivary gland          1284.0            1354.0         249.0\n",
       "skeletal muscle         1058.0            1132.0         204.0\n",
       "skin                    1286.0            1343.0         251.0\n",
       "small intestine         1304.0            1383.0         260.0\n",
       "spinal cord             1294.0            1355.0         265.0\n",
       "spleen                  1229.0            1295.0         245.0\n",
       "stomach                 1239.0            1299.0         242.0\n",
       "testis                  1478.0            1539.0         287.0\n",
       "thyroid gland           1336.0            1382.0         266.0\n",
       "urinary bladder         1306.0            1366.0         252.0\n",
       "vagina                  1323.0            1372.0         258.0"
      ]
     },
     "execution_count": 593,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Browning_expression = pd.DataFrame({ 'ancient': Browning_ancient_expression_sums, 'archaic-specific': Browning_archaic_specific_expression_sums, 'introgressed': Browning_introgressed_expression_sums })\n",
    "Browning_expression"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 594,
   "id": "9e8a5be3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.9588936542275524 4.547848246251568e-19\n"
     ]
    }
   ],
   "source": [
    "rho, p = spearmanr(Browning_expression['ancient'], Browning_expression['archaic-specific'])\n",
    "print(rho,p)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 595,
   "id": "b1c0c99b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.9357169147823144 4.920930114343845e-16\n"
     ]
    }
   ],
   "source": [
    "rho, p = spearmanr(Browning_expression['ancient'], Browning_expression['introgressed'])\n",
    "print(rho,p)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 596,
   "id": "a78922fe",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.9306003291060468 1.6145461837822605e-15\n"
     ]
    }
   ],
   "source": [
    "rho, p = spearmanr(Browning_expression['archaic-specific'], Browning_expression['introgressed'])\n",
    "print(rho,p)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "934314b8",
   "metadata": {},
   "source": [
    "# Genes of Evolutionary Significance <a class = 'anchor' id = 'evolutionarysignificantgenes'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5cd6412a",
   "metadata": {},
   "source": [
    "Let's look at SAVs among a few genes of evolutionary significance."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 597,
   "id": "56c144a2",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref_allele</th>\n",
       "      <th>alt_allele</th>\n",
       "      <th>ancestral_allele</th>\n",
       "      <th>anc_dev</th>\n",
       "      <th>variant_type</th>\n",
       "      <th>altai_gt</th>\n",
       "      <th>chagyrskaya_gt</th>\n",
       "      <th>denisovan_gt</th>\n",
       "      <th>vindija_gt</th>\n",
       "      <th>altai_gt_boolean</th>\n",
       "      <th>chagyrskaya_gt_boolean</th>\n",
       "      <th>denisovan_gt_boolean</th>\n",
       "      <th>vindija_gt_boolean</th>\n",
       "      <th>distribution</th>\n",
       "      <th>present_in_1KG</th>\n",
       "      <th>1KG_allele_count</th>\n",
       "      <th>1KG_allele_number</th>\n",
       "      <th>1KG_allele_frequency</th>\n",
       "      <th>1KG_EAS_AF</th>\n",
       "      <th>1KG_EUR_AF</th>\n",
       "      <th>1KG_AFR_AF</th>\n",
       "      <th>1KG_AMR_AF</th>\n",
       "      <th>1KG_SAS_AF</th>\n",
       "      <th>1KG_non_ASW_AFR_AF</th>\n",
       "      <th>present_in_gnomAD</th>\n",
       "      <th>gnomAD_allele_count</th>\n",
       "      <th>gnomAD_allele_number</th>\n",
       "      <th>gnomAD_allele_frequency</th>\n",
       "      <th>Vernot_introgressed</th>\n",
       "      <th>Vernot_ancestral_allele</th>\n",
       "      <th>Vernot_derived_allele</th>\n",
       "      <th>Vernot_ancestral_derived_code</th>\n",
       "      <th>Vernot_AFA_AF</th>\n",
       "      <th>Vernot_AFR_AF</th>\n",
       "      <th>Vernot_AMR_AF</th>\n",
       "      <th>Vernot_EAS_AF</th>\n",
       "      <th>Vernot_EUR_AF</th>\n",
       "      <th>Vernot_PNG_AF</th>\n",
       "      <th>Vernot_SAS_AF</th>\n",
       "      <th>Vernot_Denisovan_base</th>\n",
       "      <th>Vernot_Neanderthal_base_1</th>\n",
       "      <th>Vernot_Neanderthal_base_2</th>\n",
       "      <th>Vernot_haplotype_tag</th>\n",
       "      <th>Vernot_introgressed_AF</th>\n",
       "      <th>Browning_introgressed</th>\n",
       "      <th>Browning_ref_alt</th>\n",
       "      <th>Browning_introgressed_AF</th>\n",
       "      <th>annotation</th>\n",
       "      <th>mis_oe</th>\n",
       "      <th>mis_z</th>\n",
       "      <th>lof_oe</th>\n",
       "      <th>lof_z</th>\n",
       "      <th>phyloP</th>\n",
       "      <th>ag_delta</th>\n",
       "      <th>al_delta</th>\n",
       "      <th>dg_delta</th>\n",
       "      <th>dl_delta</th>\n",
       "      <th>delta_max</th>\n",
       "      <th>ag_pos</th>\n",
       "      <th>al_pos</th>\n",
       "      <th>dg_pos</th>\n",
       "      <th>dl_pos</th>\n",
       "      <th>Adipose_Subcutaneous</th>\n",
       "      <th>Adipose_Visceral_Omentum</th>\n",
       "      <th>Adrenal_Gland</th>\n",
       "      <th>Artery_Aorta</th>\n",
       "      <th>Artery_Coronary</th>\n",
       "      <th>Artery_Tibial</th>\n",
       "      <th>Brain_Amygdala</th>\n",
       "      <th>Brain_Anterior_cingulate_cortex_BA24</th>\n",
       "      <th>Brain_Caudate_basal_ganglia</th>\n",
       "      <th>Brain_Cerebellar_Hemisphere</th>\n",
       "      <th>Brain_Cerebellum</th>\n",
       "      <th>Brain_Cortex</th>\n",
       "      <th>Brain_Frontal_Cortex_BA9</th>\n",
       "      <th>Brain_Hippocampus</th>\n",
       "      <th>Brain_Hypothalamus</th>\n",
       "      <th>Brain_Nucleus_accumbens_basal_ganglia</th>\n",
       "      <th>Brain_Putamen_basal_ganglia</th>\n",
       "      <th>Brain_Spinal_cord_cervical_c-1</th>\n",
       "      <th>Brain_Substantia_nigra</th>\n",
       "      <th>Breast_Mammary_Tissue</th>\n",
       "      <th>Cells_Cultured_fibroblasts</th>\n",
       "      <th>Cells_EBV-transformed_lymphocytes</th>\n",
       "      <th>Colon_Sigmoid</th>\n",
       "      <th>Colon_Transverse</th>\n",
       "      <th>Esophagus_Gastroesophageal_Junction</th>\n",
       "      <th>Esophagus_Mucosa</th>\n",
       "      <th>Esophagus_Muscularis</th>\n",
       "      <th>Heart_Atrial_Appendage</th>\n",
       "      <th>Heart_Left_Ventricle</th>\n",
       "      <th>Kidney_Cortex</th>\n",
       "      <th>Liver</th>\n",
       "      <th>Lung</th>\n",
       "      <th>Minor_Salivary_Gland</th>\n",
       "      <th>Muscle_Skeletal</th>\n",
       "      <th>Nerve_Tibial</th>\n",
       "      <th>Ovary</th>\n",
       "      <th>Pancreas</th>\n",
       "      <th>Pituitary</th>\n",
       "      <th>Prostate</th>\n",
       "      <th>Skin_Not_Sun_Exposed_Suprapubic</th>\n",
       "      <th>Skin_Sun_Exposed_Lower_leg</th>\n",
       "      <th>Small_Intestine_Terminal_Ileum</th>\n",
       "      <th>Spleen</th>\n",
       "      <th>Stomach</th>\n",
       "      <th>Testis</th>\n",
       "      <th>Thyroid</th>\n",
       "      <th>Uterus</th>\n",
       "      <th>Vagina</th>\n",
       "      <th>Whole_Blood</th>\n",
       "      <th>N_GTEx_tissues</th>\n",
       "      <th>sQTL</th>\n",
       "      <th>Vernot_allele_origin</th>\n",
       "      <th>Browning_allele_origin</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>385314</th>\n",
       "      <td>chr12</td>\n",
       "      <td>113355275</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>yes</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>12.0</td>\n",
       "      <td>143230.0</td>\n",
       "      <td>0.000084</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>OAS1</td>\n",
       "      <td>1.0792</td>\n",
       "      <td>-0.42591</td>\n",
       "      <td>0.54623</td>\n",
       "      <td>1.7991</td>\n",
       "      <td>-0.137</td>\n",
       "      <td>0.26</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.26</td>\n",
       "      <td>10</td>\n",
       "      <td>1</td>\n",
       "      <td>10</td>\n",
       "      <td>-29</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        chrom        pos ref_allele alt_allele ancestral_allele  anc_dev  \\\n",
       "385314  chr12  113355275          G          T                G  derived   \n",
       "\n",
       "       variant_type altai_gt chagyrskaya_gt denisovan_gt vindija_gt  \\\n",
       "385314          snv      0/0            0/0          1/1        0/0   \n",
       "\n",
       "        altai_gt_boolean  chagyrskaya_gt_boolean  denisovan_gt_boolean  \\\n",
       "385314             False                   False                  True   \n",
       "\n",
       "        vindija_gt_boolean distribution present_in_1KG  1KG_allele_count  \\\n",
       "385314               False    Denisovan            yes               1.0   \n",
       "\n",
       "        1KG_allele_number  1KG_allele_frequency  1KG_EAS_AF  1KG_EUR_AF  \\\n",
       "385314             5096.0                   0.0         0.0         0.0   \n",
       "\n",
       "        1KG_AFR_AF  1KG_AMR_AF  1KG_SAS_AF  1KG_non_ASW_AFR_AF  \\\n",
       "385314         0.0         0.0         0.0                 0.0   \n",
       "\n",
       "       present_in_gnomAD  gnomAD_allele_count  gnomAD_allele_number  \\\n",
       "385314               yes                 12.0              143230.0   \n",
       "\n",
       "        gnomAD_allele_frequency Vernot_introgressed Vernot_ancestral_allele  \\\n",
       "385314                 0.000084                  no                     NaN   \n",
       "\n",
       "       Vernot_derived_allele  Vernot_ancestral_derived_code  Vernot_AFA_AF  \\\n",
       "385314                   NaN                            NaN            NaN   \n",
       "\n",
       "        Vernot_AFR_AF  Vernot_AMR_AF  Vernot_EAS_AF  Vernot_EUR_AF  \\\n",
       "385314            NaN            NaN            NaN            NaN   \n",
       "\n",
       "        Vernot_PNG_AF  Vernot_SAS_AF Vernot_Denisovan_base  \\\n",
       "385314            NaN            NaN                   NaN   \n",
       "\n",
       "       Vernot_Neanderthal_base_1 Vernot_Neanderthal_base_2  \\\n",
       "385314                       NaN                       NaN   \n",
       "\n",
       "       Vernot_haplotype_tag  Vernot_introgressed_AF Browning_introgressed  \\\n",
       "385314                  NaN                     NaN                    no   \n",
       "\n",
       "        Browning_ref_alt  Browning_introgressed_AF annotation  mis_oe  \\\n",
       "385314               NaN                       NaN       OAS1  1.0792   \n",
       "\n",
       "          mis_z   lof_oe   lof_z  phyloP  ag_delta  al_delta  dg_delta  \\\n",
       "385314 -0.42591  0.54623  1.7991  -0.137      0.26       0.0       0.0   \n",
       "\n",
       "        dl_delta  delta_max  ag_pos  al_pos  dg_pos  dl_pos  \\\n",
       "385314       0.0       0.26      10       1      10     -29   \n",
       "\n",
       "        Adipose_Subcutaneous  Adipose_Visceral_Omentum  Adrenal_Gland  \\\n",
       "385314                   NaN                       NaN            NaN   \n",
       "\n",
       "        Artery_Aorta  Artery_Coronary  Artery_Tibial  Brain_Amygdala  \\\n",
       "385314           NaN              NaN            NaN             NaN   \n",
       "\n",
       "        Brain_Anterior_cingulate_cortex_BA24  Brain_Caudate_basal_ganglia  \\\n",
       "385314                                   NaN                          NaN   \n",
       "\n",
       "        Brain_Cerebellar_Hemisphere  Brain_Cerebellum  Brain_Cortex  \\\n",
       "385314                          NaN               NaN           NaN   \n",
       "\n",
       "        Brain_Frontal_Cortex_BA9  Brain_Hippocampus  Brain_Hypothalamus  \\\n",
       "385314                       NaN                NaN                 NaN   \n",
       "\n",
       "        Brain_Nucleus_accumbens_basal_ganglia  Brain_Putamen_basal_ganglia  \\\n",
       "385314                                    NaN                          NaN   \n",
       "\n",
       "        Brain_Spinal_cord_cervical_c-1  Brain_Substantia_nigra  \\\n",
       "385314                             NaN                     NaN   \n",
       "\n",
       "        Breast_Mammary_Tissue  Cells_Cultured_fibroblasts  \\\n",
       "385314                    NaN                         NaN   \n",
       "\n",
       "        Cells_EBV-transformed_lymphocytes  Colon_Sigmoid  Colon_Transverse  \\\n",
       "385314                                NaN            NaN               NaN   \n",
       "\n",
       "        Esophagus_Gastroesophageal_Junction  Esophagus_Mucosa  \\\n",
       "385314                                  NaN               NaN   \n",
       "\n",
       "        Esophagus_Muscularis  Heart_Atrial_Appendage  Heart_Left_Ventricle  \\\n",
       "385314                   NaN                     NaN                   NaN   \n",
       "\n",
       "        Kidney_Cortex  Liver  Lung  Minor_Salivary_Gland  Muscle_Skeletal  \\\n",
       "385314            NaN    NaN   NaN                   NaN              NaN   \n",
       "\n",
       "        Nerve_Tibial  Ovary  Pancreas  Pituitary  Prostate  \\\n",
       "385314           NaN    NaN       NaN        NaN       NaN   \n",
       "\n",
       "        Skin_Not_Sun_Exposed_Suprapubic  Skin_Sun_Exposed_Lower_leg  \\\n",
       "385314                              NaN                         NaN   \n",
       "\n",
       "        Small_Intestine_Terminal_Ileum  Spleen  Stomach  Testis  Thyroid  \\\n",
       "385314                             NaN     NaN      NaN     NaN      NaN   \n",
       "\n",
       "        Uterus  Vagina  Whole_Blood  N_GTEx_tissues sQTL  \\\n",
       "385314     NaN     NaN          NaN             NaN   no   \n",
       "\n",
       "          Vernot_allele_origin  Browning_allele_origin  \n",
       "385314  low-confidence ancient  low-confidence ancient  "
      ]
     },
     "execution_count": 597,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_2[data_2['annotation']=='OAS1']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 598,
   "id": "f95ddb1f",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref_allele</th>\n",
       "      <th>alt_allele</th>\n",
       "      <th>ancestral_allele</th>\n",
       "      <th>anc_dev</th>\n",
       "      <th>variant_type</th>\n",
       "      <th>altai_gt</th>\n",
       "      <th>chagyrskaya_gt</th>\n",
       "      <th>denisovan_gt</th>\n",
       "      <th>vindija_gt</th>\n",
       "      <th>altai_gt_boolean</th>\n",
       "      <th>chagyrskaya_gt_boolean</th>\n",
       "      <th>denisovan_gt_boolean</th>\n",
       "      <th>vindija_gt_boolean</th>\n",
       "      <th>distribution</th>\n",
       "      <th>present_in_1KG</th>\n",
       "      <th>1KG_allele_count</th>\n",
       "      <th>1KG_allele_number</th>\n",
       "      <th>1KG_allele_frequency</th>\n",
       "      <th>1KG_EAS_AF</th>\n",
       "      <th>1KG_EUR_AF</th>\n",
       "      <th>1KG_AFR_AF</th>\n",
       "      <th>1KG_AMR_AF</th>\n",
       "      <th>1KG_SAS_AF</th>\n",
       "      <th>1KG_non_ASW_AFR_AF</th>\n",
       "      <th>present_in_gnomAD</th>\n",
       "      <th>gnomAD_allele_count</th>\n",
       "      <th>gnomAD_allele_number</th>\n",
       "      <th>gnomAD_allele_frequency</th>\n",
       "      <th>Vernot_introgressed</th>\n",
       "      <th>Vernot_ancestral_allele</th>\n",
       "      <th>Vernot_derived_allele</th>\n",
       "      <th>Vernot_ancestral_derived_code</th>\n",
       "      <th>Vernot_AFA_AF</th>\n",
       "      <th>Vernot_AFR_AF</th>\n",
       "      <th>Vernot_AMR_AF</th>\n",
       "      <th>Vernot_EAS_AF</th>\n",
       "      <th>Vernot_EUR_AF</th>\n",
       "      <th>Vernot_PNG_AF</th>\n",
       "      <th>Vernot_SAS_AF</th>\n",
       "      <th>Vernot_Denisovan_base</th>\n",
       "      <th>Vernot_Neanderthal_base_1</th>\n",
       "      <th>Vernot_Neanderthal_base_2</th>\n",
       "      <th>Vernot_haplotype_tag</th>\n",
       "      <th>Vernot_introgressed_AF</th>\n",
       "      <th>Browning_introgressed</th>\n",
       "      <th>Browning_ref_alt</th>\n",
       "      <th>Browning_introgressed_AF</th>\n",
       "      <th>annotation</th>\n",
       "      <th>mis_oe</th>\n",
       "      <th>mis_z</th>\n",
       "      <th>lof_oe</th>\n",
       "      <th>lof_z</th>\n",
       "      <th>phyloP</th>\n",
       "      <th>ag_delta</th>\n",
       "      <th>al_delta</th>\n",
       "      <th>dg_delta</th>\n",
       "      <th>dl_delta</th>\n",
       "      <th>delta_max</th>\n",
       "      <th>ag_pos</th>\n",
       "      <th>al_pos</th>\n",
       "      <th>dg_pos</th>\n",
       "      <th>dl_pos</th>\n",
       "      <th>Adipose_Subcutaneous</th>\n",
       "      <th>Adipose_Visceral_Omentum</th>\n",
       "      <th>Adrenal_Gland</th>\n",
       "      <th>Artery_Aorta</th>\n",
       "      <th>Artery_Coronary</th>\n",
       "      <th>Artery_Tibial</th>\n",
       "      <th>Brain_Amygdala</th>\n",
       "      <th>Brain_Anterior_cingulate_cortex_BA24</th>\n",
       "      <th>Brain_Caudate_basal_ganglia</th>\n",
       "      <th>Brain_Cerebellar_Hemisphere</th>\n",
       "      <th>Brain_Cerebellum</th>\n",
       "      <th>Brain_Cortex</th>\n",
       "      <th>Brain_Frontal_Cortex_BA9</th>\n",
       "      <th>Brain_Hippocampus</th>\n",
       "      <th>Brain_Hypothalamus</th>\n",
       "      <th>Brain_Nucleus_accumbens_basal_ganglia</th>\n",
       "      <th>Brain_Putamen_basal_ganglia</th>\n",
       "      <th>Brain_Spinal_cord_cervical_c-1</th>\n",
       "      <th>Brain_Substantia_nigra</th>\n",
       "      <th>Breast_Mammary_Tissue</th>\n",
       "      <th>Cells_Cultured_fibroblasts</th>\n",
       "      <th>Cells_EBV-transformed_lymphocytes</th>\n",
       "      <th>Colon_Sigmoid</th>\n",
       "      <th>Colon_Transverse</th>\n",
       "      <th>Esophagus_Gastroesophageal_Junction</th>\n",
       "      <th>Esophagus_Mucosa</th>\n",
       "      <th>Esophagus_Muscularis</th>\n",
       "      <th>Heart_Atrial_Appendage</th>\n",
       "      <th>Heart_Left_Ventricle</th>\n",
       "      <th>Kidney_Cortex</th>\n",
       "      <th>Liver</th>\n",
       "      <th>Lung</th>\n",
       "      <th>Minor_Salivary_Gland</th>\n",
       "      <th>Muscle_Skeletal</th>\n",
       "      <th>Nerve_Tibial</th>\n",
       "      <th>Ovary</th>\n",
       "      <th>Pancreas</th>\n",
       "      <th>Pituitary</th>\n",
       "      <th>Prostate</th>\n",
       "      <th>Skin_Not_Sun_Exposed_Suprapubic</th>\n",
       "      <th>Skin_Sun_Exposed_Lower_leg</th>\n",
       "      <th>Small_Intestine_Terminal_Ileum</th>\n",
       "      <th>Spleen</th>\n",
       "      <th>Stomach</th>\n",
       "      <th>Testis</th>\n",
       "      <th>Thyroid</th>\n",
       "      <th>Uterus</th>\n",
       "      <th>Vagina</th>\n",
       "      <th>Whole_Blood</th>\n",
       "      <th>N_GTEx_tissues</th>\n",
       "      <th>sQTL</th>\n",
       "      <th>Vernot_allele_origin</th>\n",
       "      <th>Browning_allele_origin</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>784210</th>\n",
       "      <td>chr2</td>\n",
       "      <td>46584859</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>yes</td>\n",
       "      <td>12.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>21.0</td>\n",
       "      <td>143324.0</td>\n",
       "      <td>0.000147</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>EPAS1</td>\n",
       "      <td>1.0198</td>\n",
       "      <td>-0.15408</td>\n",
       "      <td>0.21127</td>\n",
       "      <td>4.4975</td>\n",
       "      <td>-0.450</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.37</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.37</td>\n",
       "      <td>0</td>\n",
       "      <td>32</td>\n",
       "      <td>0</td>\n",
       "      <td>-24</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>784257</th>\n",
       "      <td>chr2</td>\n",
       "      <td>46610904</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>Altai</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>EPAS1</td>\n",
       "      <td>1.0198</td>\n",
       "      <td>-0.15408</td>\n",
       "      <td>0.21127</td>\n",
       "      <td>4.4975</td>\n",
       "      <td>-0.313</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.23</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.23</td>\n",
       "      <td>5</td>\n",
       "      <td>28</td>\n",
       "      <td>-18</td>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       chrom       pos ref_allele alt_allele ancestral_allele  anc_dev  \\\n",
       "784210  chr2  46584859          A          G                A  derived   \n",
       "784257  chr2  46610904          A          G                A  derived   \n",
       "\n",
       "       variant_type altai_gt chagyrskaya_gt denisovan_gt vindija_gt  \\\n",
       "784210          snv      0/0            0/0          1/1        0/0   \n",
       "784257          snv      0/1            0/0          0/0        0/0   \n",
       "\n",
       "        altai_gt_boolean  chagyrskaya_gt_boolean  denisovan_gt_boolean  \\\n",
       "784210             False                   False                  True   \n",
       "784257              True                   False                 False   \n",
       "\n",
       "        vindija_gt_boolean distribution present_in_1KG  1KG_allele_count  \\\n",
       "784210               False    Denisovan            yes              12.0   \n",
       "784257               False        Altai             no               NaN   \n",
       "\n",
       "        1KG_allele_number  1KG_allele_frequency  1KG_EAS_AF  1KG_EUR_AF  \\\n",
       "784210             5096.0                   0.0        0.01         0.0   \n",
       "784257                NaN                   NaN         NaN         NaN   \n",
       "\n",
       "        1KG_AFR_AF  1KG_AMR_AF  1KG_SAS_AF  1KG_non_ASW_AFR_AF  \\\n",
       "784210         0.0         0.0         0.0                 0.0   \n",
       "784257         NaN         NaN         NaN                 NaN   \n",
       "\n",
       "       present_in_gnomAD  gnomAD_allele_count  gnomAD_allele_number  \\\n",
       "784210               yes                 21.0              143324.0   \n",
       "784257                no                  NaN                   NaN   \n",
       "\n",
       "        gnomAD_allele_frequency Vernot_introgressed Vernot_ancestral_allele  \\\n",
       "784210                 0.000147                  no                     NaN   \n",
       "784257                      NaN                  no                     NaN   \n",
       "\n",
       "       Vernot_derived_allele  Vernot_ancestral_derived_code  Vernot_AFA_AF  \\\n",
       "784210                   NaN                            NaN            NaN   \n",
       "784257                   NaN                            NaN            NaN   \n",
       "\n",
       "        Vernot_AFR_AF  Vernot_AMR_AF  Vernot_EAS_AF  Vernot_EUR_AF  \\\n",
       "784210            NaN            NaN            NaN            NaN   \n",
       "784257            NaN            NaN            NaN            NaN   \n",
       "\n",
       "        Vernot_PNG_AF  Vernot_SAS_AF Vernot_Denisovan_base  \\\n",
       "784210            NaN            NaN                   NaN   \n",
       "784257            NaN            NaN                   NaN   \n",
       "\n",
       "       Vernot_Neanderthal_base_1 Vernot_Neanderthal_base_2  \\\n",
       "784210                       NaN                       NaN   \n",
       "784257                       NaN                       NaN   \n",
       "\n",
       "       Vernot_haplotype_tag  Vernot_introgressed_AF Browning_introgressed  \\\n",
       "784210                  NaN                     NaN                    no   \n",
       "784257                  NaN                     NaN                    no   \n",
       "\n",
       "        Browning_ref_alt  Browning_introgressed_AF annotation  mis_oe  \\\n",
       "784210               NaN                       NaN      EPAS1  1.0198   \n",
       "784257               NaN                       NaN      EPAS1  1.0198   \n",
       "\n",
       "          mis_z   lof_oe   lof_z  phyloP  ag_delta  al_delta  dg_delta  \\\n",
       "784210 -0.15408  0.21127  4.4975  -0.450       0.0      0.00      0.37   \n",
       "784257 -0.15408  0.21127  4.4975  -0.313       0.0      0.23      0.00   \n",
       "\n",
       "        dl_delta  delta_max  ag_pos  al_pos  dg_pos  dl_pos  \\\n",
       "784210       0.0       0.37       0      32       0     -24   \n",
       "784257       0.0       0.23       5      28     -18       4   \n",
       "\n",
       "        Adipose_Subcutaneous  Adipose_Visceral_Omentum  Adrenal_Gland  \\\n",
       "784210                   NaN                       NaN            NaN   \n",
       "784257                   NaN                       NaN            NaN   \n",
       "\n",
       "        Artery_Aorta  Artery_Coronary  Artery_Tibial  Brain_Amygdala  \\\n",
       "784210           NaN              NaN            NaN             NaN   \n",
       "784257           NaN              NaN            NaN             NaN   \n",
       "\n",
       "        Brain_Anterior_cingulate_cortex_BA24  Brain_Caudate_basal_ganglia  \\\n",
       "784210                                   NaN                          NaN   \n",
       "784257                                   NaN                          NaN   \n",
       "\n",
       "        Brain_Cerebellar_Hemisphere  Brain_Cerebellum  Brain_Cortex  \\\n",
       "784210                          NaN               NaN           NaN   \n",
       "784257                          NaN               NaN           NaN   \n",
       "\n",
       "        Brain_Frontal_Cortex_BA9  Brain_Hippocampus  Brain_Hypothalamus  \\\n",
       "784210                       NaN                NaN                 NaN   \n",
       "784257                       NaN                NaN                 NaN   \n",
       "\n",
       "        Brain_Nucleus_accumbens_basal_ganglia  Brain_Putamen_basal_ganglia  \\\n",
       "784210                                    NaN                          NaN   \n",
       "784257                                    NaN                          NaN   \n",
       "\n",
       "        Brain_Spinal_cord_cervical_c-1  Brain_Substantia_nigra  \\\n",
       "784210                             NaN                     NaN   \n",
       "784257                             NaN                     NaN   \n",
       "\n",
       "        Breast_Mammary_Tissue  Cells_Cultured_fibroblasts  \\\n",
       "784210                    NaN                         NaN   \n",
       "784257                    NaN                         NaN   \n",
       "\n",
       "        Cells_EBV-transformed_lymphocytes  Colon_Sigmoid  Colon_Transverse  \\\n",
       "784210                                NaN            NaN               NaN   \n",
       "784257                                NaN            NaN               NaN   \n",
       "\n",
       "        Esophagus_Gastroesophageal_Junction  Esophagus_Mucosa  \\\n",
       "784210                                  NaN               NaN   \n",
       "784257                                  NaN               NaN   \n",
       "\n",
       "        Esophagus_Muscularis  Heart_Atrial_Appendage  Heart_Left_Ventricle  \\\n",
       "784210                   NaN                     NaN                   NaN   \n",
       "784257                   NaN                     NaN                   NaN   \n",
       "\n",
       "        Kidney_Cortex  Liver  Lung  Minor_Salivary_Gland  Muscle_Skeletal  \\\n",
       "784210            NaN    NaN   NaN                   NaN              NaN   \n",
       "784257            NaN    NaN   NaN                   NaN              NaN   \n",
       "\n",
       "        Nerve_Tibial  Ovary  Pancreas  Pituitary  Prostate  \\\n",
       "784210           NaN    NaN       NaN        NaN       NaN   \n",
       "784257           NaN    NaN       NaN        NaN       NaN   \n",
       "\n",
       "        Skin_Not_Sun_Exposed_Suprapubic  Skin_Sun_Exposed_Lower_leg  \\\n",
       "784210                              NaN                         NaN   \n",
       "784257                              NaN                         NaN   \n",
       "\n",
       "        Small_Intestine_Terminal_Ileum  Spleen  Stomach  Testis  Thyroid  \\\n",
       "784210                             NaN     NaN      NaN     NaN      NaN   \n",
       "784257                             NaN     NaN      NaN     NaN      NaN   \n",
       "\n",
       "        Uterus  Vagina  Whole_Blood  N_GTEx_tissues sQTL  \\\n",
       "784210     NaN     NaN          NaN             NaN   no   \n",
       "784257     NaN     NaN          NaN             NaN   no   \n",
       "\n",
       "          Vernot_allele_origin  Browning_allele_origin  \n",
       "784210  low-confidence ancient  low-confidence ancient  \n",
       "784257        archaic-specific        archaic-specific  "
      ]
     },
     "execution_count": 598,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_2[data_2['annotation']=='EPAS1']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 599,
   "id": "6e4c5a2f",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref_allele</th>\n",
       "      <th>alt_allele</th>\n",
       "      <th>ancestral_allele</th>\n",
       "      <th>anc_dev</th>\n",
       "      <th>variant_type</th>\n",
       "      <th>altai_gt</th>\n",
       "      <th>chagyrskaya_gt</th>\n",
       "      <th>denisovan_gt</th>\n",
       "      <th>vindija_gt</th>\n",
       "      <th>altai_gt_boolean</th>\n",
       "      <th>chagyrskaya_gt_boolean</th>\n",
       "      <th>denisovan_gt_boolean</th>\n",
       "      <th>vindija_gt_boolean</th>\n",
       "      <th>distribution</th>\n",
       "      <th>present_in_1KG</th>\n",
       "      <th>1KG_allele_count</th>\n",
       "      <th>1KG_allele_number</th>\n",
       "      <th>1KG_allele_frequency</th>\n",
       "      <th>1KG_EAS_AF</th>\n",
       "      <th>1KG_EUR_AF</th>\n",
       "      <th>1KG_AFR_AF</th>\n",
       "      <th>1KG_AMR_AF</th>\n",
       "      <th>1KG_SAS_AF</th>\n",
       "      <th>1KG_non_ASW_AFR_AF</th>\n",
       "      <th>present_in_gnomAD</th>\n",
       "      <th>gnomAD_allele_count</th>\n",
       "      <th>gnomAD_allele_number</th>\n",
       "      <th>gnomAD_allele_frequency</th>\n",
       "      <th>Vernot_introgressed</th>\n",
       "      <th>Vernot_ancestral_allele</th>\n",
       "      <th>Vernot_derived_allele</th>\n",
       "      <th>Vernot_ancestral_derived_code</th>\n",
       "      <th>Vernot_AFA_AF</th>\n",
       "      <th>Vernot_AFR_AF</th>\n",
       "      <th>Vernot_AMR_AF</th>\n",
       "      <th>Vernot_EAS_AF</th>\n",
       "      <th>Vernot_EUR_AF</th>\n",
       "      <th>Vernot_PNG_AF</th>\n",
       "      <th>Vernot_SAS_AF</th>\n",
       "      <th>Vernot_Denisovan_base</th>\n",
       "      <th>Vernot_Neanderthal_base_1</th>\n",
       "      <th>Vernot_Neanderthal_base_2</th>\n",
       "      <th>Vernot_haplotype_tag</th>\n",
       "      <th>Vernot_introgressed_AF</th>\n",
       "      <th>Browning_introgressed</th>\n",
       "      <th>Browning_ref_alt</th>\n",
       "      <th>Browning_introgressed_AF</th>\n",
       "      <th>annotation</th>\n",
       "      <th>mis_oe</th>\n",
       "      <th>mis_z</th>\n",
       "      <th>lof_oe</th>\n",
       "      <th>lof_z</th>\n",
       "      <th>phyloP</th>\n",
       "      <th>ag_delta</th>\n",
       "      <th>al_delta</th>\n",
       "      <th>dg_delta</th>\n",
       "      <th>dl_delta</th>\n",
       "      <th>delta_max</th>\n",
       "      <th>ag_pos</th>\n",
       "      <th>al_pos</th>\n",
       "      <th>dg_pos</th>\n",
       "      <th>dl_pos</th>\n",
       "      <th>Adipose_Subcutaneous</th>\n",
       "      <th>Adipose_Visceral_Omentum</th>\n",
       "      <th>Adrenal_Gland</th>\n",
       "      <th>Artery_Aorta</th>\n",
       "      <th>Artery_Coronary</th>\n",
       "      <th>Artery_Tibial</th>\n",
       "      <th>Brain_Amygdala</th>\n",
       "      <th>Brain_Anterior_cingulate_cortex_BA24</th>\n",
       "      <th>Brain_Caudate_basal_ganglia</th>\n",
       "      <th>Brain_Cerebellar_Hemisphere</th>\n",
       "      <th>Brain_Cerebellum</th>\n",
       "      <th>Brain_Cortex</th>\n",
       "      <th>Brain_Frontal_Cortex_BA9</th>\n",
       "      <th>Brain_Hippocampus</th>\n",
       "      <th>Brain_Hypothalamus</th>\n",
       "      <th>Brain_Nucleus_accumbens_basal_ganglia</th>\n",
       "      <th>Brain_Putamen_basal_ganglia</th>\n",
       "      <th>Brain_Spinal_cord_cervical_c-1</th>\n",
       "      <th>Brain_Substantia_nigra</th>\n",
       "      <th>Breast_Mammary_Tissue</th>\n",
       "      <th>Cells_Cultured_fibroblasts</th>\n",
       "      <th>Cells_EBV-transformed_lymphocytes</th>\n",
       "      <th>Colon_Sigmoid</th>\n",
       "      <th>Colon_Transverse</th>\n",
       "      <th>Esophagus_Gastroesophageal_Junction</th>\n",
       "      <th>Esophagus_Mucosa</th>\n",
       "      <th>Esophagus_Muscularis</th>\n",
       "      <th>Heart_Atrial_Appendage</th>\n",
       "      <th>Heart_Left_Ventricle</th>\n",
       "      <th>Kidney_Cortex</th>\n",
       "      <th>Liver</th>\n",
       "      <th>Lung</th>\n",
       "      <th>Minor_Salivary_Gland</th>\n",
       "      <th>Muscle_Skeletal</th>\n",
       "      <th>Nerve_Tibial</th>\n",
       "      <th>Ovary</th>\n",
       "      <th>Pancreas</th>\n",
       "      <th>Pituitary</th>\n",
       "      <th>Prostate</th>\n",
       "      <th>Skin_Not_Sun_Exposed_Suprapubic</th>\n",
       "      <th>Skin_Sun_Exposed_Lower_leg</th>\n",
       "      <th>Small_Intestine_Terminal_Ileum</th>\n",
       "      <th>Spleen</th>\n",
       "      <th>Stomach</th>\n",
       "      <th>Testis</th>\n",
       "      <th>Thyroid</th>\n",
       "      <th>Uterus</th>\n",
       "      <th>Vagina</th>\n",
       "      <th>Whole_Blood</th>\n",
       "      <th>N_GTEx_tissues</th>\n",
       "      <th>sQTL</th>\n",
       "      <th>Vernot_allele_origin</th>\n",
       "      <th>Browning_allele_origin</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1181365</th>\n",
       "      <td>chr5</td>\n",
       "      <td>96235896</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>yes</td>\n",
       "      <td>2812.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.55</td>\n",
       "      <td>0.53</td>\n",
       "      <td>0.52</td>\n",
       "      <td>0.6</td>\n",
       "      <td>0.58</td>\n",
       "      <td>0.53</td>\n",
       "      <td>0.590643</td>\n",
       "      <td>yes</td>\n",
       "      <td>77437.0</td>\n",
       "      <td>142990.0</td>\n",
       "      <td>0.541555</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ERAP2</td>\n",
       "      <td>1.009</td>\n",
       "      <td>-0.07147</td>\n",
       "      <td>0.92859</td>\n",
       "      <td>0.46067</td>\n",
       "      <td>0.460</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.51</td>\n",
       "      <td>0.51</td>\n",
       "      <td>-37</td>\n",
       "      <td>-3</td>\n",
       "      <td>20</td>\n",
       "      <td>-3</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>32.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>ancient</td>\n",
       "      <td>ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1181375</th>\n",
       "      <td>chr5</td>\n",
       "      <td>96238551</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>yes</td>\n",
       "      <td>118.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.06</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.000975</td>\n",
       "      <td>yes</td>\n",
       "      <td>5387.0</td>\n",
       "      <td>143282.0</td>\n",
       "      <td>0.037597</td>\n",
       "      <td>yes</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.00318</td>\n",
       "      <td>0.00099</td>\n",
       "      <td>0.05043</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.05368</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.02352</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr5_95966719_96398083</td>\n",
       "      <td>0.025724</td>\n",
       "      <td>yes</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.02</td>\n",
       "      <td>ERAP2</td>\n",
       "      <td>1.009</td>\n",
       "      <td>-0.07147</td>\n",
       "      <td>0.92859</td>\n",
       "      <td>0.46067</td>\n",
       "      <td>0.455</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.53</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.53</td>\n",
       "      <td>-41</td>\n",
       "      <td>28</td>\n",
       "      <td>-5</td>\n",
       "      <td>41</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>introgressed</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1181403</th>\n",
       "      <td>chr5</td>\n",
       "      <td>96248413</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>Altai</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ERAP2</td>\n",
       "      <td>1.009</td>\n",
       "      <td>-0.07147</td>\n",
       "      <td>0.92859</td>\n",
       "      <td>0.46067</td>\n",
       "      <td>-0.376</td>\n",
       "      <td>0.24</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.24</td>\n",
       "      <td>-26</td>\n",
       "      <td>2</td>\n",
       "      <td>47</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        chrom       pos ref_allele alt_allele ancestral_allele  anc_dev  \\\n",
       "1181365  chr5  96235896          A          G                A  derived   \n",
       "1181375  chr5  96238551          T          G                T  derived   \n",
       "1181403  chr5  96248413          A          G                A  derived   \n",
       "\n",
       "        variant_type altai_gt chagyrskaya_gt denisovan_gt vindija_gt  \\\n",
       "1181365          snv      0/0            0/0          1/1        0/0   \n",
       "1181375          snv      1/1            1/1          0/0        1/1   \n",
       "1181403          snv      0/1            0/0          0/0        0/0   \n",
       "\n",
       "         altai_gt_boolean  chagyrskaya_gt_boolean  denisovan_gt_boolean  \\\n",
       "1181365             False                   False                  True   \n",
       "1181375              True                    True                 False   \n",
       "1181403              True                   False                 False   \n",
       "\n",
       "         vindija_gt_boolean distribution present_in_1KG  1KG_allele_count  \\\n",
       "1181365               False    Denisovan            yes            2812.0   \n",
       "1181375                True  Neanderthal            yes             118.0   \n",
       "1181403               False        Altai             no               NaN   \n",
       "\n",
       "         1KG_allele_number  1KG_allele_frequency  1KG_EAS_AF  1KG_EUR_AF  \\\n",
       "1181365             5096.0                  0.55        0.53        0.52   \n",
       "1181375             5096.0                  0.02        0.00        0.06   \n",
       "1181403                NaN                   NaN         NaN         NaN   \n",
       "\n",
       "         1KG_AFR_AF  1KG_AMR_AF  1KG_SAS_AF  1KG_non_ASW_AFR_AF  \\\n",
       "1181365         0.6        0.58        0.53            0.590643   \n",
       "1181375         0.0        0.05        0.02            0.000975   \n",
       "1181403         NaN         NaN         NaN                 NaN   \n",
       "\n",
       "        present_in_gnomAD  gnomAD_allele_count  gnomAD_allele_number  \\\n",
       "1181365               yes              77437.0              142990.0   \n",
       "1181375               yes               5387.0              143282.0   \n",
       "1181403                no                  NaN                   NaN   \n",
       "\n",
       "         gnomAD_allele_frequency Vernot_introgressed Vernot_ancestral_allele  \\\n",
       "1181365                 0.541555                  no                     NaN   \n",
       "1181375                 0.037597                 yes                       T   \n",
       "1181403                      NaN                  no                     NaN   \n",
       "\n",
       "        Vernot_derived_allele  Vernot_ancestral_derived_code  Vernot_AFA_AF  \\\n",
       "1181365                   NaN                            NaN            NaN   \n",
       "1181375                     G                            1.0        0.00318   \n",
       "1181403                   NaN                            NaN            NaN   \n",
       "\n",
       "         Vernot_AFR_AF  Vernot_AMR_AF  Vernot_EAS_AF  Vernot_EUR_AF  \\\n",
       "1181365            NaN            NaN            NaN            NaN   \n",
       "1181375        0.00099        0.05043            0.0        0.05368   \n",
       "1181403            NaN            NaN            NaN            NaN   \n",
       "\n",
       "         Vernot_PNG_AF  Vernot_SAS_AF Vernot_Denisovan_base  \\\n",
       "1181365            NaN            NaN                   NaN   \n",
       "1181375            0.0        0.02352                     T   \n",
       "1181403            NaN            NaN                   NaN   \n",
       "\n",
       "        Vernot_Neanderthal_base_1 Vernot_Neanderthal_base_2  \\\n",
       "1181365                       NaN                       NaN   \n",
       "1181375                         G                       NaN   \n",
       "1181403                       NaN                       NaN   \n",
       "\n",
       "           Vernot_haplotype_tag  Vernot_introgressed_AF Browning_introgressed  \\\n",
       "1181365                     NaN                     NaN                    no   \n",
       "1181375  chr5_95966719_96398083                0.025724                   yes   \n",
       "1181403                     NaN                     NaN                    no   \n",
       "\n",
       "         Browning_ref_alt  Browning_introgressed_AF annotation  mis_oe  \\\n",
       "1181365               NaN                       NaN      ERAP2   1.009   \n",
       "1181375               1.0                      0.02      ERAP2   1.009   \n",
       "1181403               NaN                       NaN      ERAP2   1.009   \n",
       "\n",
       "           mis_z   lof_oe    lof_z  phyloP  ag_delta  al_delta  dg_delta  \\\n",
       "1181365 -0.07147  0.92859  0.46067   0.460      0.00      0.00      0.00   \n",
       "1181375 -0.07147  0.92859  0.46067   0.455      0.00      0.00      0.53   \n",
       "1181403 -0.07147  0.92859  0.46067  -0.376      0.24      0.01      0.00   \n",
       "\n",
       "         dl_delta  delta_max  ag_pos  al_pos  dg_pos  dl_pos  \\\n",
       "1181365      0.51       0.51     -37      -3      20      -3   \n",
       "1181375      0.00       0.53     -41      28      -5      41   \n",
       "1181403      0.00       0.24     -26       2      47       2   \n",
       "\n",
       "         Adipose_Subcutaneous  Adipose_Visceral_Omentum  Adrenal_Gland  \\\n",
       "1181365                   1.0                       1.0            1.0   \n",
       "1181375                   0.0                       0.0            0.0   \n",
       "1181403                   NaN                       NaN            NaN   \n",
       "\n",
       "         Artery_Aorta  Artery_Coronary  Artery_Tibial  Brain_Amygdala  \\\n",
       "1181365           1.0              1.0            1.0             0.0   \n",
       "1181375           0.0              0.0            0.0             0.0   \n",
       "1181403           NaN              NaN            NaN             NaN   \n",
       "\n",
       "         Brain_Anterior_cingulate_cortex_BA24  Brain_Caudate_basal_ganglia  \\\n",
       "1181365                                   0.0                          0.0   \n",
       "1181375                                   0.0                          0.0   \n",
       "1181403                                   NaN                          NaN   \n",
       "\n",
       "         Brain_Cerebellar_Hemisphere  Brain_Cerebellum  Brain_Cortex  \\\n",
       "1181365                          1.0               0.0           1.0   \n",
       "1181375                          0.0               0.0           0.0   \n",
       "1181403                          NaN               NaN           NaN   \n",
       "\n",
       "         Brain_Frontal_Cortex_BA9  Brain_Hippocampus  Brain_Hypothalamus  \\\n",
       "1181365                       0.0                0.0                 0.0   \n",
       "1181375                       0.0                0.0                 0.0   \n",
       "1181403                       NaN                NaN                 NaN   \n",
       "\n",
       "         Brain_Nucleus_accumbens_basal_ganglia  Brain_Putamen_basal_ganglia  \\\n",
       "1181365                                    0.0                          0.0   \n",
       "1181375                                    0.0                          0.0   \n",
       "1181403                                    NaN                          NaN   \n",
       "\n",
       "         Brain_Spinal_cord_cervical_c-1  Brain_Substantia_nigra  \\\n",
       "1181365                             1.0                     0.0   \n",
       "1181375                             0.0                     0.0   \n",
       "1181403                             NaN                     NaN   \n",
       "\n",
       "         Breast_Mammary_Tissue  Cells_Cultured_fibroblasts  \\\n",
       "1181365                    1.0                         1.0   \n",
       "1181375                    0.0                         0.0   \n",
       "1181403                    NaN                         NaN   \n",
       "\n",
       "         Cells_EBV-transformed_lymphocytes  Colon_Sigmoid  Colon_Transverse  \\\n",
       "1181365                                0.0            1.0               0.0   \n",
       "1181375                                1.0            0.0               1.0   \n",
       "1181403                                NaN            NaN               NaN   \n",
       "\n",
       "         Esophagus_Gastroesophageal_Junction  Esophagus_Mucosa  \\\n",
       "1181365                                  1.0               1.0   \n",
       "1181375                                  0.0               0.0   \n",
       "1181403                                  NaN               NaN   \n",
       "\n",
       "         Esophagus_Muscularis  Heart_Atrial_Appendage  Heart_Left_Ventricle  \\\n",
       "1181365                   1.0                     1.0                   1.0   \n",
       "1181375                   0.0                     0.0                   0.0   \n",
       "1181403                   NaN                     NaN                   NaN   \n",
       "\n",
       "         Kidney_Cortex  Liver  Lung  Minor_Salivary_Gland  Muscle_Skeletal  \\\n",
       "1181365            0.0    0.0   1.0                   1.0              1.0   \n",
       "1181375            0.0    0.0   0.0                   0.0              1.0   \n",
       "1181403            NaN    NaN   NaN                   NaN              NaN   \n",
       "\n",
       "         Nerve_Tibial  Ovary  Pancreas  Pituitary  Prostate  \\\n",
       "1181365           1.0    1.0       0.0        1.0       1.0   \n",
       "1181375           0.0    0.0       0.0        0.0       0.0   \n",
       "1181403           NaN    NaN       NaN        NaN       NaN   \n",
       "\n",
       "         Skin_Not_Sun_Exposed_Suprapubic  Skin_Sun_Exposed_Lower_leg  \\\n",
       "1181365                              1.0                         1.0   \n",
       "1181375                              0.0                         0.0   \n",
       "1181403                              NaN                         NaN   \n",
       "\n",
       "         Small_Intestine_Terminal_Ileum  Spleen  Stomach  Testis  Thyroid  \\\n",
       "1181365                             0.0     0.0      1.0     1.0      1.0   \n",
       "1181375                             1.0     1.0      0.0     0.0      0.0   \n",
       "1181403                             NaN     NaN      NaN     NaN      NaN   \n",
       "\n",
       "         Uterus  Vagina  Whole_Blood  N_GTEx_tissues sQTL  \\\n",
       "1181365     1.0     1.0          1.0            32.0  yes   \n",
       "1181375     0.0     0.0          0.0             5.0  yes   \n",
       "1181403     NaN     NaN          NaN             NaN   no   \n",
       "\n",
       "        Vernot_allele_origin Browning_allele_origin  \n",
       "1181365              ancient                ancient  \n",
       "1181375         introgressed           introgressed  \n",
       "1181403     archaic-specific       archaic-specific  "
      ]
     },
     "execution_count": 599,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_2[data_2['annotation']=='ERAP2']"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "62d87d55",
   "metadata": {},
   "source": [
    "From McCoy et al. 2017"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 600,
   "id": "5314c4c2",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref_allele</th>\n",
       "      <th>alt_allele</th>\n",
       "      <th>ancestral_allele</th>\n",
       "      <th>anc_dev</th>\n",
       "      <th>variant_type</th>\n",
       "      <th>altai_gt</th>\n",
       "      <th>chagyrskaya_gt</th>\n",
       "      <th>denisovan_gt</th>\n",
       "      <th>vindija_gt</th>\n",
       "      <th>altai_gt_boolean</th>\n",
       "      <th>chagyrskaya_gt_boolean</th>\n",
       "      <th>denisovan_gt_boolean</th>\n",
       "      <th>vindija_gt_boolean</th>\n",
       "      <th>distribution</th>\n",
       "      <th>present_in_1KG</th>\n",
       "      <th>1KG_allele_count</th>\n",
       "      <th>1KG_allele_number</th>\n",
       "      <th>1KG_allele_frequency</th>\n",
       "      <th>1KG_EAS_AF</th>\n",
       "      <th>1KG_EUR_AF</th>\n",
       "      <th>1KG_AFR_AF</th>\n",
       "      <th>1KG_AMR_AF</th>\n",
       "      <th>1KG_SAS_AF</th>\n",
       "      <th>1KG_non_ASW_AFR_AF</th>\n",
       "      <th>present_in_gnomAD</th>\n",
       "      <th>gnomAD_allele_count</th>\n",
       "      <th>gnomAD_allele_number</th>\n",
       "      <th>gnomAD_allele_frequency</th>\n",
       "      <th>Vernot_introgressed</th>\n",
       "      <th>Vernot_ancestral_allele</th>\n",
       "      <th>Vernot_derived_allele</th>\n",
       "      <th>Vernot_ancestral_derived_code</th>\n",
       "      <th>Vernot_AFA_AF</th>\n",
       "      <th>Vernot_AFR_AF</th>\n",
       "      <th>Vernot_AMR_AF</th>\n",
       "      <th>Vernot_EAS_AF</th>\n",
       "      <th>Vernot_EUR_AF</th>\n",
       "      <th>Vernot_PNG_AF</th>\n",
       "      <th>Vernot_SAS_AF</th>\n",
       "      <th>Vernot_Denisovan_base</th>\n",
       "      <th>Vernot_Neanderthal_base_1</th>\n",
       "      <th>Vernot_Neanderthal_base_2</th>\n",
       "      <th>Vernot_haplotype_tag</th>\n",
       "      <th>Vernot_introgressed_AF</th>\n",
       "      <th>Browning_introgressed</th>\n",
       "      <th>Browning_ref_alt</th>\n",
       "      <th>Browning_introgressed_AF</th>\n",
       "      <th>annotation</th>\n",
       "      <th>mis_oe</th>\n",
       "      <th>mis_z</th>\n",
       "      <th>lof_oe</th>\n",
       "      <th>lof_z</th>\n",
       "      <th>phyloP</th>\n",
       "      <th>ag_delta</th>\n",
       "      <th>al_delta</th>\n",
       "      <th>dg_delta</th>\n",
       "      <th>dl_delta</th>\n",
       "      <th>delta_max</th>\n",
       "      <th>ag_pos</th>\n",
       "      <th>al_pos</th>\n",
       "      <th>dg_pos</th>\n",
       "      <th>dl_pos</th>\n",
       "      <th>Adipose_Subcutaneous</th>\n",
       "      <th>Adipose_Visceral_Omentum</th>\n",
       "      <th>Adrenal_Gland</th>\n",
       "      <th>Artery_Aorta</th>\n",
       "      <th>Artery_Coronary</th>\n",
       "      <th>Artery_Tibial</th>\n",
       "      <th>Brain_Amygdala</th>\n",
       "      <th>Brain_Anterior_cingulate_cortex_BA24</th>\n",
       "      <th>Brain_Caudate_basal_ganglia</th>\n",
       "      <th>Brain_Cerebellar_Hemisphere</th>\n",
       "      <th>Brain_Cerebellum</th>\n",
       "      <th>Brain_Cortex</th>\n",
       "      <th>Brain_Frontal_Cortex_BA9</th>\n",
       "      <th>Brain_Hippocampus</th>\n",
       "      <th>Brain_Hypothalamus</th>\n",
       "      <th>Brain_Nucleus_accumbens_basal_ganglia</th>\n",
       "      <th>Brain_Putamen_basal_ganglia</th>\n",
       "      <th>Brain_Spinal_cord_cervical_c-1</th>\n",
       "      <th>Brain_Substantia_nigra</th>\n",
       "      <th>Breast_Mammary_Tissue</th>\n",
       "      <th>Cells_Cultured_fibroblasts</th>\n",
       "      <th>Cells_EBV-transformed_lymphocytes</th>\n",
       "      <th>Colon_Sigmoid</th>\n",
       "      <th>Colon_Transverse</th>\n",
       "      <th>Esophagus_Gastroesophageal_Junction</th>\n",
       "      <th>Esophagus_Mucosa</th>\n",
       "      <th>Esophagus_Muscularis</th>\n",
       "      <th>Heart_Atrial_Appendage</th>\n",
       "      <th>Heart_Left_Ventricle</th>\n",
       "      <th>Kidney_Cortex</th>\n",
       "      <th>Liver</th>\n",
       "      <th>Lung</th>\n",
       "      <th>Minor_Salivary_Gland</th>\n",
       "      <th>Muscle_Skeletal</th>\n",
       "      <th>Nerve_Tibial</th>\n",
       "      <th>Ovary</th>\n",
       "      <th>Pancreas</th>\n",
       "      <th>Pituitary</th>\n",
       "      <th>Prostate</th>\n",
       "      <th>Skin_Not_Sun_Exposed_Suprapubic</th>\n",
       "      <th>Skin_Sun_Exposed_Lower_leg</th>\n",
       "      <th>Small_Intestine_Terminal_Ileum</th>\n",
       "      <th>Spleen</th>\n",
       "      <th>Stomach</th>\n",
       "      <th>Testis</th>\n",
       "      <th>Thyroid</th>\n",
       "      <th>Uterus</th>\n",
       "      <th>Vagina</th>\n",
       "      <th>Whole_Blood</th>\n",
       "      <th>N_GTEx_tissues</th>\n",
       "      <th>sQTL</th>\n",
       "      <th>Vernot_allele_origin</th>\n",
       "      <th>Browning_allele_origin</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>540636</th>\n",
       "      <td>chr15</td>\n",
       "      <td>84322854</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>./.</td>\n",
       "      <td>0/1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Vindija</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ADAMTSL3</td>\n",
       "      <td>0.95678</td>\n",
       "      <td>0.4755</td>\n",
       "      <td>0.5073</td>\n",
       "      <td>4.3946</td>\n",
       "      <td>0.334</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-11</td>\n",
       "      <td>40</td>\n",
       "      <td>-1</td>\n",
       "      <td>-11</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>540637</th>\n",
       "      <td>chr15</td>\n",
       "      <td>84324441</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Late Neanderthal</td>\n",
       "      <td>yes</td>\n",
       "      <td>743.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.15</td>\n",
       "      <td>0.06</td>\n",
       "      <td>0.28</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.37</td>\n",
       "      <td>0.12</td>\n",
       "      <td>0.002924</td>\n",
       "      <td>yes</td>\n",
       "      <td>28823.0</td>\n",
       "      <td>143164.0</td>\n",
       "      <td>0.201329</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ADAMTSL3</td>\n",
       "      <td>0.95678</td>\n",
       "      <td>0.4755</td>\n",
       "      <td>0.5073</td>\n",
       "      <td>4.3946</td>\n",
       "      <td>0.561</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>40</td>\n",
       "      <td>5</td>\n",
       "      <td>13</td>\n",
       "      <td>-32</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>ancient</td>\n",
       "      <td>ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>540638</th>\n",
       "      <td>chr15</td>\n",
       "      <td>84324778</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ADAMTSL3</td>\n",
       "      <td>0.95678</td>\n",
       "      <td>0.4755</td>\n",
       "      <td>0.5073</td>\n",
       "      <td>4.3946</td>\n",
       "      <td>-0.254</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>14</td>\n",
       "      <td>-31</td>\n",
       "      <td>-30</td>\n",
       "      <td>48</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>540639</th>\n",
       "      <td>chr15</td>\n",
       "      <td>84325970</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>yes</td>\n",
       "      <td>738.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.14</td>\n",
       "      <td>0.06</td>\n",
       "      <td>0.28</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.37</td>\n",
       "      <td>0.11</td>\n",
       "      <td>0.002924</td>\n",
       "      <td>yes</td>\n",
       "      <td>28868.0</td>\n",
       "      <td>77878.0</td>\n",
       "      <td>0.370682</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.14</td>\n",
       "      <td>ADAMTSL3</td>\n",
       "      <td>0.95678</td>\n",
       "      <td>0.4755</td>\n",
       "      <td>0.5073</td>\n",
       "      <td>4.3946</td>\n",
       "      <td>0.561</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-2</td>\n",
       "      <td>38</td>\n",
       "      <td>3</td>\n",
       "      <td>-4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>ancient</td>\n",
       "      <td>introgressed</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>540640</th>\n",
       "      <td>chr15</td>\n",
       "      <td>84326162</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ADAMTSL3</td>\n",
       "      <td>0.95678</td>\n",
       "      <td>0.4755</td>\n",
       "      <td>0.5073</td>\n",
       "      <td>4.3946</td>\n",
       "      <td>0.561</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.05</td>\n",
       "      <td>11</td>\n",
       "      <td>-45</td>\n",
       "      <td>2</td>\n",
       "      <td>42</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>archaic-specific</td>\n",
       "      <td>archaic-specific</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>541070</th>\n",
       "      <td>chr15</td>\n",
       "      <td>84703470</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Other</td>\n",
       "      <td>yes</td>\n",
       "      <td>670.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.13</td>\n",
       "      <td>0.09</td>\n",
       "      <td>0.25</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.17</td>\n",
       "      <td>0.18</td>\n",
       "      <td>0.003899</td>\n",
       "      <td>yes</td>\n",
       "      <td>25253.0</td>\n",
       "      <td>143126.0</td>\n",
       "      <td>0.176439</td>\n",
       "      <td>yes</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.03822</td>\n",
       "      <td>0.00397</td>\n",
       "      <td>0.17291</td>\n",
       "      <td>0.08730</td>\n",
       "      <td>0.24751</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.18200</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr15_84263508_84707851</td>\n",
       "      <td>0.138738</td>\n",
       "      <td>yes</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.13</td>\n",
       "      <td>ADAMTSL3</td>\n",
       "      <td>0.95678</td>\n",
       "      <td>0.4755</td>\n",
       "      <td>0.5073</td>\n",
       "      <td>4.3946</td>\n",
       "      <td>0.491</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.01</td>\n",
       "      <td>14</td>\n",
       "      <td>-27</td>\n",
       "      <td>14</td>\n",
       "      <td>13</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>introgressed</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>541071</th>\n",
       "      <td>chr15</td>\n",
       "      <td>84703874</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>yes</td>\n",
       "      <td>646.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.13</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.25</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.18</td>\n",
       "      <td>0.19</td>\n",
       "      <td>0.003899</td>\n",
       "      <td>yes</td>\n",
       "      <td>25535.0</td>\n",
       "      <td>143088.0</td>\n",
       "      <td>0.178457</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yes</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.13</td>\n",
       "      <td>ADAMTSL3</td>\n",
       "      <td>0.95678</td>\n",
       "      <td>0.4755</td>\n",
       "      <td>0.5073</td>\n",
       "      <td>4.3946</td>\n",
       "      <td>0.334</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-46</td>\n",
       "      <td>36</td>\n",
       "      <td>-5</td>\n",
       "      <td>-48</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>ancient</td>\n",
       "      <td>introgressed</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>541072</th>\n",
       "      <td>chr15</td>\n",
       "      <td>84705645</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>yes</td>\n",
       "      <td>19.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.014620</td>\n",
       "      <td>yes</td>\n",
       "      <td>544.0</td>\n",
       "      <td>143248.0</td>\n",
       "      <td>0.003798</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ADAMTSL3</td>\n",
       "      <td>0.95678</td>\n",
       "      <td>0.4755</td>\n",
       "      <td>0.5073</td>\n",
       "      <td>4.3946</td>\n",
       "      <td>-0.171</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-5</td>\n",
       "      <td>9</td>\n",
       "      <td>-5</td>\n",
       "      <td>39</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "      <td>low-confidence ancient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>541073</th>\n",
       "      <td>chr15</td>\n",
       "      <td>84706461</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Other</td>\n",
       "      <td>yes</td>\n",
       "      <td>665.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.13</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.28</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.18</td>\n",
       "      <td>0.18</td>\n",
       "      <td>0.003899</td>\n",
       "      <td>yes</td>\n",
       "      <td>27546.0</td>\n",
       "      <td>143200.0</td>\n",
       "      <td>0.192360</td>\n",
       "      <td>yes</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.03503</td>\n",
       "      <td>0.00397</td>\n",
       "      <td>0.17867</td>\n",
       "      <td>0.05357</td>\n",
       "      <td>0.27336</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.18200</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr15_84263508_84707851</td>\n",
       "      <td>0.138314</td>\n",
       "      <td>yes</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.13</td>\n",
       "      <td>ADAMTSL3</td>\n",
       "      <td>0.95678</td>\n",
       "      <td>0.4755</td>\n",
       "      <td>0.5073</td>\n",
       "      <td>4.3946</td>\n",
       "      <td>0.591</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.19</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.19</td>\n",
       "      <td>22</td>\n",
       "      <td>-9</td>\n",
       "      <td>49</td>\n",
       "      <td>17</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>introgressed</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>541074</th>\n",
       "      <td>chr15</td>\n",
       "      <td>84707851</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>derived</td>\n",
       "      <td>snv</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>yes</td>\n",
       "      <td>682.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.13</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.28</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.18</td>\n",
       "      <td>0.20</td>\n",
       "      <td>0.003899</td>\n",
       "      <td>yes</td>\n",
       "      <td>27763.0</td>\n",
       "      <td>143092.0</td>\n",
       "      <td>0.194022</td>\n",
       "      <td>yes</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.03822</td>\n",
       "      <td>0.00397</td>\n",
       "      <td>0.18012</td>\n",
       "      <td>0.05357</td>\n",
       "      <td>0.27336</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.19734</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr15_84263508_84707851</td>\n",
       "      <td>0.141672</td>\n",
       "      <td>yes</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.13</td>\n",
       "      <td>ADAMTSL3</td>\n",
       "      <td>0.95678</td>\n",
       "      <td>0.4755</td>\n",
       "      <td>0.5073</td>\n",
       "      <td>4.3946</td>\n",
       "      <td>0.459</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>40</td>\n",
       "      <td>28</td>\n",
       "      <td>-25</td>\n",
       "      <td>-4</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>introgressed</td>\n",
       "      <td>introgressed</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>439 rows × 117 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        chrom       pos ref_allele alt_allele ancestral_allele  anc_dev  \\\n",
       "540636  chr15  84322854          T          G                T  derived   \n",
       "540637  chr15  84324441          G          A                G  derived   \n",
       "540638  chr15  84324778          C          T                C  derived   \n",
       "540639  chr15  84325970          C          T                C  derived   \n",
       "540640  chr15  84326162          G          A                G  derived   \n",
       "...       ...       ...        ...        ...              ...      ...   \n",
       "541070  chr15  84703470          C          T                C  derived   \n",
       "541071  chr15  84703874          A          G                A  derived   \n",
       "541072  chr15  84705645          C          G                C  derived   \n",
       "541073  chr15  84706461          C          T                C  derived   \n",
       "541074  chr15  84707851          A          G                A  derived   \n",
       "\n",
       "       variant_type altai_gt chagyrskaya_gt denisovan_gt vindija_gt  \\\n",
       "540636          snv      0/0            0/0          ./.        0/1   \n",
       "540637          snv      0/0            0/1          0/0        0/1   \n",
       "540638          snv      0/0            0/0          1/1        0/0   \n",
       "540639          snv      1/1            1/1          0/0        0/1   \n",
       "540640          snv      0/0            0/0          1/1        0/0   \n",
       "...             ...      ...            ...          ...        ...   \n",
       "541070          snv      1/1            0/0          0/0        0/1   \n",
       "541071          snv      1/1            1/1          0/0        1/1   \n",
       "541072          snv      0/0            0/0          1/1        0/0   \n",
       "541073          snv      1/1            0/0          0/0        0/1   \n",
       "541074          snv      1/1            1/1          0/0        1/1   \n",
       "\n",
       "        altai_gt_boolean  chagyrskaya_gt_boolean  denisovan_gt_boolean  \\\n",
       "540636             False                   False                 False   \n",
       "540637             False                    True                 False   \n",
       "540638             False                   False                  True   \n",
       "540639              True                    True                 False   \n",
       "540640             False                   False                  True   \n",
       "...                  ...                     ...                   ...   \n",
       "541070              True                   False                 False   \n",
       "541071              True                    True                 False   \n",
       "541072             False                   False                  True   \n",
       "541073              True                   False                 False   \n",
       "541074              True                    True                 False   \n",
       "\n",
       "        vindija_gt_boolean      distribution present_in_1KG  1KG_allele_count  \\\n",
       "540636                True           Vindija             no               NaN   \n",
       "540637                True  Late Neanderthal            yes             743.0   \n",
       "540638               False         Denisovan             no               NaN   \n",
       "540639                True       Neanderthal            yes             738.0   \n",
       "540640               False         Denisovan             no               NaN   \n",
       "...                    ...               ...            ...               ...   \n",
       "541070                True             Other            yes             670.0   \n",
       "541071                True       Neanderthal            yes             646.0   \n",
       "541072               False         Denisovan            yes              19.0   \n",
       "541073                True             Other            yes             665.0   \n",
       "541074                True       Neanderthal            yes             682.0   \n",
       "\n",
       "        1KG_allele_number  1KG_allele_frequency  1KG_EAS_AF  1KG_EUR_AF  \\\n",
       "540636                NaN                   NaN         NaN         NaN   \n",
       "540637             5096.0                  0.15        0.06        0.28   \n",
       "540638                NaN                   NaN         NaN         NaN   \n",
       "540639             5096.0                  0.14        0.06        0.28   \n",
       "540640                NaN                   NaN         NaN         NaN   \n",
       "...                   ...                   ...         ...         ...   \n",
       "541070             5096.0                  0.13        0.09        0.25   \n",
       "541071             5096.0                  0.13        0.05        0.25   \n",
       "541072             5096.0                  0.00        0.00        0.00   \n",
       "541073             5096.0                  0.13        0.05        0.28   \n",
       "541074             5096.0                  0.13        0.05        0.28   \n",
       "\n",
       "        1KG_AFR_AF  1KG_AMR_AF  1KG_SAS_AF  1KG_non_ASW_AFR_AF  \\\n",
       "540636         NaN         NaN         NaN                 NaN   \n",
       "540637        0.02        0.37        0.12            0.002924   \n",
       "540638         NaN         NaN         NaN                 NaN   \n",
       "540639        0.02        0.37        0.11            0.002924   \n",
       "540640         NaN         NaN         NaN                 NaN   \n",
       "...            ...         ...         ...                 ...   \n",
       "541070        0.01        0.17        0.18            0.003899   \n",
       "541071        0.01        0.18        0.19            0.003899   \n",
       "541072        0.01        0.00        0.00            0.014620   \n",
       "541073        0.01        0.18        0.18            0.003899   \n",
       "541074        0.01        0.18        0.20            0.003899   \n",
       "\n",
       "       present_in_gnomAD  gnomAD_allele_count  gnomAD_allele_number  \\\n",
       "540636                no                  NaN                   NaN   \n",
       "540637               yes              28823.0              143164.0   \n",
       "540638                no                  NaN                   NaN   \n",
       "540639               yes              28868.0               77878.0   \n",
       "540640                no                  NaN                   NaN   \n",
       "...                  ...                  ...                   ...   \n",
       "541070               yes              25253.0              143126.0   \n",
       "541071               yes              25535.0              143088.0   \n",
       "541072               yes                544.0              143248.0   \n",
       "541073               yes              27546.0              143200.0   \n",
       "541074               yes              27763.0              143092.0   \n",
       "\n",
       "        gnomAD_allele_frequency Vernot_introgressed Vernot_ancestral_allele  \\\n",
       "540636                      NaN                  no                     NaN   \n",
       "540637                 0.201329                  no                     NaN   \n",
       "540638                      NaN                  no                     NaN   \n",
       "540639                 0.370682                  no                     NaN   \n",
       "540640                      NaN                  no                     NaN   \n",
       "...                         ...                 ...                     ...   \n",
       "541070                 0.176439                 yes                       C   \n",
       "541071                 0.178457                  no                     NaN   \n",
       "541072                 0.003798                  no                     NaN   \n",
       "541073                 0.192360                 yes                       C   \n",
       "541074                 0.194022                 yes                       A   \n",
       "\n",
       "       Vernot_derived_allele  Vernot_ancestral_derived_code  Vernot_AFA_AF  \\\n",
       "540636                   NaN                            NaN            NaN   \n",
       "540637                   NaN                            NaN            NaN   \n",
       "540638                   NaN                            NaN            NaN   \n",
       "540639                   NaN                            NaN            NaN   \n",
       "540640                   NaN                            NaN            NaN   \n",
       "...                      ...                            ...            ...   \n",
       "541070                     T                            1.0        0.03822   \n",
       "541071                   NaN                            NaN            NaN   \n",
       "541072                   NaN                            NaN            NaN   \n",
       "541073                     T                            1.0        0.03503   \n",
       "541074                     G                            1.0        0.03822   \n",
       "\n",
       "        Vernot_AFR_AF  Vernot_AMR_AF  Vernot_EAS_AF  Vernot_EUR_AF  \\\n",
       "540636            NaN            NaN            NaN            NaN   \n",
       "540637            NaN            NaN            NaN            NaN   \n",
       "540638            NaN            NaN            NaN            NaN   \n",
       "540639            NaN            NaN            NaN            NaN   \n",
       "540640            NaN            NaN            NaN            NaN   \n",
       "...               ...            ...            ...            ...   \n",
       "541070        0.00397        0.17291        0.08730        0.24751   \n",
       "541071            NaN            NaN            NaN            NaN   \n",
       "541072            NaN            NaN            NaN            NaN   \n",
       "541073        0.00397        0.17867        0.05357        0.27336   \n",
       "541074        0.00397        0.18012        0.05357        0.27336   \n",
       "\n",
       "        Vernot_PNG_AF  Vernot_SAS_AF Vernot_Denisovan_base  \\\n",
       "540636            NaN            NaN                   NaN   \n",
       "540637            NaN            NaN                   NaN   \n",
       "540638            NaN            NaN                   NaN   \n",
       "540639            NaN            NaN                   NaN   \n",
       "540640            NaN            NaN                   NaN   \n",
       "...               ...            ...                   ...   \n",
       "541070            0.0        0.18200                     C   \n",
       "541071            NaN            NaN                   NaN   \n",
       "541072            NaN            NaN                   NaN   \n",
       "541073            0.0        0.18200                     C   \n",
       "541074            0.0        0.19734                     A   \n",
       "\n",
       "       Vernot_Neanderthal_base_1 Vernot_Neanderthal_base_2  \\\n",
       "540636                       NaN                       NaN   \n",
       "540637                       NaN                       NaN   \n",
       "540638                       NaN                       NaN   \n",
       "540639                       NaN                       NaN   \n",
       "540640                       NaN                       NaN   \n",
       "...                          ...                       ...   \n",
       "541070                         T                       NaN   \n",
       "541071                       NaN                       NaN   \n",
       "541072                       NaN                       NaN   \n",
       "541073                         T                       NaN   \n",
       "541074                         G                       NaN   \n",
       "\n",
       "           Vernot_haplotype_tag  Vernot_introgressed_AF Browning_introgressed  \\\n",
       "540636                      NaN                     NaN                    no   \n",
       "540637                      NaN                     NaN                    no   \n",
       "540638                      NaN                     NaN                    no   \n",
       "540639                      NaN                     NaN                   yes   \n",
       "540640                      NaN                     NaN                    no   \n",
       "...                         ...                     ...                   ...   \n",
       "541070  chr15_84263508_84707851                0.138738                   yes   \n",
       "541071                      NaN                     NaN                   yes   \n",
       "541072                      NaN                     NaN                    no   \n",
       "541073  chr15_84263508_84707851                0.138314                   yes   \n",
       "541074  chr15_84263508_84707851                0.141672                   yes   \n",
       "\n",
       "        Browning_ref_alt  Browning_introgressed_AF annotation   mis_oe  \\\n",
       "540636               NaN                       NaN   ADAMTSL3  0.95678   \n",
       "540637               NaN                       NaN   ADAMTSL3  0.95678   \n",
       "540638               NaN                       NaN   ADAMTSL3  0.95678   \n",
       "540639               1.0                      0.14   ADAMTSL3  0.95678   \n",
       "540640               NaN                       NaN   ADAMTSL3  0.95678   \n",
       "...                  ...                       ...        ...      ...   \n",
       "541070               1.0                      0.13   ADAMTSL3  0.95678   \n",
       "541071               1.0                      0.13   ADAMTSL3  0.95678   \n",
       "541072               NaN                       NaN   ADAMTSL3  0.95678   \n",
       "541073               1.0                      0.13   ADAMTSL3  0.95678   \n",
       "541074               1.0                      0.13   ADAMTSL3  0.95678   \n",
       "\n",
       "         mis_z  lof_oe   lof_z  phyloP  ag_delta  al_delta  dg_delta  \\\n",
       "540636  0.4755  0.5073  4.3946   0.334      0.00      0.00       0.0   \n",
       "540637  0.4755  0.5073  4.3946   0.561      0.00      0.00       0.0   \n",
       "540638  0.4755  0.5073  4.3946  -0.254      0.00      0.00       0.0   \n",
       "540639  0.4755  0.5073  4.3946   0.561      0.00      0.00       0.0   \n",
       "540640  0.4755  0.5073  4.3946   0.561      0.00      0.05       0.0   \n",
       "...        ...     ...     ...     ...       ...       ...       ...   \n",
       "541070  0.4755  0.5073  4.3946   0.491      0.01      0.00       0.0   \n",
       "541071  0.4755  0.5073  4.3946   0.334      0.00      0.00       0.0   \n",
       "541072  0.4755  0.5073  4.3946  -0.171      0.00      0.00       0.0   \n",
       "541073  0.4755  0.5073  4.3946   0.591      0.00      0.19       0.0   \n",
       "541074  0.4755  0.5073  4.3946   0.459      0.00      0.00       0.0   \n",
       "\n",
       "        dl_delta  delta_max  ag_pos  al_pos  dg_pos  dl_pos  \\\n",
       "540636      0.00       0.00     -11      40      -1     -11   \n",
       "540637      0.00       0.00      40       5      13     -32   \n",
       "540638      0.00       0.00      14     -31     -30      48   \n",
       "540639      0.00       0.00      -2      38       3      -4   \n",
       "540640      0.00       0.05      11     -45       2      42   \n",
       "...          ...        ...     ...     ...     ...     ...   \n",
       "541070      0.00       0.01      14     -27      14      13   \n",
       "541071      0.00       0.00     -46      36      -5     -48   \n",
       "541072      0.00       0.00      -5       9      -5      39   \n",
       "541073      0.01       0.19      22      -9      49      17   \n",
       "541074      0.00       0.00      40      28     -25      -4   \n",
       "\n",
       "        Adipose_Subcutaneous  Adipose_Visceral_Omentum  Adrenal_Gland  \\\n",
       "540636                   NaN                       NaN            NaN   \n",
       "540637                   0.0                       0.0            0.0   \n",
       "540638                   NaN                       NaN            NaN   \n",
       "540639                   NaN                       NaN            NaN   \n",
       "540640                   NaN                       NaN            NaN   \n",
       "...                      ...                       ...            ...   \n",
       "541070                   1.0                       0.0            0.0   \n",
       "541071                   1.0                       0.0            0.0   \n",
       "541072                   NaN                       NaN            NaN   \n",
       "541073                   1.0                       0.0            0.0   \n",
       "541074                   1.0                       0.0            0.0   \n",
       "\n",
       "        Artery_Aorta  Artery_Coronary  Artery_Tibial  Brain_Amygdala  \\\n",
       "540636           NaN              NaN            NaN             NaN   \n",
       "540637           0.0              0.0            0.0             0.0   \n",
       "540638           NaN              NaN            NaN             NaN   \n",
       "540639           NaN              NaN            NaN             NaN   \n",
       "540640           NaN              NaN            NaN             NaN   \n",
       "...              ...              ...            ...             ...   \n",
       "541070           1.0              0.0            1.0             0.0   \n",
       "541071           1.0              0.0            1.0             0.0   \n",
       "541072           NaN              NaN            NaN             NaN   \n",
       "541073           1.0              0.0            1.0             0.0   \n",
       "541074           1.0              0.0            1.0             0.0   \n",
       "\n",
       "        Brain_Anterior_cingulate_cortex_BA24  Brain_Caudate_basal_ganglia  \\\n",
       "540636                                   NaN                          NaN   \n",
       "540637                                   0.0                          0.0   \n",
       "540638                                   NaN                          NaN   \n",
       "540639                                   NaN                          NaN   \n",
       "540640                                   NaN                          NaN   \n",
       "...                                      ...                          ...   \n",
       "541070                                   0.0                          0.0   \n",
       "541071                                   0.0                          0.0   \n",
       "541072                                   NaN                          NaN   \n",
       "541073                                   0.0                          0.0   \n",
       "541074                                   0.0                          0.0   \n",
       "\n",
       "        Brain_Cerebellar_Hemisphere  Brain_Cerebellum  Brain_Cortex  \\\n",
       "540636                          NaN               NaN           NaN   \n",
       "540637                          0.0               0.0           0.0   \n",
       "540638                          NaN               NaN           NaN   \n",
       "540639                          NaN               NaN           NaN   \n",
       "540640                          NaN               NaN           NaN   \n",
       "...                             ...               ...           ...   \n",
       "541070                          0.0               0.0           0.0   \n",
       "541071                          0.0               0.0           0.0   \n",
       "541072                          NaN               NaN           NaN   \n",
       "541073                          0.0               0.0           0.0   \n",
       "541074                          0.0               0.0           0.0   \n",
       "\n",
       "        Brain_Frontal_Cortex_BA9  Brain_Hippocampus  Brain_Hypothalamus  \\\n",
       "540636                       NaN                NaN                 NaN   \n",
       "540637                       0.0                0.0                 0.0   \n",
       "540638                       NaN                NaN                 NaN   \n",
       "540639                       NaN                NaN                 NaN   \n",
       "540640                       NaN                NaN                 NaN   \n",
       "...                          ...                ...                 ...   \n",
       "541070                       0.0                0.0                 0.0   \n",
       "541071                       0.0                0.0                 0.0   \n",
       "541072                       NaN                NaN                 NaN   \n",
       "541073                       0.0                0.0                 0.0   \n",
       "541074                       0.0                0.0                 0.0   \n",
       "\n",
       "        Brain_Nucleus_accumbens_basal_ganglia  Brain_Putamen_basal_ganglia  \\\n",
       "540636                                    NaN                          NaN   \n",
       "540637                                    0.0                          0.0   \n",
       "540638                                    NaN                          NaN   \n",
       "540639                                    NaN                          NaN   \n",
       "540640                                    NaN                          NaN   \n",
       "...                                       ...                          ...   \n",
       "541070                                    0.0                          0.0   \n",
       "541071                                    0.0                          0.0   \n",
       "541072                                    NaN                          NaN   \n",
       "541073                                    0.0                          0.0   \n",
       "541074                                    0.0                          0.0   \n",
       "\n",
       "        Brain_Spinal_cord_cervical_c-1  Brain_Substantia_nigra  \\\n",
       "540636                             NaN                     NaN   \n",
       "540637                             0.0                     0.0   \n",
       "540638                             NaN                     NaN   \n",
       "540639                             NaN                     NaN   \n",
       "540640                             NaN                     NaN   \n",
       "...                                ...                     ...   \n",
       "541070                             0.0                     0.0   \n",
       "541071                             0.0                     0.0   \n",
       "541072                             NaN                     NaN   \n",
       "541073                             0.0                     0.0   \n",
       "541074                             0.0                     0.0   \n",
       "\n",
       "        Breast_Mammary_Tissue  Cells_Cultured_fibroblasts  \\\n",
       "540636                    NaN                         NaN   \n",
       "540637                    0.0                         0.0   \n",
       "540638                    NaN                         NaN   \n",
       "540639                    NaN                         NaN   \n",
       "540640                    NaN                         NaN   \n",
       "...                       ...                         ...   \n",
       "541070                    0.0                         1.0   \n",
       "541071                    0.0                         1.0   \n",
       "541072                    NaN                         NaN   \n",
       "541073                    0.0                         1.0   \n",
       "541074                    0.0                         1.0   \n",
       "\n",
       "        Cells_EBV-transformed_lymphocytes  Colon_Sigmoid  Colon_Transverse  \\\n",
       "540636                                NaN            NaN               NaN   \n",
       "540637                                0.0            0.0               0.0   \n",
       "540638                                NaN            NaN               NaN   \n",
       "540639                                NaN            NaN               NaN   \n",
       "540640                                NaN            NaN               NaN   \n",
       "...                                   ...            ...               ...   \n",
       "541070                                0.0            0.0               0.0   \n",
       "541071                                0.0            0.0               0.0   \n",
       "541072                                NaN            NaN               NaN   \n",
       "541073                                0.0            0.0               0.0   \n",
       "541074                                0.0            0.0               0.0   \n",
       "\n",
       "        Esophagus_Gastroesophageal_Junction  Esophagus_Mucosa  \\\n",
       "540636                                  NaN               NaN   \n",
       "540637                                  0.0               0.0   \n",
       "540638                                  NaN               NaN   \n",
       "540639                                  NaN               NaN   \n",
       "540640                                  NaN               NaN   \n",
       "...                                     ...               ...   \n",
       "541070                                  1.0               0.0   \n",
       "541071                                  1.0               0.0   \n",
       "541072                                  NaN               NaN   \n",
       "541073                                  1.0               0.0   \n",
       "541074                                  1.0               0.0   \n",
       "\n",
       "        Esophagus_Muscularis  Heart_Atrial_Appendage  Heart_Left_Ventricle  \\\n",
       "540636                   NaN                     NaN                   NaN   \n",
       "540637                   0.0                     0.0                   0.0   \n",
       "540638                   NaN                     NaN                   NaN   \n",
       "540639                   NaN                     NaN                   NaN   \n",
       "540640                   NaN                     NaN                   NaN   \n",
       "...                      ...                     ...                   ...   \n",
       "541070                   0.0                     0.0                   0.0   \n",
       "541071                   0.0                     0.0                   0.0   \n",
       "541072                   NaN                     NaN                   NaN   \n",
       "541073                   1.0                     0.0                   0.0   \n",
       "541074                   1.0                     0.0                   0.0   \n",
       "\n",
       "        Kidney_Cortex  Liver  Lung  Minor_Salivary_Gland  Muscle_Skeletal  \\\n",
       "540636            NaN    NaN   NaN                   NaN              NaN   \n",
       "540637            0.0    0.0   0.0                   0.0              0.0   \n",
       "540638            NaN    NaN   NaN                   NaN              NaN   \n",
       "540639            NaN    NaN   NaN                   NaN              NaN   \n",
       "540640            NaN    NaN   NaN                   NaN              NaN   \n",
       "...               ...    ...   ...                   ...              ...   \n",
       "541070            0.0    0.0   0.0                   0.0              0.0   \n",
       "541071            0.0    0.0   0.0                   0.0              0.0   \n",
       "541072            NaN    NaN   NaN                   NaN              NaN   \n",
       "541073            0.0    0.0   0.0                   0.0              0.0   \n",
       "541074            0.0    0.0   0.0                   0.0              0.0   \n",
       "\n",
       "        Nerve_Tibial  Ovary  Pancreas  Pituitary  Prostate  \\\n",
       "540636           NaN    NaN       NaN        NaN       NaN   \n",
       "540637           1.0    0.0       0.0        0.0       0.0   \n",
       "540638           NaN    NaN       NaN        NaN       NaN   \n",
       "540639           NaN    NaN       NaN        NaN       NaN   \n",
       "540640           NaN    NaN       NaN        NaN       NaN   \n",
       "...              ...    ...       ...        ...       ...   \n",
       "541070           1.0    0.0       0.0        0.0       0.0   \n",
       "541071           1.0    0.0       0.0        0.0       0.0   \n",
       "541072           NaN    NaN       NaN        NaN       NaN   \n",
       "541073           1.0    0.0       0.0        0.0       0.0   \n",
       "541074           1.0    0.0       0.0        0.0       0.0   \n",
       "\n",
       "        Skin_Not_Sun_Exposed_Suprapubic  Skin_Sun_Exposed_Lower_leg  \\\n",
       "540636                              NaN                         NaN   \n",
       "540637                              0.0                         0.0   \n",
       "540638                              NaN                         NaN   \n",
       "540639                              NaN                         NaN   \n",
       "540640                              NaN                         NaN   \n",
       "...                                 ...                         ...   \n",
       "541070                              0.0                         0.0   \n",
       "541071                              0.0                         0.0   \n",
       "541072                              NaN                         NaN   \n",
       "541073                              0.0                         0.0   \n",
       "541074                              0.0                         1.0   \n",
       "\n",
       "        Small_Intestine_Terminal_Ileum  Spleen  Stomach  Testis  Thyroid  \\\n",
       "540636                             NaN     NaN      NaN     NaN      NaN   \n",
       "540637                             0.0     0.0      0.0     1.0      0.0   \n",
       "540638                             NaN     NaN      NaN     NaN      NaN   \n",
       "540639                             NaN     NaN      NaN     NaN      NaN   \n",
       "540640                             NaN     NaN      NaN     NaN      NaN   \n",
       "...                                ...     ...      ...     ...      ...   \n",
       "541070                             0.0     0.0      0.0     1.0      1.0   \n",
       "541071                             0.0     0.0      0.0     1.0      1.0   \n",
       "541072                             NaN     NaN      NaN     NaN      NaN   \n",
       "541073                             0.0     0.0      0.0     1.0      1.0   \n",
       "541074                             0.0     0.0      0.0     1.0      1.0   \n",
       "\n",
       "        Uterus  Vagina  Whole_Blood  N_GTEx_tissues sQTL  \\\n",
       "540636     NaN     NaN          NaN             NaN   no   \n",
       "540637     0.0     0.0          0.0             2.0  yes   \n",
       "540638     NaN     NaN          NaN             NaN   no   \n",
       "540639     NaN     NaN          NaN             NaN   no   \n",
       "540640     NaN     NaN          NaN             NaN   no   \n",
       "...        ...     ...          ...             ...  ...   \n",
       "541070     0.0     0.0          0.0             8.0  yes   \n",
       "541071     0.0     0.0          0.0             8.0  yes   \n",
       "541072     NaN     NaN          NaN             NaN   no   \n",
       "541073     0.0     0.0          0.0             9.0  yes   \n",
       "541074     0.0     0.0          0.0            10.0  yes   \n",
       "\n",
       "          Vernot_allele_origin  Browning_allele_origin  \n",
       "540636        archaic-specific        archaic-specific  \n",
       "540637                 ancient                 ancient  \n",
       "540638        archaic-specific        archaic-specific  \n",
       "540639                 ancient            introgressed  \n",
       "540640        archaic-specific        archaic-specific  \n",
       "...                        ...                     ...  \n",
       "541070            introgressed            introgressed  \n",
       "541071                 ancient            introgressed  \n",
       "541072  low-confidence ancient  low-confidence ancient  \n",
       "541073            introgressed            introgressed  \n",
       "541074            introgressed            introgressed  \n",
       "\n",
       "[439 rows x 117 columns]"
      ]
     },
     "execution_count": 600,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data[data['annotation']=='ADAMTSL3']"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ca920bb8",
   "metadata": {},
   "source": [
    "# Average SAVs in 1KG <a class = 'anchor' id = 'avg1KGSAVs'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7a28a7dc",
   "metadata": {},
   "source": [
    "Is the distribution of SAVs among the archaics unusual? Here, we import data on modern humans to which we have applied SpliceAI."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 601,
   "id": "ea6ef77e",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/scratch/cbrand/ipykernel_47833/2307328771.py:2: DtypeWarning: Columns (0) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  avg_SAV = pd.read_csv('../thousand_genomes/thousand_genomes_avg_SAVs/spliceai_1KG_snvs.txt', sep = '\\t', names = avg_SAV_header)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref_allele</th>\n",
       "      <th>HG00137</th>\n",
       "      <th>HG00338</th>\n",
       "      <th>HG00619</th>\n",
       "      <th>HG01198</th>\n",
       "      <th>HG01281</th>\n",
       "      <th>HG01524</th>\n",
       "      <th>HG01802</th>\n",
       "      <th>HG02142</th>\n",
       "      <th>HG02345</th>\n",
       "      <th>HG02629</th>\n",
       "      <th>HG03060</th>\n",
       "      <th>HG03190</th>\n",
       "      <th>HG03708</th>\n",
       "      <th>HG03711</th>\n",
       "      <th>HG03800</th>\n",
       "      <th>HG04014</th>\n",
       "      <th>NA11830</th>\n",
       "      <th>NA18552</th>\n",
       "      <th>NA18868</th>\n",
       "      <th>NA19011</th>\n",
       "      <th>NA19452</th>\n",
       "      <th>NA19741</th>\n",
       "      <th>NA20537</th>\n",
       "      <th>NA21141</th>\n",
       "      <th>alt_allele</th>\n",
       "      <th>annotation</th>\n",
       "      <th>ag_delta</th>\n",
       "      <th>al_delta</th>\n",
       "      <th>dg_delta</th>\n",
       "      <th>dl_delta</th>\n",
       "      <th>ag_pos</th>\n",
       "      <th>al_pos</th>\n",
       "      <th>dg_pos</th>\n",
       "      <th>dl_pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10</td>\n",
       "      <td>47460</td>\n",
       "      <td>A</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>1|0</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>C</td>\n",
       "      <td>TUBB8</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>10</td>\n",
       "      <td>-22</td>\n",
       "      <td>-43</td>\n",
       "      <td>-22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>10</td>\n",
       "      <td>47754</td>\n",
       "      <td>T</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>0|1</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>C</td>\n",
       "      <td>TUBB8</td>\n",
       "      <td>0.09</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-2</td>\n",
       "      <td>-1</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>10</td>\n",
       "      <td>47792</td>\n",
       "      <td>A</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>0|1</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>G</td>\n",
       "      <td>TUBB8</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>34</td>\n",
       "      <td>-8</td>\n",
       "      <td>-20</td>\n",
       "      <td>-7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10</td>\n",
       "      <td>47804</td>\n",
       "      <td>T</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>1|0</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>C</td>\n",
       "      <td>TUBB8</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>22</td>\n",
       "      <td>-20</td>\n",
       "      <td>-32</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>10</td>\n",
       "      <td>47876</td>\n",
       "      <td>C</td>\n",
       "      <td>1|0</td>\n",
       "      <td>0|1</td>\n",
       "      <td>./.</td>\n",
       "      <td>0|1</td>\n",
       "      <td>0|1</td>\n",
       "      <td>1|0</td>\n",
       "      <td>./.</td>\n",
       "      <td>0|1</td>\n",
       "      <td>./.</td>\n",
       "      <td>1|0</td>\n",
       "      <td>1|0</td>\n",
       "      <td>1|0</td>\n",
       "      <td>1|0</td>\n",
       "      <td>0|1</td>\n",
       "      <td>1|0</td>\n",
       "      <td>1|0</td>\n",
       "      <td>0|1</td>\n",
       "      <td>0|1</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>0|1</td>\n",
       "      <td>0|1</td>\n",
       "      <td>./.</td>\n",
       "      <td>0|1</td>\n",
       "      <td>T</td>\n",
       "      <td>TUBB8</td>\n",
       "      <td>0.03</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-7</td>\n",
       "      <td>12</td>\n",
       "      <td>-6</td>\n",
       "      <td>-8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>10</td>\n",
       "      <td>48005</td>\n",
       "      <td>G</td>\n",
       "      <td>1|0</td>\n",
       "      <td>0|1</td>\n",
       "      <td>./.</td>\n",
       "      <td>0|1</td>\n",
       "      <td>1|0</td>\n",
       "      <td>0|1</td>\n",
       "      <td>./.</td>\n",
       "      <td>0|1</td>\n",
       "      <td>./.</td>\n",
       "      <td>1|0</td>\n",
       "      <td>1|0</td>\n",
       "      <td>0|1</td>\n",
       "      <td>1|0</td>\n",
       "      <td>1|0</td>\n",
       "      <td>1|1</td>\n",
       "      <td>1|0</td>\n",
       "      <td>0|1</td>\n",
       "      <td>0|1</td>\n",
       "      <td>0|1</td>\n",
       "      <td>./.</td>\n",
       "      <td>0|1</td>\n",
       "      <td>0|1</td>\n",
       "      <td>./.</td>\n",
       "      <td>0|1</td>\n",
       "      <td>A</td>\n",
       "      <td>TUBB8</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-7</td>\n",
       "      <td>-31</td>\n",
       "      <td>2</td>\n",
       "      <td>-31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>10</td>\n",
       "      <td>48078</td>\n",
       "      <td>T</td>\n",
       "      <td>0|1</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>0|1</td>\n",
       "      <td>0|1</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>1|0</td>\n",
       "      <td>1|0</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>0|1</td>\n",
       "      <td>0|1</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>1|0</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>1|0</td>\n",
       "      <td>1|0</td>\n",
       "      <td>C</td>\n",
       "      <td>TUBB8</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>36</td>\n",
       "      <td>37</td>\n",
       "      <td>-27</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>10</td>\n",
       "      <td>48221</td>\n",
       "      <td>T</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>1|0</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>C</td>\n",
       "      <td>TUBB8</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.14</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-1</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>-2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>10</td>\n",
       "      <td>48232</td>\n",
       "      <td>G</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>0|1</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>A</td>\n",
       "      <td>TUBB8</td>\n",
       "      <td>0.23</td>\n",
       "      <td>0.14</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-6</td>\n",
       "      <td>50</td>\n",
       "      <td>-5</td>\n",
       "      <td>50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>10</td>\n",
       "      <td>48249</td>\n",
       "      <td>C</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>1|0</td>\n",
       "      <td>1|0</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>T</td>\n",
       "      <td>TUBB8</td>\n",
       "      <td>0.14</td>\n",
       "      <td>0.31</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>33</td>\n",
       "      <td>-23</td>\n",
       "      <td>-47</td>\n",
       "      <td>33</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chrom    pos ref_allele HG00137 HG00338 HG00619 HG01198 HG01281 HG01524  \\\n",
       "0    10  47460          A     ./.     ./.     ./.     ./.     ./.     ./.   \n",
       "1    10  47754          T     ./.     ./.     ./.     ./.     0|1     ./.   \n",
       "2    10  47792          A     ./.     ./.     ./.     ./.     ./.     ./.   \n",
       "3    10  47804          T     ./.     ./.     ./.     ./.     ./.     1|0   \n",
       "4    10  47876          C     1|0     0|1     ./.     0|1     0|1     1|0   \n",
       "5    10  48005          G     1|0     0|1     ./.     0|1     1|0     0|1   \n",
       "6    10  48078          T     0|1     ./.     ./.     0|1     0|1     ./.   \n",
       "7    10  48221          T     ./.     ./.     ./.     ./.     ./.     ./.   \n",
       "8    10  48232          G     ./.     ./.     ./.     ./.     ./.     ./.   \n",
       "9    10  48249          C     ./.     ./.     ./.     ./.     ./.     ./.   \n",
       "\n",
       "  HG01802 HG02142 HG02345 HG02629 HG03060 HG03190 HG03708 HG03711 HG03800  \\\n",
       "0     ./.     ./.     ./.     ./.     ./.     ./.     ./.     1|0     ./.   \n",
       "1     ./.     ./.     ./.     ./.     ./.     ./.     ./.     ./.     ./.   \n",
       "2     ./.     ./.     ./.     ./.     ./.     ./.     ./.     ./.     ./.   \n",
       "3     ./.     ./.     ./.     ./.     ./.     ./.     ./.     ./.     ./.   \n",
       "4     ./.     0|1     ./.     1|0     1|0     1|0     1|0     0|1     1|0   \n",
       "5     ./.     0|1     ./.     1|0     1|0     0|1     1|0     1|0     1|1   \n",
       "6     ./.     1|0     1|0     ./.     ./.     ./.     0|1     0|1     ./.   \n",
       "7     ./.     ./.     ./.     ./.     ./.     ./.     ./.     1|0     ./.   \n",
       "8     ./.     ./.     ./.     ./.     ./.     ./.     ./.     ./.     0|1   \n",
       "9     ./.     ./.     ./.     ./.     ./.     ./.     ./.     ./.     ./.   \n",
       "\n",
       "  HG04014 NA11830 NA18552 NA18868 NA19011 NA19452 NA19741 NA20537 NA21141  \\\n",
       "0     ./.     ./.     ./.     ./.     ./.     ./.     ./.     ./.     ./.   \n",
       "1     ./.     ./.     ./.     ./.     ./.     ./.     ./.     ./.     ./.   \n",
       "2     ./.     ./.     ./.     ./.     ./.     ./.     0|1     ./.     ./.   \n",
       "3     ./.     ./.     ./.     ./.     ./.     ./.     ./.     ./.     ./.   \n",
       "4     1|0     0|1     0|1     ./.     ./.     0|1     0|1     ./.     0|1   \n",
       "5     1|0     0|1     0|1     0|1     ./.     0|1     0|1     ./.     0|1   \n",
       "6     ./.     ./.     1|0     ./.     ./.     ./.     ./.     1|0     1|0   \n",
       "7     ./.     ./.     ./.     ./.     ./.     ./.     ./.     ./.     ./.   \n",
       "8     ./.     ./.     ./.     ./.     ./.     ./.     ./.     ./.     ./.   \n",
       "9     ./.     ./.     ./.     ./.     ./.     1|0     1|0     ./.     ./.   \n",
       "\n",
       "  alt_allele annotation  ag_delta  al_delta  dg_delta  dl_delta  ag_pos  \\\n",
       "0          C      TUBB8      0.00      0.02       0.0       0.0      10   \n",
       "1          C      TUBB8      0.09      0.00       0.0       0.0      -1   \n",
       "2          G      TUBB8      0.00      0.00       0.0       0.0      34   \n",
       "3          C      TUBB8      0.00      0.00       0.0       0.0      22   \n",
       "4          T      TUBB8      0.03      0.00       0.0       0.0      -7   \n",
       "5          A      TUBB8      0.00      0.00       0.0       0.0      -7   \n",
       "6          C      TUBB8      0.00      0.00       0.0       0.0      36   \n",
       "7          C      TUBB8      0.00      0.14       0.0       0.0      -1   \n",
       "8          A      TUBB8      0.23      0.14       0.0       0.0      -6   \n",
       "9          T      TUBB8      0.14      0.31       0.0       0.0      33   \n",
       "\n",
       "   al_pos  dg_pos  dl_pos  \n",
       "0     -22     -43     -22  \n",
       "1      -2      -1      18  \n",
       "2      -8     -20      -7  \n",
       "3     -20     -32       0  \n",
       "4      12      -6      -8  \n",
       "5     -31       2     -31  \n",
       "6      37     -27      37  \n",
       "7       5       5      -2  \n",
       "8      50      -5      50  \n",
       "9     -23     -47      33  "
      ]
     },
     "execution_count": 601,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "avg_SAV_header = ['chrom','pos','ref_allele','HG00137','HG00338','HG00619','HG01198','HG01281','HG01524','HG01802','HG02142','HG02345','HG02629','HG03060','HG03190','HG03708','HG03711','HG03800','HG04014','NA11830','NA18552','NA18868','NA19011','NA19452','NA19741','NA20537','NA21141','alt_allele','annotation','ag_delta','al_delta','dg_delta','dl_delta','ag_pos','al_pos','dg_pos','dl_pos']\n",
    "avg_SAV = pd.read_csv('../thousand_genomes/thousand_genomes_avg_SAVs/spliceai_1KG_snvs.txt', sep = '\\t', names = avg_SAV_header)\n",
    "avg_SAV.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 602,
   "id": "6cb4ca3a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "4582422"
      ]
     },
     "execution_count": 602,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "avg_SAV = avg_SAV[(avg_SAV['chrom'] != 'X')]\n",
    "len(avg_SAV)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 603,
   "id": "2449c5ef",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref_allele</th>\n",
       "      <th>HG00137</th>\n",
       "      <th>HG00338</th>\n",
       "      <th>HG00619</th>\n",
       "      <th>HG01198</th>\n",
       "      <th>HG01281</th>\n",
       "      <th>HG01524</th>\n",
       "      <th>HG01802</th>\n",
       "      <th>HG02142</th>\n",
       "      <th>HG02345</th>\n",
       "      <th>HG02629</th>\n",
       "      <th>HG03060</th>\n",
       "      <th>HG03190</th>\n",
       "      <th>HG03708</th>\n",
       "      <th>HG03711</th>\n",
       "      <th>HG03800</th>\n",
       "      <th>HG04014</th>\n",
       "      <th>NA11830</th>\n",
       "      <th>NA18552</th>\n",
       "      <th>NA18868</th>\n",
       "      <th>NA19011</th>\n",
       "      <th>NA19452</th>\n",
       "      <th>NA19741</th>\n",
       "      <th>NA20537</th>\n",
       "      <th>NA21141</th>\n",
       "      <th>alt_allele</th>\n",
       "      <th>annotation</th>\n",
       "      <th>ag_delta</th>\n",
       "      <th>al_delta</th>\n",
       "      <th>dg_delta</th>\n",
       "      <th>dl_delta</th>\n",
       "      <th>ag_pos</th>\n",
       "      <th>al_pos</th>\n",
       "      <th>dg_pos</th>\n",
       "      <th>dl_pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>10</td>\n",
       "      <td>48232</td>\n",
       "      <td>G</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>0|1</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>A</td>\n",
       "      <td>TUBB8</td>\n",
       "      <td>0.23</td>\n",
       "      <td>0.14</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-6</td>\n",
       "      <td>50</td>\n",
       "      <td>-5</td>\n",
       "      <td>50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>10</td>\n",
       "      <td>48249</td>\n",
       "      <td>C</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>1|0</td>\n",
       "      <td>1|0</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>T</td>\n",
       "      <td>TUBB8</td>\n",
       "      <td>0.14</td>\n",
       "      <td>0.31</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>33</td>\n",
       "      <td>-23</td>\n",
       "      <td>-47</td>\n",
       "      <td>33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>262</th>\n",
       "      <td>10</td>\n",
       "      <td>220176</td>\n",
       "      <td>A</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>1|0</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>G</td>\n",
       "      <td>ZMYND11</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.27</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0</td>\n",
       "      <td>39</td>\n",
       "      <td>-1</td>\n",
       "      <td>-32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>417</th>\n",
       "      <td>10</td>\n",
       "      <td>288178</td>\n",
       "      <td>C</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>1|0</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>G</td>\n",
       "      <td>DIP2C</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.28</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-38</td>\n",
       "      <td>0</td>\n",
       "      <td>-18</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>675</th>\n",
       "      <td>10</td>\n",
       "      <td>349332</td>\n",
       "      <td>T</td>\n",
       "      <td>1|0</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>C</td>\n",
       "      <td>DIP2C</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.12</td>\n",
       "      <td>0.21</td>\n",
       "      <td>-14</td>\n",
       "      <td>-2</td>\n",
       "      <td>-14</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1229</th>\n",
       "      <td>10</td>\n",
       "      <td>487391</td>\n",
       "      <td>A</td>\n",
       "      <td>1|1</td>\n",
       "      <td>1|1</td>\n",
       "      <td>1|1</td>\n",
       "      <td>1|0</td>\n",
       "      <td>1|1</td>\n",
       "      <td>1|1</td>\n",
       "      <td>1|1</td>\n",
       "      <td>1|0</td>\n",
       "      <td>1|1</td>\n",
       "      <td>0|1</td>\n",
       "      <td>./.</td>\n",
       "      <td>1|1</td>\n",
       "      <td>1|1</td>\n",
       "      <td>1|1</td>\n",
       "      <td>0|1</td>\n",
       "      <td>0|1</td>\n",
       "      <td>1|1</td>\n",
       "      <td>1|1</td>\n",
       "      <td>0|1</td>\n",
       "      <td>1|1</td>\n",
       "      <td>./.</td>\n",
       "      <td>1|1</td>\n",
       "      <td>1|1</td>\n",
       "      <td>1|1</td>\n",
       "      <td>G</td>\n",
       "      <td>DIP2C</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.03</td>\n",
       "      <td>0.46</td>\n",
       "      <td>-16</td>\n",
       "      <td>2</td>\n",
       "      <td>-16</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1996</th>\n",
       "      <td>10</td>\n",
       "      <td>662757</td>\n",
       "      <td>T</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>0|1</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>0|1</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>C</td>\n",
       "      <td>DIP2C</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.45</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1</td>\n",
       "      <td>-3</td>\n",
       "      <td>1</td>\n",
       "      <td>-3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2564</th>\n",
       "      <td>10</td>\n",
       "      <td>996086</td>\n",
       "      <td>A</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>1|0</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>G</td>\n",
       "      <td>GTPBP4</td>\n",
       "      <td>0.22</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.11</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1</td>\n",
       "      <td>20</td>\n",
       "      <td>-49</td>\n",
       "      <td>49</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2638</th>\n",
       "      <td>10</td>\n",
       "      <td>1008958</td>\n",
       "      <td>G</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>0|1</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>A</td>\n",
       "      <td>GTPBP4</td>\n",
       "      <td>0.07</td>\n",
       "      <td>0.37</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>25</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2648</th>\n",
       "      <td>10</td>\n",
       "      <td>1010007</td>\n",
       "      <td>A</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>1|1</td>\n",
       "      <td>./.</td>\n",
       "      <td>0|1</td>\n",
       "      <td>./.</td>\n",
       "      <td>1|0</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>0|1</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>1|1</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>G</td>\n",
       "      <td>GTPBP4</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.57</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-40</td>\n",
       "      <td>17</td>\n",
       "      <td>-5</td>\n",
       "      <td>-28</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     chrom      pos ref_allele HG00137 HG00338 HG00619 HG01198 HG01281  \\\n",
       "8       10    48232          G     ./.     ./.     ./.     ./.     ./.   \n",
       "9       10    48249          C     ./.     ./.     ./.     ./.     ./.   \n",
       "262     10   220176          A     ./.     ./.     ./.     ./.     ./.   \n",
       "417     10   288178          C     ./.     ./.     ./.     ./.     ./.   \n",
       "675     10   349332          T     1|0     ./.     ./.     ./.     ./.   \n",
       "1229    10   487391          A     1|1     1|1     1|1     1|0     1|1   \n",
       "1996    10   662757          T     ./.     ./.     0|1     ./.     ./.   \n",
       "2564    10   996086          A     ./.     ./.     ./.     ./.     ./.   \n",
       "2638    10  1008958          G     ./.     ./.     ./.     ./.     ./.   \n",
       "2648    10  1010007          A     ./.     ./.     1|1     ./.     0|1   \n",
       "\n",
       "     HG01524 HG01802 HG02142 HG02345 HG02629 HG03060 HG03190 HG03708 HG03711  \\\n",
       "8        ./.     ./.     ./.     ./.     ./.     ./.     ./.     ./.     ./.   \n",
       "9        ./.     ./.     ./.     ./.     ./.     ./.     ./.     ./.     ./.   \n",
       "262      ./.     ./.     ./.     ./.     ./.     ./.     ./.     ./.     ./.   \n",
       "417      ./.     ./.     ./.     ./.     1|0     ./.     ./.     ./.     ./.   \n",
       "675      ./.     ./.     ./.     ./.     ./.     ./.     ./.     ./.     ./.   \n",
       "1229     1|1     1|1     1|0     1|1     0|1     ./.     1|1     1|1     1|1   \n",
       "1996     ./.     ./.     ./.     ./.     ./.     ./.     ./.     ./.     ./.   \n",
       "2564     ./.     ./.     ./.     ./.     ./.     ./.     ./.     ./.     ./.   \n",
       "2638     ./.     ./.     ./.     ./.     ./.     ./.     ./.     ./.     ./.   \n",
       "2648     ./.     1|0     ./.     ./.     ./.     ./.     ./.     ./.     ./.   \n",
       "\n",
       "     HG03800 HG04014 NA11830 NA18552 NA18868 NA19011 NA19452 NA19741 NA20537  \\\n",
       "8        0|1     ./.     ./.     ./.     ./.     ./.     ./.     ./.     ./.   \n",
       "9        ./.     ./.     ./.     ./.     ./.     ./.     1|0     1|0     ./.   \n",
       "262      ./.     ./.     ./.     ./.     ./.     ./.     1|0     ./.     ./.   \n",
       "417      ./.     ./.     ./.     ./.     ./.     ./.     ./.     ./.     ./.   \n",
       "675      ./.     ./.     ./.     ./.     ./.     ./.     ./.     ./.     ./.   \n",
       "1229     0|1     0|1     1|1     1|1     0|1     1|1     ./.     1|1     1|1   \n",
       "1996     ./.     ./.     ./.     ./.     ./.     ./.     ./.     0|1     ./.   \n",
       "2564     ./.     1|0     ./.     ./.     ./.     ./.     ./.     ./.     ./.   \n",
       "2638     ./.     ./.     0|1     ./.     ./.     ./.     ./.     ./.     ./.   \n",
       "2648     ./.     ./.     0|1     ./.     ./.     ./.     ./.     1|1     ./.   \n",
       "\n",
       "     NA21141 alt_allele annotation  ag_delta  al_delta  dg_delta  dl_delta  \\\n",
       "8        ./.          A      TUBB8      0.23      0.14      0.00      0.00   \n",
       "9        ./.          T      TUBB8      0.14      0.31      0.00      0.00   \n",
       "262      ./.          G    ZMYND11      0.00      0.00      0.27      0.00   \n",
       "417      ./.          G      DIP2C      0.01      0.28      0.00      0.00   \n",
       "675      ./.          C      DIP2C      0.00      0.00      0.12      0.21   \n",
       "1229     1|1          G      DIP2C      0.00      0.00      0.03      0.46   \n",
       "1996     ./.          C      DIP2C      0.00      0.00      0.45      0.00   \n",
       "2564     ./.          G     GTPBP4      0.22      0.00      0.11      0.00   \n",
       "2638     ./.          A     GTPBP4      0.07      0.37      0.00      0.00   \n",
       "2648     ./.          G     GTPBP4      0.00      0.00      0.57      0.00   \n",
       "\n",
       "      ag_pos  al_pos  dg_pos  dl_pos  \n",
       "8         -6      50      -5      50  \n",
       "9         33     -23     -47      33  \n",
       "262        0      39      -1     -32  \n",
       "417      -38       0     -18       1  \n",
       "675      -14      -2     -14      -1  \n",
       "1229     -16       2     -16       2  \n",
       "1996       1      -3       1      -3  \n",
       "2564       1      20     -49      49  \n",
       "2638      25       0      25       0  \n",
       "2648     -40      17      -5     -28  "
      ]
     },
     "execution_count": 603,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "avg_SAV = avg_SAV[(avg_SAV['chrom'] != 'X')]\n",
    "avg_SAV = avg_SAV[(avg_SAV['ag_delta'] >= 0.2) | (avg_SAV['al_delta'] >= 0.2) | (avg_SAV['dg_delta'] >= 0.2) | (avg_SAV['dl_delta'] >= 0.2)]\n",
    "avg_SAV.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 604,
   "id": "c3001a9c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "14006"
      ]
     },
     "execution_count": 604,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(avg_SAV)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 605,
   "id": "ab27c476",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref_allele</th>\n",
       "      <th>HG00137</th>\n",
       "      <th>HG00338</th>\n",
       "      <th>HG00619</th>\n",
       "      <th>HG01198</th>\n",
       "      <th>HG01281</th>\n",
       "      <th>HG01524</th>\n",
       "      <th>HG01802</th>\n",
       "      <th>HG02142</th>\n",
       "      <th>HG02345</th>\n",
       "      <th>HG02629</th>\n",
       "      <th>HG03060</th>\n",
       "      <th>HG03190</th>\n",
       "      <th>HG03708</th>\n",
       "      <th>HG03711</th>\n",
       "      <th>HG03800</th>\n",
       "      <th>HG04014</th>\n",
       "      <th>NA11830</th>\n",
       "      <th>NA18552</th>\n",
       "      <th>NA18868</th>\n",
       "      <th>NA19011</th>\n",
       "      <th>NA19452</th>\n",
       "      <th>NA19741</th>\n",
       "      <th>NA20537</th>\n",
       "      <th>NA21141</th>\n",
       "      <th>alt_allele</th>\n",
       "      <th>annotation</th>\n",
       "      <th>ag_delta</th>\n",
       "      <th>al_delta</th>\n",
       "      <th>dg_delta</th>\n",
       "      <th>dl_delta</th>\n",
       "      <th>ag_pos</th>\n",
       "      <th>al_pos</th>\n",
       "      <th>dg_pos</th>\n",
       "      <th>dl_pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>10</td>\n",
       "      <td>48232</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>A</td>\n",
       "      <td>TUBB8</td>\n",
       "      <td>0.23</td>\n",
       "      <td>0.14</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-6</td>\n",
       "      <td>50</td>\n",
       "      <td>-5</td>\n",
       "      <td>50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>10</td>\n",
       "      <td>48249</td>\n",
       "      <td>C</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>T</td>\n",
       "      <td>TUBB8</td>\n",
       "      <td>0.14</td>\n",
       "      <td>0.31</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>33</td>\n",
       "      <td>-23</td>\n",
       "      <td>-47</td>\n",
       "      <td>33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>262</th>\n",
       "      <td>10</td>\n",
       "      <td>220176</td>\n",
       "      <td>A</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>G</td>\n",
       "      <td>ZMYND11</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.27</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0</td>\n",
       "      <td>39</td>\n",
       "      <td>-1</td>\n",
       "      <td>-32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>417</th>\n",
       "      <td>10</td>\n",
       "      <td>288178</td>\n",
       "      <td>C</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>G</td>\n",
       "      <td>DIP2C</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.28</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-38</td>\n",
       "      <td>0</td>\n",
       "      <td>-18</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>675</th>\n",
       "      <td>10</td>\n",
       "      <td>349332</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>C</td>\n",
       "      <td>DIP2C</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.12</td>\n",
       "      <td>0.21</td>\n",
       "      <td>-14</td>\n",
       "      <td>-2</td>\n",
       "      <td>-14</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1229</th>\n",
       "      <td>10</td>\n",
       "      <td>487391</td>\n",
       "      <td>A</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>G</td>\n",
       "      <td>DIP2C</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.03</td>\n",
       "      <td>0.46</td>\n",
       "      <td>-16</td>\n",
       "      <td>2</td>\n",
       "      <td>-16</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1996</th>\n",
       "      <td>10</td>\n",
       "      <td>662757</td>\n",
       "      <td>T</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>C</td>\n",
       "      <td>DIP2C</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.45</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1</td>\n",
       "      <td>-3</td>\n",
       "      <td>1</td>\n",
       "      <td>-3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2564</th>\n",
       "      <td>10</td>\n",
       "      <td>996086</td>\n",
       "      <td>A</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>G</td>\n",
       "      <td>GTPBP4</td>\n",
       "      <td>0.22</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.11</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1</td>\n",
       "      <td>20</td>\n",
       "      <td>-49</td>\n",
       "      <td>49</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2638</th>\n",
       "      <td>10</td>\n",
       "      <td>1008958</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>A</td>\n",
       "      <td>GTPBP4</td>\n",
       "      <td>0.07</td>\n",
       "      <td>0.37</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>25</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2648</th>\n",
       "      <td>10</td>\n",
       "      <td>1010007</td>\n",
       "      <td>A</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>G</td>\n",
       "      <td>GTPBP4</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.57</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-40</td>\n",
       "      <td>17</td>\n",
       "      <td>-5</td>\n",
       "      <td>-28</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     chrom      pos ref_allele  HG00137  HG00338  HG00619  HG01198  HG01281  \\\n",
       "8       10    48232          G        0        0        0        0        0   \n",
       "9       10    48249          C        0        0        0        0        0   \n",
       "262     10   220176          A        0        0        0        0        0   \n",
       "417     10   288178          C        0        0        0        0        0   \n",
       "675     10   349332          T        1        0        0        0        0   \n",
       "1229    10   487391          A        1        1        1        1        1   \n",
       "1996    10   662757          T        0        0        1        0        0   \n",
       "2564    10   996086          A        0        0        0        0        0   \n",
       "2638    10  1008958          G        0        0        0        0        0   \n",
       "2648    10  1010007          A        0        0        1        0        1   \n",
       "\n",
       "      HG01524  HG01802  HG02142  HG02345  HG02629  HG03060  HG03190  HG03708  \\\n",
       "8           0        0        0        0        0        0        0        0   \n",
       "9           0        0        0        0        0        0        0        0   \n",
       "262         0        0        0        0        0        0        0        0   \n",
       "417         0        0        0        0        1        0        0        0   \n",
       "675         0        0        0        0        0        0        0        0   \n",
       "1229        1        1        1        1        1        0        1        1   \n",
       "1996        0        0        0        0        0        0        0        0   \n",
       "2564        0        0        0        0        0        0        0        0   \n",
       "2638        0        0        0        0        0        0        0        0   \n",
       "2648        0        1        0        0        0        0        0        0   \n",
       "\n",
       "      HG03711  HG03800  HG04014  NA11830  NA18552  NA18868  NA19011  NA19452  \\\n",
       "8           0        1        0        0        0        0        0        0   \n",
       "9           0        0        0        0        0        0        0        1   \n",
       "262         0        0        0        0        0        0        0        1   \n",
       "417         0        0        0        0        0        0        0        0   \n",
       "675         0        0        0        0        0        0        0        0   \n",
       "1229        1        1        1        1        1        1        1        0   \n",
       "1996        0        0        0        0        0        0        0        0   \n",
       "2564        0        0        1        0        0        0        0        0   \n",
       "2638        0        0        0        1        0        0        0        0   \n",
       "2648        0        0        0        1        0        0        0        0   \n",
       "\n",
       "      NA19741  NA20537  NA21141 alt_allele annotation  ag_delta  al_delta  \\\n",
       "8           0        0        0          A      TUBB8      0.23      0.14   \n",
       "9           1        0        0          T      TUBB8      0.14      0.31   \n",
       "262         0        0        0          G    ZMYND11      0.00      0.00   \n",
       "417         0        0        0          G      DIP2C      0.01      0.28   \n",
       "675         0        0        0          C      DIP2C      0.00      0.00   \n",
       "1229        1        1        1          G      DIP2C      0.00      0.00   \n",
       "1996        1        0        0          C      DIP2C      0.00      0.00   \n",
       "2564        0        0        0          G     GTPBP4      0.22      0.00   \n",
       "2638        0        0        0          A     GTPBP4      0.07      0.37   \n",
       "2648        1        0        0          G     GTPBP4      0.00      0.00   \n",
       "\n",
       "      dg_delta  dl_delta  ag_pos  al_pos  dg_pos  dl_pos  \n",
       "8         0.00      0.00      -6      50      -5      50  \n",
       "9         0.00      0.00      33     -23     -47      33  \n",
       "262       0.27      0.00       0      39      -1     -32  \n",
       "417       0.00      0.00     -38       0     -18       1  \n",
       "675       0.12      0.21     -14      -2     -14      -1  \n",
       "1229      0.03      0.46     -16       2     -16       2  \n",
       "1996      0.45      0.00       1      -3       1      -3  \n",
       "2564      0.11      0.00       1      20     -49      49  \n",
       "2638      0.00      0.00      25       0      25       0  \n",
       "2648      0.57      0.00     -40      17      -5     -28  "
      ]
     },
     "execution_count": 605,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "avg_SAV = avg_SAV.replace({'./.': 0, '0|1': 1, '1|0': 1, '1|1': 1})\n",
    "avg_SAV.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 606,
   "id": "42634d1c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "HG00137    3266\n",
       "HG00338    3344\n",
       "HG00619    3320\n",
       "HG01198    3514\n",
       "HG01281    3493\n",
       "HG01524    3363\n",
       "HG01802    3363\n",
       "HG02142    3349\n",
       "HG02345    3385\n",
       "HG02629    4120\n",
       "HG03060    4160\n",
       "HG03190    4142\n",
       "HG03708    3368\n",
       "HG03711    3379\n",
       "HG03800    3435\n",
       "HG04014    3421\n",
       "NA11830    3200\n",
       "NA18552    3382\n",
       "NA18868    4102\n",
       "NA19011    3321\n",
       "NA19452    4078\n",
       "NA19741    3284\n",
       "NA20537    3168\n",
       "NA21141    3487\n",
       "dtype: int64"
      ]
     },
     "execution_count": 606,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "avg_SAV[['HG00137','HG00338','HG00619','HG01198','HG01281','HG01524','HG01802','HG02142','HG02345','HG02629','HG03060','HG03190','HG03708','HG03711','HG03800','HG04014','NA11830','NA18552','NA18868','NA19011','NA19452','NA19741','NA20537','NA21141']].sum(axis = 0, skipna = True)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3f9c8048",
   "metadata": {},
   "source": [
    "# Introgressed Reference Alleles <a class = 'anchor' id = 'introgressedrefs'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "84730830",
   "metadata": {},
   "source": [
    "Let's check if switching the reference and alternate alleles results in the same SpliceAI results for our introgressed variants. "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "03a88e82",
   "metadata": {},
   "source": [
    "Load in the original dataframe where the introgressed variants are the reference alleles."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 607,
   "id": "0f5fcb56",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref_allele</th>\n",
       "      <th>alt_allele</th>\n",
       "      <th>variant_type</th>\n",
       "      <th>altai_gt</th>\n",
       "      <th>chagyrskaya_gt</th>\n",
       "      <th>denisovan_gt</th>\n",
       "      <th>vindija_gt</th>\n",
       "      <th>altai_gt_boolean</th>\n",
       "      <th>chagyrskaya_gt_boolean</th>\n",
       "      <th>denisovan_gt_boolean</th>\n",
       "      <th>vindija_gt_boolean</th>\n",
       "      <th>distribution</th>\n",
       "      <th>annotation</th>\n",
       "      <th>mis_oe</th>\n",
       "      <th>mis_z</th>\n",
       "      <th>lof_oe</th>\n",
       "      <th>lof_z</th>\n",
       "      <th>phyloP</th>\n",
       "      <th>ag_delta</th>\n",
       "      <th>al_delta</th>\n",
       "      <th>dg_delta</th>\n",
       "      <th>dl_delta</th>\n",
       "      <th>delta_max</th>\n",
       "      <th>ag_pos</th>\n",
       "      <th>al_pos</th>\n",
       "      <th>dg_pos</th>\n",
       "      <th>dl_pos</th>\n",
       "      <th>ancestral_allele</th>\n",
       "      <th>temp</th>\n",
       "      <th>anc_dev</th>\n",
       "      <th>1KG_allele_count</th>\n",
       "      <th>1KG_allele_number</th>\n",
       "      <th>1KG_allele_frequency</th>\n",
       "      <th>1KG_EAS_AF</th>\n",
       "      <th>1KG_EUR_AF</th>\n",
       "      <th>1KG_AFR_AF</th>\n",
       "      <th>1KG_AMR_AF</th>\n",
       "      <th>1KG_SAS_AF</th>\n",
       "      <th>present_in_1KG</th>\n",
       "      <th>1KG_non_ASW_AFR_AF</th>\n",
       "      <th>gnomAD_allele_count</th>\n",
       "      <th>gnomAD_allele_number</th>\n",
       "      <th>gnomAD_allele_frequency</th>\n",
       "      <th>present_in_gnomAD</th>\n",
       "      <th>start</th>\n",
       "      <th>Vernot_ancestral_allele</th>\n",
       "      <th>Vernot_derived_allele</th>\n",
       "      <th>Vernot_ancestral_derived_code</th>\n",
       "      <th>Vernot_AFA_AF</th>\n",
       "      <th>Vernot_AFR_AF</th>\n",
       "      <th>Vernot_AMR_AF</th>\n",
       "      <th>Vernot_EAS_AF</th>\n",
       "      <th>Vernot_EUR_AF</th>\n",
       "      <th>Vernot_PNG_AF</th>\n",
       "      <th>Vernot_SAS_AF</th>\n",
       "      <th>Vernot_Denisovan_base</th>\n",
       "      <th>Vernot_haplotype_tag</th>\n",
       "      <th>Vernot_Neanderthal_base_1</th>\n",
       "      <th>Vernot_Neanderthal_base_2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3046711</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Other</td>\n",
       "      <td>PRDM16</td>\n",
       "      <td>0.87043</td>\n",
       "      <td>1.3471</td>\n",
       "      <td>0.081747</td>\n",
       "      <td>5.9522</td>\n",
       "      <td>0.379</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-4</td>\n",
       "      <td>35</td>\n",
       "      <td>-24</td>\n",
       "      <td>19</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>derived</td>\n",
       "      <td>112.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.06</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.01</td>\n",
       "      <td>yes</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>6632.0</td>\n",
       "      <td>143094.0</td>\n",
       "      <td>0.046347</td>\n",
       "      <td>yes</td>\n",
       "      <td>3046710.0</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.01274</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.04611</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.06362</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.01125</td>\n",
       "      <td>T</td>\n",
       "      <td>chr1_3015134_3046826</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3046826</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Other</td>\n",
       "      <td>PRDM16</td>\n",
       "      <td>0.87043</td>\n",
       "      <td>1.3471</td>\n",
       "      <td>0.081747</td>\n",
       "      <td>5.9522</td>\n",
       "      <td>0.462</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-49</td>\n",
       "      <td>-1</td>\n",
       "      <td>11</td>\n",
       "      <td>-1</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>112.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.06</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.01</td>\n",
       "      <td>yes</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>6626.0</td>\n",
       "      <td>143318.0</td>\n",
       "      <td>0.046233</td>\n",
       "      <td>yes</td>\n",
       "      <td>3046825.0</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.01274</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.04611</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.06362</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.01125</td>\n",
       "      <td>G</td>\n",
       "      <td>chr1_3015134_3046826</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3188156</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>PRDM16</td>\n",
       "      <td>0.87043</td>\n",
       "      <td>1.3471</td>\n",
       "      <td>0.081747</td>\n",
       "      <td>5.9522</td>\n",
       "      <td>0.491</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-27</td>\n",
       "      <td>9</td>\n",
       "      <td>4</td>\n",
       "      <td>-1</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>301.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.06</td>\n",
       "      <td>0.18</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.04</td>\n",
       "      <td>0.08</td>\n",
       "      <td>yes</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1706.0</td>\n",
       "      <td>143300.0</td>\n",
       "      <td>0.011905</td>\n",
       "      <td>yes</td>\n",
       "      <td>3188155.0</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.00318</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.04323</td>\n",
       "      <td>0.18254</td>\n",
       "      <td>0.00298</td>\n",
       "      <td>0.07407</td>\n",
       "      <td>0.07975</td>\n",
       "      <td>G</td>\n",
       "      <td>chr1_3066509_3209504</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3199291</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>PRDM16</td>\n",
       "      <td>0.87043</td>\n",
       "      <td>1.3471</td>\n",
       "      <td>0.081747</td>\n",
       "      <td>5.9522</td>\n",
       "      <td>0.297</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>28</td>\n",
       "      <td>30</td>\n",
       "      <td>28</td>\n",
       "      <td>-23</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>derived</td>\n",
       "      <td>447.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.09</td>\n",
       "      <td>0.25</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.13</td>\n",
       "      <td>0.10</td>\n",
       "      <td>yes</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2943.0</td>\n",
       "      <td>143256.0</td>\n",
       "      <td>0.020544</td>\n",
       "      <td>yes</td>\n",
       "      <td>3199290.0</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.00955</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.12680</td>\n",
       "      <td>0.24603</td>\n",
       "      <td>0.00596</td>\n",
       "      <td>0.11111</td>\n",
       "      <td>0.09509</td>\n",
       "      <td>C</td>\n",
       "      <td>chr1_3066509_3209504</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3199727</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>PRDM16</td>\n",
       "      <td>0.87043</td>\n",
       "      <td>1.3471</td>\n",
       "      <td>0.081747</td>\n",
       "      <td>5.9522</td>\n",
       "      <td>0.436</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>8</td>\n",
       "      <td>44</td>\n",
       "      <td>-2</td>\n",
       "      <td>44</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>derived</td>\n",
       "      <td>447.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.09</td>\n",
       "      <td>0.25</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.13</td>\n",
       "      <td>0.10</td>\n",
       "      <td>yes</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2949.0</td>\n",
       "      <td>143216.0</td>\n",
       "      <td>0.020591</td>\n",
       "      <td>yes</td>\n",
       "      <td>3199726.0</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.00955</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.12680</td>\n",
       "      <td>0.24603</td>\n",
       "      <td>0.00596</td>\n",
       "      <td>0.11111</td>\n",
       "      <td>0.09509</td>\n",
       "      <td>C</td>\n",
       "      <td>chr1_3066509_3209504</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3209504</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>PRDM16</td>\n",
       "      <td>0.87043</td>\n",
       "      <td>1.3471</td>\n",
       "      <td>0.081747</td>\n",
       "      <td>5.9522</td>\n",
       "      <td>-0.438</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>20</td>\n",
       "      <td>1</td>\n",
       "      <td>-12</td>\n",
       "      <td>-1</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>444.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.09</td>\n",
       "      <td>0.24</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.13</td>\n",
       "      <td>0.09</td>\n",
       "      <td>yes</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2939.0</td>\n",
       "      <td>143312.0</td>\n",
       "      <td>0.020508</td>\n",
       "      <td>yes</td>\n",
       "      <td>3209503.0</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.00955</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.12680</td>\n",
       "      <td>0.24405</td>\n",
       "      <td>0.00596</td>\n",
       "      <td>0.11111</td>\n",
       "      <td>0.09305</td>\n",
       "      <td>G</td>\n",
       "      <td>chr1_3066509_3209504</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3210305</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>PRDM16</td>\n",
       "      <td>0.87043</td>\n",
       "      <td>1.3471</td>\n",
       "      <td>0.081747</td>\n",
       "      <td>5.9522</td>\n",
       "      <td>-0.379</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>2</td>\n",
       "      <td>7</td>\n",
       "      <td>-4</td>\n",
       "      <td>-21</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>derived</td>\n",
       "      <td>443.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.09</td>\n",
       "      <td>0.25</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.13</td>\n",
       "      <td>0.09</td>\n",
       "      <td>yes</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2939.0</td>\n",
       "      <td>143296.0</td>\n",
       "      <td>0.020510</td>\n",
       "      <td>yes</td>\n",
       "      <td>3210304.0</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.00955</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.12680</td>\n",
       "      <td>0.24405</td>\n",
       "      <td>0.00596</td>\n",
       "      <td>0.11111</td>\n",
       "      <td>0.09100</td>\n",
       "      <td>T</td>\n",
       "      <td>chr1_3166567_3212428</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3212428</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>PRDM16</td>\n",
       "      <td>0.87043</td>\n",
       "      <td>1.3471</td>\n",
       "      <td>0.081747</td>\n",
       "      <td>5.9522</td>\n",
       "      <td>-0.445</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.01</td>\n",
       "      <td>3</td>\n",
       "      <td>11</td>\n",
       "      <td>23</td>\n",
       "      <td>11</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>derived</td>\n",
       "      <td>452.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.09</td>\n",
       "      <td>0.26</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.13</td>\n",
       "      <td>0.09</td>\n",
       "      <td>yes</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>3291.0</td>\n",
       "      <td>143212.0</td>\n",
       "      <td>0.022980</td>\n",
       "      <td>yes</td>\n",
       "      <td>3212427.0</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.00955</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.13112</td>\n",
       "      <td>0.25595</td>\n",
       "      <td>0.00696</td>\n",
       "      <td>0.11111</td>\n",
       "      <td>0.08384</td>\n",
       "      <td>A</td>\n",
       "      <td>chr1_3166567_3212428</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3240652</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>PRDM16</td>\n",
       "      <td>0.87043</td>\n",
       "      <td>1.3471</td>\n",
       "      <td>0.081747</td>\n",
       "      <td>5.9522</td>\n",
       "      <td>-0.479</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>10</td>\n",
       "      <td>50</td>\n",
       "      <td>29</td>\n",
       "      <td>9</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>derived</td>\n",
       "      <td>26.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.02</td>\n",
       "      <td>yes</td>\n",
       "      <td>0.004873</td>\n",
       "      <td>294.0</td>\n",
       "      <td>143304.0</td>\n",
       "      <td>0.002052</td>\n",
       "      <td>yes</td>\n",
       "      <td>3240651.0</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.00318</td>\n",
       "      <td>0.00397</td>\n",
       "      <td>0.00144</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.18519</td>\n",
       "      <td>0.02045</td>\n",
       "      <td>T</td>\n",
       "      <td>chr1_3195401_3250112</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3240725</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Other</td>\n",
       "      <td>PRDM16</td>\n",
       "      <td>0.87043</td>\n",
       "      <td>1.3471</td>\n",
       "      <td>0.081747</td>\n",
       "      <td>5.9522</td>\n",
       "      <td>-1.193</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-39</td>\n",
       "      <td>29</td>\n",
       "      <td>40</td>\n",
       "      <td>-2</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>derived</td>\n",
       "      <td>394.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.08</td>\n",
       "      <td>0.16</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.14</td>\n",
       "      <td>0.12</td>\n",
       "      <td>yes</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2936.0</td>\n",
       "      <td>142852.0</td>\n",
       "      <td>0.020553</td>\n",
       "      <td>yes</td>\n",
       "      <td>3240724.0</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.00955</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.13833</td>\n",
       "      <td>0.16171</td>\n",
       "      <td>0.00596</td>\n",
       "      <td>0.11111</td>\n",
       "      <td>0.11656</td>\n",
       "      <td>T</td>\n",
       "      <td>chr1_3240724_3259500</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chrom      pos ref_allele alt_allele variant_type altai_gt chagyrskaya_gt  \\\n",
       "0  chr1  3046711          T          G          snv      0/1            0/0   \n",
       "1  chr1  3046826          G          T          snv      0/1            0/0   \n",
       "2  chr1  3188156          G          A          snv      0/1            1/1   \n",
       "3  chr1  3199291          C          T          snv      0/1            1/1   \n",
       "4  chr1  3199727          C          T          snv      0/1            1/1   \n",
       "5  chr1  3209504          G          A          snv      0/1            1/1   \n",
       "6  chr1  3210305          T          A          snv      0/1            1/1   \n",
       "7  chr1  3212428          A          C          snv      0/1            1/1   \n",
       "8  chr1  3240652          C          T          snv      0/0            0/0   \n",
       "9  chr1  3240725          T          C          snv      0/1            0/0   \n",
       "\n",
       "  denisovan_gt vindija_gt  altai_gt_boolean  chagyrskaya_gt_boolean  \\\n",
       "0          0/0        1/1              True                   False   \n",
       "1          0/0        1/1              True                   False   \n",
       "2          0/0        1/1              True                    True   \n",
       "3          0/0        1/1              True                    True   \n",
       "4          0/0        1/1              True                    True   \n",
       "5          0/0        1/1              True                    True   \n",
       "6          0/0        1/1              True                    True   \n",
       "7          0/0        1/1              True                    True   \n",
       "8          1/1        0/0             False                   False   \n",
       "9          0/0        0/1              True                   False   \n",
       "\n",
       "   denisovan_gt_boolean  vindija_gt_boolean distribution annotation   mis_oe  \\\n",
       "0                 False                True        Other     PRDM16  0.87043   \n",
       "1                 False                True        Other     PRDM16  0.87043   \n",
       "2                 False                True  Neanderthal     PRDM16  0.87043   \n",
       "3                 False                True  Neanderthal     PRDM16  0.87043   \n",
       "4                 False                True  Neanderthal     PRDM16  0.87043   \n",
       "5                 False                True  Neanderthal     PRDM16  0.87043   \n",
       "6                 False                True  Neanderthal     PRDM16  0.87043   \n",
       "7                 False                True  Neanderthal     PRDM16  0.87043   \n",
       "8                  True               False    Denisovan     PRDM16  0.87043   \n",
       "9                 False                True        Other     PRDM16  0.87043   \n",
       "\n",
       "    mis_z    lof_oe   lof_z  phyloP  ag_delta  al_delta  dg_delta  dl_delta  \\\n",
       "0  1.3471  0.081747  5.9522   0.379       0.0       0.0       0.0      0.00   \n",
       "1  1.3471  0.081747  5.9522   0.462       0.0       0.0       0.0      0.00   \n",
       "2  1.3471  0.081747  5.9522   0.491       0.0       0.0       0.0      0.00   \n",
       "3  1.3471  0.081747  5.9522   0.297       0.0       0.0       0.0      0.00   \n",
       "4  1.3471  0.081747  5.9522   0.436       0.0       0.0       0.0      0.00   \n",
       "5  1.3471  0.081747  5.9522  -0.438       0.0       0.0       0.0      0.00   \n",
       "6  1.3471  0.081747  5.9522  -0.379       0.0       0.0       0.0      0.00   \n",
       "7  1.3471  0.081747  5.9522  -0.445       0.0       0.0       0.0      0.01   \n",
       "8  1.3471  0.081747  5.9522  -0.479       0.0       0.0       0.0      0.00   \n",
       "9  1.3471  0.081747  5.9522  -1.193       0.0       0.0       0.0      0.00   \n",
       "\n",
       "   delta_max  ag_pos  al_pos  dg_pos  dl_pos ancestral_allele temp  anc_dev  \\\n",
       "0       0.00      -4      35     -24      19                T    T  derived   \n",
       "1       0.00     -49      -1      11      -1                G    G  derived   \n",
       "2       0.00     -27       9       4      -1                G    G  derived   \n",
       "3       0.00      28      30      28     -23                C    C  derived   \n",
       "4       0.00       8      44      -2      44                C    C  derived   \n",
       "5       0.00      20       1     -12      -1                G    G  derived   \n",
       "6       0.00       2       7      -4     -21                T    T  derived   \n",
       "7       0.01       3      11      23      11                A    A  derived   \n",
       "8       0.00      10      50      29       9                C    C  derived   \n",
       "9       0.00     -39      29      40      -2                T    T  derived   \n",
       "\n",
       "   1KG_allele_count  1KG_allele_number  1KG_allele_frequency  1KG_EAS_AF  \\\n",
       "0             112.0             5096.0                  0.02        0.00   \n",
       "1             112.0             5096.0                  0.02        0.00   \n",
       "2             301.0             5096.0                  0.06        0.18   \n",
       "3             447.0             5096.0                  0.09        0.25   \n",
       "4             447.0             5096.0                  0.09        0.25   \n",
       "5             444.0             5096.0                  0.09        0.24   \n",
       "6             443.0             5096.0                  0.09        0.25   \n",
       "7             452.0             5096.0                  0.09        0.26   \n",
       "8              26.0             5096.0                  0.01        0.00   \n",
       "9             394.0             5096.0                  0.08        0.16   \n",
       "\n",
       "   1KG_EUR_AF  1KG_AFR_AF  1KG_AMR_AF  1KG_SAS_AF present_in_1KG  \\\n",
       "0        0.06         0.0        0.05        0.01            yes   \n",
       "1        0.06         0.0        0.05        0.01            yes   \n",
       "2        0.00         0.0        0.04        0.08            yes   \n",
       "3        0.01         0.0        0.13        0.10            yes   \n",
       "4        0.01         0.0        0.13        0.10            yes   \n",
       "5        0.01         0.0        0.13        0.09            yes   \n",
       "6        0.01         0.0        0.13        0.09            yes   \n",
       "7        0.01         0.0        0.13        0.09            yes   \n",
       "8        0.00         0.0        0.00        0.02            yes   \n",
       "9        0.01         0.0        0.14        0.12            yes   \n",
       "\n",
       "   1KG_non_ASW_AFR_AF  gnomAD_allele_count  gnomAD_allele_number  \\\n",
       "0            0.000000               6632.0              143094.0   \n",
       "1            0.000000               6626.0              143318.0   \n",
       "2            0.000000               1706.0              143300.0   \n",
       "3            0.000000               2943.0              143256.0   \n",
       "4            0.000000               2949.0              143216.0   \n",
       "5            0.000000               2939.0              143312.0   \n",
       "6            0.000000               2939.0              143296.0   \n",
       "7            0.000000               3291.0              143212.0   \n",
       "8            0.004873                294.0              143304.0   \n",
       "9            0.000000               2936.0              142852.0   \n",
       "\n",
       "   gnomAD_allele_frequency present_in_gnomAD      start  \\\n",
       "0                 0.046347               yes  3046710.0   \n",
       "1                 0.046233               yes  3046825.0   \n",
       "2                 0.011905               yes  3188155.0   \n",
       "3                 0.020544               yes  3199290.0   \n",
       "4                 0.020591               yes  3199726.0   \n",
       "5                 0.020508               yes  3209503.0   \n",
       "6                 0.020510               yes  3210304.0   \n",
       "7                 0.022980               yes  3212427.0   \n",
       "8                 0.002052               yes  3240651.0   \n",
       "9                 0.020553               yes  3240724.0   \n",
       "\n",
       "  Vernot_ancestral_allele Vernot_derived_allele  \\\n",
       "0                       T                     G   \n",
       "1                       G                     T   \n",
       "2                       G                     A   \n",
       "3                       C                     T   \n",
       "4                       C                     T   \n",
       "5                       G                     A   \n",
       "6                       T                     A   \n",
       "7                       A                     C   \n",
       "8                       C                     T   \n",
       "9                       T                     C   \n",
       "\n",
       "   Vernot_ancestral_derived_code  Vernot_AFA_AF  Vernot_AFR_AF  Vernot_AMR_AF  \\\n",
       "0                            1.0        0.01274        0.00000        0.04611   \n",
       "1                            1.0        0.01274        0.00000        0.04611   \n",
       "2                            1.0        0.00318        0.00000        0.04323   \n",
       "3                            1.0        0.00955        0.00000        0.12680   \n",
       "4                            1.0        0.00955        0.00000        0.12680   \n",
       "5                            1.0        0.00955        0.00000        0.12680   \n",
       "6                            1.0        0.00955        0.00000        0.12680   \n",
       "7                            1.0        0.00955        0.00000        0.13112   \n",
       "8                            1.0        0.00318        0.00397        0.00144   \n",
       "9                            1.0        0.00955        0.00000        0.13833   \n",
       "\n",
       "   Vernot_EAS_AF  Vernot_EUR_AF  Vernot_PNG_AF  Vernot_SAS_AF  \\\n",
       "0        0.00000        0.06362        0.00000        0.01125   \n",
       "1        0.00000        0.06362        0.00000        0.01125   \n",
       "2        0.18254        0.00298        0.07407        0.07975   \n",
       "3        0.24603        0.00596        0.11111        0.09509   \n",
       "4        0.24603        0.00596        0.11111        0.09509   \n",
       "5        0.24405        0.00596        0.11111        0.09305   \n",
       "6        0.24405        0.00596        0.11111        0.09100   \n",
       "7        0.25595        0.00696        0.11111        0.08384   \n",
       "8        0.00000        0.00000        0.18519        0.02045   \n",
       "9        0.16171        0.00596        0.11111        0.11656   \n",
       "\n",
       "  Vernot_Denisovan_base  Vernot_haplotype_tag Vernot_Neanderthal_base_1  \\\n",
       "0                     T  chr1_3015134_3046826                         T   \n",
       "1                     G  chr1_3015134_3046826                         G   \n",
       "2                     G  chr1_3066509_3209504                         G   \n",
       "3                     C  chr1_3066509_3209504                         C   \n",
       "4                     C  chr1_3066509_3209504                         C   \n",
       "5                     G  chr1_3066509_3209504                         G   \n",
       "6                     T  chr1_3166567_3212428                         T   \n",
       "7                     A  chr1_3166567_3212428                         A   \n",
       "8                     T  chr1_3195401_3250112                         C   \n",
       "9                     T  chr1_3240724_3259500                         T   \n",
       "\n",
       "  Vernot_Neanderthal_base_2  \n",
       "0                         G  \n",
       "1                         T  \n",
       "2                         A  \n",
       "3                         T  \n",
       "4                         T  \n",
       "5                         A  \n",
       "6                         A  \n",
       "7                         C  \n",
       "8                       NaN  \n",
       "9                         C  "
      ]
     },
     "execution_count": 607,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "introgressed_ref_tag_header = ['chrom','pos','ref_allele','alt_allele','variant_type','altai_gt','chagyrskaya_gt','denisovan_gt','vindija_gt','altai_gt_boolean','chagyrskaya_gt_boolean','denisovan_gt_boolean','vindija_gt_boolean','distribution','annotation','mis_oe','mis_z','lof_oe','lof_z','phyloP','ag_delta','al_delta','dg_delta','dl_delta','delta_max','ag_pos','al_pos','dg_pos','dl_pos','ancestral_allele','temp','anc_dev','1KG_allele_count','1KG_allele_number','1KG_allele_frequency','1KG_EAS_AF','1KG_EUR_AF','1KG_AFR_AF','1KG_AMR_AF','1KG_SAS_AF','present_in_1KG','1KG_non_ASW_AFR_AF','gnomAD_allele_count','gnomAD_allele_number','gnomAD_allele_frequency','present_in_gnomAD','start','Vernot_ancestral_allele','Vernot_derived_allele','Vernot_ancestral_derived_code','Vernot_AFA_AF','Vernot_AFR_AF','Vernot_AMR_AF','Vernot_EAS_AF','Vernot_EUR_AF','Vernot_PNG_AF','Vernot_SAS_AF','Vernot_Denisovan_base','Vernot_haplotype_tag','Vernot_Neanderthal_base_1','Vernot_Neanderthal_base_2']\n",
    "introgressed_ref_tag = pd.read_csv('../introgressed_refs/introgressed_ref_tag.txt', sep = '\\t', names = introgressed_ref_tag_header)\n",
    "introgressed_ref_tag.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 608,
   "id": "f3914c30",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "7977"
      ]
     },
     "execution_count": 608,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(introgressed_ref_tag)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 609,
   "id": "74495a7c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "26"
      ]
     },
     "execution_count": 609,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SAV_introgressed_ref_tag = introgressed_ref_tag[introgressed_ref_tag['delta_max']>=0.2]\n",
    "len(SAV_introgressed_ref_tag)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 610,
   "id": "77bcb72f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref_allele</th>\n",
       "      <th>alt_allele</th>\n",
       "      <th>variant_type</th>\n",
       "      <th>altai_gt</th>\n",
       "      <th>chagyrskaya_gt</th>\n",
       "      <th>denisovan_gt</th>\n",
       "      <th>vindija_gt</th>\n",
       "      <th>altai_gt_boolean</th>\n",
       "      <th>chagyrskaya_gt_boolean</th>\n",
       "      <th>denisovan_gt_boolean</th>\n",
       "      <th>vindija_gt_boolean</th>\n",
       "      <th>distribution</th>\n",
       "      <th>annotation</th>\n",
       "      <th>mis_oe</th>\n",
       "      <th>mis_z</th>\n",
       "      <th>lof_oe</th>\n",
       "      <th>lof_z</th>\n",
       "      <th>phyloP</th>\n",
       "      <th>ag_delta</th>\n",
       "      <th>al_delta</th>\n",
       "      <th>dg_delta</th>\n",
       "      <th>dl_delta</th>\n",
       "      <th>delta_max</th>\n",
       "      <th>ag_pos</th>\n",
       "      <th>al_pos</th>\n",
       "      <th>dg_pos</th>\n",
       "      <th>dl_pos</th>\n",
       "      <th>ancestral_allele</th>\n",
       "      <th>temp</th>\n",
       "      <th>anc_dev</th>\n",
       "      <th>1KG_allele_count</th>\n",
       "      <th>1KG_allele_number</th>\n",
       "      <th>1KG_allele_frequency</th>\n",
       "      <th>1KG_EAS_AF</th>\n",
       "      <th>1KG_EUR_AF</th>\n",
       "      <th>1KG_AFR_AF</th>\n",
       "      <th>1KG_AMR_AF</th>\n",
       "      <th>1KG_SAS_AF</th>\n",
       "      <th>present_in_1KG</th>\n",
       "      <th>1KG_non_ASW_AFR_AF</th>\n",
       "      <th>gnomAD_allele_count</th>\n",
       "      <th>gnomAD_allele_number</th>\n",
       "      <th>gnomAD_allele_frequency</th>\n",
       "      <th>present_in_gnomAD</th>\n",
       "      <th>start</th>\n",
       "      <th>Vernot_ancestral_allele</th>\n",
       "      <th>Vernot_derived_allele</th>\n",
       "      <th>Vernot_ancestral_derived_code</th>\n",
       "      <th>Vernot_AFA_AF</th>\n",
       "      <th>Vernot_AFR_AF</th>\n",
       "      <th>Vernot_AMR_AF</th>\n",
       "      <th>Vernot_EAS_AF</th>\n",
       "      <th>Vernot_EUR_AF</th>\n",
       "      <th>Vernot_PNG_AF</th>\n",
       "      <th>Vernot_SAS_AF</th>\n",
       "      <th>Vernot_Denisovan_base</th>\n",
       "      <th>Vernot_haplotype_tag</th>\n",
       "      <th>Vernot_Neanderthal_base_1</th>\n",
       "      <th>Vernot_Neanderthal_base_2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>295</th>\n",
       "      <td>chr1</td>\n",
       "      <td>52820680</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Other</td>\n",
       "      <td>CC2D1B</td>\n",
       "      <td>1.12620</td>\n",
       "      <td>-1.00890</td>\n",
       "      <td>0.694430</td>\n",
       "      <td>2.038800</td>\n",
       "      <td>-0.140</td>\n",
       "      <td>0.23</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.23</td>\n",
       "      <td>-22</td>\n",
       "      <td>-40</td>\n",
       "      <td>-22</td>\n",
       "      <td>34</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>derived</td>\n",
       "      <td>90.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.03</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.05</td>\n",
       "      <td>yes</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>443.0</td>\n",
       "      <td>143266.0</td>\n",
       "      <td>0.003092</td>\n",
       "      <td>yes</td>\n",
       "      <td>52820679.0</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.00637</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.02877</td>\n",
       "      <td>0.00497</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.05521</td>\n",
       "      <td>C</td>\n",
       "      <td>chr1_52325475_53062486</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>476</th>\n",
       "      <td>chr1</td>\n",
       "      <td>152944501</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Other</td>\n",
       "      <td>SPRR4</td>\n",
       "      <td>1.49550</td>\n",
       "      <td>-1.10600</td>\n",
       "      <td>0.714380</td>\n",
       "      <td>0.442850</td>\n",
       "      <td>0.383</td>\n",
       "      <td>0.23</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.23</td>\n",
       "      <td>13</td>\n",
       "      <td>2</td>\n",
       "      <td>12</td>\n",
       "      <td>-1</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>derived</td>\n",
       "      <td>2.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>yes</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>4.0</td>\n",
       "      <td>143152.0</td>\n",
       "      <td>0.000028</td>\n",
       "      <td>yes</td>\n",
       "      <td>152944500.0</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00198</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>A</td>\n",
       "      <td>chr1_152942848_153033467</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>557</th>\n",
       "      <td>chr1</td>\n",
       "      <td>183750131</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>Altai</td>\n",
       "      <td>RGL1</td>\n",
       "      <td>0.70413</td>\n",
       "      <td>2.22380</td>\n",
       "      <td>0.177530</td>\n",
       "      <td>5.116100</td>\n",
       "      <td>0.655</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.26</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.26</td>\n",
       "      <td>-7</td>\n",
       "      <td>-11</td>\n",
       "      <td>-7</td>\n",
       "      <td>-11</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>derived</td>\n",
       "      <td>50.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>yes</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>194.0</td>\n",
       "      <td>143252.0</td>\n",
       "      <td>0.001354</td>\n",
       "      <td>yes</td>\n",
       "      <td>183750130.0</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.04861</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.01852</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>C</td>\n",
       "      <td>chr1_183411087_183827254</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>800</th>\n",
       "      <td>chr1</td>\n",
       "      <td>216945756</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>ESRRG</td>\n",
       "      <td>0.56557</td>\n",
       "      <td>2.55570</td>\n",
       "      <td>0.283790</td>\n",
       "      <td>3.051600</td>\n",
       "      <td>0.655</td>\n",
       "      <td>0.20</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.20</td>\n",
       "      <td>-27</td>\n",
       "      <td>47</td>\n",
       "      <td>-27</td>\n",
       "      <td>48</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>derived</td>\n",
       "      <td>169.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.03</td>\n",
       "      <td>0.14</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.02</td>\n",
       "      <td>yes</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>746.0</td>\n",
       "      <td>143174.0</td>\n",
       "      <td>0.005210</td>\n",
       "      <td>yes</td>\n",
       "      <td>216945755.0</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.01153</td>\n",
       "      <td>0.13988</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.01840</td>\n",
       "      <td>C</td>\n",
       "      <td>chr1_216944862_216964892</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>998</th>\n",
       "      <td>chr10</td>\n",
       "      <td>17373518</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>ST8SIA6</td>\n",
       "      <td>1.08980</td>\n",
       "      <td>-0.43129</td>\n",
       "      <td>1.008600</td>\n",
       "      <td>-0.029733</td>\n",
       "      <td>0.533</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.22</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.22</td>\n",
       "      <td>33</td>\n",
       "      <td>-19</td>\n",
       "      <td>30</td>\n",
       "      <td>-19</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>derived</td>\n",
       "      <td>56.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.04</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.00</td>\n",
       "      <td>yes</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2946.0</td>\n",
       "      <td>143262.0</td>\n",
       "      <td>0.020564</td>\n",
       "      <td>yes</td>\n",
       "      <td>17373517.0</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.01274</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.01729</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.03479</td>\n",
       "      <td>0.01852</td>\n",
       "      <td>0.00307</td>\n",
       "      <td>T</td>\n",
       "      <td>chr10_17338795_17417730</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1617</th>\n",
       "      <td>chr11</td>\n",
       "      <td>84191111</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>DLG2</td>\n",
       "      <td>0.74745</td>\n",
       "      <td>2.07620</td>\n",
       "      <td>0.212280</td>\n",
       "      <td>5.712300</td>\n",
       "      <td>-0.959</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.40</td>\n",
       "      <td>0.40</td>\n",
       "      <td>1</td>\n",
       "      <td>29</td>\n",
       "      <td>1</td>\n",
       "      <td>-1</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>derived</td>\n",
       "      <td>12.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.01</td>\n",
       "      <td>yes</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>38.0</td>\n",
       "      <td>143298.0</td>\n",
       "      <td>0.000265</td>\n",
       "      <td>yes</td>\n",
       "      <td>84191110.0</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.01227</td>\n",
       "      <td>T</td>\n",
       "      <td>chr11_84051845_84211811</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1669</th>\n",
       "      <td>chr11</td>\n",
       "      <td>99345359</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>CNTN5</td>\n",
       "      <td>1.03930</td>\n",
       "      <td>-0.33557</td>\n",
       "      <td>0.327200</td>\n",
       "      <td>4.624200</td>\n",
       "      <td>-0.229</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.67</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.67</td>\n",
       "      <td>-4</td>\n",
       "      <td>48</td>\n",
       "      <td>-5</td>\n",
       "      <td>37</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>derived</td>\n",
       "      <td>177.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.03</td>\n",
       "      <td>0.11</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.07</td>\n",
       "      <td>yes</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>551.0</td>\n",
       "      <td>143130.0</td>\n",
       "      <td>0.003850</td>\n",
       "      <td>yes</td>\n",
       "      <td>99345358.0</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.00318</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.10615</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.11111</td>\n",
       "      <td>0.06748</td>\n",
       "      <td>C</td>\n",
       "      <td>chr11_99309226_99359652</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3230</th>\n",
       "      <td>chr17</td>\n",
       "      <td>14109360</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>COX10</td>\n",
       "      <td>1.03160</td>\n",
       "      <td>-0.18011</td>\n",
       "      <td>0.290420</td>\n",
       "      <td>2.728300</td>\n",
       "      <td>0.338</td>\n",
       "      <td>0.20</td>\n",
       "      <td>0.04</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.20</td>\n",
       "      <td>1</td>\n",
       "      <td>38</td>\n",
       "      <td>50</td>\n",
       "      <td>-23</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>derived</td>\n",
       "      <td>155.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.03</td>\n",
       "      <td>0.08</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.08</td>\n",
       "      <td>0.01</td>\n",
       "      <td>yes</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1030.0</td>\n",
       "      <td>143224.0</td>\n",
       "      <td>0.007192</td>\n",
       "      <td>yes</td>\n",
       "      <td>14109359.0</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.00637</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.08501</td>\n",
       "      <td>0.08333</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.16667</td>\n",
       "      <td>0.00818</td>\n",
       "      <td>G</td>\n",
       "      <td>chr17_13902596_14111504</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3473</th>\n",
       "      <td>chr18</td>\n",
       "      <td>60003587</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>TNFRSF11A</td>\n",
       "      <td>0.80508</td>\n",
       "      <td>1.28940</td>\n",
       "      <td>0.330780</td>\n",
       "      <td>3.049700</td>\n",
       "      <td>0.650</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.31</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.28</td>\n",
       "      <td>0.31</td>\n",
       "      <td>3</td>\n",
       "      <td>-40</td>\n",
       "      <td>3</td>\n",
       "      <td>33</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>444.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.09</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.25</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.12</td>\n",
       "      <td>0.10</td>\n",
       "      <td>yes</td>\n",
       "      <td>0.003899</td>\n",
       "      <td>21727.0</td>\n",
       "      <td>143202.0</td>\n",
       "      <td>0.151723</td>\n",
       "      <td>yes</td>\n",
       "      <td>60003586.0</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.01592</td>\n",
       "      <td>0.00397</td>\n",
       "      <td>0.11671</td>\n",
       "      <td>0.00198</td>\n",
       "      <td>0.24950</td>\n",
       "      <td>0.09259</td>\n",
       "      <td>0.09714</td>\n",
       "      <td>A</td>\n",
       "      <td>chr18_59569521_60089689</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3541</th>\n",
       "      <td>chr19</td>\n",
       "      <td>33410289</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>CEP89</td>\n",
       "      <td>0.93498</td>\n",
       "      <td>0.47493</td>\n",
       "      <td>0.939730</td>\n",
       "      <td>0.364400</td>\n",
       "      <td>-0.642</td>\n",
       "      <td>0.47</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.71</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.71</td>\n",
       "      <td>41</td>\n",
       "      <td>-29</td>\n",
       "      <td>5</td>\n",
       "      <td>-29</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>ancestral</td>\n",
       "      <td>4744.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.93</td>\n",
       "      <td>0.90</td>\n",
       "      <td>0.86</td>\n",
       "      <td>0.99</td>\n",
       "      <td>0.93</td>\n",
       "      <td>0.96</td>\n",
       "      <td>yes</td>\n",
       "      <td>0.999025</td>\n",
       "      <td>129412.0</td>\n",
       "      <td>143248.0</td>\n",
       "      <td>0.903412</td>\n",
       "      <td>yes</td>\n",
       "      <td>33410288.0</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.03503</td>\n",
       "      <td>0.00099</td>\n",
       "      <td>0.06916</td>\n",
       "      <td>0.09821</td>\n",
       "      <td>0.14612</td>\n",
       "      <td>0.07407</td>\n",
       "      <td>0.04090</td>\n",
       "      <td>C</td>\n",
       "      <td>chr19_33040186_33456652</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4564</th>\n",
       "      <td>chr20</td>\n",
       "      <td>62224595</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>GMEB2</td>\n",
       "      <td>0.66048</td>\n",
       "      <td>2.20200</td>\n",
       "      <td>0.049787</td>\n",
       "      <td>3.946200</td>\n",
       "      <td>-0.837</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.41</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.41</td>\n",
       "      <td>15</td>\n",
       "      <td>-44</td>\n",
       "      <td>2</td>\n",
       "      <td>-2</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>1138.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.22</td>\n",
       "      <td>0.51</td>\n",
       "      <td>0.21</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.11</td>\n",
       "      <td>0.31</td>\n",
       "      <td>yes</td>\n",
       "      <td>0.000975</td>\n",
       "      <td>23692.0</td>\n",
       "      <td>143104.0</td>\n",
       "      <td>0.165558</td>\n",
       "      <td>yes</td>\n",
       "      <td>62224594.0</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.03503</td>\n",
       "      <td>0.00099</td>\n",
       "      <td>0.10663</td>\n",
       "      <td>0.51488</td>\n",
       "      <td>0.21173</td>\n",
       "      <td>0.03704</td>\n",
       "      <td>0.30982</td>\n",
       "      <td>G</td>\n",
       "      <td>chr20_62195671_62229244</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5067</th>\n",
       "      <td>chr3</td>\n",
       "      <td>52486376</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>TNNC1</td>\n",
       "      <td>0.49480</td>\n",
       "      <td>1.76820</td>\n",
       "      <td>0.125160</td>\n",
       "      <td>2.291500</td>\n",
       "      <td>0.459</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.34</td>\n",
       "      <td>0.34</td>\n",
       "      <td>-23</td>\n",
       "      <td>2</td>\n",
       "      <td>25</td>\n",
       "      <td>-1</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>derived</td>\n",
       "      <td>40.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.03</td>\n",
       "      <td>yes</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>494.0</td>\n",
       "      <td>143210.0</td>\n",
       "      <td>0.003449</td>\n",
       "      <td>yes</td>\n",
       "      <td>52486375.0</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.00318</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00288</td>\n",
       "      <td>0.00794</td>\n",
       "      <td>0.00298</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.02658</td>\n",
       "      <td>T</td>\n",
       "      <td>chr3_52214838_52516460</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5237</th>\n",
       "      <td>chr4</td>\n",
       "      <td>38805942</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>TLR1</td>\n",
       "      <td>1.09920</td>\n",
       "      <td>-0.70157</td>\n",
       "      <td>1.411200</td>\n",
       "      <td>-1.727200</td>\n",
       "      <td>-0.219</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.38</td>\n",
       "      <td>0.38</td>\n",
       "      <td>27</td>\n",
       "      <td>-15</td>\n",
       "      <td>1</td>\n",
       "      <td>-15</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>911.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.18</td>\n",
       "      <td>0.39</td>\n",
       "      <td>0.22</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.15</td>\n",
       "      <td>0.16</td>\n",
       "      <td>yes</td>\n",
       "      <td>0.001949</td>\n",
       "      <td>19931.0</td>\n",
       "      <td>143224.0</td>\n",
       "      <td>0.139160</td>\n",
       "      <td>yes</td>\n",
       "      <td>38805941.0</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.04459</td>\n",
       "      <td>0.00198</td>\n",
       "      <td>0.14986</td>\n",
       "      <td>0.39484</td>\n",
       "      <td>0.22068</td>\n",
       "      <td>0.05556</td>\n",
       "      <td>0.16360</td>\n",
       "      <td>G</td>\n",
       "      <td>chr4_38294200_38857896</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5318</th>\n",
       "      <td>chr4</td>\n",
       "      <td>100460531</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>C4orf17</td>\n",
       "      <td>0.87519</td>\n",
       "      <td>0.60341</td>\n",
       "      <td>0.898160</td>\n",
       "      <td>0.359040</td>\n",
       "      <td>0.533</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.35</td>\n",
       "      <td>0.35</td>\n",
       "      <td>-4</td>\n",
       "      <td>-40</td>\n",
       "      <td>-1</td>\n",
       "      <td>-4</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>derived</td>\n",
       "      <td>50.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.04</td>\n",
       "      <td>0.00</td>\n",
       "      <td>yes</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>3317.0</td>\n",
       "      <td>143342.0</td>\n",
       "      <td>0.023141</td>\n",
       "      <td>yes</td>\n",
       "      <td>100460530.0</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.03602</td>\n",
       "      <td>0.00099</td>\n",
       "      <td>0.02087</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00204</td>\n",
       "      <td>G</td>\n",
       "      <td>chr4_100349203_100638613</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5437</th>\n",
       "      <td>chr4</td>\n",
       "      <td>135122662</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Other</td>\n",
       "      <td>PABPC4L</td>\n",
       "      <td>0.73213</td>\n",
       "      <td>1.47170</td>\n",
       "      <td>0.557620</td>\n",
       "      <td>1.452400</td>\n",
       "      <td>0.561</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.80</td>\n",
       "      <td>0.08</td>\n",
       "      <td>0.80</td>\n",
       "      <td>2</td>\n",
       "      <td>40</td>\n",
       "      <td>2</td>\n",
       "      <td>-2</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>yes</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.0</td>\n",
       "      <td>143316.0</td>\n",
       "      <td>0.000014</td>\n",
       "      <td>yes</td>\n",
       "      <td>135122661.0</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00099</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>G</td>\n",
       "      <td>chr4_135066013_135400700</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5765</th>\n",
       "      <td>chr5</td>\n",
       "      <td>118669326</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>TNFAIP8</td>\n",
       "      <td>0.87632</td>\n",
       "      <td>0.44541</td>\n",
       "      <td>0.204890</td>\n",
       "      <td>1.627700</td>\n",
       "      <td>0.650</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.22</td>\n",
       "      <td>0.22</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>yes</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>118669325.0</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.11111</td>\n",
       "      <td>0.00102</td>\n",
       "      <td>C</td>\n",
       "      <td>chr5_118661135_118933833</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5785</th>\n",
       "      <td>chr5</td>\n",
       "      <td>126989007</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>Altai</td>\n",
       "      <td>CTXN3</td>\n",
       "      <td>0.91784</td>\n",
       "      <td>0.19514</td>\n",
       "      <td>0.528190</td>\n",
       "      <td>0.601570</td>\n",
       "      <td>0.528</td>\n",
       "      <td>0.34</td>\n",
       "      <td>0.91</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.91</td>\n",
       "      <td>14</td>\n",
       "      <td>2</td>\n",
       "      <td>-34</td>\n",
       "      <td>2</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>derived</td>\n",
       "      <td>2.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>yes</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>10.0</td>\n",
       "      <td>143290.0</td>\n",
       "      <td>0.000070</td>\n",
       "      <td>yes</td>\n",
       "      <td>126989006.0</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00204</td>\n",
       "      <td>A</td>\n",
       "      <td>chr5_126989006_127537918</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5941</th>\n",
       "      <td>chr5</td>\n",
       "      <td>148430560</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Other</td>\n",
       "      <td>SH3TC2</td>\n",
       "      <td>1.07760</td>\n",
       "      <td>-0.71307</td>\n",
       "      <td>0.608270</td>\n",
       "      <td>2.673700</td>\n",
       "      <td>-1.235</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.32</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.32</td>\n",
       "      <td>-6</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>-41</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>derived</td>\n",
       "      <td>81.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.07</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.00</td>\n",
       "      <td>yes</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>214.0</td>\n",
       "      <td>143150.0</td>\n",
       "      <td>0.001495</td>\n",
       "      <td>yes</td>\n",
       "      <td>148430559.0</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.00318</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00720</td>\n",
       "      <td>0.07242</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.03704</td>\n",
       "      <td>0.00204</td>\n",
       "      <td>A</td>\n",
       "      <td>chr5_148339333_148539017</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6252</th>\n",
       "      <td>chr6</td>\n",
       "      <td>51889358</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>PKHD1</td>\n",
       "      <td>1.06000</td>\n",
       "      <td>-0.98178</td>\n",
       "      <td>0.580270</td>\n",
       "      <td>5.281600</td>\n",
       "      <td>-1.093</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.42</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.42</td>\n",
       "      <td>31</td>\n",
       "      <td>14</td>\n",
       "      <td>1</td>\n",
       "      <td>10</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>derived</td>\n",
       "      <td>114.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.08</td>\n",
       "      <td>yes</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2957.0</td>\n",
       "      <td>143284.0</td>\n",
       "      <td>0.020637</td>\n",
       "      <td>yes</td>\n",
       "      <td>51889357.0</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.00318</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.01009</td>\n",
       "      <td>0.00694</td>\n",
       "      <td>0.02286</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.07771</td>\n",
       "      <td>T/C</td>\n",
       "      <td>chr6_51841815_51972144</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6253</th>\n",
       "      <td>chr6</td>\n",
       "      <td>52138226</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>Altai</td>\n",
       "      <td>MCM3</td>\n",
       "      <td>0.89914</td>\n",
       "      <td>0.80451</td>\n",
       "      <td>0.535230</td>\n",
       "      <td>2.884000</td>\n",
       "      <td>0.563</td>\n",
       "      <td>0.21</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.21</td>\n",
       "      <td>-7</td>\n",
       "      <td>-40</td>\n",
       "      <td>-7</td>\n",
       "      <td>3</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>derived</td>\n",
       "      <td>782.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.15</td>\n",
       "      <td>0.24</td>\n",
       "      <td>0.17</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.09</td>\n",
       "      <td>0.29</td>\n",
       "      <td>yes</td>\n",
       "      <td>0.001949</td>\n",
       "      <td>17455.0</td>\n",
       "      <td>143064.0</td>\n",
       "      <td>0.122008</td>\n",
       "      <td>yes</td>\n",
       "      <td>52138225.0</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.01274</td>\n",
       "      <td>0.00198</td>\n",
       "      <td>0.09366</td>\n",
       "      <td>0.24206</td>\n",
       "      <td>0.16600</td>\n",
       "      <td>0.09259</td>\n",
       "      <td>0.29448</td>\n",
       "      <td>C</td>\n",
       "      <td>chr6_52138225_52183126</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6589</th>\n",
       "      <td>chr6</td>\n",
       "      <td>119524266</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Other</td>\n",
       "      <td>MAN1A1</td>\n",
       "      <td>0.77221</td>\n",
       "      <td>1.51360</td>\n",
       "      <td>0.255850</td>\n",
       "      <td>3.856000</td>\n",
       "      <td>0.430</td>\n",
       "      <td>0.30</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.29</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.30</td>\n",
       "      <td>43</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>derived</td>\n",
       "      <td>98.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.04</td>\n",
       "      <td>0.01</td>\n",
       "      <td>yes</td>\n",
       "      <td>0.002924</td>\n",
       "      <td>5549.0</td>\n",
       "      <td>143236.0</td>\n",
       "      <td>0.038740</td>\n",
       "      <td>yes</td>\n",
       "      <td>119524265.0</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.00318</td>\n",
       "      <td>0.00298</td>\n",
       "      <td>0.04179</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.04871</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.01431</td>\n",
       "      <td>C</td>\n",
       "      <td>chr6_119475382_119646998</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7258</th>\n",
       "      <td>chr7</td>\n",
       "      <td>126797408</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>Other</td>\n",
       "      <td>GRM8</td>\n",
       "      <td>0.88746</td>\n",
       "      <td>0.91440</td>\n",
       "      <td>0.429960</td>\n",
       "      <td>3.417800</td>\n",
       "      <td>0.482</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.43</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.43</td>\n",
       "      <td>47</td>\n",
       "      <td>-16</td>\n",
       "      <td>7</td>\n",
       "      <td>-16</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>derived</td>\n",
       "      <td>2.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>yes</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.0</td>\n",
       "      <td>143256.0</td>\n",
       "      <td>0.000014</td>\n",
       "      <td>yes</td>\n",
       "      <td>126797407.0</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00198</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>T</td>\n",
       "      <td>chr7_126660339_126797408</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7301</th>\n",
       "      <td>chr7</td>\n",
       "      <td>140801296</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>TMEM178B</td>\n",
       "      <td>0.50215</td>\n",
       "      <td>2.31520</td>\n",
       "      <td>0.070903</td>\n",
       "      <td>3.233300</td>\n",
       "      <td>-0.490</td>\n",
       "      <td>0.25</td>\n",
       "      <td>-0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.25</td>\n",
       "      <td>-13</td>\n",
       "      <td>-30</td>\n",
       "      <td>11</td>\n",
       "      <td>-13</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>derived</td>\n",
       "      <td>4.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>yes</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>8.0</td>\n",
       "      <td>143328.0</td>\n",
       "      <td>0.000056</td>\n",
       "      <td>yes</td>\n",
       "      <td>140801295.0</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00099</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00307</td>\n",
       "      <td>C</td>\n",
       "      <td>chr7_140306146_140804665</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7313</th>\n",
       "      <td>chr7</td>\n",
       "      <td>140958656</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>TMEM178B</td>\n",
       "      <td>0.50215</td>\n",
       "      <td>2.31520</td>\n",
       "      <td>0.070903</td>\n",
       "      <td>3.233300</td>\n",
       "      <td>0.443</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.26</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.26</td>\n",
       "      <td>-1</td>\n",
       "      <td>3</td>\n",
       "      <td>-1</td>\n",
       "      <td>3</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>derived</td>\n",
       "      <td>18.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.02</td>\n",
       "      <td>yes</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>60.0</td>\n",
       "      <td>143274.0</td>\n",
       "      <td>0.000419</td>\n",
       "      <td>yes</td>\n",
       "      <td>140958655.0</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.01840</td>\n",
       "      <td>A</td>\n",
       "      <td>chr7_140934039_141109377</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7341</th>\n",
       "      <td>chr7</td>\n",
       "      <td>157177273</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>DNAJB6</td>\n",
       "      <td>0.76273</td>\n",
       "      <td>1.17450</td>\n",
       "      <td>0.153770</td>\n",
       "      <td>3.463600</td>\n",
       "      <td>0.455</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.31</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.31</td>\n",
       "      <td>-2</td>\n",
       "      <td>46</td>\n",
       "      <td>-2</td>\n",
       "      <td>2</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>derived</td>\n",
       "      <td>57.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>yes</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>268.0</td>\n",
       "      <td>143270.0</td>\n",
       "      <td>0.001871</td>\n",
       "      <td>yes</td>\n",
       "      <td>157177272.0</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.05357</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00204</td>\n",
       "      <td>C</td>\n",
       "      <td>chr7_157116267_157177286</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7383</th>\n",
       "      <td>chr7</td>\n",
       "      <td>158543832</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Other</td>\n",
       "      <td>ESYT2</td>\n",
       "      <td>0.92723</td>\n",
       "      <td>0.55098</td>\n",
       "      <td>0.377280</td>\n",
       "      <td>3.873500</td>\n",
       "      <td>0.561</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.21</td>\n",
       "      <td>0.21</td>\n",
       "      <td>-40</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>derived</td>\n",
       "      <td>120.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.11</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.01</td>\n",
       "      <td>yes</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>481.0</td>\n",
       "      <td>143162.0</td>\n",
       "      <td>0.003360</td>\n",
       "      <td>yes</td>\n",
       "      <td>158543831.0</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.11210</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00511</td>\n",
       "      <td>A</td>\n",
       "      <td>chr7_158399204_158736324</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      chrom        pos ref_allele alt_allele variant_type altai_gt  \\\n",
       "295    chr1   52820680          C          T          snv      0/1   \n",
       "476    chr1  152944501          A          T          snv      0/1   \n",
       "557    chr1  183750131          C          A          snv      0/1   \n",
       "800    chr1  216945756          C          A          snv      0/1   \n",
       "998   chr10   17373518          T          C          snv      0/1   \n",
       "1617  chr11   84191111          T          C          snv      0/1   \n",
       "1669  chr11   99345359          C          G          snv      0/1   \n",
       "3230  chr17   14109360          T          G          snv      0/0   \n",
       "3473  chr18   60003587          G          A          snv      0/1   \n",
       "3541  chr19   33410289          T          C          snv      0/0   \n",
       "4564  chr20   62224595          G          A          snv      0/1   \n",
       "5067   chr3   52486376          T          C          snv      0/1   \n",
       "5237   chr4   38805942          G          C          snv      0/1   \n",
       "5318   chr4  100460531          A          G          snv      0/1   \n",
       "5437   chr4  135122662          G          A          snv      0/1   \n",
       "5765   chr5  118669326          G          C          snv      0/0   \n",
       "5785   chr5  126989007          A          G          snv      0/1   \n",
       "5941   chr5  148430560          A          C          snv      0/1   \n",
       "6252   chr6   51889358          T          C          snv      0/1   \n",
       "6253   chr6   52138226          C          G          snv      0/1   \n",
       "6589   chr6  119524266          C          T          snv      0/1   \n",
       "7258   chr7  126797408          T          A          snv      0/1   \n",
       "7301   chr7  140801296          T          C          snv      0/1   \n",
       "7313   chr7  140958656          A          G          snv      0/1   \n",
       "7341   chr7  157177273          C          T          snv      0/1   \n",
       "7383   chr7  158543832          C          A          snv      0/1   \n",
       "\n",
       "     chagyrskaya_gt denisovan_gt vindija_gt  altai_gt_boolean  \\\n",
       "295             0/0          0/0        0/1              True   \n",
       "476             0/0          0/0        0/1              True   \n",
       "557             0/0          0/0        0/0              True   \n",
       "800             0/1          0/0        1/1              True   \n",
       "998             1/1          0/0        1/1              True   \n",
       "1617            1/1          0/0        1/1              True   \n",
       "1669            0/1          0/0        1/1              True   \n",
       "3230            0/0          1/1        0/0             False   \n",
       "3473            1/1          1/1        1/1              True   \n",
       "3541            0/0          1/1        0/0             False   \n",
       "4564            1/1          0/0        0/1              True   \n",
       "5067            1/1          0/0        1/1              True   \n",
       "5237            1/1          0/0        1/1              True   \n",
       "5318            1/1          1/1        1/1              True   \n",
       "5437            0/0          0/0        0/1              True   \n",
       "5765            0/0          1/1        0/0             False   \n",
       "5785            0/0          0/0        0/0              True   \n",
       "5941            0/0          0/0        1/1              True   \n",
       "6252            1/1          0/1        1/1              True   \n",
       "6253            0/0          0/0        0/0              True   \n",
       "6589            0/0          0/0        1/1              True   \n",
       "7258            1/1          0/0        0/0              True   \n",
       "7301            1/1          1/1        1/1              True   \n",
       "7313            1/1          0/0        0/1              True   \n",
       "7341            1/1          0/0        1/1              True   \n",
       "7383            1/1          1/1        0/0              True   \n",
       "\n",
       "      chagyrskaya_gt_boolean  denisovan_gt_boolean  vindija_gt_boolean  \\\n",
       "295                    False                 False                True   \n",
       "476                    False                 False                True   \n",
       "557                    False                 False               False   \n",
       "800                     True                 False                True   \n",
       "998                     True                 False                True   \n",
       "1617                    True                 False                True   \n",
       "1669                    True                 False                True   \n",
       "3230                   False                  True               False   \n",
       "3473                    True                  True                True   \n",
       "3541                   False                  True               False   \n",
       "4564                    True                 False                True   \n",
       "5067                    True                 False                True   \n",
       "5237                    True                 False                True   \n",
       "5318                    True                  True                True   \n",
       "5437                   False                 False                True   \n",
       "5765                   False                  True               False   \n",
       "5785                   False                 False               False   \n",
       "5941                   False                 False                True   \n",
       "6252                    True                  True                True   \n",
       "6253                   False                 False               False   \n",
       "6589                   False                 False                True   \n",
       "7258                    True                 False               False   \n",
       "7301                    True                  True                True   \n",
       "7313                    True                 False                True   \n",
       "7341                    True                 False                True   \n",
       "7383                    True                  True               False   \n",
       "\n",
       "     distribution annotation   mis_oe    mis_z    lof_oe     lof_z  phyloP  \\\n",
       "295         Other     CC2D1B  1.12620 -1.00890  0.694430  2.038800  -0.140   \n",
       "476         Other      SPRR4  1.49550 -1.10600  0.714380  0.442850   0.383   \n",
       "557         Altai       RGL1  0.70413  2.22380  0.177530  5.116100   0.655   \n",
       "800   Neanderthal      ESRRG  0.56557  2.55570  0.283790  3.051600   0.655   \n",
       "998   Neanderthal    ST8SIA6  1.08980 -0.43129  1.008600 -0.029733   0.533   \n",
       "1617  Neanderthal       DLG2  0.74745  2.07620  0.212280  5.712300  -0.959   \n",
       "1669  Neanderthal      CNTN5  1.03930 -0.33557  0.327200  4.624200  -0.229   \n",
       "3230    Denisovan      COX10  1.03160 -0.18011  0.290420  2.728300   0.338   \n",
       "3473       Shared  TNFRSF11A  0.80508  1.28940  0.330780  3.049700   0.650   \n",
       "3541    Denisovan      CEP89  0.93498  0.47493  0.939730  0.364400  -0.642   \n",
       "4564  Neanderthal      GMEB2  0.66048  2.20200  0.049787  3.946200  -0.837   \n",
       "5067  Neanderthal      TNNC1  0.49480  1.76820  0.125160  2.291500   0.459   \n",
       "5237  Neanderthal       TLR1  1.09920 -0.70157  1.411200 -1.727200  -0.219   \n",
       "5318       Shared    C4orf17  0.87519  0.60341  0.898160  0.359040   0.533   \n",
       "5437        Other    PABPC4L  0.73213  1.47170  0.557620  1.452400   0.561   \n",
       "5765    Denisovan    TNFAIP8  0.87632  0.44541  0.204890  1.627700   0.650   \n",
       "5785        Altai      CTXN3  0.91784  0.19514  0.528190  0.601570   0.528   \n",
       "5941        Other     SH3TC2  1.07760 -0.71307  0.608270  2.673700  -1.235   \n",
       "6252       Shared      PKHD1  1.06000 -0.98178  0.580270  5.281600  -1.093   \n",
       "6253        Altai       MCM3  0.89914  0.80451  0.535230  2.884000   0.563   \n",
       "6589        Other     MAN1A1  0.77221  1.51360  0.255850  3.856000   0.430   \n",
       "7258        Other       GRM8  0.88746  0.91440  0.429960  3.417800   0.482   \n",
       "7301       Shared   TMEM178B  0.50215  2.31520  0.070903  3.233300  -0.490   \n",
       "7313  Neanderthal   TMEM178B  0.50215  2.31520  0.070903  3.233300   0.443   \n",
       "7341  Neanderthal     DNAJB6  0.76273  1.17450  0.153770  3.463600   0.455   \n",
       "7383        Other      ESYT2  0.92723  0.55098  0.377280  3.873500   0.561   \n",
       "\n",
       "      ag_delta  al_delta  dg_delta  dl_delta  delta_max  ag_pos  al_pos  \\\n",
       "295       0.23      0.00      0.00      0.01       0.23     -22     -40   \n",
       "476       0.23      0.00      0.00      0.00       0.23      13       2   \n",
       "557       0.00      0.00      0.26      0.01       0.26      -7     -11   \n",
       "800       0.20      0.00      0.00      0.00       0.20     -27      47   \n",
       "998       0.05      0.22      0.00      0.00       0.22      33     -19   \n",
       "1617      0.00      0.00      0.00      0.40       0.40       1      29   \n",
       "1669      0.00      0.00      0.67      0.00       0.67      -4      48   \n",
       "3230      0.20      0.04      0.00      0.00       0.20       1      38   \n",
       "3473      0.00      0.31      0.00      0.28       0.31       3     -40   \n",
       "3541      0.47      0.00      0.71      0.00       0.71      41     -29   \n",
       "4564      0.01      0.00      0.41      0.02       0.41      15     -44   \n",
       "5067      0.00      0.00      0.00      0.34       0.34     -23       2   \n",
       "5237      0.00      0.00      0.00      0.38       0.38      27     -15   \n",
       "5318      0.00      0.00      0.01      0.35       0.35      -4     -40   \n",
       "5437      0.00      0.00      0.80      0.08       0.80       2      40   \n",
       "5765      0.00      0.00      0.00      0.22       0.22       4       0   \n",
       "5785      0.34      0.91      0.00      0.00       0.91      14       2   \n",
       "5941      0.00      0.32      0.00      0.00       0.32      -6       1   \n",
       "6252      0.00      0.00      0.42      0.00       0.42      31      14   \n",
       "6253      0.21      0.00      0.00      0.00       0.21      -7     -40   \n",
       "6589      0.30      0.00      0.29      0.00       0.30      43       4   \n",
       "7258      0.00      0.00      0.43      0.00       0.43      47     -16   \n",
       "7301      0.25     -0.00      0.00      0.00       0.25     -13     -30   \n",
       "7313      0.00      0.00      0.26      0.00       0.26      -1       3   \n",
       "7341      0.00      0.00      0.31      0.00       0.31      -2      46   \n",
       "7383      0.00      0.00      0.00      0.21       0.21     -40       1   \n",
       "\n",
       "      dg_pos  dl_pos ancestral_allele temp    anc_dev  1KG_allele_count  \\\n",
       "295      -22      34                C    C    derived              90.0   \n",
       "476       12      -1                A    A    derived               2.0   \n",
       "557       -7     -11                C    C    derived              50.0   \n",
       "800      -27      48                C    C    derived             169.0   \n",
       "998       30     -19                T    T    derived              56.0   \n",
       "1617       1      -1                T    T    derived              12.0   \n",
       "1669      -5      37                C    C    derived             177.0   \n",
       "3230      50     -23                T    T    derived             155.0   \n",
       "3473       3      33                G    G    derived             444.0   \n",
       "3541       5     -29                C    C  ancestral            4744.0   \n",
       "4564       2      -2                G    G    derived            1138.0   \n",
       "5067      25      -1                T    T    derived              40.0   \n",
       "5237       1     -15                G    G    derived             911.0   \n",
       "5318      -1      -4                A    A    derived              50.0   \n",
       "5437       2      -2                G    G    derived               1.0   \n",
       "5765       4       0                G    G    derived               1.0   \n",
       "5785     -34       2                A    A    derived               2.0   \n",
       "5941       4     -41                A    A    derived              81.0   \n",
       "6252       1      10                T    T    derived             114.0   \n",
       "6253      -7       3                C    C    derived             782.0   \n",
       "6589       4       0                C    C    derived              98.0   \n",
       "7258       7     -16                T    T    derived               2.0   \n",
       "7301      11     -13                T    T    derived               4.0   \n",
       "7313      -1       3                A    A    derived              18.0   \n",
       "7341      -2       2                C    C    derived              57.0   \n",
       "7383       2       1                C    C    derived             120.0   \n",
       "\n",
       "      1KG_allele_number  1KG_allele_frequency  1KG_EAS_AF  1KG_EUR_AF  \\\n",
       "295              5096.0                  0.02        0.03        0.00   \n",
       "476              5096.0                  0.00        0.00        0.00   \n",
       "557              5096.0                  0.01        0.05        0.00   \n",
       "800              5096.0                  0.03        0.14        0.00   \n",
       "998              5096.0                  0.01        0.00        0.04   \n",
       "1617             5096.0                  0.00        0.00        0.00   \n",
       "1669             5096.0                  0.03        0.11        0.00   \n",
       "3230             5096.0                  0.03        0.08        0.00   \n",
       "3473             5096.0                  0.09        0.00        0.25   \n",
       "3541             5096.0                  0.93        0.90        0.86   \n",
       "4564             5096.0                  0.22        0.51        0.21   \n",
       "5067             5096.0                  0.01        0.01        0.00   \n",
       "5237             5096.0                  0.18        0.39        0.22   \n",
       "5318             5096.0                  0.01        0.00        0.02   \n",
       "5437             5096.0                  0.00        0.00        0.00   \n",
       "5765             5096.0                  0.00        0.00        0.00   \n",
       "5785             5096.0                  0.00        0.00        0.00   \n",
       "5941             5096.0                  0.02        0.07        0.00   \n",
       "6252             5096.0                  0.02        0.01        0.02   \n",
       "6253             5096.0                  0.15        0.24        0.17   \n",
       "6589             5096.0                  0.02        0.00        0.05   \n",
       "7258             5096.0                  0.00        0.00        0.00   \n",
       "7301             5096.0                  0.00        0.00        0.00   \n",
       "7313             5096.0                  0.00        0.00        0.00   \n",
       "7341             5096.0                  0.01        0.05        0.00   \n",
       "7383             5096.0                  0.02        0.11        0.00   \n",
       "\n",
       "      1KG_AFR_AF  1KG_AMR_AF  1KG_SAS_AF present_in_1KG  1KG_non_ASW_AFR_AF  \\\n",
       "295         0.00        0.00        0.05            yes            0.000000   \n",
       "476         0.00        0.00        0.00            yes            0.000000   \n",
       "557         0.00        0.00        0.00            yes            0.000000   \n",
       "800         0.00        0.01        0.02            yes            0.000000   \n",
       "998         0.00        0.02        0.00            yes            0.000000   \n",
       "1617        0.00        0.00        0.01            yes            0.000000   \n",
       "1669        0.00        0.00        0.07            yes            0.000000   \n",
       "3230        0.00        0.08        0.01            yes            0.000000   \n",
       "3473        0.01        0.12        0.10            yes            0.003899   \n",
       "3541        0.99        0.93        0.96            yes            0.999025   \n",
       "4564        0.01        0.11        0.31            yes            0.000975   \n",
       "5067        0.00        0.00        0.03            yes            0.000000   \n",
       "5237        0.01        0.15        0.16            yes            0.001949   \n",
       "5318        0.00        0.04        0.00            yes            0.000000   \n",
       "5437        0.00        0.00        0.00            yes            0.000000   \n",
       "5765        0.00        0.00        0.00            yes            0.000000   \n",
       "5785        0.00        0.00        0.00            yes            0.000000   \n",
       "5941        0.00        0.01        0.00            yes            0.000000   \n",
       "6252        0.00        0.01        0.08            yes            0.000000   \n",
       "6253        0.00        0.09        0.29            yes            0.001949   \n",
       "6589        0.00        0.04        0.01            yes            0.002924   \n",
       "7258        0.00        0.00        0.00            yes            0.000000   \n",
       "7301        0.00        0.00        0.00            yes            0.000000   \n",
       "7313        0.00        0.00        0.02            yes            0.000000   \n",
       "7341        0.00        0.00        0.00            yes            0.000000   \n",
       "7383        0.00        0.00        0.01            yes            0.000000   \n",
       "\n",
       "      gnomAD_allele_count  gnomAD_allele_number  gnomAD_allele_frequency  \\\n",
       "295                 443.0              143266.0                 0.003092   \n",
       "476                   4.0              143152.0                 0.000028   \n",
       "557                 194.0              143252.0                 0.001354   \n",
       "800                 746.0              143174.0                 0.005210   \n",
       "998                2946.0              143262.0                 0.020564   \n",
       "1617                 38.0              143298.0                 0.000265   \n",
       "1669                551.0              143130.0                 0.003850   \n",
       "3230               1030.0              143224.0                 0.007192   \n",
       "3473              21727.0              143202.0                 0.151723   \n",
       "3541             129412.0              143248.0                 0.903412   \n",
       "4564              23692.0              143104.0                 0.165558   \n",
       "5067                494.0              143210.0                 0.003449   \n",
       "5237              19931.0              143224.0                 0.139160   \n",
       "5318               3317.0              143342.0                 0.023141   \n",
       "5437                  2.0              143316.0                 0.000014   \n",
       "5765                  NaN                   NaN                      NaN   \n",
       "5785                 10.0              143290.0                 0.000070   \n",
       "5941                214.0              143150.0                 0.001495   \n",
       "6252               2957.0              143284.0                 0.020637   \n",
       "6253              17455.0              143064.0                 0.122008   \n",
       "6589               5549.0              143236.0                 0.038740   \n",
       "7258                  2.0              143256.0                 0.000014   \n",
       "7301                  8.0              143328.0                 0.000056   \n",
       "7313                 60.0              143274.0                 0.000419   \n",
       "7341                268.0              143270.0                 0.001871   \n",
       "7383                481.0              143162.0                 0.003360   \n",
       "\n",
       "     present_in_gnomAD        start Vernot_ancestral_allele  \\\n",
       "295                yes   52820679.0                       C   \n",
       "476                yes  152944500.0                       A   \n",
       "557                yes  183750130.0                       C   \n",
       "800                yes  216945755.0                       C   \n",
       "998                yes   17373517.0                       T   \n",
       "1617               yes   84191110.0                       T   \n",
       "1669               yes   99345358.0                       C   \n",
       "3230               yes   14109359.0                       T   \n",
       "3473               yes   60003586.0                       G   \n",
       "3541               yes   33410288.0                       C   \n",
       "4564               yes   62224594.0                       G   \n",
       "5067               yes   52486375.0                       T   \n",
       "5237               yes   38805941.0                       G   \n",
       "5318               yes  100460530.0                       A   \n",
       "5437               yes  135122661.0                       G   \n",
       "5765                no  118669325.0                       G   \n",
       "5785               yes  126989006.0                       A   \n",
       "5941               yes  148430559.0                       A   \n",
       "6252               yes   51889357.0                       T   \n",
       "6253               yes   52138225.0                       C   \n",
       "6589               yes  119524265.0                       C   \n",
       "7258               yes  126797407.0                       T   \n",
       "7301               yes  140801295.0                       T   \n",
       "7313               yes  140958655.0                       A   \n",
       "7341               yes  157177272.0                       C   \n",
       "7383               yes  158543831.0                       C   \n",
       "\n",
       "     Vernot_derived_allele  Vernot_ancestral_derived_code  Vernot_AFA_AF  \\\n",
       "295                      T                            1.0        0.00637   \n",
       "476                      T                            1.0        0.00000   \n",
       "557                      A                            1.0        0.00000   \n",
       "800                      A                            1.0        0.00000   \n",
       "998                      C                            1.0        0.01274   \n",
       "1617                     C                            1.0        0.00000   \n",
       "1669                     G                            1.0        0.00318   \n",
       "3230                     G                            1.0        0.00637   \n",
       "3473                     A                            1.0        0.01592   \n",
       "3541                     T                            1.0        0.03503   \n",
       "4564                     A                            1.0        0.03503   \n",
       "5067                     C                            1.0        0.00318   \n",
       "5237                     C                            1.0        0.04459   \n",
       "5318                     G                            1.0        0.00000   \n",
       "5437                     A                            1.0        0.00000   \n",
       "5765                     C                            1.0        0.00000   \n",
       "5785                     G                            1.0        0.00000   \n",
       "5941                     C                            1.0        0.00318   \n",
       "6252                     C                            1.0        0.00318   \n",
       "6253                     G                            1.0        0.01274   \n",
       "6589                     T                            1.0        0.00318   \n",
       "7258                     A                            1.0        0.00000   \n",
       "7301                     C                            1.0        0.00000   \n",
       "7313                     G                            1.0        0.00000   \n",
       "7341                     T                            1.0        0.00000   \n",
       "7383                     A                            1.0        0.00000   \n",
       "\n",
       "      Vernot_AFR_AF  Vernot_AMR_AF  Vernot_EAS_AF  Vernot_EUR_AF  \\\n",
       "295         0.00000        0.00000        0.02877        0.00497   \n",
       "476         0.00000        0.00000        0.00198        0.00000   \n",
       "557         0.00000        0.00000        0.04861        0.00000   \n",
       "800         0.00000        0.01153        0.13988        0.00000   \n",
       "998         0.00000        0.01729        0.00000        0.03479   \n",
       "1617        0.00000        0.00000        0.00000        0.00000   \n",
       "1669        0.00000        0.00000        0.10615        0.00000   \n",
       "3230        0.00000        0.08501        0.08333        0.00000   \n",
       "3473        0.00397        0.11671        0.00198        0.24950   \n",
       "3541        0.00099        0.06916        0.09821        0.14612   \n",
       "4564        0.00099        0.10663        0.51488        0.21173   \n",
       "5067        0.00000        0.00288        0.00794        0.00298   \n",
       "5237        0.00198        0.14986        0.39484        0.22068   \n",
       "5318        0.00000        0.03602        0.00099        0.02087   \n",
       "5437        0.00000        0.00000        0.00099        0.00000   \n",
       "5765        0.00000        0.00000        0.00000        0.00000   \n",
       "5785        0.00000        0.00000        0.00000        0.00000   \n",
       "5941        0.00000        0.00720        0.07242        0.00000   \n",
       "6252        0.00000        0.01009        0.00694        0.02286   \n",
       "6253        0.00198        0.09366        0.24206        0.16600   \n",
       "6589        0.00298        0.04179        0.00000        0.04871   \n",
       "7258        0.00000        0.00000        0.00198        0.00000   \n",
       "7301        0.00000        0.00000        0.00099        0.00000   \n",
       "7313        0.00000        0.00000        0.00000        0.00000   \n",
       "7341        0.00000        0.00000        0.05357        0.00000   \n",
       "7383        0.00000        0.00000        0.11210        0.00000   \n",
       "\n",
       "      Vernot_PNG_AF  Vernot_SAS_AF Vernot_Denisovan_base  \\\n",
       "295         0.00000        0.05521                     C   \n",
       "476         0.00000        0.00000                     A   \n",
       "557         0.01852        0.00000                     C   \n",
       "800         0.00000        0.01840                     C   \n",
       "998         0.01852        0.00307                     T   \n",
       "1617        0.00000        0.01227                     T   \n",
       "1669        0.11111        0.06748                     C   \n",
       "3230        0.16667        0.00818                     G   \n",
       "3473        0.09259        0.09714                     A   \n",
       "3541        0.07407        0.04090                     C   \n",
       "4564        0.03704        0.30982                     G   \n",
       "5067        0.00000        0.02658                     T   \n",
       "5237        0.05556        0.16360                     G   \n",
       "5318        0.00000        0.00204                     G   \n",
       "5437        0.00000        0.00000                     G   \n",
       "5765        0.11111        0.00102                     C   \n",
       "5785        0.00000        0.00204                     A   \n",
       "5941        0.03704        0.00204                     A   \n",
       "6252        0.00000        0.07771                   T/C   \n",
       "6253        0.09259        0.29448                     C   \n",
       "6589        0.00000        0.01431                     C   \n",
       "7258        0.00000        0.00000                     T   \n",
       "7301        0.00000        0.00307                     C   \n",
       "7313        0.00000        0.01840                     A   \n",
       "7341        0.00000        0.00204                     C   \n",
       "7383        0.00000        0.00511                     A   \n",
       "\n",
       "          Vernot_haplotype_tag Vernot_Neanderthal_base_1  \\\n",
       "295     chr1_52325475_53062486                         C   \n",
       "476   chr1_152942848_153033467                         A   \n",
       "557   chr1_183411087_183827254                         C   \n",
       "800   chr1_216944862_216964892                         C   \n",
       "998    chr10_17338795_17417730                         T   \n",
       "1617   chr11_84051845_84211811                         T   \n",
       "1669   chr11_99309226_99359652                         C   \n",
       "3230   chr17_13902596_14111504                         T   \n",
       "3473   chr18_59569521_60089689                         G   \n",
       "3541   chr19_33040186_33456652                         T   \n",
       "4564   chr20_62195671_62229244                         G   \n",
       "5067    chr3_52214838_52516460                         T   \n",
       "5237    chr4_38294200_38857896                         G   \n",
       "5318  chr4_100349203_100638613                         A   \n",
       "5437  chr4_135066013_135400700                         G   \n",
       "5765  chr5_118661135_118933833                         G   \n",
       "5785  chr5_126989006_127537918                         A   \n",
       "5941  chr5_148339333_148539017                         A   \n",
       "6252    chr6_51841815_51972144                         T   \n",
       "6253    chr6_52138225_52183126                         C   \n",
       "6589  chr6_119475382_119646998                         C   \n",
       "7258  chr7_126660339_126797408                         T   \n",
       "7301  chr7_140306146_140804665                         T   \n",
       "7313  chr7_140934039_141109377                         A   \n",
       "7341  chr7_157116267_157177286                         C   \n",
       "7383  chr7_158399204_158736324                         C   \n",
       "\n",
       "     Vernot_Neanderthal_base_2  \n",
       "295                          T  \n",
       "476                          T  \n",
       "557                          A  \n",
       "800                          A  \n",
       "998                          C  \n",
       "1617                         C  \n",
       "1669                         G  \n",
       "3230                       NaN  \n",
       "3473                         A  \n",
       "3541                       NaN  \n",
       "4564                         A  \n",
       "5067                         C  \n",
       "5237                         C  \n",
       "5318                         G  \n",
       "5437                         A  \n",
       "5765                       NaN  \n",
       "5785                         G  \n",
       "5941                         C  \n",
       "6252                         C  \n",
       "6253                         G  \n",
       "6589                         T  \n",
       "7258                         A  \n",
       "7301                         C  \n",
       "7313                         G  \n",
       "7341                         T  \n",
       "7383                         A  "
      ]
     },
     "execution_count": 610,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SAV_introgressed_ref_tag"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "95a20e83",
   "metadata": {},
   "source": [
    "Now load in the SpliceAI annotated dataframe where these variants were the alternate allele."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 611,
   "id": "8950487f",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref_allele</th>\n",
       "      <th>altai_gt</th>\n",
       "      <th>chagyrskaya_gt</th>\n",
       "      <th>denisovan_gt</th>\n",
       "      <th>vindija_gt</th>\n",
       "      <th>alt_allele</th>\n",
       "      <th>annotation</th>\n",
       "      <th>ag_delta</th>\n",
       "      <th>al_delta</th>\n",
       "      <th>dg_delta</th>\n",
       "      <th>dl_delta</th>\n",
       "      <th>ag_pos</th>\n",
       "      <th>al_pos</th>\n",
       "      <th>dg_pos</th>\n",
       "      <th>dl_pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10</td>\n",
       "      <td>1098863</td>\n",
       "      <td>G</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>A</td>\n",
       "      <td>WDR37</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-16</td>\n",
       "      <td>-15</td>\n",
       "      <td>-16</td>\n",
       "      <td>45</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10</td>\n",
       "      <td>1113464</td>\n",
       "      <td>T</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>A</td>\n",
       "      <td>WDR37</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2</td>\n",
       "      <td>48</td>\n",
       "      <td>18</td>\n",
       "      <td>-6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>1175546</td>\n",
       "      <td>G</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>A</td>\n",
       "      <td>WDR37</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>13</td>\n",
       "      <td>-30</td>\n",
       "      <td>-5</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>1230214</td>\n",
       "      <td>C</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>T</td>\n",
       "      <td>ADARB2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>23</td>\n",
       "      <td>-48</td>\n",
       "      <td>45</td>\n",
       "      <td>-22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr10</td>\n",
       "      <td>1460803</td>\n",
       "      <td>A</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>C</td>\n",
       "      <td>ADARB2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>45</td>\n",
       "      <td>-8</td>\n",
       "      <td>45</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr10</td>\n",
       "      <td>1464483</td>\n",
       "      <td>A</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>G</td>\n",
       "      <td>ADARB2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>44</td>\n",
       "      <td>-16</td>\n",
       "      <td>-47</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr10</td>\n",
       "      <td>1682361</td>\n",
       "      <td>A</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>T</td>\n",
       "      <td>ADARB2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>-47</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr10</td>\n",
       "      <td>1684310</td>\n",
       "      <td>C</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>T</td>\n",
       "      <td>ADARB2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>26</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr10</td>\n",
       "      <td>1686465</td>\n",
       "      <td>C</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>T</td>\n",
       "      <td>ADARB2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-18</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr10</td>\n",
       "      <td>1710328</td>\n",
       "      <td>G</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>A</td>\n",
       "      <td>ADARB2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-22</td>\n",
       "      <td>-41</td>\n",
       "      <td>35</td>\n",
       "      <td>-41</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   chrom      pos ref_allele altai_gt chagyrskaya_gt denisovan_gt vindija_gt  \\\n",
       "0  chr10  1098863          G      1/1            1/1          1/1        1/1   \n",
       "1  chr10  1113464          T      1/1            1/1          1/1        1/1   \n",
       "2  chr10  1175546          G      1/1            1/1          1/1        1/1   \n",
       "3  chr10  1230214          C      1/1            1/1          1/1        1/1   \n",
       "4  chr10  1460803          A      1/1            1/1          1/1        1/1   \n",
       "5  chr10  1464483          A      1/1            1/1          1/1        1/1   \n",
       "6  chr10  1682361          A      1/1            1/1          1/1        1/1   \n",
       "7  chr10  1684310          C      1/1            1/1          1/1        1/1   \n",
       "8  chr10  1686465          C      1/1            1/1          1/1        1/1   \n",
       "9  chr10  1710328          G      1/1            1/1          1/1        1/1   \n",
       "\n",
       "  alt_allele annotation  ag_delta  al_delta  dg_delta  dl_delta  ag_pos  \\\n",
       "0          A      WDR37       0.0       0.0       0.0       0.0     -16   \n",
       "1          A      WDR37       0.0       0.0       0.0       0.0       2   \n",
       "2          A      WDR37       0.0       0.0       0.0       0.0      13   \n",
       "3          T     ADARB2       0.0       0.0       0.0       0.0      23   \n",
       "4          C     ADARB2       0.0       0.0       0.0       0.0      45   \n",
       "5          G     ADARB2       0.0       0.0       0.0       0.0      44   \n",
       "6          T     ADARB2       0.0       0.0       0.0       0.0      31   \n",
       "7          T     ADARB2       0.0       0.0       0.0       0.0       9   \n",
       "8          T     ADARB2       0.0       0.0       0.0       0.0     -18   \n",
       "9          A     ADARB2       0.0       0.0       0.0       0.0     -22   \n",
       "\n",
       "   al_pos  dg_pos  dl_pos  \n",
       "0     -15     -16      45  \n",
       "1      48      18      -6  \n",
       "2     -30      -5       7  \n",
       "3     -48      45     -22  \n",
       "4      -8      45       2  \n",
       "5     -16     -47      37  \n",
       "6       1     -47       2  \n",
       "7       1      26       1  \n",
       "8       6       3       6  \n",
       "9     -41      35     -41  "
      ]
     },
     "execution_count": 611,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "swapped_header = ['chrom','pos','ref_allele','altai_gt','chagyrskaya_gt','denisovan_gt','vindija_gt','alt_allele','annotation','ag_delta','al_delta','dg_delta','dl_delta','ag_pos','al_pos','dg_pos','dl_pos']\n",
    "swapped = pd.read_csv('../introgressed_refs/spliceai_annotated_introgressed_refs.txt', sep = '\\t', names = swapped_header)\n",
    "swapped.head(10)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c22588cf",
   "metadata": {},
   "source": [
    "We need to create a delta_max column for comparison."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 612,
   "id": "71fbbb5d",
   "metadata": {},
   "outputs": [],
   "source": [
    "swapped['delta_max'] = swapped[['ag_delta','al_delta','dg_delta','dl_delta']].max(axis = 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 613,
   "id": "92fdb636",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "8709"
      ]
     },
     "execution_count": 613,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(swapped)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 614,
   "id": "90fefe24",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "24"
      ]
     },
     "execution_count": 614,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SAV_swapped = swapped[swapped['delta_max']>=0.2]\n",
    "len(SAV_swapped)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fa9fd316",
   "metadata": {},
   "source": [
    "Now merge."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 615,
   "id": "16b90b6b",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref_allele_x</th>\n",
       "      <th>alt_allele_x</th>\n",
       "      <th>variant_type</th>\n",
       "      <th>altai_gt_x</th>\n",
       "      <th>chagyrskaya_gt_x</th>\n",
       "      <th>denisovan_gt_x</th>\n",
       "      <th>vindija_gt_x</th>\n",
       "      <th>altai_gt_boolean</th>\n",
       "      <th>chagyrskaya_gt_boolean</th>\n",
       "      <th>denisovan_gt_boolean</th>\n",
       "      <th>vindija_gt_boolean</th>\n",
       "      <th>distribution</th>\n",
       "      <th>annotation</th>\n",
       "      <th>mis_oe</th>\n",
       "      <th>mis_z</th>\n",
       "      <th>lof_oe</th>\n",
       "      <th>lof_z</th>\n",
       "      <th>phyloP</th>\n",
       "      <th>ag_delta_x</th>\n",
       "      <th>al_delta_x</th>\n",
       "      <th>dg_delta_x</th>\n",
       "      <th>dl_delta_x</th>\n",
       "      <th>delta_max_x</th>\n",
       "      <th>ag_pos_x</th>\n",
       "      <th>al_pos_x</th>\n",
       "      <th>dg_pos_x</th>\n",
       "      <th>dl_pos_x</th>\n",
       "      <th>ancestral_allele</th>\n",
       "      <th>temp</th>\n",
       "      <th>anc_dev</th>\n",
       "      <th>1KG_allele_count</th>\n",
       "      <th>1KG_allele_number</th>\n",
       "      <th>1KG_allele_frequency</th>\n",
       "      <th>1KG_EAS_AF</th>\n",
       "      <th>1KG_EUR_AF</th>\n",
       "      <th>1KG_AFR_AF</th>\n",
       "      <th>1KG_AMR_AF</th>\n",
       "      <th>1KG_SAS_AF</th>\n",
       "      <th>present_in_1KG</th>\n",
       "      <th>1KG_non_ASW_AFR_AF</th>\n",
       "      <th>gnomAD_allele_count</th>\n",
       "      <th>gnomAD_allele_number</th>\n",
       "      <th>gnomAD_allele_frequency</th>\n",
       "      <th>present_in_gnomAD</th>\n",
       "      <th>start</th>\n",
       "      <th>Vernot_ancestral_allele</th>\n",
       "      <th>Vernot_derived_allele</th>\n",
       "      <th>Vernot_ancestral_derived_code</th>\n",
       "      <th>Vernot_AFA_AF</th>\n",
       "      <th>Vernot_AFR_AF</th>\n",
       "      <th>Vernot_AMR_AF</th>\n",
       "      <th>Vernot_EAS_AF</th>\n",
       "      <th>Vernot_EUR_AF</th>\n",
       "      <th>Vernot_PNG_AF</th>\n",
       "      <th>Vernot_SAS_AF</th>\n",
       "      <th>Vernot_Denisovan_base</th>\n",
       "      <th>Vernot_haplotype_tag</th>\n",
       "      <th>Vernot_Neanderthal_base_1</th>\n",
       "      <th>Vernot_Neanderthal_base_2</th>\n",
       "      <th>ref_allele_y</th>\n",
       "      <th>altai_gt_y</th>\n",
       "      <th>chagyrskaya_gt_y</th>\n",
       "      <th>denisovan_gt_y</th>\n",
       "      <th>vindija_gt_y</th>\n",
       "      <th>alt_allele_y</th>\n",
       "      <th>ag_delta_y</th>\n",
       "      <th>al_delta_y</th>\n",
       "      <th>dg_delta_y</th>\n",
       "      <th>dl_delta_y</th>\n",
       "      <th>ag_pos_y</th>\n",
       "      <th>al_pos_y</th>\n",
       "      <th>dg_pos_y</th>\n",
       "      <th>dl_pos_y</th>\n",
       "      <th>delta_max_y</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3046711</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Other</td>\n",
       "      <td>PRDM16</td>\n",
       "      <td>0.87043</td>\n",
       "      <td>1.3471</td>\n",
       "      <td>0.081747</td>\n",
       "      <td>5.9522</td>\n",
       "      <td>0.379</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-4</td>\n",
       "      <td>35</td>\n",
       "      <td>-24</td>\n",
       "      <td>19</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>derived</td>\n",
       "      <td>112.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.06</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.01</td>\n",
       "      <td>yes</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>6632.0</td>\n",
       "      <td>143094.0</td>\n",
       "      <td>0.046347</td>\n",
       "      <td>yes</td>\n",
       "      <td>3046710.0</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.01274</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.04611</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.06362</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.01125</td>\n",
       "      <td>T</td>\n",
       "      <td>chr1_3015134_3046826</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>T</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>35</td>\n",
       "      <td>-4</td>\n",
       "      <td>19</td>\n",
       "      <td>-24</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3046826</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Other</td>\n",
       "      <td>PRDM16</td>\n",
       "      <td>0.87043</td>\n",
       "      <td>1.3471</td>\n",
       "      <td>0.081747</td>\n",
       "      <td>5.9522</td>\n",
       "      <td>0.462</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-49</td>\n",
       "      <td>-1</td>\n",
       "      <td>11</td>\n",
       "      <td>-1</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>112.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.06</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.01</td>\n",
       "      <td>yes</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>6626.0</td>\n",
       "      <td>143318.0</td>\n",
       "      <td>0.046233</td>\n",
       "      <td>yes</td>\n",
       "      <td>3046825.0</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.01274</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.04611</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.06362</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.01125</td>\n",
       "      <td>G</td>\n",
       "      <td>chr1_3015134_3046826</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>G</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-49</td>\n",
       "      <td>-1</td>\n",
       "      <td>11</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3188156</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>PRDM16</td>\n",
       "      <td>0.87043</td>\n",
       "      <td>1.3471</td>\n",
       "      <td>0.081747</td>\n",
       "      <td>5.9522</td>\n",
       "      <td>0.491</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-27</td>\n",
       "      <td>9</td>\n",
       "      <td>4</td>\n",
       "      <td>-1</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>301.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.06</td>\n",
       "      <td>0.18</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.04</td>\n",
       "      <td>0.08</td>\n",
       "      <td>yes</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1706.0</td>\n",
       "      <td>143300.0</td>\n",
       "      <td>0.011905</td>\n",
       "      <td>yes</td>\n",
       "      <td>3188155.0</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.00318</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.04323</td>\n",
       "      <td>0.18254</td>\n",
       "      <td>0.00298</td>\n",
       "      <td>0.07407</td>\n",
       "      <td>0.07975</td>\n",
       "      <td>G</td>\n",
       "      <td>chr1_3066509_3209504</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>G</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>9</td>\n",
       "      <td>-27</td>\n",
       "      <td>-1</td>\n",
       "      <td>4</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3199291</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>PRDM16</td>\n",
       "      <td>0.87043</td>\n",
       "      <td>1.3471</td>\n",
       "      <td>0.081747</td>\n",
       "      <td>5.9522</td>\n",
       "      <td>0.297</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>28</td>\n",
       "      <td>30</td>\n",
       "      <td>28</td>\n",
       "      <td>-23</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>derived</td>\n",
       "      <td>447.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.09</td>\n",
       "      <td>0.25</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.13</td>\n",
       "      <td>0.10</td>\n",
       "      <td>yes</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2943.0</td>\n",
       "      <td>143256.0</td>\n",
       "      <td>0.020544</td>\n",
       "      <td>yes</td>\n",
       "      <td>3199290.0</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.00955</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.12680</td>\n",
       "      <td>0.24603</td>\n",
       "      <td>0.00596</td>\n",
       "      <td>0.11111</td>\n",
       "      <td>0.09509</td>\n",
       "      <td>C</td>\n",
       "      <td>chr1_3066509_3209504</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>C</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>30</td>\n",
       "      <td>28</td>\n",
       "      <td>-23</td>\n",
       "      <td>28</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3199727</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>PRDM16</td>\n",
       "      <td>0.87043</td>\n",
       "      <td>1.3471</td>\n",
       "      <td>0.081747</td>\n",
       "      <td>5.9522</td>\n",
       "      <td>0.436</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>8</td>\n",
       "      <td>44</td>\n",
       "      <td>-2</td>\n",
       "      <td>44</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>derived</td>\n",
       "      <td>447.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.09</td>\n",
       "      <td>0.25</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.13</td>\n",
       "      <td>0.10</td>\n",
       "      <td>yes</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2949.0</td>\n",
       "      <td>143216.0</td>\n",
       "      <td>0.020591</td>\n",
       "      <td>yes</td>\n",
       "      <td>3199726.0</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.00955</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.12680</td>\n",
       "      <td>0.24603</td>\n",
       "      <td>0.00596</td>\n",
       "      <td>0.11111</td>\n",
       "      <td>0.09509</td>\n",
       "      <td>C</td>\n",
       "      <td>chr1_3066509_3209504</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>C</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>44</td>\n",
       "      <td>8</td>\n",
       "      <td>44</td>\n",
       "      <td>-2</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3209504</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>PRDM16</td>\n",
       "      <td>0.87043</td>\n",
       "      <td>1.3471</td>\n",
       "      <td>0.081747</td>\n",
       "      <td>5.9522</td>\n",
       "      <td>-0.438</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>20</td>\n",
       "      <td>1</td>\n",
       "      <td>-12</td>\n",
       "      <td>-1</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>444.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.09</td>\n",
       "      <td>0.24</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.13</td>\n",
       "      <td>0.09</td>\n",
       "      <td>yes</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2939.0</td>\n",
       "      <td>143312.0</td>\n",
       "      <td>0.020508</td>\n",
       "      <td>yes</td>\n",
       "      <td>3209503.0</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.00955</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.12680</td>\n",
       "      <td>0.24405</td>\n",
       "      <td>0.00596</td>\n",
       "      <td>0.11111</td>\n",
       "      <td>0.09305</td>\n",
       "      <td>G</td>\n",
       "      <td>chr1_3066509_3209504</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>G</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>20</td>\n",
       "      <td>-1</td>\n",
       "      <td>-12</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3210305</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>PRDM16</td>\n",
       "      <td>0.87043</td>\n",
       "      <td>1.3471</td>\n",
       "      <td>0.081747</td>\n",
       "      <td>5.9522</td>\n",
       "      <td>-0.379</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>2</td>\n",
       "      <td>7</td>\n",
       "      <td>-4</td>\n",
       "      <td>-21</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>derived</td>\n",
       "      <td>443.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.09</td>\n",
       "      <td>0.25</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.13</td>\n",
       "      <td>0.09</td>\n",
       "      <td>yes</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2939.0</td>\n",
       "      <td>143296.0</td>\n",
       "      <td>0.020510</td>\n",
       "      <td>yes</td>\n",
       "      <td>3210304.0</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.00955</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.12680</td>\n",
       "      <td>0.24405</td>\n",
       "      <td>0.00596</td>\n",
       "      <td>0.11111</td>\n",
       "      <td>0.09100</td>\n",
       "      <td>T</td>\n",
       "      <td>chr1_3166567_3212428</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>T</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>7</td>\n",
       "      <td>2</td>\n",
       "      <td>-21</td>\n",
       "      <td>-4</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3212428</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>PRDM16</td>\n",
       "      <td>0.87043</td>\n",
       "      <td>1.3471</td>\n",
       "      <td>0.081747</td>\n",
       "      <td>5.9522</td>\n",
       "      <td>-0.445</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.01</td>\n",
       "      <td>3</td>\n",
       "      <td>11</td>\n",
       "      <td>23</td>\n",
       "      <td>11</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>derived</td>\n",
       "      <td>452.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.09</td>\n",
       "      <td>0.26</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.13</td>\n",
       "      <td>0.09</td>\n",
       "      <td>yes</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>3291.0</td>\n",
       "      <td>143212.0</td>\n",
       "      <td>0.022980</td>\n",
       "      <td>yes</td>\n",
       "      <td>3212427.0</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.00955</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.13112</td>\n",
       "      <td>0.25595</td>\n",
       "      <td>0.00696</td>\n",
       "      <td>0.11111</td>\n",
       "      <td>0.08384</td>\n",
       "      <td>A</td>\n",
       "      <td>chr1_3166567_3212428</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>A</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.0</td>\n",
       "      <td>11</td>\n",
       "      <td>3</td>\n",
       "      <td>11</td>\n",
       "      <td>23</td>\n",
       "      <td>0.01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3240652</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>PRDM16</td>\n",
       "      <td>0.87043</td>\n",
       "      <td>1.3471</td>\n",
       "      <td>0.081747</td>\n",
       "      <td>5.9522</td>\n",
       "      <td>-0.479</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>10</td>\n",
       "      <td>50</td>\n",
       "      <td>29</td>\n",
       "      <td>9</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>derived</td>\n",
       "      <td>26.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.02</td>\n",
       "      <td>yes</td>\n",
       "      <td>0.004873</td>\n",
       "      <td>294.0</td>\n",
       "      <td>143304.0</td>\n",
       "      <td>0.002052</td>\n",
       "      <td>yes</td>\n",
       "      <td>3240651.0</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.00318</td>\n",
       "      <td>0.00397</td>\n",
       "      <td>0.00144</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.18519</td>\n",
       "      <td>0.02045</td>\n",
       "      <td>T</td>\n",
       "      <td>chr1_3195401_3250112</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>C</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>50</td>\n",
       "      <td>10</td>\n",
       "      <td>9</td>\n",
       "      <td>29</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3240725</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Other</td>\n",
       "      <td>PRDM16</td>\n",
       "      <td>0.87043</td>\n",
       "      <td>1.3471</td>\n",
       "      <td>0.081747</td>\n",
       "      <td>5.9522</td>\n",
       "      <td>-1.193</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-39</td>\n",
       "      <td>29</td>\n",
       "      <td>40</td>\n",
       "      <td>-2</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>derived</td>\n",
       "      <td>394.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.08</td>\n",
       "      <td>0.16</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.14</td>\n",
       "      <td>0.12</td>\n",
       "      <td>yes</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2936.0</td>\n",
       "      <td>142852.0</td>\n",
       "      <td>0.020553</td>\n",
       "      <td>yes</td>\n",
       "      <td>3240724.0</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.00955</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.13833</td>\n",
       "      <td>0.16171</td>\n",
       "      <td>0.00596</td>\n",
       "      <td>0.11111</td>\n",
       "      <td>0.11656</td>\n",
       "      <td>T</td>\n",
       "      <td>chr1_3240724_3259500</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>T</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>29</td>\n",
       "      <td>-39</td>\n",
       "      <td>-2</td>\n",
       "      <td>40</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chrom      pos ref_allele_x alt_allele_x variant_type altai_gt_x  \\\n",
       "0  chr1  3046711            T            G          snv        0/1   \n",
       "1  chr1  3046826            G            T          snv        0/1   \n",
       "2  chr1  3188156            G            A          snv        0/1   \n",
       "3  chr1  3199291            C            T          snv        0/1   \n",
       "4  chr1  3199727            C            T          snv        0/1   \n",
       "5  chr1  3209504            G            A          snv        0/1   \n",
       "6  chr1  3210305            T            A          snv        0/1   \n",
       "7  chr1  3212428            A            C          snv        0/1   \n",
       "8  chr1  3240652            C            T          snv        0/0   \n",
       "9  chr1  3240725            T            C          snv        0/1   \n",
       "\n",
       "  chagyrskaya_gt_x denisovan_gt_x vindija_gt_x  altai_gt_boolean  \\\n",
       "0              0/0            0/0          1/1              True   \n",
       "1              0/0            0/0          1/1              True   \n",
       "2              1/1            0/0          1/1              True   \n",
       "3              1/1            0/0          1/1              True   \n",
       "4              1/1            0/0          1/1              True   \n",
       "5              1/1            0/0          1/1              True   \n",
       "6              1/1            0/0          1/1              True   \n",
       "7              1/1            0/0          1/1              True   \n",
       "8              0/0            1/1          0/0             False   \n",
       "9              0/0            0/0          0/1              True   \n",
       "\n",
       "   chagyrskaya_gt_boolean  denisovan_gt_boolean  vindija_gt_boolean  \\\n",
       "0                   False                 False                True   \n",
       "1                   False                 False                True   \n",
       "2                    True                 False                True   \n",
       "3                    True                 False                True   \n",
       "4                    True                 False                True   \n",
       "5                    True                 False                True   \n",
       "6                    True                 False                True   \n",
       "7                    True                 False                True   \n",
       "8                   False                  True               False   \n",
       "9                   False                 False                True   \n",
       "\n",
       "  distribution annotation   mis_oe   mis_z    lof_oe   lof_z  phyloP  \\\n",
       "0        Other     PRDM16  0.87043  1.3471  0.081747  5.9522   0.379   \n",
       "1        Other     PRDM16  0.87043  1.3471  0.081747  5.9522   0.462   \n",
       "2  Neanderthal     PRDM16  0.87043  1.3471  0.081747  5.9522   0.491   \n",
       "3  Neanderthal     PRDM16  0.87043  1.3471  0.081747  5.9522   0.297   \n",
       "4  Neanderthal     PRDM16  0.87043  1.3471  0.081747  5.9522   0.436   \n",
       "5  Neanderthal     PRDM16  0.87043  1.3471  0.081747  5.9522  -0.438   \n",
       "6  Neanderthal     PRDM16  0.87043  1.3471  0.081747  5.9522  -0.379   \n",
       "7  Neanderthal     PRDM16  0.87043  1.3471  0.081747  5.9522  -0.445   \n",
       "8    Denisovan     PRDM16  0.87043  1.3471  0.081747  5.9522  -0.479   \n",
       "9        Other     PRDM16  0.87043  1.3471  0.081747  5.9522  -1.193   \n",
       "\n",
       "   ag_delta_x  al_delta_x  dg_delta_x  dl_delta_x  delta_max_x  ag_pos_x  \\\n",
       "0         0.0         0.0         0.0        0.00         0.00        -4   \n",
       "1         0.0         0.0         0.0        0.00         0.00       -49   \n",
       "2         0.0         0.0         0.0        0.00         0.00       -27   \n",
       "3         0.0         0.0         0.0        0.00         0.00        28   \n",
       "4         0.0         0.0         0.0        0.00         0.00         8   \n",
       "5         0.0         0.0         0.0        0.00         0.00        20   \n",
       "6         0.0         0.0         0.0        0.00         0.00         2   \n",
       "7         0.0         0.0         0.0        0.01         0.01         3   \n",
       "8         0.0         0.0         0.0        0.00         0.00        10   \n",
       "9         0.0         0.0         0.0        0.00         0.00       -39   \n",
       "\n",
       "   al_pos_x  dg_pos_x  dl_pos_x ancestral_allele temp  anc_dev  \\\n",
       "0        35       -24        19                T    T  derived   \n",
       "1        -1        11        -1                G    G  derived   \n",
       "2         9         4        -1                G    G  derived   \n",
       "3        30        28       -23                C    C  derived   \n",
       "4        44        -2        44                C    C  derived   \n",
       "5         1       -12        -1                G    G  derived   \n",
       "6         7        -4       -21                T    T  derived   \n",
       "7        11        23        11                A    A  derived   \n",
       "8        50        29         9                C    C  derived   \n",
       "9        29        40        -2                T    T  derived   \n",
       "\n",
       "   1KG_allele_count  1KG_allele_number  1KG_allele_frequency  1KG_EAS_AF  \\\n",
       "0             112.0             5096.0                  0.02        0.00   \n",
       "1             112.0             5096.0                  0.02        0.00   \n",
       "2             301.0             5096.0                  0.06        0.18   \n",
       "3             447.0             5096.0                  0.09        0.25   \n",
       "4             447.0             5096.0                  0.09        0.25   \n",
       "5             444.0             5096.0                  0.09        0.24   \n",
       "6             443.0             5096.0                  0.09        0.25   \n",
       "7             452.0             5096.0                  0.09        0.26   \n",
       "8              26.0             5096.0                  0.01        0.00   \n",
       "9             394.0             5096.0                  0.08        0.16   \n",
       "\n",
       "   1KG_EUR_AF  1KG_AFR_AF  1KG_AMR_AF  1KG_SAS_AF present_in_1KG  \\\n",
       "0        0.06         0.0        0.05        0.01            yes   \n",
       "1        0.06         0.0        0.05        0.01            yes   \n",
       "2        0.00         0.0        0.04        0.08            yes   \n",
       "3        0.01         0.0        0.13        0.10            yes   \n",
       "4        0.01         0.0        0.13        0.10            yes   \n",
       "5        0.01         0.0        0.13        0.09            yes   \n",
       "6        0.01         0.0        0.13        0.09            yes   \n",
       "7        0.01         0.0        0.13        0.09            yes   \n",
       "8        0.00         0.0        0.00        0.02            yes   \n",
       "9        0.01         0.0        0.14        0.12            yes   \n",
       "\n",
       "   1KG_non_ASW_AFR_AF  gnomAD_allele_count  gnomAD_allele_number  \\\n",
       "0            0.000000               6632.0              143094.0   \n",
       "1            0.000000               6626.0              143318.0   \n",
       "2            0.000000               1706.0              143300.0   \n",
       "3            0.000000               2943.0              143256.0   \n",
       "4            0.000000               2949.0              143216.0   \n",
       "5            0.000000               2939.0              143312.0   \n",
       "6            0.000000               2939.0              143296.0   \n",
       "7            0.000000               3291.0              143212.0   \n",
       "8            0.004873                294.0              143304.0   \n",
       "9            0.000000               2936.0              142852.0   \n",
       "\n",
       "   gnomAD_allele_frequency present_in_gnomAD      start  \\\n",
       "0                 0.046347               yes  3046710.0   \n",
       "1                 0.046233               yes  3046825.0   \n",
       "2                 0.011905               yes  3188155.0   \n",
       "3                 0.020544               yes  3199290.0   \n",
       "4                 0.020591               yes  3199726.0   \n",
       "5                 0.020508               yes  3209503.0   \n",
       "6                 0.020510               yes  3210304.0   \n",
       "7                 0.022980               yes  3212427.0   \n",
       "8                 0.002052               yes  3240651.0   \n",
       "9                 0.020553               yes  3240724.0   \n",
       "\n",
       "  Vernot_ancestral_allele Vernot_derived_allele  \\\n",
       "0                       T                     G   \n",
       "1                       G                     T   \n",
       "2                       G                     A   \n",
       "3                       C                     T   \n",
       "4                       C                     T   \n",
       "5                       G                     A   \n",
       "6                       T                     A   \n",
       "7                       A                     C   \n",
       "8                       C                     T   \n",
       "9                       T                     C   \n",
       "\n",
       "   Vernot_ancestral_derived_code  Vernot_AFA_AF  Vernot_AFR_AF  Vernot_AMR_AF  \\\n",
       "0                            1.0        0.01274        0.00000        0.04611   \n",
       "1                            1.0        0.01274        0.00000        0.04611   \n",
       "2                            1.0        0.00318        0.00000        0.04323   \n",
       "3                            1.0        0.00955        0.00000        0.12680   \n",
       "4                            1.0        0.00955        0.00000        0.12680   \n",
       "5                            1.0        0.00955        0.00000        0.12680   \n",
       "6                            1.0        0.00955        0.00000        0.12680   \n",
       "7                            1.0        0.00955        0.00000        0.13112   \n",
       "8                            1.0        0.00318        0.00397        0.00144   \n",
       "9                            1.0        0.00955        0.00000        0.13833   \n",
       "\n",
       "   Vernot_EAS_AF  Vernot_EUR_AF  Vernot_PNG_AF  Vernot_SAS_AF  \\\n",
       "0        0.00000        0.06362        0.00000        0.01125   \n",
       "1        0.00000        0.06362        0.00000        0.01125   \n",
       "2        0.18254        0.00298        0.07407        0.07975   \n",
       "3        0.24603        0.00596        0.11111        0.09509   \n",
       "4        0.24603        0.00596        0.11111        0.09509   \n",
       "5        0.24405        0.00596        0.11111        0.09305   \n",
       "6        0.24405        0.00596        0.11111        0.09100   \n",
       "7        0.25595        0.00696        0.11111        0.08384   \n",
       "8        0.00000        0.00000        0.18519        0.02045   \n",
       "9        0.16171        0.00596        0.11111        0.11656   \n",
       "\n",
       "  Vernot_Denisovan_base  Vernot_haplotype_tag Vernot_Neanderthal_base_1  \\\n",
       "0                     T  chr1_3015134_3046826                         T   \n",
       "1                     G  chr1_3015134_3046826                         G   \n",
       "2                     G  chr1_3066509_3209504                         G   \n",
       "3                     C  chr1_3066509_3209504                         C   \n",
       "4                     C  chr1_3066509_3209504                         C   \n",
       "5                     G  chr1_3066509_3209504                         G   \n",
       "6                     T  chr1_3166567_3212428                         T   \n",
       "7                     A  chr1_3166567_3212428                         A   \n",
       "8                     T  chr1_3195401_3250112                         C   \n",
       "9                     T  chr1_3240724_3259500                         T   \n",
       "\n",
       "  Vernot_Neanderthal_base_2 ref_allele_y altai_gt_y chagyrskaya_gt_y  \\\n",
       "0                         G            G        1/1              1/1   \n",
       "1                         T            T        1/1              1/1   \n",
       "2                         A            A        1/1              1/1   \n",
       "3                         T            T        1/1              1/1   \n",
       "4                         T            T        1/1              1/1   \n",
       "5                         A            A        1/1              1/1   \n",
       "6                         A            A        1/1              1/1   \n",
       "7                         C            C        1/1              1/1   \n",
       "8                       NaN            T        1/1              1/1   \n",
       "9                         C            C        1/1              1/1   \n",
       "\n",
       "  denisovan_gt_y vindija_gt_y alt_allele_y  ag_delta_y  al_delta_y  \\\n",
       "0            1/1          1/1            T         0.0         0.0   \n",
       "1            1/1          1/1            G         0.0         0.0   \n",
       "2            1/1          1/1            G         0.0         0.0   \n",
       "3            1/1          1/1            C         0.0         0.0   \n",
       "4            1/1          1/1            C         0.0         0.0   \n",
       "5            1/1          1/1            G         0.0         0.0   \n",
       "6            1/1          1/1            T         0.0         0.0   \n",
       "7            1/1          1/1            A         0.0         0.0   \n",
       "8            1/1          1/1            C         0.0         0.0   \n",
       "9            1/1          1/1            T         0.0         0.0   \n",
       "\n",
       "   dg_delta_y  dl_delta_y  ag_pos_y  al_pos_y  dg_pos_y  dl_pos_y  delta_max_y  \n",
       "0        0.00         0.0        35        -4        19       -24         0.00  \n",
       "1        0.00         0.0        -1       -49        -1        11         0.00  \n",
       "2        0.00         0.0         9       -27        -1         4         0.00  \n",
       "3        0.00         0.0        30        28       -23        28         0.00  \n",
       "4        0.00         0.0        44         8        44        -2         0.00  \n",
       "5        0.00         0.0         1        20        -1       -12         0.00  \n",
       "6        0.00         0.0         7         2       -21        -4         0.00  \n",
       "7        0.01         0.0        11         3        11        23         0.01  \n",
       "8        0.00         0.0        50        10         9        29         0.00  \n",
       "9        0.00         0.0        29       -39        -2        40         0.00  "
      ]
     },
     "execution_count": 615,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "introgressed_ref_tag_merge = pd.merge(introgressed_ref_tag, swapped, on = ['chrom','pos','annotation'])\n",
    "introgressed_ref_tag_merge.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 616,
   "id": "f5154bbf",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "24"
      ]
     },
     "execution_count": 616,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(introgressed_ref_tag_merge[(introgressed_ref_tag_merge['delta_max_x']>=0.2) & (introgressed_ref_tag_merge['delta_max_y']>=0.2)])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 617,
   "id": "01e91e03",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref_allele_x</th>\n",
       "      <th>alt_allele_x</th>\n",
       "      <th>variant_type</th>\n",
       "      <th>altai_gt_x</th>\n",
       "      <th>chagyrskaya_gt_x</th>\n",
       "      <th>denisovan_gt_x</th>\n",
       "      <th>vindija_gt_x</th>\n",
       "      <th>altai_gt_boolean</th>\n",
       "      <th>chagyrskaya_gt_boolean</th>\n",
       "      <th>denisovan_gt_boolean</th>\n",
       "      <th>vindija_gt_boolean</th>\n",
       "      <th>distribution</th>\n",
       "      <th>annotation</th>\n",
       "      <th>mis_oe</th>\n",
       "      <th>mis_z</th>\n",
       "      <th>lof_oe</th>\n",
       "      <th>lof_z</th>\n",
       "      <th>phyloP</th>\n",
       "      <th>ag_delta_x</th>\n",
       "      <th>al_delta_x</th>\n",
       "      <th>dg_delta_x</th>\n",
       "      <th>dl_delta_x</th>\n",
       "      <th>delta_max_x</th>\n",
       "      <th>ag_pos_x</th>\n",
       "      <th>al_pos_x</th>\n",
       "      <th>dg_pos_x</th>\n",
       "      <th>dl_pos_x</th>\n",
       "      <th>ancestral_allele</th>\n",
       "      <th>temp</th>\n",
       "      <th>anc_dev</th>\n",
       "      <th>1KG_allele_count</th>\n",
       "      <th>1KG_allele_number</th>\n",
       "      <th>1KG_allele_frequency</th>\n",
       "      <th>1KG_EAS_AF</th>\n",
       "      <th>1KG_EUR_AF</th>\n",
       "      <th>1KG_AFR_AF</th>\n",
       "      <th>1KG_AMR_AF</th>\n",
       "      <th>1KG_SAS_AF</th>\n",
       "      <th>present_in_1KG</th>\n",
       "      <th>1KG_non_ASW_AFR_AF</th>\n",
       "      <th>gnomAD_allele_count</th>\n",
       "      <th>gnomAD_allele_number</th>\n",
       "      <th>gnomAD_allele_frequency</th>\n",
       "      <th>present_in_gnomAD</th>\n",
       "      <th>start</th>\n",
       "      <th>Vernot_ancestral_allele</th>\n",
       "      <th>Vernot_derived_allele</th>\n",
       "      <th>Vernot_ancestral_derived_code</th>\n",
       "      <th>Vernot_AFA_AF</th>\n",
       "      <th>Vernot_AFR_AF</th>\n",
       "      <th>Vernot_AMR_AF</th>\n",
       "      <th>Vernot_EAS_AF</th>\n",
       "      <th>Vernot_EUR_AF</th>\n",
       "      <th>Vernot_PNG_AF</th>\n",
       "      <th>Vernot_SAS_AF</th>\n",
       "      <th>Vernot_Denisovan_base</th>\n",
       "      <th>Vernot_haplotype_tag</th>\n",
       "      <th>Vernot_Neanderthal_base_1</th>\n",
       "      <th>Vernot_Neanderthal_base_2</th>\n",
       "      <th>ref_allele_y</th>\n",
       "      <th>altai_gt_y</th>\n",
       "      <th>chagyrskaya_gt_y</th>\n",
       "      <th>denisovan_gt_y</th>\n",
       "      <th>vindija_gt_y</th>\n",
       "      <th>alt_allele_y</th>\n",
       "      <th>ag_delta_y</th>\n",
       "      <th>al_delta_y</th>\n",
       "      <th>dg_delta_y</th>\n",
       "      <th>dl_delta_y</th>\n",
       "      <th>ag_pos_y</th>\n",
       "      <th>al_pos_y</th>\n",
       "      <th>dg_pos_y</th>\n",
       "      <th>dl_pos_y</th>\n",
       "      <th>delta_max_y</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>305</th>\n",
       "      <td>chr1</td>\n",
       "      <td>52820680</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Other</td>\n",
       "      <td>CC2D1B</td>\n",
       "      <td>1.12620</td>\n",
       "      <td>-1.00890</td>\n",
       "      <td>0.694430</td>\n",
       "      <td>2.038800</td>\n",
       "      <td>-0.140</td>\n",
       "      <td>0.23</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.23</td>\n",
       "      <td>-22</td>\n",
       "      <td>-40</td>\n",
       "      <td>-22</td>\n",
       "      <td>34</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>derived</td>\n",
       "      <td>90.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.03</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.05</td>\n",
       "      <td>yes</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>443.0</td>\n",
       "      <td>143266.0</td>\n",
       "      <td>0.003092</td>\n",
       "      <td>yes</td>\n",
       "      <td>52820679.0</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.00637</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.02877</td>\n",
       "      <td>0.00497</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.05521</td>\n",
       "      <td>C</td>\n",
       "      <td>chr1_52325475_53062486</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>C</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.23</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-40</td>\n",
       "      <td>-22</td>\n",
       "      <td>34</td>\n",
       "      <td>-22</td>\n",
       "      <td>0.23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>492</th>\n",
       "      <td>chr1</td>\n",
       "      <td>152944501</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Other</td>\n",
       "      <td>SPRR4</td>\n",
       "      <td>1.49550</td>\n",
       "      <td>-1.10600</td>\n",
       "      <td>0.714380</td>\n",
       "      <td>0.442850</td>\n",
       "      <td>0.383</td>\n",
       "      <td>0.23</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.23</td>\n",
       "      <td>13</td>\n",
       "      <td>2</td>\n",
       "      <td>12</td>\n",
       "      <td>-1</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>derived</td>\n",
       "      <td>2.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>yes</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>4.0</td>\n",
       "      <td>143152.0</td>\n",
       "      <td>0.000028</td>\n",
       "      <td>yes</td>\n",
       "      <td>152944500.0</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00198</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>A</td>\n",
       "      <td>chr1_152942848_153033467</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>A</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.23</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>2</td>\n",
       "      <td>13</td>\n",
       "      <td>-1</td>\n",
       "      <td>12</td>\n",
       "      <td>0.23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>575</th>\n",
       "      <td>chr1</td>\n",
       "      <td>183750131</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>Altai</td>\n",
       "      <td>RGL1</td>\n",
       "      <td>0.70413</td>\n",
       "      <td>2.22380</td>\n",
       "      <td>0.177530</td>\n",
       "      <td>5.116100</td>\n",
       "      <td>0.655</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.26</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.26</td>\n",
       "      <td>-7</td>\n",
       "      <td>-11</td>\n",
       "      <td>-7</td>\n",
       "      <td>-11</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>derived</td>\n",
       "      <td>50.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>yes</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>194.0</td>\n",
       "      <td>143252.0</td>\n",
       "      <td>0.001354</td>\n",
       "      <td>yes</td>\n",
       "      <td>183750130.0</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.04861</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.01852</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>C</td>\n",
       "      <td>chr1_183411087_183827254</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>C</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.26</td>\n",
       "      <td>-11</td>\n",
       "      <td>-7</td>\n",
       "      <td>-11</td>\n",
       "      <td>-7</td>\n",
       "      <td>0.26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1018</th>\n",
       "      <td>chr10</td>\n",
       "      <td>17373518</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>ST8SIA6</td>\n",
       "      <td>1.08980</td>\n",
       "      <td>-0.43129</td>\n",
       "      <td>1.008600</td>\n",
       "      <td>-0.029733</td>\n",
       "      <td>0.533</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.22</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.22</td>\n",
       "      <td>33</td>\n",
       "      <td>-19</td>\n",
       "      <td>30</td>\n",
       "      <td>-19</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>derived</td>\n",
       "      <td>56.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.04</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.00</td>\n",
       "      <td>yes</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2946.0</td>\n",
       "      <td>143262.0</td>\n",
       "      <td>0.020564</td>\n",
       "      <td>yes</td>\n",
       "      <td>17373517.0</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.01274</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.01729</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.03479</td>\n",
       "      <td>0.01852</td>\n",
       "      <td>0.00307</td>\n",
       "      <td>T</td>\n",
       "      <td>chr10_17338795_17417730</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>T</td>\n",
       "      <td>0.22</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-19</td>\n",
       "      <td>33</td>\n",
       "      <td>-19</td>\n",
       "      <td>30</td>\n",
       "      <td>0.22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1655</th>\n",
       "      <td>chr11</td>\n",
       "      <td>84191111</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>DLG2</td>\n",
       "      <td>0.74745</td>\n",
       "      <td>2.07620</td>\n",
       "      <td>0.212280</td>\n",
       "      <td>5.712300</td>\n",
       "      <td>-0.959</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.40</td>\n",
       "      <td>0.40</td>\n",
       "      <td>1</td>\n",
       "      <td>29</td>\n",
       "      <td>1</td>\n",
       "      <td>-1</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>derived</td>\n",
       "      <td>12.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.01</td>\n",
       "      <td>yes</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>38.0</td>\n",
       "      <td>143298.0</td>\n",
       "      <td>0.000265</td>\n",
       "      <td>yes</td>\n",
       "      <td>84191110.0</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.01227</td>\n",
       "      <td>T</td>\n",
       "      <td>chr11_84051845_84211811</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>T</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.40</td>\n",
       "      <td>0.00</td>\n",
       "      <td>29</td>\n",
       "      <td>1</td>\n",
       "      <td>-1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.40</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1707</th>\n",
       "      <td>chr11</td>\n",
       "      <td>99345359</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>CNTN5</td>\n",
       "      <td>1.03930</td>\n",
       "      <td>-0.33557</td>\n",
       "      <td>0.327200</td>\n",
       "      <td>4.624200</td>\n",
       "      <td>-0.229</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.67</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.67</td>\n",
       "      <td>-4</td>\n",
       "      <td>48</td>\n",
       "      <td>-5</td>\n",
       "      <td>37</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>derived</td>\n",
       "      <td>177.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.03</td>\n",
       "      <td>0.11</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.07</td>\n",
       "      <td>yes</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>551.0</td>\n",
       "      <td>143130.0</td>\n",
       "      <td>0.003850</td>\n",
       "      <td>yes</td>\n",
       "      <td>99345358.0</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.00318</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.10615</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.11111</td>\n",
       "      <td>0.06748</td>\n",
       "      <td>C</td>\n",
       "      <td>chr11_99309226_99359652</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>C</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.67</td>\n",
       "      <td>48</td>\n",
       "      <td>-4</td>\n",
       "      <td>37</td>\n",
       "      <td>-5</td>\n",
       "      <td>0.67</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3326</th>\n",
       "      <td>chr17</td>\n",
       "      <td>14109360</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>COX10</td>\n",
       "      <td>1.03160</td>\n",
       "      <td>-0.18011</td>\n",
       "      <td>0.290420</td>\n",
       "      <td>2.728300</td>\n",
       "      <td>0.338</td>\n",
       "      <td>0.20</td>\n",
       "      <td>0.04</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.20</td>\n",
       "      <td>1</td>\n",
       "      <td>38</td>\n",
       "      <td>50</td>\n",
       "      <td>-23</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>derived</td>\n",
       "      <td>155.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.03</td>\n",
       "      <td>0.08</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.08</td>\n",
       "      <td>0.01</td>\n",
       "      <td>yes</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1030.0</td>\n",
       "      <td>143224.0</td>\n",
       "      <td>0.007192</td>\n",
       "      <td>yes</td>\n",
       "      <td>14109359.0</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.00637</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.08501</td>\n",
       "      <td>0.08333</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.16667</td>\n",
       "      <td>0.00818</td>\n",
       "      <td>G</td>\n",
       "      <td>chr17_13902596_14111504</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>T</td>\n",
       "      <td>0.04</td>\n",
       "      <td>0.21</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>38</td>\n",
       "      <td>1</td>\n",
       "      <td>-23</td>\n",
       "      <td>50</td>\n",
       "      <td>0.21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3577</th>\n",
       "      <td>chr18</td>\n",
       "      <td>60003587</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>TNFRSF11A</td>\n",
       "      <td>0.80508</td>\n",
       "      <td>1.28940</td>\n",
       "      <td>0.330780</td>\n",
       "      <td>3.049700</td>\n",
       "      <td>0.650</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.31</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.28</td>\n",
       "      <td>0.31</td>\n",
       "      <td>3</td>\n",
       "      <td>-40</td>\n",
       "      <td>3</td>\n",
       "      <td>33</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>444.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.09</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.25</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.12</td>\n",
       "      <td>0.10</td>\n",
       "      <td>yes</td>\n",
       "      <td>0.003899</td>\n",
       "      <td>21727.0</td>\n",
       "      <td>143202.0</td>\n",
       "      <td>0.151723</td>\n",
       "      <td>yes</td>\n",
       "      <td>60003586.0</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.01592</td>\n",
       "      <td>0.00397</td>\n",
       "      <td>0.11671</td>\n",
       "      <td>0.00198</td>\n",
       "      <td>0.24950</td>\n",
       "      <td>0.09259</td>\n",
       "      <td>0.09714</td>\n",
       "      <td>A</td>\n",
       "      <td>chr18_59569521_60089689</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>G</td>\n",
       "      <td>0.30</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.29</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-40</td>\n",
       "      <td>3</td>\n",
       "      <td>33</td>\n",
       "      <td>3</td>\n",
       "      <td>0.30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3645</th>\n",
       "      <td>chr19</td>\n",
       "      <td>33410289</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>CEP89</td>\n",
       "      <td>0.93498</td>\n",
       "      <td>0.47493</td>\n",
       "      <td>0.939730</td>\n",
       "      <td>0.364400</td>\n",
       "      <td>-0.642</td>\n",
       "      <td>0.47</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.71</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.71</td>\n",
       "      <td>41</td>\n",
       "      <td>-29</td>\n",
       "      <td>5</td>\n",
       "      <td>-29</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>ancestral</td>\n",
       "      <td>4744.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.93</td>\n",
       "      <td>0.90</td>\n",
       "      <td>0.86</td>\n",
       "      <td>0.99</td>\n",
       "      <td>0.93</td>\n",
       "      <td>0.96</td>\n",
       "      <td>yes</td>\n",
       "      <td>0.999025</td>\n",
       "      <td>129412.0</td>\n",
       "      <td>143248.0</td>\n",
       "      <td>0.903412</td>\n",
       "      <td>yes</td>\n",
       "      <td>33410288.0</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.03503</td>\n",
       "      <td>0.00099</td>\n",
       "      <td>0.06916</td>\n",
       "      <td>0.09821</td>\n",
       "      <td>0.14612</td>\n",
       "      <td>0.07407</td>\n",
       "      <td>0.04090</td>\n",
       "      <td>C</td>\n",
       "      <td>chr19_33040186_33456652</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>T</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.47</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.71</td>\n",
       "      <td>-29</td>\n",
       "      <td>41</td>\n",
       "      <td>-29</td>\n",
       "      <td>5</td>\n",
       "      <td>0.71</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4708</th>\n",
       "      <td>chr20</td>\n",
       "      <td>62224595</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>GMEB2</td>\n",
       "      <td>0.66048</td>\n",
       "      <td>2.20200</td>\n",
       "      <td>0.049787</td>\n",
       "      <td>3.946200</td>\n",
       "      <td>-0.837</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.41</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.41</td>\n",
       "      <td>15</td>\n",
       "      <td>-44</td>\n",
       "      <td>2</td>\n",
       "      <td>-2</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>1138.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.22</td>\n",
       "      <td>0.51</td>\n",
       "      <td>0.21</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.11</td>\n",
       "      <td>0.31</td>\n",
       "      <td>yes</td>\n",
       "      <td>0.000975</td>\n",
       "      <td>23692.0</td>\n",
       "      <td>143104.0</td>\n",
       "      <td>0.165558</td>\n",
       "      <td>yes</td>\n",
       "      <td>62224594.0</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.03503</td>\n",
       "      <td>0.00099</td>\n",
       "      <td>0.10663</td>\n",
       "      <td>0.51488</td>\n",
       "      <td>0.21173</td>\n",
       "      <td>0.03704</td>\n",
       "      <td>0.30982</td>\n",
       "      <td>G</td>\n",
       "      <td>chr20_62195671_62229244</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>G</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.41</td>\n",
       "      <td>-44</td>\n",
       "      <td>15</td>\n",
       "      <td>-2</td>\n",
       "      <td>2</td>\n",
       "      <td>0.41</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5235</th>\n",
       "      <td>chr3</td>\n",
       "      <td>52486376</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>TNNC1</td>\n",
       "      <td>0.49480</td>\n",
       "      <td>1.76820</td>\n",
       "      <td>0.125160</td>\n",
       "      <td>2.291500</td>\n",
       "      <td>0.459</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.34</td>\n",
       "      <td>0.34</td>\n",
       "      <td>-23</td>\n",
       "      <td>2</td>\n",
       "      <td>25</td>\n",
       "      <td>-1</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>derived</td>\n",
       "      <td>40.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.03</td>\n",
       "      <td>yes</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>494.0</td>\n",
       "      <td>143210.0</td>\n",
       "      <td>0.003449</td>\n",
       "      <td>yes</td>\n",
       "      <td>52486375.0</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.00318</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00288</td>\n",
       "      <td>0.00794</td>\n",
       "      <td>0.00298</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.02658</td>\n",
       "      <td>T</td>\n",
       "      <td>chr3_52214838_52516460</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>T</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.35</td>\n",
       "      <td>0.00</td>\n",
       "      <td>2</td>\n",
       "      <td>-23</td>\n",
       "      <td>-1</td>\n",
       "      <td>25</td>\n",
       "      <td>0.35</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5409</th>\n",
       "      <td>chr4</td>\n",
       "      <td>38805942</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>TLR1</td>\n",
       "      <td>1.09920</td>\n",
       "      <td>-0.70157</td>\n",
       "      <td>1.411200</td>\n",
       "      <td>-1.727200</td>\n",
       "      <td>-0.219</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.38</td>\n",
       "      <td>0.38</td>\n",
       "      <td>27</td>\n",
       "      <td>-15</td>\n",
       "      <td>1</td>\n",
       "      <td>-15</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>911.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.18</td>\n",
       "      <td>0.39</td>\n",
       "      <td>0.22</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.15</td>\n",
       "      <td>0.16</td>\n",
       "      <td>yes</td>\n",
       "      <td>0.001949</td>\n",
       "      <td>19931.0</td>\n",
       "      <td>143224.0</td>\n",
       "      <td>0.139160</td>\n",
       "      <td>yes</td>\n",
       "      <td>38805941.0</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.04459</td>\n",
       "      <td>0.00198</td>\n",
       "      <td>0.14986</td>\n",
       "      <td>0.39484</td>\n",
       "      <td>0.22068</td>\n",
       "      <td>0.05556</td>\n",
       "      <td>0.16360</td>\n",
       "      <td>G</td>\n",
       "      <td>chr4_38294200_38857896</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>G</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.37</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-15</td>\n",
       "      <td>27</td>\n",
       "      <td>-15</td>\n",
       "      <td>1</td>\n",
       "      <td>0.37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5490</th>\n",
       "      <td>chr4</td>\n",
       "      <td>100460531</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>C4orf17</td>\n",
       "      <td>0.87519</td>\n",
       "      <td>0.60341</td>\n",
       "      <td>0.898160</td>\n",
       "      <td>0.359040</td>\n",
       "      <td>0.533</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.35</td>\n",
       "      <td>0.35</td>\n",
       "      <td>-4</td>\n",
       "      <td>-40</td>\n",
       "      <td>-1</td>\n",
       "      <td>-4</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>derived</td>\n",
       "      <td>50.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.04</td>\n",
       "      <td>0.00</td>\n",
       "      <td>yes</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>3317.0</td>\n",
       "      <td>143342.0</td>\n",
       "      <td>0.023141</td>\n",
       "      <td>yes</td>\n",
       "      <td>100460530.0</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.03602</td>\n",
       "      <td>0.00099</td>\n",
       "      <td>0.02087</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00204</td>\n",
       "      <td>G</td>\n",
       "      <td>chr4_100349203_100638613</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>A</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.35</td>\n",
       "      <td>0.01</td>\n",
       "      <td>-40</td>\n",
       "      <td>-4</td>\n",
       "      <td>-4</td>\n",
       "      <td>-1</td>\n",
       "      <td>0.35</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5611</th>\n",
       "      <td>chr4</td>\n",
       "      <td>135122662</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Other</td>\n",
       "      <td>PABPC4L</td>\n",
       "      <td>0.73213</td>\n",
       "      <td>1.47170</td>\n",
       "      <td>0.557620</td>\n",
       "      <td>1.452400</td>\n",
       "      <td>0.561</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.80</td>\n",
       "      <td>0.08</td>\n",
       "      <td>0.80</td>\n",
       "      <td>2</td>\n",
       "      <td>40</td>\n",
       "      <td>2</td>\n",
       "      <td>-2</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>yes</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.0</td>\n",
       "      <td>143316.0</td>\n",
       "      <td>0.000014</td>\n",
       "      <td>yes</td>\n",
       "      <td>135122661.0</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00099</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>G</td>\n",
       "      <td>chr4_135066013_135400700</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>G</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.08</td>\n",
       "      <td>0.80</td>\n",
       "      <td>40</td>\n",
       "      <td>2</td>\n",
       "      <td>-2</td>\n",
       "      <td>2</td>\n",
       "      <td>0.80</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5941</th>\n",
       "      <td>chr5</td>\n",
       "      <td>118669326</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Denisovan</td>\n",
       "      <td>TNFAIP8</td>\n",
       "      <td>0.87632</td>\n",
       "      <td>0.44541</td>\n",
       "      <td>0.204890</td>\n",
       "      <td>1.627700</td>\n",
       "      <td>0.650</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.22</td>\n",
       "      <td>0.22</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>derived</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>yes</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>118669325.0</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.11111</td>\n",
       "      <td>0.00102</td>\n",
       "      <td>C</td>\n",
       "      <td>chr5_118661135_118933833</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>G</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.22</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>0.22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5961</th>\n",
       "      <td>chr5</td>\n",
       "      <td>126989007</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>Altai</td>\n",
       "      <td>CTXN3</td>\n",
       "      <td>0.91784</td>\n",
       "      <td>0.19514</td>\n",
       "      <td>0.528190</td>\n",
       "      <td>0.601570</td>\n",
       "      <td>0.528</td>\n",
       "      <td>0.34</td>\n",
       "      <td>0.91</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.91</td>\n",
       "      <td>14</td>\n",
       "      <td>2</td>\n",
       "      <td>-34</td>\n",
       "      <td>2</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>derived</td>\n",
       "      <td>2.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>yes</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>10.0</td>\n",
       "      <td>143290.0</td>\n",
       "      <td>0.000070</td>\n",
       "      <td>yes</td>\n",
       "      <td>126989006.0</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00204</td>\n",
       "      <td>A</td>\n",
       "      <td>chr5_126989006_127537918</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>A</td>\n",
       "      <td>0.91</td>\n",
       "      <td>0.34</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>2</td>\n",
       "      <td>14</td>\n",
       "      <td>2</td>\n",
       "      <td>-34</td>\n",
       "      <td>0.91</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6599</th>\n",
       "      <td>chr5</td>\n",
       "      <td>148430560</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Other</td>\n",
       "      <td>SH3TC2</td>\n",
       "      <td>1.07760</td>\n",
       "      <td>-0.71307</td>\n",
       "      <td>0.608270</td>\n",
       "      <td>2.673700</td>\n",
       "      <td>-1.235</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.32</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.32</td>\n",
       "      <td>-6</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>-41</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>derived</td>\n",
       "      <td>81.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.07</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.00</td>\n",
       "      <td>yes</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>214.0</td>\n",
       "      <td>143150.0</td>\n",
       "      <td>0.001495</td>\n",
       "      <td>yes</td>\n",
       "      <td>148430559.0</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.00318</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00720</td>\n",
       "      <td>0.07242</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.03704</td>\n",
       "      <td>0.00204</td>\n",
       "      <td>A</td>\n",
       "      <td>chr5_148339333_148539017</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>A</td>\n",
       "      <td>0.32</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1</td>\n",
       "      <td>-6</td>\n",
       "      <td>-41</td>\n",
       "      <td>4</td>\n",
       "      <td>0.32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6932</th>\n",
       "      <td>chr6</td>\n",
       "      <td>51889358</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>PKHD1</td>\n",
       "      <td>1.06000</td>\n",
       "      <td>-0.98178</td>\n",
       "      <td>0.580270</td>\n",
       "      <td>5.281600</td>\n",
       "      <td>-1.093</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.42</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.42</td>\n",
       "      <td>31</td>\n",
       "      <td>14</td>\n",
       "      <td>1</td>\n",
       "      <td>10</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>derived</td>\n",
       "      <td>114.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.08</td>\n",
       "      <td>yes</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2957.0</td>\n",
       "      <td>143284.0</td>\n",
       "      <td>0.020637</td>\n",
       "      <td>yes</td>\n",
       "      <td>51889357.0</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.00318</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.01009</td>\n",
       "      <td>0.00694</td>\n",
       "      <td>0.02286</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.07771</td>\n",
       "      <td>T/C</td>\n",
       "      <td>chr6_51841815_51972144</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>T</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.42</td>\n",
       "      <td>14</td>\n",
       "      <td>31</td>\n",
       "      <td>10</td>\n",
       "      <td>1</td>\n",
       "      <td>0.42</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6933</th>\n",
       "      <td>chr6</td>\n",
       "      <td>52138226</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>Altai</td>\n",
       "      <td>MCM3</td>\n",
       "      <td>0.89914</td>\n",
       "      <td>0.80451</td>\n",
       "      <td>0.535230</td>\n",
       "      <td>2.884000</td>\n",
       "      <td>0.563</td>\n",
       "      <td>0.21</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.21</td>\n",
       "      <td>-7</td>\n",
       "      <td>-40</td>\n",
       "      <td>-7</td>\n",
       "      <td>3</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>derived</td>\n",
       "      <td>782.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.15</td>\n",
       "      <td>0.24</td>\n",
       "      <td>0.17</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.09</td>\n",
       "      <td>0.29</td>\n",
       "      <td>yes</td>\n",
       "      <td>0.001949</td>\n",
       "      <td>17455.0</td>\n",
       "      <td>143064.0</td>\n",
       "      <td>0.122008</td>\n",
       "      <td>yes</td>\n",
       "      <td>52138225.0</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.01274</td>\n",
       "      <td>0.00198</td>\n",
       "      <td>0.09366</td>\n",
       "      <td>0.24206</td>\n",
       "      <td>0.16600</td>\n",
       "      <td>0.09259</td>\n",
       "      <td>0.29448</td>\n",
       "      <td>C</td>\n",
       "      <td>chr6_52138225_52183126</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>C</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.21</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-40</td>\n",
       "      <td>-7</td>\n",
       "      <td>3</td>\n",
       "      <td>-7</td>\n",
       "      <td>0.21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7271</th>\n",
       "      <td>chr6</td>\n",
       "      <td>119524266</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Other</td>\n",
       "      <td>MAN1A1</td>\n",
       "      <td>0.77221</td>\n",
       "      <td>1.51360</td>\n",
       "      <td>0.255850</td>\n",
       "      <td>3.856000</td>\n",
       "      <td>0.430</td>\n",
       "      <td>0.30</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.29</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.30</td>\n",
       "      <td>43</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>derived</td>\n",
       "      <td>98.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.04</td>\n",
       "      <td>0.01</td>\n",
       "      <td>yes</td>\n",
       "      <td>0.002924</td>\n",
       "      <td>5549.0</td>\n",
       "      <td>143236.0</td>\n",
       "      <td>0.038740</td>\n",
       "      <td>yes</td>\n",
       "      <td>119524265.0</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.00318</td>\n",
       "      <td>0.00298</td>\n",
       "      <td>0.04179</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.04871</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.01431</td>\n",
       "      <td>C</td>\n",
       "      <td>chr6_119475382_119646998</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>C</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.30</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.29</td>\n",
       "      <td>4</td>\n",
       "      <td>43</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>0.30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7956</th>\n",
       "      <td>chr7</td>\n",
       "      <td>126797408</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>Other</td>\n",
       "      <td>GRM8</td>\n",
       "      <td>0.88746</td>\n",
       "      <td>0.91440</td>\n",
       "      <td>0.429960</td>\n",
       "      <td>3.417800</td>\n",
       "      <td>0.482</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.43</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.43</td>\n",
       "      <td>47</td>\n",
       "      <td>-16</td>\n",
       "      <td>7</td>\n",
       "      <td>-16</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>derived</td>\n",
       "      <td>2.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>yes</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.0</td>\n",
       "      <td>143256.0</td>\n",
       "      <td>0.000014</td>\n",
       "      <td>yes</td>\n",
       "      <td>126797407.0</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00198</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>T</td>\n",
       "      <td>chr7_126660339_126797408</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>T</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.43</td>\n",
       "      <td>-16</td>\n",
       "      <td>47</td>\n",
       "      <td>-16</td>\n",
       "      <td>7</td>\n",
       "      <td>0.43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7999</th>\n",
       "      <td>chr7</td>\n",
       "      <td>140801296</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>Shared</td>\n",
       "      <td>TMEM178B</td>\n",
       "      <td>0.50215</td>\n",
       "      <td>2.31520</td>\n",
       "      <td>0.070903</td>\n",
       "      <td>3.233300</td>\n",
       "      <td>-0.490</td>\n",
       "      <td>0.25</td>\n",
       "      <td>-0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.25</td>\n",
       "      <td>-13</td>\n",
       "      <td>-30</td>\n",
       "      <td>11</td>\n",
       "      <td>-13</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>derived</td>\n",
       "      <td>4.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>yes</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>8.0</td>\n",
       "      <td>143328.0</td>\n",
       "      <td>0.000056</td>\n",
       "      <td>yes</td>\n",
       "      <td>140801295.0</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00099</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00307</td>\n",
       "      <td>C</td>\n",
       "      <td>chr7_140306146_140804665</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>T</td>\n",
       "      <td>-0.00</td>\n",
       "      <td>0.25</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>27</td>\n",
       "      <td>-13</td>\n",
       "      <td>-13</td>\n",
       "      <td>11</td>\n",
       "      <td>0.25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8011</th>\n",
       "      <td>chr7</td>\n",
       "      <td>140958656</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>TMEM178B</td>\n",
       "      <td>0.50215</td>\n",
       "      <td>2.31520</td>\n",
       "      <td>0.070903</td>\n",
       "      <td>3.233300</td>\n",
       "      <td>0.443</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.26</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.26</td>\n",
       "      <td>-1</td>\n",
       "      <td>3</td>\n",
       "      <td>-1</td>\n",
       "      <td>3</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>derived</td>\n",
       "      <td>18.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.02</td>\n",
       "      <td>yes</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>60.0</td>\n",
       "      <td>143274.0</td>\n",
       "      <td>0.000419</td>\n",
       "      <td>yes</td>\n",
       "      <td>140958655.0</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.01840</td>\n",
       "      <td>A</td>\n",
       "      <td>chr7_140934039_141109377</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>A</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.26</td>\n",
       "      <td>3</td>\n",
       "      <td>-1</td>\n",
       "      <td>3</td>\n",
       "      <td>-1</td>\n",
       "      <td>0.26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8081</th>\n",
       "      <td>chr7</td>\n",
       "      <td>158543832</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Other</td>\n",
       "      <td>ESYT2</td>\n",
       "      <td>0.92723</td>\n",
       "      <td>0.55098</td>\n",
       "      <td>0.377280</td>\n",
       "      <td>3.873500</td>\n",
       "      <td>0.561</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.21</td>\n",
       "      <td>0.21</td>\n",
       "      <td>-40</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>derived</td>\n",
       "      <td>120.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.11</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.01</td>\n",
       "      <td>yes</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>481.0</td>\n",
       "      <td>143162.0</td>\n",
       "      <td>0.003360</td>\n",
       "      <td>yes</td>\n",
       "      <td>158543831.0</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.11210</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00511</td>\n",
       "      <td>A</td>\n",
       "      <td>chr7_158399204_158736324</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>C</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.22</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1</td>\n",
       "      <td>-40</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0.22</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      chrom        pos ref_allele_x alt_allele_x variant_type altai_gt_x  \\\n",
       "305    chr1   52820680            C            T          snv        0/1   \n",
       "492    chr1  152944501            A            T          snv        0/1   \n",
       "575    chr1  183750131            C            A          snv        0/1   \n",
       "1018  chr10   17373518            T            C          snv        0/1   \n",
       "1655  chr11   84191111            T            C          snv        0/1   \n",
       "1707  chr11   99345359            C            G          snv        0/1   \n",
       "3326  chr17   14109360            T            G          snv        0/0   \n",
       "3577  chr18   60003587            G            A          snv        0/1   \n",
       "3645  chr19   33410289            T            C          snv        0/0   \n",
       "4708  chr20   62224595            G            A          snv        0/1   \n",
       "5235   chr3   52486376            T            C          snv        0/1   \n",
       "5409   chr4   38805942            G            C          snv        0/1   \n",
       "5490   chr4  100460531            A            G          snv        0/1   \n",
       "5611   chr4  135122662            G            A          snv        0/1   \n",
       "5941   chr5  118669326            G            C          snv        0/0   \n",
       "5961   chr5  126989007            A            G          snv        0/1   \n",
       "6599   chr5  148430560            A            C          snv        0/1   \n",
       "6932   chr6   51889358            T            C          snv        0/1   \n",
       "6933   chr6   52138226            C            G          snv        0/1   \n",
       "7271   chr6  119524266            C            T          snv        0/1   \n",
       "7956   chr7  126797408            T            A          snv        0/1   \n",
       "7999   chr7  140801296            T            C          snv        0/1   \n",
       "8011   chr7  140958656            A            G          snv        0/1   \n",
       "8081   chr7  158543832            C            A          snv        0/1   \n",
       "\n",
       "     chagyrskaya_gt_x denisovan_gt_x vindija_gt_x  altai_gt_boolean  \\\n",
       "305               0/0            0/0          0/1              True   \n",
       "492               0/0            0/0          0/1              True   \n",
       "575               0/0            0/0          0/0              True   \n",
       "1018              1/1            0/0          1/1              True   \n",
       "1655              1/1            0/0          1/1              True   \n",
       "1707              0/1            0/0          1/1              True   \n",
       "3326              0/0            1/1          0/0             False   \n",
       "3577              1/1            1/1          1/1              True   \n",
       "3645              0/0            1/1          0/0             False   \n",
       "4708              1/1            0/0          0/1              True   \n",
       "5235              1/1            0/0          1/1              True   \n",
       "5409              1/1            0/0          1/1              True   \n",
       "5490              1/1            1/1          1/1              True   \n",
       "5611              0/0            0/0          0/1              True   \n",
       "5941              0/0            1/1          0/0             False   \n",
       "5961              0/0            0/0          0/0              True   \n",
       "6599              0/0            0/0          1/1              True   \n",
       "6932              1/1            0/1          1/1              True   \n",
       "6933              0/0            0/0          0/0              True   \n",
       "7271              0/0            0/0          1/1              True   \n",
       "7956              1/1            0/0          0/0              True   \n",
       "7999              1/1            1/1          1/1              True   \n",
       "8011              1/1            0/0          0/1              True   \n",
       "8081              1/1            1/1          0/0              True   \n",
       "\n",
       "      chagyrskaya_gt_boolean  denisovan_gt_boolean  vindija_gt_boolean  \\\n",
       "305                    False                 False                True   \n",
       "492                    False                 False                True   \n",
       "575                    False                 False               False   \n",
       "1018                    True                 False                True   \n",
       "1655                    True                 False                True   \n",
       "1707                    True                 False                True   \n",
       "3326                   False                  True               False   \n",
       "3577                    True                  True                True   \n",
       "3645                   False                  True               False   \n",
       "4708                    True                 False                True   \n",
       "5235                    True                 False                True   \n",
       "5409                    True                 False                True   \n",
       "5490                    True                  True                True   \n",
       "5611                   False                 False                True   \n",
       "5941                   False                  True               False   \n",
       "5961                   False                 False               False   \n",
       "6599                   False                 False                True   \n",
       "6932                    True                  True                True   \n",
       "6933                   False                 False               False   \n",
       "7271                   False                 False                True   \n",
       "7956                    True                 False               False   \n",
       "7999                    True                  True                True   \n",
       "8011                    True                 False                True   \n",
       "8081                    True                  True               False   \n",
       "\n",
       "     distribution annotation   mis_oe    mis_z    lof_oe     lof_z  phyloP  \\\n",
       "305         Other     CC2D1B  1.12620 -1.00890  0.694430  2.038800  -0.140   \n",
       "492         Other      SPRR4  1.49550 -1.10600  0.714380  0.442850   0.383   \n",
       "575         Altai       RGL1  0.70413  2.22380  0.177530  5.116100   0.655   \n",
       "1018  Neanderthal    ST8SIA6  1.08980 -0.43129  1.008600 -0.029733   0.533   \n",
       "1655  Neanderthal       DLG2  0.74745  2.07620  0.212280  5.712300  -0.959   \n",
       "1707  Neanderthal      CNTN5  1.03930 -0.33557  0.327200  4.624200  -0.229   \n",
       "3326    Denisovan      COX10  1.03160 -0.18011  0.290420  2.728300   0.338   \n",
       "3577       Shared  TNFRSF11A  0.80508  1.28940  0.330780  3.049700   0.650   \n",
       "3645    Denisovan      CEP89  0.93498  0.47493  0.939730  0.364400  -0.642   \n",
       "4708  Neanderthal      GMEB2  0.66048  2.20200  0.049787  3.946200  -0.837   \n",
       "5235  Neanderthal      TNNC1  0.49480  1.76820  0.125160  2.291500   0.459   \n",
       "5409  Neanderthal       TLR1  1.09920 -0.70157  1.411200 -1.727200  -0.219   \n",
       "5490       Shared    C4orf17  0.87519  0.60341  0.898160  0.359040   0.533   \n",
       "5611        Other    PABPC4L  0.73213  1.47170  0.557620  1.452400   0.561   \n",
       "5941    Denisovan    TNFAIP8  0.87632  0.44541  0.204890  1.627700   0.650   \n",
       "5961        Altai      CTXN3  0.91784  0.19514  0.528190  0.601570   0.528   \n",
       "6599        Other     SH3TC2  1.07760 -0.71307  0.608270  2.673700  -1.235   \n",
       "6932       Shared      PKHD1  1.06000 -0.98178  0.580270  5.281600  -1.093   \n",
       "6933        Altai       MCM3  0.89914  0.80451  0.535230  2.884000   0.563   \n",
       "7271        Other     MAN1A1  0.77221  1.51360  0.255850  3.856000   0.430   \n",
       "7956        Other       GRM8  0.88746  0.91440  0.429960  3.417800   0.482   \n",
       "7999       Shared   TMEM178B  0.50215  2.31520  0.070903  3.233300  -0.490   \n",
       "8011  Neanderthal   TMEM178B  0.50215  2.31520  0.070903  3.233300   0.443   \n",
       "8081        Other      ESYT2  0.92723  0.55098  0.377280  3.873500   0.561   \n",
       "\n",
       "      ag_delta_x  al_delta_x  dg_delta_x  dl_delta_x  delta_max_x  ag_pos_x  \\\n",
       "305         0.23        0.00        0.00        0.01         0.23       -22   \n",
       "492         0.23        0.00        0.00        0.00         0.23        13   \n",
       "575         0.00        0.00        0.26        0.01         0.26        -7   \n",
       "1018        0.05        0.22        0.00        0.00         0.22        33   \n",
       "1655        0.00        0.00        0.00        0.40         0.40         1   \n",
       "1707        0.00        0.00        0.67        0.00         0.67        -4   \n",
       "3326        0.20        0.04        0.00        0.00         0.20         1   \n",
       "3577        0.00        0.31        0.00        0.28         0.31         3   \n",
       "3645        0.47        0.00        0.71        0.00         0.71        41   \n",
       "4708        0.01        0.00        0.41        0.02         0.41        15   \n",
       "5235        0.00        0.00        0.00        0.34         0.34       -23   \n",
       "5409        0.00        0.00        0.00        0.38         0.38        27   \n",
       "5490        0.00        0.00        0.01        0.35         0.35        -4   \n",
       "5611        0.00        0.00        0.80        0.08         0.80         2   \n",
       "5941        0.00        0.00        0.00        0.22         0.22         4   \n",
       "5961        0.34        0.91        0.00        0.00         0.91        14   \n",
       "6599        0.00        0.32        0.00        0.00         0.32        -6   \n",
       "6932        0.00        0.00        0.42        0.00         0.42        31   \n",
       "6933        0.21        0.00        0.00        0.00         0.21        -7   \n",
       "7271        0.30        0.00        0.29        0.00         0.30        43   \n",
       "7956        0.00        0.00        0.43        0.00         0.43        47   \n",
       "7999        0.25       -0.00        0.00        0.00         0.25       -13   \n",
       "8011        0.00        0.00        0.26        0.00         0.26        -1   \n",
       "8081        0.00        0.00        0.00        0.21         0.21       -40   \n",
       "\n",
       "      al_pos_x  dg_pos_x  dl_pos_x ancestral_allele temp    anc_dev  \\\n",
       "305        -40       -22        34                C    C    derived   \n",
       "492          2        12        -1                A    A    derived   \n",
       "575        -11        -7       -11                C    C    derived   \n",
       "1018       -19        30       -19                T    T    derived   \n",
       "1655        29         1        -1                T    T    derived   \n",
       "1707        48        -5        37                C    C    derived   \n",
       "3326        38        50       -23                T    T    derived   \n",
       "3577       -40         3        33                G    G    derived   \n",
       "3645       -29         5       -29                C    C  ancestral   \n",
       "4708       -44         2        -2                G    G    derived   \n",
       "5235         2        25        -1                T    T    derived   \n",
       "5409       -15         1       -15                G    G    derived   \n",
       "5490       -40        -1        -4                A    A    derived   \n",
       "5611        40         2        -2                G    G    derived   \n",
       "5941         0         4         0                G    G    derived   \n",
       "5961         2       -34         2                A    A    derived   \n",
       "6599         1         4       -41                A    A    derived   \n",
       "6932        14         1        10                T    T    derived   \n",
       "6933       -40        -7         3                C    C    derived   \n",
       "7271         4         4         0                C    C    derived   \n",
       "7956       -16         7       -16                T    T    derived   \n",
       "7999       -30        11       -13                T    T    derived   \n",
       "8011         3        -1         3                A    A    derived   \n",
       "8081         1         2         1                C    C    derived   \n",
       "\n",
       "      1KG_allele_count  1KG_allele_number  1KG_allele_frequency  1KG_EAS_AF  \\\n",
       "305               90.0             5096.0                  0.02        0.03   \n",
       "492                2.0             5096.0                  0.00        0.00   \n",
       "575               50.0             5096.0                  0.01        0.05   \n",
       "1018              56.0             5096.0                  0.01        0.00   \n",
       "1655              12.0             5096.0                  0.00        0.00   \n",
       "1707             177.0             5096.0                  0.03        0.11   \n",
       "3326             155.0             5096.0                  0.03        0.08   \n",
       "3577             444.0             5096.0                  0.09        0.00   \n",
       "3645            4744.0             5096.0                  0.93        0.90   \n",
       "4708            1138.0             5096.0                  0.22        0.51   \n",
       "5235              40.0             5096.0                  0.01        0.01   \n",
       "5409             911.0             5096.0                  0.18        0.39   \n",
       "5490              50.0             5096.0                  0.01        0.00   \n",
       "5611               1.0             5096.0                  0.00        0.00   \n",
       "5941               1.0             5096.0                  0.00        0.00   \n",
       "5961               2.0             5096.0                  0.00        0.00   \n",
       "6599              81.0             5096.0                  0.02        0.07   \n",
       "6932             114.0             5096.0                  0.02        0.01   \n",
       "6933             782.0             5096.0                  0.15        0.24   \n",
       "7271              98.0             5096.0                  0.02        0.00   \n",
       "7956               2.0             5096.0                  0.00        0.00   \n",
       "7999               4.0             5096.0                  0.00        0.00   \n",
       "8011              18.0             5096.0                  0.00        0.00   \n",
       "8081             120.0             5096.0                  0.02        0.11   \n",
       "\n",
       "      1KG_EUR_AF  1KG_AFR_AF  1KG_AMR_AF  1KG_SAS_AF present_in_1KG  \\\n",
       "305         0.00        0.00        0.00        0.05            yes   \n",
       "492         0.00        0.00        0.00        0.00            yes   \n",
       "575         0.00        0.00        0.00        0.00            yes   \n",
       "1018        0.04        0.00        0.02        0.00            yes   \n",
       "1655        0.00        0.00        0.00        0.01            yes   \n",
       "1707        0.00        0.00        0.00        0.07            yes   \n",
       "3326        0.00        0.00        0.08        0.01            yes   \n",
       "3577        0.25        0.01        0.12        0.10            yes   \n",
       "3645        0.86        0.99        0.93        0.96            yes   \n",
       "4708        0.21        0.01        0.11        0.31            yes   \n",
       "5235        0.00        0.00        0.00        0.03            yes   \n",
       "5409        0.22        0.01        0.15        0.16            yes   \n",
       "5490        0.02        0.00        0.04        0.00            yes   \n",
       "5611        0.00        0.00        0.00        0.00            yes   \n",
       "5941        0.00        0.00        0.00        0.00            yes   \n",
       "5961        0.00        0.00        0.00        0.00            yes   \n",
       "6599        0.00        0.00        0.01        0.00            yes   \n",
       "6932        0.02        0.00        0.01        0.08            yes   \n",
       "6933        0.17        0.00        0.09        0.29            yes   \n",
       "7271        0.05        0.00        0.04        0.01            yes   \n",
       "7956        0.00        0.00        0.00        0.00            yes   \n",
       "7999        0.00        0.00        0.00        0.00            yes   \n",
       "8011        0.00        0.00        0.00        0.02            yes   \n",
       "8081        0.00        0.00        0.00        0.01            yes   \n",
       "\n",
       "      1KG_non_ASW_AFR_AF  gnomAD_allele_count  gnomAD_allele_number  \\\n",
       "305             0.000000                443.0              143266.0   \n",
       "492             0.000000                  4.0              143152.0   \n",
       "575             0.000000                194.0              143252.0   \n",
       "1018            0.000000               2946.0              143262.0   \n",
       "1655            0.000000                 38.0              143298.0   \n",
       "1707            0.000000                551.0              143130.0   \n",
       "3326            0.000000               1030.0              143224.0   \n",
       "3577            0.003899              21727.0              143202.0   \n",
       "3645            0.999025             129412.0              143248.0   \n",
       "4708            0.000975              23692.0              143104.0   \n",
       "5235            0.000000                494.0              143210.0   \n",
       "5409            0.001949              19931.0              143224.0   \n",
       "5490            0.000000               3317.0              143342.0   \n",
       "5611            0.000000                  2.0              143316.0   \n",
       "5941            0.000000                  NaN                   NaN   \n",
       "5961            0.000000                 10.0              143290.0   \n",
       "6599            0.000000                214.0              143150.0   \n",
       "6932            0.000000               2957.0              143284.0   \n",
       "6933            0.001949              17455.0              143064.0   \n",
       "7271            0.002924               5549.0              143236.0   \n",
       "7956            0.000000                  2.0              143256.0   \n",
       "7999            0.000000                  8.0              143328.0   \n",
       "8011            0.000000                 60.0              143274.0   \n",
       "8081            0.000000                481.0              143162.0   \n",
       "\n",
       "      gnomAD_allele_frequency present_in_gnomAD        start  \\\n",
       "305                  0.003092               yes   52820679.0   \n",
       "492                  0.000028               yes  152944500.0   \n",
       "575                  0.001354               yes  183750130.0   \n",
       "1018                 0.020564               yes   17373517.0   \n",
       "1655                 0.000265               yes   84191110.0   \n",
       "1707                 0.003850               yes   99345358.0   \n",
       "3326                 0.007192               yes   14109359.0   \n",
       "3577                 0.151723               yes   60003586.0   \n",
       "3645                 0.903412               yes   33410288.0   \n",
       "4708                 0.165558               yes   62224594.0   \n",
       "5235                 0.003449               yes   52486375.0   \n",
       "5409                 0.139160               yes   38805941.0   \n",
       "5490                 0.023141               yes  100460530.0   \n",
       "5611                 0.000014               yes  135122661.0   \n",
       "5941                      NaN                no  118669325.0   \n",
       "5961                 0.000070               yes  126989006.0   \n",
       "6599                 0.001495               yes  148430559.0   \n",
       "6932                 0.020637               yes   51889357.0   \n",
       "6933                 0.122008               yes   52138225.0   \n",
       "7271                 0.038740               yes  119524265.0   \n",
       "7956                 0.000014               yes  126797407.0   \n",
       "7999                 0.000056               yes  140801295.0   \n",
       "8011                 0.000419               yes  140958655.0   \n",
       "8081                 0.003360               yes  158543831.0   \n",
       "\n",
       "     Vernot_ancestral_allele Vernot_derived_allele  \\\n",
       "305                        C                     T   \n",
       "492                        A                     T   \n",
       "575                        C                     A   \n",
       "1018                       T                     C   \n",
       "1655                       T                     C   \n",
       "1707                       C                     G   \n",
       "3326                       T                     G   \n",
       "3577                       G                     A   \n",
       "3645                       C                     T   \n",
       "4708                       G                     A   \n",
       "5235                       T                     C   \n",
       "5409                       G                     C   \n",
       "5490                       A                     G   \n",
       "5611                       G                     A   \n",
       "5941                       G                     C   \n",
       "5961                       A                     G   \n",
       "6599                       A                     C   \n",
       "6932                       T                     C   \n",
       "6933                       C                     G   \n",
       "7271                       C                     T   \n",
       "7956                       T                     A   \n",
       "7999                       T                     C   \n",
       "8011                       A                     G   \n",
       "8081                       C                     A   \n",
       "\n",
       "      Vernot_ancestral_derived_code  Vernot_AFA_AF  Vernot_AFR_AF  \\\n",
       "305                             1.0        0.00637        0.00000   \n",
       "492                             1.0        0.00000        0.00000   \n",
       "575                             1.0        0.00000        0.00000   \n",
       "1018                            1.0        0.01274        0.00000   \n",
       "1655                            1.0        0.00000        0.00000   \n",
       "1707                            1.0        0.00318        0.00000   \n",
       "3326                            1.0        0.00637        0.00000   \n",
       "3577                            1.0        0.01592        0.00397   \n",
       "3645                            1.0        0.03503        0.00099   \n",
       "4708                            1.0        0.03503        0.00099   \n",
       "5235                            1.0        0.00318        0.00000   \n",
       "5409                            1.0        0.04459        0.00198   \n",
       "5490                            1.0        0.00000        0.00000   \n",
       "5611                            1.0        0.00000        0.00000   \n",
       "5941                            1.0        0.00000        0.00000   \n",
       "5961                            1.0        0.00000        0.00000   \n",
       "6599                            1.0        0.00318        0.00000   \n",
       "6932                            1.0        0.00318        0.00000   \n",
       "6933                            1.0        0.01274        0.00198   \n",
       "7271                            1.0        0.00318        0.00298   \n",
       "7956                            1.0        0.00000        0.00000   \n",
       "7999                            1.0        0.00000        0.00000   \n",
       "8011                            1.0        0.00000        0.00000   \n",
       "8081                            1.0        0.00000        0.00000   \n",
       "\n",
       "      Vernot_AMR_AF  Vernot_EAS_AF  Vernot_EUR_AF  Vernot_PNG_AF  \\\n",
       "305         0.00000        0.02877        0.00497        0.00000   \n",
       "492         0.00000        0.00198        0.00000        0.00000   \n",
       "575         0.00000        0.04861        0.00000        0.01852   \n",
       "1018        0.01729        0.00000        0.03479        0.01852   \n",
       "1655        0.00000        0.00000        0.00000        0.00000   \n",
       "1707        0.00000        0.10615        0.00000        0.11111   \n",
       "3326        0.08501        0.08333        0.00000        0.16667   \n",
       "3577        0.11671        0.00198        0.24950        0.09259   \n",
       "3645        0.06916        0.09821        0.14612        0.07407   \n",
       "4708        0.10663        0.51488        0.21173        0.03704   \n",
       "5235        0.00288        0.00794        0.00298        0.00000   \n",
       "5409        0.14986        0.39484        0.22068        0.05556   \n",
       "5490        0.03602        0.00099        0.02087        0.00000   \n",
       "5611        0.00000        0.00099        0.00000        0.00000   \n",
       "5941        0.00000        0.00000        0.00000        0.11111   \n",
       "5961        0.00000        0.00000        0.00000        0.00000   \n",
       "6599        0.00720        0.07242        0.00000        0.03704   \n",
       "6932        0.01009        0.00694        0.02286        0.00000   \n",
       "6933        0.09366        0.24206        0.16600        0.09259   \n",
       "7271        0.04179        0.00000        0.04871        0.00000   \n",
       "7956        0.00000        0.00198        0.00000        0.00000   \n",
       "7999        0.00000        0.00099        0.00000        0.00000   \n",
       "8011        0.00000        0.00000        0.00000        0.00000   \n",
       "8081        0.00000        0.11210        0.00000        0.00000   \n",
       "\n",
       "      Vernot_SAS_AF Vernot_Denisovan_base      Vernot_haplotype_tag  \\\n",
       "305         0.05521                     C    chr1_52325475_53062486   \n",
       "492         0.00000                     A  chr1_152942848_153033467   \n",
       "575         0.00000                     C  chr1_183411087_183827254   \n",
       "1018        0.00307                     T   chr10_17338795_17417730   \n",
       "1655        0.01227                     T   chr11_84051845_84211811   \n",
       "1707        0.06748                     C   chr11_99309226_99359652   \n",
       "3326        0.00818                     G   chr17_13902596_14111504   \n",
       "3577        0.09714                     A   chr18_59569521_60089689   \n",
       "3645        0.04090                     C   chr19_33040186_33456652   \n",
       "4708        0.30982                     G   chr20_62195671_62229244   \n",
       "5235        0.02658                     T    chr3_52214838_52516460   \n",
       "5409        0.16360                     G    chr4_38294200_38857896   \n",
       "5490        0.00204                     G  chr4_100349203_100638613   \n",
       "5611        0.00000                     G  chr4_135066013_135400700   \n",
       "5941        0.00102                     C  chr5_118661135_118933833   \n",
       "5961        0.00204                     A  chr5_126989006_127537918   \n",
       "6599        0.00204                     A  chr5_148339333_148539017   \n",
       "6932        0.07771                   T/C    chr6_51841815_51972144   \n",
       "6933        0.29448                     C    chr6_52138225_52183126   \n",
       "7271        0.01431                     C  chr6_119475382_119646998   \n",
       "7956        0.00000                     T  chr7_126660339_126797408   \n",
       "7999        0.00307                     C  chr7_140306146_140804665   \n",
       "8011        0.01840                     A  chr7_140934039_141109377   \n",
       "8081        0.00511                     A  chr7_158399204_158736324   \n",
       "\n",
       "     Vernot_Neanderthal_base_1 Vernot_Neanderthal_base_2 ref_allele_y  \\\n",
       "305                          C                         T            T   \n",
       "492                          A                         T            T   \n",
       "575                          C                         A            A   \n",
       "1018                         T                         C            C   \n",
       "1655                         T                         C            C   \n",
       "1707                         C                         G            G   \n",
       "3326                         T                       NaN            G   \n",
       "3577                         G                         A            A   \n",
       "3645                         T                       NaN            C   \n",
       "4708                         G                         A            A   \n",
       "5235                         T                         C            C   \n",
       "5409                         G                         C            C   \n",
       "5490                         A                         G            G   \n",
       "5611                         G                         A            A   \n",
       "5941                         G                       NaN            C   \n",
       "5961                         A                         G            G   \n",
       "6599                         A                         C            C   \n",
       "6932                         T                         C            C   \n",
       "6933                         C                         G            G   \n",
       "7271                         C                         T            T   \n",
       "7956                         T                         A            A   \n",
       "7999                         T                         C            C   \n",
       "8011                         A                         G            G   \n",
       "8081                         C                         A            A   \n",
       "\n",
       "     altai_gt_y chagyrskaya_gt_y denisovan_gt_y vindija_gt_y alt_allele_y  \\\n",
       "305         1/1              1/1            1/1          1/1            C   \n",
       "492         1/1              1/1            1/1          1/1            A   \n",
       "575         1/1              1/1            1/1          1/1            C   \n",
       "1018        1/1              1/1            1/1          1/1            T   \n",
       "1655        1/1              1/1            1/1          1/1            T   \n",
       "1707        1/1              1/1            1/1          1/1            C   \n",
       "3326        1/1              1/1            1/1          1/1            T   \n",
       "3577        1/1              1/1            1/1          1/1            G   \n",
       "3645        1/1              1/1            1/1          1/1            T   \n",
       "4708        1/1              1/1            1/1          1/1            G   \n",
       "5235        1/1              1/1            1/1          1/1            T   \n",
       "5409        1/1              1/1            1/1          1/1            G   \n",
       "5490        1/1              1/1            1/1          1/1            A   \n",
       "5611        1/1              1/1            1/1          1/1            G   \n",
       "5941        1/1              1/1            1/1          1/1            G   \n",
       "5961        1/1              1/1            1/1          1/1            A   \n",
       "6599        1/1              1/1            1/1          1/1            A   \n",
       "6932        1/1              1/1            1/1          1/1            T   \n",
       "6933        1/1              1/1            1/1          1/1            C   \n",
       "7271        1/1              1/1            1/1          1/1            C   \n",
       "7956        1/1              1/1            1/1          1/1            T   \n",
       "7999        1/1              1/1            1/1          1/1            T   \n",
       "8011        1/1              1/1            1/1          1/1            A   \n",
       "8081        1/1              1/1            1/1          1/1            C   \n",
       "\n",
       "      ag_delta_y  al_delta_y  dg_delta_y  dl_delta_y  ag_pos_y  al_pos_y  \\\n",
       "305         0.00        0.23        0.01        0.00       -40       -22   \n",
       "492         0.00        0.23        0.00        0.00         2        13   \n",
       "575         0.00        0.00        0.01        0.26       -11        -7   \n",
       "1018        0.22        0.05        0.00        0.00       -19        33   \n",
       "1655        0.00        0.00        0.40        0.00        29         1   \n",
       "1707        0.00        0.00        0.00        0.67        48        -4   \n",
       "3326        0.04        0.21        0.00        0.00        38         1   \n",
       "3577        0.30        0.00        0.29        0.00       -40         3   \n",
       "3645        0.00        0.47        0.00        0.71       -29        41   \n",
       "4708        0.00        0.01        0.02        0.41       -44        15   \n",
       "5235        0.00        0.00        0.35        0.00         2       -23   \n",
       "5409        0.00        0.00        0.37        0.00       -15        27   \n",
       "5490        0.00        0.00        0.35        0.01       -40        -4   \n",
       "5611        0.00        0.00        0.08        0.80        40         2   \n",
       "5941        0.00        0.00        0.22        0.00         0         4   \n",
       "5961        0.91        0.34        0.00        0.00         2        14   \n",
       "6599        0.32        0.00        0.00        0.00         1        -6   \n",
       "6932        0.00        0.00        0.00        0.42        14        31   \n",
       "6933        0.00        0.21        0.00        0.00       -40        -7   \n",
       "7271        0.00        0.30        0.00        0.29         4        43   \n",
       "7956        0.00        0.00        0.00        0.43       -16        47   \n",
       "7999       -0.00        0.25        0.00        0.00        27       -13   \n",
       "8011        0.00        0.00        0.00        0.26         3        -1   \n",
       "8081        0.00        0.00        0.22        0.00         1       -40   \n",
       "\n",
       "      dg_pos_y  dl_pos_y  delta_max_y  \n",
       "305         34       -22         0.23  \n",
       "492         -1        12         0.23  \n",
       "575        -11        -7         0.26  \n",
       "1018       -19        30         0.22  \n",
       "1655        -1         1         0.40  \n",
       "1707        37        -5         0.67  \n",
       "3326       -23        50         0.21  \n",
       "3577        33         3         0.30  \n",
       "3645       -29         5         0.71  \n",
       "4708        -2         2         0.41  \n",
       "5235        -1        25         0.35  \n",
       "5409       -15         1         0.37  \n",
       "5490        -4        -1         0.35  \n",
       "5611        -2         2         0.80  \n",
       "5941         0         4         0.22  \n",
       "5961         2       -34         0.91  \n",
       "6599       -41         4         0.32  \n",
       "6932        10         1         0.42  \n",
       "6933         3        -7         0.21  \n",
       "7271         0         4         0.30  \n",
       "7956       -16         7         0.43  \n",
       "7999       -13        11         0.25  \n",
       "8011         3        -1         0.26  \n",
       "8081         1         2         0.22  "
      ]
     },
     "execution_count": 617,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "introgressed_ref_tag_merge[(introgressed_ref_tag_merge['delta_max_x']>=0.2) & (introgressed_ref_tag_merge['delta_max_y']>=0.2)]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f4451b68",
   "metadata": {},
   "source": [
    "Let's identify the two positions where the deltas were different."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 618,
   "id": "7ff87e6e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>pos</th>\n",
       "      <th>_merge</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>52820680</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>152944501</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>183750131</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>216945756</td>\n",
       "      <td>left_only</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr10</td>\n",
       "      <td>17373518</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr11</td>\n",
       "      <td>84191111</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr11</td>\n",
       "      <td>99345359</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr17</td>\n",
       "      <td>14109360</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr18</td>\n",
       "      <td>60003587</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr19</td>\n",
       "      <td>33410289</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr20</td>\n",
       "      <td>62224595</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr3</td>\n",
       "      <td>52486376</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr4</td>\n",
       "      <td>38805942</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr4</td>\n",
       "      <td>100460531</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>chr4</td>\n",
       "      <td>135122662</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>chr5</td>\n",
       "      <td>118669326</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>chr5</td>\n",
       "      <td>126989007</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>chr5</td>\n",
       "      <td>148430560</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>chr6</td>\n",
       "      <td>51889358</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>chr6</td>\n",
       "      <td>52138226</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>chr6</td>\n",
       "      <td>119524266</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>chr7</td>\n",
       "      <td>126797408</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>chr7</td>\n",
       "      <td>140801296</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>chr7</td>\n",
       "      <td>140958656</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>chr7</td>\n",
       "      <td>157177273</td>\n",
       "      <td>left_only</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>chr7</td>\n",
       "      <td>158543832</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    chrom        pos     _merge\n",
       "0    chr1   52820680       both\n",
       "1    chr1  152944501       both\n",
       "2    chr1  183750131       both\n",
       "3    chr1  216945756  left_only\n",
       "4   chr10   17373518       both\n",
       "5   chr11   84191111       both\n",
       "6   chr11   99345359       both\n",
       "7   chr17   14109360       both\n",
       "8   chr18   60003587       both\n",
       "9   chr19   33410289       both\n",
       "10  chr20   62224595       both\n",
       "11   chr3   52486376       both\n",
       "12   chr4   38805942       both\n",
       "13   chr4  100460531       both\n",
       "14   chr4  135122662       both\n",
       "15   chr5  118669326       both\n",
       "16   chr5  126989007       both\n",
       "17   chr5  148430560       both\n",
       "18   chr6   51889358       both\n",
       "19   chr6   52138226       both\n",
       "20   chr6  119524266       both\n",
       "21   chr7  126797408       both\n",
       "22   chr7  140801296       both\n",
       "23   chr7  140958656       both\n",
       "24   chr7  157177273  left_only\n",
       "25   chr7  158543832       both"
      ]
     },
     "execution_count": 618,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "diffs = pd.merge(SAV_introgressed_ref_tag[['chrom','pos']], SAV_swapped[['chrom','pos']], how = 'left', indicator=True)\n",
    "diffs"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9285d5c7",
   "metadata": {},
   "source": [
    "Let's look at the actual deltas. First up is chr1: 216,945,756."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 619,
   "id": "41e4744e",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref_allele</th>\n",
       "      <th>altai_gt</th>\n",
       "      <th>chagyrskaya_gt</th>\n",
       "      <th>denisovan_gt</th>\n",
       "      <th>vindija_gt</th>\n",
       "      <th>alt_allele</th>\n",
       "      <th>annotation</th>\n",
       "      <th>ag_delta</th>\n",
       "      <th>al_delta</th>\n",
       "      <th>dg_delta</th>\n",
       "      <th>dl_delta</th>\n",
       "      <th>ag_pos</th>\n",
       "      <th>al_pos</th>\n",
       "      <th>dg_pos</th>\n",
       "      <th>dl_pos</th>\n",
       "      <th>delta_max</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3553</th>\n",
       "      <td>chr1</td>\n",
       "      <td>216945756</td>\n",
       "      <td>A</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>C</td>\n",
       "      <td>ESRRG</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.19</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>47</td>\n",
       "      <td>-27</td>\n",
       "      <td>48</td>\n",
       "      <td>-27</td>\n",
       "      <td>0.19</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     chrom        pos ref_allele altai_gt chagyrskaya_gt denisovan_gt  \\\n",
       "3553  chr1  216945756          A      1/1            1/1          1/1   \n",
       "\n",
       "     vindija_gt alt_allele annotation  ag_delta  al_delta  dg_delta  dl_delta  \\\n",
       "3553        1/1          C      ESRRG       0.0      0.19       0.0       0.0   \n",
       "\n",
       "      ag_pos  al_pos  dg_pos  dl_pos  delta_max  \n",
       "3553      47     -27      48     -27       0.19  "
      ]
     },
     "execution_count": 619,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "swapped[(swapped['chrom'] == 'chr1') & (swapped['pos'] == 216945756)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 620,
   "id": "fc727ac8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref_allele</th>\n",
       "      <th>alt_allele</th>\n",
       "      <th>variant_type</th>\n",
       "      <th>altai_gt</th>\n",
       "      <th>chagyrskaya_gt</th>\n",
       "      <th>denisovan_gt</th>\n",
       "      <th>vindija_gt</th>\n",
       "      <th>altai_gt_boolean</th>\n",
       "      <th>chagyrskaya_gt_boolean</th>\n",
       "      <th>denisovan_gt_boolean</th>\n",
       "      <th>vindija_gt_boolean</th>\n",
       "      <th>distribution</th>\n",
       "      <th>annotation</th>\n",
       "      <th>mis_oe</th>\n",
       "      <th>mis_z</th>\n",
       "      <th>lof_oe</th>\n",
       "      <th>lof_z</th>\n",
       "      <th>phyloP</th>\n",
       "      <th>ag_delta</th>\n",
       "      <th>al_delta</th>\n",
       "      <th>dg_delta</th>\n",
       "      <th>dl_delta</th>\n",
       "      <th>delta_max</th>\n",
       "      <th>ag_pos</th>\n",
       "      <th>al_pos</th>\n",
       "      <th>dg_pos</th>\n",
       "      <th>dl_pos</th>\n",
       "      <th>ancestral_allele</th>\n",
       "      <th>temp</th>\n",
       "      <th>anc_dev</th>\n",
       "      <th>1KG_allele_count</th>\n",
       "      <th>1KG_allele_number</th>\n",
       "      <th>1KG_allele_frequency</th>\n",
       "      <th>1KG_EAS_AF</th>\n",
       "      <th>1KG_EUR_AF</th>\n",
       "      <th>1KG_AFR_AF</th>\n",
       "      <th>1KG_AMR_AF</th>\n",
       "      <th>1KG_SAS_AF</th>\n",
       "      <th>present_in_1KG</th>\n",
       "      <th>1KG_non_ASW_AFR_AF</th>\n",
       "      <th>gnomAD_allele_count</th>\n",
       "      <th>gnomAD_allele_number</th>\n",
       "      <th>gnomAD_allele_frequency</th>\n",
       "      <th>present_in_gnomAD</th>\n",
       "      <th>start</th>\n",
       "      <th>Vernot_ancestral_allele</th>\n",
       "      <th>Vernot_derived_allele</th>\n",
       "      <th>Vernot_ancestral_derived_code</th>\n",
       "      <th>Vernot_AFA_AF</th>\n",
       "      <th>Vernot_AFR_AF</th>\n",
       "      <th>Vernot_AMR_AF</th>\n",
       "      <th>Vernot_EAS_AF</th>\n",
       "      <th>Vernot_EUR_AF</th>\n",
       "      <th>Vernot_PNG_AF</th>\n",
       "      <th>Vernot_SAS_AF</th>\n",
       "      <th>Vernot_Denisovan_base</th>\n",
       "      <th>Vernot_haplotype_tag</th>\n",
       "      <th>Vernot_Neanderthal_base_1</th>\n",
       "      <th>Vernot_Neanderthal_base_2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>800</th>\n",
       "      <td>chr1</td>\n",
       "      <td>216945756</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>ESRRG</td>\n",
       "      <td>0.56557</td>\n",
       "      <td>2.5557</td>\n",
       "      <td>0.28379</td>\n",
       "      <td>3.0516</td>\n",
       "      <td>0.655</td>\n",
       "      <td>0.2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.2</td>\n",
       "      <td>-27</td>\n",
       "      <td>47</td>\n",
       "      <td>-27</td>\n",
       "      <td>48</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>derived</td>\n",
       "      <td>169.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.03</td>\n",
       "      <td>0.14</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.02</td>\n",
       "      <td>yes</td>\n",
       "      <td>0.0</td>\n",
       "      <td>746.0</td>\n",
       "      <td>143174.0</td>\n",
       "      <td>0.00521</td>\n",
       "      <td>yes</td>\n",
       "      <td>216945755.0</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.01153</td>\n",
       "      <td>0.13988</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0184</td>\n",
       "      <td>C</td>\n",
       "      <td>chr1_216944862_216964892</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    chrom        pos ref_allele alt_allele variant_type altai_gt  \\\n",
       "800  chr1  216945756          C          A          snv      0/1   \n",
       "\n",
       "    chagyrskaya_gt denisovan_gt vindija_gt  altai_gt_boolean  \\\n",
       "800            0/1          0/0        1/1              True   \n",
       "\n",
       "     chagyrskaya_gt_boolean  denisovan_gt_boolean  vindija_gt_boolean  \\\n",
       "800                    True                 False                True   \n",
       "\n",
       "    distribution annotation   mis_oe   mis_z   lof_oe   lof_z  phyloP  \\\n",
       "800  Neanderthal      ESRRG  0.56557  2.5557  0.28379  3.0516   0.655   \n",
       "\n",
       "     ag_delta  al_delta  dg_delta  dl_delta  delta_max  ag_pos  al_pos  \\\n",
       "800       0.2       0.0       0.0       0.0        0.2     -27      47   \n",
       "\n",
       "     dg_pos  dl_pos ancestral_allele temp  anc_dev  1KG_allele_count  \\\n",
       "800     -27      48                C    C  derived             169.0   \n",
       "\n",
       "     1KG_allele_number  1KG_allele_frequency  1KG_EAS_AF  1KG_EUR_AF  \\\n",
       "800             5096.0                  0.03        0.14         0.0   \n",
       "\n",
       "     1KG_AFR_AF  1KG_AMR_AF  1KG_SAS_AF present_in_1KG  1KG_non_ASW_AFR_AF  \\\n",
       "800         0.0        0.01        0.02            yes                 0.0   \n",
       "\n",
       "     gnomAD_allele_count  gnomAD_allele_number  gnomAD_allele_frequency  \\\n",
       "800                746.0              143174.0                  0.00521   \n",
       "\n",
       "    present_in_gnomAD        start Vernot_ancestral_allele  \\\n",
       "800               yes  216945755.0                       C   \n",
       "\n",
       "    Vernot_derived_allele  Vernot_ancestral_derived_code  Vernot_AFA_AF  \\\n",
       "800                     A                            1.0            0.0   \n",
       "\n",
       "     Vernot_AFR_AF  Vernot_AMR_AF  Vernot_EAS_AF  Vernot_EUR_AF  \\\n",
       "800            0.0        0.01153        0.13988            0.0   \n",
       "\n",
       "     Vernot_PNG_AF  Vernot_SAS_AF Vernot_Denisovan_base  \\\n",
       "800            0.0         0.0184                     C   \n",
       "\n",
       "         Vernot_haplotype_tag Vernot_Neanderthal_base_1  \\\n",
       "800  chr1_216944862_216964892                         C   \n",
       "\n",
       "    Vernot_Neanderthal_base_2  \n",
       "800                         A  "
      ]
     },
     "execution_count": 620,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SAV_introgressed_ref_tag[(SAV_introgressed_ref_tag['chrom'] == 'chr1') & (SAV_introgressed_ref_tag['pos'] == 216945756)]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "563963e7",
   "metadata": {},
   "source": [
    "Now the other allele: chr7: 157,177,273."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 621,
   "id": "1cbd3def",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref_allele</th>\n",
       "      <th>altai_gt</th>\n",
       "      <th>chagyrskaya_gt</th>\n",
       "      <th>denisovan_gt</th>\n",
       "      <th>vindija_gt</th>\n",
       "      <th>alt_allele</th>\n",
       "      <th>annotation</th>\n",
       "      <th>ag_delta</th>\n",
       "      <th>al_delta</th>\n",
       "      <th>dg_delta</th>\n",
       "      <th>dl_delta</th>\n",
       "      <th>ag_pos</th>\n",
       "      <th>al_pos</th>\n",
       "      <th>dg_pos</th>\n",
       "      <th>dl_pos</th>\n",
       "      <th>delta_max</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>8039</th>\n",
       "      <td>chr7</td>\n",
       "      <td>157177273</td>\n",
       "      <td>T</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>C</td>\n",
       "      <td>DNAJB6</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.03</td>\n",
       "      <td>0.16</td>\n",
       "      <td>12</td>\n",
       "      <td>-2</td>\n",
       "      <td>12</td>\n",
       "      <td>-2</td>\n",
       "      <td>0.16</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     chrom        pos ref_allele altai_gt chagyrskaya_gt denisovan_gt  \\\n",
       "8039  chr7  157177273          T      1/1            1/1          1/1   \n",
       "\n",
       "     vindija_gt alt_allele annotation  ag_delta  al_delta  dg_delta  dl_delta  \\\n",
       "8039        1/1          C     DNAJB6       0.0       0.0      0.03      0.16   \n",
       "\n",
       "      ag_pos  al_pos  dg_pos  dl_pos  delta_max  \n",
       "8039      12      -2      12      -2       0.16  "
      ]
     },
     "execution_count": 621,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "swapped[(swapped['chrom'] == 'chr7') & (swapped['pos'] == 157177273)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 622,
   "id": "47eff0b7",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref_allele</th>\n",
       "      <th>alt_allele</th>\n",
       "      <th>variant_type</th>\n",
       "      <th>altai_gt</th>\n",
       "      <th>chagyrskaya_gt</th>\n",
       "      <th>denisovan_gt</th>\n",
       "      <th>vindija_gt</th>\n",
       "      <th>altai_gt_boolean</th>\n",
       "      <th>chagyrskaya_gt_boolean</th>\n",
       "      <th>denisovan_gt_boolean</th>\n",
       "      <th>vindija_gt_boolean</th>\n",
       "      <th>distribution</th>\n",
       "      <th>annotation</th>\n",
       "      <th>mis_oe</th>\n",
       "      <th>mis_z</th>\n",
       "      <th>lof_oe</th>\n",
       "      <th>lof_z</th>\n",
       "      <th>phyloP</th>\n",
       "      <th>ag_delta</th>\n",
       "      <th>al_delta</th>\n",
       "      <th>dg_delta</th>\n",
       "      <th>dl_delta</th>\n",
       "      <th>delta_max</th>\n",
       "      <th>ag_pos</th>\n",
       "      <th>al_pos</th>\n",
       "      <th>dg_pos</th>\n",
       "      <th>dl_pos</th>\n",
       "      <th>ancestral_allele</th>\n",
       "      <th>temp</th>\n",
       "      <th>anc_dev</th>\n",
       "      <th>1KG_allele_count</th>\n",
       "      <th>1KG_allele_number</th>\n",
       "      <th>1KG_allele_frequency</th>\n",
       "      <th>1KG_EAS_AF</th>\n",
       "      <th>1KG_EUR_AF</th>\n",
       "      <th>1KG_AFR_AF</th>\n",
       "      <th>1KG_AMR_AF</th>\n",
       "      <th>1KG_SAS_AF</th>\n",
       "      <th>present_in_1KG</th>\n",
       "      <th>1KG_non_ASW_AFR_AF</th>\n",
       "      <th>gnomAD_allele_count</th>\n",
       "      <th>gnomAD_allele_number</th>\n",
       "      <th>gnomAD_allele_frequency</th>\n",
       "      <th>present_in_gnomAD</th>\n",
       "      <th>start</th>\n",
       "      <th>Vernot_ancestral_allele</th>\n",
       "      <th>Vernot_derived_allele</th>\n",
       "      <th>Vernot_ancestral_derived_code</th>\n",
       "      <th>Vernot_AFA_AF</th>\n",
       "      <th>Vernot_AFR_AF</th>\n",
       "      <th>Vernot_AMR_AF</th>\n",
       "      <th>Vernot_EAS_AF</th>\n",
       "      <th>Vernot_EUR_AF</th>\n",
       "      <th>Vernot_PNG_AF</th>\n",
       "      <th>Vernot_SAS_AF</th>\n",
       "      <th>Vernot_Denisovan_base</th>\n",
       "      <th>Vernot_haplotype_tag</th>\n",
       "      <th>Vernot_Neanderthal_base_1</th>\n",
       "      <th>Vernot_Neanderthal_base_2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>7341</th>\n",
       "      <td>chr7</td>\n",
       "      <td>157177273</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>snv</td>\n",
       "      <td>0/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Neanderthal</td>\n",
       "      <td>DNAJB6</td>\n",
       "      <td>0.76273</td>\n",
       "      <td>1.1745</td>\n",
       "      <td>0.15377</td>\n",
       "      <td>3.4636</td>\n",
       "      <td>0.455</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.31</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.31</td>\n",
       "      <td>-2</td>\n",
       "      <td>46</td>\n",
       "      <td>-2</td>\n",
       "      <td>2</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>derived</td>\n",
       "      <td>57.0</td>\n",
       "      <td>5096.0</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>0.0</td>\n",
       "      <td>268.0</td>\n",
       "      <td>143270.0</td>\n",
       "      <td>0.001871</td>\n",
       "      <td>yes</td>\n",
       "      <td>157177272.0</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.05357</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00204</td>\n",
       "      <td>C</td>\n",
       "      <td>chr7_157116267_157177286</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     chrom        pos ref_allele alt_allele variant_type altai_gt  \\\n",
       "7341  chr7  157177273          C          T          snv      0/1   \n",
       "\n",
       "     chagyrskaya_gt denisovan_gt vindija_gt  altai_gt_boolean  \\\n",
       "7341            1/1          0/0        1/1              True   \n",
       "\n",
       "      chagyrskaya_gt_boolean  denisovan_gt_boolean  vindija_gt_boolean  \\\n",
       "7341                    True                 False                True   \n",
       "\n",
       "     distribution annotation   mis_oe   mis_z   lof_oe   lof_z  phyloP  \\\n",
       "7341  Neanderthal     DNAJB6  0.76273  1.1745  0.15377  3.4636   0.455   \n",
       "\n",
       "      ag_delta  al_delta  dg_delta  dl_delta  delta_max  ag_pos  al_pos  \\\n",
       "7341       0.0       0.0      0.31       0.0       0.31      -2      46   \n",
       "\n",
       "      dg_pos  dl_pos ancestral_allele temp  anc_dev  1KG_allele_count  \\\n",
       "7341      -2       2                C    C  derived              57.0   \n",
       "\n",
       "      1KG_allele_number  1KG_allele_frequency  1KG_EAS_AF  1KG_EUR_AF  \\\n",
       "7341             5096.0                  0.01        0.05         0.0   \n",
       "\n",
       "      1KG_AFR_AF  1KG_AMR_AF  1KG_SAS_AF present_in_1KG  1KG_non_ASW_AFR_AF  \\\n",
       "7341         0.0         0.0         0.0            yes                 0.0   \n",
       "\n",
       "      gnomAD_allele_count  gnomAD_allele_number  gnomAD_allele_frequency  \\\n",
       "7341                268.0              143270.0                 0.001871   \n",
       "\n",
       "     present_in_gnomAD        start Vernot_ancestral_allele  \\\n",
       "7341               yes  157177272.0                       C   \n",
       "\n",
       "     Vernot_derived_allele  Vernot_ancestral_derived_code  Vernot_AFA_AF  \\\n",
       "7341                     T                            1.0            0.0   \n",
       "\n",
       "      Vernot_AFR_AF  Vernot_AMR_AF  Vernot_EAS_AF  Vernot_EUR_AF  \\\n",
       "7341            0.0            0.0        0.05357            0.0   \n",
       "\n",
       "      Vernot_PNG_AF  Vernot_SAS_AF Vernot_Denisovan_base  \\\n",
       "7341            0.0        0.00204                     C   \n",
       "\n",
       "          Vernot_haplotype_tag Vernot_Neanderthal_base_1  \\\n",
       "7341  chr7_157116267_157177286                         C   \n",
       "\n",
       "     Vernot_Neanderthal_base_2  \n",
       "7341                         T  "
      ]
     },
     "execution_count": 622,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SAV_introgressed_ref_tag[(SAV_introgressed_ref_tag['chrom'] == 'chr7') & (SAV_introgressed_ref_tag['pos'] == 157177273)]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7ea3490d",
   "metadata": {},
   "source": [
    "# Introgressed Fixed Alleles <a class = 'anchor' id = 'introgressedfixed'></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d0a1ffbd",
   "metadata": {},
   "source": [
    "Let's take a brief look at the introgressed fixed alleles. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 623,
   "id": "77f64c55",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref_allele</th>\n",
       "      <th>altai_gt</th>\n",
       "      <th>chagyrskaya_gt</th>\n",
       "      <th>denisovan_gt</th>\n",
       "      <th>vindija_gt</th>\n",
       "      <th>alt_allele</th>\n",
       "      <th>annotation</th>\n",
       "      <th>ag_delta</th>\n",
       "      <th>al_delta</th>\n",
       "      <th>dg_delta</th>\n",
       "      <th>dl_delta</th>\n",
       "      <th>ag_pos</th>\n",
       "      <th>al_pos</th>\n",
       "      <th>dg_pos</th>\n",
       "      <th>dl_pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>2998934</td>\n",
       "      <td>C</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>T</td>\n",
       "      <td>PRDM16</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>17</td>\n",
       "      <td>29</td>\n",
       "      <td>-12</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>2999202</td>\n",
       "      <td>C</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>T</td>\n",
       "      <td>PRDM16</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>17</td>\n",
       "      <td>33</td>\n",
       "      <td>-20</td>\n",
       "      <td>-3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>2999415</td>\n",
       "      <td>C</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>T</td>\n",
       "      <td>PRDM16</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>14</td>\n",
       "      <td>2</td>\n",
       "      <td>29</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3000465</td>\n",
       "      <td>G</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>A</td>\n",
       "      <td>PRDM16</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>32</td>\n",
       "      <td>-25</td>\n",
       "      <td>-20</td>\n",
       "      <td>-34</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3046711</td>\n",
       "      <td>G</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>T</td>\n",
       "      <td>PRDM16</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>35</td>\n",
       "      <td>-4</td>\n",
       "      <td>19</td>\n",
       "      <td>-24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3046826</td>\n",
       "      <td>T</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>G</td>\n",
       "      <td>PRDM16</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-49</td>\n",
       "      <td>-1</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3188156</td>\n",
       "      <td>A</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>G</td>\n",
       "      <td>PRDM16</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>9</td>\n",
       "      <td>-27</td>\n",
       "      <td>-1</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3195402</td>\n",
       "      <td>G</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>C</td>\n",
       "      <td>PRDM16</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.04</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>41</td>\n",
       "      <td>-29</td>\n",
       "      <td>-29</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3199291</td>\n",
       "      <td>T</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>C</td>\n",
       "      <td>PRDM16</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>30</td>\n",
       "      <td>28</td>\n",
       "      <td>-23</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3199727</td>\n",
       "      <td>T</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>C</td>\n",
       "      <td>PRDM16</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>44</td>\n",
       "      <td>8</td>\n",
       "      <td>44</td>\n",
       "      <td>-2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chrom      pos ref_allele altai_gt chagyrskaya_gt denisovan_gt vindija_gt  \\\n",
       "0  chr1  2998934          C      1/1            1/1          1/1        1/1   \n",
       "1  chr1  2999202          C      1/1            1/1          1/1        1/1   \n",
       "2  chr1  2999415          C      1/1            1/1          1/1        1/1   \n",
       "3  chr1  3000465          G      1/1            1/1          1/1        1/1   \n",
       "4  chr1  3046711          G      1/1            1/1          1/1        1/1   \n",
       "5  chr1  3046826          T      1/1            1/1          1/1        1/1   \n",
       "6  chr1  3188156          A      1/1            1/1          1/1        1/1   \n",
       "7  chr1  3195402          G      1/1            1/1          1/1        1/1   \n",
       "8  chr1  3199291          T      1/1            1/1          1/1        1/1   \n",
       "9  chr1  3199727          T      1/1            1/1          1/1        1/1   \n",
       "\n",
       "  alt_allele annotation  ag_delta  al_delta  dg_delta  dl_delta  ag_pos  \\\n",
       "0          T     PRDM16       0.0      0.00       0.0       0.0      17   \n",
       "1          T     PRDM16       0.0      0.00       0.0       0.0      17   \n",
       "2          T     PRDM16       0.0      0.00       0.0       0.0      14   \n",
       "3          A     PRDM16       0.0      0.00       0.0       0.0      32   \n",
       "4          T     PRDM16       0.0      0.00       0.0       0.0      35   \n",
       "5          G     PRDM16       0.0      0.00       0.0       0.0      -1   \n",
       "6          G     PRDM16       0.0      0.00       0.0       0.0       9   \n",
       "7          C     PRDM16       0.0      0.04       0.0       0.0      41   \n",
       "8          C     PRDM16       0.0      0.00       0.0       0.0      30   \n",
       "9          C     PRDM16       0.0      0.00       0.0       0.0      44   \n",
       "\n",
       "   al_pos  dg_pos  dl_pos  \n",
       "0      29     -12      29  \n",
       "1      33     -20      -3  \n",
       "2       2      29       8  \n",
       "3     -25     -20     -34  \n",
       "4      -4      19     -24  \n",
       "5     -49      -1      11  \n",
       "6     -27      -1       4  \n",
       "7     -29     -29      31  \n",
       "8      28     -23      28  \n",
       "9       8      44      -2  "
      ]
     },
     "execution_count": 623,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fixed_header = ['chrom','pos','ref_allele','altai_gt','chagyrskaya_gt','denisovan_gt','vindija_gt','alt_allele','annotation','ag_delta','al_delta','dg_delta','dl_delta','ag_pos','al_pos','dg_pos','dl_pos']\n",
    "fixed = pd.read_csv('../fixed_refs/spliceai_annotated_fixed_refs.txt', sep = '\\t', names = fixed_header)\n",
    "fixed.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 624,
   "id": "4ca50489",
   "metadata": {},
   "outputs": [],
   "source": [
    "fixed['delta_max'] = fixed[['ag_delta','al_delta','dg_delta','dl_delta']].max(axis = 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 625,
   "id": "955481e7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "41"
      ]
     },
     "execution_count": 625,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(fixed[fixed['delta_max'] >= 0.2])"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
